<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emi Tanaka" />


<title>Selection of R-packages for review</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>/*! jQuery v3.5.1 | (c) JS Foundation and other contributors | jquery.org/license */
!function(e,t){"use strict";"object"==typeof module&&"object"==typeof module.exports?module.exports=e.document?t(e,!0):function(e){if(!e.document)throw new Error("jQuery requires a window with a document");return t(e)}:t(e)}("undefined"!=typeof window?window:this,function(C,e){"use strict";var t=[],r=Object.getPrototypeOf,s=t.slice,g=t.flat?function(e){return t.flat.call(e)}:function(e){return t.concat.apply([],e)},u=t.push,i=t.indexOf,n={},o=n.toString,v=n.hasOwnProperty,a=v.toString,l=a.call(Object),y={},m=function(e){return"function"==typeof e&&"number"!=typeof e.nodeType},x=function(e){return null!=e&&e===e.window},E=C.document,c={type:!0,src:!0,nonce:!0,noModule:!0};function b(e,t,n){var r,i,o=(n=n||E).createElement("script");if(o.text=e,t)for(r in c)(i=t[r]||t.getAttribute&&t.getAttribute(r))&&o.setAttribute(r,i);n.head.appendChild(o).parentNode.removeChild(o)}function w(e){return null==e?e+"":"object"==typeof e||"function"==typeof e?n[o.call(e)]||"object":typeof e}var f="3.5.1",S=function(e,t){return new S.fn.init(e,t)};function p(e){var t=!!e&&"length"in e&&e.length,n=w(e);return!m(e)&&!x(e)&&("array"===n||0===t||"number"==typeof t&&0<t&&t-1 in e)}S.fn=S.prototype={jquery:f,constructor:S,length:0,toArray:function(){return s.call(this)},get:function(e){return null==e?s.call(this):e<0?this[e+this.length]:this[e]},pushStack:function(e){var t=S.merge(this.constructor(),e);return t.prevObject=this,t},each:function(e){return S.each(this,e)},map:function(n){return this.pushStack(S.map(this,function(e,t){return n.call(e,t,e)}))},slice:function(){return this.pushStack(s.apply(this,arguments))},first:function(){return this.eq(0)},last:function(){return this.eq(-1)},even:function(){return this.pushStack(S.grep(this,function(e,t){return(t+1)%2}))},odd:function(){return this.pushStack(S.grep(this,function(e,t){return t%2}))},eq:function(e){var t=this.length,n=+e+(e<0?t:0);return this.pushStack(0<=n&&n<t?[this[n]]:[])},end:function(){return this.prevObject||this.constructor()},push:u,sort:t.sort,splice:t.splice},S.extend=S.fn.extend=function(){var e,t,n,r,i,o,a=arguments[0]||{},s=1,u=arguments.length,l=!1;for("boolean"==typeof a&&(l=a,a=arguments[s]||{},s++),"object"==typeof a||m(a)||(a={}),s===u&&(a=this,s--);s<u;s++)if(null!=(e=arguments[s]))for(t in e)r=e[t],"__proto__"!==t&&a!==r&&(l&&r&&(S.isPlainObject(r)||(i=Array.isArray(r)))?(n=a[t],o=i&&!Array.isArray(n)?[]:i||S.isPlainObject(n)?n:{},i=!1,a[t]=S.extend(l,o,r)):void 0!==r&&(a[t]=r));return a},S.extend({expando:"jQuery"+(f+Math.random()).replace(/\D/g,""),isReady:!0,error:function(e){throw new Error(e)},noop:function(){},isPlainObject:function(e){var t,n;return!(!e||"[object Object]"!==o.call(e))&&(!(t=r(e))||"function"==typeof(n=v.call(t,"constructor")&&t.constructor)&&a.call(n)===l)},isEmptyObject:function(e){var t;for(t in e)return!1;return!0},globalEval:function(e,t,n){b(e,{nonce:t&&t.nonce},n)},each:function(e,t){var n,r=0;if(p(e)){for(n=e.length;r<n;r++)if(!1===t.call(e[r],r,e[r]))break}else for(r in e)if(!1===t.call(e[r],r,e[r]))break;return e},makeArray:function(e,t){var n=t||[];return null!=e&&(p(Object(e))?S.merge(n,"string"==typeof e?[e]:e):u.call(n,e)),n},inArray:function(e,t,n){return null==t?-1:i.call(t,e,n)},merge:function(e,t){for(var n=+t.length,r=0,i=e.length;r<n;r++)e[i++]=t[r];return e.length=i,e},grep:function(e,t,n){for(var r=[],i=0,o=e.length,a=!n;i<o;i++)!t(e[i],i)!==a&&r.push(e[i]);return r},map:function(e,t,n){var r,i,o=0,a=[];if(p(e))for(r=e.length;o<r;o++)null!=(i=t(e[o],o,n))&&a.push(i);else for(o in e)null!=(i=t(e[o],o,n))&&a.push(i);return g(a)},guid:1,support:y}),"function"==typeof Symbol&&(S.fn[Symbol.iterator]=t[Symbol.iterator]),S.each("Boolean Number String Function Array Date RegExp Object Error Symbol".split(" "),function(e,t){n["[object "+t+"]"]=t.toLowerCase()});var d=function(n){var e,d,b,o,i,h,f,g,w,u,l,T,C,a,E,v,s,c,y,S="sizzle"+1*new Date,p=n.document,k=0,r=0,m=ue(),x=ue(),A=ue(),N=ue(),D=function(e,t){return e===t&&(l=!0),0},j={}.hasOwnProperty,t=[],q=t.pop,L=t.push,H=t.push,O=t.slice,P=function(e,t){for(var n=0,r=e.length;n<r;n++)if(e[n]===t)return n;return-1},R="checked|selected|async|autofocus|autoplay|controls|defer|disabled|hidden|ismap|loop|multiple|open|readonly|required|scoped",M="[\\x20\\t\\r\\n\\f]",I="(?:\\\\[\\da-fA-F]{1,6}"+M+"?|\\\\[^\\r\\n\\f]|[\\w-]|[^\0-\\x7f])+",W="\\["+M+"*("+I+")(?:"+M+"*([*^$|!~]?=)"+M+"*(?:'((?:\\\\.|[^\\\\'])*)'|\"((?:\\\\.|[^\\\\\"])*)\"|("+I+"))|)"+M+"*\\]",F=":("+I+")(?:\\((('((?:\\\\.|[^\\\\'])*)'|\"((?:\\\\.|[^\\\\\"])*)\")|((?:\\\\.|[^\\\\()[\\]]|"+W+")*)|.*)\\)|)",B=new RegExp(M+"+","g"),$=new RegExp("^"+M+"+|((?:^|[^\\\\])(?:\\\\.)*)"+M+"+$","g"),_=new RegExp("^"+M+"*,"+M+"*"),z=new RegExp("^"+M+"*([>+~]|"+M+")"+M+"*"),U=new RegExp(M+"|>"),X=new RegExp(F),V=new RegExp("^"+I+"$"),G={ID:new RegExp("^#("+I+")"),CLASS:new RegExp("^\\.("+I+")"),TAG:new RegExp("^("+I+"|[*])"),ATTR:new RegExp("^"+W),PSEUDO:new RegExp("^"+F),CHILD:new RegExp("^:(only|first|last|nth|nth-last)-(child|of-type)(?:\\("+M+"*(even|odd|(([+-]|)(\\d*)n|)"+M+"*(?:([+-]|)"+M+"*(\\d+)|))"+M+"*\\)|)","i"),bool:new RegExp("^(?:"+R+")$","i"),needsContext:new RegExp("^"+M+"*[>+~]|:(even|odd|eq|gt|lt|nth|first|last)(?:\\("+M+"*((?:-\\d)?\\d*)"+M+"*\\)|)(?=[^-]|$)","i")},Y=/HTML$/i,Q=/^(?:input|select|textarea|button)$/i,J=/^h\d$/i,K=/^[^{]+\{\s*\[native \w/,Z=/^(?:#([\w-]+)|(\w+)|\.([\w-]+))$/,ee=/[+~]/,te=new RegExp("\\\\[\\da-fA-F]{1,6}"+M+"?|\\\\([^\\r\\n\\f])","g"),ne=function(e,t){var n="0x"+e.slice(1)-65536;return t||(n<0?String.fromCharCode(n+65536):String.fromCharCode(n>>10|55296,1023&n|56320))},re=/([\0-\x1f\x7f]|^-?\d)|^-$|[^\0-\x1f\x7f-\uFFFF\w-]/g,ie=function(e,t){return t?"\0"===e?"\ufffd":e.slice(0,-1)+"\\"+e.charCodeAt(e.length-1).toString(16)+" ":"\\"+e},oe=function(){T()},ae=be(function(e){return!0===e.disabled&&"fieldset"===e.nodeName.toLowerCase()},{dir:"parentNode",next:"legend"});try{H.apply(t=O.call(p.childNodes),p.childNodes),t[p.childNodes.length].nodeType}catch(e){H={apply:t.length?function(e,t){L.apply(e,O.call(t))}:function(e,t){var n=e.length,r=0;while(e[n++]=t[r++]);e.length=n-1}}}function se(t,e,n,r){var i,o,a,s,u,l,c,f=e&&e.ownerDocument,p=e?e.nodeType:9;if(n=n||[],"string"!=typeof t||!t||1!==p&&9!==p&&11!==p)return n;if(!r&&(T(e),e=e||C,E)){if(11!==p&&(u=Z.exec(t)))if(i=u[1]){if(9===p){if(!(a=e.getElementById(i)))return n;if(a.id===i)return n.push(a),n}else if(f&&(a=f.getElementById(i))&&y(e,a)&&a.id===i)return n.push(a),n}else{if(u[2])return H.apply(n,e.getElementsByTagName(t)),n;if((i=u[3])&&d.getElementsByClassName&&e.getElementsByClassName)return H.apply(n,e.getElementsByClassName(i)),n}if(d.qsa&&!N[t+" "]&&(!v||!v.test(t))&&(1!==p||"object"!==e.nodeName.toLowerCase())){if(c=t,f=e,1===p&&(U.test(t)||z.test(t))){(f=ee.test(t)&&ye(e.parentNode)||e)===e&&d.scope||((s=e.getAttribute("id"))?s=s.replace(re,ie):e.setAttribute("id",s=S)),o=(l=h(t)).length;while(o--)l[o]=(s?"#"+s:":scope")+" "+xe(l[o]);c=l.join(",")}try{return H.apply(n,f.querySelectorAll(c)),n}catch(e){N(t,!0)}finally{s===S&&e.removeAttribute("id")}}}return g(t.replace($,"$1"),e,n,r)}function ue(){var r=[];return function e(t,n){return r.push(t+" ")>b.cacheLength&&delete e[r.shift()],e[t+" "]=n}}function le(e){return e[S]=!0,e}function ce(e){var t=C.createElement("fieldset");try{return!!e(t)}catch(e){return!1}finally{t.parentNode&&t.parentNode.removeChild(t),t=null}}function fe(e,t){var n=e.split("|"),r=n.length;while(r--)b.attrHandle[n[r]]=t}function pe(e,t){var n=t&&e,r=n&&1===e.nodeType&&1===t.nodeType&&e.sourceIndex-t.sourceIndex;if(r)return r;if(n)while(n=n.nextSibling)if(n===t)return-1;return e?1:-1}function de(t){return function(e){return"input"===e.nodeName.toLowerCase()&&e.type===t}}function he(n){return function(e){var t=e.nodeName.toLowerCase();return("input"===t||"button"===t)&&e.type===n}}function ge(t){return function(e){return"form"in e?e.parentNode&&!1===e.disabled?"label"in e?"label"in e.parentNode?e.parentNode.disabled===t:e.disabled===t:e.isDisabled===t||e.isDisabled!==!t&&ae(e)===t:e.disabled===t:"label"in e&&e.disabled===t}}function ve(a){return le(function(o){return o=+o,le(function(e,t){var n,r=a([],e.length,o),i=r.length;while(i--)e[n=r[i]]&&(e[n]=!(t[n]=e[n]))})})}function ye(e){return e&&"undefined"!=typeof e.getElementsByTagName&&e}for(e in d=se.support={},i=se.isXML=function(e){var t=e.namespaceURI,n=(e.ownerDocument||e).documentElement;return!Y.test(t||n&&n.nodeName||"HTML")},T=se.setDocument=function(e){var t,n,r=e?e.ownerDocument||e:p;return r!=C&&9===r.nodeType&&r.documentElement&&(a=(C=r).documentElement,E=!i(C),p!=C&&(n=C.defaultView)&&n.top!==n&&(n.addEventListener?n.addEventListener("unload",oe,!1):n.attachEvent&&n.attachEvent("onunload",oe)),d.scope=ce(function(e){return a.appendChild(e).appendChild(C.createElement("div")),"undefined"!=typeof e.querySelectorAll&&!e.querySelectorAll(":scope fieldset div").length}),d.attributes=ce(function(e){return e.className="i",!e.getAttribute("className")}),d.getElementsByTagName=ce(function(e){return e.appendChild(C.createComment("")),!e.getElementsByTagName("*").length}),d.getElementsByClassName=K.test(C.getElementsByClassName),d.getById=ce(function(e){return a.appendChild(e).id=S,!C.getElementsByName||!C.getElementsByName(S).length}),d.getById?(b.filter.ID=function(e){var t=e.replace(te,ne);return function(e){return e.getAttribute("id")===t}},b.find.ID=function(e,t){if("undefined"!=typeof t.getElementById&&E){var n=t.getElementById(e);return n?[n]:[]}}):(b.filter.ID=function(e){var n=e.replace(te,ne);return function(e){var t="undefined"!=typeof e.getAttributeNode&&e.getAttributeNode("id");return t&&t.value===n}},b.find.ID=function(e,t){if("undefined"!=typeof t.getElementById&&E){var n,r,i,o=t.getElementById(e);if(o){if((n=o.getAttributeNode("id"))&&n.value===e)return[o];i=t.getElementsByName(e),r=0;while(o=i[r++])if((n=o.getAttributeNode("id"))&&n.value===e)return[o]}return[]}}),b.find.TAG=d.getElementsByTagName?function(e,t){return"undefined"!=typeof t.getElementsByTagName?t.getElementsByTagName(e):d.qsa?t.querySelectorAll(e):void 0}:function(e,t){var n,r=[],i=0,o=t.getElementsByTagName(e);if("*"===e){while(n=o[i++])1===n.nodeType&&r.push(n);return r}return o},b.find.CLASS=d.getElementsByClassName&&function(e,t){if("undefined"!=typeof t.getElementsByClassName&&E)return t.getElementsByClassName(e)},s=[],v=[],(d.qsa=K.test(C.querySelectorAll))&&(ce(function(e){var t;a.appendChild(e).innerHTML="<a id='"+S+"'></a><select id='"+S+"-\r\\' msallowcapture=''><option selected=''></option></select>",e.querySelectorAll("[msallowcapture^='']").length&&v.push("[*^$]="+M+"*(?:''|\"\")"),e.querySelectorAll("[selected]").length||v.push("\\["+M+"*(?:value|"+R+")"),e.querySelectorAll("[id~="+S+"-]").length||v.push("~="),(t=C.createElement("input")).setAttribute("name",""),e.appendChild(t),e.querySelectorAll("[name='']").length||v.push("\\["+M+"*name"+M+"*="+M+"*(?:''|\"\")"),e.querySelectorAll(":checked").length||v.push(":checked"),e.querySelectorAll("a#"+S+"+*").length||v.push(".#.+[+~]"),e.querySelectorAll("\\\f"),v.push("[\\r\\n\\f]")}),ce(function(e){e.innerHTML="<a href='' disabled='disabled'></a><select disabled='disabled'><option/></select>";var t=C.createElement("input");t.setAttribute("type","hidden"),e.appendChild(t).setAttribute("name","D"),e.querySelectorAll("[name=d]").length&&v.push("name"+M+"*[*^$|!~]?="),2!==e.querySelectorAll(":enabled").length&&v.push(":enabled",":disabled"),a.appendChild(e).disabled=!0,2!==e.querySelectorAll(":disabled").length&&v.push(":enabled",":disabled"),e.querySelectorAll("*,:x"),v.push(",.*:")})),(d.matchesSelector=K.test(c=a.matches||a.webkitMatchesSelector||a.mozMatchesSelector||a.oMatchesSelector||a.msMatchesSelector))&&ce(function(e){d.disconnectedMatch=c.call(e,"*"),c.call(e,"[s!='']:x"),s.push("!=",F)}),v=v.length&&new RegExp(v.join("|")),s=s.length&&new RegExp(s.join("|")),t=K.test(a.compareDocumentPosition),y=t||K.test(a.contains)?function(e,t){var n=9===e.nodeType?e.documentElement:e,r=t&&t.parentNode;return e===r||!(!r||1!==r.nodeType||!(n.contains?n.contains(r):e.compareDocumentPosition&&16&e.compareDocumentPosition(r)))}:function(e,t){if(t)while(t=t.parentNode)if(t===e)return!0;return!1},D=t?function(e,t){if(e===t)return l=!0,0;var n=!e.compareDocumentPosition-!t.compareDocumentPosition;return n||(1&(n=(e.ownerDocument||e)==(t.ownerDocument||t)?e.compareDocumentPosition(t):1)||!d.sortDetached&&t.compareDocumentPosition(e)===n?e==C||e.ownerDocument==p&&y(p,e)?-1:t==C||t.ownerDocument==p&&y(p,t)?1:u?P(u,e)-P(u,t):0:4&n?-1:1)}:function(e,t){if(e===t)return l=!0,0;var n,r=0,i=e.parentNode,o=t.parentNode,a=[e],s=[t];if(!i||!o)return e==C?-1:t==C?1:i?-1:o?1:u?P(u,e)-P(u,t):0;if(i===o)return pe(e,t);n=e;while(n=n.parentNode)a.unshift(n);n=t;while(n=n.parentNode)s.unshift(n);while(a[r]===s[r])r++;return r?pe(a[r],s[r]):a[r]==p?-1:s[r]==p?1:0}),C},se.matches=function(e,t){return se(e,null,null,t)},se.matchesSelector=function(e,t){if(T(e),d.matchesSelector&&E&&!N[t+" "]&&(!s||!s.test(t))&&(!v||!v.test(t)))try{var n=c.call(e,t);if(n||d.disconnectedMatch||e.document&&11!==e.document.nodeType)return n}catch(e){N(t,!0)}return 0<se(t,C,null,[e]).length},se.contains=function(e,t){return(e.ownerDocument||e)!=C&&T(e),y(e,t)},se.attr=function(e,t){(e.ownerDocument||e)!=C&&T(e);var n=b.attrHandle[t.toLowerCase()],r=n&&j.call(b.attrHandle,t.toLowerCase())?n(e,t,!E):void 0;return void 0!==r?r:d.attributes||!E?e.getAttribute(t):(r=e.getAttributeNode(t))&&r.specified?r.value:null},se.escape=function(e){return(e+"").replace(re,ie)},se.error=function(e){throw new Error("Syntax error, unrecognized expression: "+e)},se.uniqueSort=function(e){var t,n=[],r=0,i=0;if(l=!d.detectDuplicates,u=!d.sortStable&&e.slice(0),e.sort(D),l){while(t=e[i++])t===e[i]&&(r=n.push(i));while(r--)e.splice(n[r],1)}return u=null,e},o=se.getText=function(e){var t,n="",r=0,i=e.nodeType;if(i){if(1===i||9===i||11===i){if("string"==typeof e.textContent)return e.textContent;for(e=e.firstChild;e;e=e.nextSibling)n+=o(e)}else if(3===i||4===i)return e.nodeValue}else while(t=e[r++])n+=o(t);return n},(b=se.selectors={cacheLength:50,createPseudo:le,match:G,attrHandle:{},find:{},relative:{">":{dir:"parentNode",first:!0}," ":{dir:"parentNode"},"+":{dir:"previousSibling",first:!0},"~":{dir:"previousSibling"}},preFilter:{ATTR:function(e){return e[1]=e[1].replace(te,ne),e[3]=(e[3]||e[4]||e[5]||"").replace(te,ne),"~="===e[2]&&(e[3]=" "+e[3]+" "),e.slice(0,4)},CHILD:function(e){return e[1]=e[1].toLowerCase(),"nth"===e[1].slice(0,3)?(e[3]||se.error(e[0]),e[4]=+(e[4]?e[5]+(e[6]||1):2*("even"===e[3]||"odd"===e[3])),e[5]=+(e[7]+e[8]||"odd"===e[3])):e[3]&&se.error(e[0]),e},PSEUDO:function(e){var t,n=!e[6]&&e[2];return G.CHILD.test(e[0])?null:(e[3]?e[2]=e[4]||e[5]||"":n&&X.test(n)&&(t=h(n,!0))&&(t=n.indexOf(")",n.length-t)-n.length)&&(e[0]=e[0].slice(0,t),e[2]=n.slice(0,t)),e.slice(0,3))}},filter:{TAG:function(e){var t=e.replace(te,ne).toLowerCase();return"*"===e?function(){return!0}:function(e){return e.nodeName&&e.nodeName.toLowerCase()===t}},CLASS:function(e){var t=m[e+" "];return t||(t=new RegExp("(^|"+M+")"+e+"("+M+"|$)"))&&m(e,function(e){return t.test("string"==typeof e.className&&e.className||"undefined"!=typeof e.getAttribute&&e.getAttribute("class")||"")})},ATTR:function(n,r,i){return function(e){var t=se.attr(e,n);return null==t?"!="===r:!r||(t+="","="===r?t===i:"!="===r?t!==i:"^="===r?i&&0===t.indexOf(i):"*="===r?i&&-1<t.indexOf(i):"$="===r?i&&t.slice(-i.length)===i:"~="===r?-1<(" "+t.replace(B," ")+" ").indexOf(i):"|="===r&&(t===i||t.slice(0,i.length+1)===i+"-"))}},CHILD:function(h,e,t,g,v){var y="nth"!==h.slice(0,3),m="last"!==h.slice(-4),x="of-type"===e;return 1===g&&0===v?function(e){return!!e.parentNode}:function(e,t,n){var r,i,o,a,s,u,l=y!==m?"nextSibling":"previousSibling",c=e.parentNode,f=x&&e.nodeName.toLowerCase(),p=!n&&!x,d=!1;if(c){if(y){while(l){a=e;while(a=a[l])if(x?a.nodeName.toLowerCase()===f:1===a.nodeType)return!1;u=l="only"===h&&!u&&"nextSibling"}return!0}if(u=[m?c.firstChild:c.lastChild],m&&p){d=(s=(r=(i=(o=(a=c)[S]||(a[S]={}))[a.uniqueID]||(o[a.uniqueID]={}))[h]||[])[0]===k&&r[1])&&r[2],a=s&&c.childNodes[s];while(a=++s&&a&&a[l]||(d=s=0)||u.pop())if(1===a.nodeType&&++d&&a===e){i[h]=[k,s,d];break}}else if(p&&(d=s=(r=(i=(o=(a=e)[S]||(a[S]={}))[a.uniqueID]||(o[a.uniqueID]={}))[h]||[])[0]===k&&r[1]),!1===d)while(a=++s&&a&&a[l]||(d=s=0)||u.pop())if((x?a.nodeName.toLowerCase()===f:1===a.nodeType)&&++d&&(p&&((i=(o=a[S]||(a[S]={}))[a.uniqueID]||(o[a.uniqueID]={}))[h]=[k,d]),a===e))break;return(d-=v)===g||d%g==0&&0<=d/g}}},PSEUDO:function(e,o){var t,a=b.pseudos[e]||b.setFilters[e.toLowerCase()]||se.error("unsupported pseudo: "+e);return a[S]?a(o):1<a.length?(t=[e,e,"",o],b.setFilters.hasOwnProperty(e.toLowerCase())?le(function(e,t){var n,r=a(e,o),i=r.length;while(i--)e[n=P(e,r[i])]=!(t[n]=r[i])}):function(e){return a(e,0,t)}):a}},pseudos:{not:le(function(e){var r=[],i=[],s=f(e.replace($,"$1"));return s[S]?le(function(e,t,n,r){var i,o=s(e,null,r,[]),a=e.length;while(a--)(i=o[a])&&(e[a]=!(t[a]=i))}):function(e,t,n){return r[0]=e,s(r,null,n,i),r[0]=null,!i.pop()}}),has:le(function(t){return function(e){return 0<se(t,e).length}}),contains:le(function(t){return t=t.replace(te,ne),function(e){return-1<(e.textContent||o(e)).indexOf(t)}}),lang:le(function(n){return V.test(n||"")||se.error("unsupported lang: "+n),n=n.replace(te,ne).toLowerCase(),function(e){var t;do{if(t=E?e.lang:e.getAttribute("xml:lang")||e.getAttribute("lang"))return(t=t.toLowerCase())===n||0===t.indexOf(n+"-")}while((e=e.parentNode)&&1===e.nodeType);return!1}}),target:function(e){var t=n.location&&n.location.hash;return t&&t.slice(1)===e.id},root:function(e){return e===a},focus:function(e){return e===C.activeElement&&(!C.hasFocus||C.hasFocus())&&!!(e.type||e.href||~e.tabIndex)},enabled:ge(!1),disabled:ge(!0),checked:function(e){var t=e.nodeName.toLowerCase();return"input"===t&&!!e.checked||"option"===t&&!!e.selected},selected:function(e){return e.parentNode&&e.parentNode.selectedIndex,!0===e.selected},empty:function(e){for(e=e.firstChild;e;e=e.nextSibling)if(e.nodeType<6)return!1;return!0},parent:function(e){return!b.pseudos.empty(e)},header:function(e){return J.test(e.nodeName)},input:function(e){return Q.test(e.nodeName)},button:function(e){var t=e.nodeName.toLowerCase();return"input"===t&&"button"===e.type||"button"===t},text:function(e){var t;return"input"===e.nodeName.toLowerCase()&&"text"===e.type&&(null==(t=e.getAttribute("type"))||"text"===t.toLowerCase())},first:ve(function(){return[0]}),last:ve(function(e,t){return[t-1]}),eq:ve(function(e,t,n){return[n<0?n+t:n]}),even:ve(function(e,t){for(var n=0;n<t;n+=2)e.push(n);return e}),odd:ve(function(e,t){for(var n=1;n<t;n+=2)e.push(n);return e}),lt:ve(function(e,t,n){for(var r=n<0?n+t:t<n?t:n;0<=--r;)e.push(r);return e}),gt:ve(function(e,t,n){for(var r=n<0?n+t:n;++r<t;)e.push(r);return e})}}).pseudos.nth=b.pseudos.eq,{radio:!0,checkbox:!0,file:!0,password:!0,image:!0})b.pseudos[e]=de(e);for(e in{submit:!0,reset:!0})b.pseudos[e]=he(e);function me(){}function xe(e){for(var t=0,n=e.length,r="";t<n;t++)r+=e[t].value;return r}function be(s,e,t){var u=e.dir,l=e.next,c=l||u,f=t&&"parentNode"===c,p=r++;return e.first?function(e,t,n){while(e=e[u])if(1===e.nodeType||f)return s(e,t,n);return!1}:function(e,t,n){var r,i,o,a=[k,p];if(n){while(e=e[u])if((1===e.nodeType||f)&&s(e,t,n))return!0}else while(e=e[u])if(1===e.nodeType||f)if(i=(o=e[S]||(e[S]={}))[e.uniqueID]||(o[e.uniqueID]={}),l&&l===e.nodeName.toLowerCase())e=e[u]||e;else{if((r=i[c])&&r[0]===k&&r[1]===p)return a[2]=r[2];if((i[c]=a)[2]=s(e,t,n))return!0}return!1}}function we(i){return 1<i.length?function(e,t,n){var r=i.length;while(r--)if(!i[r](e,t,n))return!1;return!0}:i[0]}function Te(e,t,n,r,i){for(var o,a=[],s=0,u=e.length,l=null!=t;s<u;s++)(o=e[s])&&(n&&!n(o,r,i)||(a.push(o),l&&t.push(s)));return a}function Ce(d,h,g,v,y,e){return v&&!v[S]&&(v=Ce(v)),y&&!y[S]&&(y=Ce(y,e)),le(function(e,t,n,r){var i,o,a,s=[],u=[],l=t.length,c=e||function(e,t,n){for(var r=0,i=t.length;r<i;r++)se(e,t[r],n);return n}(h||"*",n.nodeType?[n]:n,[]),f=!d||!e&&h?c:Te(c,s,d,n,r),p=g?y||(e?d:l||v)?[]:t:f;if(g&&g(f,p,n,r),v){i=Te(p,u),v(i,[],n,r),o=i.length;while(o--)(a=i[o])&&(p[u[o]]=!(f[u[o]]=a))}if(e){if(y||d){if(y){i=[],o=p.length;while(o--)(a=p[o])&&i.push(f[o]=a);y(null,p=[],i,r)}o=p.length;while(o--)(a=p[o])&&-1<(i=y?P(e,a):s[o])&&(e[i]=!(t[i]=a))}}else p=Te(p===t?p.splice(l,p.length):p),y?y(null,t,p,r):H.apply(t,p)})}function Ee(e){for(var i,t,n,r=e.length,o=b.relative[e[0].type],a=o||b.relative[" "],s=o?1:0,u=be(function(e){return e===i},a,!0),l=be(function(e){return-1<P(i,e)},a,!0),c=[function(e,t,n){var r=!o&&(n||t!==w)||((i=t).nodeType?u(e,t,n):l(e,t,n));return i=null,r}];s<r;s++)if(t=b.relative[e[s].type])c=[be(we(c),t)];else{if((t=b.filter[e[s].type].apply(null,e[s].matches))[S]){for(n=++s;n<r;n++)if(b.relative[e[n].type])break;return Ce(1<s&&we(c),1<s&&xe(e.slice(0,s-1).concat({value:" "===e[s-2].type?"*":""})).replace($,"$1"),t,s<n&&Ee(e.slice(s,n)),n<r&&Ee(e=e.slice(n)),n<r&&xe(e))}c.push(t)}return we(c)}return me.prototype=b.filters=b.pseudos,b.setFilters=new me,h=se.tokenize=function(e,t){var n,r,i,o,a,s,u,l=x[e+" "];if(l)return t?0:l.slice(0);a=e,s=[],u=b.preFilter;while(a){for(o in n&&!(r=_.exec(a))||(r&&(a=a.slice(r[0].length)||a),s.push(i=[])),n=!1,(r=z.exec(a))&&(n=r.shift(),i.push({value:n,type:r[0].replace($," ")}),a=a.slice(n.length)),b.filter)!(r=G[o].exec(a))||u[o]&&!(r=u[o](r))||(n=r.shift(),i.push({value:n,type:o,matches:r}),a=a.slice(n.length));if(!n)break}return t?a.length:a?se.error(e):x(e,s).slice(0)},f=se.compile=function(e,t){var n,v,y,m,x,r,i=[],o=[],a=A[e+" "];if(!a){t||(t=h(e)),n=t.length;while(n--)(a=Ee(t[n]))[S]?i.push(a):o.push(a);(a=A(e,(v=o,m=0<(y=i).length,x=0<v.length,r=function(e,t,n,r,i){var o,a,s,u=0,l="0",c=e&&[],f=[],p=w,d=e||x&&b.find.TAG("*",i),h=k+=null==p?1:Math.random()||.1,g=d.length;for(i&&(w=t==C||t||i);l!==g&&null!=(o=d[l]);l++){if(x&&o){a=0,t||o.ownerDocument==C||(T(o),n=!E);while(s=v[a++])if(s(o,t||C,n)){r.push(o);break}i&&(k=h)}m&&((o=!s&&o)&&u--,e&&c.push(o))}if(u+=l,m&&l!==u){a=0;while(s=y[a++])s(c,f,t,n);if(e){if(0<u)while(l--)c[l]||f[l]||(f[l]=q.call(r));f=Te(f)}H.apply(r,f),i&&!e&&0<f.length&&1<u+y.length&&se.uniqueSort(r)}return i&&(k=h,w=p),c},m?le(r):r))).selector=e}return a},g=se.select=function(e,t,n,r){var i,o,a,s,u,l="function"==typeof e&&e,c=!r&&h(e=l.selector||e);if(n=n||[],1===c.length){if(2<(o=c[0]=c[0].slice(0)).length&&"ID"===(a=o[0]).type&&9===t.nodeType&&E&&b.relative[o[1].type]){if(!(t=(b.find.ID(a.matches[0].replace(te,ne),t)||[])[0]))return n;l&&(t=t.parentNode),e=e.slice(o.shift().value.length)}i=G.needsContext.test(e)?0:o.length;while(i--){if(a=o[i],b.relative[s=a.type])break;if((u=b.find[s])&&(r=u(a.matches[0].replace(te,ne),ee.test(o[0].type)&&ye(t.parentNode)||t))){if(o.splice(i,1),!(e=r.length&&xe(o)))return H.apply(n,r),n;break}}}return(l||f(e,c))(r,t,!E,n,!t||ee.test(e)&&ye(t.parentNode)||t),n},d.sortStable=S.split("").sort(D).join("")===S,d.detectDuplicates=!!l,T(),d.sortDetached=ce(function(e){return 1&e.compareDocumentPosition(C.createElement("fieldset"))}),ce(function(e){return e.innerHTML="<a href='#'></a>","#"===e.firstChild.getAttribute("href")})||fe("type|href|height|width",function(e,t,n){if(!n)return e.getAttribute(t,"type"===t.toLowerCase()?1:2)}),d.attributes&&ce(function(e){return e.innerHTML="<input/>",e.firstChild.setAttribute("value",""),""===e.firstChild.getAttribute("value")})||fe("value",function(e,t,n){if(!n&&"input"===e.nodeName.toLowerCase())return e.defaultValue}),ce(function(e){return null==e.getAttribute("disabled")})||fe(R,function(e,t,n){var r;if(!n)return!0===e[t]?t.toLowerCase():(r=e.getAttributeNode(t))&&r.specified?r.value:null}),se}(C);S.find=d,S.expr=d.selectors,S.expr[":"]=S.expr.pseudos,S.uniqueSort=S.unique=d.uniqueSort,S.text=d.getText,S.isXMLDoc=d.isXML,S.contains=d.contains,S.escapeSelector=d.escape;var h=function(e,t,n){var r=[],i=void 0!==n;while((e=e[t])&&9!==e.nodeType)if(1===e.nodeType){if(i&&S(e).is(n))break;r.push(e)}return r},T=function(e,t){for(var n=[];e;e=e.nextSibling)1===e.nodeType&&e!==t&&n.push(e);return n},k=S.expr.match.needsContext;function A(e,t){return e.nodeName&&e.nodeName.toLowerCase()===t.toLowerCase()}var N=/^<([a-z][^\/\0>:\x20\t\r\n\f]*)[\x20\t\r\n\f]*\/?>(?:<\/\1>|)$/i;function D(e,n,r){return m(n)?S.grep(e,function(e,t){return!!n.call(e,t,e)!==r}):n.nodeType?S.grep(e,function(e){return e===n!==r}):"string"!=typeof n?S.grep(e,function(e){return-1<i.call(n,e)!==r}):S.filter(n,e,r)}S.filter=function(e,t,n){var r=t[0];return n&&(e=":not("+e+")"),1===t.length&&1===r.nodeType?S.find.matchesSelector(r,e)?[r]:[]:S.find.matches(e,S.grep(t,function(e){return 1===e.nodeType}))},S.fn.extend({find:function(e){var t,n,r=this.length,i=this;if("string"!=typeof e)return this.pushStack(S(e).filter(function(){for(t=0;t<r;t++)if(S.contains(i[t],this))return!0}));for(n=this.pushStack([]),t=0;t<r;t++)S.find(e,i[t],n);return 1<r?S.uniqueSort(n):n},filter:function(e){return this.pushStack(D(this,e||[],!1))},not:function(e){return this.pushStack(D(this,e||[],!0))},is:function(e){return!!D(this,"string"==typeof e&&k.test(e)?S(e):e||[],!1).length}});var j,q=/^(?:\s*(<[\w\W]+>)[^>]*|#([\w-]+))$/;(S.fn.init=function(e,t,n){var r,i;if(!e)return this;if(n=n||j,"string"==typeof e){if(!(r="<"===e[0]&&">"===e[e.length-1]&&3<=e.length?[null,e,null]:q.exec(e))||!r[1]&&t)return!t||t.jquery?(t||n).find(e):this.constructor(t).find(e);if(r[1]){if(t=t instanceof S?t[0]:t,S.merge(this,S.parseHTML(r[1],t&&t.nodeType?t.ownerDocument||t:E,!0)),N.test(r[1])&&S.isPlainObject(t))for(r in t)m(this[r])?this[r](t[r]):this.attr(r,t[r]);return this}return(i=E.getElementById(r[2]))&&(this[0]=i,this.length=1),this}return e.nodeType?(this[0]=e,this.length=1,this):m(e)?void 0!==n.ready?n.ready(e):e(S):S.makeArray(e,this)}).prototype=S.fn,j=S(E);var L=/^(?:parents|prev(?:Until|All))/,H={children:!0,contents:!0,next:!0,prev:!0};function O(e,t){while((e=e[t])&&1!==e.nodeType);return e}S.fn.extend({has:function(e){var t=S(e,this),n=t.length;return this.filter(function(){for(var e=0;e<n;e++)if(S.contains(this,t[e]))return!0})},closest:function(e,t){var n,r=0,i=this.length,o=[],a="string"!=typeof e&&S(e);if(!k.test(e))for(;r<i;r++)for(n=this[r];n&&n!==t;n=n.parentNode)if(n.nodeType<11&&(a?-1<a.index(n):1===n.nodeType&&S.find.matchesSelector(n,e))){o.push(n);break}return this.pushStack(1<o.length?S.uniqueSort(o):o)},index:function(e){return e?"string"==typeof e?i.call(S(e),this[0]):i.call(this,e.jquery?e[0]:e):this[0]&&this[0].parentNode?this.first().prevAll().length:-1},add:function(e,t){return this.pushStack(S.uniqueSort(S.merge(this.get(),S(e,t))))},addBack:function(e){return this.add(null==e?this.prevObject:this.prevObject.filter(e))}}),S.each({parent:function(e){var t=e.parentNode;return t&&11!==t.nodeType?t:null},parents:function(e){return h(e,"parentNode")},parentsUntil:function(e,t,n){return h(e,"parentNode",n)},next:function(e){return O(e,"nextSibling")},prev:function(e){return O(e,"previousSibling")},nextAll:function(e){return h(e,"nextSibling")},prevAll:function(e){return h(e,"previousSibling")},nextUntil:function(e,t,n){return h(e,"nextSibling",n)},prevUntil:function(e,t,n){return h(e,"previousSibling",n)},siblings:function(e){return T((e.parentNode||{}).firstChild,e)},children:function(e){return T(e.firstChild)},contents:function(e){return null!=e.contentDocument&&r(e.contentDocument)?e.contentDocument:(A(e,"template")&&(e=e.content||e),S.merge([],e.childNodes))}},function(r,i){S.fn[r]=function(e,t){var n=S.map(this,i,e);return"Until"!==r.slice(-5)&&(t=e),t&&"string"==typeof t&&(n=S.filter(t,n)),1<this.length&&(H[r]||S.uniqueSort(n),L.test(r)&&n.reverse()),this.pushStack(n)}});var P=/[^\x20\t\r\n\f]+/g;function R(e){return e}function M(e){throw e}function I(e,t,n,r){var i;try{e&&m(i=e.promise)?i.call(e).done(t).fail(n):e&&m(i=e.then)?i.call(e,t,n):t.apply(void 0,[e].slice(r))}catch(e){n.apply(void 0,[e])}}S.Callbacks=function(r){var e,n;r="string"==typeof r?(e=r,n={},S.each(e.match(P)||[],function(e,t){n[t]=!0}),n):S.extend({},r);var i,t,o,a,s=[],u=[],l=-1,c=function(){for(a=a||r.once,o=i=!0;u.length;l=-1){t=u.shift();while(++l<s.length)!1===s[l].apply(t[0],t[1])&&r.stopOnFalse&&(l=s.length,t=!1)}r.memory||(t=!1),i=!1,a&&(s=t?[]:"")},f={add:function(){return s&&(t&&!i&&(l=s.length-1,u.push(t)),function n(e){S.each(e,function(e,t){m(t)?r.unique&&f.has(t)||s.push(t):t&&t.length&&"string"!==w(t)&&n(t)})}(arguments),t&&!i&&c()),this},remove:function(){return S.each(arguments,function(e,t){var n;while(-1<(n=S.inArray(t,s,n)))s.splice(n,1),n<=l&&l--}),this},has:function(e){return e?-1<S.inArray(e,s):0<s.length},empty:function(){return s&&(s=[]),this},disable:function(){return a=u=[],s=t="",this},disabled:function(){return!s},lock:function(){return a=u=[],t||i||(s=t=""),this},locked:function(){return!!a},fireWith:function(e,t){return a||(t=[e,(t=t||[]).slice?t.slice():t],u.push(t),i||c()),this},fire:function(){return f.fireWith(this,arguments),this},fired:function(){return!!o}};return f},S.extend({Deferred:function(e){var o=[["notify","progress",S.Callbacks("memory"),S.Callbacks("memory"),2],["resolve","done",S.Callbacks("once memory"),S.Callbacks("once memory"),0,"resolved"],["reject","fail",S.Callbacks("once memory"),S.Callbacks("once memory"),1,"rejected"]],i="pending",a={state:function(){return i},always:function(){return s.done(arguments).fail(arguments),this},"catch":function(e){return a.then(null,e)},pipe:function(){var i=arguments;return S.Deferred(function(r){S.each(o,function(e,t){var n=m(i[t[4]])&&i[t[4]];s[t[1]](function(){var e=n&&n.apply(this,arguments);e&&m(e.promise)?e.promise().progress(r.notify).done(r.resolve).fail(r.reject):r[t[0]+"With"](this,n?[e]:arguments)})}),i=null}).promise()},then:function(t,n,r){var u=0;function l(i,o,a,s){return function(){var n=this,r=arguments,e=function(){var e,t;if(!(i<u)){if((e=a.apply(n,r))===o.promise())throw new TypeError("Thenable self-resolution");t=e&&("object"==typeof e||"function"==typeof e)&&e.then,m(t)?s?t.call(e,l(u,o,R,s),l(u,o,M,s)):(u++,t.call(e,l(u,o,R,s),l(u,o,M,s),l(u,o,R,o.notifyWith))):(a!==R&&(n=void 0,r=[e]),(s||o.resolveWith)(n,r))}},t=s?e:function(){try{e()}catch(e){S.Deferred.exceptionHook&&S.Deferred.exceptionHook(e,t.stackTrace),u<=i+1&&(a!==M&&(n=void 0,r=[e]),o.rejectWith(n,r))}};i?t():(S.Deferred.getStackHook&&(t.stackTrace=S.Deferred.getStackHook()),C.setTimeout(t))}}return S.Deferred(function(e){o[0][3].add(l(0,e,m(r)?r:R,e.notifyWith)),o[1][3].add(l(0,e,m(t)?t:R)),o[2][3].add(l(0,e,m(n)?n:M))}).promise()},promise:function(e){return null!=e?S.extend(e,a):a}},s={};return S.each(o,function(e,t){var n=t[2],r=t[5];a[t[1]]=n.add,r&&n.add(function(){i=r},o[3-e][2].disable,o[3-e][3].disable,o[0][2].lock,o[0][3].lock),n.add(t[3].fire),s[t[0]]=function(){return s[t[0]+"With"](this===s?void 0:this,arguments),this},s[t[0]+"With"]=n.fireWith}),a.promise(s),e&&e.call(s,s),s},when:function(e){var n=arguments.length,t=n,r=Array(t),i=s.call(arguments),o=S.Deferred(),a=function(t){return function(e){r[t]=this,i[t]=1<arguments.length?s.call(arguments):e,--n||o.resolveWith(r,i)}};if(n<=1&&(I(e,o.done(a(t)).resolve,o.reject,!n),"pending"===o.state()||m(i[t]&&i[t].then)))return o.then();while(t--)I(i[t],a(t),o.reject);return o.promise()}});var W=/^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;S.Deferred.exceptionHook=function(e,t){C.console&&C.console.warn&&e&&W.test(e.name)&&C.console.warn("jQuery.Deferred exception: "+e.message,e.stack,t)},S.readyException=function(e){C.setTimeout(function(){throw e})};var F=S.Deferred();function B(){E.removeEventListener("DOMContentLoaded",B),C.removeEventListener("load",B),S.ready()}S.fn.ready=function(e){return F.then(e)["catch"](function(e){S.readyException(e)}),this},S.extend({isReady:!1,readyWait:1,ready:function(e){(!0===e?--S.readyWait:S.isReady)||(S.isReady=!0)!==e&&0<--S.readyWait||F.resolveWith(E,[S])}}),S.ready.then=F.then,"complete"===E.readyState||"loading"!==E.readyState&&!E.documentElement.doScroll?C.setTimeout(S.ready):(E.addEventListener("DOMContentLoaded",B),C.addEventListener("load",B));var $=function(e,t,n,r,i,o,a){var s=0,u=e.length,l=null==n;if("object"===w(n))for(s in i=!0,n)$(e,t,s,n[s],!0,o,a);else if(void 0!==r&&(i=!0,m(r)||(a=!0),l&&(a?(t.call(e,r),t=null):(l=t,t=function(e,t,n){return l.call(S(e),n)})),t))for(;s<u;s++)t(e[s],n,a?r:r.call(e[s],s,t(e[s],n)));return i?e:l?t.call(e):u?t(e[0],n):o},_=/^-ms-/,z=/-([a-z])/g;function U(e,t){return t.toUpperCase()}function X(e){return e.replace(_,"ms-").replace(z,U)}var V=function(e){return 1===e.nodeType||9===e.nodeType||!+e.nodeType};function G(){this.expando=S.expando+G.uid++}G.uid=1,G.prototype={cache:function(e){var t=e[this.expando];return t||(t={},V(e)&&(e.nodeType?e[this.expando]=t:Object.defineProperty(e,this.expando,{value:t,configurable:!0}))),t},set:function(e,t,n){var r,i=this.cache(e);if("string"==typeof t)i[X(t)]=n;else for(r in t)i[X(r)]=t[r];return i},get:function(e,t){return void 0===t?this.cache(e):e[this.expando]&&e[this.expando][X(t)]},access:function(e,t,n){return void 0===t||t&&"string"==typeof t&&void 0===n?this.get(e,t):(this.set(e,t,n),void 0!==n?n:t)},remove:function(e,t){var n,r=e[this.expando];if(void 0!==r){if(void 0!==t){n=(t=Array.isArray(t)?t.map(X):(t=X(t))in r?[t]:t.match(P)||[]).length;while(n--)delete r[t[n]]}(void 0===t||S.isEmptyObject(r))&&(e.nodeType?e[this.expando]=void 0:delete e[this.expando])}},hasData:function(e){var t=e[this.expando];return void 0!==t&&!S.isEmptyObject(t)}};var Y=new G,Q=new G,J=/^(?:\{[\w\W]*\}|\[[\w\W]*\])$/,K=/[A-Z]/g;function Z(e,t,n){var r,i;if(void 0===n&&1===e.nodeType)if(r="data-"+t.replace(K,"-$&").toLowerCase(),"string"==typeof(n=e.getAttribute(r))){try{n="true"===(i=n)||"false"!==i&&("null"===i?null:i===+i+""?+i:J.test(i)?JSON.parse(i):i)}catch(e){}Q.set(e,t,n)}else n=void 0;return n}S.extend({hasData:function(e){return Q.hasData(e)||Y.hasData(e)},data:function(e,t,n){return Q.access(e,t,n)},removeData:function(e,t){Q.remove(e,t)},_data:function(e,t,n){return Y.access(e,t,n)},_removeData:function(e,t){Y.remove(e,t)}}),S.fn.extend({data:function(n,e){var t,r,i,o=this[0],a=o&&o.attributes;if(void 0===n){if(this.length&&(i=Q.get(o),1===o.nodeType&&!Y.get(o,"hasDataAttrs"))){t=a.length;while(t--)a[t]&&0===(r=a[t].name).indexOf("data-")&&(r=X(r.slice(5)),Z(o,r,i[r]));Y.set(o,"hasDataAttrs",!0)}return i}return"object"==typeof n?this.each(function(){Q.set(this,n)}):$(this,function(e){var t;if(o&&void 0===e)return void 0!==(t=Q.get(o,n))?t:void 0!==(t=Z(o,n))?t:void 0;this.each(function(){Q.set(this,n,e)})},null,e,1<arguments.length,null,!0)},removeData:function(e){return this.each(function(){Q.remove(this,e)})}}),S.extend({queue:function(e,t,n){var r;if(e)return t=(t||"fx")+"queue",r=Y.get(e,t),n&&(!r||Array.isArray(n)?r=Y.access(e,t,S.makeArray(n)):r.push(n)),r||[]},dequeue:function(e,t){t=t||"fx";var n=S.queue(e,t),r=n.length,i=n.shift(),o=S._queueHooks(e,t);"inprogress"===i&&(i=n.shift(),r--),i&&("fx"===t&&n.unshift("inprogress"),delete o.stop,i.call(e,function(){S.dequeue(e,t)},o)),!r&&o&&o.empty.fire()},_queueHooks:function(e,t){var n=t+"queueHooks";return Y.get(e,n)||Y.access(e,n,{empty:S.Callbacks("once memory").add(function(){Y.remove(e,[t+"queue",n])})})}}),S.fn.extend({queue:function(t,n){var e=2;return"string"!=typeof t&&(n=t,t="fx",e--),arguments.length<e?S.queue(this[0],t):void 0===n?this:this.each(function(){var e=S.queue(this,t,n);S._queueHooks(this,t),"fx"===t&&"inprogress"!==e[0]&&S.dequeue(this,t)})},dequeue:function(e){return this.each(function(){S.dequeue(this,e)})},clearQueue:function(e){return this.queue(e||"fx",[])},promise:function(e,t){var n,r=1,i=S.Deferred(),o=this,a=this.length,s=function(){--r||i.resolveWith(o,[o])};"string"!=typeof e&&(t=e,e=void 0),e=e||"fx";while(a--)(n=Y.get(o[a],e+"queueHooks"))&&n.empty&&(r++,n.empty.add(s));return s(),i.promise(t)}});var ee=/[+-]?(?:\d*\.|)\d+(?:[eE][+-]?\d+|)/.source,te=new RegExp("^(?:([+-])=|)("+ee+")([a-z%]*)$","i"),ne=["Top","Right","Bottom","Left"],re=E.documentElement,ie=function(e){return S.contains(e.ownerDocument,e)},oe={composed:!0};re.getRootNode&&(ie=function(e){return S.contains(e.ownerDocument,e)||e.getRootNode(oe)===e.ownerDocument});var ae=function(e,t){return"none"===(e=t||e).style.display||""===e.style.display&&ie(e)&&"none"===S.css(e,"display")};function se(e,t,n,r){var i,o,a=20,s=r?function(){return r.cur()}:function(){return S.css(e,t,"")},u=s(),l=n&&n[3]||(S.cssNumber[t]?"":"px"),c=e.nodeType&&(S.cssNumber[t]||"px"!==l&&+u)&&te.exec(S.css(e,t));if(c&&c[3]!==l){u/=2,l=l||c[3],c=+u||1;while(a--)S.style(e,t,c+l),(1-o)*(1-(o=s()/u||.5))<=0&&(a=0),c/=o;c*=2,S.style(e,t,c+l),n=n||[]}return n&&(c=+c||+u||0,i=n[1]?c+(n[1]+1)*n[2]:+n[2],r&&(r.unit=l,r.start=c,r.end=i)),i}var ue={};function le(e,t){for(var n,r,i,o,a,s,u,l=[],c=0,f=e.length;c<f;c++)(r=e[c]).style&&(n=r.style.display,t?("none"===n&&(l[c]=Y.get(r,"display")||null,l[c]||(r.style.display="")),""===r.style.display&&ae(r)&&(l[c]=(u=a=o=void 0,a=(i=r).ownerDocument,s=i.nodeName,(u=ue[s])||(o=a.body.appendChild(a.createElement(s)),u=S.css(o,"display"),o.parentNode.removeChild(o),"none"===u&&(u="block"),ue[s]=u)))):"none"!==n&&(l[c]="none",Y.set(r,"display",n)));for(c=0;c<f;c++)null!=l[c]&&(e[c].style.display=l[c]);return e}S.fn.extend({show:function(){return le(this,!0)},hide:function(){return le(this)},toggle:function(e){return"boolean"==typeof e?e?this.show():this.hide():this.each(function(){ae(this)?S(this).show():S(this).hide()})}});var ce,fe,pe=/^(?:checkbox|radio)$/i,de=/<([a-z][^\/\0>\x20\t\r\n\f]*)/i,he=/^$|^module$|\/(?:java|ecma)script/i;ce=E.createDocumentFragment().appendChild(E.createElement("div")),(fe=E.createElement("input")).setAttribute("type","radio"),fe.setAttribute("checked","checked"),fe.setAttribute("name","t"),ce.appendChild(fe),y.checkClone=ce.cloneNode(!0).cloneNode(!0).lastChild.checked,ce.innerHTML="<textarea>x</textarea>",y.noCloneChecked=!!ce.cloneNode(!0).lastChild.defaultValue,ce.innerHTML="<option></option>",y.option=!!ce.lastChild;var ge={thead:[1,"<table>","</table>"],col:[2,"<table><colgroup>","</colgroup></table>"],tr:[2,"<table><tbody>","</tbody></table>"],td:[3,"<table><tbody><tr>","</tr></tbody></table>"],_default:[0,"",""]};function ve(e,t){var n;return n="undefined"!=typeof e.getElementsByTagName?e.getElementsByTagName(t||"*"):"undefined"!=typeof e.querySelectorAll?e.querySelectorAll(t||"*"):[],void 0===t||t&&A(e,t)?S.merge([e],n):n}function ye(e,t){for(var n=0,r=e.length;n<r;n++)Y.set(e[n],"globalEval",!t||Y.get(t[n],"globalEval"))}ge.tbody=ge.tfoot=ge.colgroup=ge.caption=ge.thead,ge.th=ge.td,y.option||(ge.optgroup=ge.option=[1,"<select multiple='multiple'>","</select>"]);var me=/<|&#?\w+;/;function xe(e,t,n,r,i){for(var o,a,s,u,l,c,f=t.createDocumentFragment(),p=[],d=0,h=e.length;d<h;d++)if((o=e[d])||0===o)if("object"===w(o))S.merge(p,o.nodeType?[o]:o);else if(me.test(o)){a=a||f.appendChild(t.createElement("div")),s=(de.exec(o)||["",""])[1].toLowerCase(),u=ge[s]||ge._default,a.innerHTML=u[1]+S.htmlPrefilter(o)+u[2],c=u[0];while(c--)a=a.lastChild;S.merge(p,a.childNodes),(a=f.firstChild).textContent=""}else p.push(t.createTextNode(o));f.textContent="",d=0;while(o=p[d++])if(r&&-1<S.inArray(o,r))i&&i.push(o);else if(l=ie(o),a=ve(f.appendChild(o),"script"),l&&ye(a),n){c=0;while(o=a[c++])he.test(o.type||"")&&n.push(o)}return f}var be=/^key/,we=/^(?:mouse|pointer|contextmenu|drag|drop)|click/,Te=/^([^.]*)(?:\.(.+)|)/;function Ce(){return!0}function Ee(){return!1}function Se(e,t){return e===function(){try{return E.activeElement}catch(e){}}()==("focus"===t)}function ke(e,t,n,r,i,o){var a,s;if("object"==typeof t){for(s in"string"!=typeof n&&(r=r||n,n=void 0),t)ke(e,s,n,r,t[s],o);return e}if(null==r&&null==i?(i=n,r=n=void 0):null==i&&("string"==typeof n?(i=r,r=void 0):(i=r,r=n,n=void 0)),!1===i)i=Ee;else if(!i)return e;return 1===o&&(a=i,(i=function(e){return S().off(e),a.apply(this,arguments)}).guid=a.guid||(a.guid=S.guid++)),e.each(function(){S.event.add(this,t,i,r,n)})}function Ae(e,i,o){o?(Y.set(e,i,!1),S.event.add(e,i,{namespace:!1,handler:function(e){var t,n,r=Y.get(this,i);if(1&e.isTrigger&&this[i]){if(r.length)(S.event.special[i]||{}).delegateType&&e.stopPropagation();else if(r=s.call(arguments),Y.set(this,i,r),t=o(this,i),this[i](),r!==(n=Y.get(this,i))||t?Y.set(this,i,!1):n={},r!==n)return e.stopImmediatePropagation(),e.preventDefault(),n.value}else r.length&&(Y.set(this,i,{value:S.event.trigger(S.extend(r[0],S.Event.prototype),r.slice(1),this)}),e.stopImmediatePropagation())}})):void 0===Y.get(e,i)&&S.event.add(e,i,Ce)}S.event={global:{},add:function(t,e,n,r,i){var o,a,s,u,l,c,f,p,d,h,g,v=Y.get(t);if(V(t)){n.handler&&(n=(o=n).handler,i=o.selector),i&&S.find.matchesSelector(re,i),n.guid||(n.guid=S.guid++),(u=v.events)||(u=v.events=Object.create(null)),(a=v.handle)||(a=v.handle=function(e){return"undefined"!=typeof S&&S.event.triggered!==e.type?S.event.dispatch.apply(t,arguments):void 0}),l=(e=(e||"").match(P)||[""]).length;while(l--)d=g=(s=Te.exec(e[l])||[])[1],h=(s[2]||"").split(".").sort(),d&&(f=S.event.special[d]||{},d=(i?f.delegateType:f.bindType)||d,f=S.event.special[d]||{},c=S.extend({type:d,origType:g,data:r,handler:n,guid:n.guid,selector:i,needsContext:i&&S.expr.match.needsContext.test(i),namespace:h.join(".")},o),(p=u[d])||((p=u[d]=[]).delegateCount=0,f.setup&&!1!==f.setup.call(t,r,h,a)||t.addEventListener&&t.addEventListener(d,a)),f.add&&(f.add.call(t,c),c.handler.guid||(c.handler.guid=n.guid)),i?p.splice(p.delegateCount++,0,c):p.push(c),S.event.global[d]=!0)}},remove:function(e,t,n,r,i){var o,a,s,u,l,c,f,p,d,h,g,v=Y.hasData(e)&&Y.get(e);if(v&&(u=v.events)){l=(t=(t||"").match(P)||[""]).length;while(l--)if(d=g=(s=Te.exec(t[l])||[])[1],h=(s[2]||"").split(".").sort(),d){f=S.event.special[d]||{},p=u[d=(r?f.delegateType:f.bindType)||d]||[],s=s[2]&&new RegExp("(^|\\.)"+h.join("\\.(?:.*\\.|)")+"(\\.|$)"),a=o=p.length;while(o--)c=p[o],!i&&g!==c.origType||n&&n.guid!==c.guid||s&&!s.test(c.namespace)||r&&r!==c.selector&&("**"!==r||!c.selector)||(p.splice(o,1),c.selector&&p.delegateCount--,f.remove&&f.remove.call(e,c));a&&!p.length&&(f.teardown&&!1!==f.teardown.call(e,h,v.handle)||S.removeEvent(e,d,v.handle),delete u[d])}else for(d in u)S.event.remove(e,d+t[l],n,r,!0);S.isEmptyObject(u)&&Y.remove(e,"handle events")}},dispatch:function(e){var t,n,r,i,o,a,s=new Array(arguments.length),u=S.event.fix(e),l=(Y.get(this,"events")||Object.create(null))[u.type]||[],c=S.event.special[u.type]||{};for(s[0]=u,t=1;t<arguments.length;t++)s[t]=arguments[t];if(u.delegateTarget=this,!c.preDispatch||!1!==c.preDispatch.call(this,u)){a=S.event.handlers.call(this,u,l),t=0;while((i=a[t++])&&!u.isPropagationStopped()){u.currentTarget=i.elem,n=0;while((o=i.handlers[n++])&&!u.isImmediatePropagationStopped())u.rnamespace&&!1!==o.namespace&&!u.rnamespace.test(o.namespace)||(u.handleObj=o,u.data=o.data,void 0!==(r=((S.event.special[o.origType]||{}).handle||o.handler).apply(i.elem,s))&&!1===(u.result=r)&&(u.preventDefault(),u.stopPropagation()))}return c.postDispatch&&c.postDispatch.call(this,u),u.result}},handlers:function(e,t){var n,r,i,o,a,s=[],u=t.delegateCount,l=e.target;if(u&&l.nodeType&&!("click"===e.type&&1<=e.button))for(;l!==this;l=l.parentNode||this)if(1===l.nodeType&&("click"!==e.type||!0!==l.disabled)){for(o=[],a={},n=0;n<u;n++)void 0===a[i=(r=t[n]).selector+" "]&&(a[i]=r.needsContext?-1<S(i,this).index(l):S.find(i,this,null,[l]).length),a[i]&&o.push(r);o.length&&s.push({elem:l,handlers:o})}return l=this,u<t.length&&s.push({elem:l,handlers:t.slice(u)}),s},addProp:function(t,e){Object.defineProperty(S.Event.prototype,t,{enumerable:!0,configurable:!0,get:m(e)?function(){if(this.originalEvent)return e(this.originalEvent)}:function(){if(this.originalEvent)return this.originalEvent[t]},set:function(e){Object.defineProperty(this,t,{enumerable:!0,configurable:!0,writable:!0,value:e})}})},fix:function(e){return e[S.expando]?e:new S.Event(e)},special:{load:{noBubble:!0},click:{setup:function(e){var t=this||e;return pe.test(t.type)&&t.click&&A(t,"input")&&Ae(t,"click",Ce),!1},trigger:function(e){var t=this||e;return pe.test(t.type)&&t.click&&A(t,"input")&&Ae(t,"click"),!0},_default:function(e){var t=e.target;return pe.test(t.type)&&t.click&&A(t,"input")&&Y.get(t,"click")||A(t,"a")}},beforeunload:{postDispatch:function(e){void 0!==e.result&&e.originalEvent&&(e.originalEvent.returnValue=e.result)}}}},S.removeEvent=function(e,t,n){e.removeEventListener&&e.removeEventListener(t,n)},S.Event=function(e,t){if(!(this instanceof S.Event))return new S.Event(e,t);e&&e.type?(this.originalEvent=e,this.type=e.type,this.isDefaultPrevented=e.defaultPrevented||void 0===e.defaultPrevented&&!1===e.returnValue?Ce:Ee,this.target=e.target&&3===e.target.nodeType?e.target.parentNode:e.target,this.currentTarget=e.currentTarget,this.relatedTarget=e.relatedTarget):this.type=e,t&&S.extend(this,t),this.timeStamp=e&&e.timeStamp||Date.now(),this[S.expando]=!0},S.Event.prototype={constructor:S.Event,isDefaultPrevented:Ee,isPropagationStopped:Ee,isImmediatePropagationStopped:Ee,isSimulated:!1,preventDefault:function(){var e=this.originalEvent;this.isDefaultPrevented=Ce,e&&!this.isSimulated&&e.preventDefault()},stopPropagation:function(){var e=this.originalEvent;this.isPropagationStopped=Ce,e&&!this.isSimulated&&e.stopPropagation()},stopImmediatePropagation:function(){var e=this.originalEvent;this.isImmediatePropagationStopped=Ce,e&&!this.isSimulated&&e.stopImmediatePropagation(),this.stopPropagation()}},S.each({altKey:!0,bubbles:!0,cancelable:!0,changedTouches:!0,ctrlKey:!0,detail:!0,eventPhase:!0,metaKey:!0,pageX:!0,pageY:!0,shiftKey:!0,view:!0,"char":!0,code:!0,charCode:!0,key:!0,keyCode:!0,button:!0,buttons:!0,clientX:!0,clientY:!0,offsetX:!0,offsetY:!0,pointerId:!0,pointerType:!0,screenX:!0,screenY:!0,targetTouches:!0,toElement:!0,touches:!0,which:function(e){var t=e.button;return null==e.which&&be.test(e.type)?null!=e.charCode?e.charCode:e.keyCode:!e.which&&void 0!==t&&we.test(e.type)?1&t?1:2&t?3:4&t?2:0:e.which}},S.event.addProp),S.each({focus:"focusin",blur:"focusout"},function(e,t){S.event.special[e]={setup:function(){return Ae(this,e,Se),!1},trigger:function(){return Ae(this,e),!0},delegateType:t}}),S.each({mouseenter:"mouseover",mouseleave:"mouseout",pointerenter:"pointerover",pointerleave:"pointerout"},function(e,i){S.event.special[e]={delegateType:i,bindType:i,handle:function(e){var t,n=e.relatedTarget,r=e.handleObj;return n&&(n===this||S.contains(this,n))||(e.type=r.origType,t=r.handler.apply(this,arguments),e.type=i),t}}}),S.fn.extend({on:function(e,t,n,r){return ke(this,e,t,n,r)},one:function(e,t,n,r){return ke(this,e,t,n,r,1)},off:function(e,t,n){var r,i;if(e&&e.preventDefault&&e.handleObj)return r=e.handleObj,S(e.delegateTarget).off(r.namespace?r.origType+"."+r.namespace:r.origType,r.selector,r.handler),this;if("object"==typeof e){for(i in e)this.off(i,t,e[i]);return this}return!1!==t&&"function"!=typeof t||(n=t,t=void 0),!1===n&&(n=Ee),this.each(function(){S.event.remove(this,e,n,t)})}});var Ne=/<script|<style|<link/i,De=/checked\s*(?:[^=]|=\s*.checked.)/i,je=/^\s*<!(?:\[CDATA\[|--)|(?:\]\]|--)>\s*$/g;function qe(e,t){return A(e,"table")&&A(11!==t.nodeType?t:t.firstChild,"tr")&&S(e).children("tbody")[0]||e}function Le(e){return e.type=(null!==e.getAttribute("type"))+"/"+e.type,e}function He(e){return"true/"===(e.type||"").slice(0,5)?e.type=e.type.slice(5):e.removeAttribute("type"),e}function Oe(e,t){var n,r,i,o,a,s;if(1===t.nodeType){if(Y.hasData(e)&&(s=Y.get(e).events))for(i in Y.remove(t,"handle events"),s)for(n=0,r=s[i].length;n<r;n++)S.event.add(t,i,s[i][n]);Q.hasData(e)&&(o=Q.access(e),a=S.extend({},o),Q.set(t,a))}}function Pe(n,r,i,o){r=g(r);var e,t,a,s,u,l,c=0,f=n.length,p=f-1,d=r[0],h=m(d);if(h||1<f&&"string"==typeof d&&!y.checkClone&&De.test(d))return n.each(function(e){var t=n.eq(e);h&&(r[0]=d.call(this,e,t.html())),Pe(t,r,i,o)});if(f&&(t=(e=xe(r,n[0].ownerDocument,!1,n,o)).firstChild,1===e.childNodes.length&&(e=t),t||o)){for(s=(a=S.map(ve(e,"script"),Le)).length;c<f;c++)u=e,c!==p&&(u=S.clone(u,!0,!0),s&&S.merge(a,ve(u,"script"))),i.call(n[c],u,c);if(s)for(l=a[a.length-1].ownerDocument,S.map(a,He),c=0;c<s;c++)u=a[c],he.test(u.type||"")&&!Y.access(u,"globalEval")&&S.contains(l,u)&&(u.src&&"module"!==(u.type||"").toLowerCase()?S._evalUrl&&!u.noModule&&S._evalUrl(u.src,{nonce:u.nonce||u.getAttribute("nonce")},l):b(u.textContent.replace(je,""),u,l))}return n}function Re(e,t,n){for(var r,i=t?S.filter(t,e):e,o=0;null!=(r=i[o]);o++)n||1!==r.nodeType||S.cleanData(ve(r)),r.parentNode&&(n&&ie(r)&&ye(ve(r,"script")),r.parentNode.removeChild(r));return e}S.extend({htmlPrefilter:function(e){return e},clone:function(e,t,n){var r,i,o,a,s,u,l,c=e.cloneNode(!0),f=ie(e);if(!(y.noCloneChecked||1!==e.nodeType&&11!==e.nodeType||S.isXMLDoc(e)))for(a=ve(c),r=0,i=(o=ve(e)).length;r<i;r++)s=o[r],u=a[r],void 0,"input"===(l=u.nodeName.toLowerCase())&&pe.test(s.type)?u.checked=s.checked:"input"!==l&&"textarea"!==l||(u.defaultValue=s.defaultValue);if(t)if(n)for(o=o||ve(e),a=a||ve(c),r=0,i=o.length;r<i;r++)Oe(o[r],a[r]);else Oe(e,c);return 0<(a=ve(c,"script")).length&&ye(a,!f&&ve(e,"script")),c},cleanData:function(e){for(var t,n,r,i=S.event.special,o=0;void 0!==(n=e[o]);o++)if(V(n)){if(t=n[Y.expando]){if(t.events)for(r in t.events)i[r]?S.event.remove(n,r):S.removeEvent(n,r,t.handle);n[Y.expando]=void 0}n[Q.expando]&&(n[Q.expando]=void 0)}}}),S.fn.extend({detach:function(e){return Re(this,e,!0)},remove:function(e){return Re(this,e)},text:function(e){return $(this,function(e){return void 0===e?S.text(this):this.empty().each(function(){1!==this.nodeType&&11!==this.nodeType&&9!==this.nodeType||(this.textContent=e)})},null,e,arguments.length)},append:function(){return Pe(this,arguments,function(e){1!==this.nodeType&&11!==this.nodeType&&9!==this.nodeType||qe(this,e).appendChild(e)})},prepend:function(){return Pe(this,arguments,function(e){if(1===this.nodeType||11===this.nodeType||9===this.nodeType){var t=qe(this,e);t.insertBefore(e,t.firstChild)}})},before:function(){return Pe(this,arguments,function(e){this.parentNode&&this.parentNode.insertBefore(e,this)})},after:function(){return Pe(this,arguments,function(e){this.parentNode&&this.parentNode.insertBefore(e,this.nextSibling)})},empty:function(){for(var e,t=0;null!=(e=this[t]);t++)1===e.nodeType&&(S.cleanData(ve(e,!1)),e.textContent="");return this},clone:function(e,t){return e=null!=e&&e,t=null==t?e:t,this.map(function(){return S.clone(this,e,t)})},html:function(e){return $(this,function(e){var t=this[0]||{},n=0,r=this.length;if(void 0===e&&1===t.nodeType)return t.innerHTML;if("string"==typeof e&&!Ne.test(e)&&!ge[(de.exec(e)||["",""])[1].toLowerCase()]){e=S.htmlPrefilter(e);try{for(;n<r;n++)1===(t=this[n]||{}).nodeType&&(S.cleanData(ve(t,!1)),t.innerHTML=e);t=0}catch(e){}}t&&this.empty().append(e)},null,e,arguments.length)},replaceWith:function(){var n=[];return Pe(this,arguments,function(e){var t=this.parentNode;S.inArray(this,n)<0&&(S.cleanData(ve(this)),t&&t.replaceChild(e,this))},n)}}),S.each({appendTo:"append",prependTo:"prepend",insertBefore:"before",insertAfter:"after",replaceAll:"replaceWith"},function(e,a){S.fn[e]=function(e){for(var t,n=[],r=S(e),i=r.length-1,o=0;o<=i;o++)t=o===i?this:this.clone(!0),S(r[o])[a](t),u.apply(n,t.get());return this.pushStack(n)}});var Me=new RegExp("^("+ee+")(?!px)[a-z%]+$","i"),Ie=function(e){var t=e.ownerDocument.defaultView;return t&&t.opener||(t=C),t.getComputedStyle(e)},We=function(e,t,n){var r,i,o={};for(i in t)o[i]=e.style[i],e.style[i]=t[i];for(i in r=n.call(e),t)e.style[i]=o[i];return r},Fe=new RegExp(ne.join("|"),"i");function Be(e,t,n){var r,i,o,a,s=e.style;return(n=n||Ie(e))&&(""!==(a=n.getPropertyValue(t)||n[t])||ie(e)||(a=S.style(e,t)),!y.pixelBoxStyles()&&Me.test(a)&&Fe.test(t)&&(r=s.width,i=s.minWidth,o=s.maxWidth,s.minWidth=s.maxWidth=s.width=a,a=n.width,s.width=r,s.minWidth=i,s.maxWidth=o)),void 0!==a?a+"":a}function $e(e,t){return{get:function(){if(!e())return(this.get=t).apply(this,arguments);delete this.get}}}!function(){function e(){if(l){u.style.cssText="position:absolute;left:-11111px;width:60px;margin-top:1px;padding:0;border:0",l.style.cssText="position:relative;display:block;box-sizing:border-box;overflow:scroll;margin:auto;border:1px;padding:1px;width:60%;top:1%",re.appendChild(u).appendChild(l);var e=C.getComputedStyle(l);n="1%"!==e.top,s=12===t(e.marginLeft),l.style.right="60%",o=36===t(e.right),r=36===t(e.width),l.style.position="absolute",i=12===t(l.offsetWidth/3),re.removeChild(u),l=null}}function t(e){return Math.round(parseFloat(e))}var n,r,i,o,a,s,u=E.createElement("div"),l=E.createElement("div");l.style&&(l.style.backgroundClip="content-box",l.cloneNode(!0).style.backgroundClip="",y.clearCloneStyle="content-box"===l.style.backgroundClip,S.extend(y,{boxSizingReliable:function(){return e(),r},pixelBoxStyles:function(){return e(),o},pixelPosition:function(){return e(),n},reliableMarginLeft:function(){return e(),s},scrollboxSize:function(){return e(),i},reliableTrDimensions:function(){var e,t,n,r;return null==a&&(e=E.createElement("table"),t=E.createElement("tr"),n=E.createElement("div"),e.style.cssText="position:absolute;left:-11111px",t.style.height="1px",n.style.height="9px",re.appendChild(e).appendChild(t).appendChild(n),r=C.getComputedStyle(t),a=3<parseInt(r.height),re.removeChild(e)),a}}))}();var _e=["Webkit","Moz","ms"],ze=E.createElement("div").style,Ue={};function Xe(e){var t=S.cssProps[e]||Ue[e];return t||(e in ze?e:Ue[e]=function(e){var t=e[0].toUpperCase()+e.slice(1),n=_e.length;while(n--)if((e=_e[n]+t)in ze)return e}(e)||e)}var Ve=/^(none|table(?!-c[ea]).+)/,Ge=/^--/,Ye={position:"absolute",visibility:"hidden",display:"block"},Qe={letterSpacing:"0",fontWeight:"400"};function Je(e,t,n){var r=te.exec(t);return r?Math.max(0,r[2]-(n||0))+(r[3]||"px"):t}function Ke(e,t,n,r,i,o){var a="width"===t?1:0,s=0,u=0;if(n===(r?"border":"content"))return 0;for(;a<4;a+=2)"margin"===n&&(u+=S.css(e,n+ne[a],!0,i)),r?("content"===n&&(u-=S.css(e,"padding"+ne[a],!0,i)),"margin"!==n&&(u-=S.css(e,"border"+ne[a]+"Width",!0,i))):(u+=S.css(e,"padding"+ne[a],!0,i),"padding"!==n?u+=S.css(e,"border"+ne[a]+"Width",!0,i):s+=S.css(e,"border"+ne[a]+"Width",!0,i));return!r&&0<=o&&(u+=Math.max(0,Math.ceil(e["offset"+t[0].toUpperCase()+t.slice(1)]-o-u-s-.5))||0),u}function Ze(e,t,n){var r=Ie(e),i=(!y.boxSizingReliable()||n)&&"border-box"===S.css(e,"boxSizing",!1,r),o=i,a=Be(e,t,r),s="offset"+t[0].toUpperCase()+t.slice(1);if(Me.test(a)){if(!n)return a;a="auto"}return(!y.boxSizingReliable()&&i||!y.reliableTrDimensions()&&A(e,"tr")||"auto"===a||!parseFloat(a)&&"inline"===S.css(e,"display",!1,r))&&e.getClientRects().length&&(i="border-box"===S.css(e,"boxSizing",!1,r),(o=s in e)&&(a=e[s])),(a=parseFloat(a)||0)+Ke(e,t,n||(i?"border":"content"),o,r,a)+"px"}function et(e,t,n,r,i){return new et.prototype.init(e,t,n,r,i)}S.extend({cssHooks:{opacity:{get:function(e,t){if(t){var n=Be(e,"opacity");return""===n?"1":n}}}},cssNumber:{animationIterationCount:!0,columnCount:!0,fillOpacity:!0,flexGrow:!0,flexShrink:!0,fontWeight:!0,gridArea:!0,gridColumn:!0,gridColumnEnd:!0,gridColumnStart:!0,gridRow:!0,gridRowEnd:!0,gridRowStart:!0,lineHeight:!0,opacity:!0,order:!0,orphans:!0,widows:!0,zIndex:!0,zoom:!0},cssProps:{},style:function(e,t,n,r){if(e&&3!==e.nodeType&&8!==e.nodeType&&e.style){var i,o,a,s=X(t),u=Ge.test(t),l=e.style;if(u||(t=Xe(s)),a=S.cssHooks[t]||S.cssHooks[s],void 0===n)return a&&"get"in a&&void 0!==(i=a.get(e,!1,r))?i:l[t];"string"===(o=typeof n)&&(i=te.exec(n))&&i[1]&&(n=se(e,t,i),o="number"),null!=n&&n==n&&("number"!==o||u||(n+=i&&i[3]||(S.cssNumber[s]?"":"px")),y.clearCloneStyle||""!==n||0!==t.indexOf("background")||(l[t]="inherit"),a&&"set"in a&&void 0===(n=a.set(e,n,r))||(u?l.setProperty(t,n):l[t]=n))}},css:function(e,t,n,r){var i,o,a,s=X(t);return Ge.test(t)||(t=Xe(s)),(a=S.cssHooks[t]||S.cssHooks[s])&&"get"in a&&(i=a.get(e,!0,n)),void 0===i&&(i=Be(e,t,r)),"normal"===i&&t in Qe&&(i=Qe[t]),""===n||n?(o=parseFloat(i),!0===n||isFinite(o)?o||0:i):i}}),S.each(["height","width"],function(e,u){S.cssHooks[u]={get:function(e,t,n){if(t)return!Ve.test(S.css(e,"display"))||e.getClientRects().length&&e.getBoundingClientRect().width?Ze(e,u,n):We(e,Ye,function(){return Ze(e,u,n)})},set:function(e,t,n){var r,i=Ie(e),o=!y.scrollboxSize()&&"absolute"===i.position,a=(o||n)&&"border-box"===S.css(e,"boxSizing",!1,i),s=n?Ke(e,u,n,a,i):0;return a&&o&&(s-=Math.ceil(e["offset"+u[0].toUpperCase()+u.slice(1)]-parseFloat(i[u])-Ke(e,u,"border",!1,i)-.5)),s&&(r=te.exec(t))&&"px"!==(r[3]||"px")&&(e.style[u]=t,t=S.css(e,u)),Je(0,t,s)}}}),S.cssHooks.marginLeft=$e(y.reliableMarginLeft,function(e,t){if(t)return(parseFloat(Be(e,"marginLeft"))||e.getBoundingClientRect().left-We(e,{marginLeft:0},function(){return e.getBoundingClientRect().left}))+"px"}),S.each({margin:"",padding:"",border:"Width"},function(i,o){S.cssHooks[i+o]={expand:function(e){for(var t=0,n={},r="string"==typeof e?e.split(" "):[e];t<4;t++)n[i+ne[t]+o]=r[t]||r[t-2]||r[0];return n}},"margin"!==i&&(S.cssHooks[i+o].set=Je)}),S.fn.extend({css:function(e,t){return $(this,function(e,t,n){var r,i,o={},a=0;if(Array.isArray(t)){for(r=Ie(e),i=t.length;a<i;a++)o[t[a]]=S.css(e,t[a],!1,r);return o}return void 0!==n?S.style(e,t,n):S.css(e,t)},e,t,1<arguments.length)}}),((S.Tween=et).prototype={constructor:et,init:function(e,t,n,r,i,o){this.elem=e,this.prop=n,this.easing=i||S.easing._default,this.options=t,this.start=this.now=this.cur(),this.end=r,this.unit=o||(S.cssNumber[n]?"":"px")},cur:function(){var e=et.propHooks[this.prop];return e&&e.get?e.get(this):et.propHooks._default.get(this)},run:function(e){var t,n=et.propHooks[this.prop];return this.options.duration?this.pos=t=S.easing[this.easing](e,this.options.duration*e,0,1,this.options.duration):this.pos=t=e,this.now=(this.end-this.start)*t+this.start,this.options.step&&this.options.step.call(this.elem,this.now,this),n&&n.set?n.set(this):et.propHooks._default.set(this),this}}).init.prototype=et.prototype,(et.propHooks={_default:{get:function(e){var t;return 1!==e.elem.nodeType||null!=e.elem[e.prop]&&null==e.elem.style[e.prop]?e.elem[e.prop]:(t=S.css(e.elem,e.prop,""))&&"auto"!==t?t:0},set:function(e){S.fx.step[e.prop]?S.fx.step[e.prop](e):1!==e.elem.nodeType||!S.cssHooks[e.prop]&&null==e.elem.style[Xe(e.prop)]?e.elem[e.prop]=e.now:S.style(e.elem,e.prop,e.now+e.unit)}}}).scrollTop=et.propHooks.scrollLeft={set:function(e){e.elem.nodeType&&e.elem.parentNode&&(e.elem[e.prop]=e.now)}},S.easing={linear:function(e){return e},swing:function(e){return.5-Math.cos(e*Math.PI)/2},_default:"swing"},S.fx=et.prototype.init,S.fx.step={};var tt,nt,rt,it,ot=/^(?:toggle|show|hide)$/,at=/queueHooks$/;function st(){nt&&(!1===E.hidden&&C.requestAnimationFrame?C.requestAnimationFrame(st):C.setTimeout(st,S.fx.interval),S.fx.tick())}function ut(){return C.setTimeout(function(){tt=void 0}),tt=Date.now()}function lt(e,t){var n,r=0,i={height:e};for(t=t?1:0;r<4;r+=2-t)i["margin"+(n=ne[r])]=i["padding"+n]=e;return t&&(i.opacity=i.width=e),i}function ct(e,t,n){for(var r,i=(ft.tweeners[t]||[]).concat(ft.tweeners["*"]),o=0,a=i.length;o<a;o++)if(r=i[o].call(n,t,e))return r}function ft(o,e,t){var n,a,r=0,i=ft.prefilters.length,s=S.Deferred().always(function(){delete u.elem}),u=function(){if(a)return!1;for(var e=tt||ut(),t=Math.max(0,l.startTime+l.duration-e),n=1-(t/l.duration||0),r=0,i=l.tweens.length;r<i;r++)l.tweens[r].run(n);return s.notifyWith(o,[l,n,t]),n<1&&i?t:(i||s.notifyWith(o,[l,1,0]),s.resolveWith(o,[l]),!1)},l=s.promise({elem:o,props:S.extend({},e),opts:S.extend(!0,{specialEasing:{},easing:S.easing._default},t),originalProperties:e,originalOptions:t,startTime:tt||ut(),duration:t.duration,tweens:[],createTween:function(e,t){var n=S.Tween(o,l.opts,e,t,l.opts.specialEasing[e]||l.opts.easing);return l.tweens.push(n),n},stop:function(e){var t=0,n=e?l.tweens.length:0;if(a)return this;for(a=!0;t<n;t++)l.tweens[t].run(1);return e?(s.notifyWith(o,[l,1,0]),s.resolveWith(o,[l,e])):s.rejectWith(o,[l,e]),this}}),c=l.props;for(!function(e,t){var n,r,i,o,a;for(n in e)if(i=t[r=X(n)],o=e[n],Array.isArray(o)&&(i=o[1],o=e[n]=o[0]),n!==r&&(e[r]=o,delete e[n]),(a=S.cssHooks[r])&&"expand"in a)for(n in o=a.expand(o),delete e[r],o)n in e||(e[n]=o[n],t[n]=i);else t[r]=i}(c,l.opts.specialEasing);r<i;r++)if(n=ft.prefilters[r].call(l,o,c,l.opts))return m(n.stop)&&(S._queueHooks(l.elem,l.opts.queue).stop=n.stop.bind(n)),n;return S.map(c,ct,l),m(l.opts.start)&&l.opts.start.call(o,l),l.progress(l.opts.progress).done(l.opts.done,l.opts.complete).fail(l.opts.fail).always(l.opts.always),S.fx.timer(S.extend(u,{elem:o,anim:l,queue:l.opts.queue})),l}S.Animation=S.extend(ft,{tweeners:{"*":[function(e,t){var n=this.createTween(e,t);return se(n.elem,e,te.exec(t),n),n}]},tweener:function(e,t){m(e)?(t=e,e=["*"]):e=e.match(P);for(var n,r=0,i=e.length;r<i;r++)n=e[r],ft.tweeners[n]=ft.tweeners[n]||[],ft.tweeners[n].unshift(t)},prefilters:[function(e,t,n){var r,i,o,a,s,u,l,c,f="width"in t||"height"in t,p=this,d={},h=e.style,g=e.nodeType&&ae(e),v=Y.get(e,"fxshow");for(r in n.queue||(null==(a=S._queueHooks(e,"fx")).unqueued&&(a.unqueued=0,s=a.empty.fire,a.empty.fire=function(){a.unqueued||s()}),a.unqueued++,p.always(function(){p.always(function(){a.unqueued--,S.queue(e,"fx").length||a.empty.fire()})})),t)if(i=t[r],ot.test(i)){if(delete t[r],o=o||"toggle"===i,i===(g?"hide":"show")){if("show"!==i||!v||void 0===v[r])continue;g=!0}d[r]=v&&v[r]||S.style(e,r)}if((u=!S.isEmptyObject(t))||!S.isEmptyObject(d))for(r in f&&1===e.nodeType&&(n.overflow=[h.overflow,h.overflowX,h.overflowY],null==(l=v&&v.display)&&(l=Y.get(e,"display")),"none"===(c=S.css(e,"display"))&&(l?c=l:(le([e],!0),l=e.style.display||l,c=S.css(e,"display"),le([e]))),("inline"===c||"inline-block"===c&&null!=l)&&"none"===S.css(e,"float")&&(u||(p.done(function(){h.display=l}),null==l&&(c=h.display,l="none"===c?"":c)),h.display="inline-block")),n.overflow&&(h.overflow="hidden",p.always(function(){h.overflow=n.overflow[0],h.overflowX=n.overflow[1],h.overflowY=n.overflow[2]})),u=!1,d)u||(v?"hidden"in v&&(g=v.hidden):v=Y.access(e,"fxshow",{display:l}),o&&(v.hidden=!g),g&&le([e],!0),p.done(function(){for(r in g||le([e]),Y.remove(e,"fxshow"),d)S.style(e,r,d[r])})),u=ct(g?v[r]:0,r,p),r in v||(v[r]=u.start,g&&(u.end=u.start,u.start=0))}],prefilter:function(e,t){t?ft.prefilters.unshift(e):ft.prefilters.push(e)}}),S.speed=function(e,t,n){var r=e&&"object"==typeof e?S.extend({},e):{complete:n||!n&&t||m(e)&&e,duration:e,easing:n&&t||t&&!m(t)&&t};return S.fx.off?r.duration=0:"number"!=typeof r.duration&&(r.duration in S.fx.speeds?r.duration=S.fx.speeds[r.duration]:r.duration=S.fx.speeds._default),null!=r.queue&&!0!==r.queue||(r.queue="fx"),r.old=r.complete,r.complete=function(){m(r.old)&&r.old.call(this),r.queue&&S.dequeue(this,r.queue)},r},S.fn.extend({fadeTo:function(e,t,n,r){return this.filter(ae).css("opacity",0).show().end().animate({opacity:t},e,n,r)},animate:function(t,e,n,r){var i=S.isEmptyObject(t),o=S.speed(e,n,r),a=function(){var e=ft(this,S.extend({},t),o);(i||Y.get(this,"finish"))&&e.stop(!0)};return a.finish=a,i||!1===o.queue?this.each(a):this.queue(o.queue,a)},stop:function(i,e,o){var a=function(e){var t=e.stop;delete e.stop,t(o)};return"string"!=typeof i&&(o=e,e=i,i=void 0),e&&this.queue(i||"fx",[]),this.each(function(){var e=!0,t=null!=i&&i+"queueHooks",n=S.timers,r=Y.get(this);if(t)r[t]&&r[t].stop&&a(r[t]);else for(t in r)r[t]&&r[t].stop&&at.test(t)&&a(r[t]);for(t=n.length;t--;)n[t].elem!==this||null!=i&&n[t].queue!==i||(n[t].anim.stop(o),e=!1,n.splice(t,1));!e&&o||S.dequeue(this,i)})},finish:function(a){return!1!==a&&(a=a||"fx"),this.each(function(){var e,t=Y.get(this),n=t[a+"queue"],r=t[a+"queueHooks"],i=S.timers,o=n?n.length:0;for(t.finish=!0,S.queue(this,a,[]),r&&r.stop&&r.stop.call(this,!0),e=i.length;e--;)i[e].elem===this&&i[e].queue===a&&(i[e].anim.stop(!0),i.splice(e,1));for(e=0;e<o;e++)n[e]&&n[e].finish&&n[e].finish.call(this);delete t.finish})}}),S.each(["toggle","show","hide"],function(e,r){var i=S.fn[r];S.fn[r]=function(e,t,n){return null==e||"boolean"==typeof e?i.apply(this,arguments):this.animate(lt(r,!0),e,t,n)}}),S.each({slideDown:lt("show"),slideUp:lt("hide"),slideToggle:lt("toggle"),fadeIn:{opacity:"show"},fadeOut:{opacity:"hide"},fadeToggle:{opacity:"toggle"}},function(e,r){S.fn[e]=function(e,t,n){return this.animate(r,e,t,n)}}),S.timers=[],S.fx.tick=function(){var e,t=0,n=S.timers;for(tt=Date.now();t<n.length;t++)(e=n[t])()||n[t]!==e||n.splice(t--,1);n.length||S.fx.stop(),tt=void 0},S.fx.timer=function(e){S.timers.push(e),S.fx.start()},S.fx.interval=13,S.fx.start=function(){nt||(nt=!0,st())},S.fx.stop=function(){nt=null},S.fx.speeds={slow:600,fast:200,_default:400},S.fn.delay=function(r,e){return r=S.fx&&S.fx.speeds[r]||r,e=e||"fx",this.queue(e,function(e,t){var n=C.setTimeout(e,r);t.stop=function(){C.clearTimeout(n)}})},rt=E.createElement("input"),it=E.createElement("select").appendChild(E.createElement("option")),rt.type="checkbox",y.checkOn=""!==rt.value,y.optSelected=it.selected,(rt=E.createElement("input")).value="t",rt.type="radio",y.radioValue="t"===rt.value;var pt,dt=S.expr.attrHandle;S.fn.extend({attr:function(e,t){return $(this,S.attr,e,t,1<arguments.length)},removeAttr:function(e){return this.each(function(){S.removeAttr(this,e)})}}),S.extend({attr:function(e,t,n){var r,i,o=e.nodeType;if(3!==o&&8!==o&&2!==o)return"undefined"==typeof e.getAttribute?S.prop(e,t,n):(1===o&&S.isXMLDoc(e)||(i=S.attrHooks[t.toLowerCase()]||(S.expr.match.bool.test(t)?pt:void 0)),void 0!==n?null===n?void S.removeAttr(e,t):i&&"set"in i&&void 0!==(r=i.set(e,n,t))?r:(e.setAttribute(t,n+""),n):i&&"get"in i&&null!==(r=i.get(e,t))?r:null==(r=S.find.attr(e,t))?void 0:r)},attrHooks:{type:{set:function(e,t){if(!y.radioValue&&"radio"===t&&A(e,"input")){var n=e.value;return e.setAttribute("type",t),n&&(e.value=n),t}}}},removeAttr:function(e,t){var n,r=0,i=t&&t.match(P);if(i&&1===e.nodeType)while(n=i[r++])e.removeAttribute(n)}}),pt={set:function(e,t,n){return!1===t?S.removeAttr(e,n):e.setAttribute(n,n),n}},S.each(S.expr.match.bool.source.match(/\w+/g),function(e,t){var a=dt[t]||S.find.attr;dt[t]=function(e,t,n){var r,i,o=t.toLowerCase();return n||(i=dt[o],dt[o]=r,r=null!=a(e,t,n)?o:null,dt[o]=i),r}});var ht=/^(?:input|select|textarea|button)$/i,gt=/^(?:a|area)$/i;function vt(e){return(e.match(P)||[]).join(" ")}function yt(e){return e.getAttribute&&e.getAttribute("class")||""}function mt(e){return Array.isArray(e)?e:"string"==typeof e&&e.match(P)||[]}S.fn.extend({prop:function(e,t){return $(this,S.prop,e,t,1<arguments.length)},removeProp:function(e){return this.each(function(){delete this[S.propFix[e]||e]})}}),S.extend({prop:function(e,t,n){var r,i,o=e.nodeType;if(3!==o&&8!==o&&2!==o)return 1===o&&S.isXMLDoc(e)||(t=S.propFix[t]||t,i=S.propHooks[t]),void 0!==n?i&&"set"in i&&void 0!==(r=i.set(e,n,t))?r:e[t]=n:i&&"get"in i&&null!==(r=i.get(e,t))?r:e[t]},propHooks:{tabIndex:{get:function(e){var t=S.find.attr(e,"tabindex");return t?parseInt(t,10):ht.test(e.nodeName)||gt.test(e.nodeName)&&e.href?0:-1}}},propFix:{"for":"htmlFor","class":"className"}}),y.optSelected||(S.propHooks.selected={get:function(e){var t=e.parentNode;return t&&t.parentNode&&t.parentNode.selectedIndex,null},set:function(e){var t=e.parentNode;t&&(t.selectedIndex,t.parentNode&&t.parentNode.selectedIndex)}}),S.each(["tabIndex","readOnly","maxLength","cellSpacing","cellPadding","rowSpan","colSpan","useMap","frameBorder","contentEditable"],function(){S.propFix[this.toLowerCase()]=this}),S.fn.extend({addClass:function(t){var e,n,r,i,o,a,s,u=0;if(m(t))return this.each(function(e){S(this).addClass(t.call(this,e,yt(this)))});if((e=mt(t)).length)while(n=this[u++])if(i=yt(n),r=1===n.nodeType&&" "+vt(i)+" "){a=0;while(o=e[a++])r.indexOf(" "+o+" ")<0&&(r+=o+" ");i!==(s=vt(r))&&n.setAttribute("class",s)}return this},removeClass:function(t){var e,n,r,i,o,a,s,u=0;if(m(t))return this.each(function(e){S(this).removeClass(t.call(this,e,yt(this)))});if(!arguments.length)return this.attr("class","");if((e=mt(t)).length)while(n=this[u++])if(i=yt(n),r=1===n.nodeType&&" "+vt(i)+" "){a=0;while(o=e[a++])while(-1<r.indexOf(" "+o+" "))r=r.replace(" "+o+" "," ");i!==(s=vt(r))&&n.setAttribute("class",s)}return this},toggleClass:function(i,t){var o=typeof i,a="string"===o||Array.isArray(i);return"boolean"==typeof t&&a?t?this.addClass(i):this.removeClass(i):m(i)?this.each(function(e){S(this).toggleClass(i.call(this,e,yt(this),t),t)}):this.each(function(){var e,t,n,r;if(a){t=0,n=S(this),r=mt(i);while(e=r[t++])n.hasClass(e)?n.removeClass(e):n.addClass(e)}else void 0!==i&&"boolean"!==o||((e=yt(this))&&Y.set(this,"__className__",e),this.setAttribute&&this.setAttribute("class",e||!1===i?"":Y.get(this,"__className__")||""))})},hasClass:function(e){var t,n,r=0;t=" "+e+" ";while(n=this[r++])if(1===n.nodeType&&-1<(" "+vt(yt(n))+" ").indexOf(t))return!0;return!1}});var xt=/\r/g;S.fn.extend({val:function(n){var r,e,i,t=this[0];return arguments.length?(i=m(n),this.each(function(e){var t;1===this.nodeType&&(null==(t=i?n.call(this,e,S(this).val()):n)?t="":"number"==typeof t?t+="":Array.isArray(t)&&(t=S.map(t,function(e){return null==e?"":e+""})),(r=S.valHooks[this.type]||S.valHooks[this.nodeName.toLowerCase()])&&"set"in r&&void 0!==r.set(this,t,"value")||(this.value=t))})):t?(r=S.valHooks[t.type]||S.valHooks[t.nodeName.toLowerCase()])&&"get"in r&&void 0!==(e=r.get(t,"value"))?e:"string"==typeof(e=t.value)?e.replace(xt,""):null==e?"":e:void 0}}),S.extend({valHooks:{option:{get:function(e){var t=S.find.attr(e,"value");return null!=t?t:vt(S.text(e))}},select:{get:function(e){var t,n,r,i=e.options,o=e.selectedIndex,a="select-one"===e.type,s=a?null:[],u=a?o+1:i.length;for(r=o<0?u:a?o:0;r<u;r++)if(((n=i[r]).selected||r===o)&&!n.disabled&&(!n.parentNode.disabled||!A(n.parentNode,"optgroup"))){if(t=S(n).val(),a)return t;s.push(t)}return s},set:function(e,t){var n,r,i=e.options,o=S.makeArray(t),a=i.length;while(a--)((r=i[a]).selected=-1<S.inArray(S.valHooks.option.get(r),o))&&(n=!0);return n||(e.selectedIndex=-1),o}}}}),S.each(["radio","checkbox"],function(){S.valHooks[this]={set:function(e,t){if(Array.isArray(t))return e.checked=-1<S.inArray(S(e).val(),t)}},y.checkOn||(S.valHooks[this].get=function(e){return null===e.getAttribute("value")?"on":e.value})}),y.focusin="onfocusin"in C;var bt=/^(?:focusinfocus|focusoutblur)$/,wt=function(e){e.stopPropagation()};S.extend(S.event,{trigger:function(e,t,n,r){var i,o,a,s,u,l,c,f,p=[n||E],d=v.call(e,"type")?e.type:e,h=v.call(e,"namespace")?e.namespace.split("."):[];if(o=f=a=n=n||E,3!==n.nodeType&&8!==n.nodeType&&!bt.test(d+S.event.triggered)&&(-1<d.indexOf(".")&&(d=(h=d.split(".")).shift(),h.sort()),u=d.indexOf(":")<0&&"on"+d,(e=e[S.expando]?e:new S.Event(d,"object"==typeof e&&e)).isTrigger=r?2:3,e.namespace=h.join("."),e.rnamespace=e.namespace?new RegExp("(^|\\.)"+h.join("\\.(?:.*\\.|)")+"(\\.|$)"):null,e.result=void 0,e.target||(e.target=n),t=null==t?[e]:S.makeArray(t,[e]),c=S.event.special[d]||{},r||!c.trigger||!1!==c.trigger.apply(n,t))){if(!r&&!c.noBubble&&!x(n)){for(s=c.delegateType||d,bt.test(s+d)||(o=o.parentNode);o;o=o.parentNode)p.push(o),a=o;a===(n.ownerDocument||E)&&p.push(a.defaultView||a.parentWindow||C)}i=0;while((o=p[i++])&&!e.isPropagationStopped())f=o,e.type=1<i?s:c.bindType||d,(l=(Y.get(o,"events")||Object.create(null))[e.type]&&Y.get(o,"handle"))&&l.apply(o,t),(l=u&&o[u])&&l.apply&&V(o)&&(e.result=l.apply(o,t),!1===e.result&&e.preventDefault());return e.type=d,r||e.isDefaultPrevented()||c._default&&!1!==c._default.apply(p.pop(),t)||!V(n)||u&&m(n[d])&&!x(n)&&((a=n[u])&&(n[u]=null),S.event.triggered=d,e.isPropagationStopped()&&f.addEventListener(d,wt),n[d](),e.isPropagationStopped()&&f.removeEventListener(d,wt),S.event.triggered=void 0,a&&(n[u]=a)),e.result}},simulate:function(e,t,n){var r=S.extend(new S.Event,n,{type:e,isSimulated:!0});S.event.trigger(r,null,t)}}),S.fn.extend({trigger:function(e,t){return this.each(function(){S.event.trigger(e,t,this)})},triggerHandler:function(e,t){var n=this[0];if(n)return S.event.trigger(e,t,n,!0)}}),y.focusin||S.each({focus:"focusin",blur:"focusout"},function(n,r){var i=function(e){S.event.simulate(r,e.target,S.event.fix(e))};S.event.special[r]={setup:function(){var e=this.ownerDocument||this.document||this,t=Y.access(e,r);t||e.addEventListener(n,i,!0),Y.access(e,r,(t||0)+1)},teardown:function(){var e=this.ownerDocument||this.document||this,t=Y.access(e,r)-1;t?Y.access(e,r,t):(e.removeEventListener(n,i,!0),Y.remove(e,r))}}});var Tt=C.location,Ct={guid:Date.now()},Et=/\?/;S.parseXML=function(e){var t;if(!e||"string"!=typeof e)return null;try{t=(new C.DOMParser).parseFromString(e,"text/xml")}catch(e){t=void 0}return t&&!t.getElementsByTagName("parsererror").length||S.error("Invalid XML: "+e),t};var St=/\[\]$/,kt=/\r?\n/g,At=/^(?:submit|button|image|reset|file)$/i,Nt=/^(?:input|select|textarea|keygen)/i;function Dt(n,e,r,i){var t;if(Array.isArray(e))S.each(e,function(e,t){r||St.test(n)?i(n,t):Dt(n+"["+("object"==typeof t&&null!=t?e:"")+"]",t,r,i)});else if(r||"object"!==w(e))i(n,e);else for(t in e)Dt(n+"["+t+"]",e[t],r,i)}S.param=function(e,t){var n,r=[],i=function(e,t){var n=m(t)?t():t;r[r.length]=encodeURIComponent(e)+"="+encodeURIComponent(null==n?"":n)};if(null==e)return"";if(Array.isArray(e)||e.jquery&&!S.isPlainObject(e))S.each(e,function(){i(this.name,this.value)});else for(n in e)Dt(n,e[n],t,i);return r.join("&")},S.fn.extend({serialize:function(){return S.param(this.serializeArray())},serializeArray:function(){return this.map(function(){var e=S.prop(this,"elements");return e?S.makeArray(e):this}).filter(function(){var e=this.type;return this.name&&!S(this).is(":disabled")&&Nt.test(this.nodeName)&&!At.test(e)&&(this.checked||!pe.test(e))}).map(function(e,t){var n=S(this).val();return null==n?null:Array.isArray(n)?S.map(n,function(e){return{name:t.name,value:e.replace(kt,"\r\n")}}):{name:t.name,value:n.replace(kt,"\r\n")}}).get()}});var jt=/%20/g,qt=/#.*$/,Lt=/([?&])_=[^&]*/,Ht=/^(.*?):[ \t]*([^\r\n]*)$/gm,Ot=/^(?:GET|HEAD)$/,Pt=/^\/\//,Rt={},Mt={},It="*/".concat("*"),Wt=E.createElement("a");function Ft(o){return function(e,t){"string"!=typeof e&&(t=e,e="*");var n,r=0,i=e.toLowerCase().match(P)||[];if(m(t))while(n=i[r++])"+"===n[0]?(n=n.slice(1)||"*",(o[n]=o[n]||[]).unshift(t)):(o[n]=o[n]||[]).push(t)}}function Bt(t,i,o,a){var s={},u=t===Mt;function l(e){var r;return s[e]=!0,S.each(t[e]||[],function(e,t){var n=t(i,o,a);return"string"!=typeof n||u||s[n]?u?!(r=n):void 0:(i.dataTypes.unshift(n),l(n),!1)}),r}return l(i.dataTypes[0])||!s["*"]&&l("*")}function $t(e,t){var n,r,i=S.ajaxSettings.flatOptions||{};for(n in t)void 0!==t[n]&&((i[n]?e:r||(r={}))[n]=t[n]);return r&&S.extend(!0,e,r),e}Wt.href=Tt.href,S.extend({active:0,lastModified:{},etag:{},ajaxSettings:{url:Tt.href,type:"GET",isLocal:/^(?:about|app|app-storage|.+-extension|file|res|widget):$/.test(Tt.protocol),global:!0,processData:!0,async:!0,contentType:"application/x-www-form-urlencoded; charset=UTF-8",accepts:{"*":It,text:"text/plain",html:"text/html",xml:"application/xml, text/xml",json:"application/json, text/javascript"},contents:{xml:/\bxml\b/,html:/\bhtml/,json:/\bjson\b/},responseFields:{xml:"responseXML",text:"responseText",json:"responseJSON"},converters:{"* text":String,"text html":!0,"text json":JSON.parse,"text xml":S.parseXML},flatOptions:{url:!0,context:!0}},ajaxSetup:function(e,t){return t?$t($t(e,S.ajaxSettings),t):$t(S.ajaxSettings,e)},ajaxPrefilter:Ft(Rt),ajaxTransport:Ft(Mt),ajax:function(e,t){"object"==typeof e&&(t=e,e=void 0),t=t||{};var c,f,p,n,d,r,h,g,i,o,v=S.ajaxSetup({},t),y=v.context||v,m=v.context&&(y.nodeType||y.jquery)?S(y):S.event,x=S.Deferred(),b=S.Callbacks("once memory"),w=v.statusCode||{},a={},s={},u="canceled",T={readyState:0,getResponseHeader:function(e){var t;if(h){if(!n){n={};while(t=Ht.exec(p))n[t[1].toLowerCase()+" "]=(n[t[1].toLowerCase()+" "]||[]).concat(t[2])}t=n[e.toLowerCase()+" "]}return null==t?null:t.join(", ")},getAllResponseHeaders:function(){return h?p:null},setRequestHeader:function(e,t){return null==h&&(e=s[e.toLowerCase()]=s[e.toLowerCase()]||e,a[e]=t),this},overrideMimeType:function(e){return null==h&&(v.mimeType=e),this},statusCode:function(e){var t;if(e)if(h)T.always(e[T.status]);else for(t in e)w[t]=[w[t],e[t]];return this},abort:function(e){var t=e||u;return c&&c.abort(t),l(0,t),this}};if(x.promise(T),v.url=((e||v.url||Tt.href)+"").replace(Pt,Tt.protocol+"//"),v.type=t.method||t.type||v.method||v.type,v.dataTypes=(v.dataType||"*").toLowerCase().match(P)||[""],null==v.crossDomain){r=E.createElement("a");try{r.href=v.url,r.href=r.href,v.crossDomain=Wt.protocol+"//"+Wt.host!=r.protocol+"//"+r.host}catch(e){v.crossDomain=!0}}if(v.data&&v.processData&&"string"!=typeof v.data&&(v.data=S.param(v.data,v.traditional)),Bt(Rt,v,t,T),h)return T;for(i in(g=S.event&&v.global)&&0==S.active++&&S.event.trigger("ajaxStart"),v.type=v.type.toUpperCase(),v.hasContent=!Ot.test(v.type),f=v.url.replace(qt,""),v.hasContent?v.data&&v.processData&&0===(v.contentType||"").indexOf("application/x-www-form-urlencoded")&&(v.data=v.data.replace(jt,"+")):(o=v.url.slice(f.length),v.data&&(v.processData||"string"==typeof v.data)&&(f+=(Et.test(f)?"&":"?")+v.data,delete v.data),!1===v.cache&&(f=f.replace(Lt,"$1"),o=(Et.test(f)?"&":"?")+"_="+Ct.guid+++o),v.url=f+o),v.ifModified&&(S.lastModified[f]&&T.setRequestHeader("If-Modified-Since",S.lastModified[f]),S.etag[f]&&T.setRequestHeader("If-None-Match",S.etag[f])),(v.data&&v.hasContent&&!1!==v.contentType||t.contentType)&&T.setRequestHeader("Content-Type",v.contentType),T.setRequestHeader("Accept",v.dataTypes[0]&&v.accepts[v.dataTypes[0]]?v.accepts[v.dataTypes[0]]+("*"!==v.dataTypes[0]?", "+It+"; q=0.01":""):v.accepts["*"]),v.headers)T.setRequestHeader(i,v.headers[i]);if(v.beforeSend&&(!1===v.beforeSend.call(y,T,v)||h))return T.abort();if(u="abort",b.add(v.complete),T.done(v.success),T.fail(v.error),c=Bt(Mt,v,t,T)){if(T.readyState=1,g&&m.trigger("ajaxSend",[T,v]),h)return T;v.async&&0<v.timeout&&(d=C.setTimeout(function(){T.abort("timeout")},v.timeout));try{h=!1,c.send(a,l)}catch(e){if(h)throw e;l(-1,e)}}else l(-1,"No Transport");function l(e,t,n,r){var i,o,a,s,u,l=t;h||(h=!0,d&&C.clearTimeout(d),c=void 0,p=r||"",T.readyState=0<e?4:0,i=200<=e&&e<300||304===e,n&&(s=function(e,t,n){var r,i,o,a,s=e.contents,u=e.dataTypes;while("*"===u[0])u.shift(),void 0===r&&(r=e.mimeType||t.getResponseHeader("Content-Type"));if(r)for(i in s)if(s[i]&&s[i].test(r)){u.unshift(i);break}if(u[0]in n)o=u[0];else{for(i in n){if(!u[0]||e.converters[i+" "+u[0]]){o=i;break}a||(a=i)}o=o||a}if(o)return o!==u[0]&&u.unshift(o),n[o]}(v,T,n)),!i&&-1<S.inArray("script",v.dataTypes)&&(v.converters["text script"]=function(){}),s=function(e,t,n,r){var i,o,a,s,u,l={},c=e.dataTypes.slice();if(c[1])for(a in e.converters)l[a.toLowerCase()]=e.converters[a];o=c.shift();while(o)if(e.responseFields[o]&&(n[e.responseFields[o]]=t),!u&&r&&e.dataFilter&&(t=e.dataFilter(t,e.dataType)),u=o,o=c.shift())if("*"===o)o=u;else if("*"!==u&&u!==o){if(!(a=l[u+" "+o]||l["* "+o]))for(i in l)if((s=i.split(" "))[1]===o&&(a=l[u+" "+s[0]]||l["* "+s[0]])){!0===a?a=l[i]:!0!==l[i]&&(o=s[0],c.unshift(s[1]));break}if(!0!==a)if(a&&e["throws"])t=a(t);else try{t=a(t)}catch(e){return{state:"parsererror",error:a?e:"No conversion from "+u+" to "+o}}}return{state:"success",data:t}}(v,s,T,i),i?(v.ifModified&&((u=T.getResponseHeader("Last-Modified"))&&(S.lastModified[f]=u),(u=T.getResponseHeader("etag"))&&(S.etag[f]=u)),204===e||"HEAD"===v.type?l="nocontent":304===e?l="notmodified":(l=s.state,o=s.data,i=!(a=s.error))):(a=l,!e&&l||(l="error",e<0&&(e=0))),T.status=e,T.statusText=(t||l)+"",i?x.resolveWith(y,[o,l,T]):x.rejectWith(y,[T,l,a]),T.statusCode(w),w=void 0,g&&m.trigger(i?"ajaxSuccess":"ajaxError",[T,v,i?o:a]),b.fireWith(y,[T,l]),g&&(m.trigger("ajaxComplete",[T,v]),--S.active||S.event.trigger("ajaxStop")))}return T},getJSON:function(e,t,n){return S.get(e,t,n,"json")},getScript:function(e,t){return S.get(e,void 0,t,"script")}}),S.each(["get","post"],function(e,i){S[i]=function(e,t,n,r){return m(t)&&(r=r||n,n=t,t=void 0),S.ajax(S.extend({url:e,type:i,dataType:r,data:t,success:n},S.isPlainObject(e)&&e))}}),S.ajaxPrefilter(function(e){var t;for(t in e.headers)"content-type"===t.toLowerCase()&&(e.contentType=e.headers[t]||"")}),S._evalUrl=function(e,t,n){return S.ajax({url:e,type:"GET",dataType:"script",cache:!0,async:!1,global:!1,converters:{"text script":function(){}},dataFilter:function(e){S.globalEval(e,t,n)}})},S.fn.extend({wrapAll:function(e){var t;return this[0]&&(m(e)&&(e=e.call(this[0])),t=S(e,this[0].ownerDocument).eq(0).clone(!0),this[0].parentNode&&t.insertBefore(this[0]),t.map(function(){var e=this;while(e.firstElementChild)e=e.firstElementChild;return e}).append(this)),this},wrapInner:function(n){return m(n)?this.each(function(e){S(this).wrapInner(n.call(this,e))}):this.each(function(){var e=S(this),t=e.contents();t.length?t.wrapAll(n):e.append(n)})},wrap:function(t){var n=m(t);return this.each(function(e){S(this).wrapAll(n?t.call(this,e):t)})},unwrap:function(e){return this.parent(e).not("body").each(function(){S(this).replaceWith(this.childNodes)}),this}}),S.expr.pseudos.hidden=function(e){return!S.expr.pseudos.visible(e)},S.expr.pseudos.visible=function(e){return!!(e.offsetWidth||e.offsetHeight||e.getClientRects().length)},S.ajaxSettings.xhr=function(){try{return new C.XMLHttpRequest}catch(e){}};var _t={0:200,1223:204},zt=S.ajaxSettings.xhr();y.cors=!!zt&&"withCredentials"in zt,y.ajax=zt=!!zt,S.ajaxTransport(function(i){var o,a;if(y.cors||zt&&!i.crossDomain)return{send:function(e,t){var n,r=i.xhr();if(r.open(i.type,i.url,i.async,i.username,i.password),i.xhrFields)for(n in i.xhrFields)r[n]=i.xhrFields[n];for(n in i.mimeType&&r.overrideMimeType&&r.overrideMimeType(i.mimeType),i.crossDomain||e["X-Requested-With"]||(e["X-Requested-With"]="XMLHttpRequest"),e)r.setRequestHeader(n,e[n]);o=function(e){return function(){o&&(o=a=r.onload=r.onerror=r.onabort=r.ontimeout=r.onreadystatechange=null,"abort"===e?r.abort():"error"===e?"number"!=typeof r.status?t(0,"error"):t(r.status,r.statusText):t(_t[r.status]||r.status,r.statusText,"text"!==(r.responseType||"text")||"string"!=typeof r.responseText?{binary:r.response}:{text:r.responseText},r.getAllResponseHeaders()))}},r.onload=o(),a=r.onerror=r.ontimeout=o("error"),void 0!==r.onabort?r.onabort=a:r.onreadystatechange=function(){4===r.readyState&&C.setTimeout(function(){o&&a()})},o=o("abort");try{r.send(i.hasContent&&i.data||null)}catch(e){if(o)throw e}},abort:function(){o&&o()}}}),S.ajaxPrefilter(function(e){e.crossDomain&&(e.contents.script=!1)}),S.ajaxSetup({accepts:{script:"text/javascript, application/javascript, application/ecmascript, application/x-ecmascript"},contents:{script:/\b(?:java|ecma)script\b/},converters:{"text script":function(e){return S.globalEval(e),e}}}),S.ajaxPrefilter("script",function(e){void 0===e.cache&&(e.cache=!1),e.crossDomain&&(e.type="GET")}),S.ajaxTransport("script",function(n){var r,i;if(n.crossDomain||n.scriptAttrs)return{send:function(e,t){r=S("<script>").attr(n.scriptAttrs||{}).prop({charset:n.scriptCharset,src:n.url}).on("load error",i=function(e){r.remove(),i=null,e&&t("error"===e.type?404:200,e.type)}),E.head.appendChild(r[0])},abort:function(){i&&i()}}});var Ut,Xt=[],Vt=/(=)\?(?=&|$)|\?\?/;S.ajaxSetup({jsonp:"callback",jsonpCallback:function(){var e=Xt.pop()||S.expando+"_"+Ct.guid++;return this[e]=!0,e}}),S.ajaxPrefilter("json jsonp",function(e,t,n){var r,i,o,a=!1!==e.jsonp&&(Vt.test(e.url)?"url":"string"==typeof e.data&&0===(e.contentType||"").indexOf("application/x-www-form-urlencoded")&&Vt.test(e.data)&&"data");if(a||"jsonp"===e.dataTypes[0])return r=e.jsonpCallback=m(e.jsonpCallback)?e.jsonpCallback():e.jsonpCallback,a?e[a]=e[a].replace(Vt,"$1"+r):!1!==e.jsonp&&(e.url+=(Et.test(e.url)?"&":"?")+e.jsonp+"="+r),e.converters["script json"]=function(){return o||S.error(r+" was not called"),o[0]},e.dataTypes[0]="json",i=C[r],C[r]=function(){o=arguments},n.always(function(){void 0===i?S(C).removeProp(r):C[r]=i,e[r]&&(e.jsonpCallback=t.jsonpCallback,Xt.push(r)),o&&m(i)&&i(o[0]),o=i=void 0}),"script"}),y.createHTMLDocument=((Ut=E.implementation.createHTMLDocument("").body).innerHTML="<form></form><form></form>",2===Ut.childNodes.length),S.parseHTML=function(e,t,n){return"string"!=typeof e?[]:("boolean"==typeof t&&(n=t,t=!1),t||(y.createHTMLDocument?((r=(t=E.implementation.createHTMLDocument("")).createElement("base")).href=E.location.href,t.head.appendChild(r)):t=E),o=!n&&[],(i=N.exec(e))?[t.createElement(i[1])]:(i=xe([e],t,o),o&&o.length&&S(o).remove(),S.merge([],i.childNodes)));var r,i,o},S.fn.load=function(e,t,n){var r,i,o,a=this,s=e.indexOf(" ");return-1<s&&(r=vt(e.slice(s)),e=e.slice(0,s)),m(t)?(n=t,t=void 0):t&&"object"==typeof t&&(i="POST"),0<a.length&&S.ajax({url:e,type:i||"GET",dataType:"html",data:t}).done(function(e){o=arguments,a.html(r?S("<div>").append(S.parseHTML(e)).find(r):e)}).always(n&&function(e,t){a.each(function(){n.apply(this,o||[e.responseText,t,e])})}),this},S.expr.pseudos.animated=function(t){return S.grep(S.timers,function(e){return t===e.elem}).length},S.offset={setOffset:function(e,t,n){var r,i,o,a,s,u,l=S.css(e,"position"),c=S(e),f={};"static"===l&&(e.style.position="relative"),s=c.offset(),o=S.css(e,"top"),u=S.css(e,"left"),("absolute"===l||"fixed"===l)&&-1<(o+u).indexOf("auto")?(a=(r=c.position()).top,i=r.left):(a=parseFloat(o)||0,i=parseFloat(u)||0),m(t)&&(t=t.call(e,n,S.extend({},s))),null!=t.top&&(f.top=t.top-s.top+a),null!=t.left&&(f.left=t.left-s.left+i),"using"in t?t.using.call(e,f):("number"==typeof f.top&&(f.top+="px"),"number"==typeof f.left&&(f.left+="px"),c.css(f))}},S.fn.extend({offset:function(t){if(arguments.length)return void 0===t?this:this.each(function(e){S.offset.setOffset(this,t,e)});var e,n,r=this[0];return r?r.getClientRects().length?(e=r.getBoundingClientRect(),n=r.ownerDocument.defaultView,{top:e.top+n.pageYOffset,left:e.left+n.pageXOffset}):{top:0,left:0}:void 0},position:function(){if(this[0]){var e,t,n,r=this[0],i={top:0,left:0};if("fixed"===S.css(r,"position"))t=r.getBoundingClientRect();else{t=this.offset(),n=r.ownerDocument,e=r.offsetParent||n.documentElement;while(e&&(e===n.body||e===n.documentElement)&&"static"===S.css(e,"position"))e=e.parentNode;e&&e!==r&&1===e.nodeType&&((i=S(e).offset()).top+=S.css(e,"borderTopWidth",!0),i.left+=S.css(e,"borderLeftWidth",!0))}return{top:t.top-i.top-S.css(r,"marginTop",!0),left:t.left-i.left-S.css(r,"marginLeft",!0)}}},offsetParent:function(){return this.map(function(){var e=this.offsetParent;while(e&&"static"===S.css(e,"position"))e=e.offsetParent;return e||re})}}),S.each({scrollLeft:"pageXOffset",scrollTop:"pageYOffset"},function(t,i){var o="pageYOffset"===i;S.fn[t]=function(e){return $(this,function(e,t,n){var r;if(x(e)?r=e:9===e.nodeType&&(r=e.defaultView),void 0===n)return r?r[i]:e[t];r?r.scrollTo(o?r.pageXOffset:n,o?n:r.pageYOffset):e[t]=n},t,e,arguments.length)}}),S.each(["top","left"],function(e,n){S.cssHooks[n]=$e(y.pixelPosition,function(e,t){if(t)return t=Be(e,n),Me.test(t)?S(e).position()[n]+"px":t})}),S.each({Height:"height",Width:"width"},function(a,s){S.each({padding:"inner"+a,content:s,"":"outer"+a},function(r,o){S.fn[o]=function(e,t){var n=arguments.length&&(r||"boolean"!=typeof e),i=r||(!0===e||!0===t?"margin":"border");return $(this,function(e,t,n){var r;return x(e)?0===o.indexOf("outer")?e["inner"+a]:e.document.documentElement["client"+a]:9===e.nodeType?(r=e.documentElement,Math.max(e.body["scroll"+a],r["scroll"+a],e.body["offset"+a],r["offset"+a],r["client"+a])):void 0===n?S.css(e,t,i):S.style(e,t,n,i)},s,n?e:void 0,n)}})}),S.each(["ajaxStart","ajaxStop","ajaxComplete","ajaxError","ajaxSuccess","ajaxSend"],function(e,t){S.fn[t]=function(e){return this.on(t,e)}}),S.fn.extend({bind:function(e,t,n){return this.on(e,null,t,n)},unbind:function(e,t){return this.off(e,null,t)},delegate:function(e,t,n,r){return this.on(t,e,n,r)},undelegate:function(e,t,n){return 1===arguments.length?this.off(e,"**"):this.off(t,e||"**",n)},hover:function(e,t){return this.mouseenter(e).mouseleave(t||e)}}),S.each("blur focus focusin focusout resize scroll click dblclick mousedown mouseup mousemove mouseover mouseout mouseenter mouseleave change select submit keydown keypress keyup contextmenu".split(" "),function(e,n){S.fn[n]=function(e,t){return 0<arguments.length?this.on(n,null,e,t):this.trigger(n)}});var Gt=/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g;S.proxy=function(e,t){var n,r,i;if("string"==typeof t&&(n=e[t],t=e,e=n),m(e))return r=s.call(arguments,2),(i=function(){return e.apply(t||this,r.concat(s.call(arguments)))}).guid=e.guid=e.guid||S.guid++,i},S.holdReady=function(e){e?S.readyWait++:S.ready(!0)},S.isArray=Array.isArray,S.parseJSON=JSON.parse,S.nodeName=A,S.isFunction=m,S.isWindow=x,S.camelCase=X,S.type=w,S.now=Date.now,S.isNumeric=function(e){var t=S.type(e);return("number"===t||"string"===t)&&!isNaN(e-parseFloat(e))},S.trim=function(e){return null==e?"":(e+"").replace(Gt,"")},"function"==typeof define&&define.amd&&define("jquery",[],function(){return S});var Yt=C.jQuery,Qt=C.$;return S.noConflict=function(e){return C.$===S&&(C.$=Qt),e&&C.jQuery===S&&(C.jQuery=Yt),S},"undefined"==typeof e&&(C.jQuery=C.$=S),S});
</script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<style type="text/css">html{font-family:sans-serif;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%}body{margin:0}article,aside,details,figcaption,figure,footer,header,hgroup,main,menu,nav,section,summary{display:block}audio,canvas,progress,video{display:inline-block;vertical-align:baseline}audio:not([controls]){display:none;height:0}[hidden],template{display:none}a{background-color:transparent}a:active,a:hover{outline:0}abbr[title]{border-bottom:1px dotted}b,strong{font-weight:700}dfn{font-style:italic}h1{margin:.67em 0;font-size:2em}mark{color:#000;background:#ff0}small{font-size:80%}sub,sup{position:relative;font-size:75%;line-height:0;vertical-align:baseline}sup{top:-.5em}sub{bottom:-.25em}img{border:0}svg:not(:root){overflow:hidden}figure{margin:1em 40px}hr{height:0;-webkit-box-sizing:content-box;-moz-box-sizing:content-box;box-sizing:content-box}pre{overflow:auto}code,kbd,pre,samp{font-family:monospace,monospace;font-size:1em}button,input,optgroup,select,textarea{margin:0;font:inherit;color:inherit}button{overflow:visible}button,select{text-transform:none}button,html input[type=button],input[type=reset],input[type=submit]{-webkit-appearance:button;cursor:pointer}button[disabled],html input[disabled]{cursor:default}button::-moz-focus-inner,input::-moz-focus-inner{padding:0;border:0}input{line-height:normal}input[type=checkbox],input[type=radio]{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box;padding:0}input[type=number]::-webkit-inner-spin-button,input[type=number]::-webkit-outer-spin-button{height:auto}input[type=search]{-webkit-box-sizing:content-box;-moz-box-sizing:content-box;box-sizing:content-box;-webkit-appearance:textfield}input[type=search]::-webkit-search-cancel-button,input[type=search]::-webkit-search-decoration{-webkit-appearance:none}fieldset{padding:.35em .625em .75em;margin:0 2px;border:1px solid silver}legend{padding:0;border:0}textarea{overflow:auto}optgroup{font-weight:700}table{border-spacing:0;border-collapse:collapse}td,th{padding:0}@media print{*,:after,:before{color:#000!important;text-shadow:none!important;background:0 0!important;-webkit-box-shadow:none!important;box-shadow:none!important}a,a:visited{text-decoration:underline}a[href]:after{content:" (" attr(href) ")"}abbr[title]:after{content:" (" attr(title) ")"}a[href^="javascript:"]:after,a[href^="#"]:after{content:""}blockquote,pre{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}img,tr{page-break-inside:avoid}img{max-width:100%!important}h2,h3,p{orphans:3;widows:3}h2,h3{page-break-after:avoid}.navbar{display:none}.btn>.caret,.dropup>.btn>.caret{border-top-color:#000!important}.label{border:1px solid #000}.table{border-collapse:collapse!important}.table td,.table th{background-color:#fff!important}.table-bordered td,.table-bordered th{border:1px solid #ddd!important}}@font-face{font-family:'Glyphicons Halflings';src:url(data:application/vnd.ms-fontobject;base64,n04AAEFNAAACAAIABAAAAAAABQAAAAAAAAABAJABAAAEAExQAAAAAAAAAAIAAAAAAAAAAAEAAAAAAAAAJxJ/LAAAAAAAAAAAAAAAAAAAAAAAACgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzAAAADgBSAGUAZwB1AGwAYQByAAAAeABWAGUAcgBzAGkAbwBuACAAMQAuADAAMAA5ADsAUABTACAAMAAwADEALgAwADAAOQA7AGgAbwB0AGMAbwBuAHYAIAAxAC4AMAAuADcAMAA7AG0AYQBrAGUAbwB0AGYALgBsAGkAYgAyAC4ANQAuADUAOAAzADIAOQAAADgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzACAAUgBlAGcAdQBsAGEAcgAAAAAAQlNHUAAAAAAAAAAAAAAAAAAAAAADAKncAE0TAE0ZAEbuFM3pjM/SEdmjKHUbyow8ATBE40IvWA3vTu8LiABDQ+pexwUMcm1SMnNryctQSiI1K5ZnbOlXKmnVV5YvRe6RnNMFNCOs1KNVpn6yZhCJkRtVRNzEufeIq7HgSrcx4S8h/v4vnrrKc6oCNxmSk2uKlZQHBii6iKFoH0746ThvkO1kJHlxjrkxs+LWORaDQBEtiYJIR5IB9Bi1UyL4Rmr0BNigNkMzlKQmnofBHviqVzUxwdMb3NdCn69hy+pRYVKGVS/1tnsqv4LL7wCCPZZAZPT4aCShHjHJVNuXbmMrY5LeQaGnvAkXlVrJgKRAUdFjrWEah9XebPeQMj7KS7DIBAFt8ycgC5PLGUOHSE3ErGZCiViNLL5ZARfywnCoZaKQCu6NuFX42AEeKtKUGnr/Cm2Cy8tpFhBPMW5Fxi4Qm4TkDWh4IWFDClhU2hRWosUWqcKLlgyXB+lSHaWaHiWlBAR8SeSgSPCQxdVQgzUixWKSTrIQEbU94viDctkvX+VSjJuUmV8L4CXShI11esnp0pjWNZIyxKHS4wVQ2ime1P4RnhvGw0aDN1OLAXGERsB7buFpFGGBAre4QEQR0HOIO5oYH305G+KspT/FupEGGafCCwxSe6ZUa+073rXHnNdVXE6eWvibUS27XtRzkH838mYLMBmYysZTM0EM3A1fbpCBYFccN1B/EnCYu/TgCGmr7bMh8GfYL+BfcLvB0gRagC09w9elfldaIy/hNCBLRgBgtCC7jAF63wLSMAfbfAlEggYU0bUA7ACCJmTDpEmJtI78w4/BO7dN7JR7J7ZvbYaUbaILSQsRBiF3HGk5fEg6p9unwLvn98r+vnsV+372uf1xBLq4qU/45fTuqaAP+pssmCCCTF0mhEow8ZXZOS8D7Q85JsxZ+Azok7B7O/f6J8AzYBySZQB/QHYUSA+EeQhEWiS6AIQzgcsDiER4MjgMBAWDV4AgQ3g1eBgIdweCQmCjJEMkJ+PKRWyFHHmg1Wi/6xzUgA0LREoKJChwnQa9B+5RQZRB3IlBlkAnxyQNaANwHMowzlYSMCBgnbpzvqpl0iTJNCQidDI9ZrSYNIRBhHtUa5YHMHxyGEik9hDE0AKj72AbTCaxtHPUaKZdAZSnQTyjGqGLsmBStCejApUhg4uBMU6mATujEl+KdDPbI6Ag4vLr+hjY6lbjBeoLKnZl0UZgRX8gTySOeynZVz1wOq7e1hFGYIq+MhrGxDLak0PrwYzSXtcuyhXEhwOYofiW+EcI/jw8P6IY6ed+etAbuqKp5QIapT77LnAe505lMuqL79a0ut4rWexzFttsOsLDy7zvtQzcq3U1qabe7tB0wHWVXji+zDbo8x8HyIRUbXnwUcklFv51fvTymiV+MXLSmGH9d9+aXpD5X6lao41anWGig7IwIdnoBY2ht/pO9mClLo4NdXHAsefqWUKlXJkbqPOFhMoR4aiA1BXqhRNbB2Xwi+7u/jpAoOpKJ0UX24EsrzMfHXViakCNcKjBxuQX8BO0ZqjJ3xXzf+61t2VXOSgJ8xu65QKgtN6FibPmPYsXbJRHHqbgATcSZxBqGiDiU4NNNsYBsKD0MIP/OfKnlk/Lkaid/O2NbKeuQrwOB2Gq3YHyr6ALgzym5wIBnsdC1ZkoBFZSQXChZvlesPqvK2c5oHHT3Q65jYpNxnQcGF0EHbvYqoFw60WNlXIHQF2HQB7zD6lWjZ9rVqUKBXUT6hrkZOle0RFYII0V5ZYGl1JAP0Ud1fZZMvSomBzJ710j4Me8mjQDwEre5Uv2wQfk1ifDwb5ksuJQQ3xt423lbuQjvoIQByQrNDh1JxGFkOdlJvu/gFtuW0wR4cgd+ZKesSV7QkNE2kw6AV4hoIuC02LGmTomyf8PiO6CZzOTLTPQ+HW06H+tx+bQ8LmDYg1pTFrp2oJXgkZTyeRJZM0C8aE2LpFrNVDuhARsN543/FV6klQ6Tv1OoZGXLv0igKrl/CmJxRmX7JJbJ998VSIPQRyDBICzl4JJlYHbdql30NvYcOuZ7a10uWRrgoieOdgIm4rlq6vNOQBuqESLbXG5lzdJGHw2m0sDYmODXbYGTfSTGRKpssTO95fothJCjUGQgEL4yKoGAF/0SrpUDNn8CBgBcSDQByAeNkCXp4S4Ro2Xh4OeaGRgR66PVOsU8bc6TR5/xTcn4IVMLOkXSWiXxkZQCbvKfmoAvQaKjO3EDKwkwqHChCDEM5loQRPd5ACBki1TjF772oaQhQbQ5C0lcWXPFOzrfsDGUXGrpxasbG4iab6eByaQkQfm0VFlP0ZsDkvvqCL6QXMUwCjdMx1ZOyKhTJ7a1GWAdOUcJ8RSejxNVyGs31OKMyRyBVoZFjqIkmKlLQ5eHMeEL4MkUf23cQ/1SgRCJ1dk4UdBT7OoyuNgLs0oCd8RnrEIb6QdMxT2QjD4zMrJkfgx5aDMcA4orsTtKCqWb/Veyceqa5OGSmB28YwH4rFbkQaLoUN8OQQYnD3w2eXpI4ScQfbCUZiJ4yMOIKLyyTc7BQ4uXUw6Ee6/xM+4Y67ngNBknxIPwuppgIhFcwJyr6EIj+LzNj/mfR2vhhRlx0BILZoAYruF0caWQ7YxO66UmeguDREAFHYuC7HJviRgVO6ruJH59h/C/PkgSle8xNzZJULLWq9JMDTE2fjGE146a1Us6PZDGYle6ldWRqn/pdpgHKNGrGIdkRK+KPETT9nKT6kLyDI8xd9A1FgWmXWRAIHwZ37WyZHOVyCadJEmMVz0MadMjDrPho+EIochkVC2xgGiwwsQ6DMv2P7UXqT4x7CdcYGId2BJQQa85EQKmCmwcRejQ9Bm4oATENFPkxPXILHpMPUyWTI5rjNOsIlmEeMbcOCEqInpXACYQ9DDxmFo9vcmsDblcMtg4tqBerNngkIKaFJmrQAPnq1dEzsMXcwjcHdfdCibcAxxA+q/j9m3LM/O7WJka4tSidVCjsvo2lQ/2ewyoYyXwAYyr2PlRoR5MpgVmSUIrM3PQxXPbgjBOaDQFIyFMJvx3Pc5RSYj12ySVF9fwFPQu2e2KWVoL9q3Ayv3IzpGHUdvdPdrNUdicjsTQ2ISy7QU3DrEytIjvbzJnAkmANXjAFERA0MUoPF3/5KFmW14bBNOhwircYgMqoDpUMcDtCmBE82QM2YtdjVLB4kBuKho/bcwQdeboqfQartuU3CsCf+cXkgYAqp/0Ee3RorAZt0AvvOCSI4JICIlGlsV0bsSid/NIEALAAzb6HAgyWHBps6xAOwkJIGcB82CxRQq4sJf3FzA70A+TRqcqjEMETCoez3mkPcpnoALs0ugJY8kQwrC+JE5ik3w9rzrvDRjAQnqgEVvdGrNwlanR0SOKWzxOJOvLJhcd8Cl4AshACUkv9czdMkJCVQSQhp6kp7StAlpVRpK0t0SW6LHeBJnE2QchB5Ccu8kxRghZXGIgZIiSj7gEKMJDClcnX6hgoqJMwiQDigIXg3ioFLCgDgjPtYHYpsF5EiA4kcnN18MZtOrY866dEQAb0FB34OGKHGZQjwW/WDHA60cYFaI/PjpzquUqdaYGcIq+mLez3WLFFCtNBN2QJcrlcoELgiPku5R5dSlJFaCEqEZle1AQzAKC+1SotMcBNyQUFuRHRF6OlimSBgjZeTBCwLyc6A+P/oFRchXTz5ADknYJHxzrJ5pGuIKRQISU6WyKTBBjD8WozmVYWIsto1AS5rxzKlvJu4E/vwOiKxRtCWsDM+eTHUrmwrCK5BIfMzGkD+0Fk5LzBs0jMYXktNDblB06LMNJ09U8pzSLmo14MS0OMjcdrZ31pyQqxJJpRImlSvfYAK8inkYU52QY2FPEVsjoWewpwhRp5yAuNpkqhdb7ku9Seefl2D0B8SMTFD90xi4CSOwwZy9IKkpMtI3FmFUg3/kFutpQGNc3pCR7gvC4sgwbupDu3DyEN+W6YGLNM21jpB49irxy9BSlHrVDlnihGKHwPrbVFtc+h1rVQKZduxIyojccZIIcOCmhEnC7UkY68WXKQgLi2JCDQkQWJRQuk60hZp0D3rtCTINSeY9Ej2kIKYfGxwOs4j9qMM7fYZiipzgcf7TamnehqdhsiMiCawXnz4xAbyCkLAx5EGbo3Ax1u3dUIKnTxIaxwQTHehPl3V491H0+bC5zgpGz7Io+mjdhKlPJ01EeMpM7UsRJMi1nGjmJg35i6bQBAAxjO/ENJubU2mg3ONySEoWklCwdABETcs7ck3jgiuU9pcKKpbgn+3YlzV1FzIkB6pmEDOSSyDfPPlQskznctFji0kpgZjW5RZe6x9kYT4KJcXg0bNiCyif+pZACCyRMmYsfiKmN9tSO65F0R2OO6ytlEhY5Sj6uRKfFxw0ijJaAx/k3QgnAFSq27/2i4GEBA+UvTJKK/9eISNvG46Em5RZfjTYLdeD8kdXHyrwId/DQZUaMCY4gGbke2C8vfjgV/Y9kkRQOJIn/xM9INZSpiBnqX0Q9GlQPpPKAyO5y+W5NMPSRdBCUlmuxl40ZfMCnf2Cp044uI9WLFtCi4YVxKjuRCOBWIb4XbIsGdbo4qtMQnNOQz4XDSui7W/N6l54qOynCqD3DpWQ+mpD7C40D8BZEWGJX3tlAaZBMj1yjvDYKwCJBa201u6nBKE5UE+7QSEhCwrXfbRZylAaAkplhBWX50dumrElePyNMRYUrC99UmcSSNgImhFhDI4BXjMtiqkgizUGCrZ8iwFxU6fQ8GEHCFdLewwxYWxgScAYMdMLmcZR6b7rZl95eQVDGVoUKcRMM1ixXQtXNkBETZkVVPg8LoSrdetHzkuM7DjZRHP02tCxA1fmkXKF3VzfN1pc1cv/8lbTIkkYpqKM9VOhp65ktYk+Q46myFWBapDfyWUCnsnI00QTBQmuFjMZTcd0V2NQ768Fhpby04k2IzNR1wKabuGJqYWwSly6ocMFGTeeI+ejsWDYgEvr66QgqdcIbFYDNgsm0x9UHY6SCd5+7tpsLpKdvhahIDyYmEJQCqMqtCF6UlrE5GXRmbu+vtm3BFSxI6ND6UxIE7GsGMgWqghXxSnaRJuGFveTcK5ZVSPJyjUxe1dKgI6kNF7EZhIZs8y8FVqwEfbM0Xk2ltORVDKZZM40SD3qQoQe0orJEKwPfZwm3YPqwixhUMOndis6MhbmfvLBKjC8sKKIZKbJk8L11oNkCQzCgvjhyyEiQSuJcgCQSG4Mocfgc0Hkwcjal1UNgP0CBPikYqBIk9tONv4kLtBswH07vUCjEaHiFGlLf8MgXKzSgjp2HolRRccAOh0ILHz9qlGgIFkwAnzHJRjWFhlA7ROwINyB5HFj59PRZHFor6voq7l23EPNRwdWhgawqbivLSjRA4htEYUFkjESu67icTg5S0aW1sOkCiIysfJ9UnIWevOOLGpepcBxy1wEhd2WI3AZg7sr9WBmHWyasxMcvY/iOmsLtHSWNUWEGk9hScMPShasUA1AcHOtRZlqMeQ0OzYS9vQvYUjOLrzP07BUAFikcJNMi7gIxEw4pL1G54TcmmmoAQ5s7TGWErJZ2Io4yQ0ljRYhL8H5e62oDtLF8aDpnIvZ5R3GWJyAugdiiJW9hQAVTsnCBHhwu7rkBlBX6r3b7ejEY0k5GGeyKv66v+6dg7mcJTrWHbtMywbedYqCQ0FPwoytmSWsL8WTtChZCKKzEF7vP6De4x2BJkkniMgSdWhbeBSLtJZR9CTHetK1xb34AYIJ37OegYIoPVbXgJ/qDQK+bfCtxQRVKQu77WzOoM6SGL7MaZwCGJVk46aImai9fmam+WpHG+0BtQPWUgZ7RIAlPq6lkECUhZQ2gqWkMYKcYMYaIc4gYCDFHYa2d1nzp3+J1eCBay8IYZ0wQRKGAqvCuZ/UgbQPyllosq+XtfKIZOzmeJqRazpmmoP/76YfkjzV2NlXTDSBYB04SVlNQsFTbGPk1t/I4Jktu0XSgifO2ozFOiwd/0SssJDn0dn4xqk4GDTTKX73/wQyBLdqgJ+Wx6AQaba3BA9CKEzjtQYIfAsiYamapq80LAamYjinlKXUkxdpIDk0puXUEYzSalfRibAeDAKpNiqQ0FTwoxuGYzRnisyTotdVTclis1LHRQCy/qqL8oUaQzWRxilq5Mi0IJGtMY02cGLD69vGjkj3p6pGePKI8bkBv5evq8SjjyU04vJR2cQXQwSJyoinDsUJHCQ50jrFTT7yRdbdYQMB3MYCb6uBzJ9ewhXYPAIZSXfeEQBZZ3GPN3Nbhh/wkvAJLXnQMdi5NYYZ5GHE400GS5rXkOZSQsdZgIbzRnF9ueLnsfQ47wHAsirITnTlkCcuWWIUhJSbpM3wWhXNHvt2xUsKKMpdBSbJnBMcihkoDqAd1Zml/R4yrzow1Q2A5G+kzo/RhRxQS2lCSDRV8LlYLBOOoo1bF4jwJAwKMK1tWLHlu9i0j4Ig8qVm6wE1DxXwAwQwsaBWUg2pOOol2dHxyt6npwJEdLDDVYyRc2D0HbcbLUJQj8gPevQBUBOUHXPrsAPBERICpnYESeu2OHotpXQxRGlCCtLdIsu23MhZVEoJg8Qumj/UMMc34IBqTKLDTp76WzL/dMjCxK7MjhiGjeYAC/kj/jY/Rde7hpSM1xChrog6yZ7OWTuD56xBJnGFE+pT2ElSyCnJcwVzCjkqeNLfMEJqKW0G7OFIp0G+9mh50I9o8k1tpCY0xYqFNIALgIfc2me4n1bmJnRZ89oepgLPT0NTMLNZsvSCZAc3TXaNB07vail36/dBySis4m9/DR8izaLJW6bWCkVgm5T+ius3ZXq4xI+GnbveLbdRwF2mNtsrE0JjYc1AXknCOrLSu7Te/r4dPYMCl5qtiHNTn+TPbh1jCBHH+dMJNhwNgs3nT+OhQoQ0vYif56BMG6WowAcHR3DjQolxLzyVekHj00PBAaW7IIAF1EF+uRIWyXjQMAs2chdpaKPNaB+kSezYt0+CA04sOg5vx8Fr7Ofa9sUv87h7SLAUFSzbetCCZ9pmyLt6l6/TzoA1/ZBG9bIUVHLAbi/kdBFgYGyGwRQGBpkqCEg2ah9UD6EedEcEL3j4y0BQQCiExEnocA3SZboh+epgd3YsOkHskZwPuQ5OoyA0fTA5AXrHcUOQF+zkJHIA7PwCDk1gGVmGUZSSoPhNf+Tklauz98QofOlCIQ/tCD4dosHYPqtPCXB3agggQQIqQJsSkB+qn0rkQ1toJjON/OtCIB9RYv3PqRA4C4U68ZMlZn6BdgEvi2ziU+TQ6NIw3ej+AtDwMGEZk7e2IjxUWKdAxyaw9OCwSmeADTPPleyk6UhGDNXQb++W6Uk4q6F7/rg6WVTo82IoCxSIsFDrav4EPHphD3u4hR53WKVvYZUwNCCeM4PMBWzK+EfIthZOkuAwPo5C5jgoZgn6dUdvx5rIDmd58cXXdKNfw3l+wM2UjgrDJeQHhbD7HW2QDoZMCujgIUkk5Fg8VCsdyjOtnGRx8wgKRPZN5dR0zPUyfGZFVihbFRniXZFOZGKPnEQzU3AnD1KfR6weHW2XS6KbPJxUkOTZsAB9vTVp3Le1F8q5l+DMcLiIq78jxAImD2pGFw0VHfRatScGlK6SMu8leTmhUSMy8Uhdd6xBiH3Gdman4tjQGLboJfqz6fL2WKHTmrfsKZRYX6BTDjDldKMosaSTLdQS7oDisJNqAUhw1PfTlnacCO8vl8706Km1FROgLDmudzxg+EWTiArtHgLsRrAXYWdB0NmToNCJdKm0KWycZQqb+Mw76Qy29iQ5up/X7oyw8QZ75kP5F6iJAJz6KCmqxz8fEa/xnsMYcIO/vEkGRuMckhr4rIeLrKaXnmIzlNLxbFspOphkcnJdnz/Chp/Vlpj2P7jJQmQRwGnltkTV5dbF9fE3/fxoSqTROgq9wFUlbuYzYcasE0ouzBo+dDCDzxKAfhbAZYxQiHrLzV2iVexnDX/QnT1fsT/xuhu1ui5qIytgbGmRoQkeQooO8eJNNZsf0iALur8QxZFH0nCMnjerYQqG1pIfjyVZWxhVRznmmfLG00BcBWJE6hzQWRyFknuJnXuk8A5FRDCulwrWASSNoBtR+CtGdkPwYN2o7DOw/VGlCZPusRBFXODQdUM5zeHDIVuAJBLqbO/f9Qua+pDqEPk230Sob9lEZ8BHiCorjVghuI0lI4JDgHGRDD/prQ84B1pVGkIpVUAHCG+iz3Bn3qm2AVrYcYWhock4jso5+J7HfHVj4WMIQdGctq3psBCVVzupQOEioBGA2Bk+UILT7+VoX5mdxxA5fS42gISQVi/HTzrgMxu0fY6hE1ocUwwbsbWcezrY2n6S8/6cxXkOH4prpmPuFoikTzY7T85C4T2XYlbxLglSv2uLCgFv8Quk/wdesUdWPeHYIH0R729JIisN9Apdd4eB10aqwXrPt+Su9mA8k8n1sjMwnfsfF2j3jMUzXepSHmZ/BfqXvzgUNQQWOXO8YEuFBh4QTYCkOAPxywpYu1VxiDyJmKVcmJPGWk/gc3Pov02StyYDahwmzw3E1gYC9wkupyWfDqDSUMpCTH5e5N8B//lHiMuIkTNw4USHrJU67bjXGqNav6PBuQSoqTxc8avHoGmvqNtXzIaoyMIQIiiUHIM64cXieouplhNYln7qgc4wBVAYR104kO+CvKqsg4yIUlFNThVUAKZxZt1XA34h3TCUUiXVkZ0w8Hh2R0Z5L0b4LZvPd/p1gi/07h8qfwHrByuSxglc9cI4QIg2oqvC/qm0i7tjPLTgDhoWTAKDO2ONW5oe+/eKB9vZB8K6C25yCZ9RFVMnb6NRdRjyVK57CHHSkJBfnM2/j4ODUwRkqrtBBCrDsDpt8jhZdXoy/1BCqw3sSGhgGGy0a5Jw6BP/TExoCmNFYjZl248A0osgPyGEmRA+fAsqPVaNAfytu0vuQJ7rk3J4kTDTR2AlCHJ5cls26opZM4w3jMULh2YXKpcqGBtuleAlOZnaZGbD6DHzMd6i2oFeJ8z9XYmalg1Szd/ocZDc1C7Y6vcALJz2lYnTXiWEr2wawtoR4g3jvWUU2Ngjd1cewtFzEvM1NiHZPeLlIXFbBPawxNgMwwAlyNSuGF3zizVeOoC9bag1qRAQKQE/EZBWC2J8mnXAN2aTBboZ7HewnObE8CwROudZHmUM5oZ/Ugd/JZQK8lvAm43uDRAbyW8gZ+ZGq0EVerVGUKUSm/Idn8AQHdR4m7bue88WBwft9mSCeMOt1ncBwziOmJYI2ZR7ewNMPiCugmSsE4EyQ+QATJG6qORMGd4snEzc6B4shPIo4G1T7PgSm8PY5eUkPdF8JZ0VBtadbHXoJgnEhZQaODPj2gpODKJY5Yp4DOsLBFxWbvXN755KWylJm+oOd4zEL9Hpubuy2gyyfxh8oEfFutnYWdfB8PdESLWYvSqbElP9qo3u6KTmkhoacDauMNNjj0oy40DFV7Ql0aZj77xfGl7TJNHnIwgqOkenruYYNo6h724+zUQ7+vkCpZB+pGA562hYQiDxHVWOq0oDQl/QsoiY+cuI7iWq/ZIBtHcXJ7kks+h2fCNUPA82BzjnqktNts+RLdk1VSu+tqEn7QZCCsvEqk6FkfiOYkrsw092J8jsfIuEKypNjLxrKA9kiA19mxBD2suxQKCzwXGws7kEJvlhUiV9tArLIdZW0IORcxEzdzKmjtFhsjKy/44XYXdI5noQoRcvjZ1RMPACRqYg2V1+OwOepcOknRLLFdYgTkT5UApt/JhLM3jeFYprZV+Zow2g8fP+U68hkKFWJj2yBbKqsrp25xkZX1DAjUw52IMYWaOhab8Kp05VrdNftqwRrymWF4OQSjbdfzmRZirK8FMJELEgER2PHjEAN9pGfLhCUiTJFbd5LBkOBMaxLr/A1SY9dXFz4RjzoU9ExfJCmx/I9FKEGT3n2cmzl2X42L3Jh+AbQq6sA+Ss1kitoa4TAYgKHaoybHUDJ51oETdeI/9ThSmjWGkyLi5QAGWhL0BG1UsTyRGRJOldKBrYJeB8ljLJHfATWTEQBXBDnQexOHTB+Un44zExFE4vLytcu5NwpWrUxO/0ZICUGM7hGABXym0V6ZvDST0E370St9MIWQOTWngeoQHUTdCJUP04spMBMS8LSker9cReVQkULFDIZDFPrhTzBl6sed9wcZQTbL+BDqMyaN3RJPh/anbx+Iv+qgQdAa3M9Z5JmvYlh4qop+Ho1F1W5gbOE9YKLgAnWytXElU4G8GtW47lhgFE6gaSs+gs37sFvi0PPVvA5dnCBgILTwoKd/+DoL9F6inlM7H4rOTzD79KJgKlZO/Zgt22UsKhrAaXU5ZcLrAglTVKJEmNJvORGN1vqrcfSMizfpsgbIe9zno+gBoKVXgIL/VI8dB1O5o/R3Suez/gD7M781ShjKpIIORM/nxG+jjhhgPwsn2IoXsPGPqYHXA63zJ07M2GPEykQwJBYLK808qYxuIew4frk52nhCsnCYmXiR6CuapvE1IwRB4/QftDbEn+AucIr1oxrLabRj9q4ae0+fXkHnteAJwXRbVkR0mctVSwEbqhJiMSZUp9DNbEDMmjX22m3ABpkrPQQTP3S1sib5pD2VRKRd+eNAjLYyT0hGrdjWJZy24OYXRoWQAIhGBZRxuBFMjjZQhpgrWo8SiFYbojcHO8V5DyscJpLTHyx9Fimassyo5U6WNtquUMYgccaHY5amgR3PQzq3ToNM5ABnoB9kuxsebqmYZm0R9qxJbFXCQ1UPyFIbxoUraTJFDpCk0Wk9GaYJKz/6oHwEP0Q14lMtlddQsOAU9zlYdMVHiT7RQP3XCmWYDcHCGbVRHGnHuwzScA0BaSBOGkz3lM8CArjrBsyEoV6Ys4qgDK3ykQQPZ3hCRGNXQTNNXbEb6tDiTDLKOyMzRhCFT+mAUmiYbV3YQVqFVp9dorv+TsLeCykS2b5yyu8AV7IS9cxcL8z4Kfwp+xJyYLv1OsxQCZwTB4a8BZ/5EdxTBJthApqyfd9u3ifr/WILTqq5VqgwMT9SOxbSGWLQJUUWCVi4k9tho9nEsbUh7U6NUsLmkYFXOhZ0kmamaJLRNJzSj/qn4Mso6zb6iLLBXoaZ6AqeWCjHQm2lztnejYYM2eubnpBdKVLORZhudH3JF1waBJKA9+W8EhMj3Kzf0L4vi4k6RoHh3Z5YgmSZmk6ns4fjScjAoL8GoOECgqgYEBYUGFVO4FUv4/YtowhEmTs0vrvlD/CrisnoBNDAcUi/teY7OctFlmARQzjOItrrlKuPO6E2Ox93L4O/4DcgV/dZ7qR3VBwVQxP1GCieA4RIpweYJ5FoYrHxqRBdJjnqbsikA2Ictbb8vE1GYIo9dacK0REgDX4smy6GAkxlH1yCGGsk+tgiDhNKuKu3yNrMdxafmKTF632F8Vx4BNK57GvlFisrkjN9WDAtjsWA0ENT2e2nETUb/n7qwhvGnrHuf5bX6Vh/n3xffU3PeHdR+FA92i6ufT3AlyAREoNDh6chiMWTvjKjHDeRhOa9YkOQRq1vQXEMppAQVwHCuIcV2g5rBn6GmZZpTR7vnSD6ZmhdSl176gqKTXu5E+YbfL0adwNtHP7dT7t7b46DVZIkzaRJOM+S6KcrzYVg+T3wSRFRQashjfU18NutrKa/7PXbtuJvpIjbgPeqd+pjmRw6YKpnANFSQcpzTZgpSNJ6J7uiagAbir/8tNXJ/OsOnRh6iuIexxrmkIneAgz8QoLmiaJ8sLQrELVK2yn3wOHp57BAZJhDZjTBzyoRAuuZ4eoxHruY1pSb7qq79cIeAdOwin4GdgMeIMHeG+FZWYaiUQQyC5b50zKjYw97dFjAeY2I4Bnl105Iku1y0lMA1ZHolLx19uZnRdILcXKlZGQx/GdEqSsMRU1BIrFqRcV1qQOOHyxOLXEGcbRtAEsuAC2V4K3p5mFJ22IDWaEkk9ttf5Izb2LkD1MnrSwztXmmD/Qi/EmVEFBfiKGmftsPwVaIoZanlKndMZsIBOskFYpDOq3QUs9aSbAAtL5Dbokus2G4/asthNMK5UQKCOhU97oaOYNGsTah+jfCKsZnTRn5TbhFX8ghg8CBYt/BjeYYYUrtUZ5jVij/op7V5SsbA4mYTOwZ46hqdpbB6Qvq3AS2HHNkC15pTDIcDNGsMPXaBidXYPHc6PJAkRh29Vx8KcgX46LoUQBhRM+3SW6Opll/wgxxsPgKJKzr5QCmwkUxNbeg6Wj34SUnEzOemSuvS2OetRCO8Tyy+QbSKVJcqkia+GvDefFwMOmgnD7h81TUtMn+mRpyJJ349HhAnoWFTejhpYTL9G8N2nVg1qkXBeoS9Nw2fB27t7trm7d/QK7Cr4uoCeOQ7/8JfKT77KiDzLImESHw/0wf73QeHu74hxv7uihi4fTX+XEwAyQG3264dwv17aJ5N335Vt9sdrAXhPOAv8JFvzqyYXwfx8WYJaef1gMl98JRFyl5Mv5Uo/oVH5ww5OzLFsiTPDns7fS6EURSSWd/92BxMYQ8sBaH+j+wthQPdVgDGpTfi+JQIWMD8xKqULliRH01rTeyF8x8q/GBEEEBrAJMPf25UQwi0b8tmqRXY7kIvNkzrkvRWLnxoGYEJsz8u4oOyMp8cHyaybb1HdMCaLApUE+/7xLIZGP6H9xuSEXp1zLIdjk5nBaMuV/yTDRRP8Y2ww5RO6d2D94o+6ucWIqUAvgHIHXhZsmDhjVLczmZ3ca0Cb3PpKwt2UtHVQ0BgFJsqqTsnzZPlKahRUkEu4qmkJt+kqdae76ViWe3STan69yaF9+fESD2lcQshLHWVu4ovItXxO69bqC5p1nZLvI8NdQB9s9UNaJGlQ5mG947ipdDA0eTIw/A1zEdjWquIsQXXGIVEH0thC5M+W9pZe7IhAVnPJkYCCXN5a32HjN6nsvokEqRS44tGIs7s2LVTvcrHAF+RVmI8L4HUYk4x+67AxSMJKqCg8zrGOgvK9kNMdDrNiUtSWuHFpC8/p5qIQrEo/H+1l/0cAwQ2nKmpWxKcMIuHY44Y6DlkpO48tRuUGBWT0FyHwSKO72Ud+tJUfdaZ4CWNijzZtlRa8+CkmO/EwHYfPZFU/hzjFWH7vnzHRMo+aF9u8qHSAiEkA2HjoNQPEwHsDKOt6hOoK3Ce/+/9boMWDa44I6FrQhdgS7OnNaSzwxWKZMcyHi6LN4WC6sSj0qm2PSOGBTvDs/GWJS6SwEN/ULwpb4LQo9fYjUfSXRwZkynUazlSpvX9e+G2zor8l+YaMxSEomDdLHGcD6YVQPegTaA74H8+V4WvJkFUrjMLGLlvSZQWvi8/QA7yzQ8GPno//5SJHRP/OqKObPCo81s/+6WgLqykYpGAgQZhVDEBPXWgU/WzFZjKUhSFInufPRiMAUULC6T11yL45ZrRoB4DzOyJShKXaAJIBS9wzLYIoCEcJKQW8GVCx4fihqJ6mshBUXSw3wWVj3grrHQlGNGhIDNNzsxQ3M+GWn6ASobIWC+LbYOC6UpahVO13Zs2zOzZC8z7FmA05JhUGyBsF4tsG0drcggIFzgg/kpf3+CnAXKiMgIE8Jk/Mhpkc8DUJEUzDSnWlQFme3d0sHZDrg7LavtsEX3cHwjCYA17pMTfx8Ajw9hHscN67hyo+RJQ4458RmPywXykkVcW688oVUrQhahpPRvTWPnuI0B+SkQu7dCyvLRyFYlC1LG1gRCIvn3rwQeINzZQC2KXq31FaR9UmVV2QeGVqBHjmE+VMd3b1fhCynD0pQNhCG6/WCDbKPyE7NRQzL3BzQAJ0g09aUzcQA6mUp9iZFK6Sbp/YbHjo++7/Wj8S4YNa+ZdqAw1hDrKWFXv9+zaXpf8ZTDSbiqsxnwN/CzK5tPkOr4tRh2kY3Bn9JtalbIOI4b3F7F1vPQMfoDcdxMS8CW9m/NCW/HILTUVWQIPiD0j1A6bo8vsv6P1hCESl2abrSJWDrq5sSzUpwoxaCU9FtJyYH4QFMxDBpkkBR6kn0LMPO+5EJ7Z6bCiRoPedRZ/P0SSdii7ZnPAtVwwHUidcdyspwncz5uq6vvm4IEDbJVLUFCn/LvIHfooUBTkFO130FC7CmmcrKdgDJcid9mvVzsDSibOoXtIf9k6ABle3PmIxejodc4aob0QKS432srrCMndbfD454q52V01G4q913mC5HOsTzWF4h2No1av1VbcUgWAqyoZl+11PoFYnNv2HwAODeNRkHj+8SF1fcvVBu6MrehHAZK1Gm69ICcTKizykHgGFx7QdowTVAsYEF2tVc0Z6wLryz2FI1sc5By2znJAAmINndoJiB4sfPdPrTC8RnkW7KRCwxC6YvXg5ahMlQuMpoCSXjOlBy0Kij+bsCYPbGp8BdCBiLmLSAkEQRaieWo1SYvZIKJGj9Ur/eWHjiB7SOVdqMAVmpBvfRiebsFjger7DC+8kRFGtNrTrnnGD2GAJb8rQCWkUPYHhwXsjNBSkE6lGWUj5QNhK0DMNM2l+kXRZ0KLZaGsFSIdQz/HXDxf3/TE30+DgBKWGWdxElyLccJfEpjsnszECNoDGZpdwdRgCixeg9L4EPhH+RptvRMVRaahu4cySjS3P5wxAUCPkmn+rhyASpmiTaiDeggaIxYBmtLZDDhiWIJaBgzfCsAGUF1Q1SFZYyXDt9skCaxJsxK2Ms65dmdp5WAZyxik/zbrTQk5KmgxCg/f45L0jywebOWUYFJQAJia7XzCV0x89rpp/f3AVWhSPyTanqmik2SkD8A3Ml4NhIGLAjBXtPShwKYfi2eXtrDuKLk4QlSyTw1ftXgwqA2jUuopDl+5tfUWZNwBpEPXghzbBggYCw/dhy0ntds2yeHCDKkF/YxQjNIL/F/37jLPHCKBO9ibwYCmuxImIo0ijV2Wbg3kSN2psoe8IsABv3RNFaF9uMyCtCYtqcD+qNOhwMlfARQUdJ2tUX+MNJqOwIciWalZsmEjt07tfa8ma4cji9sqz+Q9hWfmMoKEbIHPOQORbhQRHIsrTYlnVTNvcq1imqmmPDdVDkJgRcTgB8Sb6epCQVmFZe+jGDiNJQLWnfx+drTKYjm0G8yH0ZAGMWzEJhUEQ4Maimgf/bkvo8PLVBsZl152y5S8+HRDfZIMCbYZ1WDp4yrdchOJw8k6R+/2pHmydK4NIK2PHdFPHtoLmHxRDwLFb7eB+M4zNZcB9NrAgjVyzLM7xyYSY13ykWfIEEd2n5/iYp3ZdrCf7fL+en+sIJu2W7E30MrAgZBD1rAAbZHPgeAMtKCg3NpSpYQUDWJu9bT3V7tOKv+NRiJc8JAKqqgCA/PNRBR7ChpiEulyQApMK1AyqcWnpSOmYh6yLiWkGJ2mklCSPIqN7UypWj3dGi5MvsHQ87MrB4VFgypJaFriaHivwcHIpmyi5LhNqtem4q0n8awM19Qk8BOS0EsqGscuuydYsIGsbT5GHnERUiMpKJl4ON7qjB4fEqlGN/hCky89232UQCiaeWpDYCJINXjT6xl4Gc7DxRCtgV0i1ma4RgWLsNtnEBRQFqZggCLiuyEydmFd7WlogpkCw5G1x4ft2psm3KAREwVwr1Gzl6RT7FDAqpVal34ewVm3VH4qn5mjGj+bYL1NgfLNeXDwtmYSpwzbruDKpTjOdgiIHDVQSb5/zBgSMbHLkxWWgghIh9QTFSDILixVwg0Eg1puooBiHAt7DzwJ7m8i8/i+jHvKf0QDnnHVkVTIqMvIQImOrzCJwhSR7qYB5gSwL6aWL9hERHCZc4G2+JrpgHNB8eCCmcIWIQ6rSdyPCyftXkDlErUkHafHRlkOIjxGbAktz75bnh50dU7YHk+Mz7wwstg6RFZb+TZuSOx1qqP5C66c0mptQmzIC2dlpte7vZrauAMm/7RfBYkGtXWGiaWTtwvAQiq2oD4YixPLXE2khB2FRaNRDTk+9sZ6K74Ia9VntCpN4BhJGJMT4Z5c5FhSepRCRWmBXqx+whVZC4me4saDs2iNqXMuCl6iAZflH8fscC1sTsy4PHeC+XYuqMBMUun5YezKbRKmEPwuK+CLzijPEQgfhahQswBBLfg/GBgBiI4QwAqzJkkyYAWtjzSg2ILgMAgqxYfwERRo3zruBL9WOryUArSD8sQOcD7fvIODJxKFS615KFPsb68USBEPPj1orNzFY2xoTtNBVTyzBhPbhFH0PI5AtlJBl2aSgNPYzxYLw7XTDBDinmVoENwiGzmngrMo8OmnRP0Z0i0Zrln9DDFcnmOoBZjABaQIbPOJYZGqX+RCMlDDbElcjaROLDoualmUIQ88Kekk3iM4OQrADcxi3rJguS4MOIBIgKgXrjd1WkbCdqxJk/4efRIFsavZA7KvvJQqp3Iid5Z0NFc5aiMRzGN3vrpBzaMy4JYde3wr96PjN90AYOIbyp6T4zj8LoE66OGcX1Ef4Z3KoWLAUF4BTg7ug/AbkG5UNQXAMkQezujSHeir2uTThgd3gpyzDrbnEdDRH2W7U6PeRvBX1ZFMP5RM+Zu6UUZZD8hDPHldVWntTCNk7To8IeOW9yn2wx0gmurwqC60AOde4r3ETi5pVMSDK8wxhoGAoEX9NLWHIR33VbrbMveii2jAJlrxwytTHbWNu8Y4N8vCCyZjAX/pcsfwXbLze2+D+u33OGBoJyAAL3jn3RuEcdp5If8O+a4NKWvxOTyDltG0IWoHhwVGe7dKkCWFT++tm+haBCikRUUMrMhYKZJKYoVuv/bsJzO8DwfVIInQq3g3BYypiz8baogH3r3GwqCwFtZnz4xMjAVOYnyOi5HWbFA8n0qz1OjSpHWFzpQOpvkNETZBGpxN8ybhtqV/DMUxd9uFZmBfKXMCn/SqkWJyKPnT6lq+4zBZni6fYRByJn6OK+OgPBGRAJluwGSk4wxjOOzyce/PKODwRlsgrVkdcsEiYrqYdXo0Er2GXi2GQZd0tNJT6c9pK1EEJG1zgDJBoTVuCXGAU8BKTvCO/cEQ1Wjk3Zzuy90JX4m3O5IlxVFhYkSUwuQB2up7jhvkm+bddRQu5F9s0XftGEJ9JSuSk+ZachCbdU45fEqbugzTIUokwoAKvpUQF/CvLbWW5BNQFqFkJg2f30E/48StNe5QwBg8zz3YAJ82FZoXBxXSv4QDooDo79NixyglO9AembuBcx5Re3CwOKTHebOPhkmFC7wNaWtoBhFuV4AkEuJ0J+1pT0tLkvFVZaNzfhs/Kd3+A9YsImlO4XK4vpCo/elHQi/9gkFg07xxnuXLt21unCIpDV+bbRxb7FC6nWYTsMFF8+1LUg4JFjVt3vqbuhHmDKbgQ4e+RGizRiO8ky05LQGMdL2IKLSNar0kNG7lHJMaXr5mLdG3nykgj6vB/KVijd1ARWkFEf3yiUw1v/WaQivVUpIDdSNrrKbjO5NPnxz6qTTGgYg03HgPhDrCFyYZTi3XQw3HXCva39mpLNFtz8AiEhxAJHpWX13gCTAwgm9YTvMeiqetdNQv6IU0hH0G+ZManTqDLPjyrOse7WiiwOJCG+J0pZYULhN8NILulmYYvmVcV2MjAfA39sGKqGdjpiPo86fecg65UPyXDIAOyOkCx5NQsLeD4gGVjTVDwOHWkbbBW0GeNjDkcSOn2Nq4cEssP54t9D749A7M1AIOBl0Fi0sSO5v3P7LCBrM6ZwFY6kp2FX6AcbGUdybnfChHPyu6WlRZ2Fwv9YM0RMI7kISRgR8HpQSJJOyTfXj/6gQKuihPtiUtlCQVPohUgzfezTg8o1b3n9pNZeco1QucaoXe40Fa5JYhqdTspFmxGtW9h5ezLFZs3j/N46f+S2rjYNC2JySXrnSAFhvAkz9a5L3pza8eYKHNoPrvBRESpxYPJdKVUxBE39nJ1chrAFpy4MMkf0qKgYALctGg1DQI1kIymyeS2AJNT4X240d3IFQb/0jQbaHJ2YRK8A+ls6WMhWmpCXYG5jqapGs5/eOJErxi2/2KWVHiPellTgh/fNl/2KYPKb7DUcAg+mCOPQFCiU9Mq/WLcU1xxC8aLePFZZlE+PCLzf7ey46INWRw2kcXySR9FDgByXzfxiNKwDFbUSMMhALPFSedyjEVM5442GZ4hTrsAEvZxIieSHGSgkwFh/nFNdrrFD4tBH4Il7fW6ur4J8Xaz7RW9jgtuPEXQsYk7gcMs2neu3zJwTyUerHKSh1iTBkj2YJh1SSOZL5pLuQbFFAvyO4k1Hxg2h99MTC6cTUkbONQIAnEfGsGkNFWRbuRyyaEZInM5pij73EA9rPIUfU4XoqQpHT9THZkW+oKFLvpyvTBMM69tN1Ydwv1LIEhHsC+ueVG+w+kyCPsvV3erRikcscHjZCkccx6VrBkBRusTDDd8847GA7p2Ucy0y0HdSRN6YIBciYa4vuXcAZbQAuSEmzw+H/AuOx+aH+tBL88H57D0MsqyiZxhOEQkF/8DR1d2hSPMj/sNOa5rxcUnBgH8ictv2J+cb4BA4v3MCShdZ2vtK30vAwkobnEWh7rsSyhmos3WC93Gn9C4nnAd/PjMMtQfyDNZsOPd6XcAsnBE/mRHtHEyJMzJfZFLE9OvQa0i9kUmToJ0ZxknTgdl/XPV8xoh0K7wNHHsnBdvFH3sv52lU7UFteseLG/VanIvcwycVA7+BE1Ulyb20BvwUWZcMTKhaCcmY3ROpvonVMV4N7yBXTL7IDtHzQ4CCcqF66LjF3xUqgErKzolLyCG6Kb7irP/MVTCCwGRxfrPGpMMGvPLgJ881PHMNMIO09T5ig7AzZTX/5PLlwnJLDAPfuHynSGhV4tPqR3gJ4kg4c06c/F1AcjGytKm2Yb5jwMotF7vro4YDLWlnMIpmPg36NgAZsGA0W1spfLSue4xxat0Gdwd0lqDBOgIaMANykwwDKejt5YaNtJYIkrSgu0KjIg0pznY0SCd1qlC6R19g97UrWDoYJGlrvCE05J/5wkjpkre727p5PTRX5FGrSBIfJqhJE/IS876PaHFkx9pGTH3oaY3jJRvLX9Iy3Edoar7cFvJqyUlOhAEiOSAyYgVEGkzHdug+oRHIEOXAExMiTSKU9A6nmRC8mp8iYhwWdP2U/5EkFAdPrZw03YA3gSyNUtMZeh7dDCu8pF5x0VORCTgKp07ehy7NZqKTpIC4UJJ89lnboyAfy5OyXzXtuDRbtAFjZRSyGFTpFrXwkpjSLIQIG3N0Vj4BtzK3wdlkBJrO18MNsgseR4BysJilI0wI6ZahLhBFA0XBmV8d4LUzEcNVb0xbLjLTETYN8OEVqNxkt10W614dd1FlFFVTIgB7/BQQp1sWlNolpIu4ekxUTBV7NmxOFKEBmmN+nA7pvF78/RII5ZHA09OAiE/66MF6HQ+qVEJCHxwymukkNvzqHEh52dULPbVasfQMgTDyBZzx4007YiKdBuUauQOt27Gmy8ISclPmEUCIcuLbkb1mzQSqIa3iE0PJh7UMYQbkpe+hXjTJKdldyt2mVPwywoODGJtBV1lJTgMsuSQBlDMwhEKIfrvsxGQjHPCEfNfMAY2oxvyKcKPUbQySkKG6tj9AQyEW3Q5rpaDJ5Sns9ScLKeizPRbvWYAw4bXkrZdmB7CQopCH8NAmqbuciZChHN8lVGaDbCnmddnqO1PQ4ieMYfcSiBE5zzMz+JV/4eyzrzTEShvqSGzgWimkNxLvUj86iAwcZuIkqdB0VaIB7wncLRmzHkiUQpPBIXbDDLHBlq7vp9xwuC9AiNkIptAYlG7Biyuk8ILdynuUM1cHWJgeB+K3wBP/ineogxkvBNNQ4AkW0hvpBOQGFfeptF2YTR75MexYDUy7Q/9uocGsx41O4IZhViw/2FvAEuGO5g2kyXBUijAggWM08bRhXg5ijgMwDJy40QeY/cQpUDZiIzmvskQpO5G1zyGZA8WByjIQU4jRoFJt56behxtHUUE/om7Rj2psYXGmq3llVOCgGYKNMo4pzwntITtapDqjvQtqpjaJwjHmDzSVGLxMt12gEXAdLi/caHSM3FPRGRf7dB7YC+cD2ho6oL2zGDCkjlf/DFoQVl8GS/56wur3rdV6ggtzZW60MRB3g+U1W8o8cvqIpMkctiGVMzXUFI7FacFLrgtdz4mTEr4aRAaQ2AFQaNeG7GX0yOJgMRYFziXdJf24kg/gBQIZMG/YcPEllRTVNoDYR6oSJ8wQNLuihfw81UpiKPm714bZX1KYjcXJdfclCUOOpvTxr9AAJevTY4HK/G7F3mUc3GOAKqh60zM0v34v+ELyhJZqhkaMA8UMMOU90f8RKEJFj7EqepBVwsRiLbwMo1J2zrE2UYJnsgIAscDmjPjnzI8a719Wxp757wqmSJBjXowhc46QN4RwKIxqEE6E5218OeK7RfcpGjWG1jD7qND+/GTk6M56Ig4yMsU6LUW1EWE+fIYycVV1thldSlbP6ltdC01y3KUfkobkt2q01YYMmxpKRvh1Z48uNKzP/IoRIZ/F6buOymSnW8gICitpJjKWBscSb9JJKaWkvEkqinAJ2kowKoqkqZftRqfRQlLtKoqvTRDi2vg/RrPD/d3a09J8JhGZlEkOM6znTsoMCsuvTmywxTCDhw5dd0GJOHCMPbsj3QLkTE3MInsZsimDQ3HkvthT7U9VA4s6G07sID0FW4SHJmRGwCl+Mu4xf0ezqeXD2PtPDnwMPo86sbwDV+9PWcgFcARUVYm3hrFQrHcgMElFGbSM2A1zUYA3baWfheJp2AINmTJLuoyYD/OwA4a6V0ChBN97E8YtDBerUECv0u0TlxR5yhJCXvJxgyM73Bb6pyq0jTFJDZ4p1Am1SA6sh8nADd1hAcGBMfq4d/UfwnmBqe0Jun1n1LzrgKuZMAnxA3NtCN7Klf4BH+14B7ibBmgt0TGUafVzI4uKlpF7v8NmgNjg90D6QE3tbx8AjSAC+OA1YJvclyPKgT27QpIEgVYpbPYGBsnyCNrGz9XUsCHkW1QAHgL2STZk12QGqmvAB0NFteERkvBIH7INDsNW9KKaAYyDMdBEMzJiWaJHZALqDxQDWRntumSDPcplyFiI1oDpT8wbwe01AHhW6+vAUUBoGhY3CT2tgwehdPqU/4Q7ZLYvhRl/ogOvR9O2+wkkPKW5vCTjD2fHRYXONCoIl4Jh1bZY0ZE1O94mMGn/dFSWBWzQ/VYk+Gezi46RgiDv3EshoTmMSlioUK6MQEN8qeyK6FRninyX8ZPeUWjjbMJChn0n/yJvrq5bh5UcCAcBYSafTFg7p0jDgrXo2QWLb3WpSOET/Hh4oSadBTvyDo10IufLzxiMLAnbZ1vcUmj3w7BQuIXjEZXifwukVxrGa9j+DXfpi12m1RbzYLg9J2wFergEwOxFyD0/JstNK06ZN2XdZSGWxcJODpQHOq4iKqjqkJUmPu1VczL5xTGUfCgLEYyNBCCbMBFT/cUP6pE/mujnHsSDeWxMbhrNilS5MyYR0nJyzanWXBeVcEQrRIhQeJA6Xt4f2eQESNeLwmC10WJVHqwx8SSyrtAAjpGjidcj1E2FYN0LObUcFQhafUKTiGmHWRHGsFCB+HEXgrzJEB5bp0QiF8ZHh11nFX8AboTD0PS4O1LqF8XBks2MpjsQnwKHF6HgaKCVLJtcr0XjqFMRGfKv8tmmykhLRzu+vqQ02+KpJBjaLt9ye1Ab+BbEBhy4EVdIJDrL2naV0o4wU8YZ2Lq04FG1mWCKC+UwkXOoAjneU/xHplMQo2cXUlrVNqJYczgYlaOEczVCs/OCgkyvLmTmdaBJc1iBLuKwmr6qtRnhowngsDxhzKFAi02tf8bmET8BO27ovJKF1plJwm3b0JpMh38+xsrXXg7U74QUM8ZCIMOpXujHntKdaRtsgyEZl5MClMVMMMZkZLNxH9+b8fH6+b8Lev30A9TuEVj9CqAdmwAAHBPbfOBFEATAPZ2CS0OH1Pj/0Q7PFUcC8hDrxESWdfgFRm+7vvWbkEppHB4T/1ApWnlTIqQwjcPl0VgS1yHSmD0OdsCVST8CQVwuiew1Y+g3QGFjNMzwRB2DSsAk26cmA8lp2wIU4p93AUBiUHFGOxOajAqD7Gm6NezNDjYzwLOaSXRBYcWipTSONHjUDXCY4mMI8XoVCR/Rrs/JLKXgEx+qkmeDlFOD1/yTQNDClRuiUyKYCllfMiQiyFkmuTz2vLsBNyRW+xz+5FElFxWB28VjYIGZ0Yd+5wIjkcoMaggxswbT0pCmckRAErbRlIlcOGdBo4djTNO8FAgQ+lT6vPS60BwTRSUAM3ddkEAZiwtEyArrkiDRnS7LJ+2hwbzd2YDQagSgACpsovmjil5wfPuXq3GuH0CyE7FK3M4FgRaFoIkaodORrPx1+JpI9psyNYIFuJogZa0/1AhOWdlHQxdAgbwacsHqPZo8u/ngAH2GmaTdhYnBfSDbBfh8CHq6Bx5bttP2+RdM+MAaYaZ0Y/ADkbNCZuAyAVQa2OcXOeICmDn9Q/eFkDeFQg5MgHEDXq/tVjj+jtd26nhaaolWxs1ixSUgOBwrDhRIGOLyOVk2/Bc0UxvseQCO2pQ2i+Krfhu/WeBovNb5dJxQtJRUDv2mCwYVpNl2efQM9xQHnK0JwLYt/U0Wf+phiA4uw8G91slC832pmOTCAoZXohg1fewCZqLBhkOUBofBWpMPsqg7XEXgPfAlDo2U5WXjtFdS87PIqClCK5nW6adCeXPkUiTGx0emOIDQqw1yFYGHEVx20xKjJVYe0O8iLmnQr3FA9nSIQilUKtJ4ZAdcTm7+ExseJauyqo30hs+1qSW211A1SFAOUgDlCGq7eTIcMAeyZkV1SQJ4j/e1Smbq4HcjqgFbLAGLyKxlMDMgZavK5NAYH19Olz3la/QCTiVelFnU6O/GCvykqS/wZJDhKN9gBtSOp/1SP5VRgJcoVj+kmf2wBgv4gjrgARBWiURYx8xENV3bEVUAAWWD3dYDKAIWk5opaCFCMR5ZjJExiCAw7gYiSZ2rkyTce4eNMY3lfGn+8p6+vBckGlKEXnA6Eota69OxDO9oOsJoy28BXOR0UoXNRaJD5ceKdlWMJlOFzDdZNpc05tkMGQtqeNF2lttZqNco1VtwXgRstLSQ6tSPChgqtGV5h2DcDReIQadaNRR6AsAYKL5gSFsCJMgfsaZ7DpKh8mg8Wz8V7H+gDnLuMxaWEIUPevIbClgap4dqmVWSrPgVYCzAoZHIa5z2Ocx1D/GvDOEqMOKLrMefWIbSWHZ6jbgA8qVBhYNHpx0P+jAgN5TB3haSifDcApp6yymEi6Ij/GsEpDYUgcHATJUYDUAmC1SCkJ4cuZXSAP2DEpQsGUjQmKJfJOvlC2x/pChkOyLW7KEoMYc5FDC4v2FGqSoRWiLsbPCiyg1U5yiHZVm1XLkHMMZL11/yxyw0UnGig3MFdZklN5FI/qiT65T+jOXOdO7XbgWurOAZR6Cv9uu1cm5LjkXX4xi6mWn5r5NjBS0gTliHhMZI2WNqSiSphEtiCAwnafS11JhseDGHYQ5+bqWiAYiAv6Jsf79/VUs4cIl+n6+WOjcgB/2l5TreoAV2717JzZbQIR0W1cl/dEqCy5kJ3ZSIHuU0vBoHooEpiHeQWVkkkOqRX27eD1FWw4BfO9CJDdKoSogQi3hAAwsPRFrN5RbX7bqLdBJ9JYMohWrgJKHSjVl1sy2xAG0E3sNyO0oCbSGOxCNBRRXTXenYKuwAoDLfnDcQaCwehUOIDiHAu5m5hMpKeKM4sIo3vxACakIxKoH2YWF2QM84e6F5C5hJU4g8uxuFOlAYnqtwxmHyNEawLW/PhoawJDrGAP0JYWHgAVUByo/bGdiv2T2EMg8gsS14/rAdzlOYazFE7w4OzxeKiWdm3nSOnQRRKXSlVo8HEAbBfyJMKqoq+SCcTSx5NDtbFwNlh8VhjGGDu7JG5/TAGAvniQSSUog0pNzTim8Owc6QTuSKSTXlQqwV3eiEnklS3LeSXYPXGK2VgeZBqNcHG6tZHvA3vTINhV0ELuQdp3t1y9+ogD8Kk/W7QoRN1UWPqM4+xdygkFDPLoTaumKReKiLWoPHOfY54m3qPx4c+4pgY3MRKKbljG8w4wvz8pxk3AqKsy4GMAkAtmRjRMsCxbb4Q2Ds0Ia9ci8cMT6DmsJG00XaHCIS+o3F8YVVeikw13w+OEDaCYYhC0ZE54kA4jpjruBr5STWeqQG6M74HHL6TZ3lXrd99ZX++7LhNatQaZosuxEf5yRA15S9gPeHskBIq3Gcw81AGb9/O53DYi/5CsQ51EmEh8Rkg4vOciClpy4d04eYsfr6fyQkBmtD+P8sNh6e+XYHJXT/lkXxT4KXU5F2sGxYyzfniMMQkb9OjDN2C8tRRgTyL7GwozH14PrEUZc6oz05Emne3Ts5EG7WolDmU8OB1LDG3VrpQxp+pT0KYV5dGtknU64JhabdqcVQbGZiAxQAnvN1u70y1AnmvOSPgLI6uB4AuDGhmAu3ATkJSw7OtS/2ToPjqkaq62/7WFG8advGlRRqxB9diP07JrXowKR9tpRa+jGJ91zxNTT1h8I2PcSfoUPtd7NejVoH03EUcqSBuFZPkMZhegHyo2ZAITovmm3zAIdGFWxoNNORiMRShgwdYwFzkPw5PA4a5MIIQpmq+nsp3YMuXt/GkXxLx/P6+ZJS0lFyz4MunC3eWSGE8xlCQrKvhKUPXr0hjpAN9ZK4PfEDrPMfMbGNWcHDzjA7ngMxTPnT7GMHar+gMQQ3NwHCv4zH4BIMYvzsdiERi6gebRmerTsVwZJTRsL8dkZgxgRxmpbgRcud+YlCIRpPwHShlUSwuipZnx9QCsEWziVazdDeKSYU5CF7UVPAhLer3CgJOQXl/zh575R5rsrmRnKAzq4POFdgbYBuEviM4+LVC15ssLNFghbTtHWerS1hDt5s4qkLUha/qpZXhWh1C6lTQAqCNQnaDjS7UGFBC6wTu8yFnKJnExCnAs3Ok9yj5KpfZESQ4lTy5pTGTnkAUpxI+yjEldJfSo4y0QhG4i4IwkRFGcjWY8+EzgYYJUK7BXQksLxAww/YYWBMhJILB9e8ePEJ4OP7z+4/wOQDl64iOYDp26DaONPxpKtBxq/aTzRGarm3VkPYTLJKx6Z/Mw2YbBGseJhPMwhhNswrIkyvV2BYzrvZbxLpKwcWJhYmFtVZ+lPEq91FzVp1HlQY1bZVLqeNR9SAUn6n0E28k/UuGkNpP1DBI5ch/EehZfjUQ9aE41NhETExoPT2gGQz0IhWJbEOvTQ4wgcXCHHFBhewYUiFHuhRSAUVmEHeCRQHQkXGFwkAgyzREJCVN7TRnTon36Zw3tPhx4EALwNdwDv+J41YSP4B2CQqz0EFgARZ4ESgBHQgROwAVn9GTI+HYexTUevLUeta4/DqKrbMVS+Yqb8hUwYCrlgKtmAq1YCrFgKrd4qpXiqZcKn1oqdWipjYKpWwVPVYqW6xUpVipKqFR3QKjagVEtAqHpxUMTitsnFaJOKx2cVhswq35RVpyiq9lFVNIKnOQVMkgqtYxVNxiqQjFS7GKlSIVIsQqPIhUWwioigFQ++KkN8VHr49HDw9Ebo9EDo9DTo9Crg9BDg9/Wx7gWx7YWwlobYrOGxWPNisAaAHEyALpkAVDIAeWAArsABVXACYuAD5cAF6wAKFQAQqgAbVAAsoAAlQAUaYAfkwAvogBWQACOgAD9AAHSAAKT4GUdMiOvFngBTwCn2AZ7Dv6B6k/90B8+yRnkV144AIBoAMTQATGgAjNAA4YABgwABZgB/mQCwyAVlwCguASlwCEuAQFwB4uAMlwBYuAJlQAUVAAhUD2KgdpUDaJgaRMDFJgX5MC1JgWJEAokQCWRAHxEAWkQBMRADpEAMkQAYROAEecC484DRpwBDTnwNOdw05tjTmiNOYwtswhYFwLA7BYG4LA2BYGOLAwRYFuLAsxYFQJAohIEyJAMwkAwiQC0JAJgkAeiQBkJAFokAPCQA0JABwcD4Dgc4cDdDgaYcDIDgYgUC6CgWgUClCgUYUAVBQBOFAEYMALgwAgDA9QYAdIn8AZzeBB2L5EcWrenUT1KXienEsuJJ7x5U8XlTjc1NVzUyXFTGb1LlpUtWlTDIjqwE4LsagowoCi2gJLKAkpoBgJQNpAIhNqaEoneI6kiiqQ6Go/n6j0cS+a2gEU8gIHJ+BwfgZX4GL+Bd/gW34FZ+BS/gUH4FN6BTegTvoEv6BJegRnYEF2A79gOvYDl2BdEjCkqkGtwXp0LNToIskOTXzh/F062yJ7AAAAEDAWAAABWhJ+KPEIJgBFxMVP7w2QJBGHASQnOBKXKFIdUK4igKA9IEaYJg);src:url(data:application/vnd.ms-fontobject;base64,n04AAEFNAAACAAIABAAAAAAABQAAAAAAAAABAJABAAAEAExQAAAAAAAAAAIAAAAAAAAAAAEAAAAAAAAAJxJ/LAAAAAAAAAAAAAAAAAAAAAAAACgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzAAAADgBSAGUAZwB1AGwAYQByAAAAeABWAGUAcgBzAGkAbwBuACAAMQAuADAAMAA5ADsAUABTACAAMAAwADEALgAwADAAOQA7AGgAbwB0AGMAbwBuAHYAIAAxAC4AMAAuADcAMAA7AG0AYQBrAGUAbwB0AGYALgBsAGkAYgAyAC4ANQAuADUAOAAzADIAOQAAADgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzACAAUgBlAGcAdQBsAGEAcgAAAAAAQlNHUAAAAAAAAAAAAAAAAAAAAAADAKncAE0TAE0ZAEbuFM3pjM/SEdmjKHUbyow8ATBE40IvWA3vTu8LiABDQ+pexwUMcm1SMnNryctQSiI1K5ZnbOlXKmnVV5YvRe6RnNMFNCOs1KNVpn6yZhCJkRtVRNzEufeIq7HgSrcx4S8h/v4vnrrKc6oCNxmSk2uKlZQHBii6iKFoH0746ThvkO1kJHlxjrkxs+LWORaDQBEtiYJIR5IB9Bi1UyL4Rmr0BNigNkMzlKQmnofBHviqVzUxwdMb3NdCn69hy+pRYVKGVS/1tnsqv4LL7wCCPZZAZPT4aCShHjHJVNuXbmMrY5LeQaGnvAkXlVrJgKRAUdFjrWEah9XebPeQMj7KS7DIBAFt8ycgC5PLGUOHSE3ErGZCiViNLL5ZARfywnCoZaKQCu6NuFX42AEeKtKUGnr/Cm2Cy8tpFhBPMW5Fxi4Qm4TkDWh4IWFDClhU2hRWosUWqcKLlgyXB+lSHaWaHiWlBAR8SeSgSPCQxdVQgzUixWKSTrIQEbU94viDctkvX+VSjJuUmV8L4CXShI11esnp0pjWNZIyxKHS4wVQ2ime1P4RnhvGw0aDN1OLAXGERsB7buFpFGGBAre4QEQR0HOIO5oYH305G+KspT/FupEGGafCCwxSe6ZUa+073rXHnNdVXE6eWvibUS27XtRzkH838mYLMBmYysZTM0EM3A1fbpCBYFccN1B/EnCYu/TgCGmr7bMh8GfYL+BfcLvB0gRagC09w9elfldaIy/hNCBLRgBgtCC7jAF63wLSMAfbfAlEggYU0bUA7ACCJmTDpEmJtI78w4/BO7dN7JR7J7ZvbYaUbaILSQsRBiF3HGk5fEg6p9unwLvn98r+vnsV+372uf1xBLq4qU/45fTuqaAP+pssmCCCTF0mhEow8ZXZOS8D7Q85JsxZ+Azok7B7O/f6J8AzYBySZQB/QHYUSA+EeQhEWiS6AIQzgcsDiER4MjgMBAWDV4AgQ3g1eBgIdweCQmCjJEMkJ+PKRWyFHHmg1Wi/6xzUgA0LREoKJChwnQa9B+5RQZRB3IlBlkAnxyQNaANwHMowzlYSMCBgnbpzvqpl0iTJNCQidDI9ZrSYNIRBhHtUa5YHMHxyGEik9hDE0AKj72AbTCaxtHPUaKZdAZSnQTyjGqGLsmBStCejApUhg4uBMU6mATujEl+KdDPbI6Ag4vLr+hjY6lbjBeoLKnZl0UZgRX8gTySOeynZVz1wOq7e1hFGYIq+MhrGxDLak0PrwYzSXtcuyhXEhwOYofiW+EcI/jw8P6IY6ed+etAbuqKp5QIapT77LnAe505lMuqL79a0ut4rWexzFttsOsLDy7zvtQzcq3U1qabe7tB0wHWVXji+zDbo8x8HyIRUbXnwUcklFv51fvTymiV+MXLSmGH9d9+aXpD5X6lao41anWGig7IwIdnoBY2ht/pO9mClLo4NdXHAsefqWUKlXJkbqPOFhMoR4aiA1BXqhRNbB2Xwi+7u/jpAoOpKJ0UX24EsrzMfHXViakCNcKjBxuQX8BO0ZqjJ3xXzf+61t2VXOSgJ8xu65QKgtN6FibPmPYsXbJRHHqbgATcSZxBqGiDiU4NNNsYBsKD0MIP/OfKnlk/Lkaid/O2NbKeuQrwOB2Gq3YHyr6ALgzym5wIBnsdC1ZkoBFZSQXChZvlesPqvK2c5oHHT3Q65jYpNxnQcGF0EHbvYqoFw60WNlXIHQF2HQB7zD6lWjZ9rVqUKBXUT6hrkZOle0RFYII0V5ZYGl1JAP0Ud1fZZMvSomBzJ710j4Me8mjQDwEre5Uv2wQfk1ifDwb5ksuJQQ3xt423lbuQjvoIQByQrNDh1JxGFkOdlJvu/gFtuW0wR4cgd+ZKesSV7QkNE2kw6AV4hoIuC02LGmTomyf8PiO6CZzOTLTPQ+HW06H+tx+bQ8LmDYg1pTFrp2oJXgkZTyeRJZM0C8aE2LpFrNVDuhARsN543/FV6klQ6Tv1OoZGXLv0igKrl/CmJxRmX7JJbJ998VSIPQRyDBICzl4JJlYHbdql30NvYcOuZ7a10uWRrgoieOdgIm4rlq6vNOQBuqESLbXG5lzdJGHw2m0sDYmODXbYGTfSTGRKpssTO95fothJCjUGQgEL4yKoGAF/0SrpUDNn8CBgBcSDQByAeNkCXp4S4Ro2Xh4OeaGRgR66PVOsU8bc6TR5/xTcn4IVMLOkXSWiXxkZQCbvKfmoAvQaKjO3EDKwkwqHChCDEM5loQRPd5ACBki1TjF772oaQhQbQ5C0lcWXPFOzrfsDGUXGrpxasbG4iab6eByaQkQfm0VFlP0ZsDkvvqCL6QXMUwCjdMx1ZOyKhTJ7a1GWAdOUcJ8RSejxNVyGs31OKMyRyBVoZFjqIkmKlLQ5eHMeEL4MkUf23cQ/1SgRCJ1dk4UdBT7OoyuNgLs0oCd8RnrEIb6QdMxT2QjD4zMrJkfgx5aDMcA4orsTtKCqWb/Veyceqa5OGSmB28YwH4rFbkQaLoUN8OQQYnD3w2eXpI4ScQfbCUZiJ4yMOIKLyyTc7BQ4uXUw6Ee6/xM+4Y67ngNBknxIPwuppgIhFcwJyr6EIj+LzNj/mfR2vhhRlx0BILZoAYruF0caWQ7YxO66UmeguDREAFHYuC7HJviRgVO6ruJH59h/C/PkgSle8xNzZJULLWq9JMDTE2fjGE146a1Us6PZDGYle6ldWRqn/pdpgHKNGrGIdkRK+KPETT9nKT6kLyDI8xd9A1FgWmXWRAIHwZ37WyZHOVyCadJEmMVz0MadMjDrPho+EIochkVC2xgGiwwsQ6DMv2P7UXqT4x7CdcYGId2BJQQa85EQKmCmwcRejQ9Bm4oATENFPkxPXILHpMPUyWTI5rjNOsIlmEeMbcOCEqInpXACYQ9DDxmFo9vcmsDblcMtg4tqBerNngkIKaFJmrQAPnq1dEzsMXcwjcHdfdCibcAxxA+q/j9m3LM/O7WJka4tSidVCjsvo2lQ/2ewyoYyXwAYyr2PlRoR5MpgVmSUIrM3PQxXPbgjBOaDQFIyFMJvx3Pc5RSYj12ySVF9fwFPQu2e2KWVoL9q3Ayv3IzpGHUdvdPdrNUdicjsTQ2ISy7QU3DrEytIjvbzJnAkmANXjAFERA0MUoPF3/5KFmW14bBNOhwircYgMqoDpUMcDtCmBE82QM2YtdjVLB4kBuKho/bcwQdeboqfQartuU3CsCf+cXkgYAqp/0Ee3RorAZt0AvvOCSI4JICIlGlsV0bsSid/NIEALAAzb6HAgyWHBps6xAOwkJIGcB82CxRQq4sJf3FzA70A+TRqcqjEMETCoez3mkPcpnoALs0ugJY8kQwrC+JE5ik3w9rzrvDRjAQnqgEVvdGrNwlanR0SOKWzxOJOvLJhcd8Cl4AshACUkv9czdMkJCVQSQhp6kp7StAlpVRpK0t0SW6LHeBJnE2QchB5Ccu8kxRghZXGIgZIiSj7gEKMJDClcnX6hgoqJMwiQDigIXg3ioFLCgDgjPtYHYpsF5EiA4kcnN18MZtOrY866dEQAb0FB34OGKHGZQjwW/WDHA60cYFaI/PjpzquUqdaYGcIq+mLez3WLFFCtNBN2QJcrlcoELgiPku5R5dSlJFaCEqEZle1AQzAKC+1SotMcBNyQUFuRHRF6OlimSBgjZeTBCwLyc6A+P/oFRchXTz5ADknYJHxzrJ5pGuIKRQISU6WyKTBBjD8WozmVYWIsto1AS5rxzKlvJu4E/vwOiKxRtCWsDM+eTHUrmwrCK5BIfMzGkD+0Fk5LzBs0jMYXktNDblB06LMNJ09U8pzSLmo14MS0OMjcdrZ31pyQqxJJpRImlSvfYAK8inkYU52QY2FPEVsjoWewpwhRp5yAuNpkqhdb7ku9Seefl2D0B8SMTFD90xi4CSOwwZy9IKkpMtI3FmFUg3/kFutpQGNc3pCR7gvC4sgwbupDu3DyEN+W6YGLNM21jpB49irxy9BSlHrVDlnihGKHwPrbVFtc+h1rVQKZduxIyojccZIIcOCmhEnC7UkY68WXKQgLi2JCDQkQWJRQuk60hZp0D3rtCTINSeY9Ej2kIKYfGxwOs4j9qMM7fYZiipzgcf7TamnehqdhsiMiCawXnz4xAbyCkLAx5EGbo3Ax1u3dUIKnTxIaxwQTHehPl3V491H0+bC5zgpGz7Io+mjdhKlPJ01EeMpM7UsRJMi1nGjmJg35i6bQBAAxjO/ENJubU2mg3ONySEoWklCwdABETcs7ck3jgiuU9pcKKpbgn+3YlzV1FzIkB6pmEDOSSyDfPPlQskznctFji0kpgZjW5RZe6x9kYT4KJcXg0bNiCyif+pZACCyRMmYsfiKmN9tSO65F0R2OO6ytlEhY5Sj6uRKfFxw0ijJaAx/k3QgnAFSq27/2i4GEBA+UvTJKK/9eISNvG46Em5RZfjTYLdeD8kdXHyrwId/DQZUaMCY4gGbke2C8vfjgV/Y9kkRQOJIn/xM9INZSpiBnqX0Q9GlQPpPKAyO5y+W5NMPSRdBCUlmuxl40ZfMCnf2Cp044uI9WLFtCi4YVxKjuRCOBWIb4XbIsGdbo4qtMQnNOQz4XDSui7W/N6l54qOynCqD3DpWQ+mpD7C40D8BZEWGJX3tlAaZBMj1yjvDYKwCJBa201u6nBKE5UE+7QSEhCwrXfbRZylAaAkplhBWX50dumrElePyNMRYUrC99UmcSSNgImhFhDI4BXjMtiqkgizUGCrZ8iwFxU6fQ8GEHCFdLewwxYWxgScAYMdMLmcZR6b7rZl95eQVDGVoUKcRMM1ixXQtXNkBETZkVVPg8LoSrdetHzkuM7DjZRHP02tCxA1fmkXKF3VzfN1pc1cv/8lbTIkkYpqKM9VOhp65ktYk+Q46myFWBapDfyWUCnsnI00QTBQmuFjMZTcd0V2NQ768Fhpby04k2IzNR1wKabuGJqYWwSly6ocMFGTeeI+ejsWDYgEvr66QgqdcIbFYDNgsm0x9UHY6SCd5+7tpsLpKdvhahIDyYmEJQCqMqtCF6UlrE5GXRmbu+vtm3BFSxI6ND6UxIE7GsGMgWqghXxSnaRJuGFveTcK5ZVSPJyjUxe1dKgI6kNF7EZhIZs8y8FVqwEfbM0Xk2ltORVDKZZM40SD3qQoQe0orJEKwPfZwm3YPqwixhUMOndis6MhbmfvLBKjC8sKKIZKbJk8L11oNkCQzCgvjhyyEiQSuJcgCQSG4Mocfgc0Hkwcjal1UNgP0CBPikYqBIk9tONv4kLtBswH07vUCjEaHiFGlLf8MgXKzSgjp2HolRRccAOh0ILHz9qlGgIFkwAnzHJRjWFhlA7ROwINyB5HFj59PRZHFor6voq7l23EPNRwdWhgawqbivLSjRA4htEYUFkjESu67icTg5S0aW1sOkCiIysfJ9UnIWevOOLGpepcBxy1wEhd2WI3AZg7sr9WBmHWyasxMcvY/iOmsLtHSWNUWEGk9hScMPShasUA1AcHOtRZlqMeQ0OzYS9vQvYUjOLrzP07BUAFikcJNMi7gIxEw4pL1G54TcmmmoAQ5s7TGWErJZ2Io4yQ0ljRYhL8H5e62oDtLF8aDpnIvZ5R3GWJyAugdiiJW9hQAVTsnCBHhwu7rkBlBX6r3b7ejEY0k5GGeyKv66v+6dg7mcJTrWHbtMywbedYqCQ0FPwoytmSWsL8WTtChZCKKzEF7vP6De4x2BJkkniMgSdWhbeBSLtJZR9CTHetK1xb34AYIJ37OegYIoPVbXgJ/qDQK+bfCtxQRVKQu77WzOoM6SGL7MaZwCGJVk46aImai9fmam+WpHG+0BtQPWUgZ7RIAlPq6lkECUhZQ2gqWkMYKcYMYaIc4gYCDFHYa2d1nzp3+J1eCBay8IYZ0wQRKGAqvCuZ/UgbQPyllosq+XtfKIZOzmeJqRazpmmoP/76YfkjzV2NlXTDSBYB04SVlNQsFTbGPk1t/I4Jktu0XSgifO2ozFOiwd/0SssJDn0dn4xqk4GDTTKX73/wQyBLdqgJ+Wx6AQaba3BA9CKEzjtQYIfAsiYamapq80LAamYjinlKXUkxdpIDk0puXUEYzSalfRibAeDAKpNiqQ0FTwoxuGYzRnisyTotdVTclis1LHRQCy/qqL8oUaQzWRxilq5Mi0IJGtMY02cGLD69vGjkj3p6pGePKI8bkBv5evq8SjjyU04vJR2cQXQwSJyoinDsUJHCQ50jrFTT7yRdbdYQMB3MYCb6uBzJ9ewhXYPAIZSXfeEQBZZ3GPN3Nbhh/wkvAJLXnQMdi5NYYZ5GHE400GS5rXkOZSQsdZgIbzRnF9ueLnsfQ47wHAsirITnTlkCcuWWIUhJSbpM3wWhXNHvt2xUsKKMpdBSbJnBMcihkoDqAd1Zml/R4yrzow1Q2A5G+kzo/RhRxQS2lCSDRV8LlYLBOOoo1bF4jwJAwKMK1tWLHlu9i0j4Ig8qVm6wE1DxXwAwQwsaBWUg2pOOol2dHxyt6npwJEdLDDVYyRc2D0HbcbLUJQj8gPevQBUBOUHXPrsAPBERICpnYESeu2OHotpXQxRGlCCtLdIsu23MhZVEoJg8Qumj/UMMc34IBqTKLDTp76WzL/dMjCxK7MjhiGjeYAC/kj/jY/Rde7hpSM1xChrog6yZ7OWTuD56xBJnGFE+pT2ElSyCnJcwVzCjkqeNLfMEJqKW0G7OFIp0G+9mh50I9o8k1tpCY0xYqFNIALgIfc2me4n1bmJnRZ89oepgLPT0NTMLNZsvSCZAc3TXaNB07vail36/dBySis4m9/DR8izaLJW6bWCkVgm5T+ius3ZXq4xI+GnbveLbdRwF2mNtsrE0JjYc1AXknCOrLSu7Te/r4dPYMCl5qtiHNTn+TPbh1jCBHH+dMJNhwNgs3nT+OhQoQ0vYif56BMG6WowAcHR3DjQolxLzyVekHj00PBAaW7IIAF1EF+uRIWyXjQMAs2chdpaKPNaB+kSezYt0+CA04sOg5vx8Fr7Ofa9sUv87h7SLAUFSzbetCCZ9pmyLt6l6/TzoA1/ZBG9bIUVHLAbi/kdBFgYGyGwRQGBpkqCEg2ah9UD6EedEcEL3j4y0BQQCiExEnocA3SZboh+epgd3YsOkHskZwPuQ5OoyA0fTA5AXrHcUOQF+zkJHIA7PwCDk1gGVmGUZSSoPhNf+Tklauz98QofOlCIQ/tCD4dosHYPqtPCXB3agggQQIqQJsSkB+qn0rkQ1toJjON/OtCIB9RYv3PqRA4C4U68ZMlZn6BdgEvi2ziU+TQ6NIw3ej+AtDwMGEZk7e2IjxUWKdAxyaw9OCwSmeADTPPleyk6UhGDNXQb++W6Uk4q6F7/rg6WVTo82IoCxSIsFDrav4EPHphD3u4hR53WKVvYZUwNCCeM4PMBWzK+EfIthZOkuAwPo5C5jgoZgn6dUdvx5rIDmd58cXXdKNfw3l+wM2UjgrDJeQHhbD7HW2QDoZMCujgIUkk5Fg8VCsdyjOtnGRx8wgKRPZN5dR0zPUyfGZFVihbFRniXZFOZGKPnEQzU3AnD1KfR6weHW2XS6KbPJxUkOTZsAB9vTVp3Le1F8q5l+DMcLiIq78jxAImD2pGFw0VHfRatScGlK6SMu8leTmhUSMy8Uhdd6xBiH3Gdman4tjQGLboJfqz6fL2WKHTmrfsKZRYX6BTDjDldKMosaSTLdQS7oDisJNqAUhw1PfTlnacCO8vl8706Km1FROgLDmudzxg+EWTiArtHgLsRrAXYWdB0NmToNCJdKm0KWycZQqb+Mw76Qy29iQ5up/X7oyw8QZ75kP5F6iJAJz6KCmqxz8fEa/xnsMYcIO/vEkGRuMckhr4rIeLrKaXnmIzlNLxbFspOphkcnJdnz/Chp/Vlpj2P7jJQmQRwGnltkTV5dbF9fE3/fxoSqTROgq9wFUlbuYzYcasE0ouzBo+dDCDzxKAfhbAZYxQiHrLzV2iVexnDX/QnT1fsT/xuhu1ui5qIytgbGmRoQkeQooO8eJNNZsf0iALur8QxZFH0nCMnjerYQqG1pIfjyVZWxhVRznmmfLG00BcBWJE6hzQWRyFknuJnXuk8A5FRDCulwrWASSNoBtR+CtGdkPwYN2o7DOw/VGlCZPusRBFXODQdUM5zeHDIVuAJBLqbO/f9Qua+pDqEPk230Sob9lEZ8BHiCorjVghuI0lI4JDgHGRDD/prQ84B1pVGkIpVUAHCG+iz3Bn3qm2AVrYcYWhock4jso5+J7HfHVj4WMIQdGctq3psBCVVzupQOEioBGA2Bk+UILT7+VoX5mdxxA5fS42gISQVi/HTzrgMxu0fY6hE1ocUwwbsbWcezrY2n6S8/6cxXkOH4prpmPuFoikTzY7T85C4T2XYlbxLglSv2uLCgFv8Quk/wdesUdWPeHYIH0R729JIisN9Apdd4eB10aqwXrPt+Su9mA8k8n1sjMwnfsfF2j3jMUzXepSHmZ/BfqXvzgUNQQWOXO8YEuFBh4QTYCkOAPxywpYu1VxiDyJmKVcmJPGWk/gc3Pov02StyYDahwmzw3E1gYC9wkupyWfDqDSUMpCTH5e5N8B//lHiMuIkTNw4USHrJU67bjXGqNav6PBuQSoqTxc8avHoGmvqNtXzIaoyMIQIiiUHIM64cXieouplhNYln7qgc4wBVAYR104kO+CvKqsg4yIUlFNThVUAKZxZt1XA34h3TCUUiXVkZ0w8Hh2R0Z5L0b4LZvPd/p1gi/07h8qfwHrByuSxglc9cI4QIg2oqvC/qm0i7tjPLTgDhoWTAKDO2ONW5oe+/eKB9vZB8K6C25yCZ9RFVMnb6NRdRjyVK57CHHSkJBfnM2/j4ODUwRkqrtBBCrDsDpt8jhZdXoy/1BCqw3sSGhgGGy0a5Jw6BP/TExoCmNFYjZl248A0osgPyGEmRA+fAsqPVaNAfytu0vuQJ7rk3J4kTDTR2AlCHJ5cls26opZM4w3jMULh2YXKpcqGBtuleAlOZnaZGbD6DHzMd6i2oFeJ8z9XYmalg1Szd/ocZDc1C7Y6vcALJz2lYnTXiWEr2wawtoR4g3jvWUU2Ngjd1cewtFzEvM1NiHZPeLlIXFbBPawxNgMwwAlyNSuGF3zizVeOoC9bag1qRAQKQE/EZBWC2J8mnXAN2aTBboZ7HewnObE8CwROudZHmUM5oZ/Ugd/JZQK8lvAm43uDRAbyW8gZ+ZGq0EVerVGUKUSm/Idn8AQHdR4m7bue88WBwft9mSCeMOt1ncBwziOmJYI2ZR7ewNMPiCugmSsE4EyQ+QATJG6qORMGd4snEzc6B4shPIo4G1T7PgSm8PY5eUkPdF8JZ0VBtadbHXoJgnEhZQaODPj2gpODKJY5Yp4DOsLBFxWbvXN755KWylJm+oOd4zEL9Hpubuy2gyyfxh8oEfFutnYWdfB8PdESLWYvSqbElP9qo3u6KTmkhoacDauMNNjj0oy40DFV7Ql0aZj77xfGl7TJNHnIwgqOkenruYYNo6h724+zUQ7+vkCpZB+pGA562hYQiDxHVWOq0oDQl/QsoiY+cuI7iWq/ZIBtHcXJ7kks+h2fCNUPA82BzjnqktNts+RLdk1VSu+tqEn7QZCCsvEqk6FkfiOYkrsw092J8jsfIuEKypNjLxrKA9kiA19mxBD2suxQKCzwXGws7kEJvlhUiV9tArLIdZW0IORcxEzdzKmjtFhsjKy/44XYXdI5noQoRcvjZ1RMPACRqYg2V1+OwOepcOknRLLFdYgTkT5UApt/JhLM3jeFYprZV+Zow2g8fP+U68hkKFWJj2yBbKqsrp25xkZX1DAjUw52IMYWaOhab8Kp05VrdNftqwRrymWF4OQSjbdfzmRZirK8FMJELEgER2PHjEAN9pGfLhCUiTJFbd5LBkOBMaxLr/A1SY9dXFz4RjzoU9ExfJCmx/I9FKEGT3n2cmzl2X42L3Jh+AbQq6sA+Ss1kitoa4TAYgKHaoybHUDJ51oETdeI/9ThSmjWGkyLi5QAGWhL0BG1UsTyRGRJOldKBrYJeB8ljLJHfATWTEQBXBDnQexOHTB+Un44zExFE4vLytcu5NwpWrUxO/0ZICUGM7hGABXym0V6ZvDST0E370St9MIWQOTWngeoQHUTdCJUP04spMBMS8LSker9cReVQkULFDIZDFPrhTzBl6sed9wcZQTbL+BDqMyaN3RJPh/anbx+Iv+qgQdAa3M9Z5JmvYlh4qop+Ho1F1W5gbOE9YKLgAnWytXElU4G8GtW47lhgFE6gaSs+gs37sFvi0PPVvA5dnCBgILTwoKd/+DoL9F6inlM7H4rOTzD79KJgKlZO/Zgt22UsKhrAaXU5ZcLrAglTVKJEmNJvORGN1vqrcfSMizfpsgbIe9zno+gBoKVXgIL/VI8dB1O5o/R3Suez/gD7M781ShjKpIIORM/nxG+jjhhgPwsn2IoXsPGPqYHXA63zJ07M2GPEykQwJBYLK808qYxuIew4frk52nhCsnCYmXiR6CuapvE1IwRB4/QftDbEn+AucIr1oxrLabRj9q4ae0+fXkHnteAJwXRbVkR0mctVSwEbqhJiMSZUp9DNbEDMmjX22m3ABpkrPQQTP3S1sib5pD2VRKRd+eNAjLYyT0hGrdjWJZy24OYXRoWQAIhGBZRxuBFMjjZQhpgrWo8SiFYbojcHO8V5DyscJpLTHyx9Fimassyo5U6WNtquUMYgccaHY5amgR3PQzq3ToNM5ABnoB9kuxsebqmYZm0R9qxJbFXCQ1UPyFIbxoUraTJFDpCk0Wk9GaYJKz/6oHwEP0Q14lMtlddQsOAU9zlYdMVHiT7RQP3XCmWYDcHCGbVRHGnHuwzScA0BaSBOGkz3lM8CArjrBsyEoV6Ys4qgDK3ykQQPZ3hCRGNXQTNNXbEb6tDiTDLKOyMzRhCFT+mAUmiYbV3YQVqFVp9dorv+TsLeCykS2b5yyu8AV7IS9cxcL8z4Kfwp+xJyYLv1OsxQCZwTB4a8BZ/5EdxTBJthApqyfd9u3ifr/WILTqq5VqgwMT9SOxbSGWLQJUUWCVi4k9tho9nEsbUh7U6NUsLmkYFXOhZ0kmamaJLRNJzSj/qn4Mso6zb6iLLBXoaZ6AqeWCjHQm2lztnejYYM2eubnpBdKVLORZhudH3JF1waBJKA9+W8EhMj3Kzf0L4vi4k6RoHh3Z5YgmSZmk6ns4fjScjAoL8GoOECgqgYEBYUGFVO4FUv4/YtowhEmTs0vrvlD/CrisnoBNDAcUi/teY7OctFlmARQzjOItrrlKuPO6E2Ox93L4O/4DcgV/dZ7qR3VBwVQxP1GCieA4RIpweYJ5FoYrHxqRBdJjnqbsikA2Ictbb8vE1GYIo9dacK0REgDX4smy6GAkxlH1yCGGsk+tgiDhNKuKu3yNrMdxafmKTF632F8Vx4BNK57GvlFisrkjN9WDAtjsWA0ENT2e2nETUb/n7qwhvGnrHuf5bX6Vh/n3xffU3PeHdR+FA92i6ufT3AlyAREoNDh6chiMWTvjKjHDeRhOa9YkOQRq1vQXEMppAQVwHCuIcV2g5rBn6GmZZpTR7vnSD6ZmhdSl176gqKTXu5E+YbfL0adwNtHP7dT7t7b46DVZIkzaRJOM+S6KcrzYVg+T3wSRFRQashjfU18NutrKa/7PXbtuJvpIjbgPeqd+pjmRw6YKpnANFSQcpzTZgpSNJ6J7uiagAbir/8tNXJ/OsOnRh6iuIexxrmkIneAgz8QoLmiaJ8sLQrELVK2yn3wOHp57BAZJhDZjTBzyoRAuuZ4eoxHruY1pSb7qq79cIeAdOwin4GdgMeIMHeG+FZWYaiUQQyC5b50zKjYw97dFjAeY2I4Bnl105Iku1y0lMA1ZHolLx19uZnRdILcXKlZGQx/GdEqSsMRU1BIrFqRcV1qQOOHyxOLXEGcbRtAEsuAC2V4K3p5mFJ22IDWaEkk9ttf5Izb2LkD1MnrSwztXmmD/Qi/EmVEFBfiKGmftsPwVaIoZanlKndMZsIBOskFYpDOq3QUs9aSbAAtL5Dbokus2G4/asthNMK5UQKCOhU97oaOYNGsTah+jfCKsZnTRn5TbhFX8ghg8CBYt/BjeYYYUrtUZ5jVij/op7V5SsbA4mYTOwZ46hqdpbB6Qvq3AS2HHNkC15pTDIcDNGsMPXaBidXYPHc6PJAkRh29Vx8KcgX46LoUQBhRM+3SW6Opll/wgxxsPgKJKzr5QCmwkUxNbeg6Wj34SUnEzOemSuvS2OetRCO8Tyy+QbSKVJcqkia+GvDefFwMOmgnD7h81TUtMn+mRpyJJ349HhAnoWFTejhpYTL9G8N2nVg1qkXBeoS9Nw2fB27t7trm7d/QK7Cr4uoCeOQ7/8JfKT77KiDzLImESHw/0wf73QeHu74hxv7uihi4fTX+XEwAyQG3264dwv17aJ5N335Vt9sdrAXhPOAv8JFvzqyYXwfx8WYJaef1gMl98JRFyl5Mv5Uo/oVH5ww5OzLFsiTPDns7fS6EURSSWd/92BxMYQ8sBaH+j+wthQPdVgDGpTfi+JQIWMD8xKqULliRH01rTeyF8x8q/GBEEEBrAJMPf25UQwi0b8tmqRXY7kIvNkzrkvRWLnxoGYEJsz8u4oOyMp8cHyaybb1HdMCaLApUE+/7xLIZGP6H9xuSEXp1zLIdjk5nBaMuV/yTDRRP8Y2ww5RO6d2D94o+6ucWIqUAvgHIHXhZsmDhjVLczmZ3ca0Cb3PpKwt2UtHVQ0BgFJsqqTsnzZPlKahRUkEu4qmkJt+kqdae76ViWe3STan69yaF9+fESD2lcQshLHWVu4ovItXxO69bqC5p1nZLvI8NdQB9s9UNaJGlQ5mG947ipdDA0eTIw/A1zEdjWquIsQXXGIVEH0thC5M+W9pZe7IhAVnPJkYCCXN5a32HjN6nsvokEqRS44tGIs7s2LVTvcrHAF+RVmI8L4HUYk4x+67AxSMJKqCg8zrGOgvK9kNMdDrNiUtSWuHFpC8/p5qIQrEo/H+1l/0cAwQ2nKmpWxKcMIuHY44Y6DlkpO48tRuUGBWT0FyHwSKO72Ud+tJUfdaZ4CWNijzZtlRa8+CkmO/EwHYfPZFU/hzjFWH7vnzHRMo+aF9u8qHSAiEkA2HjoNQPEwHsDKOt6hOoK3Ce/+/9boMWDa44I6FrQhdgS7OnNaSzwxWKZMcyHi6LN4WC6sSj0qm2PSOGBTvDs/GWJS6SwEN/ULwpb4LQo9fYjUfSXRwZkynUazlSpvX9e+G2zor8l+YaMxSEomDdLHGcD6YVQPegTaA74H8+V4WvJkFUrjMLGLlvSZQWvi8/QA7yzQ8GPno//5SJHRP/OqKObPCo81s/+6WgLqykYpGAgQZhVDEBPXWgU/WzFZjKUhSFInufPRiMAUULC6T11yL45ZrRoB4DzOyJShKXaAJIBS9wzLYIoCEcJKQW8GVCx4fihqJ6mshBUXSw3wWVj3grrHQlGNGhIDNNzsxQ3M+GWn6ASobIWC+LbYOC6UpahVO13Zs2zOzZC8z7FmA05JhUGyBsF4tsG0drcggIFzgg/kpf3+CnAXKiMgIE8Jk/Mhpkc8DUJEUzDSnWlQFme3d0sHZDrg7LavtsEX3cHwjCYA17pMTfx8Ajw9hHscN67hyo+RJQ4458RmPywXykkVcW688oVUrQhahpPRvTWPnuI0B+SkQu7dCyvLRyFYlC1LG1gRCIvn3rwQeINzZQC2KXq31FaR9UmVV2QeGVqBHjmE+VMd3b1fhCynD0pQNhCG6/WCDbKPyE7NRQzL3BzQAJ0g09aUzcQA6mUp9iZFK6Sbp/YbHjo++7/Wj8S4YNa+ZdqAw1hDrKWFXv9+zaXpf8ZTDSbiqsxnwN/CzK5tPkOr4tRh2kY3Bn9JtalbIOI4b3F7F1vPQMfoDcdxMS8CW9m/NCW/HILTUVWQIPiD0j1A6bo8vsv6P1hCESl2abrSJWDrq5sSzUpwoxaCU9FtJyYH4QFMxDBpkkBR6kn0LMPO+5EJ7Z6bCiRoPedRZ/P0SSdii7ZnPAtVwwHUidcdyspwncz5uq6vvm4IEDbJVLUFCn/LvIHfooUBTkFO130FC7CmmcrKdgDJcid9mvVzsDSibOoXtIf9k6ABle3PmIxejodc4aob0QKS432srrCMndbfD454q52V01G4q913mC5HOsTzWF4h2No1av1VbcUgWAqyoZl+11PoFYnNv2HwAODeNRkHj+8SF1fcvVBu6MrehHAZK1Gm69ICcTKizykHgGFx7QdowTVAsYEF2tVc0Z6wLryz2FI1sc5By2znJAAmINndoJiB4sfPdPrTC8RnkW7KRCwxC6YvXg5ahMlQuMpoCSXjOlBy0Kij+bsCYPbGp8BdCBiLmLSAkEQRaieWo1SYvZIKJGj9Ur/eWHjiB7SOVdqMAVmpBvfRiebsFjger7DC+8kRFGtNrTrnnGD2GAJb8rQCWkUPYHhwXsjNBSkE6lGWUj5QNhK0DMNM2l+kXRZ0KLZaGsFSIdQz/HXDxf3/TE30+DgBKWGWdxElyLccJfEpjsnszECNoDGZpdwdRgCixeg9L4EPhH+RptvRMVRaahu4cySjS3P5wxAUCPkmn+rhyASpmiTaiDeggaIxYBmtLZDDhiWIJaBgzfCsAGUF1Q1SFZYyXDt9skCaxJsxK2Ms65dmdp5WAZyxik/zbrTQk5KmgxCg/f45L0jywebOWUYFJQAJia7XzCV0x89rpp/f3AVWhSPyTanqmik2SkD8A3Ml4NhIGLAjBXtPShwKYfi2eXtrDuKLk4QlSyTw1ftXgwqA2jUuopDl+5tfUWZNwBpEPXghzbBggYCw/dhy0ntds2yeHCDKkF/YxQjNIL/F/37jLPHCKBO9ibwYCmuxImIo0ijV2Wbg3kSN2psoe8IsABv3RNFaF9uMyCtCYtqcD+qNOhwMlfARQUdJ2tUX+MNJqOwIciWalZsmEjt07tfa8ma4cji9sqz+Q9hWfmMoKEbIHPOQORbhQRHIsrTYlnVTNvcq1imqmmPDdVDkJgRcTgB8Sb6epCQVmFZe+jGDiNJQLWnfx+drTKYjm0G8yH0ZAGMWzEJhUEQ4Maimgf/bkvo8PLVBsZl152y5S8+HRDfZIMCbYZ1WDp4yrdchOJw8k6R+/2pHmydK4NIK2PHdFPHtoLmHxRDwLFb7eB+M4zNZcB9NrAgjVyzLM7xyYSY13ykWfIEEd2n5/iYp3ZdrCf7fL+en+sIJu2W7E30MrAgZBD1rAAbZHPgeAMtKCg3NpSpYQUDWJu9bT3V7tOKv+NRiJc8JAKqqgCA/PNRBR7ChpiEulyQApMK1AyqcWnpSOmYh6yLiWkGJ2mklCSPIqN7UypWj3dGi5MvsHQ87MrB4VFgypJaFriaHivwcHIpmyi5LhNqtem4q0n8awM19Qk8BOS0EsqGscuuydYsIGsbT5GHnERUiMpKJl4ON7qjB4fEqlGN/hCky89232UQCiaeWpDYCJINXjT6xl4Gc7DxRCtgV0i1ma4RgWLsNtnEBRQFqZggCLiuyEydmFd7WlogpkCw5G1x4ft2psm3KAREwVwr1Gzl6RT7FDAqpVal34ewVm3VH4qn5mjGj+bYL1NgfLNeXDwtmYSpwzbruDKpTjOdgiIHDVQSb5/zBgSMbHLkxWWgghIh9QTFSDILixVwg0Eg1puooBiHAt7DzwJ7m8i8/i+jHvKf0QDnnHVkVTIqMvIQImOrzCJwhSR7qYB5gSwL6aWL9hERHCZc4G2+JrpgHNB8eCCmcIWIQ6rSdyPCyftXkDlErUkHafHRlkOIjxGbAktz75bnh50dU7YHk+Mz7wwstg6RFZb+TZuSOx1qqP5C66c0mptQmzIC2dlpte7vZrauAMm/7RfBYkGtXWGiaWTtwvAQiq2oD4YixPLXE2khB2FRaNRDTk+9sZ6K74Ia9VntCpN4BhJGJMT4Z5c5FhSepRCRWmBXqx+whVZC4me4saDs2iNqXMuCl6iAZflH8fscC1sTsy4PHeC+XYuqMBMUun5YezKbRKmEPwuK+CLzijPEQgfhahQswBBLfg/GBgBiI4QwAqzJkkyYAWtjzSg2ILgMAgqxYfwERRo3zruBL9WOryUArSD8sQOcD7fvIODJxKFS615KFPsb68USBEPPj1orNzFY2xoTtNBVTyzBhPbhFH0PI5AtlJBl2aSgNPYzxYLw7XTDBDinmVoENwiGzmngrMo8OmnRP0Z0i0Zrln9DDFcnmOoBZjABaQIbPOJYZGqX+RCMlDDbElcjaROLDoualmUIQ88Kekk3iM4OQrADcxi3rJguS4MOIBIgKgXrjd1WkbCdqxJk/4efRIFsavZA7KvvJQqp3Iid5Z0NFc5aiMRzGN3vrpBzaMy4JYde3wr96PjN90AYOIbyp6T4zj8LoE66OGcX1Ef4Z3KoWLAUF4BTg7ug/AbkG5UNQXAMkQezujSHeir2uTThgd3gpyzDrbnEdDRH2W7U6PeRvBX1ZFMP5RM+Zu6UUZZD8hDPHldVWntTCNk7To8IeOW9yn2wx0gmurwqC60AOde4r3ETi5pVMSDK8wxhoGAoEX9NLWHIR33VbrbMveii2jAJlrxwytTHbWNu8Y4N8vCCyZjAX/pcsfwXbLze2+D+u33OGBoJyAAL3jn3RuEcdp5If8O+a4NKWvxOTyDltG0IWoHhwVGe7dKkCWFT++tm+haBCikRUUMrMhYKZJKYoVuv/bsJzO8DwfVIInQq3g3BYypiz8baogH3r3GwqCwFtZnz4xMjAVOYnyOi5HWbFA8n0qz1OjSpHWFzpQOpvkNETZBGpxN8ybhtqV/DMUxd9uFZmBfKXMCn/SqkWJyKPnT6lq+4zBZni6fYRByJn6OK+OgPBGRAJluwGSk4wxjOOzyce/PKODwRlsgrVkdcsEiYrqYdXo0Er2GXi2GQZd0tNJT6c9pK1EEJG1zgDJBoTVuCXGAU8BKTvCO/cEQ1Wjk3Zzuy90JX4m3O5IlxVFhYkSUwuQB2up7jhvkm+bddRQu5F9s0XftGEJ9JSuSk+ZachCbdU45fEqbugzTIUokwoAKvpUQF/CvLbWW5BNQFqFkJg2f30E/48StNe5QwBg8zz3YAJ82FZoXBxXSv4QDooDo79NixyglO9AembuBcx5Re3CwOKTHebOPhkmFC7wNaWtoBhFuV4AkEuJ0J+1pT0tLkvFVZaNzfhs/Kd3+A9YsImlO4XK4vpCo/elHQi/9gkFg07xxnuXLt21unCIpDV+bbRxb7FC6nWYTsMFF8+1LUg4JFjVt3vqbuhHmDKbgQ4e+RGizRiO8ky05LQGMdL2IKLSNar0kNG7lHJMaXr5mLdG3nykgj6vB/KVijd1ARWkFEf3yiUw1v/WaQivVUpIDdSNrrKbjO5NPnxz6qTTGgYg03HgPhDrCFyYZTi3XQw3HXCva39mpLNFtz8AiEhxAJHpWX13gCTAwgm9YTvMeiqetdNQv6IU0hH0G+ZManTqDLPjyrOse7WiiwOJCG+J0pZYULhN8NILulmYYvmVcV2MjAfA39sGKqGdjpiPo86fecg65UPyXDIAOyOkCx5NQsLeD4gGVjTVDwOHWkbbBW0GeNjDkcSOn2Nq4cEssP54t9D749A7M1AIOBl0Fi0sSO5v3P7LCBrM6ZwFY6kp2FX6AcbGUdybnfChHPyu6WlRZ2Fwv9YM0RMI7kISRgR8HpQSJJOyTfXj/6gQKuihPtiUtlCQVPohUgzfezTg8o1b3n9pNZeco1QucaoXe40Fa5JYhqdTspFmxGtW9h5ezLFZs3j/N46f+S2rjYNC2JySXrnSAFhvAkz9a5L3pza8eYKHNoPrvBRESpxYPJdKVUxBE39nJ1chrAFpy4MMkf0qKgYALctGg1DQI1kIymyeS2AJNT4X240d3IFQb/0jQbaHJ2YRK8A+ls6WMhWmpCXYG5jqapGs5/eOJErxi2/2KWVHiPellTgh/fNl/2KYPKb7DUcAg+mCOPQFCiU9Mq/WLcU1xxC8aLePFZZlE+PCLzf7ey46INWRw2kcXySR9FDgByXzfxiNKwDFbUSMMhALPFSedyjEVM5442GZ4hTrsAEvZxIieSHGSgkwFh/nFNdrrFD4tBH4Il7fW6ur4J8Xaz7RW9jgtuPEXQsYk7gcMs2neu3zJwTyUerHKSh1iTBkj2YJh1SSOZL5pLuQbFFAvyO4k1Hxg2h99MTC6cTUkbONQIAnEfGsGkNFWRbuRyyaEZInM5pij73EA9rPIUfU4XoqQpHT9THZkW+oKFLvpyvTBMM69tN1Ydwv1LIEhHsC+ueVG+w+kyCPsvV3erRikcscHjZCkccx6VrBkBRusTDDd8847GA7p2Ucy0y0HdSRN6YIBciYa4vuXcAZbQAuSEmzw+H/AuOx+aH+tBL88H57D0MsqyiZxhOEQkF/8DR1d2hSPMj/sNOa5rxcUnBgH8ictv2J+cb4BA4v3MCShdZ2vtK30vAwkobnEWh7rsSyhmos3WC93Gn9C4nnAd/PjMMtQfyDNZsOPd6XcAsnBE/mRHtHEyJMzJfZFLE9OvQa0i9kUmToJ0ZxknTgdl/XPV8xoh0K7wNHHsnBdvFH3sv52lU7UFteseLG/VanIvcwycVA7+BE1Ulyb20BvwUWZcMTKhaCcmY3ROpvonVMV4N7yBXTL7IDtHzQ4CCcqF66LjF3xUqgErKzolLyCG6Kb7irP/MVTCCwGRxfrPGpMMGvPLgJ881PHMNMIO09T5ig7AzZTX/5PLlwnJLDAPfuHynSGhV4tPqR3gJ4kg4c06c/F1AcjGytKm2Yb5jwMotF7vro4YDLWlnMIpmPg36NgAZsGA0W1spfLSue4xxat0Gdwd0lqDBOgIaMANykwwDKejt5YaNtJYIkrSgu0KjIg0pznY0SCd1qlC6R19g97UrWDoYJGlrvCE05J/5wkjpkre727p5PTRX5FGrSBIfJqhJE/IS876PaHFkx9pGTH3oaY3jJRvLX9Iy3Edoar7cFvJqyUlOhAEiOSAyYgVEGkzHdug+oRHIEOXAExMiTSKU9A6nmRC8mp8iYhwWdP2U/5EkFAdPrZw03YA3gSyNUtMZeh7dDCu8pF5x0VORCTgKp07ehy7NZqKTpIC4UJJ89lnboyAfy5OyXzXtuDRbtAFjZRSyGFTpFrXwkpjSLIQIG3N0Vj4BtzK3wdlkBJrO18MNsgseR4BysJilI0wI6ZahLhBFA0XBmV8d4LUzEcNVb0xbLjLTETYN8OEVqNxkt10W614dd1FlFFVTIgB7/BQQp1sWlNolpIu4ekxUTBV7NmxOFKEBmmN+nA7pvF78/RII5ZHA09OAiE/66MF6HQ+qVEJCHxwymukkNvzqHEh52dULPbVasfQMgTDyBZzx4007YiKdBuUauQOt27Gmy8ISclPmEUCIcuLbkb1mzQSqIa3iE0PJh7UMYQbkpe+hXjTJKdldyt2mVPwywoODGJtBV1lJTgMsuSQBlDMwhEKIfrvsxGQjHPCEfNfMAY2oxvyKcKPUbQySkKG6tj9AQyEW3Q5rpaDJ5Sns9ScLKeizPRbvWYAw4bXkrZdmB7CQopCH8NAmqbuciZChHN8lVGaDbCnmddnqO1PQ4ieMYfcSiBE5zzMz+JV/4eyzrzTEShvqSGzgWimkNxLvUj86iAwcZuIkqdB0VaIB7wncLRmzHkiUQpPBIXbDDLHBlq7vp9xwuC9AiNkIptAYlG7Biyuk8ILdynuUM1cHWJgeB+K3wBP/ineogxkvBNNQ4AkW0hvpBOQGFfeptF2YTR75MexYDUy7Q/9uocGsx41O4IZhViw/2FvAEuGO5g2kyXBUijAggWM08bRhXg5ijgMwDJy40QeY/cQpUDZiIzmvskQpO5G1zyGZA8WByjIQU4jRoFJt56behxtHUUE/om7Rj2psYXGmq3llVOCgGYKNMo4pzwntITtapDqjvQtqpjaJwjHmDzSVGLxMt12gEXAdLi/caHSM3FPRGRf7dB7YC+cD2ho6oL2zGDCkjlf/DFoQVl8GS/56wur3rdV6ggtzZW60MRB3g+U1W8o8cvqIpMkctiGVMzXUFI7FacFLrgtdz4mTEr4aRAaQ2AFQaNeG7GX0yOJgMRYFziXdJf24kg/gBQIZMG/YcPEllRTVNoDYR6oSJ8wQNLuihfw81UpiKPm714bZX1KYjcXJdfclCUOOpvTxr9AAJevTY4HK/G7F3mUc3GOAKqh60zM0v34v+ELyhJZqhkaMA8UMMOU90f8RKEJFj7EqepBVwsRiLbwMo1J2zrE2UYJnsgIAscDmjPjnzI8a719Wxp757wqmSJBjXowhc46QN4RwKIxqEE6E5218OeK7RfcpGjWG1jD7qND+/GTk6M56Ig4yMsU6LUW1EWE+fIYycVV1thldSlbP6ltdC01y3KUfkobkt2q01YYMmxpKRvh1Z48uNKzP/IoRIZ/F6buOymSnW8gICitpJjKWBscSb9JJKaWkvEkqinAJ2kowKoqkqZftRqfRQlLtKoqvTRDi2vg/RrPD/d3a09J8JhGZlEkOM6znTsoMCsuvTmywxTCDhw5dd0GJOHCMPbsj3QLkTE3MInsZsimDQ3HkvthT7U9VA4s6G07sID0FW4SHJmRGwCl+Mu4xf0ezqeXD2PtPDnwMPo86sbwDV+9PWcgFcARUVYm3hrFQrHcgMElFGbSM2A1zUYA3baWfheJp2AINmTJLuoyYD/OwA4a6V0ChBN97E8YtDBerUECv0u0TlxR5yhJCXvJxgyM73Bb6pyq0jTFJDZ4p1Am1SA6sh8nADd1hAcGBMfq4d/UfwnmBqe0Jun1n1LzrgKuZMAnxA3NtCN7Klf4BH+14B7ibBmgt0TGUafVzI4uKlpF7v8NmgNjg90D6QE3tbx8AjSAC+OA1YJvclyPKgT27QpIEgVYpbPYGBsnyCNrGz9XUsCHkW1QAHgL2STZk12QGqmvAB0NFteERkvBIH7INDsNW9KKaAYyDMdBEMzJiWaJHZALqDxQDWRntumSDPcplyFiI1oDpT8wbwe01AHhW6+vAUUBoGhY3CT2tgwehdPqU/4Q7ZLYvhRl/ogOvR9O2+wkkPKW5vCTjD2fHRYXONCoIl4Jh1bZY0ZE1O94mMGn/dFSWBWzQ/VYk+Gezi46RgiDv3EshoTmMSlioUK6MQEN8qeyK6FRninyX8ZPeUWjjbMJChn0n/yJvrq5bh5UcCAcBYSafTFg7p0jDgrXo2QWLb3WpSOET/Hh4oSadBTvyDo10IufLzxiMLAnbZ1vcUmj3w7BQuIXjEZXifwukVxrGa9j+DXfpi12m1RbzYLg9J2wFergEwOxFyD0/JstNK06ZN2XdZSGWxcJODpQHOq4iKqjqkJUmPu1VczL5xTGUfCgLEYyNBCCbMBFT/cUP6pE/mujnHsSDeWxMbhrNilS5MyYR0nJyzanWXBeVcEQrRIhQeJA6Xt4f2eQESNeLwmC10WJVHqwx8SSyrtAAjpGjidcj1E2FYN0LObUcFQhafUKTiGmHWRHGsFCB+HEXgrzJEB5bp0QiF8ZHh11nFX8AboTD0PS4O1LqF8XBks2MpjsQnwKHF6HgaKCVLJtcr0XjqFMRGfKv8tmmykhLRzu+vqQ02+KpJBjaLt9ye1Ab+BbEBhy4EVdIJDrL2naV0o4wU8YZ2Lq04FG1mWCKC+UwkXOoAjneU/xHplMQo2cXUlrVNqJYczgYlaOEczVCs/OCgkyvLmTmdaBJc1iBLuKwmr6qtRnhowngsDxhzKFAi02tf8bmET8BO27ovJKF1plJwm3b0JpMh38+xsrXXg7U74QUM8ZCIMOpXujHntKdaRtsgyEZl5MClMVMMMZkZLNxH9+b8fH6+b8Lev30A9TuEVj9CqAdmwAAHBPbfOBFEATAPZ2CS0OH1Pj/0Q7PFUcC8hDrxESWdfgFRm+7vvWbkEppHB4T/1ApWnlTIqQwjcPl0VgS1yHSmD0OdsCVST8CQVwuiew1Y+g3QGFjNMzwRB2DSsAk26cmA8lp2wIU4p93AUBiUHFGOxOajAqD7Gm6NezNDjYzwLOaSXRBYcWipTSONHjUDXCY4mMI8XoVCR/Rrs/JLKXgEx+qkmeDlFOD1/yTQNDClRuiUyKYCllfMiQiyFkmuTz2vLsBNyRW+xz+5FElFxWB28VjYIGZ0Yd+5wIjkcoMaggxswbT0pCmckRAErbRlIlcOGdBo4djTNO8FAgQ+lT6vPS60BwTRSUAM3ddkEAZiwtEyArrkiDRnS7LJ+2hwbzd2YDQagSgACpsovmjil5wfPuXq3GuH0CyE7FK3M4FgRaFoIkaodORrPx1+JpI9psyNYIFuJogZa0/1AhOWdlHQxdAgbwacsHqPZo8u/ngAH2GmaTdhYnBfSDbBfh8CHq6Bx5bttP2+RdM+MAaYaZ0Y/ADkbNCZuAyAVQa2OcXOeICmDn9Q/eFkDeFQg5MgHEDXq/tVjj+jtd26nhaaolWxs1ixSUgOBwrDhRIGOLyOVk2/Bc0UxvseQCO2pQ2i+Krfhu/WeBovNb5dJxQtJRUDv2mCwYVpNl2efQM9xQHnK0JwLYt/U0Wf+phiA4uw8G91slC832pmOTCAoZXohg1fewCZqLBhkOUBofBWpMPsqg7XEXgPfAlDo2U5WXjtFdS87PIqClCK5nW6adCeXPkUiTGx0emOIDQqw1yFYGHEVx20xKjJVYe0O8iLmnQr3FA9nSIQilUKtJ4ZAdcTm7+ExseJauyqo30hs+1qSW211A1SFAOUgDlCGq7eTIcMAeyZkV1SQJ4j/e1Smbq4HcjqgFbLAGLyKxlMDMgZavK5NAYH19Olz3la/QCTiVelFnU6O/GCvykqS/wZJDhKN9gBtSOp/1SP5VRgJcoVj+kmf2wBgv4gjrgARBWiURYx8xENV3bEVUAAWWD3dYDKAIWk5opaCFCMR5ZjJExiCAw7gYiSZ2rkyTce4eNMY3lfGn+8p6+vBckGlKEXnA6Eota69OxDO9oOsJoy28BXOR0UoXNRaJD5ceKdlWMJlOFzDdZNpc05tkMGQtqeNF2lttZqNco1VtwXgRstLSQ6tSPChgqtGV5h2DcDReIQadaNRR6AsAYKL5gSFsCJMgfsaZ7DpKh8mg8Wz8V7H+gDnLuMxaWEIUPevIbClgap4dqmVWSrPgVYCzAoZHIa5z2Ocx1D/GvDOEqMOKLrMefWIbSWHZ6jbgA8qVBhYNHpx0P+jAgN5TB3haSifDcApp6yymEi6Ij/GsEpDYUgcHATJUYDUAmC1SCkJ4cuZXSAP2DEpQsGUjQmKJfJOvlC2x/pChkOyLW7KEoMYc5FDC4v2FGqSoRWiLsbPCiyg1U5yiHZVm1XLkHMMZL11/yxyw0UnGig3MFdZklN5FI/qiT65T+jOXOdO7XbgWurOAZR6Cv9uu1cm5LjkXX4xi6mWn5r5NjBS0gTliHhMZI2WNqSiSphEtiCAwnafS11JhseDGHYQ5+bqWiAYiAv6Jsf79/VUs4cIl+n6+WOjcgB/2l5TreoAV2717JzZbQIR0W1cl/dEqCy5kJ3ZSIHuU0vBoHooEpiHeQWVkkkOqRX27eD1FWw4BfO9CJDdKoSogQi3hAAwsPRFrN5RbX7bqLdBJ9JYMohWrgJKHSjVl1sy2xAG0E3sNyO0oCbSGOxCNBRRXTXenYKuwAoDLfnDcQaCwehUOIDiHAu5m5hMpKeKM4sIo3vxACakIxKoH2YWF2QM84e6F5C5hJU4g8uxuFOlAYnqtwxmHyNEawLW/PhoawJDrGAP0JYWHgAVUByo/bGdiv2T2EMg8gsS14/rAdzlOYazFE7w4OzxeKiWdm3nSOnQRRKXSlVo8HEAbBfyJMKqoq+SCcTSx5NDtbFwNlh8VhjGGDu7JG5/TAGAvniQSSUog0pNzTim8Owc6QTuSKSTXlQqwV3eiEnklS3LeSXYPXGK2VgeZBqNcHG6tZHvA3vTINhV0ELuQdp3t1y9+ogD8Kk/W7QoRN1UWPqM4+xdygkFDPLoTaumKReKiLWoPHOfY54m3qPx4c+4pgY3MRKKbljG8w4wvz8pxk3AqKsy4GMAkAtmRjRMsCxbb4Q2Ds0Ia9ci8cMT6DmsJG00XaHCIS+o3F8YVVeikw13w+OEDaCYYhC0ZE54kA4jpjruBr5STWeqQG6M74HHL6TZ3lXrd99ZX++7LhNatQaZosuxEf5yRA15S9gPeHskBIq3Gcw81AGb9/O53DYi/5CsQ51EmEh8Rkg4vOciClpy4d04eYsfr6fyQkBmtD+P8sNh6e+XYHJXT/lkXxT4KXU5F2sGxYyzfniMMQkb9OjDN2C8tRRgTyL7GwozH14PrEUZc6oz05Emne3Ts5EG7WolDmU8OB1LDG3VrpQxp+pT0KYV5dGtknU64JhabdqcVQbGZiAxQAnvN1u70y1AnmvOSPgLI6uB4AuDGhmAu3ATkJSw7OtS/2ToPjqkaq62/7WFG8advGlRRqxB9diP07JrXowKR9tpRa+jGJ91zxNTT1h8I2PcSfoUPtd7NejVoH03EUcqSBuFZPkMZhegHyo2ZAITovmm3zAIdGFWxoNNORiMRShgwdYwFzkPw5PA4a5MIIQpmq+nsp3YMuXt/GkXxLx/P6+ZJS0lFyz4MunC3eWSGE8xlCQrKvhKUPXr0hjpAN9ZK4PfEDrPMfMbGNWcHDzjA7ngMxTPnT7GMHar+gMQQ3NwHCv4zH4BIMYvzsdiERi6gebRmerTsVwZJTRsL8dkZgxgRxmpbgRcud+YlCIRpPwHShlUSwuipZnx9QCsEWziVazdDeKSYU5CF7UVPAhLer3CgJOQXl/zh575R5rsrmRnKAzq4POFdgbYBuEviM4+LVC15ssLNFghbTtHWerS1hDt5s4qkLUha/qpZXhWh1C6lTQAqCNQnaDjS7UGFBC6wTu8yFnKJnExCnAs3Ok9yj5KpfZESQ4lTy5pTGTnkAUpxI+yjEldJfSo4y0QhG4i4IwkRFGcjWY8+EzgYYJUK7BXQksLxAww/YYWBMhJILB9e8ePEJ4OP7z+4/wOQDl64iOYDp26DaONPxpKtBxq/aTzRGarm3VkPYTLJKx6Z/Mw2YbBGseJhPMwhhNswrIkyvV2BYzrvZbxLpKwcWJhYmFtVZ+lPEq91FzVp1HlQY1bZVLqeNR9SAUn6n0E28k/UuGkNpP1DBI5ch/EehZfjUQ9aE41NhETExoPT2gGQz0IhWJbEOvTQ4wgcXCHHFBhewYUiFHuhRSAUVmEHeCRQHQkXGFwkAgyzREJCVN7TRnTon36Zw3tPhx4EALwNdwDv+J41YSP4B2CQqz0EFgARZ4ESgBHQgROwAVn9GTI+HYexTUevLUeta4/DqKrbMVS+Yqb8hUwYCrlgKtmAq1YCrFgKrd4qpXiqZcKn1oqdWipjYKpWwVPVYqW6xUpVipKqFR3QKjagVEtAqHpxUMTitsnFaJOKx2cVhswq35RVpyiq9lFVNIKnOQVMkgqtYxVNxiqQjFS7GKlSIVIsQqPIhUWwioigFQ++KkN8VHr49HDw9Ebo9EDo9DTo9Crg9BDg9/Wx7gWx7YWwlobYrOGxWPNisAaAHEyALpkAVDIAeWAArsABVXACYuAD5cAF6wAKFQAQqgAbVAAsoAAlQAUaYAfkwAvogBWQACOgAD9AAHSAAKT4GUdMiOvFngBTwCn2AZ7Dv6B6k/90B8+yRnkV144AIBoAMTQATGgAjNAA4YABgwABZgB/mQCwyAVlwCguASlwCEuAQFwB4uAMlwBYuAJlQAUVAAhUD2KgdpUDaJgaRMDFJgX5MC1JgWJEAokQCWRAHxEAWkQBMRADpEAMkQAYROAEecC484DRpwBDTnwNOdw05tjTmiNOYwtswhYFwLA7BYG4LA2BYGOLAwRYFuLAsxYFQJAohIEyJAMwkAwiQC0JAJgkAeiQBkJAFokAPCQA0JABwcD4Dgc4cDdDgaYcDIDgYgUC6CgWgUClCgUYUAVBQBOFAEYMALgwAgDA9QYAdIn8AZzeBB2L5EcWrenUT1KXienEsuJJ7x5U8XlTjc1NVzUyXFTGb1LlpUtWlTDIjqwE4LsagowoCi2gJLKAkpoBgJQNpAIhNqaEoneI6kiiqQ6Go/n6j0cS+a2gEU8gIHJ+BwfgZX4GL+Bd/gW34FZ+BS/gUH4FN6BTegTvoEv6BJegRnYEF2A79gOvYDl2BdEjCkqkGtwXp0LNToIskOTXzh/F062yJ7AAAAEDAWAAABWhJ+KPEIJgBFxMVP7w2QJBGHASQnOBKXKFIdUK4igKA9IEaYJg) format('embedded-opentype'),url(data:application/font-woff;base64,d09GRgABAAAAAFuAAA8AAAAAsVwAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcbSqX3EdERUYAAAF0AAAAHwAAACABRAAET1MvMgAAAZQAAABFAAAAYGe5a4ljbWFwAAAB3AAAAsAAAAZy2q3jgWN2dCAAAAScAAAABAAAAAQAKAL4Z2FzcAAABKAAAAAIAAAACP//AANnbHlmAAAEqAAATRcAAJSkfV3Cb2hlYWQAAFHAAAAANAAAADYFTS/YaGhlYQAAUfQAAAAcAAAAJApEBBFobXR4AABSEAAAAU8AAAN00scgYGxvY2EAAFNgAAACJwAAAjBv+5XObWF4cAAAVYgAAAAgAAAAIAFqANhuYW1lAABVqAAAAZ4AAAOisyygm3Bvc3QAAFdIAAAELQAACtG6o+U1d2ViZgAAW3gAAAAGAAAABsMYVFAAAAABAAAAAMw9os8AAAAA0HaBdQAAAADQdnOXeNpjYGRgYOADYgkGEGBiYGRgZBQDkixgHgMABUgASgB42mNgZulmnMDAysDCzMN0gYGBIQpCMy5hMGLaAeQDpRCACYkd6h3ux+DAoPD/P/OB/wJAdSIM1UBhRiQlCgyMADGWCwwAAAB42u2UP2hTQRzHf5ekaVPExv6JjW3fvTQ0sa3QLA5xylBLgyBx0gzSWEUaXbIoBBQyCQGHLqXUqYNdtIIgIg5FHJxEtwqtpbnfaV1E1KFaSvX5vVwGEbW6OPngk8/vvXfv7pt3v4SImojIDw6BViKxRgIVBaZwVdSv+xvXA+Iuzqcog2cOkkvDNE8Lbqs74k64i+5Sf3u8Z2AnIRLbyVCyTflVSEXVoEqrrMqrgiqqsqqqWQ5xlAc5zWOc5TwXucxVnuE5HdQhHdFRHdNJndZZndeFLc/zsKJLQ/WV6BcrCdWkwspVKZVROaw0qUqqoqZZcJhdTnGGxznHBS5xhad5VhNWCuturBTXKZ3RObuS98pb9c57k6ql9rp2v1as5deb1r6s9q1GV2IrHSt73T631424YXzjgPwqt+Rn+VG+lRvyirwsS/KCPCfPytPypDwhj8mjctRZd9acF86y89x55jxxHjkPnXstXfbt/pNjj/nwXW+cHa6/SYvZ7yEwbDYazDcIgoUGzY3h2HtqgUcs1AFPWKgTXrRQF7xkoQhRf7uF9hPFeyzUTTSwY6EoUUJY6AC8bSGMS4Ys1Au3WaiPSGGsMtkdGH2rzJgYHAaYjxIwQqtB1CnYkEZ9BM6ALOpROAfyqI/DBQudgidBETXuqRIooz4DV0AV9UV4GsyivkTEyMMmw1UYGdhkuAYjA5sMGMvIwCbDDRgZeAz1TXgcmDy3YeRhk+cOjCxsMjyAkYFNhscwMrDJ8BQ2886gXoaRhedQvyTSkDZ7uA6HLLQBI5vGntAbGHugTc53cMxC7+E4SKL+ACOzNpk3YWTWJid+iRo5NXIKM3fBItAPW55FdJLY3FeHBDr90606JCIU9Jk+Ms3/Y/8L8jUq3y79bJ/0/+ROoP4v9v/4/mj+i7HBXUd0/elU6IHfHt8Aj9EPGAAoAvgAAAAB//8AAnjaxb0JfBvVtTA+dxaN1hltI1m2ZVuSJVneLVlSHCdy9oTEWchqtrBEJRAgCYEsQNhC2EsbWmpI2dqkQBoSYgKlpaQthVL0yusrpW77aEubfq/ly+ujvJampSTW5Dvnzmi1E+jr//3+Xmbu3Llz77nnbuece865DMu0MAy5jGtiOEZkOp8lTNeUwyLP/DH+rEH41ZTDHAtB5lkOowWMPiwayNiUwwTjE46AI5xwhFrINPXYn/7ENY0dbWHfZAiTZbL8ID/InAd5xz2NpIH4STpDGonHIJNE3OP1KG4ISaSNeBuITAyRLgIxoiEUhFAnmUpEiXSRSGqAQEw0kuyFUIb0k2gnGSApyBFi0il2SI5YLGb5MdFjXCey4mNHzQ7WwLGEdZiPPgYR64we8THZHAt+wnT84D/x8YTpGPgheKH4CMEDVF9xBOIeP3EbQgGH29BGgpGkIxCMTCW9qUTA0Zsir+QUP1mt+P2KusevwIO6Bx/Iaj8/OD5O0VNrZW2EsqZBWbO1skRiEKE0DdlKKaSVO5VAuRpqk8VQJAqY7ydxaK44YJvrO2EWjOoDBoFYzQbDNkON+UbiKoRkywMWWf1j4bEY2iIY1AeMgvmEz/kVo9v4FSc/aMZMrFbjl4zWLL0+Y5FlyzNlEVYDudJohg8gPUP7kcB/mn+G6cd+5PV4Q72dXCgocWJADBgUuDTwiXiGSyZo14HOEQ2lE6k0XDIEusexDzZOMXwt1Dutz+tqmxTvlskNWXXUQIbhaurum9GrePqm9Yaeabjkiqf+bUvzDOvb2Y1E+EX2DnemcTP/zLcuu7xjQXdAtjR0Lo5n4/Hs/GtntMlysHt+29NXbH6se//WbFcyu+r28H0MwzI30DYeYTLMXIA2EG8QlHpAsyS0EfEToR0a3utIxFPJ3kiIHCCrZ66b0e2xEmL1dM9YN/MwS5p01N5jMX/BLKt/1R83l0LyC29M6+iYxo/UNg/EF7c2WyyW5tYl8WnhWg2/hyySbD5UhnDyS7OcU0dnrFw+DfGdI7v4QfYIIzOMq9hFtY55gmvC7jZ2FK7sEdrn6IXBuucYhjsGdQ8z0yEbWkkczjjsE5hNAIZrPx2zOLZDmKNXcXtg7EMqidAEEWg+SJCBBNwxvxJfc/bZa+KKf+xoKZybnq5vaqpPTye7CiF+ZFjxZ8/7Qij0hfOG/cowPA1rT1l4ymWnrKmxxqfErTVrpgwPlz1kC+Oy8NMDz6c+IO38K/x0xkPnLW8Kx6qGAoQdL+TD9V9rb+/ctn//trxz8dUrZrD/zk/ferF0cNt1BzctmX2FZPXt/jnFCQNz4Ah/iKllGiCMs1w5Lkg0kiEwj6VTXCDKsX9rMpnvIj9pcDecXAIXMnqn2dTUbN6w0XQ9ue6FV/nnXCH7S3lPWGltVcLsH75ub3ab7A8M28caNrIeOr3o5Q0yFsYL80xaa0EY/UEczV7icUMY5pnelAkmUAXmHYjvFWFGxuqlSaow3OM+/iYY7/l/hVELF4EjRqNR/bvRbOY+DUGzGR/Oh3EqmE/ugIQQguGt/eMYz/+L0cimjeZfQDI3phXMbMQsqH+CjwVz/hf4idHovgVmB8gLvjbicDcC/NypP536E/9N/puMibExdohBmNwyiaZdJGoigos7GpF222xrfnZhML/7Z+ylaqP63Hr+m7bdUkQ6/2cXqdfmvwixY+s2ksXFeXcE+iX0Z+Iow76DBNgjJ7TOdUK18iPsPflfQD+DPsZG2Aj9VmKMMJ4fYRrhIaxhTDR0Elh2vA6h/AE6xUb29mj3sjmL72petXjejPy+oel60M99tFduCI59N3221xe7apOvxs6aHs7vab1IqY2tv7q2xsHeHGml/cV06u/8S/xTjJ+JYc0bWEX0ukW6YmIbGkJRMdjJ9mYIH5QIdJF4hvRGyK7cC7ctImQRcUET99fGXOoft35GYLMQu+g2smnkgZUrH8AL/9Si217IssJ916nv14ZrJrvdxLkQvrvtBcjgPC0NXOicO8Qf4mcxPqh3hgUw3DDfdvLJXngg7N3dN2zbPJSaed3OfZnMU7dvmznp3C3bruO+Nmue0LFsy7S+6265+fCKFYdvvuW6vmlblnUI8xCXp37CrOZv4B9gauDBlYp7adcUXB5DNCwYImlXOJJKkAdvExXxVvKEYnCo+3eIskP9qrrfIYs71CccBjfXRC52udTHHdaP1A1ui/VvH1otbrLrpNXBsGX5B89QghDyimlvNB2KfkxZ5C9/em3+d1+d//IfFp2+2Oxn/s+9n/79p39S3s8idN6g0yZObwJOgKUpNB3GyU0Ls0PbRzIRq4lcarLKOJBkLRzJQD4j2090XrbA7DW8K3jNF5hlGS5e4V2D17zgss4T20egOJte5iD0bReM9yjTxnQxCRj3c5kFzGJmGbNKmwGw39IJDJcXJZGMkaAB4jyJAKw0jt5IAuIE+A+U3cVAZZrq9zhDyBrU8oosuxcGNTzCKJfla7JjNVmuSb/+tuzN2H+X4vlB+PpdfMXXmuVsNiub1T34SFbjYw5itEvVi0K0Nt9pNJUMI7SLGRhf2xipfCYf8z5OdlGKayOucFeVPeS/dbo3lBrbSMmwUiQN5/ed7g0Ds1s17IuZC5kNzM3MZ6EWCa0DtekdJfAxz+R/OX28sND7yRMTBcf++s8mQCQWHya4qBv/ufeMoWyslPA9DtMxUknxkH/yfTnm2CMYzs+Cq3r7PxY/MXomrvTEsRpfEGHa+WN8E1AHjElb7d06ddA7oK/+5Mdsv9EtPms0jv0Z5kf1FqPxWdFtfFr0kHfgDX0Y+5PRSG7RUj0tQr7rmfX8DH4G5W28kKeJLtmQsQkuwMP1pk16EV4sl7vrMJATfyUWo/GwEco4rh4XFQgaiUX9qxZHrMQqKnz/c2d8b9TysYrAuXpP/Rf/Gr8b1qwwc5a+euLa6S6sneNXToG2XrEJi4R5SGs8Sq2S3d97bsfCRaTdaLwKClRHt37mkudvXbjwVrLhuYeGhh56bvfQkHpk2CwvwClqgWwuBfndC3c8dwmstj81KkagcUgbfPY8Zje0W/82VPWJHmSq6pP8hPWpotc/EexDOK3qU+wngPhOCiO9MJRm8TJefjelrzoKnG2Bn+1NCUmPE4gHFmBN9jrTigRIpsACrc9Gstg58ULkp9467+Gf/eFnD5/31lNrt2967dhrm7bzI+VT5m+fzKhvf2MzpICEm79Bopkn07lt1762adNr127LwVqQLdJ5+lpQDcvHPQtVY5knhYrK6q8/JsiP6EuhGZdFdaNszjvpqvc+PI0CdjN0AXsFOC3ZfALDJwr4q2Xq+GF+GNbsxUg5NLLIEXi8otcDQcUts0D8eQ1iVDRAMBTsYiNdRIxE09EIBJO9A2xqgERTaW86BUFn0OD2xFO97FAgFhF6OoQ7prYt4XwSeUgQHiJyDbeke9IdQntciLQ1FlJMaYcUNvZBg+FB1ubjlnRNvl3o6IEU2w7fdNPhm/hh+FLysUu6++DLHkOkrSHYEjH0tEPe7WdD3uyDgvAgK/m4szFFR7ch0toUgBTdWHr7EpaWru6+6dmbbnqWEbV2EtxAsXiZAPTtGPSbHsotI2leoM8TePEqgSQprs7AGFf8kuOkPdZPXGb55POAW1d/jLST9v5YflasP6v/CO7+GNAPC2BMZWmsOjp2NNbfHwMCJD+LPVL+D/OYlWEEI/9jpPddOFkB5d1GSuKZYggmCCd7JUxD7EXAzxyirYnNDLdDZoFdx14kivkvGc3579Jm36reTTvDgBnaO6vzyQ6chQmlsMoIkIQ2+bBDWBud1Va4pcCn8CPqxlh/fgtG8IPaPH8C5wk6/nZDv69jurV5QhtwE0x2iqOsj9Mx8B9/0EaUdiPfOYYDCi/q9jhWRuupMDEU0+CtX0sDFxv07T/K5niBPqN9+tQjgEc31NGCXFeMcCEuQBIc/BK4CO78u7EPYvl3yaEfK3vcb6qP1R2tI7vUjVDDUdKubsSrNjYKY1qBEa2P50SJoaXiksIoLiCwnxS6EBuBde87botNfdEWwYvF/R0/u5yCqhGeEOR2ynSeyXjt6ka7neyye8kryBSWE52y+RBgogrXPZ8E1yIHoHIFUM+AbJhE7lbMtt8ApL+xmZW7PwbjAO0fAVoXQOuiSP/ksIVdFZ0aulsamKUzwPZ/NYDMJRBPCxsBqLzqHyneXF6Ej9HlIFo7+pg+jUb3unRmGpstGkm6etOuDBGA5wCMefp1gTHcdZlvPBXlOslvYTp1cd8UjYLVd/J5awNrIOKLnIt9MD9qdrKrWCvA6ALm3QV9VrsPm60Q7+RHJHP+2hqfugo/MvI2H/mqr4b9tFnKSRY1Y5Ek80Nm/WIhr1ikKnxGz9TWXrokf9xwujfvcOTtNTWnxd0F37Y2W79tteBqZ4G5qLCuomw+nSr28QESCRVLTyYKILGJOPfcnaIFOsewhRdvv+rWa/Wih0vlbX6Zb75T5C0qNKVFvH1QL/vazSWgC2s6oWXXIuUxQelKiJbowuJDQViatLmLijg9CQBMg8WiPgiw3LEeYRmm5f+XdnvkDnxLLjMLxtvX74C3OlwPQqx4xwIdpPx38LrlDphiyWUWHWKAzzxurS/xTo+P5wGFak62ap1PVFFN4v/y+xuR39WnIO7lsWfwgVsK17wxrs9K8ltIKuhkw7f/6dhK6gQokFKhWX3urrjk/rnI0pgfpGMeuQIUaEM7+GF5q2iMkCaMQwxxOzcvU0eXbsnS9XknXvP7Gtw5dwPXlFu2ecvSHEZgNDsU6x/GdXBYXyOQjzZReSedeEPY6nEv9gJR4oBQJtFO6Kd0fwC6BO4LNHDeBujB6dSNcUQC9zIv2LnAzGk99bUDrdFY+9yGFQtEo0GQPNv6vS2drj4+1jHbv3aJSMUWP+QTZrmbNTjU8wyG/iXNNpskybLcJ3CiTF5Ir+JYzmJwE0mSVhlxbtbmvweB3ulB6Til5UuUZydpgiFVeobhU0WaBqpJ198d+/XeNRTZ9/1OPfG7+2hwzd5W3D+hmyjsRcUg/+Cavb++Vh2ls3L7zT/etOnHNxeerv313vzLVqPai4nJv+K1FC6040/4udw7sAb3laSg0XCkAAs0npBO6VJabS4Elk/U+D4gTXW+j0wnrMlqNamq4tMIYB87tE10i0FR3LZNhJsb7/R561btmes8YBCRkhYNByRtKd55mqTas9FYhJnbRGHuOh3M4QTdgQSqmgRxuzGdSvZGcbMxNQGk5C3ebLjoXIOFM4l+WKHmLTJwRv9E8GWJ6dYvf/FmEyEGr+gyrr1p5zrgkz0Cw2j94Hv8Jdx7dIVegBSNtgsqGsRQEYiIBoXwD0LNvQ5d7s5Z00QzwNhqZA0b+tMG1tQq5nd84uq8R0zPvX35G8uRaze4jcOHzz0w1+Q2BIRvf6J6Kgatnrbiem+CFvAxfkrndzD9MFPP1GWTUHclpASUkCNAQkpCCcCgDSUDAhDZ+CuEkgn8J7i9nMA7pA4lISappxILKfAeSAbIcSDuN2bJcfZILqeO5rLs0MnngSHYRdrHjmaz7JEsEPw51ZqDJDmUIOZIe34WaQeegNsJn1qz8AIpT3yCjyEih/xELkuJ0lEMYTLVCiWpo5oYMleMH6USyYJcD+uOe+kWKpn1Qns34iyYDjkSLvgnZXcgVQNeqINXr48m3iS7cjm8tedyY0f1QvTnHHdsrKby/+SSbPY8/NH6vpl/Esq3Ae4ZU1HC44KFiI9o7CEgab/RqHbj7s5KAg06s39ZP/zxI/mVuF/TbTSy+3Fb8If9/cv7+wt91yy8RfP1QXtW5RzQn7qIiZyuFM5QfJ5E9uVnqT85TanFx0lkP3ukBAMprvsRyi/C8NAJL1xbIIirSvnSj4O5netb4JxmNANHPssHAcHMHsFRgEug816gDBeMbdfiuRcghqYcm0+Xxx/5IAEtN3fqFF3LzAXqwoT0PN0OVTNqxo8sxMkd5Ig6k79Zk7VxxX6gMLOZFQgvpW2RrMW1D0BDihaXQ9wVRoBxPLfpknmkeMtoB/qM9cRc9IqmMD2XUmdZ7GSRKPUZvChf8BoykriM2MnKYbOHX8R7cLdNCxSFFVQqoYswnlWtlFS2mNkhswVpZiQW1J/UKFfipHGlUkM6UKBhMz1istELIHJLMSctu3ugzfaVSOjKvUgc/THK4Sdg2Wscz69leKIkkrwuuWiOe9yGYKQXRumkC3qbRcMwrvhjNXgdZk3RxAUEhuSPvn3nnd++U/3vlVOmrJzCD8JLxV1OHRjrZifbcFDOuRNTGqdgQm1tSNJ2OcQ04YiEXuxtII1ECSQRoQGYioEsgCfchB4ghAtw7FfJre4WZ9hkVi9MtjuWqtdNDlpMrfEG9fOT6q21okg+e4As38MfGquNt7oUws6Ysarj1/efE+yst86YUVNvDdts3Pv5c8m/aP0C+f8/Qb+IMnGq09BgwN01oIOAnAdagI8mBSrqk1gxTDUBOtk2ousEtBH2z4Ir2d3f6k8PXXVlt2qN9RODxRuoJT/v27wm09jRYVc/e++iyx2tyzJb/n3J0htXP87eSsQaf2Ly0s6Zmxela88REy1cf4273mI3iXNJ7KxrZibOm9xm6rl4fqy/t27smU8tOfdW2ucBzg2UfmOIVyLIl3kpYlwphDISTXJXsctmiDtN7fNV6zelgxwnWxsVr83Aj/S5ki1jL/a0GC6+2L6Um+aoddlNFuj+bJ8mH/iaLh8I0/U51NspIEfq0dohwyFXKgm4NggwQ4rRhCOUFtxxo8XnitT4cnGfT93IS8FaT85XE3H5LMY4zIEPL1hw443wz+1UmhTJyJGxZzw+wsKkKZgUiVtKOKMEb2AKHTv61FNc01PQFwKnvsZ/9pPA4RKTASWahmh+8MxwzHxKy74IRn5LGRjsPUUwTu64UYNY38caqd7HKucZ/tHnODtENw/2UfHRMaq1UUPDJQ0OKkWCeet5fYOhII1VRz8+/Elg5j4Gxur3J8o2PJ4rg+2d08T/fwEzSVbyZ9XPro95T477lRKqUSRXQnauHNsISAl27oWi6Fv9z48JMv8r/aMMj8onCP/DuDZOuN+GPPr/+p7bx+7JlbYdppcNhzKU/1Px5aiaGDn/s1iGMaBcleKUo/v9rcxkZj7DBEKOfrayytXNLYiUdBY+pleQXdnscKlQcpzuWluxsieeyuXIK6SdxozitWyGOV3vOHHjguyCQ6fpIYy2JwvrQEF/Qa9Pdf/QqOSqCiE/EE1/XIVKTc2tzWbHnimrEd+Vyz311Ml3P0GVTj7PD5aDnsvCvH36alEaPMePcMegXs7x8igTu4B9v7G9vTHvhCu/kzIdx+BxC0ay9zRSvoS0F2lIxI+X7klU63I40gLQ3w5ep5na+SFnba3z5D64zv+QtM4n4ffG3tq4aNHGRfxgrXPMim+5487abL7xhdseIRn1KDl+7aINixdv0OD+JSPwKf5+xoP6aiTeQIDVlIhMcL1H5R9PYXvprs3fv2bO7MOplCmweuiq2JRZ1zz+9a/v2PH1Hfz9236w+ZrPXvWfAxlj4NLLHpq3c/PQ3uvmvbrjG7fe+o2y/cLdtE6VUlXi0ASb1VLUBVSUWSU4HdvAraTyS8xzM8NxvxFkXV6pUVRiJwcgC5zEeht4rwcp7ki0k41G0qlQhG1Vzlq8alEmnFi58caB5Q9vn988MLhqyVlHvLEWjtQFeupdiocF/tkkOGPW2ibWaBTkeZ/dvPWazXfOnnvL6jkRXpi85sFzZt+55ZptW3bl1cCCHZPD06MhySha7UFzjcjbp8fOecFCirzAG/yVjBX6OFIaadSjQq1nNhyIe8tVbaaSdHlXIWKacMeuZA1uxS95zILhyrxAdsXTL6m7kNQlx2P9uZf2qhufePFFbpI6/OU0WcP99RrCsrwseVot5mtytpf6Y0gm9sdeyKnPQ7onyK4nXlR/rg7H95M1upzu89DH6pgUcikoiihJ6NJKmRxV1x+MJiOA3YwhDRQrWU0u/0rvq0VYXnyCwsLeTJYBq3dAtJDavuzyoVpzZ99Z0+a0uoiFH/xcqgDR7rUFeOrUn6Cywb8ZeNMbhLV5ugP9l0zv9UN5b5mFkjzxUcpPJCn3V402pRxtJd2GrnLdhtVk9ZSZh9W91fCSH5B7ofxPiWL+j3D/uwhBRdyAyozeZwvQzs79soi+BKSnafLviZCcfrpBpLyimfLfTyJtbyruIQKD01tUwJyKEo/ybaxkSNFUMdMkhQoJyRBQFhnUkDQSXhTM+3NmY0EDM7ffLIjqWEGt8lCO6mLia3PukFnghosJD5p5SIho/VDkzQfLE+IrYoJXkD19pdP7OwG/voIUtagiWiZ4PAFTHHlTVhRZ7dYmPar+NJ+8JhmR6DFK5DV1foHoLNO/pHrvZfmWZ15RQlwvoVDKhCWNK3CCch9lfFBuAqUgpFSShmNaPj+i5++WZfKeViJfW5HnUakVL4UCNVkA4+ETfIqx4B5xSaP2L1yn0zn2ltPn4+OqZGmwwEVCaCSqG53ldtL1oLGAhdMLd09MpCCF6tD6ZnAZBY9hDaYsP0jzZ0j5ZjKsF4i1UmLuhbJMCnYJPt5VwFNvmZawXjEvLJqIH8STonZjq7BZ8gKgR20C9MDFqJAX1H64QW2NEup6qgzLP8cvppL/NNTOBTCJABOHeWoXzLhw4Wuy7gaBtjKr9kgKq8ZlRYBS32Lpxc8vIhpNDTfyNXWybMJbn2RyQ5EmWc2QF9wmSZ0KYCE+cPuYO6b15Uotj2Kd4MItLS7gtFbkTdrFND6pvEZqv5Yv7jXAus7Pg7avo7KDot50NX3CPkP+Kps8J9/3mGQIteY/LGPC+L7872SPR2br5fy8MtKBMHedGuM28/MZmPJMrGgi3Gb1S+Si1/L/zrZwO9XH1ce/z7ZQ1WSoY/+pMb5FT4ua0Wm+Jf/298nFmChEQ+Ti71est4mq9VYI6RsymoRJKYidElT2FGnDTZvqtfhGAFTbeqEw68GqtfmbVa/1IFO1/jdWr/8BDRRtQh9XNjubEm4aWVpVonpTGR7PVGc+KJNoBIWF7kYi4gUV3r1U6723i6TxUl3n3/tM27aZfKb7THiHW9VzFSwHJ05VfK6Ar7kaB0XgPPE0BSkSFKsBUpaLihEWoA9wBt8qirh2VSOkZwXEwyrxZ5jyt2rJmSo9gX7cg6jsEUGJU9z9xJPOEM3uQQxKgkh35DNATnVyrmJ3mbCNyIB/yox4wH1bg2DwN7q9kov4pFqny8oSm3RQbGgJ1QQTs6ZMLilOVYJ9v6Wha3HcJ9jddsXp9YhGUXLXt/qMDnvLpPNTXfNa60z5/yjXQOMq+lNmwh5egpYrdfZQZV9rI47xlRkuyTjpzsmCBSWNkAXVoK8sgYWqQJWbo1RLo6QH0YW6pxqfCnRgkd+RiFjUQUQ7poIaYoakgXxwFd9BuuI38H1xBxXSFb/pBDIKQFn7YB3dB36l7sG1FLaKiBdp1KxLvfswap/30lnVESgNnvjbUoT6w9N+Xoio0qcYOIM+heg940YimsucQVvli9NEcft2UZwGQwLuilj1fFr1i3NP94X+PE7Hpvtj6lBJfJ4R6NvWiaL6MgzWHxiN66DExa+dAdAbMYX6HVF8A+7rjEZIXAVbDe7PVI9rmN69JOLV1DOSvRPxWNPZBZf/Nf+Ny65BhYxxxV+77XJ2wfQ389/IQPgajXbwMsuAz/0IaQcXJavKbRqR2IqyZruXjVC2+hdee/5vdnYOedpmVtR3NGXldxSzDSIiBVpkGb9by89UpEPKrSLZmyFDzMab/wXl2CNe7s/qCtTvWgG5kpBmCBlSzDS/r8N4uwBwohRW63JTS1y32f0TQsPfXVGEHQrV8/NCfiOUVirYcBbIeA2+iF68rQIo3B/S628vYESr79ehzS7Q9LEL9UXmik9XVHb1yBO3Ngvt5935+k1efkV51mzzrM0LL3/20avnwMeKuWyOUZg2TasSqZ+KcZQiOn1Iu2Vh497ALUVZiCKt/gh6IvTIj1ZLRjWAkpHKOKovNwp00eqPROiAbiNEKieXwMLcXhVJ1/uzmLP4tfxaHR59cBdJVG1kTAgl9ze9QKUEQ946Hkb+okJ5JRDyf54Axur1D+WS49cLr0tTPEu7UmXrxcSr3XNvumv4yXzInXKH4F7Tc7p17Zt+t/qW2+93k063X7VW6lALxTY7i1nBXMxcxmzQbabxz+tJo+wijYaIGMNS8AoSMgAPt84DdHOoMPfjXhF+kuH1tZvuFQrRCN07xGcXRX9MYxYchDe5BcHj+Z4i+42WyPc8Xofi7bbZJN5nJLJ5qr6IqRtzqNlM17SpFsnkEyTWoABEjz4JXOQvzWYuwdnV5LNGOwTM5v9r4RpQ8ZXsYodks3o31JBlzbYtNotisnm22MxiwGFXam5oN1n0TA/hRvshvTSDwHff4nNzRo9Dum6PaJbMXzDz+x+Fkj4L4bFNBb1asqsgH7Dyh4DvbkPtf5yMDKzEwyoaESMSNS9P9gJVA3/RTlwoMwZvxECFWxIPNw9gi01nOHjP32esZTtmXHnxvZd8ZtakqQ7ekajbXetpNa6ocTVxJtY+uSe69OLz77zh5bDR3xjZMzUz6fxrz1nqrZGcHQHfPVefN+fiK86LeXj+Sc5lPKy+k/vCUI/DaLFYCWHr6nbXuILTIsb5imNKY/rCm28fSMxPhkN1XbNMNZGuqwOBhtTSxWuTk6bw0ZaG86b1hKddePOKuBvmiguYBn4T/yOqOyGRBt7bKUI1GjioBC8aUKwF7Q319UgcmtFGIzCJGBqwQij0ynDsfdFGc3TS3BlNfJ25xmzniMkpXXTPvCaD3ZaZvyzjmZdudBostmhb0ORZNN2sJBeed1HXkrUsywueQH+L0eCPxmsa5ZpgRJSDZ11yDv+jmbd86vxZfc1WcZJ3UkMq1BOOOVtvu/+pB+en186d3GTwWAw2jheaJs09/+LNfZft37DALyrNj1wABMuUKbODyTVnT/KYbJ3Tpq8IrNh92dkxOj5P/YpZx4/ycyiVcDYdn4JbEoKdQi9054iBKsygLW46FRGxAb0NPNCm8BSNCPjoKcj6EAus4SuP3rB+cV99/eTF6294dA8+TK6v74MHVpYNRt/I30e8QGTOOdfGWzzxcy+87a7bLjw37rHw1nPzp0KyyRSeZO+QQhInt3dYgvycjrPOv+T8s1rptaP84VeywdWX2T4ysr0/7TLIs6+x9zib56ye1dM9e/XsZmePY3NDs9zlnNVt4+WgHJbbz3Livg4P9WWgviOMm4kCRT6I8vw0NbUUEnFvOuFKoxQW1gTsvFirsF5pb7qTUCx4i7VmtToveaDxvK9uOaedVvPRpVOnNz0Q6bry7uiSdQ8t7Vy4JQKVS+XPplV2ts4bvCwZu+KzgITtxepaPRzWdpv74muvv6RO0SorX6cu/dqKn/XWnrtp/Zragz13DUCl5myiFW2Ycvb0PtsXnU+tx8pvLFbUspLX68mdegwmOif/NPDONajTGoUh6tU56HBJCTBASVvNUB5VIiKpc9kd7kludodSFz7xQbiOmMk5dOYk56gzL6uaf7N8a6MQOHm0ae6snZpFDfuT3/jdYzjzwkXXIVHoXNuCfQslQZqBZjTsoHMqrkE4jaYdgkGz2ATOgB3cPkSukD01DnV3ttb1wx+6arPqbkcNAHoFPzKUUQ+qL0k97pjbZv1I/egC9zTFbrrlFpNdmea+gIgfWW3wqkcis8ky5FAcRd1If5nNZrl2FFpungc8wpoCl1BpQV/ScS+zjlASyUTVv/AJ46gkJI4bHX4lTnloctxPZE1ckS3+jG2fKIjkQFyzuo8jvYQG1OrGvJPSTu/nSp9PHNTl4z5hK/8gtXVKF6gEKiglgcKiRlCESsQCV5QIlKWKpr34lt/wkSx/JCmP5/cBKQfl/5gd+rOS/+p91/+YCg5CXK2W4M9fu+/6xxX+vnelVuldIDCG0VQTpU9Dw4pRfei+6zWx0MLie0gPbyrkmRU7OwT16JGeyXLHqOLqAfVN1GPlBzWtFNzj0TRTCjogtP1NjIvu5habN5Aoa1k66wGpqriVetJgiGdwDZtKhnN0y4n9sXYnsqGmZfDSR15+5NLBlhoDaedEm7sxmpqRija6ZEEg2EAnTiAC8IrmFbGz1q08P9PSkjl/5bqzYqT9hMmptEXDgTqP3Wiye+sD4Wir4jCeoHbbp5hRfpB7BakUIppIlPCD30dR1GtslDz8OsqbXmejFC/v8wu5X2myq7SJ8Avzv9DFUJySf5uNvq4+Ti7W9D/OZrLChdwxmPNiBRqVjnpK/aGxRCDspVYKAW9AN1JANoo8wP4BJUlGqdgw6m1qPQ2QW3+OfU5/ieLS/NuKpDU3uf8bcAXyBal5jMR2NEAbPAZt0K3hvxHBEDlUxfIGcD+N2gNSNx36nfqlAYow0puatNpRz0e4W2oahKzQHsjf2c16ad/3t2KTtPobnX6D8C8pd0MDP+Kx7wnXqGGlLQcvikMErm6TmfsuxJXbSAxqNjOogJLQBLiKEHAE+JGTS3JoEhTrz8/CB+5YlupJ58aOat8Kv4JvregxwcU5Cp8GFAFm1FyOfto6GS2m1NGTS6CPNKkbsTdCBlnN9onMho55BX8IJZtEQ35lk+htwN5A0V3RCPoD/yXAcv6pAtbZczRUA64JmcUf4q7Q89ZHLeJVZ5D1Ps/t+0iCT3AHVtZC7JDCXfR7OSb/Xja5H3zQbZL1B+ULX1BMTEk3AseSpmnKEK4T9ekMIidUCRQFfcbj7z8gNLvzF7mbhQN8h6ZbRset+nQWdS/ZX3k7WpS8P9sfo0iGS64wV516pOhjI6TZ2dApgI5+LhxywYoWxKUrykKJsIoDsR4mSrCTg0egMPnLW/3Q5Nn8BZEuzqEI7HK3n0+zFmuO3TtWQ5WJoG9YqCD6Gc32SxnbnVPfsxvrFXK2dILl7bLthDp6glhcsfp4bYvbSmj/mQ94uBTw0E73x2jbNRCvC6VL6GCFDwU7eWQDcC5FY5s0slieRDwtAbRsbLXbaXAuu14e2OJw1dc6jQ3ZdY8v7rv2/BWZLqvFWVvvcmwZkK9f5jS4muO9yR5res4kfkRxhV03L1RfPOiPtYi8pd7jNEsOpyTwxpaY/yCZu/Amd5Or9uS3DYaeqVOhH7gZN/8I/wi1fEuLXvyNivibjuKvN+1Nc01HF/3h+ef/sOhox8MPd5SFucPjorQwXT+ytA8EmA5mamHNFDVhBI5pjZbQpugBNkO8MvRub8KVDKST1Wag7D3xlin1ZF7LFP/79nbvCXFOY+PUjrT7/otsPXXZ4exdPzuhZuL5LUXVAn7k7PbhG89uz3b41X01gbjP1xwlu5rrvvf9+pbs6E/Vu7Nk642/PYRaAiUBdrmO6CDTBLPQFA1ur0uXoBR1INDMkypKpoTqnSMx5GiEdTEaSHLs0Alvu/19/5QW9Rv1U1ridT22i+53pzumbs+XFFXYC++CGsTj5JUT/GCgRt3n78i2n71FHG4/u6X++9+raya7os3ZbDmgWfXun44e+u2NZKuGZ0HiF8M4TlMPR+EU6rPKRJ8wOU2RFUFLex3egEsz3YqEAq0cqhAAW19dBZIlVzR61tuIdTnpXH7l+uXrbjPUyep+8cl6aXKWhPHpDcXl9KiTWDNr4mBQc8Tq+NzK/OKSbsfl79o9G20R+brBXYvUg0rLHhtrc4TN81TTOWSZ0gL1ZVlOYH2ery/7XVUjFMbzYpg7UswcqJPQwBd0LKLabJ8IaCr2otcjSkIrGwootKECaUd4XH1+SdazRrfddkBU98t1htvWrbjqSqjaCguxrffM/5zDCpBALUycmajhd+R6ww4SWafuZ5eU+tPid4lgd3gt+b/Y9rQoZNmiXYPXyRHbRs8zX/f4WIFjWZJtUdSD55AP3xtXH+ZipC0EqdBGDA4CoYEU6gRLGPU11QhkLTBiEYPiqOeQgwTCl9aok1Qr5pFf71qEeNxjy/8F0GoqYPv75Yh9j3x4DuJ+uEzHRpAq2lMqb+qfTdiq6kGtzfOWsv0c7lSeMXDHBDe1MT+LUgx0Pg/p87u2UicdIvqQi8DkxhcUwUXCedMpb4NQjwY3npTmgsURJavLwCRyEcN2HfWsDVGfv/u9ZUWUx+PYFueUKwaNvbtu+Xps3eVWbN1GcgVrdMnWJ7WmJz9SD66EBidag0NF1Ukep0t5A7sFCWdhzvYwHv6L/BehXuHqfaBwBEU7hfVLcXvS4VQv+T/vaSIl7cbeMc7ekv9i8S3e1L5xxpvMGcu1EYPbKyCiijjGXcDKckm43PqU2qNWlXusZMiqF82cuVzolUHN9NNR0HZPxFPV9V0wLtvq+k4DqOwVWDlzuQLVdqFiP08cRX7aRlBVfR8cb55bWe5LExnlcsDp1vAP8Q9BucPMk1Ulh4GnN0SAdxcNHv3q9ohx1Ati4S/tkWjIDe3hQdkUGrGRaFBiUdiTSkI41UkMuuQHP+EaSQYlPQTFWJF03BNPpTu5KFAdkWgDukzsZKMG0Q1TAQQglScOaP/dsZ8+fP75D/9Uu5Gs3FY/2SxPld0DHOciXI9gqjcEidXjE+3BLosy0OcX3T7O5g65ROGyzQ2BZs7WbZVnO5ydLe32hMwTQ4wnnKXW6XW5LAa7oaXOIHoUl0FgLQLH2by8wSTWeAx2Y5PDazK3BqZbeJZwXGPaYhX87ZNszoDdaRxotXO1nNlpdvAPFWHDm8PqEE0sZxDEqGzxisFNnuCWetPcGrObN0p23tTZwMuRVodSV8+LTrOV3eRvzjQZiSjaLYS1WEJe0kNsJlZu9LFun7++wW4gRDRbaxw2nrOGm+xOj9cmtbp9ZqeTM1m8UXfQQCSTVSQox6pvtjot/FpHvIUjJovFEoYvHYV9C5Y/xN9OfcalvII37UEhTbTg/AQIaPb4Vz6j5u8/aViycMod/fkDcpu8QZbZoeBi/vbzP3XPsZvOubMtaPHkD9jt6+U2O7vqU/9C9SMvgrXpQNG/E0oJxun+CiElUa0IKQSUwERxOntKSV7ekcuh9VBZBBo3VUcB58ofKBHCwLyf9qFosz9Ibf8dGqwaBMjRig4SGOZ2UkWI7UiO9OfUPdxOYFApUZyfpY7mgEc5rtNGGk2H1lPhAk1Hp/VAMqQEHEUfEYkkUQq1JMdzsX7kklRrTrUi1wMcDjmu1YYfATj7Y+pGpPEBXuoQIj8rR9mgCl4C9yqmF7xnVWxGVniNqtpVmXBvQ6iwni5YQ8a1jYrXtc2J13HvgkvqWxuva1sbr+P2S5ceKGyBwDv2DbrToe1u6BkAJV7xnVLUaq0sJB8pFqcUIPi3yuwxi4JuLr+P30f3OkPQ72aO0xYo3/EsmO3QO5qEF8S0qQH0UsKXv0brnl9+8M7jF174+DsfvPOl1au/RL5/9DsbNnwHL2pHR1NTRxMZhJtHktOOxLxErPF6YlLvpC9YP73x+4ofw+3xVdrHcDE0dQQCmCRgvt9b35xINDf1CDcRSfJ+pYl+Sf8YcurfmXP5F/kj6J82jNsrkWiEuhVlgFfyNkB3S5MUzLhoNiwSCYcxQ7Ui4J0Xh7fmqRbaPa1tzujxkBRlsEHy0/OM4pYLPb7g9O6BQJN6l9zQ0OGyCaZz0vMTbHOzXfQ7a2tsterTcqxeInODoemdktw+1SbVhKwtW9ffe8VKadK0OVuC3bWzyKm5LeddsWTeorWyY9IMtUFutdu5g+Rn533qkocdvLs2HmhU75br/MmWtD8zA3OP2t1ea636jEzqYxJZGAwFiDEd61oTsrRuW3/3pYNi3bS+Rd+GjOfVpAPNd6y64Gsz1GaZleWIPoYL/v9mTeQBENVEguiF1aC4YeXxFETw6QyPfn0m9g8IrMFAvKM1EI11DARnbqibHk/Iojy5rSdgCyZi06y8sS024PeuO4MfwQ5Y9yKRZCqyYaF30vzeHlmUprR21tR0t0yz8KZY66zWuGvxVQB/36kP+K38t2Hu6NQ9SFJfw0AdpqPEK2qTMpf2VCqJwqPoJezTL824b8akoL+x03nhh+oNo5e77psxg9Q5LzebIKD+fsY34f2MtB9fk9v5b8PT6tYrgv4kRPwd0q9z3gdJSJ0653KjCYPwCaR5aUY63eW48O/kdo33yxX9wCiMv2QTrk8eGSI6Ag6moG9t2P/F7GRNlDjl0gw7pJ5aOXXqyqn8SENnXBmbSwUYLyqJjv3UmY1nKr4t80no0faXsaIEiF/BRaIBnItSce4OUif7W6Vm9T9H1X9Vj71BEm+RdmIJQST/ZfVdudUvh9S/qqNvqT98g9SQ3lHibZY0mRVHooyDN/FHmTgzjdozKw28NwQ0hwN6BCoPKaEk3YtKwNhwRLXuk076CGoZNXDQcRwZvreTZY9EZi+d0s4+ztv8iei04JQl6ZbDD2eHV7X4uHuFVfPrOmcs6m6Kr7hssr+1VZFcEZ/PdJkn1hOs8SXS/NFFgqt94PIZzZ3tdaL6Q5vo6piSzdy737pwsX1VyxUrF15iJ4uNkq+rbyg1Z+O8VsNC1UmcvORPRfxtPrfRwL2p/oA1eZp6Z/aGffoewaXcA/xBlKlQLfhQL/oPgBGP3qsA7IQS8qDVNswHKRSheDUvA3Q7MZoRcJMxlEygujn1QdyzfPfq3dEp/bXh5e5YXW2Ngfvza0ZF6UgFL/E0fTq4LBlvTE2qb/KuuzYSXVnjTfM1osvqMHVbm9950quIZlbqaL6YP7jk3kUtA0GnX2nvq53f3WoSsvEdDRnULgo2fN7lNZJgI8/VWi33c3bBZnGY05+dm+3qc7fNmj4YGKLj2nfqFP+g7jdDlxEV5XsJQZP6hYrS1l0VQr4c69Xueixp90gnZPmE5OF22j+SYEWHlZ0K/Hgsh/Ztsbh6h2DNRlvv6jJh9XaJaHCZDiUDKNTMkvb8vsqCyf3ZNdSmO0fa0Y4baJTtpbKzuVzeeSI7fCKr2Z0WypapnXJ4gnoWy3PoUIlIQ1TXdqhQJIXp9Wx5fYdpeWh2TY5D+YVyKd0jw3iumwi/BC3cEy4o83QlZnW79MrCgCjbhWXBlRZVVZZv4rIKpXC01HFlHdHLoeWVl6UVc/J5uGm6CViW5mulYMk+HqNYr0AyUPivLg2oMs2MPqtuhHyRyiwvNJej1Br+fcLyoAyu8D9B7bgmzUqfFobF5nKnK4+t8MPJkI/xHUNWk117jugWF+xazTAALQn6+UE9lhoI5ApGA/iuJOsrlNP28SVVuBVajXmircLel46w2bJS1Q0Ft0KDuikDFL/3pYrid1Q4FvofwRIo4R9h2ftSwc6jHAMqLcCql8YPHtlzGoByNXYN6v8hXnRaOhUvx0sVLCexwupGDR4NOYC7PePa5keIPACnuAdD7dEadRuTIiS6Lb7uskb381My5yjzF8lGCjBRqdwrWJCagfB3yCy7XT1i92hbcZ5Ci1FJkgYMDf6n+jspIsHFjJrTOdzSMuOa9DbDcj/nH9N9bIoGVgzHPWIQuFuYtaMRaq8eCKI0gEF6lPOZjBz3EEvaaxwSUT9U/8JbJZPJJLBLolH1La/RbF9AbC8JJjv/mMnssKjLRBJyqj9QXxNko0Ux/X79epfiXkm6fmKwF/en1HLc6LxloXWKvGa5rVCVL83VuiPcDEX/K5pTXOxHfx6HHB0t2FI0qI2rCZFTrvPWU67zVuS/kTsLnc7IKhFg30e4FOkqNSfH5PtkmUy6Cpiv/36k2sbqCeCFNa+URpoY0sZoYmCgCr3qgZz6s8I0gP1bYiR+D79H56NOz0EVWCTy2/fffvSCCx59W7uRV9995eqrX8GLesOXNm360iZ+T/El3uZqL+FyzSZ8XxpTiI/G0nkT4zznFZ0t4ipMz5v4q9ssqbdKUZt6u82knPCrt6PZwsnn0XySVnyPR1ZXAn72yx48bWJsu7apnI3Hy8bygUK5Js32qcytapqgmn95uexccj205vGgJ+euOeG2SORmKZr/qKzcx9SFctMJdwMUFZDJITs7dnOp1EKZCxg304Cevyfya+vlKqv6aXK1qIj3imL+L6hL+yvUlFfE0VKZ7E8gBY3M/8VoJCFgizH1W6VyC76nH6b7jiibYVxUmVIEspry/LgZIlCeP11Z4zs/AwvVwtGFEut5S1JY4lfyT0N/evOLo+rUEgjcqc9IkGpQbv3iW7Co5b+KgjvpzYdH85PLcc4X21ouwEGl/S4qnUAvoSlXUUhR1eKr2VWFTB+GMl6FsiQsVD1R3urlAAIoSn7JQkmiVVCHSpCwDH/qPepXQ0Db77CJOAImohB+RPWr31ev5g/kE+zTa4lbvZo8xdWPffQu9yJTPCNB66s+zXoJt/0L6hSoCuBIoK8fnBGG87OoRckJpLqyWe4YbpGi50g0+3I3UD85Oa0fzubfoXxPLbW3FDWzigmyJeM0tQkax7PqTy80+UxfUHPlBZIRVNQ+v0xRm8REKPoLmNr0+Uo48v9GFbXPKylqQ2IKm00QddgyWGMROCTxdLB9nCY8P7j2DjlsV/+mfr0C0r/NkeXbbpPlOTBBwT0mVz1zx9S/wJecBF9Wgv3p032iP2v4VSgfgW2G+HUEdEXU6iq4CtpLJfIN9XQG8dwa1VoO8XC2SrPDDyCOQptXgbcPvlAgBfxBoGwftQKeKFrNTASPt3pGGqDt/QRasn2kri+H6L80MJRsmVYJrAKyDItpJUy3/15WYIJqcJ9Q5N/LFJ4c3dc1URpWl9hW6mu50MUIelg4ucTPf15zs5DFo1c0VSp1tKB9jkwIyuM45kb+IP8gHed+6jO3v0KbIknzLy636E8KPTdCuUpB0wLo9JKnAO6pv0vS31EtBha/fJemkgLVVnd8KCk4qBTpQ5m7FbifBKrPJcq0pZAFVG/XbOFz+Tcq2MLrcmV28Nmi/OHskh82bau0k8eWCaPijQPWQ5lUvslwVCfHkXBMIehqUgtDNLeauH1huvZTbYmw+luPjyWoNGEuxRLR7LK5fSyXFUyK7PURQv2v8D3XOt2NJ6liBbmPGOsakw1kbeOs+31Wm5qpH+iJWSzqdPr2O7zc2TmtnrzCig6bBd/vgQmzOlz0STWIlmZEQfupogOZFHUZ7EkUnMn0RrpIMqAgHRJAOjIJ3yGw1I/MAp9q9S3Q/clADNm1wEeO+xbwg5OIYHZLY3ehG5lJk2xhco+6JWybpEVz2wrR6hZyD0QXZbeDVB+onmlimpkWprdAs4WEZDSQppsDlcdCBJJESIYFuAtUnC4GIF2C3Uu2Kv7L1bdz6FxtqxpG4TqQOqOUNAJ2HLvPWA2GgDy4O4vaDrtyl6P+1fAll+SyFcQ28GHqh7fvvf37udylf0fNwhzgz87Y+cf5x9GnF6ygHu18sAbipWeF0YPBgp2GaKeQduxxdEr3SgbH1kvH7tvqSLhedomOvZyts2dw8acu3dY/f+ucuMtCuP/e4zC4XnH3OLZ8ZuxTWxy8dJfU5dhDeKPSlJy5pn/+7u3XrJhmr9C5CuleGflGQocKnlAUaRKp0BAHV0ZwUt9VCqk6zYOgRIuMfePJzdmBdpPJ7/6B23+f+sp9NMDZevovvfYHG5dGPISQq1DojqNckchVrCcCYz/Q0hI0m3NKDRfkgsrnamo+p0CAq1FyvC3a3Nak/s5VX282x9Ufy3E39VAx6o7LpCvO2wK+ch9jNqpJCutcIOooKnYWtDK8gTRVYygRQfwgzKM5+jP2jOZdx3r32Py7rQUPOzAnoRs95NvRAR0qLGU11Taqu1bUYSzMcWjMEir067JQQHfIrLBHsrgv00/Wavd8HRLMEEYFSW3HCSNQehnrHztKqHcDyo4VfZ6gPKCR+gufwA8GegxUEo4A+gd0BASHiH6jYMLIsUdQJTs/C641KN4oCHWolCMLlMfIdtWKScjx7SM5LD9HnfmhrGI0S139UWfUnxgOXdJFW+AMcGjKr6eHAttHF5sUoeArYKDcxMSYcKA/xUDhPiEOEAPafSIUFArN0r24ynI91EPARDXvIDYyvqZaWeroBOUABQA/E+DXC7PWafDLQY2oiwpUEyj4RQtVlUp1GrM7In2p2A7VuiOW6otMiGOo5Mrp05ejVuTy6dNX/k/7mybZQ0nUmfrbx3U4KueDnlHm5wdh8FFeKnoaKKh/TK18StOPhwG9Xo5mqXAxvw/79YQwwDR+nAKQQ4izVXioB84qcppWB7IqjU45z4CE17OvF1Dw+oTFqxtz8dxwtogBnF9MjIl/in+K8s3hM9laIn0TiCbTAXL0T798bPXqx36p3chrv0O+GC9Xaj48Ecv8U8UEeBvUEsDlTepiU5OvlpeNGvpnKF0RvUooWhIjnx6GeBapXCQYTw9DNg6/OC3gZjp76oNTj9Kz6Jqobxb9NDqc08vcKReOpcsQV2K8InXFaXW3aI6Ofr1k48rp7CX7rx+v1UKPsfvzQU0Kc83i2VdILmd2/yX55zT9luN2+Cu4nKfwPcK/CvDVU+pHh8+LaldIf1fA5h3ndT6Fln9/W/9Ce1vndfvJtnPVO2xhm3qbafHVCN1X363UXHq9xuVD8OSD29Z8pZ5cZrern9cAdGW/uib/ud+VK0L9a42r6C90kL8KzxwLQw9NkIQJL0ASU8M+VG0KsUdgdvpgP/6NqqP0/gHZFUfGEijZLHpiIgvV5/Bltrj8Qd7XQd5p4P+7tJo30NMO6VGBwahSPMYiaaBYoLY6uEnciyhhh1Z/vvacG/rjpsvnpzs0B1Id6fmX8119l88XnOxe/uGrzzHcdu7UtY3+2vmXN5zUyj3ZcPl8p1sZSs6/nGXtwrV7Ka0XZdz83fwjjINpZWYw85lL8BRK4nGyIir2RiOsEyipuEcIakpGjWgBjLiHWOgj0Yi34gW1kKPxHt2Na5q+lwg1RdRSpFDNzosb44YJXnAfoEOpZW//6u1lhYA6leevezbI26zNHO811M2dc5HFxpk4i1jPC0s21/BWW5DnPQbn2X1WK43/aM2n18DfSoybbNHijFpamzXI31eRibGUOxSu/lT96YZlq1Yt20DaSBuG6knw2eusHs5EPBfNmVvHKdaQzcDfz9ZsXmLDWGXy2U5OsYSsIn8CS12jQIyD12KKqZrLPy7mSPdICmd6WGHG8NDZkkHuE4h9TU8FpmUO/VjC/EinToFyoNDz2p9XD6g78WgQdPG7Z3R0T/Z5dTM9lsL8Ktek7szl2L+gQwGgwkZHc2g5Su7NvVqwGy2Ua4KSXUwt1X4PaM5paaEu6jQ5zVFyNabxvUksVt2T/4VeamYPlLtffdQsk+2sUTY/zDXl/05W53/Bz9UK3p7LjapZ2ZxOm+UlZXrL3HHGqO8+wVroDaCTTnTxitMxmiAAYQzVJQH+nj3oIHnPaN6Zq6sNSLjBl8tKgVr2mj/9CWi9dnKca8rBQBsd5R1tzVlgrl5pbnPw6kZclCr2CHxMnHohLz+3KRQokzALyeIKFU1TNCiayJdoHvDYe7K6mZLm8S3uJ9dojuaJ62/qN/tjQxnSnhnKPw+LNrLi8ZKyJ3x1YhiI1aNAtP6NzCGzYv3DmaGh/LvQZnt0evgIhTFV0kE/PYxAnOHhCQUZdCWY5JWJwMzlAGl1mpNbDU7yyGnhRMILsYhH3VRAijrPcBU8/Cj1Y9NY6cnGVW0CjTLaz7E3epvaT/LtTV72Rs+0WVVmd0dz/MGTI5F0OsIviaqDlbbO5X6xT3PeXbXHRtf/z+fdka+eKPr8KF7IF4vBsT9MFPuPJMBTBMq9hQxXelQ+bewnf18ap4Ib+mSMrtDU5zqlD8QANa5MBGh/OwOvSDfcV2d66mfEWsbGWmIz6nsyZDWQSmqmxDneYyvjHPmRXHZxeueyRGLZzvRioKnGto9nIPkibAJA16adcOZRQr1iAP3bUyBR7T4RgAWTKxhkCYFwshq+7iV9r0whk50cmRcTg4fy5x4OmmNkHndIA2+YuMbmE9dwGYB4KFTsvnDE6Ah47r/fE3AYI+oXADpkdlENcZ8OZEEf8FFGZNxMs6ZLpG3SUFLL7Q2kcFU/A/Jsw+vWDa/7emewLaoeibaF1B9qUNnuqWK3+UfXYVL1v/omD15xxeDkPnXTOKSVcCbDGtOu0YQNpGAP7U1HU58UrqGu8xIbHtkQ3LVhb7Dx46ET3Ffcm1q0YcOizNmf3bC3VjWfAcpSv3MyTlgJ23FHQgmgvk+gk8pL0mcCDOn08MDAQlf+/SlTZ1z12fnqntOhbOTL9/ZdevbAPN+yby1f/uUtC/ixm8ZBo59LTXEW060hGrTDplNprWd58fwB/b/E27BdS/s7U+rGVCeQ46nzaw9QccnmZerGZZs3Yw9aVHt+Kh6HN4ti6lxIhT/wahnZtWwzlY9QHQ2c79C+dxzvVDKy8GqKWQERO9YAKbpsDUTLdWV5dE8PVPjvj9pqw7ah/PFVtkit7aj6G5xY9mfJrCz1j1e0BcnPol4UjtrCdbahIVtd2HaURujnFJR8CuOuUUfhrGhgKKgjCYNSvCc1WKlEp8wHUaAYynFNyzZn+2MnYv36dbMDBTonl/T/ma5IKAyEGz+4eRnVtaX6tss2o34u8mWorFtuFgm4A6qK/yp/gLEBVat5WnPDdKA574ubuFJ/IUfZ/Y2Nt6mN+ZNNTSTaeI56gKwkXerTe9DDHUw8/H35FY3nNN7GGuBKWhrV9ep+0k1WjNWVaHkW1yA+QHWNu8rtBw2a5YXuE40rs7/GA+j09V3hA98yRnFPOGr8ltGlsFdD/7tRce3LH6Trcneuiy7K7J3khKu+3qUaXPWaX7T6/Kfj9BX2eZq2XAcZT79u1ClJzUtHUqfqSMWBcZS43Ena0cUGLgpkKxB1QM+0Fxz10wgg6r5rltnFpH05pepUq3Y2HfYqeKRntmUFNz+XmcOs1H31U6cC6RTVLfCg7RNBF1UF2/wBgu0fFQtPEU1sSg3VcNsR7dWq3af87tUFn1l3ltXpaJxpNvtcZkH2WmMst3JqRpxUH+WC0E1qOGtP66s1MYv+VLu8/XFXvV/ZbunYYBeVN64ls0ur6NzpV9xzlmQwB5qC4Tq70WC0tk8dWJXeHvkD0h9zJOM0vD86/1NJMaIAolctvlByferCsqOKDKceOfUu1PsmoFCamV5mCrMUOCi6V6FJosMF22AcrKJgQDVhfYh6tepp/lYgvnCEAbJQ1L0rOpajEmRcasMiPfxhgGoVo4rwreQpV6fUJHH2e8fa1s2c13Apl1b89a58ozdoap2sjgLN9uISl7P1DrulyeIkt0zr6JjWocoPOZsaXPb6jtqBblsgsaRre2xHi4nELm0MhG1+x1SXwLpFi53b+aHRYo/IrbZtuWAKu5cSEXfybnnmUCaXGTpQr0xK2O2WWY76f+nAjNVf7nCZHU5XqIkTnpt6VtvsFlPXg1031g/VRdpkkyVpD7jnmax88QwDvg/66NnMRdRXTcGTmQc3cuINwN5IQqi0yzb+YFVHuVqI5s4ADfg5oE4ybDLd28mFSFmYvRoomsWXEdLU2Wl3GJy93ZNb/d5gqmNaqJZSO1l6PVRy0nZIj/45EetjLguh1rLqR+SK0hO6NrsqcNX8zoUdjQYDJ7tb4os6+i+Y0qpY2AWlnLRDWdGFTfGY1gV0zNAtJ7pdo24se0D88AwLY/gZmE9iuP4V5v7CSR/RThaHLh+UeBkXwU6BC7lGOevK65udTv+tS/PfW7qj3ljTcj3b9OkbV85t8xsMj7Ddj7DGpthZKwKPvso/c/1K9aLE12fMWLV1y1D9ua8lyJdWXr/bG+noCFutf/mLILe39ITUV4igr3876fpX5g2zeB52sWnIL4fXHlgeUzOx5QfIvJQyrKQE9wHUqVq+PEaOrz0wVvNbJZVSfsuMzxN4l9PkedFzw9V5Dj+nzpgoT4ZxCxJfC5RWLc74YVHxKlExCYt0JAOMatREhHBSCAtSfod6x6Ls8HCWECLwXZ9nd5Dz1T24JUdWs6fU3++fcnT49Qe+kBs+wdsMZgPXMp3U5S958snPP/EE7bvkOPCuTUDTUQ/UzirLhML9yPahoe1D5Fj5jWsaoveyP00PehdUAHk/seDVWsvDWXXXsyn/4wfpXc2V3/Qxli3jl/5hj/83avSCfpTNxOEKLmTjxOEKuxgNlsQn0xgct724mhynupNW1Ph6o3RYS3/+2TJrzLlkFz+ip3qCHKf6eqW02QJLjBYuuj4sobhCWqa/YHGEHpcnumuWSOhxeaL7sOakNR6vvmo+YcfFA8UFXEPZf9UjyudIOyNwx/i90DdsujS/FX2UAwvWSVK4NxaMhAGw3oowp/uc8CTi7D2rBgZWwb/60faR7SPsEbjkXy4G0XaqhXPwe2cePjxjxuHD6ssQuR1fq6PF0E+o2t1nePTn8TUmxz/A3crMoCc7egESuoTHYc7mYdg6etORoOhR7BBGD+qJopELrl4S6cJNRtEAsLP/OdvnJq0Wo0GolY2Et9VFB2Kf+4bZvVyxfOMz3WdFfSIryj6DwWghre7aQbdiDrkTL3A3vNDuDpk93HqXwam+bWmUJZfNn5ozKV5Pmmq8PF/jVY+2Tlk2M2RzSXKjmbQ4RZcQavEYrN/9rlXwtIQqzxQNMzPPfHYLvuPoO9TbT8bpGw5CQPGd+SyX/Cyf0Vxjd2R9NmsunnXYa8xGHzn+sSfM5J0y0DZEXWWxkXjcR75KBLNLHi7XvX2G8VOrf4Ykg0AMdBESIpo7MgAfyakA6rkqpI6UjNs0px7cMV+D5BF49Tez1VGnYmq0WIijp985m4Sn2gJR9b07riPPFo97OYbUZbxJCpot7H/lpZBicglCPN7WOfJkcHqc3ElWqvvz/1E6bIQrG+tz6WkM1SM9FBTR7FSs8KyBBytSmNEoquJNFN5EQyTiCrnKDx1h58yxCepPHU5nxGoxEQeeOZi2m80DxNxncVhr6BmEfUarxejw+WSiHhWk19bSY7aKR5MsteblJpfTLtjimBouXsm3d3djjYM+wEW0El9dM/ueVRWIsXwe43R7SgbVZqrnqoJ1X/kuF7pcgf8duv4q6vayV5U9zMV91GxO59UUjW8rHV6u799WzKMT7umRCXbYUKM+foaCcwgaoqZUtmodV3p+X7akb4dnU9B9La38RPFUG2SCC90tVA4XwEFhyOpZZrUCsgWYHsczLFBBVGNtstoN1bw0Z+O4fYIbvZVt4EUcJEKOhHeincWqONw+q6w5Go+WGOSR7LhKV+KBqbBPpfUvOf9QqkpDyVhBeyyZQGMsdA5FBUqvFMtUyGq9vjnsAJU4UcrxldP1CCaofyDkSAifoP5QwWx+SyUGxp75BzGAvtG7uQ38LehlyEQMeh0TeE6Bm7tYdXqdkt0uOb3kfYlNwmOdDyacOq/qlFo1v+PTmTi3E/glC9W11b34A22zmLzvb231Q0L2Bgg60OTW4YdstO+YOJnO38TtpH7zy9ymokWyA79qlVSn38HtpFlImFnhu3b4boNWXklOXV0Iwo7lQ1hrZyPFcwtjwFP7iEKSHSSJw509kh8kj6pr+H1jR7km9vcvqN9657vffefkv+fKxge1X+7RdjYUPIESN7gTvRkB/RMYtEkaVkdHApmdBPpnKmz0n1xSWFOyVIuLrinZwpoCRe6kyiVZoHX088F+UX4+WKS4iBTP0IWxGtZgOdMaV4KTayqHQF/VihBwTbgDXTCmKoOBJeNhwJMzEVjtjIFLuU38fPR7hqNG1JS7g/qRCuy3vmQ3W9Vu8qbVbP+SzazGRJH83MzP90Ck2m31mMjP8TiLn5uwD2Ugr2PFvPQjB5BnSJvQxGQZZEB+LopqzGzDbMmbkAPkZVJjeO5FzOSBKCgJze2ZS4Gemc9twrwY6u9H61iUQTcRvtdT9RW3tRxAWwFs2tcuJRnI6xjmBdWjbgFNRHMHiF1uHYBfUR/ut5Ug2jXAaT96+9RH/FToRwIzGbKmVJ1AZQnoabSB1yyIg7ByAridHApPMjyw0OiV6RjSbCuzwLAvFizBliWJua1tsuAgvNPbmljYbpt8lkWam7b3XZiOiKJskMOtmfScnsbPW208knwjuXrXK4Q1iKIgNyYXXDVT9C2Ye/78GQ5BEEXfFdde2RwauOysdJNL5AzCy84ard/nGAVN8alecnFdgu5Gbd5DJTL+hHZK0vApVy3OfU8XTSJg1TlssivsPYUlIqvn66PzrVTymCc4wgF6SDNR0pDf+9Gp+VnsUH5WtpHYsuhOaey8zdwLN47V8MTbm78g687+P3cx6tcAeNpjYGRgYGBk8s0/zBIfz2/zlUGeZQNQhOFCWfF0GP0/8P8c1jusIkAuBwMTSBQAYwQM6HjaY2BkYGAV+d8KJgP/XWG9wwAUQQGLAYqPBl942n1TvUoDQRCe1VM8kWARjNrZGIurBAsRBIuA2vkAFsJiKTYW4guIjT5ARMgTxCLoA1hcb5OgDyGHrY7f7M65e8fpLF++2W/nZ2eTmGfaIJi5I0qGDlZZcD51QzTTJirZPAI9JIwVA+wT8L5nOdMaV0AuMJ+icRHq8of6LSD18fzq8ds7xjpwBnQiSI9V5QVl6NwPvgM15NXn/AtWZyj3W0HjEXitOc/dIdbetPdFTZ+P6t+X7xU0/k6GJtOe1/B3arN0/pmz1J4UZc+D6ExwjD7vioeGd5HvhvU+R+DZcGZ6YBPNfAi0G97iBPwFXqph2cW8+D7kjMfwtinHb6kLb6Wygk3cZytSEoptGrlScdHtLPeri1JKueACMZfU1ViJG1Sq5E43dIt7SZZFl1zuRhb/GOs44xFVDbrJzB5tYs35OmaXTrEmkv0DajnMWQB42mNgYNCCwk0MLxheMPrhgUuY2JiUmOqY2pjWMD1hdmPOY+5hPsLCwWLEksSyiOUOawzrLrYiti/sCuxJ7Kc45DiSOPZxmnG2cG7jvMelweXDNYXrEbcBdxf3KR4OngheLd443g18fHwZfFv4NfiX8T8TEBIIEZggsEpQS7BMcJsQl5CFUI3QAWEp4RLhCyJaIldEbURXiJ4RYxEzE0sQ2yD2TzxIfJkEk4SeRJbENIkNEg8k/klqSGZITpE8InlL8p2UmVSG1A6pb9Jx0ltkjGSmyDySlZF1kc2RnSK7R/aZnJ5cmdwB+ST5SwpuCvsUjRTLFHcoOShNU9qhzKespGyhXKV8SPmBCpOKgUqcyjSVR6omqgmqe9RE1OrUnqkHqO9R/6FholGgsUZzgeYZLTUtL60WbS7tKh0OnQydXTpvdGV0O3S/6Gnopekt0ruhz6fvpl+nv0n/h4GdQYvBJUMhwwTDdYYvjFSM4oxmGd0zVjK2M84w3mYiYZJgssLkkqmO6TzTF2Z2ZjVmd8ylzP3MJ5lfsRCwcLJoszhhyWXpZdlhecZKxirHapbVPesF1ndsJGwCbBbZ/LA1sn1jZ2XXY3fFXsM+z36V/S8HD4cGh2OOTI51ThJOK5zeOUs4OzmXOS9wPuUi4JLgss7lm2uU6zY3NrcSty1u39zN3Mvct7l/8xDzMPLw88jyaPM44ynkaeEZ59niucqLyUvPKwgAn3OqOQAAAQAAARcApwARAAAAAAACAAAAAQABAAAAQAAuAAAAAHjarZK9TgJBEMf/d6CRaAyRhMLqCgsbL4ciglTGRPEjSiSKlnLycXJ86CEniU/hM9jYWPgIFkYfwd6nsDD+d1mBIIUx3mZnfzs3MzszuwDCeIYG8UUwQxmAFgxxPeeuyxrmcaNYxzTuFAewi0fFQSTxqXgM11pC8TgS2oPiCUS1d8Uh8ofiSczpYcVT5LjiCPlY8Qui+ncOr7D02y6/BTCrP/m+b5bdTrPi2I26Z9qNGtbRQBMdXMJBGRW0YOCecxEWYoiTCvxrYBunqHPdoX2bLOyrMKlZg8thDETw5K7Itci1TXlGy0124QRZZLDFU/exhxztMozlosTpMH6ZPge0L+OKGnFKjJ4WRwppHPL0PP3SI2P9jLQwFOu3GRhDfkeyDo//G7IHgzllZQxLdquvrdCyBVvat3seJlYo06gxapUxhU2JWnFygR03sSxnEkvcpf5Y5eibGq315TDp7fKWm8zbUVl71Aqq/ZtNnlkWmLnQtno9ycvXYbA6W2pF3aKfCayyC0Ja7Fr/PW70/HO4YM0OKxFvzf0C1MyPjwAAeNpt1VWUU2cYRuHsgxenQt1d8/3JOUnqAyR1d/cCLQVKO22pu7tQd3d3d3d3d3cXmGzumrWy3pWLs/NdPDMpZaWu1783l1Lpf14MnfzO6FbqVupfGkD30iR60JNe9KYP09CXfvRnAAMZxGCGMG3pW6ZjemZgKDMyEzMzC7MyG7MzB3MyF3MzD/MyH/OzAAuyEAuzCIuyGIuzBGWCRIUqOQU16jRYkqVYmmVYluVYng6GMZwRNGmxAiuyEiuzCquyGquzBmuyFmuzDuuyHuuzARuyERuzCZuyGZuzBVuyFVuzDduyHdszklGMZgd2ZAw7MZZxjGdnJrALu9LJbuzOHkxkT/Zib/ZhX/Zjfw7gQA7iYA7hUA7jcI7gSI7iaI7hWI7jeE7gRE7iZE5hEqdyGqdzBmdyFmdzDudyHudzARdyERdzCZdyGZdzBVdyFVdzDddyHddzAzdyEzdzC7dyG7dzB3dyF3dzD/dyH/fzAA/yEA/zCI/yGI/zBE/yFE/zDM/yHM/zAi/yEi/zCq/yGq/zBm/yFm/zDu/yHu/zAR/yER/zCZ/yGZ/zBV/yFV/zDd/yHd/zAz/yEz/zC7/yG7/zB3/yF3/zD/9mpYwsy7pl3bMeWc+sV9Y765NNk/XN+mX9swHZwGxQNjgb0nPkmInjR0V7Uq/OsaPL5Y7ylE3l8tQNN7kVt+rmbuHW3LrbcDvam1rtzVvdm50TxrU/DBvRtZUY1rV5a3jXFn550Wo/XDNWK3dFmh7X9LimxzU9qulRTY9qelTTo5rlKLt2wk7YiaprL+yFvbAX9pK9ZC/ZS/aSvWQv2Uv2kr1kr2KvYq9ir2KvYq9ir2KvYq9ir2Kvaq9qr2qvaq9qr2qvaq9qr2qvai+3l9vL7eX2cnu5vdxebi+3l9sr7BV2CjuFncJOYaewU9gp7NTs1LyrZq9mr2avZq9mr2avZq9mr26vbq9ur26vbq9ur26vbq9ur26vYa9hr2GvYa9hr2GvYa/R7oXuQ/eh+2j/UU7e3C3cqc/V3fYdof/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D92H7kP3ofvQfeg+dB+6D92H7kP3ofvQfRT29B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6j6nuG3Ya7U5q/0hN3nCTW3Grbu4Wrs/rP+k/6T/pP+k/6T/pP+k+6T7pPek86TzpPOk86TzpOuk66TrpOuk66TrpOlWmPu/36zrpOuk66TrpOuk66TrpOvl/Pek76TvpO+k76TvpO+k76TvpO+k76TvpO7V9t+qtVs/OaOURU6bo6PgPt6rZbwAAAAABVFDDFwAA) format('woff'),url(data:application/x-font-truetype;base64,AAEAAAAPAIAAAwBwRkZUTW0ql9wAAAD8AAAAHEdERUYBRAAEAAABGAAAACBPUy8yZ7lriQAAATgAAABgY21hcNqt44EAAAGYAAAGcmN2dCAAKAL4AAAIDAAAAARnYXNw//8AAwAACBAAAAAIZ2x5Zn1dwm8AAAgYAACUpGhlYWQFTS/YAACcvAAAADZoaGVhCkQEEQAAnPQAAAAkaG10eNLHIGAAAJ0YAAADdGxvY2Fv+5XOAACgjAAAAjBtYXhwAWoA2AAAorwAAAAgbmFtZbMsoJsAAKLcAAADonBvc3S6o+U1AACmgAAACtF3ZWJmwxhUUAAAsVQAAAAGAAAAAQAAAADMPaLPAAAAANB2gXUAAAAA0HZzlwABAAAADgAAABgAAAAAAAIAAQABARYAAQAEAAAAAgAAAAMEiwGQAAUABAMMAtAAAABaAwwC0AAAAaQAMgK4AAAAAAUAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAFVLV04AQAAg//8DwP8QAAAFFAB7AAAAAQAAAAAAAAAAAAAAIAABAAAABQAAAAMAAAAsAAAACgAAAdwAAQAAAAAEaAADAAEAAAAsAAMACgAAAdwABAGwAAAAaABAAAUAKAAgACsAoAClIAogLyBfIKwgvSISIxsl/CYBJvonCScP4APgCeAZ4CngOeBJ4FngYOBp4HngieCX4QnhGeEp4TnhRuFJ4VnhaeF54YnhleGZ4gbiCeIW4hniIeIn4jniSeJZ4mD4////AAAAIAAqAKAApSAAIC8gXyCsIL0iEiMbJfwmASb6JwknD+AB4AXgEOAg4DDgQOBQ4GDgYuBw4IDgkOEB4RDhIOEw4UDhSOFQ4WDhcOGA4ZDhl+IA4gniEOIY4iHiI+Iw4kDiUOJg+P/////j/9r/Zv9i4Ajf5N+132nfWd4F3P3aHdoZ2SHZE9kOIB0gHCAWIBAgCiAEH/4f+B/3H/Ef6x/lH3wfdh9wH2ofZB9jH10fVx9RH0sfRR9EHt4e3B7WHtUezh7NHsUevx65HrMIFQABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAAAAAACjAAAAAAAAAA1AAAAIAAAACAAAAADAAAAKgAAACsAAAAEAAAAoAAAAKAAAAAGAAAApQAAAKUAAAAHAAAgAAAAIAoAAAAIAAAgLwAAIC8AAAATAAAgXwAAIF8AAAAUAAAgrAAAIKwAAAAVAAAgvQAAIL0AAAAWAAAiEgAAIhIAAAAXAAAjGwAAIxsAAAAYAAAl/AAAJfwAAAAZAAAmAQAAJgEAAAAaAAAm+gAAJvoAAAAbAAAnCQAAJwkAAAAcAAAnDwAAJw8AAAAdAADgAQAA4AMAAAAeAADgBQAA4AkAAAAhAADgEAAA4BkAAAAmAADgIAAA4CkAAAAwAADgMAAA4DkAAAA6AADgQAAA4EkAAABEAADgUAAA4FkAAABOAADgYAAA4GAAAABYAADgYgAA4GkAAABZAADgcAAA4HkAAABhAADggAAA4IkAAABrAADgkAAA4JcAAAB1AADhAQAA4QkAAAB9AADhEAAA4RkAAACGAADhIAAA4SkAAACQAADhMAAA4TkAAACaAADhQAAA4UYAAACkAADhSAAA4UkAAACrAADhUAAA4VkAAACtAADhYAAA4WkAAAC3AADhcAAA4XkAAADBAADhgAAA4YkAAADLAADhkAAA4ZUAAADVAADhlwAA4ZkAAADbAADiAAAA4gYAAADeAADiCQAA4gkAAADlAADiEAAA4hYAAADmAADiGAAA4hkAAADtAADiIQAA4iEAAADvAADiIwAA4icAAADwAADiMAAA4jkAAAD1AADiQAAA4kkAAAD/AADiUAAA4lkAAAEJAADiYAAA4mAAAAETAAD4/wAA+P8AAAEUAAH1EQAB9REAAAEVAAH2qgAB9qoAAAEWAAYCCgAAAAABAAABAAAAAAAAAAAAAAAAAAAAAQACAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAEAAUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKAL4AAAAAf//AAIAAgAoAAABaAMgAAMABwAusQEALzyyBwQA7TKxBgXcPLIDAgDtMgCxAwAvPLIFBADtMrIHBgH8PLIBAgDtMjMRIRElMxEjKAFA/ujw8AMg/OAoAtAAAQBkAGQETARMAFsAAAEyFh8BHgEdATc+AR8BFgYPATMyFhcWFRQGDwEOASsBFx4BDwEGJi8BFRQGBwYjIiYvAS4BPQEHDgEvASY2PwEjIiYnJjU0Nj8BPgE7AScuAT8BNhYfATU0Njc2AlgPJgsLCg+eBxYIagcCB57gChECBgMCAQIRCuCeBwIHaggWB54PCikiDyYLCwoPngcWCGoHAgee4AoRAgYDAgECEQrgngcCB2oIFgeeDwopBEwDAgECEQrgngcCB2oIFgeeDwopIg8mCwsKD54HFghqBwIHnuAKEQIGAwIBAhEK4J4HAgdqCBYHng8KKSIPJgsLCg+eBxYIagcCB57gChECBgAAAAABAAAAAARMBEwAIwAAATMyFhURITIWHQEUBiMhERQGKwEiJjURISImPQE0NjMhETQ2AcLIFR0BXhUdHRX+oh0VyBUd/qIVHR0VAV4dBEwdFf6iHRXIFR3+ohUdHRUBXh0VyBUdAV4VHQAAAAABAHAAAARABEwARQAAATMyFgcBBgchMhYPAQ4BKwEVITIWDwEOASsBFRQGKwEiJj0BISImPwE+ATsBNSEiJj8BPgE7ASYnASY2OwEyHwEWMj8BNgM5+goFCP6UBgUBDAoGBngGGAp9ARMKBgZ4BhgKfQ8LlAsP/u0KBgZ4BhgKff7tCgYGeAYYCnYFBv6UCAUK+hkSpAgUCKQSBEwKCP6UBgwMCKAIDGQMCKAIDK4LDw8LrgwIoAgMZAwIoAgMDAYBbAgKEqQICKQSAAABAGQABQSMBK4AOwAAATIXFhcjNC4DIyIOAwchByEGFSEHIR4EMzI+AzUzBgcGIyInLgEnIzczNjcjNzM+ATc2AujycDwGtSM0QDkXEys4MjAPAXtk/tQGAZZk/tQJMDlCNBUWOUA0I64eYmunznYkQgzZZHABBdpkhhQ+H3UErr1oaS1LMCEPCx4uTzJkMjJkSnRCKw8PIjBKK6trdZ4wqndkLzVkV4UljQAAAgB7AAAETASwAD4ARwAAASEyHgUVHAEVFA4FKwEHITIWDwEOASsBFRQGKwEiJj0BISImPwE+ATsBNSEiJj8BPgE7ARE0NhcRMzI2NTQmIwGsAV5DakIwFgwBAQwWMEJqQ7ICASAKBgZ4BhgKigsKlQoP/vUKBgZ4BhgKdf71CgYGeAYYCnUPtstALS1ABLAaJD8yTyokCwsLJCpQMkAlGmQMCKAIDK8LDg8KrwwIoAgMZAwIoAgMAdsKD8j+1EJWVEAAAAEAyAGQBEwCvAAPAAATITIWHQEUBiMhIiY9ATQ2+gMgFR0dFfzgFR0dArwdFcgVHR0VyBUdAAAAAgDIAAAD6ASwACUAQQAAARUUBisBFRQGBx4BHQEzMhYdASE1NDY7ATU0NjcuAT0BIyImPQEXFRQWFx4BFAYHDgEdASE1NCYnLgE0Njc+AT0BA+gdFTJjUVFjMhUd/OAdFTJjUVFjMhUdyEE3HCAgHDdBAZBBNxwgIBw3QQSwlhUdZFuVIyOVW5YdFZaWFR2WW5UjI5VbZB0VlshkPGMYDDI8MgwYYzyWljxjGAwyPDIMGGM8ZAAAAAEAAAAAAAAAAAAAAAAxAAAB//IBLATCBEEAFgAAATIWFzYzMhYVFAYjISImNTQ2NyY1NDYB9261LCwueKqqeP0ST3FVQgLYBEF3YQ6teHmtclBFaw4MGZnXAAAAAgAAAGQEsASvABoAHgAAAB4BDwEBMzIWHQEhNTQ2OwEBJyY+ARYfATc2AyEnAwL2IAkKiAHTHhQe+1AeFB4B1IcKCSAkCm9wCXoBebbDBLMTIxC7/RYlFSoqFSUC6rcQJBQJEJSWEPwecAIWAAAAAAQAAABkBLAETAALABcAIwA3AAATITIWBwEGIicBJjYXARYUBwEGJjURNDYJATYWFREUBicBJjQHARYGIyEiJjcBNjIfARYyPwE2MhkEfgoFCP3MCBQI/cwIBQMBCAgI/vgICgoDjAEICAoKCP74CFwBbAgFCvuCCgUIAWwIFAikCBQIpAgUBEwKCP3JCAgCNwgK2v74CBQI/vgIBQoCJgoF/vABCAgFCv3aCgUIAQgIFID+lAgKCggBbAgIpAgIpAgAAAAD//D/8AS6BLoACQANABAAAAAyHwEWFA8BJzcTAScJAQUTA+AmDpkNDWPWXyL9mdYCZv4f/rNuBLoNmQ4mDlzWYP50/ZrWAmb8anABTwAAAAEAAAAABLAEsAAPAAABETMyFh0BITU0NjsBEQEhArz6FR384B0V+v4MBLACiv3aHRUyMhUdAiYCJgAAAAEADgAIBEwEnAAfAAABJTYWFREUBgcGLgE2NzYXEQURFAYHBi4BNjc2FxE0NgFwAoUnMFNGT4gkV09IQv2oWEFPiCRXT0hCHQP5ow8eIvzBN1EXGSltchkYEAIJm/2iKmAVGilucRoYEQJ/JioAAAACAAn/+AS7BKcAHQApAAAAMh4CFQcXFAcBFgYPAQYiJwEGIycHIi4CND4BBCIOARQeATI+ATQmAZDItoNOAQFOARMXARY7GikT/u13jgUCZLaDTk6DAXKwlFZWlLCUVlYEp06DtmQCBY15/u4aJRg6FBQBEk0BAU6Dtsi2g1tWlLCUVlaUsJQAAQBkAFgErwREABkAAAE+Ah4CFRQOAwcuBDU0PgIeAQKJMHt4dVg2Q3mEqD4+p4V4Qzhadnh5A7VESAUtU3ZAOXmAf7JVVbJ/gHk5QHZTLQVIAAAAAf/TAF4EewSUABgAAAETNjIXEyEyFgcFExYGJyUFBiY3EyUmNjMBl4MHFQeBAaUVBhH+qoIHDxH+qf6qEQ8Hgv6lEQYUAyABYRMT/p8RDPn+bxQLDPb3DAsUAZD7DBEAAv/TAF4EewSUABgAIgAAARM2MhcTITIWBwUTFgYnJQUGJjcTJSY2MwUjFwc3Fyc3IycBl4MHFQeBAaUVBhH+qoIHDxH+qf6qEQ8Hgv6lEQYUAfPwxUrBw0rA6k4DIAFhExP+nxEM+f5vFAsM9vcMCxQBkPsMEWSO4ouM5YzTAAABAAAAAASwBLAAJgAAATIWHQEUBiMVFBYXBR4BHQEUBiMhIiY9ATQ2NyU+AT0BIiY9ATQ2Alh8sD4mDAkBZgkMDwr7ggoPDAkBZgkMJj6wBLCwfPouaEsKFwbmBRcKXQoPDwpdChcF5gYXCktoLvp8sAAAAA0AAAAABLAETAAPABMAIwAnACsALwAzADcARwBLAE8AUwBXAAATITIWFREUBiMhIiY1ETQ2FxUzNSkBIgYVERQWMyEyNjURNCYzFTM1BRUzNSEVMzUFFTM1IRUzNQchIgYVERQWMyEyNjURNCYFFTM1IRUzNQUVMzUhFTM1GQR+Cg8PCvuCCg8PVWQCo/3aCg8PCgImCg8Pc2T8GGQDIGT8GGQDIGTh/doKDw8KAiYKDw/872QDIGT8GGQDIGQETA8K++YKDw8KBBoKD2RkZA8K/qIKDw8KAV4KD2RkyGRkZGTIZGRkZGQPCv6iCg8PCgFeCg9kZGRkZMhkZGRkAAAEAAAAAARMBEwADwAfAC8APwAAEyEyFhURFAYjISImNRE0NikBMhYVERQGIyEiJjURNDYBITIWFREUBiMhIiY1ETQ2KQEyFhURFAYjISImNRE0NjIBkBUdHRX+cBUdHQJtAZAVHR0V/nAVHR39vQGQFR0dFf5wFR0dAm0BkBUdHRX+cBUdHQRMHRX+cBUdHRUBkBUdHRX+cBUdHRUBkBUd/agdFf5wFR0dFQGQFR0dFf5wFR0dFQGQFR0AAAkAAAAABEwETAAPAB8ALwA/AE8AXwBvAH8AjwAAEzMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYhMzIWHQEUBisBIiY9ATQ2ATMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYhMzIWHQEUBisBIiY9ATQ2ATMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYhMzIWHQEUBisBIiY9ATQ2MsgVHR0VyBUdHQGlyBUdHRXIFR0dAaXIFR0dFcgVHR389cgVHR0VyBUdHQGlyBUdHRXIFR0dAaXIFR0dFcgVHR389cgVHR0VyBUdHQGlyBUdHRXIFR0dAaXIFR0dFcgVHR0ETB0VyBUdHRXIFR0dFcgVHR0VyBUdHRXIFR0dFcgVHf5wHRXIFR0dFcgVHR0VyBUdHRXIFR0dFcgVHR0VyBUd/nAdFcgVHR0VyBUdHRXIFR0dFcgVHR0VyBUdHRXIFR0ABgAAAAAEsARMAA8AHwAvAD8ATwBfAAATMzIWHQEUBisBIiY9ATQ2KQEyFh0BFAYjISImPQE0NgEzMhYdARQGKwEiJj0BNDYpATIWHQEUBiMhIiY9ATQ2ATMyFh0BFAYrASImPQE0NikBMhYdARQGIyEiJj0BNDYyyBUdHRXIFR0dAaUCvBUdHRX9RBUdHf6FyBUdHRXIFR0dAaUCvBUdHRX9RBUdHf6FyBUdHRXIFR0dAaUCvBUdHRX9RBUdHQRMHRXIFR0dFcgVHR0VyBUdHRXIFR3+cB0VyBUdHRXIFR0dFcgVHR0VyBUd/nAdFcgVHR0VyBUdHRXIFR0dFcgVHQAAAAABACYALAToBCAAFwAACQE2Mh8BFhQHAQYiJwEmND8BNjIfARYyAdECOwgUB7EICPzxBxUH/oAICLEHFAirBxYB3QI7CAixBxQI/PAICAGACBQHsQgIqwcAAQBuAG4EQgRCACMAAAEXFhQHCQEWFA8BBiInCQEGIi8BJjQ3CQEmND8BNjIXCQE2MgOIsggI/vUBCwgIsggVB/70/vQHFQiyCAgBC/71CAiyCBUHAQwBDAcVBDuzCBUH/vT+9AcVCLIICAEL/vUICLIIFQcBDAEMBxUIsggI/vUBDAcAAwAX/+sExQSZABkAJQBJAAAAMh4CFRQHARYUDwEGIicBBiMiLgI0PgEEIg4BFB4BMj4BNCYFMzIWHQEzMhYdARQGKwEVFAYrASImPQEjIiY9ATQ2OwE1NDYBmcSzgk1OASwICG0HFQj+1HeOYrSBTU2BAW+zmFhYmLOZWFj+vJYKD0sKDw8KSw8KlgoPSwoPDwpLDwSZTYKzYo15/tUIFQhsCAgBK01NgbTEs4JNWJmzmFhYmLOZIw8KSw8KlgoPSwoPDwpLDwqWCg9LCg8AAAMAF//rBMUEmQAZACUANQAAADIeAhUUBwEWFA8BBiInAQYjIi4CND4BBCIOARQeATI+ATQmBSEyFh0BFAYjISImPQE0NgGZxLOCTU4BLAgIbQcVCP7Ud45itIFNTYEBb7OYWFiYs5lYWP5YAV4KDw8K/qIKDw8EmU2Cs2KNef7VCBUIbAgIAStNTYG0xLOCTViZs5hYWJizmYcPCpYKDw8KlgoPAAAAAAIAFwAXBJkEsAAPAC0AAAEzMhYVERQGKwEiJjURNDYFNRYSFRQOAiIuAjU0EjcVDgEVFB4BMj4BNTQmAiZkFR0dFWQVHR0BD6fSW5vW6tabW9KnZ3xyxejFcnwEsB0V/nAVHR0VAZAVHeGmPv7ZuHXWm1tbm9Z1uAEnPqY3yHh0xXJyxXR4yAAEAGQAAASwBLAADwAfAC8APwAAATMyFhURFAYrASImNRE0NgEzMhYVERQGKwEiJjURNDYBMzIWFREUBisBIiY1ETQ2BTMyFh0BFAYrASImPQE0NgQBlgoPDwqWCg8P/t6WCg8PCpYKDw/+3pYKDw8KlgoPD/7elgoPDwqWCg8PBLAPCvuCCg8PCgR+Cg/+cA8K/RIKDw8KAu4KD/7UDwr+PgoPDwoBwgoPyA8K+goPDwr6Cg8AAAAAAgAaABsElgSWAEcATwAAATIfAhYfATcWFwcXFh8CFhUUDwIGDwEXBgcnBwYPAgYjIi8CJi8BByYnNycmLwImNTQ/AjY/ASc2Nxc3Nj8CNhIiBhQWMjY0AlghKSYFMS0Fhj0rUAMZDgGYBQWYAQ8YA1AwOIYFLDIFJisfISkmBTEtBYY8LFADGQ0ClwYGlwINGQNQLzqFBS0xBSYreLJ+frJ+BJYFmAEOGQJQMDmGBSwxBiYrHiIoJgYxLAWGPSxRAxkOApcFBZcCDhkDUTA5hgUtMAYmKiAhKCYGMC0Fhj0sUAIZDgGYBf6ZfrF+frEABwBkAAAEsAUUABMAFwAhACUAKQAtADEAAAEhMhYdASEyFh0BITU0NjMhNTQ2FxUhNQERFAYjISImNREXETMRMxEzETMRMxEzETMRAfQBLCk7ARMKD/u0DwoBEzspASwBLDsp/UQpO2RkZGRkZGRkBRQ7KWQPCktLCg9kKTtkZGT+1PzgKTs7KQMgZP1EArz9RAK8/UQCvP1EArwAAQAMAAAFCATRAB8AABMBNjIXARYGKwERFAYrASImNREhERQGKwEiJjURIyImEgJsCBUHAmAIBQqvDwr6Cg/+1A8K+goPrwoFAmoCYAcH/aAICv3BCg8PCgF3/okKDw8KAj8KAAIAZAAAA+gEsAARABcAAAERFBYzIREUBiMhIiY1ETQ2MwEjIiY9AQJYOykBLB0V/OAVHR0VA1L6FR0EsP5wKTv9dhUdHRUETBUd/nAdFfoAAwAXABcEmQSZAA8AGwAwAAAAMh4CFA4CIi4CND4BBCIOARQeATI+ATQmBTMyFhURMzIWHQEUBisBIiY1ETQ2AePq1ptbW5vW6tabW1ubAb/oxXJyxejFcnL+fDIKD68KDw8K+goPDwSZW5vW6tabW1ub1urWmztyxejFcnLF6MUNDwr+7Q8KMgoPDwoBXgoPAAAAAAL/nAAABRQEsAALAA8AACkBAyMDIQEzAzMDMwEDMwMFFP3mKfIp/eYBr9EVohTQ/p4b4BsBkP5wBLD+1AEs/nD+1AEsAAAAAAIAZAAABLAEsAAVAC8AAAEzMhYVETMyFgcBBiInASY2OwERNDYBMzIWFREUBiMhIiY1ETQ2OwEyFh0BITU0NgImyBUdvxQLDf65DSYN/rkNCxS/HQJUMgoPDwr75goPDwoyCg8DhA8EsB0V/j4XEP5wEBABkBAXAcIVHfzgDwr+ogoPDwoBXgoPDwqvrwoPAAMAFwAXBJkEmQAPABsAMQAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgUzMhYVETMyFgcDBiInAyY2OwERNDYB4+rWm1tbm9bq1ptbW5sBv+jFcnLF6MVycv58lgoPiRUKDd8NJg3fDQoViQ8EmVub1urWm1tbm9bq1ps7csXoxXJyxejFDQ8K/u0XEP7tEBABExAXARMKDwAAAAMAFwAXBJkEmQAPABsAMQAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JiUTFgYrAREUBisBIiY1ESMiJjcTNjIB4+rWm1tbm9bq1ptbW5sBv+jFcnLF6MVycv7n3w0KFYkPCpYKD4kVCg3fDSYEmVub1urWm1tbm9bq1ps7csXoxXJyxejFAf7tEBf+7QoPDwoBExcQARMQAAAAAAIAAAAABLAEsAAZADkAABMhMhYXExYVERQGBwYjISImJyY1EzQ3Ez4BBSEiBgcDBhY7ATIWHwEeATsBMjY/AT4BOwEyNicDLgHhAu4KEwO6BwgFDBn7tAweAgYBB7kDEwKX/dQKEgJXAgwKlgoTAiYCEwr6ChMCJgITCpYKDAJXAhIEsA4K/XQYGf5XDB4CBggEDRkBqRkYAowKDsgOC/4+Cw4OCpgKDg4KmAoODgsBwgsOAAMAFwAXBJkEmQAPABsAJwAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgUXFhQPAQYmNRE0NgHj6tabW1ub1urWm1tbmwG/6MVycsXoxXJy/ov9ERH9EBgYBJlbm9bq1ptbW5vW6tabO3LF6MVycsXoxV2+DCQMvgwLFQGQFQsAAQAXABcEmQSwACgAAAE3NhYVERQGIyEiJj8BJiMiDgEUHgEyPgE1MxQOAiIuAjQ+AjMyA7OHBwsPCv6WCwQHhW2BdMVycsXoxXKWW5vW6tabW1ub1nXABCSHBwQL/pYKDwsHhUxyxejFcnLFdHXWm1tbm9bq1ptbAAAAAAIAFwABBJkEsAAaADUAAAE3NhYVERQGIyEiJj8BJiMiDgEVIzQ+AjMyEzMUDgIjIicHBiY1ETQ2MyEyFg8BFjMyPgEDs4cHCw8L/pcLBAeGboF0xXKWW5vWdcDrllub1nXAnIYHCw8LAWgKBQiFboJ0xXIEJIcHBAv+lwsPCweGS3LFdHXWm1v9v3XWm1t2hggFCgFoCw8LB4VMcsUAAAAKAGQAAASwBLAADwAfAC8APwBPAF8AbwB/AI8AnwAAEyEyFhURFAYjISImNRE0NgUhIgYVERQWMyEyNjURNCYFMzIWHQEUBisBIiY9ATQ2MyEyFh0BFAYjISImPQE0NgczMhYdARQGKwEiJj0BNDYzITIWHQEUBiMhIiY9ATQ2BzMyFh0BFAYrASImPQE0NjMhMhYdARQGIyEiJj0BNDYHMzIWHQEUBisBIiY9ATQ2MyEyFh0BFAYjISImPQE0Nn0EGgoPDwr75goPDwPA/K4KDw8KA1IKDw/9CDIKDw8KMgoPD9IBwgoPDwr+PgoPD74yCg8PCjIKDw/SAcIKDw8K/j4KDw++MgoPDwoyCg8P0gHCCg8PCv4+Cg8PvjIKDw8KMgoPD9IBwgoPDwr+PgoPDwSwDwr7ggoPDwoEfgoPyA8K/K4KDw8KA1IKD2QPCjIKDw8KMgoPDwoyCg8PCjIKD8gPCjIKDw8KMgoPDwoyCg8PCjIKD8gPCjIKDw8KMgoPDwoyCg8PCjIKD8gPCjIKDw8KMgoPDwoyCg8PCjIKDwAAAAACAAAAAARMBLAAGQAjAAABNTQmIyEiBh0BIyIGFREUFjMhMjY1ETQmIyE1NDY7ATIWHQEDhHVT/tRSdmQpOzspA4QpOzsp/ageFMgUHgMgyFN1dlLIOyn9qCk7OykCWCk7lhUdHRWWAAIAZAAABEwETAAJADcAABMzMhYVESMRNDYFMhcWFREUBw4DIyIuAScuAiMiBwYjIicmNRE+ATc2HgMXHgIzMjc2fTIKD2QPA8AEBRADIUNAMRwaPyonKSxHHlVLBwgGBQ4WeDsXKC4TOQQpLUUdZ1AHBEwPCvvNBDMKDzACBhH+WwYGO1AkDQ0ODg8PDzkFAwcPAbY3VwMCAwsGFAEODg5XCAAAAwAAAAAEsASXACEAMQBBAAAAMh4CFREUBisBIiY1ETQuASAOARURFAYrASImNRE0PgEDMzIWFREUBisBIiY1ETQ2ITMyFhURFAYrASImNRE0NgHk6N6jYw8KMgoPjeT++uSNDwoyCg9joyqgCAwMCKAIDAwCYKAIDAwIoAgMDASXY6PedP7UCg8PCgEsf9FyctF//tQKDw8KASx03qP9wAwI/jQIDAwIAcwIDAwI/jQIDAwIAcwIDAAAAAACAAAA0wRHA90AFQA5AAABJTYWFREUBiclJisBIiY1ETQ2OwEyBTc2Mh8BFhQPARcWFA8BBiIvAQcGIi8BJjQ/AScmND8BNjIXAUEBAgkMDAn+/hUZ+goPDwr6GQJYeAcUByIHB3h4BwciBxQHeHgHFAciBwd3dwcHIgcUBwMurAYHCv0SCgcGrA4PCgFeCg+EeAcHIgcUB3h4BxQHIgcHd3cHByIHFAd4eAcUByIICAAAAAACAAAA0wNyA90AFQAvAAABJTYWFREUBiclJisBIiY1ETQ2OwEyJTMWFxYVFAcGDwEiLwEuATc2NTQnJjY/ATYBQQECCQwMCf7+FRn6Cg8PCvoZAdIECgZgWgYLAwkHHQcDBkhOBgMIHQcDLqwGBwr9EgoHBqwODwoBXgoPZAEJgaGafwkBAQYXBxMIZ36EaggUBxYFAAAAAAMAAADEBGID7AAbADEASwAAATMWFxYVFAYHBgcjIi8BLgE3NjU0JicmNj8BNgUlNhYVERQGJyUmKwEiJjURNDY7ATIlMxYXFhUUBwYPASIvAS4BNzY1NCcmNj8BNgPHAwsGh0RABwoDCQcqCAIGbzs3BgIJKgf9ggECCQwMCf7+FRn6Cg8PCvoZAdIECgZgWgYLAwkHHQcDBkhOBgMIHQcD7AEJs9lpy1QJAQYiBhQIlrJarEcJFAYhBb6sBgcK/RIKBwasDg8KAV4KD2QBCYGhmn8JAQEGFwcTCGd+hGoIFQYWBQAAAAANAAAAAASwBLAACQAVABkAHQAhACUALQA7AD8AQwBHAEsATwAAATMVIxUhFSMRIQEjFTMVIREjESM1IQURIREhESERBSM1MwUjNTMBMxEhETM1MwEzFSMVIzUjNTM1IzUhBREhEQcjNTMFIzUzASM1MwUhNSEB9GRk/nBkAfQCvMjI/tTIZAJY+7QBLAGQASz84GRkArxkZP1EyP4MyGQB9MhkyGRkyAEs/UQBLGRkZAOEZGT+DGRkAfT+1AEsA4RkZGQCWP4MZMgBLAEsyGT+1AEs/tQBLMhkZGT+DP4MAfRk/tRkZGRkyGTI/tQBLMhkZGT+1GRkZAAAAAAJAAAAAASwBLAAAwAHAAsADwATABcAGwAfACMAADcjETMTIxEzASMRMxMjETMBIxEzASE1IRcjNTMXIzUzBSM1M2RkZMhkZAGQyMjIZGQBLMjI/OD+1AEsyGRkyGRkASzIyMgD6PwYA+j8GAPo/BgD6PwYA+j7UGRkW1tbW1sAAAIAAAAKBKYEsAANABUAAAkBFhQHAQYiJwETNDYzBCYiBhQWMjYB9AKqCAj+MAgUCP1WAQ8KAUM7Uzs7UzsEsP1WCBQI/jAICAKqAdsKD807O1Q7OwAAAAADAAAACgXSBLAADQAZACEAAAkBFhQHAQYiJwETNDYzIQEWFAcBBiIvAQkBBCYiBhQWMjYB9AKqCAj+MAgUCP1WAQ8KAwYCqggI/jAIFAg4Aaj9RP7TO1M7O1M7BLD9VggUCP4wCAgCqgHbCg/9VggUCP4wCAg4AaoCvM07O1Q7OwAAAAABAGQAAASwBLAAJgAAASEyFREUDwEGJjURNCYjISIPAQYWMyEyFhURFAYjISImNRE0PwE2ASwDOUsSQAgKDwr9RBkSQAgFCgK8Cg8PCvyuCg8SixIEsEv8fBkSQAgFCgO2Cg8SQAgKDwr8SgoPDwoDzxkSixIAAAABAMj//wRMBLAACgAAEyEyFhURCQERNDb6AyAVHf4+/j4dBLAdFfuCAbz+QwR/FR0AAAAAAwAAAAAEsASwABUARQBVAAABISIGBwMGHwEeATMhMjY/ATYnAy4BASMiBg8BDgEjISImLwEuASsBIgYVERQWOwEyNj0BNDYzITIWHQEUFjsBMjY1ETQmASEiBg8BBhYzITI2LwEuAQM2/kQLEAFOBw45BhcKAcIKFwY+DgdTARABVpYKFgROBBYK/doKFgROBBYKlgoPDwqWCg8PCgLuCg8PCpYKDw/+sf4MChMCJgILCgJYCgsCJgITBLAPCv7TGBVsCQwMCWwVGAEtCg/+cA0JnAkNDQmcCQ0PCv12Cg8PCpYKDw8KlgoPDwoCigoP/agOCpgKDg4KmAoOAAAAAAQAAABkBLAETAAdACEAKQAxAAABMzIeAh8BMzIWFREUBiMhIiY1ETQ2OwE+BAEVMzUEIgYUFjI2NCQyFhQGIiY0AfTIOF00JAcGlik7Oyn8GCk7OymWAgknM10ByGT+z76Hh76H/u9WPDxWPARMKTs7FRQ7Kf2oKTs7KQJYKTsIG0U1K/7UZGRGh76Hh74IPFY8PFYAAAAAAgA1AAAEsASvACAAIwAACQEWFx4BHwEVITUyNi8BIQYHBh4CMxUhNTY3PgE/AQEDIQMCqQGBFCgSJQkK/l81LBFS/nk6IgsJKjIe/pM4HAwaBwcBj6wBVKIEr/waMioTFQECQkJXLd6RWSIuHAxCQhgcDCUNDQPu/VoByQAAAAADAGQAAAPwBLAAJwAyADsAAAEeBhUUDgMjITU+ATURNC4EJzUFMh4CFRQOAgclMzI2NTQuAisBETMyNjU0JisBAvEFEzUwOyodN1htbDD+DCk7AQYLFyEaAdc5dWM+Hy0tEP6Pi05pESpTPnbYUFJ9Xp8CgQEHGB0zOlIuQ3VONxpZBzMoAzsYFBwLEAkHRwEpSXNDM1s6KwkxYUopOzQb/K5lUFqBAAABAMgAAANvBLAAGQAAARcOAQcDBhYXFSE1NjcTNjQuBCcmJzUDbQJTQgeECSxK/gy6Dq0DAw8MHxUXDQYEsDkTNSj8uTEoBmFhEFIDQBEaExAJCwYHAwI5AAAAAAL/tQAABRQEsAAlAC8AAAEjNC4FKwERFBYfARUhNTI+AzURIyIOBRUjESEFIxEzByczESM3BRQyCAsZEyYYGcgyGRn+cAQOIhoWyBkYJhMZCwgyA+j7m0tLfX1LS30DhBUgFQ4IAwH8rhYZAQJkZAEFCRUOA1IBAwgOFSAVASzI/OCnpwMgpwACACH/tQSPBLAAJQAvAAABIzQuBSsBERQWHwEVITUyPgM1ESMiDgUVIxEhEwc1IRUnNxUhNQRMMggLGRMmGBnIMhkZ/nAEDiIaFsgZGCYTGQsIMgPoQ6f84KenAyADhBUgFQ4IAwH9dhYZAQJkZAEFCRUOAooBAwgOFSAVASz7gn1LS319S0sABAAAAAAEsARMAA8AHwAvAD8AABMhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2EyEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYyAlgVHR0V/agVHR0VA+gVHR0V/BgVHR0VAyAVHR0V/OAVHR0VBEwVHR0V+7QVHR0ETB0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR0ABAAAAAAEsARMAA8AHwAvAD8AABMhMhYdARQGIyEiJj0BNDYDITIWHQEUBiMhIiY9ATQ2EyEyFh0BFAYjISImPQE0NgMhMhYdARQGIyEiJj0BNDb6ArwVHR0V/UQVHR2zBEwVHR0V+7QVHR3dArwVHR0V/UQVHR2zBEwVHR0V+7QVHR0ETB0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR0ABAAAAAAEsARMAA8AHwAvAD8AAAE1NDYzITIWHQEUBiMhIiYBNTQ2MyEyFh0BFAYjISImEzU0NjMhMhYdARQGIyEiJgE1NDYzITIWHQEUBiMhIiYB9B0VAlgVHR0V/agVHf5wHRUD6BUdHRX8GBUdyB0VAyAVHR0V/OAVHf7UHRUETBUdHRX7tBUdA7ZkFR0dFWQVHR3+6WQVHR0VZBUdHf7pZBUdHRVkFR0d/ulkFR0dFWQVHR0AAAQAAAAABLAETAAPAB8ALwA/AAATITIWHQEUBiMhIiY9ATQ2EyEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2MgRMFR0dFfu0FR0dFQRMFR0dFfu0FR0dFQRMFR0dFfu0FR0dFQRMFR0dFfu0FR0dBEwdFWQVHR0VZBUd/tQdFWQVHR0VZBUd/tQdFWQVHR0VZBUd/tQdFWQVHR0VZBUdAAgAAAAABLAETAAPAB8ALwA/AE8AXwBvAH8AABMzMhYdARQGKwEiJj0BNDYpATIWHQEUBiMhIiY9ATQ2ATMyFh0BFAYrASImPQE0NikBMhYdARQGIyEiJj0BNDYBMzIWHQEUBisBIiY9ATQ2KQEyFh0BFAYjISImPQE0NgEzMhYdARQGKwEiJj0BNDYpATIWHQEUBiMhIiY9ATQ2MmQVHR0VZBUdHQFBAyAVHR0V/OAVHR3+6WQVHR0VZBUdHQFBAyAVHR0V/OAVHR3+6WQVHR0VZBUdHQFBAyAVHR0V/OAVHR3+6WQVHR0VZBUdHQFBAyAVHR0V/OAVHR0ETB0VZBUdHRVkFR0dFWQVHR0VZBUd/tQdFWQVHR0VZBUdHRVkFR0dFWQVHf7UHRVkFR0dFWQVHR0VZBUdHRVkFR3+1B0VZBUdHRVkFR0dFWQVHR0VZBUdAAAG/5wAAASwBEwAAwATACMAKgA6AEoAACEjETsCMhYdARQGKwEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2BQc1IzUzNQUhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2AZBkZJZkFR0dFWQVHR0VAfQVHR0V/gwVHR3++qfIyAHCASwVHR0V/tQVHR0VAlgVHR0V/agVHR0ETB0VZBUdHRVkFR3+1B0VZBUdHRVkFR36fUtkS68dFWQVHR0VZBUd/tQdFWQVHR0VZBUdAAAABgAAAAAFFARMAA8AEwAjACoAOgBKAAATMzIWHQEUBisBIiY9ATQ2ASMRMwEhMhYdARQGIyEiJj0BNDYFMxUjFSc3BSEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYyZBUdHRVkFR0dA2dkZPyuAfQVHR0V/gwVHR0EL8jIp6f75gEsFR0dFf7UFR0dFQJYFR0dFf2oFR0dBEwdFWQVHR0VZBUd+7QETP7UHRVkFR0dFWQVHchkS319rx0VZBUdHRVkFR3+1B0VZBUdHRVkFR0AAAAAAgAAAMgEsAPoAA8AEgAAEyEyFhURFAYjISImNRE0NgkCSwLuHywsH/0SHywsBIT+1AEsA+gsH/12HywsHwKKHyz9RAEsASwAAwAAAAAEsARMAA8AFwAfAAATITIWFREUBiMhIiY1ETQ2FxE3BScBExEEMhYUBiImNCwEWBIaGhL7qBIaGkr3ASpKASXs/NJwTk5wTgRMGhL8DBIaGhID9BIaZP0ftoOcAT7+4AH0dE5vT09vAAAAAAIA2wAFBDYEkQAWAB4AAAEyHgEVFAcOAQ8BLgQnJjU0PgIWIgYUFjI2NAKIdcZzRkWyNjYJIV5YbSk8RHOft7eCgreCBJF4ynVzj23pPz4IIWZomEiEdVijeUjDgriBgbgAAAACABcAFwSZBJkADwAXAAAAMh4CFA4CIi4CND4BAREiDgEUHgEB4+rWm1tbm9bq1ptbW5sBS3TFcnLFBJlbm9bq1ptbW5vW6tab/G8DVnLF6MVyAAACAHUAAwPfBQ8AGgA1AAABHgYVFA4DBy4DNTQ+BQMOAhceBBcWNj8BNiYnLgInJjc2IyYCKhVJT1dOPiUzVnB9P1SbfEokP0xXUEm8FykoAwEbITEcExUWAgYCCQkFEikMGiACCAgFD0iPdXdzdYdFR4BeRiYEBTpjl1lFh3ZzeHaQ/f4hS4I6JUEnIw4IBwwQIgoYBwQQQSlZtgsBAAAAAwAAAAAEywRsAAwAKgAvAAABNz4CHgEXHgEPAiUhMhcHISIGFREUFjMhMjY9ATcRFAYjISImNRE0NgkBBzcBA+hsAgYUFR0OFgoFBmz9BQGQMje7/pApOzspAfQpO8i7o/5wpbm5Azj+lqE3AWMD9XMBAgIEDw4WKgsKc8gNuzsp/gwpOzsptsj+tKW5uaUBkKW5/tf+ljKqAWMAAgAAAAAEkwRMABsANgAAASEGByMiBhURFBYzITI2NTcVFAYjISImNRE0NgUBFhQHAQYmJzUmDgMHPgY3NT4BAV4BaaQ0wyk7OykB9Ck7yLml/nClubkCfwFTCAj+rAcLARo5ZFRYGgouOUlARioTAQsETJI2Oyn+DCk7OymZZ6W5uaUBkKW5G/7TBxUH/s4GBAnLAQINFjAhO2JBNB0UBwHSCgUAAAAAAgAAAAAEnQRMAB0ANQAAASEyFwchIgYVERQWMyEyNj0BNxUUBiMhIiY1ETQ2CQE2Mh8BFhQHAQYiLwEmND8BNjIfARYyAV4BXjxDsv6jKTs7KQH0KTvIuaX+cKW5uQHKAYsHFQdlBwf97QcVB/gHB2UHFQdvCBQETBexOyn+DCk7OylFyNulubmlAZCluf4zAYsHB2UHFQf97AcH+AcVB2UHB28HAAAAAQAKAAoEpgSmADsAAAkBNjIXARYGKwEVMzU0NhcBFhQHAQYmPQEjFTMyFgcBBiInASY2OwE1IxUUBicBJjQ3ATYWHQEzNSMiJgE+AQgIFAgBBAcFCqrICggBCAgI/vgICsiqCgUH/vwIFAj++AgFCq/ICgj++AgIAQgICsivCgUDlgEICAj++AgKyK0KBAf+/AcVB/73BwQKrcgKCP74CAgBCAgKyK0KBAcBCQcVBwEEBwQKrcgKAAEAyAAAA4QETAAZAAATMzIWFREBNhYVERQGJwERFAYrASImNRE0NvpkFR0B0A8VFQ/+MB0VZBUdHQRMHRX+SgHFDggV/BgVCA4Bxf5KFR0dFQPoFR0AAAABAAAAAASwBEwAIwAAEzMyFhURATYWFREBNhYVERQGJwERFAYnAREUBisBIiY1ETQ2MmQVHQHQDxUB0A8VFQ/+MBUP/jAdFWQVHR0ETB0V/koBxQ4IFf5KAcUOCBX8GBUIDgHF/koVCA4Bxf5KFR0dFQPoFR0AAAABAJ0AGQSwBDMAFQAAAREUBicBERQGJwEmNDcBNhYVEQE2FgSwFQ/+MBUP/hQPDwHsDxUB0A8VBBr8GBUIDgHF/koVCA4B4A4qDgHgDggV/koBxQ4IAAAAAQDIABYEMwQ2AAsAABMBFhQHAQYmNRE0NvMDLhIS/NISGRkEMv4OCx4L/g4LDhUD6BUOAAIAyABkA4QD6AAPAB8AABMzMhYVERQGKwEiJjURNDYhMzIWFREUBisBIiY1ETQ2+sgVHR0VyBUdHQGlyBUdHRXIFR0dA+gdFfzgFR0dFQMgFR0dFfzgFR0dFQMgFR0AAAEAyABkBEwD6AAPAAABERQGIyEiJjURNDYzITIWBEwdFfzgFR0dFQMgFR0DtvzgFR0dFQMgFR0dAAAAAAEAAAAZBBMEMwAVAAABETQ2FwEWFAcBBiY1EQEGJjURNDYXAfQVDwHsDw/+FA8V/jAPFRUPAmQBthUIDv4gDioO/iAOCBUBtv47DggVA+gVCA4AAAH//gACBLMETwAjAAABNzIWFRMUBiMHIiY1AwEGJjUDAQYmNQM0NhcBAzQ2FwEDNDYEGGQUHgUdFWQVHQL+MQ4VAv4yDxUFFQ8B0gIVDwHSAh0ETgEdFfwYFR0BHRUBtf46DwkVAbX+OQ4JFAPoFQkP/j4BthQJDv49AbYVHQAAAQEsAAAD6ARMABkAAAEzMhYVERQGKwEiJjURAQYmNRE0NhcBETQ2A1JkFR0dFWQVHf4wDxUVDwHQHQRMHRX8GBUdHRUBtv47DggVA+gVCA7+OwG2FR0AAAIAZADIBLAESAALABsAAAkBFgYjISImNwE2MgEhMhYdARQGIyEiJj0BNDYCrgH1DwkW++4WCQ8B9Q8q/fcD6BUdHRX8GBUdHQQ5/eQPFhYPAhwP/UgdFWQVHR0VZBUdAAEAiP/8A3UESgAFAAAJAgcJAQN1/qABYMX92AIoA4T+n/6fxgIoAiYAAAAAAQE7//wEKARKAAUAAAkBJwkBNwQo/dnGAWH+n8YCI/3ZxgFhAWHGAAIAFwAXBJkEmQAPADMAAAAyHgIUDgIiLgI0PgEFIyIGHQEjIgYdARQWOwEVFBY7ATI2PQEzMjY9ATQmKwE1NCYB4+rWm1tbm9bq1ptbW5sBfWQVHZYVHR0Vlh0VZBUdlhUdHRWWHQSZW5vW6tabW1ub1urWm7odFZYdFWQVHZYVHR0Vlh0VZBUdlhUdAAAAAAIAFwAXBJkEmQAPAB8AAAAyHgIUDgIiLgI0PgEBISIGHQEUFjMhMjY9ATQmAePq1ptbW5vW6tabW1ubAkX+DBUdHRUB9BUdHQSZW5vW6tabW1ub1urWm/5+HRVkFR0dFWQVHQACABcAFwSZBJkADwAzAAAAMh4CFA4CIi4CND4BBCIPAScmIg8BBhQfAQcGFB8BFjI/ARcWMj8BNjQvATc2NC8BAePq1ptbW5vW6tabW1ubAeUZCXh4CRkJjQkJeHgJCY0JGQl4eAkZCY0JCXh4CQmNBJlbm9bq1ptbW5vW6tabrQl4eAkJjQkZCXh4CRkJjQkJeHgJCY0JGQl4eAkZCY0AAgAXABcEmQSZAA8AJAAAADIeAhQOAiIuAjQ+AQEnJiIPAQYUHwEWMjcBNjQvASYiBwHj6tabW1ub1urWm1tbmwEVVAcVCIsHB/IHFQcBdwcHiwcVBwSZW5vW6tabW1ub1urWm/4xVQcHiwgUCPEICAF3BxUIiwcHAAAAAAMAFwAXBJkEmQAPADsASwAAADIeAhQOAiIuAjQ+AQUiDgMVFDsBFjc+ATMyFhUUBgciDgUHBhY7ATI+AzU0LgMTIyIGHQEUFjsBMjY9ATQmAePq1ptbW5vW6tabW1ubAT8dPEIyIRSDHgUGHR8UFw4TARkOGhITDAIBDQ6tBx4oIxgiM0Q8OpYKDw8KlgoPDwSZW5vW6tabW1ub1urWm5ELHi9PMhkFEBQQFRIXFgcIBw4UHCoZCBEQKDhcNi9IKhsJ/eMPCpYKDw8KlgoPAAADABcAFwSZBJkADwAfAD4AAAAyHgIUDgIiLgI0PgEFIyIGHQEUFjsBMjY9ATQmAyMiBh0BFBY7ARUjIgYdARQWMyEyNj0BNCYrARE0JgHj6tabW1ub1urWm1tbmwGWlgoPDwqWCg8PCvoKDw8KS0sKDw8KAV4KDw8KSw8EmVub1urWm1tbm9bq1ptWDwqWCg8PCpYKD/7UDwoyCg/IDwoyCg8PCjIKDwETCg8AAgAAAAAEsASwAC8AXwAAATMyFh0BHgEXMzIWHQEUBisBDgEHFRQGKwEiJj0BLgEnIyImPQE0NjsBPgE3NTQ2ExUUBisBIiY9AQ4BBzMyFh0BFAYrAR4BFzU0NjsBMhYdAT4BNyMiJj0BNDY7AS4BAg2WCg9nlxvCCg8PCsIbl2cPCpYKD2eXG8IKDw8KwhuXZw+5DwqWCg9EZheoCg8PCqgXZkQPCpYKD0RmF6gKDw8KqBdmBLAPCsIbl2cPCpYKD2eXG8IKDw8KwhuXZw8KlgoPZ5cbwgoP/s2oCg8PCqgXZkQPCpYKD0RmF6gKDw8KqBdmRA8KlgoPRGYAAwAXABcEmQSZAA8AGwA/AAAAMh4CFA4CIi4CND4BBCIOARQeATI+ATQmBxcWFA8BFxYUDwEGIi8BBwYiLwEmND8BJyY0PwE2Mh8BNzYyAePq1ptbW5vW6tabW1ubAb/oxXJyxejFcnKaQAcHfHwHB0AHFQd8fAcVB0AHB3x8BwdABxUHfHwHFQSZW5vW6tabW1ub1urWmztyxejFcnLF6MVaQAcVB3x8BxUHQAcHfHwHB0AHFQd8fAcVB0AHB3x8BwAAAAMAFwAXBJkEmQAPABsAMAAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgcXFhQHAQYiLwEmND8BNjIfATc2MgHj6tabW1ub1urWm1tbmwG/6MVycsXoxXJyg2oHB/7ACBQIyggIagcVB0/FBxUEmVub1urWm1tbm9bq1ps7csXoxXJyxejFfWoHFQf+vwcHywcVB2oICE/FBwAAAAMAFwAXBJkEmQAPABgAIQAAADIeAhQOAiIuAjQ+AQUiDgEVFBcBJhcBFjMyPgE1NAHj6tabW1ub1urWm1tbmwFLdMVyQQJLafX9uGhzdMVyBJlbm9bq1ptbW5vW6tabO3LFdHhpAktB0P24PnLFdHMAAAAAAQAXAFMEsAP5ABUAABMBNhYVESEyFh0BFAYjIREUBicBJjQnAgoQFwImFR0dFf3aFxD99hACRgGrDQoV/t0dFcgVHf7dFQoNAasNJgAAAAABAAAAUwSZA/kAFQAACQEWFAcBBiY1ESEiJj0BNDYzIRE0NgJ/AgoQEP32EBf92hUdHRUCJhcD8f5VDSYN/lUNChUBIx0VyBUdASMVCgAAAAEAtwAABF0EmQAVAAAJARYGIyERFAYrASImNREhIiY3ATYyAqoBqw0KFf7dHRXIFR3+3RUKDQGrDSYEif32EBf92hUdHRUCJhcQAgoQAAAAAQC3ABcEXQSwABUAAAEzMhYVESEyFgcBBiInASY2MyERNDYCJsgVHQEjFQoN/lUNJg3+VQ0KFQEjHQSwHRX92hcQ/fYQEAIKEBcCJhUdAAABAAAAtwSZBF0AFwAACQEWFAcBBiY1EQ4DBz4ENxE0NgJ/AgoQEP32EBdesKWBJAUsW4fHfhcEVf5VDSYN/lUNChUBIwIkRHVNabGdcUYHAQYVCgACAAAAAASwBLAAFQArAAABITIWFREUBi8BBwYiLwEmND8BJyY2ASEiJjURNDYfATc2Mh8BFhQPARcWBgNSASwVHRUOXvkIFAhqBwf5Xg4I/iH+1BUdFQ5e+QgUCGoHB/leDggEsB0V/tQVCA5e+QcHaggUCPleDhX7UB0VASwVCA5e+QcHaggUCPleDhUAAAACAEkASQRnBGcAFQArAAABFxYUDwEXFgYjISImNRE0Nh8BNzYyASEyFhURFAYvAQcGIi8BJjQ/AScmNgP2agcH+V4OCBX+1BUdFQ5e+QgU/QwBLBUdFQ5e+QgUCGoHB/leDggEYGoIFAj5Xg4VHRUBLBUIDl75B/3xHRX+1BUIDl75BwdqCBQI+V4OFQAAAAADABcAFwSZBJkADwAfAC8AAAAyHgIUDgIiLgI0PgEFIyIGFxMeATsBMjY3EzYmAyMiBh0BFBY7ATI2PQE0JgHj6tabW1ub1urWm1tbmwGz0BQYBDoEIxQ2FCMEOgQYMZYKDw8KlgoPDwSZW5vW6tabW1ub1urWm7odFP7SFB0dFAEuFB3+DA8KlgoPDwqWCg8AAAAABQAAAAAEsASwAEkAVQBhAGgAbwAAATIWHwEWHwEWFxY3Nj8BNjc2MzIWHwEWHwIeATsBMhYdARQGKwEiBh0BIREjESE1NCYrASImPQE0NjsBMjY1ND8BNjc+BAUHBhY7ATI2LwEuAQUnJgYPAQYWOwEyNhMhIiY1ESkBERQGIyERAQQJFAUFFhbEFQ8dCAsmxBYXERUXMA0NDgQZCAEPCj0KDw8KMgoP/nDI/nAPCjIKDw8KPQsOCRkFDgIGFRYfAp2mBwQK2woKAzMDEP41sQgQAzMDCgrnCwMe/okKDwGQAlgPCv6JBLAEAgIKDXYNCxUJDRZ2DQoHIREQFRh7LAkLDwoyCg8PCq8BLP7UrwoPDwoyCg8GBQQwgBkUAwgWEQ55ogcKDgqVCgSqnQcECo8KDgr8cg8KAXf+iQoPAZAAAAAAAgAAAAwErwSmACsASQAAATYWFQYCDgQuAScmByYOAQ8BBiY1NDc+ATc+AScuAT4BNz4GFyYGBw4BDwEOBAcOARY2Nz4CNz4DNz4BBI0IGgItQmxhi2KORDg9EQQRMxuZGhYqCFUYEyADCQIQOjEnUmFch3vAJQgdHyaiPT44XHRZUhcYDhItIRmKcVtGYWtbKRYEBKYDEwiy/t3IlVgxEQgLCwwBAQIbG5kYEyJAJghKFRE8Hzdff4U/M0o1JSMbL0QJGCYvcSEhHjZST2c1ODwEJygeW0AxJUBff1UyFAABAF0AHgRyBM8ATwAAAQ4BHgQXLgc+ATceAwYHDgQHBicmNzY3PgQuAScWDgMmJy4BJyY+BDcGHgM3PgEuAicmPgMCjScfCic4R0IgBBsKGAoQAwEJEg5gikggBhANPkpTPhZINx8SBgsNJysiCRZOQQoVNU1bYC9QZwICBAUWITsoCAYdJzIYHw8YIiYHDyJJYlkEz0OAZVxEOSQMBzgXOB42IzElKRIqg5Gnl0o3Z0c6IAYWCwYNAwQFIDhHXGF1OWiqb0sdBxUknF0XNTQ8PEUiNWNROBYJDS5AQVUhVZloUSkAAAAAA//cAGoE1ARGABsAPwBRAAAAMh4FFA4FIi4FND4EBSYGFxYVFAYiJjU0NzYmBwYHDgEXHgQyPgM3NiYnJgUHDgEXFhcWNj8BNiYnJicuAQIGpJ17bk85HBw6T257naKde25POhwcOU9uewIPDwYIGbD4sBcIBw5GWg0ECxYyWl+DiINfWjIWCwQMWv3/Iw8JCSU4EC0OIw4DDywtCyIERi1JXGJcSSpJXGJcSS0tSVxiXEkqSVxiXEncDwYTOT58sLB8OzcTBg9FcxAxEiRGXkQxMEVeRSQSMRF1HiQPLxJEMA0EDyIPJQ8sSRIEAAAABP/cAAAE1ASwABQAJwA7AEwAACEjNy4ENTQ+BTMyFzczEzceARUUDgMHNz4BNzYmJyYlBgcOARceBBc3LgE1NDc2JhcHDgEXFhcWNj8CJyYnLgECUJQfW6l2WSwcOU9ue51SPUEglCYvbIknUGqYUi5NdiYLBAw2/VFGWg0ECxIqSExoNSlrjxcIB3wjDwkJJTgQLQ4MFgMsLQsieBRhdHpiGxVJXGJcSS0Pef5StVXWNBpacm5jGq0xiD8SMRFGckVzEDESHjxRQTkNmhKnbjs3EwZwJA8vEkQwDQQPC1YELEkSBAAAAAP/ngAABRIEqwALABgAKAAAJwE2FhcBFgYjISImJSE1NDY7ATIWHQEhAQczMhYPAQ4BKwEiJi8BJjZaAoIUOBQCghUbJfryJRsBCgFZDwqWCg8BWf5DaNAUGAQ6BCMUNhQjBDoEGGQEKh8FIfvgIEdEhEsKDw8KSwLT3x0U/BQdHRT8FB0AAAABAGQAFQSwBLAAKAAAADIWFREBHgEdARQGJyURFh0BFAYvAQcGJj0BNDcRBQYmPQE0NjcBETQCTHxYAWsPFhgR/plkGhPNzRMaZP6ZERgWDwFrBLBYPv6t/rsOMRQpFA0M+f75XRRAFRAJgIAJEBVAFF0BB/kMDRQpFDEOAUUBUz4AAAARAAAAAARMBLAAHQAnACsALwAzADcAOwA/AEMARwBLAE8AUwBXAFsAXwBjAAABMzIWHQEzMhYdASE1NDY7ATU0NjsBMhYdASE1NDYBERQGIyEiJjURFxUzNTMVMzUzFTM1MxUzNTMVMzUFFTM1MxUzNTMVMzUzFTM1MxUzNQUVMzUzFTM1MxUzNTMVMzUzFTM1A1JkFR0yFR37tB0VMh0VZBUdAfQdAQ8dFfwYFR1kZGRkZGRkZGRk/HxkZGRkZGRkZGT8fGRkZGRkZGRkZASwHRUyHRWWlhUdMhUdHRUyMhUd/nD9EhUdHRUC7shkZGRkZGRkZGRkyGRkZGRkZGRkZGTIZGRkZGRkZGRkZAAAAAMAAAAZBXcElwAZACUANwAAARcWFA8BBiY9ASMBISImPQE0NjsBATM1NDYBBycjIiY9ATQ2MyEBFxYUDwEGJj0BIyc3FzM1NDYEb/kPD/kOFZ/9qP7dFR0dFdECWPEV/amNetEVHR0VASMDGvkPD/kOFfG1jXqfFQSN5g4qDuYOCBWW/agdFWQVHQJYlhUI/piNeh0VZBUd/k3mDioO5g4IFZa1jXqWFQgAAAABAAAAAASwBEwAEgAAEyEyFhURFAYjIQERIyImNRE0NmQD6Ck7Oyn9rP7QZCk7OwRMOyn9qCk7/tQBLDspAlgpOwAAAAMAZAAABEwEsAAJABMAPwAAEzMyFh0BITU0NiEzMhYdASE1NDYBERQOBSIuBTURIRUUFRwBHgYyPgYmNTQ9AZbIFR3+1B0C0cgVHf7UHQEPBhgoTGacwJxmTCgYBgEsAwcNFB8nNkI2Jx8TDwUFAQSwHRX6+hUdHRX6+hUd/nD+1ClJalZcPigoPlxWakkpASz6CRIVKyclIRsWEAgJEBccISUnKhURCPoAAAAB//8A1ARMA8IABQAAAQcJAScBBEzG/p/+n8UCJwGbxwFh/p/HAicAAQAAAO4ETQPcAAUAAAkCNwkBBE392v3ZxgFhAWEDFf3ZAifH/p8BYQAAAAAC/1EAZAVfA+gAFAApAAABITIWFREzMhYPAQYiLwEmNjsBESElFxYGKwERIRchIiY1ESMiJj8BNjIBlALqFR2WFQgO5g4qDuYOCBWW/oP+HOYOCBWWAYHX/RIVHZYVCA7mDioD6B0V/dkVDvkPD/kOFQGRuPkOFf5wyB0VAiYVDvkPAAABAAYAAASeBLAAMAAAEzMyFh8BITIWBwMOASMhFyEyFhQGKwEVFAYiJj0BIRUUBiImPQEjIiYvAQMjIiY0NjheERwEJgOAGB4FZAUsIf2HMAIXFR0dFTIdKh3+1B0qHR8SHQYFyTYUHh4EsBYQoiUY/iUVK8gdKh0yFR0dFTIyFR0dFTIUCQoDwR0qHQAAAAACAAAAAASwBEwACwAPAAABFSE1MzQ2MyEyFhUFIREhBLD7UMg7KQEsKTv9RASw+1AD6GRkKTs7Kcj84AACAAAAAAXcBEwADAAQAAATAxEzNDYzITIWFSEVBQEhAcjIyDspASwqOgH0ASz+1PtQASwDIP5wAlgpOzspyGT9RAK8AAEBRQAAA2sErwAbAAABFxYGKwERMzIWDwEGIi8BJjY7AREjIiY/ATYyAnvmDggVlpYVCA7mDioO5g4IFZaWFQgO5g4qBKD5DhX9pxUO+Q8P+Q4VAlkVDvkPAAAAAQABAUQErwNrABsAAAEXFhQPAQYmPQEhFRQGLwEmND8BNhYdASE1NDYDqPkODvkPFf2oFQ/5Dg75DxUCWBUDYOUPKQ/lDwkUl5cUCQ/lDykP5Q8JFZWVFQkAAAAEAAAAAASwBLAACQAZAB0AIQAAAQMuASMhIgYHAwUhIgYdARQWMyEyNj0BNCYFNTMVMzUzFQSRrAUkFP1gFCQFrAQt/BgpOzspA+gpOzv+q2RkZAGQAtwXLSgV/R1kOylkKTs7KWQpO8hkZGRkAAAAA/+cAGQEsARMAAsAIwAxAAAAMhYVERQGIiY1ETQDJSMTFgYjIisBIiYnAj0BNDU0PgE7ASUBFSIuAz0BND4CNwRpKh0dKh1k/V0mLwMRFQUCVBQdBDcCCwzIAqP8GAQOIhoWFR0dCwRMHRX8rhUdHRUDUhX8mcj+7BAIHBUBUQ76AgQQDw36/tT6AQsTKRwyGigUDAEAAAACAEoAAARmBLAALAA1AAABMzIWDwEeARcTFzMyFhQGBw4EIyIuBC8BLgE0NjsBNxM+ATcnJjYDFjMyNw4BIiYCKV4UEgYSU3oPP3YRExwaEggeZGqfTzl0XFU+LwwLEhocExF2Pw96UxIGEyQyNDUxDDdGOASwFRMlE39N/rmtHSkoBwQLHBYSCg4REg4FBAgoKR2tAUdNfhQgExr7vgYGMT09AAEAFAAUBJwEnAAXAAABNwcXBxcHFycHJwcnBzcnNyc3Jxc3FzcDIOBO6rS06k7gLZubLeBO6rS06k7gLZubA7JO4C2bmy3gTuq0tOpO4C2bmy3gTuq0tAADAAAAZASwBLAAIQAtAD0AAAEzMhYdAQchMhYdARQHAw4BKwEiJi8BIyImNRE0PwI+ARcPAREzFzMTNSE3NQEzMhYVERQGKwEiJjURNDYCijIoPBwBSCg8He4QLBf6B0YfHz0tNxSRYA0xG2SWZIjW+v4+Mv12ZBUdHRVkFR0dBLBRLJZ9USxkLR3+qBghMhkZJCcBkCQbxMYcKGTU1f6JZAF3feGv/tQdFf4MFR0dFQH0FR0AAAAAAwAAAAAEsARMACAAMAA8AAABMzIWFxMWHQEUBiMhFh0BFAYrASImLwImNRE0NjsBNgUzMhYVERQGKwEiJjURNDYhByMRHwEzNSchNQMCWPoXLBDuHTwo/rgcPCgyGzENYJEUNy09fP3pZBUdHRVkFR0dAl+IZJZkMjIBwvoETCEY/qgdLWQsUXYHlixRKBzGxBskAZAnJGRkHRX+DBUdHRUB9BUdZP6J1dSv4X0BdwADAAAAZAUOBE8AGwA3AEcAAAElNh8BHgEPASEyFhQGKwEDDgEjISImNRE0NjcXERchEz4BOwEyNiYjISoDLgQnJj8BJwUzMhYVERQGKwEiJjURNDYBZAFrHxZuDQEMVAEuVGxuVGqDBhsP/qoHphwOOmQBJYMGGw/LFRMSFv44AgoCCQMHAwUDAQwRklb9T2QVHR0VZBUdHQNp5hAWcA0mD3lMkE7+rRUoog0CDRElCkj+CVkBUxUoMjIBAgIDBQIZFrdT5B0V/gwVHR0VAfQVHQAAAAP/nABkBLAETwAdADYARgAAAQUeBBURFAYjISImJwMjIiY0NjMhJyY2PwE2BxcWBw4FKgIjIRUzMhYXEyE3ESUFMzIWFREUBisBIiY1ETQ2AdsBbgIIFBANrAf+qg8bBoNqVW1sVAEuVQsBDW4WSpIRDAIDBQMHAwkDCgH+Jd0PHAaCASZq/qoCUGQVHR0VZBUdHQRP5gEFEBEXC/3zDaIoFQFTTpBMeQ8mDXAWrrcWGQIFAwICAWQoFf6tWQH37OQdFf4MFR0dFQH0FR0AAAADAGEAAARMBQ4AGwA3AEcAAAAyFh0BBR4BFREUBiMhIiYvAQMmPwE+AR8BETQXNTQmBhURHAMOBAcGLwEHEyE3ESUuAQMhMhYdARQGIyEiJj0BNDYB3pBOAVMVKKIN/fMRJQoJ5hAWcA0mD3nGMjIBAgIDBQIZFrdT7AH3Wf6tFSiWAfQVHR0V/gwVHR0FDm5UaoMGGw/+qgemHA4OAWsfFm4NAQxUAS5U1ssVExIW/jgCCgIJAwcDBQMBDBGSVv6tZAElgwYb/QsdFWQVHR0VZBUdAAP//QAGA+gFFAAPAC0ASQAAASEyNj0BNCYjISIGHQEUFgEVFAYiJjURBwYmLwEmNxM+BDMhMhYVERQGBwEDFzc2Fx4FHAIVERQWNj0BNDY3JREnAV4B9BUdHRX+DBUdHQEPTpBMeQ8mDXAWEOYBBRARFwsCDQ2iKBX9iexTtxYZAgUDAgIBMjIoFQFTWQRMHRVkFR0dFWQVHfzmalRubFQBLlQMAQ1uFh8BawIIEw8Mpgf+qg8bBgHP/q1WkhEMAQMFAwcDCQIKAv44FhITFcsPGwaDASVkAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgEBJSYGHQEhIgYdARQWMyEVFBY3JTY0AeLs1ptbW5vW7NabW1ubAob+7RAX/u0KDw8KARMXEAETEASaW5vW7NabW1ub1uzWm/453w0KFYkPCpYKD4kVCg3fDSYAAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgENAQYUFwUWNj0BITI2PQE0JiMhNTQmAeLs1ptbW5vW7NabW1ubASX+7RAQARMQFwETCg8PCv7tFwSaW5vW7NabW1ub1uzWm+jfDSYN3w0KFYkPCpYKD4kVCgAAAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgEBAyYiBwMGFjsBERQWOwEyNjURMzI2AeLs1ptbW5vW7NabW1ubAkvfDSYN3w0KFYkPCpYKD4kVCgSaW5vW7NabW1ub1uzWm/5AARMQEP7tEBf+7QoPDwoBExcAAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgEFIyIGFREjIgYXExYyNxM2JisBETQmAeLs1ptbW5vW7NabW1ubAZeWCg+JFQoN3w0mDd8NChWJDwSaW5vW7NabW1ub1uzWm7sPCv7tFxD+7RAQARMQFwETCg8AAAMAGAAYBJgEmAAPAJYApgAAADIeAhQOAiIuAjQ+ASUOAwcGJgcOAQcGFgcOAQcGFgcUFgcyHgEXHgIXHgI3Fg4BFx4CFxQGFBcWNz4CNy4BJy4BJyIOAgcGJyY2NS4BJzYuAQYHBicmNzY3HgIXHgMfAT4CJyY+ATc+AzcmNzIWMjY3LgMnND4CJiceAT8BNi4CJwYHFB4BFS4CJz4BNxYyPgEB5OjVm1xcm9Xo1ZtcXJsBZA8rHDoKDz0PFD8DAxMBAzEFCRwGIgEMFhkHECIvCxU/OR0HFBkDDRQjEwcFaHUeISQDDTAMD0UREi4oLBAzDwQBBikEAQMLGhIXExMLBhAGKBsGBxYVEwYFAgsFAwMNFwQGCQcYFgYQCCARFwkKKiFBCwQCAQMDHzcLDAUdLDgNEiEQEgg/KhADGgMKEgoRBJhcm9Xo1ZtcXJvV6NWbEQwRBwkCAwYFBycPCxcHInIWInYcCUcYChQECA4QBAkuHgQPJioRFRscBAcSCgwCch0kPiAIAQcHEAsBAgsLIxcBMQENCQIPHxkCFBkdHB4QBgEBBwoMGBENBAMMJSAQEhYXDQ4qFBkKEhIDCQsXJxQiBgEOCQwHAQ0DBAUcJAwSCwRnETIoAwEJCwsLJQcKDBEAAAAAAQAAAAIErwSFABYAAAE2FwUXNxYGBw4BJwEGIi8BJjQ3ASY2AvSkjv79kfsGUE08hjv9rA8rD28PDwJYIk8EhVxliuh+WYcrIgsW/awQEG4PKxACV2XJAAYAAABgBLAErAAPABMAIwAnADcAOwAAEyEyFh0BFAYjISImPQE0NgUjFTMFITIWHQEUBiMhIiY9ATQ2BSEVIQUhMhYdARQGIyEiJj0BNDYFIRUhZAPoKTs7KfwYKTs7BBHIyPwYA+gpOzsp/BgpOzsEEf4MAfT8GAPoKTs7KfwYKTs7BBH+1AEsBKw7KWQpOzspZCk7ZGTIOylkKTs7KWQpO2RkyDspZCk7OylkKTtkZAAAAAIAZAAABEwEsAALABEAABMhMhYUBiMhIiY0NgERBxEBIZYDhBUdHRX8fBUdHQI7yP6iA4QEsB0qHR0qHf1E/tTIAfQB9AAAAAMAAABkBLAEsAAXABsAJQAAATMyFh0BITIWFREhNSMVIRE0NjMhNTQ2FxUzNQEVFAYjISImPQEB9MgpOwEsKTv+DMj+DDspASw7KcgB9Dsp/BgpOwSwOylkOyn+cGRkAZApO2QpO2RkZP1EyCk7OynIAAAABAAAAAAEsASwABUAKwBBAFcAABMhMhYPARcWFA8BBiIvAQcGJjURNDYpATIWFREUBi8BBwYiLwEmND8BJyY2ARcWFA8BFxYGIyEiJjURNDYfATc2MgU3NhYVERQGIyEiJj8BJyY0PwE2MhcyASwVCA5exwcHaggUCMdeDhUdAzUBLBUdFQ5exwgUCGoHB8deDgj+L2oHB8deDggV/tQVHRUOXscIFALLXg4VHRX+1BUIDl7HBwdqCBQIBLAVDl7HCBQIagcHx14OCBUBLBUdHRX+1BUIDl7HBwdqCBQIx14OFf0maggUCMdeDhUdFQEsFQgOXscHzl4OCBX+1BUdFQ5exwgUCGoHBwAAAAYAAAAABKgEqAAPABsAIwA7AEMASwAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JiQyFhQGIiY0JDIWFAYjIicHFhUUBiImNTQ2PwImNTQEMhYUBiImNCQyFhQGIiY0Advy3Z9fX5/d8t2gXl6gAcbgv29vv+C/b2/+LS0gIC0gAUwtICAWDg83ETNIMykfegEJ/octICAtIAIdLSAgLSAEqF+f3fLdoF5eoN3y3Z9Xb7/gv29vv+C/BiAtISEtICAtIQqRFxwkMzMkIDEFfgEODhekIC0gIC0gIC0gIC0AAf/YAFoEuQS8AFsAACUBNjc2JicmIyIOAwcABw4EFx4BMzI3ATYnLgEjIgcGBwEOASY0NwA3PgEzMhceARcWBgcOBgcGIyImJyY2NwE2NzYzMhceARcWBgcBDgEnLgECIgHVWwgHdl8WGSJBMD8hIP6IDx4eLRMNBQlZN0ozAiQkEAcdEhoYDRr+qw8pHA4BRyIjQS4ODyw9DQ4YIwwod26La1YOOEBGdiIwGkQB/0coW2tQSE5nDxE4Qv4eDyoQEAOtAdZbZWKbEQQUGjIhH/6JDxsdNSg3HT5CMwIkJCcQFBcMGv6uDwEcKQ4BTSIjIQEINykvYyMLKnhuiWZMBxtAOU6+RAH/SBg3ISSGV121Qv4kDwIPDyYAAAACAGQAWASvBEQAGQBEAAABPgIeAhUUDgMHLgQ1ND4CHgEFIg4DIi4DIyIGFRQeAhcWFx4EMj4DNzY3PgQ1NCYCiTB7eHVYNkN5hKg+PqeFeEM4WnZ4eQEjIT8yLSohJyktPyJDbxtBMjMPBw86KzEhDSIzKUAMBAgrKT8dF2oDtURIBS1TdkA5eYB/slVVsn+AeTlAdlMtBUgtJjY1JiY1NiZvTRc4SjQxDwcOPCouGBgwKEALBAkpKkQqMhNPbQACADn/8gR3BL4AFwAuAAAAMh8BFhUUBg8BJi8BNycBFwcvASY0NwEDNxYfARYUBwEGIi8BJjQ/ARYfAQcXAQKru0KNQjgiHR8uEl/3/nvUaRONQkIBGxJpCgmNQkL+5UK6Qo1CQjcdLhJf9wGFBL5CjUJeKmsiHTUuEl/4/nvUahKNQrpCARv+RmkICY1CukL+5UJCjUK7Qjc3LxFf+AGFAAAAAAMAyAAAA+gEsAARABUAHQAAADIeAhURFAYjISImNRE0PgEHESERACIGFBYyNjQCBqqaZDo7Kf2oKTs8Zj4CWP7/Vj09Vj0EsB4uMhX8Ryk7OykDuRUzLar9RAK8/RY9Vj09VgABAAAAAASwBLAAFgAACQEWFAYiLwEBEScBBRMBJyEBJyY0NjIDhgEbDx0qDiT+6dT+zP7oywEz0gEsAQsjDx0qBKH+5g8qHQ8j/vX+1NL+zcsBGAE01AEXJA4qHQAAAAADAScAEQQJBOAAMgBAAEsAAAEVHgQXIy4DJxEXHgQVFAYHFSM1JicuASczHgEXEScuBDU0PgI3NRkBDgMVFB4DFxYXET4ENC4CArwmRVI8LAKfBA0dMydAIjxQNyiym2SWVygZA4sFV0obLkJOMCAyVWg6HSoqFQ4TJhkZCWgWKTEiGBkzNwTgTgUTLD9pQiQuLBsH/s0NBxMtPGQ+i6oMTU8QVyhrVk1iEAFPCA4ZLzlYNkZwSCoGTf4SARIEDh02Jh0rGRQIBgPQ/soCCRYgNEM0JRkAAAABAGQAZgOUBK0ASgAAATIeARUjNC4CIyIGBwYVFB4BFxYXMxUjFgYHBgc+ATM2FjMyNxcOAyMiLgEHDgEPASc+BTc+AScjNTMmJy4CPgE3NgIxVJlemSc8OxolVBQpGxoYBgPxxQgVFS02ImIWIIwiUzUyHzY4HCAXanQmJ1YYFzcEGAcTDBEJMAwk3aYXFQcKAg4tJGEErVCLTig/IhIdFSw5GkowKgkFZDKCHj4yCg8BIh6TExcIASIfBAMaDAuRAxAFDQsRCjePR2QvORQrREFMIVgAAAACABn//wSXBLAADwAfAAABMzIWDwEGIi8BJjY7AREzBRcWBisBESMRIyImPwE2MgGQlhUIDuYOKg7mDggVlsgCF+YOCBWWyJYVCA7mDioBLBYO+g8P+g4WA4QQ+Q4V/HwDhBUO+Q8AAAQAGf//A+gEsAAHABcAGwAlAAABIzUjFSMRIQEzMhYPAQYiLwEmNjsBETMFFTM1EwczFSE1NyM1IQPoZGRkASz9qJYVCA7mDioO5g4IFZbIAZFkY8jI/tTIyAEsArxkZAH0/HwWDvoPD/oOFgOEZMjI/RL6ZJb6ZAAAAAAEABn//wPoBLAADwAZACEAJQAAATMyFg8BBiIvASY2OwERMwUHMxUhNTcjNSERIzUjFSMRIQcVMzUBkJYVCA7mDioO5g4IFZbIAljIyP7UyMgBLGRkZAEsx2QBLBYO+g8P+g4WA4SW+mSW+mT7UGRkAfRkyMgAAAAEABn//wRMBLAADwAVABsAHwAAATMyFg8BBiIvASY2OwERMwEjESM1MxMjNSMRIQcVMzUBkJYVCA7mDioO5g4IFZbIAlhkZMhkZMgBLMdkASwWDvoPD/oOFgOE/gwBkGT7UGQBkGTIyAAAAAAEABn//wRMBLAADwAVABkAHwAAATMyFg8BBiIvASY2OwERMwEjNSMRIQcVMzUDIxEjNTMBkJYVCA7mDioO5g4IFZbIArxkyAEsx2QBZGTIASwWDvoPD/oOFgOE/gxkAZBkyMj7tAGQZAAAAAAFABn//wSwBLAADwATABcAGwAfAAABMzIWDwEGIi8BJjY7AREzBSM1MxMhNSETITUhEyE1IQGQlhUIDuYOKg7mDggVlsgB9MjIZP7UASxk/nABkGT+DAH0ASwWDvoPD/oOFgOEyMj+DMj+DMj+DMgABQAZ//8EsASwAA8AEwAXABsAHwAAATMyFg8BBiIvASY2OwERMwUhNSEDITUhAyE1IQMjNTMBkJYVCA7mDioO5g4IFZbIAyD+DAH0ZP5wAZBk/tQBLGTIyAEsFg76Dw/6DhYDhMjI/gzI/gzI/gzIAAIAAAAABEwETAAPAB8AAAEhMhYVERQGIyEiJjURNDYFISIGFREUFjMhMjY1ETQmAV4BkKK8u6P+cKW5uQJn/gwpOzspAfQpOzsETLuj/nClubmlAZClucg7Kf4MKTs7KQH0KTsAAAAAAwAAAAAETARMAA8AHwArAAABITIWFREUBiMhIiY1ETQ2BSEiBhURFBYzITI2NRE0JgUXFhQPAQYmNRE0NgFeAZClubml/nCju7wCZP4MKTs7KQH0KTs7/m/9ERH9EBgYBEy5pf5wpbm5pQGQo7vIOyn+DCk7OykB9Ck7gr4MJAy+DAsVAZAVCwAAAAADAAAAAARMBEwADwAfACsAAAEhMhYVERQGIyEiJjURNDYFISIGFREUFjMhMjY1ETQmBSEyFg8BBiIvASY2AV4BkKO7uaX+cKW5uQJn/gwpOzspAfQpOzv+FQGQFQsMvgwkDL4MCwRMvKL+cKW5uaUBkKO7yDsp/gwpOzspAfQpO8gYEP0REf0QGAAAAAMAAAAABEwETAAPAB8AKwAAASEyFhURFAYjISImNRE0NgUhIgYVERQWMyEyNjURNCYFFxYGIyEiJj8BNjIBXgGQpbm5pf5wo7u5Amf+DCk7OykB9Ck7O/77vgwLFf5wFQsMvgwkBEy5pf5wo7u8ogGQpbnIOyn+DCk7OykB9Ck7z/0QGBgQ/REAAAAAAgAAAAAFFARMAB8ANQAAASEyFhURFAYjISImPQE0NjMhMjY1ETQmIyEiJj0BNDYHARYUBwEGJj0BIyImPQE0NjsBNTQ2AiYBkKW5uaX+cBUdHRUBwik7Oyn+PhUdHb8BRBAQ/rwQFvoVHR0V+hYETLml/nCluR0VZBUdOykB9Ck7HRVkFR3p/uQOJg7+5A4KFZYdFcgVHZYVCgAAAQDZAAID1wSeACMAAAEXFgcGAgclMhYHIggBBwYrAScmNz4BPwEhIicmNzYANjc2MwMZCQgDA5gCASwYEQ4B/vf+8wQMDgkJCQUCUCcn/tIXCAoQSwENuwUJEASeCQoRC/5TBwEjEv7K/sUFDwgLFQnlbm4TFRRWAS/TBhAAAAACAAAAAAT+BEwAHwA1AAABITIWHQEUBiMhIgYVERQWMyEyFh0BFAYjISImNRE0NgUBFhQHAQYmPQEjIiY9ATQ2OwE1NDYBXgGQFR0dFf4+KTs7KQHCFR0dFf5wpbm5AvEBRBAQ/rwQFvoVHR0V+hYETB0VZBUdOyn+DCk7HRVkFR25pQGQpbnp/uQOJg7+5A4KFZYdFcgVHZYVCgACAAAAAASwBLAAFQAxAAABITIWFREUBi8BAQYiLwEmNDcBJyY2ASMiBhURFBYzITI2PQE3ERQGIyEiJjURNDYzIQLuAZAVHRUObf7IDykPjQ8PAThtDgj+75wpOzspAfQpO8i7o/5wpbm5pQEsBLAdFf5wFQgObf7IDw+NDykPAThtDhX+1Dsp/gwpOzsplMj+1qW5uaUBkKW5AAADAA4ADgSiBKIADwAbACMAAAAyHgIUDgIiLgI0PgEEIg4BFB4BMj4BNCYEMhYUBiImNAHh7tmdXV2d2e7ZnV1dnQHD5sJxccLmwnFx/nugcnKgcgSiXZ3Z7tmdXV2d2e7ZnUdxwubCcXHC5sJzcqBycqAAAAMAAAAABEwEsAAVAB8AIwAAATMyFhURMzIWBwEGIicBJjY7ARE0NgEhMhYdASE1NDYFFTM1AcLIFR31FAoO/oEOJw3+hQ0JFfod/oUD6BUd+7QdA2dkBLAdFf6iFg/+Vg8PAaoPFgFeFR38fB0V+voVHWQyMgAAAAMAAAAABEwErAAVAB8AIwAACQEWBisBFRQGKwEiJj0BIyImNwE+AQEhMhYdASE1NDYFFTM1AkcBeg4KFfQiFsgUGPoUCw4Bfw4n/fkD6BUd+7QdA2dkBJ7+TQ8g+hQeHRX6IQ8BrxAC/H8dFfr6FR1kMjIAAwAAAAAETARLABQAHgAiAAAJATYyHwEWFAcBBiInASY0PwE2MhcDITIWHQEhNTQ2BRUzNQGMAXEHFQeLBwf98wcVB/7cBweLCBUH1APoFR37tB0DZ2QC0wFxBweLCBUH/fMICAEjCBQIiwcH/dIdFfr6FR1kMjIABAAAAAAETASbAAkAGQAjACcAABM3NjIfAQcnJjQFNzYWFQMOASMFIiY/ASc3ASEyFh0BITU0NgUVMzWHjg4qDk3UTQ4CFtIOFQIBHRX9qxUIDtCa1P49A+gVHfu0HQNnZAP/jg4OTdRMDyqa0g4IFf2pFB4BFQ7Qm9T9Oh0V+voVHWQyMgAAAAQAAAAABEwEsAAPABkAIwAnAAABBR4BFRMUBi8BByc3JyY2EwcGIi8BJjQ/AQEhMhYdASE1NDYFFTM1AV4CVxQeARUO0JvUm9IOCMNMDyoOjg4OTf76A+gVHfu0HQNnZASwAgEdFf2rFQgO0JrUmtIOFf1QTQ4Ojg4qDk3+WB0V+voVHWQyMgACAAT/7ASwBK8ABQAIAAAlCQERIQkBFQEEsP4d/sb+cQSs/TMCq2cBFP5xAacDHPz55gO5AAAAAAIAAABkBEwEsAAVABkAAAERFAYrAREhESMiJjURNDY7AREhETMHIzUzBEwdFZb9RJYVHR0V+gH0ZMhkZAPo/K4VHQGQ/nAdFQPoFB7+1AEsyMgAAAMAAABFBN0EsAAWABoALwAAAQcBJyYiDwEhESMiJjURNDY7AREhETMHIzUzARcWFAcBBiIvASY0PwE2Mh8BATYyBEwC/tVfCRkJlf7IlhUdHRX6AfRkyGRkAbBqBwf+XAgUCMoICGoHFQdPASkHFQPolf7VXwkJk/5wHRUD6BQe/tQBLMjI/c5qBxUH/lsHB8sHFQdqCAhPASkHAAMAAAANBQcEsAAWABoAPgAAAREHJy4BBwEhESMiJjURNDY7AREhETMHIzUzARcWFA8BFxYUDwEGIi8BBwYiLwEmND8BJyY0PwE2Mh8BNzYyBExnhg8lEP72/reWFR0dFfoB9GTIZGQB9kYPD4ODDw9GDykPg4MPKQ9GDw+Dgw8PRg8pD4ODDykD6P7zZ4YPAw7+9v5wHRUD6BQe/tQBLMjI/YxGDykPg4MPKQ9GDw+Dgw8PRg8pD4ODDykPRg8Pg4MPAAADAAAAFQSXBLAAFQAZAC8AAAERISIGHQEhESMiJjURNDY7AREhETMHIzUzEzMyFh0BMzIWDwEGIi8BJjY7ATU0NgRM/qIVHf4MlhUdHRX6AfRkyGRklmQVHZYVCA7mDioO5g4IFZYdA+j+1B0Vlv5wHRUD6BQe/tQBLMjI/agdFfoVDuYODuYOFfoVHQAAAAADAAAAAASXBLAAFQAZAC8AAAERJyYiBwEhESMiJjURNDY7AREhETMHIzUzExcWBisBFRQGKwEiJj0BIyImPwE2MgRMpQ4qDv75/m6WFR0dFfoB9GTIZGTr5g4IFZYdFWQVHZYVCA7mDioD6P5wpQ8P/vf+cB0VA+gUHv7UASzIyP2F5Q8V+hQeHhT6FQ/lDwADAAAAyASwBEwACQATABcAABMhMhYdASE1NDYBERQGIyEiJjURExUhNTIETBUd+1AdBJMdFfu0FR1kAZAETB0VlpYVHf7U/doVHR0VAib+1MjIAAAGAAMAfQStBJcADwAZAB0ALQAxADsAAAEXFhQPAQYmPQEhNSE1NDYBIyImPQE0NjsBFyM1MwE3NhYdASEVIRUUBi8BJjQFIzU7AjIWHQEUBisBA6f4Dg74DhX+cAGQFf0vMhUdHRUyyGRk/oL3DhUBkP5wFQ73DwOBZGRkMxQdHRQzBI3mDioO5g4IFZbIlhUI/oUdFWQVHcjI/cvmDggVlsiWFQgO5g4qecgdFWQVHQAAAAACAGQAAASwBLAAFgBRAAABJTYWFREUBisBIiY1ES4ENRE0NiUyFh8BERQOAg8BERQGKwEiJjURLgQ1ETQ+AzMyFh8BETMRPAE+AjMyFh8BETMRND4DA14BFBklHRXIFR0EDiIaFiX+4RYZAgEVHR0LCh0VyBUdBA4iGhYBBwoTDRQZAgNkBQkVDxcZAQFkAQUJFQQxdBIUH/uuFR0dFQGNAQgbHzUeAWcfRJEZDA3+Phw/MSkLC/5BFR0dFQG/BA8uLkAcAcICBxENCxkMDf6iAV4CBxENCxkMDf6iAV4CBxENCwABAGQAAASwBEwAMwAAARUiDgMVERQWHwEVITUyNjURIREUFjMVITUyPgM1ETQmLwE1IRUiBhURIRE0JiM1BLAEDiIaFjIZGf5wSxn+DBlL/nAEDiIaFjIZGQGQSxkB9BlLBEw4AQUKFA78iBYZAQI4OA0lAYr+diUNODgBBQoUDgN4FhkBAjg4DSX+dgGKJQ04AAAABgAAAAAETARMAAwAHAAgACQAKAA0AAABITIWHQEjBTUnITchBSEyFhURFAYjISImNRE0NhcVITUBBTUlBRUhNQUVFAYjIQchJyE3MwKjAXcVHWn+2cj+cGQBd/4lASwpOzsp/tQpOzspASwCvP5wAZD8GAEsArwdFf6JZP6JZAGQyGkD6B0VlmJiyGTIOyn+DCk7OykB9Ck7ZMjI/veFo4XGyMhm+BUdZGTIAAEAEAAQBJ8EnwAmAAATNzYWHwEWBg8BHgEXNz4BHwEeAQ8BBiIuBicuBTcRohEuDosOBhF3ZvyNdxEzE8ATBxGjAw0uMUxPZWZ4O0p3RjITCwED76IRBhPCFDERdo78ZXYRBA6IDi8RogEECBUgNUNjO0qZfHNVQBAAAAACAAAAAASwBEwAIwBBAAAAMh4EHwEVFAYvAS4BPQEmIAcVFAYPAQYmPQE+BRIyHgIfARUBHgEdARQGIyEiJj0BNDY3ATU0PgIB/LimdWQ/LAkJHRTKFB2N/sKNHRTKFB0DDTE7ZnTKcFImFgEBAW0OFR0V+7QVHRUOAW0CFiYETBUhKCgiCgrIFRgDIgMiFZIYGJIVIgMiAxgVyAQNJyQrIP7kExwcCgoy/tEPMhTUFR0dFdQUMg8BLzIEDSEZAAADAAAAAASwBLAADQAdACcAAAEHIScRMxUzNTMVMzUzASEyFhQGKwEXITcjIiY0NgMhMhYdASE1NDYETMj9qMjIyMjIyPyuArwVHR0VDIn8SokMFR0dswRMFR37UB0CvMjIAfTIyMjI/OAdKh1kZB0qHf7UHRUyMhUdAAAAAwBkAAAEsARMAAkAEwAdAAABIyIGFREhETQmASMiBhURIRE0JgEhETQ2OwEyFhUCvGQpOwEsOwFnZCk7ASw7/Rv+1DspZCk7BEw7KfwYA+gpO/7UOyn9RAK8KTv84AGQKTs7KQAAAAAF/5wAAASwBEwADwATAB8AJQApAAATITIWFREUBiMhIiY1ETQ2FxEhEQUjFTMRITUzNSMRIQURByMRMwcRMxHIArx8sLB8/UR8sLAYA4T+DMjI/tTIyAEsAZBkyMhkZARMsHz+DHywsHwB9HywyP1EArzIZP7UZGQBLGT+1GQB9GT+1AEsAAAABf+cAAAEsARMAA8AEwAfACUAKQAAEyEyFhURFAYjISImNRE0NhcRIREBIzUjFSMRMxUzNTMFEQcjETMHETMRyAK8fLCwfP1EfLCwGAOE/gxkZGRkZGQBkGTIyGRkBEywfP4MfLCwfAH0fLDI/UQCvP2oyMgB9MjIZP7UZAH0ZP7UASwABP+cAAAEsARMAA8AEwAbACMAABMhMhYVERQGIyEiJjURNDYXESERBSMRMxUhESEFIxEzFSERIcgCvHywsHz9RHywsBgDhP4MyMj+1AEsAZDIyP7UASwETLB8/gx8sLB8AfR8sMj9RAK8yP7UZAH0ZP7UZAH0AAAABP+cAAAEsARMAA8AEwAWABkAABMhMhYVERQGIyEiJjURNDYXESERAS0BDQERyAK8fLCwfP1EfLCwGAOE/gz+1AEsAZD+1ARMsHz+DHywsHwB9HywyP1EArz+DJaWlpYBLAAAAAX/nAAABLAETAAPABMAFwAgACkAABMhMhYVERQGIyEiJjURNDYXESERAyERIQcjIgYVFBY7AQERMzI2NTQmI8gCvHywsHz9RHywsBgDhGT9RAK8ZIImOTYpgv4Mgik2OSYETLB8/gx8sLB8AfR8sMj9RAK8/agB9GRWQUFUASz+1FRBQVYAAAAF/5wAAASwBEwADwATAB8AJQApAAATITIWFREUBiMhIiY1ETQ2FxEhEQUjFTMRITUzNSMRIQEjESM1MwMjNTPIArx8sLB8/UR8sLAYA4T+DMjI/tTIyAEsAZBkZMjIZGQETLB8/gx8sLB8AfR8sMj9RAK8yGT+1GRkASz+DAGQZP4MZAAG/5wAAASwBEwADwATABkAHwAjACcAABMhMhYVERQGIyEiJjURNDYXESERBTMRIREzASMRIzUzBRUzNQEjNTPIArx8sLB8/UR8sLAYA4T9RMj+1GQCWGRkyP2oZAEsZGQETLB8/gx8sLB8AfR8sMj9RAK8yP5wAfT+DAGQZMjIyP7UZAAF/5wAAASwBEwADwATABwAIgAmAAATITIWFREUBiMhIiY1ETQ2FxEhEQEHIzU3NSM1IQEjESM1MwMjNTPIArx8sLB8/UR8sLAYA4T+DMdkx8gBLAGQZGTIx2RkBEywfP4MfLCwfAH0fLDI/UQCvP5wyDLIlmT+DAGQZP4MZAAAAAMACQAJBKcEpwAPABsAJQAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgchFSEVISc1NyEB4PDbnl5entvw255eXp4BxeTCcXHC5MJxcWz+1AEs/tRkZAEsBKdentvw255eXp7b8NueTHHC5MJxccLkwtDIZGTIZAAAAAAEAAkACQSnBKcADwAbACcAKwAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgcVBxcVIycjFSMRIQcVMzUB4PDbnl5entvw255eXp4BxeTCcXHC5MJxcWwyZGRklmQBLMjIBKdentvw255eXp7b8NueTHHC5MJxccLkwtBkMmQyZGQBkGRkZAAAAv/y/50EwgRBACAANgAAATIWFzYzMhYUBisBNTQmIyEiBh0BIyImNTQ2NyY1ND4BEzMyFhURMzIWDwEGIi8BJjY7ARE0NgH3brUsLC54qqp4gB0V/tQVHd5QcFZBAmKqepYKD4kVCg3fDSYN3w0KFYkPBEF3YQ6t8a36FR0dFfpzT0VrDhMSZKpi/bMPCv7tFxD0EBD0EBcBEwoPAAAAAAL/8v+cBMMEQQAcADMAAAEyFhc2MzIWFxQGBwEmIgcBIyImNTQ2NyY1ND4BExcWBisBERQGKwEiJjURIyImNzY3NjIB9m62LCsueaoBeFr+hg0lDf6DCU9xVkECYqnm3w0KFYkPCpYKD4kVCg3HGBMZBEF3YQ+teGOkHAFoEBD+k3NPRWsOExNkqWP9kuQQF/7tCg8PCgETFxDMGBMAAAABAGQAAARMBG0AGAAAJTUhATMBMwkBMwEzASEVIyIGHQEhNTQmIwK8AZD+8qr+8qr+1P7Uqv7yqv7yAZAyFR0BkB0VZGQBLAEsAU3+s/7U/tRkHRUyMhUdAAAAAAEAeQAABDcEmwAvAAABMhYXHgEVFAYHFhUUBiMiJxUyFh0BITU0NjM1BiMiJjU0Ny4BNTQ2MzIXNCY1NDYCWF6TGll7OzIJaUo3LRUd/tQdFS03SmkELzlpSgUSAqMEm3FZBoNaPWcfHRpKaR77HRUyMhUd+x5pShIUFVg1SmkCAhAFdKMAAAAGACcAFASJBJwAEQAqAEIASgBiAHsAAAEWEgIHDgEiJicmAhI3PgEyFgUiBw4BBwYWHwEWMzI3Njc2Nz4BLwEmJyYXIgcOAQcGFh8BFjMyNz4BNz4BLwEmJyYWJiIGFBYyNjciBw4BBw4BHwEWFxYzMjc+ATc2Ji8BJhciBwYHBgcOAR8BFhcWMzI3PgE3NiYvASYD8m9PT29T2dzZU29PT29T2dzZ/j0EBHmxIgQNDCQDBBcGG0dGYAsNAwkDCwccBAVQdRgEDA0iBAQWBhJROQwMAwkDCwf5Y4xjY4xjVhYGElE6CwwDCQMLBwgEBVB1GAQNDCIEjRcGG0dGYAsNAwkDCwcIBAR5sSIEDQwkAwPyb/7V/tVvU1dXU28BKwErb1NXVxwBIrF5DBYDCQEWYEZHGwMVDCMNBgSRAhh1UA0WAwkBFTpREgMVCyMMBwT6Y2OMY2MVFTpREQQVCyMMBwQCGHVQDRYDCQEkFmBGRxsDFQwjDQYEASKxeQwWAwkBAAAABQBkAAAD6ASwAAwADwAWABwAIgAAASERIzUhFSERNDYzIQEjNQMzByczNTMDISImNREFFRQGKwECvAEstP6s/oQPCgI/ASzIZKLU1KJktP51Cg8DhA8KwwMg/oTIyALzCg/+1Mj84NTUyP4MDwoBi8jDCg8AAAAABQBkAAAD6ASwAAkADAATABoAIQAAASERCQERNDYzIQEjNRMjFSM1IzcDISImPQEpARUUBisBNQK8ASz+ov3aDwoCPwEsyD6iZKLUqv6dCg8BfAIIDwqbAyD9+AFe/doERwoP/tTI/HzIyNT+ZA8KNzcKD1AAAAAAAwAAAAAEsAP0AAgAGQAfAAABIxUzFyERIzcFMzIeAhUhFSEDETM0PgIBMwMhASEEiqJkZP7UotT9EsgbGiEOASz9qMhkDiEaAnPw8PzgASwB9AMgyGQBLNTUBBErJGT+ogHCJCsRBP5w/nAB9AAAAAMAAAAABEwETAAZADIAOQAAATMyFh0BMzIWHQEUBiMhIiY9ATQ2OwE1NDYFNTIWFREUBiMhIic3ARE0NjMVFBYzITI2AQc1IzUzNQKKZBUdMhUdHRX+1BUdHRUyHQFzKTs7Kf2oARP2/ro7KVg+ASw+WP201MjIBEwdFTIdFWQVHR0VZBUdMhUd+pY7KfzgKTsE9gFGAUQpO5Y+WFj95tSiZKIAAwBkAAAEvARMABkANgA9AAABMzIWHQEzMhYdARQGIyEiJj0BNDY7ATU0NgU1MhYVESMRMxQOAiMhIiY1ETQ2MxUUFjMhMjYBBzUjNTM1AcJkFR0yFR0dFf7UFR0dFTIdAXMpO8jIDiEaG/2oKTs7KVg+ASw+WAGc1MjIBEwdFTIdFWQVHR0VZBUdMhUd+pY7Kf4M/tQkKxEEOykDICk7lj5YWP3m1KJkogAAAAP/ogAABRYE1AALABsAHwAACQEWBiMhIiY3ATYyEyMiBhcTHgE7ATI2NxM2JgMVMzUCkgJ9FyAs+wQsIBcCfRZARNAUGAQ6BCMUNhQjBDoEGODIBK37sCY3NyYEUCf+TB0U/tIUHR0UAS4UHf4MZGQAAAAACQAAAAAETARMAA8AHwAvAD8ATwBfAG8AfwCPAAABMzIWHQEUBisBIiY9ATQ2EzMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYBMzIWHQEUBisBIiY9ATQ2ITMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYBMzIWHQEUBisBIiY9ATQ2ITMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYBqfoKDw8K+goPDwr6Cg8PCvoKDw8BmvoKDw8K+goPD/zq+goPDwr6Cg8PAZr6Cg8PCvoKDw8BmvoKDw8K+goPD/zq+goPDwr6Cg8PAZr6Cg8PCvoKDw8BmvoKDw8K+goPDwRMDwqWCg8PCpYKD/7UDwqWCg8PCpYKDw8KlgoPDwqWCg/+1A8KlgoPDwqWCg8PCpYKDw8KlgoPDwqWCg8PCpYKD/7UDwqWCg8PCpYKDw8KlgoPDwqWCg8PCpYKDw8KlgoPAAAAAwAAAAAEsAUUABkAKQAzAAABMxUjFSEyFg8BBgchJi8BJjYzITUjNTM1MwEhMhYUBisBFyE3IyImNDYDITIWHQEhNTQ2ArxkZAFePjEcQiko/PwoKUIcMT4BXmRkyP4+ArwVHR0VDIn8SooNFR0dswRMFR37UB0EsMhkTzeEUzMzU4Q3T2TIZPx8HSodZGQdKh3+1B0VMjIVHQAABAAAAAAEsAUUAAUAGQArADUAAAAyFhUjNAchFhUUByEyFg8BIScmNjMhJjU0AyEyFhQGKwEVBSElNSMiJjQ2AyEyFh0BITU0NgIwUDnCPAE6EgMBSCkHIq/9WrIiCikBSAOvArwVHR0VlgET/EoBE5YVHR2zBEwVHftQHQUUOykpjSUmCBEhFpGRFiERCCb+lR0qHcjIyMgdKh39qB0VMjIVHQAEAAAAAASwBJ0ABwAUACQALgAAADIWFAYiJjQTMzIWFRQXITY1NDYzASEyFhQGKwEXITcjIiY0NgMhMhYdASE1NDYCDZZqapZqty4iKyf+vCcrI/7NArwVHR0VDYr8SokMFR0dswRMFR37UB0EnWqWamqW/us5Okxra0w6Of5yHSodZGQdKh3+1B0VMjIVHQAEAAAAAASwBRQADwAcACwANgAAATIeARUUBiImNTQ3FzcnNhMzMhYVFBchNjU0NjMBITIWFAYrARchNyMiJjQ2AyEyFh0BITU0NgJYL1szb5xvIpBvoyIfLiIrJ/68Jysj/s0CvBUdHRUNivxKiQwVHR2zBEwVHftQHQUUa4s2Tm9vTj5Rj2+jGv4KOTpMa2tMOjn+ch0qHWRkHSod/tQdFTIyFR0AAAADAAAAAASwBRIAEgAiACwAAAEFFSEUHgMXIS4BNTQ+AjcBITIWFAYrARchNyMiJjQ2AyEyFh0BITU0NgJYASz+1CU/P00T/e48PUJtj0r+ogK8FR0dFQ2K/EqJDBUdHbMETBUd+1AdBLChizlmUT9IGVO9VFShdksE/H4dKh1kZB0qHf7UHRUyMhUdAAIAyAAAA+gFFAAPACkAAAAyFh0BHgEdASE1NDY3NTQDITIWFyMVMxUjFTMVIxUzFAYjISImNRE0NgIvUjsuNv5wNi5kAZA2XBqsyMjIyMh1U/5wU3V1BRQ7KU4aXDYyMjZcGk4p/kc2LmRkZGRkU3V1UwGQU3UAAAMAZP//BEwETAAPAC8AMwAAEyEyFhURFAYjISImNRE0NgMhMhYdARQGIyEXFhQGIi8BIQcGIiY0PwEhIiY9ATQ2BQchJ5YDhBUdHRX8fBUdHQQDtgoPDwr+5eANGiUNWP30Vw0mGg3g/t8KDw8BqmQBRGQETB0V/gwVHR0VAfQVHf1EDwoyCg/gDSUbDVhYDRslDeAPCjIKD2RkZAAAAAAEAAAAAASwBEwAGQAjAC0ANwAAEyEyFh0BIzQmKwEiBhUjNCYrASIGFSM1NDYDITIWFREhETQ2ExUUBisBIiY9ASEVFAYrASImPQHIAyBTdWQ7KfopO2Q7KfopO2R1EQPoKTv7UDvxHRVkFR0D6B0VZBUdBEx1U8gpOzspKTs7KchTdf4MOyn+1AEsKTv+DDIVHR0VMjIVHR0VMgADAAEAAASpBKwADQARABsAAAkBFhQPASEBJjQ3ATYyCQMDITIWHQEhNTQ2AeACqh8fg/4f/fsgIAEnH1n+rAFWAS/+q6IDIBUd/HwdBI39VR9ZH4MCBh9ZHwEoH/5u/qoBMAFV/BsdFTIyFR0AAAAAAgCPAAAEIQSwABcALwAAAQMuASMhIgYHAwYWMyEVFBYyNj0BMzI2AyE1NDY7ATU0NjsBETMRMzIWHQEzMhYVBCG9CCcV/nAVJwi9CBMVAnEdKh19FROo/a0dFTIdFTDILxUdMhUdAocB+hMcHBP+BhMclhUdHRWWHP2MMhUdMhUdASz+1B0VMh0VAAAEAAAAAASwBLAADQAQAB8AIgAAASERFAYjIREBNTQ2MyEBIzUBIREUBiMhIiY1ETQ2MyEBIzUDhAEsDwr+if7UDwoBdwEsyP2oASwPCv12Cg8PCgF3ASzIAyD9wQoPAk8BLFQKD/7UyP4M/cEKDw8KA7YKD/7UyAAC/5wAZAUUBEcARgBWAAABMzIeAhcWFxY2NzYnJjc+ARYXFgcOASsBDgEPAQ4BKwEiJj8BBisBIicHDgErASImPwEmLwEuAT0BNDY7ATY3JyY2OwE2BSMiBh0BFBY7ATI2PQE0JgHkw0uOakkMEhEfQwoKGRMKBQ8XDCkCA1Y9Pgc4HCcDIhVkFRgDDDEqwxgpCwMiFWQVGAMaVCyfExwdFXwLLW8QBxXLdAFF+goPDwr6Cg8PBEdBa4pJDgYKISAiJRsQCAYIDCw9P1c3fCbqFB0dFEYOCEAUHR0UnUplNQcmFTIVHVdPXw4TZV8PCjIKDw8KMgoPAAb/nP/mBRQEfgAJACQANAA8AFIAYgAAASU2Fh8BFgYPASUzMhYfASEyFh0BFAYHBQYmJyYjISImPQE0NhcjIgYdARQ7ATI2NTQmJyYEIgYUFjI2NAE3PgEeARceAT8BFxYGDwEGJi8BJjYlBwYfAR4BPwE2Jy4BJy4BAoEBpxMuDiAOAxCL/CtqQ0geZgM3FR0cE/0fFyIJKjr+1D5YWLlQExIqhhALIAsSAYBALS1ALf4PmBIgHhMQHC0aPzANITNQL3wpgigJASlmHyElDR0RPRMFAhQHCxADhPcICxAmDyoNeMgiNtQdFTIVJgeEBBQPQ1g+yD5YrBwVODMQEAtEERzJLUAtLUD+24ITChESEyMgAwWzPUkrRSgJL5cvfRxYGyYrDwkLNRAhFEgJDAQAAAAAAwBkAAAEOQSwAFEAYABvAAABMzIWHQEeARcWDgIPATIeBRUUDgUjFRQGKwEiJj0BIxUUBisBIiY9ASMiJj0BNDY7AREjIiY9ATQ2OwE1NDY7ATIWHQEzNTQ2AxUhMj4CNTc0LgMjARUhMj4CNTc0LgMjAnGWCg9PaAEBIC4uEBEGEjQwOiodFyI2LUAjGg8KlgoPZA8KlgoPrwoPDwpLSwoPDwqvDwqWCg9kD9cBBxwpEwsBAQsTKRz++QFrHCkTCwEBCxMpHASwDwptIW1KLk0tHwYGAw8UKDJOLTtdPCoVCwJLCg8PCktLCg8PCksPCpYKDwJYDwqWCg9LCg8PCktLCg/+1MgVHR0LCgQOIhoW/nDIFR0dCwoEDiIaFgAAAwAEAAIEsASuABcAKQAsAAATITIWFREUBg8BDgEjISImJy4CNRE0NgQiDgQPARchNy4FAyMT1AMMVnokEhIdgVL9xFKCHAgYKHoCIIx9VkcrHQYGnAIwnAIIIClJVSGdwwSuelb+YDO3QkJXd3ZYHFrFMwGgVnqZFyYtLSUMDPPzBQ8sKDEj/sIBBQACAMgAAAOEBRQADwAZAAABMzIWFREUBiMhIiY1ETQ2ARUUBisBIiY9AQHblmesVCn+PilUrAFINhWWFTYFFKxn/gwpVFQpAfRnrPwY4RU2NhXhAAACAMgAAAOEBRQADwAZAAABMxQWMxEUBiMhIiY1ETQ2ARUUBisBIiY9AQHbYLOWVCn+PilUrAFINhWWFTYFFJaz/kIpVFQpAfRnrPwY4RU2NhXhAAACAAAAFAUOBBoAFAAaAAAJASUHFRcVJwc1NzU0Jj4CPwEnCQEFJTUFJQUO/YL+hk5klpZkAQEBBQQvkwKCAVz+ov6iAV4BXgL//uWqPOCWx5SVyJb6BA0GCgYDKEEBG/1ipqaTpaUAAAMAZAH0BLADIAAHAA8AFwAAEjIWFAYiJjQkMhYUBiImNCQyFhQGIiY0vHxYWHxYAeh8WFh8WAHofFhYfFgDIFh8WFh8WFh8WFh8WFh8WFh8AAAAAAMBkAAAArwETAAHAA8AFwAAADIWFAYiJjQSMhYUBiImNBIyFhQGIiY0Aeh8WFh8WFh8WFh8WFh8WFh8WARMWHxYWHz+yFh8WFh8/shYfFhYfAAAAAMAZABkBEwETAAPAB8ALwAAEyEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2fQO2Cg8PCvxKCg8PCgO2Cg8PCvxKCg8PCgO2Cg8PCvxKCg8PBEwPCpYKDw8KlgoP/nAPCpYKDw8KlgoP/nAPCpYKDw8KlgoPAAAABAAAAAAEsASwAA8AHwAvADMAAAEhMhYVERQGIyEiJjURNDYFISIGFREUFjMhMjY1ETQmBSEyFhURFAYjISImNRE0NhcVITUBXgH0ory7o/4Mpbm5Asv9qCk7OykCWCk7O/2xAfQVHR0V/gwVHR1HAZAEsLuj/gylubmlAfSlucg7Kf2oKTs7KQJYKTtkHRX+1BUdHRUBLBUdZMjIAAAAAAEAZABkBLAETAA7AAATITIWFAYrARUzMhYUBisBFTMyFhQGKwEVMzIWFAYjISImNDY7ATUjIiY0NjsBNSMiJjQ2OwE1IyImNDaWA+gVHR0VMjIVHR0VMjIVHR0VMjIVHR0V/BgVHR0VMjIVHR0VMjIVHR0VMjIVHR0ETB0qHcgdKh3IHSodyB0qHR0qHcgdKh3IHSodyB0qHQAAAAYBLAAFA+gEowAHAA0AEwAZAB8AKgAAAR4BBgcuATYBMhYVIiYlFAYjNDYBMhYVIiYlFAYjNDYDFRQGIiY9ARYzMgKKVz8/V1c/P/75fLB8sAK8sHyw/cB8sHywArywfLCwHSodKAMRBKNDsrJCQrKy/sCwfLB8fLB8sP7UsHywfHywfLD+05AVHR0VjgQAAAH/tQDIBJQDgQBCAAABNzYXAR4BBw4BKwEyFRQOBCsBIhE0NyYiBxYVECsBIi4DNTQzIyImJyY2NwE2HwEeAQ4BLwEHIScHBi4BNgLpRRkUASoLCAYFGg8IAQQNGyc/KZK4ChRUFQu4jjBJJxkHAgcPGQYGCAsBKhQaTBQVCiMUM7YDe7YsFCMKFgNuEwYS/tkLHw8OEw0dNkY4MhwBIBgXBAQYF/7gKjxTQyMNEw4PHwoBKBIHEwUjKBYGDMHBDAUWKCMAAAAAAgAAAAAEsASwACUAQwAAASM0LgUrAREUFh8BFSE1Mj4DNREjIg4FFSMRIQEjNC4DKwERFBYXMxUjNTI1ESMiDgMVIzUhBLAyCAsZEyYYGcgyGRn+cAQOIhoWyBkYJhMZCwgyA+j9RBkIChgQEWQZDQzIMmQREBgKCBkB9AOEFSAVDggDAfyuFhkBAmRkAQUJFQ4DUgEDCA4VIBUBLP0SDxMKBQH+VwsNATIyGQGpAQUKEw+WAAAAAAMAAAAABEwErgAdACAAMAAAATUiJy4BLwEBIwEGBw4BDwEVITUiJj8BIRcWBiMVARsBARUUBiMhIiY9ATQ2MyEyFgPoGR4OFgUE/t9F/tQSFQkfCwsBETE7EkUBJT0NISf+7IZ5AbEdFfwYFR0dFQPoFR0BLDIgDiIKCwLr/Q4jFQkTBQUyMisusKYiQTIBhwFW/qr942QVHR0VZBUdHQADAAAAAASwBLAADwBHAEoAABMhMhYVERQGIyEiJjURNDYFIyIHAQYHBgcGHQEUFjMhMjY9ATQmIyInJj8BIRcWBwYjIgYdARQWMyEyNj0BNCYnIicmJyMBJhMjEzIETBUdHRX7tBUdHQJGRg0F/tUREhImDAsJAREIDAwINxAKCj8BCjkLEQwYCAwMCAE5CAwLCBEZGQ8B/uAFDsVnBLAdFfu0FR0dFQRMFR1SDP0PIBMSEAUNMggMDAgyCAwXDhmjmR8YEQwIMggMDAgyBwwBGRskAuwM/gUBCAAABAAAAAAEsASwAAMAEwAjACcAAAEhNSEFITIWFREUBiMhIiY1ETQ2KQEyFhURFAYjISImNRE0NhcRIREEsPtQBLD7ggGQFR0dFf5wFR0dAm0BkBUdHRX+cBUdHUcBLARMZMgdFfx8FR0dFQOEFR0dFf5wFR0dFQGQFR1k/tQBLAAEAAAAAASwBLAADwAfACMAJwAAEyEyFhURFAYjISImNRE0NgEhMhYVERQGIyEiJjURNDYXESEREyE1ITIBkBUdHRX+cBUdHQJtAZAVHR0V/nAVHR1HASzI+1AEsASwHRX8fBUdHRUDhBUd/gwdFf5wFR0dFQGQFR1k/tQBLP2oZAAAAAACAAAAZASwA+gAJwArAAATITIWFREzNTQ2MyEyFh0BMxUjFRQGIyEiJj0BIxEUBiMhIiY1ETQ2AREhETIBkBUdZB0VAZAVHWRkHRX+cBUdZB0V/nAVHR0CnwEsA+gdFf6ilhUdHRWWZJYVHR0Vlv6iFR0dFQMgFR3+1P7UASwAAAQAAAAABLAEsAADABMAFwAnAAAzIxEzFyEyFhURFAYjISImNRE0NhcRIREBITIWFREUBiMhIiY1ETQ2ZGRklgGQFR0dFf5wFR0dRwEs/qIDhBUdHRX8fBUdHQSwZB0V/nAVHR0VAZAVHWT+1AEs/gwdFf5wFR0dFQGQFR0AAAAAAgBkAAAETASwACcAKwAAATMyFhURFAYrARUhMhYVERQGIyEiJjURNDYzITUjIiY1ETQ2OwE1MwcRIRECWJYVHR0VlgHCFR0dFfx8FR0dFQFelhUdHRWWZMgBLARMHRX+cBUdZB0V/nAVHR0VAZAVHWQdFQGQFR1kyP7UASwAAAAEAAAAAASwBLAAAwATABcAJwAAISMRMwUhMhYVERQGIyEiJjURNDYXESERASEyFhURFAYjISImNRE0NgSwZGT9dgGQFR0dFf5wFR0dRwEs/K4DhBUdHRX8fBUdHQSwZB0V/nAVHR0VAZAVHWT+1AEs/gwdFf5wFR0dFQGQFR0AAAEBLAAwA28EgAAPAAAJAQYjIiY1ETQ2MzIXARYUA2H+EhcSDhAQDhIXAe4OAjX+EhcbGQPoGRsX/hIOKgAAAAABAUEAMgOEBH4ACwAACQE2FhURFAYnASY0AU8B7h0qKh3+Eg4CewHuHREp/BgpER0B7g4qAAAAAAEAMgFBBH4DhAALAAATITIWBwEGIicBJjZkA+gpER3+Eg4qDv4SHREDhCod/hIODgHuHSoAAAAAAQAyASwEfgNvAAsAAAkBFgYjISImNwE2MgJ7Ae4dESn8GCkRHQHuDioDYf4SHSoqHQHuDgAAAAACAAgAAASwBCgABgAKAAABFQE1LQE1ASE1IQK8/UwBnf5jBKj84AMgAuW2/r3dwcHd+9jIAAAAAAIAAABkBLAEsAALADEAAAEjFTMVIREzNSM1IQEzND4FOwERFAYPARUhNSIuAzURMzIeBRUzESEEsMjI/tTIyAEs+1AyCAsZEyYYGWQyGRkBkAQOIhoWZBkYJhMZCwgy/OADhGRkASxkZP4MFSAVDggDAf3aFhkBAmRkAQUJFQ4CJgEDCA4VIBUBLAAAAgAAAAAETAPoACUAMQAAASM0LgUrAREUFh8BFSE1Mj4DNREjIg4FFSMRIQEjFTMVIREzNSM1IQMgMggLGRMmGBlkMhkZ/nAEDiIaFmQZGCYTGQsIMgMgASzIyP7UyMgBLAK8FSAVDggDAf3aFhkCAWRkAQUJFQ4CJgEDCA4VIBUBLPzgZGQBLGRkAAABAMgAZgNyBEoAEgAAATMyFgcJARYGKwEiJwEmNDcBNgK9oBAKDP4wAdAMChCgDQr+KQcHAdcKBEoWDP4w/jAMFgkB1wgUCAHXCQAAAQE+AGYD6ARKABIAAAEzMhcBFhQHAQYrASImNwkBJjYBU6ANCgHXBwf+KQoNoBAKDAHQ/jAMCgRKCf4pCBQI/ikJFgwB0AHQDBYAAAEAZgDIBEoDcgASAAAAFh0BFAcBBiInASY9ATQ2FwkBBDQWCf4pCBQI/ikJFgwB0AHQA3cKEKANCv4pBwcB1woNoBAKDP4wAdAAAAABAGYBPgRKA+gAEgAACQEWHQEUBicJAQYmPQE0NwE2MgJqAdcJFgz+MP4wDBYJAdcIFAPh/ikKDaAQCgwB0P4wDAoQoA0KAdcHAAAAAgDZ//kEPQSwAAUAOgAAARQGIzQ2BTMyFh8BNjc+Ah4EBgcOBgcGIiYjIgYiJy4DLwEuAT4EHgEXJyY2A+iwfLD+VmQVJgdPBQsiKFAzRyorDwURAQQSFyozTSwNOkkLDkc3EDlfNyYHBw8GDyUqPjdGMR+TDA0EsHywfLDIHBPCAQIGBwcFDx81S21DBxlLR1xKQhEFBQcHGWt0bCQjP2hJNyATBwMGBcASGAAAAAACAMgAFQOEBLAAFgAaAAATITIWFREUBisBEQcGJjURIyImNRE0NhcVITX6AlgVHR0Vlv8TGpYVHR2rASwEsB0V/nAVHf4MsgkQFQKKHRUBkBUdZGRkAAAAAgDIABkETASwAA4AEgAAEyEyFhURBRElIREjETQ2ARU3NfoC7ic9/UQCWP1EZB8BDWQEsFEs/Ft1A7Z9/BgEARc0/V1kFGQAAQAAAAECTW/DBF9fDzz1AB8EsAAAAADQdnOXAAAAANB2c5f/Uf+cBdwFFAAAAAgAAgAAAAAAAAABAAAFFP+FAAAFFP9R/tQF3AABAAAAAAAAAAAAAAAAAAAAowG4ACgAAAAAAZAAAASwAAAEsABkBLAAAASwAAAEsABwAooAAAUUAAACigAABRQAAAGxAAABRQAAANgAAADYAAAAogAAAQQAAABIAAABBAAAAUUAAASwAGQEsAB7BLAAyASwAMgB9AAABLD/8gSwAAAEsAAABLD/8ASwAAAEsAAOBLAACQSwAGQEsP/TBLD/0wSwAAAEsAAABLAAAASwAAAEsAAABLAAJgSwAG4EsAAXBLAAFwSwABcEsABkBLAAGgSwAGQEsAAMBLAAZASwABcEsP+cBLAAZASwABcEsAAXBLAAAASwABcEsAAXBLAAFwSwAGQEsAAABLAAZASwAAAEsAAABLAAAASwAAAEsAAABLAAAASwAAAEsAAABLAAZASwAMgEsAAABLAAAASwADUEsABkBLAAyASw/7UEsAAhBLAAAASwAAAEsAAABLAAAASwAAAEsP+cBLAAAASwAAAEsAAABLAA2wSwABcEsAB1BLAAAASwAAAEsAAABLAACgSwAMgEsAAABLAAnQSwAMgEsADIBLAAyASwAAAEsP/+BLABLASwAGQEsACIBLABOwSwABcEsAAXBLAAFwSwABcEsAAXBLAAFwSwAAAEsAAXBLAAFwSwABcEsAAXBLAAAASwALcEsAC3BLAAAASwAAAEsABJBLAAFwSwAAAEsAAABLAAXQSw/9wEsP/cBLD/nwSwAGQEsAAABLAAAASwAAAEsABkBLD//wSwAAAEsP9RBLAABgSwAAAEsAAABLABRQSwAAEEsAAABLD/nASwAEoEsAAUBLAAAASwAAAEsAAABLD/nASwAGEEsP/9BLAAFgSwABYEsAAWBLAAFgSwABgEsAAABMQAAASwAGQAAAAAAAD/2ABkADkAyAAAAScAZAAZABkAGQAZABkAGQAZAAAAAAAAAAAAAADZAAAAAAAOAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAMAZABkAAAAEAAAAAAAZP+c/5z/nP+c/5z/nP+c/5wACQAJ//L/8gBkAHkAJwBkAGQAAAAAAGT/ogAAAAAAAAAAAAAAAADIAGQAAAABAI8AAP+c/5wAZAAEAMgAyAAAAGQBkABkAAAAZAEs/7UAAAAAAAAAAAAAAAAAAABkAAABLAFBADIAMgAIAAAAAADIAT4AZgBmANkAyADIAAAAKgAqACoAKgCyAOgA6AFOAU4BTgFOAU4BTgFOAU4BTgFOAU4BTgFOAU4BpAIGAiICfgKGAqwC5ANGA24DjAPEBAgEMgRiBKIE3AVcBboGcgb0ByAHYgfKCB4IYgi+CTYJhAm2Cd4KKApMCpQK4gswC4oLygwIDFgNKg1eDbAODg5oDrQPKA+mD+YQEhBUEJAQqhEqEXYRthIKEjgSfBLAExoTdBPQFCoU1BU8FagVzBYEFjYWYBawFv4XUhemGAIYLhhqGJYYsBjgGP4ZKBloGZQZxBnaGe4aNhpoGrga9hteG7QcMhyUHOIdHB1EHWwdlB28HeYeLh52HsAfYh/SIEYgviEyIXYhuCJAIpYiuCMOIyIjOCN6I8Ij4CQCJDAkXiSWJOIlNCVgJbwmFCZ+JuYnUCe8J/goNChwKKwpoCnMKiYqSiqEKworeiwILGgsuizsLRwtiC30LiguZi6iLtgvDi9GL34vsi/4MD4whDDSMRIxYDGuMegyJDJeMpoy3jMiMz4zaDO2NBg0YDSoNNI1LDWeNeg2PjZ8Ntw3GjdON5I31DgQOEI4hjjIOQo5SjmIOcw6HDpsOpo63jugO9w8GDxQPKI8+D0yPew+Oj6MPtQ/KD9uP6o/+kBIQIBAxkECQX5CGEKoQu5DGENCQ3ZDoEPKRBBEYESuRPZFWkW2RgZGdEa0RvZHNkd2R7ZH9kgWSDJITkhqSIZIzEkSSThJXkmESapKAkouSlIAAQAAARcApwARAAAAAAACAAAAAQABAAAAQAAuAAAAAAAAABAAxgABAAAAAAATABIAAAADAAEECQAAAGoAEgADAAEECQABACgAfAADAAEECQACAA4ApAADAAEECQADAEwAsgADAAEECQAEADgA/gADAAEECQAFAHgBNgADAAEECQAGADYBrgADAAEECQAIABYB5AADAAEECQAJABYB+gADAAEECQALACQCEAADAAEECQAMACQCNAADAAEECQATACQCWAADAAEECQDIABYCfAADAAEECQDJADACkgADAAEECdkDABoCwnd3dy5nbHlwaGljb25zLmNvbQBDAG8AcAB5AHIAaQBnAGgAdAAgAKkAIAAyADAAMQA0ACAAYgB5ACAASgBhAG4AIABLAG8AdgBhAHIAaQBrAC4AIABBAGwAbAAgAHIAaQBnAGgAdABzACAAcgBlAHMAZQByAHYAZQBkAC4ARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzAFIAZQBnAHUAbABhAHIAMQAuADAAMAA5ADsAVQBLAFcATgA7AEcATABZAFAASABJAEMATwBOAFMASABhAGwAZgBsAGkAbgBnAHMALQBSAGUAZwB1AGwAYQByAEcATABZAFAASABJAEMATwBOAFMAIABIAGEAbABmAGwAaQBuAGcAcwAgAFIAZQBnAHUAbABhAHIAVgBlAHIAcwBpAG8AbgAgADEALgAwADAAOQA7AFAAUwAgADAAMAAxAC4AMAAwADkAOwBoAG8AdABjAG8AbgB2ACAAMQAuADAALgA3ADAAOwBtAGEAawBlAG8AdABmAC4AbABpAGIAMgAuADUALgA1ADgAMwAyADkARwBMAFkAUABIAEkAQwBPAE4AUwBIAGEAbABmAGwAaQBuAGcAcwAtAFIAZQBnAHUAbABhAHIASgBhAG4AIABLAG8AdgBhAHIAaQBrAEoAYQBuACAASwBvAHYAYQByAGkAawB3AHcAdwAuAGcAbAB5AHAAaABpAGMAbwBuAHMALgBjAG8AbQB3AHcAdwAuAGcAbAB5AHAAaABpAGMAbwBuAHMALgBjAG8AbQB3AHcAdwAuAGcAbAB5AHAAaABpAGMAbwBuAHMALgBjAG8AbQBXAGUAYgBmAG8AbgB0ACAAMQAuADAAVwBlAGQAIABPAGMAdAAgADIAOQAgADAANgA6ADMANgA6ADAANwAgADIAMAAxADQARgBvAG4AdAAgAFMAcQB1AGkAcgByAGUAbAAAAAIAAAAAAAD/tQAyAAAAAAAAAAAAAAAAAAAAAAAAAAABFwAAAQIBAwADAA0ADgEEAJYBBQEGAQcBCAEJAQoBCwEMAQ0BDgEPARABEQESARMA7wEUARUBFgEXARgBGQEaARsBHAEdAR4BHwEgASEBIgEjASQBJQEmAScBKAEpASoBKwEsAS0BLgEvATABMQEyATMBNAE1ATYBNwE4ATkBOgE7ATwBPQE+AT8BQAFBAUIBQwFEAUUBRgFHAUgBSQFKAUsBTAFNAU4BTwFQAVEBUgFTAVQBVQFWAVcBWAFZAVoBWwFcAV0BXgFfAWABYQFiAWMBZAFlAWYBZwFoAWkBagFrAWwBbQFuAW8BcAFxAXIBcwF0AXUBdgF3AXgBeQF6AXsBfAF9AX4BfwGAAYEBggGDAYQBhQGGAYcBiAGJAYoBiwGMAY0BjgGPAZABkQGSAZMBlAGVAZYBlwGYAZkBmgGbAZwBnQGeAZ8BoAGhAaIBowGkAaUBpgGnAagBqQGqAasBrAGtAa4BrwGwAbEBsgGzAbQBtQG2AbcBuAG5AboBuwG8Ab0BvgG/AcABwQHCAcMBxAHFAcYBxwHIAckBygHLAcwBzQHOAc8B0AHRAdIB0wHUAdUB1gHXAdgB2QHaAdsB3AHdAd4B3wHgAeEB4gHjAeQB5QHmAecB6AHpAeoB6wHsAe0B7gHvAfAB8QHyAfMB9AH1AfYB9wH4AfkB+gH7AfwB/QH+Af8CAAIBAgICAwIEAgUCBgIHAggCCQIKAgsCDAINAg4CDwIQAhECEgZnbHlwaDEGZ2x5cGgyB3VuaTAwQTAHdW5pMjAwMAd1bmkyMDAxB3VuaTIwMDIHdW5pMjAwMwd1bmkyMDA0B3VuaTIwMDUHdW5pMjAwNgd1bmkyMDA3B3VuaTIwMDgHdW5pMjAwOQd1bmkyMDBBB3VuaTIwMkYHdW5pMjA1RgRFdXJvB3VuaTIwQkQHdW5pMjMxQgd1bmkyNUZDB3VuaTI2MDEHdW5pMjZGQQd1bmkyNzA5B3VuaTI3MEYHdW5pRTAwMQd1bmlFMDAyB3VuaUUwMDMHdW5pRTAwNQd1bmlFMDA2B3VuaUUwMDcHdW5pRTAwOAd1bmlFMDA5B3VuaUUwMTAHdW5pRTAxMQd1bmlFMDEyB3VuaUUwMTMHdW5pRTAxNAd1bmlFMDE1B3VuaUUwMTYHdW5pRTAxNwd1bmlFMDE4B3VuaUUwMTkHdW5pRTAyMAd1bmlFMDIxB3VuaUUwMjIHdW5pRTAyMwd1bmlFMDI0B3VuaUUwMjUHdW5pRTAyNgd1bmlFMDI3B3VuaUUwMjgHdW5pRTAyOQd1bmlFMDMwB3VuaUUwMzEHdW5pRTAzMgd1bmlFMDMzB3VuaUUwMzQHdW5pRTAzNQd1bmlFMDM2B3VuaUUwMzcHdW5pRTAzOAd1bmlFMDM5B3VuaUUwNDAHdW5pRTA0MQd1bmlFMDQyB3VuaUUwNDMHdW5pRTA0NAd1bmlFMDQ1B3VuaUUwNDYHdW5pRTA0Nwd1bmlFMDQ4B3VuaUUwNDkHdW5pRTA1MAd1bmlFMDUxB3VuaUUwNTIHdW5pRTA1Mwd1bmlFMDU0B3VuaUUwNTUHdW5pRTA1Ngd1bmlFMDU3B3VuaUUwNTgHdW5pRTA1OQd1bmlFMDYwB3VuaUUwNjIHdW5pRTA2Mwd1bmlFMDY0B3VuaUUwNjUHdW5pRTA2Ngd1bmlFMDY3B3VuaUUwNjgHdW5pRTA2OQd1bmlFMDcwB3VuaUUwNzEHdW5pRTA3Mgd1bmlFMDczB3VuaUUwNzQHdW5pRTA3NQd1bmlFMDc2B3VuaUUwNzcHdW5pRTA3OAd1bmlFMDc5B3VuaUUwODAHdW5pRTA4MQd1bmlFMDgyB3VuaUUwODMHdW5pRTA4NAd1bmlFMDg1B3VuaUUwODYHdW5pRTA4Nwd1bmlFMDg4B3VuaUUwODkHdW5pRTA5MAd1bmlFMDkxB3VuaUUwOTIHdW5pRTA5Mwd1bmlFMDk0B3VuaUUwOTUHdW5pRTA5Ngd1bmlFMDk3B3VuaUUxMDEHdW5pRTEwMgd1bmlFMTAzB3VuaUUxMDQHdW5pRTEwNQd1bmlFMTA2B3VuaUUxMDcHdW5pRTEwOAd1bmlFMTA5B3VuaUUxMTAHdW5pRTExMQd1bmlFMTEyB3VuaUUxMTMHdW5pRTExNAd1bmlFMTE1B3VuaUUxMTYHdW5pRTExNwd1bmlFMTE4B3VuaUUxMTkHdW5pRTEyMAd1bmlFMTIxB3VuaUUxMjIHdW5pRTEyMwd1bmlFMTI0B3VuaUUxMjUHdW5pRTEyNgd1bmlFMTI3B3VuaUUxMjgHdW5pRTEyOQd1bmlFMTMwB3VuaUUxMzEHdW5pRTEzMgd1bmlFMTMzB3VuaUUxMzQHdW5pRTEzNQd1bmlFMTM2B3VuaUUxMzcHdW5pRTEzOAd1bmlFMTM5B3VuaUUxNDAHdW5pRTE0MQd1bmlFMTQyB3VuaUUxNDMHdW5pRTE0NAd1bmlFMTQ1B3VuaUUxNDYHdW5pRTE0OAd1bmlFMTQ5B3VuaUUxNTAHdW5pRTE1MQd1bmlFMTUyB3VuaUUxNTMHdW5pRTE1NAd1bmlFMTU1B3VuaUUxNTYHdW5pRTE1Nwd1bmlFMTU4B3VuaUUxNTkHdW5pRTE2MAd1bmlFMTYxB3VuaUUxNjIHdW5pRTE2Mwd1bmlFMTY0B3VuaUUxNjUHdW5pRTE2Ngd1bmlFMTY3B3VuaUUxNjgHdW5pRTE2OQd1bmlFMTcwB3VuaUUxNzEHdW5pRTE3Mgd1bmlFMTczB3VuaUUxNzQHdW5pRTE3NQd1bmlFMTc2B3VuaUUxNzcHdW5pRTE3OAd1bmlFMTc5B3VuaUUxODAHdW5pRTE4MQd1bmlFMTgyB3VuaUUxODMHdW5pRTE4NAd1bmlFMTg1B3VuaUUxODYHdW5pRTE4Nwd1bmlFMTg4B3VuaUUxODkHdW5pRTE5MAd1bmlFMTkxB3VuaUUxOTIHdW5pRTE5Mwd1bmlFMTk0B3VuaUUxOTUHdW5pRTE5Nwd1bmlFMTk4B3VuaUUxOTkHdW5pRTIwMAd1bmlFMjAxB3VuaUUyMDIHdW5pRTIwMwd1bmlFMjA0B3VuaUUyMDUHdW5pRTIwNgd1bmlFMjA5B3VuaUUyMTAHdW5pRTIxMQd1bmlFMjEyB3VuaUUyMTMHdW5pRTIxNAd1bmlFMjE1B3VuaUUyMTYHdW5pRTIxOAd1bmlFMjE5B3VuaUUyMjEHdW5pRTIyMwd1bmlFMjI0B3VuaUUyMjUHdW5pRTIyNgd1bmlFMjI3B3VuaUUyMzAHdW5pRTIzMQd1bmlFMjMyB3VuaUUyMzMHdW5pRTIzNAd1bmlFMjM1B3VuaUUyMzYHdW5pRTIzNwd1bmlFMjM4B3VuaUUyMzkHdW5pRTI0MAd1bmlFMjQxB3VuaUUyNDIHdW5pRTI0Mwd1bmlFMjQ0B3VuaUUyNDUHdW5pRTI0Ngd1bmlFMjQ3B3VuaUUyNDgHdW5pRTI0OQd1bmlFMjUwB3VuaUUyNTEHdW5pRTI1Mgd1bmlFMjUzB3VuaUUyNTQHdW5pRTI1NQd1bmlFMjU2B3VuaUUyNTcHdW5pRTI1OAd1bmlFMjU5B3VuaUUyNjAHdW5pRjhGRgZ1MUY1MTEGdTFGNkFBAAAAAAFUUMMXAAA=) format('truetype'),url(data:image/svg+xml;base64,<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" >
<svg xmlns="http://www.w3.org/2000/svg">
<metadata></metadata>
<defs>
<font id="glyphicons_halflingsregular" horiz-adv-x="1200" >
<font-face units-per-em="1200" ascent="960" descent="-240" />
<missing-glyph horiz-adv-x="500" />
<glyph horiz-adv-x="0" />
<glyph horiz-adv-x="400" />
<glyph unicode=" " />
<glyph unicode="*" d="M600 1100q15 0 34 -1.5t30 -3.5l11 -1q10 -2 17.5 -10.5t7.5 -18.5v-224l158 158q7 7 18 8t19 -6l106 -106q7 -8 6 -19t-8 -18l-158 -158h224q10 0 18.5 -7.5t10.5 -17.5q6 -41 6 -75q0 -15 -1.5 -34t-3.5 -30l-1 -11q-2 -10 -10.5 -17.5t-18.5 -7.5h-224l158 -158 q7 -7 8 -18t-6 -19l-106 -106q-8 -7 -19 -6t-18 8l-158 158v-224q0 -10 -7.5 -18.5t-17.5 -10.5q-41 -6 -75 -6q-15 0 -34 1.5t-30 3.5l-11 1q-10 2 -17.5 10.5t-7.5 18.5v224l-158 -158q-7 -7 -18 -8t-19 6l-106 106q-7 8 -6 19t8 18l158 158h-224q-10 0 -18.5 7.5 t-10.5 17.5q-6 41 -6 75q0 15 1.5 34t3.5 30l1 11q2 10 10.5 17.5t18.5 7.5h224l-158 158q-7 7 -8 18t6 19l106 106q8 7 19 6t18 -8l158 -158v224q0 10 7.5 18.5t17.5 10.5q41 6 75 6z" />
<glyph unicode="+" d="M450 1100h200q21 0 35.5 -14.5t14.5 -35.5v-350h350q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-350v-350q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v350h-350q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5 h350v350q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xa0;" />
<glyph unicode="&#xa5;" d="M825 1100h250q10 0 12.5 -5t-5.5 -13l-364 -364q-6 -6 -11 -18h268q10 0 13 -6t-3 -14l-120 -160q-6 -8 -18 -14t-22 -6h-125v-100h275q10 0 13 -6t-3 -14l-120 -160q-6 -8 -18 -14t-22 -6h-125v-174q0 -11 -7.5 -18.5t-18.5 -7.5h-148q-11 0 -18.5 7.5t-7.5 18.5v174 h-275q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h125v100h-275q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h118q-5 12 -11 18l-364 364q-8 8 -5.5 13t12.5 5h250q25 0 43 -18l164 -164q8 -8 18 -8t18 8l164 164q18 18 43 18z" />
<glyph unicode="&#x2000;" horiz-adv-x="650" />
<glyph unicode="&#x2001;" horiz-adv-x="1300" />
<glyph unicode="&#x2002;" horiz-adv-x="650" />
<glyph unicode="&#x2003;" horiz-adv-x="1300" />
<glyph unicode="&#x2004;" horiz-adv-x="433" />
<glyph unicode="&#x2005;" horiz-adv-x="325" />
<glyph unicode="&#x2006;" horiz-adv-x="216" />
<glyph unicode="&#x2007;" horiz-adv-x="216" />
<glyph unicode="&#x2008;" horiz-adv-x="162" />
<glyph unicode="&#x2009;" horiz-adv-x="260" />
<glyph unicode="&#x200a;" horiz-adv-x="72" />
<glyph unicode="&#x202f;" horiz-adv-x="260" />
<glyph unicode="&#x205f;" horiz-adv-x="325" />
<glyph unicode="&#x20ac;" d="M744 1198q242 0 354 -189q60 -104 66 -209h-181q0 45 -17.5 82.5t-43.5 61.5t-58 40.5t-60.5 24t-51.5 7.5q-19 0 -40.5 -5.5t-49.5 -20.5t-53 -38t-49 -62.5t-39 -89.5h379l-100 -100h-300q-6 -50 -6 -100h406l-100 -100h-300q9 -74 33 -132t52.5 -91t61.5 -54.5t59 -29 t47 -7.5q22 0 50.5 7.5t60.5 24.5t58 41t43.5 61t17.5 80h174q-30 -171 -128 -278q-107 -117 -274 -117q-206 0 -324 158q-36 48 -69 133t-45 204h-217l100 100h112q1 47 6 100h-218l100 100h134q20 87 51 153.5t62 103.5q117 141 297 141z" />
<glyph unicode="&#x20bd;" d="M428 1200h350q67 0 120 -13t86 -31t57 -49.5t35 -56.5t17 -64.5t6.5 -60.5t0.5 -57v-16.5v-16.5q0 -36 -0.5 -57t-6.5 -61t-17 -65t-35 -57t-57 -50.5t-86 -31.5t-120 -13h-178l-2 -100h288q10 0 13 -6t-3 -14l-120 -160q-6 -8 -18 -14t-22 -6h-138v-175q0 -11 -5.5 -18 t-15.5 -7h-149q-10 0 -17.5 7.5t-7.5 17.5v175h-267q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h117v100h-267q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h117v475q0 10 7.5 17.5t17.5 7.5zM600 1000v-300h203q64 0 86.5 33t22.5 119q0 84 -22.5 116t-86.5 32h-203z" />
<glyph unicode="&#x2212;" d="M250 700h800q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#x231b;" d="M1000 1200v-150q0 -21 -14.5 -35.5t-35.5 -14.5h-50v-100q0 -91 -49.5 -165.5t-130.5 -109.5q81 -35 130.5 -109.5t49.5 -165.5v-150h50q21 0 35.5 -14.5t14.5 -35.5v-150h-800v150q0 21 14.5 35.5t35.5 14.5h50v150q0 91 49.5 165.5t130.5 109.5q-81 35 -130.5 109.5 t-49.5 165.5v100h-50q-21 0 -35.5 14.5t-14.5 35.5v150h800zM400 1000v-100q0 -60 32.5 -109.5t87.5 -73.5q28 -12 44 -37t16 -55t-16 -55t-44 -37q-55 -24 -87.5 -73.5t-32.5 -109.5v-150h400v150q0 60 -32.5 109.5t-87.5 73.5q-28 12 -44 37t-16 55t16 55t44 37 q55 24 87.5 73.5t32.5 109.5v100h-400z" />
<glyph unicode="&#x25fc;" horiz-adv-x="500" d="M0 0z" />
<glyph unicode="&#x2601;" d="M503 1089q110 0 200.5 -59.5t134.5 -156.5q44 14 90 14q120 0 205 -86.5t85 -206.5q0 -121 -85 -207.5t-205 -86.5h-750q-79 0 -135.5 57t-56.5 137q0 69 42.5 122.5t108.5 67.5q-2 12 -2 37q0 153 108 260.5t260 107.5z" />
<glyph unicode="&#x26fa;" d="M774 1193.5q16 -9.5 20.5 -27t-5.5 -33.5l-136 -187l467 -746h30q20 0 35 -18.5t15 -39.5v-42h-1200v42q0 21 15 39.5t35 18.5h30l468 746l-135 183q-10 16 -5.5 34t20.5 28t34 5.5t28 -20.5l111 -148l112 150q9 16 27 20.5t34 -5zM600 200h377l-182 112l-195 534v-646z " />
<glyph unicode="&#x2709;" d="M25 1100h1150q10 0 12.5 -5t-5.5 -13l-564 -567q-8 -8 -18 -8t-18 8l-564 567q-8 8 -5.5 13t12.5 5zM18 882l264 -264q8 -8 8 -18t-8 -18l-264 -264q-8 -8 -13 -5.5t-5 12.5v550q0 10 5 12.5t13 -5.5zM918 618l264 264q8 8 13 5.5t5 -12.5v-550q0 -10 -5 -12.5t-13 5.5 l-264 264q-8 8 -8 18t8 18zM818 482l364 -364q8 -8 5.5 -13t-12.5 -5h-1150q-10 0 -12.5 5t5.5 13l364 364q8 8 18 8t18 -8l164 -164q8 -8 18 -8t18 8l164 164q8 8 18 8t18 -8z" />
<glyph unicode="&#x270f;" d="M1011 1210q19 0 33 -13l153 -153q13 -14 13 -33t-13 -33l-99 -92l-214 214l95 96q13 14 32 14zM1013 800l-615 -614l-214 214l614 614zM317 96l-333 -112l110 335z" />
<glyph unicode="&#xe001;" d="M700 650v-550h250q21 0 35.5 -14.5t14.5 -35.5v-50h-800v50q0 21 14.5 35.5t35.5 14.5h250v550l-500 550h1200z" />
<glyph unicode="&#xe002;" d="M368 1017l645 163q39 15 63 0t24 -49v-831q0 -55 -41.5 -95.5t-111.5 -63.5q-79 -25 -147 -4.5t-86 75t25.5 111.5t122.5 82q72 24 138 8v521l-600 -155v-606q0 -42 -44 -90t-109 -69q-79 -26 -147 -5.5t-86 75.5t25.5 111.5t122.5 82.5q72 24 138 7v639q0 38 14.5 59 t53.5 34z" />
<glyph unicode="&#xe003;" d="M500 1191q100 0 191 -39t156.5 -104.5t104.5 -156.5t39 -191l-1 -2l1 -5q0 -141 -78 -262l275 -274q23 -26 22.5 -44.5t-22.5 -42.5l-59 -58q-26 -20 -46.5 -20t-39.5 20l-275 274q-119 -77 -261 -77l-5 1l-2 -1q-100 0 -191 39t-156.5 104.5t-104.5 156.5t-39 191 t39 191t104.5 156.5t156.5 104.5t191 39zM500 1022q-88 0 -162 -43t-117 -117t-43 -162t43 -162t117 -117t162 -43t162 43t117 117t43 162t-43 162t-117 117t-162 43z" />
<glyph unicode="&#xe005;" d="M649 949q48 68 109.5 104t121.5 38.5t118.5 -20t102.5 -64t71 -100.5t27 -123q0 -57 -33.5 -117.5t-94 -124.5t-126.5 -127.5t-150 -152.5t-146 -174q-62 85 -145.5 174t-150 152.5t-126.5 127.5t-93.5 124.5t-33.5 117.5q0 64 28 123t73 100.5t104 64t119 20 t120.5 -38.5t104.5 -104z" />
<glyph unicode="&#xe006;" d="M407 800l131 353q7 19 17.5 19t17.5 -19l129 -353h421q21 0 24 -8.5t-14 -20.5l-342 -249l130 -401q7 -20 -0.5 -25.5t-24.5 6.5l-343 246l-342 -247q-17 -12 -24.5 -6.5t-0.5 25.5l130 400l-347 251q-17 12 -14 20.5t23 8.5h429z" />
<glyph unicode="&#xe007;" d="M407 800l131 353q7 19 17.5 19t17.5 -19l129 -353h421q21 0 24 -8.5t-14 -20.5l-342 -249l130 -401q7 -20 -0.5 -25.5t-24.5 6.5l-343 246l-342 -247q-17 -12 -24.5 -6.5t-0.5 25.5l130 400l-347 251q-17 12 -14 20.5t23 8.5h429zM477 700h-240l197 -142l-74 -226 l193 139l195 -140l-74 229l192 140h-234l-78 211z" />
<glyph unicode="&#xe008;" d="M600 1200q124 0 212 -88t88 -212v-250q0 -46 -31 -98t-69 -52v-75q0 -10 6 -21.5t15 -17.5l358 -230q9 -5 15 -16.5t6 -21.5v-93q0 -10 -7.5 -17.5t-17.5 -7.5h-1150q-10 0 -17.5 7.5t-7.5 17.5v93q0 10 6 21.5t15 16.5l358 230q9 6 15 17.5t6 21.5v75q-38 0 -69 52 t-31 98v250q0 124 88 212t212 88z" />
<glyph unicode="&#xe009;" d="M25 1100h1150q10 0 17.5 -7.5t7.5 -17.5v-1050q0 -10 -7.5 -17.5t-17.5 -7.5h-1150q-10 0 -17.5 7.5t-7.5 17.5v1050q0 10 7.5 17.5t17.5 7.5zM100 1000v-100h100v100h-100zM875 1000h-550q-10 0 -17.5 -7.5t-7.5 -17.5v-350q0 -10 7.5 -17.5t17.5 -7.5h550 q10 0 17.5 7.5t7.5 17.5v350q0 10 -7.5 17.5t-17.5 7.5zM1000 1000v-100h100v100h-100zM100 800v-100h100v100h-100zM1000 800v-100h100v100h-100zM100 600v-100h100v100h-100zM1000 600v-100h100v100h-100zM875 500h-550q-10 0 -17.5 -7.5t-7.5 -17.5v-350q0 -10 7.5 -17.5 t17.5 -7.5h550q10 0 17.5 7.5t7.5 17.5v350q0 10 -7.5 17.5t-17.5 7.5zM100 400v-100h100v100h-100zM1000 400v-100h100v100h-100zM100 200v-100h100v100h-100zM1000 200v-100h100v100h-100z" />
<glyph unicode="&#xe010;" d="M50 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM650 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400 q0 21 14.5 35.5t35.5 14.5zM50 500h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM650 500h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400 q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe011;" d="M50 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200 q0 21 14.5 35.5t35.5 14.5zM850 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM50 700h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200 q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 700h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM850 700h200q21 0 35.5 -14.5t14.5 -35.5v-200 q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM50 300h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 300h200 q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM850 300h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5 t35.5 14.5z" />
<glyph unicode="&#xe012;" d="M50 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 1100h700q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v200 q0 21 14.5 35.5t35.5 14.5zM50 700h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 700h700q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-700 q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM50 300h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 300h700q21 0 35.5 -14.5t14.5 -35.5v-200 q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe013;" d="M465 477l571 571q8 8 18 8t17 -8l177 -177q8 -7 8 -17t-8 -18l-783 -784q-7 -8 -17.5 -8t-17.5 8l-384 384q-8 8 -8 18t8 17l177 177q7 8 17 8t18 -8l171 -171q7 -7 18 -7t18 7z" />
<glyph unicode="&#xe014;" d="M904 1083l178 -179q8 -8 8 -18.5t-8 -17.5l-267 -268l267 -268q8 -7 8 -17.5t-8 -18.5l-178 -178q-8 -8 -18.5 -8t-17.5 8l-268 267l-268 -267q-7 -8 -17.5 -8t-18.5 8l-178 178q-8 8 -8 18.5t8 17.5l267 268l-267 268q-8 7 -8 17.5t8 18.5l178 178q8 8 18.5 8t17.5 -8 l268 -267l268 268q7 7 17.5 7t18.5 -7z" />
<glyph unicode="&#xe015;" d="M507 1177q98 0 187.5 -38.5t154.5 -103.5t103.5 -154.5t38.5 -187.5q0 -141 -78 -262l300 -299q8 -8 8 -18.5t-8 -18.5l-109 -108q-7 -8 -17.5 -8t-18.5 8l-300 299q-119 -77 -261 -77q-98 0 -188 38.5t-154.5 103t-103 154.5t-38.5 188t38.5 187.5t103 154.5 t154.5 103.5t188 38.5zM506.5 1023q-89.5 0 -165.5 -44t-120 -120.5t-44 -166t44 -165.5t120 -120t165.5 -44t166 44t120.5 120t44 165.5t-44 166t-120.5 120.5t-166 44zM425 900h150q10 0 17.5 -7.5t7.5 -17.5v-75h75q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5 t-17.5 -7.5h-75v-75q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v75h-75q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h75v75q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe016;" d="M507 1177q98 0 187.5 -38.5t154.5 -103.5t103.5 -154.5t38.5 -187.5q0 -141 -78 -262l300 -299q8 -8 8 -18.5t-8 -18.5l-109 -108q-7 -8 -17.5 -8t-18.5 8l-300 299q-119 -77 -261 -77q-98 0 -188 38.5t-154.5 103t-103 154.5t-38.5 188t38.5 187.5t103 154.5 t154.5 103.5t188 38.5zM506.5 1023q-89.5 0 -165.5 -44t-120 -120.5t-44 -166t44 -165.5t120 -120t165.5 -44t166 44t120.5 120t44 165.5t-44 166t-120.5 120.5t-166 44zM325 800h350q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-350q-10 0 -17.5 7.5 t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe017;" d="M550 1200h100q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM800 975v166q167 -62 272 -209.5t105 -331.5q0 -117 -45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5 t-184.5 123t-123 184.5t-45.5 224q0 184 105 331.5t272 209.5v-166q-103 -55 -165 -155t-62 -220q0 -116 57 -214.5t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5q0 120 -62 220t-165 155z" />
<glyph unicode="&#xe018;" d="M1025 1200h150q10 0 17.5 -7.5t7.5 -17.5v-1150q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v1150q0 10 7.5 17.5t17.5 7.5zM725 800h150q10 0 17.5 -7.5t7.5 -17.5v-750q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v750 q0 10 7.5 17.5t17.5 7.5zM425 500h150q10 0 17.5 -7.5t7.5 -17.5v-450q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v450q0 10 7.5 17.5t17.5 7.5zM125 300h150q10 0 17.5 -7.5t7.5 -17.5v-250q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5 v250q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe019;" d="M600 1174q33 0 74 -5l38 -152l5 -1q49 -14 94 -39l5 -2l134 80q61 -48 104 -105l-80 -134l3 -5q25 -44 39 -93l1 -6l152 -38q5 -43 5 -73q0 -34 -5 -74l-152 -38l-1 -6q-15 -49 -39 -93l-3 -5l80 -134q-48 -61 -104 -105l-134 81l-5 -3q-44 -25 -94 -39l-5 -2l-38 -151 q-43 -5 -74 -5q-33 0 -74 5l-38 151l-5 2q-49 14 -94 39l-5 3l-134 -81q-60 48 -104 105l80 134l-3 5q-25 45 -38 93l-2 6l-151 38q-6 42 -6 74q0 33 6 73l151 38l2 6q13 48 38 93l3 5l-80 134q47 61 105 105l133 -80l5 2q45 25 94 39l5 1l38 152q43 5 74 5zM600 815 q-89 0 -152 -63t-63 -151.5t63 -151.5t152 -63t152 63t63 151.5t-63 151.5t-152 63z" />
<glyph unicode="&#xe020;" d="M500 1300h300q41 0 70.5 -29.5t29.5 -70.5v-100h275q10 0 17.5 -7.5t7.5 -17.5v-75h-1100v75q0 10 7.5 17.5t17.5 7.5h275v100q0 41 29.5 70.5t70.5 29.5zM500 1200v-100h300v100h-300zM1100 900v-800q0 -41 -29.5 -70.5t-70.5 -29.5h-700q-41 0 -70.5 29.5t-29.5 70.5 v800h900zM300 800v-700h100v700h-100zM500 800v-700h100v700h-100zM700 800v-700h100v700h-100zM900 800v-700h100v700h-100z" />
<glyph unicode="&#xe021;" d="M18 618l620 608q8 7 18.5 7t17.5 -7l608 -608q8 -8 5.5 -13t-12.5 -5h-175v-575q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v375h-300v-375q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v575h-175q-10 0 -12.5 5t5.5 13z" />
<glyph unicode="&#xe022;" d="M600 1200v-400q0 -41 29.5 -70.5t70.5 -29.5h300v-650q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v1100q0 21 14.5 35.5t35.5 14.5h450zM1000 800h-250q-21 0 -35.5 14.5t-14.5 35.5v250z" />
<glyph unicode="&#xe023;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM525 900h50q10 0 17.5 -7.5t7.5 -17.5v-275h175q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe024;" d="M1300 0h-538l-41 400h-242l-41 -400h-538l431 1200h209l-21 -300h162l-20 300h208zM515 800l-27 -300h224l-27 300h-170z" />
<glyph unicode="&#xe025;" d="M550 1200h200q21 0 35.5 -14.5t14.5 -35.5v-450h191q20 0 25.5 -11.5t-7.5 -27.5l-327 -400q-13 -16 -32 -16t-32 16l-327 400q-13 16 -7.5 27.5t25.5 11.5h191v450q0 21 14.5 35.5t35.5 14.5zM1125 400h50q10 0 17.5 -7.5t7.5 -17.5v-350q0 -10 -7.5 -17.5t-17.5 -7.5 h-1050q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h50q10 0 17.5 -7.5t7.5 -17.5v-175h900v175q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe026;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM525 900h150q10 0 17.5 -7.5t7.5 -17.5v-275h137q21 0 26 -11.5t-8 -27.5l-223 -275q-13 -16 -32 -16t-32 16l-223 275q-13 16 -8 27.5t26 11.5h137v275q0 10 7.5 17.5t17.5 7.5z " />
<glyph unicode="&#xe027;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM632 914l223 -275q13 -16 8 -27.5t-26 -11.5h-137v-275q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v275h-137q-21 0 -26 11.5t8 27.5l223 275q13 16 32 16 t32 -16z" />
<glyph unicode="&#xe028;" d="M225 1200h750q10 0 19.5 -7t12.5 -17l186 -652q7 -24 7 -49v-425q0 -12 -4 -27t-9 -17q-12 -6 -37 -6h-1100q-12 0 -27 4t-17 8q-6 13 -6 38l1 425q0 25 7 49l185 652q3 10 12.5 17t19.5 7zM878 1000h-556q-10 0 -19 -7t-11 -18l-87 -450q-2 -11 4 -18t16 -7h150 q10 0 19.5 -7t11.5 -17l38 -152q2 -10 11.5 -17t19.5 -7h250q10 0 19.5 7t11.5 17l38 152q2 10 11.5 17t19.5 7h150q10 0 16 7t4 18l-87 450q-2 11 -11 18t-19 7z" />
<glyph unicode="&#xe029;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM540 820l253 -190q17 -12 17 -30t-17 -30l-253 -190q-16 -12 -28 -6.5t-12 26.5v400q0 21 12 26.5t28 -6.5z" />
<glyph unicode="&#xe030;" d="M947 1060l135 135q7 7 12.5 5t5.5 -13v-362q0 -10 -7.5 -17.5t-17.5 -7.5h-362q-11 0 -13 5.5t5 12.5l133 133q-109 76 -238 76q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5h150q0 -117 -45.5 -224 t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5q192 0 347 -117z" />
<glyph unicode="&#xe031;" d="M947 1060l135 135q7 7 12.5 5t5.5 -13v-361q0 -11 -7.5 -18.5t-18.5 -7.5h-361q-11 0 -13 5.5t5 12.5l134 134q-110 75 -239 75q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5h-150q0 117 45.5 224t123 184.5t184.5 123t224 45.5q192 0 347 -117zM1027 600h150 q0 -117 -45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5q-192 0 -348 118l-134 -134q-7 -8 -12.5 -5.5t-5.5 12.5v360q0 11 7.5 18.5t18.5 7.5h360q10 0 12.5 -5.5t-5.5 -12.5l-133 -133q110 -76 240 -76q116 0 214.5 57t155.5 155.5t57 214.5z" />
<glyph unicode="&#xe032;" d="M125 1200h1050q10 0 17.5 -7.5t7.5 -17.5v-1150q0 -10 -7.5 -17.5t-17.5 -7.5h-1050q-10 0 -17.5 7.5t-7.5 17.5v1150q0 10 7.5 17.5t17.5 7.5zM1075 1000h-850q-10 0 -17.5 -7.5t-7.5 -17.5v-850q0 -10 7.5 -17.5t17.5 -7.5h850q10 0 17.5 7.5t7.5 17.5v850 q0 10 -7.5 17.5t-17.5 7.5zM325 900h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 900h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5zM325 700h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 700h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5zM325 500h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 500h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5zM325 300h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 300h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe033;" d="M900 800v200q0 83 -58.5 141.5t-141.5 58.5h-300q-82 0 -141 -59t-59 -141v-200h-100q-41 0 -70.5 -29.5t-29.5 -70.5v-600q0 -41 29.5 -70.5t70.5 -29.5h900q41 0 70.5 29.5t29.5 70.5v600q0 41 -29.5 70.5t-70.5 29.5h-100zM400 800v150q0 21 15 35.5t35 14.5h200 q20 0 35 -14.5t15 -35.5v-150h-300z" />
<glyph unicode="&#xe034;" d="M125 1100h50q10 0 17.5 -7.5t7.5 -17.5v-1075h-100v1075q0 10 7.5 17.5t17.5 7.5zM1075 1052q4 0 9 -2q16 -6 16 -23v-421q0 -6 -3 -12q-33 -59 -66.5 -99t-65.5 -58t-56.5 -24.5t-52.5 -6.5q-26 0 -57.5 6.5t-52.5 13.5t-60 21q-41 15 -63 22.5t-57.5 15t-65.5 7.5 q-85 0 -160 -57q-7 -5 -15 -5q-6 0 -11 3q-14 7 -14 22v438q22 55 82 98.5t119 46.5q23 2 43 0.5t43 -7t32.5 -8.5t38 -13t32.5 -11q41 -14 63.5 -21t57 -14t63.5 -7q103 0 183 87q7 8 18 8z" />
<glyph unicode="&#xe035;" d="M600 1175q116 0 227 -49.5t192.5 -131t131 -192.5t49.5 -227v-300q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v300q0 127 -70.5 231.5t-184.5 161.5t-245 57t-245 -57t-184.5 -161.5t-70.5 -231.5v-300q0 -10 -7.5 -17.5t-17.5 -7.5h-50 q-10 0 -17.5 7.5t-7.5 17.5v300q0 116 49.5 227t131 192.5t192.5 131t227 49.5zM220 500h160q8 0 14 -6t6 -14v-460q0 -8 -6 -14t-14 -6h-160q-8 0 -14 6t-6 14v460q0 8 6 14t14 6zM820 500h160q8 0 14 -6t6 -14v-460q0 -8 -6 -14t-14 -6h-160q-8 0 -14 6t-6 14v460 q0 8 6 14t14 6z" />
<glyph unicode="&#xe036;" d="M321 814l258 172q9 6 15 2.5t6 -13.5v-750q0 -10 -6 -13.5t-15 2.5l-258 172q-21 14 -46 14h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h250q25 0 46 14zM900 668l120 120q7 7 17 7t17 -7l34 -34q7 -7 7 -17t-7 -17l-120 -120l120 -120q7 -7 7 -17 t-7 -17l-34 -34q-7 -7 -17 -7t-17 7l-120 119l-120 -119q-7 -7 -17 -7t-17 7l-34 34q-7 7 -7 17t7 17l119 120l-119 120q-7 7 -7 17t7 17l34 34q7 8 17 8t17 -8z" />
<glyph unicode="&#xe037;" d="M321 814l258 172q9 6 15 2.5t6 -13.5v-750q0 -10 -6 -13.5t-15 2.5l-258 172q-21 14 -46 14h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h250q25 0 46 14zM766 900h4q10 -1 16 -10q96 -129 96 -290q0 -154 -90 -281q-6 -9 -17 -10l-3 -1q-9 0 -16 6 l-29 23q-7 7 -8.5 16.5t4.5 17.5q72 103 72 229q0 132 -78 238q-6 8 -4.5 18t9.5 17l29 22q7 5 15 5z" />
<glyph unicode="&#xe038;" d="M967 1004h3q11 -1 17 -10q135 -179 135 -396q0 -105 -34 -206.5t-98 -185.5q-7 -9 -17 -10h-3q-9 0 -16 6l-42 34q-8 6 -9 16t5 18q111 150 111 328q0 90 -29.5 176t-84.5 157q-6 9 -5 19t10 16l42 33q7 5 15 5zM321 814l258 172q9 6 15 2.5t6 -13.5v-750q0 -10 -6 -13.5 t-15 2.5l-258 172q-21 14 -46 14h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h250q25 0 46 14zM766 900h4q10 -1 16 -10q96 -129 96 -290q0 -154 -90 -281q-6 -9 -17 -10l-3 -1q-9 0 -16 6l-29 23q-7 7 -8.5 16.5t4.5 17.5q72 103 72 229q0 132 -78 238 q-6 8 -4.5 18.5t9.5 16.5l29 22q7 5 15 5z" />
<glyph unicode="&#xe039;" d="M500 900h100v-100h-100v-100h-400v-100h-100v600h500v-300zM1200 700h-200v-100h200v-200h-300v300h-200v300h-100v200h600v-500zM100 1100v-300h300v300h-300zM800 1100v-300h300v300h-300zM300 900h-100v100h100v-100zM1000 900h-100v100h100v-100zM300 500h200v-500 h-500v500h200v100h100v-100zM800 300h200v-100h-100v-100h-200v100h-100v100h100v200h-200v100h300v-300zM100 400v-300h300v300h-300zM300 200h-100v100h100v-100zM1200 200h-100v100h100v-100zM700 0h-100v100h100v-100zM1200 0h-300v100h300v-100z" />
<glyph unicode="&#xe040;" d="M100 200h-100v1000h100v-1000zM300 200h-100v1000h100v-1000zM700 200h-200v1000h200v-1000zM900 200h-100v1000h100v-1000zM1200 200h-200v1000h200v-1000zM400 0h-300v100h300v-100zM600 0h-100v91h100v-91zM800 0h-100v91h100v-91zM1100 0h-200v91h200v-91z" />
<glyph unicode="&#xe041;" d="M500 1200l682 -682q8 -8 8 -18t-8 -18l-464 -464q-8 -8 -18 -8t-18 8l-682 682l1 475q0 10 7.5 17.5t17.5 7.5h474zM319.5 1024.5q-29.5 29.5 -71 29.5t-71 -29.5t-29.5 -71.5t29.5 -71.5t71 -29.5t71 29.5t29.5 71.5t-29.5 71.5z" />
<glyph unicode="&#xe042;" d="M500 1200l682 -682q8 -8 8 -18t-8 -18l-464 -464q-8 -8 -18 -8t-18 8l-682 682l1 475q0 10 7.5 17.5t17.5 7.5h474zM800 1200l682 -682q8 -8 8 -18t-8 -18l-464 -464q-8 -8 -18 -8t-18 8l-56 56l424 426l-700 700h150zM319.5 1024.5q-29.5 29.5 -71 29.5t-71 -29.5 t-29.5 -71.5t29.5 -71.5t71 -29.5t71 29.5t29.5 71.5t-29.5 71.5z" />
<glyph unicode="&#xe043;" d="M300 1200h825q75 0 75 -75v-900q0 -25 -18 -43l-64 -64q-8 -8 -13 -5.5t-5 12.5v950q0 10 -7.5 17.5t-17.5 7.5h-700q-25 0 -43 -18l-64 -64q-8 -8 -5.5 -13t12.5 -5h700q10 0 17.5 -7.5t7.5 -17.5v-950q0 -10 -7.5 -17.5t-17.5 -7.5h-850q-10 0 -17.5 7.5t-7.5 17.5v975 q0 25 18 43l139 139q18 18 43 18z" />
<glyph unicode="&#xe044;" d="M250 1200h800q21 0 35.5 -14.5t14.5 -35.5v-1150l-450 444l-450 -445v1151q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe045;" d="M822 1200h-444q-11 0 -19 -7.5t-9 -17.5l-78 -301q-7 -24 7 -45l57 -108q6 -9 17.5 -15t21.5 -6h450q10 0 21.5 6t17.5 15l62 108q14 21 7 45l-83 301q-1 10 -9 17.5t-19 7.5zM1175 800h-150q-10 0 -21 -6.5t-15 -15.5l-78 -156q-4 -9 -15 -15.5t-21 -6.5h-550 q-10 0 -21 6.5t-15 15.5l-78 156q-4 9 -15 15.5t-21 6.5h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-650q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h750q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5 t7.5 17.5v650q0 10 -7.5 17.5t-17.5 7.5zM850 200h-500q-10 0 -19.5 -7t-11.5 -17l-38 -152q-2 -10 3.5 -17t15.5 -7h600q10 0 15.5 7t3.5 17l-38 152q-2 10 -11.5 17t-19.5 7z" />
<glyph unicode="&#xe046;" d="M500 1100h200q56 0 102.5 -20.5t72.5 -50t44 -59t25 -50.5l6 -20h150q41 0 70.5 -29.5t29.5 -70.5v-600q0 -41 -29.5 -70.5t-70.5 -29.5h-1000q-41 0 -70.5 29.5t-29.5 70.5v600q0 41 29.5 70.5t70.5 29.5h150q2 8 6.5 21.5t24 48t45 61t72 48t102.5 21.5zM900 800v-100 h100v100h-100zM600 730q-95 0 -162.5 -67.5t-67.5 -162.5t67.5 -162.5t162.5 -67.5t162.5 67.5t67.5 162.5t-67.5 162.5t-162.5 67.5zM600 603q43 0 73 -30t30 -73t-30 -73t-73 -30t-73 30t-30 73t30 73t73 30z" />
<glyph unicode="&#xe047;" d="M681 1199l385 -998q20 -50 60 -92q18 -19 36.5 -29.5t27.5 -11.5l10 -2v-66h-417v66q53 0 75 43.5t5 88.5l-82 222h-391q-58 -145 -92 -234q-11 -34 -6.5 -57t25.5 -37t46 -20t55 -6v-66h-365v66q56 24 84 52q12 12 25 30.5t20 31.5l7 13l399 1006h93zM416 521h340 l-162 457z" />
<glyph unicode="&#xe048;" d="M753 641q5 -1 14.5 -4.5t36 -15.5t50.5 -26.5t53.5 -40t50.5 -54.5t35.5 -70t14.5 -87q0 -67 -27.5 -125.5t-71.5 -97.5t-98.5 -66.5t-108.5 -40.5t-102 -13h-500v89q41 7 70.5 32.5t29.5 65.5v827q0 24 -0.5 34t-3.5 24t-8.5 19.5t-17 13.5t-28 12.5t-42.5 11.5v71 l471 -1q57 0 115.5 -20.5t108 -57t80.5 -94t31 -124.5q0 -51 -15.5 -96.5t-38 -74.5t-45 -50.5t-38.5 -30.5zM400 700h139q78 0 130.5 48.5t52.5 122.5q0 41 -8.5 70.5t-29.5 55.5t-62.5 39.5t-103.5 13.5h-118v-350zM400 200h216q80 0 121 50.5t41 130.5q0 90 -62.5 154.5 t-156.5 64.5h-159v-400z" />
<glyph unicode="&#xe049;" d="M877 1200l2 -57q-83 -19 -116 -45.5t-40 -66.5l-132 -839q-9 -49 13 -69t96 -26v-97h-500v97q186 16 200 98l173 832q3 17 3 30t-1.5 22.5t-9 17.5t-13.5 12.5t-21.5 10t-26 8.5t-33.5 10q-13 3 -19 5v57h425z" />
<glyph unicode="&#xe050;" d="M1300 900h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-200v-850q0 -22 25 -34.5t50 -13.5l25 -2v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v850h-200q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h1000v-300zM175 1000h-75v-800h75l-125 -167l-125 167h75v800h-75l125 167z" />
<glyph unicode="&#xe051;" d="M1100 900h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-200v-650q0 -22 25 -34.5t50 -13.5l25 -2v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v650h-200q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h1000v-300zM1167 50l-167 -125v75h-800v-75l-167 125l167 125v-75h800v75z" />
<glyph unicode="&#xe052;" d="M50 1100h600q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 800h1000q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM50 500h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe053;" d="M250 1100h700q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 800h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM250 500h700q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe054;" d="M500 950v100q0 21 14.5 35.5t35.5 14.5h600q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5zM100 650v100q0 21 14.5 35.5t35.5 14.5h1000q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000 q-21 0 -35.5 14.5t-14.5 35.5zM300 350v100q0 21 14.5 35.5t35.5 14.5h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5zM0 50v100q0 21 14.5 35.5t35.5 14.5h1100q21 0 35.5 -14.5t14.5 -35.5v-100 q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5z" />
<glyph unicode="&#xe055;" d="M50 1100h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 800h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM50 500h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe056;" d="M50 1100h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 1100h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM50 800h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 800h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 500h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 500h800q21 0 35.5 -14.5t14.5 -35.5v-100 q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 200h800 q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe057;" d="M400 0h-100v1100h100v-1100zM550 1100h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM550 800h500q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-500 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM267 550l-167 -125v75h-200v100h200v75zM550 500h300q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM550 200h600 q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe058;" d="M50 1100h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM900 0h-100v1100h100v-1100zM50 800h500q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-500 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM1100 600h200v-100h-200v-75l-167 125l167 125v-75zM50 500h300q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h600 q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe059;" d="M75 1000h750q31 0 53 -22t22 -53v-650q0 -31 -22 -53t-53 -22h-750q-31 0 -53 22t-22 53v650q0 31 22 53t53 22zM1200 300l-300 300l300 300v-600z" />
<glyph unicode="&#xe060;" d="M44 1100h1112q18 0 31 -13t13 -31v-1012q0 -18 -13 -31t-31 -13h-1112q-18 0 -31 13t-13 31v1012q0 18 13 31t31 13zM100 1000v-737l247 182l298 -131l-74 156l293 318l236 -288v500h-1000zM342 884q56 0 95 -39t39 -94.5t-39 -95t-95 -39.5t-95 39.5t-39 95t39 94.5 t95 39z" />
<glyph unicode="&#xe062;" d="M648 1169q117 0 216 -60t156.5 -161t57.5 -218q0 -115 -70 -258q-69 -109 -158 -225.5t-143 -179.5l-54 -62q-9 8 -25.5 24.5t-63.5 67.5t-91 103t-98.5 128t-95.5 148q-60 132 -60 249q0 88 34 169.5t91.5 142t137 96.5t166.5 36zM652.5 974q-91.5 0 -156.5 -65 t-65 -157t65 -156.5t156.5 -64.5t156.5 64.5t65 156.5t-65 157t-156.5 65z" />
<glyph unicode="&#xe063;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 173v854q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57z" />
<glyph unicode="&#xe064;" d="M554 1295q21 -72 57.5 -143.5t76 -130t83 -118t82.5 -117t70 -116t49.5 -126t18.5 -136.5q0 -71 -25.5 -135t-68.5 -111t-99 -82t-118.5 -54t-125.5 -23q-84 5 -161.5 34t-139.5 78.5t-99 125t-37 164.5q0 69 18 136.5t49.5 126.5t69.5 116.5t81.5 117.5t83.5 119 t76.5 131t58.5 143zM344 710q-23 -33 -43.5 -70.5t-40.5 -102.5t-17 -123q1 -37 14.5 -69.5t30 -52t41 -37t38.5 -24.5t33 -15q21 -7 32 -1t13 22l6 34q2 10 -2.5 22t-13.5 19q-5 4 -14 12t-29.5 40.5t-32.5 73.5q-26 89 6 271q2 11 -6 11q-8 1 -15 -10z" />
<glyph unicode="&#xe065;" d="M1000 1013l108 115q2 1 5 2t13 2t20.5 -1t25 -9.5t28.5 -21.5q22 -22 27 -43t0 -32l-6 -10l-108 -115zM350 1100h400q50 0 105 -13l-187 -187h-368q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5v182l200 200v-332 q0 -165 -93.5 -257.5t-256.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5zM1009 803l-362 -362l-161 -50l55 170l355 355z" />
<glyph unicode="&#xe066;" d="M350 1100h361q-164 -146 -216 -200h-195q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5l200 153v-103q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5z M824 1073l339 -301q8 -7 8 -17.5t-8 -17.5l-340 -306q-7 -6 -12.5 -4t-6.5 11v203q-26 1 -54.5 0t-78.5 -7.5t-92 -17.5t-86 -35t-70 -57q10 59 33 108t51.5 81.5t65 58.5t68.5 40.5t67 24.5t56 13.5t40 4.5v210q1 10 6.5 12.5t13.5 -4.5z" />
<glyph unicode="&#xe067;" d="M350 1100h350q60 0 127 -23l-178 -177h-349q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5v69l200 200v-219q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5z M643 639l395 395q7 7 17.5 7t17.5 -7l101 -101q7 -7 7 -17.5t-7 -17.5l-531 -532q-7 -7 -17.5 -7t-17.5 7l-248 248q-7 7 -7 17.5t7 17.5l101 101q7 7 17.5 7t17.5 -7l111 -111q8 -7 18 -7t18 7z" />
<glyph unicode="&#xe068;" d="M318 918l264 264q8 8 18 8t18 -8l260 -264q7 -8 4.5 -13t-12.5 -5h-170v-200h200v173q0 10 5 12t13 -5l264 -260q8 -7 8 -17.5t-8 -17.5l-264 -265q-8 -7 -13 -5t-5 12v173h-200v-200h170q10 0 12.5 -5t-4.5 -13l-260 -264q-8 -8 -18 -8t-18 8l-264 264q-8 8 -5.5 13 t12.5 5h175v200h-200v-173q0 -10 -5 -12t-13 5l-264 265q-8 7 -8 17.5t8 17.5l264 260q8 7 13 5t5 -12v-173h200v200h-175q-10 0 -12.5 5t5.5 13z" />
<glyph unicode="&#xe069;" d="M250 1100h100q21 0 35.5 -14.5t14.5 -35.5v-438l464 453q15 14 25.5 10t10.5 -25v-1000q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v1000q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe070;" d="M50 1100h100q21 0 35.5 -14.5t14.5 -35.5v-438l464 453q15 14 25.5 10t10.5 -25v-438l464 453q15 14 25.5 10t10.5 -25v-1000q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5 t-14.5 35.5v1000q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe071;" d="M1200 1050v-1000q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -10.5 -25t-25.5 10l-492 480q-15 14 -15 35t15 35l492 480q15 14 25.5 10t10.5 -25v-438l464 453q15 14 25.5 10t10.5 -25z" />
<glyph unicode="&#xe072;" d="M243 1074l814 -498q18 -11 18 -26t-18 -26l-814 -498q-18 -11 -30.5 -4t-12.5 28v1000q0 21 12.5 28t30.5 -4z" />
<glyph unicode="&#xe073;" d="M250 1000h200q21 0 35.5 -14.5t14.5 -35.5v-800q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v800q0 21 14.5 35.5t35.5 14.5zM650 1000h200q21 0 35.5 -14.5t14.5 -35.5v-800q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v800 q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe074;" d="M1100 950v-800q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v800q0 21 14.5 35.5t35.5 14.5h800q21 0 35.5 -14.5t14.5 -35.5z" />
<glyph unicode="&#xe075;" d="M500 612v438q0 21 10.5 25t25.5 -10l492 -480q15 -14 15 -35t-15 -35l-492 -480q-15 -14 -25.5 -10t-10.5 25v438l-464 -453q-15 -14 -25.5 -10t-10.5 25v1000q0 21 10.5 25t25.5 -10z" />
<glyph unicode="&#xe076;" d="M1048 1102l100 1q20 0 35 -14.5t15 -35.5l5 -1000q0 -21 -14.5 -35.5t-35.5 -14.5l-100 -1q-21 0 -35.5 14.5t-14.5 35.5l-2 437l-463 -454q-14 -15 -24.5 -10.5t-10.5 25.5l-2 437l-462 -455q-15 -14 -25.5 -9.5t-10.5 24.5l-5 1000q0 21 10.5 25.5t25.5 -10.5l466 -450 l-2 438q0 20 10.5 24.5t25.5 -9.5l466 -451l-2 438q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe077;" d="M850 1100h100q21 0 35.5 -14.5t14.5 -35.5v-1000q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v438l-464 -453q-15 -14 -25.5 -10t-10.5 25v1000q0 21 10.5 25t25.5 -10l464 -453v438q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe078;" d="M686 1081l501 -540q15 -15 10.5 -26t-26.5 -11h-1042q-22 0 -26.5 11t10.5 26l501 540q15 15 36 15t36 -15zM150 400h1000q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe079;" d="M885 900l-352 -353l352 -353l-197 -198l-552 552l552 550z" />
<glyph unicode="&#xe080;" d="M1064 547l-551 -551l-198 198l353 353l-353 353l198 198z" />
<glyph unicode="&#xe081;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM650 900h-100q-21 0 -35.5 -14.5t-14.5 -35.5v-150h-150 q-21 0 -35.5 -14.5t-14.5 -35.5v-100q0 -21 14.5 -35.5t35.5 -14.5h150v-150q0 -21 14.5 -35.5t35.5 -14.5h100q21 0 35.5 14.5t14.5 35.5v150h150q21 0 35.5 14.5t14.5 35.5v100q0 21 -14.5 35.5t-35.5 14.5h-150v150q0 21 -14.5 35.5t-35.5 14.5z" />
<glyph unicode="&#xe082;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM850 700h-500q-21 0 -35.5 -14.5t-14.5 -35.5v-100q0 -21 14.5 -35.5 t35.5 -14.5h500q21 0 35.5 14.5t14.5 35.5v100q0 21 -14.5 35.5t-35.5 14.5z" />
<glyph unicode="&#xe083;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM741.5 913q-12.5 0 -21.5 -9l-120 -120l-120 120q-9 9 -21.5 9 t-21.5 -9l-141 -141q-9 -9 -9 -21.5t9 -21.5l120 -120l-120 -120q-9 -9 -9 -21.5t9 -21.5l141 -141q9 -9 21.5 -9t21.5 9l120 120l120 -120q9 -9 21.5 -9t21.5 9l141 141q9 9 9 21.5t-9 21.5l-120 120l120 120q9 9 9 21.5t-9 21.5l-141 141q-9 9 -21.5 9z" />
<glyph unicode="&#xe084;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM546 623l-84 85q-7 7 -17.5 7t-18.5 -7l-139 -139q-7 -8 -7 -18t7 -18 l242 -241q7 -8 17.5 -8t17.5 8l375 375q7 7 7 17.5t-7 18.5l-139 139q-7 7 -17.5 7t-17.5 -7z" />
<glyph unicode="&#xe085;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM588 941q-29 0 -59 -5.5t-63 -20.5t-58 -38.5t-41.5 -63t-16.5 -89.5 q0 -25 20 -25h131q30 -5 35 11q6 20 20.5 28t45.5 8q20 0 31.5 -10.5t11.5 -28.5q0 -23 -7 -34t-26 -18q-1 0 -13.5 -4t-19.5 -7.5t-20 -10.5t-22 -17t-18.5 -24t-15.5 -35t-8 -46q-1 -8 5.5 -16.5t20.5 -8.5h173q7 0 22 8t35 28t37.5 48t29.5 74t12 100q0 47 -17 83 t-42.5 57t-59.5 34.5t-64 18t-59 4.5zM675 400h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe086;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM675 1000h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5 t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5zM675 700h-250q-10 0 -17.5 -7.5t-7.5 -17.5v-50q0 -10 7.5 -17.5t17.5 -7.5h75v-200h-75q-10 0 -17.5 -7.5t-7.5 -17.5v-50q0 -10 7.5 -17.5t17.5 -7.5h350q10 0 17.5 7.5t7.5 17.5v50q0 10 -7.5 17.5 t-17.5 7.5h-75v275q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe087;" d="M525 1200h150q10 0 17.5 -7.5t7.5 -17.5v-194q103 -27 178.5 -102.5t102.5 -178.5h194q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-194q-27 -103 -102.5 -178.5t-178.5 -102.5v-194q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v194 q-103 27 -178.5 102.5t-102.5 178.5h-194q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h194q27 103 102.5 178.5t178.5 102.5v194q0 10 7.5 17.5t17.5 7.5zM700 893v-168q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v168q-68 -23 -119 -74 t-74 -119h168q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-168q23 -68 74 -119t119 -74v168q0 10 7.5 17.5t17.5 7.5h150q10 0 17.5 -7.5t7.5 -17.5v-168q68 23 119 74t74 119h-168q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h168 q-23 68 -74 119t-119 74z" />
<glyph unicode="&#xe088;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM759 823l64 -64q7 -7 7 -17.5t-7 -17.5l-124 -124l124 -124q7 -7 7 -17.5t-7 -17.5l-64 -64q-7 -7 -17.5 -7t-17.5 7l-124 124l-124 -124q-7 -7 -17.5 -7t-17.5 7l-64 64 q-7 7 -7 17.5t7 17.5l124 124l-124 124q-7 7 -7 17.5t7 17.5l64 64q7 7 17.5 7t17.5 -7l124 -124l124 124q7 7 17.5 7t17.5 -7z" />
<glyph unicode="&#xe089;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM782 788l106 -106q7 -7 7 -17.5t-7 -17.5l-320 -321q-8 -7 -18 -7t-18 7l-202 203q-8 7 -8 17.5t8 17.5l106 106q7 8 17.5 8t17.5 -8l79 -79l197 197q7 7 17.5 7t17.5 -7z" />
<glyph unicode="&#xe090;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5q0 -120 65 -225 l587 587q-105 65 -225 65zM965 819l-584 -584q104 -62 219 -62q116 0 214.5 57t155.5 155.5t57 214.5q0 115 -62 219z" />
<glyph unicode="&#xe091;" d="M39 582l522 427q16 13 27.5 8t11.5 -26v-291h550q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-550v-291q0 -21 -11.5 -26t-27.5 8l-522 427q-16 13 -16 32t16 32z" />
<glyph unicode="&#xe092;" d="M639 1009l522 -427q16 -13 16 -32t-16 -32l-522 -427q-16 -13 -27.5 -8t-11.5 26v291h-550q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5h550v291q0 21 11.5 26t27.5 -8z" />
<glyph unicode="&#xe093;" d="M682 1161l427 -522q13 -16 8 -27.5t-26 -11.5h-291v-550q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v550h-291q-21 0 -26 11.5t8 27.5l427 522q13 16 32 16t32 -16z" />
<glyph unicode="&#xe094;" d="M550 1200h200q21 0 35.5 -14.5t14.5 -35.5v-550h291q21 0 26 -11.5t-8 -27.5l-427 -522q-13 -16 -32 -16t-32 16l-427 522q-13 16 -8 27.5t26 11.5h291v550q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe095;" d="M639 1109l522 -427q16 -13 16 -32t-16 -32l-522 -427q-16 -13 -27.5 -8t-11.5 26v291q-94 -2 -182 -20t-170.5 -52t-147 -92.5t-100.5 -135.5q5 105 27 193.5t67.5 167t113 135t167 91.5t225.5 42v262q0 21 11.5 26t27.5 -8z" />
<glyph unicode="&#xe096;" d="M850 1200h300q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -10.5 -25t-24.5 10l-94 94l-249 -249q-8 -7 -18 -7t-18 7l-106 106q-7 8 -7 18t7 18l249 249l-94 94q-14 14 -10 24.5t25 10.5zM350 0h-300q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 10.5 25t24.5 -10l94 -94l249 249 q8 7 18 7t18 -7l106 -106q7 -8 7 -18t-7 -18l-249 -249l94 -94q14 -14 10 -24.5t-25 -10.5z" />
<glyph unicode="&#xe097;" d="M1014 1120l106 -106q7 -8 7 -18t-7 -18l-249 -249l94 -94q14 -14 10 -24.5t-25 -10.5h-300q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 10.5 25t24.5 -10l94 -94l249 249q8 7 18 7t18 -7zM250 600h300q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -10.5 -25t-24.5 10l-94 94 l-249 -249q-8 -7 -18 -7t-18 7l-106 106q-7 8 -7 18t7 18l249 249l-94 94q-14 14 -10 24.5t25 10.5z" />
<glyph unicode="&#xe101;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM704 900h-208q-20 0 -32 -14.5t-8 -34.5l58 -302q4 -20 21.5 -34.5 t37.5 -14.5h54q20 0 37.5 14.5t21.5 34.5l58 302q4 20 -8 34.5t-32 14.5zM675 400h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe102;" d="M260 1200q9 0 19 -2t15 -4l5 -2q22 -10 44 -23l196 -118q21 -13 36 -24q29 -21 37 -12q11 13 49 35l196 118q22 13 45 23q17 7 38 7q23 0 47 -16.5t37 -33.5l13 -16q14 -21 18 -45l25 -123l8 -44q1 -9 8.5 -14.5t17.5 -5.5h61q10 0 17.5 -7.5t7.5 -17.5v-50 q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 -7.5t-7.5 -17.5v-175h-400v300h-200v-300h-400v175q0 10 -7.5 17.5t-17.5 7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5h61q11 0 18 3t7 8q0 4 9 52l25 128q5 25 19 45q2 3 5 7t13.5 15t21.5 19.5t26.5 15.5 t29.5 7zM915 1079l-166 -162q-7 -7 -5 -12t12 -5h219q10 0 15 7t2 17l-51 149q-3 10 -11 12t-15 -6zM463 917l-177 157q-8 7 -16 5t-11 -12l-51 -143q-3 -10 2 -17t15 -7h231q11 0 12.5 5t-5.5 12zM500 0h-375q-10 0 -17.5 7.5t-7.5 17.5v375h400v-400zM1100 400v-375 q0 -10 -7.5 -17.5t-17.5 -7.5h-375v400h400z" />
<glyph unicode="&#xe103;" d="M1165 1190q8 3 21 -6.5t13 -17.5q-2 -178 -24.5 -323.5t-55.5 -245.5t-87 -174.5t-102.5 -118.5t-118 -68.5t-118.5 -33t-120 -4.5t-105 9.5t-90 16.5q-61 12 -78 11q-4 1 -12.5 0t-34 -14.5t-52.5 -40.5l-153 -153q-26 -24 -37 -14.5t-11 43.5q0 64 42 102q8 8 50.5 45 t66.5 58q19 17 35 47t13 61q-9 55 -10 102.5t7 111t37 130t78 129.5q39 51 80 88t89.5 63.5t94.5 45t113.5 36t129 31t157.5 37t182 47.5zM1116 1098q-8 9 -22.5 -3t-45.5 -50q-38 -47 -119 -103.5t-142 -89.5l-62 -33q-56 -30 -102 -57t-104 -68t-102.5 -80.5t-85.5 -91 t-64 -104.5q-24 -56 -31 -86t2 -32t31.5 17.5t55.5 59.5q25 30 94 75.5t125.5 77.5t147.5 81q70 37 118.5 69t102 79.5t99 111t86.5 148.5q22 50 24 60t-6 19z" />
<glyph unicode="&#xe104;" d="M653 1231q-39 -67 -54.5 -131t-10.5 -114.5t24.5 -96.5t47.5 -80t63.5 -62.5t68.5 -46.5t65 -30q-4 7 -17.5 35t-18.5 39.5t-17 39.5t-17 43t-13 42t-9.5 44.5t-2 42t4 43t13.5 39t23 38.5q96 -42 165 -107.5t105 -138t52 -156t13 -159t-19 -149.5q-13 -55 -44 -106.5 t-68 -87t-78.5 -64.5t-72.5 -45t-53 -22q-72 -22 -127 -11q-31 6 -13 19q6 3 17 7q13 5 32.5 21t41 44t38.5 63.5t21.5 81.5t-6.5 94.5t-50 107t-104 115.5q10 -104 -0.5 -189t-37 -140.5t-65 -93t-84 -52t-93.5 -11t-95 24.5q-80 36 -131.5 114t-53.5 171q-2 23 0 49.5 t4.5 52.5t13.5 56t27.5 60t46 64.5t69.5 68.5q-8 -53 -5 -102.5t17.5 -90t34 -68.5t44.5 -39t49 -2q31 13 38.5 36t-4.5 55t-29 64.5t-36 75t-26 75.5q-15 85 2 161.5t53.5 128.5t85.5 92.5t93.5 61t81.5 25.5z" />
<glyph unicode="&#xe105;" d="M600 1094q82 0 160.5 -22.5t140 -59t116.5 -82.5t94.5 -95t68 -95t42.5 -82.5t14 -57.5t-14 -57.5t-43 -82.5t-68.5 -95t-94.5 -95t-116.5 -82.5t-140 -59t-159.5 -22.5t-159.5 22.5t-140 59t-116.5 82.5t-94.5 95t-68.5 95t-43 82.5t-14 57.5t14 57.5t42.5 82.5t68 95 t94.5 95t116.5 82.5t140 59t160.5 22.5zM888 829q-15 15 -18 12t5 -22q25 -57 25 -119q0 -124 -88 -212t-212 -88t-212 88t-88 212q0 59 23 114q8 19 4.5 22t-17.5 -12q-70 -69 -160 -184q-13 -16 -15 -40.5t9 -42.5q22 -36 47 -71t70 -82t92.5 -81t113 -58.5t133.5 -24.5 t133.5 24t113 58.5t92.5 81.5t70 81.5t47 70.5q11 18 9 42.5t-14 41.5q-90 117 -163 189zM448 727l-35 -36q-15 -15 -19.5 -38.5t4.5 -41.5q37 -68 93 -116q16 -13 38.5 -11t36.5 17l35 34q14 15 12.5 33.5t-16.5 33.5q-44 44 -89 117q-11 18 -28 20t-32 -12z" />
<glyph unicode="&#xe106;" d="M592 0h-148l31 120q-91 20 -175.5 68.5t-143.5 106.5t-103.5 119t-66.5 110t-22 76q0 21 14 57.5t42.5 82.5t68 95t94.5 95t116.5 82.5t140 59t160.5 22.5q61 0 126 -15l32 121h148zM944 770l47 181q108 -85 176.5 -192t68.5 -159q0 -26 -19.5 -71t-59.5 -102t-93 -112 t-129 -104.5t-158 -75.5l46 173q77 49 136 117t97 131q11 18 9 42.5t-14 41.5q-54 70 -107 130zM310 824q-70 -69 -160 -184q-13 -16 -15 -40.5t9 -42.5q18 -30 39 -60t57 -70.5t74 -73t90 -61t105 -41.5l41 154q-107 18 -178.5 101.5t-71.5 193.5q0 59 23 114q8 19 4.5 22 t-17.5 -12zM448 727l-35 -36q-15 -15 -19.5 -38.5t4.5 -41.5q37 -68 93 -116q16 -13 38.5 -11t36.5 17l12 11l22 86l-3 4q-44 44 -89 117q-11 18 -28 20t-32 -12z" />
<glyph unicode="&#xe107;" d="M-90 100l642 1066q20 31 48 28.5t48 -35.5l642 -1056q21 -32 7.5 -67.5t-50.5 -35.5h-1294q-37 0 -50.5 34t7.5 66zM155 200h345v75q0 10 7.5 17.5t17.5 7.5h150q10 0 17.5 -7.5t7.5 -17.5v-75h345l-445 723zM496 700h208q20 0 32 -14.5t8 -34.5l-58 -252 q-4 -20 -21.5 -34.5t-37.5 -14.5h-54q-20 0 -37.5 14.5t-21.5 34.5l-58 252q-4 20 8 34.5t32 14.5z" />
<glyph unicode="&#xe108;" d="M650 1200q62 0 106 -44t44 -106v-339l363 -325q15 -14 26 -38.5t11 -44.5v-41q0 -20 -12 -26.5t-29 5.5l-359 249v-263q100 -93 100 -113v-64q0 -21 -13 -29t-32 1l-205 128l-205 -128q-19 -9 -32 -1t-13 29v64q0 20 100 113v263l-359 -249q-17 -12 -29 -5.5t-12 26.5v41 q0 20 11 44.5t26 38.5l363 325v339q0 62 44 106t106 44z" />
<glyph unicode="&#xe109;" d="M850 1200h100q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-150h-1100v150q0 21 14.5 35.5t35.5 14.5h50v50q0 21 14.5 35.5t35.5 14.5h100q21 0 35.5 -14.5t14.5 -35.5v-50h500v50q0 21 14.5 35.5t35.5 14.5zM1100 800v-750q0 -21 -14.5 -35.5 t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v750h1100zM100 600v-100h100v100h-100zM300 600v-100h100v100h-100zM500 600v-100h100v100h-100zM700 600v-100h100v100h-100zM900 600v-100h100v100h-100zM100 400v-100h100v100h-100zM300 400v-100h100v100h-100zM500 400 v-100h100v100h-100zM700 400v-100h100v100h-100zM900 400v-100h100v100h-100zM100 200v-100h100v100h-100zM300 200v-100h100v100h-100zM500 200v-100h100v100h-100zM700 200v-100h100v100h-100zM900 200v-100h100v100h-100z" />
<glyph unicode="&#xe110;" d="M1135 1165l249 -230q15 -14 15 -35t-15 -35l-249 -230q-14 -14 -24.5 -10t-10.5 25v150h-159l-600 -600h-291q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h209l600 600h241v150q0 21 10.5 25t24.5 -10zM522 819l-141 -141l-122 122h-209q-21 0 -35.5 14.5 t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h291zM1135 565l249 -230q15 -14 15 -35t-15 -35l-249 -230q-14 -14 -24.5 -10t-10.5 25v150h-241l-181 181l141 141l122 -122h159v150q0 21 10.5 25t24.5 -10z" />
<glyph unicode="&#xe111;" d="M100 1100h1000q41 0 70.5 -29.5t29.5 -70.5v-600q0 -41 -29.5 -70.5t-70.5 -29.5h-596l-304 -300v300h-100q-41 0 -70.5 29.5t-29.5 70.5v600q0 41 29.5 70.5t70.5 29.5z" />
<glyph unicode="&#xe112;" d="M150 1200h200q21 0 35.5 -14.5t14.5 -35.5v-250h-300v250q0 21 14.5 35.5t35.5 14.5zM850 1200h200q21 0 35.5 -14.5t14.5 -35.5v-250h-300v250q0 21 14.5 35.5t35.5 14.5zM1100 800v-300q0 -41 -3 -77.5t-15 -89.5t-32 -96t-58 -89t-89 -77t-129 -51t-174 -20t-174 20 t-129 51t-89 77t-58 89t-32 96t-15 89.5t-3 77.5v300h300v-250v-27v-42.5t1.5 -41t5 -38t10 -35t16.5 -30t25.5 -24.5t35 -19t46.5 -12t60 -4t60 4.5t46.5 12.5t35 19.5t25 25.5t17 30.5t10 35t5 38t2 40.5t-0.5 42v25v250h300z" />
<glyph unicode="&#xe113;" d="M1100 411l-198 -199l-353 353l-353 -353l-197 199l551 551z" />
<glyph unicode="&#xe114;" d="M1101 789l-550 -551l-551 551l198 199l353 -353l353 353z" />
<glyph unicode="&#xe115;" d="M404 1000h746q21 0 35.5 -14.5t14.5 -35.5v-551h150q21 0 25 -10.5t-10 -24.5l-230 -249q-14 -15 -35 -15t-35 15l-230 249q-14 14 -10 24.5t25 10.5h150v401h-381zM135 984l230 -249q14 -14 10 -24.5t-25 -10.5h-150v-400h385l215 -200h-750q-21 0 -35.5 14.5 t-14.5 35.5v550h-150q-21 0 -25 10.5t10 24.5l230 249q14 15 35 15t35 -15z" />
<glyph unicode="&#xe116;" d="M56 1200h94q17 0 31 -11t18 -27l38 -162h896q24 0 39 -18.5t10 -42.5l-100 -475q-5 -21 -27 -42.5t-55 -21.5h-633l48 -200h535q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-50q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v50h-300v-50 q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v50h-31q-18 0 -32.5 10t-20.5 19l-5 10l-201 961h-54q-20 0 -35 14.5t-15 35.5t15 35.5t35 14.5z" />
<glyph unicode="&#xe117;" d="M1200 1000v-100h-1200v100h200q0 41 29.5 70.5t70.5 29.5h300q41 0 70.5 -29.5t29.5 -70.5h500zM0 800h1200v-800h-1200v800z" />
<glyph unicode="&#xe118;" d="M200 800l-200 -400v600h200q0 41 29.5 70.5t70.5 29.5h300q42 0 71 -29.5t29 -70.5h500v-200h-1000zM1500 700l-300 -700h-1200l300 700h1200z" />
<glyph unicode="&#xe119;" d="M635 1184l230 -249q14 -14 10 -24.5t-25 -10.5h-150v-601h150q21 0 25 -10.5t-10 -24.5l-230 -249q-14 -15 -35 -15t-35 15l-230 249q-14 14 -10 24.5t25 10.5h150v601h-150q-21 0 -25 10.5t10 24.5l230 249q14 15 35 15t35 -15z" />
<glyph unicode="&#xe120;" d="M936 864l249 -229q14 -15 14 -35.5t-14 -35.5l-249 -229q-15 -15 -25.5 -10.5t-10.5 24.5v151h-600v-151q0 -20 -10.5 -24.5t-25.5 10.5l-249 229q-14 15 -14 35.5t14 35.5l249 229q15 15 25.5 10.5t10.5 -25.5v-149h600v149q0 21 10.5 25.5t25.5 -10.5z" />
<glyph unicode="&#xe121;" d="M1169 400l-172 732q-5 23 -23 45.5t-38 22.5h-672q-20 0 -38 -20t-23 -41l-172 -739h1138zM1100 300h-1000q-41 0 -70.5 -29.5t-29.5 -70.5v-100q0 -41 29.5 -70.5t70.5 -29.5h1000q41 0 70.5 29.5t29.5 70.5v100q0 41 -29.5 70.5t-70.5 29.5zM800 100v100h100v-100h-100 zM1000 100v100h100v-100h-100z" />
<glyph unicode="&#xe122;" d="M1150 1100q21 0 35.5 -14.5t14.5 -35.5v-850q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v850q0 21 14.5 35.5t35.5 14.5zM1000 200l-675 200h-38l47 -276q3 -16 -5.5 -20t-29.5 -4h-7h-84q-20 0 -34.5 14t-18.5 35q-55 337 -55 351v250v6q0 16 1 23.5t6.5 14 t17.5 6.5h200l675 250v-850zM0 750v-250q-4 0 -11 0.5t-24 6t-30 15t-24 30t-11 48.5v50q0 26 10.5 46t25 30t29 16t25.5 7z" />
<glyph unicode="&#xe123;" d="M553 1200h94q20 0 29 -10.5t3 -29.5l-18 -37q83 -19 144 -82.5t76 -140.5l63 -327l118 -173h17q19 0 33 -14.5t14 -35t-13 -40.5t-31 -27q-8 -4 -23 -9.5t-65 -19.5t-103 -25t-132.5 -20t-158.5 -9q-57 0 -115 5t-104 12t-88.5 15.5t-73.5 17.5t-54.5 16t-35.5 12l-11 4 q-18 8 -31 28t-13 40.5t14 35t33 14.5h17l118 173l63 327q15 77 76 140t144 83l-18 32q-6 19 3.5 32t28.5 13zM498 110q50 -6 102 -6q53 0 102 6q-12 -49 -39.5 -79.5t-62.5 -30.5t-63 30.5t-39 79.5z" />
<glyph unicode="&#xe124;" d="M800 946l224 78l-78 -224l234 -45l-180 -155l180 -155l-234 -45l78 -224l-224 78l-45 -234l-155 180l-155 -180l-45 234l-224 -78l78 224l-234 45l180 155l-180 155l234 45l-78 224l224 -78l45 234l155 -180l155 180z" />
<glyph unicode="&#xe125;" d="M650 1200h50q40 0 70 -40.5t30 -84.5v-150l-28 -125h328q40 0 70 -40.5t30 -84.5v-100q0 -45 -29 -74l-238 -344q-16 -24 -38 -40.5t-45 -16.5h-250q-7 0 -42 25t-66 50l-31 25h-61q-45 0 -72.5 18t-27.5 57v400q0 36 20 63l145 196l96 198q13 28 37.5 48t51.5 20z M650 1100l-100 -212l-150 -213v-375h100l136 -100h214l250 375v125h-450l50 225v175h-50zM50 800h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v500q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe126;" d="M600 1100h250q23 0 45 -16.5t38 -40.5l238 -344q29 -29 29 -74v-100q0 -44 -30 -84.5t-70 -40.5h-328q28 -118 28 -125v-150q0 -44 -30 -84.5t-70 -40.5h-50q-27 0 -51.5 20t-37.5 48l-96 198l-145 196q-20 27 -20 63v400q0 39 27.5 57t72.5 18h61q124 100 139 100z M50 1000h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v500q0 21 14.5 35.5t35.5 14.5zM636 1000l-136 -100h-100v-375l150 -213l100 -212h50v175l-50 225h450v125l-250 375h-214z" />
<glyph unicode="&#xe127;" d="M356 873l363 230q31 16 53 -6l110 -112q13 -13 13.5 -32t-11.5 -34l-84 -121h302q84 0 138 -38t54 -110t-55 -111t-139 -39h-106l-131 -339q-6 -21 -19.5 -41t-28.5 -20h-342q-7 0 -90 81t-83 94v525q0 17 14 35.5t28 28.5zM400 792v-503l100 -89h293l131 339 q6 21 19.5 41t28.5 20h203q21 0 30.5 25t0.5 50t-31 25h-456h-7h-6h-5.5t-6 0.5t-5 1.5t-5 2t-4 2.5t-4 4t-2.5 4.5q-12 25 5 47l146 183l-86 83zM50 800h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v500 q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe128;" d="M475 1103l366 -230q2 -1 6 -3.5t14 -10.5t18 -16.5t14.5 -20t6.5 -22.5v-525q0 -13 -86 -94t-93 -81h-342q-15 0 -28.5 20t-19.5 41l-131 339h-106q-85 0 -139.5 39t-54.5 111t54 110t138 38h302l-85 121q-11 15 -10.5 34t13.5 32l110 112q22 22 53 6zM370 945l146 -183 q17 -22 5 -47q-2 -2 -3.5 -4.5t-4 -4t-4 -2.5t-5 -2t-5 -1.5t-6 -0.5h-6h-6.5h-6h-475v-100h221q15 0 29 -20t20 -41l130 -339h294l106 89v503l-342 236zM1050 800h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5 v500q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe129;" d="M550 1294q72 0 111 -55t39 -139v-106l339 -131q21 -6 41 -19.5t20 -28.5v-342q0 -7 -81 -90t-94 -83h-525q-17 0 -35.5 14t-28.5 28l-9 14l-230 363q-16 31 6 53l112 110q13 13 32 13.5t34 -11.5l121 -84v302q0 84 38 138t110 54zM600 972v203q0 21 -25 30.5t-50 0.5 t-25 -31v-456v-7v-6v-5.5t-0.5 -6t-1.5 -5t-2 -5t-2.5 -4t-4 -4t-4.5 -2.5q-25 -12 -47 5l-183 146l-83 -86l236 -339h503l89 100v293l-339 131q-21 6 -41 19.5t-20 28.5zM450 200h500q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-500 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe130;" d="M350 1100h500q21 0 35.5 14.5t14.5 35.5v100q0 21 -14.5 35.5t-35.5 14.5h-500q-21 0 -35.5 -14.5t-14.5 -35.5v-100q0 -21 14.5 -35.5t35.5 -14.5zM600 306v-106q0 -84 -39 -139t-111 -55t-110 54t-38 138v302l-121 -84q-15 -12 -34 -11.5t-32 13.5l-112 110 q-22 22 -6 53l230 363q1 2 3.5 6t10.5 13.5t16.5 17t20 13.5t22.5 6h525q13 0 94 -83t81 -90v-342q0 -15 -20 -28.5t-41 -19.5zM308 900l-236 -339l83 -86l183 146q22 17 47 5q2 -1 4.5 -2.5t4 -4t2.5 -4t2 -5t1.5 -5t0.5 -6v-5.5v-6v-7v-456q0 -22 25 -31t50 0.5t25 30.5 v203q0 15 20 28.5t41 19.5l339 131v293l-89 100h-503z" />
<glyph unicode="&#xe131;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM914 632l-275 223q-16 13 -27.5 8t-11.5 -26v-137h-275 q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h275v-137q0 -21 11.5 -26t27.5 8l275 223q16 13 16 32t-16 32z" />
<glyph unicode="&#xe132;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM561 855l-275 -223q-16 -13 -16 -32t16 -32l275 -223q16 -13 27.5 -8 t11.5 26v137h275q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5h-275v137q0 21 -11.5 26t-27.5 -8z" />
<glyph unicode="&#xe133;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM855 639l-223 275q-13 16 -32 16t-32 -16l-223 -275q-13 -16 -8 -27.5 t26 -11.5h137v-275q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v275h137q21 0 26 11.5t-8 27.5z" />
<glyph unicode="&#xe134;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM675 900h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-275h-137q-21 0 -26 -11.5 t8 -27.5l223 -275q13 -16 32 -16t32 16l223 275q13 16 8 27.5t-26 11.5h-137v275q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe135;" d="M600 1176q116 0 222.5 -46t184 -123.5t123.5 -184t46 -222.5t-46 -222.5t-123.5 -184t-184 -123.5t-222.5 -46t-222.5 46t-184 123.5t-123.5 184t-46 222.5t46 222.5t123.5 184t184 123.5t222.5 46zM627 1101q-15 -12 -36.5 -20.5t-35.5 -12t-43 -8t-39 -6.5 q-15 -3 -45.5 0t-45.5 -2q-20 -7 -51.5 -26.5t-34.5 -34.5q-3 -11 6.5 -22.5t8.5 -18.5q-3 -34 -27.5 -91t-29.5 -79q-9 -34 5 -93t8 -87q0 -9 17 -44.5t16 -59.5q12 0 23 -5t23.5 -15t19.5 -14q16 -8 33 -15t40.5 -15t34.5 -12q21 -9 52.5 -32t60 -38t57.5 -11 q7 -15 -3 -34t-22.5 -40t-9.5 -38q13 -21 23 -34.5t27.5 -27.5t36.5 -18q0 -7 -3.5 -16t-3.5 -14t5 -17q104 -2 221 112q30 29 46.5 47t34.5 49t21 63q-13 8 -37 8.5t-36 7.5q-15 7 -49.5 15t-51.5 19q-18 0 -41 -0.5t-43 -1.5t-42 -6.5t-38 -16.5q-51 -35 -66 -12 q-4 1 -3.5 25.5t0.5 25.5q-6 13 -26.5 17.5t-24.5 6.5q1 15 -0.5 30.5t-7 28t-18.5 11.5t-31 -21q-23 -25 -42 4q-19 28 -8 58q6 16 22 22q6 -1 26 -1.5t33.5 -4t19.5 -13.5q7 -12 18 -24t21.5 -20.5t20 -15t15.5 -10.5l5 -3q2 12 7.5 30.5t8 34.5t-0.5 32q-3 18 3.5 29 t18 22.5t15.5 24.5q6 14 10.5 35t8 31t15.5 22.5t34 22.5q-6 18 10 36q8 0 24 -1.5t24.5 -1.5t20 4.5t20.5 15.5q-10 23 -31 42.5t-37.5 29.5t-49 27t-43.5 23q0 1 2 8t3 11.5t1.5 10.5t-1 9.5t-4.5 4.5q31 -13 58.5 -14.5t38.5 2.5l12 5q5 28 -9.5 46t-36.5 24t-50 15 t-41 20q-18 -4 -37 0zM613 994q0 -17 8 -42t17 -45t9 -23q-8 1 -39.5 5.5t-52.5 10t-37 16.5q3 11 16 29.5t16 25.5q10 -10 19 -10t14 6t13.5 14.5t16.5 12.5z" />
<glyph unicode="&#xe136;" d="M756 1157q164 92 306 -9l-259 -138l145 -232l251 126q6 -89 -34 -156.5t-117 -110.5q-60 -34 -127 -39.5t-126 16.5l-596 -596q-15 -16 -36.5 -16t-36.5 16l-111 110q-15 15 -15 36.5t15 37.5l600 599q-34 101 5.5 201.5t135.5 154.5z" />
<glyph unicode="&#xe137;" horiz-adv-x="1220" d="M100 1196h1000q41 0 70.5 -29.5t29.5 -70.5v-100q0 -41 -29.5 -70.5t-70.5 -29.5h-1000q-41 0 -70.5 29.5t-29.5 70.5v100q0 41 29.5 70.5t70.5 29.5zM1100 1096h-200v-100h200v100zM100 796h1000q41 0 70.5 -29.5t29.5 -70.5v-100q0 -41 -29.5 -70.5t-70.5 -29.5h-1000 q-41 0 -70.5 29.5t-29.5 70.5v100q0 41 29.5 70.5t70.5 29.5zM1100 696h-500v-100h500v100zM100 396h1000q41 0 70.5 -29.5t29.5 -70.5v-100q0 -41 -29.5 -70.5t-70.5 -29.5h-1000q-41 0 -70.5 29.5t-29.5 70.5v100q0 41 29.5 70.5t70.5 29.5zM1100 296h-300v-100h300v100z " />
<glyph unicode="&#xe138;" d="M150 1200h900q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM700 500v-300l-200 -200v500l-350 500h900z" />
<glyph unicode="&#xe139;" d="M500 1200h200q41 0 70.5 -29.5t29.5 -70.5v-100h300q41 0 70.5 -29.5t29.5 -70.5v-400h-500v100h-200v-100h-500v400q0 41 29.5 70.5t70.5 29.5h300v100q0 41 29.5 70.5t70.5 29.5zM500 1100v-100h200v100h-200zM1200 400v-200q0 -41 -29.5 -70.5t-70.5 -29.5h-1000 q-41 0 -70.5 29.5t-29.5 70.5v200h1200z" />
<glyph unicode="&#xe140;" d="M50 1200h300q21 0 25 -10.5t-10 -24.5l-94 -94l199 -199q7 -8 7 -18t-7 -18l-106 -106q-8 -7 -18 -7t-18 7l-199 199l-94 -94q-14 -14 -24.5 -10t-10.5 25v300q0 21 14.5 35.5t35.5 14.5zM850 1200h300q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -10.5 -25t-24.5 10l-94 94 l-199 -199q-8 -7 -18 -7t-18 7l-106 106q-7 8 -7 18t7 18l199 199l-94 94q-14 14 -10 24.5t25 10.5zM364 470l106 -106q7 -8 7 -18t-7 -18l-199 -199l94 -94q14 -14 10 -24.5t-25 -10.5h-300q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 10.5 25t24.5 -10l94 -94l199 199 q8 7 18 7t18 -7zM1071 271l94 94q14 14 24.5 10t10.5 -25v-300q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -25 10.5t10 24.5l94 94l-199 199q-7 8 -7 18t7 18l106 106q8 7 18 7t18 -7z" />
<glyph unicode="&#xe141;" d="M596 1192q121 0 231.5 -47.5t190 -127t127 -190t47.5 -231.5t-47.5 -231.5t-127 -190.5t-190 -127t-231.5 -47t-231.5 47t-190.5 127t-127 190.5t-47 231.5t47 231.5t127 190t190.5 127t231.5 47.5zM596 1010q-112 0 -207.5 -55.5t-151 -151t-55.5 -207.5t55.5 -207.5 t151 -151t207.5 -55.5t207.5 55.5t151 151t55.5 207.5t-55.5 207.5t-151 151t-207.5 55.5zM454.5 905q22.5 0 38.5 -16t16 -38.5t-16 -39t-38.5 -16.5t-38.5 16.5t-16 39t16 38.5t38.5 16zM754.5 905q22.5 0 38.5 -16t16 -38.5t-16 -39t-38 -16.5q-14 0 -29 10l-55 -145 q17 -23 17 -51q0 -36 -25.5 -61.5t-61.5 -25.5t-61.5 25.5t-25.5 61.5q0 32 20.5 56.5t51.5 29.5l122 126l1 1q-9 14 -9 28q0 23 16 39t38.5 16zM345.5 709q22.5 0 38.5 -16t16 -38.5t-16 -38.5t-38.5 -16t-38.5 16t-16 38.5t16 38.5t38.5 16zM854.5 709q22.5 0 38.5 -16 t16 -38.5t-16 -38.5t-38.5 -16t-38.5 16t-16 38.5t16 38.5t38.5 16z" />
<glyph unicode="&#xe142;" d="M546 173l469 470q91 91 99 192q7 98 -52 175.5t-154 94.5q-22 4 -47 4q-34 0 -66.5 -10t-56.5 -23t-55.5 -38t-48 -41.5t-48.5 -47.5q-376 -375 -391 -390q-30 -27 -45 -41.5t-37.5 -41t-32 -46.5t-16 -47.5t-1.5 -56.5q9 -62 53.5 -95t99.5 -33q74 0 125 51l548 548 q36 36 20 75q-7 16 -21.5 26t-32.5 10q-26 0 -50 -23q-13 -12 -39 -38l-341 -338q-15 -15 -35.5 -15.5t-34.5 13.5t-14 34.5t14 34.5q327 333 361 367q35 35 67.5 51.5t78.5 16.5q14 0 29 -1q44 -8 74.5 -35.5t43.5 -68.5q14 -47 2 -96.5t-47 -84.5q-12 -11 -32 -32 t-79.5 -81t-114.5 -115t-124.5 -123.5t-123 -119.5t-96.5 -89t-57 -45q-56 -27 -120 -27q-70 0 -129 32t-93 89q-48 78 -35 173t81 163l511 511q71 72 111 96q91 55 198 55q80 0 152 -33q78 -36 129.5 -103t66.5 -154q17 -93 -11 -183.5t-94 -156.5l-482 -476 q-15 -15 -36 -16t-37 14t-17.5 34t14.5 35z" />
<glyph unicode="&#xe143;" d="M649 949q48 68 109.5 104t121.5 38.5t118.5 -20t102.5 -64t71 -100.5t27 -123q0 -57 -33.5 -117.5t-94 -124.5t-126.5 -127.5t-150 -152.5t-146 -174q-62 85 -145.5 174t-150 152.5t-126.5 127.5t-93.5 124.5t-33.5 117.5q0 64 28 123t73 100.5t104 64t119 20 t120.5 -38.5t104.5 -104zM896 972q-33 0 -64.5 -19t-56.5 -46t-47.5 -53.5t-43.5 -45.5t-37.5 -19t-36 19t-40 45.5t-43 53.5t-54 46t-65.5 19q-67 0 -122.5 -55.5t-55.5 -132.5q0 -23 13.5 -51t46 -65t57.5 -63t76 -75l22 -22q15 -14 44 -44t50.5 -51t46 -44t41 -35t23 -12 t23.5 12t42.5 36t46 44t52.5 52t44 43q4 4 12 13q43 41 63.5 62t52 55t46 55t26 46t11.5 44q0 79 -53 133.5t-120 54.5z" />
<glyph unicode="&#xe144;" d="M776.5 1214q93.5 0 159.5 -66l141 -141q66 -66 66 -160q0 -42 -28 -95.5t-62 -87.5l-29 -29q-31 53 -77 99l-18 18l95 95l-247 248l-389 -389l212 -212l-105 -106l-19 18l-141 141q-66 66 -66 159t66 159l283 283q65 66 158.5 66zM600 706l105 105q10 -8 19 -17l141 -141 q66 -66 66 -159t-66 -159l-283 -283q-66 -66 -159 -66t-159 66l-141 141q-66 66 -66 159.5t66 159.5l55 55q29 -55 75 -102l18 -17l-95 -95l247 -248l389 389z" />
<glyph unicode="&#xe145;" d="M603 1200q85 0 162 -15t127 -38t79 -48t29 -46v-953q0 -41 -29.5 -70.5t-70.5 -29.5h-600q-41 0 -70.5 29.5t-29.5 70.5v953q0 21 30 46.5t81 48t129 37.5t163 15zM300 1000v-700h600v700h-600zM600 254q-43 0 -73.5 -30.5t-30.5 -73.5t30.5 -73.5t73.5 -30.5t73.5 30.5 t30.5 73.5t-30.5 73.5t-73.5 30.5z" />
<glyph unicode="&#xe146;" d="M902 1185l283 -282q15 -15 15 -36t-14.5 -35.5t-35.5 -14.5t-35 15l-36 35l-279 -267v-300l-212 210l-308 -307l-280 -203l203 280l307 308l-210 212h300l267 279l-35 36q-15 14 -15 35t14.5 35.5t35.5 14.5t35 -15z" />
<glyph unicode="&#xe148;" d="M700 1248v-78q38 -5 72.5 -14.5t75.5 -31.5t71 -53.5t52 -84t24 -118.5h-159q-4 36 -10.5 59t-21 45t-40 35.5t-64.5 20.5v-307l64 -13q34 -7 64 -16.5t70 -32t67.5 -52.5t47.5 -80t20 -112q0 -139 -89 -224t-244 -97v-77h-100v79q-150 16 -237 103q-40 40 -52.5 93.5 t-15.5 139.5h139q5 -77 48.5 -126t117.5 -65v335l-27 8q-46 14 -79 26.5t-72 36t-63 52t-40 72.5t-16 98q0 70 25 126t67.5 92t94.5 57t110 27v77h100zM600 754v274q-29 -4 -50 -11t-42 -21.5t-31.5 -41.5t-10.5 -65q0 -29 7 -50.5t16.5 -34t28.5 -22.5t31.5 -14t37.5 -10 q9 -3 13 -4zM700 547v-310q22 2 42.5 6.5t45 15.5t41.5 27t29 42t12 59.5t-12.5 59.5t-38 44.5t-53 31t-66.5 24.5z" />
<glyph unicode="&#xe149;" d="M561 1197q84 0 160.5 -40t123.5 -109.5t47 -147.5h-153q0 40 -19.5 71.5t-49.5 48.5t-59.5 26t-55.5 9q-37 0 -79 -14.5t-62 -35.5q-41 -44 -41 -101q0 -26 13.5 -63t26.5 -61t37 -66q6 -9 9 -14h241v-100h-197q8 -50 -2.5 -115t-31.5 -95q-45 -62 -99 -112 q34 10 83 17.5t71 7.5q32 1 102 -16t104 -17q83 0 136 30l50 -147q-31 -19 -58 -30.5t-55 -15.5t-42 -4.5t-46 -0.5q-23 0 -76 17t-111 32.5t-96 11.5q-39 -3 -82 -16t-67 -25l-23 -11l-55 145q4 3 16 11t15.5 10.5t13 9t15.5 12t14.5 14t17.5 18.5q48 55 54 126.5 t-30 142.5h-221v100h166q-23 47 -44 104q-7 20 -12 41.5t-6 55.5t6 66.5t29.5 70.5t58.5 71q97 88 263 88z" />
<glyph unicode="&#xe150;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM935 1184l230 -249q14 -14 10 -24.5t-25 -10.5h-150v-900h-200v900h-150q-21 0 -25 10.5t10 24.5l230 249q14 15 35 15t35 -15z" />
<glyph unicode="&#xe151;" d="M1000 700h-100v100h-100v-100h-100v500h300v-500zM400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM801 1100v-200h100v200h-100zM1000 350l-200 -250h200v-100h-300v150l200 250h-200v100h300v-150z " />
<glyph unicode="&#xe152;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1000 1050l-200 -250h200v-100h-300v150l200 250h-200v100h300v-150zM1000 0h-100v100h-100v-100h-100v500h300v-500zM801 400v-200h100v200h-100z " />
<glyph unicode="&#xe153;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1000 700h-100v400h-100v100h200v-500zM1100 0h-100v100h-200v400h300v-500zM901 400v-200h100v200h-100z" />
<glyph unicode="&#xe154;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1100 700h-100v100h-200v400h300v-500zM901 1100v-200h100v200h-100zM1000 0h-100v400h-100v100h200v-500z" />
<glyph unicode="&#xe155;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM900 1000h-200v200h200v-200zM1000 700h-300v200h300v-200zM1100 400h-400v200h400v-200zM1200 100h-500v200h500v-200z" />
<glyph unicode="&#xe156;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1200 1000h-500v200h500v-200zM1100 700h-400v200h400v-200zM1000 400h-300v200h300v-200zM900 100h-200v200h200v-200z" />
<glyph unicode="&#xe157;" d="M350 1100h400q162 0 256 -93.5t94 -256.5v-400q0 -165 -93.5 -257.5t-256.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5z" />
<glyph unicode="&#xe158;" d="M350 1100h400q165 0 257.5 -92.5t92.5 -257.5v-400q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-163 0 -256.5 92.5t-93.5 257.5v400q0 163 94 256.5t256 93.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5zM440 770l253 -190q17 -12 17 -30t-17 -30l-253 -190q-16 -12 -28 -6.5t-12 26.5v400q0 21 12 26.5t28 -6.5z" />
<glyph unicode="&#xe159;" d="M350 1100h400q163 0 256.5 -94t93.5 -256v-400q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 163 92.5 256.5t257.5 93.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5zM350 700h400q21 0 26.5 -12t-6.5 -28l-190 -253q-12 -17 -30 -17t-30 17l-190 253q-12 16 -6.5 28t26.5 12z" />
<glyph unicode="&#xe160;" d="M350 1100h400q165 0 257.5 -92.5t92.5 -257.5v-400q0 -163 -92.5 -256.5t-257.5 -93.5h-400q-163 0 -256.5 94t-93.5 256v400q0 165 92.5 257.5t257.5 92.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5zM580 693l190 -253q12 -16 6.5 -28t-26.5 -12h-400q-21 0 -26.5 12t6.5 28l190 253q12 17 30 17t30 -17z" />
<glyph unicode="&#xe161;" d="M550 1100h400q165 0 257.5 -92.5t92.5 -257.5v-400q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h450q41 0 70.5 29.5t29.5 70.5v500q0 41 -29.5 70.5t-70.5 29.5h-450q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM338 867l324 -284q16 -14 16 -33t-16 -33l-324 -284q-16 -14 -27 -9t-11 26v150h-250q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5h250v150q0 21 11 26t27 -9z" />
<glyph unicode="&#xe162;" d="M793 1182l9 -9q8 -10 5 -27q-3 -11 -79 -225.5t-78 -221.5l300 1q24 0 32.5 -17.5t-5.5 -35.5q-1 0 -133.5 -155t-267 -312.5t-138.5 -162.5q-12 -15 -26 -15h-9l-9 8q-9 11 -4 32q2 9 42 123.5t79 224.5l39 110h-302q-23 0 -31 19q-10 21 6 41q75 86 209.5 237.5 t228 257t98.5 111.5q9 16 25 16h9z" />
<glyph unicode="&#xe163;" d="M350 1100h400q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-450q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h450q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400 q0 165 92.5 257.5t257.5 92.5zM938 867l324 -284q16 -14 16 -33t-16 -33l-324 -284q-16 -14 -27 -9t-11 26v150h-250q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5h250v150q0 21 11 26t27 -9z" />
<glyph unicode="&#xe164;" d="M750 1200h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -10.5 -25t-24.5 10l-109 109l-312 -312q-15 -15 -35.5 -15t-35.5 15l-141 141q-15 15 -15 35.5t15 35.5l312 312l-109 109q-14 14 -10 24.5t25 10.5zM456 900h-156q-41 0 -70.5 -29.5t-29.5 -70.5v-500 q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5v148l200 200v-298q0 -165 -93.5 -257.5t-256.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5h300z" />
<glyph unicode="&#xe165;" d="M600 1186q119 0 227.5 -46.5t187 -125t125 -187t46.5 -227.5t-46.5 -227.5t-125 -187t-187 -125t-227.5 -46.5t-227.5 46.5t-187 125t-125 187t-46.5 227.5t46.5 227.5t125 187t187 125t227.5 46.5zM600 1022q-115 0 -212 -56.5t-153.5 -153.5t-56.5 -212t56.5 -212 t153.5 -153.5t212 -56.5t212 56.5t153.5 153.5t56.5 212t-56.5 212t-153.5 153.5t-212 56.5zM600 794q80 0 137 -57t57 -137t-57 -137t-137 -57t-137 57t-57 137t57 137t137 57z" />
<glyph unicode="&#xe166;" d="M450 1200h200q21 0 35.5 -14.5t14.5 -35.5v-350h245q20 0 25 -11t-9 -26l-383 -426q-14 -15 -33.5 -15t-32.5 15l-379 426q-13 15 -8.5 26t25.5 11h250v350q0 21 14.5 35.5t35.5 14.5zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5z M900 200v-50h100v50h-100z" />
<glyph unicode="&#xe167;" d="M583 1182l378 -435q14 -15 9 -31t-26 -16h-244v-250q0 -20 -17 -35t-39 -15h-200q-20 0 -32 14.5t-12 35.5v250h-250q-20 0 -25.5 16.5t8.5 31.5l383 431q14 16 33.5 17t33.5 -14zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5z M900 200v-50h100v50h-100z" />
<glyph unicode="&#xe168;" d="M396 723l369 369q7 7 17.5 7t17.5 -7l139 -139q7 -8 7 -18.5t-7 -17.5l-525 -525q-7 -8 -17.5 -8t-17.5 8l-292 291q-7 8 -7 18t7 18l139 139q8 7 18.5 7t17.5 -7zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5zM900 200v-50h100v50 h-100z" />
<glyph unicode="&#xe169;" d="M135 1023l142 142q14 14 35 14t35 -14l77 -77l-212 -212l-77 76q-14 15 -14 36t14 35zM655 855l210 210q14 14 24.5 10t10.5 -25l-2 -599q-1 -20 -15.5 -35t-35.5 -15l-597 -1q-21 0 -25 10.5t10 24.5l208 208l-154 155l212 212zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5 v-250h-1100v250q0 21 14.5 35.5t35.5 14.5zM900 200v-50h100v50h-100z" />
<glyph unicode="&#xe170;" d="M350 1200l599 -2q20 -1 35 -15.5t15 -35.5l1 -597q0 -21 -10.5 -25t-24.5 10l-208 208l-155 -154l-212 212l155 154l-210 210q-14 14 -10 24.5t25 10.5zM524 512l-76 -77q-15 -14 -36 -14t-35 14l-142 142q-14 14 -14 35t14 35l77 77zM50 300h1000q21 0 35.5 -14.5 t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5zM900 200v-50h100v50h-100z" />
<glyph unicode="&#xe171;" d="M1200 103l-483 276l-314 -399v423h-399l1196 796v-1096zM483 424v-230l683 953z" />
<glyph unicode="&#xe172;" d="M1100 1000v-850q0 -21 -14.5 -35.5t-35.5 -14.5h-150v400h-700v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200z" />
<glyph unicode="&#xe173;" d="M1100 1000l-2 -149l-299 -299l-95 95q-9 9 -21.5 9t-21.5 -9l-149 -147h-312v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM1132 638l106 -106q7 -7 7 -17.5t-7 -17.5l-420 -421q-8 -7 -18 -7 t-18 7l-202 203q-8 7 -8 17.5t8 17.5l106 106q7 8 17.5 8t17.5 -8l79 -79l297 297q7 7 17.5 7t17.5 -7z" />
<glyph unicode="&#xe174;" d="M1100 1000v-269l-103 -103l-134 134q-15 15 -33.5 16.5t-34.5 -12.5l-266 -266h-329v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM1202 572l70 -70q15 -15 15 -35.5t-15 -35.5l-131 -131 l131 -131q15 -15 15 -35.5t-15 -35.5l-70 -70q-15 -15 -35.5 -15t-35.5 15l-131 131l-131 -131q-15 -15 -35.5 -15t-35.5 15l-70 70q-15 15 -15 35.5t15 35.5l131 131l-131 131q-15 15 -15 35.5t15 35.5l70 70q15 15 35.5 15t35.5 -15l131 -131l131 131q15 15 35.5 15 t35.5 -15z" />
<glyph unicode="&#xe175;" d="M1100 1000v-300h-350q-21 0 -35.5 -14.5t-14.5 -35.5v-150h-500v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM850 600h100q21 0 35.5 -14.5t14.5 -35.5v-250h150q21 0 25 -10.5t-10 -24.5 l-230 -230q-14 -14 -35 -14t-35 14l-230 230q-14 14 -10 24.5t25 10.5h150v250q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe176;" d="M1100 1000v-400l-165 165q-14 15 -35 15t-35 -15l-263 -265h-402v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM935 565l230 -229q14 -15 10 -25.5t-25 -10.5h-150v-250q0 -20 -14.5 -35 t-35.5 -15h-100q-21 0 -35.5 15t-14.5 35v250h-150q-21 0 -25 10.5t10 25.5l230 229q14 15 35 15t35 -15z" />
<glyph unicode="&#xe177;" d="M50 1100h1100q21 0 35.5 -14.5t14.5 -35.5v-150h-1200v150q0 21 14.5 35.5t35.5 14.5zM1200 800v-550q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v550h1200zM100 500v-200h400v200h-400z" />
<glyph unicode="&#xe178;" d="M935 1165l248 -230q14 -14 14 -35t-14 -35l-248 -230q-14 -14 -24.5 -10t-10.5 25v150h-400v200h400v150q0 21 10.5 25t24.5 -10zM200 800h-50q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h50v-200zM400 800h-100v200h100v-200zM18 435l247 230 q14 14 24.5 10t10.5 -25v-150h400v-200h-400v-150q0 -21 -10.5 -25t-24.5 10l-247 230q-15 14 -15 35t15 35zM900 300h-100v200h100v-200zM1000 500h51q20 0 34.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-34.5 -14.5h-51v200z" />
<glyph unicode="&#xe179;" d="M862 1073l276 116q25 18 43.5 8t18.5 -41v-1106q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v397q-4 1 -11 5t-24 17.5t-30 29t-24 42t-11 56.5v359q0 31 18.5 65t43.5 52zM550 1200q22 0 34.5 -12.5t14.5 -24.5l1 -13v-450q0 -28 -10.5 -59.5 t-25 -56t-29 -45t-25.5 -31.5l-10 -11v-447q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v447q-4 4 -11 11.5t-24 30.5t-30 46t-24 55t-11 60v450q0 2 0.5 5.5t4 12t8.5 15t14.5 12t22.5 5.5q20 0 32.5 -12.5t14.5 -24.5l3 -13v-350h100v350v5.5t2.5 12 t7 15t15 12t25.5 5.5q23 0 35.5 -12.5t13.5 -24.5l1 -13v-350h100v350q0 2 0.5 5.5t3 12t7 15t15 12t24.5 5.5z" />
<glyph unicode="&#xe180;" d="M1200 1100v-56q-4 0 -11 -0.5t-24 -3t-30 -7.5t-24 -15t-11 -24v-888q0 -22 25 -34.5t50 -13.5l25 -2v-56h-400v56q75 0 87.5 6.5t12.5 43.5v394h-500v-394q0 -37 12.5 -43.5t87.5 -6.5v-56h-400v56q4 0 11 0.5t24 3t30 7.5t24 15t11 24v888q0 22 -25 34.5t-50 13.5 l-25 2v56h400v-56q-75 0 -87.5 -6.5t-12.5 -43.5v-394h500v394q0 37 -12.5 43.5t-87.5 6.5v56h400z" />
<glyph unicode="&#xe181;" d="M675 1000h375q21 0 35.5 -14.5t14.5 -35.5v-150h-105l-295 -98v98l-200 200h-400l100 100h375zM100 900h300q41 0 70.5 -29.5t29.5 -70.5v-500q0 -41 -29.5 -70.5t-70.5 -29.5h-300q-41 0 -70.5 29.5t-29.5 70.5v500q0 41 29.5 70.5t70.5 29.5zM100 800v-200h300v200 h-300zM1100 535l-400 -133v163l400 133v-163zM100 500v-200h300v200h-300zM1100 398v-248q0 -21 -14.5 -35.5t-35.5 -14.5h-375l-100 -100h-375l-100 100h400l200 200h105z" />
<glyph unicode="&#xe182;" d="M17 1007l162 162q17 17 40 14t37 -22l139 -194q14 -20 11 -44.5t-20 -41.5l-119 -118q102 -142 228 -268t267 -227l119 118q17 17 42.5 19t44.5 -12l192 -136q19 -14 22.5 -37.5t-13.5 -40.5l-163 -162q-3 -1 -9.5 -1t-29.5 2t-47.5 6t-62.5 14.5t-77.5 26.5t-90 42.5 t-101.5 60t-111 83t-119 108.5q-74 74 -133.5 150.5t-94.5 138.5t-60 119.5t-34.5 100t-15 74.5t-4.5 48z" />
<glyph unicode="&#xe183;" d="M600 1100q92 0 175 -10.5t141.5 -27t108.5 -36.5t81.5 -40t53.5 -37t31 -27l9 -10v-200q0 -21 -14.5 -33t-34.5 -9l-202 34q-20 3 -34.5 20t-14.5 38v146q-141 24 -300 24t-300 -24v-146q0 -21 -14.5 -38t-34.5 -20l-202 -34q-20 -3 -34.5 9t-14.5 33v200q3 4 9.5 10.5 t31 26t54 37.5t80.5 39.5t109 37.5t141 26.5t175 10.5zM600 795q56 0 97 -9.5t60 -23.5t30 -28t12 -24l1 -10v-50l365 -303q14 -15 24.5 -40t10.5 -45v-212q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v212q0 20 10.5 45t24.5 40l365 303v50 q0 4 1 10.5t12 23t30 29t60 22.5t97 10z" />
<glyph unicode="&#xe184;" d="M1100 700l-200 -200h-600l-200 200v500h200v-200h200v200h200v-200h200v200h200v-500zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-12l137 -100h-950l137 100h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5 t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe185;" d="M700 1100h-100q-41 0 -70.5 -29.5t-29.5 -70.5v-1000h300v1000q0 41 -29.5 70.5t-70.5 29.5zM1100 800h-100q-41 0 -70.5 -29.5t-29.5 -70.5v-700h300v700q0 41 -29.5 70.5t-70.5 29.5zM400 0h-300v400q0 41 29.5 70.5t70.5 29.5h100q41 0 70.5 -29.5t29.5 -70.5v-400z " />
<glyph unicode="&#xe186;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 700h-200v-100h200v-300h-300v100h200v100h-200v300h300v-100zM900 700v-300l-100 -100h-200v500h200z M700 700v-300h100v300h-100z" />
<glyph unicode="&#xe187;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 300h-100v200h-100v-200h-100v500h100v-200h100v200h100v-500zM900 700v-300l-100 -100h-200v500h200z M700 700v-300h100v300h-100z" />
<glyph unicode="&#xe188;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 700h-200v-300h200v-100h-300v500h300v-100zM900 700h-200v-300h200v-100h-300v500h300v-100z" />
<glyph unicode="&#xe189;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 400l-300 150l300 150v-300zM900 550l-300 -150v300z" />
<glyph unicode="&#xe190;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM900 300h-700v500h700v-500zM800 700h-130q-38 0 -66.5 -43t-28.5 -108t27 -107t68 -42h130v300zM300 700v-300 h130q41 0 68 42t27 107t-28.5 108t-66.5 43h-130z" />
<glyph unicode="&#xe191;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 700h-200v-100h200v-300h-300v100h200v100h-200v300h300v-100zM900 300h-100v400h-100v100h200v-500z M700 300h-100v100h100v-100z" />
<glyph unicode="&#xe192;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM300 700h200v-400h-300v500h100v-100zM900 300h-100v400h-100v100h200v-500zM300 600v-200h100v200h-100z M700 300h-100v100h100v-100z" />
<glyph unicode="&#xe193;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 500l-199 -200h-100v50l199 200v150h-200v100h300v-300zM900 300h-100v400h-100v100h200v-500zM701 300h-100 v100h100v-100z" />
<glyph unicode="&#xe194;" d="M600 1191q120 0 229.5 -47t188.5 -126t126 -188.5t47 -229.5t-47 -229.5t-126 -188.5t-188.5 -126t-229.5 -47t-229.5 47t-188.5 126t-126 188.5t-47 229.5t47 229.5t126 188.5t188.5 126t229.5 47zM600 1021q-114 0 -211 -56.5t-153.5 -153.5t-56.5 -211t56.5 -211 t153.5 -153.5t211 -56.5t211 56.5t153.5 153.5t56.5 211t-56.5 211t-153.5 153.5t-211 56.5zM800 700h-300v-200h300v-100h-300l-100 100v200l100 100h300v-100z" />
<glyph unicode="&#xe195;" d="M600 1191q120 0 229.5 -47t188.5 -126t126 -188.5t47 -229.5t-47 -229.5t-126 -188.5t-188.5 -126t-229.5 -47t-229.5 47t-188.5 126t-126 188.5t-47 229.5t47 229.5t126 188.5t188.5 126t229.5 47zM600 1021q-114 0 -211 -56.5t-153.5 -153.5t-56.5 -211t56.5 -211 t153.5 -153.5t211 -56.5t211 56.5t153.5 153.5t56.5 211t-56.5 211t-153.5 153.5t-211 56.5zM800 700v-100l-50 -50l100 -100v-50h-100l-100 100h-150v-100h-100v400h300zM500 700v-100h200v100h-200z" />
<glyph unicode="&#xe197;" d="M503 1089q110 0 200.5 -59.5t134.5 -156.5q44 14 90 14q120 0 205 -86.5t85 -207t-85 -207t-205 -86.5h-128v250q0 21 -14.5 35.5t-35.5 14.5h-300q-21 0 -35.5 -14.5t-14.5 -35.5v-250h-222q-80 0 -136 57.5t-56 136.5q0 69 43 122.5t108 67.5q-2 19 -2 37q0 100 49 185 t134 134t185 49zM525 500h150q10 0 17.5 -7.5t7.5 -17.5v-275h137q21 0 26 -11.5t-8 -27.5l-223 -244q-13 -16 -32 -16t-32 16l-223 244q-13 16 -8 27.5t26 11.5h137v275q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe198;" d="M502 1089q110 0 201 -59.5t135 -156.5q43 15 89 15q121 0 206 -86.5t86 -206.5q0 -99 -60 -181t-150 -110l-378 360q-13 16 -31.5 16t-31.5 -16l-381 -365h-9q-79 0 -135.5 57.5t-56.5 136.5q0 69 43 122.5t108 67.5q-2 19 -2 38q0 100 49 184.5t133.5 134t184.5 49.5z M632 467l223 -228q13 -16 8 -27.5t-26 -11.5h-137v-275q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v275h-137q-21 0 -26 11.5t8 27.5q199 204 223 228q19 19 31.5 19t32.5 -19z" />
<glyph unicode="&#xe199;" d="M700 100v100h400l-270 300h170l-270 300h170l-300 333l-300 -333h170l-270 -300h170l-270 -300h400v-100h-50q-21 0 -35.5 -14.5t-14.5 -35.5v-50h400v50q0 21 -14.5 35.5t-35.5 14.5h-50z" />
<glyph unicode="&#xe200;" d="M600 1179q94 0 167.5 -56.5t99.5 -145.5q89 -6 150.5 -71.5t61.5 -155.5q0 -61 -29.5 -112.5t-79.5 -82.5q9 -29 9 -55q0 -74 -52.5 -126.5t-126.5 -52.5q-55 0 -100 30v-251q21 0 35.5 -14.5t14.5 -35.5v-50h-300v50q0 21 14.5 35.5t35.5 14.5v251q-45 -30 -100 -30 q-74 0 -126.5 52.5t-52.5 126.5q0 18 4 38q-47 21 -75.5 65t-28.5 97q0 74 52.5 126.5t126.5 52.5q5 0 23 -2q0 2 -1 10t-1 13q0 116 81.5 197.5t197.5 81.5z" />
<glyph unicode="&#xe201;" d="M1010 1010q111 -111 150.5 -260.5t0 -299t-150.5 -260.5q-83 -83 -191.5 -126.5t-218.5 -43.5t-218.5 43.5t-191.5 126.5q-111 111 -150.5 260.5t0 299t150.5 260.5q83 83 191.5 126.5t218.5 43.5t218.5 -43.5t191.5 -126.5zM476 1065q-4 0 -8 -1q-121 -34 -209.5 -122.5 t-122.5 -209.5q-4 -12 2.5 -23t18.5 -14l36 -9q3 -1 7 -1q23 0 29 22q27 96 98 166q70 71 166 98q11 3 17.5 13.5t3.5 22.5l-9 35q-3 13 -14 19q-7 4 -15 4zM512 920q-4 0 -9 -2q-80 -24 -138.5 -82.5t-82.5 -138.5q-4 -13 2 -24t19 -14l34 -9q4 -1 8 -1q22 0 28 21 q18 58 58.5 98.5t97.5 58.5q12 3 18 13.5t3 21.5l-9 35q-3 12 -14 19q-7 4 -15 4zM719.5 719.5q-49.5 49.5 -119.5 49.5t-119.5 -49.5t-49.5 -119.5t49.5 -119.5t119.5 -49.5t119.5 49.5t49.5 119.5t-49.5 119.5zM855 551q-22 0 -28 -21q-18 -58 -58.5 -98.5t-98.5 -57.5 q-11 -4 -17 -14.5t-3 -21.5l9 -35q3 -12 14 -19q7 -4 15 -4q4 0 9 2q80 24 138.5 82.5t82.5 138.5q4 13 -2.5 24t-18.5 14l-34 9q-4 1 -8 1zM1000 515q-23 0 -29 -22q-27 -96 -98 -166q-70 -71 -166 -98q-11 -3 -17.5 -13.5t-3.5 -22.5l9 -35q3 -13 14 -19q7 -4 15 -4 q4 0 8 1q121 34 209.5 122.5t122.5 209.5q4 12 -2.5 23t-18.5 14l-36 9q-3 1 -7 1z" />
<glyph unicode="&#xe202;" d="M700 800h300v-380h-180v200h-340v-200h-380v755q0 10 7.5 17.5t17.5 7.5h575v-400zM1000 900h-200v200zM700 300h162l-212 -212l-212 212h162v200h100v-200zM520 0h-395q-10 0 -17.5 7.5t-7.5 17.5v395zM1000 220v-195q0 -10 -7.5 -17.5t-17.5 -7.5h-195z" />
<glyph unicode="&#xe203;" d="M700 800h300v-520l-350 350l-550 -550v1095q0 10 7.5 17.5t17.5 7.5h575v-400zM1000 900h-200v200zM862 200h-162v-200h-100v200h-162l212 212zM480 0h-355q-10 0 -17.5 7.5t-7.5 17.5v55h380v-80zM1000 80v-55q0 -10 -7.5 -17.5t-17.5 -7.5h-155v80h180z" />
<glyph unicode="&#xe204;" d="M1162 800h-162v-200h100l100 -100h-300v300h-162l212 212zM200 800h200q27 0 40 -2t29.5 -10.5t23.5 -30t7 -57.5h300v-100h-600l-200 -350v450h100q0 36 7 57.5t23.5 30t29.5 10.5t40 2zM800 400h240l-240 -400h-800l300 500h500v-100z" />
<glyph unicode="&#xe205;" d="M650 1100h100q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h50v50q0 21 14.5 35.5t35.5 14.5zM1000 850v150q41 0 70.5 -29.5t29.5 -70.5v-800 q0 -41 -29.5 -70.5t-70.5 -29.5h-600q-1 0 -20 4l246 246l-326 326v324q0 41 29.5 70.5t70.5 29.5v-150q0 -62 44 -106t106 -44h300q62 0 106 44t44 106zM412 250l-212 -212v162h-200v100h200v162z" />
<glyph unicode="&#xe206;" d="M450 1100h100q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h50v50q0 21 14.5 35.5t35.5 14.5zM800 850v150q41 0 70.5 -29.5t29.5 -70.5v-500 h-200v-300h200q0 -36 -7 -57.5t-23.5 -30t-29.5 -10.5t-40 -2h-600q-41 0 -70.5 29.5t-29.5 70.5v800q0 41 29.5 70.5t70.5 29.5v-150q0 -62 44 -106t106 -44h300q62 0 106 44t44 106zM1212 250l-212 -212v162h-200v100h200v162z" />
<glyph unicode="&#xe209;" d="M658 1197l637 -1104q23 -38 7 -65.5t-60 -27.5h-1276q-44 0 -60 27.5t7 65.5l637 1104q22 39 54 39t54 -39zM704 800h-208q-20 0 -32 -14.5t-8 -34.5l58 -302q4 -20 21.5 -34.5t37.5 -14.5h54q20 0 37.5 14.5t21.5 34.5l58 302q4 20 -8 34.5t-32 14.5zM500 300v-100h200 v100h-200z" />
<glyph unicode="&#xe210;" d="M425 1100h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM425 800h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5 t17.5 7.5zM825 800h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM25 500h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150 q0 10 7.5 17.5t17.5 7.5zM425 500h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM825 500h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5 v150q0 10 7.5 17.5t17.5 7.5zM25 200h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM425 200h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5 t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM825 200h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe211;" d="M700 1200h100v-200h-100v-100h350q62 0 86.5 -39.5t-3.5 -94.5l-66 -132q-41 -83 -81 -134h-772q-40 51 -81 134l-66 132q-28 55 -3.5 94.5t86.5 39.5h350v100h-100v200h100v100h200v-100zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-12l137 -100 h-950l138 100h-13q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe212;" d="M600 1300q40 0 68.5 -29.5t28.5 -70.5h-194q0 41 28.5 70.5t68.5 29.5zM443 1100h314q18 -37 18 -75q0 -8 -3 -25h328q41 0 44.5 -16.5t-30.5 -38.5l-175 -145h-678l-178 145q-34 22 -29 38.5t46 16.5h328q-3 17 -3 25q0 38 18 75zM250 700h700q21 0 35.5 -14.5 t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-150v-200l275 -200h-950l275 200v200h-150q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe213;" d="M600 1181q75 0 128 -53t53 -128t-53 -128t-128 -53t-128 53t-53 128t53 128t128 53zM602 798h46q34 0 55.5 -28.5t21.5 -86.5q0 -76 39 -183h-324q39 107 39 183q0 58 21.5 86.5t56.5 28.5h45zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-13 l138 -100h-950l137 100h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe214;" d="M600 1300q47 0 92.5 -53.5t71 -123t25.5 -123.5q0 -78 -55.5 -133.5t-133.5 -55.5t-133.5 55.5t-55.5 133.5q0 62 34 143l144 -143l111 111l-163 163q34 26 63 26zM602 798h46q34 0 55.5 -28.5t21.5 -86.5q0 -76 39 -183h-324q39 107 39 183q0 58 21.5 86.5t56.5 28.5h45 zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-13l138 -100h-950l137 100h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe215;" d="M600 1200l300 -161v-139h-300q0 -57 18.5 -108t50 -91.5t63 -72t70 -67.5t57.5 -61h-530q-60 83 -90.5 177.5t-30.5 178.5t33 164.5t87.5 139.5t126 96.5t145.5 41.5v-98zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-13l138 -100h-950l137 100 h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe216;" d="M600 1300q41 0 70.5 -29.5t29.5 -70.5v-78q46 -26 73 -72t27 -100v-50h-400v50q0 54 27 100t73 72v78q0 41 29.5 70.5t70.5 29.5zM400 800h400q54 0 100 -27t72 -73h-172v-100h200v-100h-200v-100h200v-100h-200v-100h200q0 -83 -58.5 -141.5t-141.5 -58.5h-400 q-83 0 -141.5 58.5t-58.5 141.5v400q0 83 58.5 141.5t141.5 58.5z" />
<glyph unicode="&#xe218;" d="M150 1100h900q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v500q0 21 14.5 35.5t35.5 14.5zM125 400h950q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-283l224 -224q13 -13 13 -31.5t-13 -32 t-31.5 -13.5t-31.5 13l-88 88h-524l-87 -88q-13 -13 -32 -13t-32 13.5t-13 32t13 31.5l224 224h-289q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM541 300l-100 -100h324l-100 100h-124z" />
<glyph unicode="&#xe219;" d="M200 1100h800q83 0 141.5 -58.5t58.5 -141.5v-200h-100q0 41 -29.5 70.5t-70.5 29.5h-250q-41 0 -70.5 -29.5t-29.5 -70.5h-100q0 41 -29.5 70.5t-70.5 29.5h-250q-41 0 -70.5 -29.5t-29.5 -70.5h-100v200q0 83 58.5 141.5t141.5 58.5zM100 600h1000q41 0 70.5 -29.5 t29.5 -70.5v-300h-1200v300q0 41 29.5 70.5t70.5 29.5zM300 100v-50q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v50h200zM1100 100v-50q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v50h200z" />
<glyph unicode="&#xe221;" d="M480 1165l682 -683q31 -31 31 -75.5t-31 -75.5l-131 -131h-481l-517 518q-32 31 -32 75.5t32 75.5l295 296q31 31 75.5 31t76.5 -31zM108 794l342 -342l303 304l-341 341zM250 100h800q21 0 35.5 -14.5t14.5 -35.5v-50h-900v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe223;" d="M1057 647l-189 506q-8 19 -27.5 33t-40.5 14h-400q-21 0 -40.5 -14t-27.5 -33l-189 -506q-8 -19 1.5 -33t30.5 -14h625v-150q0 -21 14.5 -35.5t35.5 -14.5t35.5 14.5t14.5 35.5v150h125q21 0 30.5 14t1.5 33zM897 0h-595v50q0 21 14.5 35.5t35.5 14.5h50v50 q0 21 14.5 35.5t35.5 14.5h48v300h200v-300h47q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-50z" />
<glyph unicode="&#xe224;" d="M900 800h300v-575q0 -10 -7.5 -17.5t-17.5 -7.5h-375v591l-300 300v84q0 10 7.5 17.5t17.5 7.5h375v-400zM1200 900h-200v200zM400 600h300v-575q0 -10 -7.5 -17.5t-17.5 -7.5h-650q-10 0 -17.5 7.5t-7.5 17.5v950q0 10 7.5 17.5t17.5 7.5h375v-400zM700 700h-200v200z " />
<glyph unicode="&#xe225;" d="M484 1095h195q75 0 146 -32.5t124 -86t89.5 -122.5t48.5 -142q18 -14 35 -20q31 -10 64.5 6.5t43.5 48.5q10 34 -15 71q-19 27 -9 43q5 8 12.5 11t19 -1t23.5 -16q41 -44 39 -105q-3 -63 -46 -106.5t-104 -43.5h-62q-7 -55 -35 -117t-56 -100l-39 -234q-3 -20 -20 -34.5 t-38 -14.5h-100q-21 0 -33 14.5t-9 34.5l12 70q-49 -14 -91 -14h-195q-24 0 -65 8l-11 -64q-3 -20 -20 -34.5t-38 -14.5h-100q-21 0 -33 14.5t-9 34.5l26 157q-84 74 -128 175l-159 53q-19 7 -33 26t-14 40v50q0 21 14.5 35.5t35.5 14.5h124q11 87 56 166l-111 95 q-16 14 -12.5 23.5t24.5 9.5h203q116 101 250 101zM675 1000h-250q-10 0 -17.5 -7.5t-7.5 -17.5v-50q0 -10 7.5 -17.5t17.5 -7.5h250q10 0 17.5 7.5t7.5 17.5v50q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe226;" d="M641 900l423 247q19 8 42 2.5t37 -21.5l32 -38q14 -15 12.5 -36t-17.5 -34l-139 -120h-390zM50 1100h106q67 0 103 -17t66 -71l102 -212h823q21 0 35.5 -14.5t14.5 -35.5v-50q0 -21 -14 -40t-33 -26l-737 -132q-23 -4 -40 6t-26 25q-42 67 -100 67h-300q-62 0 -106 44 t-44 106v200q0 62 44 106t106 44zM173 928h-80q-19 0 -28 -14t-9 -35v-56q0 -51 42 -51h134q16 0 21.5 8t5.5 24q0 11 -16 45t-27 51q-18 28 -43 28zM550 727q-32 0 -54.5 -22.5t-22.5 -54.5t22.5 -54.5t54.5 -22.5t54.5 22.5t22.5 54.5t-22.5 54.5t-54.5 22.5zM130 389 l152 130q18 19 34 24t31 -3.5t24.5 -17.5t25.5 -28q28 -35 50.5 -51t48.5 -13l63 5l48 -179q13 -61 -3.5 -97.5t-67.5 -79.5l-80 -69q-47 -40 -109 -35.5t-103 51.5l-130 151q-40 47 -35.5 109.5t51.5 102.5zM380 377l-102 -88q-31 -27 2 -65l37 -43q13 -15 27.5 -19.5 t31.5 6.5l61 53q19 16 14 49q-2 20 -12 56t-17 45q-11 12 -19 14t-23 -8z" />
<glyph unicode="&#xe227;" d="M625 1200h150q10 0 17.5 -7.5t7.5 -17.5v-109q79 -33 131 -87.5t53 -128.5q1 -46 -15 -84.5t-39 -61t-46 -38t-39 -21.5l-17 -6q6 0 15 -1.5t35 -9t50 -17.5t53 -30t50 -45t35.5 -64t14.5 -84q0 -59 -11.5 -105.5t-28.5 -76.5t-44 -51t-49.5 -31.5t-54.5 -16t-49.5 -6.5 t-43.5 -1v-75q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v75h-100v-75q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v75h-175q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h75v600h-75q-10 0 -17.5 7.5t-7.5 17.5v150 q0 10 7.5 17.5t17.5 7.5h175v75q0 10 7.5 17.5t17.5 7.5h150q10 0 17.5 -7.5t7.5 -17.5v-75h100v75q0 10 7.5 17.5t17.5 7.5zM400 900v-200h263q28 0 48.5 10.5t30 25t15 29t5.5 25.5l1 10q0 4 -0.5 11t-6 24t-15 30t-30 24t-48.5 11h-263zM400 500v-200h363q28 0 48.5 10.5 t30 25t15 29t5.5 25.5l1 10q0 4 -0.5 11t-6 24t-15 30t-30 24t-48.5 11h-363z" />
<glyph unicode="&#xe230;" d="M212 1198h780q86 0 147 -61t61 -147v-416q0 -51 -18 -142.5t-36 -157.5l-18 -66q-29 -87 -93.5 -146.5t-146.5 -59.5h-572q-82 0 -147 59t-93 147q-8 28 -20 73t-32 143.5t-20 149.5v416q0 86 61 147t147 61zM600 1045q-70 0 -132.5 -11.5t-105.5 -30.5t-78.5 -41.5 t-57 -45t-36 -41t-20.5 -30.5l-6 -12l156 -243h560l156 243q-2 5 -6 12.5t-20 29.5t-36.5 42t-57 44.5t-79 42t-105 29.5t-132.5 12zM762 703h-157l195 261z" />
<glyph unicode="&#xe231;" d="M475 1300h150q103 0 189 -86t86 -189v-500q0 -41 -42 -83t-83 -42h-450q-41 0 -83 42t-42 83v500q0 103 86 189t189 86zM700 300v-225q0 -21 -27 -48t-48 -27h-150q-21 0 -48 27t-27 48v225h300z" />
<glyph unicode="&#xe232;" d="M475 1300h96q0 -150 89.5 -239.5t239.5 -89.5v-446q0 -41 -42 -83t-83 -42h-450q-41 0 -83 42t-42 83v500q0 103 86 189t189 86zM700 300v-225q0 -21 -27 -48t-48 -27h-150q-21 0 -48 27t-27 48v225h300z" />
<glyph unicode="&#xe233;" d="M1294 767l-638 -283l-378 170l-78 -60v-224l100 -150v-199l-150 148l-150 -149v200l100 150v250q0 4 -0.5 10.5t0 9.5t1 8t3 8t6.5 6l47 40l-147 65l642 283zM1000 380l-350 -166l-350 166v147l350 -165l350 165v-147z" />
<glyph unicode="&#xe234;" d="M250 800q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM650 800q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM1050 800q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44z" />
<glyph unicode="&#xe235;" d="M550 1100q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM550 700q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM550 300q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44z" />
<glyph unicode="&#xe236;" d="M125 1100h950q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-950q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM125 700h950q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-950q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5 t17.5 7.5zM125 300h950q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-950q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe237;" d="M350 1200h500q162 0 256 -93.5t94 -256.5v-500q0 -165 -93.5 -257.5t-256.5 -92.5h-500q-165 0 -257.5 92.5t-92.5 257.5v500q0 165 92.5 257.5t257.5 92.5zM900 1000h-600q-41 0 -70.5 -29.5t-29.5 -70.5v-600q0 -41 29.5 -70.5t70.5 -29.5h600q41 0 70.5 29.5 t29.5 70.5v600q0 41 -29.5 70.5t-70.5 29.5zM350 900h500q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -14.5 -35.5t-35.5 -14.5h-500q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 14.5 35.5t35.5 14.5zM400 800v-200h400v200h-400z" />
<glyph unicode="&#xe238;" d="M150 1100h1000q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-200h50q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-200h50q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-200h50q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5 t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5h50v200h-50q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5h50v200h-50q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5h50v200h-50q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe239;" d="M650 1187q87 -67 118.5 -156t0 -178t-118.5 -155q-87 66 -118.5 155t0 178t118.5 156zM300 800q124 0 212 -88t88 -212q-124 0 -212 88t-88 212zM1000 800q0 -124 -88 -212t-212 -88q0 124 88 212t212 88zM300 500q124 0 212 -88t88 -212q-124 0 -212 88t-88 212z M1000 500q0 -124 -88 -212t-212 -88q0 124 88 212t212 88zM700 199v-144q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v142q40 -4 43 -4q17 0 57 6z" />
<glyph unicode="&#xe240;" d="M745 878l69 19q25 6 45 -12l298 -295q11 -11 15 -26.5t-2 -30.5q-5 -14 -18 -23.5t-28 -9.5h-8q1 0 1 -13q0 -29 -2 -56t-8.5 -62t-20 -63t-33 -53t-51 -39t-72.5 -14h-146q-184 0 -184 288q0 24 10 47q-20 4 -62 4t-63 -4q11 -24 11 -47q0 -288 -184 -288h-142 q-48 0 -84.5 21t-56 51t-32 71.5t-16 75t-3.5 68.5q0 13 2 13h-7q-15 0 -27.5 9.5t-18.5 23.5q-6 15 -2 30.5t15 25.5l298 296q20 18 46 11l76 -19q20 -5 30.5 -22.5t5.5 -37.5t-22.5 -31t-37.5 -5l-51 12l-182 -193h891l-182 193l-44 -12q-20 -5 -37.5 6t-22.5 31t6 37.5 t31 22.5z" />
<glyph unicode="&#xe241;" d="M1200 900h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-200v-850q0 -22 25 -34.5t50 -13.5l25 -2v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v850h-200q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h1000v-300zM500 450h-25q0 15 -4 24.5t-9 14.5t-17 7.5t-20 3t-25 0.5h-100v-425q0 -11 12.5 -17.5t25.5 -7.5h12v-50h-200v50q50 0 50 25v425h-100q-17 0 -25 -0.5t-20 -3t-17 -7.5t-9 -14.5t-4 -24.5h-25v150h500v-150z" />
<glyph unicode="&#xe242;" d="M1000 300v50q-25 0 -55 32q-14 14 -25 31t-16 27l-4 11l-289 747h-69l-300 -754q-18 -35 -39 -56q-9 -9 -24.5 -18.5t-26.5 -14.5l-11 -5v-50h273v50q-49 0 -78.5 21.5t-11.5 67.5l69 176h293l61 -166q13 -34 -3.5 -66.5t-55.5 -32.5v-50h312zM412 691l134 342l121 -342 h-255zM1100 150v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h1000q21 0 35.5 -14.5t14.5 -35.5z" />
<glyph unicode="&#xe243;" d="M50 1200h1100q21 0 35.5 -14.5t14.5 -35.5v-1100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v1100q0 21 14.5 35.5t35.5 14.5zM611 1118h-70q-13 0 -18 -12l-299 -753q-17 -32 -35 -51q-18 -18 -56 -34q-12 -5 -12 -18v-50q0 -8 5.5 -14t14.5 -6 h273q8 0 14 6t6 14v50q0 8 -6 14t-14 6q-55 0 -71 23q-10 14 0 39l63 163h266l57 -153q11 -31 -6 -55q-12 -17 -36 -17q-8 0 -14 -6t-6 -14v-50q0 -8 6 -14t14 -6h313q8 0 14 6t6 14v50q0 7 -5.5 13t-13.5 7q-17 0 -42 25q-25 27 -40 63h-1l-288 748q-5 12 -19 12zM639 611 h-197l103 264z" />
<glyph unicode="&#xe244;" d="M1200 1100h-1200v100h1200v-100zM50 1000h400q21 0 35.5 -14.5t14.5 -35.5v-900q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v900q0 21 14.5 35.5t35.5 14.5zM650 1000h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400 q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM700 900v-300h300v300h-300z" />
<glyph unicode="&#xe245;" d="M50 1200h400q21 0 35.5 -14.5t14.5 -35.5v-900q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v900q0 21 14.5 35.5t35.5 14.5zM650 700h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400 q0 21 14.5 35.5t35.5 14.5zM700 600v-300h300v300h-300zM1200 0h-1200v100h1200v-100z" />
<glyph unicode="&#xe246;" d="M50 1000h400q21 0 35.5 -14.5t14.5 -35.5v-350h100v150q0 21 14.5 35.5t35.5 14.5h400q21 0 35.5 -14.5t14.5 -35.5v-150h100v-100h-100v-150q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v150h-100v-350q0 -21 -14.5 -35.5t-35.5 -14.5h-400 q-21 0 -35.5 14.5t-14.5 35.5v800q0 21 14.5 35.5t35.5 14.5zM700 700v-300h300v300h-300z" />
<glyph unicode="&#xe247;" d="M100 0h-100v1200h100v-1200zM250 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM300 1000v-300h300v300h-300zM250 500h900q21 0 35.5 -14.5t14.5 -35.5v-400 q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe248;" d="M600 1100h150q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-150v-100h450q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5h350v100h-150q-21 0 -35.5 14.5 t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5h150v100h100v-100zM400 1000v-300h300v300h-300z" />
<glyph unicode="&#xe249;" d="M1200 0h-100v1200h100v-1200zM550 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM600 1000v-300h300v300h-300zM50 500h900q21 0 35.5 -14.5t14.5 -35.5v-400 q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe250;" d="M865 565l-494 -494q-23 -23 -41 -23q-14 0 -22 13.5t-8 38.5v1000q0 25 8 38.5t22 13.5q18 0 41 -23l494 -494q14 -14 14 -35t-14 -35z" />
<glyph unicode="&#xe251;" d="M335 635l494 494q29 29 50 20.5t21 -49.5v-1000q0 -41 -21 -49.5t-50 20.5l-494 494q-14 14 -14 35t14 35z" />
<glyph unicode="&#xe252;" d="M100 900h1000q41 0 49.5 -21t-20.5 -50l-494 -494q-14 -14 -35 -14t-35 14l-494 494q-29 29 -20.5 50t49.5 21z" />
<glyph unicode="&#xe253;" d="M635 865l494 -494q29 -29 20.5 -50t-49.5 -21h-1000q-41 0 -49.5 21t20.5 50l494 494q14 14 35 14t35 -14z" />
<glyph unicode="&#xe254;" d="M700 741v-182l-692 -323v221l413 193l-413 193v221zM1200 0h-800v200h800v-200z" />
<glyph unicode="&#xe255;" d="M1200 900h-200v-100h200v-100h-300v300h200v100h-200v100h300v-300zM0 700h50q0 21 4 37t9.5 26.5t18 17.5t22 11t28.5 5.5t31 2t37 0.5h100v-550q0 -22 -25 -34.5t-50 -13.5l-25 -2v-100h400v100q-4 0 -11 0.5t-24 3t-30 7t-24 15t-11 24.5v550h100q25 0 37 -0.5t31 -2 t28.5 -5.5t22 -11t18 -17.5t9.5 -26.5t4 -37h50v300h-800v-300z" />
<glyph unicode="&#xe256;" d="M800 700h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-100v-550q0 -22 25 -34.5t50 -14.5l25 -1v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v550h-100q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h800v-300zM1100 200h-200v-100h200v-100h-300v300h200v100h-200v100h300v-300z" />
<glyph unicode="&#xe257;" d="M701 1098h160q16 0 21 -11t-7 -23l-464 -464l464 -464q12 -12 7 -23t-21 -11h-160q-13 0 -23 9l-471 471q-7 8 -7 18t7 18l471 471q10 9 23 9z" />
<glyph unicode="&#xe258;" d="M339 1098h160q13 0 23 -9l471 -471q7 -8 7 -18t-7 -18l-471 -471q-10 -9 -23 -9h-160q-16 0 -21 11t7 23l464 464l-464 464q-12 12 -7 23t21 11z" />
<glyph unicode="&#xe259;" d="M1087 882q11 -5 11 -21v-160q0 -13 -9 -23l-471 -471q-8 -7 -18 -7t-18 7l-471 471q-9 10 -9 23v160q0 16 11 21t23 -7l464 -464l464 464q12 12 23 7z" />
<glyph unicode="&#xe260;" d="M618 993l471 -471q9 -10 9 -23v-160q0 -16 -11 -21t-23 7l-464 464l-464 -464q-12 -12 -23 -7t-11 21v160q0 13 9 23l471 471q8 7 18 7t18 -7z" />
<glyph unicode="&#xf8ff;" d="M1000 1200q0 -124 -88 -212t-212 -88q0 124 88 212t212 88zM450 1000h100q21 0 40 -14t26 -33l79 -194q5 1 16 3q34 6 54 9.5t60 7t65.5 1t61 -10t56.5 -23t42.5 -42t29 -64t5 -92t-19.5 -121.5q-1 -7 -3 -19.5t-11 -50t-20.5 -73t-32.5 -81.5t-46.5 -83t-64 -70 t-82.5 -50q-13 -5 -42 -5t-65.5 2.5t-47.5 2.5q-14 0 -49.5 -3.5t-63 -3.5t-43.5 7q-57 25 -104.5 78.5t-75 111.5t-46.5 112t-26 90l-7 35q-15 63 -18 115t4.5 88.5t26 64t39.5 43.5t52 25.5t58.5 13t62.5 2t59.5 -4.5t55.5 -8l-147 192q-12 18 -5.5 30t27.5 12z" />
<glyph unicode="&#x1f511;" d="M250 1200h600q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-150v-500l-255 -178q-19 -9 -32 -1t-13 29v650h-150q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM400 1100v-100h300v100h-300z" />
<glyph unicode="&#x1f6aa;" d="M250 1200h750q39 0 69.5 -40.5t30.5 -84.5v-933l-700 -117v950l600 125h-700v-1000h-100v1025q0 23 15.5 49t34.5 26zM500 525v-100l100 20v100z" />
</font>
</defs></svg> ) format('svg')}.glyphicon{position:relative;top:1px;display:inline-block;font-family:'Glyphicons Halflings';font-style:normal;font-weight:400;line-height:1;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}.glyphicon-asterisk:before{content:"\2a"}.glyphicon-plus:before{content:"\2b"}.glyphicon-eur:before,.glyphicon-euro:before{content:"\20ac"}.glyphicon-minus:before{content:"\2212"}.glyphicon-cloud:before{content:"\2601"}.glyphicon-envelope:before{content:"\2709"}.glyphicon-pencil:before{content:"\270f"}.glyphicon-glass:before{content:"\e001"}.glyphicon-music:before{content:"\e002"}.glyphicon-search:before{content:"\e003"}.glyphicon-heart:before{content:"\e005"}.glyphicon-star:before{content:"\e006"}.glyphicon-star-empty:before{content:"\e007"}.glyphicon-user:before{content:"\e008"}.glyphicon-film:before{content:"\e009"}.glyphicon-th-large:before{content:"\e010"}.glyphicon-th:before{content:"\e011"}.glyphicon-th-list:before{content:"\e012"}.glyphicon-ok:before{content:"\e013"}.glyphicon-remove:before{content:"\e014"}.glyphicon-zoom-in:before{content:"\e015"}.glyphicon-zoom-out:before{content:"\e016"}.glyphicon-off:before{content:"\e017"}.glyphicon-signal:before{content:"\e018"}.glyphicon-cog:before{content:"\e019"}.glyphicon-trash:before{content:"\e020"}.glyphicon-home:before{content:"\e021"}.glyphicon-file:before{content:"\e022"}.glyphicon-time:before{content:"\e023"}.glyphicon-road:before{content:"\e024"}.glyphicon-download-alt:before{content:"\e025"}.glyphicon-download:before{content:"\e026"}.glyphicon-upload:before{content:"\e027"}.glyphicon-inbox:before{content:"\e028"}.glyphicon-play-circle:before{content:"\e029"}.glyphicon-repeat:before{content:"\e030"}.glyphicon-refresh:before{content:"\e031"}.glyphicon-list-alt:before{content:"\e032"}.glyphicon-lock:before{content:"\e033"}.glyphicon-flag:before{content:"\e034"}.glyphicon-headphones:before{content:"\e035"}.glyphicon-volume-off:before{content:"\e036"}.glyphicon-volume-down:before{content:"\e037"}.glyphicon-volume-up:before{content:"\e038"}.glyphicon-qrcode:before{content:"\e039"}.glyphicon-barcode:before{content:"\e040"}.glyphicon-tag:before{content:"\e041"}.glyphicon-tags:before{content:"\e042"}.glyphicon-book:before{content:"\e043"}.glyphicon-bookmark:before{content:"\e044"}.glyphicon-print:before{content:"\e045"}.glyphicon-camera:before{content:"\e046"}.glyphicon-font:before{content:"\e047"}.glyphicon-bold:before{content:"\e048"}.glyphicon-italic:before{content:"\e049"}.glyphicon-text-height:before{content:"\e050"}.glyphicon-text-width:before{content:"\e051"}.glyphicon-align-left:before{content:"\e052"}.glyphicon-align-center:before{content:"\e053"}.glyphicon-align-right:before{content:"\e054"}.glyphicon-align-justify:before{content:"\e055"}.glyphicon-list:before{content:"\e056"}.glyphicon-indent-left:before{content:"\e057"}.glyphicon-indent-right:before{content:"\e058"}.glyphicon-facetime-video:before{content:"\e059"}.glyphicon-picture:before{content:"\e060"}.glyphicon-map-marker:before{content:"\e062"}.glyphicon-adjust:before{content:"\e063"}.glyphicon-tint:before{content:"\e064"}.glyphicon-edit:before{content:"\e065"}.glyphicon-share:before{content:"\e066"}.glyphicon-check:before{content:"\e067"}.glyphicon-move:before{content:"\e068"}.glyphicon-step-backward:before{content:"\e069"}.glyphicon-fast-backward:before{content:"\e070"}.glyphicon-backward:before{content:"\e071"}.glyphicon-play:before{content:"\e072"}.glyphicon-pause:before{content:"\e073"}.glyphicon-stop:before{content:"\e074"}.glyphicon-forward:before{content:"\e075"}.glyphicon-fast-forward:before{content:"\e076"}.glyphicon-step-forward:before{content:"\e077"}.glyphicon-eject:before{content:"\e078"}.glyphicon-chevron-left:before{content:"\e079"}.glyphicon-chevron-right:before{content:"\e080"}.glyphicon-plus-sign:before{content:"\e081"}.glyphicon-minus-sign:before{content:"\e082"}.glyphicon-remove-sign:before{content:"\e083"}.glyphicon-ok-sign:before{content:"\e084"}.glyphicon-question-sign:before{content:"\e085"}.glyphicon-info-sign:before{content:"\e086"}.glyphicon-screenshot:before{content:"\e087"}.glyphicon-remove-circle:before{content:"\e088"}.glyphicon-ok-circle:before{content:"\e089"}.glyphicon-ban-circle:before{content:"\e090"}.glyphicon-arrow-left:before{content:"\e091"}.glyphicon-arrow-right:before{content:"\e092"}.glyphicon-arrow-up:before{content:"\e093"}.glyphicon-arrow-down:before{content:"\e094"}.glyphicon-share-alt:before{content:"\e095"}.glyphicon-resize-full:before{content:"\e096"}.glyphicon-resize-small:before{content:"\e097"}.glyphicon-exclamation-sign:before{content:"\e101"}.glyphicon-gift:before{content:"\e102"}.glyphicon-leaf:before{content:"\e103"}.glyphicon-fire:before{content:"\e104"}.glyphicon-eye-open:before{content:"\e105"}.glyphicon-eye-close:before{content:"\e106"}.glyphicon-warning-sign:before{content:"\e107"}.glyphicon-plane:before{content:"\e108"}.glyphicon-calendar:before{content:"\e109"}.glyphicon-random:before{content:"\e110"}.glyphicon-comment:before{content:"\e111"}.glyphicon-magnet:before{content:"\e112"}.glyphicon-chevron-up:before{content:"\e113"}.glyphicon-chevron-down:before{content:"\e114"}.glyphicon-retweet:before{content:"\e115"}.glyphicon-shopping-cart:before{content:"\e116"}.glyphicon-folder-close:before{content:"\e117"}.glyphicon-folder-open:before{content:"\e118"}.glyphicon-resize-vertical:before{content:"\e119"}.glyphicon-resize-horizontal:before{content:"\e120"}.glyphicon-hdd:before{content:"\e121"}.glyphicon-bullhorn:before{content:"\e122"}.glyphicon-bell:before{content:"\e123"}.glyphicon-certificate:before{content:"\e124"}.glyphicon-thumbs-up:before{content:"\e125"}.glyphicon-thumbs-down:before{content:"\e126"}.glyphicon-hand-right:before{content:"\e127"}.glyphicon-hand-left:before{content:"\e128"}.glyphicon-hand-up:before{content:"\e129"}.glyphicon-hand-down:before{content:"\e130"}.glyphicon-circle-arrow-right:before{content:"\e131"}.glyphicon-circle-arrow-left:before{content:"\e132"}.glyphicon-circle-arrow-up:before{content:"\e133"}.glyphicon-circle-arrow-down:before{content:"\e134"}.glyphicon-globe:before{content:"\e135"}.glyphicon-wrench:before{content:"\e136"}.glyphicon-tasks:before{content:"\e137"}.glyphicon-filter:before{content:"\e138"}.glyphicon-briefcase:before{content:"\e139"}.glyphicon-fullscreen:before{content:"\e140"}.glyphicon-dashboard:before{content:"\e141"}.glyphicon-paperclip:before{content:"\e142"}.glyphicon-heart-empty:before{content:"\e143"}.glyphicon-link:before{content:"\e144"}.glyphicon-phone:before{content:"\e145"}.glyphicon-pushpin:before{content:"\e146"}.glyphicon-usd:before{content:"\e148"}.glyphicon-gbp:before{content:"\e149"}.glyphicon-sort:before{content:"\e150"}.glyphicon-sort-by-alphabet:before{content:"\e151"}.glyphicon-sort-by-alphabet-alt:before{content:"\e152"}.glyphicon-sort-by-order:before{content:"\e153"}.glyphicon-sort-by-order-alt:before{content:"\e154"}.glyphicon-sort-by-attributes:before{content:"\e155"}.glyphicon-sort-by-attributes-alt:before{content:"\e156"}.glyphicon-unchecked:before{content:"\e157"}.glyphicon-expand:before{content:"\e158"}.glyphicon-collapse-down:before{content:"\e159"}.glyphicon-collapse-up:before{content:"\e160"}.glyphicon-log-in:before{content:"\e161"}.glyphicon-flash:before{content:"\e162"}.glyphicon-log-out:before{content:"\e163"}.glyphicon-new-window:before{content:"\e164"}.glyphicon-record:before{content:"\e165"}.glyphicon-save:before{content:"\e166"}.glyphicon-open:before{content:"\e167"}.glyphicon-saved:before{content:"\e168"}.glyphicon-import:before{content:"\e169"}.glyphicon-export:before{content:"\e170"}.glyphicon-send:before{content:"\e171"}.glyphicon-floppy-disk:before{content:"\e172"}.glyphicon-floppy-saved:before{content:"\e173"}.glyphicon-floppy-remove:before{content:"\e174"}.glyphicon-floppy-save:before{content:"\e175"}.glyphicon-floppy-open:before{content:"\e176"}.glyphicon-credit-card:before{content:"\e177"}.glyphicon-transfer:before{content:"\e178"}.glyphicon-cutlery:before{content:"\e179"}.glyphicon-header:before{content:"\e180"}.glyphicon-compressed:before{content:"\e181"}.glyphicon-earphone:before{content:"\e182"}.glyphicon-phone-alt:before{content:"\e183"}.glyphicon-tower:before{content:"\e184"}.glyphicon-stats:before{content:"\e185"}.glyphicon-sd-video:before{content:"\e186"}.glyphicon-hd-video:before{content:"\e187"}.glyphicon-subtitles:before{content:"\e188"}.glyphicon-sound-stereo:before{content:"\e189"}.glyphicon-sound-dolby:before{content:"\e190"}.glyphicon-sound-5-1:before{content:"\e191"}.glyphicon-sound-6-1:before{content:"\e192"}.glyphicon-sound-7-1:before{content:"\e193"}.glyphicon-copyright-mark:before{content:"\e194"}.glyphicon-registration-mark:before{content:"\e195"}.glyphicon-cloud-download:before{content:"\e197"}.glyphicon-cloud-upload:before{content:"\e198"}.glyphicon-tree-conifer:before{content:"\e199"}.glyphicon-tree-deciduous:before{content:"\e200"}.glyphicon-cd:before{content:"\e201"}.glyphicon-save-file:before{content:"\e202"}.glyphicon-open-file:before{content:"\e203"}.glyphicon-level-up:before{content:"\e204"}.glyphicon-copy:before{content:"\e205"}.glyphicon-paste:before{content:"\e206"}.glyphicon-alert:before{content:"\e209"}.glyphicon-equalizer:before{content:"\e210"}.glyphicon-king:before{content:"\e211"}.glyphicon-queen:before{content:"\e212"}.glyphicon-pawn:before{content:"\e213"}.glyphicon-bishop:before{content:"\e214"}.glyphicon-knight:before{content:"\e215"}.glyphicon-baby-formula:before{content:"\e216"}.glyphicon-tent:before{content:"\26fa"}.glyphicon-blackboard:before{content:"\e218"}.glyphicon-bed:before{content:"\e219"}.glyphicon-apple:before{content:"\f8ff"}.glyphicon-erase:before{content:"\e221"}.glyphicon-hourglass:before{content:"\231b"}.glyphicon-lamp:before{content:"\e223"}.glyphicon-duplicate:before{content:"\e224"}.glyphicon-piggy-bank:before{content:"\e225"}.glyphicon-scissors:before{content:"\e226"}.glyphicon-bitcoin:before{content:"\e227"}.glyphicon-btc:before{content:"\e227"}.glyphicon-xbt:before{content:"\e227"}.glyphicon-yen:before{content:"\00a5"}.glyphicon-jpy:before{content:"\00a5"}.glyphicon-ruble:before{content:"\20bd"}.glyphicon-rub:before{content:"\20bd"}.glyphicon-scale:before{content:"\e230"}.glyphicon-ice-lolly:before{content:"\e231"}.glyphicon-ice-lolly-tasted:before{content:"\e232"}.glyphicon-education:before{content:"\e233"}.glyphicon-option-horizontal:before{content:"\e234"}.glyphicon-option-vertical:before{content:"\e235"}.glyphicon-menu-hamburger:before{content:"\e236"}.glyphicon-modal-window:before{content:"\e237"}.glyphicon-oil:before{content:"\e238"}.glyphicon-grain:before{content:"\e239"}.glyphicon-sunglasses:before{content:"\e240"}.glyphicon-text-size:before{content:"\e241"}.glyphicon-text-color:before{content:"\e242"}.glyphicon-text-background:before{content:"\e243"}.glyphicon-object-align-top:before{content:"\e244"}.glyphicon-object-align-bottom:before{content:"\e245"}.glyphicon-object-align-horizontal:before{content:"\e246"}.glyphicon-object-align-left:before{content:"\e247"}.glyphicon-object-align-vertical:before{content:"\e248"}.glyphicon-object-align-right:before{content:"\e249"}.glyphicon-triangle-right:before{content:"\e250"}.glyphicon-triangle-left:before{content:"\e251"}.glyphicon-triangle-bottom:before{content:"\e252"}.glyphicon-triangle-top:before{content:"\e253"}.glyphicon-console:before{content:"\e254"}.glyphicon-superscript:before{content:"\e255"}.glyphicon-subscript:before{content:"\e256"}.glyphicon-menu-left:before{content:"\e257"}.glyphicon-menu-right:before{content:"\e258"}.glyphicon-menu-down:before{content:"\e259"}.glyphicon-menu-up:before{content:"\e260"}*{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box}:after,:before{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box}html{font-size:10px;-webkit-tap-highlight-color:rgba(0,0,0,0)}body{font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:14px;line-height:1.42857143;color:#333;background-color:#fff}button,input,select,textarea{font-family:inherit;font-size:inherit;line-height:inherit}a{color:#337ab7;text-decoration:none}a:focus,a:hover{color:#23527c;text-decoration:underline}a:focus{outline:thin dotted;outline:5px auto -webkit-focus-ring-color;outline-offset:-2px}figure{margin:0}img{vertical-align:middle}.carousel-inner>.item>a>img,.carousel-inner>.item>img,.img-responsive,.thumbnail a>img,.thumbnail>img{display:block;max-width:100%;height:auto}.img-rounded{border-radius:6px}.img-thumbnail{display:inline-block;max-width:100%;height:auto;padding:4px;line-height:1.42857143;background-color:#fff;border:1px solid #ddd;border-radius:4px;-webkit-transition:all .2s ease-in-out;-o-transition:all .2s ease-in-out;transition:all .2s ease-in-out}.img-circle{border-radius:50%}hr{margin-top:20px;margin-bottom:20px;border:0;border-top:1px solid #eee}.sr-only{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0,0,0,0);border:0}.sr-only-focusable:active,.sr-only-focusable:focus{position:static;width:auto;height:auto;margin:0;overflow:visible;clip:auto}[role=button]{cursor:pointer}.h1,.h2,.h3,.h4,.h5,.h6,h1,h2,h3,h4,h5,h6{font-family:inherit;font-weight:500;line-height:1.1;color:inherit}.h1 .small,.h1 small,.h2 .small,.h2 small,.h3 .small,.h3 small,.h4 .small,.h4 small,.h5 .small,.h5 small,.h6 .small,.h6 small,h1 .small,h1 small,h2 .small,h2 small,h3 .small,h3 small,h4 .small,h4 small,h5 .small,h5 small,h6 .small,h6 small{font-weight:400;line-height:1;color:#777}.h1,.h2,.h3,h1,h2,h3{margin-top:20px;margin-bottom:10px}.h1 .small,.h1 small,.h2 .small,.h2 small,.h3 .small,.h3 small,h1 .small,h1 small,h2 .small,h2 small,h3 .small,h3 small{font-size:65%}.h4,.h5,.h6,h4,h5,h6{margin-top:10px;margin-bottom:10px}.h4 .small,.h4 small,.h5 .small,.h5 small,.h6 .small,.h6 small,h4 .small,h4 small,h5 .small,h5 small,h6 .small,h6 small{font-size:75%}.h1,h1{font-size:36px}.h2,h2{font-size:30px}.h3,h3{font-size:24px}.h4,h4{font-size:18px}.h5,h5{font-size:14px}.h6,h6{font-size:12px}p{margin:0 0 10px}.lead{margin-bottom:20px;font-size:16px;font-weight:300;line-height:1.4}@media (min-width:768px){.lead{font-size:21px}}.small,small{font-size:85%}.mark,mark{padding:.2em;background-color:#fcf8e3}.text-left{text-align:left}.text-right{text-align:right}.text-center{text-align:center}.text-justify{text-align:justify}.text-nowrap{white-space:nowrap}.text-lowercase{text-transform:lowercase}.text-uppercase{text-transform:uppercase}.text-capitalize{text-transform:capitalize}.text-muted{color:#777}.text-primary{color:#337ab7}a.text-primary:focus,a.text-primary:hover{color:#286090}.text-success{color:#3c763d}a.text-success:focus,a.text-success:hover{color:#2b542c}.text-info{color:#31708f}a.text-info:focus,a.text-info:hover{color:#245269}.text-warning{color:#8a6d3b}a.text-warning:focus,a.text-warning:hover{color:#66512c}.text-danger{color:#a94442}a.text-danger:focus,a.text-danger:hover{color:#843534}.bg-primary{color:#fff;background-color:#337ab7}a.bg-primary:focus,a.bg-primary:hover{background-color:#286090}.bg-success{background-color:#dff0d8}a.bg-success:focus,a.bg-success:hover{background-color:#c1e2b3}.bg-info{background-color:#d9edf7}a.bg-info:focus,a.bg-info:hover{background-color:#afd9ee}.bg-warning{background-color:#fcf8e3}a.bg-warning:focus,a.bg-warning:hover{background-color:#f7ecb5}.bg-danger{background-color:#f2dede}a.bg-danger:focus,a.bg-danger:hover{background-color:#e4b9b9}.page-header{padding-bottom:9px;margin:40px 0 20px;border-bottom:1px solid #eee}ol,ul{margin-top:0;margin-bottom:10px}ol ol,ol ul,ul ol,ul ul{margin-bottom:0}.list-unstyled{padding-left:0;list-style:none}.list-inline{padding-left:0;margin-left:-5px;list-style:none}.list-inline>li{display:inline-block;padding-right:5px;padding-left:5px}dl{margin-top:0;margin-bottom:20px}dd,dt{line-height:1.42857143}dt{font-weight:700}dd{margin-left:0}@media (min-width:768px){.dl-horizontal dt{float:left;width:160px;overflow:hidden;clear:left;text-align:right;text-overflow:ellipsis;white-space:nowrap}.dl-horizontal dd{margin-left:180px}}abbr[data-original-title],abbr[title]{cursor:help;border-bottom:1px dotted #777}.initialism{font-size:90%;text-transform:uppercase}blockquote{padding:10px 20px;margin:0 0 20px;font-size:17.5px;border-left:5px solid #eee}blockquote ol:last-child,blockquote p:last-child,blockquote ul:last-child{margin-bottom:0}blockquote .small,blockquote footer,blockquote small{display:block;font-size:80%;line-height:1.42857143;color:#777}blockquote .small:before,blockquote footer:before,blockquote small:before{content:'\2014 \00A0'}.blockquote-reverse,blockquote.pull-right{padding-right:15px;padding-left:0;text-align:right;border-right:5px solid #eee;border-left:0}.blockquote-reverse .small:before,.blockquote-reverse footer:before,.blockquote-reverse small:before,blockquote.pull-right .small:before,blockquote.pull-right footer:before,blockquote.pull-right small:before{content:''}.blockquote-reverse .small:after,.blockquote-reverse footer:after,.blockquote-reverse small:after,blockquote.pull-right .small:after,blockquote.pull-right footer:after,blockquote.pull-right small:after{content:'\00A0 \2014'}address{margin-bottom:20px;font-style:normal;line-height:1.42857143}code,kbd,pre,samp{font-family:monospace}code{padding:2px 4px;font-size:90%;color:#c7254e;background-color:#f9f2f4;border-radius:4px}kbd{padding:2px 4px;font-size:90%;color:#fff;background-color:#333;border-radius:3px;-webkit-box-shadow:inset 0 -1px 0 rgba(0,0,0,.25);box-shadow:inset 0 -1px 0 rgba(0,0,0,.25)}kbd kbd{padding:0;font-size:100%;font-weight:700;-webkit-box-shadow:none;box-shadow:none}pre{display:block;padding:9.5px;margin:0 0 10px;font-size:13px;line-height:1.42857143;color:#333;word-break:break-all;word-wrap:break-word;background-color:#f5f5f5;border:1px solid #ccc;border-radius:4px}pre code{padding:0;font-size:inherit;color:inherit;white-space:pre-wrap;background-color:transparent;border-radius:0}.pre-scrollable{max-height:340px;overflow-y:scroll}.container{padding-right:15px;padding-left:15px;margin-right:auto;margin-left:auto}@media (min-width:768px){.container{width:750px}}@media (min-width:992px){.container{width:970px}}@media (min-width:1200px){.container{width:1170px}}.container-fluid{padding-right:15px;padding-left:15px;margin-right:auto;margin-left:auto}.row{margin-right:-15px;margin-left:-15px}.col-lg-1,.col-lg-10,.col-lg-11,.col-lg-12,.col-lg-2,.col-lg-3,.col-lg-4,.col-lg-5,.col-lg-6,.col-lg-7,.col-lg-8,.col-lg-9,.col-md-1,.col-md-10,.col-md-11,.col-md-12,.col-md-2,.col-md-3,.col-md-4,.col-md-5,.col-md-6,.col-md-7,.col-md-8,.col-md-9,.col-sm-1,.col-sm-10,.col-sm-11,.col-sm-12,.col-sm-2,.col-sm-3,.col-sm-4,.col-sm-5,.col-sm-6,.col-sm-7,.col-sm-8,.col-sm-9,.col-xs-1,.col-xs-10,.col-xs-11,.col-xs-12,.col-xs-2,.col-xs-3,.col-xs-4,.col-xs-5,.col-xs-6,.col-xs-7,.col-xs-8,.col-xs-9{position:relative;min-height:1px;padding-right:15px;padding-left:15px}.col-xs-1,.col-xs-10,.col-xs-11,.col-xs-12,.col-xs-2,.col-xs-3,.col-xs-4,.col-xs-5,.col-xs-6,.col-xs-7,.col-xs-8,.col-xs-9{float:left}.col-xs-12{width:100%}.col-xs-11{width:91.66666667%}.col-xs-10{width:83.33333333%}.col-xs-9{width:75%}.col-xs-8{width:66.66666667%}.col-xs-7{width:58.33333333%}.col-xs-6{width:50%}.col-xs-5{width:41.66666667%}.col-xs-4{width:33.33333333%}.col-xs-3{width:25%}.col-xs-2{width:16.66666667%}.col-xs-1{width:8.33333333%}.col-xs-pull-12{right:100%}.col-xs-pull-11{right:91.66666667%}.col-xs-pull-10{right:83.33333333%}.col-xs-pull-9{right:75%}.col-xs-pull-8{right:66.66666667%}.col-xs-pull-7{right:58.33333333%}.col-xs-pull-6{right:50%}.col-xs-pull-5{right:41.66666667%}.col-xs-pull-4{right:33.33333333%}.col-xs-pull-3{right:25%}.col-xs-pull-2{right:16.66666667%}.col-xs-pull-1{right:8.33333333%}.col-xs-pull-0{right:auto}.col-xs-push-12{left:100%}.col-xs-push-11{left:91.66666667%}.col-xs-push-10{left:83.33333333%}.col-xs-push-9{left:75%}.col-xs-push-8{left:66.66666667%}.col-xs-push-7{left:58.33333333%}.col-xs-push-6{left:50%}.col-xs-push-5{left:41.66666667%}.col-xs-push-4{left:33.33333333%}.col-xs-push-3{left:25%}.col-xs-push-2{left:16.66666667%}.col-xs-push-1{left:8.33333333%}.col-xs-push-0{left:auto}.col-xs-offset-12{margin-left:100%}.col-xs-offset-11{margin-left:91.66666667%}.col-xs-offset-10{margin-left:83.33333333%}.col-xs-offset-9{margin-left:75%}.col-xs-offset-8{margin-left:66.66666667%}.col-xs-offset-7{margin-left:58.33333333%}.col-xs-offset-6{margin-left:50%}.col-xs-offset-5{margin-left:41.66666667%}.col-xs-offset-4{margin-left:33.33333333%}.col-xs-offset-3{margin-left:25%}.col-xs-offset-2{margin-left:16.66666667%}.col-xs-offset-1{margin-left:8.33333333%}.col-xs-offset-0{margin-left:0}@media (min-width:768px){.col-sm-1,.col-sm-10,.col-sm-11,.col-sm-12,.col-sm-2,.col-sm-3,.col-sm-4,.col-sm-5,.col-sm-6,.col-sm-7,.col-sm-8,.col-sm-9{float:left}.col-sm-12{width:100%}.col-sm-11{width:91.66666667%}.col-sm-10{width:83.33333333%}.col-sm-9{width:75%}.col-sm-8{width:66.66666667%}.col-sm-7{width:58.33333333%}.col-sm-6{width:50%}.col-sm-5{width:41.66666667%}.col-sm-4{width:33.33333333%}.col-sm-3{width:25%}.col-sm-2{width:16.66666667%}.col-sm-1{width:8.33333333%}.col-sm-pull-12{right:100%}.col-sm-pull-11{right:91.66666667%}.col-sm-pull-10{right:83.33333333%}.col-sm-pull-9{right:75%}.col-sm-pull-8{right:66.66666667%}.col-sm-pull-7{right:58.33333333%}.col-sm-pull-6{right:50%}.col-sm-pull-5{right:41.66666667%}.col-sm-pull-4{right:33.33333333%}.col-sm-pull-3{right:25%}.col-sm-pull-2{right:16.66666667%}.col-sm-pull-1{right:8.33333333%}.col-sm-pull-0{right:auto}.col-sm-push-12{left:100%}.col-sm-push-11{left:91.66666667%}.col-sm-push-10{left:83.33333333%}.col-sm-push-9{left:75%}.col-sm-push-8{left:66.66666667%}.col-sm-push-7{left:58.33333333%}.col-sm-push-6{left:50%}.col-sm-push-5{left:41.66666667%}.col-sm-push-4{left:33.33333333%}.col-sm-push-3{left:25%}.col-sm-push-2{left:16.66666667%}.col-sm-push-1{left:8.33333333%}.col-sm-push-0{left:auto}.col-sm-offset-12{margin-left:100%}.col-sm-offset-11{margin-left:91.66666667%}.col-sm-offset-10{margin-left:83.33333333%}.col-sm-offset-9{margin-left:75%}.col-sm-offset-8{margin-left:66.66666667%}.col-sm-offset-7{margin-left:58.33333333%}.col-sm-offset-6{margin-left:50%}.col-sm-offset-5{margin-left:41.66666667%}.col-sm-offset-4{margin-left:33.33333333%}.col-sm-offset-3{margin-left:25%}.col-sm-offset-2{margin-left:16.66666667%}.col-sm-offset-1{margin-left:8.33333333%}.col-sm-offset-0{margin-left:0}}@media (min-width:992px){.col-md-1,.col-md-10,.col-md-11,.col-md-12,.col-md-2,.col-md-3,.col-md-4,.col-md-5,.col-md-6,.col-md-7,.col-md-8,.col-md-9{float:left}.col-md-12{width:100%}.col-md-11{width:91.66666667%}.col-md-10{width:83.33333333%}.col-md-9{width:75%}.col-md-8{width:66.66666667%}.col-md-7{width:58.33333333%}.col-md-6{width:50%}.col-md-5{width:41.66666667%}.col-md-4{width:33.33333333%}.col-md-3{width:25%}.col-md-2{width:16.66666667%}.col-md-1{width:8.33333333%}.col-md-pull-12{right:100%}.col-md-pull-11{right:91.66666667%}.col-md-pull-10{right:83.33333333%}.col-md-pull-9{right:75%}.col-md-pull-8{right:66.66666667%}.col-md-pull-7{right:58.33333333%}.col-md-pull-6{right:50%}.col-md-pull-5{right:41.66666667%}.col-md-pull-4{right:33.33333333%}.col-md-pull-3{right:25%}.col-md-pull-2{right:16.66666667%}.col-md-pull-1{right:8.33333333%}.col-md-pull-0{right:auto}.col-md-push-12{left:100%}.col-md-push-11{left:91.66666667%}.col-md-push-10{left:83.33333333%}.col-md-push-9{left:75%}.col-md-push-8{left:66.66666667%}.col-md-push-7{left:58.33333333%}.col-md-push-6{left:50%}.col-md-push-5{left:41.66666667%}.col-md-push-4{left:33.33333333%}.col-md-push-3{left:25%}.col-md-push-2{left:16.66666667%}.col-md-push-1{left:8.33333333%}.col-md-push-0{left:auto}.col-md-offset-12{margin-left:100%}.col-md-offset-11{margin-left:91.66666667%}.col-md-offset-10{margin-left:83.33333333%}.col-md-offset-9{margin-left:75%}.col-md-offset-8{margin-left:66.66666667%}.col-md-offset-7{margin-left:58.33333333%}.col-md-offset-6{margin-left:50%}.col-md-offset-5{margin-left:41.66666667%}.col-md-offset-4{margin-left:33.33333333%}.col-md-offset-3{margin-left:25%}.col-md-offset-2{margin-left:16.66666667%}.col-md-offset-1{margin-left:8.33333333%}.col-md-offset-0{margin-left:0}}@media (min-width:1200px){.col-lg-1,.col-lg-10,.col-lg-11,.col-lg-12,.col-lg-2,.col-lg-3,.col-lg-4,.col-lg-5,.col-lg-6,.col-lg-7,.col-lg-8,.col-lg-9{float:left}.col-lg-12{width:100%}.col-lg-11{width:91.66666667%}.col-lg-10{width:83.33333333%}.col-lg-9{width:75%}.col-lg-8{width:66.66666667%}.col-lg-7{width:58.33333333%}.col-lg-6{width:50%}.col-lg-5{width:41.66666667%}.col-lg-4{width:33.33333333%}.col-lg-3{width:25%}.col-lg-2{width:16.66666667%}.col-lg-1{width:8.33333333%}.col-lg-pull-12{right:100%}.col-lg-pull-11{right:91.66666667%}.col-lg-pull-10{right:83.33333333%}.col-lg-pull-9{right:75%}.col-lg-pull-8{right:66.66666667%}.col-lg-pull-7{right:58.33333333%}.col-lg-pull-6{right:50%}.col-lg-pull-5{right:41.66666667%}.col-lg-pull-4{right:33.33333333%}.col-lg-pull-3{right:25%}.col-lg-pull-2{right:16.66666667%}.col-lg-pull-1{right:8.33333333%}.col-lg-pull-0{right:auto}.col-lg-push-12{left:100%}.col-lg-push-11{left:91.66666667%}.col-lg-push-10{left:83.33333333%}.col-lg-push-9{left:75%}.col-lg-push-8{left:66.66666667%}.col-lg-push-7{left:58.33333333%}.col-lg-push-6{left:50%}.col-lg-push-5{left:41.66666667%}.col-lg-push-4{left:33.33333333%}.col-lg-push-3{left:25%}.col-lg-push-2{left:16.66666667%}.col-lg-push-1{left:8.33333333%}.col-lg-push-0{left:auto}.col-lg-offset-12{margin-left:100%}.col-lg-offset-11{margin-left:91.66666667%}.col-lg-offset-10{margin-left:83.33333333%}.col-lg-offset-9{margin-left:75%}.col-lg-offset-8{margin-left:66.66666667%}.col-lg-offset-7{margin-left:58.33333333%}.col-lg-offset-6{margin-left:50%}.col-lg-offset-5{margin-left:41.66666667%}.col-lg-offset-4{margin-left:33.33333333%}.col-lg-offset-3{margin-left:25%}.col-lg-offset-2{margin-left:16.66666667%}.col-lg-offset-1{margin-left:8.33333333%}.col-lg-offset-0{margin-left:0}}table{background-color:transparent}caption{padding-top:8px;padding-bottom:8px;color:#777;text-align:left}th{}.table{width:100%;max-width:100%;margin-bottom:20px}.table>tbody>tr>td,.table>tbody>tr>th,.table>tfoot>tr>td,.table>tfoot>tr>th,.table>thead>tr>td,.table>thead>tr>th{padding:8px;line-height:1.42857143;vertical-align:top;border-top:1px solid #ddd}.table>thead>tr>th{vertical-align:bottom;border-bottom:2px solid #ddd}.table>caption+thead>tr:first-child>td,.table>caption+thead>tr:first-child>th,.table>colgroup+thead>tr:first-child>td,.table>colgroup+thead>tr:first-child>th,.table>thead:first-child>tr:first-child>td,.table>thead:first-child>tr:first-child>th{border-top:0}.table>tbody+tbody{border-top:2px solid #ddd}.table .table{background-color:#fff}.table-condensed>tbody>tr>td,.table-condensed>tbody>tr>th,.table-condensed>tfoot>tr>td,.table-condensed>tfoot>tr>th,.table-condensed>thead>tr>td,.table-condensed>thead>tr>th{padding:5px}.table-bordered{border:1px solid #ddd}.table-bordered>tbody>tr>td,.table-bordered>tbody>tr>th,.table-bordered>tfoot>tr>td,.table-bordered>tfoot>tr>th,.table-bordered>thead>tr>td,.table-bordered>thead>tr>th{border:1px solid #ddd}.table-bordered>thead>tr>td,.table-bordered>thead>tr>th{border-bottom-width:2px}.table-striped>tbody>tr:nth-of-type(odd){background-color:#f9f9f9}.table-hover>tbody>tr:hover{background-color:#f5f5f5}table col[class*=col-]{position:static;display:table-column;float:none}table td[class*=col-],table th[class*=col-]{position:static;display:table-cell;float:none}.table>tbody>tr.active>td,.table>tbody>tr.active>th,.table>tbody>tr>td.active,.table>tbody>tr>th.active,.table>tfoot>tr.active>td,.table>tfoot>tr.active>th,.table>tfoot>tr>td.active,.table>tfoot>tr>th.active,.table>thead>tr.active>td,.table>thead>tr.active>th,.table>thead>tr>td.active,.table>thead>tr>th.active{background-color:#f5f5f5}.table-hover>tbody>tr.active:hover>td,.table-hover>tbody>tr.active:hover>th,.table-hover>tbody>tr:hover>.active,.table-hover>tbody>tr>td.active:hover,.table-hover>tbody>tr>th.active:hover{background-color:#e8e8e8}.table>tbody>tr.success>td,.table>tbody>tr.success>th,.table>tbody>tr>td.success,.table>tbody>tr>th.success,.table>tfoot>tr.success>td,.table>tfoot>tr.success>th,.table>tfoot>tr>td.success,.table>tfoot>tr>th.success,.table>thead>tr.success>td,.table>thead>tr.success>th,.table>thead>tr>td.success,.table>thead>tr>th.success{background-color:#dff0d8}.table-hover>tbody>tr.success:hover>td,.table-hover>tbody>tr.success:hover>th,.table-hover>tbody>tr:hover>.success,.table-hover>tbody>tr>td.success:hover,.table-hover>tbody>tr>th.success:hover{background-color:#d0e9c6}.table>tbody>tr.info>td,.table>tbody>tr.info>th,.table>tbody>tr>td.info,.table>tbody>tr>th.info,.table>tfoot>tr.info>td,.table>tfoot>tr.info>th,.table>tfoot>tr>td.info,.table>tfoot>tr>th.info,.table>thead>tr.info>td,.table>thead>tr.info>th,.table>thead>tr>td.info,.table>thead>tr>th.info{background-color:#d9edf7}.table-hover>tbody>tr.info:hover>td,.table-hover>tbody>tr.info:hover>th,.table-hover>tbody>tr:hover>.info,.table-hover>tbody>tr>td.info:hover,.table-hover>tbody>tr>th.info:hover{background-color:#c4e3f3}.table>tbody>tr.warning>td,.table>tbody>tr.warning>th,.table>tbody>tr>td.warning,.table>tbody>tr>th.warning,.table>tfoot>tr.warning>td,.table>tfoot>tr.warning>th,.table>tfoot>tr>td.warning,.table>tfoot>tr>th.warning,.table>thead>tr.warning>td,.table>thead>tr.warning>th,.table>thead>tr>td.warning,.table>thead>tr>th.warning{background-color:#fcf8e3}.table-hover>tbody>tr.warning:hover>td,.table-hover>tbody>tr.warning:hover>th,.table-hover>tbody>tr:hover>.warning,.table-hover>tbody>tr>td.warning:hover,.table-hover>tbody>tr>th.warning:hover{background-color:#faf2cc}.table>tbody>tr.danger>td,.table>tbody>tr.danger>th,.table>tbody>tr>td.danger,.table>tbody>tr>th.danger,.table>tfoot>tr.danger>td,.table>tfoot>tr.danger>th,.table>tfoot>tr>td.danger,.table>tfoot>tr>th.danger,.table>thead>tr.danger>td,.table>thead>tr.danger>th,.table>thead>tr>td.danger,.table>thead>tr>th.danger{background-color:#f2dede}.table-hover>tbody>tr.danger:hover>td,.table-hover>tbody>tr.danger:hover>th,.table-hover>tbody>tr:hover>.danger,.table-hover>tbody>tr>td.danger:hover,.table-hover>tbody>tr>th.danger:hover{background-color:#ebcccc}.table-responsive{min-height:.01%;overflow-x:auto}@media screen and (max-width:767px){.table-responsive{width:100%;margin-bottom:15px;overflow-y:hidden;-ms-overflow-style:-ms-autohiding-scrollbar;border:1px solid #ddd}.table-responsive>.table{margin-bottom:0}.table-responsive>.table>tbody>tr>td,.table-responsive>.table>tbody>tr>th,.table-responsive>.table>tfoot>tr>td,.table-responsive>.table>tfoot>tr>th,.table-responsive>.table>thead>tr>td,.table-responsive>.table>thead>tr>th{white-space:nowrap}.table-responsive>.table-bordered{border:0}.table-responsive>.table-bordered>tbody>tr>td:first-child,.table-responsive>.table-bordered>tbody>tr>th:first-child,.table-responsive>.table-bordered>tfoot>tr>td:first-child,.table-responsive>.table-bordered>tfoot>tr>th:first-child,.table-responsive>.table-bordered>thead>tr>td:first-child,.table-responsive>.table-bordered>thead>tr>th:first-child{border-left:0}.table-responsive>.table-bordered>tbody>tr>td:last-child,.table-responsive>.table-bordered>tbody>tr>th:last-child,.table-responsive>.table-bordered>tfoot>tr>td:last-child,.table-responsive>.table-bordered>tfoot>tr>th:last-child,.table-responsive>.table-bordered>thead>tr>td:last-child,.table-responsive>.table-bordered>thead>tr>th:last-child{border-right:0}.table-responsive>.table-bordered>tbody>tr:last-child>td,.table-responsive>.table-bordered>tbody>tr:last-child>th,.table-responsive>.table-bordered>tfoot>tr:last-child>td,.table-responsive>.table-bordered>tfoot>tr:last-child>th{border-bottom:0}}fieldset{min-width:0;padding:0;margin:0;border:0}legend{display:block;width:100%;padding:0;margin-bottom:20px;font-size:21px;line-height:inherit;color:#333;border:0;border-bottom:1px solid #e5e5e5}label{display:inline-block;max-width:100%;margin-bottom:5px;font-weight:700}input[type=search]{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box}input[type=checkbox],input[type=radio]{margin:4px 0 0;margin-top:1px\9;line-height:normal}input[type=file]{display:block}input[type=range]{display:block;width:100%}select[multiple],select[size]{height:auto}input[type=file]:focus,input[type=checkbox]:focus,input[type=radio]:focus{outline:thin dotted;outline:5px auto -webkit-focus-ring-color;outline-offset:-2px}output{display:block;padding-top:7px;font-size:14px;line-height:1.42857143;color:#555}.form-control{display:block;width:100%;height:34px;padding:6px 12px;font-size:14px;line-height:1.42857143;color:#555;background-color:#fff;background-image:none;border:1px solid #ccc;border-radius:4px;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075);-webkit-transition:border-color ease-in-out .15s,-webkit-box-shadow ease-in-out .15s;-o-transition:border-color ease-in-out .15s,box-shadow ease-in-out .15s;transition:border-color ease-in-out .15s,box-shadow ease-in-out .15s}.form-control:focus{border-color:#66afe9;outline:0;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 8px rgba(102,175,233,.6);box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 8px rgba(102,175,233,.6)}.form-control::-moz-placeholder{color:#999;opacity:1}.form-control:-ms-input-placeholder{color:#999}.form-control::-webkit-input-placeholder{color:#999}.form-control[disabled],.form-control[readonly],fieldset[disabled] .form-control{background-color:#eee;opacity:1}.form-control[disabled],fieldset[disabled] .form-control{cursor:not-allowed}textarea.form-control{height:auto}input[type=search]{-webkit-appearance:none}@media screen and (-webkit-min-device-pixel-ratio:0){input[type=date].form-control,input[type=time].form-control,input[type=datetime-local].form-control,input[type=month].form-control{line-height:34px}.input-group-sm input[type=date],.input-group-sm input[type=time],.input-group-sm input[type=datetime-local],.input-group-sm input[type=month],input[type=date].input-sm,input[type=time].input-sm,input[type=datetime-local].input-sm,input[type=month].input-sm{line-height:30px}.input-group-lg input[type=date],.input-group-lg input[type=time],.input-group-lg input[type=datetime-local],.input-group-lg input[type=month],input[type=date].input-lg,input[type=time].input-lg,input[type=datetime-local].input-lg,input[type=month].input-lg{line-height:46px}}.form-group{margin-bottom:15px}.checkbox,.radio{position:relative;display:block;margin-top:10px;margin-bottom:10px}.checkbox label,.radio label{min-height:20px;padding-left:20px;margin-bottom:0;font-weight:400;cursor:pointer}.checkbox input[type=checkbox],.checkbox-inline input[type=checkbox],.radio input[type=radio],.radio-inline input[type=radio]{position:absolute;margin-top:4px\9;margin-left:-20px}.checkbox+.checkbox,.radio+.radio{margin-top:-5px}.checkbox-inline,.radio-inline{position:relative;display:inline-block;padding-left:20px;margin-bottom:0;font-weight:400;vertical-align:middle;cursor:pointer}.checkbox-inline+.checkbox-inline,.radio-inline+.radio-inline{margin-top:0;margin-left:10px}fieldset[disabled] input[type=checkbox],fieldset[disabled] input[type=radio],input[type=checkbox].disabled,input[type=checkbox][disabled],input[type=radio].disabled,input[type=radio][disabled]{cursor:not-allowed}.checkbox-inline.disabled,.radio-inline.disabled,fieldset[disabled] .checkbox-inline,fieldset[disabled] .radio-inline{cursor:not-allowed}.checkbox.disabled label,.radio.disabled label,fieldset[disabled] .checkbox label,fieldset[disabled] .radio label{cursor:not-allowed}.form-control-static{min-height:34px;padding-top:7px;padding-bottom:7px;margin-bottom:0}.form-control-static.input-lg,.form-control-static.input-sm{padding-right:0;padding-left:0}.input-sm{height:30px;padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}select.input-sm{height:30px;line-height:30px}select[multiple].input-sm,textarea.input-sm{height:auto}.form-group-sm .form-control{height:30px;padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}.form-group-sm select.form-control{height:30px;line-height:30px}.form-group-sm select[multiple].form-control,.form-group-sm textarea.form-control{height:auto}.form-group-sm .form-control-static{height:30px;min-height:32px;padding:6px 10px;font-size:12px;line-height:1.5}.input-lg{height:46px;padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}select.input-lg{height:46px;line-height:46px}select[multiple].input-lg,textarea.input-lg{height:auto}.form-group-lg .form-control{height:46px;padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}.form-group-lg select.form-control{height:46px;line-height:46px}.form-group-lg select[multiple].form-control,.form-group-lg textarea.form-control{height:auto}.form-group-lg .form-control-static{height:46px;min-height:38px;padding:11px 16px;font-size:18px;line-height:1.3333333}.has-feedback{position:relative}.has-feedback .form-control{padding-right:42.5px}.form-control-feedback{position:absolute;top:0;right:0;z-index:2;display:block;width:34px;height:34px;line-height:34px;text-align:center;pointer-events:none}.form-group-lg .form-control+.form-control-feedback,.input-group-lg+.form-control-feedback,.input-lg+.form-control-feedback{width:46px;height:46px;line-height:46px}.form-group-sm .form-control+.form-control-feedback,.input-group-sm+.form-control-feedback,.input-sm+.form-control-feedback{width:30px;height:30px;line-height:30px}.has-success .checkbox,.has-success .checkbox-inline,.has-success .control-label,.has-success .help-block,.has-success .radio,.has-success .radio-inline,.has-success.checkbox label,.has-success.checkbox-inline label,.has-success.radio label,.has-success.radio-inline label{color:#3c763d}.has-success .form-control{border-color:#3c763d;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-success .form-control:focus{border-color:#2b542c;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #67b168;box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #67b168}.has-success .input-group-addon{color:#3c763d;background-color:#dff0d8;border-color:#3c763d}.has-success .form-control-feedback{color:#3c763d}.has-warning .checkbox,.has-warning .checkbox-inline,.has-warning .control-label,.has-warning .help-block,.has-warning .radio,.has-warning .radio-inline,.has-warning.checkbox label,.has-warning.checkbox-inline label,.has-warning.radio label,.has-warning.radio-inline label{color:#8a6d3b}.has-warning .form-control{border-color:#8a6d3b;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-warning .form-control:focus{border-color:#66512c;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #c0a16b;box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #c0a16b}.has-warning .input-group-addon{color:#8a6d3b;background-color:#fcf8e3;border-color:#8a6d3b}.has-warning .form-control-feedback{color:#8a6d3b}.has-error .checkbox,.has-error .checkbox-inline,.has-error .control-label,.has-error .help-block,.has-error .radio,.has-error .radio-inline,.has-error.checkbox label,.has-error.checkbox-inline label,.has-error.radio label,.has-error.radio-inline label{color:#a94442}.has-error .form-control{border-color:#a94442;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-error .form-control:focus{border-color:#843534;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #ce8483;box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #ce8483}.has-error .input-group-addon{color:#a94442;background-color:#f2dede;border-color:#a94442}.has-error .form-control-feedback{color:#a94442}.has-feedback label~.form-control-feedback{top:25px}.has-feedback label.sr-only~.form-control-feedback{top:0}.help-block{display:block;margin-top:5px;margin-bottom:10px;color:#737373}@media (min-width:768px){.form-inline .form-group{display:inline-block;margin-bottom:0;vertical-align:middle}.form-inline .form-control{display:inline-block;width:auto;vertical-align:middle}.form-inline .form-control-static{display:inline-block}.form-inline .input-group{display:inline-table;vertical-align:middle}.form-inline .input-group .form-control,.form-inline .input-group .input-group-addon,.form-inline .input-group .input-group-btn{width:auto}.form-inline .input-group>.form-control{width:100%}.form-inline .control-label{margin-bottom:0;vertical-align:middle}.form-inline .checkbox,.form-inline .radio{display:inline-block;margin-top:0;margin-bottom:0;vertical-align:middle}.form-inline .checkbox label,.form-inline .radio label{padding-left:0}.form-inline .checkbox input[type=checkbox],.form-inline .radio input[type=radio]{position:relative;margin-left:0}.form-inline .has-feedback .form-control-feedback{top:0}}.form-horizontal .checkbox,.form-horizontal .checkbox-inline,.form-horizontal .radio,.form-horizontal .radio-inline{padding-top:7px;margin-top:0;margin-bottom:0}.form-horizontal .checkbox,.form-horizontal .radio{min-height:27px}.form-horizontal .form-group{margin-right:-15px;margin-left:-15px}@media (min-width:768px){.form-horizontal .control-label{padding-top:7px;margin-bottom:0;text-align:right}}.form-horizontal .has-feedback .form-control-feedback{right:15px}@media (min-width:768px){.form-horizontal .form-group-lg .control-label{padding-top:14.33px;font-size:18px}}@media (min-width:768px){.form-horizontal .form-group-sm .control-label{padding-top:6px;font-size:12px}}.btn{display:inline-block;padding:6px 12px;margin-bottom:0;font-size:14px;font-weight:400;line-height:1.42857143;text-align:center;white-space:nowrap;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-image:none;border:1px solid transparent;border-radius:4px}.btn.active.focus,.btn.active:focus,.btn.focus,.btn:active.focus,.btn:active:focus,.btn:focus{outline:thin dotted;outline:5px auto -webkit-focus-ring-color;outline-offset:-2px}.btn.focus,.btn:focus,.btn:hover{color:#333;text-decoration:none}.btn.active,.btn:active{background-image:none;outline:0;-webkit-box-shadow:inset 0 3px 5px rgba(0,0,0,.125);box-shadow:inset 0 3px 5px rgba(0,0,0,.125)}.btn.disabled,.btn[disabled],fieldset[disabled] .btn{cursor:not-allowed;filter:alpha(opacity=65);-webkit-box-shadow:none;box-shadow:none;opacity:.65}a.btn.disabled,fieldset[disabled] a.btn{pointer-events:none}.btn-default{color:#333;background-color:#fff;border-color:#ccc}.btn-default.focus,.btn-default:focus{color:#333;background-color:#e6e6e6;border-color:#8c8c8c}.btn-default:hover{color:#333;background-color:#e6e6e6;border-color:#adadad}.btn-default.active,.btn-default:active,.open>.dropdown-toggle.btn-default{color:#333;background-color:#e6e6e6;border-color:#adadad}.btn-default.active.focus,.btn-default.active:focus,.btn-default.active:hover,.btn-default:active.focus,.btn-default:active:focus,.btn-default:active:hover,.open>.dropdown-toggle.btn-default.focus,.open>.dropdown-toggle.btn-default:focus,.open>.dropdown-toggle.btn-default:hover{color:#333;background-color:#d4d4d4;border-color:#8c8c8c}.btn-default.active,.btn-default:active,.open>.dropdown-toggle.btn-default{background-image:none}.btn-default.disabled,.btn-default.disabled.active,.btn-default.disabled.focus,.btn-default.disabled:active,.btn-default.disabled:focus,.btn-default.disabled:hover,.btn-default[disabled],.btn-default[disabled].active,.btn-default[disabled].focus,.btn-default[disabled]:active,.btn-default[disabled]:focus,.btn-default[disabled]:hover,fieldset[disabled] .btn-default,fieldset[disabled] .btn-default.active,fieldset[disabled] .btn-default.focus,fieldset[disabled] .btn-default:active,fieldset[disabled] .btn-default:focus,fieldset[disabled] .btn-default:hover{background-color:#fff;border-color:#ccc}.btn-default .badge{color:#fff;background-color:#333}.btn-primary{color:#fff;background-color:#337ab7;border-color:#2e6da4}.btn-primary.focus,.btn-primary:focus{color:#fff;background-color:#286090;border-color:#122b40}.btn-primary:hover{color:#fff;background-color:#286090;border-color:#204d74}.btn-primary.active,.btn-primary:active,.open>.dropdown-toggle.btn-primary{color:#fff;background-color:#286090;border-color:#204d74}.btn-primary.active.focus,.btn-primary.active:focus,.btn-primary.active:hover,.btn-primary:active.focus,.btn-primary:active:focus,.btn-primary:active:hover,.open>.dropdown-toggle.btn-primary.focus,.open>.dropdown-toggle.btn-primary:focus,.open>.dropdown-toggle.btn-primary:hover{color:#fff;background-color:#204d74;border-color:#122b40}.btn-primary.active,.btn-primary:active,.open>.dropdown-toggle.btn-primary{background-image:none}.btn-primary.disabled,.btn-primary.disabled.active,.btn-primary.disabled.focus,.btn-primary.disabled:active,.btn-primary.disabled:focus,.btn-primary.disabled:hover,.btn-primary[disabled],.btn-primary[disabled].active,.btn-primary[disabled].focus,.btn-primary[disabled]:active,.btn-primary[disabled]:focus,.btn-primary[disabled]:hover,fieldset[disabled] .btn-primary,fieldset[disabled] .btn-primary.active,fieldset[disabled] .btn-primary.focus,fieldset[disabled] .btn-primary:active,fieldset[disabled] .btn-primary:focus,fieldset[disabled] .btn-primary:hover{background-color:#337ab7;border-color:#2e6da4}.btn-primary .badge{color:#337ab7;background-color:#fff}.btn-success{color:#fff;background-color:#5cb85c;border-color:#4cae4c}.btn-success.focus,.btn-success:focus{color:#fff;background-color:#449d44;border-color:#255625}.btn-success:hover{color:#fff;background-color:#449d44;border-color:#398439}.btn-success.active,.btn-success:active,.open>.dropdown-toggle.btn-success{color:#fff;background-color:#449d44;border-color:#398439}.btn-success.active.focus,.btn-success.active:focus,.btn-success.active:hover,.btn-success:active.focus,.btn-success:active:focus,.btn-success:active:hover,.open>.dropdown-toggle.btn-success.focus,.open>.dropdown-toggle.btn-success:focus,.open>.dropdown-toggle.btn-success:hover{color:#fff;background-color:#398439;border-color:#255625}.btn-success.active,.btn-success:active,.open>.dropdown-toggle.btn-success{background-image:none}.btn-success.disabled,.btn-success.disabled.active,.btn-success.disabled.focus,.btn-success.disabled:active,.btn-success.disabled:focus,.btn-success.disabled:hover,.btn-success[disabled],.btn-success[disabled].active,.btn-success[disabled].focus,.btn-success[disabled]:active,.btn-success[disabled]:focus,.btn-success[disabled]:hover,fieldset[disabled] .btn-success,fieldset[disabled] .btn-success.active,fieldset[disabled] .btn-success.focus,fieldset[disabled] .btn-success:active,fieldset[disabled] .btn-success:focus,fieldset[disabled] .btn-success:hover{background-color:#5cb85c;border-color:#4cae4c}.btn-success .badge{color:#5cb85c;background-color:#fff}.btn-info{color:#fff;background-color:#5bc0de;border-color:#46b8da}.btn-info.focus,.btn-info:focus{color:#fff;background-color:#31b0d5;border-color:#1b6d85}.btn-info:hover{color:#fff;background-color:#31b0d5;border-color:#269abc}.btn-info.active,.btn-info:active,.open>.dropdown-toggle.btn-info{color:#fff;background-color:#31b0d5;border-color:#269abc}.btn-info.active.focus,.btn-info.active:focus,.btn-info.active:hover,.btn-info:active.focus,.btn-info:active:focus,.btn-info:active:hover,.open>.dropdown-toggle.btn-info.focus,.open>.dropdown-toggle.btn-info:focus,.open>.dropdown-toggle.btn-info:hover{color:#fff;background-color:#269abc;border-color:#1b6d85}.btn-info.active,.btn-info:active,.open>.dropdown-toggle.btn-info{background-image:none}.btn-info.disabled,.btn-info.disabled.active,.btn-info.disabled.focus,.btn-info.disabled:active,.btn-info.disabled:focus,.btn-info.disabled:hover,.btn-info[disabled],.btn-info[disabled].active,.btn-info[disabled].focus,.btn-info[disabled]:active,.btn-info[disabled]:focus,.btn-info[disabled]:hover,fieldset[disabled] .btn-info,fieldset[disabled] .btn-info.active,fieldset[disabled] .btn-info.focus,fieldset[disabled] .btn-info:active,fieldset[disabled] .btn-info:focus,fieldset[disabled] .btn-info:hover{background-color:#5bc0de;border-color:#46b8da}.btn-info .badge{color:#5bc0de;background-color:#fff}.btn-warning{color:#fff;background-color:#f0ad4e;border-color:#eea236}.btn-warning.focus,.btn-warning:focus{color:#fff;background-color:#ec971f;border-color:#985f0d}.btn-warning:hover{color:#fff;background-color:#ec971f;border-color:#d58512}.btn-warning.active,.btn-warning:active,.open>.dropdown-toggle.btn-warning{color:#fff;background-color:#ec971f;border-color:#d58512}.btn-warning.active.focus,.btn-warning.active:focus,.btn-warning.active:hover,.btn-warning:active.focus,.btn-warning:active:focus,.btn-warning:active:hover,.open>.dropdown-toggle.btn-warning.focus,.open>.dropdown-toggle.btn-warning:focus,.open>.dropdown-toggle.btn-warning:hover{color:#fff;background-color:#d58512;border-color:#985f0d}.btn-warning.active,.btn-warning:active,.open>.dropdown-toggle.btn-warning{background-image:none}.btn-warning.disabled,.btn-warning.disabled.active,.btn-warning.disabled.focus,.btn-warning.disabled:active,.btn-warning.disabled:focus,.btn-warning.disabled:hover,.btn-warning[disabled],.btn-warning[disabled].active,.btn-warning[disabled].focus,.btn-warning[disabled]:active,.btn-warning[disabled]:focus,.btn-warning[disabled]:hover,fieldset[disabled] .btn-warning,fieldset[disabled] .btn-warning.active,fieldset[disabled] .btn-warning.focus,fieldset[disabled] .btn-warning:active,fieldset[disabled] .btn-warning:focus,fieldset[disabled] .btn-warning:hover{background-color:#f0ad4e;border-color:#eea236}.btn-warning .badge{color:#f0ad4e;background-color:#fff}.btn-danger{color:#fff;background-color:#d9534f;border-color:#d43f3a}.btn-danger.focus,.btn-danger:focus{color:#fff;background-color:#c9302c;border-color:#761c19}.btn-danger:hover{color:#fff;background-color:#c9302c;border-color:#ac2925}.btn-danger.active,.btn-danger:active,.open>.dropdown-toggle.btn-danger{color:#fff;background-color:#c9302c;border-color:#ac2925}.btn-danger.active.focus,.btn-danger.active:focus,.btn-danger.active:hover,.btn-danger:active.focus,.btn-danger:active:focus,.btn-danger:active:hover,.open>.dropdown-toggle.btn-danger.focus,.open>.dropdown-toggle.btn-danger:focus,.open>.dropdown-toggle.btn-danger:hover{color:#fff;background-color:#ac2925;border-color:#761c19}.btn-danger.active,.btn-danger:active,.open>.dropdown-toggle.btn-danger{background-image:none}.btn-danger.disabled,.btn-danger.disabled.active,.btn-danger.disabled.focus,.btn-danger.disabled:active,.btn-danger.disabled:focus,.btn-danger.disabled:hover,.btn-danger[disabled],.btn-danger[disabled].active,.btn-danger[disabled].focus,.btn-danger[disabled]:active,.btn-danger[disabled]:focus,.btn-danger[disabled]:hover,fieldset[disabled] .btn-danger,fieldset[disabled] .btn-danger.active,fieldset[disabled] .btn-danger.focus,fieldset[disabled] .btn-danger:active,fieldset[disabled] .btn-danger:focus,fieldset[disabled] .btn-danger:hover{background-color:#d9534f;border-color:#d43f3a}.btn-danger .badge{color:#d9534f;background-color:#fff}.btn-link{font-weight:400;color:#337ab7;border-radius:0}.btn-link,.btn-link.active,.btn-link:active,.btn-link[disabled],fieldset[disabled] .btn-link{background-color:transparent;-webkit-box-shadow:none;box-shadow:none}.btn-link,.btn-link:active,.btn-link:focus,.btn-link:hover{border-color:transparent}.btn-link:focus,.btn-link:hover{color:#23527c;text-decoration:underline;background-color:transparent}.btn-link[disabled]:focus,.btn-link[disabled]:hover,fieldset[disabled] .btn-link:focus,fieldset[disabled] .btn-link:hover{color:#777;text-decoration:none}.btn-group-lg>.btn,.btn-lg{padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}.btn-group-sm>.btn,.btn-sm{padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}.btn-group-xs>.btn,.btn-xs{padding:1px 5px;font-size:12px;line-height:1.5;border-radius:3px}.btn-block{display:block;width:100%}.btn-block+.btn-block{margin-top:5px}input[type=button].btn-block,input[type=reset].btn-block,input[type=submit].btn-block{width:100%}.fade{opacity:0;-webkit-transition:opacity .15s linear;-o-transition:opacity .15s linear;transition:opacity .15s linear}.fade.in{opacity:1}.collapse{display:none}.collapse.in{display:block}tr.collapse.in{display:table-row}tbody.collapse.in{display:table-row-group}.collapsing{position:relative;height:0;overflow:hidden;-webkit-transition-timing-function:ease;-o-transition-timing-function:ease;transition-timing-function:ease;-webkit-transition-duration:.35s;-o-transition-duration:.35s;transition-duration:.35s;-webkit-transition-property:height,visibility;-o-transition-property:height,visibility;transition-property:height,visibility}.caret{display:inline-block;width:0;height:0;margin-left:2px;vertical-align:middle;border-top:4px dashed;border-top:4px solid\9;border-right:4px solid transparent;border-left:4px solid transparent}.dropdown,.dropup{position:relative}.dropdown-toggle:focus{outline:0}.dropdown-menu{position:absolute;top:100%;left:0;z-index:1000;display:none;float:left;min-width:160px;padding:5px 0;margin:2px 0 0;font-size:14px;text-align:left;list-style:none;background-color:#fff;-webkit-background-clip:padding-box;background-clip:padding-box;border:1px solid #ccc;border:1px solid rgba(0,0,0,.15);border-radius:4px;-webkit-box-shadow:0 6px 12px rgba(0,0,0,.175);box-shadow:0 6px 12px rgba(0,0,0,.175)}.dropdown-menu.pull-right{right:0;left:auto}.dropdown-menu .divider{height:1px;margin:9px 0;overflow:hidden;background-color:#e5e5e5}.dropdown-menu>li>a{display:block;padding:3px 20px;clear:both;font-weight:400;line-height:1.42857143;color:#333;white-space:nowrap}.dropdown-menu>li>a:focus,.dropdown-menu>li>a:hover{color:#262626;text-decoration:none;background-color:#f5f5f5}.dropdown-menu>.active>a,.dropdown-menu>.active>a:focus,.dropdown-menu>.active>a:hover{color:#fff;text-decoration:none;background-color:#337ab7;outline:0}.dropdown-menu>.disabled>a,.dropdown-menu>.disabled>a:focus,.dropdown-menu>.disabled>a:hover{color:#777}.dropdown-menu>.disabled>a:focus,.dropdown-menu>.disabled>a:hover{text-decoration:none;cursor:not-allowed;background-color:transparent;background-image:none;filter:progid:DXImageTransform.Microsoft.gradient(enabled=false)}.open>.dropdown-menu{display:block}.open>a{outline:0}.dropdown-menu-right{right:0;left:auto}.dropdown-menu-left{right:auto;left:0}.dropdown-header{display:block;padding:3px 20px;font-size:12px;line-height:1.42857143;color:#777;white-space:nowrap}.dropdown-backdrop{position:fixed;top:0;right:0;bottom:0;left:0;z-index:990}.pull-right>.dropdown-menu{right:0;left:auto}.dropup .caret,.navbar-fixed-bottom .dropdown .caret{content:"";border-top:0;border-bottom:4px dashed;border-bottom:4px solid\9}.dropup .dropdown-menu,.navbar-fixed-bottom .dropdown .dropdown-menu{top:auto;bottom:100%;margin-bottom:2px}@media (min-width:768px){.navbar-right .dropdown-menu{right:0;left:auto}.navbar-right .dropdown-menu-left{right:auto;left:0}}.btn-group,.btn-group-vertical{position:relative;display:inline-block;vertical-align:middle}.btn-group-vertical>.btn,.btn-group>.btn{position:relative;float:left}.btn-group-vertical>.btn.active,.btn-group-vertical>.btn:active,.btn-group-vertical>.btn:focus,.btn-group-vertical>.btn:hover,.btn-group>.btn.active,.btn-group>.btn:active,.btn-group>.btn:focus,.btn-group>.btn:hover{z-index:2}.btn-group .btn+.btn,.btn-group .btn+.btn-group,.btn-group .btn-group+.btn,.btn-group .btn-group+.btn-group{margin-left:-1px}.btn-toolbar{margin-left:-5px}.btn-toolbar .btn,.btn-toolbar .btn-group,.btn-toolbar .input-group{float:left}.btn-toolbar>.btn,.btn-toolbar>.btn-group,.btn-toolbar>.input-group{margin-left:5px}.btn-group>.btn:not(:first-child):not(:last-child):not(.dropdown-toggle){border-radius:0}.btn-group>.btn:first-child{margin-left:0}.btn-group>.btn:first-child:not(:last-child):not(.dropdown-toggle){border-top-right-radius:0;border-bottom-right-radius:0}.btn-group>.btn:last-child:not(:first-child),.btn-group>.dropdown-toggle:not(:first-child){border-top-left-radius:0;border-bottom-left-radius:0}.btn-group>.btn-group{float:left}.btn-group>.btn-group:not(:first-child):not(:last-child)>.btn{border-radius:0}.btn-group>.btn-group:first-child:not(:last-child)>.btn:last-child,.btn-group>.btn-group:first-child:not(:last-child)>.dropdown-toggle{border-top-right-radius:0;border-bottom-right-radius:0}.btn-group>.btn-group:last-child:not(:first-child)>.btn:first-child{border-top-left-radius:0;border-bottom-left-radius:0}.btn-group .dropdown-toggle:active,.btn-group.open .dropdown-toggle{outline:0}.btn-group>.btn+.dropdown-toggle{padding-right:8px;padding-left:8px}.btn-group>.btn-lg+.dropdown-toggle{padding-right:12px;padding-left:12px}.btn-group.open .dropdown-toggle{-webkit-box-shadow:inset 0 3px 5px rgba(0,0,0,.125);box-shadow:inset 0 3px 5px rgba(0,0,0,.125)}.btn-group.open .dropdown-toggle.btn-link{-webkit-box-shadow:none;box-shadow:none}.btn .caret{margin-left:0}.btn-lg .caret{border-width:5px 5px 0;border-bottom-width:0}.dropup .btn-lg .caret{border-width:0 5px 5px}.btn-group-vertical>.btn,.btn-group-vertical>.btn-group,.btn-group-vertical>.btn-group>.btn{display:block;float:none;width:100%;max-width:100%}.btn-group-vertical>.btn-group>.btn{float:none}.btn-group-vertical>.btn+.btn,.btn-group-vertical>.btn+.btn-group,.btn-group-vertical>.btn-group+.btn,.btn-group-vertical>.btn-group+.btn-group{margin-top:-1px;margin-left:0}.btn-group-vertical>.btn:not(:first-child):not(:last-child){border-radius:0}.btn-group-vertical>.btn:first-child:not(:last-child){border-top-right-radius:4px;border-bottom-right-radius:0;border-bottom-left-radius:0}.btn-group-vertical>.btn:last-child:not(:first-child){border-top-left-radius:0;border-top-right-radius:0;border-bottom-left-radius:4px}.btn-group-vertical>.btn-group:not(:first-child):not(:last-child)>.btn{border-radius:0}.btn-group-vertical>.btn-group:first-child:not(:last-child)>.btn:last-child,.btn-group-vertical>.btn-group:first-child:not(:last-child)>.dropdown-toggle{border-bottom-right-radius:0;border-bottom-left-radius:0}.btn-group-vertical>.btn-group:last-child:not(:first-child)>.btn:first-child{border-top-left-radius:0;border-top-right-radius:0}.btn-group-justified{display:table;width:100%;table-layout:fixed;border-collapse:separate}.btn-group-justified>.btn,.btn-group-justified>.btn-group{display:table-cell;float:none;width:1%}.btn-group-justified>.btn-group .btn{width:100%}.btn-group-justified>.btn-group .dropdown-menu{left:auto}[data-toggle=buttons]>.btn input[type=checkbox],[data-toggle=buttons]>.btn input[type=radio],[data-toggle=buttons]>.btn-group>.btn input[type=checkbox],[data-toggle=buttons]>.btn-group>.btn input[type=radio]{position:absolute;clip:rect(0,0,0,0);pointer-events:none}.input-group{position:relative;display:table;border-collapse:separate}.input-group[class*=col-]{float:none;padding-right:0;padding-left:0}.input-group .form-control{position:relative;z-index:2;float:left;width:100%;margin-bottom:0}.input-group-lg>.form-control,.input-group-lg>.input-group-addon,.input-group-lg>.input-group-btn>.btn{height:46px;padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}select.input-group-lg>.form-control,select.input-group-lg>.input-group-addon,select.input-group-lg>.input-group-btn>.btn{height:46px;line-height:46px}select[multiple].input-group-lg>.form-control,select[multiple].input-group-lg>.input-group-addon,select[multiple].input-group-lg>.input-group-btn>.btn,textarea.input-group-lg>.form-control,textarea.input-group-lg>.input-group-addon,textarea.input-group-lg>.input-group-btn>.btn{height:auto}.input-group-sm>.form-control,.input-group-sm>.input-group-addon,.input-group-sm>.input-group-btn>.btn{height:30px;padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}select.input-group-sm>.form-control,select.input-group-sm>.input-group-addon,select.input-group-sm>.input-group-btn>.btn{height:30px;line-height:30px}select[multiple].input-group-sm>.form-control,select[multiple].input-group-sm>.input-group-addon,select[multiple].input-group-sm>.input-group-btn>.btn,textarea.input-group-sm>.form-control,textarea.input-group-sm>.input-group-addon,textarea.input-group-sm>.input-group-btn>.btn{height:auto}.input-group .form-control,.input-group-addon,.input-group-btn{display:table-cell}.input-group .form-control:not(:first-child):not(:last-child),.input-group-addon:not(:first-child):not(:last-child),.input-group-btn:not(:first-child):not(:last-child){border-radius:0}.input-group-addon,.input-group-btn{width:1%;white-space:nowrap;vertical-align:middle}.input-group-addon{padding:6px 12px;font-size:14px;font-weight:400;line-height:1;color:#555;text-align:center;background-color:#eee;border:1px solid #ccc;border-radius:4px}.input-group-addon.input-sm{padding:5px 10px;font-size:12px;border-radius:3px}.input-group-addon.input-lg{padding:10px 16px;font-size:18px;border-radius:6px}.input-group-addon input[type=checkbox],.input-group-addon input[type=radio]{margin-top:0}.input-group .form-control:first-child,.input-group-addon:first-child,.input-group-btn:first-child>.btn,.input-group-btn:first-child>.btn-group>.btn,.input-group-btn:first-child>.dropdown-toggle,.input-group-btn:last-child>.btn-group:not(:last-child)>.btn,.input-group-btn:last-child>.btn:not(:last-child):not(.dropdown-toggle){border-top-right-radius:0;border-bottom-right-radius:0}.input-group-addon:first-child{border-right:0}.input-group .form-control:last-child,.input-group-addon:last-child,.input-group-btn:first-child>.btn-group:not(:first-child)>.btn,.input-group-btn:first-child>.btn:not(:first-child),.input-group-btn:last-child>.btn,.input-group-btn:last-child>.btn-group>.btn,.input-group-btn:last-child>.dropdown-toggle{border-top-left-radius:0;border-bottom-left-radius:0}.input-group-addon:last-child{border-left:0}.input-group-btn{position:relative;font-size:0;white-space:nowrap}.input-group-btn>.btn{position:relative}.input-group-btn>.btn+.btn{margin-left:-1px}.input-group-btn>.btn:active,.input-group-btn>.btn:focus,.input-group-btn>.btn:hover{z-index:2}.input-group-btn:first-child>.btn,.input-group-btn:first-child>.btn-group{margin-right:-1px}.input-group-btn:last-child>.btn,.input-group-btn:last-child>.btn-group{z-index:2;margin-left:-1px}.nav{padding-left:0;margin-bottom:0;list-style:none}.nav>li{position:relative;display:block}.nav>li>a{position:relative;display:block;padding:10px 15px}.nav>li>a:focus,.nav>li>a:hover{text-decoration:none;background-color:#eee}.nav>li.disabled>a{color:#777}.nav>li.disabled>a:focus,.nav>li.disabled>a:hover{color:#777;text-decoration:none;cursor:not-allowed;background-color:transparent}.nav .open>a,.nav .open>a:focus,.nav .open>a:hover{background-color:#eee;border-color:#337ab7}.nav .nav-divider{height:1px;margin:9px 0;overflow:hidden;background-color:#e5e5e5}.nav>li>a>img{max-width:none}.nav-tabs{border-bottom:1px solid #ddd}.nav-tabs>li{float:left;margin-bottom:-1px}.nav-tabs>li>a{margin-right:2px;line-height:1.42857143;border:1px solid transparent;border-radius:4px 4px 0 0}.nav-tabs>li>a:hover{border-color:#eee #eee #ddd}.nav-tabs>li.active>a,.nav-tabs>li.active>a:focus,.nav-tabs>li.active>a:hover{color:#555;cursor:default;background-color:#fff;border:1px solid #ddd;border-bottom-color:transparent}.nav-tabs.nav-justified{width:100%;border-bottom:0}.nav-tabs.nav-justified>li{float:none}.nav-tabs.nav-justified>li>a{margin-bottom:5px;text-align:center}.nav-tabs.nav-justified>.dropdown .dropdown-menu{top:auto;left:auto}@media (min-width:768px){.nav-tabs.nav-justified>li{display:table-cell;width:1%}.nav-tabs.nav-justified>li>a{margin-bottom:0}}.nav-tabs.nav-justified>li>a{margin-right:0;border-radius:4px}.nav-tabs.nav-justified>.active>a,.nav-tabs.nav-justified>.active>a:focus,.nav-tabs.nav-justified>.active>a:hover{border:1px solid #ddd}@media (min-width:768px){.nav-tabs.nav-justified>li>a{border-bottom:1px solid #ddd;border-radius:4px 4px 0 0}.nav-tabs.nav-justified>.active>a,.nav-tabs.nav-justified>.active>a:focus,.nav-tabs.nav-justified>.active>a:hover{border-bottom-color:#fff}}.nav-pills>li{float:left}.nav-pills>li>a{border-radius:4px}.nav-pills>li+li{margin-left:2px}.nav-pills>li.active>a,.nav-pills>li.active>a:focus,.nav-pills>li.active>a:hover{color:#fff;background-color:#337ab7}.nav-stacked>li{float:none}.nav-stacked>li+li{margin-top:2px;margin-left:0}.nav-justified{width:100%}.nav-justified>li{float:none}.nav-justified>li>a{margin-bottom:5px;text-align:center}.nav-justified>.dropdown .dropdown-menu{top:auto;left:auto}@media (min-width:768px){.nav-justified>li{display:table-cell;width:1%}.nav-justified>li>a{margin-bottom:0}}.nav-tabs-justified{border-bottom:0}.nav-tabs-justified>li>a{margin-right:0;border-radius:4px}.nav-tabs-justified>.active>a,.nav-tabs-justified>.active>a:focus,.nav-tabs-justified>.active>a:hover{border:1px solid #ddd}@media (min-width:768px){.nav-tabs-justified>li>a{border-bottom:1px solid #ddd;border-radius:4px 4px 0 0}.nav-tabs-justified>.active>a,.nav-tabs-justified>.active>a:focus,.nav-tabs-justified>.active>a:hover{border-bottom-color:#fff}}.tab-content>.tab-pane{display:none}.tab-content>.active{display:block}.nav-tabs .dropdown-menu{margin-top:-1px;border-top-left-radius:0;border-top-right-radius:0}.navbar{position:relative;min-height:50px;margin-bottom:20px;border:1px solid transparent}@media (min-width:768px){.navbar{border-radius:4px}}@media (min-width:768px){.navbar-header{float:left}}.navbar-collapse{padding-right:15px;padding-left:15px;overflow-x:visible;-webkit-overflow-scrolling:touch;border-top:1px solid transparent;-webkit-box-shadow:inset 0 1px 0 rgba(255,255,255,.1);box-shadow:inset 0 1px 0 rgba(255,255,255,.1)}.navbar-collapse.in{overflow-y:auto}@media (min-width:768px){.navbar-collapse{width:auto;border-top:0;-webkit-box-shadow:none;box-shadow:none}.navbar-collapse.collapse{display:block!important;height:auto!important;padding-bottom:0;overflow:visible!important}.navbar-collapse.in{overflow-y:visible}.navbar-fixed-bottom .navbar-collapse,.navbar-fixed-top .navbar-collapse,.navbar-static-top .navbar-collapse{padding-right:0;padding-left:0}}.navbar-fixed-bottom .navbar-collapse,.navbar-fixed-top .navbar-collapse{max-height:340px}@media (max-device-width:480px) and (orientation:landscape){.navbar-fixed-bottom .navbar-collapse,.navbar-fixed-top .navbar-collapse{max-height:200px}}.container-fluid>.navbar-collapse,.container-fluid>.navbar-header,.container>.navbar-collapse,.container>.navbar-header{margin-right:-15px;margin-left:-15px}@media (min-width:768px){.container-fluid>.navbar-collapse,.container-fluid>.navbar-header,.container>.navbar-collapse,.container>.navbar-header{margin-right:0;margin-left:0}}.navbar-static-top{z-index:1000;border-width:0 0 1px}@media (min-width:768px){.navbar-static-top{border-radius:0}}.navbar-fixed-bottom,.navbar-fixed-top{position:fixed;right:0;left:0;z-index:1030}@media (min-width:768px){.navbar-fixed-bottom,.navbar-fixed-top{border-radius:0}}.navbar-fixed-top{top:0;border-width:0 0 1px}.navbar-fixed-bottom{bottom:0;margin-bottom:0;border-width:1px 0 0}.navbar-brand{float:left;height:50px;padding:15px 15px;font-size:18px;line-height:20px}.navbar-brand:focus,.navbar-brand:hover{text-decoration:none}.navbar-brand>img{display:block}@media (min-width:768px){.navbar>.container .navbar-brand,.navbar>.container-fluid .navbar-brand{margin-left:-15px}}.navbar-toggle{position:relative;float:right;padding:9px 10px;margin-top:8px;margin-right:15px;margin-bottom:8px;background-color:transparent;background-image:none;border:1px solid transparent;border-radius:4px}.navbar-toggle:focus{outline:0}.navbar-toggle .icon-bar{display:block;width:22px;height:2px;border-radius:1px}.navbar-toggle .icon-bar+.icon-bar{margin-top:4px}@media (min-width:768px){.navbar-toggle{display:none}}.navbar-nav{margin:7.5px -15px}.navbar-nav>li>a{padding-top:10px;padding-bottom:10px;line-height:20px}@media (max-width:767px){.navbar-nav .open .dropdown-menu{position:static;float:none;width:auto;margin-top:0;background-color:transparent;border:0;-webkit-box-shadow:none;box-shadow:none}.navbar-nav .open .dropdown-menu .dropdown-header,.navbar-nav .open .dropdown-menu>li>a{padding:5px 15px 5px 25px}.navbar-nav .open .dropdown-menu>li>a{line-height:20px}.navbar-nav .open .dropdown-menu>li>a:focus,.navbar-nav .open .dropdown-menu>li>a:hover{background-image:none}}@media (min-width:768px){.navbar-nav{float:left;margin:0}.navbar-nav>li{float:left}.navbar-nav>li>a{padding-top:15px;padding-bottom:15px}}.navbar-form{padding:10px 15px;margin-top:8px;margin-right:-15px;margin-bottom:8px;margin-left:-15px;border-top:1px solid transparent;border-bottom:1px solid transparent;-webkit-box-shadow:inset 0 1px 0 rgba(255,255,255,.1),0 1px 0 rgba(255,255,255,.1);box-shadow:inset 0 1px 0 rgba(255,255,255,.1),0 1px 0 rgba(255,255,255,.1)}@media (min-width:768px){.navbar-form .form-group{display:inline-block;margin-bottom:0;vertical-align:middle}.navbar-form .form-control{display:inline-block;width:auto;vertical-align:middle}.navbar-form .form-control-static{display:inline-block}.navbar-form .input-group{display:inline-table;vertical-align:middle}.navbar-form .input-group .form-control,.navbar-form .input-group .input-group-addon,.navbar-form .input-group .input-group-btn{width:auto}.navbar-form .input-group>.form-control{width:100%}.navbar-form .control-label{margin-bottom:0;vertical-align:middle}.navbar-form .checkbox,.navbar-form .radio{display:inline-block;margin-top:0;margin-bottom:0;vertical-align:middle}.navbar-form .checkbox label,.navbar-form .radio label{padding-left:0}.navbar-form .checkbox input[type=checkbox],.navbar-form .radio input[type=radio]{position:relative;margin-left:0}.navbar-form .has-feedback .form-control-feedback{top:0}}@media (max-width:767px){.navbar-form .form-group{margin-bottom:5px}.navbar-form .form-group:last-child{margin-bottom:0}}@media (min-width:768px){.navbar-form{width:auto;padding-top:0;padding-bottom:0;margin-right:0;margin-left:0;border:0;-webkit-box-shadow:none;box-shadow:none}}.navbar-nav>li>.dropdown-menu{margin-top:0;border-top-left-radius:0;border-top-right-radius:0}.navbar-fixed-bottom .navbar-nav>li>.dropdown-menu{margin-bottom:0;border-top-left-radius:4px;border-top-right-radius:4px;border-bottom-right-radius:0;border-bottom-left-radius:0}.navbar-btn{margin-top:8px;margin-bottom:8px}.navbar-btn.btn-sm{margin-top:10px;margin-bottom:10px}.navbar-btn.btn-xs{margin-top:14px;margin-bottom:14px}.navbar-text{margin-top:15px;margin-bottom:15px}@media (min-width:768px){.navbar-text{float:left;margin-right:15px;margin-left:15px}}@media (min-width:768px){.navbar-left{float:left!important}.navbar-right{float:right!important;margin-right:-15px}.navbar-right~.navbar-right{margin-right:0}}.navbar-default{background-color:#f8f8f8;border-color:#e7e7e7}.navbar-default .navbar-brand{color:#777}.navbar-default .navbar-brand:focus,.navbar-default .navbar-brand:hover{color:#5e5e5e;background-color:transparent}.navbar-default .navbar-text{color:#777}.navbar-default .navbar-nav>li>a{color:#777}.navbar-default .navbar-nav>li>a:focus,.navbar-default .navbar-nav>li>a:hover{color:#333;background-color:transparent}.navbar-default .navbar-nav>.active>a,.navbar-default .navbar-nav>.active>a:focus,.navbar-default .navbar-nav>.active>a:hover{color:#555;background-color:#e7e7e7}.navbar-default .navbar-nav>.disabled>a,.navbar-default .navbar-nav>.disabled>a:focus,.navbar-default .navbar-nav>.disabled>a:hover{color:#ccc;background-color:transparent}.navbar-default .navbar-toggle{border-color:#ddd}.navbar-default .navbar-toggle:focus,.navbar-default .navbar-toggle:hover{background-color:#ddd}.navbar-default .navbar-toggle .icon-bar{background-color:#888}.navbar-default .navbar-collapse,.navbar-default .navbar-form{border-color:#e7e7e7}.navbar-default .navbar-nav>.open>a,.navbar-default .navbar-nav>.open>a:focus,.navbar-default .navbar-nav>.open>a:hover{color:#555;background-color:#e7e7e7}@media (max-width:767px){.navbar-default .navbar-nav .open .dropdown-menu>li>a{color:#777}.navbar-default .navbar-nav .open .dropdown-menu>li>a:focus,.navbar-default .navbar-nav .open .dropdown-menu>li>a:hover{color:#333;background-color:transparent}.navbar-default .navbar-nav .open .dropdown-menu>.active>a,.navbar-default .navbar-nav .open .dropdown-menu>.active>a:focus,.navbar-default .navbar-nav .open .dropdown-menu>.active>a:hover{color:#555;background-color:#e7e7e7}.navbar-default .navbar-nav .open .dropdown-menu>.disabled>a,.navbar-default .navbar-nav .open .dropdown-menu>.disabled>a:focus,.navbar-default .navbar-nav .open .dropdown-menu>.disabled>a:hover{color:#ccc;background-color:transparent}}.navbar-default .navbar-link{color:#777}.navbar-default .navbar-link:hover{color:#333}.navbar-default .btn-link{color:#777}.navbar-default .btn-link:focus,.navbar-default .btn-link:hover{color:#333}.navbar-default .btn-link[disabled]:focus,.navbar-default .btn-link[disabled]:hover,fieldset[disabled] .navbar-default .btn-link:focus,fieldset[disabled] .navbar-default .btn-link:hover{color:#ccc}.navbar-inverse{background-color:#222;border-color:#080808}.navbar-inverse .navbar-brand{color:#9d9d9d}.navbar-inverse .navbar-brand:focus,.navbar-inverse .navbar-brand:hover{color:#fff;background-color:transparent}.navbar-inverse .navbar-text{color:#9d9d9d}.navbar-inverse .navbar-nav>li>a{color:#9d9d9d}.navbar-inverse .navbar-nav>li>a:focus,.navbar-inverse .navbar-nav>li>a:hover{color:#fff;background-color:transparent}.navbar-inverse .navbar-nav>.active>a,.navbar-inverse .navbar-nav>.active>a:focus,.navbar-inverse .navbar-nav>.active>a:hover{color:#fff;background-color:#080808}.navbar-inverse .navbar-nav>.disabled>a,.navbar-inverse .navbar-nav>.disabled>a:focus,.navbar-inverse .navbar-nav>.disabled>a:hover{color:#444;background-color:transparent}.navbar-inverse .navbar-toggle{border-color:#333}.navbar-inverse .navbar-toggle:focus,.navbar-inverse .navbar-toggle:hover{background-color:#333}.navbar-inverse .navbar-toggle .icon-bar{background-color:#fff}.navbar-inverse .navbar-collapse,.navbar-inverse .navbar-form{border-color:#101010}.navbar-inverse .navbar-nav>.open>a,.navbar-inverse .navbar-nav>.open>a:focus,.navbar-inverse .navbar-nav>.open>a:hover{color:#fff;background-color:#080808}@media (max-width:767px){.navbar-inverse .navbar-nav .open .dropdown-menu>.dropdown-header{border-color:#080808}.navbar-inverse .navbar-nav .open .dropdown-menu .divider{background-color:#080808}.navbar-inverse .navbar-nav .open .dropdown-menu>li>a{color:#9d9d9d}.navbar-inverse .navbar-nav .open .dropdown-menu>li>a:focus,.navbar-inverse .navbar-nav .open .dropdown-menu>li>a:hover{color:#fff;background-color:transparent}.navbar-inverse .navbar-nav .open .dropdown-menu>.active>a,.navbar-inverse .navbar-nav .open .dropdown-menu>.active>a:focus,.navbar-inverse .navbar-nav .open .dropdown-menu>.active>a:hover{color:#fff;background-color:#080808}.navbar-inverse .navbar-nav .open .dropdown-menu>.disabled>a,.navbar-inverse .navbar-nav .open .dropdown-menu>.disabled>a:focus,.navbar-inverse .navbar-nav .open .dropdown-menu>.disabled>a:hover{color:#444;background-color:transparent}}.navbar-inverse .navbar-link{color:#9d9d9d}.navbar-inverse .navbar-link:hover{color:#fff}.navbar-inverse .btn-link{color:#9d9d9d}.navbar-inverse .btn-link:focus,.navbar-inverse .btn-link:hover{color:#fff}.navbar-inverse .btn-link[disabled]:focus,.navbar-inverse .btn-link[disabled]:hover,fieldset[disabled] .navbar-inverse .btn-link:focus,fieldset[disabled] .navbar-inverse .btn-link:hover{color:#444}.breadcrumb{padding:8px 15px;margin-bottom:20px;list-style:none;background-color:#f5f5f5;border-radius:4px}.breadcrumb>li{display:inline-block}.breadcrumb>li+li:before{padding:0 5px;color:#ccc;content:"/\00a0"}.breadcrumb>.active{color:#777}.pagination{display:inline-block;padding-left:0;margin:20px 0;border-radius:4px}.pagination>li{display:inline}.pagination>li>a,.pagination>li>span{position:relative;float:left;padding:6px 12px;margin-left:-1px;line-height:1.42857143;color:#337ab7;text-decoration:none;background-color:#fff;border:1px solid #ddd}.pagination>li:first-child>a,.pagination>li:first-child>span{margin-left:0;border-top-left-radius:4px;border-bottom-left-radius:4px}.pagination>li:last-child>a,.pagination>li:last-child>span{border-top-right-radius:4px;border-bottom-right-radius:4px}.pagination>li>a:focus,.pagination>li>a:hover,.pagination>li>span:focus,.pagination>li>span:hover{z-index:3;color:#23527c;background-color:#eee;border-color:#ddd}.pagination>.active>a,.pagination>.active>a:focus,.pagination>.active>a:hover,.pagination>.active>span,.pagination>.active>span:focus,.pagination>.active>span:hover{z-index:2;color:#fff;cursor:default;background-color:#337ab7;border-color:#337ab7}.pagination>.disabled>a,.pagination>.disabled>a:focus,.pagination>.disabled>a:hover,.pagination>.disabled>span,.pagination>.disabled>span:focus,.pagination>.disabled>span:hover{color:#777;cursor:not-allowed;background-color:#fff;border-color:#ddd}.pagination-lg>li>a,.pagination-lg>li>span{padding:10px 16px;font-size:18px;line-height:1.3333333}.pagination-lg>li:first-child>a,.pagination-lg>li:first-child>span{border-top-left-radius:6px;border-bottom-left-radius:6px}.pagination-lg>li:last-child>a,.pagination-lg>li:last-child>span{border-top-right-radius:6px;border-bottom-right-radius:6px}.pagination-sm>li>a,.pagination-sm>li>span{padding:5px 10px;font-size:12px;line-height:1.5}.pagination-sm>li:first-child>a,.pagination-sm>li:first-child>span{border-top-left-radius:3px;border-bottom-left-radius:3px}.pagination-sm>li:last-child>a,.pagination-sm>li:last-child>span{border-top-right-radius:3px;border-bottom-right-radius:3px}.pager{padding-left:0;margin:20px 0;text-align:center;list-style:none}.pager li{display:inline}.pager li>a,.pager li>span{display:inline-block;padding:5px 14px;background-color:#fff;border:1px solid #ddd;border-radius:15px}.pager li>a:focus,.pager li>a:hover{text-decoration:none;background-color:#eee}.pager .next>a,.pager .next>span{float:right}.pager .previous>a,.pager .previous>span{float:left}.pager .disabled>a,.pager .disabled>a:focus,.pager .disabled>a:hover,.pager .disabled>span{color:#777;cursor:not-allowed;background-color:#fff}.label{display:inline;padding:.2em .6em .3em;font-size:75%;font-weight:700;line-height:1;color:#fff;text-align:center;white-space:nowrap;vertical-align:baseline;border-radius:.25em}a.label:focus,a.label:hover{color:#fff;text-decoration:none;cursor:pointer}.label:empty{display:none}.btn .label{position:relative;top:-1px}.label-default{background-color:#777}.label-default[href]:focus,.label-default[href]:hover{background-color:#5e5e5e}.label-primary{background-color:#337ab7}.label-primary[href]:focus,.label-primary[href]:hover{background-color:#286090}.label-success{background-color:#5cb85c}.label-success[href]:focus,.label-success[href]:hover{background-color:#449d44}.label-info{background-color:#5bc0de}.label-info[href]:focus,.label-info[href]:hover{background-color:#31b0d5}.label-warning{background-color:#f0ad4e}.label-warning[href]:focus,.label-warning[href]:hover{background-color:#ec971f}.label-danger{background-color:#d9534f}.label-danger[href]:focus,.label-danger[href]:hover{background-color:#c9302c}.badge{display:inline-block;min-width:10px;padding:3px 7px;font-size:12px;font-weight:700;line-height:1;color:#fff;text-align:center;white-space:nowrap;vertical-align:middle;background-color:#777;border-radius:10px}.badge:empty{display:none}.btn .badge{position:relative;top:-1px}.btn-group-xs>.btn .badge,.btn-xs .badge{top:0;padding:1px 5px}a.badge:focus,a.badge:hover{color:#fff;text-decoration:none;cursor:pointer}.list-group-item.active>.badge,.nav-pills>.active>a>.badge{color:#337ab7;background-color:#fff}.list-group-item>.badge{float:right}.list-group-item>.badge+.badge{margin-right:5px}.nav-pills>li>a>.badge{margin-left:3px}.jumbotron{padding-top:30px;padding-bottom:30px;margin-bottom:30px;color:inherit;background-color:#eee}.jumbotron .h1,.jumbotron h1{color:inherit}.jumbotron p{margin-bottom:15px;font-size:21px;font-weight:200}.jumbotron>hr{border-top-color:#d5d5d5}.container .jumbotron,.container-fluid .jumbotron{border-radius:6px}.jumbotron .container{max-width:100%}@media screen and (min-width:768px){.jumbotron{padding-top:48px;padding-bottom:48px}.container .jumbotron,.container-fluid .jumbotron{padding-right:60px;padding-left:60px}.jumbotron .h1,.jumbotron h1{font-size:63px}}.thumbnail{display:block;padding:4px;margin-bottom:20px;line-height:1.42857143;background-color:#fff;border:1px solid #ddd;border-radius:4px;-webkit-transition:border .2s ease-in-out;-o-transition:border .2s ease-in-out;transition:border .2s ease-in-out}.thumbnail a>img,.thumbnail>img{margin-right:auto;margin-left:auto}a.thumbnail.active,a.thumbnail:focus,a.thumbnail:hover{border-color:#337ab7}.thumbnail .caption{padding:9px;color:#333}.alert{padding:15px;margin-bottom:20px;border:1px solid transparent;border-radius:4px}.alert h4{margin-top:0;color:inherit}.alert .alert-link{font-weight:700}.alert>p,.alert>ul{margin-bottom:0}.alert>p+p{margin-top:5px}.alert-dismissable,.alert-dismissible{padding-right:35px}.alert-dismissable .close,.alert-dismissible .close{position:relative;top:-2px;right:-21px;color:inherit}.alert-success{color:#3c763d;background-color:#dff0d8;border-color:#d6e9c6}.alert-success hr{border-top-color:#c9e2b3}.alert-success .alert-link{color:#2b542c}.alert-info{color:#31708f;background-color:#d9edf7;border-color:#bce8f1}.alert-info hr{border-top-color:#a6e1ec}.alert-info .alert-link{color:#245269}.alert-warning{color:#8a6d3b;background-color:#fcf8e3;border-color:#faebcc}.alert-warning hr{border-top-color:#f7e1b5}.alert-warning .alert-link{color:#66512c}.alert-danger{color:#a94442;background-color:#f2dede;border-color:#ebccd1}.alert-danger hr{border-top-color:#e4b9c0}.alert-danger .alert-link{color:#843534}@-webkit-keyframes progress-bar-stripes{from{background-position:40px 0}to{background-position:0 0}}@-o-keyframes progress-bar-stripes{from{background-position:40px 0}to{background-position:0 0}}@keyframes progress-bar-stripes{from{background-position:40px 0}to{background-position:0 0}}.progress{height:20px;margin-bottom:20px;overflow:hidden;background-color:#f5f5f5;border-radius:4px;-webkit-box-shadow:inset 0 1px 2px rgba(0,0,0,.1);box-shadow:inset 0 1px 2px rgba(0,0,0,.1)}.progress-bar{float:left;width:0;height:100%;font-size:12px;line-height:20px;color:#fff;text-align:center;background-color:#337ab7;-webkit-box-shadow:inset 0 -1px 0 rgba(0,0,0,.15);box-shadow:inset 0 -1px 0 rgba(0,0,0,.15);-webkit-transition:width .6s ease;-o-transition:width .6s ease;transition:width .6s ease}.progress-bar-striped,.progress-striped .progress-bar{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);-webkit-background-size:40px 40px;background-size:40px 40px}.progress-bar.active,.progress.active .progress-bar{-webkit-animation:progress-bar-stripes 2s linear infinite;-o-animation:progress-bar-stripes 2s linear infinite;animation:progress-bar-stripes 2s linear infinite}.progress-bar-success{background-color:#5cb85c}.progress-striped .progress-bar-success{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.progress-bar-info{background-color:#5bc0de}.progress-striped .progress-bar-info{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.progress-bar-warning{background-color:#f0ad4e}.progress-striped .progress-bar-warning{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.progress-bar-danger{background-color:#d9534f}.progress-striped .progress-bar-danger{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.media{margin-top:15px}.media:first-child{margin-top:0}.media,.media-body{overflow:hidden;zoom:1}.media-body{width:10000px}.media-object{display:block}.media-object.img-thumbnail{max-width:none}.media-right,.media>.pull-right{padding-left:10px}.media-left,.media>.pull-left{padding-right:10px}.media-body,.media-left,.media-right{display:table-cell;vertical-align:top}.media-middle{vertical-align:middle}.media-bottom{vertical-align:bottom}.media-heading{margin-top:0;margin-bottom:5px}.media-list{padding-left:0;list-style:none}.list-group{padding-left:0;margin-bottom:20px}.list-group-item{position:relative;display:block;padding:10px 15px;margin-bottom:-1px;background-color:#fff;border:1px solid #ddd}.list-group-item:first-child{border-top-left-radius:4px;border-top-right-radius:4px}.list-group-item:last-child{margin-bottom:0;border-bottom-right-radius:4px;border-bottom-left-radius:4px}a.list-group-item,button.list-group-item{color:#555}a.list-group-item .list-group-item-heading,button.list-group-item .list-group-item-heading{color:#333}a.list-group-item:focus,a.list-group-item:hover,button.list-group-item:focus,button.list-group-item:hover{color:#555;text-decoration:none;background-color:#f5f5f5}button.list-group-item{width:100%;text-align:left}.list-group-item.disabled,.list-group-item.disabled:focus,.list-group-item.disabled:hover{color:#777;cursor:not-allowed;background-color:#eee}.list-group-item.disabled .list-group-item-heading,.list-group-item.disabled:focus .list-group-item-heading,.list-group-item.disabled:hover .list-group-item-heading{color:inherit}.list-group-item.disabled .list-group-item-text,.list-group-item.disabled:focus .list-group-item-text,.list-group-item.disabled:hover .list-group-item-text{color:#777}.list-group-item.active,.list-group-item.active:focus,.list-group-item.active:hover{z-index:2;color:#fff;background-color:#337ab7;border-color:#337ab7}.list-group-item.active .list-group-item-heading,.list-group-item.active .list-group-item-heading>.small,.list-group-item.active .list-group-item-heading>small,.list-group-item.active:focus .list-group-item-heading,.list-group-item.active:focus .list-group-item-heading>.small,.list-group-item.active:focus .list-group-item-heading>small,.list-group-item.active:hover .list-group-item-heading,.list-group-item.active:hover .list-group-item-heading>.small,.list-group-item.active:hover .list-group-item-heading>small{color:inherit}.list-group-item.active .list-group-item-text,.list-group-item.active:focus .list-group-item-text,.list-group-item.active:hover .list-group-item-text{color:#c7ddef}.list-group-item-success{color:#3c763d;background-color:#dff0d8}a.list-group-item-success,button.list-group-item-success{color:#3c763d}a.list-group-item-success .list-group-item-heading,button.list-group-item-success .list-group-item-heading{color:inherit}a.list-group-item-success:focus,a.list-group-item-success:hover,button.list-group-item-success:focus,button.list-group-item-success:hover{color:#3c763d;background-color:#d0e9c6}a.list-group-item-success.active,a.list-group-item-success.active:focus,a.list-group-item-success.active:hover,button.list-group-item-success.active,button.list-group-item-success.active:focus,button.list-group-item-success.active:hover{color:#fff;background-color:#3c763d;border-color:#3c763d}.list-group-item-info{color:#31708f;background-color:#d9edf7}a.list-group-item-info,button.list-group-item-info{color:#31708f}a.list-group-item-info .list-group-item-heading,button.list-group-item-info .list-group-item-heading{color:inherit}a.list-group-item-info:focus,a.list-group-item-info:hover,button.list-group-item-info:focus,button.list-group-item-info:hover{color:#31708f;background-color:#c4e3f3}a.list-group-item-info.active,a.list-group-item-info.active:focus,a.list-group-item-info.active:hover,button.list-group-item-info.active,button.list-group-item-info.active:focus,button.list-group-item-info.active:hover{color:#fff;background-color:#31708f;border-color:#31708f}.list-group-item-warning{color:#8a6d3b;background-color:#fcf8e3}a.list-group-item-warning,button.list-group-item-warning{color:#8a6d3b}a.list-group-item-warning .list-group-item-heading,button.list-group-item-warning .list-group-item-heading{color:inherit}a.list-group-item-warning:focus,a.list-group-item-warning:hover,button.list-group-item-warning:focus,button.list-group-item-warning:hover{color:#8a6d3b;background-color:#faf2cc}a.list-group-item-warning.active,a.list-group-item-warning.active:focus,a.list-group-item-warning.active:hover,button.list-group-item-warning.active,button.list-group-item-warning.active:focus,button.list-group-item-warning.active:hover{color:#fff;background-color:#8a6d3b;border-color:#8a6d3b}.list-group-item-danger{color:#a94442;background-color:#f2dede}a.list-group-item-danger,button.list-group-item-danger{color:#a94442}a.list-group-item-danger .list-group-item-heading,button.list-group-item-danger .list-group-item-heading{color:inherit}a.list-group-item-danger:focus,a.list-group-item-danger:hover,button.list-group-item-danger:focus,button.list-group-item-danger:hover{color:#a94442;background-color:#ebcccc}a.list-group-item-danger.active,a.list-group-item-danger.active:focus,a.list-group-item-danger.active:hover,button.list-group-item-danger.active,button.list-group-item-danger.active:focus,button.list-group-item-danger.active:hover{color:#fff;background-color:#a94442;border-color:#a94442}.list-group-item-heading{margin-top:0;margin-bottom:5px}.list-group-item-text{margin-bottom:0;line-height:1.3}.panel{margin-bottom:20px;background-color:#fff;border:1px solid transparent;border-radius:4px;-webkit-box-shadow:0 1px 1px rgba(0,0,0,.05);box-shadow:0 1px 1px rgba(0,0,0,.05)}.panel-body{padding:15px}.panel-heading{padding:10px 15px;border-bottom:1px solid transparent;border-top-left-radius:3px;border-top-right-radius:3px}.panel-heading>.dropdown .dropdown-toggle{color:inherit}.panel-title{margin-top:0;margin-bottom:0;font-size:16px;color:inherit}.panel-title>.small,.panel-title>.small>a,.panel-title>a,.panel-title>small,.panel-title>small>a{color:inherit}.panel-footer{padding:10px 15px;background-color:#f5f5f5;border-top:1px solid #ddd;border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.list-group,.panel>.panel-collapse>.list-group{margin-bottom:0}.panel>.list-group .list-group-item,.panel>.panel-collapse>.list-group .list-group-item{border-width:1px 0;border-radius:0}.panel>.list-group:first-child .list-group-item:first-child,.panel>.panel-collapse>.list-group:first-child .list-group-item:first-child{border-top:0;border-top-left-radius:3px;border-top-right-radius:3px}.panel>.list-group:last-child .list-group-item:last-child,.panel>.panel-collapse>.list-group:last-child .list-group-item:last-child{border-bottom:0;border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.panel-heading+.panel-collapse>.list-group .list-group-item:first-child{border-top-left-radius:0;border-top-right-radius:0}.panel-heading+.list-group .list-group-item:first-child{border-top-width:0}.list-group+.panel-footer{border-top-width:0}.panel>.panel-collapse>.table,.panel>.table,.panel>.table-responsive>.table{margin-bottom:0}.panel>.panel-collapse>.table caption,.panel>.table caption,.panel>.table-responsive>.table caption{padding-right:15px;padding-left:15px}.panel>.table-responsive:first-child>.table:first-child,.panel>.table:first-child{border-top-left-radius:3px;border-top-right-radius:3px}.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child,.panel>.table:first-child>tbody:first-child>tr:first-child,.panel>.table:first-child>thead:first-child>tr:first-child{border-top-left-radius:3px;border-top-right-radius:3px}.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child td:first-child,.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child th:first-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child td:first-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child th:first-child,.panel>.table:first-child>tbody:first-child>tr:first-child td:first-child,.panel>.table:first-child>tbody:first-child>tr:first-child th:first-child,.panel>.table:first-child>thead:first-child>tr:first-child td:first-child,.panel>.table:first-child>thead:first-child>tr:first-child th:first-child{border-top-left-radius:3px}.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child td:last-child,.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child th:last-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child td:last-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child th:last-child,.panel>.table:first-child>tbody:first-child>tr:first-child td:last-child,.panel>.table:first-child>tbody:first-child>tr:first-child th:last-child,.panel>.table:first-child>thead:first-child>tr:first-child td:last-child,.panel>.table:first-child>thead:first-child>tr:first-child th:last-child{border-top-right-radius:3px}.panel>.table-responsive:last-child>.table:last-child,.panel>.table:last-child{border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child,.panel>.table:last-child>tbody:last-child>tr:last-child,.panel>.table:last-child>tfoot:last-child>tr:last-child{border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child td:first-child,.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child th:first-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child td:first-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child th:first-child,.panel>.table:last-child>tbody:last-child>tr:last-child td:first-child,.panel>.table:last-child>tbody:last-child>tr:last-child th:first-child,.panel>.table:last-child>tfoot:last-child>tr:last-child td:first-child,.panel>.table:last-child>tfoot:last-child>tr:last-child th:first-child{border-bottom-left-radius:3px}.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child td:last-child,.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child th:last-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child td:last-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child th:last-child,.panel>.table:last-child>tbody:last-child>tr:last-child td:last-child,.panel>.table:last-child>tbody:last-child>tr:last-child th:last-child,.panel>.table:last-child>tfoot:last-child>tr:last-child td:last-child,.panel>.table:last-child>tfoot:last-child>tr:last-child th:last-child{border-bottom-right-radius:3px}.panel>.panel-body+.table,.panel>.panel-body+.table-responsive,.panel>.table+.panel-body,.panel>.table-responsive+.panel-body{border-top:1px solid #ddd}.panel>.table>tbody:first-child>tr:first-child td,.panel>.table>tbody:first-child>tr:first-child th{border-top:0}.panel>.table-bordered,.panel>.table-responsive>.table-bordered{border:0}.panel>.table-bordered>tbody>tr>td:first-child,.panel>.table-bordered>tbody>tr>th:first-child,.panel>.table-bordered>tfoot>tr>td:first-child,.panel>.table-bordered>tfoot>tr>th:first-child,.panel>.table-bordered>thead>tr>td:first-child,.panel>.table-bordered>thead>tr>th:first-child,.panel>.table-responsive>.table-bordered>tbody>tr>td:first-child,.panel>.table-responsive>.table-bordered>tbody>tr>th:first-child,.panel>.table-responsive>.table-bordered>tfoot>tr>td:first-child,.panel>.table-responsive>.table-bordered>tfoot>tr>th:first-child,.panel>.table-responsive>.table-bordered>thead>tr>td:first-child,.panel>.table-responsive>.table-bordered>thead>tr>th:first-child{border-left:0}.panel>.table-bordered>tbody>tr>td:last-child,.panel>.table-bordered>tbody>tr>th:last-child,.panel>.table-bordered>tfoot>tr>td:last-child,.panel>.table-bordered>tfoot>tr>th:last-child,.panel>.table-bordered>thead>tr>td:last-child,.panel>.table-bordered>thead>tr>th:last-child,.panel>.table-responsive>.table-bordered>tbody>tr>td:last-child,.panel>.table-responsive>.table-bordered>tbody>tr>th:last-child,.panel>.table-responsive>.table-bordered>tfoot>tr>td:last-child,.panel>.table-responsive>.table-bordered>tfoot>tr>th:last-child,.panel>.table-responsive>.table-bordered>thead>tr>td:last-child,.panel>.table-responsive>.table-bordered>thead>tr>th:last-child{border-right:0}.panel>.table-bordered>tbody>tr:first-child>td,.panel>.table-bordered>tbody>tr:first-child>th,.panel>.table-bordered>thead>tr:first-child>td,.panel>.table-bordered>thead>tr:first-child>th,.panel>.table-responsive>.table-bordered>tbody>tr:first-child>td,.panel>.table-responsive>.table-bordered>tbody>tr:first-child>th,.panel>.table-responsive>.table-bordered>thead>tr:first-child>td,.panel>.table-responsive>.table-bordered>thead>tr:first-child>th{border-bottom:0}.panel>.table-bordered>tbody>tr:last-child>td,.panel>.table-bordered>tbody>tr:last-child>th,.panel>.table-bordered>tfoot>tr:last-child>td,.panel>.table-bordered>tfoot>tr:last-child>th,.panel>.table-responsive>.table-bordered>tbody>tr:last-child>td,.panel>.table-responsive>.table-bordered>tbody>tr:last-child>th,.panel>.table-responsive>.table-bordered>tfoot>tr:last-child>td,.panel>.table-responsive>.table-bordered>tfoot>tr:last-child>th{border-bottom:0}.panel>.table-responsive{margin-bottom:0;border:0}.panel-group{margin-bottom:20px}.panel-group .panel{margin-bottom:0;border-radius:4px}.panel-group .panel+.panel{margin-top:5px}.panel-group .panel-heading{border-bottom:0}.panel-group .panel-heading+.panel-collapse>.list-group,.panel-group .panel-heading+.panel-collapse>.panel-body{border-top:1px solid #ddd}.panel-group .panel-footer{border-top:0}.panel-group .panel-footer+.panel-collapse .panel-body{border-bottom:1px solid #ddd}.panel-default{border-color:#ddd}.panel-default>.panel-heading{color:#333;background-color:#f5f5f5;border-color:#ddd}.panel-default>.panel-heading+.panel-collapse>.panel-body{border-top-color:#ddd}.panel-default>.panel-heading .badge{color:#f5f5f5;background-color:#333}.panel-default>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#ddd}.panel-primary{border-color:#337ab7}.panel-primary>.panel-heading{color:#fff;background-color:#337ab7;border-color:#337ab7}.panel-primary>.panel-heading+.panel-collapse>.panel-body{border-top-color:#337ab7}.panel-primary>.panel-heading .badge{color:#337ab7;background-color:#fff}.panel-primary>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#337ab7}.panel-success{border-color:#d6e9c6}.panel-success>.panel-heading{color:#3c763d;background-color:#dff0d8;border-color:#d6e9c6}.panel-success>.panel-heading+.panel-collapse>.panel-body{border-top-color:#d6e9c6}.panel-success>.panel-heading .badge{color:#dff0d8;background-color:#3c763d}.panel-success>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#d6e9c6}.panel-info{border-color:#bce8f1}.panel-info>.panel-heading{color:#31708f;background-color:#d9edf7;border-color:#bce8f1}.panel-info>.panel-heading+.panel-collapse>.panel-body{border-top-color:#bce8f1}.panel-info>.panel-heading .badge{color:#d9edf7;background-color:#31708f}.panel-info>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#bce8f1}.panel-warning{border-color:#faebcc}.panel-warning>.panel-heading{color:#8a6d3b;background-color:#fcf8e3;border-color:#faebcc}.panel-warning>.panel-heading+.panel-collapse>.panel-body{border-top-color:#faebcc}.panel-warning>.panel-heading .badge{color:#fcf8e3;background-color:#8a6d3b}.panel-warning>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#faebcc}.panel-danger{border-color:#ebccd1}.panel-danger>.panel-heading{color:#a94442;background-color:#f2dede;border-color:#ebccd1}.panel-danger>.panel-heading+.panel-collapse>.panel-body{border-top-color:#ebccd1}.panel-danger>.panel-heading .badge{color:#f2dede;background-color:#a94442}.panel-danger>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#ebccd1}.embed-responsive{position:relative;display:block;height:0;padding:0;overflow:hidden}.embed-responsive .embed-responsive-item,.embed-responsive embed,.embed-responsive iframe,.embed-responsive object,.embed-responsive video{position:absolute;top:0;bottom:0;left:0;width:100%;height:100%;border:0}.embed-responsive-16by9{padding-bottom:56.25%}.embed-responsive-4by3{padding-bottom:75%}.well{min-height:20px;padding:19px;margin-bottom:20px;background-color:#f5f5f5;border:1px solid #e3e3e3;border-radius:4px;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.05);box-shadow:inset 0 1px 1px rgba(0,0,0,.05)}.well blockquote{border-color:#ddd;border-color:rgba(0,0,0,.15)}.well-lg{padding:24px;border-radius:6px}.well-sm{padding:9px;border-radius:3px}.close{float:right;font-size:21px;font-weight:700;line-height:1;color:#000;text-shadow:0 1px 0 #fff;filter:alpha(opacity=20);opacity:.2}.close:focus,.close:hover{color:#000;text-decoration:none;cursor:pointer;filter:alpha(opacity=50);opacity:.5}button.close{-webkit-appearance:none;padding:0;cursor:pointer;background:0 0;border:0}.modal-open{overflow:hidden}.modal{position:fixed;top:0;right:0;bottom:0;left:0;z-index:1050;display:none;overflow:hidden;-webkit-overflow-scrolling:touch;outline:0}.modal.fade .modal-dialog{-webkit-transition:-webkit-transform .3s ease-out;-o-transition:-o-transform .3s ease-out;transition:transform .3s ease-out;-webkit-transform:translate(0,-25%);-ms-transform:translate(0,-25%);-o-transform:translate(0,-25%);transform:translate(0,-25%)}.modal.in .modal-dialog{-webkit-transform:translate(0,0);-ms-transform:translate(0,0);-o-transform:translate(0,0);transform:translate(0,0)}.modal-open .modal{overflow-x:hidden;overflow-y:auto}.modal-dialog{position:relative;width:auto;margin:10px}.modal-content{position:relative;background-color:#fff;-webkit-background-clip:padding-box;background-clip:padding-box;border:1px solid #999;border:1px solid rgba(0,0,0,.2);border-radius:6px;outline:0;-webkit-box-shadow:0 3px 9px rgba(0,0,0,.5);box-shadow:0 3px 9px rgba(0,0,0,.5)}.modal-backdrop{position:fixed;top:0;right:0;bottom:0;left:0;z-index:1040;background-color:#000}.modal-backdrop.fade{filter:alpha(opacity=0);opacity:0}.modal-backdrop.in{filter:alpha(opacity=50);opacity:.5}.modal-header{min-height:16.43px;padding:15px;border-bottom:1px solid #e5e5e5}.modal-header .close{margin-top:-2px}.modal-title{margin:0;line-height:1.42857143}.modal-body{position:relative;padding:15px}.modal-footer{padding:15px;text-align:right;border-top:1px solid #e5e5e5}.modal-footer .btn+.btn{margin-bottom:0;margin-left:5px}.modal-footer .btn-group .btn+.btn{margin-left:-1px}.modal-footer .btn-block+.btn-block{margin-left:0}.modal-scrollbar-measure{position:absolute;top:-9999px;width:50px;height:50px;overflow:scroll}@media (min-width:768px){.modal-dialog{width:600px;margin:30px auto}.modal-content{-webkit-box-shadow:0 5px 15px rgba(0,0,0,.5);box-shadow:0 5px 15px rgba(0,0,0,.5)}.modal-sm{width:300px}}@media (min-width:992px){.modal-lg{width:900px}}.tooltip{position:absolute;z-index:1070;display:block;font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:12px;font-style:normal;font-weight:400;line-height:1.42857143;text-align:left;text-align:start;text-decoration:none;text-shadow:none;text-transform:none;letter-spacing:normal;word-break:normal;word-spacing:normal;word-wrap:normal;white-space:normal;filter:alpha(opacity=0);opacity:0;line-break:auto}.tooltip.in{filter:alpha(opacity=90);opacity:.9}.tooltip.top{padding:5px 0;margin-top:-3px}.tooltip.right{padding:0 5px;margin-left:3px}.tooltip.bottom{padding:5px 0;margin-top:3px}.tooltip.left{padding:0 5px;margin-left:-3px}.tooltip-inner{max-width:200px;padding:3px 8px;color:#fff;text-align:center;background-color:#000;border-radius:4px}.tooltip-arrow{position:absolute;width:0;height:0;border-color:transparent;border-style:solid}.tooltip.top .tooltip-arrow{bottom:0;left:50%;margin-left:-5px;border-width:5px 5px 0;border-top-color:#000}.tooltip.top-left .tooltip-arrow{right:5px;bottom:0;margin-bottom:-5px;border-width:5px 5px 0;border-top-color:#000}.tooltip.top-right .tooltip-arrow{bottom:0;left:5px;margin-bottom:-5px;border-width:5px 5px 0;border-top-color:#000}.tooltip.right .tooltip-arrow{top:50%;left:0;margin-top:-5px;border-width:5px 5px 5px 0;border-right-color:#000}.tooltip.left .tooltip-arrow{top:50%;right:0;margin-top:-5px;border-width:5px 0 5px 5px;border-left-color:#000}.tooltip.bottom .tooltip-arrow{top:0;left:50%;margin-left:-5px;border-width:0 5px 5px;border-bottom-color:#000}.tooltip.bottom-left .tooltip-arrow{top:0;right:5px;margin-top:-5px;border-width:0 5px 5px;border-bottom-color:#000}.tooltip.bottom-right .tooltip-arrow{top:0;left:5px;margin-top:-5px;border-width:0 5px 5px;border-bottom-color:#000}.popover{position:absolute;top:0;left:0;z-index:1060;display:none;max-width:276px;padding:1px;font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:14px;font-style:normal;font-weight:400;line-height:1.42857143;text-align:left;text-align:start;text-decoration:none;text-shadow:none;text-transform:none;letter-spacing:normal;word-break:normal;word-spacing:normal;word-wrap:normal;white-space:normal;background-color:#fff;-webkit-background-clip:padding-box;background-clip:padding-box;border:1px solid #ccc;border:1px solid rgba(0,0,0,.2);border-radius:6px;-webkit-box-shadow:0 5px 10px rgba(0,0,0,.2);box-shadow:0 5px 10px rgba(0,0,0,.2);line-break:auto}.popover.top{margin-top:-10px}.popover.right{margin-left:10px}.popover.bottom{margin-top:10px}.popover.left{margin-left:-10px}.popover-title{padding:8px 14px;margin:0;font-size:14px;background-color:#f7f7f7;border-bottom:1px solid #ebebeb;border-radius:5px 5px 0 0}.popover-content{padding:9px 14px}.popover>.arrow,.popover>.arrow:after{position:absolute;display:block;width:0;height:0;border-color:transparent;border-style:solid}.popover>.arrow{border-width:11px}.popover>.arrow:after{content:"";border-width:10px}.popover.top>.arrow{bottom:-11px;left:50%;margin-left:-11px;border-top-color:#999;border-top-color:rgba(0,0,0,.25);border-bottom-width:0}.popover.top>.arrow:after{bottom:1px;margin-left:-10px;content:" ";border-top-color:#fff;border-bottom-width:0}.popover.right>.arrow{top:50%;left:-11px;margin-top:-11px;border-right-color:#999;border-right-color:rgba(0,0,0,.25);border-left-width:0}.popover.right>.arrow:after{bottom:-10px;left:1px;content:" ";border-right-color:#fff;border-left-width:0}.popover.bottom>.arrow{top:-11px;left:50%;margin-left:-11px;border-top-width:0;border-bottom-color:#999;border-bottom-color:rgba(0,0,0,.25)}.popover.bottom>.arrow:after{top:1px;margin-left:-10px;content:" ";border-top-width:0;border-bottom-color:#fff}.popover.left>.arrow{top:50%;right:-11px;margin-top:-11px;border-right-width:0;border-left-color:#999;border-left-color:rgba(0,0,0,.25)}.popover.left>.arrow:after{right:1px;bottom:-10px;content:" ";border-right-width:0;border-left-color:#fff}.carousel{position:relative}.carousel-inner{position:relative;width:100%;overflow:hidden}.carousel-inner>.item{position:relative;display:none;-webkit-transition:.6s ease-in-out left;-o-transition:.6s ease-in-out left;transition:.6s ease-in-out left}.carousel-inner>.item>a>img,.carousel-inner>.item>img{line-height:1}@media all and (transform-3d),(-webkit-transform-3d){.carousel-inner>.item{-webkit-transition:-webkit-transform .6s ease-in-out;-o-transition:-o-transform .6s ease-in-out;transition:transform .6s ease-in-out;-webkit-backface-visibility:hidden;backface-visibility:hidden;-webkit-perspective:1000px;perspective:1000px}.carousel-inner>.item.active.right,.carousel-inner>.item.next{left:0;-webkit-transform:translate3d(100%,0,0);transform:translate3d(100%,0,0)}.carousel-inner>.item.active.left,.carousel-inner>.item.prev{left:0;-webkit-transform:translate3d(-100%,0,0);transform:translate3d(-100%,0,0)}.carousel-inner>.item.active,.carousel-inner>.item.next.left,.carousel-inner>.item.prev.right{left:0;-webkit-transform:translate3d(0,0,0);transform:translate3d(0,0,0)}}.carousel-inner>.active,.carousel-inner>.next,.carousel-inner>.prev{display:block}.carousel-inner>.active{left:0}.carousel-inner>.next,.carousel-inner>.prev{position:absolute;top:0;width:100%}.carousel-inner>.next{left:100%}.carousel-inner>.prev{left:-100%}.carousel-inner>.next.left,.carousel-inner>.prev.right{left:0}.carousel-inner>.active.left{left:-100%}.carousel-inner>.active.right{left:100%}.carousel-control{position:absolute;top:0;bottom:0;left:0;width:15%;font-size:20px;color:#fff;text-align:center;text-shadow:0 1px 2px rgba(0,0,0,.6);filter:alpha(opacity=50);opacity:.5}.carousel-control.left{background-image:-webkit-linear-gradient(left,rgba(0,0,0,.5) 0,rgba(0,0,0,.0001) 100%);background-image:-o-linear-gradient(left,rgba(0,0,0,.5) 0,rgba(0,0,0,.0001) 100%);background-image:-webkit-gradient(linear,left top,right top,from(rgba(0,0,0,.5)),to(rgba(0,0,0,.0001)));background-image:linear-gradient(to right,rgba(0,0,0,.5) 0,rgba(0,0,0,.0001) 100%);filter:progid:DXImageTransform.Microsoft.gradient(startColorstr='#80000000', endColorstr='#00000000', GradientType=1);background-repeat:repeat-x}.carousel-control.right{right:0;left:auto;background-image:-webkit-linear-gradient(left,rgba(0,0,0,.0001) 0,rgba(0,0,0,.5) 100%);background-image:-o-linear-gradient(left,rgba(0,0,0,.0001) 0,rgba(0,0,0,.5) 100%);background-image:-webkit-gradient(linear,left top,right top,from(rgba(0,0,0,.0001)),to(rgba(0,0,0,.5)));background-image:linear-gradient(to right,rgba(0,0,0,.0001) 0,rgba(0,0,0,.5) 100%);filter:progid:DXImageTransform.Microsoft.gradient(startColorstr='#00000000', endColorstr='#80000000', GradientType=1);background-repeat:repeat-x}.carousel-control:focus,.carousel-control:hover{color:#fff;text-decoration:none;filter:alpha(opacity=90);outline:0;opacity:.9}.carousel-control .glyphicon-chevron-left,.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next,.carousel-control .icon-prev{position:absolute;top:50%;z-index:5;display:inline-block;margin-top:-10px}.carousel-control .glyphicon-chevron-left,.carousel-control .icon-prev{left:50%;margin-left:-10px}.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next{right:50%;margin-right:-10px}.carousel-control .icon-next,.carousel-control .icon-prev{width:20px;height:20px;font-family:serif;line-height:1}.carousel-control .icon-prev:before{content:'\2039'}.carousel-control .icon-next:before{content:'\203a'}.carousel-indicators{position:absolute;bottom:10px;left:50%;z-index:15;width:60%;padding-left:0;margin-left:-30%;text-align:center;list-style:none}.carousel-indicators li{display:inline-block;width:10px;height:10px;margin:1px;text-indent:-999px;cursor:pointer;background-color:#000\9;background-color:rgba(0,0,0,0);border:1px solid #fff;border-radius:10px}.carousel-indicators .active{width:12px;height:12px;margin:0;background-color:#fff}.carousel-caption{position:absolute;right:15%;bottom:20px;left:15%;z-index:10;padding-top:20px;padding-bottom:20px;color:#fff;text-align:center;text-shadow:0 1px 2px rgba(0,0,0,.6)}.carousel-caption .btn{text-shadow:none}@media screen and (min-width:768px){.carousel-control .glyphicon-chevron-left,.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next,.carousel-control .icon-prev{width:30px;height:30px;margin-top:-15px;font-size:30px}.carousel-control .glyphicon-chevron-left,.carousel-control .icon-prev{margin-left:-15px}.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next{margin-right:-15px}.carousel-caption{right:20%;left:20%;padding-bottom:30px}.carousel-indicators{bottom:20px}}.btn-group-vertical>.btn-group:after,.btn-group-vertical>.btn-group:before,.btn-toolbar:after,.btn-toolbar:before,.clearfix:after,.clearfix:before,.container-fluid:after,.container-fluid:before,.container:after,.container:before,.dl-horizontal dd:after,.dl-horizontal dd:before,.form-horizontal .form-group:after,.form-horizontal .form-group:before,.modal-footer:after,.modal-footer:before,.nav:after,.nav:before,.navbar-collapse:after,.navbar-collapse:before,.navbar-header:after,.navbar-header:before,.navbar:after,.navbar:before,.pager:after,.pager:before,.panel-body:after,.panel-body:before,.row:after,.row:before{display:table;content:" "}.btn-group-vertical>.btn-group:after,.btn-toolbar:after,.clearfix:after,.container-fluid:after,.container:after,.dl-horizontal dd:after,.form-horizontal .form-group:after,.modal-footer:after,.nav:after,.navbar-collapse:after,.navbar-header:after,.navbar:after,.pager:after,.panel-body:after,.row:after{clear:both}.center-block{display:block;margin-right:auto;margin-left:auto}.pull-right{float:right!important}.pull-left{float:left!important}.hide{display:none!important}.show{display:block!important}.invisible{visibility:hidden}.text-hide{font:0/0 a;color:transparent;text-shadow:none;background-color:transparent;border:0}.hidden{display:none!important}.affix{position:fixed}@-ms-viewport{width:device-width}.visible-lg,.visible-md,.visible-sm,.visible-xs{display:none!important}.visible-lg-block,.visible-lg-inline,.visible-lg-inline-block,.visible-md-block,.visible-md-inline,.visible-md-inline-block,.visible-sm-block,.visible-sm-inline,.visible-sm-inline-block,.visible-xs-block,.visible-xs-inline,.visible-xs-inline-block{display:none!important}@media (max-width:767px){.visible-xs{display:block!important}table.visible-xs{display:table!important}tr.visible-xs{display:table-row!important}td.visible-xs,th.visible-xs{display:table-cell!important}}@media (max-width:767px){.visible-xs-block{display:block!important}}@media (max-width:767px){.visible-xs-inline{display:inline!important}}@media (max-width:767px){.visible-xs-inline-block{display:inline-block!important}}@media (min-width:768px) and (max-width:991px){.visible-sm{display:block!important}table.visible-sm{display:table!important}tr.visible-sm{display:table-row!important}td.visible-sm,th.visible-sm{display:table-cell!important}}@media (min-width:768px) and (max-width:991px){.visible-sm-block{display:block!important}}@media (min-width:768px) and (max-width:991px){.visible-sm-inline{display:inline!important}}@media (min-width:768px) and (max-width:991px){.visible-sm-inline-block{display:inline-block!important}}@media (min-width:992px) and (max-width:1199px){.visible-md{display:block!important}table.visible-md{display:table!important}tr.visible-md{display:table-row!important}td.visible-md,th.visible-md{display:table-cell!important}}@media (min-width:992px) and (max-width:1199px){.visible-md-block{display:block!important}}@media (min-width:992px) and (max-width:1199px){.visible-md-inline{display:inline!important}}@media (min-width:992px) and (max-width:1199px){.visible-md-inline-block{display:inline-block!important}}@media (min-width:1200px){.visible-lg{display:block!important}table.visible-lg{display:table!important}tr.visible-lg{display:table-row!important}td.visible-lg,th.visible-lg{display:table-cell!important}}@media (min-width:1200px){.visible-lg-block{display:block!important}}@media (min-width:1200px){.visible-lg-inline{display:inline!important}}@media (min-width:1200px){.visible-lg-inline-block{display:inline-block!important}}@media (max-width:767px){.hidden-xs{display:none!important}}@media (min-width:768px) and (max-width:991px){.hidden-sm{display:none!important}}@media (min-width:992px) and (max-width:1199px){.hidden-md{display:none!important}}@media (min-width:1200px){.hidden-lg{display:none!important}}.visible-print{display:none!important}@media print{.visible-print{display:block!important}table.visible-print{display:table!important}tr.visible-print{display:table-row!important}td.visible-print,th.visible-print{display:table-cell!important}}.visible-print-block{display:none!important}@media print{.visible-print-block{display:block!important}}.visible-print-inline{display:none!important}@media print{.visible-print-inline{display:inline!important}}.visible-print-inline-block{display:none!important}@media print{.visible-print-inline-block{display:inline-block!important}}@media print{.hidden-print{display:none!important}}
</style>
<script>/*!
 * Bootstrap v3.3.5 (http://getbootstrap.com)
 * Copyright 2011-2015 Twitter, Inc.
 * Licensed under the MIT license
 */
if("undefined"==typeof jQuery)throw new Error("Bootstrap's JavaScript requires jQuery");+function(a){"use strict";var b=a.fn.jquery.split(" ")[0].split(".");if(b[0]<2&&b[1]<9||1==b[0]&&9==b[1]&&b[2]<1)throw new Error("Bootstrap's JavaScript requires jQuery version 1.9.1 or higher")}(jQuery),+function(a){"use strict";function b(){var a=document.createElement("bootstrap"),b={WebkitTransition:"webkitTransitionEnd",MozTransition:"transitionend",OTransition:"oTransitionEnd otransitionend",transition:"transitionend"};for(var c in b)if(void 0!==a.style[c])return{end:b[c]};return!1}a.fn.emulateTransitionEnd=function(b){var c=!1,d=this;a(this).one("bsTransitionEnd",function(){c=!0});var e=function(){c||a(d).trigger(a.support.transition.end)};return setTimeout(e,b),this},a(function(){a.support.transition=b(),a.support.transition&&(a.event.special.bsTransitionEnd={bindType:a.support.transition.end,delegateType:a.support.transition.end,handle:function(b){return a(b.target).is(this)?b.handleObj.handler.apply(this,arguments):void 0}})})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var c=a(this),e=c.data("bs.alert");e||c.data("bs.alert",e=new d(this)),"string"==typeof b&&e[b].call(c)})}var c='[data-dismiss="alert"]',d=function(b){a(b).on("click",c,this.close)};d.VERSION="3.3.5",d.TRANSITION_DURATION=150,d.prototype.close=function(b){function c(){g.detach().trigger("closed.bs.alert").remove()}var e=a(this),f=e.attr("data-target");f||(f=e.attr("href"),f=f&&f.replace(/.*(?=#[^\s]*$)/,""));var g=a(f);b&&b.preventDefault(),g.length||(g=e.closest(".alert")),g.trigger(b=a.Event("close.bs.alert")),b.isDefaultPrevented()||(g.removeClass("in"),a.support.transition&&g.hasClass("fade")?g.one("bsTransitionEnd",c).emulateTransitionEnd(d.TRANSITION_DURATION):c())};var e=a.fn.alert;a.fn.alert=b,a.fn.alert.Constructor=d,a.fn.alert.noConflict=function(){return a.fn.alert=e,this},a(document).on("click.bs.alert.data-api",c,d.prototype.close)}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.button"),f="object"==typeof b&&b;e||d.data("bs.button",e=new c(this,f)),"toggle"==b?e.toggle():b&&e.setState(b)})}var c=function(b,d){this.$element=a(b),this.options=a.extend({},c.DEFAULTS,d),this.isLoading=!1};c.VERSION="3.3.5",c.DEFAULTS={loadingText:"loading..."},c.prototype.setState=function(b){var c="disabled",d=this.$element,e=d.is("input")?"val":"html",f=d.data();b+="Text",null==f.resetText&&d.data("resetText",d[e]()),setTimeout(a.proxy(function(){d[e](null==f[b]?this.options[b]:f[b]),"loadingText"==b?(this.isLoading=!0,d.addClass(c).attr(c,c)):this.isLoading&&(this.isLoading=!1,d.removeClass(c).removeAttr(c))},this),0)},c.prototype.toggle=function(){var a=!0,b=this.$element.closest('[data-toggle="buttons"]');if(b.length){var c=this.$element.find("input");"radio"==c.prop("type")?(c.prop("checked")&&(a=!1),b.find(".active").removeClass("active"),this.$element.addClass("active")):"checkbox"==c.prop("type")&&(c.prop("checked")!==this.$element.hasClass("active")&&(a=!1),this.$element.toggleClass("active")),c.prop("checked",this.$element.hasClass("active")),a&&c.trigger("change")}else this.$element.attr("aria-pressed",!this.$element.hasClass("active")),this.$element.toggleClass("active")};var d=a.fn.button;a.fn.button=b,a.fn.button.Constructor=c,a.fn.button.noConflict=function(){return a.fn.button=d,this},a(document).on("click.bs.button.data-api",'[data-toggle^="button"]',function(c){var d=a(c.target);d.hasClass("btn")||(d=d.closest(".btn")),b.call(d,"toggle"),a(c.target).is('input[type="radio"]')||a(c.target).is('input[type="checkbox"]')||c.preventDefault()}).on("focus.bs.button.data-api blur.bs.button.data-api",'[data-toggle^="button"]',function(b){a(b.target).closest(".btn").toggleClass("focus",/^focus(in)?$/.test(b.type))})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.carousel"),f=a.extend({},c.DEFAULTS,d.data(),"object"==typeof b&&b),g="string"==typeof b?b:f.slide;e||d.data("bs.carousel",e=new c(this,f)),"number"==typeof b?e.to(b):g?e[g]():f.interval&&e.pause().cycle()})}var c=function(b,c){this.$element=a(b),this.$indicators=this.$element.find(".carousel-indicators"),this.options=c,this.paused=null,this.sliding=null,this.interval=null,this.$active=null,this.$items=null,this.options.keyboard&&this.$element.on("keydown.bs.carousel",a.proxy(this.keydown,this)),"hover"==this.options.pause&&!("ontouchstart"in document.documentElement)&&this.$element.on("mouseenter.bs.carousel",a.proxy(this.pause,this)).on("mouseleave.bs.carousel",a.proxy(this.cycle,this))};c.VERSION="3.3.5",c.TRANSITION_DURATION=600,c.DEFAULTS={interval:5e3,pause:"hover",wrap:!0,keyboard:!0},c.prototype.keydown=function(a){if(!/input|textarea/i.test(a.target.tagName)){switch(a.which){case 37:this.prev();break;case 39:this.next();break;default:return}a.preventDefault()}},c.prototype.cycle=function(b){return b||(this.paused=!1),this.interval&&clearInterval(this.interval),this.options.interval&&!this.paused&&(this.interval=setInterval(a.proxy(this.next,this),this.options.interval)),this},c.prototype.getItemIndex=function(a){return this.$items=a.parent().children(".item"),this.$items.index(a||this.$active)},c.prototype.getItemForDirection=function(a,b){var c=this.getItemIndex(b),d="prev"==a&&0===c||"next"==a&&c==this.$items.length-1;if(d&&!this.options.wrap)return b;var e="prev"==a?-1:1,f=(c+e)%this.$items.length;return this.$items.eq(f)},c.prototype.to=function(a){var b=this,c=this.getItemIndex(this.$active=this.$element.find(".item.active"));return a>this.$items.length-1||0>a?void 0:this.sliding?this.$element.one("slid.bs.carousel",function(){b.to(a)}):c==a?this.pause().cycle():this.slide(a>c?"next":"prev",this.$items.eq(a))},c.prototype.pause=function(b){return b||(this.paused=!0),this.$element.find(".next, .prev").length&&a.support.transition&&(this.$element.trigger(a.support.transition.end),this.cycle(!0)),this.interval=clearInterval(this.interval),this},c.prototype.next=function(){return this.sliding?void 0:this.slide("next")},c.prototype.prev=function(){return this.sliding?void 0:this.slide("prev")},c.prototype.slide=function(b,d){var e=this.$element.find(".item.active"),f=d||this.getItemForDirection(b,e),g=this.interval,h="next"==b?"left":"right",i=this;if(f.hasClass("active"))return this.sliding=!1;var j=f[0],k=a.Event("slide.bs.carousel",{relatedTarget:j,direction:h});if(this.$element.trigger(k),!k.isDefaultPrevented()){if(this.sliding=!0,g&&this.pause(),this.$indicators.length){this.$indicators.find(".active").removeClass("active");var l=a(this.$indicators.children()[this.getItemIndex(f)]);l&&l.addClass("active")}var m=a.Event("slid.bs.carousel",{relatedTarget:j,direction:h});return a.support.transition&&this.$element.hasClass("slide")?(f.addClass(b),f[0].offsetWidth,e.addClass(h),f.addClass(h),e.one("bsTransitionEnd",function(){f.removeClass([b,h].join(" ")).addClass("active"),e.removeClass(["active",h].join(" ")),i.sliding=!1,setTimeout(function(){i.$element.trigger(m)},0)}).emulateTransitionEnd(c.TRANSITION_DURATION)):(e.removeClass("active"),f.addClass("active"),this.sliding=!1,this.$element.trigger(m)),g&&this.cycle(),this}};var d=a.fn.carousel;a.fn.carousel=b,a.fn.carousel.Constructor=c,a.fn.carousel.noConflict=function(){return a.fn.carousel=d,this};var e=function(c){var d,e=a(this),f=a(e.attr("data-target")||(d=e.attr("href"))&&d.replace(/.*(?=#[^\s]+$)/,""));if(f.hasClass("carousel")){var g=a.extend({},f.data(),e.data()),h=e.attr("data-slide-to");h&&(g.interval=!1),b.call(f,g),h&&f.data("bs.carousel").to(h),c.preventDefault()}};a(document).on("click.bs.carousel.data-api","[data-slide]",e).on("click.bs.carousel.data-api","[data-slide-to]",e),a(window).on("load",function(){a('[data-ride="carousel"]').each(function(){var c=a(this);b.call(c,c.data())})})}(jQuery),+function(a){"use strict";function b(b){var c,d=b.attr("data-target")||(c=b.attr("href"))&&c.replace(/.*(?=#[^\s]+$)/,"");return a(d)}function c(b){return this.each(function(){var c=a(this),e=c.data("bs.collapse"),f=a.extend({},d.DEFAULTS,c.data(),"object"==typeof b&&b);!e&&f.toggle&&/show|hide/.test(b)&&(f.toggle=!1),e||c.data("bs.collapse",e=new d(this,f)),"string"==typeof b&&e[b]()})}var d=function(b,c){this.$element=a(b),this.options=a.extend({},d.DEFAULTS,c),this.$trigger=a('[data-toggle="collapse"][href="#'+b.id+'"],[data-toggle="collapse"][data-target="#'+b.id+'"]'),this.transitioning=null,this.options.parent?this.$parent=this.getParent():this.addAriaAndCollapsedClass(this.$element,this.$trigger),this.options.toggle&&this.toggle()};d.VERSION="3.3.5",d.TRANSITION_DURATION=350,d.DEFAULTS={toggle:!0},d.prototype.dimension=function(){var a=this.$element.hasClass("width");return a?"width":"height"},d.prototype.show=function(){if(!this.transitioning&&!this.$element.hasClass("in")){var b,e=this.$parent&&this.$parent.children(".panel").children(".in, .collapsing");if(!(e&&e.length&&(b=e.data("bs.collapse"),b&&b.transitioning))){var f=a.Event("show.bs.collapse");if(this.$element.trigger(f),!f.isDefaultPrevented()){e&&e.length&&(c.call(e,"hide"),b||e.data("bs.collapse",null));var g=this.dimension();this.$element.removeClass("collapse").addClass("collapsing")[g](0).attr("aria-expanded",!0),this.$trigger.removeClass("collapsed").attr("aria-expanded",!0),this.transitioning=1;var h=function(){this.$element.removeClass("collapsing").addClass("collapse in")[g](""),this.transitioning=0,this.$element.trigger("shown.bs.collapse")};if(!a.support.transition)return h.call(this);var i=a.camelCase(["scroll",g].join("-"));this.$element.one("bsTransitionEnd",a.proxy(h,this)).emulateTransitionEnd(d.TRANSITION_DURATION)[g](this.$element[0][i])}}}},d.prototype.hide=function(){if(!this.transitioning&&this.$element.hasClass("in")){var b=a.Event("hide.bs.collapse");if(this.$element.trigger(b),!b.isDefaultPrevented()){var c=this.dimension();this.$element[c](this.$element[c]())[0].offsetHeight,this.$element.addClass("collapsing").removeClass("collapse in").attr("aria-expanded",!1),this.$trigger.addClass("collapsed").attr("aria-expanded",!1),this.transitioning=1;var e=function(){this.transitioning=0,this.$element.removeClass("collapsing").addClass("collapse").trigger("hidden.bs.collapse")};return a.support.transition?void this.$element[c](0).one("bsTransitionEnd",a.proxy(e,this)).emulateTransitionEnd(d.TRANSITION_DURATION):e.call(this)}}},d.prototype.toggle=function(){this[this.$element.hasClass("in")?"hide":"show"]()},d.prototype.getParent=function(){return a(this.options.parent).find('[data-toggle="collapse"][data-parent="'+this.options.parent+'"]').each(a.proxy(function(c,d){var e=a(d);this.addAriaAndCollapsedClass(b(e),e)},this)).end()},d.prototype.addAriaAndCollapsedClass=function(a,b){var c=a.hasClass("in");a.attr("aria-expanded",c),b.toggleClass("collapsed",!c).attr("aria-expanded",c)};var e=a.fn.collapse;a.fn.collapse=c,a.fn.collapse.Constructor=d,a.fn.collapse.noConflict=function(){return a.fn.collapse=e,this},a(document).on("click.bs.collapse.data-api",'[data-toggle="collapse"]',function(d){var e=a(this);e.attr("data-target")||d.preventDefault();var f=b(e),g=f.data("bs.collapse"),h=g?"toggle":e.data();c.call(f,h)})}(jQuery),+function(a){"use strict";function b(b){var c=b.attr("data-target");c||(c=b.attr("href"),c=c&&/#[A-Za-z]/.test(c)&&c.replace(/.*(?=#[^\s]*$)/,""));var d=c&&a(c);return d&&d.length?d:b.parent()}function c(c){c&&3===c.which||(a(e).remove(),a(f).each(function(){var d=a(this),e=b(d),f={relatedTarget:this};e.hasClass("open")&&(c&&"click"==c.type&&/input|textarea/i.test(c.target.tagName)&&a.contains(e[0],c.target)||(e.trigger(c=a.Event("hide.bs.dropdown",f)),c.isDefaultPrevented()||(d.attr("aria-expanded","false"),e.removeClass("open").trigger("hidden.bs.dropdown",f))))}))}function d(b){return this.each(function(){var c=a(this),d=c.data("bs.dropdown");d||c.data("bs.dropdown",d=new g(this)),"string"==typeof b&&d[b].call(c)})}var e=".dropdown-backdrop",f='[data-toggle="dropdown"]',g=function(b){a(b).on("click.bs.dropdown",this.toggle)};g.VERSION="3.3.5",g.prototype.toggle=function(d){var e=a(this);if(!e.is(".disabled, :disabled")){var f=b(e),g=f.hasClass("open");if(c(),!g){"ontouchstart"in document.documentElement&&!f.closest(".navbar-nav").length&&a(document.createElement("div")).addClass("dropdown-backdrop").insertAfter(a(this)).on("click",c);var h={relatedTarget:this};if(f.trigger(d=a.Event("show.bs.dropdown",h)),d.isDefaultPrevented())return;e.trigger("focus").attr("aria-expanded","true"),f.toggleClass("open").trigger("shown.bs.dropdown",h)}return!1}},g.prototype.keydown=function(c){if(/(38|40|27|32)/.test(c.which)&&!/input|textarea/i.test(c.target.tagName)){var d=a(this);if(c.preventDefault(),c.stopPropagation(),!d.is(".disabled, :disabled")){var e=b(d),g=e.hasClass("open");if(!g&&27!=c.which||g&&27==c.which)return 27==c.which&&e.find(f).trigger("focus"),d.trigger("click");var h=" li:not(.disabled):visible a",i=e.find(".dropdown-menu"+h);if(i.length){var j=i.index(c.target);38==c.which&&j>0&&j--,40==c.which&&j<i.length-1&&j++,~j||(j=0),i.eq(j).trigger("focus")}}}};var h=a.fn.dropdown;a.fn.dropdown=d,a.fn.dropdown.Constructor=g,a.fn.dropdown.noConflict=function(){return a.fn.dropdown=h,this},a(document).on("click.bs.dropdown.data-api",c).on("click.bs.dropdown.data-api",".dropdown form",function(a){a.stopPropagation()}).on("click.bs.dropdown.data-api",f,g.prototype.toggle).on("keydown.bs.dropdown.data-api",f,g.prototype.keydown).on("keydown.bs.dropdown.data-api",".dropdown-menu",g.prototype.keydown)}(jQuery),+function(a){"use strict";function b(b,d){return this.each(function(){var e=a(this),f=e.data("bs.modal"),g=a.extend({},c.DEFAULTS,e.data(),"object"==typeof b&&b);f||e.data("bs.modal",f=new c(this,g)),"string"==typeof b?f[b](d):g.show&&f.show(d)})}var c=function(b,c){this.options=c,this.$body=a(document.body),this.$element=a(b),this.$dialog=this.$element.find(".modal-dialog"),this.$backdrop=null,this.isShown=null,this.originalBodyPad=null,this.scrollbarWidth=0,this.ignoreBackdropClick=!1,this.options.remote&&this.$element.find(".modal-content").load(this.options.remote,a.proxy(function(){this.$element.trigger("loaded.bs.modal")},this))};c.VERSION="3.3.5",c.TRANSITION_DURATION=300,c.BACKDROP_TRANSITION_DURATION=150,c.DEFAULTS={backdrop:!0,keyboard:!0,show:!0},c.prototype.toggle=function(a){return this.isShown?this.hide():this.show(a)},c.prototype.show=function(b){var d=this,e=a.Event("show.bs.modal",{relatedTarget:b});this.$element.trigger(e),this.isShown||e.isDefaultPrevented()||(this.isShown=!0,this.checkScrollbar(),this.setScrollbar(),this.$body.addClass("modal-open"),this.escape(),this.resize(),this.$element.on("click.dismiss.bs.modal",'[data-dismiss="modal"]',a.proxy(this.hide,this)),this.$dialog.on("mousedown.dismiss.bs.modal",function(){d.$element.one("mouseup.dismiss.bs.modal",function(b){a(b.target).is(d.$element)&&(d.ignoreBackdropClick=!0)})}),this.backdrop(function(){var e=a.support.transition&&d.$element.hasClass("fade");d.$element.parent().length||d.$element.appendTo(d.$body),d.$element.show().scrollTop(0),d.adjustDialog(),e&&d.$element[0].offsetWidth,d.$element.addClass("in"),d.enforceFocus();var f=a.Event("shown.bs.modal",{relatedTarget:b});e?d.$dialog.one("bsTransitionEnd",function(){d.$element.trigger("focus").trigger(f)}).emulateTransitionEnd(c.TRANSITION_DURATION):d.$element.trigger("focus").trigger(f)}))},c.prototype.hide=function(b){b&&b.preventDefault(),b=a.Event("hide.bs.modal"),this.$element.trigger(b),this.isShown&&!b.isDefaultPrevented()&&(this.isShown=!1,this.escape(),this.resize(),a(document).off("focusin.bs.modal"),this.$element.removeClass("in").off("click.dismiss.bs.modal").off("mouseup.dismiss.bs.modal"),this.$dialog.off("mousedown.dismiss.bs.modal"),a.support.transition&&this.$element.hasClass("fade")?this.$element.one("bsTransitionEnd",a.proxy(this.hideModal,this)).emulateTransitionEnd(c.TRANSITION_DURATION):this.hideModal())},c.prototype.enforceFocus=function(){a(document).off("focusin.bs.modal").on("focusin.bs.modal",a.proxy(function(a){this.$element[0]===a.target||this.$element.has(a.target).length||this.$element.trigger("focus")},this))},c.prototype.escape=function(){this.isShown&&this.options.keyboard?this.$element.on("keydown.dismiss.bs.modal",a.proxy(function(a){27==a.which&&this.hide()},this)):this.isShown||this.$element.off("keydown.dismiss.bs.modal")},c.prototype.resize=function(){this.isShown?a(window).on("resize.bs.modal",a.proxy(this.handleUpdate,this)):a(window).off("resize.bs.modal")},c.prototype.hideModal=function(){var a=this;this.$element.hide(),this.backdrop(function(){a.$body.removeClass("modal-open"),a.resetAdjustments(),a.resetScrollbar(),a.$element.trigger("hidden.bs.modal")})},c.prototype.removeBackdrop=function(){this.$backdrop&&this.$backdrop.remove(),this.$backdrop=null},c.prototype.backdrop=function(b){var d=this,e=this.$element.hasClass("fade")?"fade":"";if(this.isShown&&this.options.backdrop){var f=a.support.transition&&e;if(this.$backdrop=a(document.createElement("div")).addClass("modal-backdrop "+e).appendTo(this.$body),this.$element.on("click.dismiss.bs.modal",a.proxy(function(a){return this.ignoreBackdropClick?void(this.ignoreBackdropClick=!1):void(a.target===a.currentTarget&&("static"==this.options.backdrop?this.$element[0].focus():this.hide()))},this)),f&&this.$backdrop[0].offsetWidth,this.$backdrop.addClass("in"),!b)return;f?this.$backdrop.one("bsTransitionEnd",b).emulateTransitionEnd(c.BACKDROP_TRANSITION_DURATION):b()}else if(!this.isShown&&this.$backdrop){this.$backdrop.removeClass("in");var g=function(){d.removeBackdrop(),b&&b()};a.support.transition&&this.$element.hasClass("fade")?this.$backdrop.one("bsTransitionEnd",g).emulateTransitionEnd(c.BACKDROP_TRANSITION_DURATION):g()}else b&&b()},c.prototype.handleUpdate=function(){this.adjustDialog()},c.prototype.adjustDialog=function(){var a=this.$element[0].scrollHeight>document.documentElement.clientHeight;this.$element.css({paddingLeft:!this.bodyIsOverflowing&&a?this.scrollbarWidth:"",paddingRight:this.bodyIsOverflowing&&!a?this.scrollbarWidth:""})},c.prototype.resetAdjustments=function(){this.$element.css({paddingLeft:"",paddingRight:""})},c.prototype.checkScrollbar=function(){var a=window.innerWidth;if(!a){var b=document.documentElement.getBoundingClientRect();a=b.right-Math.abs(b.left)}this.bodyIsOverflowing=document.body.clientWidth<a,this.scrollbarWidth=this.measureScrollbar()},c.prototype.setScrollbar=function(){var a=parseInt(this.$body.css("padding-right")||0,10);this.originalBodyPad=document.body.style.paddingRight||"",this.bodyIsOverflowing&&this.$body.css("padding-right",a+this.scrollbarWidth)},c.prototype.resetScrollbar=function(){this.$body.css("padding-right",this.originalBodyPad)},c.prototype.measureScrollbar=function(){var a=document.createElement("div");a.className="modal-scrollbar-measure",this.$body.append(a);var b=a.offsetWidth-a.clientWidth;return this.$body[0].removeChild(a),b};var d=a.fn.modal;a.fn.modal=b,a.fn.modal.Constructor=c,a.fn.modal.noConflict=function(){return a.fn.modal=d,this},a(document).on("click.bs.modal.data-api",'[data-toggle="modal"]',function(c){var d=a(this),e=d.attr("href"),f=a(d.attr("data-target")||e&&e.replace(/.*(?=#[^\s]+$)/,"")),g=f.data("bs.modal")?"toggle":a.extend({remote:!/#/.test(e)&&e},f.data(),d.data());d.is("a")&&c.preventDefault(),f.one("show.bs.modal",function(a){a.isDefaultPrevented()||f.one("hidden.bs.modal",function(){d.is(":visible")&&d.trigger("focus")})}),b.call(f,g,this)})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.tooltip"),f="object"==typeof b&&b;(e||!/destroy|hide/.test(b))&&(e||d.data("bs.tooltip",e=new c(this,f)),"string"==typeof b&&e[b]())})}var c=function(a,b){this.type=null,this.options=null,this.enabled=null,this.timeout=null,this.hoverState=null,this.$element=null,this.inState=null,this.init("tooltip",a,b)};c.VERSION="3.3.5",c.TRANSITION_DURATION=150,c.DEFAULTS={animation:!0,placement:"top",selector:!1,template:'<div class="tooltip" role="tooltip"><div class="tooltip-arrow"></div><div class="tooltip-inner"></div></div>',trigger:"hover focus",title:"",delay:0,html:!1,container:!1,viewport:{selector:"body",padding:0}},c.prototype.init=function(b,c,d){if(this.enabled=!0,this.type=b,this.$element=a(c),this.options=this.getOptions(d),this.$viewport=this.options.viewport&&a(a.isFunction(this.options.viewport)?this.options.viewport.call(this,this.$element):this.options.viewport.selector||this.options.viewport),this.inState={click:!1,hover:!1,focus:!1},this.$element[0]instanceof document.constructor&&!this.options.selector)throw new Error("`selector` option must be specified when initializing "+this.type+" on the window.document object!");for(var e=this.options.trigger.split(" "),f=e.length;f--;){var g=e[f];if("click"==g)this.$element.on("click."+this.type,this.options.selector,a.proxy(this.toggle,this));else if("manual"!=g){var h="hover"==g?"mouseenter":"focusin",i="hover"==g?"mouseleave":"focusout";this.$element.on(h+"."+this.type,this.options.selector,a.proxy(this.enter,this)),this.$element.on(i+"."+this.type,this.options.selector,a.proxy(this.leave,this))}}this.options.selector?this._options=a.extend({},this.options,{trigger:"manual",selector:""}):this.fixTitle()},c.prototype.getDefaults=function(){return c.DEFAULTS},c.prototype.getOptions=function(b){return b=a.extend({},this.getDefaults(),this.$element.data(),b),b.delay&&"number"==typeof b.delay&&(b.delay={show:b.delay,hide:b.delay}),b},c.prototype.getDelegateOptions=function(){var b={},c=this.getDefaults();return this._options&&a.each(this._options,function(a,d){c[a]!=d&&(b[a]=d)}),b},c.prototype.enter=function(b){var c=b instanceof this.constructor?b:a(b.currentTarget).data("bs."+this.type);return c||(c=new this.constructor(b.currentTarget,this.getDelegateOptions()),a(b.currentTarget).data("bs."+this.type,c)),b instanceof a.Event&&(c.inState["focusin"==b.type?"focus":"hover"]=!0),c.tip().hasClass("in")||"in"==c.hoverState?void(c.hoverState="in"):(clearTimeout(c.timeout),c.hoverState="in",c.options.delay&&c.options.delay.show?void(c.timeout=setTimeout(function(){"in"==c.hoverState&&c.show()},c.options.delay.show)):c.show())},c.prototype.isInStateTrue=function(){for(var a in this.inState)if(this.inState[a])return!0;return!1},c.prototype.leave=function(b){var c=b instanceof this.constructor?b:a(b.currentTarget).data("bs."+this.type);return c||(c=new this.constructor(b.currentTarget,this.getDelegateOptions()),a(b.currentTarget).data("bs."+this.type,c)),b instanceof a.Event&&(c.inState["focusout"==b.type?"focus":"hover"]=!1),c.isInStateTrue()?void 0:(clearTimeout(c.timeout),c.hoverState="out",c.options.delay&&c.options.delay.hide?void(c.timeout=setTimeout(function(){"out"==c.hoverState&&c.hide()},c.options.delay.hide)):c.hide())},c.prototype.show=function(){var b=a.Event("show.bs."+this.type);if(this.hasContent()&&this.enabled){this.$element.trigger(b);var d=a.contains(this.$element[0].ownerDocument.documentElement,this.$element[0]);if(b.isDefaultPrevented()||!d)return;var e=this,f=this.tip(),g=this.getUID(this.type);this.setContent(),f.attr("id",g),this.$element.attr("aria-describedby",g),this.options.animation&&f.addClass("fade");var h="function"==typeof this.options.placement?this.options.placement.call(this,f[0],this.$element[0]):this.options.placement,i=/\s?auto?\s?/i,j=i.test(h);j&&(h=h.replace(i,"")||"top"),f.detach().css({top:0,left:0,display:"block"}).addClass(h).data("bs."+this.type,this),this.options.container?f.appendTo(this.options.container):f.insertAfter(this.$element),this.$element.trigger("inserted.bs."+this.type);var k=this.getPosition(),l=f[0].offsetWidth,m=f[0].offsetHeight;if(j){var n=h,o=this.getPosition(this.$viewport);h="bottom"==h&&k.bottom+m>o.bottom?"top":"top"==h&&k.top-m<o.top?"bottom":"right"==h&&k.right+l>o.width?"left":"left"==h&&k.left-l<o.left?"right":h,f.removeClass(n).addClass(h)}var p=this.getCalculatedOffset(h,k,l,m);this.applyPlacement(p,h);var q=function(){var a=e.hoverState;e.$element.trigger("shown.bs."+e.type),e.hoverState=null,"out"==a&&e.leave(e)};a.support.transition&&this.$tip.hasClass("fade")?f.one("bsTransitionEnd",q).emulateTransitionEnd(c.TRANSITION_DURATION):q()}},c.prototype.applyPlacement=function(b,c){var d=this.tip(),e=d[0].offsetWidth,f=d[0].offsetHeight,g=parseInt(d.css("margin-top"),10),h=parseInt(d.css("margin-left"),10);isNaN(g)&&(g=0),isNaN(h)&&(h=0),b.top+=g,b.left+=h,a.offset.setOffset(d[0],a.extend({using:function(a){d.css({top:Math.round(a.top),left:Math.round(a.left)})}},b),0),d.addClass("in");var i=d[0].offsetWidth,j=d[0].offsetHeight;"top"==c&&j!=f&&(b.top=b.top+f-j);var k=this.getViewportAdjustedDelta(c,b,i,j);k.left?b.left+=k.left:b.top+=k.top;var l=/top|bottom/.test(c),m=l?2*k.left-e+i:2*k.top-f+j,n=l?"offsetWidth":"offsetHeight";d.offset(b),this.replaceArrow(m,d[0][n],l)},c.prototype.replaceArrow=function(a,b,c){this.arrow().css(c?"left":"top",50*(1-a/b)+"%").css(c?"top":"left","")},c.prototype.setContent=function(){var a=this.tip(),b=this.getTitle();a.find(".tooltip-inner")[this.options.html?"html":"text"](b),a.removeClass("fade in top bottom left right")},c.prototype.hide=function(b){function d(){"in"!=e.hoverState&&f.detach(),e.$element.removeAttr("aria-describedby").trigger("hidden.bs."+e.type),b&&b()}var e=this,f=a(this.$tip),g=a.Event("hide.bs."+this.type);return this.$element.trigger(g),g.isDefaultPrevented()?void 0:(f.removeClass("in"),a.support.transition&&f.hasClass("fade")?f.one("bsTransitionEnd",d).emulateTransitionEnd(c.TRANSITION_DURATION):d(),this.hoverState=null,this)},c.prototype.fixTitle=function(){var a=this.$element;(a.attr("title")||"string"!=typeof a.attr("data-original-title"))&&a.attr("data-original-title",a.attr("title")||"").attr("title","")},c.prototype.hasContent=function(){return this.getTitle()},c.prototype.getPosition=function(b){b=b||this.$element;var c=b[0],d="BODY"==c.tagName,e=c.getBoundingClientRect();null==e.width&&(e=a.extend({},e,{width:e.right-e.left,height:e.bottom-e.top}));var f=d?{top:0,left:0}:b.offset(),g={scroll:d?document.documentElement.scrollTop||document.body.scrollTop:b.scrollTop()},h=d?{width:a(window).width(),height:a(window).height()}:null;return a.extend({},e,g,h,f)},c.prototype.getCalculatedOffset=function(a,b,c,d){return"bottom"==a?{top:b.top+b.height,left:b.left+b.width/2-c/2}:"top"==a?{top:b.top-d,left:b.left+b.width/2-c/2}:"left"==a?{top:b.top+b.height/2-d/2,left:b.left-c}:{top:b.top+b.height/2-d/2,left:b.left+b.width}},c.prototype.getViewportAdjustedDelta=function(a,b,c,d){var e={top:0,left:0};if(!this.$viewport)return e;var f=this.options.viewport&&this.options.viewport.padding||0,g=this.getPosition(this.$viewport);if(/right|left/.test(a)){var h=b.top-f-g.scroll,i=b.top+f-g.scroll+d;h<g.top?e.top=g.top-h:i>g.top+g.height&&(e.top=g.top+g.height-i)}else{var j=b.left-f,k=b.left+f+c;j<g.left?e.left=g.left-j:k>g.right&&(e.left=g.left+g.width-k)}return e},c.prototype.getTitle=function(){var a,b=this.$element,c=this.options;return a=b.attr("data-original-title")||("function"==typeof c.title?c.title.call(b[0]):c.title)},c.prototype.getUID=function(a){do a+=~~(1e6*Math.random());while(document.getElementById(a));return a},c.prototype.tip=function(){if(!this.$tip&&(this.$tip=a(this.options.template),1!=this.$tip.length))throw new Error(this.type+" `template` option must consist of exactly 1 top-level element!");return this.$tip},c.prototype.arrow=function(){return this.$arrow=this.$arrow||this.tip().find(".tooltip-arrow")},c.prototype.enable=function(){this.enabled=!0},c.prototype.disable=function(){this.enabled=!1},c.prototype.toggleEnabled=function(){this.enabled=!this.enabled},c.prototype.toggle=function(b){var c=this;b&&(c=a(b.currentTarget).data("bs."+this.type),c||(c=new this.constructor(b.currentTarget,this.getDelegateOptions()),a(b.currentTarget).data("bs."+this.type,c))),b?(c.inState.click=!c.inState.click,c.isInStateTrue()?c.enter(c):c.leave(c)):c.tip().hasClass("in")?c.leave(c):c.enter(c)},c.prototype.destroy=function(){var a=this;clearTimeout(this.timeout),this.hide(function(){a.$element.off("."+a.type).removeData("bs."+a.type),a.$tip&&a.$tip.detach(),a.$tip=null,a.$arrow=null,a.$viewport=null})};var d=a.fn.tooltip;a.fn.tooltip=b,a.fn.tooltip.Constructor=c,a.fn.tooltip.noConflict=function(){return a.fn.tooltip=d,this}}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.popover"),f="object"==typeof b&&b;(e||!/destroy|hide/.test(b))&&(e||d.data("bs.popover",e=new c(this,f)),"string"==typeof b&&e[b]())})}var c=function(a,b){this.init("popover",a,b)};if(!a.fn.tooltip)throw new Error("Popover requires tooltip.js");c.VERSION="3.3.5",c.DEFAULTS=a.extend({},a.fn.tooltip.Constructor.DEFAULTS,{placement:"right",trigger:"click",content:"",template:'<div class="popover" role="tooltip"><div class="arrow"></div><h3 class="popover-title"></h3><div class="popover-content"></div></div>'}),c.prototype=a.extend({},a.fn.tooltip.Constructor.prototype),c.prototype.constructor=c,c.prototype.getDefaults=function(){return c.DEFAULTS},c.prototype.setContent=function(){var a=this.tip(),b=this.getTitle(),c=this.getContent();a.find(".popover-title")[this.options.html?"html":"text"](b),a.find(".popover-content").children().detach().end()[this.options.html?"string"==typeof c?"html":"append":"text"](c),a.removeClass("fade top bottom left right in"),a.find(".popover-title").html()||a.find(".popover-title").hide()},c.prototype.hasContent=function(){return this.getTitle()||this.getContent()},c.prototype.getContent=function(){var a=this.$element,b=this.options;return a.attr("data-content")||("function"==typeof b.content?b.content.call(a[0]):b.content)},c.prototype.arrow=function(){return this.$arrow=this.$arrow||this.tip().find(".arrow")};var d=a.fn.popover;a.fn.popover=b,a.fn.popover.Constructor=c,a.fn.popover.noConflict=function(){return a.fn.popover=d,this}}(jQuery),+function(a){"use strict";function b(c,d){this.$body=a(document.body),this.$scrollElement=a(a(c).is(document.body)?window:c),this.options=a.extend({},b.DEFAULTS,d),this.selector=(this.options.target||"")+" .nav li > a",this.offsets=[],this.targets=[],this.activeTarget=null,this.scrollHeight=0,this.$scrollElement.on("scroll.bs.scrollspy",a.proxy(this.process,this)),this.refresh(),this.process()}function c(c){return this.each(function(){var d=a(this),e=d.data("bs.scrollspy"),f="object"==typeof c&&c;e||d.data("bs.scrollspy",e=new b(this,f)),"string"==typeof c&&e[c]()})}b.VERSION="3.3.5",b.DEFAULTS={offset:10},b.prototype.getScrollHeight=function(){return this.$scrollElement[0].scrollHeight||Math.max(this.$body[0].scrollHeight,document.documentElement.scrollHeight)},b.prototype.refresh=function(){var b=this,c="offset",d=0;this.offsets=[],this.targets=[],this.scrollHeight=this.getScrollHeight(),a.isWindow(this.$scrollElement[0])||(c="position",d=this.$scrollElement.scrollTop()),this.$body.find(this.selector).map(function(){var b=a(this),e=b.data("target")||b.attr("href"),f=/^#./.test(e)&&a(e);return f&&f.length&&f.is(":visible")&&[[f[c]().top+d,e]]||null}).sort(function(a,b){return a[0]-b[0]}).each(function(){b.offsets.push(this[0]),b.targets.push(this[1])})},b.prototype.process=function(){var a,b=this.$scrollElement.scrollTop()+this.options.offset,c=this.getScrollHeight(),d=this.options.offset+c-this.$scrollElement.height(),e=this.offsets,f=this.targets,g=this.activeTarget;if(this.scrollHeight!=c&&this.refresh(),b>=d)return g!=(a=f[f.length-1])&&this.activate(a);if(g&&b<e[0])return this.activeTarget=null,this.clear();for(a=e.length;a--;)g!=f[a]&&b>=e[a]&&(void 0===e[a+1]||b<e[a+1])&&this.activate(f[a])},b.prototype.activate=function(b){this.activeTarget=b,this.clear();var c=this.selector+'[data-target="'+b+'"],'+this.selector+'[href="'+b+'"]',d=a(c).parents("li").addClass("active");d.parent(".dropdown-menu").length&&(d=d.closest("li.dropdown").addClass("active")),
d.trigger("activate.bs.scrollspy")},b.prototype.clear=function(){a(this.selector).parentsUntil(this.options.target,".active").removeClass("active")};var d=a.fn.scrollspy;a.fn.scrollspy=c,a.fn.scrollspy.Constructor=b,a.fn.scrollspy.noConflict=function(){return a.fn.scrollspy=d,this},a(window).on("load.bs.scrollspy.data-api",function(){a('[data-spy="scroll"]').each(function(){var b=a(this);c.call(b,b.data())})})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.tab");e||d.data("bs.tab",e=new c(this)),"string"==typeof b&&e[b]()})}var c=function(b){this.element=a(b)};c.VERSION="3.3.5",c.TRANSITION_DURATION=150,c.prototype.show=function(){var b=this.element,c=b.closest("ul:not(.dropdown-menu)"),d=b.data("target");if(d||(d=b.attr("href"),d=d&&d.replace(/.*(?=#[^\s]*$)/,"")),!b.parent("li").hasClass("active")){var e=c.find(".active:last a"),f=a.Event("hide.bs.tab",{relatedTarget:b[0]}),g=a.Event("show.bs.tab",{relatedTarget:e[0]});if(e.trigger(f),b.trigger(g),!g.isDefaultPrevented()&&!f.isDefaultPrevented()){var h=a(d);this.activate(b.closest("li"),c),this.activate(h,h.parent(),function(){e.trigger({type:"hidden.bs.tab",relatedTarget:b[0]}),b.trigger({type:"shown.bs.tab",relatedTarget:e[0]})})}}},c.prototype.activate=function(b,d,e){function f(){g.removeClass("active").find("> .dropdown-menu > .active").removeClass("active").end().find('[data-toggle="tab"]').attr("aria-expanded",!1),b.addClass("active").find('[data-toggle="tab"]').attr("aria-expanded",!0),h?(b[0].offsetWidth,b.addClass("in")):b.removeClass("fade"),b.parent(".dropdown-menu").length&&b.closest("li.dropdown").addClass("active").end().find('[data-toggle="tab"]').attr("aria-expanded",!0),e&&e()}var g=d.find("> .active"),h=e&&a.support.transition&&(g.length&&g.hasClass("fade")||!!d.find("> .fade").length);g.length&&h?g.one("bsTransitionEnd",f).emulateTransitionEnd(c.TRANSITION_DURATION):f(),g.removeClass("in")};var d=a.fn.tab;a.fn.tab=b,a.fn.tab.Constructor=c,a.fn.tab.noConflict=function(){return a.fn.tab=d,this};var e=function(c){c.preventDefault(),b.call(a(this),"show")};a(document).on("click.bs.tab.data-api",'[data-toggle="tab"]',e).on("click.bs.tab.data-api",'[data-toggle="pill"]',e)}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.affix"),f="object"==typeof b&&b;e||d.data("bs.affix",e=new c(this,f)),"string"==typeof b&&e[b]()})}var c=function(b,d){this.options=a.extend({},c.DEFAULTS,d),this.$target=a(this.options.target).on("scroll.bs.affix.data-api",a.proxy(this.checkPosition,this)).on("click.bs.affix.data-api",a.proxy(this.checkPositionWithEventLoop,this)),this.$element=a(b),this.affixed=null,this.unpin=null,this.pinnedOffset=null,this.checkPosition()};c.VERSION="3.3.5",c.RESET="affix affix-top affix-bottom",c.DEFAULTS={offset:0,target:window},c.prototype.getState=function(a,b,c,d){var e=this.$target.scrollTop(),f=this.$element.offset(),g=this.$target.height();if(null!=c&&"top"==this.affixed)return c>e?"top":!1;if("bottom"==this.affixed)return null!=c?e+this.unpin<=f.top?!1:"bottom":a-d>=e+g?!1:"bottom";var h=null==this.affixed,i=h?e:f.top,j=h?g:b;return null!=c&&c>=e?"top":null!=d&&i+j>=a-d?"bottom":!1},c.prototype.getPinnedOffset=function(){if(this.pinnedOffset)return this.pinnedOffset;this.$element.removeClass(c.RESET).addClass("affix");var a=this.$target.scrollTop(),b=this.$element.offset();return this.pinnedOffset=b.top-a},c.prototype.checkPositionWithEventLoop=function(){setTimeout(a.proxy(this.checkPosition,this),1)},c.prototype.checkPosition=function(){if(this.$element.is(":visible")){var b=this.$element.height(),d=this.options.offset,e=d.top,f=d.bottom,g=Math.max(a(document).height(),a(document.body).height());"object"!=typeof d&&(f=e=d),"function"==typeof e&&(e=d.top(this.$element)),"function"==typeof f&&(f=d.bottom(this.$element));var h=this.getState(g,b,e,f);if(this.affixed!=h){null!=this.unpin&&this.$element.css("top","");var i="affix"+(h?"-"+h:""),j=a.Event(i+".bs.affix");if(this.$element.trigger(j),j.isDefaultPrevented())return;this.affixed=h,this.unpin="bottom"==h?this.getPinnedOffset():null,this.$element.removeClass(c.RESET).addClass(i).trigger(i.replace("affix","affixed")+".bs.affix")}"bottom"==h&&this.$element.offset({top:g-b-f})}};var d=a.fn.affix;a.fn.affix=b,a.fn.affix.Constructor=c,a.fn.affix.noConflict=function(){return a.fn.affix=d,this},a(window).on("load",function(){a('[data-spy="affix"]').each(function(){var c=a(this),d=c.data();d.offset=d.offset||{},null!=d.offsetBottom&&(d.offset.bottom=d.offsetBottom),null!=d.offsetTop&&(d.offset.top=d.offsetTop),b.call(c,d)})})}(jQuery);</script>
<script>/**
* @preserve HTML5 Shiv 3.7.2 | @afarkas @jdalton @jon_neal @rem | MIT/GPL2 Licensed
*/
// Only run this code in IE 8
if (!!window.navigator.userAgent.match("MSIE 8")) {
!function(a,b){function c(a,b){var c=a.createElement("p"),d=a.getElementsByTagName("head")[0]||a.documentElement;return c.innerHTML="x<style>"+b+"</style>",d.insertBefore(c.lastChild,d.firstChild)}function d(){var a=t.elements;return"string"==typeof a?a.split(" "):a}function e(a,b){var c=t.elements;"string"!=typeof c&&(c=c.join(" ")),"string"!=typeof a&&(a=a.join(" ")),t.elements=c+" "+a,j(b)}function f(a){var b=s[a[q]];return b||(b={},r++,a[q]=r,s[r]=b),b}function g(a,c,d){if(c||(c=b),l)return c.createElement(a);d||(d=f(c));var e;return e=d.cache[a]?d.cache[a].cloneNode():p.test(a)?(d.cache[a]=d.createElem(a)).cloneNode():d.createElem(a),!e.canHaveChildren||o.test(a)||e.tagUrn?e:d.frag.appendChild(e)}function h(a,c){if(a||(a=b),l)return a.createDocumentFragment();c=c||f(a);for(var e=c.frag.cloneNode(),g=0,h=d(),i=h.length;i>g;g++)e.createElement(h[g]);return e}function i(a,b){b.cache||(b.cache={},b.createElem=a.createElement,b.createFrag=a.createDocumentFragment,b.frag=b.createFrag()),a.createElement=function(c){return t.shivMethods?g(c,a,b):b.createElem(c)},a.createDocumentFragment=Function("h,f","return function(){var n=f.cloneNode(),c=n.createElement;h.shivMethods&&("+d().join().replace(/[\w\-:]+/g,function(a){return b.createElem(a),b.frag.createElement(a),'c("'+a+'")'})+");return n}")(t,b.frag)}function j(a){a||(a=b);var d=f(a);return!t.shivCSS||k||d.hasCSS||(d.hasCSS=!!c(a,"article,aside,dialog,figcaption,figure,footer,header,hgroup,main,nav,section{display:block}mark{background:#FF0;color:#000}template{display:none}")),l||i(a,d),a}var k,l,m="3.7.2",n=a.html5||{},o=/^<|^(?:button|map|select|textarea|object|iframe|option|optgroup)$/i,p=/^(?:a|b|code|div|fieldset|h1|h2|h3|h4|h5|h6|i|label|li|ol|p|q|span|strong|style|table|tbody|td|th|tr|ul)$/i,q="_html5shiv",r=0,s={};!function(){try{var a=b.createElement("a");a.innerHTML="<xyz></xyz>",k="hidden"in a,l=1==a.childNodes.length||function(){b.createElement("a");var a=b.createDocumentFragment();return"undefined"==typeof a.cloneNode||"undefined"==typeof a.createDocumentFragment||"undefined"==typeof a.createElement}()}catch(c){k=!0,l=!0}}();var t={elements:n.elements||"abbr article aside audio bdi canvas data datalist details dialog figcaption figure footer header hgroup main mark meter nav output picture progress section summary template time video",version:m,shivCSS:n.shivCSS!==!1,supportsUnknownElements:l,shivMethods:n.shivMethods!==!1,type:"default",shivDocument:j,createElement:g,createDocumentFragment:h,addElements:e};a.html5=t,j(b)}(this,document);
};
</script>
<script>/*! Respond.js v1.4.2: min/max-width media query polyfill * Copyright 2013 Scott Jehl
 * Licensed under https://github.com/scottjehl/Respond/blob/master/LICENSE-MIT
 *  */

// Only run this code in IE 8
if (!!window.navigator.userAgent.match("MSIE 8")) {
!function(a){"use strict";a.matchMedia=a.matchMedia||function(a){var b,c=a.documentElement,d=c.firstElementChild||c.firstChild,e=a.createElement("body"),f=a.createElement("div");return f.id="mq-test-1",f.style.cssText="position:absolute;top:-100em",e.style.background="none",e.appendChild(f),function(a){return f.innerHTML='&shy;<style media="'+a+'"> #mq-test-1 { width: 42px; }</style>',c.insertBefore(e,d),b=42===f.offsetWidth,c.removeChild(e),{matches:b,media:a}}}(a.document)}(this),function(a){"use strict";function b(){u(!0)}var c={};a.respond=c,c.update=function(){};var d=[],e=function(){var b=!1;try{b=new a.XMLHttpRequest}catch(c){b=new a.ActiveXObject("Microsoft.XMLHTTP")}return function(){return b}}(),f=function(a,b){var c=e();c&&(c.open("GET",a,!0),c.onreadystatechange=function(){4!==c.readyState||200!==c.status&&304!==c.status||b(c.responseText)},4!==c.readyState&&c.send(null))};if(c.ajax=f,c.queue=d,c.regex={media:/@media[^\{]+\{([^\{\}]*\{[^\}\{]*\})+/gi,keyframes:/@(?:\-(?:o|moz|webkit)\-)?keyframes[^\{]+\{(?:[^\{\}]*\{[^\}\{]*\})+[^\}]*\}/gi,urls:/(url\()['"]?([^\/\)'"][^:\)'"]+)['"]?(\))/g,findStyles:/@media *([^\{]+)\{([\S\s]+?)$/,only:/(only\s+)?([a-zA-Z]+)\s?/,minw:/\([\s]*min\-width\s*:[\s]*([\s]*[0-9\.]+)(px|em)[\s]*\)/,maxw:/\([\s]*max\-width\s*:[\s]*([\s]*[0-9\.]+)(px|em)[\s]*\)/},c.mediaQueriesSupported=a.matchMedia&&null!==a.matchMedia("only all")&&a.matchMedia("only all").matches,!c.mediaQueriesSupported){var g,h,i,j=a.document,k=j.documentElement,l=[],m=[],n=[],o={},p=30,q=j.getElementsByTagName("head")[0]||k,r=j.getElementsByTagName("base")[0],s=q.getElementsByTagName("link"),t=function(){var a,b=j.createElement("div"),c=j.body,d=k.style.fontSize,e=c&&c.style.fontSize,f=!1;return b.style.cssText="position:absolute;font-size:1em;width:1em",c||(c=f=j.createElement("body"),c.style.background="none"),k.style.fontSize="100%",c.style.fontSize="100%",c.appendChild(b),f&&k.insertBefore(c,k.firstChild),a=b.offsetWidth,f?k.removeChild(c):c.removeChild(b),k.style.fontSize=d,e&&(c.style.fontSize=e),a=i=parseFloat(a)},u=function(b){var c="clientWidth",d=k[c],e="CSS1Compat"===j.compatMode&&d||j.body[c]||d,f={},o=s[s.length-1],r=(new Date).getTime();if(b&&g&&p>r-g)return a.clearTimeout(h),h=a.setTimeout(u,p),void 0;g=r;for(var v in l)if(l.hasOwnProperty(v)){var w=l[v],x=w.minw,y=w.maxw,z=null===x,A=null===y,B="em";x&&(x=parseFloat(x)*(x.indexOf(B)>-1?i||t():1)),y&&(y=parseFloat(y)*(y.indexOf(B)>-1?i||t():1)),w.hasquery&&(z&&A||!(z||e>=x)||!(A||y>=e))||(f[w.media]||(f[w.media]=[]),f[w.media].push(m[w.rules]))}for(var C in n)n.hasOwnProperty(C)&&n[C]&&n[C].parentNode===q&&q.removeChild(n[C]);n.length=0;for(var D in f)if(f.hasOwnProperty(D)){var E=j.createElement("style"),F=f[D].join("\n");E.type="text/css",E.media=D,q.insertBefore(E,o.nextSibling),E.styleSheet?E.styleSheet.cssText=F:E.appendChild(j.createTextNode(F)),n.push(E)}},v=function(a,b,d){var e=a.replace(c.regex.keyframes,"").match(c.regex.media),f=e&&e.length||0;b=b.substring(0,b.lastIndexOf("/"));var g=function(a){return a.replace(c.regex.urls,"$1"+b+"$2$3")},h=!f&&d;b.length&&(b+="/"),h&&(f=1);for(var i=0;f>i;i++){var j,k,n,o;h?(j=d,m.push(g(a))):(j=e[i].match(c.regex.findStyles)&&RegExp.$1,m.push(RegExp.$2&&g(RegExp.$2))),n=j.split(","),o=n.length;for(var p=0;o>p;p++)k=n[p],l.push({media:k.split("(")[0].match(c.regex.only)&&RegExp.$2||"all",rules:m.length-1,hasquery:k.indexOf("(")>-1,minw:k.match(c.regex.minw)&&parseFloat(RegExp.$1)+(RegExp.$2||""),maxw:k.match(c.regex.maxw)&&parseFloat(RegExp.$1)+(RegExp.$2||"")})}u()},w=function(){if(d.length){var b=d.shift();f(b.href,function(c){v(c,b.href,b.media),o[b.href]=!0,a.setTimeout(function(){w()},0)})}},x=function(){for(var b=0;b<s.length;b++){var c=s[b],e=c.href,f=c.media,g=c.rel&&"stylesheet"===c.rel.toLowerCase();e&&g&&!o[e]&&(c.styleSheet&&c.styleSheet.rawCssText?(v(c.styleSheet.rawCssText,e,f),o[e]=!0):(!/^([a-zA-Z:]*\/\/)/.test(e)&&!r||e.replace(RegExp.$1,"").split("/")[0]===a.location.host)&&("//"===e.substring(0,2)&&(e=a.location.protocol+e),d.push({href:e,media:f})))}w()};x(),c.update=x,c.getEmValue=t,a.addEventListener?a.addEventListener("resize",b,!1):a.attachEvent&&a.attachEvent("onresize",b)}}(this);
};
</script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script>

/**
 * jQuery Plugin: Sticky Tabs
 *
 * @author Aidan Lister <aidan@php.net>
 * adapted by Ruben Arslan to activate parent tabs too
 * http://www.aidanlister.com/2014/03/persisting-the-tab-state-in-bootstrap/
 */
(function($) {
  "use strict";
  $.fn.rmarkdownStickyTabs = function() {
    var context = this;
    // Show the tab corresponding with the hash in the URL, or the first tab
    var showStuffFromHash = function() {
      var hash = window.location.hash;
      var selector = hash ? 'a[href="' + hash + '"]' : 'li.active > a';
      var $selector = $(selector, context);
      if($selector.data('toggle') === "tab") {
        $selector.tab('show');
        // walk up the ancestors of this element, show any hidden tabs
        $selector.parents('.section.tabset').each(function(i, elm) {
          var link = $('a[href="#' + $(elm).attr('id') + '"]');
          if(link.data('toggle') === "tab") {
            link.tab("show");
          }
        });
      }
    };


    // Set the correct tab when the page loads
    showStuffFromHash(context);

    // Set the correct tab when a user uses their back/forward button
    $(window).on('hashchange', function() {
      showStuffFromHash(context);
    });

    // Change the URL when tabs are clicked
    $('a', context).on('click', function(e) {
      history.pushState(null, null, this.href);
      showStuffFromHash(context);
    });

    return this;
  };
}(jQuery));

window.buildTabsets = function(tocID) {

  // build a tabset from a section div with the .tabset class
  function buildTabset(tabset) {

    // check for fade and pills options
    var fade = tabset.hasClass("tabset-fade");
    var pills = tabset.hasClass("tabset-pills");
    var navClass = pills ? "nav-pills" : "nav-tabs";

    // determine the heading level of the tabset and tabs
    var match = tabset.attr('class').match(/level(\d) /);
    if (match === null)
      return;
    var tabsetLevel = Number(match[1]);
    var tabLevel = tabsetLevel + 1;

    // find all subheadings immediately below
    var tabs = tabset.find("div.section.level" + tabLevel);
    if (!tabs.length)
      return;

    // create tablist and tab-content elements
    var tabList = $('<ul class="nav ' + navClass + '" role="tablist"></ul>');
    $(tabs[0]).before(tabList);
    var tabContent = $('<div class="tab-content"></div>');
    $(tabs[0]).before(tabContent);

    // build the tabset
    var activeTab = 0;
    tabs.each(function(i) {

      // get the tab div
      var tab = $(tabs[i]);

      // get the id then sanitize it for use with bootstrap tabs
      var id = tab.attr('id');

      // see if this is marked as the active tab
      if (tab.hasClass('active'))
        activeTab = i;

      // remove any table of contents entries associated with
      // this ID (since we'll be removing the heading element)
      $("div#" + tocID + " li a[href='#" + id + "']").parent().remove();

      // sanitize the id for use with bootstrap tabs
      id = id.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_');
      tab.attr('id', id);

      // get the heading element within it, grab it's text, then remove it
      var heading = tab.find('h' + tabLevel + ':first');
      var headingText = heading.html();
      heading.remove();

      // build and append the tab list item
      var a = $('<a role="tab" data-toggle="tab">' + headingText + '</a>');
      a.attr('href', '#' + id);
      a.attr('aria-controls', id);
      var li = $('<li role="presentation"></li>');
      li.append(a);
      tabList.append(li);

      // set it's attributes
      tab.attr('role', 'tabpanel');
      tab.addClass('tab-pane');
      tab.addClass('tabbed-pane');
      if (fade)
        tab.addClass('fade');

      // move it into the tab content div
      tab.detach().appendTo(tabContent);
    });

    // set active tab
    $(tabList.children('li')[activeTab]).addClass('active');
    var active = $(tabContent.children('div.section')[activeTab]);
    active.addClass('active');
    if (fade)
      active.addClass('in');

    if (tabset.hasClass("tabset-sticky"))
      tabset.rmarkdownStickyTabs();
  }

  // convert section divs with the .tabset class to tabsets
  var tabsets = $("div.section.tabset");
  tabsets.each(function(i) {
    buildTabset($(tabsets[i]));
  });
};

</script>
<style type="text/css">.hljs-literal {
color: #990073;
}
.hljs-number {
color: #099;
}
.hljs-comment {
color: #998;
font-style: italic;
}
.hljs-keyword {
color: #900;
font-weight: bold;
}
.hljs-string {
color: #d14;
}
</style>
<script src="data:application/javascript;base64,/*! highlight.js v9.12.0 | BSD3 License | git.io/hljslicense */
!function(e){var n="object"==typeof window&&window||"object"==typeof self&&self;"undefined"!=typeof exports?e(exports):n&&(n.hljs=e({}),"function"==typeof define&&define.amd&&define([],function(){return n.hljs}))}(function(e){function n(e){return e.replace(/&/g,"&amp;").replace(/</g,"&lt;").replace(/>/g,"&gt;")}function t(e){return e.nodeName.toLowerCase()}function r(e,n){var t=e&&e.exec(n);return t&&0===t.index}function a(e){return k.test(e)}function i(e){var n,t,r,i,o=e.className+" ";if(o+=e.parentNode?e.parentNode.className:"",t=B.exec(o))return w(t[1])?t[1]:"no-highlight";for(o=o.split(/\s+/),n=0,r=o.length;r>n;n++)if(i=o[n],a(i)||w(i))return i}function o(e){var n,t={},r=Array.prototype.slice.call(arguments,1);for(n in e)t[n]=e[n];return r.forEach(function(e){for(n in e)t[n]=e[n]}),t}function u(e){var n=[];return function r(e,a){for(var i=e.firstChild;i;i=i.nextSibling)3===i.nodeType?a+=i.nodeValue.length:1===i.nodeType&&(n.push({event:"start",offset:a,node:i}),a=r(i,a),t(i).match(/br|hr|img|input/)||n.push({event:"stop",offset:a,node:i}));return a}(e,0),n}function c(e,r,a){function i(){return e.length&&r.length?e[0].offset!==r[0].offset?e[0].offset<r[0].offset?e:r:"start"===r[0].event?e:r:e.length?e:r}function o(e){function r(e){return" "+e.nodeName+'="'+n(e.value).replace('"',"&quot;")+'"'}s+="<"+t(e)+E.map.call(e.attributes,r).join("")+">"}function u(e){s+="</"+t(e)+">"}function c(e){("start"===e.event?o:u)(e.node)}for(var l=0,s="",f=[];e.length||r.length;){var g=i();if(s+=n(a.substring(l,g[0].offset)),l=g[0].offset,g===e){f.reverse().forEach(u);do c(g.splice(0,1)[0]),g=i();while(g===e&&g.length&&g[0].offset===l);f.reverse().forEach(o)}else"start"===g[0].event?f.push(g[0].node):f.pop(),c(g.splice(0,1)[0])}return s+n(a.substr(l))}function l(e){return e.v&&!e.cached_variants&&(e.cached_variants=e.v.map(function(n){return o(e,{v:null},n)})),e.cached_variants||e.eW&&[o(e)]||[e]}function s(e){function n(e){return e&&e.source||e}function t(t,r){return new RegExp(n(t),"m"+(e.cI?"i":"")+(r?"g":""))}function r(a,i){if(!a.compiled){if(a.compiled=!0,a.k=a.k||a.bK,a.k){var o={},u=function(n,t){e.cI&&(t=t.toLowerCase()),t.split(" ").forEach(function(e){var t=e.split("|");o[t[0]]=[n,t[1]?Number(t[1]):1]})};"string"==typeof a.k?u("keyword",a.k):x(a.k).forEach(function(e){u(e,a.k[e])}),a.k=o}a.lR=t(a.l||/\w+/,!0),i&&(a.bK&&(a.b="\\b("+a.bK.split(" ").join("|")+")\\b"),a.b||(a.b=/\B|\b/),a.bR=t(a.b),a.e||a.eW||(a.e=/\B|\b/),a.e&&(a.eR=t(a.e)),a.tE=n(a.e)||"",a.eW&&i.tE&&(a.tE+=(a.e?"|":"")+i.tE)),a.i&&(a.iR=t(a.i)),null==a.r&&(a.r=1),a.c||(a.c=[]),a.c=Array.prototype.concat.apply([],a.c.map(function(e){return l("self"===e?a:e)})),a.c.forEach(function(e){r(e,a)}),a.starts&&r(a.starts,i);var c=a.c.map(function(e){return e.bK?"\\.?("+e.b+")\\.?":e.b}).concat([a.tE,a.i]).map(n).filter(Boolean);a.t=c.length?t(c.join("|"),!0):{exec:function(){return null}}}}r(e)}function f(e,t,a,i){function o(e,n){var t,a;for(t=0,a=n.c.length;a>t;t++)if(r(n.c[t].bR,e))return n.c[t]}function u(e,n){if(r(e.eR,n)){for(;e.endsParent&&e.parent;)e=e.parent;return e}return e.eW?u(e.parent,n):void 0}function c(e,n){return!a&&r(n.iR,e)}function l(e,n){var t=N.cI?n[0].toLowerCase():n[0];return e.k.hasOwnProperty(t)&&e.k[t]}function p(e,n,t,r){var a=r?"":I.classPrefix,i='<span class="'+a,o=t?"":C;return i+=e+'">',i+n+o}function h(){var e,t,r,a;if(!E.k)return n(k);for(a="",t=0,E.lR.lastIndex=0,r=E.lR.exec(k);r;)a+=n(k.substring(t,r.index)),e=l(E,r),e?(B+=e[1],a+=p(e[0],n(r[0]))):a+=n(r[0]),t=E.lR.lastIndex,r=E.lR.exec(k);return a+n(k.substr(t))}function d(){var e="string"==typeof E.sL;if(e&&!y[E.sL])return n(k);var t=e?f(E.sL,k,!0,x[E.sL]):g(k,E.sL.length?E.sL:void 0);return E.r>0&&(B+=t.r),e&&(x[E.sL]=t.top),p(t.language,t.value,!1,!0)}function b(){L+=null!=E.sL?d():h(),k=""}function v(e){L+=e.cN?p(e.cN,"",!0):"",E=Object.create(e,{parent:{value:E}})}function m(e,n){if(k+=e,null==n)return b(),0;var t=o(n,E);if(t)return t.skip?k+=n:(t.eB&&(k+=n),b(),t.rB||t.eB||(k=n)),v(t,n),t.rB?0:n.length;var r=u(E,n);if(r){var a=E;a.skip?k+=n:(a.rE||a.eE||(k+=n),b(),a.eE&&(k=n));do E.cN&&(L+=C),E.skip||(B+=E.r),E=E.parent;while(E!==r.parent);return r.starts&&v(r.starts,""),a.rE?0:n.length}if(c(n,E))throw new Error('Illegal lexeme "'+n+'" for mode "'+(E.cN||"<unnamed>")+'"');return k+=n,n.length||1}var N=w(e);if(!N)throw new Error('Unknown language: "'+e+'"');s(N);var R,E=i||N,x={},L="";for(R=E;R!==N;R=R.parent)R.cN&&(L=p(R.cN,"",!0)+L);var k="",B=0;try{for(var M,j,O=0;;){if(E.t.lastIndex=O,M=E.t.exec(t),!M)break;j=m(t.substring(O,M.index),M[0]),O=M.index+j}for(m(t.substr(O)),R=E;R.parent;R=R.parent)R.cN&&(L+=C);return{r:B,value:L,language:e,top:E}}catch(T){if(T.message&&-1!==T.message.indexOf("Illegal"))return{r:0,value:n(t)};throw T}}function g(e,t){t=t||I.languages||x(y);var r={r:0,value:n(e)},a=r;return t.filter(w).forEach(function(n){var t=f(n,e,!1);t.language=n,t.r>a.r&&(a=t),t.r>r.r&&(a=r,r=t)}),a.language&&(r.second_best=a),r}function p(e){return I.tabReplace||I.useBR?e.replace(M,function(e,n){return I.useBR&&"\n"===e?"<br>":I.tabReplace?n.replace(/\t/g,I.tabReplace):""}):e}function h(e,n,t){var r=n?L[n]:t,a=[e.trim()];return e.match(/\bhljs\b/)||a.push("hljs"),-1===e.indexOf(r)&&a.push(r),a.join(" ").trim()}function d(e){var n,t,r,o,l,s=i(e);a(s)||(I.useBR?(n=document.createElementNS("http://www.w3.org/1999/xhtml","div"),n.innerHTML=e.innerHTML.replace(/\n/g,"").replace(/<br[ \/]*>/g,"\n")):n=e,l=n.textContent,r=s?f(s,l,!0):g(l),t=u(n),t.length&&(o=document.createElementNS("http://www.w3.org/1999/xhtml","div"),o.innerHTML=r.value,r.value=c(t,u(o),l)),r.value=p(r.value),e.innerHTML=r.value,e.className=h(e.className,s,r.language),e.result={language:r.language,re:r.r},r.second_best&&(e.second_best={language:r.second_best.language,re:r.second_best.r}))}function b(e){I=o(I,e)}function v(){if(!v.called){v.called=!0;var e=document.querySelectorAll("pre code");E.forEach.call(e,d)}}function m(){addEventListener("DOMContentLoaded",v,!1),addEventListener("load",v,!1)}function N(n,t){var r=y[n]=t(e);r.aliases&&r.aliases.forEach(function(e){L[e]=n})}function R(){return x(y)}function w(e){return e=(e||"").toLowerCase(),y[e]||y[L[e]]}var E=[],x=Object.keys,y={},L={},k=/^(no-?highlight|plain|text)$/i,B=/\blang(?:uage)?-([\w-]+)\b/i,M=/((^(<[^>]+>|\t|)+|(?:\n)))/gm,C="</span>",I={classPrefix:"hljs-",tabReplace:null,useBR:!1,languages:void 0};return e.highlight=f,e.highlightAuto=g,e.fixMarkup=p,e.highlightBlock=d,e.configure=b,e.initHighlighting=v,e.initHighlightingOnLoad=m,e.registerLanguage=N,e.listLanguages=R,e.getLanguage=w,e.inherit=o,e.IR="[a-zA-Z]\\w*",e.UIR="[a-zA-Z_]\\w*",e.NR="\\b\\d+(\\.\\d+)?",e.CNR="(-?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)",e.BNR="\\b(0b[01]+)",e.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|-|-=|/=|/|:|;|<<|<<=|<=|<|===|==|=|>>>=|>>=|>=|>>>|>>|>|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~",e.BE={b:"\\\\[\\s\\S]",r:0},e.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[e.BE]},e.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[e.BE]},e.PWM={b:/\b(a|an|the|are|I'm|isn't|don't|doesn't|won't|but|just|should|pretty|simply|enough|gonna|going|wtf|so|such|will|you|your|they|like|more)\b/},e.C=function(n,t,r){var a=e.inherit({cN:"comment",b:n,e:t,c:[]},r||{});return a.c.push(e.PWM),a.c.push({cN:"doctag",b:"(?:TODO|FIXME|NOTE|BUG|XXX):",r:0}),a},e.CLCM=e.C("//","$"),e.CBCM=e.C("/\\*","\\*/"),e.HCM=e.C("#","$"),e.NM={cN:"number",b:e.NR,r:0},e.CNM={cN:"number",b:e.CNR,r:0},e.BNM={cN:"number",b:e.BNR,r:0},e.CSSNM={cN:"number",b:e.NR+"(%|em|ex|ch|rem|vw|vh|vmin|vmax|cm|mm|in|pt|pc|px|deg|grad|rad|turn|s|ms|Hz|kHz|dpi|dpcm|dppx)?",r:0},e.RM={cN:"regexp",b:/\//,e:/\/[gimuy]*/,i:/\n/,c:[e.BE,{b:/\[/,e:/\]/,r:0,c:[e.BE]}]},e.TM={cN:"title",b:e.IR,r:0},e.UTM={cN:"title",b:e.UIR,r:0},e.METHOD_GUARD={b:"\\.\\s*"+e.UIR,r:0},e});hljs.registerLanguage("sql",function(e){var t=e.C("--","$");return{cI:!0,i:/[<>{}*#]/,c:[{bK:"begin end start commit rollback savepoint lock alter create drop rename call delete do handler insert load replace select truncate update set show pragma grant merge describe use explain help declare prepare execute deallocate release unlock purge reset change stop analyze cache flush optimize repair kill install uninstall checksum restore check backup revoke comment",e:/;/,eW:!0,l:/[\w\.]+/,k:{keyword:"abort abs absolute acc acce accep accept access accessed accessible account acos action activate add addtime admin administer advanced advise aes_decrypt aes_encrypt after agent aggregate ali alia alias allocate allow alter always analyze ancillary and any anydata anydataset anyschema anytype apply archive archived archivelog are as asc ascii asin assembly assertion associate asynchronous at atan atn2 attr attri attrib attribu attribut attribute attributes audit authenticated authentication authid authors auto autoallocate autodblink autoextend automatic availability avg backup badfile basicfile before begin beginning benchmark between bfile bfile_base big bigfile bin binary_double binary_float binlog bit_and bit_count bit_length bit_or bit_xor bitmap blob_base block blocksize body both bound buffer_cache buffer_pool build bulk by byte byteordermark bytes cache caching call calling cancel capacity cascade cascaded case cast catalog category ceil ceiling chain change changed char_base char_length character_length characters characterset charindex charset charsetform charsetid check checksum checksum_agg child choose chr chunk class cleanup clear client clob clob_base clone close cluster_id cluster_probability cluster_set clustering coalesce coercibility col collate collation collect colu colum column column_value columns columns_updated comment commit compact compatibility compiled complete composite_limit compound compress compute concat concat_ws concurrent confirm conn connec connect connect_by_iscycle connect_by_isleaf connect_by_root connect_time connection consider consistent constant constraint constraints constructor container content contents context contributors controlfile conv convert convert_tz corr corr_k corr_s corresponding corruption cos cost count count_big counted covar_pop covar_samp cpu_per_call cpu_per_session crc32 create creation critical cross cube cume_dist curdate current current_date current_time current_timestamp current_user cursor curtime customdatum cycle data database databases datafile datafiles datalength date_add date_cache date_format date_sub dateadd datediff datefromparts datename datepart datetime2fromparts day day_to_second dayname dayofmonth dayofweek dayofyear days db_role_change dbtimezone ddl deallocate declare decode decompose decrement decrypt deduplicate def defa defau defaul default defaults deferred defi defin define degrees delayed delegate delete delete_all delimited demand dense_rank depth dequeue des_decrypt des_encrypt des_key_file desc descr descri describ describe descriptor deterministic diagnostics difference dimension direct_load directory disable disable_all disallow disassociate discardfile disconnect diskgroup distinct distinctrow distribute distributed div do document domain dotnet double downgrade drop dumpfile duplicate duration each edition editionable editions element ellipsis else elsif elt empty enable enable_all enclosed encode encoding encrypt end end-exec endian enforced engine engines enqueue enterprise entityescaping eomonth error errors escaped evalname evaluate event eventdata events except exception exceptions exchange exclude excluding execu execut execute exempt exists exit exp expire explain export export_set extended extent external external_1 external_2 externally extract failed failed_login_attempts failover failure far fast feature_set feature_value fetch field fields file file_name_convert filesystem_like_logging final finish first first_value fixed flash_cache flashback floor flush following follows for forall force form forma format found found_rows freelist freelists freepools fresh from from_base64 from_days ftp full function general generated get get_format get_lock getdate getutcdate global global_name globally go goto grant grants greatest group group_concat group_id grouping grouping_id groups gtid_subtract guarantee guard handler hash hashkeys having hea head headi headin heading heap help hex hierarchy high high_priority hosts hour http id ident_current ident_incr ident_seed identified identity idle_time if ifnull ignore iif ilike ilm immediate import in include including increment index indexes indexing indextype indicator indices inet6_aton inet6_ntoa inet_aton inet_ntoa infile initial initialized initially initrans inmemory inner innodb input insert install instance instantiable instr interface interleaved intersect into invalidate invisible is is_free_lock is_ipv4 is_ipv4_compat is_not is_not_null is_used_lock isdate isnull isolation iterate java join json json_exists keep keep_duplicates key keys kill language large last last_day last_insert_id last_value lax lcase lead leading least leaves left len lenght length less level levels library like like2 like4 likec limit lines link list listagg little ln load load_file lob lobs local localtime localtimestamp locate locator lock locked log log10 log2 logfile logfiles logging logical logical_reads_per_call logoff logon logs long loop low low_priority lower lpad lrtrim ltrim main make_set makedate maketime managed management manual map mapping mask master master_pos_wait match matched materialized max maxextents maximize maxinstances maxlen maxlogfiles maxloghistory maxlogmembers maxsize maxtrans md5 measures median medium member memcompress memory merge microsecond mid migration min minextents minimum mining minus minute minvalue missing mod mode model modification modify module monitoring month months mount move movement multiset mutex name name_const names nan national native natural nav nchar nclob nested never new newline next nextval no no_write_to_binlog noarchivelog noaudit nobadfile nocheck nocompress nocopy nocycle nodelay nodiscardfile noentityescaping noguarantee nokeep nologfile nomapping nomaxvalue nominimize nominvalue nomonitoring none noneditionable nonschema noorder nopr nopro noprom nopromp noprompt norely noresetlogs noreverse normal norowdependencies noschemacheck noswitch not nothing notice notrim novalidate now nowait nth_value nullif nulls num numb numbe nvarchar nvarchar2 object ocicoll ocidate ocidatetime ociduration ociinterval ociloblocator ocinumber ociref ocirefcursor ocirowid ocistring ocitype oct octet_length of off offline offset oid oidindex old on online only opaque open operations operator optimal optimize option optionally or oracle oracle_date oradata ord ordaudio orddicom orddoc order ordimage ordinality ordvideo organization orlany orlvary out outer outfile outline output over overflow overriding package pad parallel parallel_enable parameters parent parse partial partition partitions pascal passing password password_grace_time password_lock_time password_reuse_max password_reuse_time password_verify_function patch path patindex pctincrease pctthreshold pctused pctversion percent percent_rank percentile_cont percentile_disc performance period period_add period_diff permanent physical pi pipe pipelined pivot pluggable plugin policy position post_transaction pow power pragma prebuilt precedes preceding precision prediction prediction_cost prediction_details prediction_probability prediction_set prepare present preserve prior priority private private_sga privileges procedural procedure procedure_analyze processlist profiles project prompt protection public publishingservername purge quarter query quick quiesce quota quotename radians raise rand range rank raw read reads readsize rebuild record records recover recovery recursive recycle redo reduced ref reference referenced references referencing refresh regexp_like register regr_avgx regr_avgy regr_count regr_intercept regr_r2 regr_slope regr_sxx regr_sxy reject rekey relational relative relaylog release release_lock relies_on relocate rely rem remainder rename repair repeat replace replicate replication required reset resetlogs resize resource respect restore restricted result result_cache resumable resume retention return returning returns reuse reverse revoke right rlike role roles rollback rolling rollup round row row_count rowdependencies rowid rownum rows rtrim rules safe salt sample save savepoint sb1 sb2 sb4 scan schema schemacheck scn scope scroll sdo_georaster sdo_topo_geometry search sec_to_time second section securefile security seed segment select self sequence sequential serializable server servererror session session_user sessions_per_user set sets settings sha sha1 sha2 share shared shared_pool short show shrink shutdown si_averagecolor si_colorhistogram si_featurelist si_positionalcolor si_stillimage si_texture siblings sid sign sin size size_t sizes skip slave sleep smalldatetimefromparts smallfile snapshot some soname sort soundex source space sparse spfile split sql sql_big_result sql_buffer_result sql_cache sql_calc_found_rows sql_small_result sql_variant_property sqlcode sqldata sqlerror sqlname sqlstate sqrt square standalone standby start starting startup statement static statistics stats_binomial_test stats_crosstab stats_ks_test stats_mode stats_mw_test stats_one_way_anova stats_t_test_ stats_t_test_indep stats_t_test_one stats_t_test_paired stats_wsr_test status std stddev stddev_pop stddev_samp stdev stop storage store stored str str_to_date straight_join strcmp strict string struct stuff style subdate subpartition subpartitions substitutable substr substring subtime subtring_index subtype success sum suspend switch switchoffset switchover sync synchronous synonym sys sys_xmlagg sysasm sysaux sysdate sysdatetimeoffset sysdba sysoper system system_user sysutcdatetime table tables tablespace tan tdo template temporary terminated tertiary_weights test than then thread through tier ties time time_format time_zone timediff timefromparts timeout timestamp timestampadd timestampdiff timezone_abbr timezone_minute timezone_region to to_base64 to_date to_days to_seconds todatetimeoffset trace tracking transaction transactional translate translation treat trigger trigger_nestlevel triggers trim truncate try_cast try_convert try_parse type ub1 ub2 ub4 ucase unarchived unbounded uncompress under undo unhex unicode uniform uninstall union unique unix_timestamp unknown unlimited unlock unpivot unrecoverable unsafe unsigned until untrusted unusable unused update updated upgrade upped upper upsert url urowid usable usage use use_stored_outlines user user_data user_resources users using utc_date utc_timestamp uuid uuid_short validate validate_password_strength validation valist value values var var_samp varcharc vari varia variab variabl variable variables variance varp varraw varrawc varray verify version versions view virtual visible void wait wallet warning warnings week weekday weekofyear wellformed when whene whenev wheneve whenever where while whitespace with within without work wrapped xdb xml xmlagg xmlattributes xmlcast xmlcolattval xmlelement xmlexists xmlforest xmlindex xmlnamespaces xmlpi xmlquery xmlroot xmlschema xmlserialize xmltable xmltype xor year year_to_month years yearweek",literal:"true false null",built_in:"array bigint binary bit blob boolean char character date dec decimal float int int8 integer interval number numeric real record serial serial8 smallint text varchar varying void"},c:[{cN:"string",b:"'",e:"'",c:[e.BE,{b:"''"}]},{cN:"string",b:'"',e:'"',c:[e.BE,{b:'""'}]},{cN:"string",b:"`",e:"`",c:[e.BE]},e.CNM,e.CBCM,t]},e.CBCM,t]}});hljs.registerLanguage("r",function(e){var r="([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*";return{c:[e.HCM,{b:r,l:r,k:{keyword:"function if in break next repeat else for return switch while try tryCatch stop warning require library attach detach source setMethod setGeneric setGroupGeneric setClass ...",literal:"NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 NA_complex_|10"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{b:"`",e:"`",r:0},{cN:"string",c:[e.BE],v:[{b:'"',e:'"'},{b:"'",e:"'"}]}]}});hljs.registerLanguage("perl",function(e){var t="getpwent getservent quotemeta msgrcv scalar kill dbmclose undef lc ma syswrite tr send umask sysopen shmwrite vec qx utime local oct semctl localtime readpipe do return format read sprintf dbmopen pop getpgrp not getpwnam rewinddir qqfileno qw endprotoent wait sethostent bless s|0 opendir continue each sleep endgrent shutdown dump chomp connect getsockname die socketpair close flock exists index shmgetsub for endpwent redo lstat msgctl setpgrp abs exit select print ref gethostbyaddr unshift fcntl syscall goto getnetbyaddr join gmtime symlink semget splice x|0 getpeername recv log setsockopt cos last reverse gethostbyname getgrnam study formline endhostent times chop length gethostent getnetent pack getprotoent getservbyname rand mkdir pos chmod y|0 substr endnetent printf next open msgsnd readdir use unlink getsockopt getpriority rindex wantarray hex system getservbyport endservent int chr untie rmdir prototype tell listen fork shmread ucfirst setprotoent else sysseek link getgrgid shmctl waitpid unpack getnetbyname reset chdir grep split require caller lcfirst until warn while values shift telldir getpwuid my getprotobynumber delete and sort uc defined srand accept package seekdir getprotobyname semop our rename seek if q|0 chroot sysread setpwent no crypt getc chown sqrt write setnetent setpriority foreach tie sin msgget map stat getlogin unless elsif truncate exec keys glob tied closedirioctl socket readlink eval xor readline binmode setservent eof ord bind alarm pipe atan2 getgrent exp time push setgrent gt lt or ne m|0 break given say state when",r={cN:"subst",b:"[$@]\\{",e:"\\}",k:t},s={b:"->{",e:"}"},n={v:[{b:/\$\d/},{b:/[\$%@](\^\w\b|#\w+(::\w+)*|{\w+}|\w+(::\w*)*)/},{b:/[\$%@][^\s\w{]/,r:0}]},i=[e.BE,r,n],o=[n,e.HCM,e.C("^\\=\\w","\\=cut",{eW:!0}),s,{cN:"string",c:i,v:[{b:"q[qwxr]?\\s*\\(",e:"\\)",r:5},{b:"q[qwxr]?\\s*\\[",e:"\\]",r:5},{b:"q[qwxr]?\\s*\\{",e:"\\}",r:5},{b:"q[qwxr]?\\s*\\|",e:"\\|",r:5},{b:"q[qwxr]?\\s*\\<",e:"\\>",r:5},{b:"qw\\s+q",e:"q",r:5},{b:"'",e:"'",c:[e.BE]},{b:'"',e:'"'},{b:"`",e:"`",c:[e.BE]},{b:"{\\w+}",c:[],r:0},{b:"-?\\w+\\s*\\=\\>",c:[],r:0}]},{cN:"number",b:"(\\b0[0-7_]+)|(\\b0x[0-9a-fA-F_]+)|(\\b[1-9][0-9_]*(\\.[0-9_]+)?)|[0_]\\b",r:0},{b:"(\\/\\/|"+e.RSR+"|\\b(split|return|print|reverse|grep)\\b)\\s*",k:"split return print reverse grep",r:0,c:[e.HCM,{cN:"regexp",b:"(s|tr|y)/(\\\\.|[^/])*/(\\\\.|[^/])*/[a-z]*",r:10},{cN:"regexp",b:"(m|qr)?/",e:"/[a-z]*",c:[e.BE],r:0}]},{cN:"function",bK:"sub",e:"(\\s*\\(.*?\\))?[;{]",eE:!0,r:5,c:[e.TM]},{b:"-\\w\\b",r:0},{b:"^__DATA__$",e:"^__END__$",sL:"mojolicious",c:[{b:"^@@.*",e:"$",cN:"comment"}]}];return r.c=o,s.c=o,{aliases:["pl","pm"],l:/[\w\.]+/,k:t,c:o}});hljs.registerLanguage("ini",function(e){var b={cN:"string",c:[e.BE],v:[{b:"'''",e:"'''",r:10},{b:'"""',e:'"""',r:10},{b:'"',e:'"'},{b:"'",e:"'"}]};return{aliases:["toml"],cI:!0,i:/\S/,c:[e.C(";","$"),e.HCM,{cN:"section",b:/^\s*\[+/,e:/\]+/},{b:/^[a-z0-9\[\]_-]+\s*=\s*/,e:"$",rB:!0,c:[{cN:"attr",b:/[a-z0-9\[\]_-]+/},{b:/=/,eW:!0,r:0,c:[{cN:"literal",b:/\bon|off|true|false|yes|no\b/},{cN:"variable",v:[{b:/\$[\w\d"][\w\d_]*/},{b:/\$\{(.*?)}/}]},b,{cN:"number",b:/([\+\-]+)?[\d]+_[\d_]+/},e.NM]}]}]}});hljs.registerLanguage("diff",function(e){return{aliases:["patch"],c:[{cN:"meta",r:10,v:[{b:/^@@ +\-\d+,\d+ +\+\d+,\d+ +@@$/},{b:/^\*\*\* +\d+,\d+ +\*\*\*\*$/},{b:/^\-\-\- +\d+,\d+ +\-\-\-\-$/}]},{cN:"comment",v:[{b:/Index: /,e:/$/},{b:/={3,}/,e:/$/},{b:/^\-{3}/,e:/$/},{b:/^\*{3} /,e:/$/},{b:/^\+{3}/,e:/$/},{b:/\*{5}/,e:/\*{5}$/}]},{cN:"addition",b:"^\\+",e:"$"},{cN:"deletion",b:"^\\-",e:"$"},{cN:"addition",b:"^\\!",e:"$"}]}});hljs.registerLanguage("go",function(e){var t={keyword:"break default func interface select case map struct chan else goto package switch const fallthrough if range type continue for import return var go defer bool byte complex64 complex128 float32 float64 int8 int16 int32 int64 string uint8 uint16 uint32 uint64 int uint uintptr rune",literal:"true false iota nil",built_in:"append cap close complex copy imag len make new panic print println real recover delete"};return{aliases:["golang"],k:t,i:"</",c:[e.CLCM,e.CBCM,{cN:"string",v:[e.QSM,{b:"'",e:"[^\\\\]'"},{b:"`",e:"`"}]},{cN:"number",v:[{b:e.CNR+"[dflsi]",r:1},e.CNM]},{b:/:=/},{cN:"function",bK:"func",e:/\s*\{/,eE:!0,c:[e.TM,{cN:"params",b:/\(/,e:/\)/,k:t,i:/["']/}]}]}});hljs.registerLanguage("bash",function(e){var t={cN:"variable",v:[{b:/\$[\w\d#@][\w\d_]*/},{b:/\$\{(.*?)}/}]},s={cN:"string",b:/"/,e:/"/,c:[e.BE,t,{cN:"variable",b:/\$\(/,e:/\)/,c:[e.BE]}]},a={cN:"string",b:/'/,e:/'/};return{aliases:["sh","zsh"],l:/\b-?[a-z\._]+\b/,k:{keyword:"if then else elif fi for while in do done case esac function",literal:"true false",built_in:"break cd continue eval exec exit export getopts hash pwd readonly return shift test times trap umask unset alias bind builtin caller command declare echo enable help let local logout mapfile printf read readarray source type typeset ulimit unalias set shopt autoload bg bindkey bye cap chdir clone comparguments compcall compctl compdescribe compfiles compgroups compquote comptags comptry compvalues dirs disable disown echotc echoti emulate fc fg float functions getcap getln history integer jobs kill limit log noglob popd print pushd pushln rehash sched setcap setopt stat suspend ttyctl unfunction unhash unlimit unsetopt vared wait whence where which zcompile zformat zftp zle zmodload zparseopts zprof zpty zregexparse zsocket zstyle ztcp",_:"-ne -eq -lt -gt -f -d -e -s -l -a"},c:[{cN:"meta",b:/^#![^\n]+sh\s*$/,r:10},{cN:"function",b:/\w[\w\d_]*\s*\(\s*\)\s*\{/,rB:!0,c:[e.inherit(e.TM,{b:/\w[\w\d_]*/})],r:0},e.HCM,s,a,t]}});hljs.registerLanguage("python",function(e){var r={keyword:"and elif is global as in if from raise for except finally print import pass return exec else break not with class assert yield try while continue del or def lambda async await nonlocal|10 None True False",built_in:"Ellipsis NotImplemented"},b={cN:"meta",b:/^(>>>|\.\.\.) /},c={cN:"subst",b:/\{/,e:/\}/,k:r,i:/#/},a={cN:"string",c:[e.BE],v:[{b:/(u|b)?r?'''/,e:/'''/,c:[b],r:10},{b:/(u|b)?r?"""/,e:/"""/,c:[b],r:10},{b:/(fr|rf|f)'''/,e:/'''/,c:[b,c]},{b:/(fr|rf|f)"""/,e:/"""/,c:[b,c]},{b:/(u|r|ur)'/,e:/'/,r:10},{b:/(u|r|ur)"/,e:/"/,r:10},{b:/(b|br)'/,e:/'/},{b:/(b|br)"/,e:/"/},{b:/(fr|rf|f)'/,e:/'/,c:[c]},{b:/(fr|rf|f)"/,e:/"/,c:[c]},e.ASM,e.QSM]},s={cN:"number",r:0,v:[{b:e.BNR+"[lLjJ]?"},{b:"\\b(0o[0-7]+)[lLjJ]?"},{b:e.CNR+"[lLjJ]?"}]},i={cN:"params",b:/\(/,e:/\)/,c:["self",b,s,a]};return c.c=[a,s,b],{aliases:["py","gyp"],k:r,i:/(<\/|->|\?)|=>/,c:[b,s,a,e.HCM,{v:[{cN:"function",bK:"def"},{cN:"class",bK:"class"}],e:/:/,i:/[${=;\n,]/,c:[e.UTM,i,{b:/->/,eW:!0,k:"None"}]},{cN:"meta",b:/^[\t ]*@/,e:/$/},{b:/\b(print|exec)\(/}]}});hljs.registerLanguage("julia",function(e){var r={keyword:"in isa where baremodule begin break catch ccall const continue do else elseif end export false finally for function global if import importall let local macro module quote return true try using while type immutable abstract bitstype typealias ",literal:"true false ARGS C_NULL DevNull ENDIAN_BOM ENV I Inf Inf16 Inf32 Inf64 InsertionSort JULIA_HOME LOAD_PATH MergeSort NaN NaN16 NaN32 NaN64 PROGRAM_FILE QuickSort RoundDown RoundFromZero RoundNearest RoundNearestTiesAway RoundNearestTiesUp RoundToZero RoundUp STDERR STDIN STDOUT VERSION catalan e|0 eu|0 eulergamma golden im nothing pi γ π φ ",built_in:"ANY AbstractArray AbstractChannel AbstractFloat AbstractMatrix AbstractRNG AbstractSerializer AbstractSet AbstractSparseArray AbstractSparseMatrix AbstractSparseVector AbstractString AbstractUnitRange AbstractVecOrMat AbstractVector Any ArgumentError Array AssertionError Associative Base64DecodePipe Base64EncodePipe Bidiagonal BigFloat BigInt BitArray BitMatrix BitVector Bool BoundsError BufferStream CachingPool CapturedException CartesianIndex CartesianRange Cchar Cdouble Cfloat Channel Char Cint Cintmax_t Clong Clonglong ClusterManager Cmd CodeInfo Colon Complex Complex128 Complex32 Complex64 CompositeException Condition ConjArray ConjMatrix ConjVector Cptrdiff_t Cshort Csize_t Cssize_t Cstring Cuchar Cuint Cuintmax_t Culong Culonglong Cushort Cwchar_t Cwstring DataType Date DateFormat DateTime DenseArray DenseMatrix DenseVecOrMat DenseVector Diagonal Dict DimensionMismatch Dims DirectIndexString Display DivideError DomainError EOFError EachLine Enum Enumerate ErrorException Exception ExponentialBackOff Expr Factorization FileMonitor Float16 Float32 Float64 Function Future GlobalRef GotoNode HTML Hermitian IO IOBuffer IOContext IOStream IPAddr IPv4 IPv6 IndexCartesian IndexLinear IndexStyle InexactError InitError Int Int128 Int16 Int32 Int64 Int8 IntSet Integer InterruptException InvalidStateException Irrational KeyError LabelNode LinSpace LineNumberNode LoadError LowerTriangular MIME Matrix MersenneTwister Method MethodError MethodTable Module NTuple NewvarNode NullException Nullable Number ObjectIdDict OrdinalRange OutOfMemoryError OverflowError Pair ParseError PartialQuickSort PermutedDimsArray Pipe PollingFileWatcher ProcessExitedException Ptr QuoteNode RandomDevice Range RangeIndex Rational RawFD ReadOnlyMemoryError Real ReentrantLock Ref Regex RegexMatch RemoteChannel RemoteException RevString RoundingMode RowVector SSAValue SegmentationFault SerializationState Set SharedArray SharedMatrix SharedVector Signed SimpleVector Slot SlotNumber SparseMatrixCSC SparseVector StackFrame StackOverflowError StackTrace StepRange StepRangeLen StridedArray StridedMatrix StridedVecOrMat StridedVector String SubArray SubString SymTridiagonal Symbol Symmetric SystemError TCPSocket Task Text TextDisplay Timer Tridiagonal Tuple Type TypeError TypeMapEntry TypeMapLevel TypeName TypeVar TypedSlot UDPSocket UInt UInt128 UInt16 UInt32 UInt64 UInt8 UndefRefError UndefVarError UnicodeError UniformScaling Union UnionAll UnitRange Unsigned UpperTriangular Val Vararg VecElement VecOrMat Vector VersionNumber Void WeakKeyDict WeakRef WorkerConfig WorkerPool "},t="[A-Za-z_\\u00A1-\\uFFFF][A-Za-z_0-9\\u00A1-\\uFFFF]*",a={l:t,k:r,i:/<\//},n={cN:"number",b:/(\b0x[\d_]*(\.[\d_]*)?|0x\.\d[\d_]*)p[-+]?\d+|\b0[box][a-fA-F0-9][a-fA-F0-9_]*|(\b\d[\d_]*(\.[\d_]*)?|\.\d[\d_]*)([eEfF][-+]?\d+)?/,r:0},o={cN:"string",b:/'(.|\\[xXuU][a-zA-Z0-9]+)'/},i={cN:"subst",b:/\$\(/,e:/\)/,k:r},l={cN:"variable",b:"\\$"+t},c={cN:"string",c:[e.BE,i,l],v:[{b:/\w*"""/,e:/"""\w*/,r:10},{b:/\w*"/,e:/"\w*/}]},s={cN:"string",c:[e.BE,i,l],b:"`",e:"`"},d={cN:"meta",b:"@"+t},u={cN:"comment",v:[{b:"#=",e:"=#",r:10},{b:"#",e:"$"}]};return a.c=[n,o,c,s,d,u,e.HCM,{cN:"keyword",b:"\\b(((abstract|primitive)\\s+)type|(mutable\\s+)?struct)\\b"},{b:/<:/}],i.c=a.c,a});hljs.registerLanguage("coffeescript",function(e){var c={keyword:"in if for while finally new do return else break catch instanceof throw try this switch continue typeof delete debugger super yield import export from as default await then unless until loop of by when and or is isnt not",literal:"true false null undefined yes no on off",built_in:"npm require console print module global window document"},n="[A-Za-z$_][0-9A-Za-z$_]*",r={cN:"subst",b:/#\{/,e:/}/,k:c},i=[e.BNM,e.inherit(e.CNM,{starts:{e:"(\\s*/)?",r:0}}),{cN:"string",v:[{b:/'''/,e:/'''/,c:[e.BE]},{b:/'/,e:/'/,c:[e.BE]},{b:/"""/,e:/"""/,c:[e.BE,r]},{b:/"/,e:/"/,c:[e.BE,r]}]},{cN:"regexp",v:[{b:"///",e:"///",c:[r,e.HCM]},{b:"//[gim]*",r:0},{b:/\/(?![ *])(\\\/|.)*?\/[gim]*(?=\W|$)/}]},{b:"@"+n},{sL:"javascript",eB:!0,eE:!0,v:[{b:"```",e:"```"},{b:"`",e:"`"}]}];r.c=i;var s=e.inherit(e.TM,{b:n}),t="(\\(.*\\))?\\s*\\B[-=]>",o={cN:"params",b:"\\([^\\(]",rB:!0,c:[{b:/\(/,e:/\)/,k:c,c:["self"].concat(i)}]};return{aliases:["coffee","cson","iced"],k:c,i:/\/\*/,c:i.concat([e.C("###","###"),e.HCM,{cN:"function",b:"^\\s*"+n+"\\s*=\\s*"+t,e:"[-=]>",rB:!0,c:[s,o]},{b:/[:\(,=]\s*/,r:0,c:[{cN:"function",b:t,e:"[-=]>",rB:!0,c:[o]}]},{cN:"class",bK:"class",e:"$",i:/[:="\[\]]/,c:[{bK:"extends",eW:!0,i:/[:="\[\]]/,c:[s]},s]},{b:n+":",e:":",rB:!0,rE:!0,r:0}])}});hljs.registerLanguage("cpp",function(t){var e={cN:"keyword",b:"\\b[a-z\\d_]*_t\\b"},r={cN:"string",v:[{b:'(u8?|U)?L?"',e:'"',i:"\\n",c:[t.BE]},{b:'(u8?|U)?R"',e:'"',c:[t.BE]},{b:"'\\\\?.",e:"'",i:"."}]},s={cN:"number",v:[{b:"\\b(0b[01']+)"},{b:"(-?)\\b([\\d']+(\\.[\\d']*)?|\\.[\\d']+)(u|U|l|L|ul|UL|f|F|b|B)"},{b:"(-?)(\\b0[xX][a-fA-F0-9']+|(\\b[\\d']+(\\.[\\d']*)?|\\.[\\d']+)([eE][-+]?[\\d']+)?)"}],r:0},i={cN:"meta",b:/#\s*[a-z]+\b/,e:/$/,k:{"meta-keyword":"if else elif endif define undef warning error line pragma ifdef ifndef include"},c:[{b:/\\\n/,r:0},t.inherit(r,{cN:"meta-string"}),{cN:"meta-string",b:/<[^\n>]*>/,e:/$/,i:"\\n"},t.CLCM,t.CBCM]},a=t.IR+"\\s*\\(",c={keyword:"int float while private char catch import module export virtual operator sizeof dynamic_cast|10 typedef const_cast|10 const for static_cast|10 union namespace unsigned long volatile static protected bool template mutable if public friend do goto auto void enum else break extern using asm case typeid short reinterpret_cast|10 default double register explicit signed typename try this switch continue inline delete alignof constexpr decltype noexcept static_assert thread_local restrict _Bool complex _Complex _Imaginary atomic_bool atomic_char atomic_schar atomic_uchar atomic_short atomic_ushort atomic_int atomic_uint atomic_long atomic_ulong atomic_llong atomic_ullong new throw return and or not",built_in:"std string cin cout cerr clog stdin stdout stderr stringstream istringstream ostringstream auto_ptr deque list queue stack vector map set bitset multiset multimap unordered_set unordered_map unordered_multiset unordered_multimap array shared_ptr abort abs acos asin atan2 atan calloc ceil cosh cos exit exp fabs floor fmod fprintf fputs free frexp fscanf isalnum isalpha iscntrl isdigit isgraph islower isprint ispunct isspace isupper isxdigit tolower toupper labs ldexp log10 log malloc realloc memchr memcmp memcpy memset modf pow printf putchar puts scanf sinh sin snprintf sprintf sqrt sscanf strcat strchr strcmp strcpy strcspn strlen strncat strncmp strncpy strpbrk strrchr strspn strstr tanh tan vfprintf vprintf vsprintf endl initializer_list unique_ptr",literal:"true false nullptr NULL"},n=[e,t.CLCM,t.CBCM,s,r];return{aliases:["c","cc","h","c++","h++","hpp"],k:c,i:"</",c:n.concat([i,{b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:c,c:["self",e]},{b:t.IR+"::",k:c},{v:[{b:/=/,e:/;/},{b:/\(/,e:/\)/},{bK:"new throw return else",e:/;/}],k:c,c:n.concat([{b:/\(/,e:/\)/,k:c,c:n.concat(["self"]),r:0}]),r:0},{cN:"function",b:"("+t.IR+"[\\*&\\s]+)+"+a,rB:!0,e:/[{;=]/,eE:!0,k:c,i:/[^\w\s\*&]/,c:[{b:a,rB:!0,c:[t.TM],r:0},{cN:"params",b:/\(/,e:/\)/,k:c,r:0,c:[t.CLCM,t.CBCM,r,s,e]},t.CLCM,t.CBCM,i]},{cN:"class",bK:"class struct",e:/[{;:]/,c:[{b:/</,e:/>/,c:["self"]},t.TM]}]),exports:{preprocessor:i,strings:r,k:c}}});hljs.registerLanguage("ruby",function(e){var b="[a-zA-Z_]\\w*[!?=]?|[-+~]\\@|<<|>>|=~|===?|<=>|[<>]=?|\\*\\*|[-/+%^&*~`|]|\\[\\]=?",r={keyword:"and then defined module in return redo if BEGIN retry end for self when next until do begin unless END rescue else break undef not super class case require yield alias while ensure elsif or include attr_reader attr_writer attr_accessor",literal:"true false nil"},c={cN:"doctag",b:"@[A-Za-z]+"},a={b:"#<",e:">"},s=[e.C("#","$",{c:[c]}),e.C("^\\=begin","^\\=end",{c:[c],r:10}),e.C("^__END__","\\n$")],n={cN:"subst",b:"#\\{",e:"}",k:r},t={cN:"string",c:[e.BE,n],v:[{b:/'/,e:/'/},{b:/"/,e:/"/},{b:/`/,e:/`/},{b:"%[qQwWx]?\\(",e:"\\)"},{b:"%[qQwWx]?\\[",e:"\\]"},{b:"%[qQwWx]?{",e:"}"},{b:"%[qQwWx]?<",e:">"},{b:"%[qQwWx]?/",e:"/"},{b:"%[qQwWx]?%",e:"%"},{b:"%[qQwWx]?-",e:"-"},{b:"%[qQwWx]?\\|",e:"\\|"},{b:/\B\?(\\\d{1,3}|\\x[A-Fa-f0-9]{1,2}|\\u[A-Fa-f0-9]{4}|\\?\S)\b/},{b:/<<(-?)\w+$/,e:/^\s*\w+$/}]},i={cN:"params",b:"\\(",e:"\\)",endsParent:!0,k:r},d=[t,a,{cN:"class",bK:"class module",e:"$|;",i:/=/,c:[e.inherit(e.TM,{b:"[A-Za-z_]\\w*(::\\w+)*(\\?|\\!)?"}),{b:"<\\s*",c:[{b:"("+e.IR+"::)?"+e.IR}]}].concat(s)},{cN:"function",bK:"def",e:"$|;",c:[e.inherit(e.TM,{b:b}),i].concat(s)},{b:e.IR+"::"},{cN:"symbol",b:e.UIR+"(\\!|\\?)?:",r:0},{cN:"symbol",b:":(?!\\s)",c:[t,{b:b}],r:0},{cN:"number",b:"(\\b0[0-7_]+)|(\\b0x[0-9a-fA-F_]+)|(\\b[1-9][0-9_]*(\\.[0-9_]+)?)|[0_]\\b",r:0},{b:"(\\$\\W)|((\\$|\\@\\@?)(\\w+))"},{cN:"params",b:/\|/,e:/\|/,k:r},{b:"("+e.RSR+"|unless)\\s*",k:"unless",c:[a,{cN:"regexp",c:[e.BE,n],i:/\n/,v:[{b:"/",e:"/[a-z]*"},{b:"%r{",e:"}[a-z]*"},{b:"%r\\(",e:"\\)[a-z]*"},{b:"%r!",e:"![a-z]*"},{b:"%r\\[",e:"\\][a-z]*"}]}].concat(s),r:0}].concat(s);n.c=d,i.c=d;var l="[>?]>",o="[\\w#]+\\(\\w+\\):\\d+:\\d+>",u="(\\w+-)?\\d+\\.\\d+\\.\\d(p\\d+)?[^>]+>",w=[{b:/^\s*=>/,starts:{e:"$",c:d}},{cN:"meta",b:"^("+l+"|"+o+"|"+u+")",starts:{e:"$",c:d}}];return{aliases:["rb","gemspec","podspec","thor","irb"],k:r,i:/\/\*/,c:s.concat(w).concat(d)}});hljs.registerLanguage("yaml",function(e){var b="true false yes no null",a="^[ \\-]*",r="[a-zA-Z_][\\w\\-]*",t={cN:"attr",v:[{b:a+r+":"},{b:a+'"'+r+'":'},{b:a+"'"+r+"':"}]},c={cN:"template-variable",v:[{b:"{{",e:"}}"},{b:"%{",e:"}"}]},l={cN:"string",r:0,v:[{b:/'/,e:/'/},{b:/"/,e:/"/},{b:/\S+/}],c:[e.BE,c]};return{cI:!0,aliases:["yml","YAML","yaml"],c:[t,{cN:"meta",b:"^---s*$",r:10},{cN:"string",b:"[\\|>] *$",rE:!0,c:l.c,e:t.v[0].b},{b:"<%[%=-]?",e:"[%-]?%>",sL:"ruby",eB:!0,eE:!0,r:0},{cN:"type",b:"!!"+e.UIR},{cN:"meta",b:"&"+e.UIR+"$"},{cN:"meta",b:"\\*"+e.UIR+"$"},{cN:"bullet",b:"^ *-",r:0},e.HCM,{bK:b,k:{literal:b}},e.CNM,l]}});hljs.registerLanguage("css",function(e){var c="[a-zA-Z-][a-zA-Z0-9_-]*",t={b:/[A-Z\_\.\-]+\s*:/,rB:!0,e:";",eW:!0,c:[{cN:"attribute",b:/\S/,e:":",eE:!0,starts:{eW:!0,eE:!0,c:[{b:/[\w-]+\(/,rB:!0,c:[{cN:"built_in",b:/[\w-]+/},{b:/\(/,e:/\)/,c:[e.ASM,e.QSM]}]},e.CSSNM,e.QSM,e.ASM,e.CBCM,{cN:"number",b:"#[0-9A-Fa-f]+"},{cN:"meta",b:"!important"}]}}]};return{cI:!0,i:/[=\/|'\$]/,c:[e.CBCM,{cN:"selector-id",b:/#[A-Za-z0-9_-]+/},{cN:"selector-class",b:/\.[A-Za-z0-9_-]+/},{cN:"selector-attr",b:/\[/,e:/\]/,i:"$"},{cN:"selector-pseudo",b:/:(:)?[a-zA-Z0-9\_\-\+\(\)"'.]+/},{b:"@(font-face|page)",l:"[a-z-]+",k:"font-face page"},{b:"@",e:"[{;]",i:/:/,c:[{cN:"keyword",b:/\w+/},{b:/\s/,eW:!0,eE:!0,r:0,c:[e.ASM,e.QSM,e.CSSNM]}]},{cN:"selector-tag",b:c,r:0},{b:"{",e:"}",i:/\S/,c:[e.CBCM,t]}]}});hljs.registerLanguage("fortran",function(e){var t={cN:"params",b:"\\(",e:"\\)"},n={literal:".False. .True.",keyword:"kind do while private call intrinsic where elsewhere type endtype endmodule endselect endinterface end enddo endif if forall endforall only contains default return stop then public subroutine|10 function program .and. .or. .not. .le. .eq. .ge. .gt. .lt. goto save else use module select case access blank direct exist file fmt form formatted iostat name named nextrec number opened rec recl sequential status unformatted unit continue format pause cycle exit c_null_char c_alert c_backspace c_form_feed flush wait decimal round iomsg synchronous nopass non_overridable pass protected volatile abstract extends import non_intrinsic value deferred generic final enumerator class associate bind enum c_int c_short c_long c_long_long c_signed_char c_size_t c_int8_t c_int16_t c_int32_t c_int64_t c_int_least8_t c_int_least16_t c_int_least32_t c_int_least64_t c_int_fast8_t c_int_fast16_t c_int_fast32_t c_int_fast64_t c_intmax_t C_intptr_t c_float c_double c_long_double c_float_complex c_double_complex c_long_double_complex c_bool c_char c_null_ptr c_null_funptr c_new_line c_carriage_return c_horizontal_tab c_vertical_tab iso_c_binding c_loc c_funloc c_associated  c_f_pointer c_ptr c_funptr iso_fortran_env character_storage_size error_unit file_storage_size input_unit iostat_end iostat_eor numeric_storage_size output_unit c_f_procpointer ieee_arithmetic ieee_support_underflow_control ieee_get_underflow_mode ieee_set_underflow_mode newunit contiguous recursive pad position action delim readwrite eor advance nml interface procedure namelist include sequence elemental pure integer real character complex logical dimension allocatable|10 parameter external implicit|10 none double precision assign intent optional pointer target in out common equivalence data",built_in:"alog alog10 amax0 amax1 amin0 amin1 amod cabs ccos cexp clog csin csqrt dabs dacos dasin datan datan2 dcos dcosh ddim dexp dint dlog dlog10 dmax1 dmin1 dmod dnint dsign dsin dsinh dsqrt dtan dtanh float iabs idim idint idnint ifix isign max0 max1 min0 min1 sngl algama cdabs cdcos cdexp cdlog cdsin cdsqrt cqabs cqcos cqexp cqlog cqsin cqsqrt dcmplx dconjg derf derfc dfloat dgamma dimag dlgama iqint qabs qacos qasin qatan qatan2 qcmplx qconjg qcos qcosh qdim qerf qerfc qexp qgamma qimag qlgama qlog qlog10 qmax1 qmin1 qmod qnint qsign qsin qsinh qsqrt qtan qtanh abs acos aimag aint anint asin atan atan2 char cmplx conjg cos cosh exp ichar index int log log10 max min nint sign sin sinh sqrt tan tanh print write dim lge lgt lle llt mod nullify allocate deallocate adjustl adjustr all allocated any associated bit_size btest ceiling count cshift date_and_time digits dot_product eoshift epsilon exponent floor fraction huge iand ibclr ibits ibset ieor ior ishft ishftc lbound len_trim matmul maxexponent maxloc maxval merge minexponent minloc minval modulo mvbits nearest pack present product radix random_number random_seed range repeat reshape rrspacing scale scan selected_int_kind selected_real_kind set_exponent shape size spacing spread sum system_clock tiny transpose trim ubound unpack verify achar iachar transfer dble entry dprod cpu_time command_argument_count get_command get_command_argument get_environment_variable is_iostat_end ieee_arithmetic ieee_support_underflow_control ieee_get_underflow_mode ieee_set_underflow_mode is_iostat_eor move_alloc new_line selected_char_kind same_type_as extends_type_ofacosh asinh atanh bessel_j0 bessel_j1 bessel_jn bessel_y0 bessel_y1 bessel_yn erf erfc erfc_scaled gamma log_gamma hypot norm2 atomic_define atomic_ref execute_command_line leadz trailz storage_size merge_bits bge bgt ble blt dshiftl dshiftr findloc iall iany iparity image_index lcobound ucobound maskl maskr num_images parity popcnt poppar shifta shiftl shiftr this_image"};return{cI:!0,aliases:["f90","f95"],k:n,i:/\/\*/,c:[e.inherit(e.ASM,{cN:"string",r:0}),e.inherit(e.QSM,{cN:"string",r:0}),{cN:"function",bK:"subroutine function program",i:"[${=\\n]",c:[e.UTM,t]},e.C("!","$",{r:0}),{cN:"number",b:"(?=\\b|\\+|\\-|\\.)(?=\\.\\d|\\d)(?:\\d+)?(?:\\.?\\d*)(?:[de][+-]?\\d+)?\\b\\.?",r:0}]}});hljs.registerLanguage("awk",function(e){var r={cN:"variable",v:[{b:/\$[\w\d#@][\w\d_]*/},{b:/\$\{(.*?)}/}]},b="BEGIN END if else while do for in break continue delete next nextfile function func exit|10",n={cN:"string",c:[e.BE],v:[{b:/(u|b)?r?'''/,e:/'''/,r:10},{b:/(u|b)?r?"""/,e:/"""/,r:10},{b:/(u|r|ur)'/,e:/'/,r:10},{b:/(u|r|ur)"/,e:/"/,r:10},{b:/(b|br)'/,e:/'/},{b:/(b|br)"/,e:/"/},e.ASM,e.QSM]};return{k:{keyword:b},c:[r,n,e.RM,e.HCM,e.NM]}});hljs.registerLanguage("makefile",function(e){var i={cN:"variable",v:[{b:"\\$\\("+e.UIR+"\\)",c:[e.BE]},{b:/\$[@%<?\^\+\*]/}]},r={cN:"string",b:/"/,e:/"/,c:[e.BE,i]},a={cN:"variable",b:/\$\([\w-]+\s/,e:/\)/,k:{built_in:"subst patsubst strip findstring filter filter-out sort word wordlist firstword lastword dir notdir suffix basename addsuffix addprefix join wildcard realpath abspath error warning shell origin flavor foreach if or and call eval file value"},c:[i]},n={b:"^"+e.UIR+"\\s*[:+?]?=",i:"\\n",rB:!0,c:[{b:"^"+e.UIR,e:"[:+?]?=",eE:!0}]},t={cN:"meta",b:/^\.PHONY:/,e:/$/,k:{"meta-keyword":".PHONY"},l:/[\.\w]+/},l={cN:"section",b:/^[^\s]+:/,e:/$/,c:[i]};return{aliases:["mk","mak"],k:"define endef undefine ifdef ifndef ifeq ifneq else endif include -include sinclude override export unexport private vpath",l:/[\w-]+/,c:[e.HCM,i,r,a,n,t,l]}});hljs.registerLanguage("java",function(e){var a="[À-ʸa-zA-Z_$][À-ʸa-zA-Z_$0-9]*",t=a+"(<"+a+"(\\s*,\\s*"+a+")*>)?",r="false synchronized int abstract float private char boolean static null if const for true while long strictfp finally protected import native final void enum else break transient catch instanceof byte super volatile case assert short package default double public try this switch continue throws protected public private module requires exports do",s="\\b(0[bB]([01]+[01_]+[01]+|[01]+)|0[xX]([a-fA-F0-9]+[a-fA-F0-9_]+[a-fA-F0-9]+|[a-fA-F0-9]+)|(([\\d]+[\\d_]+[\\d]+|[\\d]+)(\\.([\\d]+[\\d_]+[\\d]+|[\\d]+))?|\\.([\\d]+[\\d_]+[\\d]+|[\\d]+))([eE][-+]?\\d+)?)[lLfF]?",c={cN:"number",b:s,r:0};return{aliases:["jsp"],k:r,i:/<\/|#/,c:[e.C("/\\*\\*","\\*/",{r:0,c:[{b:/\w+@/,r:0},{cN:"doctag",b:"@[A-Za-z]+"}]}),e.CLCM,e.CBCM,e.ASM,e.QSM,{cN:"class",bK:"class interface",e:/[{;=]/,eE:!0,k:"class interface",i:/[:"\[\]]/,c:[{bK:"extends implements"},e.UTM]},{bK:"new throw return else",r:0},{cN:"function",b:"("+t+"\\s+)+"+e.UIR+"\\s*\\(",rB:!0,e:/[{;=]/,eE:!0,k:r,c:[{b:e.UIR+"\\s*\\(",rB:!0,r:0,c:[e.UTM]},{cN:"params",b:/\(/,e:/\)/,k:r,r:0,c:[e.ASM,e.QSM,e.CNM,e.CBCM]},e.CLCM,e.CBCM]},c,{cN:"meta",b:"@[A-Za-z]+"}]}});hljs.registerLanguage("stan",function(e){return{c:[e.HCM,e.CLCM,e.CBCM,{b:e.UIR,l:e.UIR,k:{name:"for in while repeat until if then else",symbol:"bernoulli bernoulli_logit binomial binomial_logit beta_binomial hypergeometric categorical categorical_logit ordered_logistic neg_binomial neg_binomial_2 neg_binomial_2_log poisson poisson_log multinomial normal exp_mod_normal skew_normal student_t cauchy double_exponential logistic gumbel lognormal chi_square inv_chi_square scaled_inv_chi_square exponential inv_gamma weibull frechet rayleigh wiener pareto pareto_type_2 von_mises uniform multi_normal multi_normal_prec multi_normal_cholesky multi_gp multi_gp_cholesky multi_student_t gaussian_dlm_obs dirichlet lkj_corr lkj_corr_cholesky wishart inv_wishart","selector-tag":"int real vector simplex unit_vector ordered positive_ordered row_vector matrix cholesky_factor_corr cholesky_factor_cov corr_matrix cov_matrix",title:"functions model data parameters quantities transformed generated",literal:"true false"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0}]}});hljs.registerLanguage("javascript",function(e){var r="[A-Za-z$_][0-9A-Za-z$_]*",t={keyword:"in of if for while finally var new function do return void else break catch instanceof with throw case default try this switch continue typeof delete let yield const export super debugger as async await static import from as",literal:"true false null undefined NaN Infinity",built_in:"eval isFinite isNaN parseFloat parseInt decodeURI decodeURIComponent encodeURI encodeURIComponent escape unescape Object Function Boolean Error EvalError InternalError RangeError ReferenceError StopIteration SyntaxError TypeError URIError Number Math Date String RegExp Array Float32Array Float64Array Int16Array Int32Array Int8Array Uint16Array Uint32Array Uint8Array Uint8ClampedArray ArrayBuffer DataView JSON Intl arguments require module console window document Symbol Set Map WeakSet WeakMap Proxy Reflect Promise"},a={cN:"number",v:[{b:"\\b(0[bB][01]+)"},{b:"\\b(0[oO][0-7]+)"},{b:e.CNR}],r:0},n={cN:"subst",b:"\\$\\{",e:"\\}",k:t,c:[]},c={cN:"string",b:"`",e:"`",c:[e.BE,n]};n.c=[e.ASM,e.QSM,c,a,e.RM];var s=n.c.concat([e.CBCM,e.CLCM]);return{aliases:["js","jsx"],k:t,c:[{cN:"meta",r:10,b:/^\s*['"]use (strict|asm)['"]/},{cN:"meta",b:/^#!/,e:/$/},e.ASM,e.QSM,c,e.CLCM,e.CBCM,a,{b:/[{,]\s*/,r:0,c:[{b:r+"\\s*:",rB:!0,r:0,c:[{cN:"attr",b:r,r:0}]}]},{b:"("+e.RSR+"|\\b(case|return|throw)\\b)\\s*",k:"return throw case",c:[e.CLCM,e.CBCM,e.RM,{cN:"function",b:"(\\(.*?\\)|"+r+")\\s*=>",rB:!0,e:"\\s*=>",c:[{cN:"params",v:[{b:r},{b:/\(\s*\)/},{b:/\(/,e:/\)/,eB:!0,eE:!0,k:t,c:s}]}]},{b:/</,e:/(\/\w+|\w+\/)>/,sL:"xml",c:[{b:/<\w+\s*\/>/,skip:!0},{b:/<\w+/,e:/(\/\w+|\w+\/)>/,skip:!0,c:[{b:/<\w+\s*\/>/,skip:!0},"self"]}]}],r:0},{cN:"function",bK:"function",e:/\{/,eE:!0,c:[e.inherit(e.TM,{b:r}),{cN:"params",b:/\(/,e:/\)/,eB:!0,eE:!0,c:s}],i:/\[|%/},{b:/\$[(.]/},e.METHOD_GUARD,{cN:"class",bK:"class",e:/[{;=]/,eE:!0,i:/[:"\[\]]/,c:[{bK:"extends"},e.UTM]},{bK:"constructor",e:/\{/,eE:!0}],i:/#(?!!)/}});hljs.registerLanguage("tex",function(c){var e={cN:"tag",b:/\\/,r:0,c:[{cN:"name",v:[{b:/[a-zA-Zа-яА-я]+[*]?/},{b:/[^a-zA-Zа-яА-я0-9]/}],starts:{eW:!0,r:0,c:[{cN:"string",v:[{b:/\[/,e:/\]/},{b:/\{/,e:/\}/}]},{b:/\s*=\s*/,eW:!0,r:0,c:[{cN:"number",b:/-?\d*\.?\d+(pt|pc|mm|cm|in|dd|cc|ex|em)?/}]}]}}]};return{c:[e,{cN:"formula",c:[e],r:0,v:[{b:/\$\$/,e:/\$\$/},{b:/\$/,e:/\$/}]},c.C("%","$",{r:0})]}});hljs.registerLanguage("xml",function(s){var e="[A-Za-z0-9\\._:-]+",t={eW:!0,i:/</,r:0,c:[{cN:"attr",b:e,r:0},{b:/=\s*/,r:0,c:[{cN:"string",endsParent:!0,v:[{b:/"/,e:/"/},{b:/'/,e:/'/},{b:/[^\s"'=<>`]+/}]}]}]};return{aliases:["html","xhtml","rss","atom","xjb","xsd","xsl","plist"],cI:!0,c:[{cN:"meta",b:"<!DOCTYPE",e:">",r:10,c:[{b:"\\[",e:"\\]"}]},s.C("<!--","-->",{r:10}),{b:"<\\!\\[CDATA\\[",e:"\\]\\]>",r:10},{b:/<\?(php)?/,e:/\?>/,sL:"php",c:[{b:"/\\*",e:"\\*/",skip:!0}]},{cN:"tag",b:"<style(?=\\s|>|$)",e:">",k:{name:"style"},c:[t],starts:{e:"</style>",rE:!0,sL:["css","xml"]}},{cN:"tag",b:"<script(?=\\s|>|$)",e:">",k:{name:"script"},c:[t],starts:{e:"</script>",rE:!0,sL:["actionscript","javascript","handlebars","xml"]}},{cN:"meta",v:[{b:/<\?xml/,e:/\?>/,r:10},{b:/<\?\w+/,e:/\?>/}]},{cN:"tag",b:"</?",e:"/?>",c:[{cN:"name",b:/[^\/><\s]+/,r:0},t]}]}});hljs.registerLanguage("markdown",function(e){return{aliases:["md","mkdown","mkd"],c:[{cN:"section",v:[{b:"^#{1,6}",e:"$"},{b:"^.+?\\n[=-]{2,}$"}]},{b:"<",e:">",sL:"xml",r:0},{cN:"bullet",b:"^([*+-]|(\\d+\\.))\\s+"},{cN:"strong",b:"[*_]{2}.+?[*_]{2}"},{cN:"emphasis",v:[{b:"\\*.+?\\*"},{b:"_.+?_",r:0}]},{cN:"quote",b:"^>\\s+",e:"$"},{cN:"code",v:[{b:"^```w*s*$",e:"^```s*$"},{b:"`.+?`"},{b:"^( {4}|	)",e:"$",r:0}]},{b:"^[-\\*]{3,}",e:"$"},{b:"\\[.+?\\][\\(\\[].*?[\\)\\]]",rB:!0,c:[{cN:"string",b:"\\[",e:"\\]",eB:!0,rE:!0,r:0},{cN:"link",b:"\\]\\(",e:"\\)",eB:!0,eE:!0},{cN:"symbol",b:"\\]\\[",e:"\\]",eB:!0,eE:!0}],r:10},{b:/^\[[^\n]+\]:/,rB:!0,c:[{cN:"symbol",b:/\[/,e:/\]/,eB:!0,eE:!0},{cN:"link",b:/:\s*/,e:/$/,eB:!0}]}]}});hljs.registerLanguage("json",function(e){var i={literal:"true false null"},n=[e.QSM,e.CNM],r={e:",",eW:!0,eE:!0,c:n,k:i},t={b:"{",e:"}",c:[{cN:"attr",b:/"/,e:/"/,c:[e.BE],i:"\\n"},e.inherit(r,{b:/:/})],i:"\\S"},c={b:"\\[",e:"\\]",c:[e.inherit(r)],i:"\\S"};return n.splice(n.length,0,t,c),{c:n,k:i,i:"\\S"}});"></script>
<script>(function() {
  // If window.HTMLWidgets is already defined, then use it; otherwise create a
  // new object. This allows preceding code to set options that affect the
  // initialization process (though none currently exist).
  window.HTMLWidgets = window.HTMLWidgets || {};

  // See if we're running in a viewer pane. If not, we're in a web browser.
  var viewerMode = window.HTMLWidgets.viewerMode =
      /\bviewer_pane=1\b/.test(window.location);

  // See if we're running in Shiny mode. If not, it's a static document.
  // Note that static widgets can appear in both Shiny and static modes, but
  // obviously, Shiny widgets can only appear in Shiny apps/documents.
  var shinyMode = window.HTMLWidgets.shinyMode =
      typeof(window.Shiny) !== "undefined" && !!window.Shiny.outputBindings;

  // We can't count on jQuery being available, so we implement our own
  // version if necessary.
  function querySelectorAll(scope, selector) {
    if (typeof(jQuery) !== "undefined" && scope instanceof jQuery) {
      return scope.find(selector);
    }
    if (scope.querySelectorAll) {
      return scope.querySelectorAll(selector);
    }
  }

  function asArray(value) {
    if (value === null)
      return [];
    if ($.isArray(value))
      return value;
    return [value];
  }

  // Implement jQuery's extend
  function extend(target /*, ... */) {
    if (arguments.length == 1) {
      return target;
    }
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var prop in source) {
        if (source.hasOwnProperty(prop)) {
          target[prop] = source[prop];
        }
      }
    }
    return target;
  }

  // IE8 doesn't support Array.forEach.
  function forEach(values, callback, thisArg) {
    if (values.forEach) {
      values.forEach(callback, thisArg);
    } else {
      for (var i = 0; i < values.length; i++) {
        callback.call(thisArg, values[i], i, values);
      }
    }
  }

  // Replaces the specified method with the return value of funcSource.
  //
  // Note that funcSource should not BE the new method, it should be a function
  // that RETURNS the new method. funcSource receives a single argument that is
  // the overridden method, it can be called from the new method. The overridden
  // method can be called like a regular function, it has the target permanently
  // bound to it so "this" will work correctly.
  function overrideMethod(target, methodName, funcSource) {
    var superFunc = target[methodName] || function() {};
    var superFuncBound = function() {
      return superFunc.apply(target, arguments);
    };
    target[methodName] = funcSource(superFuncBound);
  }

  // Add a method to delegator that, when invoked, calls
  // delegatee.methodName. If there is no such method on
  // the delegatee, but there was one on delegator before
  // delegateMethod was called, then the original version
  // is invoked instead.
  // For example:
  //
  // var a = {
  //   method1: function() { console.log('a1'); }
  //   method2: function() { console.log('a2'); }
  // };
  // var b = {
  //   method1: function() { console.log('b1'); }
  // };
  // delegateMethod(a, b, "method1");
  // delegateMethod(a, b, "method2");
  // a.method1();
  // a.method2();
  //
  // The output would be "b1", "a2".
  function delegateMethod(delegator, delegatee, methodName) {
    var inherited = delegator[methodName];
    delegator[methodName] = function() {
      var target = delegatee;
      var method = delegatee[methodName];

      // The method doesn't exist on the delegatee. Instead,
      // call the method on the delegator, if it exists.
      if (!method) {
        target = delegator;
        method = inherited;
      }

      if (method) {
        return method.apply(target, arguments);
      }
    };
  }

  // Implement a vague facsimilie of jQuery's data method
  function elementData(el, name, value) {
    if (arguments.length == 2) {
      return el["htmlwidget_data_" + name];
    } else if (arguments.length == 3) {
      el["htmlwidget_data_" + name] = value;
      return el;
    } else {
      throw new Error("Wrong number of arguments for elementData: " +
        arguments.length);
    }
  }

  // http://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex
  function escapeRegExp(str) {
    return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
  }

  function hasClass(el, className) {
    var re = new RegExp("\\b" + escapeRegExp(className) + "\\b");
    return re.test(el.className);
  }

  // elements - array (or array-like object) of HTML elements
  // className - class name to test for
  // include - if true, only return elements with given className;
  //   if false, only return elements *without* given className
  function filterByClass(elements, className, include) {
    var results = [];
    for (var i = 0; i < elements.length; i++) {
      if (hasClass(elements[i], className) == include)
        results.push(elements[i]);
    }
    return results;
  }

  function on(obj, eventName, func) {
    if (obj.addEventListener) {
      obj.addEventListener(eventName, func, false);
    } else if (obj.attachEvent) {
      obj.attachEvent(eventName, func);
    }
  }

  function off(obj, eventName, func) {
    if (obj.removeEventListener)
      obj.removeEventListener(eventName, func, false);
    else if (obj.detachEvent) {
      obj.detachEvent(eventName, func);
    }
  }

  // Translate array of values to top/right/bottom/left, as usual with
  // the "padding" CSS property
  // https://developer.mozilla.org/en-US/docs/Web/CSS/padding
  function unpackPadding(value) {
    if (typeof(value) === "number")
      value = [value];
    if (value.length === 1) {
      return {top: value[0], right: value[0], bottom: value[0], left: value[0]};
    }
    if (value.length === 2) {
      return {top: value[0], right: value[1], bottom: value[0], left: value[1]};
    }
    if (value.length === 3) {
      return {top: value[0], right: value[1], bottom: value[2], left: value[1]};
    }
    if (value.length === 4) {
      return {top: value[0], right: value[1], bottom: value[2], left: value[3]};
    }
  }

  // Convert an unpacked padding object to a CSS value
  function paddingToCss(paddingObj) {
    return paddingObj.top + "px " + paddingObj.right + "px " + paddingObj.bottom + "px " + paddingObj.left + "px";
  }

  // Makes a number suitable for CSS
  function px(x) {
    if (typeof(x) === "number")
      return x + "px";
    else
      return x;
  }

  // Retrieves runtime widget sizing information for an element.
  // The return value is either null, or an object with fill, padding,
  // defaultWidth, defaultHeight fields.
  function sizingPolicy(el) {
    var sizingEl = document.querySelector("script[data-for='" + el.id + "'][type='application/htmlwidget-sizing']");
    if (!sizingEl)
      return null;
    var sp = JSON.parse(sizingEl.textContent || sizingEl.text || "{}");
    if (viewerMode) {
      return sp.viewer;
    } else {
      return sp.browser;
    }
  }

  // @param tasks Array of strings (or falsy value, in which case no-op).
  //   Each element must be a valid JavaScript expression that yields a
  //   function. Or, can be an array of objects with "code" and "data"
  //   properties; in this case, the "code" property should be a string
  //   of JS that's an expr that yields a function, and "data" should be
  //   an object that will be added as an additional argument when that
  //   function is called.
  // @param target The object that will be "this" for each function
  //   execution.
  // @param args Array of arguments to be passed to the functions. (The
  //   same arguments will be passed to all functions.)
  function evalAndRun(tasks, target, args) {
    if (tasks) {
      forEach(tasks, function(task) {
        var theseArgs = args;
        if (typeof(task) === "object") {
          theseArgs = theseArgs.concat([task.data]);
          task = task.code;
        }
        var taskFunc = tryEval(task);
        if (typeof(taskFunc) !== "function") {
          throw new Error("Task must be a function! Source:\n" + task);
        }
        taskFunc.apply(target, theseArgs);
      });
    }
  }

  // Attempt eval() both with and without enclosing in parentheses.
  // Note that enclosing coerces a function declaration into
  // an expression that eval() can parse
  // (otherwise, a SyntaxError is thrown)
  function tryEval(code) {
    var result = null;
    try {
      result = eval("(" + code + ")");
    } catch(error) {
      if (!error instanceof SyntaxError) {
        throw error;
      }
      try {
        result = eval(code);
      } catch(e) {
        if (e instanceof SyntaxError) {
          throw error;
        } else {
          throw e;
        }
      }
    }
    return result;
  }

  function initSizing(el) {
    var sizing = sizingPolicy(el);
    if (!sizing)
      return;

    var cel = document.getElementById("htmlwidget_container");
    if (!cel)
      return;

    if (typeof(sizing.padding) !== "undefined") {
      document.body.style.margin = "0";
      document.body.style.padding = paddingToCss(unpackPadding(sizing.padding));
    }

    if (sizing.fill) {
      document.body.style.overflow = "hidden";
      document.body.style.width = "100%";
      document.body.style.height = "100%";
      document.documentElement.style.width = "100%";
      document.documentElement.style.height = "100%";
      if (cel) {
        cel.style.position = "absolute";
        var pad = unpackPadding(sizing.padding);
        cel.style.top = pad.top + "px";
        cel.style.right = pad.right + "px";
        cel.style.bottom = pad.bottom + "px";
        cel.style.left = pad.left + "px";
        el.style.width = "100%";
        el.style.height = "100%";
      }

      return {
        getWidth: function() { return cel.offsetWidth; },
        getHeight: function() { return cel.offsetHeight; }
      };

    } else {
      el.style.width = px(sizing.width);
      el.style.height = px(sizing.height);

      return {
        getWidth: function() { return el.offsetWidth; },
        getHeight: function() { return el.offsetHeight; }
      };
    }
  }

  // Default implementations for methods
  var defaults = {
    find: function(scope) {
      return querySelectorAll(scope, "." + this.name);
    },
    renderError: function(el, err) {
      var $el = $(el);

      this.clearError(el);

      // Add all these error classes, as Shiny does
      var errClass = "shiny-output-error";
      if (err.type !== null) {
        // use the classes of the error condition as CSS class names
        errClass = errClass + " " + $.map(asArray(err.type), function(type) {
          return errClass + "-" + type;
        }).join(" ");
      }
      errClass = errClass + " htmlwidgets-error";

      // Is el inline or block? If inline or inline-block, just display:none it
      // and add an inline error.
      var display = $el.css("display");
      $el.data("restore-display-mode", display);

      if (display === "inline" || display === "inline-block") {
        $el.hide();
        if (err.message !== "") {
          var errorSpan = $("<span>").addClass(errClass);
          errorSpan.text(err.message);
          $el.after(errorSpan);
        }
      } else if (display === "block") {
        // If block, add an error just after the el, set visibility:none on the
        // el, and position the error to be on top of the el.
        // Mark it with a unique ID and CSS class so we can remove it later.
        $el.css("visibility", "hidden");
        if (err.message !== "") {
          var errorDiv = $("<div>").addClass(errClass).css("position", "absolute")
            .css("top", el.offsetTop)
            .css("left", el.offsetLeft)
            // setting width can push out the page size, forcing otherwise
            // unnecessary scrollbars to appear and making it impossible for
            // the element to shrink; so use max-width instead
            .css("maxWidth", el.offsetWidth)
            .css("height", el.offsetHeight);
          errorDiv.text(err.message);
          $el.after(errorDiv);

          // Really dumb way to keep the size/position of the error in sync with
          // the parent element as the window is resized or whatever.
          var intId = setInterval(function() {
            if (!errorDiv[0].parentElement) {
              clearInterval(intId);
              return;
            }
            errorDiv
              .css("top", el.offsetTop)
              .css("left", el.offsetLeft)
              .css("maxWidth", el.offsetWidth)
              .css("height", el.offsetHeight);
          }, 500);
        }
      }
    },
    clearError: function(el) {
      var $el = $(el);
      var display = $el.data("restore-display-mode");
      $el.data("restore-display-mode", null);

      if (display === "inline" || display === "inline-block") {
        if (display)
          $el.css("display", display);
        $(el.nextSibling).filter(".htmlwidgets-error").remove();
      } else if (display === "block"){
        $el.css("visibility", "inherit");
        $(el.nextSibling).filter(".htmlwidgets-error").remove();
      }
    },
    sizing: {}
  };

  // Called by widget bindings to register a new type of widget. The definition
  // object can contain the following properties:
  // - name (required) - A string indicating the binding name, which will be
  //   used by default as the CSS classname to look for.
  // - initialize (optional) - A function(el) that will be called once per
  //   widget element; if a value is returned, it will be passed as the third
  //   value to renderValue.
  // - renderValue (required) - A function(el, data, initValue) that will be
  //   called with data. Static contexts will cause this to be called once per
  //   element; Shiny apps will cause this to be called multiple times per
  //   element, as the data changes.
  window.HTMLWidgets.widget = function(definition) {
    if (!definition.name) {
      throw new Error("Widget must have a name");
    }
    if (!definition.type) {
      throw new Error("Widget must have a type");
    }
    // Currently we only support output widgets
    if (definition.type !== "output") {
      throw new Error("Unrecognized widget type '" + definition.type + "'");
    }
    // TODO: Verify that .name is a valid CSS classname

    // Support new-style instance-bound definitions. Old-style class-bound
    // definitions have one widget "object" per widget per type/class of
    // widget; the renderValue and resize methods on such widget objects
    // take el and instance arguments, because the widget object can't
    // store them. New-style instance-bound definitions have one widget
    // object per widget instance; the definition that's passed in doesn't
    // provide renderValue or resize methods at all, just the single method
    //   factory(el, width, height)
    // which returns an object that has renderValue(x) and resize(w, h).
    // This enables a far more natural programming style for the widget
    // author, who can store per-instance state using either OO-style
    // instance fields or functional-style closure variables (I guess this
    // is in contrast to what can only be called C-style pseudo-OO which is
    // what we required before).
    if (definition.factory) {
      definition = createLegacyDefinitionAdapter(definition);
    }

    if (!definition.renderValue) {
      throw new Error("Widget must have a renderValue function");
    }

    // For static rendering (non-Shiny), use a simple widget registration
    // scheme. We also use this scheme for Shiny apps/documents that also
    // contain static widgets.
    window.HTMLWidgets.widgets = window.HTMLWidgets.widgets || [];
    // Merge defaults into the definition; don't mutate the original definition.
    var staticBinding = extend({}, defaults, definition);
    overrideMethod(staticBinding, "find", function(superfunc) {
      return function(scope) {
        var results = superfunc(scope);
        // Filter out Shiny outputs, we only want the static kind
        return filterByClass(results, "html-widget-output", false);
      };
    });
    window.HTMLWidgets.widgets.push(staticBinding);

    if (shinyMode) {
      // Shiny is running. Register the definition with an output binding.
      // The definition itself will not be the output binding, instead
      // we will make an output binding object that delegates to the
      // definition. This is because we foolishly used the same method
      // name (renderValue) for htmlwidgets definition and Shiny bindings
      // but they actually have quite different semantics (the Shiny
      // bindings receive data that includes lots of metadata that it
      // strips off before calling htmlwidgets renderValue). We can't
      // just ignore the difference because in some widgets it's helpful
      // to call this.renderValue() from inside of resize(), and if
      // we're not delegating, then that call will go to the Shiny
      // version instead of the htmlwidgets version.

      // Merge defaults with definition, without mutating either.
      var bindingDef = extend({}, defaults, definition);

      // This object will be our actual Shiny binding.
      var shinyBinding = new Shiny.OutputBinding();

      // With a few exceptions, we'll want to simply use the bindingDef's
      // version of methods if they are available, otherwise fall back to
      // Shiny's defaults. NOTE: If Shiny's output bindings gain additional
      // methods in the future, and we want them to be overrideable by
      // HTMLWidget binding definitions, then we'll need to add them to this
      // list.
      delegateMethod(shinyBinding, bindingDef, "getId");
      delegateMethod(shinyBinding, bindingDef, "onValueChange");
      delegateMethod(shinyBinding, bindingDef, "onValueError");
      delegateMethod(shinyBinding, bindingDef, "renderError");
      delegateMethod(shinyBinding, bindingDef, "clearError");
      delegateMethod(shinyBinding, bindingDef, "showProgress");

      // The find, renderValue, and resize are handled differently, because we
      // want to actually decorate the behavior of the bindingDef methods.

      shinyBinding.find = function(scope) {
        var results = bindingDef.find(scope);

        // Only return elements that are Shiny outputs, not static ones
        var dynamicResults = results.filter(".html-widget-output");

        // It's possible that whatever caused Shiny to think there might be
        // new dynamic outputs, also caused there to be new static outputs.
        // Since there might be lots of different htmlwidgets bindings, we
        // schedule execution for later--no need to staticRender multiple
        // times.
        if (results.length !== dynamicResults.length)
          scheduleStaticRender();

        return dynamicResults;
      };

      // Wrap renderValue to handle initialization, which unfortunately isn't
      // supported natively by Shiny at the time of this writing.

      shinyBinding.renderValue = function(el, data) {
        Shiny.renderDependencies(data.deps);
        // Resolve strings marked as javascript literals to objects
        if (!(data.evals instanceof Array)) data.evals = [data.evals];
        for (var i = 0; data.evals && i < data.evals.length; i++) {
          window.HTMLWidgets.evaluateStringMember(data.x, data.evals[i]);
        }
        if (!bindingDef.renderOnNullValue) {
          if (data.x === null) {
            el.style.visibility = "hidden";
            return;
          } else {
            el.style.visibility = "inherit";
          }
        }
        if (!elementData(el, "initialized")) {
          initSizing(el);

          elementData(el, "initialized", true);
          if (bindingDef.initialize) {
            var result = bindingDef.initialize(el, el.offsetWidth,
              el.offsetHeight);
            elementData(el, "init_result", result);
          }
        }
        bindingDef.renderValue(el, data.x, elementData(el, "init_result"));
        evalAndRun(data.jsHooks.render, elementData(el, "init_result"), [el, data.x]);
      };

      // Only override resize if bindingDef implements it
      if (bindingDef.resize) {
        shinyBinding.resize = function(el, width, height) {
          // Shiny can call resize before initialize/renderValue have been
          // called, which doesn't make sense for widgets.
          if (elementData(el, "initialized")) {
            bindingDef.resize(el, width, height, elementData(el, "init_result"));
          }
        };
      }

      Shiny.outputBindings.register(shinyBinding, bindingDef.name);
    }
  };

  var scheduleStaticRenderTimerId = null;
  function scheduleStaticRender() {
    if (!scheduleStaticRenderTimerId) {
      scheduleStaticRenderTimerId = setTimeout(function() {
        scheduleStaticRenderTimerId = null;
        window.HTMLWidgets.staticRender();
      }, 1);
    }
  }

  // Render static widgets after the document finishes loading
  // Statically render all elements that are of this widget's class
  window.HTMLWidgets.staticRender = function() {
    var bindings = window.HTMLWidgets.widgets || [];
    forEach(bindings, function(binding) {
      var matches = binding.find(document.documentElement);
      forEach(matches, function(el) {
        var sizeObj = initSizing(el, binding);

        if (hasClass(el, "html-widget-static-bound"))
          return;
        el.className = el.className + " html-widget-static-bound";

        var initResult;
        if (binding.initialize) {
          initResult = binding.initialize(el,
            sizeObj ? sizeObj.getWidth() : el.offsetWidth,
            sizeObj ? sizeObj.getHeight() : el.offsetHeight
          );
          elementData(el, "init_result", initResult);
        }

        if (binding.resize) {
          var lastSize = {
            w: sizeObj ? sizeObj.getWidth() : el.offsetWidth,
            h: sizeObj ? sizeObj.getHeight() : el.offsetHeight
          };
          var resizeHandler = function(e) {
            var size = {
              w: sizeObj ? sizeObj.getWidth() : el.offsetWidth,
              h: sizeObj ? sizeObj.getHeight() : el.offsetHeight
            };
            if (size.w === 0 && size.h === 0)
              return;
            if (size.w === lastSize.w && size.h === lastSize.h)
              return;
            lastSize = size;
            binding.resize(el, size.w, size.h, initResult);
          };

          on(window, "resize", resizeHandler);

          // This is needed for cases where we're running in a Shiny
          // app, but the widget itself is not a Shiny output, but
          // rather a simple static widget. One example of this is
          // an rmarkdown document that has runtime:shiny and widget
          // that isn't in a render function. Shiny only knows to
          // call resize handlers for Shiny outputs, not for static
          // widgets, so we do it ourselves.
          if (window.jQuery) {
            window.jQuery(document).on(
              "shown.htmlwidgets shown.bs.tab.htmlwidgets shown.bs.collapse.htmlwidgets",
              resizeHandler
            );
            window.jQuery(document).on(
              "hidden.htmlwidgets hidden.bs.tab.htmlwidgets hidden.bs.collapse.htmlwidgets",
              resizeHandler
            );
          }

          // This is needed for the specific case of ioslides, which
          // flips slides between display:none and display:block.
          // Ideally we would not have to have ioslide-specific code
          // here, but rather have ioslides raise a generic event,
          // but the rmarkdown package just went to CRAN so the
          // window to getting that fixed may be long.
          if (window.addEventListener) {
            // It's OK to limit this to window.addEventListener
            // browsers because ioslides itself only supports
            // such browsers.
            on(document, "slideenter", resizeHandler);
            on(document, "slideleave", resizeHandler);
          }
        }

        var scriptData = document.querySelector("script[data-for='" + el.id + "'][type='application/json']");
        if (scriptData) {
          var data = JSON.parse(scriptData.textContent || scriptData.text);
          // Resolve strings marked as javascript literals to objects
          if (!(data.evals instanceof Array)) data.evals = [data.evals];
          for (var k = 0; data.evals && k < data.evals.length; k++) {
            window.HTMLWidgets.evaluateStringMember(data.x, data.evals[k]);
          }
          binding.renderValue(el, data.x, initResult);
          evalAndRun(data.jsHooks.render, initResult, [el, data.x]);
        }
      });
    });

    invokePostRenderHandlers();
  }


  function has_jQuery3() {
    if (!window.jQuery) {
      return false;
    }
    var $version = window.jQuery.fn.jquery;
    var $major_version = parseInt($version.split(".")[0]);
    return $major_version >= 3;
  }

  /*
  / Shiny 1.4 bumped jQuery from 1.x to 3.x which means jQuery's
  / on-ready handler (i.e., $(fn)) is now asyncronous (i.e., it now
  / really means $(setTimeout(fn)).
  / https://jquery.com/upgrade-guide/3.0/#breaking-change-document-ready-handlers-are-now-asynchronous
  /
  / Since Shiny uses $() to schedule initShiny, shiny>=1.4 calls initShiny
  / one tick later than it did before, which means staticRender() is
  / called renderValue() earlier than (advanced) widget authors might be expecting.
  / https://github.com/rstudio/shiny/issues/2630
  /
  / For a concrete example, leaflet has some methods (e.g., updateBounds)
  / which reference Shiny methods registered in initShiny (e.g., setInputValue).
  / Since leaflet is privy to this life-cycle, it knows to use setTimeout() to
  / delay execution of those methods (until Shiny methods are ready)
  / https://github.com/rstudio/leaflet/blob/18ec981/javascript/src/index.js#L266-L268
  /
  / Ideally widget authors wouldn't need to use this setTimeout() hack that
  / leaflet uses to call Shiny methods on a staticRender(). In the long run,
  / the logic initShiny should be broken up so that method registration happens
  / right away, but binding happens later.
  */
  function maybeStaticRenderLater() {
    if (shinyMode && has_jQuery3()) {
      window.jQuery(window.HTMLWidgets.staticRender);
    } else {
      window.HTMLWidgets.staticRender();
    }
  }

  if (document.addEventListener) {
    document.addEventListener("DOMContentLoaded", function() {
      document.removeEventListener("DOMContentLoaded", arguments.callee, false);
      maybeStaticRenderLater();
    }, false);
  } else if (document.attachEvent) {
    document.attachEvent("onreadystatechange", function() {
      if (document.readyState === "complete") {
        document.detachEvent("onreadystatechange", arguments.callee);
        maybeStaticRenderLater();
      }
    });
  }


  window.HTMLWidgets.getAttachmentUrl = function(depname, key) {
    // If no key, default to the first item
    if (typeof(key) === "undefined")
      key = 1;

    var link = document.getElementById(depname + "-" + key + "-attachment");
    if (!link) {
      throw new Error("Attachment " + depname + "/" + key + " not found in document");
    }
    return link.getAttribute("href");
  };

  window.HTMLWidgets.dataframeToD3 = function(df) {
    var names = [];
    var length;
    for (var name in df) {
        if (df.hasOwnProperty(name))
            names.push(name);
        if (typeof(df[name]) !== "object" || typeof(df[name].length) === "undefined") {
            throw new Error("All fields must be arrays");
        } else if (typeof(length) !== "undefined" && length !== df[name].length) {
            throw new Error("All fields must be arrays of the same length");
        }
        length = df[name].length;
    }
    var results = [];
    var item;
    for (var row = 0; row < length; row++) {
        item = {};
        for (var col = 0; col < names.length; col++) {
            item[names[col]] = df[names[col]][row];
        }
        results.push(item);
    }
    return results;
  };

  window.HTMLWidgets.transposeArray2D = function(array) {
      if (array.length === 0) return array;
      var newArray = array[0].map(function(col, i) {
          return array.map(function(row) {
              return row[i]
          })
      });
      return newArray;
  };
  // Split value at splitChar, but allow splitChar to be escaped
  // using escapeChar. Any other characters escaped by escapeChar
  // will be included as usual (including escapeChar itself).
  function splitWithEscape(value, splitChar, escapeChar) {
    var results = [];
    var escapeMode = false;
    var currentResult = "";
    for (var pos = 0; pos < value.length; pos++) {
      if (!escapeMode) {
        if (value[pos] === splitChar) {
          results.push(currentResult);
          currentResult = "";
        } else if (value[pos] === escapeChar) {
          escapeMode = true;
        } else {
          currentResult += value[pos];
        }
      } else {
        currentResult += value[pos];
        escapeMode = false;
      }
    }
    if (currentResult !== "") {
      results.push(currentResult);
    }
    return results;
  }
  // Function authored by Yihui/JJ Allaire
  window.HTMLWidgets.evaluateStringMember = function(o, member) {
    var parts = splitWithEscape(member, '.', '\\');
    for (var i = 0, l = parts.length; i < l; i++) {
      var part = parts[i];
      // part may be a character or 'numeric' member name
      if (o !== null && typeof o === "object" && part in o) {
        if (i == (l - 1)) { // if we are at the end of the line then evalulate
          if (typeof o[part] === "string")
            o[part] = tryEval(o[part]);
        } else { // otherwise continue to next embedded object
          o = o[part];
        }
      }
    }
  };

  // Retrieve the HTMLWidget instance (i.e. the return value of an
  // HTMLWidget binding's initialize() or factory() function)
  // associated with an element, or null if none.
  window.HTMLWidgets.getInstance = function(el) {
    return elementData(el, "init_result");
  };

  // Finds the first element in the scope that matches the selector,
  // and returns the HTMLWidget instance (i.e. the return value of
  // an HTMLWidget binding's initialize() or factory() function)
  // associated with that element, if any. If no element matches the
  // selector, or the first matching element has no HTMLWidget
  // instance associated with it, then null is returned.
  //
  // The scope argument is optional, and defaults to window.document.
  window.HTMLWidgets.find = function(scope, selector) {
    if (arguments.length == 1) {
      selector = scope;
      scope = document;
    }

    var el = scope.querySelector(selector);
    if (el === null) {
      return null;
    } else {
      return window.HTMLWidgets.getInstance(el);
    }
  };

  // Finds all elements in the scope that match the selector, and
  // returns the HTMLWidget instances (i.e. the return values of
  // an HTMLWidget binding's initialize() or factory() function)
  // associated with the elements, in an array. If elements that
  // match the selector don't have an associated HTMLWidget
  // instance, the returned array will contain nulls.
  //
  // The scope argument is optional, and defaults to window.document.
  window.HTMLWidgets.findAll = function(scope, selector) {
    if (arguments.length == 1) {
      selector = scope;
      scope = document;
    }

    var nodes = scope.querySelectorAll(selector);
    var results = [];
    for (var i = 0; i < nodes.length; i++) {
      results.push(window.HTMLWidgets.getInstance(nodes[i]));
    }
    return results;
  };

  var postRenderHandlers = [];
  function invokePostRenderHandlers() {
    while (postRenderHandlers.length) {
      var handler = postRenderHandlers.shift();
      if (handler) {
        handler();
      }
    }
  }

  // Register the given callback function to be invoked after the
  // next time static widgets are rendered.
  window.HTMLWidgets.addPostRenderHandler = function(callback) {
    postRenderHandlers.push(callback);
  };

  // Takes a new-style instance-bound definition, and returns an
  // old-style class-bound definition. This saves us from having
  // to rewrite all the logic in this file to accomodate both
  // types of definitions.
  function createLegacyDefinitionAdapter(defn) {
    var result = {
      name: defn.name,
      type: defn.type,
      initialize: function(el, width, height) {
        return defn.factory(el, width, height);
      },
      renderValue: function(el, x, instance) {
        return instance.renderValue(x);
      },
      resize: function(el, width, height, instance) {
        return instance.resize(width, height);
      }
    };

    if (defn.find)
      result.find = defn.find;
    if (defn.renderError)
      result.renderError = defn.renderError;
    if (defn.clearError)
      result.clearError = defn.clearError;

    return result;
  }
})();

</script>
<style type="text/css">.dt-crosstalk-fade {
opacity: 0.2;
}
html body div.DTS div.dataTables_scrollBody {
background: none;
}

table.dataTable {
display: table;
}
</style>
<script>(function() {

// some helper functions: using a global object DTWidget so that it can be used
// in JS() code, e.g. datatable(options = list(foo = JS('code'))); unlike R's
// dynamic scoping, when 'code' is eval()'ed, JavaScript does not know objects
// from the "parent frame", e.g. JS('DTWidget') will not work unless it was made
// a global object
var DTWidget = {};

// 123456666.7890 -> 123,456,666.7890
var markInterval = function(d, digits, interval, mark, decMark, precision) {
  x = precision ? d.toPrecision(digits) : d.toFixed(digits);
  if (!/^-?[\d.]+$/.test(x)) return x;
  var xv = x.split('.');
  if (xv.length > 2) return x;  // should have at most one decimal point
  xv[0] = xv[0].replace(new RegExp('\\B(?=(\\d{' + interval + '})+(?!\\d))', 'g'), mark);
  return xv.join(decMark);
};

DTWidget.formatCurrency = function(data, currency, digits, interval, mark, decMark, before) {
  var d = parseFloat(data);
  if (isNaN(d)) return '';
  var res = markInterval(d, digits, interval, mark, decMark);
  res = before ? (/^-/.test(res) ? '-' + currency + res.replace(/^-/, '') : currency + res) :
    res + currency;
  return res;
};

DTWidget.formatString = function(data, prefix, suffix) {
  var d = data;
  if (d === null) return '';
  return prefix + d + suffix;
};

DTWidget.formatPercentage = function(data, digits, interval, mark, decMark) {
  var d = parseFloat(data);
  if (isNaN(d)) return '';
  return markInterval(d * 100, digits, interval, mark, decMark) + '%';
};

DTWidget.formatRound = function(data, digits, interval, mark, decMark) {
  var d = parseFloat(data);
  if (isNaN(d)) return '';
  return markInterval(d, digits, interval, mark, decMark);
};

DTWidget.formatSignif = function(data, digits, interval, mark, decMark) {
  var d = parseFloat(data);
  if (isNaN(d)) return '';
  return markInterval(d, digits, interval, mark, decMark, true);
};

DTWidget.formatDate = function(data, method, params) {
  var d = data;
  if (d === null) return '';
  // (new Date('2015-10-28')).toDateString() may return 2015-10-27 because the
  // actual time created could be like 'Tue Oct 27 2015 19:00:00 GMT-0500 (CDT)',
  // i.e. the date-only string is treated as UTC time instead of local time
  if ((method === 'toDateString' || method === 'toLocaleDateString') && /^\d{4,}\D\d{2}\D\d{2}$/.test(d)) {
    d = d.split(/\D/);
    d = new Date(d[0], d[1] - 1, d[2]);
  } else {
    d = new Date(d);
  }
  return d[method].apply(d, params);
};

window.DTWidget = DTWidget;

var transposeArray2D = function(a) {
  return a.length === 0 ? a : HTMLWidgets.transposeArray2D(a);
};

var crosstalkPluginsInstalled = false;

function maybeInstallCrosstalkPlugins() {
  if (crosstalkPluginsInstalled)
    return;
  crosstalkPluginsInstalled = true;

  $.fn.dataTable.ext.afnFiltering.push(
    function(oSettings, aData, iDataIndex) {
      var ctfilter = oSettings.nTable.ctfilter;
      if (ctfilter && !ctfilter[iDataIndex])
        return false;

      var ctselect = oSettings.nTable.ctselect;
      if (ctselect && !ctselect[iDataIndex])
        return false;

      return true;
    }
  );
}

HTMLWidgets.widget({
  name: "datatables",
  type: "output",
  renderOnNullValue: true,
  initialize: function(el, width, height) {
    $(el).html('&nbsp;');
    return {
      data: null,
      ctfilterHandle: new crosstalk.FilterHandle(),
      ctfilterSubscription: null,
      ctselectHandle: new crosstalk.SelectionHandle(),
      ctselectSubscription: null
    };
  },
  renderValue: function(el, data, instance) {
    if (el.offsetWidth === 0 || el.offsetHeight === 0) {
      instance.data = data;
      return;
    }
    instance.data = null;
    var $el = $(el);
    $el.empty();

    if (data === null) {
      $el.append('&nbsp;');
      // clear previous Shiny inputs (if any)
      for (var i in instance.clearInputs) instance.clearInputs[i]();
      instance.clearInputs = {};
      return;
    }

    var crosstalkOptions = data.crosstalkOptions;
    if (!crosstalkOptions) crosstalkOptions = {
      'key': null, 'group': null
    };
    if (crosstalkOptions.group) {
      maybeInstallCrosstalkPlugins();
      instance.ctfilterHandle.setGroup(crosstalkOptions.group);
      instance.ctselectHandle.setGroup(crosstalkOptions.group);
    }

    // If we are in a flexdashboard scroll layout then we:
    //  (a) Always want to use pagination (otherwise we'll have
    //      a "double scroll bar" effect on the phone); and
    //  (b) Never want to fill the container (we want the pagination
    //      level to determine the size of the container)
    if (window.FlexDashboard && !window.FlexDashboard.isFillPage()) {
      data.options.paging = true;
      data.fillContainer = false;
    }

    // if we are in the viewer then we always want to fillContainer and
    // and autoHideNavigation (unless the user has explicitly set these)
    if (window.HTMLWidgets.viewerMode) {
      if (!data.hasOwnProperty("fillContainer"))
        data.fillContainer = true;
      if (!data.hasOwnProperty("autoHideNavigation"))
        data.autoHideNavigation = true;
    }

    // propagate fillContainer to instance (so we have it in resize)
    instance.fillContainer = data.fillContainer;

    var cells = data.data;

    if (cells instanceof Array) cells = transposeArray2D(cells);

    $el.append(data.container);
    var $table = $el.find('table');
    if (data.class) $table.addClass(data.class);
    if (data.caption) $table.prepend(data.caption);

    if (!data.selection) data.selection = {
      mode: 'none', selected: null, target: 'row', selectable: null
    };
    if (HTMLWidgets.shinyMode && data.selection.mode !== 'none' &&
        data.selection.target === 'row+column') {
      if ($table.children('tfoot').length === 0) {
        $table.append($('<tfoot>'));
        $table.find('thead tr').clone().appendTo($table.find('tfoot'));
      }
    }

    // column filters
    var filterRow;
    switch (data.filter) {
      case 'top':
        $table.children('thead').append(data.filterHTML);
        filterRow = $table.find('thead tr:last td');
        break;
      case 'bottom':
        if ($table.children('tfoot').length === 0) {
          $table.append($('<tfoot>'));
        }
        $table.children('tfoot').prepend(data.filterHTML);
        filterRow = $table.find('tfoot tr:first td');
        break;
    }

    var options = { searchDelay: 1000 };
    if (cells !== null) $.extend(options, {
      data: cells
    });

    // options for fillContainer
    var bootstrapActive = typeof($.fn.popover) != 'undefined';
    if (instance.fillContainer) {

      // force scrollX/scrollY and turn off autoWidth
      options.scrollX = true;
      options.scrollY = "100px"; // can be any value, we'll adjust below

      // if we aren't paginating then move around the info/filter controls
      // to save space at the bottom and rephrase the info callback
      if (data.options.paging === false) {

        // we know how to do this cleanly for bootstrap, not so much
        // for other themes/layouts
        if (bootstrapActive) {
          options.dom = "<'row'<'col-sm-4'i><'col-sm-8'f>>" +
                        "<'row'<'col-sm-12'tr>>";
        }

        options.fnInfoCallback = function(oSettings, iStart, iEnd,
                                           iMax, iTotal, sPre) {
          return Number(iTotal).toLocaleString() + " records";
        };
      }
    }

    // auto hide navigation if requested
    // Note, this only works on client-side processing mode as on server-side,
    // cells (data.data) is null; In addition, we require the pageLength option
    // being provided explicitly to enable this. Despite we may be able to deduce
    // the default value of pageLength, it may complicate things so we'd rather
    // put this responsiblity to users and warn them on the R side.
    if (data.autoHideNavigation === true && data.options.paging !== false) {
      // strip all nav if length >= cells
      if ((cells instanceof Array) && data.options.pageLength >= cells.length)
        options.dom = bootstrapActive ? "<'row'<'col-sm-12'tr>>" : "t";
      // alternatively lean things out for flexdashboard mobile portrait
      else if (bootstrapActive && window.FlexDashboard && window.FlexDashboard.isMobilePhone())
        options.dom = "<'row'<'col-sm-12'f>>" +
                      "<'row'<'col-sm-12'tr>>"  +
                      "<'row'<'col-sm-12'p>>";
    }

    $.extend(true, options, data.options || {});

    var searchCols = options.searchCols;
    if (searchCols) {
      searchCols = searchCols.map(function(x) {
        return x === null ? '' : x.search;
      });
      // FIXME: this means I don't respect the escapeRegex setting
      delete options.searchCols;
    }

    // server-side processing?
    var server = options.serverSide === true;

    // use the dataSrc function to pre-process JSON data returned from R
    var DT_rows_all = [], DT_rows_current = [];
    if (server && HTMLWidgets.shinyMode && typeof options.ajax === 'object' &&
        /^session\/[\da-z]+\/dataobj/.test(options.ajax.url) && !options.ajax.dataSrc) {
      options.ajax.dataSrc = function(json) {
        DT_rows_all = $.makeArray(json.DT_rows_all);
        DT_rows_current = $.makeArray(json.DT_rows_current);
        var data = json.data;
        if (!colReorderEnabled()) return data;
        var table = $table.DataTable(), order = table.colReorder.order(), flag = true, i, j, row;
        for (i = 0; i < order.length; ++i) if (order[i] !== i) flag = false;
        if (flag) return data;
        for (i = 0; i < data.length; ++i) {
          row = data[i].slice();
          for (j = 0; j < order.length; ++j) data[i][j] = row[order[j]];
        }
        return data;
      };
    }

    var thiz = this;
    if (instance.fillContainer) $table.on('init.dt', function(e) {
      thiz.fillAvailableHeight(el, $(el).innerHeight());
    });
    // If the page contains serveral datatables and one of which enables colReorder,
    // the table.colReorder.order() function will exist but throws error when called.
    // So it seems like the only way to know if colReorder is enabled or not is to
    // check the options.
    var colReorderEnabled = function() { return "colReorder" in options; };
    var table = $table.DataTable(options);
    $el.data('datatable', table);

    // Unregister previous Crosstalk event subscriptions, if they exist
    if (instance.ctfilterSubscription) {
      instance.ctfilterHandle.off("change", instance.ctfilterSubscription);
      instance.ctfilterSubscription = null;
    }
    if (instance.ctselectSubscription) {
      instance.ctselectHandle.off("change", instance.ctselectSubscription);
      instance.ctselectSubscription = null;
    }

    if (!crosstalkOptions.group) {
      $table[0].ctfilter = null;
      $table[0].ctselect = null;
    } else {
      var key = crosstalkOptions.key;
      function keysToMatches(keys) {
        if (!keys) {
          return null;
        } else {
          var selectedKeys = {};
          for (var i = 0; i < keys.length; i++) {
            selectedKeys[keys[i]] = true;
          }
          var matches = {};
          for (var j = 0; j < key.length; j++) {
            if (selectedKeys[key[j]])
              matches[j] = true;
          }
          return matches;
        }
      }

      function applyCrosstalkFilter(e) {
        $table[0].ctfilter = keysToMatches(e.value);
        table.draw();
      }
      instance.ctfilterSubscription = instance.ctfilterHandle.on("change", applyCrosstalkFilter);
      applyCrosstalkFilter({value: instance.ctfilterHandle.filteredKeys});

      function applyCrosstalkSelection(e) {
        if (e.sender !== instance.ctselectHandle) {
          table
            .rows('.' + selClass, {search: 'applied'})
            .nodes()
            .to$()
            .removeClass(selClass);
          if (selectedRows)
            changeInput('rows_selected', selectedRows(), void 0, true);
        }

        if (e.sender !== instance.ctselectHandle && e.value && e.value.length) {
          var matches = keysToMatches(e.value);

          // persistent selection with plotly (& leaflet)
          var ctOpts = crosstalk.var("plotlyCrosstalkOpts").get() || {};
          if (ctOpts.persistent === true) {
            var matches = $.extend(matches, $table[0].ctselect);
          }

          $table[0].ctselect = matches;
          table.draw();
        } else {
          if ($table[0].ctselect) {
            $table[0].ctselect = null;
            table.draw();
          }
        }
      }
      instance.ctselectSubscription = instance.ctselectHandle.on("change", applyCrosstalkSelection);
      // TODO: This next line doesn't seem to work when renderDataTable is used
      applyCrosstalkSelection({value: instance.ctselectHandle.value});
    }

    var inArray = function(val, array) {
      return $.inArray(val, $.makeArray(array)) > -1;
    };

    // encode + to %2B when searching in the table on server side, because
    // shiny::parseQueryString() treats + as spaces, and DataTables does not
    // encode + to %2B (or % to %25) when sending the request
    var encode_plus = function(x) {
      return server ? x.replace(/%/g, '%25').replace(/\+/g, '%2B') : x;
    };

    // search the i-th column
    var searchColumn = function(i, value) {
      var regex = false, ci = true;
      if (options.search) {
        regex = options.search.regex,
        ci = options.search.caseInsensitive !== false;
      }
      return table.column(i).search(encode_plus(value), regex, !regex, ci);
    };

    if (data.filter !== 'none') {

      filterRow.each(function(i, td) {

        var $td = $(td), type = $td.data('type'), filter;
        var $input = $td.children('div').first().children('input');
        $input.prop('disabled', !table.settings()[0].aoColumns[i].bSearchable || type === 'disabled');
        $input.on('input blur', function() {
          $input.next('span').toggle(Boolean($input.val()));
        });
        // Bootstrap sets pointer-events to none and we won't be able to click
        // the clear button
        $input.next('span').css('pointer-events', 'auto').hide().click(function() {
          $(this).hide().prev('input').val('').trigger('input').focus();
        });
        var searchCol;  // search string for this column
        if (searchCols && searchCols[i]) {
          searchCol = searchCols[i];
          $input.val(searchCol).trigger('input');
        }
        var $x = $td.children('div').last();

        // remove the overflow: hidden attribute of the scrollHead
        // (otherwise the scrolling table body obscures the filters)
        // The workaround and the discussion from
        // https://github.com/rstudio/DT/issues/554#issuecomment-518007347
        // Otherwise the filter selection will not be anchored to the values
        // when the columns number is many and scrollX is enabled.
        var scrollHead = $(el).find('.dataTables_scrollHead,.dataTables_scrollFoot');
        var cssOverflowHead = scrollHead.css('overflow');
        var scrollBody = $(el).find('.dataTables_scrollBody');
        var cssOverflowBody = scrollBody.css('overflow');
        var scrollTable = $(el).find('.dataTables_scroll');
        var cssOverflowTable = scrollTable.css('overflow');
        if (cssOverflowHead === 'hidden') {
          $x.on('show hide', function(e) {
            if (e.type === 'show') {
              scrollHead.css('overflow', 'visible');
              scrollBody.css('overflow', 'visible');
              scrollTable.css('overflow-x', 'scroll');
            } else {
              scrollHead.css('overflow', cssOverflowHead);
              scrollBody.css('overflow', cssOverflowBody);
              scrollTable.css('overflow-x', cssOverflowTable);
            }
          });
          $x.css('z-index', 25);
        }

        if (inArray(type, ['factor', 'logical'])) {
          $input.on({
            click: function() {
              $input.parent().hide(); $x.show().trigger('show'); filter[0].selectize.focus();
            },
            input: function() {
              if ($input.val() === '') filter[0].selectize.setValue([]);
            }
          });
          var $input2 = $x.children('select');
          filter = $input2.selectize({
            options: $input2.data('options').map(function(v, i) {
              return ({text: v, value: v});
            }),
            plugins: ['remove_button'],
            hideSelected: true,
            onChange: function(value) {
              if (value === null) value = []; // compatibility with jQuery 3.0
              $input.val(value.length ? JSON.stringify(value) : '');
              if (value.length) $input.trigger('input');
              $input.attr('title', $input.val());
              if (server) {
                table.column(i).search(value.length ? encode_plus(JSON.stringify(value)) : '').draw();
                return;
              }
              // turn off filter if nothing selected
              $td.data('filter', value.length > 0);
              table.draw();  // redraw table, and filters will be applied
            }
          });
          if (searchCol) filter[0].selectize.setValue(JSON.parse(searchCol));
          filter[0].selectize.on('blur', function() {
            $x.hide().trigger('hide'); $input.parent().show(); $input.trigger('blur');
          });
          filter.next('div').css('margin-bottom', 'auto');
        } else if (type === 'character') {
          var fun = function() {
            searchColumn(i, $input.val()).draw();
          };
          if (server) {
            fun = $.fn.dataTable.util.throttle(fun, options.searchDelay);
          }
          $input.on('input', fun);
        } else if (inArray(type, ['number', 'integer', 'date', 'time'])) {
          var $x0 = $x;
          $x = $x0.children('div').first();
          $x0.css({
            'background-color': '#fff',
            'border': '1px #ddd solid',
            'border-radius': '4px',
            'padding': '20px 20px 10px 20px'
          });
          var $spans = $x0.children('span').css({
            'margin-top': '10px',
            'white-space': 'nowrap'
          });
          var $span1 = $spans.first(), $span2 = $spans.last();
          var r1 = +$x.data('min'), r2 = +$x.data('max');
          // when the numbers are too small or have many decimal places, the
          // slider may have numeric precision problems (#150)
          var scale = Math.pow(10, Math.max(0, +$x.data('scale') || 0));
          r1 = Math.round(r1 * scale); r2 = Math.round(r2 * scale);
          var scaleBack = function(x, scale) {
            if (scale === 1) return x;
            var d = Math.round(Math.log(scale) / Math.log(10));
            // to avoid problems like 3.423/100 -> 0.034230000000000003
            return (x / scale).toFixed(d);
          };
          var slider_min = function() {
            return filter.noUiSlider('options').range.min;
          };
          var slider_max = function() {
            return filter.noUiSlider('options').range.max;
          };
          $input.on({
            focus: function() {
              $x0.show().trigger('show');
              // first, make sure the slider div leaves at least 20px between
              // the two (slider value) span's
              $x0.width(Math.max(160, $span1.outerWidth() + $span2.outerWidth() + 20));
              // then, if the input is really wide, make the slider the same
              // width as the input
              if ($x0.outerWidth() < $input.outerWidth()) {
                $x0.outerWidth($input.outerWidth());
              }
              // make sure the slider div does not reach beyond the right margin
              if ($(window).width() < $x0.offset().left + $x0.width()) {
                $x0.offset({
                  'left': $input.offset().left + $input.outerWidth() - $x0.outerWidth()
                });
              }
            },
            blur: function() {
              $x0.hide().trigger('hide');
            },
            input: function() {
              if ($input.val() === '') filter.val([slider_min(), slider_max()]);
            },
            change: function() {
              var v = $input.val().replace(/\s/g, '');
              if (v === '') return;
              v = v.split('...');
              if (v.length !== 2) {
                $input.parent().addClass('has-error');
                return;
              }
              if (v[0] === '') v[0] = slider_min();
              if (v[1] === '') v[1] = slider_max();
              $input.parent().removeClass('has-error');
              // treat date as UTC time at midnight
              var strTime = function(x) {
                var s = type === 'date' ? 'T00:00:00Z' : '';
                var t = new Date(x + s).getTime();
                // add 10 minutes to date since it does not hurt the date, and
                // it helps avoid the tricky floating point arithmetic problems,
                // e.g. sometimes the date may be a few milliseconds earlier
                // than the midnight due to precision problems in noUiSlider
                return type === 'date' ? t + 3600000 : t;
              };
              if (inArray(type, ['date', 'time'])) {
                v[0] = strTime(v[0]);
                v[1] = strTime(v[1]);
              }
              if (v[0] != slider_min()) v[0] *= scale;
              if (v[1] != slider_max()) v[1] *= scale;
              filter.val(v);
            }
          });
          var formatDate = function(d, isoFmt) {
            d = scaleBack(d, scale);
            if (type === 'number') return d;
            if (type === 'integer') return parseInt(d);
            var x = new Date(+d);
            var fmt = ('filterDateFmt' in data) ? data.filterDateFmt[i] : undefined;
            if (fmt !== undefined && isoFmt === false) return x[fmt.method].apply(x, fmt.params);
            if (type === 'date') {
              var pad0 = function(x) {
                return ('0' + x).substr(-2, 2);
              };
              return x.getUTCFullYear() + '-' + pad0(1 + x.getUTCMonth())
                      + '-' + pad0(x.getUTCDate());
            } else {
              return x.toISOString();
            }
          };
          var opts = type === 'date' ? { step: 60 * 60 * 1000 } :
                     type === 'integer' ? { step: 1 } : {};
          filter = $x.noUiSlider($.extend({
            start: [r1, r2],
            range: {min: r1, max: r2},
            connect: true
          }, opts));
          if (scale > 1) (function() {
            var t1 = r1, t2 = r2;
            var val = filter.val();
            while (val[0] > r1 || val[1] < r2) {
              if (val[0] > r1) {
                t1 -= val[0] - r1;
              }
              if (val[1] < r2) {
                t2 += r2 - val[1];
              }
              filter = $x.noUiSlider($.extend({
                start: [t1, t2],
                range: {min: t1, max: t2},
                connect: true
              }, opts), true);
              val = filter.val();
            }
            r1  = t1; r2 = t2;
          })();
          var updateSliderText = function(v1, v2) {
            $span1.text(formatDate(v1, false)); $span2.text(formatDate(v2, false));
          };
          updateSliderText(r1, r2);
          var updateSlider = function(e) {
            var val = filter.val();
            // turn off filter if in full range
            $td.data('filter', val[0] > slider_min() || val[1] < slider_max());
            var v1 = formatDate(val[0]), v2 = formatDate(val[1]), ival;
            if ($td.data('filter')) {
              ival = v1 + ' ... ' + v2;
              $input.attr('title', ival).val(ival).trigger('input');
            } else {
              $input.attr('title', '').val('');
            }
            updateSliderText(val[0], val[1]);
            if (e.type === 'slide') return;  // no searching when sliding only
            if (server) {
              table.column(i).search($td.data('filter') ? ival : '').draw();
              return;
            }
            table.draw();
          };
          filter.on({
            set: updateSlider,
            slide: updateSlider
          });
        }

        // server-side processing will be handled by R (or whatever server
        // language you use); the following code is only needed for client-side
        // processing
        if (server) {
          // if a search string has been pre-set, search now
          if (searchCol) searchColumn(i, searchCol).draw();
          return;
        }

        var customFilter = function(settings, data, dataIndex) {
          // there is no way to attach a search function to a specific table,
          // and we need to make sure a global search function is not applied to
          // all tables (i.e. a range filter in a previous table should not be
          // applied to the current table); we use the settings object to
          // determine if we want to perform searching on the current table,
          // since settings.sTableId will be different to different tables
          if (table.settings()[0] !== settings) return true;
          // no filter on this column or no need to filter this column
          if (typeof filter === 'undefined' || !$td.data('filter')) return true;

          var r = filter.val(), v, r0, r1;
          var i_data = function(i) {
            if (!colReorderEnabled()) return i;
            var order = table.colReorder.order(), k;
            for (k = 0; k < order.length; ++k) if (order[k] === i) return k;
            return i; // in theory it will never be here...
          }
          v = data[i_data(i)];
          if (type === 'number' || type === 'integer') {
            v = parseFloat(v);
            // how to handle NaN? currently exclude these rows
            if (isNaN(v)) return(false);
            r0 = parseFloat(scaleBack(r[0], scale))
            r1 = parseFloat(scaleBack(r[1], scale));
            if (v >= r0 && v <= r1) return true;
          } else if (type === 'date' || type === 'time') {
            v = new Date(v);
            r0 = new Date(r[0] / scale); r1 = new Date(r[1] / scale);
            if (v >= r0 && v <= r1) return true;
          } else if (type === 'factor') {
            if (r.length === 0 || inArray(v, r)) return true;
          } else if (type === 'logical') {
            if (r.length === 0) return true;
            if (inArray(v === '' ? 'na' : v, r)) return true;
          }
          return false;
        };

        $.fn.dataTable.ext.search.push(customFilter);

        // search for the preset search strings if it is non-empty
        if (searchCol) {
          if (inArray(type, ['factor', 'logical'])) {
            filter[0].selectize.setValue(JSON.parse(searchCol));
          } else if (type === 'character') {
            $input.trigger('input');
          } else if (inArray(type, ['number', 'integer', 'date', 'time'])) {
            $input.trigger('change');
          }
        }

      });

    }

    // highlight search keywords
    var highlight = function() {
      var body = $(table.table().body());
      // removing the old highlighting first
      body.unhighlight();

      // don't highlight the "not found" row, so we get the rows using the api
      if (table.rows({ filter: 'applied' }).data().length === 0) return;
      // highlight global search keywords
      body.highlight($.trim(table.search()).split(/\s+/));
      // then highlight keywords from individual column filters
      if (filterRow) filterRow.each(function(i, td) {
        var $td = $(td), type = $td.data('type');
        if (type !== 'character') return;
        var $input = $td.children('div').first().children('input');
        var column = table.column(i).nodes().to$(),
            val = $.trim($input.val());
        if (type !== 'character' || val === '') return;
        column.highlight(val.split(/\s+/));
      });
    };

    if (options.searchHighlight) {
      table
      .on('draw.dt.dth column-visibility.dt.dth column-reorder.dt.dth', highlight)
      .on('destroy', function() {
        // remove event handler
        table.off('draw.dt.dth column-visibility.dt.dth column-reorder.dt.dth');
      });

      // Set the option for escaping regex characters in our search string.  This will be used
      // for all future matching.
      jQuery.fn.highlight.options.escapeRegex = (!options.search || !options.search.regex);

      // initial highlight for state saved conditions and initial states
      highlight();
    }

    // run the callback function on the table instance
    if (typeof data.callback === 'function') data.callback(table);

    // double click to edit the cell, row, column, or all cells
    if (data.editable) table.on('dblclick.dt', 'tbody td', function(e) {
      // only bring up the editor when the cell itself is dbclicked, and ignore
      // other dbclick events bubbled up (e.g. from the <input>)
      if (e.target !== this) return;
      var target = [], immediate = false;
      switch (data.editable.target) {
        case 'cell':
          target = [this];
          immediate = true;  // edit will take effect immediately
          break;
        case 'row':
          target = table.cells(table.cell(this).index().row, '*').nodes();
          break;
        case 'column':
          target = table.cells('*', table.cell(this).index().column).nodes();
          break;
        case 'all':
          target = table.cells().nodes();
          break;
        default:
          throw 'The editable parameter must be "cell", "row", "column", or "all"';
      }
      var disableCols = data.editable.disable ? data.editable.disable.columns : null;
      for (var i = 0; i < target.length; i++) {
        (function(cell, current) {
          var $cell = $(cell), html = $cell.html();
          var _cell = table.cell(cell), value = _cell.data();
          var $input = $('<input type="text">'), changed = false;
          if (!immediate) {
            $cell.data('input', $input).data('html', html);
            $input.attr('title', 'Hit Ctrl+Enter to finish editing, or Esc to cancel');
          }
          $input.val(value);
          if (inArray(_cell.index().column, disableCols)) {
            $input.attr('readonly', '').css('filter', 'invert(25%)');
          }
          $cell.empty().append($input);
          if (cell === current) $input.focus();
          $input.css('width', '100%');

          if (immediate) $input.on('change', function() {
            changed = true;
            var valueNew = $input.val();
            if (valueNew != value) {
              _cell.data(valueNew);
              if (HTMLWidgets.shinyMode) {
                changeInput('cell_edit', [cellInfo(cell)], 'DT.cellInfo', null, {priority: "event"});
              }
              // for server-side processing, users have to call replaceData() to update the table
              if (!server) table.draw(false);
            } else {
              $cell.html(html);
            }
            $input.remove();
          }).on('blur', function() {
            if (!changed) $input.trigger('change');
          }).on('keyup', function(e) {
            // hit Escape to cancel editing
            if (e.keyCode === 27) $input.trigger('blur');
          });

          // bulk edit (row, column, or all)
          if (!immediate) $input.on('keyup', function(e) {
            var removeInput = function($cell, restore) {
              $cell.data('input').remove();
              if (restore) $cell.html($cell.data('html'));
            }
            if (e.keyCode === 27) {
              for (var i = 0; i < target.length; i++) {
                removeInput($(target[i]), true);
              }
            } else if (e.keyCode === 13 && e.ctrlKey) {
              // Ctrl + Enter
              var cell, $cell, _cell, cellData = [];
              for (var i = 0; i < target.length; i++) {
                cell = target[i]; $cell = $(cell); _cell = table.cell(cell);
                _cell.data($cell.data('input').val());
                HTMLWidgets.shinyMode && cellData.push(cellInfo(cell));
                removeInput($cell, false);
              }
              if (HTMLWidgets.shinyMode) {
                changeInput('cell_edit', cellData, 'DT.cellInfo', null, {priority: "event"});
              }
              if (!server) table.draw(false);
            }
          });
        })(target[i], this);
      }
    });

    // interaction with shiny
    if (!HTMLWidgets.shinyMode && !crosstalkOptions.group) return;

    var methods = {};
    var shinyData = {};

    methods.updateCaption = function(caption) {
      if (!caption) return;
      $table.children('caption').replaceWith(caption);
    }

    // register clear functions to remove input values when the table is removed
    instance.clearInputs = {};

    var changeInput = function(id, value, type, noCrosstalk, opts) {
      var event = id;
      id = el.id + '_' + id;
      if (type) id = id + ':' + type;
      // do not update if the new value is the same as old value
      if (event !== 'cell_edit' && !/_clicked$/.test(event) && shinyData.hasOwnProperty(id) && shinyData[id] === JSON.stringify(value))
        return;
      shinyData[id] = JSON.stringify(value);
      if (HTMLWidgets.shinyMode && Shiny.setInputValue) {
        Shiny.setInputValue(id, value, opts);
        if (!instance.clearInputs[id]) instance.clearInputs[id] = function() {
          Shiny.setInputValue(id, null);
        }
      }

      // HACK
      if (event === "rows_selected" && !noCrosstalk) {
        if (crosstalkOptions.group) {
          var keys = crosstalkOptions.key;
          var selectedKeys = null;
          if (value) {
            selectedKeys = [];
            for (var i = 0; i < value.length; i++) {
              // The value array's contents use 1-based row numbers, so we must
              // convert to 0-based before indexing into the keys array.
              selectedKeys.push(keys[value[i] - 1]);
            }
          }
          instance.ctselectHandle.set(selectedKeys);
        }
      }
    };

    var addOne = function(x) {
      return x.map(function(i) { return 1 + i; });
    };

    var unique = function(x) {
      var ux = [];
      $.each(x, function(i, el){
        if ($.inArray(el, ux) === -1) ux.push(el);
      });
      return ux;
    }

    // change the row index of a cell
    var tweakCellIndex = function(cell) {
      var info = cell.index();
      // some cell may not be valid. e.g, #759
      // when using the RowGroup extension, datatables will
      // generate the row label and the cells are not part of
      // the data thus contain no row/col info
      if (info === undefined)
        return {row: null, col: null};
      if (server) {
        info.row = DT_rows_current[info.row];
      } else {
        info.row += 1;
      }
      return {row: info.row, col: info.column};
    }

    var cleanSelectedValues = function() {
      changeInput('rows_selected', []);
      changeInput('columns_selected', []);
      changeInput('cells_selected', transposeArray2D([]), 'shiny.matrix');
    }
    // #828 we should clean the selection on the server-side when the table reloads
    cleanSelectedValues();

    // a flag to indicates if select extension is initialized or not
    var flagSelectExt = table.settings()[0]._select !== undefined;
    // the Select extension should only be used in the client mode and
    // when the selection.mode is set to none
    if (data.selection.mode === 'none' && !server && flagSelectExt) {
      var updateRowsSelected = function() {
        var rows = table.rows({selected: true});
        var selected = [];
        $.each(rows.indexes().toArray(), function(i, v) {
          selected.push(v + 1);
        });
        changeInput('rows_selected', selected);
      }
      var updateColsSelected = function() {
        var columns = table.columns({selected: true});
        changeInput('columns_selected', columns.indexes().toArray());
      }
      var updateCellsSelected = function() {
        var cells = table.cells({selected: true});
        var selected = [];
        cells.every(function() {
          var row = this.index().row;
          var col = this.index().column;
          selected = selected.concat([[row + 1, col]]);
        });
        changeInput('cells_selected', transposeArray2D(selected), 'shiny.matrix');
      }
      table.on('select deselect', function(e, dt, type, indexes) {
        updateRowsSelected();
        updateColsSelected();
        updateCellsSelected();
      })
    }

    var selMode = data.selection.mode, selTarget = data.selection.target;
    var selDisable = data.selection.selectable === false;
    if (inArray(selMode, ['single', 'multiple'])) {
      var selClass = inArray(data.style, ['bootstrap', 'bootstrap4']) ? 'active' : 'selected';
      // selected1: row indices; selected2: column indices
      var initSel = function(x) {
        if (x === null || typeof x === 'boolean' || selTarget === 'cell') {
          return {rows: [], cols: []};
        } else if (selTarget === 'row') {
          return {rows: $.makeArray(x), cols: []};
        } else if (selTarget === 'column') {
          return {rows: [], cols: $.makeArray(x)};
        } else if (selTarget === 'row+column') {
          return {rows: $.makeArray(x.rows), cols: $.makeArray(x.cols)};
        }
      }
      var selected = data.selection.selected;
      var selected1 = initSel(selected).rows, selected2 = initSel(selected).cols;
      // selectable should contain either all positive or all non-positive values, not both
      // positive values indicate "selectable" while non-positive values means "nonselectable"
      // the assertion is performed on R side. (only column indicides could be zero which indicates
      // the row name)
      var selectable = data.selection.selectable;
      var selectable1 = initSel(selectable).rows, selectable2 = initSel(selectable).cols;

      // After users reorder the rows or filter the table, we cannot use the table index
      // directly. Instead, we need this function to find out the rows between the two clicks.
      // If user filter the table again between the start click and the end click, the behavior
      // would be undefined, but it should not be a problem.
      var shiftSelRowsIndex = function(start, end) {
        var indexes = server ? DT_rows_all : table.rows({ search: 'applied' }).indexes().toArray();
        start = indexes.indexOf(start); end = indexes.indexOf(end);
        // if start is larger than end, we need to swap
        if (start > end) {
          var tmp = end; end = start; start = tmp;
        }
        return indexes.slice(start, end + 1);
      }

      var serverRowIndex = function(clientRowIndex) {
        return server ? DT_rows_current[clientRowIndex] : clientRowIndex + 1;
      }

      // row, column, or cell selection
      var lastClickedRow;
      if (inArray(selTarget, ['row', 'row+column'])) {
        // Get the current selected rows. It will also
        // update the selected1's value based on the current row selection state
        // Note we can't put this function inside selectRows() directly,
        // the reason is method.selectRows() will override selected1's value but this
        // function will add rows to selected1 (keep the existing selection), which is
        // inconsistent with column and cell selection.
        var selectedRows = function() {
          var rows = table.rows('.' + selClass);
          var idx = rows.indexes().toArray();
          if (!server) {
            selected1 = addOne(idx);
            return selected1;
          }
          idx = idx.map(function(i) {
            return DT_rows_current[i];
          });
          selected1 = selMode === 'multiple' ? unique(selected1.concat(idx)) : idx;
          return selected1;
        }
        // Change selected1's value based on selectable1, then refresh the row state
        var onlyKeepSelectableRows = function() {
          if (selDisable) { // users can't select; useful when only want backend select
            selected1 = [];
            return;
          }
          if (selectable1.length === 0) return;
          var nonselectable = selectable1[0] <= 0;
          if (nonselectable) {
            // should make selectable1 positive
            selected1 = $(selected1).not(selectable1.map(function(i) { return -i; })).get();
          } else {
            selected1 = $(selected1).filter(selectable1).get();
          }
        }
        // Change selected1's value based on selectable1, then
        // refresh the row selection state according to values in selected1
        var selectRows = function(ignoreSelectable) {
          if (!ignoreSelectable) onlyKeepSelectableRows();
          table.$('tr.' + selClass).removeClass(selClass);
          if (selected1.length === 0) return;
          if (server) {
            table.rows({page: 'current'}).every(function() {
              if (inArray(DT_rows_current[this.index()], selected1)) {
                $(this.node()).addClass(selClass);
              }
            });
          } else {
            var selected0 = selected1.map(function(i) { return i - 1; });
            $(table.rows(selected0).nodes()).addClass(selClass);
          }
        }
        table.on('mousedown.dt', 'tbody tr', function(e) {
          var $this = $(this), thisRow = table.row(this);
          if (selMode === 'multiple') {
            if (e.shiftKey && lastClickedRow !== undefined) {
              // select or de-select depends on the last clicked row's status
              var flagSel = !$this.hasClass(selClass);
              var crtClickedRow = serverRowIndex(thisRow.index());
              if (server) {
                var rowsIndex = shiftSelRowsIndex(lastClickedRow, crtClickedRow);
                // update current page's selClass
                rowsIndex.map(function(i) {
                  var rowIndex = DT_rows_current.indexOf(i);
                  if (rowIndex >= 0) {
                    var row = table.row(rowIndex).nodes().to$();
                    var flagRowSel = !row.hasClass(selClass);
                    if (flagSel === flagRowSel) row.toggleClass(selClass);
                  }
                });
                // update selected1
                if (flagSel) {
                  selected1 = unique(selected1.concat(rowsIndex));
                } else {
                  selected1 = selected1.filter(function(index) {
                    return !inArray(index, rowsIndex);
                  });
                }
              } else {
                // js starts from 0
                shiftSelRowsIndex(lastClickedRow - 1, crtClickedRow - 1).map(function(value) {
                  var row = table.row(value).nodes().to$();
                  var flagRowSel = !row.hasClass(selClass);
                  if (flagSel === flagRowSel) row.toggleClass(selClass);
                });
              }
              e.preventDefault();
            } else {
              $this.toggleClass(selClass);
            }
          } else {
            if ($this.hasClass(selClass)) {
              $this.removeClass(selClass);
            } else {
              table.$('tr.' + selClass).removeClass(selClass);
              $this.addClass(selClass);
            }
          }
          if (server && !$this.hasClass(selClass)) {
            var id = DT_rows_current[thisRow.index()];
            // remove id from selected1 since its class .selected has been removed
            if (inArray(id, selected1)) selected1.splice($.inArray(id, selected1), 1);
          }
          selectedRows(); // update selected1's value based on selClass
          selectRows(false); // only keep the selectable rows
          changeInput('rows_selected', selected1);
          changeInput('row_last_clicked', serverRowIndex(thisRow.index()), null, null, {priority: 'event'});
          lastClickedRow = serverRowIndex(thisRow.index());
        });
        selectRows(false);  // in case users have specified pre-selected rows
        // restore selected rows after the table is redrawn (e.g. sort/search/page);
        // client-side tables will preserve the selections automatically; for
        // server-side tables, we have to *real* row indices are in `selected1`
        changeInput('rows_selected', selected1);
        if (server) table.on('draw.dt', function(e) { selectRows(false); });
        methods.selectRows = function(selected, ignoreSelectable) {
          selected1 = $.makeArray(selected);
          selectRows(ignoreSelectable);
          changeInput('rows_selected', selected1);
        }
      }

      if (inArray(selTarget, ['column', 'row+column'])) {
        if (selTarget === 'row+column') {
          $(table.columns().footer()).css('cursor', 'pointer');
        }
        // update selected2's value based on selectable2
        var onlyKeepSelectableCols = function() {
          if (selDisable) { // users can't select; useful when only want backend select
            selected2 = [];
            return;
          }
          if (selectable2.length === 0) return;
          var nonselectable = selectable2[0] <= 0;
          if (nonselectable) {
            // need to make selectable2 positive
            selected2 = $(selected2).not(selectable2.map(function(i) { return -i; })).get();
          } else {
            selected2 = $(selected2).filter(selectable2).get();
          }
        }
        // update selected2 and then
        // refresh the col selection state according to values in selected2
        var selectCols = function(ignoreSelectable) {
          if (!ignoreSelectable) onlyKeepSelectableCols();
          // if selected2 is not a valide index (e.g., larger than the column number)
          // table.columns(selected2) will fail and result in a blank table
          // this is different from the table.rows(), where the out-of-range indexes
          // doesn't affect at all
          selected2 = $(selected2).filter(table.columns().indexes()).get();
          table.columns().nodes().flatten().to$().removeClass(selClass);
          if (selected2.length > 0)
            table.columns(selected2).nodes().flatten().to$().addClass(selClass);
        }
        var callback = function() {
          var colIdx = selTarget === 'column' ? table.cell(this).index().column :
              $.inArray(this, table.columns().footer()),
              thisCol = $(table.column(colIdx).nodes());
          if (colIdx === -1) return;
          if (thisCol.hasClass(selClass)) {
            thisCol.removeClass(selClass);
            selected2.splice($.inArray(colIdx, selected2), 1);
          } else {
            if (selMode === 'single') $(table.cells().nodes()).removeClass(selClass);
            thisCol.addClass(selClass);
            selected2 = selMode === 'single' ? [colIdx] : unique(selected2.concat([colIdx]));
          }
          selectCols(false); // update selected2 based on selectable
          changeInput('columns_selected', selected2);
        }
        if (selTarget === 'column') {
          $(table.table().body()).on('click.dt', 'td', callback);
        } else {
          $(table.table().footer()).on('click.dt', 'tr th', callback);
        }
        selectCols(false);  // in case users have specified pre-selected columns
        changeInput('columns_selected', selected2);
        if (server) table.on('draw.dt', function(e) { selectCols(false); });
        methods.selectColumns = function(selected, ignoreSelectable) {
          selected2 = $.makeArray(selected);
          selectCols(ignoreSelectable);
          changeInput('columns_selected', selected2);
        }
      }

      if (selTarget === 'cell') {
        var selected3 = [], selectable3 = [];
        if (selected !== null) selected3 = selected;
        if (selectable !== null && typeof selectable !== 'boolean') selectable3 = selectable;
        var findIndex = function(ij, sel) {
          for (var i = 0; i < sel.length; i++) {
            if (ij[0] === sel[i][0] && ij[1] === sel[i][1]) return i;
          }
          return -1;
        }
         // Change selected3's value based on selectable3, then refresh the cell state
        var onlyKeepSelectableCells = function() {
          if (selDisable) { // users can't select; useful when only want backend select
            selected3 = [];
            return;
          }
          if (selectable3.length === 0) return;
          var nonselectable = selectable3[0][0] <= 0;
          var out = [];
          if (nonselectable) {
            selected3.map(function(ij) {
              // should make selectable3 positive
              if (findIndex([-ij[0], -ij[1]], selectable3) === -1) { out.push(ij); }
            });
          } else {
            selected3.map(function(ij) {
              if (findIndex(ij, selectable3) > -1) { out.push(ij); }
            });
          }
          selected3 = out;
        }
        // Change selected3's value based on selectable3, then
        // refresh the cell selection state according to values in selected3
        var selectCells = function(ignoreSelectable) {
          if (!ignoreSelectable) onlyKeepSelectableCells();
          table.$('td.' + selClass).removeClass(selClass);
          if (selected3.length === 0) return;
          if (server) {
            table.cells({page: 'current'}).every(function() {
              var info = tweakCellIndex(this);
              if (findIndex([info.row, info.col], selected3) > -1)
                $(this.node()).addClass(selClass);
            });
          } else {
            selected3.map(function(ij) {
              $(table.cell(ij[0] - 1, ij[1]).node()).addClass(selClass);
            });
          }
        };
        table.on('click.dt', 'tbody td', function() {
          var $this = $(this), info = tweakCellIndex(table.cell(this));
          if ($this.hasClass(selClass)) {
            $this.removeClass(selClass);
            selected3.splice(findIndex([info.row, info.col], selected3), 1);
          } else {
            if (selMode === 'single') $(table.cells().nodes()).removeClass(selClass);
            $this.addClass(selClass);
            selected3 = selMode === 'single' ? [[info.row, info.col]] :
              unique(selected3.concat([[info.row, info.col]]));
          }
          selectCells(false); // must call this to update selected3 based on selectable3
          changeInput('cells_selected', transposeArray2D(selected3), 'shiny.matrix');
        });
        selectCells(false);  // in case users have specified pre-selected columns
        changeInput('cells_selected', transposeArray2D(selected3), 'shiny.matrix');

        if (server) table.on('draw.dt', function(e) { selectCells(false); });
        methods.selectCells = function(selected, ignoreSelectable) {
          selected3 = selected ? selected : [];
          selectCells(ignoreSelectable);
          changeInput('cells_selected', transposeArray2D(selected3), 'shiny.matrix');
        }
      }
    }

    // expose some table info to Shiny
    var updateTableInfo = function(e, settings) {
      // TODO: is anyone interested in the page info?
      // changeInput('page_info', table.page.info());
      var updateRowInfo = function(id, modifier) {
        var idx;
        if (server) {
          idx = modifier.page === 'current' ? DT_rows_current : DT_rows_all;
        } else {
          var rows = table.rows($.extend({
            search: 'applied',
            page: 'all'
          }, modifier));
          idx = addOne(rows.indexes().toArray());
        }
        changeInput('rows' + '_' + id, idx);
      };
      updateRowInfo('current', {page: 'current'});
      updateRowInfo('all', {});
    }
    table.on('draw.dt', updateTableInfo);
    updateTableInfo();

    // state info
    table.on('draw.dt column-visibility.dt', function() {
      changeInput('state', table.state());
    });
    changeInput('state', table.state());

    // search info
    var updateSearchInfo = function() {
      changeInput('search', table.search());
      if (filterRow) changeInput('search_columns', filterRow.toArray().map(function(td) {
        return $(td).find('input').first().val();
      }));
    }
    table.on('draw.dt', updateSearchInfo);
    updateSearchInfo();

    var cellInfo = function(thiz) {
      var info = tweakCellIndex(table.cell(thiz));
      info.value = table.cell(thiz).data();
      return info;
    }
    // the current cell clicked on
    table.on('click.dt', 'tbody td', function() {
      changeInput('cell_clicked', cellInfo(this), null, null, {priority: 'event'});
    })
    changeInput('cell_clicked', {});

    // do not trigger table selection when clicking on links unless they have classes
    table.on('click.dt', 'tbody td a', function(e) {
      if (this.className === '') e.stopPropagation();
    });

    methods.addRow = function(data, rowname, resetPaging) {
      var n = table.columns().indexes().length, d = n - data.length;
      if (d === 1) {
        data = rowname.concat(data)
      } else if (d !== 0) {
        console.log(data);
        console.log(table.columns().indexes());
        throw 'New data must be of the same length as current data (' + n + ')';
      };
      table.row.add(data).draw(resetPaging);
    }

    methods.updateSearch = function(keywords) {
      if (keywords.global !== null)
        $(table.table().container()).find('input[type=search]').first()
             .val(keywords.global).trigger('input');
      var columns = keywords.columns;
      if (!filterRow || columns === null) return;
      filterRow.toArray().map(function(td, i) {
        var v = typeof columns === 'string' ? columns : columns[i];
        if (typeof v === 'undefined') {
          console.log('The search keyword for column ' + i + ' is undefined')
          return;
        }
        $(td).find('input').first().val(v);
        searchColumn(i, v);
      });
      table.draw();
    }

    methods.hideCols = function(hide, reset) {
      if (reset) table.columns().visible(true, false);
      table.columns(hide).visible(false);
    }

    methods.showCols = function(show, reset) {
      if (reset) table.columns().visible(false, false);
      table.columns(show).visible(true);
    }

    methods.colReorder = function(order, origOrder) {
      table.colReorder.order(order, origOrder);
    }

    methods.selectPage = function(page) {
      if (table.page.info().pages < page || page < 1) {
        throw 'Selected page is out of range';
      };
      table.page(page - 1).draw(false);
    }

    methods.reloadData = function(resetPaging, clearSelection) {
      // empty selections first if necessary
      if (methods.selectRows && inArray('row', clearSelection)) methods.selectRows([]);
      if (methods.selectColumns && inArray('column', clearSelection)) methods.selectColumns([]);
      if (methods.selectCells && inArray('cell', clearSelection)) methods.selectCells([]);
      table.ajax.reload(null, resetPaging);
    }

    table.shinyMethods = methods;
  },
  resize: function(el, width, height, instance) {
    if (instance.data) this.renderValue(el, instance.data, instance);

    // dynamically adjust height if fillContainer = TRUE
    if (instance.fillContainer)
      this.fillAvailableHeight(el, height);

    this.adjustWidth(el);
  },

  // dynamically set the scroll body to fill available height
  // (used with fillContainer = TRUE)
  fillAvailableHeight: function(el, availableHeight) {

    // see how much of the table is occupied by header/footer elements
    // and use that to compute a target scroll body height
    var dtWrapper = $(el).find('div.dataTables_wrapper');
    var dtScrollBody = $(el).find($('div.dataTables_scrollBody'));
    var framingHeight = dtWrapper.innerHeight() - dtScrollBody.innerHeight();
    var scrollBodyHeight = availableHeight - framingHeight;

    // set the height
    dtScrollBody.height(scrollBodyHeight + 'px');
  },

  // adjust the width of columns; remove the hard-coded widths on table and the
  // scroll header when scrollX/Y are enabled
  adjustWidth: function(el) {
    var $el = $(el), table = $el.data('datatable');
    if (table) table.columns.adjust();
    $el.find('.dataTables_scrollHeadInner').css('width', '')
        .children('table').css('margin-left', '');
  }
});

  if (!HTMLWidgets.shinyMode) return;

  Shiny.addCustomMessageHandler('datatable-calls', function(data) {
    var id = data.id;
    var el = document.getElementById(id);
    var table = el ? $(el).data('datatable') : null;
    if (!table) {
      console.log("Couldn't find table with id " + id);
      return;
    }

    var methods = table.shinyMethods, call = data.call;
    if (methods[call.method]) {
      methods[call.method].apply(table, call.args);
    } else {
      console.log("Unknown method " + call.method);
    }
  });

})();
</script>
<style type="text/css">table.dataTable{width:100%;margin:0 auto;clear:both;border-collapse:separate;border-spacing:0}table.dataTable thead th,table.dataTable tfoot th{font-weight:bold}table.dataTable thead th,table.dataTable thead td{padding:10px 18px;border-bottom:1px solid #111}table.dataTable thead th:active,table.dataTable thead td:active{outline:none}table.dataTable tfoot th,table.dataTable tfoot td{padding:10px 18px 6px 18px;border-top:1px solid #111}table.dataTable thead .sorting,table.dataTable thead .sorting_asc,table.dataTable thead .sorting_desc,table.dataTable thead .sorting_asc_disabled,table.dataTable thead .sorting_desc_disabled{cursor:pointer;*cursor:hand;background-repeat:no-repeat;background-position:center right}table.dataTable thead .sorting{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABMAAAATCAQAAADYWf5HAAAAkElEQVQoz7XQMQ5AQBCF4dWQSJxC5wwax1Cq1e7BAdxD5SL+Tq/QCM1oNiJidwox0355mXnG/DrEtIQ6azioNZQxI0ykPhTQIwhCR+BmBYtlK7kLJYwWCcJA9M4qdrZrd8pPjZWPtOqdRQy320YSV17OatFC4euts6z39GYMKRPCTKY9UnPQ6P+GtMRfGtPnBCiqhAeJPmkqAAAAAElFTkSuQmCC)}table.dataTable thead .sorting_asc{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABMAAAATCAYAAAByUDbMAAAAZ0lEQVQ4y2NgGLKgquEuFxBPAGI2ahhWCsS/gDibUoO0gPgxEP8H4ttArEyuQYxAPBdqEAxPBImTY5gjEL9DM+wTENuQahAvEO9DMwiGdwAxOymGJQLxTyD+jgWDxCMZRsEoGAVoAADeemwtPcZI2wAAAABJRU5ErkJggg==)}table.dataTable thead .sorting_desc{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABMAAAATCAYAAAByUDbMAAAAZUlEQVQ4y2NgGAWjYBSggaqGu5FA/BOIv2PBIPFEUgxjB+IdQPwfC94HxLykus4GiD+hGfQOiB3J8SojEE9EM2wuSJzcsFMG4ttQgx4DsRalkZENxL+AuJQaMcsGxBOAmGvopk8AVz1sLZgg0bsAAAAASUVORK5CYII=)}table.dataTable thead .sorting_asc_disabled{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABMAAAATCAQAAADYWf5HAAAAW0lEQVQoz2NgoCm4w3Vnwh02wspK7/y6k01Ikdadx3f+37l9RxmfIsY7c4GKQHDiHUbcyhzvvIMq+3THBpci3jv7oIpAcMcdduzKEu/8vPMdDn/eiWQYBYMKAAC3ykIEuYQJUgAAAABJRU5ErkJggg==)}table.dataTable thead .sorting_desc_disabled{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABMAAAATCAQAAADYWf5HAAAAWUlEQVQoz2NgGAWDCtyJvPPzznc4/HknEbsy9js77vyHw313eHGZZ3PnE1TRuzuOuK1lvDMRqmzuHUZ87lO+cxuo6PEdLUIeyb7z604pYf+y3Zlwh4u2YQoAc7ZCBHH4jigAAAAASUVORK5CYII=)}table.dataTable tbody tr{background-color:#ffffff}table.dataTable tbody tr.selected{background-color:#B0BED9}table.dataTable tbody th,table.dataTable tbody td{padding:8px 10px}table.dataTable.row-border tbody th,table.dataTable.row-border tbody td,table.dataTable.display tbody th,table.dataTable.display tbody td{border-top:1px solid #ddd}table.dataTable.row-border tbody tr:first-child th,table.dataTable.row-border tbody tr:first-child td,table.dataTable.display tbody tr:first-child th,table.dataTable.display tbody tr:first-child td{border-top:none}table.dataTable.cell-border tbody th,table.dataTable.cell-border tbody td{border-top:1px solid #ddd;border-right:1px solid #ddd}table.dataTable.cell-border tbody tr th:first-child,table.dataTable.cell-border tbody tr td:first-child{border-left:1px solid #ddd}table.dataTable.cell-border tbody tr:first-child th,table.dataTable.cell-border tbody tr:first-child td{border-top:none}table.dataTable.stripe tbody tr.odd,table.dataTable.display tbody tr.odd{background-color:#f9f9f9}table.dataTable.stripe tbody tr.odd.selected,table.dataTable.display tbody tr.odd.selected{background-color:#acbad4}table.dataTable.hover tbody tr:hover,table.dataTable.display tbody tr:hover{background-color:#f6f6f6}table.dataTable.hover tbody tr:hover.selected,table.dataTable.display tbody tr:hover.selected{background-color:#aab7d1}table.dataTable.order-column tbody tr>.sorting_1,table.dataTable.order-column tbody tr>.sorting_2,table.dataTable.order-column tbody tr>.sorting_3,table.dataTable.display tbody tr>.sorting_1,table.dataTable.display tbody tr>.sorting_2,table.dataTable.display tbody tr>.sorting_3{background-color:#fafafa}table.dataTable.order-column tbody tr.selected>.sorting_1,table.dataTable.order-column tbody tr.selected>.sorting_2,table.dataTable.order-column tbody tr.selected>.sorting_3,table.dataTable.display tbody tr.selected>.sorting_1,table.dataTable.display tbody tr.selected>.sorting_2,table.dataTable.display tbody tr.selected>.sorting_3{background-color:#acbad5}table.dataTable.display tbody tr.odd>.sorting_1,table.dataTable.order-column.stripe tbody tr.odd>.sorting_1{background-color:#f1f1f1}table.dataTable.display tbody tr.odd>.sorting_2,table.dataTable.order-column.stripe tbody tr.odd>.sorting_2{background-color:#f3f3f3}table.dataTable.display tbody tr.odd>.sorting_3,table.dataTable.order-column.stripe tbody tr.odd>.sorting_3{background-color:whitesmoke}table.dataTable.display tbody tr.odd.selected>.sorting_1,table.dataTable.order-column.stripe tbody tr.odd.selected>.sorting_1{background-color:#a6b4cd}table.dataTable.display tbody tr.odd.selected>.sorting_2,table.dataTable.order-column.stripe tbody tr.odd.selected>.sorting_2{background-color:#a8b5cf}table.dataTable.display tbody tr.odd.selected>.sorting_3,table.dataTable.order-column.stripe tbody tr.odd.selected>.sorting_3{background-color:#a9b7d1}table.dataTable.display tbody tr.even>.sorting_1,table.dataTable.order-column.stripe tbody tr.even>.sorting_1{background-color:#fafafa}table.dataTable.display tbody tr.even>.sorting_2,table.dataTable.order-column.stripe tbody tr.even>.sorting_2{background-color:#fcfcfc}table.dataTable.display tbody tr.even>.sorting_3,table.dataTable.order-column.stripe tbody tr.even>.sorting_3{background-color:#fefefe}table.dataTable.display tbody tr.even.selected>.sorting_1,table.dataTable.order-column.stripe tbody tr.even.selected>.sorting_1{background-color:#acbad5}table.dataTable.display tbody tr.even.selected>.sorting_2,table.dataTable.order-column.stripe tbody tr.even.selected>.sorting_2{background-color:#aebcd6}table.dataTable.display tbody tr.even.selected>.sorting_3,table.dataTable.order-column.stripe tbody tr.even.selected>.sorting_3{background-color:#afbdd8}table.dataTable.display tbody tr:hover>.sorting_1,table.dataTable.order-column.hover tbody tr:hover>.sorting_1{background-color:#eaeaea}table.dataTable.display tbody tr:hover>.sorting_2,table.dataTable.order-column.hover tbody tr:hover>.sorting_2{background-color:#ececec}table.dataTable.display tbody tr:hover>.sorting_3,table.dataTable.order-column.hover tbody tr:hover>.sorting_3{background-color:#efefef}table.dataTable.display tbody tr:hover.selected>.sorting_1,table.dataTable.order-column.hover tbody tr:hover.selected>.sorting_1{background-color:#a2aec7}table.dataTable.display tbody tr:hover.selected>.sorting_2,table.dataTable.order-column.hover tbody tr:hover.selected>.sorting_2{background-color:#a3b0c9}table.dataTable.display tbody tr:hover.selected>.sorting_3,table.dataTable.order-column.hover tbody tr:hover.selected>.sorting_3{background-color:#a5b2cb}table.dataTable.no-footer{border-bottom:1px solid #111}table.dataTable.nowrap th,table.dataTable.nowrap td{white-space:nowrap}table.dataTable.compact thead th,table.dataTable.compact thead td{padding:4px 17px 4px 4px}table.dataTable.compact tfoot th,table.dataTable.compact tfoot td{padding:4px}table.dataTable.compact tbody th,table.dataTable.compact tbody td{padding:4px}table.dataTable th.dt-left,table.dataTable td.dt-left{text-align:left}table.dataTable th.dt-center,table.dataTable td.dt-center,table.dataTable td.dataTables_empty{text-align:center}table.dataTable th.dt-right,table.dataTable td.dt-right{text-align:right}table.dataTable th.dt-justify,table.dataTable td.dt-justify{text-align:justify}table.dataTable th.dt-nowrap,table.dataTable td.dt-nowrap{white-space:nowrap}table.dataTable thead th.dt-head-left,table.dataTable thead td.dt-head-left,table.dataTable tfoot th.dt-head-left,table.dataTable tfoot td.dt-head-left{text-align:left}table.dataTable thead th.dt-head-center,table.dataTable thead td.dt-head-center,table.dataTable tfoot th.dt-head-center,table.dataTable tfoot td.dt-head-center{text-align:center}table.dataTable thead th.dt-head-right,table.dataTable thead td.dt-head-right,table.dataTable tfoot th.dt-head-right,table.dataTable tfoot td.dt-head-right{text-align:right}table.dataTable thead th.dt-head-justify,table.dataTable thead td.dt-head-justify,table.dataTable tfoot th.dt-head-justify,table.dataTable tfoot td.dt-head-justify{text-align:justify}table.dataTable thead th.dt-head-nowrap,table.dataTable thead td.dt-head-nowrap,table.dataTable tfoot th.dt-head-nowrap,table.dataTable tfoot td.dt-head-nowrap{white-space:nowrap}table.dataTable tbody th.dt-body-left,table.dataTable tbody td.dt-body-left{text-align:left}table.dataTable tbody th.dt-body-center,table.dataTable tbody td.dt-body-center{text-align:center}table.dataTable tbody th.dt-body-right,table.dataTable tbody td.dt-body-right{text-align:right}table.dataTable tbody th.dt-body-justify,table.dataTable tbody td.dt-body-justify{text-align:justify}table.dataTable tbody th.dt-body-nowrap,table.dataTable tbody td.dt-body-nowrap{white-space:nowrap}table.dataTable,table.dataTable th,table.dataTable td{box-sizing:content-box}.dataTables_wrapper{position:relative;clear:both;*zoom:1;zoom:1}.dataTables_wrapper .dataTables_length{float:left}.dataTables_wrapper .dataTables_filter{float:right;text-align:right}.dataTables_wrapper .dataTables_filter input{margin-left:0.5em}.dataTables_wrapper .dataTables_info{clear:both;float:left;padding-top:0.755em}.dataTables_wrapper .dataTables_paginate{float:right;text-align:right;padding-top:0.25em}.dataTables_wrapper .dataTables_paginate .paginate_button{box-sizing:border-box;display:inline-block;min-width:1.5em;padding:0.5em 1em;margin-left:2px;text-align:center;text-decoration:none !important;cursor:pointer;*cursor:hand;color:#333 !important;border:1px solid transparent;border-radius:2px}.dataTables_wrapper .dataTables_paginate .paginate_button.current,.dataTables_wrapper .dataTables_paginate .paginate_button.current:hover{color:#333 !important;border:1px solid #979797;background-color:white;background:-webkit-gradient(linear, left top, left bottom, color-stop(0%, #fff), color-stop(100%, #dcdcdc));background:-webkit-linear-gradient(top, #fff 0%, #dcdcdc 100%);background:-moz-linear-gradient(top, #fff 0%, #dcdcdc 100%);background:-ms-linear-gradient(top, #fff 0%, #dcdcdc 100%);background:-o-linear-gradient(top, #fff 0%, #dcdcdc 100%);background:linear-gradient(to bottom, #fff 0%, #dcdcdc 100%)}.dataTables_wrapper .dataTables_paginate .paginate_button.disabled,.dataTables_wrapper .dataTables_paginate .paginate_button.disabled:hover,.dataTables_wrapper .dataTables_paginate .paginate_button.disabled:active{cursor:default;color:#666 !important;border:1px solid transparent;background:transparent;box-shadow:none}.dataTables_wrapper .dataTables_paginate .paginate_button:hover{color:white !important;border:1px solid #111;background-color:#585858;background:-webkit-gradient(linear, left top, left bottom, color-stop(0%, #585858), color-stop(100%, #111));background:-webkit-linear-gradient(top, #585858 0%, #111 100%);background:-moz-linear-gradient(top, #585858 0%, #111 100%);background:-ms-linear-gradient(top, #585858 0%, #111 100%);background:-o-linear-gradient(top, #585858 0%, #111 100%);background:linear-gradient(to bottom, #585858 0%, #111 100%)}.dataTables_wrapper .dataTables_paginate .paginate_button:active{outline:none;background-color:#2b2b2b;background:-webkit-gradient(linear, left top, left bottom, color-stop(0%, #2b2b2b), color-stop(100%, #0c0c0c));background:-webkit-linear-gradient(top, #2b2b2b 0%, #0c0c0c 100%);background:-moz-linear-gradient(top, #2b2b2b 0%, #0c0c0c 100%);background:-ms-linear-gradient(top, #2b2b2b 0%, #0c0c0c 100%);background:-o-linear-gradient(top, #2b2b2b 0%, #0c0c0c 100%);background:linear-gradient(to bottom, #2b2b2b 0%, #0c0c0c 100%);box-shadow:inset 0 0 3px #111}.dataTables_wrapper .dataTables_paginate .ellipsis{padding:0 1em}.dataTables_wrapper .dataTables_processing{position:absolute;top:50%;left:50%;width:100%;height:40px;margin-left:-50%;margin-top:-25px;padding-top:20px;text-align:center;font-size:1.2em;background-color:white;background:-webkit-gradient(linear, left top, right top, color-stop(0%, rgba(255,255,255,0)), color-stop(25%, rgba(255,255,255,0.9)), color-stop(75%, rgba(255,255,255,0.9)), color-stop(100%, rgba(255,255,255,0)));background:-webkit-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);background:-moz-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);background:-ms-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);background:-o-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);background:linear-gradient(to right, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%)}.dataTables_wrapper .dataTables_length,.dataTables_wrapper .dataTables_filter,.dataTables_wrapper .dataTables_info,.dataTables_wrapper .dataTables_processing,.dataTables_wrapper .dataTables_paginate{color:#333}.dataTables_wrapper .dataTables_scroll{clear:both}.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody{*margin-top:-1px;-webkit-overflow-scrolling:touch}.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>thead>tr>th,.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>thead>tr>td,.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>tbody>tr>th,.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>tbody>tr>td{vertical-align:middle}.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>thead>tr>th>div.dataTables_sizing,.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>thead>tr>td>div.dataTables_sizing,.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>tbody>tr>th>div.dataTables_sizing,.dataTables_wrapper .dataTables_scroll div.dataTables_scrollBody>table>tbody>tr>td>div.dataTables_sizing{height:0;overflow:hidden;margin:0 !important;padding:0 !important}.dataTables_wrapper.no-footer .dataTables_scrollBody{border-bottom:1px solid #111}.dataTables_wrapper.no-footer div.dataTables_scrollHead table.dataTable,.dataTables_wrapper.no-footer div.dataTables_scrollBody>table{border-bottom:none}.dataTables_wrapper:after{visibility:hidden;display:block;content:"";clear:both;height:0}@media screen and (max-width: 767px){.dataTables_wrapper .dataTables_info,.dataTables_wrapper .dataTables_paginate{float:none;text-align:center}.dataTables_wrapper .dataTables_paginate{margin-top:0.5em}}@media screen and (max-width: 640px){.dataTables_wrapper .dataTables_length,.dataTables_wrapper .dataTables_filter{float:none;text-align:center}.dataTables_wrapper .dataTables_filter{margin-top:0.5em}}
</style>
<style type="text/css">
table.dataTable tr.selected td, table.dataTable td.selected {
background-color: #b0bed9 !important;
}

.dataTables_scrollBody .dataTables_sizing {
visibility: hidden;
}

div.datatables {
color: #333;
}
</style>
<script>/*!
   Copyright 2008-2019 SpryMedia Ltd.

 This source file is free software, available under the following license:
   MIT license - http://datatables.net/license

 This source file is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE. See the license files for details.

 For details please refer to: http://www.datatables.net
 DataTables 1.10.20
 ©2008-2019 SpryMedia Ltd - datatables.net/license
*/
var $jscomp=$jscomp||{};$jscomp.scope={};$jscomp.findInternal=function(f,z,y){f instanceof String&&(f=String(f));for(var p=f.length,H=0;H<p;H++){var L=f[H];if(z.call(y,L,H,f))return{i:H,v:L}}return{i:-1,v:void 0}};$jscomp.ASSUME_ES5=!1;$jscomp.ASSUME_NO_NATIVE_MAP=!1;$jscomp.ASSUME_NO_NATIVE_SET=!1;$jscomp.SIMPLE_FROUND_POLYFILL=!1;
$jscomp.defineProperty=$jscomp.ASSUME_ES5||"function"==typeof Object.defineProperties?Object.defineProperty:function(f,z,y){f!=Array.prototype&&f!=Object.prototype&&(f[z]=y.value)};$jscomp.getGlobal=function(f){return"undefined"!=typeof window&&window===f?f:"undefined"!=typeof global&&null!=global?global:f};$jscomp.global=$jscomp.getGlobal(this);
$jscomp.polyfill=function(f,z,y,p){if(z){y=$jscomp.global;f=f.split(".");for(p=0;p<f.length-1;p++){var H=f[p];H in y||(y[H]={});y=y[H]}f=f[f.length-1];p=y[f];z=z(p);z!=p&&null!=z&&$jscomp.defineProperty(y,f,{configurable:!0,writable:!0,value:z})}};$jscomp.polyfill("Array.prototype.find",function(f){return f?f:function(f,y){return $jscomp.findInternal(this,f,y).v}},"es6","es3");
(function(f){"function"===typeof define&&define.amd?define(["jquery"],function(z){return f(z,window,document)}):"object"===typeof exports?module.exports=function(z,y){z||(z=window);y||(y="undefined"!==typeof window?require("jquery"):require("jquery")(z));return f(y,z,z.document)}:f(jQuery,window,document)})(function(f,z,y,p){function H(a){var b,c,d={};f.each(a,function(e,h){(b=e.match(/^([^A-Z]+?)([A-Z])/))&&-1!=="a aa ai ao as b fn i m o s ".indexOf(b[1]+" ")&&(c=e.replace(b[0],b[2].toLowerCase()),
d[c]=e,"o"===b[1]&&H(a[e]))});a._hungarianMap=d}function L(a,b,c){a._hungarianMap||H(a);var d;f.each(b,function(e,h){d=a._hungarianMap[e];d===p||!c&&b[d]!==p||("o"===d.charAt(0)?(b[d]||(b[d]={}),f.extend(!0,b[d],b[e]),L(a[d],b[d],c)):b[d]=b[e])})}function Ga(a){var b=q.defaults.oLanguage,c=b.sDecimal;c&&Ha(c);if(a){var d=a.sZeroRecords;!a.sEmptyTable&&d&&"No data available in table"===b.sEmptyTable&&M(a,a,"sZeroRecords","sEmptyTable");!a.sLoadingRecords&&d&&"Loading..."===b.sLoadingRecords&&M(a,a,
"sZeroRecords","sLoadingRecords");a.sInfoThousands&&(a.sThousands=a.sInfoThousands);(a=a.sDecimal)&&c!==a&&Ha(a)}}function jb(a){F(a,"ordering","bSort");F(a,"orderMulti","bSortMulti");F(a,"orderClasses","bSortClasses");F(a,"orderCellsTop","bSortCellsTop");F(a,"order","aaSorting");F(a,"orderFixed","aaSortingFixed");F(a,"paging","bPaginate");F(a,"pagingType","sPaginationType");F(a,"pageLength","iDisplayLength");F(a,"searching","bFilter");"boolean"===typeof a.sScrollX&&(a.sScrollX=a.sScrollX?"100%":
"");"boolean"===typeof a.scrollX&&(a.scrollX=a.scrollX?"100%":"");if(a=a.aoSearchCols)for(var b=0,c=a.length;b<c;b++)a[b]&&L(q.models.oSearch,a[b])}function kb(a){F(a,"orderable","bSortable");F(a,"orderData","aDataSort");F(a,"orderSequence","asSorting");F(a,"orderDataType","sortDataType");var b=a.aDataSort;"number"!==typeof b||f.isArray(b)||(a.aDataSort=[b])}function lb(a){if(!q.__browser){var b={};q.__browser=b;var c=f("<div/>").css({position:"fixed",top:0,left:-1*f(z).scrollLeft(),height:1,width:1,
overflow:"hidden"}).append(f("<div/>").css({position:"absolute",top:1,left:1,width:100,overflow:"scroll"}).append(f("<div/>").css({width:"100%",height:10}))).appendTo("body"),d=c.children(),e=d.children();b.barWidth=d[0].offsetWidth-d[0].clientWidth;b.bScrollOversize=100===e[0].offsetWidth&&100!==d[0].clientWidth;b.bScrollbarLeft=1!==Math.round(e.offset().left);b.bBounding=c[0].getBoundingClientRect().width?!0:!1;c.remove()}f.extend(a.oBrowser,q.__browser);a.oScroll.iBarWidth=q.__browser.barWidth}
function mb(a,b,c,d,e,h){var g=!1;if(c!==p){var k=c;g=!0}for(;d!==e;)a.hasOwnProperty(d)&&(k=g?b(k,a[d],d,a):a[d],g=!0,d+=h);return k}function Ia(a,b){var c=q.defaults.column,d=a.aoColumns.length;c=f.extend({},q.models.oColumn,c,{nTh:b?b:y.createElement("th"),sTitle:c.sTitle?c.sTitle:b?b.innerHTML:"",aDataSort:c.aDataSort?c.aDataSort:[d],mData:c.mData?c.mData:d,idx:d});a.aoColumns.push(c);c=a.aoPreSearchCols;c[d]=f.extend({},q.models.oSearch,c[d]);ma(a,d,f(b).data())}function ma(a,b,c){b=a.aoColumns[b];
var d=a.oClasses,e=f(b.nTh);if(!b.sWidthOrig){b.sWidthOrig=e.attr("width")||null;var h=(e.attr("style")||"").match(/width:\s*(\d+[pxem%]+)/);h&&(b.sWidthOrig=h[1])}c!==p&&null!==c&&(kb(c),L(q.defaults.column,c,!0),c.mDataProp===p||c.mData||(c.mData=c.mDataProp),c.sType&&(b._sManualType=c.sType),c.className&&!c.sClass&&(c.sClass=c.className),c.sClass&&e.addClass(c.sClass),f.extend(b,c),M(b,c,"sWidth","sWidthOrig"),c.iDataSort!==p&&(b.aDataSort=[c.iDataSort]),M(b,c,"aDataSort"));var g=b.mData,k=U(g),
l=b.mRender?U(b.mRender):null;c=function(a){return"string"===typeof a&&-1!==a.indexOf("@")};b._bAttrSrc=f.isPlainObject(g)&&(c(g.sort)||c(g.type)||c(g.filter));b._setter=null;b.fnGetData=function(a,b,c){var d=k(a,b,p,c);return l&&b?l(d,b,a,c):d};b.fnSetData=function(a,b,c){return Q(g)(a,b,c)};"number"!==typeof g&&(a._rowReadObject=!0);a.oFeatures.bSort||(b.bSortable=!1,e.addClass(d.sSortableNone));a=-1!==f.inArray("asc",b.asSorting);c=-1!==f.inArray("desc",b.asSorting);b.bSortable&&(a||c)?a&&!c?(b.sSortingClass=
d.sSortableAsc,b.sSortingClassJUI=d.sSortJUIAscAllowed):!a&&c?(b.sSortingClass=d.sSortableDesc,b.sSortingClassJUI=d.sSortJUIDescAllowed):(b.sSortingClass=d.sSortable,b.sSortingClassJUI=d.sSortJUI):(b.sSortingClass=d.sSortableNone,b.sSortingClassJUI="")}function aa(a){if(!1!==a.oFeatures.bAutoWidth){var b=a.aoColumns;Ja(a);for(var c=0,d=b.length;c<d;c++)b[c].nTh.style.width=b[c].sWidth}b=a.oScroll;""===b.sY&&""===b.sX||na(a);A(a,null,"column-sizing",[a])}function ba(a,b){a=oa(a,"bVisible");return"number"===
typeof a[b]?a[b]:null}function ca(a,b){a=oa(a,"bVisible");b=f.inArray(b,a);return-1!==b?b:null}function W(a){var b=0;f.each(a.aoColumns,function(a,d){d.bVisible&&"none"!==f(d.nTh).css("display")&&b++});return b}function oa(a,b){var c=[];f.map(a.aoColumns,function(a,e){a[b]&&c.push(e)});return c}function Ka(a){var b=a.aoColumns,c=a.aoData,d=q.ext.type.detect,e,h,g;var k=0;for(e=b.length;k<e;k++){var f=b[k];var n=[];if(!f.sType&&f._sManualType)f.sType=f._sManualType;else if(!f.sType){var m=0;for(h=
d.length;m<h;m++){var w=0;for(g=c.length;w<g;w++){n[w]===p&&(n[w]=I(a,w,k,"type"));var u=d[m](n[w],a);if(!u&&m!==d.length-1)break;if("html"===u)break}if(u){f.sType=u;break}}f.sType||(f.sType="string")}}}function nb(a,b,c,d){var e,h,g,k=a.aoColumns;if(b)for(e=b.length-1;0<=e;e--){var l=b[e];var n=l.targets!==p?l.targets:l.aTargets;f.isArray(n)||(n=[n]);var m=0;for(h=n.length;m<h;m++)if("number"===typeof n[m]&&0<=n[m]){for(;k.length<=n[m];)Ia(a);d(n[m],l)}else if("number"===typeof n[m]&&0>n[m])d(k.length+
n[m],l);else if("string"===typeof n[m]){var w=0;for(g=k.length;w<g;w++)("_all"==n[m]||f(k[w].nTh).hasClass(n[m]))&&d(w,l)}}if(c)for(e=0,a=c.length;e<a;e++)d(e,c[e])}function R(a,b,c,d){var e=a.aoData.length,h=f.extend(!0,{},q.models.oRow,{src:c?"dom":"data",idx:e});h._aData=b;a.aoData.push(h);for(var g=a.aoColumns,k=0,l=g.length;k<l;k++)g[k].sType=null;a.aiDisplayMaster.push(e);b=a.rowIdFn(b);b!==p&&(a.aIds[b]=h);!c&&a.oFeatures.bDeferRender||La(a,e,c,d);return e}function pa(a,b){var c;b instanceof
f||(b=f(b));return b.map(function(b,e){c=Ma(a,e);return R(a,c.data,e,c.cells)})}function I(a,b,c,d){var e=a.iDraw,h=a.aoColumns[c],g=a.aoData[b]._aData,k=h.sDefaultContent,f=h.fnGetData(g,d,{settings:a,row:b,col:c});if(f===p)return a.iDrawError!=e&&null===k&&(O(a,0,"Requested unknown parameter "+("function"==typeof h.mData?"{function}":"'"+h.mData+"'")+" for row "+b+", column "+c,4),a.iDrawError=e),k;if((f===g||null===f)&&null!==k&&d!==p)f=k;else if("function"===typeof f)return f.call(g);return null===
f&&"display"==d?"":f}function ob(a,b,c,d){a.aoColumns[c].fnSetData(a.aoData[b]._aData,d,{settings:a,row:b,col:c})}function Na(a){return f.map(a.match(/(\\.|[^\.])+/g)||[""],function(a){return a.replace(/\\\./g,".")})}function U(a){if(f.isPlainObject(a)){var b={};f.each(a,function(a,c){c&&(b[a]=U(c))});return function(a,c,h,g){var d=b[c]||b._;return d!==p?d(a,c,h,g):a}}if(null===a)return function(a){return a};if("function"===typeof a)return function(b,c,h,g){return a(b,c,h,g)};if("string"!==typeof a||
-1===a.indexOf(".")&&-1===a.indexOf("[")&&-1===a.indexOf("("))return function(b,c){return b[a]};var c=function(a,b,h){if(""!==h){var d=Na(h);for(var e=0,l=d.length;e<l;e++){h=d[e].match(da);var n=d[e].match(X);if(h){d[e]=d[e].replace(da,"");""!==d[e]&&(a=a[d[e]]);n=[];d.splice(0,e+1);d=d.join(".");if(f.isArray(a))for(e=0,l=a.length;e<l;e++)n.push(c(a[e],b,d));a=h[0].substring(1,h[0].length-1);a=""===a?n:n.join(a);break}else if(n){d[e]=d[e].replace(X,"");a=a[d[e]]();continue}if(null===a||a[d[e]]===
p)return p;a=a[d[e]]}}return a};return function(b,e){return c(b,e,a)}}function Q(a){if(f.isPlainObject(a))return Q(a._);if(null===a)return function(){};if("function"===typeof a)return function(b,d,e){a(b,"set",d,e)};if("string"!==typeof a||-1===a.indexOf(".")&&-1===a.indexOf("[")&&-1===a.indexOf("("))return function(b,d){b[a]=d};var b=function(a,d,e){e=Na(e);var c=e[e.length-1];for(var g,k,l=0,n=e.length-1;l<n;l++){g=e[l].match(da);k=e[l].match(X);if(g){e[l]=e[l].replace(da,"");a[e[l]]=[];c=e.slice();
c.splice(0,l+1);g=c.join(".");if(f.isArray(d))for(k=0,n=d.length;k<n;k++)c={},b(c,d[k],g),a[e[l]].push(c);else a[e[l]]=d;return}k&&(e[l]=e[l].replace(X,""),a=a[e[l]](d));if(null===a[e[l]]||a[e[l]]===p)a[e[l]]={};a=a[e[l]]}if(c.match(X))a[c.replace(X,"")](d);else a[c.replace(da,"")]=d};return function(c,d){return b(c,d,a)}}function Oa(a){return J(a.aoData,"_aData")}function qa(a){a.aoData.length=0;a.aiDisplayMaster.length=0;a.aiDisplay.length=0;a.aIds={}}function ra(a,b,c){for(var d=-1,e=0,h=a.length;e<
h;e++)a[e]==b?d=e:a[e]>b&&a[e]--; -1!=d&&c===p&&a.splice(d,1)}function ea(a,b,c,d){var e=a.aoData[b],h,g=function(c,d){for(;c.childNodes.length;)c.removeChild(c.firstChild);c.innerHTML=I(a,b,d,"display")};if("dom"!==c&&(c&&"auto"!==c||"dom"!==e.src)){var k=e.anCells;if(k)if(d!==p)g(k[d],d);else for(c=0,h=k.length;c<h;c++)g(k[c],c)}else e._aData=Ma(a,e,d,d===p?p:e._aData).data;e._aSortData=null;e._aFilterData=null;g=a.aoColumns;if(d!==p)g[d].sType=null;else{c=0;for(h=g.length;c<h;c++)g[c].sType=null;
Pa(a,e)}}function Ma(a,b,c,d){var e=[],h=b.firstChild,g,k=0,l,n=a.aoColumns,m=a._rowReadObject;d=d!==p?d:m?{}:[];var w=function(a,b){if("string"===typeof a){var c=a.indexOf("@");-1!==c&&(c=a.substring(c+1),Q(a)(d,b.getAttribute(c)))}},u=function(a){if(c===p||c===k)g=n[k],l=f.trim(a.innerHTML),g&&g._bAttrSrc?(Q(g.mData._)(d,l),w(g.mData.sort,a),w(g.mData.type,a),w(g.mData.filter,a)):m?(g._setter||(g._setter=Q(g.mData)),g._setter(d,l)):d[k]=l;k++};if(h)for(;h;){var q=h.nodeName.toUpperCase();if("TD"==
q||"TH"==q)u(h),e.push(h);h=h.nextSibling}else for(e=b.anCells,h=0,q=e.length;h<q;h++)u(e[h]);(b=b.firstChild?b:b.nTr)&&(b=b.getAttribute("id"))&&Q(a.rowId)(d,b);return{data:d,cells:e}}function La(a,b,c,d){var e=a.aoData[b],h=e._aData,g=[],k,l;if(null===e.nTr){var n=c||y.createElement("tr");e.nTr=n;e.anCells=g;n._DT_RowIndex=b;Pa(a,e);var m=0;for(k=a.aoColumns.length;m<k;m++){var w=a.aoColumns[m];var p=(l=c?!1:!0)?y.createElement(w.sCellType):d[m];p._DT_CellIndex={row:b,column:m};g.push(p);if(l||
!(c&&!w.mRender&&w.mData===m||f.isPlainObject(w.mData)&&w.mData._===m+".display"))p.innerHTML=I(a,b,m,"display");w.sClass&&(p.className+=" "+w.sClass);w.bVisible&&!c?n.appendChild(p):!w.bVisible&&c&&p.parentNode.removeChild(p);w.fnCreatedCell&&w.fnCreatedCell.call(a.oInstance,p,I(a,b,m),h,b,m)}A(a,"aoRowCreatedCallback",null,[n,h,b,g])}e.nTr.setAttribute("role","row")}function Pa(a,b){var c=b.nTr,d=b._aData;if(c){if(a=a.rowIdFn(d))c.id=a;d.DT_RowClass&&(a=d.DT_RowClass.split(" "),b.__rowc=b.__rowc?
ta(b.__rowc.concat(a)):a,f(c).removeClass(b.__rowc.join(" ")).addClass(d.DT_RowClass));d.DT_RowAttr&&f(c).attr(d.DT_RowAttr);d.DT_RowData&&f(c).data(d.DT_RowData)}}function pb(a){var b,c,d=a.nTHead,e=a.nTFoot,h=0===f("th, td",d).length,g=a.oClasses,k=a.aoColumns;h&&(c=f("<tr/>").appendTo(d));var l=0;for(b=k.length;l<b;l++){var n=k[l];var m=f(n.nTh).addClass(n.sClass);h&&m.appendTo(c);a.oFeatures.bSort&&(m.addClass(n.sSortingClass),!1!==n.bSortable&&(m.attr("tabindex",a.iTabIndex).attr("aria-controls",
a.sTableId),Qa(a,n.nTh,l)));n.sTitle!=m[0].innerHTML&&m.html(n.sTitle);Ra(a,"header")(a,m,n,g)}h&&fa(a.aoHeader,d);f(d).find(">tr").attr("role","row");f(d).find(">tr>th, >tr>td").addClass(g.sHeaderTH);f(e).find(">tr>th, >tr>td").addClass(g.sFooterTH);if(null!==e)for(a=a.aoFooter[0],l=0,b=a.length;l<b;l++)n=k[l],n.nTf=a[l].cell,n.sClass&&f(n.nTf).addClass(n.sClass)}function ha(a,b,c){var d,e,h=[],g=[],k=a.aoColumns.length;if(b){c===p&&(c=!1);var l=0;for(d=b.length;l<d;l++){h[l]=b[l].slice();h[l].nTr=
b[l].nTr;for(e=k-1;0<=e;e--)a.aoColumns[e].bVisible||c||h[l].splice(e,1);g.push([])}l=0;for(d=h.length;l<d;l++){if(a=h[l].nTr)for(;e=a.firstChild;)a.removeChild(e);e=0;for(b=h[l].length;e<b;e++){var n=k=1;if(g[l][e]===p){a.appendChild(h[l][e].cell);for(g[l][e]=1;h[l+k]!==p&&h[l][e].cell==h[l+k][e].cell;)g[l+k][e]=1,k++;for(;h[l][e+n]!==p&&h[l][e].cell==h[l][e+n].cell;){for(c=0;c<k;c++)g[l+c][e+n]=1;n++}f(h[l][e].cell).attr("rowspan",k).attr("colspan",n)}}}}}function S(a){var b=A(a,"aoPreDrawCallback",
"preDraw",[a]);if(-1!==f.inArray(!1,b))K(a,!1);else{b=[];var c=0,d=a.asStripeClasses,e=d.length,h=a.oLanguage,g=a.iInitDisplayStart,k="ssp"==D(a),l=a.aiDisplay;a.bDrawing=!0;g!==p&&-1!==g&&(a._iDisplayStart=k?g:g>=a.fnRecordsDisplay()?0:g,a.iInitDisplayStart=-1);g=a._iDisplayStart;var n=a.fnDisplayEnd();if(a.bDeferLoading)a.bDeferLoading=!1,a.iDraw++,K(a,!1);else if(!k)a.iDraw++;else if(!a.bDestroying&&!qb(a))return;if(0!==l.length)for(h=k?a.aoData.length:n,k=k?0:g;k<h;k++){var m=l[k],w=a.aoData[m];
null===w.nTr&&La(a,m);var u=w.nTr;if(0!==e){var q=d[c%e];w._sRowStripe!=q&&(f(u).removeClass(w._sRowStripe).addClass(q),w._sRowStripe=q)}A(a,"aoRowCallback",null,[u,w._aData,c,k,m]);b.push(u);c++}else c=h.sZeroRecords,1==a.iDraw&&"ajax"==D(a)?c=h.sLoadingRecords:h.sEmptyTable&&0===a.fnRecordsTotal()&&(c=h.sEmptyTable),b[0]=f("<tr/>",{"class":e?d[0]:""}).append(f("<td />",{valign:"top",colSpan:W(a),"class":a.oClasses.sRowEmpty}).html(c))[0];A(a,"aoHeaderCallback","header",[f(a.nTHead).children("tr")[0],
Oa(a),g,n,l]);A(a,"aoFooterCallback","footer",[f(a.nTFoot).children("tr")[0],Oa(a),g,n,l]);d=f(a.nTBody);d.children().detach();d.append(f(b));A(a,"aoDrawCallback","draw",[a]);a.bSorted=!1;a.bFiltered=!1;a.bDrawing=!1}}function V(a,b){var c=a.oFeatures,d=c.bFilter;c.bSort&&rb(a);d?ia(a,a.oPreviousSearch):a.aiDisplay=a.aiDisplayMaster.slice();!0!==b&&(a._iDisplayStart=0);a._drawHold=b;S(a);a._drawHold=!1}function sb(a){var b=a.oClasses,c=f(a.nTable);c=f("<div/>").insertBefore(c);var d=a.oFeatures,e=
f("<div/>",{id:a.sTableId+"_wrapper","class":b.sWrapper+(a.nTFoot?"":" "+b.sNoFooter)});a.nHolding=c[0];a.nTableWrapper=e[0];a.nTableReinsertBefore=a.nTable.nextSibling;for(var h=a.sDom.split(""),g,k,l,n,m,p,u=0;u<h.length;u++){g=null;k=h[u];if("<"==k){l=f("<div/>")[0];n=h[u+1];if("'"==n||'"'==n){m="";for(p=2;h[u+p]!=n;)m+=h[u+p],p++;"H"==m?m=b.sJUIHeader:"F"==m&&(m=b.sJUIFooter);-1!=m.indexOf(".")?(n=m.split("."),l.id=n[0].substr(1,n[0].length-1),l.className=n[1]):"#"==m.charAt(0)?l.id=m.substr(1,
m.length-1):l.className=m;u+=p}e.append(l);e=f(l)}else if(">"==k)e=e.parent();else if("l"==k&&d.bPaginate&&d.bLengthChange)g=tb(a);else if("f"==k&&d.bFilter)g=ub(a);else if("r"==k&&d.bProcessing)g=vb(a);else if("t"==k)g=wb(a);else if("i"==k&&d.bInfo)g=xb(a);else if("p"==k&&d.bPaginate)g=yb(a);else if(0!==q.ext.feature.length)for(l=q.ext.feature,p=0,n=l.length;p<n;p++)if(k==l[p].cFeature){g=l[p].fnInit(a);break}g&&(l=a.aanFeatures,l[k]||(l[k]=[]),l[k].push(g),e.append(g))}c.replaceWith(e);a.nHolding=
null}function fa(a,b){b=f(b).children("tr");var c,d,e;a.splice(0,a.length);var h=0;for(e=b.length;h<e;h++)a.push([]);h=0;for(e=b.length;h<e;h++){var g=b[h];for(c=g.firstChild;c;){if("TD"==c.nodeName.toUpperCase()||"TH"==c.nodeName.toUpperCase()){var k=1*c.getAttribute("colspan");var l=1*c.getAttribute("rowspan");k=k&&0!==k&&1!==k?k:1;l=l&&0!==l&&1!==l?l:1;var n=0;for(d=a[h];d[n];)n++;var m=n;var p=1===k?!0:!1;for(d=0;d<k;d++)for(n=0;n<l;n++)a[h+n][m+d]={cell:c,unique:p},a[h+n].nTr=g}c=c.nextSibling}}}
function ua(a,b,c){var d=[];c||(c=a.aoHeader,b&&(c=[],fa(c,b)));b=0;for(var e=c.length;b<e;b++)for(var h=0,g=c[b].length;h<g;h++)!c[b][h].unique||d[h]&&a.bSortCellsTop||(d[h]=c[b][h].cell);return d}function va(a,b,c){A(a,"aoServerParams","serverParams",[b]);if(b&&f.isArray(b)){var d={},e=/(.*?)\[\]$/;f.each(b,function(a,b){(a=b.name.match(e))?(a=a[0],d[a]||(d[a]=[]),d[a].push(b.value)):d[b.name]=b.value});b=d}var h=a.ajax,g=a.oInstance,k=function(b){A(a,null,"xhr",[a,b,a.jqXHR]);c(b)};if(f.isPlainObject(h)&&
h.data){var l=h.data;var n="function"===typeof l?l(b,a):l;b="function"===typeof l&&n?n:f.extend(!0,b,n);delete h.data}n={data:b,success:function(b){var c=b.error||b.sError;c&&O(a,0,c);a.json=b;k(b)},dataType:"json",cache:!1,type:a.sServerMethod,error:function(b,c,d){d=A(a,null,"xhr",[a,null,a.jqXHR]);-1===f.inArray(!0,d)&&("parsererror"==c?O(a,0,"Invalid JSON response",1):4===b.readyState&&O(a,0,"Ajax error",7));K(a,!1)}};a.oAjaxData=b;A(a,null,"preXhr",[a,b]);a.fnServerData?a.fnServerData.call(g,
a.sAjaxSource,f.map(b,function(a,b){return{name:b,value:a}}),k,a):a.sAjaxSource||"string"===typeof h?a.jqXHR=f.ajax(f.extend(n,{url:h||a.sAjaxSource})):"function"===typeof h?a.jqXHR=h.call(g,b,k,a):(a.jqXHR=f.ajax(f.extend(n,h)),h.data=l)}function qb(a){return a.bAjaxDataGet?(a.iDraw++,K(a,!0),va(a,zb(a),function(b){Ab(a,b)}),!1):!0}function zb(a){var b=a.aoColumns,c=b.length,d=a.oFeatures,e=a.oPreviousSearch,h=a.aoPreSearchCols,g=[],k=Y(a);var l=a._iDisplayStart;var n=!1!==d.bPaginate?a._iDisplayLength:
-1;var m=function(a,b){g.push({name:a,value:b})};m("sEcho",a.iDraw);m("iColumns",c);m("sColumns",J(b,"sName").join(","));m("iDisplayStart",l);m("iDisplayLength",n);var p={draw:a.iDraw,columns:[],order:[],start:l,length:n,search:{value:e.sSearch,regex:e.bRegex}};for(l=0;l<c;l++){var u=b[l];var sa=h[l];n="function"==typeof u.mData?"function":u.mData;p.columns.push({data:n,name:u.sName,searchable:u.bSearchable,orderable:u.bSortable,search:{value:sa.sSearch,regex:sa.bRegex}});m("mDataProp_"+l,n);d.bFilter&&
(m("sSearch_"+l,sa.sSearch),m("bRegex_"+l,sa.bRegex),m("bSearchable_"+l,u.bSearchable));d.bSort&&m("bSortable_"+l,u.bSortable)}d.bFilter&&(m("sSearch",e.sSearch),m("bRegex",e.bRegex));d.bSort&&(f.each(k,function(a,b){p.order.push({column:b.col,dir:b.dir});m("iSortCol_"+a,b.col);m("sSortDir_"+a,b.dir)}),m("iSortingCols",k.length));b=q.ext.legacy.ajax;return null===b?a.sAjaxSource?g:p:b?g:p}function Ab(a,b){var c=function(a,c){return b[a]!==p?b[a]:b[c]},d=wa(a,b),e=c("sEcho","draw"),h=c("iTotalRecords",
"recordsTotal");c=c("iTotalDisplayRecords","recordsFiltered");if(e){if(1*e<a.iDraw)return;a.iDraw=1*e}qa(a);a._iRecordsTotal=parseInt(h,10);a._iRecordsDisplay=parseInt(c,10);e=0;for(h=d.length;e<h;e++)R(a,d[e]);a.aiDisplay=a.aiDisplayMaster.slice();a.bAjaxDataGet=!1;S(a);a._bInitComplete||xa(a,b);a.bAjaxDataGet=!0;K(a,!1)}function wa(a,b){a=f.isPlainObject(a.ajax)&&a.ajax.dataSrc!==p?a.ajax.dataSrc:a.sAjaxDataProp;return"data"===a?b.aaData||b[a]:""!==a?U(a)(b):b}function ub(a){var b=a.oClasses,c=
a.sTableId,d=a.oLanguage,e=a.oPreviousSearch,h=a.aanFeatures,g='<input type="search" class="'+b.sFilterInput+'"/>',k=d.sSearch;k=k.match(/_INPUT_/)?k.replace("_INPUT_",g):k+g;b=f("<div/>",{id:h.f?null:c+"_filter","class":b.sFilter}).append(f("<label/>").append(k));h=function(){var b=this.value?this.value:"";b!=e.sSearch&&(ia(a,{sSearch:b,bRegex:e.bRegex,bSmart:e.bSmart,bCaseInsensitive:e.bCaseInsensitive}),a._iDisplayStart=0,S(a))};g=null!==a.searchDelay?a.searchDelay:"ssp"===D(a)?400:0;var l=f("input",
b).val(e.sSearch).attr("placeholder",d.sSearchPlaceholder).on("keyup.DT search.DT input.DT paste.DT cut.DT",g?Sa(h,g):h).on("keypress.DT",function(a){if(13==a.keyCode)return!1}).attr("aria-controls",c);f(a.nTable).on("search.dt.DT",function(b,c){if(a===c)try{l[0]!==y.activeElement&&l.val(e.sSearch)}catch(w){}});return b[0]}function ia(a,b,c){var d=a.oPreviousSearch,e=a.aoPreSearchCols,h=function(a){d.sSearch=a.sSearch;d.bRegex=a.bRegex;d.bSmart=a.bSmart;d.bCaseInsensitive=a.bCaseInsensitive},g=function(a){return a.bEscapeRegex!==
p?!a.bEscapeRegex:a.bRegex};Ka(a);if("ssp"!=D(a)){Bb(a,b.sSearch,c,g(b),b.bSmart,b.bCaseInsensitive);h(b);for(b=0;b<e.length;b++)Cb(a,e[b].sSearch,b,g(e[b]),e[b].bSmart,e[b].bCaseInsensitive);Db(a)}else h(b);a.bFiltered=!0;A(a,null,"search",[a])}function Db(a){for(var b=q.ext.search,c=a.aiDisplay,d,e,h=0,g=b.length;h<g;h++){for(var k=[],l=0,n=c.length;l<n;l++)e=c[l],d=a.aoData[e],b[h](a,d._aFilterData,e,d._aData,l)&&k.push(e);c.length=0;f.merge(c,k)}}function Cb(a,b,c,d,e,h){if(""!==b){var g=[],k=
a.aiDisplay;d=Ta(b,d,e,h);for(e=0;e<k.length;e++)b=a.aoData[k[e]]._aFilterData[c],d.test(b)&&g.push(k[e]);a.aiDisplay=g}}function Bb(a,b,c,d,e,h){e=Ta(b,d,e,h);var g=a.oPreviousSearch.sSearch,k=a.aiDisplayMaster;h=[];0!==q.ext.search.length&&(c=!0);var f=Eb(a);if(0>=b.length)a.aiDisplay=k.slice();else{if(f||c||d||g.length>b.length||0!==b.indexOf(g)||a.bSorted)a.aiDisplay=k.slice();b=a.aiDisplay;for(c=0;c<b.length;c++)e.test(a.aoData[b[c]]._sFilterRow)&&h.push(b[c]);a.aiDisplay=h}}function Ta(a,b,
c,d){a=b?a:Ua(a);c&&(a="^(?=.*?"+f.map(a.match(/"[^"]+"|[^ ]+/g)||[""],function(a){if('"'===a.charAt(0)){var b=a.match(/^"(.*)"$/);a=b?b[1]:a}return a.replace('"',"")}).join(")(?=.*?")+").*$");return new RegExp(a,d?"i":"")}function Eb(a){var b=a.aoColumns,c,d,e=q.ext.type.search;var h=!1;var g=0;for(c=a.aoData.length;g<c;g++){var k=a.aoData[g];if(!k._aFilterData){var f=[];var n=0;for(d=b.length;n<d;n++){h=b[n];if(h.bSearchable){var m=I(a,g,n,"filter");e[h.sType]&&(m=e[h.sType](m));null===m&&(m="");
"string"!==typeof m&&m.toString&&(m=m.toString())}else m="";m.indexOf&&-1!==m.indexOf("&")&&(ya.innerHTML=m,m=$b?ya.textContent:ya.innerText);m.replace&&(m=m.replace(/[\r\n\u2028]/g,""));f.push(m)}k._aFilterData=f;k._sFilterRow=f.join("  ");h=!0}}return h}function Fb(a){return{search:a.sSearch,smart:a.bSmart,regex:a.bRegex,caseInsensitive:a.bCaseInsensitive}}function Gb(a){return{sSearch:a.search,bSmart:a.smart,bRegex:a.regex,bCaseInsensitive:a.caseInsensitive}}function xb(a){var b=a.sTableId,c=a.aanFeatures.i,
d=f("<div/>",{"class":a.oClasses.sInfo,id:c?null:b+"_info"});c||(a.aoDrawCallback.push({fn:Hb,sName:"information"}),d.attr("role","status").attr("aria-live","polite"),f(a.nTable).attr("aria-describedby",b+"_info"));return d[0]}function Hb(a){var b=a.aanFeatures.i;if(0!==b.length){var c=a.oLanguage,d=a._iDisplayStart+1,e=a.fnDisplayEnd(),h=a.fnRecordsTotal(),g=a.fnRecordsDisplay(),k=g?c.sInfo:c.sInfoEmpty;g!==h&&(k+=" "+c.sInfoFiltered);k+=c.sInfoPostFix;k=Ib(a,k);c=c.fnInfoCallback;null!==c&&(k=c.call(a.oInstance,
a,d,e,h,g,k));f(b).html(k)}}function Ib(a,b){var c=a.fnFormatNumber,d=a._iDisplayStart+1,e=a._iDisplayLength,h=a.fnRecordsDisplay(),g=-1===e;return b.replace(/_START_/g,c.call(a,d)).replace(/_END_/g,c.call(a,a.fnDisplayEnd())).replace(/_MAX_/g,c.call(a,a.fnRecordsTotal())).replace(/_TOTAL_/g,c.call(a,h)).replace(/_PAGE_/g,c.call(a,g?1:Math.ceil(d/e))).replace(/_PAGES_/g,c.call(a,g?1:Math.ceil(h/e)))}function ja(a){var b=a.iInitDisplayStart,c=a.aoColumns;var d=a.oFeatures;var e=a.bDeferLoading;if(a.bInitialised){sb(a);
pb(a);ha(a,a.aoHeader);ha(a,a.aoFooter);K(a,!0);d.bAutoWidth&&Ja(a);var h=0;for(d=c.length;h<d;h++){var g=c[h];g.sWidth&&(g.nTh.style.width=B(g.sWidth))}A(a,null,"preInit",[a]);V(a);c=D(a);if("ssp"!=c||e)"ajax"==c?va(a,[],function(c){var d=wa(a,c);for(h=0;h<d.length;h++)R(a,d[h]);a.iInitDisplayStart=b;V(a);K(a,!1);xa(a,c)},a):(K(a,!1),xa(a))}else setTimeout(function(){ja(a)},200)}function xa(a,b){a._bInitComplete=!0;(b||a.oInit.aaData)&&aa(a);A(a,null,"plugin-init",[a,b]);A(a,"aoInitComplete","init",
[a,b])}function Va(a,b){b=parseInt(b,10);a._iDisplayLength=b;Wa(a);A(a,null,"length",[a,b])}function tb(a){var b=a.oClasses,c=a.sTableId,d=a.aLengthMenu,e=f.isArray(d[0]),h=e?d[0]:d;d=e?d[1]:d;e=f("<select/>",{name:c+"_length","aria-controls":c,"class":b.sLengthSelect});for(var g=0,k=h.length;g<k;g++)e[0][g]=new Option("number"===typeof d[g]?a.fnFormatNumber(d[g]):d[g],h[g]);var l=f("<div><label/></div>").addClass(b.sLength);a.aanFeatures.l||(l[0].id=c+"_length");l.children().append(a.oLanguage.sLengthMenu.replace("_MENU_",
e[0].outerHTML));f("select",l).val(a._iDisplayLength).on("change.DT",function(b){Va(a,f(this).val());S(a)});f(a.nTable).on("length.dt.DT",function(b,c,d){a===c&&f("select",l).val(d)});return l[0]}function yb(a){var b=a.sPaginationType,c=q.ext.pager[b],d="function"===typeof c,e=function(a){S(a)};b=f("<div/>").addClass(a.oClasses.sPaging+b)[0];var h=a.aanFeatures;d||c.fnInit(a,b,e);h.p||(b.id=a.sTableId+"_paginate",a.aoDrawCallback.push({fn:function(a){if(d){var b=a._iDisplayStart,g=a._iDisplayLength,
f=a.fnRecordsDisplay(),m=-1===g;b=m?0:Math.ceil(b/g);g=m?1:Math.ceil(f/g);f=c(b,g);var p;m=0;for(p=h.p.length;m<p;m++)Ra(a,"pageButton")(a,h.p[m],m,f,b,g)}else c.fnUpdate(a,e)},sName:"pagination"}));return b}function Xa(a,b,c){var d=a._iDisplayStart,e=a._iDisplayLength,h=a.fnRecordsDisplay();0===h||-1===e?d=0:"number"===typeof b?(d=b*e,d>h&&(d=0)):"first"==b?d=0:"previous"==b?(d=0<=e?d-e:0,0>d&&(d=0)):"next"==b?d+e<h&&(d+=e):"last"==b?d=Math.floor((h-1)/e)*e:O(a,0,"Unknown paging action: "+b,5);b=
a._iDisplayStart!==d;a._iDisplayStart=d;b&&(A(a,null,"page",[a]),c&&S(a));return b}function vb(a){return f("<div/>",{id:a.aanFeatures.r?null:a.sTableId+"_processing","class":a.oClasses.sProcessing}).html(a.oLanguage.sProcessing).insertBefore(a.nTable)[0]}function K(a,b){a.oFeatures.bProcessing&&f(a.aanFeatures.r).css("display",b?"block":"none");A(a,null,"processing",[a,b])}function wb(a){var b=f(a.nTable);b.attr("role","grid");var c=a.oScroll;if(""===c.sX&&""===c.sY)return a.nTable;var d=c.sX,e=c.sY,
h=a.oClasses,g=b.children("caption"),k=g.length?g[0]._captionSide:null,l=f(b[0].cloneNode(!1)),n=f(b[0].cloneNode(!1)),m=b.children("tfoot");m.length||(m=null);l=f("<div/>",{"class":h.sScrollWrapper}).append(f("<div/>",{"class":h.sScrollHead}).css({overflow:"hidden",position:"relative",border:0,width:d?d?B(d):null:"100%"}).append(f("<div/>",{"class":h.sScrollHeadInner}).css({"box-sizing":"content-box",width:c.sXInner||"100%"}).append(l.removeAttr("id").css("margin-left",0).append("top"===k?g:null).append(b.children("thead"))))).append(f("<div/>",
{"class":h.sScrollBody}).css({position:"relative",overflow:"auto",width:d?B(d):null}).append(b));m&&l.append(f("<div/>",{"class":h.sScrollFoot}).css({overflow:"hidden",border:0,width:d?d?B(d):null:"100%"}).append(f("<div/>",{"class":h.sScrollFootInner}).append(n.removeAttr("id").css("margin-left",0).append("bottom"===k?g:null).append(b.children("tfoot")))));b=l.children();var p=b[0];h=b[1];var u=m?b[2]:null;if(d)f(h).on("scroll.DT",function(a){a=this.scrollLeft;p.scrollLeft=a;m&&(u.scrollLeft=a)});
f(h).css(e&&c.bCollapse?"max-height":"height",e);a.nScrollHead=p;a.nScrollBody=h;a.nScrollFoot=u;a.aoDrawCallback.push({fn:na,sName:"scrolling"});return l[0]}function na(a){var b=a.oScroll,c=b.sX,d=b.sXInner,e=b.sY;b=b.iBarWidth;var h=f(a.nScrollHead),g=h[0].style,k=h.children("div"),l=k[0].style,n=k.children("table");k=a.nScrollBody;var m=f(k),w=k.style,u=f(a.nScrollFoot).children("div"),q=u.children("table"),t=f(a.nTHead),r=f(a.nTable),v=r[0],za=v.style,T=a.nTFoot?f(a.nTFoot):null,A=a.oBrowser,
x=A.bScrollOversize,ac=J(a.aoColumns,"nTh"),Ya=[],y=[],z=[],C=[],G,H=function(a){a=a.style;a.paddingTop="0";a.paddingBottom="0";a.borderTopWidth="0";a.borderBottomWidth="0";a.height=0};var D=k.scrollHeight>k.clientHeight;if(a.scrollBarVis!==D&&a.scrollBarVis!==p)a.scrollBarVis=D,aa(a);else{a.scrollBarVis=D;r.children("thead, tfoot").remove();if(T){var E=T.clone().prependTo(r);var F=T.find("tr");E=E.find("tr")}var I=t.clone().prependTo(r);t=t.find("tr");D=I.find("tr");I.find("th, td").removeAttr("tabindex");
c||(w.width="100%",h[0].style.width="100%");f.each(ua(a,I),function(b,c){G=ba(a,b);c.style.width=a.aoColumns[G].sWidth});T&&N(function(a){a.style.width=""},E);h=r.outerWidth();""===c?(za.width="100%",x&&(r.find("tbody").height()>k.offsetHeight||"scroll"==m.css("overflow-y"))&&(za.width=B(r.outerWidth()-b)),h=r.outerWidth()):""!==d&&(za.width=B(d),h=r.outerWidth());N(H,D);N(function(a){z.push(a.innerHTML);Ya.push(B(f(a).css("width")))},D);N(function(a,b){-1!==f.inArray(a,ac)&&(a.style.width=Ya[b])},
t);f(D).height(0);T&&(N(H,E),N(function(a){C.push(a.innerHTML);y.push(B(f(a).css("width")))},E),N(function(a,b){a.style.width=y[b]},F),f(E).height(0));N(function(a,b){a.innerHTML='<div class="dataTables_sizing">'+z[b]+"</div>";a.childNodes[0].style.height="0";a.childNodes[0].style.overflow="hidden";a.style.width=Ya[b]},D);T&&N(function(a,b){a.innerHTML='<div class="dataTables_sizing">'+C[b]+"</div>";a.childNodes[0].style.height="0";a.childNodes[0].style.overflow="hidden";a.style.width=y[b]},E);r.outerWidth()<
h?(F=k.scrollHeight>k.offsetHeight||"scroll"==m.css("overflow-y")?h+b:h,x&&(k.scrollHeight>k.offsetHeight||"scroll"==m.css("overflow-y"))&&(za.width=B(F-b)),""!==c&&""===d||O(a,1,"Possible column misalignment",6)):F="100%";w.width=B(F);g.width=B(F);T&&(a.nScrollFoot.style.width=B(F));!e&&x&&(w.height=B(v.offsetHeight+b));c=r.outerWidth();n[0].style.width=B(c);l.width=B(c);d=r.height()>k.clientHeight||"scroll"==m.css("overflow-y");e="padding"+(A.bScrollbarLeft?"Left":"Right");l[e]=d?b+"px":"0px";T&&
(q[0].style.width=B(c),u[0].style.width=B(c),u[0].style[e]=d?b+"px":"0px");r.children("colgroup").insertBefore(r.children("thead"));m.trigger("scroll");!a.bSorted&&!a.bFiltered||a._drawHold||(k.scrollTop=0)}}function N(a,b,c){for(var d=0,e=0,h=b.length,g,k;e<h;){g=b[e].firstChild;for(k=c?c[e].firstChild:null;g;)1===g.nodeType&&(c?a(g,k,d):a(g,d),d++),g=g.nextSibling,k=c?k.nextSibling:null;e++}}function Ja(a){var b=a.nTable,c=a.aoColumns,d=a.oScroll,e=d.sY,h=d.sX,g=d.sXInner,k=c.length,l=oa(a,"bVisible"),
n=f("th",a.nTHead),m=b.getAttribute("width"),p=b.parentNode,u=!1,q,t=a.oBrowser;d=t.bScrollOversize;(q=b.style.width)&&-1!==q.indexOf("%")&&(m=q);for(q=0;q<l.length;q++){var r=c[l[q]];null!==r.sWidth&&(r.sWidth=Jb(r.sWidthOrig,p),u=!0)}if(d||!u&&!h&&!e&&k==W(a)&&k==n.length)for(q=0;q<k;q++)l=ba(a,q),null!==l&&(c[l].sWidth=B(n.eq(q).width()));else{k=f(b).clone().css("visibility","hidden").removeAttr("id");k.find("tbody tr").remove();var v=f("<tr/>").appendTo(k.find("tbody"));k.find("thead, tfoot").remove();
k.append(f(a.nTHead).clone()).append(f(a.nTFoot).clone());k.find("tfoot th, tfoot td").css("width","");n=ua(a,k.find("thead")[0]);for(q=0;q<l.length;q++)r=c[l[q]],n[q].style.width=null!==r.sWidthOrig&&""!==r.sWidthOrig?B(r.sWidthOrig):"",r.sWidthOrig&&h&&f(n[q]).append(f("<div/>").css({width:r.sWidthOrig,margin:0,padding:0,border:0,height:1}));if(a.aoData.length)for(q=0;q<l.length;q++)u=l[q],r=c[u],f(Kb(a,u)).clone(!1).append(r.sContentPadding).appendTo(v);f("[name]",k).removeAttr("name");r=f("<div/>").css(h||
e?{position:"absolute",top:0,left:0,height:1,right:0,overflow:"hidden"}:{}).append(k).appendTo(p);h&&g?k.width(g):h?(k.css("width","auto"),k.removeAttr("width"),k.width()<p.clientWidth&&m&&k.width(p.clientWidth)):e?k.width(p.clientWidth):m&&k.width(m);for(q=e=0;q<l.length;q++)p=f(n[q]),g=p.outerWidth()-p.width(),p=t.bBounding?Math.ceil(n[q].getBoundingClientRect().width):p.outerWidth(),e+=p,c[l[q]].sWidth=B(p-g);b.style.width=B(e);r.remove()}m&&(b.style.width=B(m));!m&&!h||a._reszEvt||(b=function(){f(z).on("resize.DT-"+
a.sInstance,Sa(function(){aa(a)}))},d?setTimeout(b,1E3):b(),a._reszEvt=!0)}function Jb(a,b){if(!a)return 0;a=f("<div/>").css("width",B(a)).appendTo(b||y.body);b=a[0].offsetWidth;a.remove();return b}function Kb(a,b){var c=Lb(a,b);if(0>c)return null;var d=a.aoData[c];return d.nTr?d.anCells[b]:f("<td/>").html(I(a,c,b,"display"))[0]}function Lb(a,b){for(var c,d=-1,e=-1,h=0,g=a.aoData.length;h<g;h++)c=I(a,h,b,"display")+"",c=c.replace(bc,""),c=c.replace(/&nbsp;/g," "),c.length>d&&(d=c.length,e=h);return e}
function B(a){return null===a?"0px":"number"==typeof a?0>a?"0px":a+"px":a.match(/\d$/)?a+"px":a}function Y(a){var b=[],c=a.aoColumns;var d=a.aaSortingFixed;var e=f.isPlainObject(d);var h=[];var g=function(a){a.length&&!f.isArray(a[0])?h.push(a):f.merge(h,a)};f.isArray(d)&&g(d);e&&d.pre&&g(d.pre);g(a.aaSorting);e&&d.post&&g(d.post);for(a=0;a<h.length;a++){var k=h[a][0];g=c[k].aDataSort;d=0;for(e=g.length;d<e;d++){var l=g[d];var n=c[l].sType||"string";h[a]._idx===p&&(h[a]._idx=f.inArray(h[a][1],c[l].asSorting));
b.push({src:k,col:l,dir:h[a][1],index:h[a]._idx,type:n,formatter:q.ext.type.order[n+"-pre"]})}}return b}function rb(a){var b,c=[],d=q.ext.type.order,e=a.aoData,h=0,g=a.aiDisplayMaster;Ka(a);var k=Y(a);var f=0;for(b=k.length;f<b;f++){var n=k[f];n.formatter&&h++;Mb(a,n.col)}if("ssp"!=D(a)&&0!==k.length){f=0;for(b=g.length;f<b;f++)c[g[f]]=f;h===k.length?g.sort(function(a,b){var d,h=k.length,g=e[a]._aSortData,f=e[b]._aSortData;for(d=0;d<h;d++){var l=k[d];var m=g[l.col];var n=f[l.col];m=m<n?-1:m>n?1:0;
if(0!==m)return"asc"===l.dir?m:-m}m=c[a];n=c[b];return m<n?-1:m>n?1:0}):g.sort(function(a,b){var h,g=k.length,f=e[a]._aSortData,l=e[b]._aSortData;for(h=0;h<g;h++){var m=k[h];var n=f[m.col];var p=l[m.col];m=d[m.type+"-"+m.dir]||d["string-"+m.dir];n=m(n,p);if(0!==n)return n}n=c[a];p=c[b];return n<p?-1:n>p?1:0})}a.bSorted=!0}function Nb(a){var b=a.aoColumns,c=Y(a);a=a.oLanguage.oAria;for(var d=0,e=b.length;d<e;d++){var h=b[d];var g=h.asSorting;var k=h.sTitle.replace(/<.*?>/g,"");var f=h.nTh;f.removeAttribute("aria-sort");
h.bSortable&&(0<c.length&&c[0].col==d?(f.setAttribute("aria-sort","asc"==c[0].dir?"ascending":"descending"),h=g[c[0].index+1]||g[0]):h=g[0],k+="asc"===h?a.sSortAscending:a.sSortDescending);f.setAttribute("aria-label",k)}}function Za(a,b,c,d){var e=a.aaSorting,h=a.aoColumns[b].asSorting,g=function(a,b){var c=a._idx;c===p&&(c=f.inArray(a[1],h));return c+1<h.length?c+1:b?null:0};"number"===typeof e[0]&&(e=a.aaSorting=[e]);c&&a.oFeatures.bSortMulti?(c=f.inArray(b,J(e,"0")),-1!==c?(b=g(e[c],!0),null===
b&&1===e.length&&(b=0),null===b?e.splice(c,1):(e[c][1]=h[b],e[c]._idx=b)):(e.push([b,h[0],0]),e[e.length-1]._idx=0)):e.length&&e[0][0]==b?(b=g(e[0]),e.length=1,e[0][1]=h[b],e[0]._idx=b):(e.length=0,e.push([b,h[0]]),e[0]._idx=0);V(a);"function"==typeof d&&d(a)}function Qa(a,b,c,d){var e=a.aoColumns[c];$a(b,{},function(b){!1!==e.bSortable&&(a.oFeatures.bProcessing?(K(a,!0),setTimeout(function(){Za(a,c,b.shiftKey,d);"ssp"!==D(a)&&K(a,!1)},0)):Za(a,c,b.shiftKey,d))})}function Aa(a){var b=a.aLastSort,
c=a.oClasses.sSortColumn,d=Y(a),e=a.oFeatures,h;if(e.bSort&&e.bSortClasses){e=0;for(h=b.length;e<h;e++){var g=b[e].src;f(J(a.aoData,"anCells",g)).removeClass(c+(2>e?e+1:3))}e=0;for(h=d.length;e<h;e++)g=d[e].src,f(J(a.aoData,"anCells",g)).addClass(c+(2>e?e+1:3))}a.aLastSort=d}function Mb(a,b){var c=a.aoColumns[b],d=q.ext.order[c.sSortDataType],e;d&&(e=d.call(a.oInstance,a,b,ca(a,b)));for(var h,g=q.ext.type.order[c.sType+"-pre"],k=0,f=a.aoData.length;k<f;k++)if(c=a.aoData[k],c._aSortData||(c._aSortData=
[]),!c._aSortData[b]||d)h=d?e[k]:I(a,k,b,"sort"),c._aSortData[b]=g?g(h):h}function Ba(a){if(a.oFeatures.bStateSave&&!a.bDestroying){var b={time:+new Date,start:a._iDisplayStart,length:a._iDisplayLength,order:f.extend(!0,[],a.aaSorting),search:Fb(a.oPreviousSearch),columns:f.map(a.aoColumns,function(b,d){return{visible:b.bVisible,search:Fb(a.aoPreSearchCols[d])}})};A(a,"aoStateSaveParams","stateSaveParams",[a,b]);a.oSavedState=b;a.fnStateSaveCallback.call(a.oInstance,a,b)}}function Ob(a,b,c){var d,
e,h=a.aoColumns;b=function(b){if(b&&b.time){var g=A(a,"aoStateLoadParams","stateLoadParams",[a,b]);if(-1===f.inArray(!1,g)&&(g=a.iStateDuration,!(0<g&&b.time<+new Date-1E3*g||b.columns&&h.length!==b.columns.length))){a.oLoadedState=f.extend(!0,{},b);b.start!==p&&(a._iDisplayStart=b.start,a.iInitDisplayStart=b.start);b.length!==p&&(a._iDisplayLength=b.length);b.order!==p&&(a.aaSorting=[],f.each(b.order,function(b,c){a.aaSorting.push(c[0]>=h.length?[0,c[1]]:c)}));b.search!==p&&f.extend(a.oPreviousSearch,
Gb(b.search));if(b.columns)for(d=0,e=b.columns.length;d<e;d++)g=b.columns[d],g.visible!==p&&(h[d].bVisible=g.visible),g.search!==p&&f.extend(a.aoPreSearchCols[d],Gb(g.search));A(a,"aoStateLoaded","stateLoaded",[a,b])}}c()};if(a.oFeatures.bStateSave){var g=a.fnStateLoadCallback.call(a.oInstance,a,b);g!==p&&b(g)}else c()}function Ca(a){var b=q.settings;a=f.inArray(a,J(b,"nTable"));return-1!==a?b[a]:null}function O(a,b,c,d){c="DataTables warning: "+(a?"table id="+a.sTableId+" - ":"")+c;d&&(c+=". For more information about this error, please see http://datatables.net/tn/"+
d);if(b)z.console&&console.log&&console.log(c);else if(b=q.ext,b=b.sErrMode||b.errMode,a&&A(a,null,"error",[a,d,c]),"alert"==b)alert(c);else{if("throw"==b)throw Error(c);"function"==typeof b&&b(a,d,c)}}function M(a,b,c,d){f.isArray(c)?f.each(c,function(c,d){f.isArray(d)?M(a,b,d[0],d[1]):M(a,b,d)}):(d===p&&(d=c),b[c]!==p&&(a[d]=b[c]))}function ab(a,b,c){var d;for(d in b)if(b.hasOwnProperty(d)){var e=b[d];f.isPlainObject(e)?(f.isPlainObject(a[d])||(a[d]={}),f.extend(!0,a[d],e)):c&&"data"!==d&&"aaData"!==
d&&f.isArray(e)?a[d]=e.slice():a[d]=e}return a}function $a(a,b,c){f(a).on("click.DT",b,function(b){f(a).blur();c(b)}).on("keypress.DT",b,function(a){13===a.which&&(a.preventDefault(),c(a))}).on("selectstart.DT",function(){return!1})}function E(a,b,c,d){c&&a[b].push({fn:c,sName:d})}function A(a,b,c,d){var e=[];b&&(e=f.map(a[b].slice().reverse(),function(b,c){return b.fn.apply(a.oInstance,d)}));null!==c&&(b=f.Event(c+".dt"),f(a.nTable).trigger(b,d),e.push(b.result));return e}function Wa(a){var b=a._iDisplayStart,
c=a.fnDisplayEnd(),d=a._iDisplayLength;b>=c&&(b=c-d);b-=b%d;if(-1===d||0>b)b=0;a._iDisplayStart=b}function Ra(a,b){a=a.renderer;var c=q.ext.renderer[b];return f.isPlainObject(a)&&a[b]?c[a[b]]||c._:"string"===typeof a?c[a]||c._:c._}function D(a){return a.oFeatures.bServerSide?"ssp":a.ajax||a.sAjaxSource?"ajax":"dom"}function ka(a,b){var c=Pb.numbers_length,d=Math.floor(c/2);b<=c?a=Z(0,b):a<=d?(a=Z(0,c-2),a.push("ellipsis"),a.push(b-1)):(a>=b-1-d?a=Z(b-(c-2),b):(a=Z(a-d+2,a+d-1),a.push("ellipsis"),
a.push(b-1)),a.splice(0,0,"ellipsis"),a.splice(0,0,0));a.DT_el="span";return a}function Ha(a){f.each({num:function(b){return Da(b,a)},"num-fmt":function(b){return Da(b,a,bb)},"html-num":function(b){return Da(b,a,Ea)},"html-num-fmt":function(b){return Da(b,a,Ea,bb)}},function(b,c){C.type.order[b+a+"-pre"]=c;b.match(/^html\-/)&&(C.type.search[b+a]=C.type.search.html)})}function Qb(a){return function(){var b=[Ca(this[q.ext.iApiIndex])].concat(Array.prototype.slice.call(arguments));return q.ext.internal[a].apply(this,
b)}}var q=function(a){this.$=function(a,b){return this.api(!0).$(a,b)};this._=function(a,b){return this.api(!0).rows(a,b).data()};this.api=function(a){return a?new v(Ca(this[C.iApiIndex])):new v(this)};this.fnAddData=function(a,b){var c=this.api(!0);a=f.isArray(a)&&(f.isArray(a[0])||f.isPlainObject(a[0]))?c.rows.add(a):c.row.add(a);(b===p||b)&&c.draw();return a.flatten().toArray()};this.fnAdjustColumnSizing=function(a){var b=this.api(!0).columns.adjust(),c=b.settings()[0],d=c.oScroll;a===p||a?b.draw(!1):
(""!==d.sX||""!==d.sY)&&na(c)};this.fnClearTable=function(a){var b=this.api(!0).clear();(a===p||a)&&b.draw()};this.fnClose=function(a){this.api(!0).row(a).child.hide()};this.fnDeleteRow=function(a,b,c){var d=this.api(!0);a=d.rows(a);var e=a.settings()[0],h=e.aoData[a[0][0]];a.remove();b&&b.call(this,e,h);(c===p||c)&&d.draw();return h};this.fnDestroy=function(a){this.api(!0).destroy(a)};this.fnDraw=function(a){this.api(!0).draw(a)};this.fnFilter=function(a,b,c,d,e,f){e=this.api(!0);null===b||b===p?
e.search(a,c,d,f):e.column(b).search(a,c,d,f);e.draw()};this.fnGetData=function(a,b){var c=this.api(!0);if(a!==p){var d=a.nodeName?a.nodeName.toLowerCase():"";return b!==p||"td"==d||"th"==d?c.cell(a,b).data():c.row(a).data()||null}return c.data().toArray()};this.fnGetNodes=function(a){var b=this.api(!0);return a!==p?b.row(a).node():b.rows().nodes().flatten().toArray()};this.fnGetPosition=function(a){var b=this.api(!0),c=a.nodeName.toUpperCase();return"TR"==c?b.row(a).index():"TD"==c||"TH"==c?(a=b.cell(a).index(),
[a.row,a.columnVisible,a.column]):null};this.fnIsOpen=function(a){return this.api(!0).row(a).child.isShown()};this.fnOpen=function(a,b,c){return this.api(!0).row(a).child(b,c).show().child()[0]};this.fnPageChange=function(a,b){a=this.api(!0).page(a);(b===p||b)&&a.draw(!1)};this.fnSetColumnVis=function(a,b,c){a=this.api(!0).column(a).visible(b);(c===p||c)&&a.columns.adjust().draw()};this.fnSettings=function(){return Ca(this[C.iApiIndex])};this.fnSort=function(a){this.api(!0).order(a).draw()};this.fnSortListener=
function(a,b,c){this.api(!0).order.listener(a,b,c)};this.fnUpdate=function(a,b,c,d,e){var h=this.api(!0);c===p||null===c?h.row(b).data(a):h.cell(b,c).data(a);(e===p||e)&&h.columns.adjust();(d===p||d)&&h.draw();return 0};this.fnVersionCheck=C.fnVersionCheck;var b=this,c=a===p,d=this.length;c&&(a={});this.oApi=this.internal=C.internal;for(var e in q.ext.internal)e&&(this[e]=Qb(e));this.each(function(){var e={},g=1<d?ab(e,a,!0):a,k=0,l;e=this.getAttribute("id");var n=!1,m=q.defaults,w=f(this);if("table"!=
this.nodeName.toLowerCase())O(null,0,"Non-table node initialisation ("+this.nodeName+")",2);else{jb(m);kb(m.column);L(m,m,!0);L(m.column,m.column,!0);L(m,f.extend(g,w.data()),!0);var u=q.settings;k=0;for(l=u.length;k<l;k++){var t=u[k];if(t.nTable==this||t.nTHead&&t.nTHead.parentNode==this||t.nTFoot&&t.nTFoot.parentNode==this){var v=g.bRetrieve!==p?g.bRetrieve:m.bRetrieve;if(c||v)return t.oInstance;if(g.bDestroy!==p?g.bDestroy:m.bDestroy){t.oInstance.fnDestroy();break}else{O(t,0,"Cannot reinitialise DataTable",
3);return}}if(t.sTableId==this.id){u.splice(k,1);break}}if(null===e||""===e)this.id=e="DataTables_Table_"+q.ext._unique++;var r=f.extend(!0,{},q.models.oSettings,{sDestroyWidth:w[0].style.width,sInstance:e,sTableId:e});r.nTable=this;r.oApi=b.internal;r.oInit=g;u.push(r);r.oInstance=1===b.length?b:w.dataTable();jb(g);Ga(g.oLanguage);g.aLengthMenu&&!g.iDisplayLength&&(g.iDisplayLength=f.isArray(g.aLengthMenu[0])?g.aLengthMenu[0][0]:g.aLengthMenu[0]);g=ab(f.extend(!0,{},m),g);M(r.oFeatures,g,"bPaginate bLengthChange bFilter bSort bSortMulti bInfo bProcessing bAutoWidth bSortClasses bServerSide bDeferRender".split(" "));
M(r,g,["asStripeClasses","ajax","fnServerData","fnFormatNumber","sServerMethod","aaSorting","aaSortingFixed","aLengthMenu","sPaginationType","sAjaxSource","sAjaxDataProp","iStateDuration","sDom","bSortCellsTop","iTabIndex","fnStateLoadCallback","fnStateSaveCallback","renderer","searchDelay","rowId",["iCookieDuration","iStateDuration"],["oSearch","oPreviousSearch"],["aoSearchCols","aoPreSearchCols"],["iDisplayLength","_iDisplayLength"]]);M(r.oScroll,g,[["sScrollX","sX"],["sScrollXInner","sXInner"],
["sScrollY","sY"],["bScrollCollapse","bCollapse"]]);M(r.oLanguage,g,"fnInfoCallback");E(r,"aoDrawCallback",g.fnDrawCallback,"user");E(r,"aoServerParams",g.fnServerParams,"user");E(r,"aoStateSaveParams",g.fnStateSaveParams,"user");E(r,"aoStateLoadParams",g.fnStateLoadParams,"user");E(r,"aoStateLoaded",g.fnStateLoaded,"user");E(r,"aoRowCallback",g.fnRowCallback,"user");E(r,"aoRowCreatedCallback",g.fnCreatedRow,"user");E(r,"aoHeaderCallback",g.fnHeaderCallback,"user");E(r,"aoFooterCallback",g.fnFooterCallback,
"user");E(r,"aoInitComplete",g.fnInitComplete,"user");E(r,"aoPreDrawCallback",g.fnPreDrawCallback,"user");r.rowIdFn=U(g.rowId);lb(r);var x=r.oClasses;f.extend(x,q.ext.classes,g.oClasses);w.addClass(x.sTable);r.iInitDisplayStart===p&&(r.iInitDisplayStart=g.iDisplayStart,r._iDisplayStart=g.iDisplayStart);null!==g.iDeferLoading&&(r.bDeferLoading=!0,e=f.isArray(g.iDeferLoading),r._iRecordsDisplay=e?g.iDeferLoading[0]:g.iDeferLoading,r._iRecordsTotal=e?g.iDeferLoading[1]:g.iDeferLoading);var y=r.oLanguage;
f.extend(!0,y,g.oLanguage);y.sUrl&&(f.ajax({dataType:"json",url:y.sUrl,success:function(a){Ga(a);L(m.oLanguage,a);f.extend(!0,y,a);ja(r)},error:function(){ja(r)}}),n=!0);null===g.asStripeClasses&&(r.asStripeClasses=[x.sStripeOdd,x.sStripeEven]);e=r.asStripeClasses;var z=w.children("tbody").find("tr").eq(0);-1!==f.inArray(!0,f.map(e,function(a,b){return z.hasClass(a)}))&&(f("tbody tr",this).removeClass(e.join(" ")),r.asDestroyStripes=e.slice());e=[];u=this.getElementsByTagName("thead");0!==u.length&&
(fa(r.aoHeader,u[0]),e=ua(r));if(null===g.aoColumns)for(u=[],k=0,l=e.length;k<l;k++)u.push(null);else u=g.aoColumns;k=0;for(l=u.length;k<l;k++)Ia(r,e?e[k]:null);nb(r,g.aoColumnDefs,u,function(a,b){ma(r,a,b)});if(z.length){var B=function(a,b){return null!==a.getAttribute("data-"+b)?b:null};f(z[0]).children("th, td").each(function(a,b){var c=r.aoColumns[a];if(c.mData===a){var d=B(b,"sort")||B(b,"order");b=B(b,"filter")||B(b,"search");if(null!==d||null!==b)c.mData={_:a+".display",sort:null!==d?a+".@data-"+
d:p,type:null!==d?a+".@data-"+d:p,filter:null!==b?a+".@data-"+b:p},ma(r,a)}})}var C=r.oFeatures;e=function(){if(g.aaSorting===p){var a=r.aaSorting;k=0;for(l=a.length;k<l;k++)a[k][1]=r.aoColumns[k].asSorting[0]}Aa(r);C.bSort&&E(r,"aoDrawCallback",function(){if(r.bSorted){var a=Y(r),b={};f.each(a,function(a,c){b[c.src]=c.dir});A(r,null,"order",[r,a,b]);Nb(r)}});E(r,"aoDrawCallback",function(){(r.bSorted||"ssp"===D(r)||C.bDeferRender)&&Aa(r)},"sc");a=w.children("caption").each(function(){this._captionSide=
f(this).css("caption-side")});var b=w.children("thead");0===b.length&&(b=f("<thead/>").appendTo(w));r.nTHead=b[0];b=w.children("tbody");0===b.length&&(b=f("<tbody/>").appendTo(w));r.nTBody=b[0];b=w.children("tfoot");0===b.length&&0<a.length&&(""!==r.oScroll.sX||""!==r.oScroll.sY)&&(b=f("<tfoot/>").appendTo(w));0===b.length||0===b.children().length?w.addClass(x.sNoFooter):0<b.length&&(r.nTFoot=b[0],fa(r.aoFooter,r.nTFoot));if(g.aaData)for(k=0;k<g.aaData.length;k++)R(r,g.aaData[k]);else(r.bDeferLoading||
"dom"==D(r))&&pa(r,f(r.nTBody).children("tr"));r.aiDisplay=r.aiDisplayMaster.slice();r.bInitialised=!0;!1===n&&ja(r)};g.bStateSave?(C.bStateSave=!0,E(r,"aoDrawCallback",Ba,"state_save"),Ob(r,g,e)):e()}});b=null;return this},C,t,x,cb={},Rb=/[\r\n\u2028]/g,Ea=/<.*?>/g,cc=/^\d{2,4}[\.\/\-]\d{1,2}[\.\/\-]\d{1,2}([T ]{1}\d{1,2}[:\.]\d{2}([\.:]\d{2})?)?$/,dc=/(\/|\.|\*|\+|\?|\||\(|\)|\[|\]|\{|\}|\\|\$|\^|\-)/g,bb=/[',$£€¥%\u2009\u202F\u20BD\u20a9\u20BArfkɃΞ]/gi,P=function(a){return a&&!0!==a&&"-"!==a?!1:
!0},Sb=function(a){var b=parseInt(a,10);return!isNaN(b)&&isFinite(a)?b:null},Tb=function(a,b){cb[b]||(cb[b]=new RegExp(Ua(b),"g"));return"string"===typeof a&&"."!==b?a.replace(/\./g,"").replace(cb[b],"."):a},db=function(a,b,c){var d="string"===typeof a;if(P(a))return!0;b&&d&&(a=Tb(a,b));c&&d&&(a=a.replace(bb,""));return!isNaN(parseFloat(a))&&isFinite(a)},Ub=function(a,b,c){return P(a)?!0:P(a)||"string"===typeof a?db(a.replace(Ea,""),b,c)?!0:null:null},J=function(a,b,c){var d=[],e=0,h=a.length;if(c!==
p)for(;e<h;e++)a[e]&&a[e][b]&&d.push(a[e][b][c]);else for(;e<h;e++)a[e]&&d.push(a[e][b]);return d},la=function(a,b,c,d){var e=[],h=0,g=b.length;if(d!==p)for(;h<g;h++)a[b[h]][c]&&e.push(a[b[h]][c][d]);else for(;h<g;h++)e.push(a[b[h]][c]);return e},Z=function(a,b){var c=[];if(b===p){b=0;var d=a}else d=b,b=a;for(a=b;a<d;a++)c.push(a);return c},Vb=function(a){for(var b=[],c=0,d=a.length;c<d;c++)a[c]&&b.push(a[c]);return b},ta=function(a){a:{if(!(2>a.length)){var b=a.slice().sort();for(var c=b[0],d=1,
e=b.length;d<e;d++){if(b[d]===c){b=!1;break a}c=b[d]}}b=!0}if(b)return a.slice();b=[];e=a.length;var h,g=0;d=0;a:for(;d<e;d++){c=a[d];for(h=0;h<g;h++)if(b[h]===c)continue a;b.push(c);g++}return b};q.util={throttle:function(a,b){var c=b!==p?b:200,d,e;return function(){var b=this,g=+new Date,f=arguments;d&&g<d+c?(clearTimeout(e),e=setTimeout(function(){d=p;a.apply(b,f)},c)):(d=g,a.apply(b,f))}},escapeRegex:function(a){return a.replace(dc,"\\$1")}};var F=function(a,b,c){a[b]!==p&&(a[c]=a[b])},da=/\[.*?\]$/,
X=/\(\)$/,Ua=q.util.escapeRegex,ya=f("<div>")[0],$b=ya.textContent!==p,bc=/<.*?>/g,Sa=q.util.throttle,Wb=[],G=Array.prototype,ec=function(a){var b,c=q.settings,d=f.map(c,function(a,b){return a.nTable});if(a){if(a.nTable&&a.oApi)return[a];if(a.nodeName&&"table"===a.nodeName.toLowerCase()){var e=f.inArray(a,d);return-1!==e?[c[e]]:null}if(a&&"function"===typeof a.settings)return a.settings().toArray();"string"===typeof a?b=f(a):a instanceof f&&(b=a)}else return[];if(b)return b.map(function(a){e=f.inArray(this,
d);return-1!==e?c[e]:null}).toArray()};var v=function(a,b){if(!(this instanceof v))return new v(a,b);var c=[],d=function(a){(a=ec(a))&&c.push.apply(c,a)};if(f.isArray(a))for(var e=0,h=a.length;e<h;e++)d(a[e]);else d(a);this.context=ta(c);b&&f.merge(this,b);this.selector={rows:null,cols:null,opts:null};v.extend(this,this,Wb)};q.Api=v;f.extend(v.prototype,{any:function(){return 0!==this.count()},concat:G.concat,context:[],count:function(){return this.flatten().length},each:function(a){for(var b=0,c=
this.length;b<c;b++)a.call(this,this[b],b,this);return this},eq:function(a){var b=this.context;return b.length>a?new v(b[a],this[a]):null},filter:function(a){var b=[];if(G.filter)b=G.filter.call(this,a,this);else for(var c=0,d=this.length;c<d;c++)a.call(this,this[c],c,this)&&b.push(this[c]);return new v(this.context,b)},flatten:function(){var a=[];return new v(this.context,a.concat.apply(a,this.toArray()))},join:G.join,indexOf:G.indexOf||function(a,b){b=b||0;for(var c=this.length;b<c;b++)if(this[b]===
a)return b;return-1},iterator:function(a,b,c,d){var e=[],h,g,f=this.context,l,n=this.selector;"string"===typeof a&&(d=c,c=b,b=a,a=!1);var m=0;for(h=f.length;m<h;m++){var q=new v(f[m]);if("table"===b){var u=c.call(q,f[m],m);u!==p&&e.push(u)}else if("columns"===b||"rows"===b)u=c.call(q,f[m],this[m],m),u!==p&&e.push(u);else if("column"===b||"column-rows"===b||"row"===b||"cell"===b){var t=this[m];"column-rows"===b&&(l=Fa(f[m],n.opts));var x=0;for(g=t.length;x<g;x++)u=t[x],u="cell"===b?c.call(q,f[m],u.row,
u.column,m,x):c.call(q,f[m],u,m,x,l),u!==p&&e.push(u)}}return e.length||d?(a=new v(f,a?e.concat.apply([],e):e),b=a.selector,b.rows=n.rows,b.cols=n.cols,b.opts=n.opts,a):this},lastIndexOf:G.lastIndexOf||function(a,b){return this.indexOf.apply(this.toArray.reverse(),arguments)},length:0,map:function(a){var b=[];if(G.map)b=G.map.call(this,a,this);else for(var c=0,d=this.length;c<d;c++)b.push(a.call(this,this[c],c));return new v(this.context,b)},pluck:function(a){return this.map(function(b){return b[a]})},
pop:G.pop,push:G.push,reduce:G.reduce||function(a,b){return mb(this,a,b,0,this.length,1)},reduceRight:G.reduceRight||function(a,b){return mb(this,a,b,this.length-1,-1,-1)},reverse:G.reverse,selector:null,shift:G.shift,slice:function(){return new v(this.context,this)},sort:G.sort,splice:G.splice,toArray:function(){return G.slice.call(this)},to$:function(){return f(this)},toJQuery:function(){return f(this)},unique:function(){return new v(this.context,ta(this))},unshift:G.unshift});v.extend=function(a,
b,c){if(c.length&&b&&(b instanceof v||b.__dt_wrapper)){var d,e=function(a,b,c){return function(){var d=b.apply(a,arguments);v.extend(d,d,c.methodExt);return d}};var h=0;for(d=c.length;h<d;h++){var g=c[h];b[g.name]="function"===g.type?e(a,g.val,g):"object"===g.type?{}:g.val;b[g.name].__dt_wrapper=!0;v.extend(a,b[g.name],g.propExt)}}};v.register=t=function(a,b){if(f.isArray(a))for(var c=0,d=a.length;c<d;c++)v.register(a[c],b);else{d=a.split(".");var e=Wb,h;a=0;for(c=d.length;a<c;a++){var g=(h=-1!==
d[a].indexOf("()"))?d[a].replace("()",""):d[a];a:{var k=0;for(var l=e.length;k<l;k++)if(e[k].name===g){k=e[k];break a}k=null}k||(k={name:g,val:{},methodExt:[],propExt:[],type:"object"},e.push(k));a===c-1?(k.val=b,k.type="function"===typeof b?"function":f.isPlainObject(b)?"object":"other"):e=h?k.methodExt:k.propExt}}};v.registerPlural=x=function(a,b,c){v.register(a,c);v.register(b,function(){var a=c.apply(this,arguments);return a===this?this:a instanceof v?a.length?f.isArray(a[0])?new v(a.context,
a[0]):a[0]:p:a})};var fc=function(a,b){if("number"===typeof a)return[b[a]];var c=f.map(b,function(a,b){return a.nTable});return f(c).filter(a).map(function(a){a=f.inArray(this,c);return b[a]}).toArray()};t("tables()",function(a){return a?new v(fc(a,this.context)):this});t("table()",function(a){a=this.tables(a);var b=a.context;return b.length?new v(b[0]):a});x("tables().nodes()","table().node()",function(){return this.iterator("table",function(a){return a.nTable},1)});x("tables().body()","table().body()",
function(){return this.iterator("table",function(a){return a.nTBody},1)});x("tables().header()","table().header()",function(){return this.iterator("table",function(a){return a.nTHead},1)});x("tables().footer()","table().footer()",function(){return this.iterator("table",function(a){return a.nTFoot},1)});x("tables().containers()","table().container()",function(){return this.iterator("table",function(a){return a.nTableWrapper},1)});t("draw()",function(a){return this.iterator("table",function(b){"page"===
a?S(b):("string"===typeof a&&(a="full-hold"===a?!1:!0),V(b,!1===a))})});t("page()",function(a){return a===p?this.page.info().page:this.iterator("table",function(b){Xa(b,a)})});t("page.info()",function(a){if(0===this.context.length)return p;a=this.context[0];var b=a._iDisplayStart,c=a.oFeatures.bPaginate?a._iDisplayLength:-1,d=a.fnRecordsDisplay(),e=-1===c;return{page:e?0:Math.floor(b/c),pages:e?1:Math.ceil(d/c),start:b,end:a.fnDisplayEnd(),length:c,recordsTotal:a.fnRecordsTotal(),recordsDisplay:d,
serverSide:"ssp"===D(a)}});t("page.len()",function(a){return a===p?0!==this.context.length?this.context[0]._iDisplayLength:p:this.iterator("table",function(b){Va(b,a)})});var Xb=function(a,b,c){if(c){var d=new v(a);d.one("draw",function(){c(d.ajax.json())})}if("ssp"==D(a))V(a,b);else{K(a,!0);var e=a.jqXHR;e&&4!==e.readyState&&e.abort();va(a,[],function(c){qa(a);c=wa(a,c);for(var d=0,e=c.length;d<e;d++)R(a,c[d]);V(a,b);K(a,!1)})}};t("ajax.json()",function(){var a=this.context;if(0<a.length)return a[0].json});
t("ajax.params()",function(){var a=this.context;if(0<a.length)return a[0].oAjaxData});t("ajax.reload()",function(a,b){return this.iterator("table",function(c){Xb(c,!1===b,a)})});t("ajax.url()",function(a){var b=this.context;if(a===p){if(0===b.length)return p;b=b[0];return b.ajax?f.isPlainObject(b.ajax)?b.ajax.url:b.ajax:b.sAjaxSource}return this.iterator("table",function(b){f.isPlainObject(b.ajax)?b.ajax.url=a:b.ajax=a})});t("ajax.url().load()",function(a,b){return this.iterator("table",function(c){Xb(c,
!1===b,a)})});var eb=function(a,b,c,d,e){var h=[],g,k,l;var n=typeof b;b&&"string"!==n&&"function"!==n&&b.length!==p||(b=[b]);n=0;for(k=b.length;n<k;n++){var m=b[n]&&b[n].split&&!b[n].match(/[\[\(:]/)?b[n].split(","):[b[n]];var q=0;for(l=m.length;q<l;q++)(g=c("string"===typeof m[q]?f.trim(m[q]):m[q]))&&g.length&&(h=h.concat(g))}a=C.selector[a];if(a.length)for(n=0,k=a.length;n<k;n++)h=a[n](d,e,h);return ta(h)},fb=function(a){a||(a={});a.filter&&a.search===p&&(a.search=a.filter);return f.extend({search:"none",
order:"current",page:"all"},a)},gb=function(a){for(var b=0,c=a.length;b<c;b++)if(0<a[b].length)return a[0]=a[b],a[0].length=1,a.length=1,a.context=[a.context[b]],a;a.length=0;return a},Fa=function(a,b){var c=[],d=a.aiDisplay;var e=a.aiDisplayMaster;var h=b.search;var g=b.order;b=b.page;if("ssp"==D(a))return"removed"===h?[]:Z(0,e.length);if("current"==b)for(g=a._iDisplayStart,a=a.fnDisplayEnd();g<a;g++)c.push(d[g]);else if("current"==g||"applied"==g)if("none"==h)c=e.slice();else if("applied"==h)c=
d.slice();else{if("removed"==h){var k={};g=0;for(a=d.length;g<a;g++)k[d[g]]=null;c=f.map(e,function(a){return k.hasOwnProperty(a)?null:a})}}else if("index"==g||"original"==g)for(g=0,a=a.aoData.length;g<a;g++)"none"==h?c.push(g):(e=f.inArray(g,d),(-1===e&&"removed"==h||0<=e&&"applied"==h)&&c.push(g));return c},gc=function(a,b,c){var d;return eb("row",b,function(b){var e=Sb(b),g=a.aoData;if(null!==e&&!c)return[e];d||(d=Fa(a,c));if(null!==e&&-1!==f.inArray(e,d))return[e];if(null===b||b===p||""===b)return d;
if("function"===typeof b)return f.map(d,function(a){var c=g[a];return b(a,c._aData,c.nTr)?a:null});if(b.nodeName){e=b._DT_RowIndex;var k=b._DT_CellIndex;if(e!==p)return g[e]&&g[e].nTr===b?[e]:[];if(k)return g[k.row]&&g[k.row].nTr===b.parentNode?[k.row]:[];e=f(b).closest("*[data-dt-row]");return e.length?[e.data("dt-row")]:[]}if("string"===typeof b&&"#"===b.charAt(0)&&(e=a.aIds[b.replace(/^#/,"")],e!==p))return[e.idx];e=Vb(la(a.aoData,d,"nTr"));return f(e).filter(b).map(function(){return this._DT_RowIndex}).toArray()},
a,c)};t("rows()",function(a,b){a===p?a="":f.isPlainObject(a)&&(b=a,a="");b=fb(b);var c=this.iterator("table",function(c){return gc(c,a,b)},1);c.selector.rows=a;c.selector.opts=b;return c});t("rows().nodes()",function(){return this.iterator("row",function(a,b){return a.aoData[b].nTr||p},1)});t("rows().data()",function(){return this.iterator(!0,"rows",function(a,b){return la(a.aoData,b,"_aData")},1)});x("rows().cache()","row().cache()",function(a){return this.iterator("row",function(b,c){b=b.aoData[c];
return"search"===a?b._aFilterData:b._aSortData},1)});x("rows().invalidate()","row().invalidate()",function(a){return this.iterator("row",function(b,c){ea(b,c,a)})});x("rows().indexes()","row().index()",function(){return this.iterator("row",function(a,b){return b},1)});x("rows().ids()","row().id()",function(a){for(var b=[],c=this.context,d=0,e=c.length;d<e;d++)for(var h=0,g=this[d].length;h<g;h++){var f=c[d].rowIdFn(c[d].aoData[this[d][h]]._aData);b.push((!0===a?"#":"")+f)}return new v(c,b)});x("rows().remove()",
"row().remove()",function(){var a=this;this.iterator("row",function(b,c,d){var e=b.aoData,h=e[c],g,f;e.splice(c,1);var l=0;for(g=e.length;l<g;l++){var n=e[l];var m=n.anCells;null!==n.nTr&&(n.nTr._DT_RowIndex=l);if(null!==m)for(n=0,f=m.length;n<f;n++)m[n]._DT_CellIndex.row=l}ra(b.aiDisplayMaster,c);ra(b.aiDisplay,c);ra(a[d],c,!1);0<b._iRecordsDisplay&&b._iRecordsDisplay--;Wa(b);c=b.rowIdFn(h._aData);c!==p&&delete b.aIds[c]});this.iterator("table",function(a){for(var b=0,d=a.aoData.length;b<d;b++)a.aoData[b].idx=
b});return this});t("rows.add()",function(a){var b=this.iterator("table",function(b){var c,d=[];var g=0;for(c=a.length;g<c;g++){var f=a[g];f.nodeName&&"TR"===f.nodeName.toUpperCase()?d.push(pa(b,f)[0]):d.push(R(b,f))}return d},1),c=this.rows(-1);c.pop();f.merge(c,b);return c});t("row()",function(a,b){return gb(this.rows(a,b))});t("row().data()",function(a){var b=this.context;if(a===p)return b.length&&this.length?b[0].aoData[this[0]]._aData:p;var c=b[0].aoData[this[0]];c._aData=a;f.isArray(a)&&c.nTr.id&&
Q(b[0].rowId)(a,c.nTr.id);ea(b[0],this[0],"data");return this});t("row().node()",function(){var a=this.context;return a.length&&this.length?a[0].aoData[this[0]].nTr||null:null});t("row.add()",function(a){a instanceof f&&a.length&&(a=a[0]);var b=this.iterator("table",function(b){return a.nodeName&&"TR"===a.nodeName.toUpperCase()?pa(b,a)[0]:R(b,a)});return this.row(b[0])});var hc=function(a,b,c,d){var e=[],h=function(b,c){if(f.isArray(b)||b instanceof f)for(var d=0,g=b.length;d<g;d++)h(b[d],c);else b.nodeName&&
"tr"===b.nodeName.toLowerCase()?e.push(b):(d=f("<tr><td/></tr>").addClass(c),f("td",d).addClass(c).html(b)[0].colSpan=W(a),e.push(d[0]))};h(c,d);b._details&&b._details.detach();b._details=f(e);b._detailsShow&&b._details.insertAfter(b.nTr)},hb=function(a,b){var c=a.context;c.length&&(a=c[0].aoData[b!==p?b:a[0]])&&a._details&&(a._details.remove(),a._detailsShow=p,a._details=p)},Yb=function(a,b){var c=a.context;c.length&&a.length&&(a=c[0].aoData[a[0]],a._details&&((a._detailsShow=b)?a._details.insertAfter(a.nTr):
a._details.detach(),ic(c[0])))},ic=function(a){var b=new v(a),c=a.aoData;b.off("draw.dt.DT_details column-visibility.dt.DT_details destroy.dt.DT_details");0<J(c,"_details").length&&(b.on("draw.dt.DT_details",function(d,e){a===e&&b.rows({page:"current"}).eq(0).each(function(a){a=c[a];a._detailsShow&&a._details.insertAfter(a.nTr)})}),b.on("column-visibility.dt.DT_details",function(b,e,f,g){if(a===e)for(e=W(e),f=0,g=c.length;f<g;f++)b=c[f],b._details&&b._details.children("td[colspan]").attr("colspan",
e)}),b.on("destroy.dt.DT_details",function(d,e){if(a===e)for(d=0,e=c.length;d<e;d++)c[d]._details&&hb(b,d)}))};t("row().child()",function(a,b){var c=this.context;if(a===p)return c.length&&this.length?c[0].aoData[this[0]]._details:p;!0===a?this.child.show():!1===a?hb(this):c.length&&this.length&&hc(c[0],c[0].aoData[this[0]],a,b);return this});t(["row().child.show()","row().child().show()"],function(a){Yb(this,!0);return this});t(["row().child.hide()","row().child().hide()"],function(){Yb(this,!1);
return this});t(["row().child.remove()","row().child().remove()"],function(){hb(this);return this});t("row().child.isShown()",function(){var a=this.context;return a.length&&this.length?a[0].aoData[this[0]]._detailsShow||!1:!1});var jc=/^([^:]+):(name|visIdx|visible)$/,Zb=function(a,b,c,d,e){c=[];d=0;for(var f=e.length;d<f;d++)c.push(I(a,e[d],b));return c},kc=function(a,b,c){var d=a.aoColumns,e=J(d,"sName"),h=J(d,"nTh");return eb("column",b,function(b){var g=Sb(b);if(""===b)return Z(d.length);if(null!==
g)return[0<=g?g:d.length+g];if("function"===typeof b){var l=Fa(a,c);return f.map(d,function(c,d){return b(d,Zb(a,d,0,0,l),h[d])?d:null})}var n="string"===typeof b?b.match(jc):"";if(n)switch(n[2]){case "visIdx":case "visible":g=parseInt(n[1],10);if(0>g){var m=f.map(d,function(a,b){return a.bVisible?b:null});return[m[m.length+g]]}return[ba(a,g)];case "name":return f.map(e,function(a,b){return a===n[1]?b:null});default:return[]}if(b.nodeName&&b._DT_CellIndex)return[b._DT_CellIndex.column];g=f(h).filter(b).map(function(){return f.inArray(this,
h)}).toArray();if(g.length||!b.nodeName)return g;g=f(b).closest("*[data-dt-column]");return g.length?[g.data("dt-column")]:[]},a,c)};t("columns()",function(a,b){a===p?a="":f.isPlainObject(a)&&(b=a,a="");b=fb(b);var c=this.iterator("table",function(c){return kc(c,a,b)},1);c.selector.cols=a;c.selector.opts=b;return c});x("columns().header()","column().header()",function(a,b){return this.iterator("column",function(a,b){return a.aoColumns[b].nTh},1)});x("columns().footer()","column().footer()",function(a,
b){return this.iterator("column",function(a,b){return a.aoColumns[b].nTf},1)});x("columns().data()","column().data()",function(){return this.iterator("column-rows",Zb,1)});x("columns().dataSrc()","column().dataSrc()",function(){return this.iterator("column",function(a,b){return a.aoColumns[b].mData},1)});x("columns().cache()","column().cache()",function(a){return this.iterator("column-rows",function(b,c,d,e,f){return la(b.aoData,f,"search"===a?"_aFilterData":"_aSortData",c)},1)});x("columns().nodes()",
"column().nodes()",function(){return this.iterator("column-rows",function(a,b,c,d,e){return la(a.aoData,e,"anCells",b)},1)});x("columns().visible()","column().visible()",function(a,b){var c=this,d=this.iterator("column",function(b,c){if(a===p)return b.aoColumns[c].bVisible;var d=b.aoColumns,e=d[c],h=b.aoData,n;if(a!==p&&e.bVisible!==a){if(a){var m=f.inArray(!0,J(d,"bVisible"),c+1);d=0;for(n=h.length;d<n;d++){var q=h[d].nTr;b=h[d].anCells;q&&q.insertBefore(b[c],b[m]||null)}}else f(J(b.aoData,"anCells",
c)).detach();e.bVisible=a}});a!==p&&this.iterator("table",function(d){ha(d,d.aoHeader);ha(d,d.aoFooter);d.aiDisplay.length||f(d.nTBody).find("td[colspan]").attr("colspan",W(d));Ba(d);c.iterator("column",function(c,d){A(c,null,"column-visibility",[c,d,a,b])});(b===p||b)&&c.columns.adjust()});return d});x("columns().indexes()","column().index()",function(a){return this.iterator("column",function(b,c){return"visible"===a?ca(b,c):c},1)});t("columns.adjust()",function(){return this.iterator("table",function(a){aa(a)},
1)});t("column.index()",function(a,b){if(0!==this.context.length){var c=this.context[0];if("fromVisible"===a||"toData"===a)return ba(c,b);if("fromData"===a||"toVisible"===a)return ca(c,b)}});t("column()",function(a,b){return gb(this.columns(a,b))});var lc=function(a,b,c){var d=a.aoData,e=Fa(a,c),h=Vb(la(d,e,"anCells")),g=f([].concat.apply([],h)),k,l=a.aoColumns.length,n,m,q,u,t,v;return eb("cell",b,function(b){var c="function"===typeof b;if(null===b||b===p||c){n=[];m=0;for(q=e.length;m<q;m++)for(k=
e[m],u=0;u<l;u++)t={row:k,column:u},c?(v=d[k],b(t,I(a,k,u),v.anCells?v.anCells[u]:null)&&n.push(t)):n.push(t);return n}if(f.isPlainObject(b))return b.column!==p&&b.row!==p&&-1!==f.inArray(b.row,e)?[b]:[];c=g.filter(b).map(function(a,b){return{row:b._DT_CellIndex.row,column:b._DT_CellIndex.column}}).toArray();if(c.length||!b.nodeName)return c;v=f(b).closest("*[data-dt-row]");return v.length?[{row:v.data("dt-row"),column:v.data("dt-column")}]:[]},a,c)};t("cells()",function(a,b,c){f.isPlainObject(a)&&
(a.row===p?(c=a,a=null):(c=b,b=null));f.isPlainObject(b)&&(c=b,b=null);if(null===b||b===p)return this.iterator("table",function(b){return lc(b,a,fb(c))});var d=c?{page:c.page,order:c.order,search:c.search}:{},e=this.columns(b,d),h=this.rows(a,d),g,k,l,n;d=this.iterator("table",function(a,b){a=[];g=0;for(k=h[b].length;g<k;g++)for(l=0,n=e[b].length;l<n;l++)a.push({row:h[b][g],column:e[b][l]});return a},1);d=c&&c.selected?this.cells(d,c):d;f.extend(d.selector,{cols:b,rows:a,opts:c});return d});x("cells().nodes()",
"cell().node()",function(){return this.iterator("cell",function(a,b,c){return(a=a.aoData[b])&&a.anCells?a.anCells[c]:p},1)});t("cells().data()",function(){return this.iterator("cell",function(a,b,c){return I(a,b,c)},1)});x("cells().cache()","cell().cache()",function(a){a="search"===a?"_aFilterData":"_aSortData";return this.iterator("cell",function(b,c,d){return b.aoData[c][a][d]},1)});x("cells().render()","cell().render()",function(a){return this.iterator("cell",function(b,c,d){return I(b,c,d,a)},
1)});x("cells().indexes()","cell().index()",function(){return this.iterator("cell",function(a,b,c){return{row:b,column:c,columnVisible:ca(a,c)}},1)});x("cells().invalidate()","cell().invalidate()",function(a){return this.iterator("cell",function(b,c,d){ea(b,c,a,d)})});t("cell()",function(a,b,c){return gb(this.cells(a,b,c))});t("cell().data()",function(a){var b=this.context,c=this[0];if(a===p)return b.length&&c.length?I(b[0],c[0].row,c[0].column):p;ob(b[0],c[0].row,c[0].column,a);ea(b[0],c[0].row,
"data",c[0].column);return this});t("order()",function(a,b){var c=this.context;if(a===p)return 0!==c.length?c[0].aaSorting:p;"number"===typeof a?a=[[a,b]]:a.length&&!f.isArray(a[0])&&(a=Array.prototype.slice.call(arguments));return this.iterator("table",function(b){b.aaSorting=a.slice()})});t("order.listener()",function(a,b,c){return this.iterator("table",function(d){Qa(d,a,b,c)})});t("order.fixed()",function(a){if(!a){var b=this.context;b=b.length?b[0].aaSortingFixed:p;return f.isArray(b)?{pre:b}:
b}return this.iterator("table",function(b){b.aaSortingFixed=f.extend(!0,{},a)})});t(["columns().order()","column().order()"],function(a){var b=this;return this.iterator("table",function(c,d){var e=[];f.each(b[d],function(b,c){e.push([c,a])});c.aaSorting=e})});t("search()",function(a,b,c,d){var e=this.context;return a===p?0!==e.length?e[0].oPreviousSearch.sSearch:p:this.iterator("table",function(e){e.oFeatures.bFilter&&ia(e,f.extend({},e.oPreviousSearch,{sSearch:a+"",bRegex:null===b?!1:b,bSmart:null===
c?!0:c,bCaseInsensitive:null===d?!0:d}),1)})});x("columns().search()","column().search()",function(a,b,c,d){return this.iterator("column",function(e,h){var g=e.aoPreSearchCols;if(a===p)return g[h].sSearch;e.oFeatures.bFilter&&(f.extend(g[h],{sSearch:a+"",bRegex:null===b?!1:b,bSmart:null===c?!0:c,bCaseInsensitive:null===d?!0:d}),ia(e,e.oPreviousSearch,1))})});t("state()",function(){return this.context.length?this.context[0].oSavedState:null});t("state.clear()",function(){return this.iterator("table",
function(a){a.fnStateSaveCallback.call(a.oInstance,a,{})})});t("state.loaded()",function(){return this.context.length?this.context[0].oLoadedState:null});t("state.save()",function(){return this.iterator("table",function(a){Ba(a)})});q.versionCheck=q.fnVersionCheck=function(a){var b=q.version.split(".");a=a.split(".");for(var c,d,e=0,f=a.length;e<f;e++)if(c=parseInt(b[e],10)||0,d=parseInt(a[e],10)||0,c!==d)return c>d;return!0};q.isDataTable=q.fnIsDataTable=function(a){var b=f(a).get(0),c=!1;if(a instanceof
q.Api)return!0;f.each(q.settings,function(a,e){a=e.nScrollHead?f("table",e.nScrollHead)[0]:null;var d=e.nScrollFoot?f("table",e.nScrollFoot)[0]:null;if(e.nTable===b||a===b||d===b)c=!0});return c};q.tables=q.fnTables=function(a){var b=!1;f.isPlainObject(a)&&(b=a.api,a=a.visible);var c=f.map(q.settings,function(b){if(!a||a&&f(b.nTable).is(":visible"))return b.nTable});return b?new v(c):c};q.camelToHungarian=L;t("$()",function(a,b){b=this.rows(b).nodes();b=f(b);return f([].concat(b.filter(a).toArray(),
b.find(a).toArray()))});f.each(["on","one","off"],function(a,b){t(b+"()",function(){var a=Array.prototype.slice.call(arguments);a[0]=f.map(a[0].split(/\s/),function(a){return a.match(/\.dt\b/)?a:a+".dt"}).join(" ");var d=f(this.tables().nodes());d[b].apply(d,a);return this})});t("clear()",function(){return this.iterator("table",function(a){qa(a)})});t("settings()",function(){return new v(this.context,this.context)});t("init()",function(){var a=this.context;return a.length?a[0].oInit:null});t("data()",
function(){return this.iterator("table",function(a){return J(a.aoData,"_aData")}).flatten()});t("destroy()",function(a){a=a||!1;return this.iterator("table",function(b){var c=b.nTableWrapper.parentNode,d=b.oClasses,e=b.nTable,h=b.nTBody,g=b.nTHead,k=b.nTFoot,l=f(e);h=f(h);var n=f(b.nTableWrapper),m=f.map(b.aoData,function(a){return a.nTr}),p;b.bDestroying=!0;A(b,"aoDestroyCallback","destroy",[b]);a||(new v(b)).columns().visible(!0);n.off(".DT").find(":not(tbody *)").off(".DT");f(z).off(".DT-"+b.sInstance);
e!=g.parentNode&&(l.children("thead").detach(),l.append(g));k&&e!=k.parentNode&&(l.children("tfoot").detach(),l.append(k));b.aaSorting=[];b.aaSortingFixed=[];Aa(b);f(m).removeClass(b.asStripeClasses.join(" "));f("th, td",g).removeClass(d.sSortable+" "+d.sSortableAsc+" "+d.sSortableDesc+" "+d.sSortableNone);h.children().detach();h.append(m);g=a?"remove":"detach";l[g]();n[g]();!a&&c&&(c.insertBefore(e,b.nTableReinsertBefore),l.css("width",b.sDestroyWidth).removeClass(d.sTable),(p=b.asDestroyStripes.length)&&
h.children().each(function(a){f(this).addClass(b.asDestroyStripes[a%p])}));c=f.inArray(b,q.settings);-1!==c&&q.settings.splice(c,1)})});f.each(["column","row","cell"],function(a,b){t(b+"s().every()",function(a){var c=this.selector.opts,e=this;return this.iterator(b,function(d,f,k,l,n){a.call(e[b](f,"cell"===b?k:c,"cell"===b?c:p),f,k,l,n)})})});t("i18n()",function(a,b,c){var d=this.context[0];a=U(a)(d.oLanguage);a===p&&(a=b);c!==p&&f.isPlainObject(a)&&(a=a[c]!==p?a[c]:a._);return a.replace("%d",c)});
q.version="1.10.20";q.settings=[];q.models={};q.models.oSearch={bCaseInsensitive:!0,sSearch:"",bRegex:!1,bSmart:!0};q.models.oRow={nTr:null,anCells:null,_aData:[],_aSortData:null,_aFilterData:null,_sFilterRow:null,_sRowStripe:"",src:null,idx:-1};q.models.oColumn={idx:null,aDataSort:null,asSorting:null,bSearchable:null,bSortable:null,bVisible:null,_sManualType:null,_bAttrSrc:!1,fnCreatedCell:null,fnGetData:null,fnSetData:null,mData:null,mRender:null,nTh:null,nTf:null,sClass:null,sContentPadding:null,
sDefaultContent:null,sName:null,sSortDataType:"std",sSortingClass:null,sSortingClassJUI:null,sTitle:null,sType:null,sWidth:null,sWidthOrig:null};q.defaults={aaData:null,aaSorting:[[0,"asc"]],aaSortingFixed:[],ajax:null,aLengthMenu:[10,25,50,100],aoColumns:null,aoColumnDefs:null,aoSearchCols:[],asStripeClasses:null,bAutoWidth:!0,bDeferRender:!1,bDestroy:!1,bFilter:!0,bInfo:!0,bLengthChange:!0,bPaginate:!0,bProcessing:!1,bRetrieve:!1,bScrollCollapse:!1,bServerSide:!1,bSort:!0,bSortMulti:!0,bSortCellsTop:!1,
bSortClasses:!0,bStateSave:!1,fnCreatedRow:null,fnDrawCallback:null,fnFooterCallback:null,fnFormatNumber:function(a){return a.toString().replace(/\B(?=(\d{3})+(?!\d))/g,this.oLanguage.sThousands)},fnHeaderCallback:null,fnInfoCallback:null,fnInitComplete:null,fnPreDrawCallback:null,fnRowCallback:null,fnServerData:null,fnServerParams:null,fnStateLoadCallback:function(a){try{return JSON.parse((-1===a.iStateDuration?sessionStorage:localStorage).getItem("DataTables_"+a.sInstance+"_"+location.pathname))}catch(b){}},
fnStateLoadParams:null,fnStateLoaded:null,fnStateSaveCallback:function(a,b){try{(-1===a.iStateDuration?sessionStorage:localStorage).setItem("DataTables_"+a.sInstance+"_"+location.pathname,JSON.stringify(b))}catch(c){}},fnStateSaveParams:null,iStateDuration:7200,iDeferLoading:null,iDisplayLength:10,iDisplayStart:0,iTabIndex:0,oClasses:{},oLanguage:{oAria:{sSortAscending:": activate to sort column ascending",sSortDescending:": activate to sort column descending"},oPaginate:{sFirst:"First",sLast:"Last",
sNext:"Next",sPrevious:"Previous"},sEmptyTable:"No data available in table",sInfo:"Showing _START_ to _END_ of _TOTAL_ entries",sInfoEmpty:"Showing 0 to 0 of 0 entries",sInfoFiltered:"(filtered from _MAX_ total entries)",sInfoPostFix:"",sDecimal:"",sThousands:",",sLengthMenu:"Show _MENU_ entries",sLoadingRecords:"Loading...",sProcessing:"Processing...",sSearch:"Search:",sSearchPlaceholder:"",sUrl:"",sZeroRecords:"No matching records found"},oSearch:f.extend({},q.models.oSearch),sAjaxDataProp:"data",
sAjaxSource:null,sDom:"lfrtip",searchDelay:null,sPaginationType:"simple_numbers",sScrollX:"",sScrollXInner:"",sScrollY:"",sServerMethod:"GET",renderer:null,rowId:"DT_RowId"};H(q.defaults);q.defaults.column={aDataSort:null,iDataSort:-1,asSorting:["asc","desc"],bSearchable:!0,bSortable:!0,bVisible:!0,fnCreatedCell:null,mData:null,mRender:null,sCellType:"td",sClass:"",sContentPadding:"",sDefaultContent:null,sName:"",sSortDataType:"std",sTitle:null,sType:null,sWidth:null};H(q.defaults.column);q.models.oSettings=
{oFeatures:{bAutoWidth:null,bDeferRender:null,bFilter:null,bInfo:null,bLengthChange:null,bPaginate:null,bProcessing:null,bServerSide:null,bSort:null,bSortMulti:null,bSortClasses:null,bStateSave:null},oScroll:{bCollapse:null,iBarWidth:0,sX:null,sXInner:null,sY:null},oLanguage:{fnInfoCallback:null},oBrowser:{bScrollOversize:!1,bScrollbarLeft:!1,bBounding:!1,barWidth:0},ajax:null,aanFeatures:[],aoData:[],aiDisplay:[],aiDisplayMaster:[],aIds:{},aoColumns:[],aoHeader:[],aoFooter:[],oPreviousSearch:{},
aoPreSearchCols:[],aaSorting:null,aaSortingFixed:[],asStripeClasses:null,asDestroyStripes:[],sDestroyWidth:0,aoRowCallback:[],aoHeaderCallback:[],aoFooterCallback:[],aoDrawCallback:[],aoRowCreatedCallback:[],aoPreDrawCallback:[],aoInitComplete:[],aoStateSaveParams:[],aoStateLoadParams:[],aoStateLoaded:[],sTableId:"",nTable:null,nTHead:null,nTFoot:null,nTBody:null,nTableWrapper:null,bDeferLoading:!1,bInitialised:!1,aoOpenRows:[],sDom:null,searchDelay:null,sPaginationType:"two_button",iStateDuration:0,
aoStateSave:[],aoStateLoad:[],oSavedState:null,oLoadedState:null,sAjaxSource:null,sAjaxDataProp:null,bAjaxDataGet:!0,jqXHR:null,json:p,oAjaxData:p,fnServerData:null,aoServerParams:[],sServerMethod:null,fnFormatNumber:null,aLengthMenu:null,iDraw:0,bDrawing:!1,iDrawError:-1,_iDisplayLength:10,_iDisplayStart:0,_iRecordsTotal:0,_iRecordsDisplay:0,oClasses:{},bFiltered:!1,bSorted:!1,bSortCellsTop:null,oInit:null,aoDestroyCallback:[],fnRecordsTotal:function(){return"ssp"==D(this)?1*this._iRecordsTotal:
this.aiDisplayMaster.length},fnRecordsDisplay:function(){return"ssp"==D(this)?1*this._iRecordsDisplay:this.aiDisplay.length},fnDisplayEnd:function(){var a=this._iDisplayLength,b=this._iDisplayStart,c=b+a,d=this.aiDisplay.length,e=this.oFeatures,f=e.bPaginate;return e.bServerSide?!1===f||-1===a?b+d:Math.min(b+a,this._iRecordsDisplay):!f||c>d||-1===a?d:c},oInstance:null,sInstance:null,iTabIndex:0,nScrollHead:null,nScrollFoot:null,aLastSort:[],oPlugins:{},rowIdFn:null,rowId:null};q.ext=C={buttons:{},
classes:{},builder:"-source-",errMode:"alert",feature:[],search:[],selector:{cell:[],column:[],row:[]},internal:{},legacy:{ajax:null},pager:{},renderer:{pageButton:{},header:{}},order:{},type:{detect:[],search:{},order:{}},_unique:0,fnVersionCheck:q.fnVersionCheck,iApiIndex:0,oJUIClasses:{},sVersion:q.version};f.extend(C,{afnFiltering:C.search,aTypes:C.type.detect,ofnSearch:C.type.search,oSort:C.type.order,afnSortData:C.order,aoFeatures:C.feature,oApi:C.internal,oStdClasses:C.classes,oPagination:C.pager});
f.extend(q.ext.classes,{sTable:"dataTable",sNoFooter:"no-footer",sPageButton:"paginate_button",sPageButtonActive:"current",sPageButtonDisabled:"disabled",sStripeOdd:"odd",sStripeEven:"even",sRowEmpty:"dataTables_empty",sWrapper:"dataTables_wrapper",sFilter:"dataTables_filter",sInfo:"dataTables_info",sPaging:"dataTables_paginate paging_",sLength:"dataTables_length",sProcessing:"dataTables_processing",sSortAsc:"sorting_asc",sSortDesc:"sorting_desc",sSortable:"sorting",sSortableAsc:"sorting_asc_disabled",
sSortableDesc:"sorting_desc_disabled",sSortableNone:"sorting_disabled",sSortColumn:"sorting_",sFilterInput:"",sLengthSelect:"",sScrollWrapper:"dataTables_scroll",sScrollHead:"dataTables_scrollHead",sScrollHeadInner:"dataTables_scrollHeadInner",sScrollBody:"dataTables_scrollBody",sScrollFoot:"dataTables_scrollFoot",sScrollFootInner:"dataTables_scrollFootInner",sHeaderTH:"",sFooterTH:"",sSortJUIAsc:"",sSortJUIDesc:"",sSortJUI:"",sSortJUIAscAllowed:"",sSortJUIDescAllowed:"",sSortJUIWrapper:"",sSortIcon:"",
sJUIHeader:"",sJUIFooter:""});var Pb=q.ext.pager;f.extend(Pb,{simple:function(a,b){return["previous","next"]},full:function(a,b){return["first","previous","next","last"]},numbers:function(a,b){return[ka(a,b)]},simple_numbers:function(a,b){return["previous",ka(a,b),"next"]},full_numbers:function(a,b){return["first","previous",ka(a,b),"next","last"]},first_last_numbers:function(a,b){return["first",ka(a,b),"last"]},_numbers:ka,numbers_length:7});f.extend(!0,q.ext.renderer,{pageButton:{_:function(a,b,
c,d,e,h){var g=a.oClasses,k=a.oLanguage.oPaginate,l=a.oLanguage.oAria.paginate||{},n,m,q=0,t=function(b,d){var p,r=g.sPageButtonDisabled,u=function(b){Xa(a,b.data.action,!0)};var w=0;for(p=d.length;w<p;w++){var v=d[w];if(f.isArray(v)){var x=f("<"+(v.DT_el||"div")+"/>").appendTo(b);t(x,v)}else{n=null;m=v;x=a.iTabIndex;switch(v){case "ellipsis":b.append('<span class="ellipsis">&#x2026;</span>');break;case "first":n=k.sFirst;0===e&&(x=-1,m+=" "+r);break;case "previous":n=k.sPrevious;0===e&&(x=-1,m+=
" "+r);break;case "next":n=k.sNext;e===h-1&&(x=-1,m+=" "+r);break;case "last":n=k.sLast;e===h-1&&(x=-1,m+=" "+r);break;default:n=v+1,m=e===v?g.sPageButtonActive:""}null!==n&&(x=f("<a>",{"class":g.sPageButton+" "+m,"aria-controls":a.sTableId,"aria-label":l[v],"data-dt-idx":q,tabindex:x,id:0===c&&"string"===typeof v?a.sTableId+"_"+v:null}).html(n).appendTo(b),$a(x,{action:v},u),q++)}}};try{var v=f(b).find(y.activeElement).data("dt-idx")}catch(mc){}t(f(b).empty(),d);v!==p&&f(b).find("[data-dt-idx="+
v+"]").focus()}}});f.extend(q.ext.type.detect,[function(a,b){b=b.oLanguage.sDecimal;return db(a,b)?"num"+b:null},function(a,b){if(a&&!(a instanceof Date)&&!cc.test(a))return null;b=Date.parse(a);return null!==b&&!isNaN(b)||P(a)?"date":null},function(a,b){b=b.oLanguage.sDecimal;return db(a,b,!0)?"num-fmt"+b:null},function(a,b){b=b.oLanguage.sDecimal;return Ub(a,b)?"html-num"+b:null},function(a,b){b=b.oLanguage.sDecimal;return Ub(a,b,!0)?"html-num-fmt"+b:null},function(a,b){return P(a)||"string"===
typeof a&&-1!==a.indexOf("<")?"html":null}]);f.extend(q.ext.type.search,{html:function(a){return P(a)?a:"string"===typeof a?a.replace(Rb," ").replace(Ea,""):""},string:function(a){return P(a)?a:"string"===typeof a?a.replace(Rb," "):a}});var Da=function(a,b,c,d){if(0!==a&&(!a||"-"===a))return-Infinity;b&&(a=Tb(a,b));a.replace&&(c&&(a=a.replace(c,"")),d&&(a=a.replace(d,"")));return 1*a};f.extend(C.type.order,{"date-pre":function(a){a=Date.parse(a);return isNaN(a)?-Infinity:a},"html-pre":function(a){return P(a)?
"":a.replace?a.replace(/<.*?>/g,"").toLowerCase():a+""},"string-pre":function(a){return P(a)?"":"string"===typeof a?a.toLowerCase():a.toString?a.toString():""},"string-asc":function(a,b){return a<b?-1:a>b?1:0},"string-desc":function(a,b){return a<b?1:a>b?-1:0}});Ha("");f.extend(!0,q.ext.renderer,{header:{_:function(a,b,c,d){f(a.nTable).on("order.dt.DT",function(e,f,g,k){a===f&&(e=c.idx,b.removeClass(c.sSortingClass+" "+d.sSortAsc+" "+d.sSortDesc).addClass("asc"==k[e]?d.sSortAsc:"desc"==k[e]?d.sSortDesc:
c.sSortingClass))})},jqueryui:function(a,b,c,d){f("<div/>").addClass(d.sSortJUIWrapper).append(b.contents()).append(f("<span/>").addClass(d.sSortIcon+" "+c.sSortingClassJUI)).appendTo(b);f(a.nTable).on("order.dt.DT",function(e,f,g,k){a===f&&(e=c.idx,b.removeClass(d.sSortAsc+" "+d.sSortDesc).addClass("asc"==k[e]?d.sSortAsc:"desc"==k[e]?d.sSortDesc:c.sSortingClass),b.find("span."+d.sSortIcon).removeClass(d.sSortJUIAsc+" "+d.sSortJUIDesc+" "+d.sSortJUI+" "+d.sSortJUIAscAllowed+" "+d.sSortJUIDescAllowed).addClass("asc"==
k[e]?d.sSortJUIAsc:"desc"==k[e]?d.sSortJUIDesc:c.sSortingClassJUI))})}}});var ib=function(a){return"string"===typeof a?a.replace(/</g,"&lt;").replace(/>/g,"&gt;").replace(/"/g,"&quot;"):a};q.render={number:function(a,b,c,d,e){return{display:function(f){if("number"!==typeof f&&"string"!==typeof f)return f;var g=0>f?"-":"",h=parseFloat(f);if(isNaN(h))return ib(f);h=h.toFixed(c);f=Math.abs(h);h=parseInt(f,10);f=c?b+(f-h).toFixed(c).substring(2):"";return g+(d||"")+h.toString().replace(/\B(?=(\d{3})+(?!\d))/g,
a)+f+(e||"")}}},text:function(){return{display:ib,filter:ib}}};f.extend(q.ext.internal,{_fnExternApiFunc:Qb,_fnBuildAjax:va,_fnAjaxUpdate:qb,_fnAjaxParameters:zb,_fnAjaxUpdateDraw:Ab,_fnAjaxDataSrc:wa,_fnAddColumn:Ia,_fnColumnOptions:ma,_fnAdjustColumnSizing:aa,_fnVisibleToColumnIndex:ba,_fnColumnIndexToVisible:ca,_fnVisbleColumns:W,_fnGetColumns:oa,_fnColumnTypes:Ka,_fnApplyColumnDefs:nb,_fnHungarianMap:H,_fnCamelToHungarian:L,_fnLanguageCompat:Ga,_fnBrowserDetect:lb,_fnAddData:R,_fnAddTr:pa,_fnNodeToDataIndex:function(a,
b){return b._DT_RowIndex!==p?b._DT_RowIndex:null},_fnNodeToColumnIndex:function(a,b,c){return f.inArray(c,a.aoData[b].anCells)},_fnGetCellData:I,_fnSetCellData:ob,_fnSplitObjNotation:Na,_fnGetObjectDataFn:U,_fnSetObjectDataFn:Q,_fnGetDataMaster:Oa,_fnClearTable:qa,_fnDeleteIndex:ra,_fnInvalidate:ea,_fnGetRowElements:Ma,_fnCreateTr:La,_fnBuildHead:pb,_fnDrawHead:ha,_fnDraw:S,_fnReDraw:V,_fnAddOptionsHtml:sb,_fnDetectHeader:fa,_fnGetUniqueThs:ua,_fnFeatureHtmlFilter:ub,_fnFilterComplete:ia,_fnFilterCustom:Db,
_fnFilterColumn:Cb,_fnFilter:Bb,_fnFilterCreateSearch:Ta,_fnEscapeRegex:Ua,_fnFilterData:Eb,_fnFeatureHtmlInfo:xb,_fnUpdateInfo:Hb,_fnInfoMacros:Ib,_fnInitialise:ja,_fnInitComplete:xa,_fnLengthChange:Va,_fnFeatureHtmlLength:tb,_fnFeatureHtmlPaginate:yb,_fnPageChange:Xa,_fnFeatureHtmlProcessing:vb,_fnProcessingDisplay:K,_fnFeatureHtmlTable:wb,_fnScrollDraw:na,_fnApplyToChildren:N,_fnCalculateColumnWidths:Ja,_fnThrottle:Sa,_fnConvertToWidth:Jb,_fnGetWidestNode:Kb,_fnGetMaxLenString:Lb,_fnStringToCss:B,
_fnSortFlatten:Y,_fnSort:rb,_fnSortAria:Nb,_fnSortListener:Za,_fnSortAttachListener:Qa,_fnSortingClasses:Aa,_fnSortData:Mb,_fnSaveState:Ba,_fnLoadState:Ob,_fnSettingsFromNode:Ca,_fnLog:O,_fnMap:M,_fnBindAction:$a,_fnCallbackReg:E,_fnCallbackFire:A,_fnLengthOverflow:Wa,_fnRenderer:Ra,_fnDataSource:D,_fnRowAttributes:Pa,_fnExtend:ab,_fnCalculateEnd:function(){}});f.fn.dataTable=q;q.$=f;f.fn.dataTableSettings=q.settings;f.fn.dataTableExt=q.ext;f.fn.DataTable=function(a){return f(this).dataTable(a).api()};
f.each(q,function(a,b){f.fn.DataTable[a]=b});return f.fn.dataTable});
</script>
<style type="text/css">
.container-fluid.crosstalk-bscols {
margin-left: -30px;
margin-right: -30px;
white-space: normal;
}

body > .container-fluid.crosstalk-bscols {
margin-left: auto;
margin-right: auto;
}
.crosstalk-input-checkboxgroup .crosstalk-options-group .crosstalk-options-column {
display: inline-block;
padding-right: 12px;
vertical-align: top;
}
@media only screen and (max-width:480px) {
.crosstalk-input-checkboxgroup .crosstalk-options-group .crosstalk-options-column {
display: block;
padding-right: inherit;
}
}
</style>
<script>!function o(u,a,l){function s(n,e){if(!a[n]){if(!u[n]){var t="function"==typeof require&&require;if(!e&&t)return t(n,!0);if(f)return f(n,!0);var r=new Error("Cannot find module '"+n+"'");throw r.code="MODULE_NOT_FOUND",r}var i=a[n]={exports:{}};u[n][0].call(i.exports,function(e){var t=u[n][1][e];return s(t||e)},i,i.exports,o,u,a,l)}return a[n].exports}for(var f="function"==typeof require&&require,e=0;e<l.length;e++)s(l[e]);return s}({1:[function(e,t,n){"use strict";Object.defineProperty(n,"__esModule",{value:!0});var r=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}();var i=function(){function e(){!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,e),this._types={},this._seq=0}return r(e,[{key:"on",value:function(e,t){var n=this._types[e];n||(n=this._types[e]={});var r="sub"+this._seq++;return n[r]=t,r}},{key:"off",value:function(e,t){var n=this._types[e];if("function"==typeof t){for(var r in n)if(n.hasOwnProperty(r)&&n[r]===t)return delete n[r],r;return!1}if("string"==typeof t)return!(!n||!n[t])&&(delete n[t],t);throw new Error("Unexpected type for listener")}},{key:"trigger",value:function(e,t,n){var r=this._types[e];for(var i in r)r.hasOwnProperty(i)&&r[i].call(n,t)}}]),e}();n.default=i},{}],2:[function(e,t,n){"use strict";Object.defineProperty(n,"__esModule",{value:!0}),n.FilterHandle=void 0;var r=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}(),i=l(e("./events")),o=l(e("./filterset")),u=l(e("./group")),a=function(e){{if(e&&e.__esModule)return e;var t={};if(null!=e)for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n]);return t.default=e,t}}(e("./util"));function l(e){return e&&e.__esModule?e:{default:e}}var s=1;n.FilterHandle=function(){function n(e,t){!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,n),this._eventRelay=new i.default,this._emitter=new a.SubscriptionTracker(this._eventRelay),this._group=null,this._filterSet=null,this._filterVar=null,this._varOnChangeSub=null,this._extraInfo=a.extend({sender:this},t),this._id="filter"+s++,this.setGroup(e)}return r(n,[{key:"setGroup",value:function(e){var t,n,r=this;if(this._group!==e&&((this._group||e)&&(this._filterVar&&(this._filterVar.off("change",this._varOnChangeSub),this.clear(),this._varOnChangeSub=null,this._filterVar=null,this._filterSet=null),this._group=e))){e=(0,u.default)(e),this._filterSet=(t=e.var("filterset"),(n=t.get())||(n=new o.default,t.set(n)),n),this._filterVar=(0,u.default)(e).var("filter");var i=this._filterVar.on("change",function(e){r._eventRelay.trigger("change",e,r)});this._varOnChangeSub=i}}},{key:"_mergeExtraInfo",value:function(e){return a.extend({},this._extraInfo?this._extraInfo:null,e||null)}},{key:"close",value:function(){this._emitter.removeAllListeners(),this.clear(),this.setGroup(null)}},{key:"clear",value:function(e){this._filterSet&&(this._filterSet.clear(this._id),this._onChange(e))}},{key:"set",value:function(e,t){this._filterSet&&(this._filterSet.update(this._id,e),this._onChange(t))}},{key:"on",value:function(e,t){return this._emitter.on(e,t)}},{key:"off",value:function(e,t){return this._emitter.off(e,t)}},{key:"_onChange",value:function(e){this._filterSet&&this._filterVar.set(this._filterSet.value,this._mergeExtraInfo(e))}},{key:"filteredKeys",get:function(){return this._filterSet?this._filterSet.value:null}}]),n}()},{"./events":1,"./filterset":3,"./group":4,"./util":11}],3:[function(e,t,n){"use strict";Object.defineProperty(n,"__esModule",{value:!0});var r=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}(),a=e("./util");function l(e,t){return e===t?0:e<t?-1:t<e?1:void 0}var i=function(){function e(){!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,e),this.reset()}return r(e,[{key:"reset",value:function(){this._handles={},this._keys={},this._value=null,this._activeHandles=0}},{key:"update",value:function(e,t){null!==t&&(t=t.slice(0)).sort(l);var n=(0,a.diffSortedLists)(this._handles[e],t),r=n.added,i=n.removed;this._handles[e]=t;for(var o=0;o<r.length;o++)this._keys[r[o]]=(this._keys[r[o]]||0)+1;for(var u=0;u<i.length;u++)this._keys[i[u]]--;this._updateValue(t)}},{key:"_updateValue",value:function(){var e=0<arguments.length&&void 0!==arguments[0]?arguments[0]:this._allKeys,t=Object.keys(this._handles).length;if(0===t)this._value=null;else{this._value=[];for(var n=0;n<e.length;n++){this._keys[e[n]]===t&&this._value.push(e[n])}}}},{key:"clear",value:function(e){if(void 0!==this._handles[e]){var t=this._handles[e];t||(t=[]);for(var n=0;n<t.length;n++)this._keys[t[n]]--;delete this._handles[e],this._updateValue()}}},{key:"value",get:function(){return this._value}},{key:"_allKeys",get:function(){var e=Object.keys(this._keys);return e.sort(l),e}}]),e}();n.default=i},{"./util":11}],4:[function(l,e,s){(function(e){"use strict";Object.defineProperty(s,"__esModule",{value:!0});var n=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}(),r="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(e){return typeof e}:function(e){return e&&"function"==typeof Symbol&&e.constructor===Symbol&&e!==Symbol.prototype?"symbol":typeof e};s.default=function e(t){{if(t&&"string"==typeof t)return u.hasOwnProperty(t)||(u[t]=new a(t)),u[t];if("object"===(void 0===t?"undefined":r(t))&&t._vars&&t.var)return t;if(Array.isArray(t)&&1==t.length&&"string"==typeof t[0])return e(t[0]);throw new Error("Invalid groupName argument")}};var t,i=l("./var"),o=(t=i)&&t.__esModule?t:{default:t};e.__crosstalk_groups=e.__crosstalk_groups||{};var u=e.__crosstalk_groups;var a=function(){function t(e){!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,t),this.name=e,this._vars={}}return n(t,[{key:"var",value:function(e){if(!e||"string"!=typeof e)throw new Error("Invalid var name");return this._vars.hasOwnProperty(e)||(this._vars[e]=new o.default(this,e)),this._vars[e]}},{key:"has",value:function(e){if(!e||"string"!=typeof e)throw new Error("Invalid var name");return this._vars.hasOwnProperty(e)}}]),t}()}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{"./var":12}],5:[function(f,e,c){(function(e){"use strict";Object.defineProperty(c,"__esModule",{value:!0});var t,n=f("./group"),r=(t=n)&&t.__esModule?t:{default:t},i=f("./selection"),o=f("./filter"),u=f("./input");f("./input_selectize"),f("./input_checkboxgroup"),f("./input_slider");var a=(0,r.default)("default");function l(e){return a.var(e)}e.Shiny&&e.Shiny.addCustomMessageHandler("update-client-value",function(e){"string"==typeof e.group?(0,r.default)(e.group).var(e.name).set(e.value):l(e.name).set(e.value)});var s={group:r.default,var:l,has:function(e){return a.has(e)},SelectionHandle:i.SelectionHandle,FilterHandle:o.FilterHandle,bind:u.bind};c.default=s,e.crosstalk=s}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{"./filter":2,"./group":4,"./input":6,"./input_checkboxgroup":7,"./input_selectize":8,"./input_slider":9,"./selection":10}],6:[function(e,t,a){(function(t){"use strict";Object.defineProperty(a,"__esModule",{value:!0}),a.register=function(e){r[e.className]=e,t.document&&"complete"!==t.document.readyState?o(function(){n()}):t.document&&setTimeout(n,100)},a.bind=n;var o=t.jQuery,r={};function n(){Object.keys(r).forEach(function(e){var n=r[e];o("."+n.className).not(".crosstalk-input-bound").each(function(e,t){i(n,t)})})}function i(e,t){var n=o(t).find("script[type='application/json'][data-for='"+t.id.replace(/([!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~])/g,"\\$1")+"']"),r=JSON.parse(n[0].innerText),i=e.factory(t,r);o(t).data("crosstalk-instance",i),o(t).addClass("crosstalk-input-bound")}if(t.Shiny){var e=new t.Shiny.InputBinding,u=t.jQuery;u.extend(e,{find:function(e){return u(e).find(".crosstalk-input")},initialize:function(e){var t,n;u(e).hasClass("crosstalk-input-bound")||(n=o(t=e),Object.keys(r).forEach(function(e){n.hasClass(e)&&!n.hasClass("crosstalk-input-bound")&&i(r[e],t)}))},getId:function(e){return e.id},getValue:function(e){},setValue:function(e,t){},receiveMessage:function(e,t){},subscribe:function(e,t){u(e).data("crosstalk-instance").resume()},unsubscribe:function(e){u(e).data("crosstalk-instance").suspend()}}),t.Shiny.inputBindings.register(e,"crosstalk.inputBinding")}}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{}],7:[function(r,e,t){(function(e){"use strict";var t=function(e){{if(e&&e.__esModule)return e;var t={};if(null!=e)for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n]);return t.default=e,t}}(r("./input")),n=r("./filter");var a=e.jQuery;t.register({className:"crosstalk-input-checkboxgroup",factory:function(e,r){var i=new n.FilterHandle(r.group),o=void 0,u=a(e);return u.on("change","input[type='checkbox']",function(){var e=u.find("input[type='checkbox']:checked");if(0===e.length)o=null,i.clear();else{var t={};e.each(function(){r.map[this.value].forEach(function(e){t[e]=!0})});var n=Object.keys(t);n.sort(),o=n,i.set(n)}}),{suspend:function(){i.clear()},resume:function(){o&&i.set(o)}}}})}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{"./filter":2,"./input":6}],8:[function(r,e,t){(function(e){"use strict";var t=n(r("./input")),l=n(r("./util")),s=r("./filter");function n(e){if(e&&e.__esModule)return e;var t={};if(null!=e)for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n]);return t.default=e,t}var f=e.jQuery;t.register({className:"crosstalk-input-select",factory:function(e,n){var t=l.dataframeToD3(n.items),r={options:[{value:"",label:"(All)"}].concat(t),valueField:"value",labelField:"label",searchField:"label"},i=f(e).find("select")[0],o=f(i).selectize(r)[0].selectize,u=new s.FilterHandle(n.group),a=void 0;return o.on("change",function(){if(0===o.items.length)a=null,u.clear();else{var t={};o.items.forEach(function(e){n.map[e].forEach(function(e){t[e]=!0})});var e=Object.keys(t);e.sort(),a=e,u.set(e)}}),{suspend:function(){u.clear()},resume:function(){a&&u.set(a)}}}})}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{"./filter":2,"./input":6,"./util":11}],9:[function(n,e,t){(function(e){"use strict";var d=function(e,t){if(Array.isArray(e))return e;if(Symbol.iterator in Object(e))return function(e,t){var n=[],r=!0,i=!1,o=void 0;try{for(var u,a=e[Symbol.iterator]();!(r=(u=a.next()).done)&&(n.push(u.value),!t||n.length!==t);r=!0);}catch(e){i=!0,o=e}finally{try{!r&&a.return&&a.return()}finally{if(i)throw o}}return n}(e,t);throw new TypeError("Invalid attempt to destructure non-iterable instance")},t=function(e){{if(e&&e.__esModule)return e;var t={};if(null!=e)for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n]);return t.default=e,t}}(n("./input")),a=n("./filter");var v=e.jQuery,p=e.strftime;function y(e,t){for(var n=e.toString();n.length<t;)n="0"+n;return n}t.register({className:"crosstalk-input-slider",factory:function(e,l){var s=new a.FilterHandle(l.group),t={},f=v(e).find("input"),n=f.data("data-type"),r=f.data("time-format"),i=f.data("round"),o=void 0;if("date"===n)o=p.utc(),t.prettify=function(e){return o(r,new Date(e))};else if("datetime"===n){var u=f.data("timezone");o=u?p.timezone(u):p,t.prettify=function(e){return o(r,new Date(e))}}else"number"===n&&void 0!==i&&(t.prettify=function(e){var t=Math.pow(10,i);return Math.round(e*t)/t});function c(){var e=f.data("ionRangeSlider").result,t=void 0,n=f.data("data-type");return t="date"===n?function(e){return(t=new Date(+e))instanceof Date?t.getUTCFullYear()+"-"+y(t.getUTCMonth()+1,2)+"-"+y(t.getUTCDate(),2):null;var t}:"datetime"===n?function(e){return+e/1e3}:function(e){return+e},"double"===f.data("ionRangeSlider").options.type?[t(e.from),t(e.to)]:t(e.from)}f.ionRangeSlider(t);var h=null;return f.on("change.crosstalkSliderInput",function(e){if(!f.data("updating")&&!f.data("animating")){for(var t=c(),n=d(t,2),r=n[0],i=n[1],o=[],u=0;u<l.values.length;u++){var a=l.values[u];r<=a&&a<=i&&o.push(l.keys[u])}o.sort(),s.set(o),h=o}}),{suspend:function(){s.clear()},resume:function(){h&&s.set(h)}}}})}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{"./filter":2,"./input":6}],10:[function(e,t,n){"use strict";Object.defineProperty(n,"__esModule",{value:!0}),n.SelectionHandle=void 0;var r=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}(),i=a(e("./events")),o=a(e("./group")),u=function(e){{if(e&&e.__esModule)return e;var t={};if(null!=e)for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n]);return t.default=e,t}}(e("./util"));function a(e){return e&&e.__esModule?e:{default:e}}n.SelectionHandle=function(){function n(){var e=0<arguments.length&&void 0!==arguments[0]?arguments[0]:null,t=1<arguments.length&&void 0!==arguments[1]?arguments[1]:null;!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,n),this._eventRelay=new i.default,this._emitter=new u.SubscriptionTracker(this._eventRelay),this._group=null,this._var=null,this._varOnChangeSub=null,this._extraInfo=u.extend({sender:this},t),this.setGroup(e)}return r(n,[{key:"setGroup",value:function(e){var t=this;if(this._group!==e&&(this._group||e)&&(this._var&&(this._var.off("change",this._varOnChangeSub),this._var=null,this._varOnChangeSub=null),this._group=e)){this._var=(0,o.default)(e).var("selection");var n=this._var.on("change",function(e){t._eventRelay.trigger("change",e,t)});this._varOnChangeSub=n}}},{key:"_mergeExtraInfo",value:function(e){return u.extend({},this._extraInfo?this._extraInfo:null,e||null)}},{key:"set",value:function(e,t){this._var&&this._var.set(e,this._mergeExtraInfo(t))}},{key:"clear",value:function(e){this._var&&this.set(void 0,this._mergeExtraInfo(e))}},{key:"on",value:function(e,t){return this._emitter.on(e,t)}},{key:"off",value:function(e,t){return this._emitter.off(e,t)}},{key:"close",value:function(){this._emitter.removeAllListeners(),this.setGroup(null)}},{key:"value",get:function(){return this._var?this._var.get():null}}]),n}()},{"./events":1,"./group":4,"./util":11}],11:[function(e,t,n){"use strict";Object.defineProperty(n,"__esModule",{value:!0});var r=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}(),l="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(e){return typeof e}:function(e){return e&&"function"==typeof Symbol&&e.constructor===Symbol&&e!==Symbol.prototype?"symbol":typeof e};function u(e){for(var t=1;t<e.length;t++)if(e[t]<=e[t-1])throw new Error("List is not sorted or contains duplicate")}n.extend=function(e){for(var t=arguments.length,n=Array(1<t?t-1:0),r=1;r<t;r++)n[r-1]=arguments[r];for(var i=0;i<n.length;i++){var o=n[i];if(null!=o)for(var u in o)o.hasOwnProperty(u)&&(e[u]=o[u])}return e},n.checkSorted=u,n.diffSortedLists=function(e,t){var n=0,r=0;e||(e=[]);t||(t=[]);var i=[],o=[];u(e),u(t);for(;n<e.length&&r<t.length;)e[n]===t[r]?(n++,r++):e[n]<t[r]?i.push(e[n++]):o.push(t[r++]);n<e.length&&(i=i.concat(e.slice(n)));r<t.length&&(o=o.concat(t.slice(r)));return{removed:i,added:o}},n.dataframeToD3=function(e){var t=[],n=void 0;for(var r in e){if(e.hasOwnProperty(r)&&t.push(r),"object"!==l(e[r])||void 0===e[r].length)throw new Error("All fields must be arrays");if(void 0!==n&&n!==e[r].length)throw new Error("All fields must be arrays of the same length");n=e[r].length}for(var i=[],o=void 0,u=0;u<n;u++){o={};for(var a=0;a<t.length;a++)o[t[a]]=e[t[a]][u];i.push(o)}return i};n.SubscriptionTracker=function(){function t(e){!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,t),this._emitter=e,this._subs={}}return r(t,[{key:"on",value:function(e,t){var n=this._emitter.on(e,t);return this._subs[n]=e,n}},{key:"off",value:function(e,t){var n=this._emitter.off(e,t);return n&&delete this._subs[n],n}},{key:"removeAllListeners",value:function(){var t=this,n=this._subs;this._subs={},Object.keys(n).forEach(function(e){t._emitter.off(n[e],e)})}}]),t}()},{}],12:[function(a,e,l){(function(o){"use strict";Object.defineProperty(l,"__esModule",{value:!0});var e,u="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(e){return typeof e}:function(e){return e&&"function"==typeof Symbol&&e.constructor===Symbol&&e!==Symbol.prototype?"symbol":typeof e},t=function(){function r(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(e,t,n){return t&&r(e.prototype,t),n&&r(e,n),e}}(),n=a("./events"),i=(e=n)&&e.__esModule?e:{default:e};var r=function(){function r(e,t,n){!function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,r),this._group=e,this._name=t,this._value=n,this._events=new i.default}return t(r,[{key:"get",value:function(){return this._value}},{key:"set",value:function(e,t){if(this._value!==e){var n=this._value;this._value=e;var r={};if(t&&"object"===(void 0===t?"undefined":u(t)))for(var i in t)t.hasOwnProperty(i)&&(r[i]=t[i]);r.oldValue=n,r.value=e,this._events.trigger("change",r,this),o.Shiny&&o.Shiny.onInputChange&&o.Shiny.onInputChange(".clientValue-"+(null!==this._group.name?this._group.name+"-":"")+this._name,void 0===e?null:e)}}},{key:"on",value:function(e,t){return this._events.on(e,t)}},{key:"off",value:function(e,t){return this._events.off(e,t)}}]),r}();l.default=r}).call(this,"undefined"!=typeof global?global:"undefined"!=typeof self?self:"undefined"!=typeof window?window:{})},{"./events":1}]},{},[5]);
//# sourceMappingURL=crosstalk.min.js.map</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type="text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">Selection of R-packages for review</h1>
<h4 class="author">Emi Tanaka</h4>
<h4 class="date">24/03/2021</h4>

</div>


<style type="text/css">
.tooltip {
  position: relative;
  display: inline-block;
  border-bottom: 1px dotted black;
  opacity: 1;
}

.tooltip .tooltiptext {
  visibility: hidden;
  width: 120px;
  background-color: black;
  color: #fff;
  text-align: center;
  border-radius: 6px;
  padding: 5px 0;

  /* Position the tooltip */
  position: absolute;
  z-index: 1;
}

.tooltip:hover .tooltiptext {
  visibility: visible;
}
</style>
<div id="results" class="section level2 tabset">
<h2 class="tabset">Results</h2>
<div id="adaptive" class="section level3">
<h3>Adaptive</h3>
<div id="htmlwidget-340bb5902321e8dda338" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-340bb5902321e8dda338">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37"],["<a href='https://cran.r-project.org/web/packages/activegp/index.html'>activegp<\/a>","<a href='https://cran.r-project.org/web/packages/ADCT/index.html'>ADCT<\/a>","<a href='https://cran.r-project.org/web/packages/adpss/index.html'>adpss<\/a>","<a href='https://cran.r-project.org/web/packages/BayesCTDesign/index.html'>BayesCTDesign<\/a>","<a href='https://cran.r-project.org/web/packages/BDEsize/index.html'>BDEsize<\/a>","<a href='https://cran.r-project.org/web/packages/BDP2/index.html'>BDP2<\/a>","<a href='https://cran.r-project.org/web/packages/blockRAR/index.html'>blockRAR<\/a>","<a href='https://cran.r-project.org/web/packages/blocksdesign/index.html'>blocksdesign<\/a>","<a href='https://cran.r-project.org/web/packages/blrm/index.html'>blrm<\/a>","<a href='https://cran.r-project.org/web/packages/clinfun/index.html'>clinfun<\/a>","<a href='https://cran.r-project.org/web/packages/CompAREdesign/index.html'>CompAREdesign<\/a>","<a href='https://cran.r-project.org/web/packages/dani/index.html'>dani<\/a>","<a href='https://cran.r-project.org/web/packages/dfcomb/index.html'>dfcomb<\/a>","<a href='https://cran.r-project.org/web/packages/dfmta/index.html'>dfmta<\/a>","<a href='https://cran.r-project.org/web/packages/earlygating/index.html'>earlygating<\/a>","<a href='https://cran.r-project.org/web/packages/easypower/index.html'>easypower<\/a>","<a href='https://cran.r-project.org/web/packages/GreedyExperimentalDesign/index.html'>GreedyExperimentalDesign<\/a>","<a href='https://cran.r-project.org/web/packages/iAdapt/index.html'>iAdapt<\/a>","<a href='https://cran.r-project.org/web/packages/MANOVA.RM/index.html'>MANOVA.RM<\/a>","<a href='https://cran.r-project.org/web/packages/MinEDfind/index.html'>MinEDfind<\/a>","<a href='https://cran.r-project.org/web/packages/nparMD/index.html'>nparMD<\/a>","<a href='https://cran.r-project.org/web/packages/OptimalRerandExpDesigns/index.html'>OptimalRerandExpDesigns<\/a>","<a href='https://cran.r-project.org/web/packages/pauwels2014/index.html'>pauwels2014<\/a>","<a href='https://cran.r-project.org/web/packages/ph2bye/index.html'>ph2bye<\/a>","<a href='https://cran.r-project.org/web/packages/ph2mult/index.html'>ph2mult<\/a>","<a href='https://cran.r-project.org/web/packages/radiant.design/index.html'>radiant.design<\/a>","<a href='https://cran.r-project.org/web/packages/rdpower/index.html'>rdpower<\/a>","<a href='https://cran.r-project.org/web/packages/rpact/index.html'>rpact<\/a>","<a href='https://cran.r-project.org/web/packages/seqDesign/index.html'>seqDesign<\/a>","<a href='https://cran.r-project.org/web/packages/seqmon/index.html'>seqmon<\/a>","<a href='https://cran.r-project.org/web/packages/SMARTp/index.html'>SMARTp<\/a>","<a href='https://cran.r-project.org/web/packages/soptdmaeA/index.html'>soptdmaeA<\/a>","<a href='https://cran.r-project.org/web/packages/SPCDAnalyze/index.html'>SPCDAnalyze<\/a>","<a href='https://cran.r-project.org/web/packages/ssize.fdr/index.html'>ssize.fdr<\/a>","<a href='https://cran.r-project.org/web/packages/SteppedPower/index.html'>SteppedPower<\/a>","<a href='https://cran.r-project.org/web/packages/tgp/index.html'>tgp<\/a>","<a href='https://cran.r-project.org/web/packages/toxtestD/index.html'>toxtestD<\/a>"],["Gaussian Process Based Design and Analysis for the Active Subspace Method","Adaptive Design in Clinical Trials","Design and Analysis of Locally or Globally Efficient Adaptive Designs","Two Arm Bayesian Clinical Trial Design with and Without Historical Control Data","Efficient Determination of Sample Size in Balanced Design of Experiments","Bayesian Adaptive Designs for Phase II Trials with Binary Endpoint","Block Design for Response-Adaptive Randomization","Nested and Crossed Block Designs for Factorial and Unstructured Treatment Sets","Dose Escalation Design in Phase I Oncology Trial Using Bayesian Logistic Regression Modeling","Clinical Trial Design and Data Analysis Functions","Statistical Functions for the Design of Studies with Composite Endpoints","Design and Analysis of Non-Inferiority Trials","Phase I/II Adaptive Dose-Finding Design for Combination Studies","Phase I/II Adaptive Dose-Finding Design for MTA","Properties of Bayesian Early Gating Designs","Sample Size Estimation for Experimental Designs","Greedy Experimental Design Construction","Two-Stage Adaptive Dose-Finding Clinical Trial Design","Resampling-Based Analysis of Multivariate Data and Repeated Measures Designs","A Bayesian Design for Minimum Effective Dosing-Finding Trial","Nonparametric Analysis of Multivariate Data in Factorial Designs","Optimal Rerandomization Experimental Designs","Bayesian Experimental Design for Systems Biology","Phase II Clinical Trial Design Using Bayesian Methods","Phase II Clinical Trial Design for Multinomial Endpoints","Design Menu for Radiant: Business Analytics using R and Shiny","Power Calculations for RD Designs","Confirmatory Adaptive Clinical Trial Design and Analysis","Simulation and Group Sequential Monitoring of Randomized Two-Stage Treatment Efficacy Trials with Time-to-Event Endpoints","Group Sequential Design Class for Clinical Trials","Sample Size for SMART Designs in Non-Surgical Periodontal Trials","Sequential Optimal Designs for Two-Colour cDNA Microarray Experiments","Design and Analyze Studies using the Sequential Parallel Comparison Design","Sample Size Calculations for Microarray Experiments","Power Calculation for Stepped Wedge Designs","Bayesian Treed Gaussian Process Models","Experimental design for binary toxicity tests"],["The active subspace method is a sensitivity analysis technique that finds important linear combinations of input variables for a simulator. This package provides functions allowing estimation of the active subspace without gradient information using Gaussian processes as well as sequential experimental design tools to minimize the amount of data required to do so. Implements Wycoff et al. (2019) <arXiv:1907.11572>.","Existing adaptive design methods in clinical trials. The package includes power, stopping boundaries (sample size) calculation functions for two-group group sequential designs, adaptive design with coprimary endpoints, biomarker-informed adaptive design, etc.","Provides the functions for planning and conducting a clinical trial with adaptive sample size determination. Maximal statistical efficiency will be exploited even when dramatic or multiple adaptations are made. Such a trial consists of adaptive determination of sample size at an interim analysis and implementation of frequentist statistical test at the interim and final analysis with a prefixed significance level. The required assumptions for the stage-wise test statistics are independent and stationary increments and normality. Predetermination of adaptation rule is not required.","A set of functions to help clinical trial researchers calculate power and sample size for two-arm Bayesian randomized clinical trials that do or do not incorporate historical control data. At some point during the design process, a clinical trial researcher who is designing a basic two-arm Bayesian randomized clinical trial needs to make decisions about power and sample size within the context of hypothesized treatment effects. Through simulation, the simple_sim() function will estimate power and other user specified clinical trial characteristics at user specified sample sizes given user defined scenarios about treatment effect,control group characteristics, and outcome. If the clinical trial researcher has access to historical control data, then the researcher can design a two-arm Bayesian randomized clinical trial that incorporates the historical data. In such a case, the researcher needs to work through the potential consequences of historical and randomized control differences on trial characteristics, in addition to working through issues regarding power in the context of sample size, treatment effect size, and outcome. If a researcher designs a clinical trial that will incorporate historical control data, the researcher needs the randomized controls to be from the same population as the historical controls. What if this is not the case when the designed trial is implemented? During the design phase, the researcher needs to investigate the negative effects of possible historic/randomized control differences on power, type one error, and other trial characteristics. Using this information, the researcher should design the trial to mitigate these negative effects. Through simulation, the historic_sim() function will estimate power and other user specified clinical trial characteristics at user specified sample sizes given user defined scenarios about historical and randomized control differences as well as treatment effects and outcomes. The results from historic_sim() and simple_sim() can be printed with print_table() and graphed with plot_table() methods. Outcomes considered are Gaussian, Poisson, Bernoulli, Lognormal, Weibull, and Piecewise Exponential.","Provides the sample size in balanced design of experiments and three graphs ; detectable standardized effect size vs power, sample size vs detectable standardized effect size, and sample size vs power. Sample size is computed in order to detect a certain standardized effect size with power at the significance level. Three graphs show the mutual relationship between the sample size, power and the detectable standardized effect size. By investigating those graphs, it can be checked that which effects are sensitive to the efficient sample size determination. Lenth,R.V.(2006-9) <http://www.stat.uiowa.edu/~rlenth/Power> Lim, Yong Bin (1998) Marvin, A., Kastenbaum, A. and Hoel, D.G. (1970) <doi:10.2307/2334851> Montgomery, Douglas C. (2013, ISBN: 0849323312).","Tools and workflow to choose design parameters in Bayesian adaptive single-arm phase II trial designs with binary endpoint (response, success) with possible stopping for efficacy and futility at interim analyses. Also contains routines to determine and visualize operating characteristics. See Kopp-Schneider et al. (2018) <doi:10.1002/bimj.201700209>.","Computes power for response-adaptive randomization with a block design that captures both the time and treatment effect. T. Chandereng, R. Chappell (2019) <arXiv:1904.07758>.","Constructs treatment and block designs for linear treatment models with crossed or nested block factors. The treatment design can be any feasible linear model and the block design can be any feasible combination of crossed or nested block factors. The block design is a sum of one or more block factors and the block design is optimized sequentially with the levels of each successive block factor optimized conditional on all previously optimized block factors. D-optimality is used throughout except for square or rectangular lattice block designs which are constructed algebraically using mutually orthogonal Latin squares. Crossed block designs with interaction effects are optimized using a weighting scheme which allows for differential weighting of first and second-order block effects. Outputs include a table showing the allocation of treatments to blocks and tables showing the achieved D-efficiency factors for each block and treatment design. Edmondson, R.N. Multi-level Block Designs for Comparative Experiments. JABES 25, 500–522 (2020) <doi:10.1007/s13253-020-00416-0>.","Plan dose escalation design using adaptive Bayesian logistic regression modeling in Phase I oncology trial.","Utilities to make your clinical collaborations easier if not fun. It contains functions for designing studies such as Simon 2-stage and group sequential designs and for data analysis such as Jonckheere-Terpstra test and estimating survival quantiles.","It has been designed to calculate the required sample size in randomized clinical trials with composite endpoints. This package also includes functions to calculate the probability of observing the composite endpoint and the expected effect on the composite endpoint, among others. The methods implemented can be found in Bofill & Gómez (2019) <doi:10.1002/sim.8092> and Gómez & Lagakos (2013) <doi:10.1002/sim.5547>.","Provides tools to help the design and analysis of resilient non-inferiority trials. These include functions for sample size calculations and analyses of trials, with either a risk difference, risk ratio or arc-sine difference margin, and a function to run simulations to design a trial with the methods described in Quartagno et al. (2019) <arXiv:1905.00241>.","Phase I/II adaptive dose-finding design for combination studies where toxicity rates are supposed to increase with both agents.","Phase I/II adaptive dose-finding design for single-agent Molecularly Targeted Agent (MTA), according to the paper \"Phase I/II Dose-Finding Design for Molecularly Targeted Agent: Plateau Determination using Adaptive Randomization\", Riviere Marie-Karelle et al. (2016) <doi:10.1177/0962280216631763>.","Computes the most important properties of four 'Bayesian' early gating designs (two single arm and two randomized controlled designs), such as minimum required number of successes in the experimental group to make a GO decision, operating characteristics and average operating characteristics with respect to the sample size. These might aid in deciding what design to use for the early phase trial.","Power analysis is used in the estimation of sample sizes for experimental designs. Most programs and R packages will only output the highest recommended sample size to the user. Often the user input can be complicated and computing multiple power analyses for different treatment comparisons can be time consuming. This package simplifies the user input and allows the user to view all of the sample size recommendations or just the ones they want to see. The calculations used to calculate the recommended sample sizes are from the 'pwr' package.","Computes experimental designs for a two-arm experiment with covariates via a number of methods. (0) complete randomization and randomization with forced-balance. (1) Greedily optimizing a balance objective function via pairwise switching. This optimization provides lower variance for the treatment effect estimator (and higher power) while preserving a design that is close to complete randomization. We return all iterations of the designs for use in a permutation test. (2) The second is via numerical optimization (via 'gurobi' which must be installed, see <https://www.gurobi.com/documentation/9.1/quickstart_windows/r_ins_the_r_package.html>) a la Bertsimas and Kallus. (3) rerandomization, (4) Karp's method for one covariate, (5) exhaustive enumeration to find the optimal solution (only for small sample sizes) (6) Binary pair matching using the 'nbpMatching' library (7) Binary pair matching plus (1) to further optimize balance (8) Binary pair matching plus (3) to further optimize balance (9) Hadamard designs We also allow for three objective functions: Mahalanobis distance, Sum of absolute differences standardized and Kernel distances via the 'kernlab' library.","Simulate and implement early phase two-stage adaptive dose-finding design for binary and quasi-continuous toxicity endpoints. See Chiuzan et al. (2018) for further reading <DOI:10.1080/19466315.2018.1462727>.","Implemented are various tests for semi-parametric repeated measures and general MANOVA designs that do neither assume multivariate normality nor covariance homogeneity, i.e., the procedures are applicable for a wide range of general multivariate factorial designs. In addition to asymptotic inference methods, novel bootstrap and permutation approaches are implemented as well. These provide more accurate results in case of small to moderate sample sizes. Furthermore, post-hoc comparisons are provided for the multivariate analyses. Friedrich, S., Konietschke, F. and Pauly, M. (2018) <arXiv:1801.08002>.","The nonparametric two-stage Bayesian adaptive design is a novel phase II clinical trial design for finding the minimum effective dose (MinED). This design is motivated by the top priority and concern of clinicians when testing a new drug, which is to effectively treat patients and minimize the chance of exposing them to subtherapeutic or overly toxic doses. It is used to design single-agent trials.","Analysis of multivariate data with two-way completely randomized factorial design. The analysis is based on fully nonparametric, rank-based methods and uses test statistics based on the Dempster's ANOVA, Wilk's Lambda, Lawley-Hotelling and Bartlett-Nanda-Pillai criteria. The multivariate response is allowed to be ordinal, quantitative, binary or a mixture of the different variable types. The package offers two functions performing the analysis, one for small and the other for large sample sizes. The underlying methodology is largely described in Bathke and Harrar (2016) <doi:10.1007/978-3-319-39065-9_7> and in Munzel and Brunner (2000) <doi:10.1016/S0378-3758(99)00212-8>.","This is a tool to find the optimal rerandomization threshold in non-sequential experiments. We offer three procedures based on assumptions made on the residuals distribution: (1) normality assumed (2) excess kurtosis assumed (3) entire distribution assumed. Illustrations are included. Also included is a routine to unbiasedly estimate Frobenius norms of variance-covariance matrices. Details of the method can be found in \"Optimal Rerandomization via a Criterion that Provides Insurance Against Failed Experiments\" Adam Kapelner, Abba M. Krieger, Michael Sklar and David Azriel (2020) <arXiv:1905.03337>.","Implementation of a Bayesian active learning strategy to carry out sequential experimental design in the context of biochemical network kinetic parameter estimation. This package gathers functions and pre-computed data sets to reproduce results presented in Pauwels E. et. al published in BMC Systems Biology, 2014. Scripts are given to compute all results from scratch or to draw pictures based on pre-computed data sets.","Calculate the Bayesian posterior/predictive probability and determine the sample size and stopping boundaries for single-arm Phase II design.","Provide multinomial design methods under intersection-union test (IUT) and union-intersection test (UIT) scheme for Phase II trial. The design types include : Minimax (minimize the maximum sample size), Optimal (minimize the expected sample size), Admissible (minimize the Bayesian risk) and Maxpower (maximize the exact power level).","The Radiant Design menu includes interfaces for design of experiments, sampling, and sample size calculation. The application extends the functionality in radiant.data.","The regression discontinuity (RD) design is a popular quasi-experimental design for causal inference and policy evaluation. The 'rdpower' package provides tools to perform power, sample size and MDE calculations in RD designs: rdpower() calculates the power of an RD design, rdsampsi() calculates the required sample size to achieve a desired power and rdmde() calculates minimum detectable effects. See Cattaneo, Titiunik and Vazquez-Bare (2019) <https://rdpackages.github.io/references/Cattaneo-Titiunik-VazquezBare_2019_Stata.pdf> for further methodological details.","Design and analysis of confirmatory adaptive clinical trials with continuous, binary, and survival endpoints according to the methods described in the monograph by Wassmer and Brannath (2016) <doi:10.1007/978-3-319-32562-0>. This includes classical group sequential as well as multi-stage adaptive hypotheses tests that are based on the combination testing principle.","A modification of the preventive vaccine efficacy trial design of Gilbert, Grove et al. (2011, Statistical Communications in Infectious Diseases) is implemented, with application generally to individual-randomized clinical trials with multiple active treatment groups and a shared control group, and a study endpoint that is a time-to-event endpoint subject to right-censoring. The design accounts for the issues that the efficacy of the treatment/vaccine groups may take time to accrue while the multiple treatment administrations/vaccinations are given; there is interest in assessing the durability of treatment efficacy over time; and group sequential monitoring of each treatment group for potential harm, non-efficacy/efficacy futility, and high efficacy is warranted. The design divides the trial into two stages of time periods, where each treatment is first evaluated for efficacy in the first stage of follow-up, and, if and only if it shows significant treatment efficacy in stage one, it is evaluated for longer-term durability of efficacy in stage two. The package produces plots and tables describing operating characteristics of a specified design including an unconditional power for intention-to-treat and per-protocol/as-treated analyses; trial duration; probabilities of the different possible trial monitoring outcomes (e.g., stopping early for non-efficacy); unconditional power for comparing treatment efficacies; and distributions of numbers of endpoint events occurring after the treatments/vaccinations are given, useful as input parameters for the design of studies of the association of biomarkers with a clinical outcome (surrogate endpoint problem). The code can be used for a single active treatment versus control design and for a single-stage design.","S4 class object for creating and managing group sequential designs. It calculates the efficacy and futility boundaries at each look. It allows modifying the design and tracking the design update history.","Sample size calculation to detect dynamic treatment regime (DTR) effects based on change in clinical attachment level (CAL) outcomes from a non-surgical chronic periodontitis treatments study. The experiment is performed under a Sequential Multiple Assignment Randomized Trial (SMART) design. The clustered tooth (sub-unit) level CAL outcomes are skewed, spatially-referenced, and non-randomly missing. The implemented algorithm is available in Xu et al. (2019+) <arXiv:1902.09386>.","Computes sequential A-, MV-, D- and E-optimal or near-optimal block and row-column designs for two-colour cDNA microarray experiments using the linear fixed effects and mixed effects models where the interest is in a comparison of all possible elementary treatment contrasts. The package also provides an optional method of using the graphical user interface (GUI) R package 'tcltk' to ensure that it is user friendly.","Programs to find the sample size or power of studies using the Sequential Parallel Comparison Design (SPCD) and programs to analyze such studies. This is a clinical trial design where patients initially on placebo who did not respond are re-randomized between placebo and active drug in a second phase and the results of the two phases are pooled. The method of analyzing binary data with this design is described in Fava,Evins, Dorer and Schoenfeld(2003) <doi:10.1159/000069738>, and the method of analyzing continuous data is described in Chen, Yang, Hung and Wang (2011) <doi:10.1016/j.cct.2011.04.006>.","This package contains a set of functions that calculates appropriate sample sizes for one-sample t-tests, two-sample t-tests, and F-tests for microarray experiments based on desired power while controlling for false discovery rates. For all tests, the standard deviations (variances) among genes can be assumed fixed or random. This is also true for effect sizes among genes in one-sample and two sample experiments. Functions also output a chart of power versus sample size, a table of power at different sample sizes, and a table of critical test values at different sample sizes.","Tools for power and sample size calculation as well as design diagnostics for longitudinal mixed model settings, with a focus on stepped wedge designs. All calculations are oracle estimates i.e. assume random effect variances to be known (or guessed) in advance. The method is introduced in Hussey and Hughes (2007) <doi:10.1016/j.cct.2006.05.007>, extensions are discussed in Li et al. (2020) <doi:10.1177/0962280220932962>.","Bayesian nonstationary, semiparametric nonlinear regression and design by treed Gaussian processes (GPs) with jumps to the limiting linear model (LLM). Special cases also implemented include Bayesian linear models, CART, treed linear models, stationary separable and isotropic GPs, and GP single-index models. Provides 1-d and 2-d plotting functions (with projection and slice capabilities) and tree drawing, designed for visualization of tgp-class output. Sensitivity analysis and multi-resolution models are supported. Sequential experimental design and adaptive sampling functions are also provided, including ALM, ALC, and expected improvement. The latter supports derivative-free optimization of noisy black-box functions.","Calculates sample size and dose allocation for binary toxicity tests, using the Fish Embryo Toxicity Test as example. An optimal test design is obtained by running (i) spoD (calculate the number of individuals to test under control conditions), (ii) setD (estimate the minimal sample size per treatment given the users precision requirements) and (iii) doseD (construct an individual dose scheme)."],["no","no","no","no","no","no","no","yes","no","no","no","no","yes","yes","no","yes","no","no","no","no","no","no","no","yes","no","no","no","no","yes","no","no","yes","no","yes","no","yes","yes"],["no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no"],["<div class='tooltip'>C_GP<span class='tooltiptext'>Computes the integral (uniform measure) over the [0,1] box domain of the outer product of the gradients of a Gaussian process. The corresponding matrix is the C matrix central in the active subspace methodology.<\/span><\/div>, <div class='tooltip'>C_GP_ci<span class='tooltiptext'>CI on Eigenvalues via Monte Carlo/GP<\/span><\/div>, <div class='tooltip'>C_GP_cpp<span class='tooltiptext'>Computes Int(kappa_i(X, design) . kappa_j(design, X)). This function is preferred for initialization<\/span><\/div>, <div class='tooltip'>C_Q<span class='tooltiptext'>Active subspace for second order linear model<\/span><\/div>, <div class='tooltip'>C_tr<span class='tooltiptext'>Expected variance of trace of C<\/span><\/div>, <div class='tooltip'>C_var<span class='tooltiptext'>Element-wise Cn+1 variance<\/span><\/div>, <div class='tooltip'>C_var2<span class='tooltiptext'>Defined as E[(C - E[C])^2], where A^2 = AA (not elementwise multiplication).<\/span><\/div>, <div class='tooltip'>domain_to_R<span class='tooltiptext'>Given an m dimensional function whose inputs live in bounded intervals [a1, b1], ..., [am, bm], return a wrapped version of the function whose inputs live in R^m. Transformed using the logit function.<\/span><\/div>, <div class='tooltip'>domain_to_unit<span class='tooltiptext'>Given an m dimensional function whose inputs live in bounded intervals [a1, b1], ..., [am, bm], return a wrapped version of the function whose inputs live in [-1, 1], ..., [-1, 1].<\/span><\/div>, <div class='tooltip'>grad_est_subspace<span class='tooltiptext'>Looks between [-1, 1]<\/span><\/div>, <div class='tooltip'>logLikHessian<span class='tooltiptext'>Hessian of the log-likelihood with respect to lengthscales hyperparametersWorks for homGP and hetGP models from the hetGP package for now.<\/span><\/div>, <div class='tooltip'>n11_2_01<span class='tooltiptext'>f:[-1, 1] -> R Becomes f:[0,1] -> R<\/span><\/div>, <div class='tooltip'>subspace_dist<span class='tooltiptext'>Get the distance between subspaces defined as the ranges of A and B<\/span><\/div>, <div class='tooltip'>update_C2<span class='tooltiptext'>Update Constantine's C, using update formula<\/span><\/div>, <div class='tooltip'>W_kappa_ij<span class='tooltiptext'>Computes Int(kappa_i(X, design) . kappa_j(design, X)). This function is preferred for initialization<\/span><\/div>, <div class='tooltip'>W_kappa_ij_up<span class='tooltiptext'>Computes Int(kappa_i(X, design) . kappa_j(design, X)). This function is preferred for updates<\/span><\/div>","<div class='tooltip'>BioInfo.Power<span class='tooltiptext'>Given the Biomarker-Informed design information, returns the overall power and probability of the arm is selected as the winner.<\/span><\/div>, <div class='tooltip'>CopriEndpt.Power<span class='tooltiptext'>Given the group sequential design information, returns the overall power.<\/span><\/div>, <div class='tooltip'>OneArm.CondPower<span class='tooltiptext'>Given the group sequential design information, returns the conditional power.<\/span><\/div>, <div class='tooltip'>TwoArms.CondPower<span class='tooltiptext'>Given the group sequential design information, returns the conditional power.<\/span><\/div>, <div class='tooltip'>TwoGrpCopriEndpt.SimPower<span class='tooltiptext'>Given the group sequential design information, returns the simulated overall power.<\/span><\/div>","<div class='tooltip'>adaptive_analysis_norm_global<span class='tooltiptext'>adaptive_analysis_norm_global performs an globally efficient adaptive test, a Frequentist adaptive test with the specified significance levelwith full flexibility.Normality with known variance is assumed for the test statistic(more accurately, the test statistic is assumed to follow Brownian motion.)Null hypothesis is fixed at 0 without loss of generality.Exact p-value, median unbiased estimate and confidence limits proposed by Gao et al. (2013) can also be calculated.For detailed illustration, see vignette(\"adpss_ex\").<\/span><\/div>, <div class='tooltip'>adaptive_analysis_norm_local<span class='tooltiptext'>adaptive_analysis_norm_local performs an locally efficient adaptive test, a Frequentist adaptive test with the specified significance levelwith full flexibility.Normality with known variance is assumed for the test statistic(more accurately, the test statistic is assumed to follow Brownian motion.)Null hypothesis is fixed at 0 without loss of generality.No procedure to calculate p-value or confidence intervals is employed.For detailed illustration, see vignette(\"adpss_ex\").<\/span><\/div>, <div class='tooltip'>sample_size_norm_global<span class='tooltiptext'>sample_size_norm_global calculates the power if the time of the finalanalysis is given and otherwise the sample size.The computed power for effect_size is an approximate lower bound.Sample size is also calculated on the basis of the bound.<\/span><\/div>, <div class='tooltip'>sample_size_norm_local<span class='tooltiptext'>sample_size_norm_local calculates the power if the time of the finalanalysis is given and otherwise the sample size.The computed power for effect_size is an approximate lower bound.Sample size is also calculated on the basis of the bound.<\/span><\/div>, <div class='tooltip'>work_test_norm_global<span class='tooltiptext'>This function is used as a routine by adaptive_analysis_norm_global and sample_size_norm_global.<\/span><\/div>","<div class='tooltip'>genbernoullidata<span class='tooltiptext'>genlogisticdata() function used mainly internally bylogistictrialsimulator() function to generate data for a two-armclinical trial, experimental and control groups. Can be used to generaterandom trial data.<\/span><\/div>, <div class='tooltip'>gengaussiandata<span class='tooltiptext'>gengaussiandata() function used mainly internally bygaussiantrialsimulator() function to generate data for a two-armclinical trial, experimental and control groups. Can be used to generaterandom trial data.<\/span><\/div>, <div class='tooltip'>genlognormaldata<span class='tooltiptext'>genlognormaldata() function used mainly internally bylognormaltrialsimulator() and lognormaltrialsimulatornohist() functionsto generate data for a two-arm clinical trial, experimental and control groups.Can be used to generate random trial data.<\/span><\/div>, <div class='tooltip'>genpoissondata<span class='tooltiptext'>genpoissondata() function mainly used internally bypoissontrialsimulator() function to generate data for a two-armclinical trial, experimental and control groups. Can be used to generaterandom trial data.<\/span><\/div>, <div class='tooltip'>genpwedata<span class='tooltiptext'>genpwedata() function used mainly internally bypwetrialsimulator() function to generate data for a two-armclinical trial, experimental and control groups. Can be used to generaterandom trial data.<\/span><\/div>, <div class='tooltip'>genweibulldata<span class='tooltiptext'>genweibulldata() function used mainly internally byweibulltrialsimulator() and weibulltrialsimulatornohist() functions to generate data for a two-arm clinical trial, experimental and control groups. Can be used to generate random trial data.<\/span><\/div>, <div class='tooltip'>historic_sim<span class='tooltiptext'>historic_sim() returns an S3 object of class bayes_ctd_array, whichwill contain simulation results for power, statistic estimation, bias,variance, and mse as requested by user.<\/span><\/div>, <div class='tooltip'>plot_table<span class='tooltiptext'>plot_table() takes an S3 object of class bayes_ctd_array, and createsa line plot from a one or two dimensional slice of the data generated by a clinicaltrial simulation using historic_sim() or simple_sim(). This methodhas been replaced with plot(). The method plot_table() has beenretained for continuity with original release of BayesCTDesign.The plottedresults can be smoothed or unsmoothed.<\/span><\/div>, <div class='tooltip'>print_table<span class='tooltiptext'>print_table() takes an S3 object of class bayes_ctd_array, and printsa two dimensional slice from the data generated by a clinical trial simulationusing historic_sim() or simple_sim(). This method has been replacedwith print(). The method print_table() has been retained forcontinuity with original release of BayesCTDesign.<\/span><\/div>, <div class='tooltip'>simple_sim<span class='tooltiptext'>simple_sim() returns an S3 object of class bayes_ctd_array, whichwill contain simulation results for power, statistic estimation, bias, variance,and mse as requested by user.<\/span><\/div>","<div class='tooltip'>BDEsizeApp<span class='tooltiptext'>Shiny App for efficient determination of the size of experiments in Balanced design of experiments<\/span><\/div>, <div class='tooltip'>fsize<span class='tooltiptext'>Detectable minimum effect size is calculated using the distribution function of noncentral F-distribution with noncentrality parameter.<\/span><\/div>, <div class='tooltip'>plots.2levFr<span class='tooltiptext'>Three graphs in 2 level fractional factorial design are for investigating the mutual relationship between the sample size, power \"1-beta\" and the detectable standardized effect size \"delta\"<\/span><\/div>, <div class='tooltip'>plots.Block<span class='tooltiptext'>Three graphs in randomized complete block design are for investigating the mutual relationship between the sample size, power \"1-beta\" and the detectable standardized effect size \"delta\"<\/span><\/div>, <div class='tooltip'>plots.Full<span class='tooltiptext'>Three graphs in full factorial design are for investigating the mutual relationship between the sample size, power \"1-beta\" and the detectable standardized effect size \"delta\"<\/span><\/div>, <div class='tooltip'>plots.Split<span class='tooltiptext'>Three graphs in split-plot design are for investigating the mutual relationship between the sample size, power \"1-beta\" and the detectable standardized effect size \"delta\"<\/span><\/div>, <div class='tooltip'>Size.2levFr<span class='tooltiptext'>Sample size in 2 level fractional factorial design is computed in order to detect a certain standardized effect size \"delta\" with power \"1-beta\" at the significance level \"alpha\".The model for fractional factorial design contains only main effects in resolution III and IV.<\/span><\/div>, <div class='tooltip'>Size.Block<span class='tooltiptext'>Sample size in randomized complete block design is computed in order to detect a certain standardized effect size \"delta\" with power \"1-beta\" at the significance level \"alpha\".<\/span><\/div>, <div class='tooltip'>Size.Full<span class='tooltiptext'>Sample size in full factorial design is computed in order to detect a certain standardized effect size \"delta\" with power \"1-beta\" at the significance level \"alpha\".<\/span><\/div>, <div class='tooltip'>Size.Split<span class='tooltiptext'>Sample size in split-plot design is computed in order to detect a certain standardized effect size \"delta\" with power \"1-beta\" at the significance level \"alpha\".<\/span><\/div>, <div class='tooltip'>sizelist<span class='tooltiptext'>Model is built on the number of factor and order.<\/span><\/div>, <div class='tooltip'>sizelist.split<span class='tooltiptext'>Model is built on the number of factor and order.<\/span><\/div>","<div class='tooltip'>BDP2<span class='tooltiptext'>Determines the operating characteristics of a single-arm trial with a binaryendpoint (response, success) and interim efficacy and futility analyses.Declaration of efficacy and futility (including possibly early stopping)is based on the posterior probability that the true response rateis at least pE , pF respectively.<\/span><\/div>, <div class='tooltip'>BDP2_simulate<span class='tooltiptext'>NOTE: Usually function BDP2 will be preferred.Determines the operating characteristics of a single-arm trial with a binaryendpoint (success - failure) and interim futility analyses.The user can choose among 10 futility criteria, which are based on predictiveor conditional power (the latter either assuming H1 or the MLE), posterioror predictive probabilities, tail probabilites (under H0 or H1), constant successrates, or arbitrary user-defined futility bounds.Assumptions:Endpoint (success/no success) data available for all study patients.In case of Bayesian analysis: Beta-binomial model. Prior distribution = Beta(shape1,shape2) (uniform if prior.mean=0.5,prior.sampleSize=2)In case of predictive or conditional power, tail probabilities, or rates: One-sidedtesting in the final analysis.<\/span><\/div>, <div class='tooltip'>BDP2workflow<span class='tooltiptext'>Starts a shiny app in the web browser. It provides a workflow to choose design parameters single-arm trial with a binaryendpoint (response, success) and interim efficacy and futility analyses as well as routines to determine and visualize operating characteristics. Also Word/pdf/html reports can be generated.<\/span><\/div>, <div class='tooltip'>crit_general<span class='tooltiptext'>Calculates CritBoundaries for different types of futility criteria<\/span><\/div>, <div class='tooltip'>pFstop<span class='tooltiptext'>Determines the operating characteristics of a single-arm trial with a binary endpoint (success - failure) on the basis of analytical derivations. The design allows for futility stopping.Outputs are probability of futility stopping (P.futil) at each interim analysis cumulative stopping probability up to the interim (P.futil.cum).<\/span><\/div>, <div class='tooltip'>pFstopEcall<span class='tooltiptext'>Determines the operating characteristics of a single-arm trial with a binary endpoint (success - failure) on the basis of analytical derivations. The design allows for futility stopping and evaluates efficacy at interim.Outputs are probability of calling efficacy (P.effic) at each interim analysis cumulative probability of calling efficacy up to the interim (P.effic.cum). probability of futility stopping (P.futil) at each interim analysis cumulative stopping probability up to the interim (P.futil.cum).<\/span><\/div>, <div class='tooltip'>pFstopEstop<span class='tooltiptext'>Determines the operating characteristics of a single-arm trial with a binary endpoint (success - failure) on the basis of analytical derivations. The design allows for stopping for futility and stopping for efficacy.Outputs are probability of stopping for efficacy (P.effic) at each interim analysis cumulative probability of stopping for efficacy up to the interim (P.effic.cum). probability of futility stopping (P.futil) at each interim analysis cumulative stopping probability up to the interim (P.futil.cum).<\/span><\/div>, <div class='tooltip'>plot.cE_vs_pEcall<span class='tooltiptext'>Plot objects returned by plotBDP2 with arguments x=\"cE\" and either y=\"PEcall\" or y=\"PEstop\" which can computationally be relatively expensive.<\/span><\/div>, <div class='tooltip'>plot.cE_vs_pEstop<span class='tooltiptext'>Plot objects returned by plotBDP2 with arguments x=\"cE\" and either y=\"PEcall\" or y=\"PEstop\" which can computationally be relatively expensive.<\/span><\/div>, <div class='tooltip'>plotBDP2<span class='tooltiptext'>Output of desired plots<\/span><\/div>","<div class='tooltip'>%>%<span class='tooltiptext'>See %>% for more details.<\/span><\/div>, <div class='tooltip'>bdpbinomial<span class='tooltiptext'>bdpbinomial is used for estimating posterior samples from a binomial outcome where an informative prior is used. The prior weight is determined using a discount function. This code is modeled after the methodologies developed in Haddad et al. (2017).<\/span><\/div>, <div class='tooltip'>binomialbayes<span class='tooltiptext'>Simulation for binomial counts for block design for response-adaptive randomization with time as a confounding<\/span><\/div>, <div class='tooltip'>binomialfreq<span class='tooltiptext'>Simulation for binomial counts for block design for response-adaptive randomization with time as a confounding<\/span><\/div>, <div class='tooltip'>prop_strata<span class='tooltiptext'>Computing the proportion of treatment difference for stratified data. The stratification is done over time.<\/span><\/div>","<div class='tooltip'>A_bound<span class='tooltiptext'>Finds upper A-efficiency bounds for regular block designs.<\/span><\/div>, <div class='tooltip'>blocks<span class='tooltiptext'>Constructs randomized multi-level nested block designs for unstructured treatment sets.<\/span><\/div>, <div class='tooltip'>design<span class='tooltiptext'>Constructs general D-optimal block and treatment designs for any feasible linear treatment model andany feasible combination of block factors.<\/span><\/div>, <div class='tooltip'>GraecoLatin<span class='tooltiptext'>Constructs mutually orthogonal Graeco-Latin squares for the following N:i) any odd valued Nii) any prime-power N = p**q where p and q can be chosen from rrrrrrrr prime p maximum q 2 13 3 8 5 6 7 5 11 4 13 17 19 23 3 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 2 Any prime >97 1 iii) any even valued N <= 30 except for 6 or 2<\/span><\/div>, <div class='tooltip'>HCF<span class='tooltiptext'>Finds the highest common factor (hcf) of a set of integer numbers greater than zero (Euclidean algorithm).<\/span><\/div>, <div class='tooltip'>isPrime<span class='tooltiptext'>Tests if a given number is prime and returns TRUE or FALSE<\/span><\/div>, <div class='tooltip'>isPrimePower<span class='tooltiptext'>Tests if a given number N is a prime power and returns either the base prime p and power q or p = 0 and q = 0.<\/span><\/div>, <div class='tooltip'>MOLS<span class='tooltiptext'>Constructs r sets of mutually orthogonal Latin squares (MOLS) of dimension p**q for prime p andinteger power q where r < p**q. Memory issues mean that the maximum size of the exponent q for specific p is restricted to the values shown in the table below: rrrrrrrr prime p maximum q 2 13 3 8 5 6 7 5 11 4 13 17 19 23 3 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 2 Any prime >97 1<\/span><\/div>","<div class='tooltip'>blrm_combo_ms<span class='tooltiptext'>Provides dose escalation design in Phase I oncology trial using adaptive Bayesian Logistic Regression Modeling given prior and observed cohorts<\/span><\/div>, <div class='tooltip'>blrm_combo_ss<span class='tooltiptext'>Provides dose escalation design in Phase I oncology trial using adaptive Bayesian Logistic Regression Modeling given prior and observed cohorts<\/span><\/div>, <div class='tooltip'>blrm_mono_ms<span class='tooltiptext'>Provides dose escalation design in Phase I oncology trial using adaptive Bayesian Logistic Regression Modeling given prior and observed cohorts<\/span><\/div>, <div class='tooltip'>blrm_mono_ss<span class='tooltiptext'>Provides dose escalation design in Phase I oncology trial using adaptive Bayesian Logistic Regression Modeling given prior and observed cohorts<\/span><\/div>","<div class='tooltip'>aucVardiTest<span class='tooltiptext'>Permutation test for comparing growth curves across tow groups under dependent right censoring.<\/span><\/div>, <div class='tooltip'>bdrycross.prob<span class='tooltiptext'>Computes a stopping rule and its operating characteristics for toxicity monitoring based repeated significance testing.<\/span><\/div>, <div class='tooltip'>calogrank<span class='tooltiptext'>Logrank test to compare survival curves adjusting for covariates<\/span><\/div>, <div class='tooltip'>coxphCPE<span class='tooltiptext'>Calculates the Concordance Probability Estimate for a Cox model.<\/span><\/div>, <div class='tooltip'>coxphERR<span class='tooltiptext'>Calculates the contribution of a subset of covariates to the explained relative risk derived from the full Cox proportional hazards model.<\/span><\/div>, <div class='tooltip'>coxphQuantile<span class='tooltiptext'>Draws a quantile curve of survival distribution as a function of covariate.<\/span><\/div>, <div class='tooltip'>CPS.ssize<span class='tooltiptext'>Calculates sample size, effect size and power based on Fisher's exact test<\/span><\/div>, <div class='tooltip'>deltaAUC<span class='tooltiptext'>Conducts the test<\/span><\/div>, <div class='tooltip'>fe.mdor<span class='tooltiptext'>Calculates sample size, effect size and power based on Fisher's exact test<\/span><\/div>, <div class='tooltip'>fe.power<span class='tooltiptext'>Calculates sample size, effect size and power based on Fisher's exact test<\/span><\/div>, <div class='tooltip'>fe.ssize<span class='tooltiptext'>Calculates sample size, effect size and power based on Fisher's exact test<\/span><\/div>, <div class='tooltip'>gsdesign.binomial<span class='tooltiptext'>Functions to calculate sample size for group sequential designs<\/span><\/div>, <div class='tooltip'>gsdesign.normal<span class='tooltiptext'>Functions to calculate sample size for group sequential designs<\/span><\/div>, <div class='tooltip'>gsdesign.survival<span class='tooltiptext'>Functions to calculate sample size for group sequential designs<\/span><\/div>, <div class='tooltip'>jonckheere.test<span class='tooltiptext'>Jonckheere-Terpstra test to test for ordered differences among classes<\/span><\/div>, <div class='tooltip'>mdrr<span class='tooltiptext'>Calculates sample size, effect size and power based on Fisher's exact test<\/span><\/div>, <div class='tooltip'>oc.twostage.bdry<span class='tooltiptext'>Calculates the operating characteristics of a two-stage boundary.<\/span><\/div>, <div class='tooltip'>or2pcase<span class='tooltiptext'>Calculates sample size, effect size and power based on Fisher's exact test<\/span><\/div>, <div class='tooltip'>permlogrank<span class='tooltiptext'>Small sample survdiff using permutation reference distributions.<\/span><\/div>, <div class='tooltip'>ph2simon<span class='tooltiptext'>Calculates Optimal and Minimax 2-stage Phase II designs given by Richard Simon<\/span><\/div>, <div class='tooltip'>ph2single<span class='tooltiptext'>Calculates the exact one stage Phase II design<\/span><\/div>, <div class='tooltip'>power.ladesign<span class='tooltiptext'>Functions to calculate the power of rank tests for animal studies.<\/span><\/div>, <div class='tooltip'>pselect<span class='tooltiptext'>Calculates the probability of selecting the treatment with the higher response rate under the pick the winner rule.<\/span><\/div>, <div class='tooltip'>roc.area.test<span class='tooltiptext'>Computes the nonparametric area under the ROC curve and its variance based on U-statistic theory (DDCP).<\/span><\/div>, <div class='tooltip'>roc.curve<span class='tooltiptext'>Computes the empricial ROC curve for a diagnostic tool.<\/span><\/div>, <div class='tooltip'>roc.perm.test<span class='tooltiptext'>Computes the test statistic and permutation reference distribution for comparing paired or unpaired ROC curves.<\/span><\/div>, <div class='tooltip'>toxbdry<span class='tooltiptext'>Computes a stopping rule and its operating characteristics for toxicity monitoring based repeated significance testing.<\/span><\/div>, <div class='tooltip'>twostage.inference<span class='tooltiptext'>Calculates the p-value, UMVUE and CI for the data from a study using a two stage design for response.<\/span><\/div>","<div class='tooltip'>ARE_cbe<span class='tooltiptext'>The composite endpoint is assumed to be a binary endpoint formed by a combination of two events (E1 and E2). We assume that the endpoint 1 is more relevant for the clinical question than endpoint 2. This function calculates the ARE method for binary endpoints. The method quantifies the differences in efficiency of using the composite or the relevant as primary endpoint to lead the trial and, moreover, provides a decision rule to choose the primary endpoint. If the ARE is larger than 1, the composite endpoint may be considered the best option as primary endpoint. Otherwise, the relevant endpoint is preferred.<\/span><\/div>, <div class='tooltip'>CopulaSelection<span class='tooltiptext'>Build a copula class object from the given family and the corresponding dependence parameter from the given correlation<\/span><\/div>, <div class='tooltip'>dClayton<span class='tooltiptext'>returns the density function of Clayton's copula<\/span><\/div>, <div class='tooltip'>dFGM<span class='tooltiptext'>returns the density function of the copulaFGM<\/span><\/div>, <div class='tooltip'>dFrank<span class='tooltiptext'>returns the density function of Frank's copula<\/span><\/div>, <div class='tooltip'>dGumbel<span class='tooltiptext'>returns the density function of Gumbel's copula<\/span><\/div>, <div class='tooltip'>dNormal<span class='tooltiptext'>returns the density function of Normal copula<\/span><\/div>, <div class='tooltip'>effectsize_cbe<span class='tooltiptext'>This function calculates different effect measures for binary composite outcomes. The composite endpoint is assumed to be a binary endpoint formed by a combination of two events (E1 and E2).The effect size is calculated on the basis of anticipated information on the composite components and the correlation between them. The function allows to compute the effect size in terms of the risk difference, risk ratio and odds ratio.<\/span><\/div>, <div class='tooltip'>effectsize_tte<span class='tooltiptext'>This function calculates different effect measures for time-to-event composite outcomes. The composite endpoint is assumed to be a time-to-event endpoint formed by a combination of two events (E1 and E2).The effect size is calculated on the basis of anticipated information on the composite components and the correlation between them.Marginal distributions are assumed for mThe function allows to compute the effect size in terms of the geometric average hazard ratio, the average hazard ratio, the ratio of restricted mean survival times and the median survival time ratio.<\/span><\/div>, <div class='tooltip'>lower_corr<span class='tooltiptext'>Pearson's correlation between two binary outcomes takes values between two bounds defined according to the probabilities of the binary outcomes.This function calculates the lower bound of the correlation based on the probabilities of two binary outcomes.<\/span><\/div>, <div class='tooltip'>MarginalsSelection<span class='tooltiptext'>Returns the distribution and parameters of the marginal Weibull distributions<\/span><\/div>, <div class='tooltip'>prob_cbe<span class='tooltiptext'>This function calculates the probability of the composite binary endpoint formed by a combination of two events (E1 and E2).This probability is calculated by means of the probabilities of the composite components (E1 and E2) and the correlation between them in terms of Pearson's correlation coefficient.<\/span><\/div>, <div class='tooltip'>samplesize_cbe<span class='tooltiptext'>This function calculates the required sample size for trials with a composite binary endpoint as primary endpoint.The primary endpoint is assumed to be a composite binary endpoint formed by a combination of two events (E1 and E2).The sample size is computed to evaluate differences between two groups in terms of the risk difference, risk ratio or odds ratio.The sample size is calculated on the basis of anticipated information on the composite components and the correlation between them.<\/span><\/div>, <div class='tooltip'>Sstar<span class='tooltiptext'>Returns the value of the survival function of S* at point x given the marginal distributions and the bivariate distributions via copula<\/span><\/div>, <div class='tooltip'>Sstar_func_perc<span class='tooltiptext'>Given a percentile perc, returns the difference between this percentile and the value of survival functionUseful for uniroot function<\/span><\/div>, <div class='tooltip'>upper_corr<span class='tooltiptext'>Pearson's correlation between two binary outcomes takes values between two bounds defined according to the probabilities of the binary outcomes.This function calculates the upper bound of the correlation based on the probabilities of two binary outcomes.<\/span><\/div>","<div class='tooltip'>figure.modify.margin<span class='tooltiptext'>A function for plotting power and type 1 error of a \"Modify margin\" Non-inferiority trial design after running simulations .<\/span><\/div>, <div class='tooltip'>sample.size.NI<span class='tooltiptext'>A function for calculating sample size of a non-inferiority trial.<\/span><\/div>, <div class='tooltip'>simulations.modify.margin<span class='tooltiptext'>A function for running simulations to investigate power and type 1 error of a \"Modify margin\" Non-inferiority trial design.<\/span><\/div>, <div class='tooltip'>test.NI<span class='tooltiptext'>A function for testing non-inferiority on either the risk difference, log-risk ratio or arc-sine difference scale.<\/span><\/div>","<div class='tooltip'>CombIncrease_next<span class='tooltiptext'>CombIncrease_next is used to determine the next or recommended combination in a phase I combination clinical trial using the design proposed by Riviere et al. entitled \"A Bayesian dose-finding design for drug combination clinical trials based on the logistic model\".<\/span><\/div>, <div class='tooltip'>CombIncrease_sim<span class='tooltiptext'>CombIncrease_sim is used to generate simulation replicates of phase I clinical trial for combination studies where the toxicity and efficacy of both agents is assumed to increase with the dose using the design proposed by Riviere et al. entitled \"A Bayesian dose-finding design for drug combination clinical trials based on the logistic model\".<\/span><\/div>","<div class='tooltip'>mtaBin_next<span class='tooltiptext'>mtaBin_next is used to determine the next optimal dose to administer in a Phase I/II clinical trial for Molecularly Targeted Agent using the design proposed by Riviere et al. entitled \"Phase I/II Dose-Finding Design for Molecularly Targeted Agent: Plateau Determination using Adaptive Randomization\".<\/span><\/div>, <div class='tooltip'>mtaBin_sim<span class='tooltiptext'>mtaBin_sim is used to generate simulation replicates of Phase I/II clinical trial for Molecularly Targeted Agent using the design proposed by Riviere et al. entitled \"Phase I/II Dose-Finding Design for Molecularly Targeted Agent: Plateau Determination using Adaptive Randomization\".<\/span><\/div>","<div class='tooltip'>avg_oc_wr_ne<span class='tooltiptext'>Function for calculating the average operating characteristics of two single arm bayesian designs for early gating with respect to the sample size in the experimental group and possible historical data.<\/span><\/div>, <div class='tooltip'>avg_oc_wr_ne_rct<span class='tooltiptext'>Function for calculating the average operating characteristics of two RCT bayesian designs for early gating with respect to the sample size in the experimental group, the sample size in the control group and possible historical data.<\/span><\/div>, <div class='tooltip'>avg_oc_wr_ph<span class='tooltiptext'>Function for calculating the average operating characteristics of a single arm bayesian designs for early gating with respect to the historic target.<\/span><\/div>, <div class='tooltip'>beta_par<span class='tooltiptext'>Function for calculating the parameters of the beta distribution used to average the operating characteristics, given historical data.<\/span><\/div>, <div class='tooltip'>oc<span class='tooltiptext'>Function for calculating the operating characteristics of the single arm bayesian designs in setting 1 and 2 for early gating.<\/span><\/div>, <div class='tooltip'>oc_rct<span class='tooltiptext'>Function for calculating the operating characteristics of the RCT bayesian designs in setting 3 and 4 for early gating.<\/span><\/div>, <div class='tooltip'>req_resp<span class='tooltiptext'>Function for calculating the minimum required number of responders in the experimental group to make a GO decision in Settings 1 and 2.<\/span><\/div>, <div class='tooltip'>req_resp_rct<span class='tooltiptext'>Function for calculating the minimum required number of responders in the experimental group to make a GO decision in Settings 3 and 4.<\/span><\/div>","<div class='tooltip'>n.multiway<span class='tooltiptext'>Sample size calculations for factorial ANOVAs<\/span><\/div>, <div class='tooltip'>n.oneway<span class='tooltiptext'>Calculates the required sample size for a one-way ANOVA.<\/span><\/div>","<div class='tooltip'>binaryMatchExperimentalDesignSearch<span class='tooltiptext'>This method creates an object of type binary_experimental_design and will find pairs. You can thenuse the function resultsBinaryMatchSearch to create randomized allocation vectors. For one columnin X, we just sort to find the pairs trivially.<\/span><\/div>, <div class='tooltip'>binaryMatchFollowedByGreedyExperimentalDesignSearch<span class='tooltiptext'>This method creates an object of type binary_then_greedy_experimental_design and will find optimal matched pairs whichare then greedily switched in order to further minimize a balance metric. You can thenuse the function resultsBinaryMatchThenGreedySearch to obtain the randomized allocation vectors. For one columnin X, the matching just sorts the values to find the pairs trivially.<\/span><\/div>, <div class='tooltip'>binaryMatchFollowedByRerandomizationDesignSearch<span class='tooltiptext'>This method creates an object of type binary_then_rerandomization_experimental_design and will find optimal matched pairs whichare then rerandomized in order to further minimize a balance metric. You can thenuse the function resultsBinaryMatchThenRerandomizationSearch to obtain the randomized allocation vectors. For one columnin X, the matching just sorts the values to find the pairs trivially.<\/span><\/div>, <div class='tooltip'>complete_randomization<span class='tooltiptext'>Implements complete randomization (without forced balance)<\/span><\/div>, <div class='tooltip'>complete_randomization_with_forced_balanced<span class='tooltiptext'>Implements forced balanced randomization<\/span><\/div>, <div class='tooltip'>compute_gram_matrix<span class='tooltiptext'>Computes the Gram Matrix for a user-specified kernel using the library kernlab. Note that this function automatically standardizes the columns of the data entered.<\/span><\/div>, <div class='tooltip'>compute_objective_val<span class='tooltiptext'>Returns the objective value given a design vector as well an an objective function.This is sometimes duplicated in Java. However, within Java, tricks are played to makeoptimization go faster so Java's objective values may not always be the same as the trueobjective function (e.g. logs or constants dropped).<\/span><\/div>, <div class='tooltip'>compute_randomization_metrics<span class='tooltiptext'>Computes Randomization Metrics (explained in paper) about a design algorithm<\/span><\/div>, <div class='tooltip'>generate_stdzied_design_matrix<span class='tooltiptext'>This function is useful for debugging.<\/span><\/div>, <div class='tooltip'>greedy_orthogonalization_curation<span class='tooltiptext'>This function takes a set of allocation vectors and pares them down one-by-oneby eliminating the vector that can result in the largest reduction in Avg[ |r_ij| ].It is recommended to begin with a set of unmirrored vectors for speed. Then add the mirrors laterfor whichever subset you wish.<\/span><\/div>, <div class='tooltip'>greedy_orthogonalization_curation2<span class='tooltiptext'>This function takes a set of allocation vectors and pares them down one-by-oneby eliminating the vector that can result in the largest reduction in Avg[ |r_ij| ].It is recommended to begin with a set of unmirrored vectors for speed. Then add the mirrors laterfor whichever subset you wish.<\/span><\/div>, <div class='tooltip'>hadamardExperimentalDesign<span class='tooltiptext'>This method returns unique designs according to a Hadamard matrix<\/span><\/div>, <div class='tooltip'>initGreedyExperimentalDesignObject<span class='tooltiptext'>This method creates an object of type greedy_experimental_design and will immediately initiatea search through $1_T$ space for forced balance designs.<\/span><\/div>, <div class='tooltip'>initKarpExperimentalDesignObject<span class='tooltiptext'>This method creates an object of type karp_experimental_design and will immediately initiatea search through $1_T$ space. Note that the Karp search only works for one covariate (i.e. $p=1$) and the objective \"abs_sum_diff\".<\/span><\/div>, <div class='tooltip'>initOptimalExperimentalDesignObject<span class='tooltiptext'>This method creates an object of type optimal_experimental_design and will immediately initiatea search through $1_T$ space. Since this search takes exponential time, for most machines, this method is futile beyond 28 samples. You've been warned!<\/span><\/div>, <div class='tooltip'>initRerandomizationExperimentalDesignObject<span class='tooltiptext'>This method creates an object of type rerandomization_experimental_design and will immediately initiatea search through $1_T$ space for forced-balance designs.<\/span><\/div>, <div class='tooltip'>plot_obj_val_by_iter<span class='tooltiptext'>Plots the objective value by iteration<\/span><\/div>, <div class='tooltip'>plot_obj_val_order_statistic<span class='tooltiptext'>Plots an order statistic of the object value as a function of number of searches<\/span><\/div>, <div class='tooltip'>resultsBinaryMatchSearch<span class='tooltiptext'>Returns unique allocation vectors that are binary matched<\/span><\/div>, <div class='tooltip'>resultsBinaryMatchThenGreedySearch<span class='tooltiptext'>Returns unique allocation vectors that are binary matched<\/span><\/div>, <div class='tooltip'>resultsBinaryMatchThenRerandomizationSearch<span class='tooltiptext'>Returns unique allocation vectors that are binary matched<\/span><\/div>, <div class='tooltip'>resultsGreedySearch<span class='tooltiptext'>Returns the results (thus far) of the greedy design search<\/span><\/div>, <div class='tooltip'>resultsKarpSearch<span class='tooltiptext'>Returns the results (thus far) of the karp design search<\/span><\/div>, <div class='tooltip'>resultsOptimalSearch<span class='tooltiptext'>Returns the results (thus far) of the optimal design search<\/span><\/div>, <div class='tooltip'>resultsRerandomizationSearch<span class='tooltiptext'>Returns the results (thus far) of the rerandomization design search<\/span><\/div>, <div class='tooltip'>searchTimeElapsed<span class='tooltiptext'>Returns the amount of time elapsed<\/span><\/div>, <div class='tooltip'>standardize_data_matrix<span class='tooltiptext'>Standardizes the columns of a data matrix.<\/span><\/div>, <div class='tooltip'>startSearch<span class='tooltiptext'>Once begun, this function cannot be run again.<\/span><\/div>, <div class='tooltip'>stopSearch<span class='tooltiptext'>Once stopped, it cannot be restarted.<\/span><\/div>","<div class='tooltip'>beta.ab<span class='tooltiptext'>Function beta.ab() returns parameters alpha and beta for generating beta r.v. (per dose)<\/span><\/div>, <div class='tooltip'>dlt.prob<span class='tooltiptext'>Calculate DLT probability corresponding to average nTTP for each dose<\/span><\/div>, <div class='tooltip'>eff.stg1<span class='tooltiptext'>Function eff.stg1() uses a beta-binomial distribution to generate outcomes (Ys) corresponding to acceptable dose assignments from stage 1.<\/span><\/div>, <div class='tooltip'>eff.stg1.nTTP<span class='tooltiptext'>Function eff.stg1.nTTP() uses a beta-binomial distribution to generate outcomes (Ys) corresponding to acceptable dose assignments from stage 1.<\/span><\/div>, <div class='tooltip'>get.thresh<span class='tooltiptext'>Obtain average nTTP at each dose level<\/span><\/div>, <div class='tooltip'>LRtox<span class='tooltiptext'>Function LRtox() calculates the likelihood of safety for a single dose and designates whether to escalate to the next dose (safe) or stop dose escalation and move onto stage 2 (unsafe).<\/span><\/div>, <div class='tooltip'>LRtox.nTTP<span class='tooltiptext'>(nTTP) Function LRtox.nTTP() calculates the likelihood of safety for a single dose and designates whether to escalate to the next dose (safe) or stop dose escalation and move onto stage 2 (unsafe).<\/span><\/div>, <div class='tooltip'>rand.prob<span class='tooltiptext'>Function rand.prob() calculates the updated randomization probabilities based on observed efficacies up to that point.It also gives the dose allocation for the next enrolled patient based on these probabilities.<\/span><\/div>, <div class='tooltip'>rand.stg2<span class='tooltiptext'>Function rand.stg2() fits a linear regression for the continuous efficacy outcomes,computes the randomization probabilities/dose and allocates the next patient to a dose thatis considered acceptably safe and has the most promising efficacy. Dose safety is still monitored using LR and dosesthat become unacceptable are discarded (never re-visited).<\/span><\/div>, <div class='tooltip'>rand.stg2.nTTP<span class='tooltiptext'>Function rand.stg2.nTTP() fits a linear regression for the continuous efficacy outcomes, computes the randomization probabilities/dose and allocates the next patient to a dose that is considered acceptably safe and has the highest efficacy. Dose safety (with nTTP) is still monitored using LR and doses that become unacceptable are discarded (never re-visited).<\/span><\/div>, <div class='tooltip'>runExample<span class='tooltiptext'>Launch Shiny app<\/span><\/div>, <div class='tooltip'>safe.dose<span class='tooltiptext'>Function safe.dose() distinguishes acceptable from unacceptable doses<\/span><\/div>, <div class='tooltip'>safe.dose.nTTP<span class='tooltiptext'>Function safe.dose.nTTP() distinguishes acceptable from unacceptable doses<\/span><\/div>, <div class='tooltip'>sim.plot<span class='tooltiptext'>Generate plots for estimated percent allocation and response per dose.<\/span><\/div>, <div class='tooltip'>sim.summary<span class='tooltiptext'>Results from simulated trials (using sim.trials() function) displayed in tabular and/or graphical format<\/span><\/div>, <div class='tooltip'>sim.trials<span class='tooltiptext'>Results are displayed in a matrix format, where each row represents one trial simulation.<\/span><\/div>, <div class='tooltip'>sim.trials.nTTP<span class='tooltiptext'>Results are displayed in a matrix format, where each row represents one trial simulation<\/span><\/div>, <div class='tooltip'>tox.profile<span class='tooltiptext'>Gives toxicity profile (number of dose-limiting toxicities) and likelihood ratio per dose based on binary toxicity.<\/span><\/div>, <div class='tooltip'>tox.profile.nTTP<span class='tooltiptext'>The normalized total toxicity profiles (nTTP) are calculated by combining multiple toxicity grades and their weights. The nTTPs are considered a quasi-continuous toxicity measure that follows a normal distribution truncated to [0, 1].The likelihood ratio per dose are based on nTTP toxicity.<\/span><\/div>","<div class='tooltip'>conf.reg<span class='tooltiptext'>The conf.reg() function calculates confidence regions for contrasts in multivariate factorial designs.In the two-dimensional case, confidence ellipsoids can be plotted via the generic plot() function.<\/span><\/div>, <div class='tooltip'>GUI.MANOVA<span class='tooltiptext'>This function provides a graphical user interface for calculating statisticaltests for multivariate data.<\/span><\/div>, <div class='tooltip'>GUI.MANOVAwide<span class='tooltiptext'>This function provides a graphical user interface for calculating statisticaltests for multivariate data.<\/span><\/div>, <div class='tooltip'>GUI.RM<span class='tooltiptext'>This function provides a graphical user interface for calculating statisticaltests in repeated measures designs.<\/span><\/div>, <div class='tooltip'>MANOVA<span class='tooltiptext'>The MANOVA function calculates the Wald-type statistic (WTS) and a modified ANOVA-type statistic (MATS) as well as resampling versions of these test statistics for semi-parametric multivariate data.<\/span><\/div>, <div class='tooltip'>MANOVA.wide<span class='tooltiptext'>The MANOVA.wide function calculates the Wald-type statistic (WTS) and a modified ANOVA-type statistic (MATS) as well as resampling versions of these test statistics for semi-parametric multivariate data provided in wide format.<\/span><\/div>, <div class='tooltip'>multRM<span class='tooltiptext'>The multRM() function calculates the Wald-type statistic (WTS) and the modified ANOVA-type statistic (MATS) as well as resampling versions of these test statistics for multivariatesemi-parametric repeated measures designs.<\/span><\/div>, <div class='tooltip'>RM<span class='tooltiptext'>The RM() function calculates the Wald-type statistic (WTS), the ANOVA-type statistic (ATS) as well as resampling versions of these test statistics for semi-parametric repeated measures designs.<\/span><\/div>, <div class='tooltip'>simCI<span class='tooltiptext'>Multivariate post-hoc comparisons and simultaneous confidence intervals for contrasts in multivariate factorial designs<\/span><\/div>","<div class='tooltip'>get.OC.MinED<span class='tooltiptext'>Obtain the operating characteristics of the nonparametric two-stage Bayesian adaptive design for minimum effective dose (MinED)-based dosing-finding trials<\/span><\/div>, <div class='tooltip'>next.MinED<span class='tooltiptext'>Determine the dose for the next cohort of new patients for single-agent trials that aim to find a MinED<\/span><\/div>, <div class='tooltip'>plot.MinED<span class='tooltiptext'>Plot the objects returned by other functions, including (1) operating characteristics of the design, including selection percentage and the number of patients treated at each dose; (2) the estimates of toxicity and response probability for each dose in the admissable set and corresponding 95% credible interval<\/span><\/div>, <div class='tooltip'>select.MinED<span class='tooltiptext'>Select the minimum effective dose (MinED) when the trial is completed<\/span><\/div>","<div class='tooltip'>nparml<span class='tooltiptext'>Analysis of multivariate data with two-way completely randomizedfactorial design - version for large samples. The analysis is based on fully nonparametric, rank-based methods anduses an F-approximation for 'Dempster's ANOVA' criterion and a chisquare-approximation for 'Lawley-Hotelling's'criterion. This approximations are given by the asymtotic distributionof these statistics under true null-hypothesis. In contrast to the normal-approximated test (nrbtest2) it is designed fordata with large samples (see details) while the number of factorial levels is allowed to be small. The multivariate response is allowed to be ordinal, quantitative, binary or a mixtureof the different variable types. The test statistics are constructed using nonparametric relative effect estimators.<\/span><\/div>, <div class='tooltip'>nparms<span class='tooltiptext'>Analysis of multivariate data with two-way completely randomizedfactorial design - version for small samples. The analysis is based on fully nonparametric, rank-based methods anduses a N(0,1)-approximation for test statistics based on 'Dempster's ANOVA', 'Wilk's Lambda', 'Lawley-Hotelling' and'Bartlett-Nanda-Pillai' criterion. This approximation is established by the asymptotic distributionof these four statistics under true null-hypothesis if one of the explanatory factors has a large numberof levels. The multivariate response is allowed to be ordinal, quantitative, binary or a mixtureof the different variable types. The test statistics are constructed using nonparametric relative effect estimators.<\/span><\/div>","<div class='tooltip'>complete_randomization_plus_one_min_one<span class='tooltiptext'>Implements the complete randomization design (CRD) AKA Bernoulli Trial<\/span><\/div>, <div class='tooltip'>complete_randomization_with_forced_balance_plus_one_min_one<span class='tooltiptext'>Implements the balanced complete randomization design (BCRD)<\/span><\/div>, <div class='tooltip'>compute_objective_val_plus_one_min_one_enc<span class='tooltiptext'>Returns the objective value given a design vector as well an an objective function.This is code duplication since this is implemented within Java. This is only to berun if...<\/span><\/div>, <div class='tooltip'>frob_norm_sq<span class='tooltiptext'>Compute naive / vanilla squared Frobenius Norm of matrix A<\/span><\/div>, <div class='tooltip'>frob_norm_sq_debiased<span class='tooltiptext'>Compute debiased Frobenius Norm of matrix Sigmahat (Appendix 5.8). Notethat for S <= 2, it returns the naive estimate.<\/span><\/div>, <div class='tooltip'>frob_norm_sq_debiased_times_matrix<span class='tooltiptext'>Compute debiased Frobenius Norm of matrix P times Sigmahat (Appendix 5.9). Notethat for S <= 2, it returns the naive estimate.<\/span><\/div>, <div class='tooltip'>generate_W_base_and_sort<span class='tooltiptext'>Generates the base vectors to be used when locating the optimal rerandomization threshold<\/span><\/div>, <div class='tooltip'>optimal_rerandomization_exact<span class='tooltiptext'>Finds the optimal rerandomization threshold based on a user-defined quantileand a function that generates the non-linear component of the response<\/span><\/div>, <div class='tooltip'>optimal_rerandomization_normality_assumed<span class='tooltiptext'>Finds the optimal rerandomization threshold based on a user-defined quantileand a function that generates the non-linear component of the response<\/span><\/div>, <div class='tooltip'>optimal_rerandomization_tail_approx<span class='tooltiptext'>Finds the optimal rerandomization threshold based on a user-defined quantileand kurtosis based on an approximation of tail standard errors<\/span><\/div>","<div class='tooltip'>active_design<span class='tooltiptext'>Simulates the active design process.<\/span><\/div>, <div class='tooltip'>add_infinitesimal<span class='tooltiptext'>Adds a small deterministic amount to a vector.<\/span><\/div>, <div class='tooltip'>add_noise<span class='tooltiptext'>Specifies a noise generative process for the simulations. This describes how the true dynamics of the system is perturbed by noise.<\/span><\/div>, <div class='tooltip'>armijo<span class='tooltiptext'>Implements Armijo condition<\/span><\/div>, <div class='tooltip'>BFGS_special<span class='tooltiptext'>Gradients are computed using finite differences.<\/span><\/div>, <div class='tooltip'>compute_gradient<span class='tooltiptext'>Computes a gradient based on finite difference approximation<\/span><\/div>, <div class='tooltip'>compute_gradient_coordinate<span class='tooltiptext'>Comute the derivative of a single coordinate based on finite difference approximation.<\/span><\/div>, <div class='tooltip'>compute_mean_risks<span class='tooltiptext'>Used to summarize information contained in lists given by read_knobjs.<\/span><\/div>, <div class='tooltip'>dmvnorm<span class='tooltiptext'>Gaussian multivariate density.<\/span><\/div>, <div class='tooltip'>dream6_design<span class='tooltiptext'>Simulates the active design process.<\/span><\/div>, <div class='tooltip'>estimate_risk_dream6<span class='tooltiptext'>Implements the strategy of Dream6 challenge winning team.<\/span><\/div>, <div class='tooltip'>estimate_risk_out_all<span class='tooltiptext'>Estimates the expected risk associated to a given experiment for all possible observations to be performed.<\/span><\/div>, <div class='tooltip'>eval_kn_log_like<span class='tooltiptext'>This evaluates the likelihood of a parameter value by comparing corresponding kinetics to a given time course matrix.<\/span><\/div>, <div class='tooltip'>eval_log_like_knobj<span class='tooltiptext'>Computes the posterior value associated to a given parameter value for a given knowledge list.<\/span><\/div>, <div class='tooltip'>generate_our_knowledge<span class='tooltiptext'>This function defines a new knowledge list. Its main purpose is to give a value to all the global parameters that have to be set up before running the simulations.<\/span><\/div>, <div class='tooltip'>generate_sample<span class='tooltiptext'>This is an implementation of MH algorithm to sample from the posterior distribution. The proposal is a mixture between a gaussian proposal and a single coordinate proposal. The step size is diminished when the rejection rate is too high.<\/span><\/div>, <div class='tooltip'>log_likelihood<span class='tooltiptext'>Noise is assumed to be independent for each entry. The default likelihood assumes the same heteroscedastic noise as the model used in add_noise.<\/span><\/div>, <div class='tooltip'>log_normalize<span class='tooltiptext'>Binding to a C implementation of addition and normalization in log space.<\/span><\/div>, <div class='tooltip'>log_prior<span class='tooltiptext'>User defined log prior based on the parameter value only.<\/span><\/div>, <div class='tooltip'>proj_grad<span class='tooltiptext'>A projected gradient scheme for least square estimation on the positive orthant.<\/span><\/div>, <div class='tooltip'>random_design<span class='tooltiptext'>Simulates a randim design process.<\/span><\/div>, <div class='tooltip'>read_knobjs<span class='tooltiptext'>Allows to reproduce the figures of the paper (see also vignette).<\/span><\/div>, <div class='tooltip'>reverse_params<span class='tooltiptext'>Least square minimization to map back log transformed parameters to their original values<\/span><\/div>, <div class='tooltip'>risk_theta_fun<span class='tooltiptext'>Compute the risk between two different parameter values.<\/span><\/div>, <div class='tooltip'>risk_theta_vect<span class='tooltiptext'>The function computes the expected risk based on the empirical distribution defined by the sample thetas_trans.<\/span><\/div>, <div class='tooltip'>sample_function<span class='tooltiptext'>This generates a posterior sample based on the data and global parameters defined in its input.<\/span><\/div>, <div class='tooltip'>sample_function_multi_mod_weight<span class='tooltiptext'>Generate a posterior sample using multiple local search maximization and sampling based on different initializations.<\/span><\/div>, <div class='tooltip'>sample_function_single_mod<span class='tooltiptext'>Generate a posterior sample using a single local search maximization and sampling.<\/span><\/div>, <div class='tooltip'>simulate_experiment<span class='tooltiptext'>This function simulates the kinetics of the system, in accordance with a chosen molecular perturbation.<\/span><\/div>, <div class='tooltip'>simulate_experiment_no_transform<span class='tooltiptext'>Simulate the kinetics of the system using the ode solver of package deSolve.<\/span><\/div>, <div class='tooltip'>transform_params<span class='tooltiptext'>Allows to go from log space to physical space for kinetic parameters. Known parameter values are also added here.<\/span><\/div>","<div class='tooltip'>bayes.design<span class='tooltiptext'>Make animation plots to present sequential monitor stopping rule using Beta-Binomial Bayesian model<\/span><\/div>, <div class='tooltip'>BB.aniplot<span class='tooltiptext'>Make animation plots to present sequential monitor the patients using Beta-Binomial Bayesian model<\/span><\/div>, <div class='tooltip'>DT.design<span class='tooltiptext'>The design function to sequentially monitor sample size and stopping boundary for both futility and efficacy<\/span><\/div>, <div class='tooltip'>MultPostP<span class='tooltiptext'>Thall, Simon and Estey's criterion function for determining the trial decision boundaries for efficacy (futility) and safety (toxicity).<\/span><\/div>, <div class='tooltip'>MultPostP.design<span class='tooltiptext'>The design function to sequentially monitor sample size and boundary based on Thall, Simon and Estey's criterion.<\/span><\/div>, <div class='tooltip'>PostP<span class='tooltiptext'>Thall and Simon's criterion function for determining the trial decision boundaries based on the posterior probability.<\/span><\/div>, <div class='tooltip'>PostP.design<span class='tooltiptext'>The design function to sequentially monitor sample size and boundary based on Thall and Simon's criterion.<\/span><\/div>, <div class='tooltip'>PredP<span class='tooltiptext'>Lee and Liu's criterion function for determining thetrial decision cutoffs based on the predictive probability.<\/span><\/div>, <div class='tooltip'>PredP.design<span class='tooltiptext'>The design function to sequentially monitor sample size and boundary based on Lee and Liu's criterion.<\/span><\/div>, <div class='tooltip'>prior<span class='tooltiptext'>The prior function to calculate the Beta prior parameters<\/span><\/div>","<div class='tooltip'>binom.design<span class='tooltiptext'>Search criterion to find the Optimal, Minimax, Admissible andMaximized power design stopping boundary and corresponding sample size<\/span><\/div>, <div class='tooltip'>binom.power<span class='tooltiptext'>Calculate the type I error or power of a two-stage design<\/span><\/div>, <div class='tooltip'>IUT.design<span class='tooltiptext'>Search the type I error or power of a multinomial (response and disease progression) single- or two-stage design under IUT:H_0: p_1 \\le p_{01} \\ OR \\ p_2 \\ge p_{02} \\ versus \\ H_1: p_1 \\ge p_{11} > p_{01} \\ AND \\ p_2 \\le p_{12} < p_{02}<\/span><\/div>, <div class='tooltip'>IUT.power<span class='tooltiptext'>Calculate the type I error or power of a multinomial (response and disease progression) single- or two-stage design under IUT:H_0: p_1 \\le p_{01} \\ OR \\ p_2 \\ge p_{02} \\ versus \\ H_1: p_1 \\ge p_{11} > p_{01} \\ AND \\ p_2 \\le p_{12} < p_{02}<\/span><\/div>, <div class='tooltip'>UIT.design<span class='tooltiptext'>Search the type I error or power of a multinomial (response and disease progression) single- or two-stage design under IUT:H_0: p_1 \\le p_{01} \\ AND \\ p_2 \\ge p_{02} \\ versus \\ H_1: p_1 \\ge p_{11} > p_{01} \\ OR \\ p_2 \\le p_{12} < p_{02}<\/span><\/div>, <div class='tooltip'>UIT.power<span class='tooltiptext'>Calculate the type I error or power of a multinomial (response and disease progression) single- or two-stage design under UIT:H_0: p_1 \\le p_{01} \\ AND \\ p_2 \\ge p_{02} \\ versus \\ H_1: p_1 \\ge p_{11} > p_{01} \\ OR \\ p_2 \\le p_{12} < p_{02}(Note: original Zee et al. (1999) set up the correct hypotheses, but did not make a match decision.)<\/span><\/div>","<div class='tooltip'>doe<span class='tooltiptext'>Create (partial) factorial design<\/span><\/div>, <div class='tooltip'>estimable<span class='tooltiptext'>A function to determine which coefficients can be estimated based on a partial factorial design. Adapted from a function written by Blakeley McShane at https://github.com/fzettelmeyer/mktg482/blob/master/R/expdesign.R<\/span><\/div>, <div class='tooltip'>radiant.design<span class='tooltiptext'>Launch radiant.design in the default web browser<\/span><\/div>, <div class='tooltip'>radiant.design_viewer<span class='tooltiptext'>Launch radiant.design in the Rstudio viewer<\/span><\/div>, <div class='tooltip'>radiant.design_window<span class='tooltiptext'>Launch radiant.design in an Rstudio window<\/span><\/div>, <div class='tooltip'>randomizer<span class='tooltiptext'>Randomize cases into experimental conditions<\/span><\/div>, <div class='tooltip'>sample_size<span class='tooltiptext'>Sample size calculation<\/span><\/div>, <div class='tooltip'>sample_size_comp<span class='tooltiptext'>Sample size calculation for comparisons<\/span><\/div>, <div class='tooltip'>sampling<span class='tooltiptext'>Simple random sampling<\/span><\/div>","<div class='tooltip'>rdmde<span class='tooltiptext'>rdmde() performs MDE calculations for RD designs.<\/span><\/div>, <div class='tooltip'>rdpower<span class='tooltiptext'>rdpower() performs power calculations for RD designs.<\/span><\/div>, <div class='tooltip'>rdsampsi<span class='tooltiptext'>rdsampsi() performs sample size calculations for RD designs.<\/span><\/div>","<div class='tooltip'>getAccrualTime<span class='tooltiptext'>Returns an AccrualTime object that contains the accrual time and the accrual intensity.<\/span><\/div>, <div class='tooltip'>getAnalysisResults<span class='tooltiptext'>Calculates and returns the analysis results for the specified design and data.<\/span><\/div>, <div class='tooltip'>getAvailablePlotTypes<span class='tooltiptext'>Function to identify the available plot types of an object.<\/span><\/div>, <div class='tooltip'>getClosedCombinationTestResults<span class='tooltiptext'>Calculates and returns the results from the closed combination test.<\/span><\/div>, <div class='tooltip'>getClosedConditionalDunnettTestResults<span class='tooltiptext'>Calculates and returns the results from the closed conditional Dunnett test.<\/span><\/div>, <div class='tooltip'>getConditionalPower<span class='tooltiptext'>Calculates and returns the conditional power.<\/span><\/div>, <div class='tooltip'>getConditionalRejectionProbabilities<span class='tooltiptext'>Calculates the conditional rejection probabilities (CRP) for given test results.<\/span><\/div>, <div class='tooltip'>getData<span class='tooltiptext'>Returns the aggregated simulation data.<\/span><\/div>, <div class='tooltip'>getDataset<span class='tooltiptext'>Creates a dataset object and returns it.<\/span><\/div>, <div class='tooltip'>getDesignCharacteristics<span class='tooltiptext'>Calculates the characteristics of a design and returns it.<\/span><\/div>, <div class='tooltip'>getDesignConditionalDunnett<span class='tooltiptext'>Defines the design to perform an analysis with the conditional Dunnett test.<\/span><\/div>, <div class='tooltip'>getDesignFisher<span class='tooltiptext'>Performs Fisher's combination test and returns critical values for this design.<\/span><\/div>, <div class='tooltip'>getDesignGroupSequential<span class='tooltiptext'>Provides adjusted boundaries and defines a group sequential design.<\/span><\/div>, <div class='tooltip'>getDesignInverseNormal<span class='tooltiptext'>Provides adjusted boundaries and defines a group sequential design for its use in the inverse normal combination test.<\/span><\/div>, <div class='tooltip'>getDesignSet<span class='tooltiptext'>Creates a trial design set object and returns it.<\/span><\/div>, <div class='tooltip'>getEventProbabilities<span class='tooltiptext'>Returns the event probabilities for specified parameters at given time vector.<\/span><\/div>, <div class='tooltip'>getFinalConfidenceInterval<span class='tooltiptext'>Returns the final confidence interval for the parameter of interest. It is based on the prototype case, i.e., the test for testing a mean for normally distributed variables.<\/span><\/div>, <div class='tooltip'>getFinalPValue<span class='tooltiptext'>Returns the final p-value for given stage results.<\/span><\/div>, <div class='tooltip'>getHazardRatioByPi<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getLambdaByMedian<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getLambdaByPi<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getLambdaStepFunction<span class='tooltiptext'>Calculates the lambda step values for a given time vector.<\/span><\/div>, <div class='tooltip'>getLogLevel<span class='tooltiptext'>Returns the current rpact log level.<\/span><\/div>, <div class='tooltip'>getMedianByLambda<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getMedianByPi<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getNumberOfSubjects<span class='tooltiptext'>Returns the number of recruited subjects at given time vector.<\/span><\/div>, <div class='tooltip'>getOutputFormat<span class='tooltiptext'>With this function the format of the standard outputs of all rpact objects can be shown and written to a file.<\/span><\/div>, <div class='tooltip'>getParameterCaption<span class='tooltiptext'>Returns the parameter caption for a given object and parameter name.<\/span><\/div>, <div class='tooltip'>getParameterName<span class='tooltiptext'>Returns the parameter name for a given object and parameter caption.<\/span><\/div>, <div class='tooltip'>getPiByLambda<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getPiByMedian<span class='tooltiptext'>Functions to convert pi, lambda and median values into each other.<\/span><\/div>, <div class='tooltip'>getPiecewiseExponentialDistribution<span class='tooltiptext'>Distribution function, quantile function and random number generation for the piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>getPiecewiseExponentialQuantile<span class='tooltiptext'>Distribution function, quantile function and random number generation for the piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>getPiecewiseExponentialRandomNumbers<span class='tooltiptext'>Distribution function, quantile function and random number generation for the piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>getPiecewiseSurvivalTime<span class='tooltiptext'>Returns a PiecewiseSurvivalTime object that contains the all relevant parameters of an exponential survival time cumulative distribution function.Use names to obtain the field names.<\/span><\/div>, <div class='tooltip'>getPowerAndAverageSampleNumber<span class='tooltiptext'>Returns the power and average sample number of the specified design.<\/span><\/div>, <div class='tooltip'>getPowerMeans<span class='tooltiptext'>Returns the power, stopping probabilities, and expected sample size for testing means in one or two samples at given sample size.<\/span><\/div>, <div class='tooltip'>getPowerRates<span class='tooltiptext'>Returns the power, stopping probabilities, and expected sample size for testing rates in one or two samples at given sample sizes.<\/span><\/div>, <div class='tooltip'>getPowerSurvival<span class='tooltiptext'>Returns the power, stopping probabilities, and expected sample size for testing the hazard ratio in a two treatment groups survival design.<\/span><\/div>, <div class='tooltip'>getRawData<span class='tooltiptext'>Returns the raw survival data which was generated for simulation.<\/span><\/div>, <div class='tooltip'>getRepeatedConfidenceIntervals<span class='tooltiptext'>Calculates and returns the lower and upper limit of the repeated confidence intervals of the trial.<\/span><\/div>, <div class='tooltip'>getRepeatedPValues<span class='tooltiptext'>Calculates the repeated p-values for a given test results.<\/span><\/div>, <div class='tooltip'>getSampleSizeMeans<span class='tooltiptext'>Returns the sample size for testing means in one or two samples.<\/span><\/div>, <div class='tooltip'>getSampleSizeRates<span class='tooltiptext'>Returns the sample size for testing rates in one or two samples.<\/span><\/div>, <div class='tooltip'>getSampleSizeSurvival<span class='tooltiptext'>Returns the sample size for testing the hazard ratio in a two treatment groups survival design.<\/span><\/div>, <div class='tooltip'>getSimulationMeans<span class='tooltiptext'>Returns the simulated power, stopping probabilities, conditional power, and expected sample size for testing means in a one or two treatment groups testing situation.<\/span><\/div>, <div class='tooltip'>getSimulationMultiArmMeans<span class='tooltiptext'>Returns the simulated power, stopping probabilities, conditional power, and expected sample size for testing means in a multi-arm treatment groups testing situation.<\/span><\/div>, <div class='tooltip'>getSimulationMultiArmRates<span class='tooltiptext'>Returns the simulated power, stopping probabilities, conditional power, and expected sample size for testing rates in a multi-arm treatment groups testing situation.<\/span><\/div>, <div class='tooltip'>getSimulationMultiArmSurvival<span class='tooltiptext'>Returns the simulated power, stopping probabilities, conditional power, and expected sample size for testing survival in a multi-arm treatment groups testing situation. In contrast to getSimulationSurvival() (where survival times are simulated), normallydistributed logrank test statistics are simulated.<\/span><\/div>, <div class='tooltip'>getSimulationRates<span class='tooltiptext'>Returns the simulated power, stopping probabilities, conditional power, and expected sample size for testing rates in a one or two treatment groups testing situation.<\/span><\/div>, <div class='tooltip'>getSimulationSurvival<span class='tooltiptext'>Returns the analysis times, power, stopping probabilities, conditional power, and expected sample size for testing the hazard ratio in a two treatment groups survival design.<\/span><\/div>, <div class='tooltip'>getStageResults<span class='tooltiptext'>Returns summary statistics and p-values for a given data set and a given design.<\/span><\/div>, <div class='tooltip'>getTestActions<span class='tooltiptext'>Returns test actions.<\/span><\/div>, <div class='tooltip'>plotTypes<span class='tooltiptext'>Function to identify the available plot types of an object.<\/span><\/div>, <div class='tooltip'>ppwexp<span class='tooltiptext'>Distribution function, quantile function and random number generation for the piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>printCitation<span class='tooltiptext'>How to cite rpact and R in publications.<\/span><\/div>, <div class='tooltip'>qpwexp<span class='tooltiptext'>Distribution function, quantile function and random number generation for the piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>readDataset<span class='tooltiptext'>Reads a data file and returns it as dataset object.<\/span><\/div>, <div class='tooltip'>readDatasets<span class='tooltiptext'>Reads a data file and returns it as a list of dataset objects.<\/span><\/div>, <div class='tooltip'>resetLogLevel<span class='tooltiptext'>Resets the rpact log level.<\/span><\/div>, <div class='tooltip'>rpwexp<span class='tooltiptext'>Distribution function, quantile function and random number generation for the piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>setLogLevel<span class='tooltiptext'>Sets the rpact log level.<\/span><\/div>, <div class='tooltip'>setOutputFormat<span class='tooltiptext'>With this function the format of the standard outputs of all rpact objects can be changed and set user defined respectively.<\/span><\/div>, <div class='tooltip'>testPackage<span class='tooltiptext'>This function allows the installed package rpact to be tested.<\/span><\/div>, <div class='tooltip'>writeDataset<span class='tooltiptext'>Writes a dataset to a CSV file.<\/span><\/div>, <div class='tooltip'>writeDatasets<span class='tooltiptext'>Writes a list of datasets to a CSV file.<\/span><\/div>","<div class='tooltip'>censTrial<span class='tooltiptext'>censTrial `correctly censors' treatment arms in data-sets generated by simTrial by including pre-unblinded follow-up data only according to the monitoring conclusions as reported by monitorTrial.<\/span><\/div>, <div class='tooltip'>getBlockSize<span class='tooltiptext'>getBlockSize returns the minimum block size (possibly within a specified range) that is compatible with a trial's overall treatment assignment totals.<\/span><\/div>, <div class='tooltip'>monitorTrial<span class='tooltiptext'>monitorTrial applies a group sequential monitoring procedure to data-sets generated by simTrial, which may result in modification or termination of each simulated trial.<\/span><\/div>, <div class='tooltip'>rankTrial<span class='tooltiptext'>rankTrial assesses the probability of correctly selecting the winning most efficacious (individual and/or pooled) treatment arm, and assesses power to detect relative treatment efficacy in head-to-head comparisons of (individual and/or pooled) treatment arms.<\/span><\/div>, <div class='tooltip'>simTrial<span class='tooltiptext'>simTrial generates independent time-to-event data-sets according to a user-specified trial design. The user makes assumptions about the enrollment, dropout, and infection processes in each treatment arm.<\/span><\/div>, <div class='tooltip'>VEpowerPP<span class='tooltiptext'>VEpowerPP computes unconditional power to detect positive treatment (vaccine) efficacy in per-protocol cohorts identified in simTrial-generated data-sets.<\/span><\/div>","<div class='tooltip'>alphaspend<span class='tooltiptext'>Calculates the upper boundaries for efficacy at each look time%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>alphaspendf<span class='tooltiptext'>The default alpha spending function%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>betaspend<span class='tooltiptext'>Calculates the lower boundaries for futility at each look%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>betaspendf<span class='tooltiptext'>The default beta spending function%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>calcBoundaries<span class='tooltiptext'>Calculates the efficacy and futility boundaries. This only needs to be done once for a new design.<\/span><\/div>, <div class='tooltip'>curtail<span class='tooltiptext'>Calculates the probability to declare efficacy at the end of study given the Z value at the current look%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>curtailDesign<span class='tooltiptext'>calculates the probability for efficacy given the Z value%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>getProbabilities<span class='tooltiptext'>Calculates the cumulative probabilities to declare efficacy and futility under the null hypothesis and the alternative hypothesis. It also returns the p-values for declaring efficacy and futility.<\/span><\/div>, <div class='tooltip'>plotBoundaries<span class='tooltiptext'>Plots the efficacy and futility boundaries%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>print<span class='tooltiptext'>'print' displays the look times, the base alpha and beta spending functions, and the noncentrality parameter<\/span><\/div>, <div class='tooltip'>seqmon<span class='tooltiptext'>Finds the probability that a sequence of standard normal random variables z_1, z_2,\\ldots,z_m derived from a normal stochastic process with independent increments will cross a lower and and upper boundary.<\/span><\/div>, <div class='tooltip'>sequential.design<span class='tooltiptext'>The S4 sequential design class%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setAlphaspendfString<span class='tooltiptext'>Sets the expression of the base alpha spending function as a string. This function is only used if one needs to display the base alpha spending function as a string. This function DOES NOT update the base alpha spending function. One can use setBaseAlphaspendf() to change the base alpha spending function. The spending functions and their string expressions should be defined only once per object. They should not be updated during any interim update to the design.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setBaseAlphaspendf<span class='tooltiptext'>Sets the base alpha spending function.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setBaseBetaspendf<span class='tooltiptext'>Sets the base beta spending function.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setBetaspendfString<span class='tooltiptext'>Sets the expression of the base beta spending function as a string.This function is only used if one needs to display the base beta spending function as a string. This function DOES NOT update the base beta spending function. One can use setBaseBetaspendf() to change the base beta spending function. The spending functions and their string expressions should be defined only once per object. They should not be updated during any interim update to the design.<\/span><\/div>, <div class='tooltip'>setCurrentLook<span class='tooltiptext'>Sets the current look number. The curent look is the one that last took place.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setDatestamp<span class='tooltiptext'>Sets the date stamp of the design object%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setNoncentrality<span class='tooltiptext'>Sets the noncentrality parameter.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>setTimes<span class='tooltiptext'>Sets the look times. It is to be called only for the inital design, not for updating the design.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'>Shows the cumulative probability for efficacy and futility under the null and alternative hypotheses, the corresponding p-values, and the boundaries for Z at each look%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>updateDesign<span class='tooltiptext'>Updates the design. This can be done in the process of the study, when the future look times need to be changed from those originally planned.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>","<div class='tooltip'>CAR_cov_teeth<span class='tooltiptext'>The covariance matrix of individual teeth measures for each subject follows a Conditional Autoregressive model (CAR) density<\/span><\/div>, <div class='tooltip'>MC_var_yibar_mis<span class='tooltiptext'>The estimated Monte Carlo mean and variance of the average change in clinical attachment level (CAL) for each subject<\/span><\/div>, <div class='tooltip'>SampleSize_SMARTp<span class='tooltiptext'>Sample size calculations to detect desired DTR effects, which includes (i) a single regime, (ii) difference between two regimes, and (iii) a specific regime is the best,based on CAL changes under the proposed clustered, two-stage, SMART trial given type I and type II error rates<\/span><\/div>","<div class='tooltip'>cmatbrcd.mae<span class='tooltiptext'>Computes the information matrix (C-matrix) for treatment effects under either the linear fixed effects model or the linear mixed effects model setting for a given block or row-column design.<\/span><\/div>, <div class='tooltip'>fixparsoptd.mae<span class='tooltiptext'>Creates a tcltk window that allow the users to set or fix values for the parametric combinations and initial optimal or near-optimal design to compute sequential optimal or near-optimal block or row-column designs.<\/span><\/div>, <div class='tooltip'>graphsoptd.mae<span class='tooltiptext'>Creates the graphical layout of resultant sequential A-, MV-, D- or E-optimal or near-optimal block or row-column design on a separate pop-up GUI tcltk window.<\/span><\/div>, <div class='tooltip'>mmenusoptd.mae<span class='tooltiptext'>Alternative to directly using the function soptdmaeA on the R console, the function mmenusoptd.mae creates the main menu tcltk window with the list of four optimality criteria (A, MV, D and E) for search of sequential optimal or near-optimal block or row-column designs for two-colour cDNA microarray experiments.<\/span><\/div>, <div class='tooltip'>print.soptdmaeA<span class='tooltiptext'>Used to compute sequential A-, MV-, D- or E-optimal or near-optimal block and row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using the array exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>print.summary.soptdmaeA<span class='tooltiptext'>Used to compute sequential A-, MV-, D- or E-optimal or near-optimal block and row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using the array exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>seqAoptbrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>seqDoptbrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>seqEoptbrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>seqMVoptbrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>soptdmaeA<span class='tooltiptext'>Used to compute sequential A-, MV-, D- or E-optimal or near-optimal block and row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using the array exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>soptdmaeA.default<span class='tooltiptext'>Used to compute sequential A-, MV-, D- or E-optimal or near-optimal block and row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using the array exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>summary.soptdmaeA<span class='tooltiptext'>Used to compute sequential A-, MV-, D- or E-optimal or near-optimal block and row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using the array exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>","<div class='tooltip'>createTestData<span class='tooltiptext'>Helper function for example that creates a test data set in the proper format<\/span><\/div>, <div class='tooltip'>placeboNonResponder<span class='tooltiptext'>The input to SPCDcontinuous requires a logical field which says that the patient was not a placebo responder in the the first phase of the study. Usually this is determined by whether their outcome was greater than a specified value but could have other criteria. This an example program which specifies non-responders as patients whose first phase results were greater than a constant.<\/span><\/div>, <div class='tooltip'>runGls<span class='tooltiptext'>The function SPCDFcontinuous requires an analysis function which estimates the treatment effect and it's standard error for each phase of the study. This function uses the so called cLDA model to analyze the data. In the situation where their are only pre-test and post-test measurements this is roughly equivalent to an analysis of covariance. This function can either be used as is or be used as a model for a user supplied function.<\/span><\/div>, <div class='tooltip'>runSlopes<span class='tooltiptext'>The function SPCDFcontinuous requires an analysis function which estimates the treatment effect and it's standard error for each phase of the study. This function uses the random slopes model to analyze the data.This is a random effects model with a random slope and intercept for each patient. The treatment effect is the difference in the mean slopes which modeled as the time-treatment interaction. This function can either be used as is or be used as a model for a user supplied function.<\/span><\/div>, <div class='tooltip'>SPCDbinary<span class='tooltiptext'>Patients are randomized in three groups, patients who receive placebo in phase 1 and again in phase 2 of the study, patients who receive placebo in phase 1 and active in phase 2 and patients who receive active therapy in phase 1 and are not included in phase 2. A response criteria is determined and the phase 2 data of patients who respond in phase 1 is eliminated. Each phase is analyzed separately and the results are pooled. The idea of the design is that by removing the placebo responders in the analysis in the second phase the treatment effect will be greater and their will be an increase in power, both from the reuse of patients and from enrichment in the second phase<\/span><\/div>, <div class='tooltip'>SPCDcontinuous<span class='tooltiptext'>Patients are randomized in three groups, patients who receive placebo in phase 1 and again in phase 2 of the study, patients who receive placebo in phase 1 and active in phase 2 and patients who receive active therapy in phase 1 and are not included in phase 2. A response criteria is determined and the phase 2 data of patients who respond in phase 1 is eliminated. Each phase is analyzed separately and the results are pooled. The idea of the design is that by removing the placebo responders in the analysis in the second phase the treatment effect will be greater and their will be an increase in power, both from the reuse of patients and from enrichment in the second phase<\/span><\/div>, <div class='tooltip'>SPCDPower<span class='tooltiptext'>Patients are randomized in three groups, patients who receive placebo in phase 1 and again in phase 2 of the study, patients who receive placebo in phase 1 and active in phase 2 and patients who receive active therapy in phase 1 and are not included in phase 2. A response criteria is determined and the phase 2 data of patients who respond in phase 1 is eliminated. Each phase is analyzed separately and the results are pooled. Calculates power or sample size as a function of the alternative hypothesis, posed in terms of response rates or effect sizes, for both binary and continuous outcomes.<\/span><\/div>","<div class='tooltip'>ssize.F<span class='tooltiptext'>This function calculates appropriate sample sizes for multi-sample microarray experiments for a desired power. Sample size calculations are performed at controlled false discovery rates and user-specified proportions of non-differentially expressed genes, design matrix, and standard deviation. A graph of power versus sample size is created.<\/span><\/div>, <div class='tooltip'>ssize.Fvary<span class='tooltiptext'>This function calculates appropriate sample sizes for multi-sample microarray experiments in which standard deviations vary among genes. Sample sizes are determined based on a desired power, a controlled false discovery rate, and user-specified proportions of non-differentially expressed genes and design matrix. A graph of power versus sample size is created.<\/span><\/div>, <div class='tooltip'>ssize.oneSamp<span class='tooltiptext'>This function calculates appropriate sample sizes for one-sample microarray experiments for a desired power. Sample size calculations are performed at controlled false discovery rates and user-specified proportions of non-differentially expressed genes, effect size, and standard deviation. A graph of power versus sample size is created.<\/span><\/div>, <div class='tooltip'>ssize.oneSampVary<span class='tooltiptext'>This function calculates appropriate sample sizes for two-sample microarray experiments in which effect sizes as well as variances vary among genes. Sample sizes are determined based on a desired power, a controlled false discovery rate, and user-specified proportions of non-differentially expressed genes. Outputs a graph of power versus sample size. A graph of power versus sample size is created.<\/span><\/div>, <div class='tooltip'>ssize.twoSamp<span class='tooltiptext'>This function calculates appropriate sample sizes for two-sample microarray experiments for a desired power. Sample size calculations are performed at controlled false discovery rates, user-specified proportions of non-differentially expressed genes, effect size, and standard deviation. A graph of power versus sample size is created.<\/span><\/div>, <div class='tooltip'>ssize.twoSampVary<span class='tooltiptext'>This function calculates appropriate sample sizes for two-sample microarray experiments in which the differences between mean treatment expression levels (delta.g for gene g) as well as standard deviations vary among genes. Sample sizes are determined based on a desired power, a controlled false discovery rate, and user-specified proportions of non-differentially expressed genes. A graph of power versus sample size is created.<\/span><\/div>","<div class='tooltip'>compute_wlsPower<span class='tooltiptext'>This function calls `construct_DesMat` and `construct_CovMat` toconstruct the design and covariance matrix, repectively. These matrices areused to calculate the variance of the treatment effect estimator which isthen used to calculate the power to detect the assumed treatment effect.<\/span><\/div>, <div class='tooltip'>construct_CovBlk<span class='tooltiptext'>Constructs the covariance matrixfor multiple measurements of the same cluster.This function is not designed to be used directly.<\/span><\/div>, <div class='tooltip'>construct_CovMat<span class='tooltiptext'>Construct a Covariance Matrixconstructs a (block diagonal) covariance matrix.This function calls `construct_CovBlk`(or `construct_CovSubMat` in case of repeatedobservations of the same individuals) for each block.<\/span><\/div>, <div class='tooltip'>construct_DesMat<span class='tooltiptext'>Construct the Design MatrixConstructs the design matrix with one column for every (fixed)parameter to be estimated and one row for every cluster for every timepoint.This function calls `construct_trtMat` to construct a matrix with`#cluster` columns and `#timepoints` rows, indicating treatment statusfore each cluster at each timepoint. This is then transformed into the firstcolumn of the design matrix. `construct_CovMat` further calls`construct_timeajust` to get the fixed effect(s) of the timepoints.Note: Unlike the usual notation, the treatment effect is in the first column(for easier access by higher level functions).<\/span><\/div>, <div class='tooltip'>construct_timeadjust<span class='tooltiptext'>Construct the time period adjustment in the design matrix<\/span><\/div>, <div class='tooltip'>construct_trtMat<span class='tooltiptext'>Constructs a matrix of `#cluster` rows and `#timepoint` columns, indicatingtreatment status in each cluster at each timepoint.<\/span><\/div>, <div class='tooltip'>tTestPwr<span class='tooltiptext'>Compute Power of a Wald Testcomputes the power of a scaled Wald test given a standard error,an effect size, the degrees of freedom of the t-distribution anda significance level.Computes the exact power, see second example<\/span><\/div>, <div class='tooltip'>wlsPower<span class='tooltiptext'>This is the main function of the SteppedPower package.It calls the constructor functions for the design matrix andcovariance matrix, and then calculates the variance of theintervention effect estimator. The latter is then usedto compute the power of a Wald test of a (given) intervention effect.<\/span><\/div>","<div class='tooltip'>bcart<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>bgp<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>bgpllm<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>blm<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>btgp<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>btgpllm<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>btlm<span class='tooltiptext'>The seven functions described below implement Bayesian regression models of varying complexity: linear model, linear CART, Gaussian process (GP), GP with jumps to the limiting linear model (LLM), treed GP, and treed GP LLM.<\/span><\/div>, <div class='tooltip'>default.itemps<span class='tooltiptext'>Parameterized by the minimum desired inverse temperature, this function generates a ladder of inverse temperatures k[1:m] starting at k[1] = 1, with m steps down to the final temperature k[m] = k.min progressing sigmoidally, harmonically or geometrically. The output is in a format convenient for the b* functions in the tgp package (e.g. btgp), including stochastic approximation parameters c_0c0 and n_0n0 for tuning the uniform pseudo-prior output by this function<\/span><\/div>, <div class='tooltip'>dopt.gp<span class='tooltiptext'>Create sequential D-Optimal design for a stationary Gaussian process model of fixed parameterization by subsampling from a list of candidates<\/span><\/div>, <div class='tooltip'>exp2d.rand<span class='tooltiptext'>A Random subsample of data(exp2d), or Latin Hypercube sampled data evaluated with exp2d.Z<\/span><\/div>, <div class='tooltip'>exp2d.Z<span class='tooltiptext'>Evaluate the functional (mean) response for the 2-d exponential data (truth) at the X inputs, and randomly sample noisy Z--values having normal error with standard deviation provided.<\/span><\/div>, <div class='tooltip'>fried.bool<span class='tooltiptext'>Generate X and Y values from the 10-dim first Friedman data set used to validate the Multivariate Adaptive Regression Splines (MARS) model, and a variation involving boolean indicators. This test function has three non-linear and interacting variables, along with two linear, and five which are irrelevant. The version with indicators has parts of the response turned on based on the setting of the indicators<\/span><\/div>, <div class='tooltip'>friedman.1.data<span class='tooltiptext'>Generate X and Y values from the 10-dim first Friedman data set used to validate the Multivariate Adaptive Regression Splines (MARS) model, and a variation involving boolean indicators. This test function has three non-linear and interacting variables, along with two linear, and five which are irrelevant. The version with indicators has parts of the response turned on based on the setting of the indicators<\/span><\/div>, <div class='tooltip'>hist2bar<span class='tooltiptext'>Functions for making barplots summarizing the progress of importance tempering. The itemps.barplot function can be used to make a histogram of the inverse temperatures visited in the trans-temporal Markov chain. The hist2bar function is useful for making a histogram of integer-valued samples (e.g., tree heights) encountered in one or several Markov chains<\/span><\/div>, <div class='tooltip'>interp.loess<span class='tooltiptext'>Use the loess function to interpolate the two-dimensional x, y, and z data onto a uniform grid. The output produced is an object directly usable by the plotting functions persp, image, and contour, etc. This function is designed as an alternative to the interp functions from the akima library.<\/span><\/div>, <div class='tooltip'>itemps.barplot<span class='tooltiptext'>Functions for making barplots summarizing the progress of importance tempering. The itemps.barplot function can be used to make a histogram of the inverse temperatures visited in the trans-temporal Markov chain. The hist2bar function is useful for making a histogram of integer-valued samples (e.g., tree heights) encountered in one or several Markov chains<\/span><\/div>, <div class='tooltip'>lhs<span class='tooltiptext'>Draw a (random) Latin Hypercube (LH) sample of size n from in the region outlined by the provided rectangle<\/span><\/div>, <div class='tooltip'>mapT<span class='tooltiptext'>Plot the maximum a' posteriori (MAP) tree from a \"tgp\"-class object, or add one on top of an existing plot. Like plot.tgp, projections and slices of trees can be plotted as specified<\/span><\/div>, <div class='tooltip'>mean0.range1<span class='tooltiptext'>Internal Treed Gaussian Process Model functions<\/span><\/div>, <div class='tooltip'>optim.ptgpf<span class='tooltiptext'>Optimize (minimize) a noisy black-box function (i.e., a function which may not be differentiable, and may not execute deterministically). A b* tgp model is used as a surrogate for adaptive sampling via improvement (and other) statistics. Note that this function is intended as a skeleton to be tailored as required for a particular application<\/span><\/div>, <div class='tooltip'>optim.step.tgp<span class='tooltiptext'>Optimize (minimize) a noisy black-box function (i.e., a function which may not be differentiable, and may not execute deterministically). A b* tgp model is used as a surrogate for adaptive sampling via improvement (and other) statistics. Note that this function is intended as a skeleton to be tailored as required for a particular application<\/span><\/div>, <div class='tooltip'>partition<span class='tooltiptext'>Partition data according to the maximum a' posteriori (MAP) tree contained in a \"tgp\"-class object.<\/span><\/div>, <div class='tooltip'>sens<span class='tooltiptext'>Fully Bayesian Monte Carlo sensitivity analysis scheme, based upon any of the regression models contained in the tgp package. Random Latin hypercube samples are drawn at each MCMC iteration in order to estimate main effects as well as 1st order and total sensitivity indices.<\/span><\/div>, <div class='tooltip'>tgp.default.params<span class='tooltiptext'>Construct a default list of parameters to the b* functions-- the interfaces to treed Gaussian process modeling<\/span><\/div>, <div class='tooltip'>tgp.design<span class='tooltiptext'>Based on the maximum a' posteriori (MAP) treed partition extracted from a \"tgp\"-class object, calculate independent sequential treed D-Optimal designs in each of the regions.<\/span><\/div>, <div class='tooltip'>tgp.trees<span class='tooltiptext'>Plot the maximum a' posteriori (MAP) tree as a function of tree height, and show the log posterior probabilities for comparison.<\/span><\/div>","<div class='tooltip'>betaBino<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>ConfA<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>doseD<span class='tooltiptext'>This function calculates optimal concentration points according to the EC target specification, spontaneous lethality (SL), immunity (IY) and the accepted type I and type II error levels.<\/span><\/div>, <div class='tooltip'>DW2s<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>EstM2Sn<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>InterpolxA<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>loglM2S<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>OptNBinoA<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>plan<span class='tooltiptext'>Functions for internal usage<\/span><\/div>, <div class='tooltip'>setD<span class='tooltiptext'>\"setD\" estimates the minimal sample size given the users precision requirements including the accepted errors of the first and second type.<\/span><\/div>, <div class='tooltip'>spoD<span class='tooltiptext'>The function \"spoD\" offers two services. In the planning process the number of individuals to test under control conditions is calculated, together with a proposal for partitioning the total data set into subgroups in order to identify the amount of biological variation between experiments. In the analysis process, the spontaneous lethality together with its 95% confidence interval and the biological variation are computed from the users data.<\/span><\/div>, <div class='tooltip'>TV.model<span class='tooltiptext'>Functions for internal usage<\/span><\/div>"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Description<\/th>\n      <th>taskview_doe<\/th>\n      <th>taskview_survey<\/th>\n      <th>func<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="doe" class="section level3">
<h3>DoE</h3>
<div id="htmlwidget-340bb5902321e8dda338" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-340bb5902321e8dda338">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138"],["<a href='https://cran.r-project.org/web/packages/acebayes/index.html'>acebayes<\/a>","<a href='https://cran.r-project.org/web/packages/adoptr/index.html'>adoptr<\/a>","<a href='https://cran.r-project.org/web/packages/agricolae/index.html'>agricolae<\/a>","<a href='https://cran.r-project.org/web/packages/AlgDesign/index.html'>AlgDesign<\/a>","<a href='https://cran.r-project.org/web/packages/ALTopt/index.html'>ALTopt<\/a>","<a href='https://cran.r-project.org/web/packages/Aoptbdtvc/index.html'>Aoptbdtvc<\/a>","<a href='https://cran.r-project.org/web/packages/augmentedRCBD/index.html'>augmentedRCBD<\/a>","<a href='https://cran.r-project.org/web/packages/bacistool/index.html'>bacistool<\/a>","<a href='https://cran.r-project.org/web/packages/BatchExperiments/index.html'>BatchExperiments<\/a>","<a href='https://cran.r-project.org/web/packages/baymedr/index.html'>baymedr<\/a>","<a href='https://cran.r-project.org/web/packages/BCHM/index.html'>BCHM<\/a>","<a href='https://cran.r-project.org/web/packages/bcrm/index.html'>bcrm<\/a>","<a href='https://cran.r-project.org/web/packages/bdpv/index.html'>bdpv<\/a>","<a href='https://cran.r-project.org/web/packages/BHH2/index.html'>BHH2<\/a>","<a href='https://cran.r-project.org/web/packages/binGroup/index.html'>binGroup<\/a>","<a href='https://cran.r-project.org/web/packages/blocksdesign/index.html'>blocksdesign<\/a>","<a href='https://cran.r-project.org/web/packages/blockTools/index.html'>blockTools<\/a>","<a href='https://cran.r-project.org/web/packages/BOIN/index.html'>BOIN<\/a>","<a href='https://cran.r-project.org/web/packages/Boptbd/index.html'>Boptbd<\/a>","<a href='https://cran.r-project.org/web/packages/bwsTools/index.html'>bwsTools<\/a>","<a href='https://cran.r-project.org/web/packages/cofad/index.html'>cofad<\/a>","<a href='https://cran.r-project.org/web/packages/CombinS/index.html'>CombinS<\/a>","<a href='https://cran.r-project.org/web/packages/conf.design/index.html'>conf.design<\/a>","<a href='https://cran.r-project.org/web/packages/crossdes/index.html'>crossdes<\/a>","<a href='https://cran.r-project.org/web/packages/Crossover/index.html'>Crossover<\/a>","<a href='https://cran.r-project.org/web/packages/cvcrand/index.html'>cvcrand<\/a>","<a href='https://cran.r-project.org/web/packages/dae/index.html'>dae<\/a>","<a href='https://cran.r-project.org/web/packages/daewr/index.html'>daewr<\/a>","<a href='https://cran.r-project.org/web/packages/demu/index.html'>demu<\/a>","<a href='https://cran.r-project.org/web/packages/designGG/index.html'>designGG<\/a>","<a href='https://cran.r-project.org/web/packages/designmatch/index.html'>designmatch<\/a>","<a href='https://cran.r-project.org/web/packages/designr/index.html'>designr<\/a>","<a href='https://cran.r-project.org/web/packages/desplot/index.html'>desplot<\/a>","<a href='https://cran.r-project.org/web/packages/dfcrm/index.html'>dfcrm<\/a>","<a href='https://cran.r-project.org/web/packages/dfpk/index.html'>dfpk<\/a>","<a href='https://cran.r-project.org/web/packages/DiceDesign/index.html'>DiceDesign<\/a>","<a href='https://cran.r-project.org/web/packages/DiceKriging/index.html'>DiceKriging<\/a>","<a href='https://cran.r-project.org/web/packages/DiceView/index.html'>DiceView<\/a>","<a href='https://cran.r-project.org/web/packages/docopulae/index.html'>docopulae<\/a>","<a href='https://cran.r-project.org/web/packages/DoE.base/index.html'>DoE.base<\/a>","<a href='https://cran.r-project.org/web/packages/DoE.MIParray/index.html'>DoE.MIParray<\/a>","<a href='https://cran.r-project.org/web/packages/DoE.multi.response/index.html'>DoE.multi.response<\/a>","<a href='https://cran.r-project.org/web/packages/DoE.wrapper/index.html'>DoE.wrapper<\/a>","<a href='https://cran.r-project.org/web/packages/dosedesignR/index.html'>dosedesignR<\/a>","<a href='https://cran.r-project.org/web/packages/DoseFinding/index.html'>DoseFinding<\/a>","<a href='https://cran.r-project.org/web/packages/eda4treeR/index.html'>eda4treeR<\/a>","<a href='https://cran.r-project.org/web/packages/eefAnalytics/index.html'>eefAnalytics<\/a>","<a href='https://cran.r-project.org/web/packages/EngrExpt/index.html'>EngrExpt<\/a>","<a href='https://cran.r-project.org/web/packages/estimatr/index.html'>estimatr<\/a>","<a href='https://cran.r-project.org/web/packages/ExpDes/index.html'>ExpDes<\/a>","<a href='https://cran.r-project.org/web/packages/ExpDes.pt/index.html'>ExpDes.pt<\/a>","<a href='https://cran.r-project.org/web/packages/experDesign/index.html'>experDesign<\/a>","<a href='https://cran.r-project.org/web/packages/experiment/index.html'>experiment<\/a>","<a href='https://cran.r-project.org/web/packages/ExpertChoice/index.html'>ExpertChoice<\/a>","<a href='https://cran.r-project.org/web/packages/ez/index.html'>ez<\/a>","<a href='https://cran.r-project.org/web/packages/factoptd/index.html'>factoptd<\/a>","<a href='https://cran.r-project.org/web/packages/factorEx/index.html'>factorEx<\/a>","<a href='https://cran.r-project.org/web/packages/faux/index.html'>faux<\/a>","<a href='https://cran.r-project.org/web/packages/FMC/index.html'>FMC<\/a>","<a href='https://cran.r-project.org/web/packages/FrF2/index.html'>FrF2<\/a>","<a href='https://cran.r-project.org/web/packages/FrF2.catlg128/index.html'>FrF2.catlg128<\/a>","<a href='https://cran.r-project.org/web/packages/gen2stage/index.html'>gen2stage<\/a>","<a href='https://cran.r-project.org/web/packages/GFD/index.html'>GFD<\/a>","<a href='https://cran.r-project.org/web/packages/GRCdata/index.html'>GRCdata<\/a>","<a href='https://cran.r-project.org/web/packages/GreedyExperimentalDesignJARs/index.html'>GreedyExperimentalDesignJARs<\/a>","<a href='https://cran.r-project.org/web/packages/hctrial/index.html'>hctrial<\/a>","<a href='https://cran.r-project.org/web/packages/hiPOD/index.html'>hiPOD<\/a>","<a href='https://cran.r-project.org/web/packages/ibd/index.html'>ibd<\/a>","<a href='https://cran.r-project.org/web/packages/ICAOD/index.html'>ICAOD<\/a>","<a href='https://cran.r-project.org/web/packages/idefix/index.html'>idefix<\/a>","<a href='https://cran.r-project.org/web/packages/IsoCheck/index.html'>IsoCheck<\/a>","<a href='https://cran.r-project.org/web/packages/ispd/index.html'>ispd<\/a>","<a href='https://cran.r-project.org/web/packages/ivdesign/index.html'>ivdesign<\/a>","<a href='https://cran.r-project.org/web/packages/Keyboard/index.html'>Keyboard<\/a>","<a href='https://cran.r-project.org/web/packages/labstats/index.html'>labstats<\/a>","<a href='https://cran.r-project.org/web/packages/LDOD/index.html'>LDOD<\/a>","<a href='https://cran.r-project.org/web/packages/LHD/index.html'>LHD<\/a>","<a href='https://cran.r-project.org/web/packages/MaxPro/index.html'>MaxPro<\/a>","<a href='https://cran.r-project.org/web/packages/microsamplingDesign/index.html'>microsamplingDesign<\/a>","<a href='https://cran.r-project.org/web/packages/mixexp/index.html'>mixexp<\/a>","<a href='https://cran.r-project.org/web/packages/mkssd/index.html'>mkssd<\/a>","<a href='https://cran.r-project.org/web/packages/MOLHD/index.html'>MOLHD<\/a>","<a href='https://cran.r-project.org/web/packages/mxkssd/index.html'>mxkssd<\/a>","<a href='https://cran.r-project.org/web/packages/NBDesign/index.html'>NBDesign<\/a>","<a href='https://cran.r-project.org/web/packages/NNTbiomarker/index.html'>NNTbiomarker<\/a>","<a href='https://cran.r-project.org/web/packages/nparsurv/index.html'>nparsurv<\/a>","<a href='https://cran.r-project.org/web/packages/npIntFactRep/index.html'>npIntFactRep<\/a>","<a href='https://cran.r-project.org/web/packages/nppbib/index.html'>nppbib<\/a>","<a href='https://cran.r-project.org/web/packages/OBsMD/index.html'>OBsMD<\/a>","<a href='https://cran.r-project.org/web/packages/odr/index.html'>odr<\/a>","<a href='https://cran.r-project.org/web/packages/OPDOE/index.html'>OPDOE<\/a>","<a href='https://cran.r-project.org/web/packages/Opt5PL/index.html'>Opt5PL<\/a>","<a href='https://cran.r-project.org/web/packages/optbdmaeAT/index.html'>optbdmaeAT<\/a>","<a href='https://cran.r-project.org/web/packages/optDesignSlopeInt/index.html'>optDesignSlopeInt<\/a>","<a href='https://cran.r-project.org/web/packages/OptGS/index.html'>OptGS<\/a>","<a href='https://cran.r-project.org/web/packages/OptimalDesign/index.html'>OptimalDesign<\/a>","<a href='https://cran.r-project.org/web/packages/optrcdmaeAT/index.html'>optrcdmaeAT<\/a>","<a href='https://cran.r-project.org/web/packages/OWEA/index.html'>OWEA<\/a>","<a href='https://cran.r-project.org/web/packages/PBIBD/index.html'>PBIBD<\/a>","<a href='https://cran.r-project.org/web/packages/PGM2/index.html'>PGM2<\/a>","<a href='https://cran.r-project.org/web/packages/ph2bayes/index.html'>ph2bayes<\/a>","<a href='https://cran.r-project.org/web/packages/phase1PRMD/index.html'>phase1PRMD<\/a>","<a href='https://cran.r-project.org/web/packages/phase1RMD/index.html'>phase1RMD<\/a>","<a href='https://cran.r-project.org/web/packages/pid/index.html'>pid<\/a>","<a href='https://cran.r-project.org/web/packages/pipe.design/index.html'>pipe.design<\/a>","<a href='https://cran.r-project.org/web/packages/PopED/index.html'>PopED<\/a>","<a href='https://cran.r-project.org/web/packages/powerAnalysis/index.html'>powerAnalysis<\/a>","<a href='https://cran.r-project.org/web/packages/qtlDesign/index.html'>qtlDesign<\/a>","<a href='https://cran.r-project.org/web/packages/rankFD/index.html'>rankFD<\/a>","<a href='https://cran.r-project.org/web/packages/RcmdrPlugin.DoE/index.html'>RcmdrPlugin.DoE<\/a>","<a href='https://cran.r-project.org/web/packages/rdlocrand/index.html'>rdlocrand<\/a>","<a href='https://cran.r-project.org/web/packages/rdmulti/index.html'>rdmulti<\/a>","<a href='https://cran.r-project.org/web/packages/rdrobust/index.html'>rdrobust<\/a>","<a href='https://cran.r-project.org/web/packages/rld/index.html'>rld<\/a>","<a href='https://cran.r-project.org/web/packages/rodd/index.html'>rodd<\/a>","<a href='https://cran.r-project.org/web/packages/rsm/index.html'>rsm<\/a>","<a href='https://cran.r-project.org/web/packages/rsurface/index.html'>rsurface<\/a>","<a href='https://cran.r-project.org/web/packages/samplingDataCRT/index.html'>samplingDataCRT<\/a>","<a href='https://cran.r-project.org/web/packages/scdhlm/index.html'>scdhlm<\/a>","<a href='https://cran.r-project.org/web/packages/sciplot/index.html'>sciplot<\/a>","<a href='https://cran.r-project.org/web/packages/Skillings.Mack/index.html'>Skillings.Mack<\/a>","<a href='https://cran.r-project.org/web/packages/skpr/index.html'>skpr<\/a>","<a href='https://cran.r-project.org/web/packages/SLHD/index.html'>SLHD<\/a>","<a href='https://cran.r-project.org/web/packages/sp23design/index.html'>sp23design<\/a>","<a href='https://cran.r-project.org/web/packages/STARTdesign/index.html'>STARTdesign<\/a>","<a href='https://cran.r-project.org/web/packages/Superpower/index.html'>Superpower<\/a>","<a href='https://cran.r-project.org/web/packages/support.CEs/index.html'>support.CEs<\/a>","<a href='https://cran.r-project.org/web/packages/swCRTdesign/index.html'>swCRTdesign<\/a>","<a href='https://cran.r-project.org/web/packages/ThreeArmedTrials/index.html'>ThreeArmedTrials<\/a>","<a href='https://cran.r-project.org/web/packages/TraitStats/index.html'>TraitStats<\/a>","<a href='https://cran.r-project.org/web/packages/trialr/index.html'>trialr<\/a>","<a href='https://cran.r-project.org/web/packages/tsdf/index.html'>tsdf<\/a>","<a href='https://cran.r-project.org/web/packages/unrepx/index.html'>unrepx<\/a>","<a href='https://cran.r-project.org/web/packages/vdg/index.html'>vdg<\/a>","<a href='https://cran.r-project.org/web/packages/Vdgraph/index.html'>Vdgraph<\/a>","<a href='https://cran.r-project.org/web/packages/VdgRsm/index.html'>VdgRsm<\/a>","<a href='https://cran.r-project.org/web/packages/visit/index.html'>visit<\/a>","<a href='https://cran.r-project.org/web/packages/VNM/index.html'>VNM<\/a>"],["Optimal Bayesian Experimental Design using the ACE Algorithm","Adaptive Optimal Two-Stage Designs in R","Statistical Procedures for Agricultural Research","Algorithmic Experimental Design","Optimal Experimental Designs for Accelerated Life Testing","A-Optimal Block Designs for Comparing Test Treatments with Controls","Analysis of Augmented Randomised Complete Block Designs","Bayesian Classification and Information Sharing (BaCIS) Tool for the Design of Multi-Group Phase II Clinical Trials","Statistical Experiments on Batch Computing Clusters","Computation of Bayes Factors for Common Biomedical Designs","Clinical Trial Calculation Based on BCHM Design","Bayesian Continual Reassessment Method for Phase I Dose-Escalation Trials","Inference and Design for Predictive Values in Diagnostic Tests","Useful Functions for Box, Hunter and Hunter II","Evaluation and Experimental Design for Binomial Group Testing","Nested and Crossed Block Designs for Factorial and Unstructured Treatment Sets","Block, Assign, and Diagnose Potential Interference in Randomized Experiments","Bayesian Optimal INterval (BOIN) Design for Single-Agent and Drug- Combination Phase I Clinical Trials","Bayesian Optimal Block Designs","Tools for Case 1 Best-Worst Scaling (MaxDiff) Designs","Contrast Analyses for Factorial Designs","Construction Methods of some Series of PBIB Designs","Construction of factorial designs","Construction of Crossover Designs","Analysis and Search of Crossover Designs","Efficient Design and Analysis of Cluster Randomized Trials","Functions Useful in the Design and ANOVA of Experiments","Design and Analysis of Experiments with R","Optimal Design Emulators via Point Processes","Computational tool for designing genetical genomics experiments","Matched Samples that are Balanced and Representative by Design","Balanced Factorial Designs","Plotting Field Plans for Agricultural Experiments","Dose-Finding by the Continual Reassessment Method","Bayesian Dose-Finding Designs using Pharmacokinetics (PK) for Phase I Clinical Trials","Designs of Computer Experiments","Kriging Methods for Computer Experiments","Methods for Visualization of Computer Experiments Design and Surrogate","Optimal Designs for Copula Models","Full Factorials, Orthogonal Arrays and Base Utilities for DoE Packages","Creation of Arrays by Mixed Integer Programming","Construct Multi-Response Experimental Designs","Wrapper Package for Design of Experiments Functionality","Interactive Designing of Dose Finding Studies","Planning and Analyzing Dose Finding Experiments","Experimental Design and Analysis for Tree Improvement","Robust Analytical Methods for Evaluating Educational Interventions using Randomised Controlled Trials Designs","Data sets from \"Introductory Statistics for Engineering Experimentation\"","Fast Estimators for Design-Based Inference","Experimental Designs Package","Pacote Experimental Designs (Portugues)","Design Experiments for Batches","R Package for Designing and Analyzing Randomized Experiments","Design of Discrete Choice and Conjoint Analysis","Easy Analysis and Visualization of Factorial Experiments","Factorial Optimal Designs for Two-Colour cDNA Microarray Experiments","Design and Analysis for Factorial Experiments","Simulation for Factorial Designs","Factorial Experiments with Minimum Level Changes","Fractional Factorial Designs with 2-Level Factors","Catalogues of resolution IV 128 run 2-level fractional factorials up to 33 factors that do have 5-letter words","Generalized Two-Stage Designs for Phase II Single-Arm Studies","Tests for General Factorial Designs","Parameter Inference and Optimal Designs for Grouped and/or Right-Censored Count Data","GreedyExperimentalDesign JARs","Using Historical Controls for Designing Phase II Clinical Trials","hierarchical Pooled Optimal Design","Incomplete Block Designs","Optimal Designs for Nonlinear Statistical Models by Imperialist Competitive Algorithm (ICA)","Efficient Designs for Discrete Choice Experiments","Isomorphism Check for Multi-Stage Factorial Designs with Randomization Restrictions","Incomplete Split-Plot Designs","Hypothesis Testing in Cluster-Randomized Encouragement Designs","Bayesian Designs for Early Phase Clinical Trials","Data Sets for the Book \"Experimental Design for Laboratory Biologists\"","Finding Locally D-optimal optimal designs for some nonlinear and generalized linear models","Latin Hypercube Designs (LHDs)","Maximum Projection Designs","Finding Optimal Microsampling Designs for Non-Compartmental Pharmacokinetic Analysis","Design and Analysis of Mixture Experiments","Efficient multi-level k-circulant supersaturated designs","Multiple Objective Latin Hypercube Design","Efficient mixed-level k-circulant supersaturated designs","Design and Monitoring of Clinical Trials with Negative Binomial Endpoint","Calculate Design Parameters for Biomarker Validation Studies","Nonparametric Tests for Main Effects, Simple Effects and Interaction Effect in a Factorial Design with Censored Data","Nonparametric Interaction Tests for Factorial Designs with Repeated Measures","Nonparametric Partially-Balanced Incomplete Block Design Analysis","Objective Bayesian Model Discrimination in Follow-Up Designs","Optimal Design and Statistical Power of Multilevel Randomized Trials","Optimal Design of Experiments","Optimal Designs for the 5-Parameter Logistic Model","Optimal Block Designs for Two-Colour cDNA Microarray Experiments","Optimal Designs for Estimating the Slope Divided by the Intercept","Near-Optimal and Balanced Group-Sequential Designs for Clinical Trials with Continuous Outcomes","A Toolbox for Computing Efficient Designs of Experiments","Optimal Row-Column Designs for Two-Colour cDNA Microarray Experiments","Optimal Weight Exchange Algorithm for Optimal Designs for Three Models","Partially Balanced Incomplete Block Designs","Nested Resolvable Designs and their Associated Uniform Designs","Bayesian Single-Arm Phase II Designs","Personalized Repeated Measurement Design for Phase I Clinical Trials","Repeated Measurement Design for Phase I Clinical Trial","Process Improvement using Data","Dual-Agent Dose Escalation for Phase I Trials using the PIPE Design","Population (and Individual) Optimal Experimental Design","Power Analysis in Experimental Design","Design of QTL experiments","Rank-Based Tests for General Factorial Designs","R Commander Plugin for (industrial) Design of Experiments","Local Randomization Methods for RD Designs","Analysis of RD Designs with Multiple Cutoffs or Scores","Robust Data-Driven Statistical Inference in Regression-Discontinuity Designs","Analyze and Design Repeated Low-Dose Challenge Experiments","Optimal Discriminating Designs","Response-Surface Analysis","Design of Rotatable Central Composite Experiments and Response Surface Analysis","Sampling Data Within Different Study Designs for Cluster Randomized Trials","Estimating Hierarchical Linear Models for Single-Case Designs","Scientific Graphing Functions for Factorial Designs","The Skillings-Mack Test Statistic for Block Designs with Missing Observations","Design of Experiments Suite: Generate and Evaluate Optimal Designs","Maximin-Distance (Sliced) Latin Hypercube Designs","Design and Simulation of seamless Phase II-III Clinical Trials","Single to Double Arm Transition Design for Phase II Clinical Trials","Simulation-Based Power Analysis for Factorial Designs","Basic Functions for Supporting an Implementation of Choice Experiments","Stepped Wedge Cluster Randomized Trial (SW CRT) Design","Design and Analysis of Clinical Non-Inferiority or Superiority Trials with Active and Placebo Control","Statistical Data Analysis for Randomized Block Design Experiments","Clinical Trial Designs in 'rstan'","Two-/Three-Stage Designs for Phase 1&2 Clinical Trials","Analysis and Graphics for Unreplicated Experiments","Variance Dispersion Graphs and Fraction of Design Space Plots","Variance dispersion graphs and Fraction of design space plots for response surface designs","Plots of Scaled Prediction Variances for Response Surface Designs","Phase I Dose Escalation Study Design for Vaccines","Finding Multiple-Objective Optimal Designs for the 4-Parameter Logistic Model"],["Optimal Bayesian experimental design using the approximate coordinate exchange (ACE) algorithm. See <doi:10.18637/jss.v095.i13>.","Optimize one or two-arm, two-stage designs for clinical trials with respect to several pre-implemented objective criteria or implement custom objectives. Optimization under uncertainty and conditional (given stage-one outcome) constraints are supported. See Pilz M, Kunzmann K, Herrmann C, Rauch G, Kieser M. A variational approach to optimal two-stage designs. Statistics in Medicine. 2019;38(21):4159–4171. <doi:10.1002/sim.8291> for details.","Original idea was presented in the thesis \"A statistical analysis tool for agricultural research\" to obtain the degree of Master on science, National Engineering University (UNI), Lima-Peru. Some experimental data for the examples come from the CIP and others research. Agricolae offers extensive functionality on experimental design especially for agricultural and plant breeding experiments, which can also be useful for other purposes. It supports planning of lattice, Alpha, Cyclic, Complete Block, Latin Square, Graeco-Latin Squares, augmented block, factorial, split and strip plot designs. There are also various analysis facilities for experimental data, e.g. treatment comparison procedures and several non-parametric tests comparison, biodiversity indexes and consensus cluster.","Algorithmic experimental designs. Calculates exact and approximate theory experimental designs for D,A, and I criteria. Very large designs may be created. Experimental designs may be blocked or blocked designs created from a candidate list, using several criteria. The blocking can be done when whole and within plot factors interact.","Creates the optimal (D, U and I) designs for the accelerated life testing with right censoring or interval censoring. It uses generalized linear model (GLM) approach to derive the asymptotic variance-covariance matrix of regression coefficients. The failure time distribution is assumed to follow Weibull distribution with a known shape parameter and log-linear link functions are used to model the relationship between failure time parameters and stress variables. The acceleration model may have multiple stress factors, although most ALTs involve only two or less stress factors. ALTopt package also provides several plotting functions including contour plot, Fraction of Use Space (FUS) plot and Variance Dispersion graphs of Use Space (VDUS) plot. For more details, see Seo and Pan (2015) <doi:10.32614/RJ-2015-029>.","A collection of functions to construct A-optimal block designs for comparing test treatments with one or more control(s). Mainly A-optimal balanced treatment incomplete block designs, weighted A-optimal balanced treatment incomplete block designs, A-optimal group divisible treatment designs and A-optimal balanced bipartite block designs can be constructed using the package. The designs are constructed using algorithms based on linear integer programming. To the best of our knowledge, these facilities to construct A-optimal block designs for comparing test treatments with one or more controls are not available in the existing R packages. For more details on designs for tests versus control(s) comparisons, please see Hedayat, A. S. and Majumdar, D. (1984) <doi:10.1080/00401706.1984.10487989> A-Optimal Incomplete Block Designs for Control-Test Treatment Comparisons, Technometrics, 26, 363-370 and Mandal, B. N. , Gupta, V. K., Parsad, Rajender. (2017) <doi:10.1080/03610926.2015.1071394> Balanced treatment incomplete block designs through integer programming. Communications in Statistics - Theory and Methods 46(8), 3728-3737.","Functions for analysis of data generated from experiments in augmented randomised complete block design according to Federer, W.T. (1961) <doi:10.2307/2527837>. Computes analysis of variance, adjusted means, descriptive statistics, genetic variability statistics etc. Further includes data visualization and report generation functions.","Provides the design of multi-group phase II clinical trials with binary outcomes using the hierarchical Bayesian classification and information sharing (BaCIS) model. Subgroups are classified into two clusters on the basis of their outcomes mimicking the hypothesis testing framework. Subsequently, information sharing takes place within subgroups in the same cluster, rather than across all subgroups. This method can be applied to the design and analysis of multi-group clinical trials with binary outcomes. Reference: Nan Chen and J. Jack Lee (2019) <doi:10.1002/bimj.201700275>.","Extends the BatchJobs package to run statistical experiments on batch computing clusters. For further details see the project web page.","BAYesian inference for MEDical designs in R. Functions for the computation of Bayes factors for common biomedical research designs. Implemented are functions to test the equivalence (equiv_bf), non-inferiority (infer_bf), and superiority (super_bf) of an experimental group compared to a control group on a continuous outcome measure. Bayes factors for these three tests can be computed based on raw data (x, y) or summary statistics (n_x, n_y, mean_x, mean_y, sd_x, sd_y [or ci_margin and ci_level]).","Users can estimate the treatment effect for multiple subgroups basket trials based on the Bayesian Cluster Hierarchical Model (BCHM). In this model, a Bayesian non-parametric method is applied to dynamically calculate the number of clusters by conducting the multiple cluster classification based on subgroup outcomes. Hierarchical model is used to compute the posterior probability of treatment effect with the borrowing strength determined by the Bayesian non-parametric clustering and the similarities between subgroups. To use this package, 'JAGS' software and 'rjags' package are required, and users need to pre-install them.","Implements a wide variety of one- and two-parameter Bayesian CRM designs. The program can run interactively, allowing the user to enter outcomes after each cohort has been recruited, or via simulation to assess operating characteristics. See Sweeting et al. (2013): <doi:10.18637/jss.v054.i13>.","Computation of asymptotic confidence intervals for negative and positive predictive values in binary diagnostic tests in case-control studies. Experimental design for hypothesis tests on predictive values.","Functions and data sets reproducing some examples in Box, Hunter and Hunter II. Useful for statistical design of experiments, especially factorial experiments.","Methods for estimation and hypothesis testing of proportions in group testing designs: methods for estimating a proportion in a single population (assuming sensitivity and specificity equal to 1 in designs with equal group sizes), as well as hypothesis tests and functions for experimental design for this situation. For estimating one proportion or the difference of proportions, a number of confidence interval methods are included, which can deal with various different pool sizes. Further, regression methods are implemented for simple pooling and matrix pooling designs. Methods for identification of positive items in group testing designs: Optimal testing configurations can be found for hierarchical and array-based algorithms. Operating characteristics can be calculated for testing configurations across a wide variety of situations.","Constructs treatment and block designs for linear treatment models with crossed or nested block factors. The treatment design can be any feasible linear model and the block design can be any feasible combination of crossed or nested block factors. The block design is a sum of one or more block factors and the block design is optimized sequentially with the levels of each successive block factor optimized conditional on all previously optimized block factors. D-optimality is used throughout except for square or rectangular lattice block designs which are constructed algebraically using mutually orthogonal Latin squares. Crossed block designs with interaction effects are optimized using a weighting scheme which allows for differential weighting of first and second-order block effects. Outputs include a table showing the allocation of treatments to blocks and tables showing the achieved D-efficiency factors for each block and treatment design. Edmondson, R.N. Multi-level Block Designs for Comparative Experiments. JABES 25, 500–522 (2020) <doi:10.1007/s13253-020-00416-0>.","Blocks units into experimental blocks, with one unit per treatment condition, by creating a measure of multivariate distance between all possible pairs of units. Maximum, minimum, or an allowable range of differences between units on one variable can be set. Randomly assign units to treatment conditions. Diagnose potential interference between units assigned to different treatment conditions. Write outputs to .tex and .csv files.","The Bayesian optimal interval (BOIN) design is a novel phase I clinical trial design for finding the maximum tolerated dose (MTD). It can be used to design both single-agent and drug-combination trials. The BOIN design is motivated by the top priority and concern of clinicians when testing a new drug, which is to effectively treat patients and minimize the chance of exposing them to subtherapeutic or overly toxic doses. The prominent advantage of the BOIN design is that it achieves simplicity and superior performance at the same time. The BOIN design is algorithm-based and can be implemented in a simple way similar to the traditional 3+3 design. The BOIN design yields an average performance that is comparable to that of the continual reassessment method (CRM, one of the best model-based designs) in terms of selecting the MTD, but has a substantially lower risk of assigning patients to subtherapeutic or overly toxic doses. For tutorial, please check Yan et al. (2020) <doi:10.18637/jss.v094.i13>.","Computes Bayesian A- and D-optimal block designs under the linear mixed effects model settings using block/array exchange algorithm of Debusho, Gemechu and Haines (2018) <doi:10.1080/03610918.2018.1429617> where the interest is in a comparison of all possible elementary treatment contrasts. The package also provides an optional method of using the graphical user interface (GUI) R package 'tcltk' to ensure that it is user friendly.","Tools to design best-worst scaling designs (i.e., balanced incomplete block designs) and to analyze data from these designs, using aggregate and individual methods such as: difference scores, Louviere, Lings, Islam, Gudergan, & Flynn (2013) <doi:10.1016/j.ijresmar.2012.10.002>; analytical estimation, Lipovetsky & Conklin (2014) <doi:10.1016/j.jocm.2014.02.001>; empirical Bayes, Lipovetsky & Conklin (2015) <doi:10.1142/S1793536915500028>; Elo, Hollis (2018) <doi:10.3758/s13428-017-0898-2>; and network-based measures.","Contrast analysis for factorial designs is an alternative to the classical ANOVA approach with the advantage of testing focused hypothesis. The method is mainly based on Rosenthal, Rosnow and Rubin (2000, ISBN:978-0521659802) and Sedlmeier and Renkewitz (2018, ISBN:978-3868943214).","Series of partially balanced incomplete block designs (PBIB) based on the combinatory method (S) introduced in (Imane Rezgui et al, 2014) <doi:10.3844/jmssp.2014.45.48>; and it gives their associated U-type design.","This small library contains a series of simple tools for constructing and manipulating confounded and fractional factorial designs.","Contains functions for the construction of carryover balanced crossover designs. In addition contains functions to check given designs for balance.","Generate and analyse crossover designs from combinatorial or search algorithms as well as from literature and a GUI to access them.","Constrained randomization by Raab and Butcher (2001) <doi:10.1002/1097-0258(20010215)20:3%3C351::AID-SIM797%3E3.0.CO;2-C> is suitable for cluster randomized trials (CRTs) with a small number of clusters (e.g., 20 or fewer). The procedure of constrained randomization is based on the baseline values of some cluster-level covariates specified. The intervention effect on the individual outcome can then be analyzed through clustered permutation test introduced by Gail, et al. (1996) <doi:10.1002/(SICI)1097-0258(19960615)15:11%3C1069::AID-SIM220%3E3.0.CO;2-Q>. Motivated from Li, et al. (2016) <doi:10.1002/sim.7410>, the package performs constrained randomization on the baseline values of cluster-level covariates and clustered permutation test on the individual-level outcomes for cluster randomized trials.","The content falls into the following groupings: (i) Data, (ii) Factor manipulation functions, (iii) Design functions, (iv) ANOVA functions, (v) Matrix functions, (vi) Projector and canonical efficiency functions, and (vii) Miscellaneous functions. There is a vignette describing how to use the design functions for randomizing and assessing designs available as a vignette called 'DesignNotes'. The ANOVA functions facilitate the extraction of information when the 'Error' function has been used in the call to 'aov'. The package 'dae' can also be installed from <http://chris.brien.name/rpackages/>.","Contains Data frames and functions used in the book \"Design and Analysis of Experiments with R\".","Implements the Determinantal point process (DPP) based optimal design emulator described in Pratola, Lin and Craigmile (2018) <arXiv:1804.02089> for Gaussian process regression models. See <http://www.matthewpratola.com/software> for more information and examples.","The package provides R scripts for designing genetical genomics experiments.","Includes functions for the construction of matched samples that are balanced and representative by design. Among others, these functions can be used for matching in observational studies with treated and control units, with cases and controls, in related settings with instrumental variables, and in discontinuity designs. Also, they can be used for the design of randomized experiments, for example, for matching before randomization. By default, 'designmatch' uses the 'GLPK' optimization solver, but its performance is greatly enhanced by the 'Gurobi' optimization solver and its associated R interface. For their installation, please follow the instructions at <http://user.gurobi.com/download/gurobi-optimizer> and <http://www.gurobi.com/documentation/7.0/refman/r_api_overview.html>. We have also included directions in the gurobi_installation file in the inst folder.","Generate balanced factorial designs with crossed and nested random and fixed effects <https://github.com/mmrabe/designr>.","A function for plotting maps of agricultural field experiments that are laid out in grids.","Provides functions to run the CRM and TITE-CRM in phase I trials and calibration tools for trial planning purposes.","Statistical methods involving PK measures are provided, in the dose allocation process during a Phase I clinical trials. These methods, proposed by Ursino et al, (2017) <doi:10.1002/bimj.201600084>, enter pharmacokinetics (PK) in the dose finding designs in different ways, including covariates models, dependent variable or hierarchical models. This package provides functions to generate data from several scenarios and functions to run simulations which their objective is to determine the maximum tolerated dose (MTD).","Space-Filling Designs and space-filling criteria (distance-based and uniformity-based), with emphasis to computer experiments; <doi:10.18637/jss.v065.i11>.","Estimation, validation and prediction of kriging models. Important functions : km, print.km, plot.km, predict.km.","View 2D/3D sections, contour plots, mesh of excursion sets for computer experiments designs, surrogates or test functions.","A direct approach to optimal designs for copula models based on the Fisher information. Provides flexible functions for building joint PDFs, evaluating the Fisher information and finding optimal designs. It includes an extensible solution to summation and integration called 'nint', functions for transforming, plotting and comparing designs, as well as a set of tools for common low-level tasks.","Creates full factorial experimental designs and designs based on orthogonal arrays for (industrial) experiments. Provides diverse quality criteria. Provides utility functions for the class design, which is also used by other packages for designed experiments.","'CRAN' packages 'DoE.base' and 'Rmosek' and non-'CRAN' package 'gurobi' are enhanced with functionality for the creation of optimized arrays for experimentation, where optimization is in terms of generalized minimum aberration. It is also possible to optimally extend existing arrays to larger run size. Optimization requires the availability of at least one of the commercial products 'Gurobi' or 'Mosek' (free academic licenses available for both). For installing 'Gurobi' and its R package 'gurobi', follow instructions at <http://www.gurobi.com/downloads/gurobi-optimizer> and <http://www.gurobi.com/documentation/7.5/refman/r_api_overview.html> (or higher version). For installing 'Mosek' and its R package 'Rmosek', follow instructions at <https://www.mosek.com/downloads/> and <http://docs.mosek.com/8.1/rmosek/install-interface.html>, or use the functionality in the stump CRAN R package 'Rmosek'.","Construct multi-response experimental designs, such as a Unique Factor Central Composite Design (UF-CCD), given information (from screening or expert knowledge) about which factors are related to each response variable (Wilmina M. Marget & Max D. Morris, 2019 <doi:10.1080/00401706.2018.1549102>).","Various kinds of designs for (industrial) experiments can be created. The package uses, and sometimes enhances, design generation routines from other packages. So far, response surface designs from package rsm, latin hypercube samples from packages lhs and DiceDesign, and D-optimal designs from package AlgDesign have been implemented.","Provides the user with an interactive application which can be used to facilitate the planning of dose finding studies by applying the theory of optimal experimental design (e.g., Ling (2006) <doi:10.1007/0-387-33706-7>).","The DoseFinding package provides functions for the design and analysis of dose-finding experiments (with focus on pharmaceutical Phase II clinical trials). It provides functions for: multiple contrast tests, fitting non-linear dose-response models (using Bayesian and non-Bayesian estimation), calculating optimal designs and an implementation of the MCPMod methodology (Pinheiro et al. (2014) <doi:10.1002/sim.6052>).","Provides data sets and R Codes for Williams, E.R., Matheson, A.C. and Harwood, C.E. (2002). Experimental Design and Analysis for Tree Improvement, CSIRO Publishing.","Analysing data from evaluations of educational interventions using a randomised controlled trial design. Various analytical tools to perform sensitivity analysis using different methods are supported (e.g. frequentist models with bootstrapping and permutations options, Bayesian models). The included commands can be used for simple randomised trials, cluster randomised trials and multisite trials. The methods can also be used more widely beyond education trials. This package can be used to evaluate other intervention designs using Frequentist and Bayesian multilevel models.","Datasets from Nelson, Coffin and Copeland \"Introductory Statistics for Engineering Experimentation\" (Elsevier, 2003) with sample code.","Fast procedures for small set of commonly-used, design-appropriate estimators with robust standard errors and confidence intervals. Includes estimators for linear regression, instrumental variables regression, difference-in-means, Horvitz-Thompson estimation, and regression improving precision of experimental estimates by interacting treatment with centered pre-treatment covariates introduced by Lin (2013) <doi:10.1214/12-AOAS583>.","Package for analysis of simple experimental designs (CRD, RBD and LSD), experiments in double factorial schemes (in CRD and RBD), experiments in a split plot in time schemes (in CRD and RBD), experiments in double factorial schemes with an additional treatment (in CRD and RBD), experiments in triple factorial scheme (in CRD and RBD) and experiments in triple factorial schemes with an additional treatment (in CRD and RBD), performing the analysis of variance and means comparison by fitting regression models until the third power (quantitative treatments) or by a multiple comparison test, Tukey test, test of Student-Newman-Keuls (SNK), Scott-Knott, Duncan test, t test (LSD) and Bonferroni t test (protected LSD) - for qualitative treatments; residual analysis (Ferreira, Cavalcanti and Nogueira, 2014) <doi:10.4236/am.2014.519280>.","Pacote para análise de delineamentos experimentais (DIC, DBC e DQL), experimentos em esquema fatorial duplo (em DIC e DBC), experimentos em parcelas subdivididas (em DIC e DBC), experimentos em esquema fatorial duplo com um tratamento adicional (em DIC e DBC), experimentos em fatorial triplo (em DIC e DBC) e experimentos em esquema fatorial triplo com um tratamento adicional (em DIC e DBC), fazendo analise de variancia e comparacao de multiplas medias (para tratamentos qualitativos), ou ajustando modelos de regressao ate a terceira potencia (para tratamentos quantitativos); analise de residuos (Ferreira, Cavalcanti and Nogueira, 2014) <doi:10.4236/am.2014.519280>.","Distributes samples in batches while making batches homogeneous according to their description. Allows for an arbitrary number of variables, both numeric and categorical. For quality control it provides functions to subset a representative sample.","Provides various statistical methods for designing and analyzing randomized experiments. One functionality of the package is the implementation of randomized-block and matched-pair designs based on possibly multivariate pre-treatment covariates. The package also provides the tools to analyze various randomized experiments including cluster randomized experiments, two-stage randomized experiments, randomized experiments with noncompliance, and randomized experiments with missing data.","Supports designing efficient discrete choice experiments (DCEs). Experimental designs can be formed on the basis of orthogonal arrays or search methods for optimal designs (Federov or mixed integer programs). Various methods for converting these experimental designs into a discrete choice experiment. Many efficiency measures! Draws from literature of Kuhfeld (2010) and Street et. al (2005) <doi:10.1016/j.ijresmar.2005.09.003>.","Facilitates easy analysis of factorial experiments, including purely within-Ss designs (a.k.a. \"repeated measures\"), purely between-Ss designs, and mixed within-and-between-Ss designs. The functions in this package aim to provide simple, intuitive and consistent specification of data analysis and visualization. Visualization functions also include design visualization for pre-analysis data auditing, and correlation matrix visualization. Finally, this package includes functions for non-parametric analysis, including permutation tests and bootstrap resampling. The bootstrap function obtains predictions either by cell means or by more advanced/powerful mixed effects models, yielding predictions and confidence intervals that may be easily visualized at any level of the experiment's design.","Computes factorial A-, D- and E-optimal designs for two-colour cDNA microarray experiments.","Provides design-based and model-based estimators for the population average marginal component effects in general factorial experiments, including conjoint analysis. The package also implements a series of recommendations offered in de la Cuesta, Egami, and Imai (2019+), and Egami and Imai (2019) <doi:10.1080/01621459.2018.1476246>.","Create datasets with factorial structure through simulation by specifying variable parameters. Extended documentation at <https://debruine.github.io/faux/>. Described in DeBruine (2020) <doi:10.5281/zenodo.2669586>.","Generate cost effective minimally changed run sequences for symmetrical as well as asymmetrical factorial designs.","Regular and non-regular Fractional Factorial 2-level designs can be created. Furthermore, analysis tools for Fractional Factorial designs with 2-level factors are offered (main effects and interaction plots for all factors simultaneously, cube plot for looking at the simultaneous effects of three factors, full or half normal plot, alias structure in a more readable format than with the built-in function alias).","This package provides catalogues of resolution IV regular fractional factorial designs in 128 runs for up to 33 2-level factors. The catalogues are complete, excluding resolution IV designs without 5-letter words, because these do not add value for a search for clear designs. The previous package version 1.0 with complete catalogues up to 24 runs (24 runs and a namespace added later) can be downloaded from the authors website.","One can find single-stage and two-stage designs for a phase II single-arm study with either efficacy or safety/toxicity endpoints as described in Kim and Wong (2019) <doi:10.29220/CSAM.2019.26.2.163>.","Implemented are the Wald-type statistic, a permuted version thereof as well as the ANOVA-type statistic for general factorial designs, even with non-normal error terms and/or heteroscedastic variances, for crossed designs with an arbitrary number of factors and nested designs with up to three factors. Friedrich et al. (2017) <doi:10.18637/jss.v079.c01>.","We implement two main functions. The first function uses a given grouped and/or right-censored grouping scheme and empirical data to infer parameters, and implements chi-square goodness-of-fit tests. The second function searches for the global optimal grouping scheme of grouped and/or right-censored count responses in surveys.","These are GreedyExperimentalDesign Java dependency libraries. Note: this package has no functionality of its own and should not be installed as a standalone package without GreedyExperimentalDesign.","Provides functions for designing phase II clinical trials adjusting for the heterogeneity of the population using known subgroups or historical controls.","Based on hierarchical modeling, this package provides a few practical functions to find and present the optimal designs for a pooled NGS design.","A collection of several utility functions related to binary incomplete block designs. The package contains function to generate A- and D-efficient binary incomplete block designs with given numbers of treatments, number of blocks and block size. The package also contains function to generate an incomplete block design with specified concurrence matrix. There are functions to generate balanced treatment incomplete block designs and incomplete block designs for test versus control treatments comparisons with specified concurrence matrix. Package also allows performing analysis of variance of data and computing estimated marginal means of factors from experiments using a connected incomplete block design. Tests of hypothesis of treatment contrasts in incomplete block design set up is supported.","Finds optimal designs for nonlinear models using a metaheuristic algorithm called Imperialist Competitive Algorithm (ICA). See, for details, Masoudi et al. (2017) <doi:10.1016/j.csda.2016.06.014> and Masoudi et al. (2019) <doi:10.1080/10618600.2019.1601097>.","Generates efficient designs for discrete choice experiments based on the multinomial logit model, and individually adapted designs for the mixed multinomial logit model. The generated designs can be presented on screen and choice data can be gathered using a shiny application. Traets F, Sanchez G, and Vandebroek M (2020) <doi:10.18637/jss.v096.i03>.","Contains functions to check the isomorphism of multi-stage factorial designs with randomisation restrictions based on balanced spreads and balanced covering stars of PG(n-1,2) as described in Spencer, Ranjan and Mendivil (2019) <doi:10.1007/s42519-019-0064-5>.","A collection of several functions related to construction and analysis of incomplete split-plot designs. The package contains functions to obtain and analyze incomplete split-plot designs for three kinds of situations namely (i) when blocks are complete with respect to main plot treatments and main plots are incomplete with respect to subplot treatments, (ii) when blocks are incomplete with respect to main plot treatments and main plots are complete with respect to subplot treatments and (iii) when blocks are incomplete with respect to main plot treatments and main plots are incomplete with respect to subplot treatments.","An implementation of randomization-based hypothesis testing for three different estimands in a cluster-randomized encouragement experiment. The three estimands include (1) testing a cluster-level constant proportional treatment effect (Fisher's sharp null hypothesis), (2) pooled effect ratio, and (3) average cluster effect ratio. To test the third estimand, user needs to install 'Gurobi' (>= 9.0.1) optimizer via its R API. Please refer to <https://www.gurobi.com/documentation/9.0/refman/ins_the_r_package.html>.","We developed a package 'Keyboard' for designing single-agent, drug-combination, or phase I/II dose-finding clinical trials. The 'Keyboard' designs are novel early phase trial designs that can be implemented simply and transparently, similar to the 3+3 design, but yield excellent performance, comparable to those of more-complicated, model-based designs (Yan F, Mandrekar SJ, Yuan Y (2017) <doi:10.1158/1078-0432.CCR-17-0220>, Li DH, Whitmore JB, Guo W, Ji Y. (2017) <doi:10.1158/1078-0432.CCR-16-1125>, Liu S, Johnson VE (2016) <doi:10.1093/biostatistics/kxv040>, Zhou Y, Lee JJ, Yuan Y (2019) <doi:10.1002/sim.8475>, Pan H, Lin R, Yuan Y (2020) <doi:10.1016/j.cct.2020.105972). The 'Keyboard' package provides tools for designing, conducting, and analyzing single-agent, drug-combination, and phase I/II dose-finding clinical trials.","Contains data sets to accompany the book: Lazic SE (2016). \"Experimental Design for Laboratory Biologists: Maximising Information and Improving Reproducibility\". Cambridge University Press.","this package provides functions for Finding Locally D-optimal designs for Logistic, Negative Binomial, Poisson, Michaelis-Menten, Exponential, Log-Linear, Emax, Richards, Weibull and Inverse Quadratic regression models and also functions for auto-constructing Fisher information matrix and Frechet derivative based on some input variables and without user-interfere.","Contains different algorithms and construction methods for efficient Latin hypercube designs (LHDs) with flexible sizes. Our package is comprehensive since it is capable of generating maximin distance LHDs, maximum projection LHDs, and orthogonal and nearly orthogonal LHDs. Documentations for all the algorithms and construction methods include informative explanation along with corresponding references. This package is particularly useful in the area of Design and Analysis of Experiments (DAE). More specifically, design of computer experiments.","Generate maximum projection (MaxPro) designs for quantitative and/or qualitative factors. Details of the MaxPro criterion can be found in: (1) Joseph, Gul, and Ba. (2015) \"Maximum Projection Designs for Computer Experiments\", Biometrika, 102, 371-380, and (2) Joseph, Gul, and Ba. (2018) \"Designing Computer Experiments with Multiple Types of Factors: The MaxPro Approach\", Journal of Quality Technology, to appear.","Find optimal microsampling designs for non-compartmental pharacokinetic analysis using a general simulation methodology: Algorithm III of Barnett, Helen, Helena Geys, Tom Jacobs, and Thomas Jaki. (2017) \"Optimal Designs for Non-Compartmental Analysis of Pharmacokinetic Studies. (currently unpublished)\" This methodology consist of (1) specifying a pharmacokinetic model including variability among animals; (2) generating possible sampling times; (3) evaluating performance of each time point choice on simulated data; (4) generating possible schemes given a time point choice and additional constraints and finally (5) evaluating scheme performance on simulated data. The default settings differ from the article of Barnett and others, in the default pharmacokinetic model used and the parameterization of variability among animals. Details can be found in the package vignette. A 'shiny' web application is included, which guides users from model parametrization to optimal microsampling scheme.","Functions for creating designs for mixture experiments, making ternary contour plots, and making mixture effect plots.","mkssd is a package that generates efficient balanced non-aliased multi-level k-circulant supersaturated designs by interchanging the elements of the generator vector. The package tries to generate a supersaturated design that has chisquare efficiency more than user specified efficiency level (mef). The package also displays the progress of generation of an efficient multi-level k-circulant design through a progress bar. The progress of 100% means that one full round of interchange is completed. More than one full round (typically 4-5 rounds) of interchange may be required for larger designs.","Generate the optimal maximin distance, minimax distance (only for low dimensions), and maximum projection designs within the class of Latin hypercube designs efficiently for computer experiments. Generate Pareto front optimal designs for each two of the three criteria and all the three criteria within the class of Latin hypercube designs efficiently. Provide criterion computing functions. References of this package can be found in Morris, M. D. and Mitchell, T. J. (1995) <doi:10.1016/0378-3758(94)00035-T>, Lu Lu and Christine M. Anderson-CookTimothy J. Robinson (2011) <doi:10.1198/Tech.2011.10087>, Joseph, V. R., Gul, E., and Ba, S. (2015) <doi:10.1093/biomet/asv002>.","mxkssd is a package that generates efficient balanced mixed-level k-circulant supersaturated designs by interchanging the elements of the generator vector. The package tries to generate a supersaturated design that has EfNOD efficiency more than user specified efficiency level (mef). The package also displays the progress of generation of an efficient mixed-level k-circulant design through a progress bar. The progress of 100 per cent means that one full round of interchange is completed. More than one full round (typically 4-5 rounds) of interchange may be required for larger designs.","Calculate various functions needed for design and monitoring clinical trials with negative binomial endpoint with variable follow-up. This version has a few changes compared to the previous version 1.0.0, including (1) correct a typo in Type 1 censoring, mtbnull=bnull and (2) restructure the code to account for shape parameter equal to zero, i.e. Poisson scenario.","Helps a clinical trial team discuss the clinical goals of a well-defined biomarker with a diagnostic, staging, prognostic, or predictive purpose. From this discussion will come a statistical plan for a (non-randomized) validation trial. Both prospective and retrospective trials are supported. In a specific focused discussion, investigators should determine the range of \"discomfort\" for the NNT, number needed to treat. The meaning of the discomfort range, [NNTlower, NNTupper], is that within this range most physicians would feel discomfort either in treating or withholding treatment. A pair of NNT values bracketing that range, NNTpos and NNTneg, become the targets of the study's design. If the trial can demonstrate that a positive biomarker test yields an NNT less than NNTlower, and that a negative biomarker test yields an NNT less than NNTlower, then the biomarker may be useful for patients. A highlight of the package is visualization of a \"contra-Bayes\" theorem, which produces criteria for retrospective case-controls studies.","Nonparametric Tests for Main Effects, Simple Effects and Interaction Effect with Censored Data and Two Factorial Influencing Variables.","Returns nonparametric aligned rank tests for the interaction in two-way factorial designs, on R data sets with repeated measures in 'wide' format. Five ANOVAs tables are reported. A PARAMETRIC one on the original data, one for a CHECK upon the interaction alignments, and three aligned rank tests: one on the aligned REGULAR, one on the FRIEDMAN, and one on the KOCH ranks. In these rank tests, only the resulting values for the interaction are relevant.","Implements a nonparametric statistical test for rank or score data from partially-balanced incomplete block-design experiments.","Implements the objective Bayesian methodology proposed in Consonni and Deldossi in order to choose the optimal experiment that better discriminate between competing models, see Deldossi and Nai Ruscone (2020) <doi:10.18637/jss.v094.i02>.","Calculate the optimal sample allocation that produces smallest variance of a treatment effect or the highest statistical power for experimental studies under a budget constraint, perform power analyses with and without accommodating cost structures of sampling, and calculate the relative efficiency between two sample allocations. The references for the proposed methods are: (1) Shen, Z. (2019). Optimal sample allocation in multilevel Experiments. (Doctoral dissertation). University of Cincinnati, Cincinnati, OH. (2) Shen, Z., & Kelcey, B. (in press). Optimal sample allocation under unequal costs in cluster-randomized trials. Journal of Educational and Behavioral Statistics. (3) Champely., S. (2018). pwr: Basic functions for power analysis (Version 1.2-2) [Software]. Available from <https://CRAN.R-project.org/package=pwr>.","Several function related to Experimental Design are implemented here, see \"Optimal Experimental Design with R\" by Rasch D. et. al (ISBN 9781439816974).","Obtain and evaluate various optimal designs for the 3, 4, and 5-parameter logistic models. The optimal designs are obtained based on the numerical algorithm in Hyun, Wong, Yang (2018) <doi:10.18637/jss.v083.i05>.","Computes A-, MV-, D- and E-optimal or near-optimal block designs for two-colour cDNA microarray experiments using the linear fixed effects and mixed effects models where the interest is in a comparison of all possible elementary treatment contrasts. The algorithms used in this package are based on the treatment exchange and array exchange algorithms of Debusho, Gemechu and Haines (2016, unpublished). The package also provides an optional method of using the graphical user interface (GUI) R package tcltk to ensure that it is user friendly.","Compute optimal experimental designs that measure the slope divided by the intercept.","Functions to find near-optimal multi-stage designs for continuous outcomes.","Algorithms for D-, A-, I-, and c-optimal designs. Some of the functions in this package require the 'gurobi' software and its accompanying R package. For their installation, please follow the instructions at <https://www.gurobi.com> and the file gurobi_inst.txt, respectively.","Computes A-, MV-, D- and E-optimal or near-optimal row-column designs for two-colour cDNA microarray experiments using the linear fixed effects and mixed effects models where the interest is in a comparison of all pairwise treatment contrasts. The algorithms used in this package are based on the array exchange and treatment exchange algorithms adopted from Debusho, Gemechu and Haines (2016, unpublished) algorithms after adjusting for the row-column designs setup. The package also provides an optional method of using the graphical user interface (GUI) R package tcltk to ensure that it is user friendly.","An implementation of optimal weight exchange algorithm Yang(2013) <doi:10.1080/01621459.2013.806268> for three models. They are Crossover model with subject dropout, crossover model with proportional first order residual effects and interference model. You can use it to find either A-opt or D-opt approximate designs. Exact designs can be automatically rounded from approximate designs and relative efficiency is provided as well.","The PBIB designs are important type of incomplete block designs having wide area of their applications for example in agricultural experiments, in plant breeding, in sample surveys etc. This package constructs various series of PBIB designs and assists in checking all the necessary conditions of PBIB designs and the association scheme on which these designs are based on. It also assists in calculating the efficiencies of PBIB designs with any number of associate classes. The package also constructs Youden-m square designs which are Row-Column designs for the two-way elimination of heterogeneity. The incomplete columns of these Youden-m square designs constitute PBIB designs. With the present functionality, the package will be of immense importance for the researchers as it will help them to construct PBIB designs, to check if their PBIB designs and association scheme satisfy various necessary conditions for the existence, to calculate the efficiencies of PBIB designs based on any association scheme and to construct Youden-m square designs for the two-way elimination of heterogeneity. R. C. Bose and K. R. Nair (1939) <http://www.jstor.org/stable/40383923>.","Construction method of nested resolvable designs from a projective geometry defined on Galois field of order 2. The obtained Resolvable designs are used to build uniform design. The presented results are based on <https://eudml.org/doc/219563> and A. Boudraa et al. (See references).","An implementation of Bayesian single-arm phase II design methods for binary outcome based on posterior probability (Thall and Simon (1994) <doi:10.2307/2533377>) and predictive probability (Lee and Liu (2008) <doi:10.1177/1740774508089279>).","Implements Bayesian phase I repeated measurement design that accounts for multidimensional toxicity endpoints and longitudinal efficacy measure from multiple treatment cycles. The package provides flags to fit a variety of model-based phase I design, including 1 stage models with or without individualized dose modification, 3-stage models with or without individualized dose modification, etc. Functions are provided to recommend dosage selection based on the data collected in the available patient cohorts and to simulate trial characteristics given design parameters. Yin, Jun, et al. (2017) <doi:10.1002/sim.7134>.","Implements our Bayesian phase I repeated measurement design that accounts for multidimensional toxicity endpoints from multiple treatment cycles. The package also provides a novel design to account for both multidimensional toxicity endpoints and early-stage efficacy endpoints in the phase I design. For both designs, functions are provided to recommend the next dosage selection based on the data collected in the available patient cohorts and to simulate trial characteristics given design parameters. Yin, Jun, et al. (2017) <doi:10.1002/sim.7134>.","A collection of scripts and data files for the statistics text: \"Process Improvement using Data\" <https://learnche.org/pid> and the online course \"Experimentation for Improvement\" found on Coursera. The package contains code for designed experiments, data sets and other convenience functions used in the book.","Implements the Product of Independent beta Probabilities dose Escalation (PIPE) design for dual-agent Phase I trials as described in Mander AP, Sweeting MJ (2015) <DOI:10.1002/sim.6434>.","Optimal experimental designs for both population and individual studies based on nonlinear mixed-effect models. Often this is based on a computation of the Fisher Information Matrix. This package was developed for pharmacometric problems, and examples and predefined models are available for these types of systems. The methods are described in Nyberg et al. (2012) <doi:10.1016/j.cmpb.2012.05.005>, and Foracchia et al. (2004) <doi:10.1016/S0169-2607(03)00073-7>.","Basic functions for power analysis and effect size calculation.","Tools for the design of QTL experiments","The rankFD() function calculates the Wald-type statistic (WTS) and the ANOVA-type statistic (ATS) for nonparametric factorial designs, e.g., for count, ordinal or score data in a crossed design with an arbitrary number of factors.","The package provides a platform-independent GUI for design of experiments. It is implemented as a plugin to the R-Commander, which is a more general graphical user interface for statistics in R based on tcl/tk. DoE functionality can be accessed through the menu Design that is added to the R-Commander menus.","The regression discontinuity (RD) design is a popular quasi-experimental design for causal inference and policy evaluation. Under the local randomization approach, RD designs can be interpreted as randomized experiments inside a window around the cutoff. This package provides tools to perform randomization inference for RD designs under local randomization: rdrandinf() to perform hypothesis testing using randomization inference, rdwinselect() to select a window around the cutoff in which randomization is likely to hold, rdsensitivity() to assess the sensitivity of the results to different window lengths and null hypotheses and rdrbounds() to construct Rosenbaum bounds for sensitivity to unobserved confounders. See Cattaneo, Titiunik and Vazquez-Bare (2016) <https://rdpackages.github.io/references/Cattaneo-Titiunik-VazquezBare_2016_Stata.pdf> for further methodological details.","The regression discontinuity (RD) design is a popular quasi-experimental design for causal inference and policy evaluation. The 'rdmulti' package provides tools to analyze RD designs with multiple cutoffs or scores: rdmc() estimates pooled and cutoff specific effects for multi-cutoff designs, rdmcplot() draws RD plots for multi-cutoff designs and rdms() estimates effects in cumulative cutoffs or multi-score designs. See Cattaneo, Titiunik and Vazquez-Bare (2020) <https://rdpackages.github.io/references/Cattaneo-Titiunik-VazquezBare_2020_Stata.pdf> for further methodological details.","Regression-discontinuity (RD) designs are quasi-experimental research designs popular in social, behavioral and natural sciences. The RD design is usually employed to study the (local) causal effect of a treatment, intervention or policy. This package provides tools for data-driven graphical and analytical statistical inference in RD designs: rdrobust() to construct local-polynomial point estimators and robust confidence intervals for average treatment effects at the cutoff in Sharp, Fuzzy and Kink RD settings, rdbwselect() to perform bandwidth selection for the different procedures implemented, and rdplot() to conduct exploratory data analysis (RD plots).","Analyzes data from repeated low-dose challenge experiments and provide vaccine efficacy estimates. In addition, this package can provide guidance to design repeated low-dose challenge studies.","A collection of functions for numerical construction of optimal discriminating designs. At the current moment T-optimal designs (which maximize the lower bound for the power of F-test for regression model discrimination), KL-optimal designs (for lognormal errors) and their robust analogues can be calculated with the package.","Provides functions to generate response-surface designs, fit first- and second-order response-surface models, make surface plots, obtain the path of steepest ascent, and do canonical analysis. A good reference on these methods is Chapter 10 of Wu, C-F J and Hamada, M (2009) \"Experiments: Planning, Analysis, and Parameter Design Optimization\" ISBN 978-0-471-69946-0.","Produces tables with the level of replication (number of replicates) and the experimental uncoded values of the quantitative factors to be used for rotatable Central Composite Design (CCD) experimentation and a 2-D contour plot of the corresponding variance of the predicted response according to Mead et al. (2012) <doi:10.1017/CBO9781139020879> design_ccd(), and analyzes CCD data with response surface methodology ccd_analysis(). A rotatable CCD provides values of the variance of the predicted response that are concentrically distributed around the average treatment combination used in the experimentation, which with uniform precision (implied by the use of several replicates at the average treatment combination) improves greatly the search and finding of an optimum response. These properties of a rotatable CCD represent undeniable advantages over the classical factorial design, as discussed by Panneton et al. (1999) <doi:10.13031/2013.13267> and Mead et al. (2012) <doi:10.1017/CBO9781139020879.018> among others.","Package provides the possibility to sampling complete datasets from a normal distribution to simulate cluster randomized trails for different study designs.","Provides a set of tools for estimating hierarchical linear models and effect sizes based on data from single-case designs. Functions are provided for calculating standardized mean difference effect sizes that are directly comparable to standardized mean differences estimated from between-subjects randomized experiments, as described in Hedges, Pustejovsky, and Shadish (2012) <DOI:10.1002/jrsm.1052>; Hedges, Pustejovsky, and Shadish (2013) <DOI:10.1002/jrsm.1086>; and Pustejovsky, Hedges, and Shadish (2014) <DOI:10.3102/1076998614547577>. Includes an interactive web interface.","A collection of functions that creates graphs with error bars for data collected from one-way or higher factorial designs.","A generalization of the statistic used in Friedman's ANOVA method and in Durbin's rank test. This nonparametric statistical test is useful for the data obtained from block designs with missing observations occurring randomly. A resulting p-value is based on the chi-squared distribution and Monte Carlo method.","Generates and evaluates D, I, A, Alias, E, T, and G optimal designs. Supports generation and evaluation of blocked and split/split-split/.../N-split plot designs. Includes parametric and Monte Carlo power evaluation functions, and supports calculating power for censored responses. Provides a framework to evaluate power using functions provided in other packages or written by the user. Includes a Shiny graphical user interface that displays the underlying code used to create and evaluate the design to improve ease-of-use and make analyses more reproducible.","Generate the optimal Latin Hypercube Designs (LHDs) for computer experiments with quantitative factors and the optimal Sliced Latin Hypercube Designs (SLHDs) for computer experiments with both quantitative and qualitative factors. Details of the algorithm can be found in Ba, S., Brenneman, W. A. and Myers, W. R. (2015), \"Optimal Sliced Latin Hypercube Designs,\" Technometrics. Important function in this package is \"maximinSLHD\".","Provides methods for generating, exploring and executing seamless Phase II-III designs of Lai, Lavori and Shih using generalized likelihood ratio statistics. Includes pdf and source files that describe the entire R implementation with the relevant mathematical details.","The package is used for calibrating the design parameters for single-to-double arm transition design proposed by Shi and Yin (2017). The calibration is performed via numerical enumeration to find the optimal design that satisfies the constraints on the type I and II error rates.","Functions to perform simulations of ANOVA designs of up to three factors. Calculates the observed power and average observed effect size for all main effects and interactions in the ANOVA, and all simple comparisons between conditions. Includes functions for analytic power calculations and additional helper functions that compute effect sizes for ANOVA designs, observed error rates in the simulations, and functions to plot power curves. Please see Lakens, D., & Caldwell, A. R. (2019). \"Simulation-Based Power-Analysis for Factorial ANOVA Designs\". <doi:10.31234/osf.io/baxsf>.","Provides basic functions that support an implementation of (discrete) choice experiments (CEs). CEs is a question-based survey method measuring people's preferences for goods/services and their characteristics. Refer to Louviere et al. (2000) <doi:10.1017/CBO9780511753831> for details on CEs, and Aizaki (2012) <doi:10.18637/jss.v050.c02> for the package.","A set of tools for examining the design and analysis aspects of stepped wedge cluster randomized trials (SW CRT) based on a repeated cross-sectional sampling scheme (Hussey MA and Hughes JP (2007) Contemporary Clinical Trials 28:182-191. <doi:10.1016/j.cct.2006.05.007>).","Design and analyze three-arm non-inferiority or superiority trials which follow a gold-standard design, i.e. trials with an experimental treatment, an active, and a placebo control. Method for the following distributions are implemented: Poisson (Mielke and Munk (2009) <arXiv:0912.4169>), negative binomial (Muetze et al. (2016) <doi:10.1002/sim.6738>), normal (Pigeot et al. (2003) <doi:10.1002/sim.1450>; Hasler et al. (2009) <doi:10.1002/sim.3052>), binary (Friede and Kieser (2007) <doi:10.1002/sim.2543>), nonparametric (Muetze et al. (2017) <doi:10.1002/sim.7176>), exponential (Mielke and Munk (2009) <arXiv:0912.4169>).","Functions for analysis of bulk data generated from experiments in Randomized block design as per Panse and Sukhatme (1954) <https://books.google.co.in/books?id=Efo9AAAAYAAJ>. Computes analysis of variance; Descriptive statistics parameter like Mean, Minimum, Maximum, CV, Standard error of mean, Standard Error of deviation, CD; Genetic parameter statistics Genotypic Coefficient of Variation, Phenotypic Coefficient of Variation, Heritability in broad sense, Genetic Advance and Genetic Advance per cent mean; Variance and Co-variance matrix of genotypic, phenotypic and environmental; Correlation of genotypic, phenotypic and environmental. Further includes directly publication ready tables.","A collection of clinical trial designs and methods, implemented in 'rstan' and R, including: the Continual Reassessment Method by O'Quigley et al. (1990) <doi:10.2307/2531628>; EffTox by Thall & Cook (2004) <doi:10.1111/j.0006-341X.2004.00218.x>; the two-parameter logistic method of Neuenschwander, Branson & Sponer (2008) <doi:10.1002/sim.3230>; and the Augmented Binary method by Wason & Seaman (2013) <doi:10.1002/sim.5867>; and more. We provide functions to aid model-fitting and analysis. The 'rstan' implementations may also serve as a cookbook to anyone looking to extend or embellish these models. We hope that this package encourages the use of Bayesian methods in clinical trials. There is a preponderance of early phase trial designs because this is where Bayesian methods are used most. If there is a method you would like implemented, please get in touch.","Calculate optimal Zhong's two-/three-stage Phase II designs (see Zhong (2012) <doi:10.1016/j.cct.2012.07.006>). Generate Target Toxicity decision table for Phase I dose-finding (two-/three-stage). This package also allows users to run dose-finding simulations based on customized decision table.","Provides half-normal plots, reference plots, and Pareto plots of effects from an unreplicated experiment, along with various pseudo-standard-error measures, simulated reference distributions, and other tools. Many of these methods are described in Daniel C. (1959) <doi:10.1080/00401706.1959.10489866> and/or Lenth R.V. (1989) <doi:10.1080/00401706.1989.10488595>, but some new approaches are added and integrated in one package.","Facilities for constructing variance dispersion graphs, fraction- of-design-space plots and similar graphics for exploring the properties of experimental designs. The design region is explored via random sampling, which allows for more flexibility than traditional variance dispersion graphs. A formula interface is leveraged to provide access to complex model formulae. Graphics can be constructed simultaneously for multiple experimental designs and/or multiple model formulae. Instead of using pointwise optimization to find the minimum and maximum scaled prediction variance curves, which can be inaccurate and time consuming, this package uses quantile regression as an alternative.","Uses a modification of the published FORTRAN code in \"A Computer Program for Generating Variance Dispersion Graphs\" by G. Vining, Journal of Quality Technology, Vol. 25 No. 1 January 1993, to produce variance dispersion graphs. Also produces fraction of design space plots, and contains data frames for several minimal run response surface designs.","Functions for creating variance dispersion graphs, fraction of design space plots, and contour plots of scaled prediction variances for second-order response surface designs in spherical and cuboidal regions. Also, some standard response surface designs can be generated.","A Bayesian Phase I cancer vaccine trial design is implemented in this package. The design allows simultaneous evaluation of safety and immunogenicity outcomes in the context of vaccine studies. See Wang (2019) <DOI:10.1002/sim.8021> for the details of the Phase I cancer vaccine trial design.","Provide tools for finding multiple-objective optimal designs for estimating the shape of dose-response, the ED50 (the dose producing an effect midway between the expected responses at the extreme doses) and the MED (the minimum effective dose level) for the 2,3,4-parameter logistic models and for evaluating its efficiencies for the three objectives. The acronym VNM stands for V-algorithm using Newton Raphson method to search multiple-objective optimal design."],["yes","no","yes","yes","yes","no","no","no","yes","no","no","yes","no","yes","no","yes","yes","yes","no","no","no","yes","yes","yes","no","no","yes","yes","no","yes","yes","no","yes","yes","yes","yes","yes","yes","yes","yes","yes","no","yes","no","yes","no","no","yes","no","no","no","no","yes","no","yes","no","no","no","yes","yes","yes","no","no","no","no","no","yes","yes","yes","no","no","no","no","no","no","yes","no","yes","no","yes","yes","no","yes","no","no","no","no","no","no","yes","yes","no","yes","yes","yes","yes","yes","no","yes","yes","yes","no","no","yes","yes","yes","yes","yes","no","yes","no","no","no","no","yes","yes","yes","no","no","no","no","yes","yes","yes","no","no","yes","no","yes","no","no","no","yes","yes","yes","yes","no","yes"],["no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no"],["<div class='tooltip'>ace<span class='tooltiptext'>These functions implement the approximate coordinate exchange (ACE) algorithm (Overstall & Woods, 2017) for finding optimal Bayesian experimental designs by maximising an approximation to an intractable expected utility function.<\/span><\/div>, <div class='tooltip'>aceglm<span class='tooltiptext'>Functions implementing the approximate coordinate exchange (ACE) algorithm (Overstall & Woods, 2017) for finding Bayesian optimal experimental designs for generalised linear models (GLMs).<\/span><\/div>, <div class='tooltip'>acenlm<span class='tooltiptext'>Functions implementing the approximate coordinate exchange algorithm (Overstall & Woods, 2017) for finding optimal Bayesian designs for non-linear regression models.<\/span><\/div>, <div class='tooltip'>acephase1<span class='tooltiptext'>These functions implement the approximate coordinate exchange (ACE) algorithm (Overstall & Woods, 2017) for finding optimal Bayesian experimental designs by maximising an approximation to an intractable expected utility function.<\/span><\/div>, <div class='tooltip'>acephase2<span class='tooltiptext'>These functions implement the approximate coordinate exchange (ACE) algorithm (Overstall & Woods, 2017) for finding optimal Bayesian experimental designs by maximising an approximation to an intractable expected utility function.<\/span><\/div>, <div class='tooltip'>assess<span class='tooltiptext'>Calculates approximations to the expected utility for two designs.<\/span><\/div>, <div class='tooltip'>inideshlrnsel<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>inideshlrsig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>inideslrnsel<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>inideslrsig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdesbeetle<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdescomp15bad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdescomp15sig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdescomp15sigDRS<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdescomp18bad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeshlrbaa<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeshlrbad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeshlrnsel<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeshlrsig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeslinmod<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeslrbaa<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeslrbad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeslrnsel<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>optdeslrsig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>pace<span class='tooltiptext'>These functions implement the approximate coordinate exchange (ACE) algorithm (Overstall & Woods, 2017) for finding optimal Bayesian experimental designs by maximising an approximation to an intractable expected utility function.<\/span><\/div>, <div class='tooltip'>paceglm<span class='tooltiptext'>Functions implementing the approximate coordinate exchange (ACE) algorithm (Overstall & Woods, 2017) for finding Bayesian optimal experimental designs for generalised linear models (GLMs).<\/span><\/div>, <div class='tooltip'>pacenlm<span class='tooltiptext'>Functions implementing the approximate coordinate exchange algorithm (Overstall & Woods, 2017) for finding optimal Bayesian designs for non-linear regression models.<\/span><\/div>, <div class='tooltip'>utilbeetle<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilcomp15bad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilcomp15sig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilcomp15sigDRS<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilcomp18bad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilhlrbaa<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilhlrbad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilhlrnsel<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilhlrsig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utilityglm<span class='tooltiptext'>Generates an approximate utility function for generalised linear models and non-linear regression models.<\/span><\/div>, <div class='tooltip'>utilitynlm<span class='tooltiptext'>Generates an approximate utility function for generalised linear models and non-linear regression models.<\/span><\/div>, <div class='tooltip'>utillinmod<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utillrbaa<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utillrbad<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utillrnsel<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>, <div class='tooltip'>utillrsig<span class='tooltiptext'>This suite of functions implement the examples in Overstall & Woods (2017).<\/span><\/div>","<div class='tooltip'>AverageN2<span class='tooltiptext'>Implements the L1-norm of the design's stage-two sample size function.The average of the stage-two sample size without weighting withthe data distribution is computed.This can be interpreted as integration over a unifrom prior onthe continuation region.<\/span><\/div>, <div class='tooltip'>Binomial<span class='tooltiptext'>Implements the normal approximation for a test on rates.The reponse rate in the control group,htmlr<sub>C<\/sub>r_C, has to be specified byrate_control.The null hypothesis is:htmlr<sub>E<\/sub> &le; r<sub>C<\/sub>r_E <= r_C,where htmlr<sub>E<\/sub>r_E denotes the response ratein the invervention group.It is tested against the alternativehtmlr<sub>E<\/sub> > r<sub>C<\/sub>r_E > r_C.The test statistic is given ashtmlX<sub>1<\/sub> = &radic;n (r<sub>E<\/sub> - r<sub>C<\/sub>) / &radic;(2 r<sub>0<\/sub> (1-r<sub>0<\/sub>))X_1 = \\sqrt{n}(r_E - r_C) / \\sqrt{2 r_0 (1- r_0)},where htmlr<sub>0<\/sub>r_0 denotes the mean betweenhtmlr<sub>E<\/sub>r_E andhtmlr<sub>C<\/sub>r_C in the two-armed case,and htmlr<sub>E<\/sub>r_E in the one-armed case.#'All priors have to be defined for the rate differencehtmlr<sub>E<\/sub> - r<sub>C<\/sub>r_E - r_C.<\/span><\/div>, <div class='tooltip'>bounds<span class='tooltiptext'>bounds() returns the range of the support of a prior or data distribution.<\/span><\/div>, <div class='tooltip'>c2<span class='tooltiptext'>Methods to access the stage-two critical values of aTwoStageDesign.c2 returns the stage-two critical value conditional on the stage-one teststatistic.<\/span><\/div>, <div class='tooltip'>composite<span class='tooltiptext'>composite defines new composite scores by point-wise evaluation ofscores in any valid numerical expression.<\/span><\/div>, <div class='tooltip'>condition<span class='tooltiptext'>Restrict an object of class Prior to a sub-interval andre-normalize the PDF.<\/span><\/div>, <div class='tooltip'>ConditionalPower<span class='tooltiptext'>This score evaluates htmlP[X<sub>2<\/sub> > c2(design, X<sub>1<\/sub>) | X<sub>1<\/sub> = x<sub>1<\/sub>]\\boldsymbol{P}[X_2 > c_2(design, X_1)|X_1 = x_1].Note that the distribution of htmlX<sub>2<\/sub>X_2 is the posterior predictive afterobserving htmlX<sub>1<\/sub> = x<sub>1<\/sub>X_1 = x_1.<\/span><\/div>, <div class='tooltip'>ConditionalSampleSize<span class='tooltiptext'>This score simply evaluates n(d, x1) for a design d and thefirst-stage outcome x1.The data distribution and prior are only relevant when it is integrated.<\/span><\/div>, <div class='tooltip'>ContinuousPrior<span class='tooltiptext'>ContinuousPrior is a sub-class of Prior implementinga generic representation of continuous prior distributions over a compactinterval on the real line.<\/span><\/div>, <div class='tooltip'>cumulative_distribution_function<span class='tooltiptext'>cumulative_distribution_function evaluates the cumulative distributionfunction of a specific distribution dist at a point x.<\/span><\/div>, <div class='tooltip'>evaluate<span class='tooltiptext'>In adoptr scores are used to assess the performance of a design.This can be done either conditionally on the observed stage-one outcomeor unconditionally.Consequently, score objects are either of class ConditionalScore orUnconditionalScore.<\/span><\/div>, <div class='tooltip'>expectation<span class='tooltiptext'>Computes the expected value of a vectorized, univariate function fwith respect to a distribution dist.I.e., htmlE[f(X)]\\boldsymbol{E}\\big[f(X)\\big]E[f(X)].<\/span><\/div>, <div class='tooltip'>expected<span class='tooltiptext'>In adoptr scores are used to assess the performance of a design.This can be done either conditionally on the observed stage-one outcomeor unconditionally.Consequently, score objects are either of class ConditionalScore orUnconditionalScore.<\/span><\/div>, <div class='tooltip'>ExpectedSampleSize<span class='tooltiptext'>This score simply evaluates n(d, x1) for a design d and thefirst-stage outcome x1.The data distribution and prior are only relevant when it is integrated.<\/span><\/div>, <div class='tooltip'>get_initial_design<span class='tooltiptext'>The optimization method minimize requires an initialdesign for optimization.The function get_initial_design provides an initial guess based on afixed design that fulfills constraints on type I error rate and power.Note that a situation-specific initial design may be much more efficient.<\/span><\/div>, <div class='tooltip'>get_lower_boundary_design<span class='tooltiptext'>The optimization method minimize is based on the packagenloptr. This requires upper and lower boundaries for optimization.Such boundaries can be computed via lower_boundary_designrespectively upper_boundary_design.They are implemented by default in minimize.Note that minimize allows the user to define its ownboundary designs, too.<\/span><\/div>, <div class='tooltip'>get_upper_boundary_design<span class='tooltiptext'>The optimization method minimize is based on the packagenloptr. This requires upper and lower boundaries for optimization.Such boundaries can be computed via lower_boundary_designrespectively upper_boundary_design.They are implemented by default in minimize.Note that minimize allows the user to define its ownboundary designs, too.<\/span><\/div>, <div class='tooltip'>GroupSequentialDesign<span class='tooltiptext'>Group-sequential designs are a sub-class of the TwoStageDesignclass with constant stage-two sample size.See TwoStageDesign for slot details.Any group-sequential design can be converted to a fully flexibleTwoStageDesign (see examples section).<\/span><\/div>, <div class='tooltip'>make_fixed<span class='tooltiptext'>The methods make_fixed and make_tunable can be used to modifythe 'tunability' status of parameters in a TwoStageDesignobject.Tunable parameters are optimized over, non-tunable ('fixed') parameters areconsidered given and not altered during optimization.<\/span><\/div>, <div class='tooltip'>make_tunable<span class='tooltiptext'>The methods make_fixed and make_tunable can be used to modifythe 'tunability' status of parameters in a TwoStageDesignobject.Tunable parameters are optimized over, non-tunable ('fixed') parameters areconsidered given and not altered during optimization.<\/span><\/div>, <div class='tooltip'>minimize<span class='tooltiptext'>minimize takes an unconditional score anda constraint set (or no constraint) and solves the correspondingminimization problem usinghttps://cran.r-project.org/package=nloptrnloptr(using COBYLA by default).An initial design has to be defined. It is also possible to definelower- and upper-boundary designs. If this is not done, the boundaries aredetermined automatically heuristically.<\/span><\/div>, <div class='tooltip'>n<span class='tooltiptext'>Methods to access the stage-one, stage-two, or overall sample size of aTwoStageDesign.n1 returns the first-stage sample size of a design,n2 the stage-two sample size conditional on the stage-one teststatistic and n the overall sample size n1 + n2.Internally, objects of the class TwoStageDesign allow non-natural,real sample sizes to allow smooth optimization (cf. minimize fordetails).The optional argument round allows to switch between the internalreal representation and a rounded version (rounding to the next positiveinteger).<\/span><\/div>, <div class='tooltip'>n1<span class='tooltiptext'>Methods to access the stage-one, stage-two, or overall sample size of aTwoStageDesign.n1 returns the first-stage sample size of a design,n2 the stage-two sample size conditional on the stage-one teststatistic and n the overall sample size n1 + n2.Internally, objects of the class TwoStageDesign allow non-natural,real sample sizes to allow smooth optimization (cf. minimize fordetails).The optional argument round allows to switch between the internalreal representation and a rounded version (rounding to the next positiveinteger).<\/span><\/div>, <div class='tooltip'>N1<span class='tooltiptext'>N1 is a class that computes the n1 value of a design.This can be used as a score in minimize.<\/span><\/div>, <div class='tooltip'>n2<span class='tooltiptext'>Methods to access the stage-one, stage-two, or overall sample size of aTwoStageDesign.n1 returns the first-stage sample size of a design,n2 the stage-two sample size conditional on the stage-one teststatistic and n the overall sample size n1 + n2.Internally, objects of the class TwoStageDesign allow non-natural,real sample sizes to allow smooth optimization (cf. minimize fordetails).The optional argument round allows to switch between the internalreal representation and a rounded version (rounding to the next positiveinteger).<\/span><\/div>, <div class='tooltip'>Normal<span class='tooltiptext'>Implements a normal data distribution for z-values given an observed z-valueand stage size.Standard deviation is 1 and mean html&theta; &radic;n\\theta\\sqrt n wherehtml&theta;\\theta is the standardized effect size.The option two_armed can be set to decide whether a one-arm or atwo-arm design should be computed.<\/span><\/div>, <div class='tooltip'>OneStageDesign<span class='tooltiptext'>OneStageDesign implements a one-stage design as special case ofa two-stage design, i.e. as sub-class of TwoStageDesign.This is possible by defining htmln<sub>2<\/sub> = 0n_2=0,htmlc = c<sub>1<\/sub><sup>f<\/sup> = c<sub>1<\/sub><sup>e<\/sup>c = c_1^f = c_1^e,htmlc<sub>2<\/sub>(x<sub>1<\/sub>) = ifelse(x<sub>1<\/sub> < c, Inf, -Inf)c_2(x_1) = ifelse(x_1 < c, Inf, -Inf).No integration pivots etc are required (set to NaN).<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>PointMassPrior<span class='tooltiptext'>PointMassPrior is a sub-class of Priorrepresenting a univariate prior over a discrete set of points with positiveprobability mass.<\/span><\/div>, <div class='tooltip'>posterior<span class='tooltiptext'>Return posterior distribution given observing stage-one outcome.<\/span><\/div>, <div class='tooltip'>Power<span class='tooltiptext'>This score evaluates htmlP[X<sub>2<\/sub> > c2(design, X<sub>1<\/sub>) | X<sub>1<\/sub> = x<sub>1<\/sub>]\\boldsymbol{P}[X_2 > c_2(design, X_1)|X_1 = x_1].Note that the distribution of htmlX<sub>2<\/sub>X_2 is the posterior predictive afterobserving htmlX<sub>1<\/sub> = x<sub>1<\/sub>X_1 = x_1.<\/span><\/div>, <div class='tooltip'>predictive_cdf<span class='tooltiptext'>predictive_cdf() evaluates the predictive CDF of the model specifiedby a DataDistribution dist andPrior at the given stage-one outcome.<\/span><\/div>, <div class='tooltip'>predictive_pdf<span class='tooltiptext'>predictive_pdf() evaluates the predictive PDF of the model specifiedby a DataDistribution dist andPrior at the given stage-one outcome.<\/span><\/div>, <div class='tooltip'>probability_density_function<span class='tooltiptext'>probability_density_function evaluates the probability densityfunction of a specific distribution dist at a point x.<\/span><\/div>, <div class='tooltip'>quantile<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>simulate<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>subject_to<span class='tooltiptext'>subject_to(...) can be used to generate an object of classConstraintsCollection from an arbitrary number of (un)conditionalconstraints.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>tunable_parameters<span class='tooltiptext'>Get tunable parameters of a design as numeric vector viatunable_parameters or update a design object with a suitablevector of values for its tunable parameters.<\/span><\/div>, <div class='tooltip'>TwoStageDesign<span class='tooltiptext'>TwoStageDesign is the fundamental design class of theadoptr package.Formally, we represent a generic two-stage design as a five-tuplehtml(n<sub>1<\/sub>, c<sub>1<\/sub><sup>f<\/sup>, c<sub>1<\/sub><sup>e<\/sup>, n<sub>2<\/sub>(&middot;), c<sub>2<\/sub>(&middot;))\\big(n_1, c_1^f, c_1^e, n_2(\\cdot), c_2(\\cdot)\\big).Here, htmln<sub>1<\/sub>n_1 is the first-stage samplesize (per group), htmlc<sub>1<\/sub><sup>f<\/sup>c_1^fand htmlc<sub>1<\/sub><sup>e<\/sup>c_1^e areboundaries for early stopping for futility and efficacy, respectively.Since the trial design is a two-stage design, the elementshtmln<sub>2<\/sub>(&middot;)n_2(\\cdot) (stage-two samplesize) and htmlc<sub>2<\/sub>(&middot;)c_2(\\cdot)(stage-two critical value) are functions of the first-stage outcomehtmlX<sub>1<\/sub>=x<sub>1<\/sub>X_1=x_1.htmlX<sub>1<\/sub>X_1 denotes the first-stage teststatistic. A brief description on this definition of two-stage designs can beread https://kkmann.github.io/adoptr/articles/adoptr.htmlhere.For available methods, see the 'See Also' section at the end of this page.<\/span><\/div>, <div class='tooltip'>update<span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>AMMI<span class='tooltiptext'>Additive Main Effects and Multiplicative Interaction Models (AMMI) are widely used to analyze main effects and genotype by environment (GEN, ENV) interactions in multilocation variety trials. Furthermore, this function generates data to biplot, triplot graphs and analysis.<\/span><\/div>, <div class='tooltip'>AMMI.contour<span class='tooltiptext'>Draws a polygon or a circumference around the center of the Biplot with a proportional radio at the longest distance of the genotype.<\/span><\/div>, <div class='tooltip'>audpc<span class='tooltiptext'>Area Under Disease Progress Curve. The AUDPC measures the disease throughout a period. The AUDPC is the area that is determined by the sum of trapezes under the curve.<\/span><\/div>, <div class='tooltip'>audps<span class='tooltiptext'>A better estimate of disease progress is the area under the disease progress stairs (AUDPS). The AUDPS approach improves the estimation of disease progress by giving a weight closer to optimal to the first and last observations.<\/span><\/div>, <div class='tooltip'>bar.err<span class='tooltiptext'>It plots bars of the averages of treatments and standard error or standard deviance. It uses the objects generated by a procedure of comparison like LSD, HSD, Kruskal and Waller-Duncan.<\/span><\/div>, <div class='tooltip'>bar.group<span class='tooltiptext'>It plots bars of the averages of treatments to compare. It uses the objects generated by a procedure of comparison like LSD, HSD, Kruskall, Waller-Duncan, Friedman or Durbin. It can also display the 'average' value over each bar in a bar chart.<\/span><\/div>, <div class='tooltip'>BIB.test<span class='tooltiptext'>Analysis of variance BIB and comparison mean adjusted.<\/span><\/div>, <div class='tooltip'>carolina<span class='tooltiptext'>Statistic analysis of the Carolina I, II and III genetic designs.<\/span><\/div>, <div class='tooltip'>consensus<span class='tooltiptext'>The criterion of the consensus is to produce many trees by means of boostrap and to such calculate the relative frequency with members of the clusters.<\/span><\/div>, <div class='tooltip'>correl<span class='tooltiptext'>An exact correlation for ties or without ties. Methods of Kendall, Spearman and Pearson.<\/span><\/div>, <div class='tooltip'>correlation<span class='tooltiptext'>It obtains the coefficients of correlation and p-value between all the variables of a data table. The methods to apply are Pearson, Spearman , Kendall and lin's concordance index. In case of not specifying the method, the Pearson method will be used. The results are similar to SAS.<\/span><\/div>, <div class='tooltip'>cv.model<span class='tooltiptext'>It obtains the coefficient of variation of the experiment obtained by models lm() or aov()<\/span><\/div>, <div class='tooltip'>cv.similarity<span class='tooltiptext'>This process consists of finding the coefficient of the distances of similarity of binary tables (1 and 0) as used for scoring molecular marker data for presence and absence of PCR amplification products.<\/span><\/div>, <div class='tooltip'>DAU.test<span class='tooltiptext'>Analysis of variance Augmented block and comparison mean adjusted.<\/span><\/div>, <div class='tooltip'>delete.na<span class='tooltiptext'>In many situations it is required to omit the rows or columns less or greater with NA of the matrix.<\/span><\/div>, <div class='tooltip'>design.ab<span class='tooltiptext'>It generates a design of blocks, randomize and latin square for combined n. factors uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.alpha<span class='tooltiptext'>Generates an alpha designs starting from the alpha design fixing under the series formulated by Patterson and Williams. These designs are generated by the alpha arrangements. They are similar to the lattice designs, but the tables are rectangular s by k (with s blocks and k<s columns. The number of treatments should be equal to s*k and all the experimental units r*s*k (r replications).<\/span><\/div>, <div class='tooltip'>design.bib<span class='tooltiptext'>Creates Randomized Balanced Incomplete Block Design. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.crd<span class='tooltiptext'>It generates completely a randomized design with equal or different repetition. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.cyclic<span class='tooltiptext'>The cyclic design is a incomplete blocks designs, it is generated from a incomplete block initial of the size k, the plan is generated and randomized. The efficient and robust cyclic designs for 6 to 30 treatments, replications <= 10.<\/span><\/div>, <div class='tooltip'>design.dau<span class='tooltiptext'>These are designs for two types of treatments: the control treatments (common) and the increased treatments. The common treatments are applied in complete randomized blocks, and the increased treatments, at random. Each treatment should be applied in any block once only. It is understood that the common treatments are of a greater interest; the standard error of the difference is much smaller than when between two increased ones in different blocks.<\/span><\/div>, <div class='tooltip'>design.graeco<span class='tooltiptext'>A graeco - latin square is a KxK pattern that permits the study of k treatments simultaneously with three different blocking variables, each at k levels. The function is only for squares of the odd numbers and even numbers (4, 8, 10 and 12)<\/span><\/div>, <div class='tooltip'>design.lattice<span class='tooltiptext'>SIMPLE and TRIPLE lattice designs. It randomizes treatments in k x k lattice.<\/span><\/div>, <div class='tooltip'>design.lsd<span class='tooltiptext'>It generates Latin Square Design. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.mat<span class='tooltiptext'>Generate the design matrix from the fieldbook generated by an experimental plan or a dataframe for analysis.<\/span><\/div>, <div class='tooltip'>design.rcbd<span class='tooltiptext'>It generates Randomized Complete Block Design. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.split<span class='tooltiptext'>It generates split plot design. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.strip<span class='tooltiptext'>It generates strip plot design. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>design.youden<span class='tooltiptext'>Such designs are referred to as Youden squares since they were introduced by Youden (1937) after Yates (1936) considered the special case of column equal to number treatment minus 1. \"Random\" uses the methods of number generation in R. The seed is by set.seed(seed, kinds).<\/span><\/div>, <div class='tooltip'>diffograph<span class='tooltiptext'>It plots bars of the averages of treatments to compare. It uses the objects generated by a procedure of comparison like LSD (Fisher), duncan, Tukey (HSD), Student Newman Keul (SNK), Scheffe, Ryan, Einot and Gabriel and Welsch (REGW), Kruskal Wallis, Friedman and Waerden.<\/span><\/div>, <div class='tooltip'>duncan.test<span class='tooltiptext'>This test is adapted from the Newman-Keuls method. Duncan's test does not control family wise error rate at the specified alpha level. It has more power than the other post tests, but only because it doesn't control the error rate properly. The Experimentwise Error Rate at: 1-(1-alpha)^(a-1); where \"a\" is the number of means and is the Per-Comparison Error Rate. Duncan's procedure is only very slightly more conservative than LSD. The level by alpha default is 0.05.<\/span><\/div>, <div class='tooltip'>durbin.test<span class='tooltiptext'>A multiple comparison of the Durbin test for the balanced incomplete blocks for sensorial or categorical evaluation. It forms groups according to the demanded ones for level of significance (alpha); by default, 0.05.<\/span><\/div>, <div class='tooltip'>friedman<span class='tooltiptext'>The data consist of b-blocks mutually independent k-variate random variables Xij, i=1,..,b; j=1,..k. The random variable X is in block i and is associated with treatment j. It makes the multiple comparison of the Friedman test with or without ties. A first result is obtained by friedman.test of R.<\/span><\/div>, <div class='tooltip'>graph.freq<span class='tooltiptext'>In many situations it has intervals of class defined with its respective frequencies. By means of this function, the graphic of frequency is obtained and it is possible to superpose the normal distribution, polygon of frequency, Ojiva and to construct the table of complete frequency.<\/span><\/div>, <div class='tooltip'>hcut<span class='tooltiptext'>It shows dendrogram of a consensus of a tree generated by hclust.<\/span><\/div>, <div class='tooltip'>hgroups<span class='tooltiptext'>Returns a vector with group memberships. This function is used by the function consensus of clusters.<\/span><\/div>, <div class='tooltip'>HSD.test<span class='tooltiptext'>It makes multiple comparisons of treatments by means of Tukey. The level by alpha default is 0.05.<\/span><\/div>, <div class='tooltip'>index.AMMI<span class='tooltiptext'>calculate AMMI stability value (ASV) and Yield stability index (YSI).<\/span><\/div>, <div class='tooltip'>index.bio<span class='tooltiptext'>Scientists use a formula called the biodiversity index to describe the amount ofspecies diversity in a given area.<\/span><\/div>, <div class='tooltip'>index.smith<span class='tooltiptext'>Smith's index of soil heterogeneity is used primarily to derive optimum plot size. The index gives a single value as a quantitative measure of soil heterogeneity in an area. Graph CV for plot size and shape.<\/span><\/div>, <div class='tooltip'>intervals.freq<span class='tooltiptext'>List class intervals.<\/span><\/div>, <div class='tooltip'>join.freq<span class='tooltiptext'>In many situations it is required to join classes because of the low .frequency in the intervals. In this process, it is required to join the intervals and ad the .frequencies of them.<\/span><\/div>, <div class='tooltip'>kendall<span class='tooltiptext'>Correlation of Kendall two set. Compute exact p-value with ties.<\/span><\/div>, <div class='tooltip'>kruskal<span class='tooltiptext'>It makes the multiple comparison with Kruskal-Wallis. The alpha parameter by default is 0.05. Post hoc test is using the criterium Fisher's least significant difference. The adjustment methods include the Bonferroni correction and others.<\/span><\/div>, <div class='tooltip'>kurtosis<span class='tooltiptext'>It obtains the value of the kurtosis for a normally distributed variable. The result is similar to SAS.<\/span><\/div>, <div class='tooltip'>lastC<span class='tooltiptext'>A special function for the group of treatments in the multiple comparison tests. Use plot.group.<\/span><\/div>, <div class='tooltip'>lateblight<span class='tooltiptext'>LATEBLIGHT is a mathematical model that simulates the effect of weather,host growth and resistance, and fungicide use on asexual development andgrowth of Phytophthora infestans on potato foliage.<\/span><\/div>, <div class='tooltip'>lineXtester<span class='tooltiptext'>It makes the Line x Tester Genetic Analysis. It also estimates the general and specific combinatory ability effects and the line and tester genetic contribution.<\/span><\/div>, <div class='tooltip'>LSD.test<span class='tooltiptext'>Multiple comparisons of treatments by means of LSD and a grouping of treatments. The level by alpha default is 0.05. Returns p-values adjusted using one of several methods<\/span><\/div>, <div class='tooltip'>Median.test<span class='tooltiptext'>A nonparametric test for several independent samples. The median test is designed to examine whether several samples came from populations having the same median.<\/span><\/div>, <div class='tooltip'>montecarlo<span class='tooltiptext'>Random generation form data, use function density and parameters<\/span><\/div>, <div class='tooltip'>nonadditivity<span class='tooltiptext'>The resistance for the transformable nonadditivity, due to J. W. Tukey, is based on the detection of a curvilinear relation between y-est(y) and est(y). A freedom degree for the transformable nonadditivity.<\/span><\/div>, <div class='tooltip'>normal.freq<span class='tooltiptext'>A normal distribution graph elaborated from the histogram previously constructed. The average and variance are obtained from the data grouped in the histogram.<\/span><\/div>, <div class='tooltip'>ogive.freq<span class='tooltiptext'>It plots the cumulative relative .frequencies with the intervals of classes defined in the histogram.<\/span><\/div>, <div class='tooltip'>order.group<span class='tooltiptext'>This function allows us to compare the treatments averages or the adding of their ranges with the minimal significant difference which can vary from one comparison to another one.<\/span><\/div>, <div class='tooltip'>orderPvalue<span class='tooltiptext'>When there are treatments and their respective values, these can be compared with a minimal difference of meaning.<\/span><\/div>, <div class='tooltip'>path.analysis<span class='tooltiptext'>If the cause and effect relationship is well defined, it is possible to represent the whole system of variables in a diagram form known as path-analysis. The function calculates the direct and indirect effects and uses the variables correlation or covariance.<\/span><\/div>, <div class='tooltip'>PBIB.test<span class='tooltiptext'>Analysis of variance PBIB and comparison mean adjusted. Applied to resoluble designs: Lattices and alpha design.<\/span><\/div>, <div class='tooltip'>plot.AMMI<span class='tooltiptext'>Biplot AMMI.<\/span><\/div>, <div class='tooltip'>plot.graph.freq<span class='tooltiptext'>In many situations it has intervals of class defined with its respective frequencies. By means of this function, the graphic of frequency is obtained and it is possible to superpose the normal distribution, polygon of frequency, Ojiva and to construct the table of complete frequency.<\/span><\/div>, <div class='tooltip'>plot.group<span class='tooltiptext'>It plots bars of the averages of treatments to compare. It uses the objects generated by a procedure of comparison like LSD, HSD, Kruskall, Waller-Duncan, Friedman or Durbin. It can also display the 'average' value over each bar in a bar chart.<\/span><\/div>, <div class='tooltip'>polygon.freq<span class='tooltiptext'>The polygon is constructed single or on a histogram. It is necessary to execute the function previously hist.<\/span><\/div>, <div class='tooltip'>reg.homog<span class='tooltiptext'>It makes the regressions homogeneity test for a group of treatments where each observation presents a linearly dependent reply from another one. There is a linear function in every treatment. The objective is to find out if the linear models of each treatment come from the same population.<\/span><\/div>, <div class='tooltip'>REGW.test<span class='tooltiptext'>Multiple range tests for all pairwise comparisons, to obtain a confident inequalities multiple range tests.<\/span><\/div>, <div class='tooltip'>resampling.cv<span class='tooltiptext'>This process finds the curve of CV for a different number of markers which allows us to determine the number of optimal markers for a given relative variability. A method of the curvature.<\/span><\/div>, <div class='tooltip'>resampling.model<span class='tooltiptext'>This process consists of finding the values of P-value by means of a re-sampling (permutation) process along with the values obtained by variance analysis.<\/span><\/div>, <div class='tooltip'>scheffe.test<span class='tooltiptext'>Scheffe 1959, method is very general in that all possible contrasts can be tested for significance and confidence intervals can be constructed for the corresponding linear. The test is conservative.<\/span><\/div>, <div class='tooltip'>similarity<span class='tooltiptext'>It finds the similarity matrix of binary tables (1 and 0).<\/span><\/div>, <div class='tooltip'>simulation.model<span class='tooltiptext'>This process consists of validating the variance analysis results using a simulation process of the experiment. The validation consists of comparing the calculated values of each source of variation of the simulated data with respect to the calculated values of the original data. If in more than 50 percent of the cases they are higher than the real one, then it is considered favorable and the probability reported by the ANOVA is accepted, since the P-Value is the probability of (F > F.value).<\/span><\/div>, <div class='tooltip'>skewness<span class='tooltiptext'>It returns the skewness of a distribution. It is similar to SAS.<\/span><\/div>, <div class='tooltip'>SNK.test<span class='tooltiptext'>SNK is derived from Tukey, but it is less conservative (finds more differences). Tukey controls the error for all comparisons, where SNK only controls for comparisons under consideration. The level by alpha default is 0.05.<\/span><\/div>, <div class='tooltip'>sp.plot<span class='tooltiptext'>The variance analysis of a split plot design is divided into two parts: the plot-factor analysis and the sub-plot factor analysis.<\/span><\/div>, <div class='tooltip'>ssp.plot<span class='tooltiptext'>The variance analysis of a split-split plot design is divided into three parts: the main-plot, subplot and sub-subplot analysis.<\/span><\/div>, <div class='tooltip'>stability.nonpar<span class='tooltiptext'>A method based on the statistical ranges of the study variable per environment for the stability analysis.<\/span><\/div>, <div class='tooltip'>stability.par<span class='tooltiptext'>This procedure calculates the stability variations as well as the statistics of selection for the yield and the stability. The averages of the genotype through the different environment repetitions are required for the calculations. The mean square error must be calculated from the joint variance analysis.<\/span><\/div>, <div class='tooltip'>stat.freq<span class='tooltiptext'>By this process the variance and central measures ar found: average, medium and mode of grouped data.<\/span><\/div>, <div class='tooltip'>strip.plot<span class='tooltiptext'>The variance analysis of a strip-plot design is divided into three parts: the horizontal-factor analysis, the vertical-factor analysis, and the interaction analysis.<\/span><\/div>, <div class='tooltip'>sturges.freq<span class='tooltiptext'>if k=0 then classes: k = 1 + log(n,2). if k > 0, fixed nclass.<\/span><\/div>, <div class='tooltip'>summary.graph.freq<span class='tooltiptext'>It finds the absolute, relative and accumulated frequencies with the class intervals defined from a previously calculated histogram by the \"hist\" of R function.<\/span><\/div>, <div class='tooltip'>table.freq<span class='tooltiptext'>It finds the absolute, relative and accumulated frequencies with the class intervals defined from a previously calculated histogram by the \"hist\" of R function.<\/span><\/div>, <div class='tooltip'>tapply.stat<span class='tooltiptext'>This process lies in finding statistics which consist of more than one variable, grouped or crossed by factors. The table must be organized by columns between variables and factors.<\/span><\/div>, <div class='tooltip'>vark<span class='tooltiptext'>The Kendall method in order to find the K variance.<\/span><\/div>, <div class='tooltip'>waerden.test<span class='tooltiptext'>A nonparametric test for several independent samples.<\/span><\/div>, <div class='tooltip'>waller<span class='tooltiptext'>A Bayes rule for the symmetric multiple comparisons problem.<\/span><\/div>, <div class='tooltip'>waller.test<span class='tooltiptext'>The Waller-Duncan k-ratio t test is performed on all main effect means in the MEANS statement. See the K-RATIO option for information on controlling details of the test.<\/span><\/div>, <div class='tooltip'>weatherSeverity<span class='tooltiptext'>Weather and Severity<\/span><\/div>, <div class='tooltip'>zigzag<span class='tooltiptext'>applied to designs: complete block, latin square, graeco, split plot, strip plot, lattice, alpha lattice, Augmented block, cyclic, Balanced Incomplete Block and factorial.<\/span><\/div>","<div class='tooltip'>actuals<span class='tooltiptext'>Outputs the actual argument list of a function<\/span><\/div>, <div class='tooltip'>efficient.rounding<span class='tooltiptext'>A vector of proportions is efficiently rounded to integers.<\/span><\/div>, <div class='tooltip'>eval.blockdesign<span class='tooltiptext'>A blocked design is evaluated.<\/span><\/div>, <div class='tooltip'>eval.design<span class='tooltiptext'>A design is evaluated.<\/span><\/div>, <div class='tooltip'>expand.formula<span class='tooltiptext'>Formulas are expanded to accommodate special functions for continuous and mixture variables.<\/span><\/div>, <div class='tooltip'>gen.factorial<span class='tooltiptext'>A full factorial design is generated.<\/span><\/div>, <div class='tooltip'>gen.mixture<span class='tooltiptext'>Creates a candidate list of mixture variables.<\/span><\/div>, <div class='tooltip'>optBlock<span class='tooltiptext'>Blocking of experimental designs using various criteria.<\/span><\/div>, <div class='tooltip'>optFederov<span class='tooltiptext'>Calculates an exact or approximate algorithmic design for one of three criteria, using Federov's exchange algorithm.<\/span><\/div>, <div class='tooltip'>optMonteCarlo<span class='tooltiptext'>Finds a design using the specified criterion via Federov's algorithm applied to a random subset of all possible candidate points.<\/span><\/div>","<div class='tooltip'>alteval.ic<span class='tooltiptext'>alteval.ic calculates the objective function value (D, U or I) for a given design with interval censoring plan.<\/span><\/div>, <div class='tooltip'>alteval.rc<span class='tooltiptext'>alteval.rc calculates the objective function value (D, U or I) for a given design with right censoring plan.<\/span><\/div>, <div class='tooltip'>altopt.ic<span class='tooltiptext'>altopt.ic creates D, U or I optimal designof the accelerated life testing with interval censoring plan.<\/span><\/div>, <div class='tooltip'>altopt.rc<span class='tooltiptext'>altopt.rc creates D, U or I optimal designof the accelerated life testing with right censoring plan.<\/span><\/div>, <div class='tooltip'>compare.fus<span class='tooltiptext'>compare.fus draws the FUS plots of multiple designs on asingle frame.<\/span><\/div>, <div class='tooltip'>compare.vdus<span class='tooltiptext'>compare.vdus draws the VDUS plots of multiple designs on asingle frame.<\/span><\/div>, <div class='tooltip'>convert.stress.level<span class='tooltiptext'>Convert the stress levels from the actual levels to standardized levels, and vice versa.<\/span><\/div>, <div class='tooltip'>design.plot<span class='tooltiptext'>design.plot draws design plot as a form of a bubble plotof any two stress factors which are specified by xAxis and yAxis.The size of each bubble indicates the relative magnitude of allocation oneach design point.<\/span><\/div>, <div class='tooltip'>pv.contour.ic<span class='tooltiptext'>pv.contour.ic draws the contour plot of prediction variancefor a given design with interval censoring plan. Either useCond oruse region (useLower and useUpper) should beprovided.<\/span><\/div>, <div class='tooltip'>pv.contour.rc<span class='tooltiptext'>pv.contour.rc draws the contour plot of prediction variancefor a given design with right censoring plan. Either useCond oruse region (useLower and useUpper) should beprovided.<\/span><\/div>, <div class='tooltip'>pv.fus.ic<span class='tooltiptext'>pv.fus.ic draws the FUS plot of prediction variancefor a given design with interval censoring plan. The use region(useLower and useUpper) should beprovided.<\/span><\/div>, <div class='tooltip'>pv.fus.rc<span class='tooltiptext'>pv.fus.rc draws the FUS plot of prediction variancefor a given design with right censoring plan. The use region(useLower and useUpper) should beprovided.<\/span><\/div>, <div class='tooltip'>pv.vdus.ic<span class='tooltiptext'>pv.vdus.ic draws the VDUS plot of prediction variancefor a given design with interval censoring plan. The use region(useLower and useUpper) should beprovided.<\/span><\/div>, <div class='tooltip'>pv.vdus.rc<span class='tooltiptext'>pv.vdus.rc draws the VDUS plot of prediction variancefor a given design with right censoring plan. The use region(useLower and useUpper) should beprovided.<\/span><\/div>","<div class='tooltip'>alternate.sol<span class='tooltiptext'>This function returns an alternate solution for a particular row of incidence matrix with specified properties.<\/span><\/div>, <div class='tooltip'>alternate.sol2<span class='tooltiptext'>This function returns an alternate solution for a particular row of incidence matrix with specified properties.<\/span><\/div>, <div class='tooltip'>alternate.sol3<span class='tooltiptext'>This function returns an alternate solution for a particular row of incidence matrix with specified properties.<\/span><\/div>, <div class='tooltip'>aoptbbpb<span class='tooltiptext'>This function generates A-optimal balanced bipartite block (BBPB) designs for tests vs controls comparisons with specified parameters<\/span><\/div>, <div class='tooltip'>aoptgdtd<span class='tooltiptext'>This function generates A-optimal group divisible treatment (GDT) designs for test vs control comparisons with specified parameters<\/span><\/div>, <div class='tooltip'>bbpbwq<span class='tooltiptext'>This function returns incidence matrix of a BBPB design with given parameters.<\/span><\/div>, <div class='tooltip'>block.design<span class='tooltiptext'>This function generates the block contents from a given incidence matrix<\/span><\/div>, <div class='tooltip'>btibts<span class='tooltiptext'>This function returns incidence matrix of a BTIB design with given parameters.<\/span><\/div>, <div class='tooltip'>check.group<span class='tooltiptext'>This function checks group<\/span><\/div>, <div class='tooltip'>g<span class='tooltiptext'>This function computes the value of g for specified parameters.<\/span><\/div>, <div class='tooltip'>gbbpb<span class='tooltiptext'>This function computes the value of g for specified parameters.<\/span><\/div>, <div class='tooltip'>getr0<span class='tooltiptext'>This function returns the value of r0 for specified parameters.<\/span><\/div>, <div class='tooltip'>getrow<span class='tooltiptext'>This function returns a particular row of incidence matrix with specified properties.<\/span><\/div>, <div class='tooltip'>getrow2<span class='tooltiptext'>This function returns a particular row of incidence matrix with specified properties.<\/span><\/div>, <div class='tooltip'>getrow3<span class='tooltiptext'>This function returns a particular row of incidence matrix with specified properties.<\/span><\/div>, <div class='tooltip'>getts<span class='tooltiptext'>This function gives the value of t and s which minimize the g function.<\/span><\/div>, <div class='tooltip'>getwq<span class='tooltiptext'>This function gives the value of w and q which minimize the g function.<\/span><\/div>, <div class='tooltip'>wtaoptbtib<span class='tooltiptext'>This function generates weighted A-optimal balanced treatment incomplete block design for test vs control comparisons with specified parameters<\/span><\/div>","<div class='tooltip'>augmentedRCBD<span class='tooltiptext'>augmentedRCBD is a function for analysis of variance of an augmentedrandomised block design (Federer, 1956; Federer, 1961) and the generation aswell as comparison of the adjusted means of the treatments/genotypes.<\/span><\/div>, <div class='tooltip'>augmentedRCBD.bulk<span class='tooltiptext'>augmentedRCBD.bulk is a wrapper around the functionsaugmentedRCBD, describe.augmentedRCBD,freqdist.augmentedRCBD and gva.augmentedRCBD. It will carry outthese analyses for multiple traits/characters from the input data as a dataframe object.<\/span><\/div>, <div class='tooltip'>describe.augmentedRCBD<span class='tooltiptext'>describe.augmentedRCBD computes descriptive statistics from theadjusted means in an object of class augmentedRCBD.<\/span><\/div>, <div class='tooltip'>freqdist.augmentedRCBD<span class='tooltiptext'>freqdist.augmentedRCBD plots frequency distribution from an object ofclass augmentedRCBD along with the corresponding normal curve and checkmeans with standard errors (if specified by argument highlight.check).<\/span><\/div>, <div class='tooltip'>gva.augmentedRCBD<span class='tooltiptext'>gva.augmentedRCBD performs genetic variability analysis on an object ofclass augmentedRCBD.<\/span><\/div>, <div class='tooltip'>report.augmentedRCBD<span class='tooltiptext'>report.augmentedRCBD generates a tidy report from an object ofclass augmentedRCBD as docx MS word file using theofficer package.<\/span><\/div>, <div class='tooltip'>report.augmentedRCBD.bulk<span class='tooltiptext'>report.augmentedRCBD.bulk generates a tidy report from an object ofclass augmentedRCBD.bulk as docx MS word file using theofficer package.<\/span><\/div>","<div class='tooltip'>bacisCheckDIC<span class='tooltiptext'>In this function, the classification model is applied using the input parameter values and the DIC value is calculated.<\/span><\/div>, <div class='tooltip'>bacisClassification<span class='tooltiptext'>The classification model is conducted based on the BaCIS method and the subgroupos are classified into two clusters: high respone rate cluster and low response rate cluster.<\/span><\/div>, <div class='tooltip'>bacisOneTrial<span class='tooltiptext'>The bacisOneTrial function takes data and parameter values as input. It conducts a trial computation based on the BaCIS model. It calls the JAGS for the Bayesian MCMC sampling for the subgroup classification and hierarchical model information borrowing. It illustrates plots of the classficaiton results and the posterior response distributions of subgroups, and returns the inference results.<\/span><\/div>, <div class='tooltip'>bacisPlotClassification<span class='tooltiptext'>The classification model is conducted based on the BaCIS method and the posterior density of \\theta is plotted.<\/span><\/div>, <div class='tooltip'>bacisSubgroupPosterior<span class='tooltiptext'>In this function, a trial computation is conducted based on the BaCIS model. It calls the JAGS for the Bayesian MCMC sampling for the subgroup classification and hierarchical model information borrowing. The response rate posterior distributions of subgroups are returned from this function.<\/span><\/div>, <div class='tooltip'>bacisThetaPosterior<span class='tooltiptext'>The classification model is conducted based on the BaCIS method and the posterior distribution of \\theta is returned for further analyses.<\/span><\/div>","<div class='tooltip'>addAlgorithm<span class='tooltiptext'>Add an algorithm to registry and stores it on disk.<\/span><\/div>, <div class='tooltip'>addExperiments<span class='tooltiptext'>Add experiments for running algorithms on problemsto the registry, so they can be executed later.<\/span><\/div>, <div class='tooltip'>addProblem<span class='tooltiptext'>Add a algorithm to problem and stores it on disk.<\/span><\/div>, <div class='tooltip'>findExperiments<span class='tooltiptext'>Find job ids by querying problem/algorithm ids, problem/algorithm parameters orreplication number.<\/span><\/div>, <div class='tooltip'>generateProblemInstance<span class='tooltiptext'>Calls the dynamic problem function on the static problem part andthereby creates the problem instance.The seeding mechanism is identical to execution on the slave.<\/span><\/div>, <div class='tooltip'>getAlgorithm<span class='tooltiptext'>The requested object is loaded from disk.<\/span><\/div>, <div class='tooltip'>getAlgorithmIds<span class='tooltiptext'>Get algorithm ids for jobs.<\/span><\/div>, <div class='tooltip'>getExperimentParts<span class='tooltiptext'>Get all parts which define an Experiment.<\/span><\/div>, <div class='tooltip'>getIndex<span class='tooltiptext'>Creates a list of factor to use in functions like tapply, byor aggregate.<\/span><\/div>, <div class='tooltip'>getProblem<span class='tooltiptext'>The requested object is loaded from disk.<\/span><\/div>, <div class='tooltip'>getProblemIds<span class='tooltiptext'>Get problem ids for jobs.<\/span><\/div>, <div class='tooltip'>getResultVars<span class='tooltiptext'>Useful helper for e.g. package plyr and such.<\/span><\/div>, <div class='tooltip'>makeDesign<span class='tooltiptext'>Create a parameter design for either a problem or an algorithm that youcan use in addExperiments.All parameters in design and exhaustive be primitivein the sense that either is.atomic is TRUE or is.factor is TRUE.Be aware of R's default behaviour of converting strings into factors if you use the designparameter. See option stringsAsFactors in data.frame to turn this off.<\/span><\/div>, <div class='tooltip'>makeExperimentRegistry<span class='tooltiptext'>Note that if you don't want links in your paths (file.dir, work.dir) to get resolved and havecomplete control over the way the path is used internally, pass an absolute path which begins with /.Every object is a list that contains the passed arguments of the constructor.<\/span><\/div>, <div class='tooltip'>reduceResultsExperiments<span class='tooltiptext'>Generates a data.frame with one row per job id. The columns are: ids of problem and algorithm(named prob and algo), one column per parameter of problem or algorithm (named by the parameter name),the replication number (named repl) and all columns defined in the function to collect the values.Note that you cannot rely on the order of the columns.If a parameter does not have a setting for a certain job / experiment it is set to NA.Have a look at getResultVars if you want to use something like ddply on theresults.The rows are ordered as ids and named with ids, so one can easily index them.<\/span><\/div>, <div class='tooltip'>reduceResultsExperimentsParallel<span class='tooltiptext'>Basically the same as reduceResultsExperiments but creates a few (hopefully short) jobsto reduce the results in parallel. The function internally calls batchMapQuick,does busy-waiting tillall jobs are done and cleans all temporary files up.The rows are ordered as ids and named with ids, so one can easily index them.<\/span><\/div>, <div class='tooltip'>removeAlgorithm<span class='tooltiptext'>THIS DELETES ALL FILES REGARDING THIS ALGORITHM, INCLUDING ALL JOBS AND RESULTS!<\/span><\/div>, <div class='tooltip'>removeExperiments<span class='tooltiptext'>THIS DELETES ALL FILES REGARDING THE JOBS, INCLUDING RESULTS!If you really know what you are doing, you may set forceto TRUE to omit sanity checks on running jobs.<\/span><\/div>, <div class='tooltip'>removeProblem<span class='tooltiptext'>THIS DELETES ALL FILES REGARDING THIS PROBLEM, INCLUDING ALL JOBS AND RESULTS!<\/span><\/div>, <div class='tooltip'>summarizeExperiments<span class='tooltiptext'>A data.frame is returned that contains summary informationabout the selected experiments. The data.frame is constructedby building the columns prob, <prob.pars>, algo, <algo.pars>, repl.Now only the columns in show will be selected, how many of such experimentsexist will be counted in a new column .count.<\/span><\/div>","<div class='tooltip'>equiv_bf<span class='tooltiptext'>equiv_bf computes a Bayes factor for equivalence designs with acontinuous dependent variable.<\/span><\/div>, <div class='tooltip'>get_bf<span class='tooltiptext'>get_bf extracts the Bayes factor from an S4 object (i.e.,baymedrSuperiority, baymedrEquivalence,baymedrNonInferiority), created from the functionssuper_bf, equiv_bf, or infer_bf.<\/span><\/div>, <div class='tooltip'>infer_bf<span class='tooltiptext'>infer_bf computes a Bayes factor for non-inferiority designswith a continuous dependent variable.<\/span><\/div>, <div class='tooltip'>super_bf<span class='tooltiptext'>super_bf computes a Bayes factor for superiority designs with acontinuous dependent variable.<\/span><\/div>","<div class='tooltip'>BCHM<span class='tooltiptext'>The BCHM computation is based on the Bayesian Cluster Hierarchical Model (BCHM) to perform the non-parametric Bayesian clustering and posterior probability calculation with the Bayesian hierarchical model for binary response data in several subgroups. Due to the potential heterogeneity among subgroups, the exchangeability assumption across all subgroups may not hold. A Bayesian non-parametric method is applied to calculate the number of clusters by conducting the multiple cluster classification based on subgroup outcomes. Due to the MCMC sampling, the clustering result is dynamic. A cluster matrix (Similarity Matrix) is constructed to depict the strength of association between any two subgroups to be classified into the same cluster. The Bayesian hierarchical model is used to compute the posterior probability of treatment effect with the borrowing strength determined by the similarity matrix values calculated from the Bayesian non-parametric clustering.<\/span><\/div>, <div class='tooltip'>BCHMplot_cluster<span class='tooltiptext'>plot the observed response rate versus subgroup ID with clusters coded by the color of dots.<\/span><\/div>, <div class='tooltip'>BCHMplot_post_dist<span class='tooltiptext'>plot the posterior distribution by subgroup ID<\/span><\/div>, <div class='tooltip'>BCHMplot_post_value<span class='tooltiptext'>plot the posterior response rate with its highest probability density (HPD) interval by subgroup ID<\/span><\/div>","<div class='tooltip'>bcrm<span class='tooltiptext'>Implements a wide variety of Bayesian CRM designs, including 1-parameter, 2-parameter and Escalation With Overdose Control (EWOC) designs. The programcan run interactively, allowing the user to enter outcomes after each cohorthas been recruited, or via simulation to assess operating characteristics.<\/span><\/div>, <div class='tooltip'>find.x<span class='tooltiptext'>Given a posterior (or prior) sample of the parameters, this function invertsthe given functional form to obtain samples from the MTD distribution.<\/span><\/div>, <div class='tooltip'>getprior<span class='tooltiptext'>A sample of specified size is obtained from the prior distribution.<\/span><\/div>, <div class='tooltip'>Posterior.BRugs<span class='tooltiptext'>If ff = \"logit2\" (i.e. a two-parameter logistic model is used), a matrix of dimensionsproduction.itr-by-2 is returned (the first and second columns containing the posterior samples for theintercept and slope parameters respectively). Otherwise, a vector of length production.itris returned.<\/span><\/div>, <div class='tooltip'>Posterior.exact<span class='tooltiptext'>Returns posterior mean parameter value and summaries of distributions for probability of DLT at each dose level<\/span><\/div>, <div class='tooltip'>Posterior.exact.sim<span class='tooltiptext'>Returns posterior mean parameter value and summaries of distributions for probability of DLT at each dose level<\/span><\/div>, <div class='tooltip'>Posterior.R2WinBUGS<span class='tooltiptext'>If ff = \"logit2\" (i.e. a two-parameter logistic model is used), a matrix of dimensionsproduction.itr-by-2 is returned (the first and second columns containing the posterior samples for theintercept and slope parameters respectively). Otherwise, a vector of length production.itris returned.<\/span><\/div>, <div class='tooltip'>Posterior.rjags<span class='tooltiptext'>If ff = \"logit2\" (i.e. a two-parameter logistic model is used), a matrix of dimensionsproduction.itr-by-2 is returned (the first and second columns containing the posterior samples for theintercept and slope parameters respectively). Otherwise, a vector of length production.itris returned.<\/span><\/div>, <div class='tooltip'>threep3<span class='tooltiptext'>All possible pathways of a standard 3+3 design (may be escalation-only(see Storer 1989, Reiner et al. 1999) or permit dose de-escalation (see Chang et al.(2006))) are calculated and assigned a probability of occurring. This facilitates the calculation of operating characteristics, using print.threep3 andplot.threep3.<\/span><\/div>","<div class='tooltip'>AOppvnpv<span class='tooltiptext'>For internal use. Functions to compute sample size (to reach a pre-specified power) and optimal allocation of true positives and true negatives in case-control designs for binary diagnostic tests (Mercaldo et al. 2007).<\/span><\/div>, <div class='tooltip'>BDtest<span class='tooltiptext'>This function computes confidence intervals for negative and positive predictive values. Confidence intervals for sensitivity, specificity are computed for completeness. All methods assume that data are obtained by binomial sampling, with the number of true positives and true negatives in the study fixed by design. The methods to compute negative and positive predictive values (NPV, PPV) assume that prevalence is a known quantity, based on external knowledge.<\/span><\/div>, <div class='tooltip'>CIlnpv<span class='tooltiptext'>Computes asymptotic confidence intervals for negative and positive predictive values under the assumption of binomial sampling and known prevalence, according to Mercaldo et al. (2007). The standard logit intervals and and adjusted version are available, where the standard logit intervals are recommended.<\/span><\/div>, <div class='tooltip'>CIlnpvak<span class='tooltiptext'>Computes asymptotic confidence intervals for negative and positive predictive values under the assumption of binomial sampling and known prevalence, according to Mercaldo et al. (2007). The standard logit intervals and and adjusted version are available, where the standard logit intervals are recommended.<\/span><\/div>, <div class='tooltip'>CIlppv<span class='tooltiptext'>Computes asymptotic confidence intervals for negative and positive predictive values under the assumption of binomial sampling and known prevalence, according to Mercaldo et al. (2007). The standard logit intervals and and adjusted version are available, where the standard logit intervals are recommended.<\/span><\/div>, <div class='tooltip'>CIlppvak<span class='tooltiptext'>Computes asymptotic confidence intervals for negative and positive predictive values under the assumption of binomial sampling and known prevalence, according to Mercaldo et al. (2007). The standard logit intervals and and adjusted version are available, where the standard logit intervals are recommended.<\/span><\/div>, <div class='tooltip'>CIpvBI<span class='tooltiptext'>Computes confidence intervals for negative and positive predictive values by simulation from the posterior beta-distribution (Stamey and Holt, 2010), assuming a case-control design to estimate sensitivity and specificity, while prevalence estimates of an external study and/or prior knowledge concerning prevalence may be introduced additionally.<\/span><\/div>, <div class='tooltip'>CIpvBII<span class='tooltiptext'>Computes confidence intervals for negative and positive predictive values by simulation from the posterior beta-distribution (Stamey and Holt, 2010), assuming a case-control design to estimate sensitivity and specificity, while prevalence estimates of an external study and/or prior knowledge concerning prevalence may be introduced additionally.<\/span><\/div>, <div class='tooltip'>CombCInpv<span class='tooltiptext'>Computes asymptotic confidence intervals for negative and positive predictive values under the assumption of binomial sampling and known prevalence, according to Mercaldo et al. (2007). The standard logit intervals and and adjusted version are available, where the standard logit intervals are recommended.<\/span><\/div>, <div class='tooltip'>CombCIppv<span class='tooltiptext'>Computes asymptotic confidence intervals for negative and positive predictive values under the assumption of binomial sampling and known prevalence, according to Mercaldo et al. (2007). The standard logit intervals and and adjusted version are available, where the standard logit intervals are recommended.<\/span><\/div>, <div class='tooltip'>nNPV<span class='tooltiptext'>For internal use. Functions to compute sample size (to reach a pre-specified power) and optimal allocation of true positives and true negatives in case-control designs for binary diagnostic tests (Mercaldo et al. 2007).<\/span><\/div>, <div class='tooltip'>nPPV<span class='tooltiptext'>For internal use. Functions to compute sample size (to reach a pre-specified power) and optimal allocation of true positives and true negatives in case-control designs for binary diagnostic tests (Mercaldo et al. 2007).<\/span><\/div>, <div class='tooltip'>nPV<span class='tooltiptext'>Functions to compute sample size (to reach a pre-specified power) and optimal allocation of true positives and true negatives in case-control designs (Steinberg et al., 2008) for binary diagnostic tests (Mercaldo et al. 2007).<\/span><\/div>, <div class='tooltip'>plotnPV<span class='tooltiptext'>The function creates a plot from the results of the function nPV.<\/span><\/div>, <div class='tooltip'>plotnPV2<span class='tooltiptext'>The function creates a plot from the results of the function nPV.<\/span><\/div>, <div class='tooltip'>simPV<span class='tooltiptext'>The function draws data under the binomial assumption and computes the asymptotic confidence bounds (lower bounds only!) for the positive and negative predictive values.Output are the power (probability to exclude NPV0/PPV0), the realized coverage probability, 0.1,0.2, and 0.5-quantiles of the realized distribution of confidence bounds.<\/span><\/div>, <div class='tooltip'>simPVmat<span class='tooltiptext'>Simulate the power (probability to exclude NPV0/PPV0), the coverage probability, and 0.1, 0.2, and 0.5-quantiles of the distribution of (lower!) asymptotic confidence bounds for predictive values. Different experimental setups may be compared. The function draws data under the binomial assumption and computes the asymptotic confidence bounds (lower bounds only!) for the positive and negative predictive values.<\/span><\/div>","<div class='tooltip'>anovaPlot<span class='tooltiptext'>Dots plot displaying the deviations of factor levels from the mean showing the residuals as reference distribution.<\/span><\/div>, <div class='tooltip'>dotPlot<span class='tooltiptext'>Displays an one-dimensional scatter plot with stacking similar to stem-and-leaf plot or histograms.<\/span><\/div>, <div class='tooltip'>dots<span class='tooltiptext'>The function adds to the current plot an one-dimensional scatter plot with stacking similar to a stem-leaf plot or histograms but using characters. .<\/span><\/div>, <div class='tooltip'>ffDesMatrix<span class='tooltiptext'>The function generates the design matrix provided thenumber of 2-levels design factors and defining relations.<\/span><\/div>, <div class='tooltip'>ffFullMatrix<span class='tooltiptext'>The function builds the full matrix with the constant term, main effects andinteractions from a design matrix.<\/span><\/div>, <div class='tooltip'>lambdaPlot<span class='tooltiptext'>Trace regression coefficients' t-values or F-ratios for different values of \\lambda\\lambda inthe Box-Cox transformation.<\/span><\/div>, <div class='tooltip'>permtest<span class='tooltiptext'>Permutation test for means and variance comparisons.<\/span><\/div>, <div class='tooltip'>subsets<span class='tooltiptext'>Generates all different subsets of size rr chosen from nn different elements.<\/span><\/div>","<div class='tooltip'>accuracy.dorf<span class='tooltiptext'>Calculate the accuracy measures for each individual in a pool used with informative Dorfman testing.<\/span><\/div>, <div class='tooltip'>Array.Measures<span class='tooltiptext'>Calculate the expected number of tests and accuracy measuresfor each individual using array testing without master pooling<\/span><\/div>, <div class='tooltip'>beta.dist<span class='tooltiptext'>Get the expected value of order statistics, E(p(i)), from a beta distribution by specifying an average probability and shape parameters for the beta distribution.<\/span><\/div>, <div class='tooltip'>bgtCI<span class='tooltiptext'>Calculates the point estimate, the exact Clopper-Pearson and Blaker CI, the Score test derived Wilson and Agresti-Coull CI, the asymptotic second-order corrected interval fo Cai and the Wald CI for a single binomial proportion estimated from a binomial group testing trial. Assumes equal group sizes, an assay method classifying a group as positive if at least one unit in the group is positive, individuals units randomly assigned to the groups.<\/span><\/div>, <div class='tooltip'>bgtPower<span class='tooltiptext'>Closed calculation of the Power to reject a hypothesis in a binomial group testing experiment using confidence intervals for decision. Closed calculation of bias of the point estimator for a given experimental design n, s and the true, unknown proportion.<\/span><\/div>, <div class='tooltip'>bgtTest<span class='tooltiptext'>Calculates p values for hypotheses tests of binomial proportions estimated from binomial group testing experiments against a threshold proportionin the hypotheses. Exact test, Score test and Wald test are available methods. Assumes equal group sizes, 100 percent sensitivity and specificity of the assays to test the groups,and individuals units randomly assigned to the groups with identical true probability of success.<\/span><\/div>, <div class='tooltip'>bgtvs<span class='tooltiptext'>Calculates confidence intervals for a single proportion in binomial group testingif groups of different size are evaluated<\/span><\/div>, <div class='tooltip'>bgtWidth<span class='tooltiptext'>Calculation of expected value of the width of confidence intervals for one proportion in binomial group testing,in dependence of the number of groups, group size, confidence level and an assumed true proportion.Available for the confidence interval methods in bgtCI(binGroup).<\/span><\/div>, <div class='tooltip'>binAC<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>binBlaker<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>binCI<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>binCP<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>binDesign<span class='tooltiptext'>This function increases the sample size until a maximal sample size or a prespecified power is achieved.<\/span><\/div>, <div class='tooltip'>binPower<span class='tooltiptext'>Closed calculation of the power to reject a hypothesis using confidence intervals for a single binomial proportion, for specified sample size n, conf.level and an assumed absolute difference to thethreshold parameter under the null hypothesis.<\/span><\/div>, <div class='tooltip'>binSOC<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>binTest<span class='tooltiptext'>Calculates p-values for hypothesis tests of a single binomial proportion.<\/span><\/div>, <div class='tooltip'>binWald<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>binWidth<span class='tooltiptext'>Calculation of expected value of the width of confidence intervals in a binomial experiment,in dependence of the number of trials (number of individuals under observation), confidence level and an assumed true proportion.Available for the confidence interval methods in binCI(binGroup).<\/span><\/div>, <div class='tooltip'>binWilson<span class='tooltiptext'>Calculates the exact Clopper-Pearson and Blaker, the asymptotic second-order corrected,Wilson, Agresti-Coull and Wald confidence interval for a single binomial proportion<\/span><\/div>, <div class='tooltip'>characteristics.pool<span class='tooltiptext'>Calculate the expectation and variation of the testingexpenditure of a pool used with informative Dorfman testing.<\/span><\/div>, <div class='tooltip'>EM<span class='tooltiptext'>gtreg is a function to fit the group testing regression model specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>EM.halving<span class='tooltiptext'>gtreg.halving is a function to fit the group testing regression model under the halving protocol specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>EM.mp<span class='tooltiptext'>gtreg.mp is a function to fit the group testing regression model in the matrix pooling setting specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>EM.ret<span class='tooltiptext'>gtreg is a function to fit the group testing regression model specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>estDesign<span class='tooltiptext'>Find the group size s for a fixed number of assays n and an assumed true proportion p.tr for whichthe mean squared error (mse) of the point estimator is minimal and bias is within a restriction. For experimental design in binomial group testingas recommended by Swallow (1985), if main objective is estimation.<\/span><\/div>, <div class='tooltip'>gt.control<span class='tooltiptext'>Auxiliary function to control fitting parameters of EM algorithm used internally in gtreg.mp and EM.mp or gtreg, EM, and EM.ret.<\/span><\/div>, <div class='tooltip'>gtreg<span class='tooltiptext'>gtreg is a function to fit the group testing regression model specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>gtreg.fit<span class='tooltiptext'>gtreg is a function to fit the group testing regression model specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>gtreg.halving<span class='tooltiptext'>gtreg.halving is a function to fit the group testing regression model under the halving protocol specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>gtreg.mp<span class='tooltiptext'>gtreg.mp is a function to fit the group testing regression model in the matrix pooling setting specified through a symbolic description of the linear predictor and descriptions of the group testing setting.<\/span><\/div>, <div class='tooltip'>hierarchical.desc2<span class='tooltiptext'>Calculate operating characteristics for hierarchical group testing with up to four stages, given a vector of individual probabilities and a testing configuration.<\/span><\/div>, <div class='tooltip'>Inf.Array<span class='tooltiptext'>Find the optimal testing configuration (OTC) forinformative array testing without master pooling and calculatethe associated operating characteristics.<\/span><\/div>, <div class='tooltip'>Inf.D3<span class='tooltiptext'>Find the optimal testing configuration (OTC) for informative three-stage hierarchical testing and calculate the associated operating charcteristics.<\/span><\/div>, <div class='tooltip'>Inf.Dorf<span class='tooltiptext'>Find the optimal testing configuration (OTC) for informative two-stage hierarchical (Dorfman) testing and calculate the associated operating characteristics.<\/span><\/div>, <div class='tooltip'>inf.dorf.measures<span class='tooltiptext'>Calculate the expected number of tests and accuracy measures for each individual using informative two-stage hierarchical (Dorfman) testing, given a vector of individual probabilities and a testing configuration.<\/span><\/div>, <div class='tooltip'>Informative.array.prob<span class='tooltiptext'>Arrange a vector of individual risk probabilities in amatrix for informative array testing without master pooling.<\/span><\/div>, <div class='tooltip'>MasterPool.Array.Measures<span class='tooltiptext'>Calculate the expected number of tests and accuracy measuresfor each individual using array testing with master pooling.<\/span><\/div>, <div class='tooltip'>nDesign<span class='tooltiptext'>Increasing number of groups (assays, bulk samples) for a fixed group size in a binomial group testing design,until a pre-specified power is achieved. At the same time, bias of the estimator is controlled. A hypotheticalthreshold proportion p.hyp and the absolute difference delta to be detected have to be specified.<\/span><\/div>, <div class='tooltip'>NI.A2M<span class='tooltiptext'>Find the optimal testing configuration fornon-informative array testing with master pooling andcalculate the associated operating characteristics.<\/span><\/div>, <div class='tooltip'>NI.Array<span class='tooltiptext'>Find the optimal testing configuration (OTC) fornon-informative array testing without master pooling andcalculate the associated operating characteristics.<\/span><\/div>, <div class='tooltip'>NI.D3<span class='tooltiptext'>Find the optimal testing configuration (OTC) fornon-informative three-stage hierarchical testing and calculatethe associated operating characteristics.<\/span><\/div>, <div class='tooltip'>NI.Dorf<span class='tooltiptext'>Find the optimal testing configuration (OTC) for non-informative two-stage hierarchical (Dorfman) testing and calculate the associated operating characteristics.<\/span><\/div>, <div class='tooltip'>opt.info.dorf<span class='tooltiptext'>Find the characteristics of an informative two-stage hierarchical (Dorfman) decoding process using Optimal Dorfman (OD), Thresholded Optimal Dorfman (TOD), or Pool-Specific Optimal Dorfman (PSOD) algorithms.<\/span><\/div>, <div class='tooltip'>opt.pool.size<span class='tooltiptext'>Find the optimal common pool size for Optimal Dorfman(OD) or Thresholded Optimal Dorfman (TOD) testing.<\/span><\/div>, <div class='tooltip'>OTC<span class='tooltiptext'>Find the optimal testing configuration (OTC) for standard group testingalgorithms and calculate the associated operating characteristics.<\/span><\/div>, <div class='tooltip'>p.vec.func<span class='tooltiptext'>Generate a vector of individual risk probabilitiesusing an overall probability of disease (i.e., the expected value of order statistics from a beta distribution)for use with informative group testing algorithms.<\/span><\/div>, <div class='tooltip'>pool.specific.dorf<span class='tooltiptext'>Find the set of optimal pool sizes for Pool-Specific Optimal Dorfman (PSOD) testing.<\/span><\/div>, <div class='tooltip'>pooledBin<span class='tooltiptext'>Calculates confidence intervals for a single proportion based on pooled testing experiments containing various different pool sizes.<\/span><\/div>, <div class='tooltip'>pooledBinDiff<span class='tooltiptext'>The function calculates confidence intervals for the difference of two proportions based on pooled testing experiments containing various different pool sizes.<\/span><\/div>, <div class='tooltip'>predict.gt<span class='tooltiptext'>Obtains predictions for individual observations and optionally estimates standard errors of those predictions from objects of class \"gt\" or \"gt.mp\" returned by gtreg and gtreg.mp, respectively.<\/span><\/div>, <div class='tooltip'>residuals.gt<span class='tooltiptext'>residuals.gt is a function which extracts model residuals from objects of class \"gt\" returned by gtreg.<\/span><\/div>, <div class='tooltip'>sDesign<span class='tooltiptext'>Increasing of group size s for a fixed number of groups n in a binomial group testing design,until a pre-specified power is achieved. Control of bias during iteration. A hypotheticalthreshold proportion p.hyp and the absolute difference delta to be detected have to be specified.<\/span><\/div>, <div class='tooltip'>sim.gt<span class='tooltiptext'>Simulates data in group testing form ready to be fit by gtreg.<\/span><\/div>, <div class='tooltip'>sim.halving<span class='tooltiptext'>Simulates data in group testing form ready to be fit by gtreg.halving.<\/span><\/div>, <div class='tooltip'>sim.mp<span class='tooltiptext'>Simulates data in group testing form ready to be fit by gtreg.mp.<\/span><\/div>, <div class='tooltip'>thresh.val.dorf<span class='tooltiptext'>Find the optimal threshold value for Thresholded OptimalDorfman (TOD) testing.<\/span><\/div>","<div class='tooltip'>A_bound<span class='tooltiptext'>Finds upper A-efficiency bounds for regular block designs.<\/span><\/div>, <div class='tooltip'>blocks<span class='tooltiptext'>Constructs randomized multi-level nested block designs for unstructured treatment sets.<\/span><\/div>, <div class='tooltip'>design<span class='tooltiptext'>Constructs general D-optimal block and treatment designs for any feasible linear treatment model andany feasible combination of block factors.<\/span><\/div>, <div class='tooltip'>GraecoLatin<span class='tooltiptext'>Constructs mutually orthogonal Graeco-Latin squares for the following N:i) any odd valued Nii) any prime-power N = p**q where p and q can be chosen from rrrrrrrr prime p maximum q 2 13 3 8 5 6 7 5 11 4 13 17 19 23 3 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 2 Any prime >97 1 iii) any even valued N <= 30 except for 6 or 2<\/span><\/div>, <div class='tooltip'>HCF<span class='tooltiptext'>Finds the highest common factor (hcf) of a set of integer numbers greater than zero (Euclidean algorithm).<\/span><\/div>, <div class='tooltip'>isPrime<span class='tooltiptext'>Tests if a given number is prime and returns TRUE or FALSE<\/span><\/div>, <div class='tooltip'>isPrimePower<span class='tooltiptext'>Tests if a given number N is a prime power and returns either the base prime p and power q or p = 0 and q = 0.<\/span><\/div>, <div class='tooltip'>MOLS<span class='tooltiptext'>Constructs r sets of mutually orthogonal Latin squares (MOLS) of dimension p**q for prime p andinteger power q where r < p**q. Memory issues mean that the maximum size of the exponent q for specific p is restricted to the values shown in the table below: rrrrrrrr prime p maximum q 2 13 3 8 5 6 7 5 11 4 13 17 19 23 3 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 2 Any prime >97 1<\/span><\/div>","<div class='tooltip'>assg2xBalance<span class='tooltiptext'>Calculate several balance statistics for experimental units assigned to treatment conditions. Naturally accepts output from the assignment function, and passes it to xBalance from library(RItools). Provides balance summaries for the entire experiment and by group.<\/span><\/div>, <div class='tooltip'>assignment<span class='tooltiptext'>Using an output object from block, assign elements of each row to treatment condition columns. Each element is equally likely to be assigned to each column.<\/span><\/div>, <div class='tooltip'>block<span class='tooltiptext'>Block units into experimental blocks, with one unit per treatment condition. Blocking begins by creating a measure of multivariate distance between all possible pairs of units. Maximum, minimum, or an allowable range of differences between units on one variable can be set.<\/span><\/div>, <div class='tooltip'>block2seqblock<span class='tooltiptext'>Converts output objects from the block and assignment functionsinto an object in the format of one output by the seqblock function. This allows theuser to block and assign multiple units at the beginning of an experiment (using block andassignment) and then sequentially block and assign more units to the experiment over time(using seqblock).<\/span><\/div>, <div class='tooltip'>createBlockIDs<span class='tooltiptext'>Creates a vector of integers which represent unique blocks in an object output from block or assignment.<\/span><\/div>, <div class='tooltip'>diagnose<span class='tooltiptext'>List all pairs of units assigned to different treatment conditions whose difference on a specified variable falls within a specified range.<\/span><\/div>, <div class='tooltip'>invertRIconfInt<span class='tooltiptext'>Using an output object from seqblock or any other matrix or dataframe that includes a treatment and an outcome variable for multiple units, as well as blocking and non-blocking variables for the respective unit(s), invertRIconfInt calculates treatment effect confidence intervals by inverting the randomization inference test.<\/span><\/div>, <div class='tooltip'>outCSV<span class='tooltiptext'>Exports output from block or assignment to a set of .csv files using write.csv.<\/span><\/div>, <div class='tooltip'>outTeX<span class='tooltiptext'>Exports output from block or assignment to a set of .tex files using xtable.<\/span><\/div>, <div class='tooltip'>seqblock<span class='tooltiptext'>Sequentially assign units into experimental conditions. Blocking begins by creating a measure of multivariate distance between a current unit and one or multiple prior, already-assigned unit(s). Then, average distance between current unit and each treatment condition is calculated, and random assignment is biased toward conditions more dissimilar to current unit. Argument values can be specified either as argument to the function, or via a query. The query directly asks the user to identify the blocking variables and to input, one-by-one, each unit's variable values.<\/span><\/div>","<div class='tooltip'>get.boundary<span class='tooltiptext'>Use this function to generate the optimal dose escalation and deescalation boundaries for conducting the trial.<\/span><\/div>, <div class='tooltip'>get.oc<span class='tooltiptext'>Obtain the operating characteristics of the BOIN design for single agent trials by simulating trials.<\/span><\/div>, <div class='tooltip'>get.oc.comb<span class='tooltiptext'>Obtain the operating characteristics of the BOIN design or waterfall design for drug combination trials. The BOIN design is to find a MTD, and the waterfall design is to find the MTD contour (i.e., multple MTDs in the dose matrix)<\/span><\/div>, <div class='tooltip'>next.comb<span class='tooltiptext'>Determine the dose combination for the next cohort of new patients for drug-combination trials that aim to find a MTD<\/span><\/div>, <div class='tooltip'>next.subtrial<span class='tooltiptext'>Determine the starting dose and the dose-searching space for next subtrial afterthe current subtrial is completed when using the waterfall design<\/span><\/div>, <div class='tooltip'>select.mtd<span class='tooltiptext'>Select the maximum tolerated dose (MTD) when the single-agent trial is completed<\/span><\/div>, <div class='tooltip'>select.mtd.comb<span class='tooltiptext'>Select the maximum tolerated dose (MTD) or MTD contour after the drug combination trial iscompleted using the BOIN design or waterfall design<\/span><\/div>","<div class='tooltip'>Baoptbd<span class='tooltiptext'>This function is for internal usage only<\/span><\/div>, <div class='tooltip'>Bdoptbd<span class='tooltiptext'>This function is for internal usage only<\/span><\/div>, <div class='tooltip'>Boptbd<span class='tooltiptext'>The function Boptbd is used to compute Bayesian A- or D-optimal block designs under the linear mixed effects model settings using array/block exchange algorithm of Debusho, Gemechu and Haines (2018).<\/span><\/div>, <div class='tooltip'>Boptbd.default<span class='tooltiptext'>The function Boptbd is used to compute Bayesian A- or D-optimal block designs under the linear mixed effects model settings using array/block exchange algorithm of Debusho, Gemechu and Haines (2018).<\/span><\/div>, <div class='tooltip'>cmatbd<span class='tooltiptext'>The function cmatbd computes the information matrix (C-matrix) for treatment effects under either the linear fixed effects model or the linear mixed effects model setting for a given block design of size 2.<\/span><\/div>, <div class='tooltip'>fixparBbd<span class='tooltiptext'>The function fixparbd creates a tcltk widow that allows users to set or fix values for the parametric combinations to compute Bayesian optimal block designs.<\/span><\/div>, <div class='tooltip'>graphoptBbd<span class='tooltiptext'>The function graphoptBbd creates the graphical layout of resultant Bayesian A-, or D-optimal block design on a separate pop-up GUI tcltk window.<\/span><\/div>, <div class='tooltip'>intcbd<span class='tooltiptext'>The function intcbd generates a random initial connected block design for a given number of blocks b of size k = 2 and the number of treatments v.<\/span><\/div>, <div class='tooltip'>mmenuBbd<span class='tooltiptext'>Alternative to directly using the function Boptbd on the R console, the function mmenuBbd creates the main menu tcltk window with the list of two optimality criteria (A and D) for search of Bayesian optimal block designs.<\/span><\/div>, <div class='tooltip'>print.Boptbd<span class='tooltiptext'>The function Boptbd is used to compute Bayesian A- or D-optimal block designs under the linear mixed effects model settings using array/block exchange algorithm of Debusho, Gemechu and Haines (2018).<\/span><\/div>, <div class='tooltip'>print.summary.Boptbd<span class='tooltiptext'>The function Boptbd is used to compute Bayesian A- or D-optimal block designs under the linear mixed effects model settings using array/block exchange algorithm of Debusho, Gemechu and Haines (2018).<\/span><\/div>, <div class='tooltip'>summary.Boptbd<span class='tooltiptext'>The function Boptbd is used to compute Bayesian A- or D-optimal block designs under the linear mixed effects model settings using array/block exchange algorithm of Debusho, Gemechu and Haines (2018).<\/span><\/div>","<div class='tooltip'>ae_mnl<span class='tooltiptext'>This uses Equations 7, 10, 12, 13, and 18 from Lipovetsky & Conklin (2014) to take vectors of total times shown to participants, total times selected as best, and total times selected as worst. It uses their closed-form solution to calculate utility coefficients—as well as their standard errors and confidence intervals—and choice probabilities.<\/span><\/div>, <div class='tooltip'>diffscoring<span class='tooltiptext'>Calculate best-worst scores for each respondent-item combination. This is simply taking the total number of times a respondent selected the item as \"best\" and subtracting from that the number of times a respondent selected the item as \"worst\" (Louviere et al., 2013).<\/span><\/div>, <div class='tooltip'>e_bayescoring<span class='tooltiptext'>Individual utilities from empirical bayes estimations. Instead of doing the computationally-expensive hierarchical Bayesian multinomial logistic regression model, Lipovetsky & Conklin (2015) show an empirical Bayes way to calculate this analytically. This function calculates choice probabilities shown using Equation 10 in Lipovetsky & Conklin (2015) and transforms them to be on a linear regression coefficient scale. Default values for the E and alpha parameters are those performing best in their empirical example.<\/span><\/div>, <div class='tooltip'>elo<span class='tooltiptext'>Calculate aggregate best-worst scores using Elo scoring. This specific application comes from Hollis (2018). It makes each individual/block pairwise comparisons and updates Elo scores based on who won and lost those comparisons. No ties are considered, which occurs between all of the items that have not been selected as either best or worst. Hollis (2018) also recommends adding two \"dummy items\": one that defeats every other item, and one that loses to every other item. This is employed here. The default K is 30, per Hollis (2018). Since Elo is temporal in nature, Hollis also recommends running various iterations, each with a different randomization of the order of matchups. The default is the 100 used by Hollis. These are averaged together to calculate individual Elo best-worst scores. Elo scores are all initialized at 1000.<\/span><\/div>, <div class='tooltip'>eloscoring<span class='tooltiptext'>Calculate individual best-worst scores using Elo scoring. This specific application comes from Hollis (2018). It makes each block pairwise comparisons and updates Elo scores based on who won and lost those comparisons. No ties are considered, which occurs between all of the items that have not been selected as either best or worst. Hollis (2018) also recommends adding two \"dummy items\": one that defeats every other item, and one that loses to every other item. This is employed here. The default K is 30, per Hollis (2018). Since Elo is temporal in nature, Hollis also recommends running various iterations, each with a different randomization of the order of matchups. The default is the 100 used by Hollis. These are averaged together to calculate individual Elo best-worst scores. Elo scores are all initialized at 1000.<\/span><\/div>, <div class='tooltip'>make_bibd<span class='tooltiptext'>This function generates a balanced incomplete block design. It takes one argument, the design number from the bibds data.frame object. See bibds.<\/span><\/div>, <div class='tooltip'>prscoring<span class='tooltiptext'>Calculate best-worst scores for each respondent-item combination. This uses the page rank method. It works virtually the same as the walkscoring method; the only difference is that there is a parameter for randomly \"teleporting\" to a random node. This can help prevent random walks from getting \"stuck\" in certain areas. Page rank scores using are calculated and synthesized in the same way as the method used in walkscoring(). See White (2019).<\/span><\/div>, <div class='tooltip'>walkscoring<span class='tooltiptext'>Calculate best-worst scores for each respondent-item combination. This uses the walkscoring method described in White (2019).<\/span><\/div>","<div class='tooltip'>calc_contrast<span class='tooltiptext'>Calculate contrast analysis for factorial designs<\/span><\/div>, <div class='tooltip'>print.cofad_bw<span class='tooltiptext'>Output of between-subject design contrast analysis<\/span><\/div>, <div class='tooltip'>print.cofad_mx<span class='tooltiptext'>Output of a mixed design contrast analysis<\/span><\/div>, <div class='tooltip'>print.cofad_wi<span class='tooltiptext'>Output of a within subject design contrast analysis<\/span><\/div>, <div class='tooltip'>summary.cofad_bw<span class='tooltiptext'>Summary of between subject design contrast analysis<\/span><\/div>, <div class='tooltip'>summary.cofad_mx<span class='tooltiptext'>Summary of a mixed design contrast analysis<\/span><\/div>, <div class='tooltip'>summary.cofad_wi<span class='tooltiptext'>Summary of within subject design contrast analysis<\/span><\/div>","<div class='tooltip'>CombS<span class='tooltiptext'>The application of the Combinatory Method (s), with s chosen in [2, l-1],on rectangular association scheme to obtain the configuration and theparameters of the PBIB design associated.<\/span><\/div>, <div class='tooltip'>GPBIB4A<span class='tooltiptext'>Gives the configuration and the parametres of the design obtained bythe first construction method of GPBIB_4 (see 3.1.1 of the paperrezgui et al (2015)).<\/span><\/div>, <div class='tooltip'>GPBIB4B<span class='tooltiptext'>Gives the configuration and the parametres of the design obtained bythe seconde construction method of GPBIB_4 (see 3.1.2 of the paperrezgui et al (2015)).<\/span><\/div>, <div class='tooltip'>GPBIB5<span class='tooltiptext'>gives the configuration and the parametres of the design obtainedby the construction method of GPBIB_5 (see 3.2 of the paper rezguiet al (2015)).<\/span><\/div>, <div class='tooltip'>GPBIB7A<span class='tooltiptext'>gives the configuration and the parametres of the design obtained bythe first construction method of GPBIB_7 (see 3.3.1 of the paperrezgui et al (2015))<\/span><\/div>, <div class='tooltip'>GPBIB7B<span class='tooltiptext'>Gives the configuration and the parametres of the design obtained bythe seconde construction method of GPBIB_7 (see 3.3.2 of the paperrezgui et al (2015)).<\/span><\/div>, <div class='tooltip'>UType<span class='tooltiptext'>Applies the Fang algorithm on our constructed designs to obtainthe configuration and the parameters of the U-type design associated.<\/span><\/div>","<div class='tooltip'>conf.design<span class='tooltiptext'>Construct designs with specified treatment contrasts confounded withblocks. All treatment factors must have the sampe (prime) number oflevels.<\/span><\/div>, <div class='tooltip'>conf.set<span class='tooltiptext'>Find minimal complete sets of confounded effects from a defining set for symmetric confounded factorial designs. Useful for checking if a low order interaction will be unintentionally confounded with blocks. As in the usual convention, only effects whose leading factor has an index of one are listed. All factors must have the same (prime) number of levels.<\/span><\/div>, <div class='tooltip'>direct.sum<span class='tooltiptext'>Constructs the direct sum of two or more designs. Each plot of one design is matched with every plot of the other. (This might also be called the Cartesian product of two designs).<\/span><\/div>, <div class='tooltip'>factorize<span class='tooltiptext'>The default method factorizes positive numeric integer arguments, returning a vector of prime factors. The factor method can be used to generate pseudo-factors. It accepts a factor, f, as principal argument and returns a data frame with factors fa, fb, each with a prime number of levels such that f is model equivalent to join(fa, fb, ).<\/span><\/div>, <div class='tooltip'>join<span class='tooltiptext'>Joins two or more factors together into a single composite factor defining the subclasses. In a model formula join(f1, f2, f3) is equivalent to f1:f2:f3.<\/span><\/div>, <div class='tooltip'>primes<span class='tooltiptext'>Generate a table of prime numbers.<\/span><\/div>, <div class='tooltip'>rjoin<span class='tooltiptext'>Combine two or more designs with the same names into a single design by row concatenation.<\/span><\/div>","<div class='tooltip'>allcombs<span class='tooltiptext'>The function constructs a row-column design with subjects as rows and periods as columns. Each subject gets each treatment at most once. All possible treatment orders are assigned to the subjects.<\/span><\/div>, <div class='tooltip'>balmin.RMD<span class='tooltiptext'>The function constructs a row-column design with subjects as rows and periods as columns. The design is incomplete, i.e. no subject gets all the treatments. The design is balanced for carryover effects but will in general not be a balanced block design.<\/span><\/div>, <div class='tooltip'>des.MOLS<span class='tooltiptext'>The function constructs row-column designs based on complete sets of mutually orthogonal latin squares. Each subject may get each tratment at most once. The design is a generalized Youden design that is also balanced for carryover effects.<\/span><\/div>, <div class='tooltip'>find.BIB<span class='tooltiptext'>The function optBlock of the library AlgDesign is used to search for balanced incomplete block designs (BIBDs). The design is assigned to a matrix where rows represent blocks (subjects) and columns represent periods.<\/span><\/div>, <div class='tooltip'>get.plan<span class='tooltiptext'>This menu based function constructs simple experimental designs for repeated measurements with one or two block variables. It is assumed that each subject is assigned to each treatment at most once. A maximum number of subjects in the study is also requested. Five possible construction methods available.These construction methods and the characteristics of the resulting designs are described in Wakeling and MacFie (1995).See also Jones and Kenward (1989), Ch. 5, for a discussion of these designs. The function is demonstrated in more detail in Sailer (2005).<\/span><\/div>, <div class='tooltip'>isCbalanced<span class='tooltiptext'>The function checks whether a block design is balanced for first order carryover effects (residual effects). The user specifies whether there is a preperiod. The design is checked and the left neighbour incidence matrix is given.<\/span><\/div>, <div class='tooltip'>isGYD<span class='tooltiptext'>A function to check a simple block or a row-column design for balance. The rows and columns of the design are blocking variables. It is checked which type of balance the design fulfills. Optionally, incidence and concurrence matrices are given.<\/span><\/div>, <div class='tooltip'>MOLS<span class='tooltiptext'>The function constructs sets of mutually othogonal latin squares (MOLS) using Galois fields. The construction works for prime powers only.<\/span><\/div>, <div class='tooltip'>williams<span class='tooltiptext'>The function constructs williams designs. Williams designs are row-column designs. They are used if each of the treatments in the study is given to each of the subjects. If the number of treatments to be tested is even, the design is a latin square, otherwise it consists of two latin squares.<\/span><\/div>, <div class='tooltip'>williams.BIB<span class='tooltiptext'>Patterson (1951) combined balanced incomplete block designs (BIBDs) with Williams designs to get carryover balanced generalized Youden designs.<\/span><\/div>","<div class='tooltip'>buildSummaryTable<span class='tooltiptext'>Build Summary Table For All Examples From Literature<\/span><\/div>, <div class='tooltip'>canonicalOrder<span class='tooltiptext'>Sorts sequences of a design into a canonical order.<\/span><\/div>, <div class='tooltip'>contrMat2<span class='tooltiptext'>Function to read in a cross-over design and create the design matrix X, the variance of each pairwise comparison and the efficicency of each pairwise comparison.<\/span><\/div>, <div class='tooltip'>CrossoverGUI<span class='tooltiptext'>Starts a graphical user interface for accessing and creating crossover designs.<\/span><\/div>, <div class='tooltip'>design.efficiency<span class='tooltiptext'>Function to read in a cross-over design and create the design matrix X, the variance of each pairwise comparison and the efficicency of each pairwise comparison.<\/span><\/div>, <div class='tooltip'>general.carryover<span class='tooltiptext'>Calculate variances of paramater contrasts<\/span><\/div>, <div class='tooltip'>getDesign<span class='tooltiptext'>Extract Design from a CrossoverSearchResult<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'>Plots information about the search algorithm and its process.<\/span><\/div>, <div class='tooltip'>print<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>searchCrossOverDesign<span class='tooltiptext'>Search for a Cross-Over Design<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>cptest<span class='tooltiptext'>cptest performs a clustered permutation test on the individual-level outcome data for clusterrandomized trials (CRTs). The type of the outcome can be specified by the user to be \"continuous\" or\"binary\".Linear regression (for outcome type \"continuous\") or logistic regression (for outcome type \"binary\") is applied to the outcome regressed on covariates specified. Cluster residual means are computed. Within the constrained space,the contrast statistic between the treatment and control arms is created from the randomization schemes and the cluster residual means. The permutation test is then conducted by comparing the contrast statistic for the scheme actually utilized to all other schemes in the constrained space.<\/span><\/div>, <div class='tooltip'>cvrall<span class='tooltiptext'>cvrall performs constrained randomization for cluster randomizedtrials (CRTs), especially suited for CRTs with a small number of clusters. In constrained randomization,a randomization scheme is randomly sampled from a subset of all possible randomization schemesbased on the value of a balancing criterion called a balance score. The cvrall function has two choices of \"l1\" and \"l2\" metrics for balance score.The cvrall function enumerates all randomization schemes or chooses the unique ones among some simulated randomization schemes as specified by the user.Some cluster-level continuous or categorical covariates are then used to calculate the balance scores for the unique schemes. A subset of the randomization schemes is chosen based on a user-specified cutoff at a certain quantile of the distribution of the balance scores or based on a fixed number of schemes with the smallest balance scores. The cvrall functiontreats the subset as the constrained space of randomization schemes and samples one scheme from the constrained space as the final chosen scheme.<\/span><\/div>, <div class='tooltip'>cvrcov<span class='tooltiptext'>cvrcov performs covariate-by-covariate constrained randomization for cluster randomizedtrials (CRTs), especially suited for CRTs with a small number of clusters. In constrained randomization,a randomization scheme is randomly sampled from a subset of all possible randomization schemesbased on the constraints on each covariate. The cvrcov function enumerates all randomization schemes or simulates a fixed size of unique randomization schemes as specified by the user.A subset of the randomization schemes is chosen based on user-specified covariate-by-covariate constraints. cvrcov treats the subset as the constrained space of randomization schemes and samples one scheme from the constrained space as the final chosen scheme.<\/span><\/div>","<div class='tooltip'>Ameasures<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>as.data.frame.pstructure<span class='tooltiptext'>Coerces a pstructure.object, which is of class pstructure, to a data.frame. One can choose whether or not to include the marginality matrix in the data.frame. The aliasing component is excluded.<\/span><\/div>, <div class='tooltip'>as.numfac<span class='tooltiptext'>Converts a factor to a numeric vector with approximately the numeric values of its levels. Hence, the levels of the factor must be numeric values, stored as characters. It uses the method described in factor. Use as.numeric to convert a factor to a numeric vector with integers 1, 2, ... corresponding to the positions in the list of levels. You can also use fac.recode to recode the levels to numeric values. If a numeric is supplied, it is left unchanged.<\/span><\/div>, <div class='tooltip'>blockboundary.plot<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>blockboundaryPlot<span class='tooltiptext'>This function plots a block boundary on a plot produced by designPlot. It allows control of the starting unit, through rstart and cstart, and the number of rows (nrows) and columns (ncolumns) from the starting unit that the blocks to be plotted are to cover.<\/span><\/div>, <div class='tooltip'>correct.degfree<span class='tooltiptext'>Check the degrees of freedom in an object of class \"projector\".<\/span><\/div>, <div class='tooltip'>decomp.relate<span class='tooltiptext'>Two decompositions produced by proj2.eigen are compared by computing all pairs of crossproduct sums of eigenvectors from the two decompositions. It is most useful when the calls to proj2.eigen have the same Q1.<\/span><\/div>, <div class='tooltip'>degfree<span class='tooltiptext'>Extracts the degrees of freedom from or replaces them in an object of class \"projector\".<\/span><\/div>, <div class='tooltip'>degfree<-<span class='tooltiptext'>Extracts the degrees of freedom from or replaces them in an object of class \"projector\".<\/span><\/div>, <div class='tooltip'>design.plot<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>designAmeasures<span class='tooltiptext'>Calculates the average variance of pairwise differences between (of elementary contrasts of) preductions, possibly for different subgroups of the predictions, from the variance matrix for the predictions. If groups are specified then the A-optimality measures are calculated for the differences between predictions within each group and for those between predictions from different groups. If groupsizes are specified, but groups are not, the predictions will be sequentially broken into groups of the size specified by the elements of groupsizes. The groups can be named.<\/span><\/div>, <div class='tooltip'>designAnatomy<span class='tooltiptext'>Computes the canonical efficiency factors for the joint decomposition of two or more structures or sets of mutually orthogonally projectors (Brien and Bailey, 2009; Brien, 2017; Brien, 2019), orthogonalizing projectors in a set to those earlier in the set of projectors with which they are partially aliased. The results can be summarized in the form of a decomposition table that shows the confounding between sources from different sets. For examples of its use also see the vignette daeDesignNotes.pdf.<\/span><\/div>, <div class='tooltip'>designBlocksGGPlot<span class='tooltiptext'>This function adds block boundaries to a plot produced by designGGPlot. It allows control of the starting unit, through originrow and origincolumn, and the number of rows (nrows) and columns (ncolumns) from the starting unit that the blocks to be plotted are to cover.<\/span><\/div>, <div class='tooltip'>designGGPlot<span class='tooltiptext'>Plots the labels in a grid of cells specified by row.factors and column.factors. The cells can be coloured by the values of the column specified by column.name and can be divided into facets by specifying multiple row and or column factors.<\/span><\/div>, <div class='tooltip'>designLatinSqrSys<span class='tooltiptext'>Generates a systematic plan for a Latin Square design using the method of cycling the integers 1 to the number of treatments. The start of the cycle for each row, or the first column, can be specified as a vector of integers.<\/span><\/div>, <div class='tooltip'>designPlot<span class='tooltiptext'>This function uses labels, usually derived from treatment and blocking factors from an experimental design and stored in a matrix, to build a graphical representation of the matrix, highlighting the position of certain labels . It is a modified version of the function supplied with DiGGer. It includes more control over the labelling of the rows and columns of the design and allows for more flexible plotting of designs with unequal block size.<\/span><\/div>, <div class='tooltip'>designPlotlabels<span class='tooltiptext'>Plots the labels in a grid specified by grid.xand grid.y. The labels can be coloured by the values of the column specified by column.name.<\/span><\/div>, <div class='tooltip'>designRandomize<span class='tooltiptext'>A systematic design is specified by a set of allocated factors that have been assigned to a set of recipient factors. In textbook designs the allocated factors are the treatment factors and the recipient factors are the factors indexing the units. To obtain a randomized layout for a systematic design it is necessary to provide (i) the systematic arrangement of the allocated factors, (ii) a list of the recipient factors or a data.frame with their values, and (iii) the nesting of the recipient factors for the design being randomized. Given this information, the allocated factors will be randomized to the recipient factors, taking into account the nesting between the recipient factors for the design. However, allocated factors that have different values associated with those recipient factors that are in the except vector will remain unchanged from the systematic design. Also, if allocated is NULL then a random permutation of the recipient factors is produced that is consistent with their nesting as specified by nested.recipients. For examples of its use also see the vignette daeDesignNotes.pdf.<\/span><\/div>, <div class='tooltip'>designTwophaseAnatomies<span class='tooltiptext'>Uses designAnatomy to obtain the four species of designs, described by Brien (2019), that are associated with a standard two-phase design: the anatomies for the (i) two-phase, (ii) first-phase, (iii) cross-phase, treatments, and (iv) combined-units designs. (The names of the last two designs in Brien (2019) were cross-phase and second-phase designs.) For the standard two-phase design, the first-phase design is the design that allocates first-phase treatments to first-phase units. The cross-phase, treatments design allocates the first-phase treatments to the second-phase units and the combined-units design allocates the the first-phase units to the second-phase units. The two-phase design combines the other three species of designs. However, it is not mandatory that the three formula correspond to second-phase units, first-phase units and first-phase treatments, respectively, as is implied above; this is just the correspondence for a standard two-phase design. The essential requirement is that three structure formulae are supplied. For example, if there are both first- and second-phase treatments in a two-phase design, the third formula might involve the treatment factors from both phases. In this case, the default anatomy titles when printing occurs will not be correct, but can be modifed using the titles argument.<\/span><\/div>, <div class='tooltip'>detect.diff<span class='tooltiptext'>Computes the delta that is detectable for specified replication, power, alpha.<\/span><\/div>, <div class='tooltip'>efficiencies<span class='tooltiptext'>Produces a list containing the canonical efficiency factors for the joint decomposition of two or more sets of projectors (Brien and Bailey, 2009) obtained using designAnatomy or projs.2canon.<\/span><\/div>, <div class='tooltip'>efficiencies.p2canon<span class='tooltiptext'>Produces a list containing the canonical efficiency factors for the joint decomposition of two or more sets of projectors (Brien and Bailey, 2009) obtained using designAnatomy or projs.2canon.<\/span><\/div>, <div class='tooltip'>efficiencies.pcanon<span class='tooltiptext'>Produces a list containing the canonical efficiency factors for the joint decomposition of two or more sets of projectors (Brien and Bailey, 2009) obtained using designAnatomy or projs.2canon.<\/span><\/div>, <div class='tooltip'>efficiency.criteria<span class='tooltiptext'>Computes efficiency criteria from a set of efficiency factors.<\/span><\/div>, <div class='tooltip'>elements<span class='tooltiptext'>Elements of the array x corresponding to the rows of the two dimensional object subscripts are extracted. The number of columns of subscripts corresponds to the number of dimensions of x. The effect of supplying less columns in subscripts than the number of dimensions in x is the same as for \"[\".<\/span><\/div>, <div class='tooltip'>extab<span class='tooltiptext'>Expands the values in table to a vector according to the index.factors that are considered to index the table, either in standard or Yates order. The order of the values in the vector is determined by the order of the values of the index.factors.<\/span><\/div>, <div class='tooltip'>fac.ar1mat<span class='tooltiptext'>Form the correlation matrix for a (generalized) factor where the correlation between the levels follows an autocorrelation of order 1 (ar1) pattern.<\/span><\/div>, <div class='tooltip'>fac.combine<span class='tooltiptext'>Combines several factors into one whose levels are the combinations of the used levels of the individual factors.<\/span><\/div>, <div class='tooltip'>fac.divide<span class='tooltiptext'>Takes a factor and divides it into several individual factors as if the levels in the original combined.factor are numbered from one to its number of levels and correspond to the numbering of the levels combinations of the individual factors when these are arranged in standard or Yates order.<\/span><\/div>, <div class='tooltip'>fac.gen<span class='tooltiptext'>Generate all combinations of several factors and, optionally, replicate them.<\/span><\/div>, <div class='tooltip'>fac.match<span class='tooltiptext'>Match, for each combination of a set of columns in x, the rows that has the same combination in table. The argument multiples.allow controls what happens when there are multple matches in table of a combination in x.<\/span><\/div>, <div class='tooltip'>fac.meanop<span class='tooltiptext'>Computes the symmetric projection matrix that produces the means corresponding to a (generalized) factor.<\/span><\/div>, <div class='tooltip'>fac.multinested<span class='tooltiptext'>Creates several factors, one for each level of nesting.fac and each of whose values are either (i) generated within those of the level of nesting.fac or (ii) using the values of nested.fac within the levels of the nesting.fac. For (i), all elements having the same level of nesting.fac are numbered from 1 to the number of different elements having that level. For (ii), the values of nested.fac for a level of nesting.fac are copied. In both cases, for the values of nested.fac not equal to the level of the values of nested.fac for which a nested factor is being created, the levels are set to outlevel and labelled using outlabel. A factor is not created for a level of nesting.fac with label equal to outlabel. The names of the factors are equal to the levels of nesting.fac; optionally fac.prefix is added to the beginning of the names of the factors. The function is used to split up a nested term into separate terms for each level of nesting.fac.<\/span><\/div>, <div class='tooltip'>fac.nested<span class='tooltiptext'>Creates a nested factor whose levels are generated within those of the factor nesting.fac. All elements of nesting.fac having the same level are numbered from 1 to the number of different elements having that level.<\/span><\/div>, <div class='tooltip'>fac.recode<span class='tooltiptext'>Recodes the levels and values of a factor using each value in the newlevels vector to replace the corresponding value in the vector of levels of the factor.<\/span><\/div>, <div class='tooltip'>fac.sumop<span class='tooltiptext'>Computes the matrix that produces the sums corresponding to a (generalized) factor.<\/span><\/div>, <div class='tooltip'>fac.uselogical<span class='tooltiptext'>Forms a two-level factor from a logical object. It can be used to recode a factor when the resulting factor is to have only two levels.<\/span><\/div>, <div class='tooltip'>fac.vcmat<span class='tooltiptext'>Form the variance matrix for a (generalized) factor whose effects for its different levels are independently and identically distributed, with their variance given by the variance component; elements of the matrix will equal either zero or sigma2 and displays compound symmetry.<\/span><\/div>, <div class='tooltip'>fitted.aovlist<span class='tooltiptext'>Extracts the fitted values as the sum of the effects for all the fitted terms in the model, stopping at error.term if this is specified. It is a method for the generic function fitted.<\/span><\/div>, <div class='tooltip'>fitted.errors<span class='tooltiptext'>An alias for the generic function fitted. When it is available, the method fitted.aovlist extracts the fitted values, which is provided in the dae package to cover aovlist objects.<\/span><\/div>, <div class='tooltip'>get.daeTolerance<span class='tooltiptext'>A function that gets the vector of values such that, in dae functions, values less than it are considered to be zero.<\/span><\/div>, <div class='tooltip'>harmonic.mean<span class='tooltiptext'>A function to calcuate the harmonic mean of a set of nonzero numbers.<\/span><\/div>, <div class='tooltip'>interaction.ABC.plot<span class='tooltiptext'>Plots a function (the mean by default) of the response for the combinations of the three factors specified as the x.factor (plotted on the x axis of each plot), the groups.factor (plotted as separate lines in each plot) and the trace.factor (its levels are plotted in different plots). Interaction plots for more than three factors can be produced by using fac.combine to combine all but two of them into a single factor that is specified as the trace.factor.<\/span><\/div>, <div class='tooltip'>is.allzero<span class='tooltiptext'>A single-line function that tests whether all elements are zero (approximately).<\/span><\/div>, <div class='tooltip'>is.projector<span class='tooltiptext'>Tests whether an object is a valid object of class \"projector\".<\/span><\/div>, <div class='tooltip'>marginality<span class='tooltiptext'>Produces (i) a marginality matrix for the formula in a call to pstructure.formula or (ii) a list containing the marginlity matrices, one for each formula in the formulae argument of a call to designAnatomy. A marginality matrix for a set of terms is a square matrix with a row and a column for each ternon-aliased term. Its elements are zeroes and ones, the entry in the ith row and jth column indicates whether or not the ith term is marginal to the jth term i.e. the column space of the ith term is a subspace of that for the jth term and so the source for the jth term will be orthogonal to that for the ith term.<\/span><\/div>, <div class='tooltip'>marginality.pcanon<span class='tooltiptext'>Produces (i) a marginality matrix for the formula in a call to pstructure.formula or (ii) a list containing the marginlity matrices, one for each formula in the formulae argument of a call to designAnatomy. A marginality matrix for a set of terms is a square matrix with a row and a column for each ternon-aliased term. Its elements are zeroes and ones, the entry in the ith row and jth column indicates whether or not the ith term is marginal to the jth term i.e. the column space of the ith term is a subspace of that for the jth term and so the source for the jth term will be orthogonal to that for the ith term.<\/span><\/div>, <div class='tooltip'>marginality.pstructure<span class='tooltiptext'>Produces (i) a marginality matrix for the formula in a call to pstructure.formula or (ii) a list containing the marginlity matrices, one for each formula in the formulae argument of a call to designAnatomy. A marginality matrix for a set of terms is a square matrix with a row and a column for each ternon-aliased term. Its elements are zeroes and ones, the entry in the ith row and jth column indicates whether or not the ith term is marginal to the jth term i.e. the column space of the ith term is a subspace of that for the jth term and so the source for the jth term will be orthogonal to that for the ith term.<\/span><\/div>, <div class='tooltip'>mat.ar1<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the ar1 pattern. The matrix is banded and has diagonal elements equal to one and the off-diagonal element in the ith row and jth column equal to \\rho^krho^(k) where k = |i- j|k = abs(i- j).<\/span><\/div>, <div class='tooltip'>mat.ar2<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the ar2 pattern. The resulting matrix is banded.<\/span><\/div>, <div class='tooltip'>mat.ar3<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the ar3 pattern. The resulting matrix is banded.<\/span><\/div>, <div class='tooltip'>mat.arma<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the arma pattern. The resulting matrix is banded.<\/span><\/div>, <div class='tooltip'>mat.banded<span class='tooltiptext'>Takes the first value in x and places it down the diagonal of the matrix. Takes the second value in x and places it down the first subdiagonal, both below and above the diagonal of the matrix. The third value is placed in the second subdiagonal and so on, until the bands for which there are elements in x have been filled. All other elements in the matrix will be zero.<\/span><\/div>, <div class='tooltip'>mat.dirprod<span class='tooltiptext'>Form the direct product of the m \\times nm x n matrix A and the p \\times qp x q matrix B. It is also called the Kroneker product and the right direct product. It is defined to be the result of replacing each element of A, a_{ij}a_{ij}, with a_{ij}\\bold{B}a_{ij}\\bold{B}. The result matrix is mp \\times nqmp \\times nq. The method employed uses the rep function to form two mp \\times nqmp \\times nq matrices: (i) the direct product of A and J, and (ii) the direct product of J and B, where each J is a matrix of ones whose dimensions are those required to produce an mp \\times nqmp \\times nq matrix. Then the elementwise product of these two matrices is taken to yield the result.<\/span><\/div>, <div class='tooltip'>mat.dirsum<span class='tooltiptext'>The direct sum is the partitioned matrices whose diagonal submatrices are the matrices from which the direct sum is to be formed and whose off-diagonal submatrices are conformable matrices of zeroes. The resulting matrix is m \\times nm x n, where mm is the sum of the numbers of rows of the contributing matrices and nn is the sum of their numbers of columns.<\/span><\/div>, <div class='tooltip'>mat.exp<span class='tooltiptext'>Form the correlation matrix of order equal to the length of coordinates. The matrix has diagonal elements equal to one and the off-diagonal element in the ith row and jth column equal to \\rho^krho^(k) where k = |coordinate[i]- coordinate[j]|k = abs(coordinate[i]- coordinate[j]).<\/span><\/div>, <div class='tooltip'>mat.gau<span class='tooltiptext'>Form the correlation matrix of order equal to the length of coordinates. The matrix has diagonal elements equal to one and the off-diagonal element in the ith row and jth column equal to \\rho^krho^(k) where k = (coordinate[i]- coordinate[j])^2k = (coordinate[i]- coordinate[j])^2.<\/span><\/div>, <div class='tooltip'>mat.I<span class='tooltiptext'>Form the unit or identity matrix of order order.<\/span><\/div>, <div class='tooltip'>mat.J<span class='tooltiptext'>Form the square matrix of ones of order order.<\/span><\/div>, <div class='tooltip'>mat.ma1<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the ma1 pattern. The matrix is banded and has diagonal elements equal to one and subdiagonal element equal to -MAparameter / (1 + MAparameter*MAparameter).<\/span><\/div>, <div class='tooltip'>mat.ma2<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the ma2 pattern. The resulting matrix is banded.<\/span><\/div>, <div class='tooltip'>mat.ncssvar<span class='tooltiptext'>Calculates the variance matrix of the random effects for a natural cubic smoothing spline. It is the tri-diagonal matrix \\bold{G}_sG_s given by Verbyla et al., (1999) multiplied by the variance component for the random spline effects.<\/span><\/div>, <div class='tooltip'>mat.random<span class='tooltiptext'>For n observations, compute the variance matrix of the random effects. The matrix can be specified using a formula for the random effects and a list of values of the variance components for the terms specified in the random formula. If a matrix specifying the variances of the nuisance random effects is supplied then it is returned as the value from the function.<\/span><\/div>, <div class='tooltip'>mat.sar<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the sar pattern. The resulting matrix is banded.<\/span><\/div>, <div class='tooltip'>mat.sar2<span class='tooltiptext'>Form the correlation matrix of order order whose correlations follow the sar2 pattern, a pattern used in crop competition models. The resulting matrix is banded and is a constrained AR3 matrix.<\/span><\/div>, <div class='tooltip'>mat.Vpred<span class='tooltiptext'>For n observations, w effects to be predicted, f nuiscance fixed effects and r nuisance random effects, the variances of a set of predicted effects is calculated using the incidence matrix for the effects to be predicted and, optionally, a variance matrix of the effects, an incidence matrix for the nuisance fixed factors and covariates, the variance matrix of the nuisance random effects in the mixed model and the residual variance matrix. This function has been superseded by mat.Vpredicts, which allows the use of both matrices and formulae.<\/span><\/div>, <div class='tooltip'>mat.Vpredicts<span class='tooltiptext'>For n observations, w effects to be predicted, f nuiscance fixed effects, r nuisance random effects and n residuals, the variances of a set of predicted effects is calculated using the incidence matrix for the effects to be predicted and, optionally, a variance matrix of these effects, an incidence matrix for the nuisance fixed factors and covariates, the variance matrix of the nuisance random effects and the residual variance matrix. The matrices can be supplied directly or using formulae and a matrix specifying the variances of the nuisance random effects. The difference between mat.Vpredicts and mat.Vpred is that the former has different names for equivalent arguments and the latter does not allow for the use of formulae.<\/span><\/div>, <div class='tooltip'>mpone<span class='tooltiptext'>Converts the first two levels of a factor into the numeric values -1 and +1.<\/span><\/div>, <div class='tooltip'>no.reps<span class='tooltiptext'>Computes the number of pure replicates required in an experiment to achieve a specified power.<\/span><\/div>, <div class='tooltip'>power.exp<span class='tooltiptext'>Computes the power for an experiment.<\/span><\/div>, <div class='tooltip'>print.aliasing<span class='tooltiptext'>Prints an aliasing data.frame.<\/span><\/div>, <div class='tooltip'>print.projector<span class='tooltiptext'>Print an object of class \"projector\", displaying the matrix and its degrees of freedom (rank).<\/span><\/div>, <div class='tooltip'>print.pstructure<span class='tooltiptext'>Prints a pstructure.object, which is of class pstructure. The df, terms and sources are coerced into a data.frame and printed; the marginality matrix is printed separately.<\/span><\/div>, <div class='tooltip'>print.summary.p2canon<span class='tooltiptext'>Prints a summary.p2canon object, which is also a data.frame, in a pretty format.<\/span><\/div>, <div class='tooltip'>print.summary.pcanon<span class='tooltiptext'>Prints a summary.pcanon object, which is also a data.frame, in a pretty format.<\/span><\/div>, <div class='tooltip'>proj2.combine<span class='tooltiptext'>The canonical relationship between a pair of projectors is established by decomposing the range of Q1 into a part that pertains to Q2 and a part that is orthogonal to Q2. It also produces the nonzero canonical efficiency factors for the joint decomposition of Q1 and Q and the corresponding eigenvectors of Q1 (James and Wilkinson, 1971). Q1 and Q2 may be nonorthogonal.<\/span><\/div>, <div class='tooltip'>proj2.decomp<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>proj2.efficiency<span class='tooltiptext'>Computes the canonical efficiency factors for the joint decomposition of two projectors (James and Wilkinson, 1971).<\/span><\/div>, <div class='tooltip'>proj2.eigen<span class='tooltiptext'>Computes the canonical efficiency factors for the joint decomposition of two projectors and the eigenvectors corresponding to the first projector (James and Wilkinson, 1971).<\/span><\/div>, <div class='tooltip'>proj2.ops<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>projector<span class='tooltiptext'>The class \"projector\" is the subclass of the class \"matrix\" in which matrices are square, symmetric and idempotent. The function projector tests whether a matrix satisfies these criteria and if it does creates a \"projector\" object, computing the projector's degrees of freedom and adding them to the object.<\/span><\/div>, <div class='tooltip'>projs.2canon<span class='tooltiptext'>Computes the canonical efficiency factors for the joint decomposition of two structures or sets of mutually orthogonally projectors (Brien and Bailey, 2009), orthogonalizing projectors in the Q2 list to those earlier in the list of projectors with which they are partially aliased. The results can be summarized in the form of a skeleton ANOVA table.<\/span><\/div>, <div class='tooltip'>projs.canon<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>projs.combine.p2canon<span class='tooltiptext'>Extracts, from a p2canon object obtained using projs.2canon, the projectors that give the combined canonical decomposition of two sets of projectors (Brien and Bailey, 2009).<\/span><\/div>, <div class='tooltip'>projs.structure<span class='tooltiptext'>These functions have been renamed and deprecated in dae.<\/span><\/div>, <div class='tooltip'>pstructure<span class='tooltiptext'>Constructs a pstructure.object that includes a set of mutually orthogonal projectors, one for each term in the formula. These are used to specify a structure, or an orthogonal decomposition of the data space. There are three methods available for orthogonalizing the projectors corresponding to the terms in the formula: differencing, eigenmethods or the default hybrid method. It is possible to use this function to find out what sources are associated with the terms in a model and to determine the marginality between terms in the model. The marginality matrix can be saved.<\/span><\/div>, <div class='tooltip'>pstructure.formula<span class='tooltiptext'>Constructs a pstructure.object that includes a set of mutually orthogonal projectors, one for each term in the formula. These are used to specify a structure, or an orthogonal decomposition of the data space. There are three methods available for orthogonalizing the projectors corresponding to the terms in the formula: differencing, eigenmethods or the default hybrid method. It is possible to use this function to find out what sources are associated with the terms in a model and to determine the marginality between terms in the model. The marginality matrix can be saved.<\/span><\/div>, <div class='tooltip'>qqyeffects<span class='tooltiptext'>Produces a half or full normal plot of the Yates effects from a 2^k2^k factorial experiment.<\/span><\/div>, <div class='tooltip'>rep.data.frame<span class='tooltiptext'>Replicate the rows of a data.frame by repeating each row consecutively and/or repeating all rows as a group.<\/span><\/div>, <div class='tooltip'>resid.errors<span class='tooltiptext'>An alias for the generic function residuals. When it is available, the method residuals.aovlist extracts residuals, which is provided in the package dae to cover aovlist objects.<\/span><\/div>, <div class='tooltip'>residuals.aovlist<span class='tooltiptext'>Extracts the residuals from error.term or, if error.term is not specified, the last error.term in the analysis. It is a method for the generic function residuals.<\/span><\/div>, <div class='tooltip'>rmvnorm<span class='tooltiptext'>Generates a vector of random values from an n-dimensional multivariate normal distribution whose mean is given by the n-vector mean and variance by the n x n symmetric matrix V. It uses the method described by Ripley (1987, p.98)<\/span><\/div>, <div class='tooltip'>set.daeTolerance<span class='tooltiptext'>A function that sets the values such that, in dae functions, values less than it are considered to be zero. The values are stored in a vector named daeTolerance in the daeEnv environment. The vector is of length two and, initially, both values are set to .Machine$double.eps ^ 0.5 (about 1.5E-08). One value is named element.tol and is used for elements of matrices; the second is named element.eigen and is used for eigenvalues and quantities based on them, such as efficiency factors.<\/span><\/div>, <div class='tooltip'>strength<span class='tooltiptext'>Generates paper strength values for an experiment with different temperatures.<\/span><\/div>, <div class='tooltip'>summary.p2canon<span class='tooltiptext'>Produces a summary of the efficiency criteria computed from the canonical efficiency factors for the joint decomposition of two sets of projectors (Brien and Bailey, 2009) obtained using projs.2canon. It takes the form of a decomposition or skeleton ANOVA table.<\/span><\/div>, <div class='tooltip'>summary.pcanon<span class='tooltiptext'>Gives the anatomy of a design in a table; it summarizes the joint decomposition of two or more sets of projectors (Brien and Bailey, 2009) obtained using designAnatomy. It includes the efficiency criteria computed from the canonical efficiency factors for the joint decomposition. The labels in the table may be Terms or Sources. The terms are those that would be included in a mixed model for an experiment based on the design. The sources are the orthogonal subspaces, derived from the terms, that make up the decomposition and the degrees of freedom and efficiency factors relate to these subspaces. The table displays how the information for the different sources allowed for in the design are related. For more information about the notation used for sources see the labels argument of designAnatomy. It is possible to supply an object that is a pcanon.object produced in versions prior to 3.0-0 using projs.canon.<\/span><\/div>, <div class='tooltip'>tukey.1df<span class='tooltiptext'>Performs Tukey's one-degree-of-freedom-test-for-nonadditivity on a set of residuals from an analysis of variance.<\/span><\/div>, <div class='tooltip'>yates.effects<span class='tooltiptext'>Extracts Yates effects from an aov object or aovlist object.<\/span><\/div>, <div class='tooltip'>Zncsspline<span class='tooltiptext'>Calculates the design matrix, Z, of the random effects for a natural cubic smoothing spline as described by Verbyla et al., (1999). An initial design matrix, \\bold{\\Delta} \\bold{\\Delta}^{-1} \\bold{\\Delta}\\bold{Delta} \\bold{Delta}^{-1} \\bold{Delta}, based on the knot points is computed. It can then be post multiplied by the power of the tri-diagonal matrix \\bold{G}_sG_S that is proportional to the variance matrix of the random spline effects. If the power is set to 0.5 then the random spline effects based on the resulting Z matrix will be independent with variance \\sigma_s^2sigma_s^2.<\/span><\/div>","<div class='tooltip'>Altscreen<span class='tooltiptext'>Recalls Jones and Montgomery's 16 run screening designs from data frames<\/span><\/div>, <div class='tooltip'>BIBsize<span class='tooltiptext'>This function computes the number of blocks, treatment frequency and lambda for a potential BIB design<\/span><\/div>, <div class='tooltip'>colormap<span class='tooltiptext'>This function makes a colormap of the correlations of a design matrix stored in the data frame design<\/span><\/div>, <div class='tooltip'>DefScreen<span class='tooltiptext'>Recalls Jones and Nachtsheim's Definitive screening designs for 3-level factors and 3-level factors with added 2-level categorical factors.<\/span><\/div>, <div class='tooltip'>EEw1s1<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 1 whole plot factor and 1 sub-plot factor from a catalog<\/span><\/div>, <div class='tooltip'>EEw1s2<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 1 whole plot factor and 2 sub-plot factors from a catalog<\/span><\/div>, <div class='tooltip'>EEw1s3<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 1 whole plot factor and 3 sub-plot factors from a catalog<\/span><\/div>, <div class='tooltip'>EEw2s1<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 2 whole plot factors and 1 sub-plot factor from a catalog<\/span><\/div>, <div class='tooltip'>EEw2s2<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 2 whole plot factors and 1 sub-plot factor from a catalog<\/span><\/div>, <div class='tooltip'>EEw2s3<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 2 whole plot factors and 1 sub-plot factor from a catalog<\/span><\/div>, <div class='tooltip'>EEw3<span class='tooltiptext'>Recalls Jones and Goos JQT Estimation Equivalent Response Surface Designs for 3 whole plot factors and 1-2 sub-plot factors from a catalog<\/span><\/div>, <div class='tooltip'>Fcrit<span class='tooltiptext'>Gets F-distribution critical values<\/span><\/div>, <div class='tooltip'>fhstep<span class='tooltiptext'>This function performs a single step of a hierarchical forward stepwise regression by entering additionalterm(s) to a model already created by ihstep or fhstep. If an interaction or quadratic term is entered first, the parent main effects are also entered into the model. This function is called by HierAFS.R<\/span><\/div>, <div class='tooltip'>fhstepDS<span class='tooltiptext'>This function performs a single step of a forward stepwise regression by entering an additional2nd order term to a model already created by FitDefSc.R or fhstepDS.R This function is called by FitDefSc.R<\/span><\/div>, <div class='tooltip'>firstm<span class='tooltiptext'>This function finds the first term to enter a hierarchical forward stepwise regression. If the term is an interaction or quadratic term, the parent main effects are also included. This function is called by ihstep.R<\/span><\/div>, <div class='tooltip'>FitDefSc<span class='tooltiptext'>This function performs fits a model to a Definitive Screeing Design by first restricting main effects to the smallest main effects and those significant at at least the .20 level in a main effects model. Next forward stepwise selection is used to enter 2 factor interactions and quadratic effects.<\/span><\/div>, <div class='tooltip'>fnextrm<span class='tooltiptext'>This function finds the first term to enter a hierarchical forward stepwise regression. If the term is an interaction or quadratic term, the parent main effects are also included. This function is called by ihstep.R<\/span><\/div>, <div class='tooltip'>fntrmDS<span class='tooltiptext'>This function finds the first term to enter a hierarchical forward stepwise regression. If the term is an interaction or quadratic term, the parent main effects are also included. This function is called by ihstep.R<\/span><\/div>, <div class='tooltip'>Fpower<span class='tooltiptext'>Calculates the power for the non-central F-distribution<\/span><\/div>, <div class='tooltip'>Fpower1<span class='tooltiptext'>Calculates the power for one-way ANOVA<\/span><\/div>, <div class='tooltip'>Fpower2<span class='tooltiptext'>Calculates the power for a two-way ANOVA<\/span><\/div>, <div class='tooltip'>fullnormal<span class='tooltiptext'>This function makes a full normal plot of the elements of the vector called effects<\/span><\/div>, <div class='tooltip'>gapstat<span class='tooltiptext'>This function computes the gap statistic which is used to test for an outlier using Daniels method<\/span><\/div>, <div class='tooltip'>Gaptest<span class='tooltiptext'>This function uses Daniel's Method to find an outlier in an unreplicated 2^{(k-p)} design.<\/span><\/div>, <div class='tooltip'>halfnorm<span class='tooltiptext'>This function makes a half normal plot of the elements of the vector called effects<\/span><\/div>, <div class='tooltip'>HierAFS<span class='tooltiptext'>This function performs a hierarchical forward stepwise regression. If an interaction or quadratic term is entered in the model, the parent main effects are also entered into the model.<\/span><\/div>, <div class='tooltip'>ihstep<span class='tooltiptext'>This function performs the first step of a hierarchical forward stepwise regression. If an interaction or quadratic term is entered first, the parent main effects are also entered into the model. This function is called by HierAFS.R<\/span><\/div>, <div class='tooltip'>LenthPlot<span class='tooltiptext'>Plot of the factor effects with significance levels based on robustestimation of contrast standard errors.<\/span><\/div>, <div class='tooltip'>LGB<span class='tooltiptext'>This function uses the LGB Method to detect significant effects in unreplicated fractional factorials.<\/span><\/div>, <div class='tooltip'>LGBc<span class='tooltiptext'>This function uses the LGB Method to detect significant effects in unreplicated fractional factorials.<\/span><\/div>, <div class='tooltip'>mod<span class='tooltiptext'>Gets mod of a to base b<\/span><\/div>, <div class='tooltip'>ModelRobust<span class='tooltiptext'>Recalls Li and Nachtsheim's model robust factorial designs from a catalog of data frames<\/span><\/div>, <div class='tooltip'>OptPB<span class='tooltiptext'>Selects the columns from a Plackett-Burman Design Produced by FrF2 that will minimize model dependence for main effects and two factor interactions and returns the design in a data frame<\/span><\/div>, <div class='tooltip'>PBDes<span class='tooltiptext'>Creates a 12, 20, or 24 run Plackett-Burman design in a data frame with numeric factor levels by cyclicallyrotating the factor leves in the first row<\/span><\/div>, <div class='tooltip'>Tukey1df<span class='tooltiptext'>This function performs Tukey's single degree of freedom test for interaction in an unreplicated two-factor design<\/span><\/div>, <div class='tooltip'>vci<span class='tooltiptext'>function for getting confidence intervals on variance components estimated by the method of moments<\/span><\/div>","<div class='tooltip'>generalized.wendland<span class='tooltiptext'>generalized.wendland() is a helper function that constructs a correlation matrix according to the generalized Wendland model with lengthscales given by the parameter vector theta. When kap=0 the correlation model corresponds to the Askey correlation model. The design must have been already formated in distlist format using the function makedistlist().<\/span><\/div>, <div class='tooltip'>getranges<span class='tooltiptext'>getranges() is a helper function to get the lower/upper bounds of variables in a design matrix, used for rescaling the inputs to the [0,1] hypercube.<\/span><\/div>, <div class='tooltip'>makedistlist<span class='tooltiptext'>makedistlist() is a helper function used to setup the difference matrices that are used by the DPP models.<\/span><\/div>, <div class='tooltip'>matern32<span class='tooltiptext'>matern32() is a helper function that constructs a correlation matrix according to the Matern model with parameter \\nu=3/2 and lengthscales given by the parameter vector theta. The design must have been already formated in distlist format using the function makedistlist().<\/span><\/div>, <div class='tooltip'>matern52<span class='tooltiptext'>matern52() is a helper function that constructs a correlation matrix according to the Matern model with parameter \\nu=5/2 and lengthscales given by the parameter vector theta. The design must have been already formated in distlist format using the function makedistlist().<\/span><\/div>, <div class='tooltip'>remove.projections<span class='tooltiptext'>remove.projections() is a helper function to identify all lower-dimensional marginal projection points of the existing design points indexed by curpts. This function can be used to remove a subset of points from the candidate set in order to enforce non-collapsingness of when sequentially adding design points.<\/span><\/div>, <div class='tooltip'>rhomat<span class='tooltiptext'>rhomat() is a helper function that constructs a correlation matrix according to the squared exponential model with parameterized by correlation parameters rho taking values in [0,1) and the exponent parameter alpha. The default of alpha=2 results in the Gaussian correlation while selecting alpha=1 corresponds to the Exponential correlation model. The design must have been already formated in distlist format using the function makedistlist().<\/span><\/div>, <div class='tooltip'>scaledesign<span class='tooltiptext'>scaledesign() is a helper function to rescale a design to the [0,1] hypercube using variable ranges previously extracted by a call to getranges().<\/span><\/div>, <div class='tooltip'>sim.dpp.modal<span class='tooltiptext'>sim.dpp.modal() uses the DPP-based design emulator of Pratola et al. (2018)to draw a sample of the n-run optimal design for a Gaussian processregression model with stationary correlation function r(x,x^\\prime), where theentries of R are formed by evaluating r(x,x^\\prime) over a grid of candidate locations.<\/span><\/div>, <div class='tooltip'>sim.dpp.modal.fast<span class='tooltiptext'>sim.dpp.modal.fast() is similar to sim.dpp.modal but is a C++ codepath that makes use ofSPAM's sparse matrices to enable faster computation. It implements the DPP-based design emulator of Pratola et al. (2018) to draw a sample of the n-run optimal design for a Gaussian process regression model with compact correlation function r(x,x^\\prime), where the entries of R are formed by evaluating r(x,x^\\prime) over a grid of candidate locations.<\/span><\/div>, <div class='tooltip'>sim.dpp.modal.fast.seq<span class='tooltiptext'>sim.dpp.modal.fast.seq() is similar to sim.dpp.modal.fast but sequentially selects n additional points to add to the design given that the points in curpts are alread in the design from previous sequential iterations. It uses the C++ codepath that makes use ofSPAM's sparse matrices to enable faster computation. It implements the DPP-based design emulator of Pratola et al. (2018) to draw a sequential sample of the n-run additional optimal design points for a Gaussian process regression model with compact correlation function r(x,x^\\prime), where the entries of R are formed by evaluating r(x,x^\\prime) over a grid of candidate locations.<\/span><\/div>, <div class='tooltip'>sim.dpp.modal.np<span class='tooltiptext'>sim.dpp.modal.np() uses sim.dpp.modal.nystrom.kmeans() to draw a design of n points in p dimensions using the kmeans-based Nystrom approximation of Zhang and Kwok (2010) and the DPP-based design emulator of Pratola et al. (2018). The design constructed assumes a Gaussian processregression model with stationary correlation function r(x,x^\\prime), where theentries of R are formed by evaluating r(x,x^\\prime) over a set of landmarks chosen by the kmeans algorithm, and the resulting eigenvectors are projected into the higher dimensional space using the Nystrom approximation. Additional options for sim.dpp.modal.nystrom.kmeans() can be passed to alter the construction of the landmark set.<\/span><\/div>, <div class='tooltip'>sim.dpp.modal.nystrom<span class='tooltiptext'>sim.dpp.modal.nystrom() uses the DPP-based design emulator of Pratola et al. (2018)to draw a sample of the n-run optimal design for a Gaussian processregression model with stationary correlation function r(x,x^\\prime), where theentries of R are formed by evaluating r(x,x^\\prime) over a grid of candidate locations. This function uses a grid-based Nystrom approximation based on the passed matrix X to avoid constructing a large correlation matrix if dimension ngrid^p and its subsequent eigendecomposition.<\/span><\/div>, <div class='tooltip'>sim.dpp.modal.nystrom.kmeans<span class='tooltiptext'>sim.dpp.modal.nystrom.kmeans() uses the kmeans-based Nystrom approximation of Zhang and Kwok (2010) to select n design sites from the observational dataset Xin using the DPP-based design emulator of Pratola et al. (2018). The design constructed assumes a Gaussian processregression model with stationary correlation function r(x,x^\\prime), where theentries of R are formed by evaluating r(x,x^\\prime) over a set of landmarks chosen by the kmeans algorithm, and the resulting eigenvectors are projected into the higher dimensional space using the Nystrom approximation. Additional options for the MiniBatchKmeans() algorithm from package ClusterR can be passed to alter the construction of the landmark set.<\/span><\/div>, <div class='tooltip'>sim.dpp.modal.seq<span class='tooltiptext'>sim.dpp.modal.seq() is similar to sim.dpp.modal but sequentially selects n additional points to add to the design given that the points in curpts are alread in the design from previous sequential iterations. It implements the DPP-based design emulator of Pratola et al. (2018) to draw a sequential sample of n-run additional optimal design points for a Gaussian processregression model with correlation function r(x,x^\\prime), where the entries of R are formed by evaluatingr(x,x^\\prime) over a grid of candidate locations. As is typical, R is formed based on all of the candidate grid points.<\/span><\/div>, <div class='tooltip'>unscalemat<span class='tooltiptext'>unscalemat() is a helper function to rescale a matrix back to its original ranges. Typically this is used to rescale the posterior samples of the parameters back to their original scale.<\/span><\/div>, <div class='tooltip'>wendland1<span class='tooltiptext'>wendland1() is a helper function that constructs a correlation matrix according to the Wendland 1 model with lengthscales given by the parameter vector theta. The design must have been already formated in distlist format using the function makedistlist().<\/span><\/div>, <div class='tooltip'>wendland2<span class='tooltiptext'>wendland2() is a helper function that constructs a correlation matrix according to the Wendland 2 model with lengthscales given by the parameter vector theta. The design must have been already formated in distlist format using the function makedistlist().<\/span><\/div>","<div class='tooltip'>acceptanceProbability<span class='tooltiptext'>Compute the acceptance probability for each updated design. It depends on the current temperature value of simulated annealing process. This is a subfunction needed for designGG, but is not directly used.<\/span><\/div>, <div class='tooltip'>arrayUpdate<span class='tooltiptext'>Update the allocation of samples on the arrays. This is a subfunction needed for updateDesign, but is not directly used.<\/span><\/div>, <div class='tooltip'>conditionAllocation<span class='tooltiptext'>This is a subfunction used by initialDesign but is not directly used. In the experiment where samples are profiled in pairs, the samples are firstly selected and paired on each array and then the selected samples are randomly allocated into different conditions.<\/span><\/div>, <div class='tooltip'>conditionCombination<span class='tooltiptext'>Generate a matrix indicating all possible levels for environmental factors with dimension nConditions * nEnvFactors. This is a subfunction needed for designScore, but is not directly used.<\/span><\/div>, <div class='tooltip'>conditionLevel<span class='tooltiptext'>Describe the levels of all environmental factors for each RIL/strain in the experiment. This is a subfunction needed for designScore, but is not directly used.<\/span><\/div>, <div class='tooltip'>conditionUpdate<span class='tooltiptext'>Update the allocation of samples onto different conditions. This is a subfunction needed for updateDesign, but is not directly used.<\/span><\/div>, <div class='tooltip'>designGG<span class='tooltiptext'>Main function to search and display A- and D- optimal designs for single- or two-channel genetical genomics experiments. Simulated annealing or Metropolis Hastings used to find the best design.<\/span><\/div>, <div class='tooltip'>designScore<span class='tooltiptext'>According to the current experimental design, the Fisher information matrix is obtained and then either the A- or D- optimality score is computed.<\/span><\/div>, <div class='tooltip'>experimentDesignTable<span class='tooltiptext'>This function generates two .csv files which descibe how samples are allocated samples into different conditions and paired on arrays.<\/span><\/div>, <div class='tooltip'>initialDesign<span class='tooltiptext'>Allocate RILs (or strains) into different conditional and pair RILs (or strains) into slides.<\/span><\/div>, <div class='tooltip'>interactionLevel<span class='tooltiptext'>Generate levels for all interacting factors for all RILs (or strains). This is a subfunction needed for designScore, but is not directly used.<\/span><\/div>, <div class='tooltip'>nSlidesANDnTuple<span class='tooltiptext'>Internal designGG functions. These are generally not to be called by the user.<\/span><\/div>, <div class='tooltip'>pairLevel<span class='tooltiptext'>Pair levels for two RILs (or strains) allocated into one slide (bTwoColorArray=TRUE). It is a subfunction needed for designScore function, but is not directly used.<\/span><\/div>, <div class='tooltip'>plotAllScores<span class='tooltiptext'>Plot all scores and the temperature at each iteration during the simulated annealing process.<\/span><\/div>, <div class='tooltip'>temperatureStep<span class='tooltiptext'>Calculate the temperature decreasing step for simulated annealing process. This is a subfunction needed for designGG, but is not directly used.<\/span><\/div>, <div class='tooltip'>updateDesign<span class='tooltiptext'>Updates current experimental design (including array.allocation and condition.allocation).<\/span><\/div>, <div class='tooltip'>variableNames<span class='tooltiptext'>Generate variable names for genetic, environmental factors and interacting terms.<\/span><\/div>, <div class='tooltip'>variableNumber<span class='tooltiptext'>When nEnvFactors = 1 and nLevels = 1, there is no environmetal pertubation in the experimental. Then we re-define nEnvFactors to be 0 within the algorithm. nEnvFactors = 0, only genetic factor is considered. nEnvFactors > 1, genetic and environmental facotrs, and all possible interacting factors are considered.<\/span><\/div>","<div class='tooltip'>absstddif<span class='tooltiptext'>Function for calculating absolute differences in means between the covariates in the treatment and control groups in terms of the original units of the covariates. Here, the absolute differences in means are normalized by the simple average of the treated and control standard deviations before matching (see Rosenbaum and Rubin 1985 for details).<\/span><\/div>, <div class='tooltip'>bmatch<span class='tooltiptext'>Function for optimal bipartite matching in observational studies that directly balances the observed covariates. bmatch allows the user to enforce different forms of covariate balance in the matched samples such as moment balance (e.g., of means, variances and correlations), distributional balance (e.g., fine balance, near-fine balance, strength-k balancing) and exact matching. In observational studies where an instrumental variable is available, bmatch can be used to handle weak instruments and strengthen them by means of the far options (see Yang et al. 2014 for an example). bmatch can also be used in discontinuity designs by matching units in a neighborhood of the discontinuity (see Keele et al. 2015 for details). In any of these settings, bmatch either minimizes the total sum of covariate distances between matched units, maximizes the total number of matched units, or optimizes a combination of the two, subject to matching, covariate balancing, and representativeness constraints (see the examples below).<\/span><\/div>, <div class='tooltip'>cardmatch<span class='tooltiptext'>Function for optimal cardinality matching in observational studies. cardmatch finds the largest matched sample that is balanced and representative by design. The formulation of cardmatch has been simplified to handle larger data than bmatch or nmatch. Similar to bmatch or nmatch, the performance of cardmatch is greatly enhanced by using the solver options cplex or gurobi.<\/span><\/div>, <div class='tooltip'>distmat<span class='tooltiptext'>Function for building a normalized rank-based Mahalanobis distance matrix with a penalty for caliper violation.<\/span><\/div>, <div class='tooltip'>distmatch<span class='tooltiptext'>Function for optimal distance matching in observational studies. distmatch minimizes the total sum of covariate distances between matches. distmatch is a wrapper to bmatch.<\/span><\/div>, <div class='tooltip'>ecdfplot<span class='tooltiptext'>Function that plots the empirical cumulative distribution function of a given covariate for treated units and matched controls. ecdfplot can be used to visually inspect the balance of the entire empirical distribution function of the covariate in question.<\/span><\/div>, <div class='tooltip'>finetab<span class='tooltiptext'>Function for tabulating the marginal distributions of a nominal covariate for the treated units and matched controls.<\/span><\/div>, <div class='tooltip'>loveplot<span class='tooltiptext'>Function that creates a Love plot for assessing covariate balance after matching.<\/span><\/div>, <div class='tooltip'>meantab<span class='tooltiptext'>Function for tabulating the means and other basic statistics useful to assess covariate balance after matching.<\/span><\/div>, <div class='tooltip'>nmatch<span class='tooltiptext'>Function for optimal nonbipartite matching in randomized experiments and observational studies that directly balances the observed covariates. nmatch allows the user to enforce different forms of covariate balance in the matched samples, such as moment balance (e.g., of means, variances, and correlations), distributional balance (e.g., fine balance, near-fine balance, strength-k balancing), and exact matching. Among others, nmatch can be used in the design of randomized experiments for matching before randomization (Greevy et al. 2004, Zou and Zubizarreta 2016), and in observational studies for matching with doses and strengthening an instrumental variable (Baiocchi et al. 2010, Lu et al. 2011).<\/span><\/div>, <div class='tooltip'>pairsplot<span class='tooltiptext'>Function for visualizing matched pairs in two dimensions.<\/span><\/div>","<div class='tooltip'>contrast.names<span class='tooltiptext'>This function can be used to retrieve contrast codes based on experimental codes / planned observations.Use this function to retrieve the names for contrasts in an experimental design, such as used by lm, lmer and many other regression models.<\/span><\/div>, <div class='tooltip'>design.codes<span class='tooltiptext'>These functions return useful summaries of a factor design, including the design matrix itself as well as other parameters and a list of random factors as experimental units.<\/span><\/div>, <div class='tooltip'>design.contrasts<span class='tooltiptext'>This function can be used to retrieve contrast codes based on experimental codes / planned observations.Use this function to retrieve the names for contrasts in an experimental design, such as used by lm, lmer and many other regression models.<\/span><\/div>, <div class='tooltip'>design.formula<span class='tooltiptext'>These functions return useful summaries of a factor design, including the design matrix itself as well as other parameters and a list of random factors as experimental units.<\/span><\/div>, <div class='tooltip'>design.units<span class='tooltiptext'>These functions return useful summaries of a factor design, including the design matrix itself as well as other parameters and a list of random factors as experimental units.<\/span><\/div>, <div class='tooltip'>factor.design<span class='tooltiptext'>The main function of this package is to create factorial designs with this function.<\/span><\/div>, <div class='tooltip'>fixed.factor<span class='tooltiptext'>This function creates an instance of fixedFactor to be used in a factorDesign. Fixed factors typically relate to (quasi-)experimental factors such as experimental conditions/manipulations, subject/item characteristics ect.<\/span><\/div>, <div class='tooltip'>fixed.factors<span class='tooltiptext'>From a given design, extract contained random or fixed factors as a list.<\/span><\/div>, <div class='tooltip'>is.designFactor<span class='tooltiptext'>Check if argument is a design factor (either random or fixed factor), specifically a random factor, a fixed factor or a factor design.<\/span><\/div>, <div class='tooltip'>is.factorDesign<span class='tooltiptext'>Check if argument is a design factor (either random or fixed factor), specifically a random factor, a fixed factor or a factor design.<\/span><\/div>, <div class='tooltip'>is.fixedFactor<span class='tooltiptext'>Check if argument is a design factor (either random or fixed factor), specifically a random factor, a fixed factor or a factor design.<\/span><\/div>, <div class='tooltip'>is.randomFactor<span class='tooltiptext'>Check if argument is a design factor (either random or fixed factor), specifically a random factor, a fixed factor or a factor design.<\/span><\/div>, <div class='tooltip'>nobs<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>output.design<span class='tooltiptext'>These functions return useful summaries of a factor design, including the design matrix itself as well as other parameters and a list of random factors as experimental units.<\/span><\/div>, <div class='tooltip'>random.factor<span class='tooltiptext'>This function creates an instance of randomFactor to be used in a factorDesign. A random factor is typically related to an experimental unit such as Subject, Item, Experimenter, ect. and does not have preset levels.<\/span><\/div>, <div class='tooltip'>random.factors<span class='tooltiptext'>From a given design, extract contained random or fixed factors as a list.<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>show.factorContainer<span class='tooltiptext'>Output a design factor summary<\/span><\/div>, <div class='tooltip'>simLMM<span class='tooltiptext'>This function simulates artificial data from a linear mixed-effects model.<\/span><\/div>, <div class='tooltip'>subset<span class='tooltiptext'>Subset factor designs<\/span><\/div>, <div class='tooltip'>write.design<span class='tooltiptext'>This function writes a design into a set of files. For each random factor, a unit list is created that contains a list of all levels (instances) of the random factor and the factor levels to which that level is assigned. Moreover, code_files are created that contain a complete set of experimental codes.<\/span><\/div>, <div class='tooltip'>write.design.csv<span class='tooltiptext'>This function writes a design into a set of files. For each random factor, a unit list is created that contains a list of all levels (instances) of the random factor and the factor levels to which that level is assigned. Moreover, code_files are created that contain a complete set of experimental codes.<\/span><\/div>, <div class='tooltip'>write.design.json<span class='tooltiptext'>This function writes a design into a set of files. For each random factor, a unit list is created that contains a list of all levels (instances) of the random factor and the factor levels to which that level is assigned. Moreover, code_files are created that contain a complete set of experimental codes.<\/span><\/div>, <div class='tooltip'>write.design.xlsx<span class='tooltiptext'>This function writes a design into a set of files. For each random factor, a unit list is created that contains a list of all levels (instances) of the random factor and the factor levels to which that level is assigned. Moreover, code_files are created that contain a complete set of experimental codes.<\/span><\/div>","<div class='tooltip'>desplot<span class='tooltiptext'>Use this function to plot the layout of a rectangular latticefield experiment and also the observed data values.<\/span><\/div>, <div class='tooltip'>geom_tileborder<span class='tooltiptext'>`geom_tileborder` draws a border between tiles of different classes.The required aesthetics are `aes(x,y,grp)`, where `grp` is the groupingclassification that separates tiles.<\/span><\/div>, <div class='tooltip'>ggdesplot<span class='tooltiptext'>Use this function to plot the layout of a rectangular latticefield experiment and also the observed data values.<\/span><\/div>, <div class='tooltip'>panel.outlinelevelplot<span class='tooltiptext'>This is a panel function for desplot which fills cells witha background color and adds outlines around blocks of cells.<\/span><\/div>, <div class='tooltip'>RedGrayBlue<span class='tooltiptext'>A function to create a Red-Gray-Blue palette.<\/span><\/div>","<div class='tooltip'>cohere<span class='tooltiptext'>Returns a message on the coherence status of a two-stage CRM design.<\/span><\/div>, <div class='tooltip'>crm<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>crmh<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>crmhlgt<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>crmht<span class='tooltiptext'>titecrm is used to compute a dose for the next patient in a phase I trial according to the TITE-CRM.<\/span><\/div>, <div class='tooltip'>crmht2<span class='tooltiptext'>titecrm is used to compute a dose for the next patient in a phase I trial according to the TITE-CRM.<\/span><\/div>, <div class='tooltip'>crmht2lgt<span class='tooltiptext'>titecrm is used to compute a dose for the next patient in a phase I trial according to the TITE-CRM.<\/span><\/div>, <div class='tooltip'>crmhtlgt<span class='tooltiptext'>titecrm is used to compute a dose for the next patient in a phase I trial according to the TITE-CRM.<\/span><\/div>, <div class='tooltip'>crmsens<span class='tooltiptext'>Evaluate the model sensitivity in the CRM by indifference intervals.<\/span><\/div>, <div class='tooltip'>crmsim<span class='tooltiptext'>crmsim is used to generate simulation replicates of phase I trial using the (group) CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>getinit<span class='tooltiptext'>Returns an initial design that is compatible with the specified CRM setup when used in a two-stage design.<\/span><\/div>, <div class='tooltip'>getn<span class='tooltiptext'>Sample size calculator for a one-stage Bayesian CRM (see Details for design specification).<\/span><\/div>, <div class='tooltip'>getprior<span class='tooltiptext'>Returns a vector of initial guesses of toxicity probabilities associated the doses for a given model sensitivity (set of indifference intervals).<\/span><\/div>, <div class='tooltip'>lcrm<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>lcrmlgt<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>mtite<span class='tooltiptext'>titesim is used to generate simulation replicates of phase I trial using the TITE-CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>mtrials<span class='tooltiptext'>crmsim is used to generate simulation replicates of phase I trial using the (group) CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>myjitter<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>nopt<span class='tooltiptext'>Sample size calculator for a one-stage Bayesian CRM (see Details for design specification).<\/span><\/div>, <div class='tooltip'>onetite<span class='tooltiptext'>titesim is used to generate simulation replicates of phase I trial using the TITE-CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>onetrial<span class='tooltiptext'>crmsim is used to generate simulation replicates of phase I trial using the (group) CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>plot.mtd<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>plot.sim<span class='tooltiptext'>crmsim is used to generate simulation replicates of phase I trial using the (group) CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>print.crmsize<span class='tooltiptext'>Sample size calculator for a one-stage Bayesian CRM (see Details for design specification).<\/span><\/div>, <div class='tooltip'>print.dxcrm<span class='tooltiptext'>Evaluate the model sensitivity in the CRM by indifference intervals.<\/span><\/div>, <div class='tooltip'>print.mtd<span class='tooltiptext'>crm is used to compute a dose for the next patient in a phase I trial according to the CRM.<\/span><\/div>, <div class='tooltip'>print.sim<span class='tooltiptext'>crmsim is used to generate simulation replicates of phase I trial using the (group) CRM under a specified dose-toxicity configuration.<\/span><\/div>, <div class='tooltip'>titecrm<span class='tooltiptext'>titecrm is used to compute a dose for the next patient in a phase I trial according to the TITE-CRM.<\/span><\/div>, <div class='tooltip'>titesim<span class='tooltiptext'>titesim is used to generate simulation replicates of phase I trial using the TITE-CRM under a specified dose-toxicity configuration.<\/span><\/div>","<div class='tooltip'>AUC.estim<span class='tooltiptext'>The AUC.estim function uses a compartmentl method or a non-compartmental method to estimate the AUC. In the field of pharmacokinetics, the area under the curve (AUC) is the area under the curve (mathematically known as definite integral) in a plot of concentration of drug in blood plasma against time. AUC is computed as the doses over the second pharmacokinetic's parameter, the clearance (CL).<\/span><\/div>, <div class='tooltip'>dtox<span class='tooltiptext'>The DTOX model enables us to estimate posterior probability of toxicity p_T versus dose directly. The dose-toxicity model is given by:p_T(d_k,\\boldsymbol{\\beta}) = \\Phi(-\\beta_0 + \\beta_1 log(d_k))where \\beta_q \\sim U(l_q, u_q) \\forall q = 0,1 andbeta0mean = c(l_0, u_0),beta1mean = c(l_1, u_1)where default choices are beta0mean = c(0, 16.71) and beta1mean = c(0, 6.43). So the default choices for model's priors are given by betapriors = c(l_0 = 0, u_0 = 16.71, l_1 = 0, u_1 = 6.43)Finally, the DTOX model has the following stopping rule in toxicity:ifP(p_T(dose) > theta) > probthen, no dose is suggested and the trial is stopped.<\/span><\/div>, <div class='tooltip'>invlogit<span class='tooltiptext'>Inverse-logit function, transforms continuous values to the range (0, 1)<\/span><\/div>, <div class='tooltip'>nextDose<span class='tooltiptext'>nextDose is used to perform parameter estimation at each step during a dose-finding trial. Determines the next or recommended dose level in a phase I clinical trial.<\/span><\/div>, <div class='tooltip'>nsim<span class='tooltiptext'>nsim is used to simulate a single or \"n\" prospective clinical trial(s) using the PK data and then link them to toxicity under a specified dose-toxicity configuration. The objective is to determine the maximum tolerated dose (MTD).<\/span><\/div>, <div class='tooltip'>pk.estim<span class='tooltiptext'>Estimation of the pharmacokinetic's (PK) measure of exposure.<\/span><\/div>, <div class='tooltip'>pkcov<span class='tooltiptext'>The PKCOV model is a modification of the method proposed by Piantadosi and Liu (1996) who suggested to use the AUC as covariate for p_T, probability of toxicity, through the logit link. Therefore, the dose-toxicity model is:logit(p_T(d_k, \\Delta {z_d}_k, \\boldsymbol{\\beta})) = -\\beta_0 + \\beta_1 log(d_k) + \\beta_2 \\Delta {z_d}_kwhere \\boldsymbol{\\beta} = (\\beta_1, \\beta_2), \\beta_0 is a constantwith \\beta_0 = -beta0mean,\\beta_1 \\sim U(l_1, u_1),\\beta_2 \\sim U(0,5),beta1mean = c(l_1, u_1)where default choices are beta0mean = -14.76, beta1mean = c(0, 8.23) and \\Delta {z_d}_k is the difference between the logarithm of population AUC at dose d_k and z, the logarithm of AUC of the subject at the same dose. Therefore, the default choices for model's priors are given by betapriors = c(beta0mean = -14.76, l_1 = 0, u_1 = 8.23)Finally, the PKCOV model has the following stopping rule in toxicity:ifP(p_T(dose) > theta) > probthen, no dose is suggested and the trial is stopped.<\/span><\/div>, <div class='tooltip'>pkcrm<span class='tooltiptext'>The PKCRM model is a combination of PKLIM as given below: z_{i} \\vert \\boldsymbol{\\beta}, \\nu \\sim N \\left( \\beta_0 + \\beta_1 \\log d_{i}, \\nu^{2} \\right)where \\boldsymbol{\\beta} = (\\beta_0, \\beta_1) are the regression parameters and \\nu is the standard deviation,and the Continual Reassessment Method's (CRM) model:p_T(d_k, \\beta) = d^\\beta_k(CRM)The default choices of the priors are:\\boldsymbol{\\beta} \\vert \\nu \\sim N(m, \\nu*beta0),\\nu \\sim Beta(1,1),m = (-log(CL_{pop}), 1) where Cl_{pop} is the population clearance.where default choices are Cl_{pop} = 10 and beta0 = 10000. Therefore, the default choices for model's priors are given by betapriors = c(Cl_{pop} = 10, beta0 = 10000)For the CRM model:Skeleton CRM = (0.01, 0.05, 0.1, 0.2, 0.35, 0.45) and\\beta \\sim N(0, 1.34)Finally, the PKCRM model has the following stopping rule in toxicity:ifP(p_T(dose) > theta) > probthen, no dose is suggested and the trial is stopped.<\/span><\/div>, <div class='tooltip'>pklogit<span class='tooltiptext'>The PKLOGIT model, inspired by Whitehead et al. (2007), uses z_i instead of dose d_i as a covariate in a logistic regression model for p_T. Therefore, we have that:\\mbox{logit}(p_T(z, \\boldsymbol{\\beta})) = -\\beta_2 + \\beta_3 zwith a bivariate Uniform distribution as prior distribution for \\boldsymbol{\\beta} = (\\beta_2, \\beta_3) and the hierarchical model of PK-toxicity for z_i given as:z_{i} \\vert \\boldsymbol{\\beta}, \\nu \\sim N \\left( \\beta_0 + \\beta_1 \\log d_{i}, \\nu^{2} \\right)where \\boldsymbol{\\beta} = (\\beta_0,\\beta_1) are the regression parameters and \\nu is the standard deviation.The default choices of the priors are:\\boldsymbol{\\beta} \\vert \\nu \\sim N(m, \\nu*beta0),\\nu \\sim Beta(1,1),m = (-log(CL_{pop}), 1), where Cl_{pop} is the population clearance.\\beta_2 \\sim U(0, beta2mean),\\beta_3 \\sim U(0, beta3mean)where default choices are Cl_{pop} = 10, beta0 = 10000, beta2mean = 20 and beta3mean = 10. Therefore, the default choices for model's priors are given by betapriors = c(Cl_{pop} = 10, beta0 = 10000, beta2mean = 20, beta3mean = 10)Finally, the PKLOGIT model has the following stopping rule in toxicity:ifP(p_T(dose) > theta) > probthen, no dose is suggested and the trial is stopped.<\/span><\/div>, <div class='tooltip'>pkpop<span class='tooltiptext'>The PKPOP model is a variation of the PKLOGIT model which replaced AUCs (z_j) with AUC of the population (z_{k,pop}), where z_{k,pop} is the mean value of the logarithm of AUC at dose k, predicted by the hierarchical model:z_{i} \\vert \\boldsymbol{\\beta}, \\nu \\sim N \\left( \\beta_0 + \\beta_1 \\log d_{i}, \\nu^{2} \\right)where \\boldsymbol{\\beta} = (\\beta_0,\\beta_1) are the regression parameters and \\nu is the standard deviation.and the logistic regression model:\\mbox{logit}(p_T(z_{k,pop}, \\boldsymbol{\\beta})) = -\\beta_3 + \\beta_4 z_{k,pop}with a bivariate Uniform distribution as prior distribution for the parameters \\boldsymbol{\\beta} = (\\beta_3, \\beta_4).The default choices of the priors are:\\boldsymbol{\\beta} \\vert \\nu \\sim N(m, \\nu*beta0),\\nu \\sim Beta(1,1),m = (-log(CL_{pop}), 1), where Cl_{pop} is the population clearance.\\beta_3 \\sim U(0, beta3mean),\\beta_4 \\sim U(0, beta4mean)where default choices are Cl_{pop} = 10, beta0 = 10000, beta3mean = 10 and beta4mean = 5. Therefore, the default choices for model's priors are given by betapriors = c(Cl_{pop} = 10, beta0 = 10000, beta3mean = 10, beta4mean = 5)Finally, the PKPOP model has the following stopping rule in toxicity:ifP(p_T(dose) > theta) > probthen, no dose is suggested and the trial is stopped.<\/span><\/div>, <div class='tooltip'>pktox<span class='tooltiptext'>The PKTOX model is essentially the PKLOGIT model with a probit regression model replacing the logistic regression, that is given by:p_T(z, \\boldsymbol{\\beta}) = \\Phi(-\\beta_2 + \\beta_3z)with a bivariate Uniform distribution as prior distribution for the parameters \\boldsymbol{\\beta} = (\\beta_2, \\beta_3) and the hierarchical model of PK-toxicity for z_i given as:z_{i} \\vert \\boldsymbol{\\beta}, \\nu \\sim N \\left( \\beta_0 + \\beta_1 \\log d_{i}, \\nu^{2} \\right)where \\boldsymbol{\\beta} = (\\beta_0,\\beta_1) are the regression parameters and \\nu is the standard deviation.The default choices of the priors are:\\boldsymbol{\\beta} \\vert \\nu \\sim N(m, \\nu*beta0),\\nu \\sim Beta(1,1),m = (-log(CL_{pop}), 1), where Cl_{pop} is the population clearance.\\beta_2 \\sim U(0, beta2mean),\\beta_3 \\sim U(0, beta3mean)where default choices are Cl_{pop} = 10, beta0 = 10000, beta2mean = 20 and beta3mean = 10. Therefore, the default choices for model's priors are given by betapriors = c(Cl_{pop} = 10, beta0 = 10000, beta2mean = 20, beta3mean = 10)Finally, the PKTOX model has the following stopping rule in toxicity:ifP(p_T(dose) > theta) > probthen, no dose is suggested and the trial is stopped.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>sim.data<span class='tooltiptext'>This function can be used to generate and store PK and toxicity data in order to be used for simulation according to the dose-finding model.<\/span><\/div>","<div class='tooltip'>coverage<span class='tooltiptext'>Compute the coverage measure<\/span><\/div>, <div class='tooltip'>discrepancyCriteria<span class='tooltiptext'>Compute discrepancy criteria.<\/span><\/div>, <div class='tooltip'>discrepESE_LHS<span class='tooltiptext'>The objective is to produce low-discrepancy LHS. ESE is a powerful genetic algorithm to produce space-filling designs. It has been adapted here to main discrepancy criteria.<\/span><\/div>, <div class='tooltip'>discrepSA_LHS<span class='tooltiptext'>The objective is to produce low-discrepancy LHS. SA is an efficient algorithm to produce space-filling designs. It has been adapted here to main discrepancy criteria.<\/span><\/div>, <div class='tooltip'>dmaxDesign<span class='tooltiptext'>Space-Filling Designs with nn experiments based on covariance matrix in [0,1]^d^d.<\/span><\/div>, <div class='tooltip'>factDesign<span class='tooltiptext'>Create a factorial design with nn = pow(levels,dimension) experimentsin [0,1]^d^d.<\/span><\/div>, <div class='tooltip'>faureprimeDesign<span class='tooltiptext'>Generate a Faure sequence with n=p^u-1n=p^u-1 experiments in [0,1]^d^d or other domains (see the details) where pp is the first prime number equal or larger than dd and uu is an exponent, usually 2.<\/span><\/div>, <div class='tooltip'>lhsDesign<span class='tooltiptext'>Simple (random) Latin Hypercube Design (randomized or centered) with nn experiments in [0,1]^d^d.<\/span><\/div>, <div class='tooltip'>maximinESE_LHS<span class='tooltiptext'>The objective is to produce maximin LHS. ESE is a powerful genetic algorithm allowing to produce space-filling designs.<\/span><\/div>, <div class='tooltip'>maximinSA_LHS<span class='tooltiptext'>The objective is to produce maximin LHS. SA is an efficient algorithm to produce space-filling designs.<\/span><\/div>, <div class='tooltip'>meshRatio<span class='tooltiptext'>The meshRatio criterion is the ratio between themaximimum and the minimum distance between two points of the experimental design.<\/span><\/div>, <div class='tooltip'>mindist<span class='tooltiptext'>Compute the mindist criterion (also called maximin)<\/span><\/div>, <div class='tooltip'>mstCriteria<span class='tooltiptext'>Compute both the mean and the standard deviation of the Minimal Spanning Tree (MST)<\/span><\/div>, <div class='tooltip'>nolhDesign<span class='tooltiptext'>This function generates a NOLH design of dimension 2 to 29 and normalizes it to the selected range. The design is extracted from Cioppa's NOLHdesigns list.<\/span><\/div>, <div class='tooltip'>nolhdrDesign<span class='tooltiptext'>This function generates a NOLH design of dimension 2 to 29 and normalizes it to the selected range. From 2 to 7 input variables, the design is extracted from Cioppa's NOLHdesigns list and from 8 to 29 input variables it is extracted from De Rainville's NOLHDRdesigns list.<\/span><\/div>, <div class='tooltip'>olhDesign<span class='tooltiptext'>Generate the Orthogonal Latin Hypercube (OLH) designs proposed by Nguyen in 2008. These OLHs have a latin structure, an orthogonality between the main terms and the interactions (+ squares) and low correlations between the interactions (+ squares). Very larges matrices can be obtained as the number of input variables and hence the number of lines is unconstrained. When the number of input variables is a power of 2, OLHs have d columns and n = 2d + 1 lines (experiments). A vertical truncature is applied when the number of input variables is not a power of 2. Various normalizations can be applied.<\/span><\/div>, <div class='tooltip'>phiP<span class='tooltiptext'>Compute the \\phi_p criterion (strongly linked to mindist criterion)<\/span><\/div>, <div class='tooltip'>rss2d<span class='tooltiptext'>For a 2-dimensional design, the 2D radial scanning statistic (RSS) scans angularly the domain. In each direction, it compares the distribution of projected points to their theoretical distribution under the assumption that all design points are drawn from uniform distribution. For a d-dimensional design, all pairs of dimensions are scanned.The RSS detects the defects of low discrepancy sequences or orthogonal arrays, and can be used for selecting space-filling designs.<\/span><\/div>, <div class='tooltip'>rss3d<span class='tooltiptext'>For a 3-dimensional design, the 3D radial scanning statistic (RSS) scans angularly the domain. In each direction, it compares the distribution of projected points to their theoretical distribution under the assumption that all design points are drawn from uniform distribution. For a d-dimensional design, all triplets of dimensions are scanned. The RSS detects the defects of low discrepancy sequences or orthogonal arrays, and can be used for selecting space-filling designs.<\/span><\/div>, <div class='tooltip'>runif.faure<span class='tooltiptext'>Generate a Faure sequence with nn experiments in [0,1]^d^d.<\/span><\/div>, <div class='tooltip'>scaleDesign<span class='tooltiptext'>This function scales the values of the design points to values comprised in [0,1]. The scaling can be made by the Rosenblatt transformation (uniformization by applying the empirical cumulative distribution function) or by translating the design from maximum and minimum values (given for each variable).<\/span><\/div>, <div class='tooltip'>straussDesign<span class='tooltiptext'>Space-Filling Designs based on Strauss process<\/span><\/div>, <div class='tooltip'>unif.test.quantile<span class='tooltiptext'>Computes the quantile of a uniformity test at a given significance level (see available tests and levels below).<\/span><\/div>, <div class='tooltip'>unif.test.statistic<span class='tooltiptext'>Computes the statistic of a uniformity test (see available tests below).<\/span><\/div>, <div class='tooltip'>unscaleDesign<span class='tooltiptext'>This function unscales the values of a scaled design (values in [0,1]). The unscaling can be made by the inverse Rosenblatt transformation (by applying the empirical quantile function given by another design) or by translating the design from maximum and minimum values (given for each variable).<\/span><\/div>, <div class='tooltip'>wspDesign<span class='tooltiptext'>The WSP (Wooton, Sergent, Phan-Tan-Luu) algorithm is an iterative algorithm based on suppression of some experiments from an initial design in each step. WSP leads to a space filling design<\/span><\/div>, <div class='tooltip'>xDRDN<span class='tooltiptext'>Extract a design contained in a list (i.e. with a design item), adjust the range, give it dimnames and finally round the values to a certain number of digits. Colnames will look like (A,B,C), (X1,X2,X3), (X01,X02,X03), (X001,X002,X003).<\/span><\/div>","<div class='tooltip'>branin<span class='tooltiptext'>Branin-Hoo 2-dimensional test function.<\/span><\/div>, <div class='tooltip'>camelback<span class='tooltiptext'>Camelback 2-dimensional test function.<\/span><\/div>, <div class='tooltip'>checkNames<span class='tooltiptext'>Tests if the names of a second matrix are equal to a given matrix up to a permutation, and permute its columns accordingly. When the second one has no column names, the names of the first one are used in the same order.<\/span><\/div>, <div class='tooltip'>checkNamesList<span class='tooltiptext'>Tests if the names of a second matrix are equal to a given matrix up to a permutation, and permute its columns accordingly. When the second one has no column names, the names of the first one are used in the same order.<\/span><\/div>, <div class='tooltip'>coef<span class='tooltiptext'>Get or set coefficients values.<\/span><\/div>, <div class='tooltip'>computeAuxVariables<span class='tooltiptext'>Computes or updates some auxiliary variables used for kriging (see below). This is useful in several situations : when all parameters are known (as for one basic step in Bayesian analysis), or when some new data is added but one does not want to re-estimate the model coefficients. On the other hand, computeAuxVariables is not used during the estimation of covariance parameters, since this function requires to compute the trend coefficients at each optimization step; the alternative given by (Park, Baek, 2001) is preferred.<\/span><\/div>, <div class='tooltip'>covMat1Mat2<span class='tooltiptext'>Computes the cross covariance matrix between two sets of locations for a spatial random process with a given covariance structure. Typically the two sets are a learning set and a test set.<\/span><\/div>, <div class='tooltip'>covMatrix<span class='tooltiptext'>Computes the covariance matrix at a set of locations for a spatial random process with a given covariance structure.<\/span><\/div>, <div class='tooltip'>covMatrixDerivative<span class='tooltiptext'>Computes a partial derivative of the covariance matrix C in function covMatrix.<\/span><\/div>, <div class='tooltip'>covparam2vect<span class='tooltiptext'>Gather the covariance parameters in a single vector. This is useful in the estimation step. Not for direct use.<\/span><\/div>, <div class='tooltip'>covParametersBounds<span class='tooltiptext'>Default boundaries for covariance parameters.<\/span><\/div>, <div class='tooltip'>covStruct.create<span class='tooltiptext'>Creates a covariance structure.<\/span><\/div>, <div class='tooltip'>covVector.dx<span class='tooltiptext'>Computes the gradient of the covariance vector c(x) computed by covMat1Mat2 with respect to x, for a given covariance structure.<\/span><\/div>, <div class='tooltip'>cv<span class='tooltiptext'>Multiple fold cross validation for a km object without noisy observations.<\/span><\/div>, <div class='tooltip'>drop.response<span class='tooltiptext'>Drop the response in the formula specifying the linear trend.<\/span><\/div>, <div class='tooltip'>goldsteinPrice<span class='tooltiptext'>Goldstein price 2-dimensional test function.<\/span><\/div>, <div class='tooltip'>hartman3<span class='tooltiptext'>Hartman 3-dimensional test function.<\/span><\/div>, <div class='tooltip'>hartman6<span class='tooltiptext'>Hartman 6-dimensional test function.<\/span><\/div>, <div class='tooltip'>inputnames<span class='tooltiptext'>Get the names of the input variables.<\/span><\/div>, <div class='tooltip'>kernelname<span class='tooltiptext'>Get the name of the underlying tensor-product covariance structure.<\/span><\/div>, <div class='tooltip'>km<span class='tooltiptext'>km is used to fit kriging models when parameters are unknown, or to create km objects otherwise. In both cases, the result is a km object. If parameters are unknown, they are estimated by Maximum Likelihood. As a beta version, Penalized Maximum Likelihood Estimation is also possible if some penalty is given, or Leave-One-Out for noise-free observations.<\/span><\/div>, <div class='tooltip'>km1Nugget.init<span class='tooltiptext'>km1Nugget.init is used to give good initial values to fit kriging models when there is an unknown nugget effect to be estimated.<\/span><\/div>, <div class='tooltip'>kmData<span class='tooltiptext'>kmData is equivalent to km, except for the interface with the data. In kmData, the user must supply both the design and the response within a single data.frame data. To supply them separately, use km.<\/span><\/div>, <div class='tooltip'>kmEstimate<span class='tooltiptext'>kmEstimate is used to fit kriging models. This function should not be called directly, due to the environments defined in km to avoid computing twice nxn matrices. Call km instead.<\/span><\/div>, <div class='tooltip'>kmNoNugget.init<span class='tooltiptext'>kmNoNugget.init is used to give initial values to fit kriging models when there is no nugget effect nor noisy observations.<\/span><\/div>, <div class='tooltip'>kmNuggets.init<span class='tooltiptext'>kmNuggets.init is used to give initial values to fit kriging models, in presence of noisy observations.<\/span><\/div>, <div class='tooltip'>leaveOneOut.km<span class='tooltiptext'>Cross validation by leave-one-out for a km object without noisy observations.<\/span><\/div>, <div class='tooltip'>leaveOneOutFun<span class='tooltiptext'>Returns the mean of the squared leave-one-out errors, computed with Dubrule's formula.<\/span><\/div>, <div class='tooltip'>leaveOneOutGrad<span class='tooltiptext'>Returns the analytical gradient of leaveOneOutFun.<\/span><\/div>, <div class='tooltip'>logLik.km<span class='tooltiptext'>Returns the log-likelihood value of a km object.<\/span><\/div>, <div class='tooltip'>logLikFun<span class='tooltiptext'>Returns the concentrated log-likelihood, obtained from the likelihood by plugging in the estimators of the parameters that can be expressed in function of the other ones.<\/span><\/div>, <div class='tooltip'>logLikGrad<span class='tooltiptext'>Returns the exact gradient vector of the concentrated log-likelihood.<\/span><\/div>, <div class='tooltip'>ninput<span class='tooltiptext'>Get the spatial dimension (number of input variables).<\/span><\/div>, <div class='tooltip'>nuggetflag<span class='tooltiptext'>Get a boolean indicating whether there is a nugget effect.<\/span><\/div>, <div class='tooltip'>nuggetvalue<span class='tooltiptext'>Get or set the nugget value.<\/span><\/div>, <div class='tooltip'>nuggetvalue<-<span class='tooltiptext'>Get or set the nugget value.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'>Three plots are currently available, based on the leaveOneOut.km results: one plot of fitted values against response values, one plot of standardized residuals, and one qqplot of standardized residuals.<\/span><\/div>, <div class='tooltip'>plot.km<span class='tooltiptext'>Three plots are currently available, based on the leaveOneOut.km results: one plot of fitted values against response values, one plot of standardized residuals, and one qqplot of standardized residuals.<\/span><\/div>, <div class='tooltip'>predict<span class='tooltiptext'>Predicted values and (marginal of joint) conditional variances based on a km model. 95 % confidence intervals are given, based on strong assumptions: Gaussian process assumption, specific prior distribution on the trend parameters, known covariance parameters. This might be abusive in particular in the case where estimated covariance parameters are plugged in.<\/span><\/div>, <div class='tooltip'>predict.km<span class='tooltiptext'>Predicted values and (marginal of joint) conditional variances based on a km model. 95 % confidence intervals are given, based on strong assumptions: Gaussian process assumption, specific prior distribution on the trend parameters, known covariance parameters. This might be abusive in particular in the case where estimated covariance parameters are plugged in.<\/span><\/div>, <div class='tooltip'>SCAD<span class='tooltiptext'>Smoothly Clipped Absolute Deviation function.<\/span><\/div>, <div class='tooltip'>SCAD.derivative<span class='tooltiptext'>Derivative of SCAD function.<\/span><\/div>, <div class='tooltip'>scalingFun<span class='tooltiptext'>Parametric transformation of the input space variables. The transformation is obtained coordinatewise by integrating piecewise affine marginal \"densities\" parametrized by a vector of knots and a matrix of density values at the knots. See references for more detail.<\/span><\/div>, <div class='tooltip'>scalingFun1d<span class='tooltiptext'>Parametric transformation of the input space variable. The transformation is obtained coordinatewise by integrating piecewise affine marginal \"density\" parametrized by a vector of knots and a matrix of density values at the knots. See references for more detail.<\/span><\/div>, <div class='tooltip'>scalingGrad<span class='tooltiptext'>Gradient of the Scaling function (marginal in dimension k) of Xiong et al. with respect to eta<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>simulate<span class='tooltiptext'>simulate is used to simulate Gaussian process values at any given set of points for a specified km object.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>trend.deltax<span class='tooltiptext'>Computes the gradient of the vector of trend basis functions f(x)=(f1(x);...;fp(x))<\/span><\/div>, <div class='tooltip'>trendMatrix.update<span class='tooltiptext'>Updates the trend linear model matrix when new data are given.<\/span><\/div>, <div class='tooltip'>update<span class='tooltiptext'>Update a km object when one or many newobservations are added. Many, but not all, fields of thekm object need to be recalculated when new observations are added. It is also possible to modify the k last (existing) observations.<\/span><\/div>, <div class='tooltip'>vect2covparam<span class='tooltiptext'>Extract the covariance parameters from a single vector. Not for direct use.<\/span><\/div>","<div class='tooltip'>Apply.fun<span class='tooltiptext'>Emulate parallel apply on a function, from mclapply. Returns a vector or array or list of values obtained by applying a function to margins of an array or matrix.<\/span><\/div>, <div class='tooltip'>are_in.mesh<span class='tooltiptext'>Checks if some points belong to a given mesh<\/span><\/div>, <div class='tooltip'>contourview<span class='tooltiptext'>Plot a contour view of a kriging or modelPredict model. It is useful for a better understanding of a model behaviour.<\/span><\/div>, <div class='tooltip'>is_in.mesh<span class='tooltiptext'>Checks if some point belongs to a given mesh<\/span><\/div>, <div class='tooltip'>is_in.p<span class='tooltiptext'>Test if points are in a hull<\/span><\/div>, <div class='tooltip'>Memoize.fun<span class='tooltiptext'>Before each call of a function, check that the cache holds the results and returns it if available. Otherwise, compute f and cache the result for next evluations.<\/span><\/div>, <div class='tooltip'>mesh_exsets<span class='tooltiptext'>Search excursion set of nD function, sampled by a mesh<\/span><\/div>, <div class='tooltip'>mesh_roots<span class='tooltiptext'>Multi Dimensional Multiple Roots (Zero) Finding, sampled by a mesh<\/span><\/div>, <div class='tooltip'>min_dist<span class='tooltiptext'>Minimal distance between one point to many points<\/span><\/div>, <div class='tooltip'>plot_mesh<span class='tooltiptext'>Plot a one dimensional mesh<\/span><\/div>, <div class='tooltip'>plot2d_mesh<span class='tooltiptext'>Plot a two dimensional mesh<\/span><\/div>, <div class='tooltip'>plot3d_mesh<span class='tooltiptext'>Plot a three dimensional mesh<\/span><\/div>, <div class='tooltip'>points_in.mesh<span class='tooltiptext'>Extract points of mesh which belong to the mesh triangulation (may not contain all points)<\/span><\/div>, <div class='tooltip'>points_out.mesh<span class='tooltiptext'>Extract points of mesh which do not belong to the mesh triangulation (may not contain all points)<\/span><\/div>, <div class='tooltip'>root<span class='tooltiptext'>Search one root with given precision (on y). Iterate over uniroot as long as necessary.<\/span><\/div>, <div class='tooltip'>roots<span class='tooltiptext'>Search multiple roots of 1D function, sampled/splitted by a (1D) mesh<\/span><\/div>, <div class='tooltip'>sectionview<span class='tooltiptext'>Plot one section view per dimension of a kriging, modelPredict model or function. It is useful for a better understanding of a model behaviour (including uncertainty).<\/span><\/div>, <div class='tooltip'>sectionview3d<span class='tooltiptext'>Plot a 3-D view of a kriging or modelPredict model. It is useful for a better understanding of a model behaviour.<\/span><\/div>, <div class='tooltip'>Vectorize.funD<span class='tooltiptext'>Vectorize a d-dimensional (input) function, in the same way that base::Vectorize for 1-dimensional functions.<\/span><\/div>","<div class='tooltip'>buildf<span class='tooltiptext'>buildf builds a joint probability density or mass function from marginal distributions and a copula.<\/span><\/div>, <div class='tooltip'>Defficiency<span class='tooltiptext'>Defficiency computes the D-, D_s or D_A-efficiency measure for a design with respect to a reference design.<\/span><\/div>, <div class='tooltip'>Deriv2Logf<span class='tooltiptext'>DerivLogf/Deriv2Logf builds a function that evaluates to the first/second derivative of log(f(y, theta, ...)) with respect to theta[[i]]/theta[[i]] and theta[[j]].<\/span><\/div>, <div class='tooltip'>DerivLogf<span class='tooltiptext'>DerivLogf/Deriv2Logf builds a function that evaluates to the first/second derivative of log(f(y, theta, ...)) with respect to theta[[i]]/theta[[i]] and theta[[j]].<\/span><\/div>, <div class='tooltip'>design<span class='tooltiptext'>design creates a custom design object.<\/span><\/div>, <div class='tooltip'>Dsensitivity<span class='tooltiptext'>Dsensitivity builds a sensitivity function for the D-, D_s or D_A-optimality criterion which relies on defaults to speed up evaluation.Wynn for instance requires this behaviour/protocol.<\/span><\/div>, <div class='tooltip'>fisherI<span class='tooltiptext'>fisherI utilizes nint_integrate to evaluate the Fisher information.<\/span><\/div>, <div class='tooltip'>getM<span class='tooltiptext'>getM returns the Fisher information corresponding to a model and a design.<\/span><\/div>, <div class='tooltip'>grow.grid<span class='tooltiptext'>grow.grid creates a data frame like expand.grid.The order of rows is adjusted to represent a growing grid with respect to resolution.<\/span><\/div>, <div class='tooltip'>integrateA<span class='tooltiptext'>integrateA is a tolerance wrapper for integrate.It allows integrate to reach the maximum number of subdivisions.<\/span><\/div>, <div class='tooltip'>nint_expandSpace<span class='tooltiptext'>nint_expandSpace expands a space or list structure of spaces to a list of true subspaces.<\/span><\/div>, <div class='tooltip'>nint_funcDim<span class='tooltiptext'>nint_funcDim defines a functionally dependent dimension.It shall depend solely on the previous dimensions.<\/span><\/div>, <div class='tooltip'>nint_gridDim<span class='tooltiptext'>nint_gridDim is defined by a sequence of values.Together with other grid dimensions it defines a dense grid.<\/span><\/div>, <div class='tooltip'>nint_integrate<span class='tooltiptext'>nint_integrate performs summation and integration of a scalar-valued function over a space or list structure of spaces.<\/span><\/div>, <div class='tooltip'>nint_integrateNCube<span class='tooltiptext'>Interface to the integration over interval dimensions.<\/span><\/div>, <div class='tooltip'>nint_integrateNCube_cubature<span class='tooltiptext'>Interface to the integration over interval dimensions.<\/span><\/div>, <div class='tooltip'>nint_integrateNCube_integrate<span class='tooltiptext'>Interface to the integration over interval dimensions.<\/span><\/div>, <div class='tooltip'>nint_integrateNCube_SparseGrid<span class='tooltiptext'>Interface to the integration over interval dimensions.<\/span><\/div>, <div class='tooltip'>nint_integrateNFunc<span class='tooltiptext'>Inferface to the integration over function dimensions.<\/span><\/div>, <div class='tooltip'>nint_integrateNFunc_recursive<span class='tooltiptext'>Inferface to the integration over function dimensions.<\/span><\/div>, <div class='tooltip'>nint_intvDim<span class='tooltiptext'>nint_intvDim defines a fixed interval.The bounds may be (negative) Inf.<\/span><\/div>, <div class='tooltip'>nint_scatDim<span class='tooltiptext'>nint_scatDim is defined by a sequence of values.Together with other scatter dimensions it defines a sparse grid.<\/span><\/div>, <div class='tooltip'>nint_space<span class='tooltiptext'>nint_space defines an n-dimensional space as a list of dimensions.A space may consist of subspaces.A space without subspaces is called true subspace.<\/span><\/div>, <div class='tooltip'>nint_tanTransform<span class='tooltiptext'>nint_tanTransform creates the transformation g(x) = atan((x - center)/scale) to be used in nint_transform.<\/span><\/div>, <div class='tooltip'>nint_transform<span class='tooltiptext'>nint_transform applies monotonic transformations to an integrand and a space or list structure of spaces.Common use cases include the probability integral transform, the transformation of infinite limits to finite ones and function dimensions to interval dimensions.<\/span><\/div>, <div class='tooltip'>nint_validateSpace<span class='tooltiptext'>nint_validateSpace performs a couple of checks on a space or list structure of spaces to ensure it is properly defined.<\/span><\/div>, <div class='tooltip'>numDeriv2Logf<span class='tooltiptext'>numDerivLogf/numDeriv2Logf builds a function that evaluates to the first/second derivative of log(f(y, theta, ...)) with respect to theta[[i]]/theta[[i]] and theta[[j]].<\/span><\/div>, <div class='tooltip'>numDerivLogf<span class='tooltiptext'>numDerivLogf/numDeriv2Logf builds a function that evaluates to the first/second derivative of log(f(y, theta, ...)) with respect to theta[[i]]/theta[[i]] and theta[[j]].<\/span><\/div>, <div class='tooltip'>param<span class='tooltiptext'>param creates an initial parametric model object.Unlike other model statements this function does not perform any computation.<\/span><\/div>, <div class='tooltip'>reduce<span class='tooltiptext'>reduce drops insignificant points and merges points in a certain neighbourhood.<\/span><\/div>, <div class='tooltip'>rowmatch<span class='tooltiptext'>rowmatch returns a vector of the positions of (first) matches of the rows of its first argument in the rows of its second.<\/span><\/div>, <div class='tooltip'>roworder<span class='tooltiptext'>roworder returns a permutation which rearranges the rows of its first argument into ascending order.<\/span><\/div>, <div class='tooltip'>rowsduplicated<span class='tooltiptext'>rowsduplicated determines which rows of a matrix are duplicates of rows with smaller subscripts, and returns a logical vector indicating which rows are duplicates.<\/span><\/div>, <div class='tooltip'>seq1<span class='tooltiptext'>seq1 is similar to seq, however by is strictly 1 by default and integer(0) is returned if the range is empty.<\/span><\/div>, <div class='tooltip'>wDefficiency<span class='tooltiptext'>wDefficiency computes the weighted D-, D_s or D_A-efficiency measure for a design with respect to a reference design.<\/span><\/div>, <div class='tooltip'>wDsensitivity<span class='tooltiptext'>wDsensitivity builds a sensitivity function for the weighted D-, D_s or D_A-optimality criterion which relies on defaults to speed up evaluation.Wynn for instance requires this behaviour/protocol.<\/span><\/div>, <div class='tooltip'>Wynn<span class='tooltiptext'>Wynn finds an optimal design using a sensitivity function and a Wynn-algorithm.<\/span><\/div>","<div class='tooltip'>add.response<span class='tooltiptext'>This function allows to add numeric response variables to an experimental plan of class design. The responses are added both to the data frame and to its desnum attribute; the response.names element of the design.info attribute is updated - the function is still experimental.<\/span><\/div>, <div class='tooltip'>aov<span class='tooltiptext'>Methods for automatic linear models for data frames of class design<\/span><\/div>, <div class='tooltip'>aov.default<span class='tooltiptext'>Methods for automatic linear models for data frames of class design<\/span><\/div>, <div class='tooltip'>aov.design<span class='tooltiptext'>Methods for automatic linear models for data frames of class design<\/span><\/div>, <div class='tooltip'>change.contr<span class='tooltiptext'>The function allows to switch between qualitative and quantitative factors and different contrast settings.<\/span><\/div>, <div class='tooltip'>CME.EM08<span class='tooltiptext'>Generic function and methods for creating half normal effects plots<\/span><\/div>, <div class='tooltip'>CME.LW98<span class='tooltiptext'>Generic function and methods for creating half normal effects plots<\/span><\/div>, <div class='tooltip'>col.remove<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>contr.FrF2<span class='tooltiptext'>Contrasts for orthogonal Fractional Factorial 2-level designs<\/span><\/div>, <div class='tooltip'>contr.XuWu<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>contr.XuWuPoly<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>corrPlot<span class='tooltiptext'>Function corrplot plots absolute or squared values of correlations between model matrix columns of main effects up to three-factor interactions for factorial designs.<\/span><\/div>, <div class='tooltip'>cross.design<span class='tooltiptext'>This function generates cartesian products of two or more experimental designs.<\/span><\/div>, <div class='tooltip'>data2design<span class='tooltiptext'>function to convert matrix, data frame or object of class planordesign to class design (allowing use of convenience functions, particularly plotting with mosaic plots)<\/span><\/div>, <div class='tooltip'>des.recode<span class='tooltiptext'>Utility functions for DoE packages, not intended for direct use<\/span><\/div>, <div class='tooltip'>design.info<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>design.info<-<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>desnum<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>desnum<-<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>expansive.replace<span class='tooltiptext'>Expansive replacement for two orthogonal arrays<\/span><\/div>, <div class='tooltip'>export.design<span class='tooltiptext'>Function for exporting a design object<\/span><\/div>, <div class='tooltip'>fac.design<span class='tooltiptext'>Function for creating full factorial designs with arbitrary numbers of levels, and potentially with blocking<\/span><\/div>, <div class='tooltip'>factor.names<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>factor.names<-<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>generators<span class='tooltiptext'>Utility functions for DoE packages, not intended for direct use<\/span><\/div>, <div class='tooltip'>getblock<span class='tooltiptext'>Function getblock creates block factors for designs with replications, repeated measurements or split plot designs. Function rerandomize.design rerandomizes an experimental design.<\/span><\/div>, <div class='tooltip'>GR<span class='tooltiptext'>Function GR calculates generalized resolution, function GRind calculatesmore detailed generalized resolution values, squared canonical correlationsand average R-squared values, the print method for class GRind appropriately prints thedetailed GRind values. Function SCFTs calculates squared canonical correlations forfactorial designs. SCFTs includes more projections than GRind (all full resolutionprojections or even all projections) and decides on regularity of the design,based on a conjecture.<\/span><\/div>, <div class='tooltip'>GRind<span class='tooltiptext'>Function GR calculates generalized resolution, function GRind calculatesmore detailed generalized resolution values, squared canonical correlationsand average R-squared values, the print method for class GRind appropriately prints thedetailed GRind values. Function SCFTs calculates squared canonical correlations forfactorial designs. SCFTs includes more projections than GRind (all full resolutionprojections or even all projections) and decides on regularity of the design,based on a conjecture.<\/span><\/div>, <div class='tooltip'>GWLP<span class='tooltiptext'>Functions needed for fast calculation of GWLP using the formulae from Xu and Wu (2001)<\/span><\/div>, <div class='tooltip'>halfnormal<span class='tooltiptext'>Generic function and methods for creating half normal effects plots<\/span><\/div>, <div class='tooltip'>ICFT<span class='tooltiptext'>Function ICFTs calculates interaction contribution frequency tables, function ICFT does the same for an entire (usually small) design with more detail.<\/span><\/div>, <div class='tooltip'>ICFTs<span class='tooltiptext'>Function ICFTs calculates interaction contribution frequency tables, function ICFT does the same for an entire (usually small) design with more detail.<\/span><\/div>, <div class='tooltip'>iscube<span class='tooltiptext'>These functions identify the positions for cube points or star points and can reduce a central composite design to its cube portion (with center points).<\/span><\/div>, <div class='tooltip'>isstar<span class='tooltiptext'>These functions identify the positions for cube points or star points and can reduce a central composite design to its cube portion (with center points).<\/span><\/div>, <div class='tooltip'>length2<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>length3<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>length4<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>length5<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>lengths<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>lm<span class='tooltiptext'>Methods for automatic linear models for data frames of class design<\/span><\/div>, <div class='tooltip'>lm.default<span class='tooltiptext'>Methods for automatic linear models for data frames of class design<\/span><\/div>, <div class='tooltip'>lm.design<span class='tooltiptext'>Methods for automatic linear models for data frames of class design<\/span><\/div>, <div class='tooltip'>lowerbound_AR<span class='tooltiptext'>The functions serve the calculation of lower bounds for the worst case confounding. lowerbound_AR is intended for direct use, lowerbounds and lowerbound_chi2 are internal functions.<\/span><\/div>, <div class='tooltip'>make.formulas<span class='tooltiptext'>Utility functions for DoE packages, not intended for direct use<\/span><\/div>, <div class='tooltip'>make.generators<span class='tooltiptext'>Utility functions for DoE packages, not intended for direct use<\/span><\/div>, <div class='tooltip'>ME.Lenth<span class='tooltiptext'>Generic function and methods for creating half normal effects plots<\/span><\/div>, <div class='tooltip'>oa_feasible<span class='tooltiptext'>The function checks necessary conditions for the existence of an array of specified strength<\/span><\/div>, <div class='tooltip'>oa.design<span class='tooltiptext'>Function for accessing orthogonal arrays, allowing limited optimal allocation of columns<\/span><\/div>, <div class='tooltip'>oa.max3<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>oa.max4<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>oa.maxGR<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>oa.min3<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>oa.min34<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>oa.min4<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>oa.minRelProjAberr<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>ord<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>origin<span class='tooltiptext'>Function for accessing orthogonal arrays, allowing limited optimal allocation of columns<\/span><\/div>, <div class='tooltip'>P2.2<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>P3.3<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>P4.4<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>param.design<span class='tooltiptext'>The functions create parameter designs for robustness experiments and signal-to-noise investigations with inner and outer arrays and facilitate their formatting and data aggregation.<\/span><\/div>, <div class='tooltip'>paramtowide<span class='tooltiptext'>The functions create parameter designs for robustness experiments and signal-to-noise investigations with inner and outer arrays and facilitate their formatting and data aggregation.<\/span><\/div>, <div class='tooltip'>pickcube<span class='tooltiptext'>These functions identify the positions for cube points or star points and can reduce a central composite design to its cube portion (with center points).<\/span><\/div>, <div class='tooltip'>planor2design<span class='tooltiptext'>function to convert matrix, data frame or object of class planordesign to class design (allowing use of convenience functions, particularly plotting with mosaic plots)<\/span><\/div>, <div class='tooltip'>plot.design<span class='tooltiptext'>The plot method for class design objects; other methods are part of a separate help page.<\/span><\/div>, <div class='tooltip'>qua.design<span class='tooltiptext'>The function allows to switch between qualitative and quantitative factors and different contrast settings.<\/span><\/div>, <div class='tooltip'>redesign<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>reptolong<span class='tooltiptext'>Convenience functions to reshape a design with repeated measurements from long to wide or vice versa<\/span><\/div>, <div class='tooltip'>reptowide<span class='tooltiptext'>Convenience functions to reshape a design with repeated measurements from long to wide or vice versa<\/span><\/div>, <div class='tooltip'>rerandomize.design<span class='tooltiptext'>Function getblock creates block factors for designs with replications, repeated measurements or split plot designs. Function rerandomize.design rerandomizes an experimental design.<\/span><\/div>, <div class='tooltip'>response.names<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>response.names<-<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>run.order<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>run.order<-<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>, <div class='tooltip'>SCFTs<span class='tooltiptext'>Function GR calculates generalized resolution, function GRind calculatesmore detailed generalized resolution values, squared canonical correlationsand average R-squared values, the print method for class GRind appropriately prints thedetailed GRind values. Function SCFTs calculates squared canonical correlations forfactorial designs. SCFTs includes more projections than GRind (all full resolutionprojections or even all projections) and decides on regularity of the design,based on a conjecture.<\/span><\/div>, <div class='tooltip'>show.oas<span class='tooltiptext'>This function allows to inspect the list of available orthogonal arrays, optionally specifying selection criteria<\/span><\/div>, <div class='tooltip'>SN<span class='tooltiptext'>Function for the signal-to-noise ratio 10 * log10(mean^2/var)<\/span><\/div>, <div class='tooltip'>tupleSel<span class='tooltiptext'>Functions length2, length3, length4 and length5 calculate the numbers of generalized wordsof lengths 2, 3, 4, and 5 respectively, lengths calculates them all. Functions P3.3 and P4.4calculate projection frequency tables, functions oa.min3, oa.min4, oa.min34, oa.maxGR (deprecated), oa.minRelProjAberr,oa.max3 and oa.max4 determine column allocations with minimum or maximum aliasing.Function nchoosek is an auxiliary functionfor calculating all subsets without replacement.<\/span><\/div>, <div class='tooltip'>undesign<span class='tooltiptext'>Convenience functions to quickly access and modify attributes of data frames of the class design; methods for the class are described in a separate help topic<\/span><\/div>","<div class='tooltip'>contr.XuWu<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>gurobi_MIParray<span class='tooltiptext'>The functions create an array with specified minimum resolution and optimized word length pattern based on mixed integer programming with the commercial software Gurobi (free academic license available) or Mosek (free academic license available).Creation is done from scratch, or using a user-specified starting value, or extending an existing array. Important: Installation of Gurobi and/of Mosek as well as the corresponding R packages is necessary. The R package gurobi comes with the software, the current version of the R package Rmosek has to be obtained from vendor's website (CRAN version is outdated!).<\/span><\/div>, <div class='tooltip'>gurobi_MIPcontinue<span class='tooltiptext'>These functions continue optimization for a MIP-based array from a stored state.<\/span><\/div>, <div class='tooltip'>gurobi_MIPsearch<span class='tooltiptext'>The functions search through different orderings of the nlevels vector with the goal to create an array with minimum resolution and optimized shortest word length. They create the orders and call gurobi_MIParray or mosek_MIParray for each order.<\/span><\/div>, <div class='tooltip'>GWLP<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>ICFTs<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>length2<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>length3<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>length4<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>length5<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>lowerbound_AR<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>mosek_MIParray<span class='tooltiptext'>The functions create an array with specified minimum resolution and optimized word length pattern based on mixed integer programming with the commercial software Gurobi (free academic license available) or Mosek (free academic license available).Creation is done from scratch, or using a user-specified starting value, or extending an existing array. Important: Installation of Gurobi and/of Mosek as well as the corresponding R packages is necessary. The R package gurobi comes with the software, the current version of the R package Rmosek has to be obtained from vendor's website (CRAN version is outdated!).<\/span><\/div>, <div class='tooltip'>mosek_MIPcontinue<span class='tooltiptext'>These functions continue optimization for a MIP-based array from a stored state.<\/span><\/div>, <div class='tooltip'>mosek_MIPsearch<span class='tooltiptext'>The functions search through different orderings of the nlevels vector with the goal to create an array with minimum resolution and optimized shortest word length. They create the orders and call gurobi_MIParray or mosek_MIParray for each order.<\/span><\/div>, <div class='tooltip'>oa_feasible<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>, <div class='tooltip'>print.oa<span class='tooltiptext'>The function suppresses printing of voluminous info attached as attributes to oa objects.<\/span><\/div>, <div class='tooltip'>SCFTs<span class='tooltiptext'>These functions from DoE.base are exported from DoE.MIParray, because they are especially important for its use.<\/span><\/div>","<div class='tooltip'>ufactors<span class='tooltiptext'>This function generates the unique factors for a unique factor CCD.<\/span><\/div>, <div class='tooltip'>ufccd<span class='tooltiptext'>This function generates the design matrix for a unique factor CCD.<\/span><\/div>","<div class='tooltip'>bbd.design<span class='tooltiptext'>Function for generating Box-Behnken designs, making use of package rsm<\/span><\/div>, <div class='tooltip'>ccd.augment<span class='tooltiptext'>Function for augmenting an existing fractional factorial with a star portion in case of a late decision for a sequential procedure.<\/span><\/div>, <div class='tooltip'>ccd.design<span class='tooltiptext'>Function for accessing central composite designs from package rsm, with automatic creation of an appropriate cube portion<\/span><\/div>, <div class='tooltip'>code.design<span class='tooltiptext'>These functions support response surface analysis with package rsm. Function rsmformula creates a model formula for use with function rsm, using the FO, TWI and PQ notation. Function code.design prepares a fractional factorial 2-level design with center points from package FrF2 or a ccd, bbd or lhs design from this package for convenient use with package rsm functionality, function decode.design reverses the coding action.<\/span><\/div>, <div class='tooltip'>compare<span class='tooltiptext'>One function for calculating the S-optimality criterion is provided here. This help file documents this function and also describes optimality criteria from other related packages, referring to space filling optimality like the S criterion or to model-based optimality like the well-known D-criterion.<\/span><\/div>, <div class='tooltip'>decode.design<span class='tooltiptext'>These functions support response surface analysis with package rsm. Function rsmformula creates a model formula for use with function rsm, using the FO, TWI and PQ notation. Function code.design prepares a fractional factorial 2-level design with center points from package FrF2 or a ccd, bbd or lhs design from this package for convenient use with package rsm functionality, function decode.design reverses the coding action.<\/span><\/div>, <div class='tooltip'>Dopt.augment<span class='tooltiptext'>Function for comfortably augmenting a design with D-optimal additional points; this functionality is still somewhat experimental.<\/span><\/div>, <div class='tooltip'>Dopt.design<span class='tooltiptext'>Function for comfortably creating a D-optimal design with or without blocking based on functions optFederov or optBlock from package AlgDesign; this functionality is still somewhat experimental.<\/span><\/div>, <div class='tooltip'>lhs.augment<span class='tooltiptext'>Functions for comfortably accessing latin hypercube sampling designs from package lhs or space-filling designs from package DiceDesign, which are useful for quantitative factors with many possible levels. In particular, they can be used in computer experiments. Most of the designs are random samples.<\/span><\/div>, <div class='tooltip'>lhs.design<span class='tooltiptext'>Functions for comfortably accessing latin hypercube sampling designs from package lhs or space-filling designs from package DiceDesign, which are useful for quantitative factors with many possible levels. In particular, they can be used in computer experiments. Most of the designs are random samples.<\/span><\/div>, <div class='tooltip'>rsmformula<span class='tooltiptext'>These functions support response surface analysis with package rsm. Function rsmformula creates a model formula for use with function rsm, using the FO, TWI and PQ notation. Function code.design prepares a fractional factorial 2-level design with center points from package FrF2 or a ccd, bbd or lhs design from this package for convenient use with package rsm functionality, function decode.design reverses the coding action.<\/span><\/div>, <div class='tooltip'>Scalc<span class='tooltiptext'>One function for calculating the S-optimality criterion is provided here. This help file documents this function and also describes optimality criteria from other related packages, referring to space filling optimality like the S criterion or to model-based optimality like the well-known D-criterion.<\/span><\/div>","<div class='tooltip'>run_dosedesignR<span class='tooltiptext'>Starts the dosedesignR application in the client's browser.<\/span><\/div>","<div class='tooltip'>betaMod<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>betaModGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>bFitMod<span class='tooltiptext'>For type = \"Bayes\", MCMC sampling from the posterior distribution of the dose-response model is done. The function assumes a multivariate normal distribution for resp with covariance matrix S, and this is taken as likelihood function and combined with the prior distributions specified in prior to form the posterior distribution. For type = \"bootstrap\", a multivariate normal distribution for resp with covariance matrix S is assumed, and a large number of samples is drawn from this distribution. For each draw the fitMod function with type = \"general\" is used to fit the draws from the multivariate normal distribution.<\/span><\/div>, <div class='tooltip'>calcCrit<span class='tooltiptext'>Given a set of models (with full parameter values and model probabilities) the optDesign function calculates the optimal design for estimating the dose-response model parameters (D-optimal) or the design for estimating the target dose (TD-optimal design) (see Dette, Bretz, Pepelyshev and Pinheiro (2008)), or a mixture of these two criteria. The design can be plotted (together with the candidate models) using plot.design. calcCrit calculates the design criterion for a discrete set of design(s). rndDesign provides efficient rounding for the calculated continous design to a finite sample size.<\/span><\/div>, <div class='tooltip'>defBnds<span class='tooltiptext'>Calculates reasonable bounds for non-linear parameters for the built-in non-linear regression model based on the dose range under investigation. For the logistic model the first row corresponds to the ED50 parameter and the second row to the delta parameter. For the sigmoid Emax model the first row corresponds to the ED50 parameter and the second row to the h parameter, while for the beta model first and second row correspond to the delta1 and delta2 parameters. See logistic, sigEmax and betaMod for details.<\/span><\/div>, <div class='tooltip'>ED<span class='tooltiptext'>The TD (target dose) is defined as the dose that achieves a target effect of Delta over placebo (if there are multiple such doses, the smallest is chosen): TD_\\Delta = \\min \\{x|f(x) > f(0)+\\Delta\\}TD = min {x|f(x) > f(0)+Delta} If a decreasing trend is beneficial the definition of the TD is TD_\\Delta = \\min \\{x|f(x) < f(0)-\\Delta\\}TD = min {x|f(x) < f(0)-Delta} When \\DeltaDelta is the clinical relevance threshold, then the TD is similar to the usual definition of the minimum effective dose (MED). The ED (effective dose) is defined as the dose that achieves a certain percentage p of the full effect size (within the observed dose-range!) over placebo (if there are multiple such doses, the smallest is chosen). ED_p=\\min\\{x|f(x) > f(0) + p(f(dmax)-f(0)) EDp=min {x|f(x) > f(0) + p(f(dmax)-f(0))} Note that this definition of the EDp is different from traditional definition based on the Emax model, where the EDp is defined relative to the asymptotic maximum effect (rather than the maximum effect in the observed dose-range).<\/span><\/div>, <div class='tooltip'>emax<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>emaxGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>exponential<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>exponentialGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>fitMod<span class='tooltiptext'>Fits a dose-response model. Built-in dose-response models are \"linlog\", \"linear\", \"quadratic\", \"emax\", \"exponential\", \"sigEmax\", \"betaMod\" and \"logistic\" (see drmodels). When type = \"normal\" ordinary least squares is used and additional additive covariates can be specified in addCovars. The underlying assumption is hence normally distributed data and homoscedastic variance. For type = \"general\" a generalized least squares criterion is used (f(dose,\\theta)-resp)'S^{-1}(f(dose,\\theta)-resp)(f(dose,theta)-resp)'S^{-1}(f(dose,theta)-resp) and an inverse weighting matrix is specified in S, type = \"general\" is primarily of interest, when fitting a model to AN(C)OVA type estimates obtained in a first stage fit, then resp contains the estimates and S is the estimated covariance matrix for the estimates in resp. Statistical inference (e.g. confidence intervals) rely on asymptotic normality of the first stage estimates, which makes this method of interest only for sufficiently large sample size for the first stage fit. A modified model-selection criterion can be applied to these model fits (see also Pinheiro et al. 2013 for details). For details on the implemented numerical optimizer see the Details section below.<\/span><\/div>, <div class='tooltip'>gAIC<span class='tooltiptext'>Fits a dose-response model. Built-in dose-response models are \"linlog\", \"linear\", \"quadratic\", \"emax\", \"exponential\", \"sigEmax\", \"betaMod\" and \"logistic\" (see drmodels). When type = \"normal\" ordinary least squares is used and additional additive covariates can be specified in addCovars. The underlying assumption is hence normally distributed data and homoscedastic variance. For type = \"general\" a generalized least squares criterion is used (f(dose,\\theta)-resp)'S^{-1}(f(dose,\\theta)-resp)(f(dose,theta)-resp)'S^{-1}(f(dose,theta)-resp) and an inverse weighting matrix is specified in S, type = \"general\" is primarily of interest, when fitting a model to AN(C)OVA type estimates obtained in a first stage fit, then resp contains the estimates and S is the estimated covariance matrix for the estimates in resp. Statistical inference (e.g. confidence intervals) rely on asymptotic normality of the first stage estimates, which makes this method of interest only for sufficiently large sample size for the first stage fit. A modified model-selection criterion can be applied to these model fits (see also Pinheiro et al. 2013 for details). For details on the implemented numerical optimizer see the Details section below.<\/span><\/div>, <div class='tooltip'>getResp<span class='tooltiptext'>The Mods functions allows to define a set of dose-response models. The function is used as input object for a number of other different functions. The dose-response models used in this package (see drmodels for details) are of form f(d) = \\theta_0+\\theta_1 f^0(d,\\theta_2)f(d) = theta0+theta1 f0(d,theta2) where the parameter \\theta_2theta2 is the only non-linear parameter and can be one- or two-dimensional, depending on the used model. One needs to hand over the effect at placebo and the maximum effect in the dose range, from which \\theta_0,\\theta_1theta0,theta1 are then back-calculated, the output object is of class \"Mods\". This object can form the input for other functions to extract the mean response (getResp) or target doses (TD and ED) corresponding to the models. It is also needed as input to the functions powMCT, optDesign Some models, for example the beta model (scal) and the linlog model (off) have parameters that are not estimated by the code, they need to be specified via the addArgs argument. NOTE: If a decreasing effect is beneficial for the considered response variable it needs to specified here, either by using direction = \"decreasing\" or by specifying a negative \"maxEff\" argument.<\/span><\/div>, <div class='tooltip'>guesst<span class='tooltiptext'>Calculates guesstimates for standardized model parameter(s) using the general approach described in Pinheiro et al. (2006).<\/span><\/div>, <div class='tooltip'>linear<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>linearGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>linInt<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>linIntGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>linlog<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>linlogGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>logistic<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>logisticGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>MCPMod<span class='tooltiptext'>Tests for a dose-response effect using a model-based multiple contrast test (see MCTtest), selects one (or several) model(s) from the significant shapes, fits them using fitMod. For details on the method see Bretz et al. (2005).<\/span><\/div>, <div class='tooltip'>MCTtest<span class='tooltiptext'>This function performs a multiple contrast test. The contrasts are either directly specified in contMat or optimal contrasts derived from the models argument. The directionality of the data (i.e. whether an increase or decrease in the response variable is beneficial is inferred from the models object, see Mods). For type = \"normal\" an ANCOVA model based on a homoscedastic normality assumption (with additive covariates specified in addCovars) is fitted. For type = \"general\" it is assumed multivariate normally distributed estimates are specified in resp with covariance given by S, and the contrast test statistic is calculated based on this assumption. Degrees of freedom specified in df.<\/span><\/div>, <div class='tooltip'>Mods<span class='tooltiptext'>The Mods functions allows to define a set of dose-response models. The function is used as input object for a number of other different functions. The dose-response models used in this package (see drmodels for details) are of form f(d) = \\theta_0+\\theta_1 f^0(d,\\theta_2)f(d) = theta0+theta1 f0(d,theta2) where the parameter \\theta_2theta2 is the only non-linear parameter and can be one- or two-dimensional, depending on the used model. One needs to hand over the effect at placebo and the maximum effect in the dose range, from which \\theta_0,\\theta_1theta0,theta1 are then back-calculated, the output object is of class \"Mods\". This object can form the input for other functions to extract the mean response (getResp) or target doses (TD and ED) corresponding to the models. It is also needed as input to the functions powMCT, optDesign Some models, for example the beta model (scal) and the linlog model (off) have parameters that are not estimated by the code, they need to be specified via the addArgs argument. NOTE: If a decreasing effect is beneficial for the considered response variable it needs to specified here, either by using direction = \"decreasing\" or by specifying a negative \"maxEff\" argument.<\/span><\/div>, <div class='tooltip'>mvtnorm.control<span class='tooltiptext'>Returns a list (an object of class \"GenzBretz\") with control parameters for the pmvt and qmvt functions from the mvtnorm package. Note that the DoseFinding package always uses \"GenzBretz\" algorithm. See the mvtnorm documentation for more information.<\/span><\/div>, <div class='tooltip'>optContr<span class='tooltiptext'>This function calculates a contrast vectors that are optimal for detecting certain alternatives. The contrast is optimal in the sense of maximizing the non-centrality parameter of the underlying contrast test statistic: \\frac{c'\\mu}{\\sqrt{c'Sc}}c'mu/sqrt(c'Sc). Here \\mumu is the mean vector under the alternative and SS the covariance matrix associated with the estimate of \\mumu. The optimal contrast is given by c^{opt} \\propto S^{-1}\\left(\\mu - \\frac{\\mu^{\\prime}S^{-1} 1}{1^\\prime S^{-1} 1}\\right),c propto S^(-1) (mu - mu'S^(-1)1)/(1'S^(-1)1), see Pinheiro et al. (2013). Note that the directionality (i.e. whether in \"increase\" in the response variable is beneficial or a \"decrease\", is inferred from the specified models object, see Mods for details). Constrained contrasts (type = \"constrained\") add the additional constraint in the optimization that the sign of the contrast coefficient for control and active treatments need to be different. The quadratic programming algorithm from the quadprog package is used to calculate the contrasts.<\/span><\/div>, <div class='tooltip'>optDesign<span class='tooltiptext'>Given a set of models (with full parameter values and model probabilities) the optDesign function calculates the optimal design for estimating the dose-response model parameters (D-optimal) or the design for estimating the target dose (TD-optimal design) (see Dette, Bretz, Pepelyshev and Pinheiro (2008)), or a mixture of these two criteria. The design can be plotted (together with the candidate models) using plot.design. calcCrit calculates the design criterion for a discrete set of design(s). rndDesign provides efficient rounding for the calculated continous design to a finite sample size.<\/span><\/div>, <div class='tooltip'>planMod<span class='tooltiptext'>This function evaluates, the performance metrics for fitting dose-response models (using asymptotic approximations or simulations). Note that some metrics are available via the print method and others only via the summary method applied to planMod objects. The implemented metrics are Root of the mean-squared error to estimate the placebo-adjusted dose-response averaged over the used dose-levels, i.e. a rather discrete set (dRMSE). Available via the print method of planMod objects. Root of the mean-squared error to estimate the placebo-adjusted dose-response (cRMSE) averaged over fine (almost continuous) grid at 101 equally spaced values between placebo and the maximum dose. NOTE: Available via the summary method applied to planMod objects. Ratio of the placebo-adjusted mean-squared error (at the observed doses) of model-based vs ANOVA approach (Eff-vs-ANOVA). This can be interpreted on the sample size scale. NOTE: Available via the summary method applied to planMod objects. Power that the (unadjusted) one-sided 1-alpha confidence interval comparing the dose with maximum effect vs placebo is larger than tau. By default alpha = 0.025 and tau = 0 (Pow(maxDose)). Available via the print method of planMod objects. Probability that the EDp estimate is within the true [EDpLB, EDpUB] (by default p=0.5, pLB=0.25 and pUB=0.75). This metric gives an idea on the ability to characterize the increasing part of the dose-response curve (P(EDp)). Available via the print method of planMod objects. Length of the quantile range for a target dose (TD or EDp). This is calculated by taking the difference of the dUB and dLB quantile of the empirical distribution of the dose estimates. (lengthTDCI and lengthEDpCI). It is NOT calculated by calculating confidence interval lengths in each simulated data-set and taking the mean. NOTE: Available via the summary method of planMod objects. A plot method exists to summarize dose-response and dose estimations graphically.<\/span><\/div>, <div class='tooltip'>powMCT<span class='tooltiptext'>Calculate power for a multiple contrast test for a set of specified alternatives.<\/span><\/div>, <div class='tooltip'>powN<span class='tooltiptext'>The sampSize function implements a bisection search algorithm for sample size calculation. The user can hand over a general target function (via targFunc) that is then iterated so that a certain target is achieved. The sampSizeMCT is a convenience wrapper of sampSize for multiple contrast tests using the power as target function. The targN functions calculates a general target function for different given sample sizes. The powN function is a convenience wrapper of targN for multiple contrast tests using the power as target function.<\/span><\/div>, <div class='tooltip'>quadratic<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>quadraticGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>rndDesign<span class='tooltiptext'>Given a set of models (with full parameter values and model probabilities) the optDesign function calculates the optimal design for estimating the dose-response model parameters (D-optimal) or the design for estimating the target dose (TD-optimal design) (see Dette, Bretz, Pepelyshev and Pinheiro (2008)), or a mixture of these two criteria. The design can be plotted (together with the candidate models) using plot.design. calcCrit calculates the design criterion for a discrete set of design(s). rndDesign provides efficient rounding for the calculated continous design to a finite sample size.<\/span><\/div>, <div class='tooltip'>sampSize<span class='tooltiptext'>The sampSize function implements a bisection search algorithm for sample size calculation. The user can hand over a general target function (via targFunc) that is then iterated so that a certain target is achieved. The sampSizeMCT is a convenience wrapper of sampSize for multiple contrast tests using the power as target function. The targN functions calculates a general target function for different given sample sizes. The powN function is a convenience wrapper of targN for multiple contrast tests using the power as target function.<\/span><\/div>, <div class='tooltip'>sampSizeMCT<span class='tooltiptext'>The sampSize function implements a bisection search algorithm for sample size calculation. The user can hand over a general target function (via targFunc) that is then iterated so that a certain target is achieved. The sampSizeMCT is a convenience wrapper of sampSize for multiple contrast tests using the power as target function. The targN functions calculates a general target function for different given sample sizes. The powN function is a convenience wrapper of targN for multiple contrast tests using the power as target function.<\/span><\/div>, <div class='tooltip'>sigEmax<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>sigEmaxGrad<span class='tooltiptext'>Dose-response model functions and gradients. Below are the definitions of the model functions: Emax model f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d). Sigmoid Emax Model f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h). Exponential Model f(d,\\theta)=E_0+E_1(\\exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1). Beta model f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 here B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2). and scalscal is a fixed dose scaling parameter. Linear Model f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d. Linear in log Model f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), here offoff is a fixed offset parameter. Logistic Model f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)). Quadratic Model f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.Linear Interpolation model The linInt model provides linear interpolation at the values defined by the nodes vector. In virtually all situations the nodes vector is equal to the doses used in the analysis. For example the Mods and the fitMod function automatically use the doses that are used in the context of the function call as nodes. The guesstimates specified in the Mods function need to be the treatment effects at the active doses standardized to the interval [0,1] (see the examples in the Mods function).<\/span><\/div>, <div class='tooltip'>targN<span class='tooltiptext'>The sampSize function implements a bisection search algorithm for sample size calculation. The user can hand over a general target function (via targFunc) that is then iterated so that a certain target is achieved. The sampSizeMCT is a convenience wrapper of sampSize for multiple contrast tests using the power as target function. The targN functions calculates a general target function for different given sample sizes. The powN function is a convenience wrapper of targN for multiple contrast tests using the power as target function.<\/span><\/div>, <div class='tooltip'>TD<span class='tooltiptext'>The TD (target dose) is defined as the dose that achieves a target effect of Delta over placebo (if there are multiple such doses, the smallest is chosen): TD_\\Delta = \\min \\{x|f(x) > f(0)+\\Delta\\}TD = min {x|f(x) > f(0)+Delta} If a decreasing trend is beneficial the definition of the TD is TD_\\Delta = \\min \\{x|f(x) < f(0)-\\Delta\\}TD = min {x|f(x) < f(0)-Delta} When \\DeltaDelta is the clinical relevance threshold, then the TD is similar to the usual definition of the minimum effective dose (MED). The ED (effective dose) is defined as the dose that achieves a certain percentage p of the full effect size (within the observed dose-range!) over placebo (if there are multiple such doses, the smallest is chosen). ED_p=\\min\\{x|f(x) > f(0) + p(f(dmax)-f(0)) EDp=min {x|f(x) > f(0) + p(f(dmax)-f(0))} Note that this definition of the EDp is different from traditional definition based on the Emax model, where the EDp is defined relative to the asymptotic maximum effect (rather than the maximum effect in the observed dose-range).<\/span><\/div>","<div class='tooltip'><span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>caceCRTBoot<span class='tooltiptext'>These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.<\/span><\/div>, <div class='tooltip'>caceMSTBoot<span class='tooltiptext'>These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.<\/span><\/div>, <div class='tooltip'>caceSRTBoot<span class='tooltiptext'>These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.<\/span><\/div>, <div class='tooltip'>ComparePlot<span class='tooltiptext'>It generates bar plot that compares the effect size from eefAnalytics' methods.<\/span><\/div>, <div class='tooltip'>crtBayes<span class='tooltiptext'>crtBayes performs analysis of cluster randomised education trials using a multilevel model under a Bayesian setting,assuming vague priors.<\/span><\/div>, <div class='tooltip'>crtFREQ<span class='tooltiptext'>crtFREQ performs analysis of cluster randomised education trials using a multilevel model under a frequentist setting.<\/span><\/div>, <div class='tooltip'>mlmbayes<span class='tooltiptext'>These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.These functions are marked as defunct and have been removed from eefAnalytics.<\/span><\/div>, <div class='tooltip'>mstBayes<span class='tooltiptext'>mstBayes performs analysis of multisite randomised education trials using a multilevel model under a Bayesian settingassuming vague priors.<\/span><\/div>, <div class='tooltip'>mstFREQ<span class='tooltiptext'>mstFREQ performs analysis of multisite randomised education trials using a multilevel model under a frequentist setting.<\/span><\/div>, <div class='tooltip'>srtBayes<span class='tooltiptext'>srtBayes performs analysis of educational trials under the assumption of independent errors among pupils using Bayesian framework with Stan.This can also be used with schools as fixed effects.<\/span><\/div>, <div class='tooltip'>srtFREQ<span class='tooltiptext'>srtFREQ performs analysis of educational trials under the assumption of independent errors among pupils.This can also be used with schools as fixed effects.<\/span><\/div>","<div class='tooltip'><span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>commarobust<span class='tooltiptext'>Build lm_robust object from lm fit<\/span><\/div>, <div class='tooltip'>declaration_to_condition_pr_mat<span class='tooltiptext'>Builds condition probability matrices for Horvitz-Thompson estimation fromrandomizr declaration<\/span><\/div>, <div class='tooltip'>difference_in_means<span class='tooltiptext'>Difference-in-means estimators that selects the appropriatepoint estimate, standard errors, and degrees of freedom for a variety ofdesigns: unit randomized, cluster randomized, block randomized,block-cluster randomized, matched-pairs, and matched-pair clusterrandomized designs<\/span><\/div>, <div class='tooltip'>extract.iv_robust<span class='tooltiptext'>Prepares a \"lm_robust\" or \"iv_robust\" object for the texregpackage. This is largely a clone of the extract.lmmethod.<\/span><\/div>, <div class='tooltip'>extract.lm_robust<span class='tooltiptext'>Prepares a \"lm_robust\" or \"iv_robust\" object for the texregpackage. This is largely a clone of the extract.lmmethod.<\/span><\/div>, <div class='tooltip'>gen_pr_matrix_cluster<span class='tooltiptext'>Generate condition probability matrix given clusters and probabilities<\/span><\/div>, <div class='tooltip'>glance<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsglance, tidy<\/span><\/div>, <div class='tooltip'>horvitz_thompson<span class='tooltiptext'>Horvitz-Thompson estimators that are unbiased for designs inwhich the randomization scheme is known<\/span><\/div>, <div class='tooltip'>iv_robust<span class='tooltiptext'>This formula estimates an instrumental variables regressionusing two-stage least squares with a variety of options for robuststandard errors<\/span><\/div>, <div class='tooltip'>lh_robust<span class='tooltiptext'>This function fits a linear model with robust standard errors and performs linear hypothesis test.<\/span><\/div>, <div class='tooltip'>lm_lin<span class='tooltiptext'>This function is a wrapper for lm_robust thatis useful for estimating treatment effects with pre-treatment covariatedata. This implements the method described by Lin (2013).<\/span><\/div>, <div class='tooltip'>lm_robust<span class='tooltiptext'>This formula fits a linear model, provides a variety ofoptions for robust standard errors, and conducts coefficient tests<\/span><\/div>, <div class='tooltip'>lm_robust_fit<span class='tooltiptext'>Internal method that creates linear fits<\/span><\/div>, <div class='tooltip'>permutations_to_condition_pr_mat<span class='tooltiptext'>Builds condition probability matrices for Horvitz-Thompson estimation frompermutation matrix<\/span><\/div>, <div class='tooltip'>starprep<span class='tooltiptext'>Prepare model fits for stargazer<\/span><\/div>, <div class='tooltip'>tidy<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsglance, tidy<\/span><\/div>","<div class='tooltip'>anscombetukey<span class='tooltiptext'>anscombetukey Performs the test for homogeneity of variances of Anscombe and Tukey (1963).<\/span><\/div>, <div class='tooltip'>bartlett<span class='tooltiptext'>bartlett Performs the test for homogeneity ofvariances of Bartlett (1937).<\/span><\/div>, <div class='tooltip'>ccboot<span class='tooltiptext'>ccboot Performs the Ramos and Ferreira (2009)multiple comparison bootstrap test.<\/span><\/div>, <div class='tooltip'>ccF<span class='tooltiptext'>ccF Performs the Calinski and Corsten test based onthe F distribution.<\/span><\/div>, <div class='tooltip'>crd<span class='tooltiptext'>crd Analyses balanced experiments in CompletelyRandomized Design under one single factor, considering afixed model.<\/span><\/div>, <div class='tooltip'>duncan<span class='tooltiptext'>duncan Performs the test of Duncan for multiplecomparison of means.<\/span><\/div>, <div class='tooltip'>fat2.ad.crd<span class='tooltiptext'>fat2.ad.crd Analyses experiments in balancedCompletely Randomized Design in double factorial schemewith an additional treatment, considering a fixed model.<\/span><\/div>, <div class='tooltip'>fat2.ad.rbd<span class='tooltiptext'>fat2.ad.rbd Analyses experiments in balancedRandomized Blocks Designs in double factorial schemewith an additional treatment, considering a fixed model.<\/span><\/div>, <div class='tooltip'>fat2.crd<span class='tooltiptext'>fat2.crd Analyses experiments in balancedCompletely Randomized Design in double factorialscheme, considering a fixed model.<\/span><\/div>, <div class='tooltip'>fat2.rbd<span class='tooltiptext'>fat2.rbd Analyses experiments in balancedRandomized Blocks Designs in double factorial scheme,considering a fixed model.<\/span><\/div>, <div class='tooltip'>fat3.ad.crd<span class='tooltiptext'>fat3.ad.crd Analyses experiments in balancedCompletely Randomized Design in triple factorialscheme with an additional treatment, considering afixed model.<\/span><\/div>, <div class='tooltip'>fat3.ad.rbd<span class='tooltiptext'>fat3.ad.rbd Analyses experiments in balancedRandomized Blocks Designs in triple factorial scheme withan additional treatment, considering a fixed model.<\/span><\/div>, <div class='tooltip'>fat3.crd<span class='tooltiptext'>fat3.crd Analyses experiments in balanced CompletelyRandomized Design in triple factorial scheme, consideringa fixed model.<\/span><\/div>, <div class='tooltip'>fat3.rbd<span class='tooltiptext'>fat3.rbd Analyses experiments in balanced RandomizedBlocks Designs in triple factorial scheme, considering afixed model.<\/span><\/div>, <div class='tooltip'>ginv<span class='tooltiptext'>ginv Computes the Moore-Penrose generalized inverseof a matrix X.<\/span><\/div>, <div class='tooltip'>graphics<span class='tooltiptext'>graphics Plots from regression models fitted in ANOVA.<\/span><\/div>, <div class='tooltip'>han<span class='tooltiptext'>han Performs the test for homogeneity of variances ofHan (1969).<\/span><\/div>, <div class='tooltip'>lastC<span class='tooltiptext'>lastC A special function for the group of treatmentsin the multiple comparison tests. Use order.group.<\/span><\/div>, <div class='tooltip'>latsd<span class='tooltiptext'>lastd Analyses experiments in balanced Latin SquareDesign, considering a fixed model.<\/span><\/div>, <div class='tooltip'>layard<span class='tooltiptext'>layard Performs the test for homogeneity of variancesof Layard for Jackknife (1973).<\/span><\/div>, <div class='tooltip'>levene<span class='tooltiptext'>levene Performs the test for homogeneity of variancesof Levene (1960).<\/span><\/div>, <div class='tooltip'>lsd<span class='tooltiptext'>lsd Performs the t test (LSD) for multiple comparisonof means.<\/span><\/div>, <div class='tooltip'>lsdb<span class='tooltiptext'>lsdb Performs the t test (LSD) with Bonferroni'sprotection, for multiple comparison of means.<\/span><\/div>, <div class='tooltip'>oneilldbc<span class='tooltiptext'>oneilldbc Performs the test for homogeneity ofvariances of ONeill and Mathews (2002).<\/span><\/div>, <div class='tooltip'>oneillmathews<span class='tooltiptext'>oneillmathews Performs the test for homogeneity ofvariances of ONeill and Mathews (2000).<\/span><\/div>, <div class='tooltip'>order.group<span class='tooltiptext'>order.group It orders the groups of means.<\/span><\/div>, <div class='tooltip'>order.stat.SNK<span class='tooltiptext'>order.stat.SNK Orders the groups of means accordingto the test of SNK.<\/span><\/div>, <div class='tooltip'>plotres<span class='tooltiptext'>plotres Residual plots for a output model. Four setsof plots are produced: (1) Histogram, (2) normal probabilityplot for the residual, (3) Standardized Residuals versusFitted Values, and (4) box-plot (Standardized Residuals).<\/span><\/div>, <div class='tooltip'>rbd<span class='tooltiptext'>rbd Analyses experiments in balanced RandomizedBlocks Designs under one single factor, considering a fixedmodel.<\/span><\/div>, <div class='tooltip'>reg.nl<span class='tooltiptext'>reg.nl Adjusts non-linear regression models in Anova(Models: Power, Exponential, Logistic, Gompertz).<\/span><\/div>, <div class='tooltip'>reg.poly<span class='tooltiptext'>reg.poly Fits sequential regression models until thethird power.<\/span><\/div>, <div class='tooltip'>samiuddin<span class='tooltiptext'>samiuddin Performs the test for homogeneity ofvariances of Samiuddin (1976).<\/span><\/div>, <div class='tooltip'>scottknott<span class='tooltiptext'>scottknott Performs the test of Scott-Knott, formultiple comparison of means.<\/span><\/div>, <div class='tooltip'>snk<span class='tooltiptext'>snk Performs the test of SNK, for multiplecomparison of means.<\/span><\/div>, <div class='tooltip'>split2.crd<span class='tooltiptext'>split2.crd Analyses experiments in Split-plot schemein balanced Completely Randomized Design, considering afixed model.<\/span><\/div>, <div class='tooltip'>split2.rbd<span class='tooltiptext'>split2.rbd Analyses experiments in Split-plot schemein balanced Randomized Blocks Design, considering afixed model.<\/span><\/div>, <div class='tooltip'>strip<span class='tooltiptext'>strip Analysis Strip-plot experiments.<\/span><\/div>, <div class='tooltip'>tapply.stat<span class='tooltiptext'>tapply.stat This process lies in finding statisticswhich consist of more than one variable, grouped or crossedby factors. The table must be organized by columns betweenvariables and factors.<\/span><\/div>, <div class='tooltip'>tukey<span class='tooltiptext'>tukey Performs the test of Tukey, for multiplecomparison of means.<\/span><\/div>","<div class='tooltip'>anscombetukey<span class='tooltiptext'>anscombetukey Realiza o teste de Anscombe e Tukey(1963) para o teste de homegenidade de variancias em umdelineamento em blocos casualizados.<\/span><\/div>, <div class='tooltip'>bartlett<span class='tooltiptext'>bartlett Executa o teste de homogeneidade devariancias de Bartlett (1937) para um delineamentointeiramente casualizado.<\/span><\/div>, <div class='tooltip'>ccboot<span class='tooltiptext'>ccboot Realiza o teste de Bootstrap de Ramos eFerreira (2009) para comparacao multipla de medias.<\/span><\/div>, <div class='tooltip'>ccF<span class='tooltiptext'>ccF Realiza o teste de Calinski \\& Corsten baseadona distribuicao F.<\/span><\/div>, <div class='tooltip'>dbc<span class='tooltiptext'>dbc Analisa experimentos em Delineamento em BlocosCasualizados balanceado com um so fator, considerando omodelo fixo.<\/span><\/div>, <div class='tooltip'>dic<span class='tooltiptext'>dic Analisa experimentos em Delineamento InteiramenteCasualizado balanceado com um so fator, considerando omodelo fixo.<\/span><\/div>, <div class='tooltip'>dql<span class='tooltiptext'>dql Analisa experimentos em Delineamento em QuadradoLatino balanceado com um so fator, considerando o modelofixo.<\/span><\/div>, <div class='tooltip'>duncan<span class='tooltiptext'>duncan Realiza o teste de Duncan para comparacaomultipla de medias.<\/span><\/div>, <div class='tooltip'>faixas<span class='tooltiptext'>faixas Analisa experimentos em faixas.<\/span><\/div>, <div class='tooltip'>fat2.ad.dbc<span class='tooltiptext'>fat2.ad.dbc Analisa experimentos em fatorial duplocom um tratamento adicional em Delineamento em BlocosCasualizados balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat2.ad.dic<span class='tooltiptext'>fat2.ad.dic Analisa experimentos em fatorial duplocom um tratamento adicional em Delineamento InteiramenteCasualizado balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat2.dbc<span class='tooltiptext'>fat2.dbc Analisa experimentos em fatorial duploem Delineamento em Blocos Casualizados balanceado,considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat2.dic<span class='tooltiptext'>fat2.dic Analisa experimentos em fatorial duploem Delineamento InteiramenteCasualizado balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat3.ad.dbc<span class='tooltiptext'>fat3.ad.dbc Analisa experimentos em fatorial triplocom um tratamento adicional em Delineamento em BlocosCasualizados balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat3.ad.dic<span class='tooltiptext'>fat3.ad.dic Analisa experimentos em fatorial triplocom um tratamento adicional em Delineamento InteiramenteCasualizado balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat3.dbc<span class='tooltiptext'>fat3.dbc Analisa experimentos em fatorial triploem Delineamento em BlocosCasualizados balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>fat3.dic<span class='tooltiptext'>fat3.dic Analisa experimentos em fatorial triploem Delineamento InteiramenteCasualizado balanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>ginv<span class='tooltiptext'>ginv Inversa generalizada<\/span><\/div>, <div class='tooltip'>graficos<span class='tooltiptext'>graphicos Constroi graficos para os modelos deregressao ajustados na ANOVA.<\/span><\/div>, <div class='tooltip'>han<span class='tooltiptext'>han Realiza o teste de Han (1969) para homogeneidadede variancias em um delineamento em blocos casualizados.<\/span><\/div>, <div class='tooltip'>lastC<span class='tooltiptext'>lastC A special function for the group of treatmentsin the multiple comparison tests. Use order.group.<\/span><\/div>, <div class='tooltip'>layard<span class='tooltiptext'>layard Executa o teste de homogeneidade de varianciasde Layard (1973) por Jackknife para um delineamentointeiramente casualizado.<\/span><\/div>, <div class='tooltip'>levene<span class='tooltiptext'>levene Executa o teste de homogeneidade de varianciasde Levene (1960) para um delineamento inteiramentecasualizado.<\/span><\/div>, <div class='tooltip'>lsd<span class='tooltiptext'>lsd Realiza o teste t (LSD) para comparacao multiplade medias.<\/span><\/div>, <div class='tooltip'>lsdb<span class='tooltiptext'>lsdb Realiza o teste t (LSD), com protecao deBonferroni, para comparacao multipla de medias.<\/span><\/div>, <div class='tooltip'>oneilldbc<span class='tooltiptext'>oneilldbc Realiza o teste de ONeill e Mathews (2002)para o teste de homegenidade de variancias para umdelineamento em blocos casualizados.<\/span><\/div>, <div class='tooltip'>oneillmathews<span class='tooltiptext'>oneillmathews Executa o teste de homogeneidade devariancias de ONeill e Mathews (2000) para um delineamentointeiramente casualizado.<\/span><\/div>, <div class='tooltip'>order.group<span class='tooltiptext'>order.group It orders the groups of means.<\/span><\/div>, <div class='tooltip'>order.stat.SNK<span class='tooltiptext'>order.stat.SNK Orders the groups of means accordingto the test of SNK.<\/span><\/div>, <div class='tooltip'>plotres<span class='tooltiptext'>plotres Graficos dos residuos da saida do modelo.Conjunto de quatro graficos sao produzidos: (1) Histograma,(2) Grafico da probabilidade normal dos residuos com bandasde confianca pela estatistica de ordem, (3) ResiduosPadronizados versus Valores Ajustados e (4) box-plot(Residuos Padronizados).<\/span><\/div>, <div class='tooltip'>psub2.dbc<span class='tooltiptext'>psub2.dbc Analisa experimentos em esquema de parcelassubdivididas em Delineamento em Blocos Casualizadosbalanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>psub2.dic<span class='tooltiptext'>psub2.dic Analisa experimentos em esquema de parcelassubdivididas em Delineamento Inteiramente Casualizadobalanceado, considerando o modelo fixo.<\/span><\/div>, <div class='tooltip'>reg.nl<span class='tooltiptext'>reg.nl Ajusta modelos de regressao nao-linear na ANAVA.<\/span><\/div>, <div class='tooltip'>reg.poly<span class='tooltiptext'>reg.poly Realiza ajustes de modelos sequenciais deregressao ate o terceiro grau.<\/span><\/div>, <div class='tooltip'>samiuddin<span class='tooltiptext'>samiuddin Executa o teste de homogeneidade devariancias de Samiuddin (1976) para um delineamentointeiramente casualizado.<\/span><\/div>, <div class='tooltip'>scottknott<span class='tooltiptext'>scottknott Aplica o teste Scott-Knott para acomparacao multipla de medias.<\/span><\/div>, <div class='tooltip'>snk<span class='tooltiptext'>snk Realiza o teste Student Newman Keuls (SNK) paracomparacao multipla de medias.<\/span><\/div>, <div class='tooltip'>tapply.stat<span class='tooltiptext'>tapply.stat This process lies in finding statisticswhich consist of more than one variable, grouped or crossedby factors. The table must be organized by columns betweenvariables and factors.<\/span><\/div>, <div class='tooltip'>tukey<span class='tooltiptext'>tukey Produz a comparacao multipla de tratamentospara as medias pela proposta de Tukey. Por padrao consideraa significancia de 0.05.<\/span><\/div>","<div class='tooltip'>batch_names<span class='tooltiptext'>Given an index return the name of the batches the samples are in<\/span><\/div>, <div class='tooltip'>check_index<span class='tooltiptext'>Report the statistics for each subset and variable compared to the original.<\/span><\/div>, <div class='tooltip'>create_subset<span class='tooltiptext'>Index of the samples grouped by batches.<\/span><\/div>, <div class='tooltip'>design<span class='tooltiptext'>Given some samples it distribute them in several batches, trying to haveequal number of samples per batch. It can handle both numeric andcategorical data.<\/span><\/div>, <div class='tooltip'>distribution<span class='tooltiptext'>Checks if all the values are maximally distributed in the several batches.Aimed for categorical variables.<\/span><\/div>, <div class='tooltip'>entropy<span class='tooltiptext'>Calculates the entropy of a category. It uses the amount of categories toscale between 0 and 1.<\/span><\/div>, <div class='tooltip'>evaluate_entropy<span class='tooltiptext'>Looks if the nominal or character columns are equally distributed accordingto the entropy and taking into account the independence between batches.If any column is different in each row it is assumed to be the sample namesand thus omitted.<\/span><\/div>, <div class='tooltip'>evaluate_independence<span class='tooltiptext'>Looks the independence between the categories and the batches.<\/span><\/div>, <div class='tooltip'>evaluate_index<span class='tooltiptext'>Measures several indicators per group<\/span><\/div>, <div class='tooltip'>evaluate_mad<span class='tooltiptext'>Looks for the median absolute deviation values in each subgroup.<\/span><\/div>, <div class='tooltip'>evaluate_mean<span class='tooltiptext'>Looks for the mean of the numeric values<\/span><\/div>, <div class='tooltip'>evaluate_na<span class='tooltiptext'>Looks how are NA distributed in each subset<\/span><\/div>, <div class='tooltip'>evaluate_orig<span class='tooltiptext'>Measure some summary statistics of the whole cohort of samples<\/span><\/div>, <div class='tooltip'>evaluate_sd<span class='tooltiptext'>Looks for the standard deviation of the numeric values<\/span><\/div>, <div class='tooltip'>extreme_cases<span class='tooltiptext'>Subset some samples that are mostly different.<\/span><\/div>, <div class='tooltip'>inspect<span class='tooltiptext'>Given the index and the data of the samples append the batch assignment<\/span><\/div>, <div class='tooltip'>optimum_batches<span class='tooltiptext'>Calculates the optimum values for number of batches or size of the batches.If you need to do several batches it can be better to distribute it evenlyand add replicates.<\/span><\/div>, <div class='tooltip'>optimum_subset<span class='tooltiptext'>Calculates the optimum values for number of batches or size of the batches.If you need to do several batches it can be better to distribute it evenlyand add replicates.<\/span><\/div>, <div class='tooltip'>qcSubset<span class='tooltiptext'>Select randomly some samples from an index<\/span><\/div>, <div class='tooltip'>replicates<span class='tooltiptext'>To ensure that the batches are comparable some samples are processed in eachbatch. This function allows to take into account that effect.It uses the most different samples as controls as defined with extreme_cases().<\/span><\/div>, <div class='tooltip'>sizes_batches<span class='tooltiptext'>Calculates the optimum values for number of batches or size of the batches.If you need to do several batches it can be better to distribute it evenlyand add replicates.<\/span><\/div>, <div class='tooltip'>spatial<span class='tooltiptext'>This function assumes that to process the batch the samples are distributes ina plate. Sometimes you know in advance the<\/span><\/div>, <div class='tooltip'>use_index<span class='tooltiptext'>Convert a given factor to an accepted index<\/span><\/div>","<div class='tooltip'>ATEbounds<span class='tooltiptext'>This function computes the sharp bounds on the average treatment effect whensome of the outcome data are missing. The confidence intervals for thebounds are also computed.<\/span><\/div>, <div class='tooltip'>ATEcluster<span class='tooltiptext'>This function estimates various average treatment effect incluster-randomized experiments without using pre-treatment covariates. Thetreatment variable is assumed to be binary. Currently, only the matched-pairdesign is allowed. The details of the methods for this design are given inImai, King, and Nall (2007).<\/span><\/div>, <div class='tooltip'>ATEnocov<span class='tooltiptext'>This function computes the standard ``difference-in-means'' estimate of theaverage treatment effect in randomized experiments without usingpre-treatment covariates. The treatment variable is assumed to be binary.Currently, the two designs are allowed: complete randomized design andmatched-pair design.<\/span><\/div>, <div class='tooltip'>ATOPnoassumption<span class='tooltiptext'>This function computes the no assumption bounds on the average treatment effect among always-observed pairs (ATOP) whensome of the outcome data are missing. The confidence intervals for theATOP are also computed.<\/span><\/div>, <div class='tooltip'>ATOPobs<span class='tooltiptext'>This function computes the bounds on the average treatment effect among always-observed pairs (ATOP) with pre-specified sensivity parameters whensome of the outcome data are missing. The sensivity parameters characterizes the degree of the within-pair similarity and the dependence between the potential missing indicators and the treatment. The confidence intervals for theATOP are also computed.<\/span><\/div>, <div class='tooltip'>ATOPsens<span class='tooltiptext'>This function computes the bounds on the average treatment effect among always-observed pairs (ATOP) with pre-specified sensivity parameters whensome of the outcome data are missing. The sensivity parameter characterizes the degree of the within-pair similarity.The confidence intervals for theATOP are also computed.<\/span><\/div>, <div class='tooltip'>AUPEC<span class='tooltiptext'>This function estimates AUPEC. The details of the methods for this design are given in Imai and Li (2019).<\/span><\/div>, <div class='tooltip'>CACEcluster<span class='tooltiptext'>This function estimates various complier average causal effect incluster-randomized experiments without using pre-treatment covariates whenunit-level noncompliance exists. Both the encouragement and treatmentvariables are assumed to be binary. Currently, only the matched-pair designis allowed. The details of the methods for this design are given in Imai,King, and Nall (2007).<\/span><\/div>, <div class='tooltip'>CADErand<span class='tooltiptext'>This function computes the point estimates and variance estimates of the complier average direct effect (CADE) and the complier average spillover effect (CASE).The estimators calculated using this function are either individual weighted or cluster-weighted. The point estimates and variances of ITT effects are also included.<\/span><\/div>, <div class='tooltip'>CADEreg<span class='tooltiptext'>This function computes the point estimates of the complier average direct effect (CADE) and four different variance estimates: the HC2 variance, the cluster-robust variance, the cluster-robust HC2 variance and the variance proposed in the reference. The estimators calculated using this function are cluster-weighted, i.e., the weights are equal for each cluster. To obtain the indivudal-weighted estimators, please multiply the recieved treatment and the outcome by n_jJ/N, where n_j is the number of individuals in cluster j, J is the number of clusters and N is the total number of individuals.<\/span><\/div>, <div class='tooltip'>PAPD<span class='tooltiptext'>This function estimates the Population Average Prescription Difference with a budgetconstraint. The details of the methods for this design are given in Imai and Li (2019).<\/span><\/div>, <div class='tooltip'>PAPE<span class='tooltiptext'>This function estimates the Population Average Prescription Effect with and without a budgetconstraint. The details of the methods for this design are given in Imai and Li (2019).<\/span><\/div>, <div class='tooltip'>randomize<span class='tooltiptext'>This function can be used to randomize the treatment assignment forrandomized experiments. In addition to the complete randomization, itimplements randomized-block and matched-pair designs.<\/span><\/div>","<div class='tooltip'>augment_levels<span class='tooltiptext'>Augments the full factorial design with a column summarising the levels of that design. Importantly, it also adds the B-matrix as an attribute.<\/span><\/div>, <div class='tooltip'>check_overshadow<span class='tooltiptext'>Check Overshadow - Pareto Dominate Solutions<\/span><\/div>, <div class='tooltip'>construct_question_frame<span class='tooltiptext'>Convert from choice_sets to a question data<\/span><\/div>, <div class='tooltip'>dce_efficiency<span class='tooltiptext'>Efficiency Measures for Discrete Choice Experiments<\/span><\/div>, <div class='tooltip'>fractional_factorial_efficiency<span class='tooltiptext'>Fractional Factorial Design Efficiency<\/span><\/div>, <div class='tooltip'>full_factorial<span class='tooltiptext'>Generates the full factorial design with all the factors coded using standardised orthogonal contrast coding.<\/span><\/div>, <div class='tooltip'>modulo_method<span class='tooltiptext'>Modulo Method - Described by Street et al.<\/span><\/div>, <div class='tooltip'>search_design<span class='tooltiptext'>Returns a consistent fractional factorial design from the input fractional factorial design. The key advantage of this function is that it ensures factors are coded and enchances the attributes of the output.<\/span><\/div>","<div class='tooltip'>ezANOVA<span class='tooltiptext'>This function provides easy analysis of data from factorial experiments, including purely within-Ss designs (a.k.a. repeated measures), purely between-Ss designs, and mixed within-and-between-Ss designs, yielding ANOVA results, generalized effect sizes and assumption checks.<\/span><\/div>, <div class='tooltip'>ezBoot<span class='tooltiptext'>This function is used to compute bootstrap resampled predictions for each cell in a specified experimental design, using either cell means or mixed effects modelling to obtain predictions. The results can be visualized using ezPlot2.<\/span><\/div>, <div class='tooltip'>ezCor<span class='tooltiptext'>This function provides simultaneous visualization of a correlation matrix, scatter-plot with linear fits, and univariate density plots for multiple variables.<\/span><\/div>, <div class='tooltip'>ezDesign<span class='tooltiptext'>This function provides easy visualization of the balance of data in a data set given a specified experimental design. This function is useful for identifying missing data and other issues (see examples).<\/span><\/div>, <div class='tooltip'>ezMixed<span class='tooltiptext'>This function provides assessment of fixed effects and their interactions via generalized mixed effects modelling, or generalized additive mixed effects modelling for effects involving numeric predictors to account for potentially non-linear effects of such predictors. See Details section below for implementation notes.<\/span><\/div>, <div class='tooltip'>ezMixedProgress<span class='tooltiptext'>When running ezMixed with a value supplied to the progress_dir argument, summary results are saved to file. ezMixedProgress retrieves those results, even from partial or discontinued runs.<\/span><\/div>, <div class='tooltip'>ezPerm<span class='tooltiptext'>This function provides easy non-parametric permutation test analysis of data from factorial experiments, including purely within-Ss designs (a.k.a. repeated measures), purely between-Ss designs, and mixed within-and-between-Ss designs.<\/span><\/div>, <div class='tooltip'>ezPlot<span class='tooltiptext'>This function provides easy visualization of any given user-requested effect from factorial experiments, including purely within-Ss designs (a.k.a. repeated measures), purely between-Ss designs, and mixed within-and-between-Ss designs. By default, Fisher's Least Significant Difference is computed to provide error bars that facilitate visual post-hoc multiple comparisons (see Warning section below).<\/span><\/div>, <div class='tooltip'>ezPlot2<span class='tooltiptext'>This function provides easy visualization of any given user-requested effect from the bootstrap predictions computed by ezPredict or ezBoot.<\/span><\/div>, <div class='tooltip'>ezPrecis<span class='tooltiptext'>This function provides a structure summary of a given data frame.<\/span><\/div>, <div class='tooltip'>ezPredict<span class='tooltiptext'>This function computes the predicted values from the fixed effects of a mixed effects model.<\/span><\/div>, <div class='tooltip'>ezResample<span class='tooltiptext'>This function resamples data (useful when bootstrapping and used by ezBoot).<\/span><\/div>, <div class='tooltip'>ezStats<span class='tooltiptext'>This function provides easy computation of descriptive statistics (between-Ss means, between-Ss SD, Fisher's Least Significant Difference) for data from factorial experiments, including purely within-Ss designs (a.k.a. repeated measures), purely between-Ss designs, and mixed within-and-between-Ss designs.<\/span><\/div>","<div class='tooltip'>factoptd<span class='tooltiptext'>Used to compute factorial A-, D- or E-optimal designs for two-colour cDNA microarray experiments.<\/span><\/div>, <div class='tooltip'>factoptd.default<span class='tooltiptext'>Used to compute factorial A-, D- or E-optimal designs for two-colour cDNA microarray experiments.<\/span><\/div>, <div class='tooltip'>print.factoptd<span class='tooltiptext'>Used to compute factorial A-, D- or E-optimal designs for two-colour cDNA microarray experiments.<\/span><\/div>","<div class='tooltip'>decompose_pAMCE<span class='tooltiptext'>See examples in `model_pAMCE`.<\/span><\/div>, <div class='tooltip'>design_pAMCE<span class='tooltiptext'>design_pAMCE implements the design-based approach to estimate the pAMCE. See de la Cuesta, Egami, and Imai (2019+) for details. More examples are available at the GitHub page of factorEx.<\/span><\/div>, <div class='tooltip'>diagnose_pAMCE<span class='tooltiptext'>Diagnose modeling assumptions for the model-based approach<\/span><\/div>, <div class='tooltip'>model_pAMCE<span class='tooltiptext'>model_pAMCE implements the model-based approach to estimate the pAMCE. See de la Cuesta, Egami, and Imai (2019+) for details. More examples are available at the GitHub page of factorEx.<\/span><\/div>, <div class='tooltip'>plot_decompose<span class='tooltiptext'>See examples in `model_pAMCE`.<\/span><\/div>, <div class='tooltip'>plot_diagnose<span class='tooltiptext'>Plotting diagnostic checks<\/span><\/div>, <div class='tooltip'>weights_pAMCE<span class='tooltiptext'>Computing weights for estimating the population AMCE using a design-based approach<\/span><\/div>","<div class='tooltip'>%>%<span class='tooltiptext'>See dplyr::%>% for details.<\/span><\/div>, <div class='tooltip'>average_r2tau_0<span class='tooltiptext'>Average r to Random Intercept SD<\/span><\/div>, <div class='tooltip'>beta2norm<span class='tooltiptext'>Convert beta to normal<\/span><\/div>, <div class='tooltip'>binom2norm<span class='tooltiptext'>Convert a binomial distribution to a normal (gaussian) distribution with specified mu and sd<\/span><\/div>, <div class='tooltip'>check_design<span class='tooltiptext'>Specify any number of within- and between-subject factors with any number of levels.<\/span><\/div>, <div class='tooltip'>check_mixed_design<span class='tooltiptext'>Get error terms from an existing data table.<\/span><\/div>, <div class='tooltip'>check_sim_stats<span class='tooltiptext'>Generates a table of the correlations and means of numeric columns in a data frame. If data was generated by sim_design and has a \"design\" attribute, between, within, dv and id are retrieved from that, unless overridden (use between = 0 to<\/span><\/div>, <div class='tooltip'>codebook<span class='tooltiptext'>See ../doc/codebook.htmlvignette(\"codebook\", package = \"faux\") for details.<\/span><\/div>, <div class='tooltip'>cormat<span class='tooltiptext'>cormat makes a correlation matrix from a single number, vars\\*vars matrix, vars\\*vars vector, or a vars\\*(vars-1)/2 vector.<\/span><\/div>, <div class='tooltip'>cormat_from_triangle<span class='tooltiptext'>cormat_from_triangle makes a correlation matrix from a vector of the upper right triangle<\/span><\/div>, <div class='tooltip'>faux_options<span class='tooltiptext'>Global faux options are used, for example, to set the default separator for cell names.<\/span><\/div>, <div class='tooltip'>fix_name_labels<span class='tooltiptext'>Fixes if a factor list does not have named levels or has special characters in the names<\/span><\/div>, <div class='tooltip'>gamma2norm<span class='tooltiptext'>Convert gamma to normal<\/span><\/div>, <div class='tooltip'>get_design_long<span class='tooltiptext'>Makes a best guess at the design of a long-format data frame.<\/span><\/div>, <div class='tooltip'>get_params<span class='tooltiptext'>Generates a table of the correlations and means of numeric columns in a data frame. If data was generated by sim_design and has a \"design\" attribute, between, within, dv and id are retrieved from that, unless overridden (use between = 0 to<\/span><\/div>, <div class='tooltip'>getcols<span class='tooltiptext'>Get columns from a data table by specifying the index, column name as a string, or unquoted column name. Returns the column names or indices.<\/span><\/div>, <div class='tooltip'>interactive_design<span class='tooltiptext'>Set design interactively<\/span><\/div>, <div class='tooltip'>is_pos_def<span class='tooltiptext'>is_pos_def makes a correlation matrix from a vector<\/span><\/div>, <div class='tooltip'>json_design<span class='tooltiptext'>Convert a design list to JSON notation for archiving (e.g. in scienceverse)<\/span><\/div>, <div class='tooltip'>long2wide<span class='tooltiptext'>Convert data from long to wide format<\/span><\/div>, <div class='tooltip'>make_id<span class='tooltiptext'>Make IDs with fixed length and a prefix (e.g., S001, S002, ..., S100).<\/span><\/div>, <div class='tooltip'>messy<span class='tooltiptext'>Insert NA or another replacement value for some proportion of specified columns to simulate missing data.<\/span><\/div>, <div class='tooltip'>nested_list<span class='tooltiptext'>Output a nested list in RMarkdown list format<\/span><\/div>, <div class='tooltip'>norm2beta<span class='tooltiptext'>Convert normal to beta<\/span><\/div>, <div class='tooltip'>norm2binom<span class='tooltiptext'>Convert normal to binomial<\/span><\/div>, <div class='tooltip'>norm2gamma<span class='tooltiptext'>Convert normal to gamma<\/span><\/div>, <div class='tooltip'>norm2likert<span class='tooltiptext'>Convert normal to likert<\/span><\/div>, <div class='tooltip'>norm2pois<span class='tooltiptext'>Convert normal to poisson<\/span><\/div>, <div class='tooltip'>norm2trunc<span class='tooltiptext'>Convert a normal (gaussian) distribution to a truncated normal distribution with specified minimum and maximum<\/span><\/div>, <div class='tooltip'>norm2unif<span class='tooltiptext'>Convert a normal (gaussian) distribution to a uniform distribution with specified minimum and maximum<\/span><\/div>, <div class='tooltip'>plot_design<span class='tooltiptext'>Plots the specified within and between design. See ../doc/plots.htmlvignette(\"plots\", package = \"faux\") for examples and details.<\/span><\/div>, <div class='tooltip'>pos_def_limits<span class='tooltiptext'>pos_def_limits returns min and max possible values for a positive definite matrix with a specified missing value<\/span><\/div>, <div class='tooltip'>readline_check<span class='tooltiptext'>Check readline input<\/span><\/div>, <div class='tooltip'>rnorm_multi<span class='tooltiptext'>Make normally distributed vectors with specified relationships. See ../doc/rnorm_multi.htmlvignette(\"rnorm_multi\", package = \"faux\") for details.<\/span><\/div>, <div class='tooltip'>rnorm_pre<span class='tooltiptext'>rnorm_pre Produces a random normally distributed vector with the specified correlation to one or more existing vectors<\/span><\/div>, <div class='tooltip'>sample_from_pop<span class='tooltiptext'>Sample Parameters from Population Parameters<\/span><\/div>, <div class='tooltip'>sim_data<span class='tooltiptext'>Simulate data from design (internal)<\/span><\/div>, <div class='tooltip'>sim_design<span class='tooltiptext'>Generates a data table with a specified within and between design. See ../doc/sim_design.htmlvignette(\"sim_design\", package = \"faux\") for examples and details.<\/span><\/div>, <div class='tooltip'>sim_df<span class='tooltiptext'>Produces a data table with the same distributions and correlations as an existing data table Only returns numeric columns and simulates all numeric variables from a continuous normal distribution (for now).<\/span><\/div>, <div class='tooltip'>sim_joint_dist<span class='tooltiptext'>This function is mainly used internally, such as for simulating missing data patterns, but is available in case anyone finds it useful.<\/span><\/div>, <div class='tooltip'>sim_mixed_cc<span class='tooltiptext'>Makes a basic cross-classified design with random intercepts for subjects and items. See ../doc/sim_mixed.htmlvignette(\"sim_mixed\", package = \"faux\") for examples and details.<\/span><\/div>, <div class='tooltip'>sim_mixed_df<span class='tooltiptext'>sim_mixed_df() produces a data table with the same distributions of by-subject and by-item random intercepts as an existing data table.<\/span><\/div>, <div class='tooltip'>std_alpha2average_r<span class='tooltiptext'>Standardized Alpha to Average R<\/span><\/div>, <div class='tooltip'>trunc2norm<span class='tooltiptext'>Convert a truncated normal distribution to a normal (gaussian) distribution<\/span><\/div>, <div class='tooltip'>unif2norm<span class='tooltiptext'>Convert a uniform distribution to a normal (gaussian) distribution with specified mu and sd<\/span><\/div>, <div class='tooltip'>unique_pairs<span class='tooltiptext'>Make unique pairs of level names for correlations<\/span><\/div>, <div class='tooltip'>wide2long<span class='tooltiptext'>Convert data from wide to long format<\/span><\/div>","<div class='tooltip'>gen.level<span class='tooltiptext'>Generate coded integers for giventotal number of levels of a factor.<\/span><\/div>, <div class='tooltip'>minimal.factorial<span class='tooltiptext'>Generate minimally changed run sequences for a given asymmetricalor symmetrical factorial design.<\/span><\/div>","<div class='tooltip'>add.center<span class='tooltiptext'>This function adds center points to a 2-level fractional factorial design. All factors must be quantitative!<\/span><\/div>, <div class='tooltip'>aliases<span class='tooltiptext'>Functions to examine the alias structure of a fractional factorial 2-level design<\/span><\/div>, <div class='tooltip'>aliasprint<span class='tooltiptext'>Functions to examine the alias structure of a fractional factorial 2-level design<\/span><\/div>, <div class='tooltip'>all.2fis.clear.catlg<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>blockgencreate<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>, <div class='tooltip'>blockpick<span class='tooltiptext'>Functions to investigate potential assignments of blocks and show alias information of resulting designs, meant for expert users<\/span><\/div>, <div class='tooltip'>blockpick.big<span class='tooltiptext'>Functions to investigate potential assignments of blocks and show alias information of resulting designs, meant for expert users<\/span><\/div>, <div class='tooltip'>BsProb.design<span class='tooltiptext'>The function calculates Bayesian posterior probabilities according to Box and Meyer (1993) for screening experiments with 2-level factors. The function is modified from function BsProb in packge BsMD with the purpose of providing usage comfort for class design objects.<\/span><\/div>, <div class='tooltip'>CIG<span class='tooltiptext'>Function CIG creates a clear interactions graph (CIG) from a catlg design (design name must be given). Function CIGstatic allows to create a static graph from a dynamically-adjusted one.<\/span><\/div>, <div class='tooltip'>CIGstatic<span class='tooltiptext'>Function CIG creates a clear interactions graph (CIG) from a catlg design (design name must be given). Function CIGstatic allows to create a static graph from a dynamically-adjusted one.<\/span><\/div>, <div class='tooltip'>clear.2fis<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>colpick<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>, <div class='tooltip'>compromise<span class='tooltiptext'>Addelman (1962) and Ke and Wu (2005) discuss compromise plans of different types. Their creation is supported by the function compromise.<\/span><\/div>, <div class='tooltip'>cubecorners<span class='tooltiptext'>A cube plot for the combined effect of three factors is produced (function cubePlot). Functions cubedraw, cubecorners, cubelabel and myscatterplot3d are not intended for users.<\/span><\/div>, <div class='tooltip'>cubedraw<span class='tooltiptext'>A cube plot for the combined effect of three factors is produced (function cubePlot). Functions cubedraw, cubecorners, cubelabel and myscatterplot3d are not intended for users.<\/span><\/div>, <div class='tooltip'>cubelabel<span class='tooltiptext'>A cube plot for the combined effect of three factors is produced (function cubePlot). Functions cubedraw, cubecorners, cubelabel and myscatterplot3d are not intended for users.<\/span><\/div>, <div class='tooltip'>cubePlot<span class='tooltiptext'>A cube plot for the combined effect of three factors is produced (function cubePlot). Functions cubedraw, cubecorners, cubelabel and myscatterplot3d are not intended for users.<\/span><\/div>, <div class='tooltip'>DanielPlot<span class='tooltiptext'>The function is modified from the same-name function in packge BsMD with the purpose of providing more usage comfort (correct effect sizes in case of factors, automatic annotation, automatic labelling of the most significant factors only).<\/span><\/div>, <div class='tooltip'>dominating<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>FF_from_X<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>, <div class='tooltip'>fold.design<span class='tooltiptext'>This function creates a foldover design for a 2-level fractional factorial. The purpose is to dealias (some) effects. Per default, all factors are folded upon, which makes the resulting design at least resolution IV. Different foldover versions can be requested.<\/span><\/div>, <div class='tooltip'>FrF2<span class='tooltiptext'>Regular fractional factorial 2-level designs are provided. Apart from obtaining the usual minimum aberration designs in a fixed number of runs, it is possible to request highest number of free 2-factor interactions instead of minimum aberration or to request the smallest design that fulfills certain requirements (e.g. resolution V with 8 factors).<\/span><\/div>, <div class='tooltip'>FrF2.currentlychecked<span class='tooltiptext'>Regular fractional factorial 2-level designs are provided. Apart from obtaining the usual minimum aberration designs in a fixed number of runs, it is possible to request highest number of free 2-factor interactions instead of minimum aberration or to request the smallest design that fulfills certain requirements (e.g. resolution V with 8 factors).<\/span><\/div>, <div class='tooltip'>FrF2Large<span class='tooltiptext'>Large regular fractional factorial 2-level designs in 8192 or more runs are provided: Resolution V designs in 8096 to 32768 runs with up to 120 factors according to the suggestion by Sanchez and Sanchez 2005 are automatically created (these are not necessarily optimal). Furthermore, manual generation of large regular fractional factorial designs via specification of generators is possible.<\/span><\/div>, <div class='tooltip'>gen2CIG<span class='tooltiptext'>Function CIG creates a clear interactions graph (CIG) from a catlg design (design name must be given). Function CIGstatic allows to create a static graph from a dynamically-adjusted one.<\/span><\/div>, <div class='tooltip'>IAPlot<span class='tooltiptext'>Main effects plots and interaction plots are produced. The other documented functions are not intended for users.<\/span><\/div>, <div class='tooltip'>leftadjust<span class='tooltiptext'>Functions to restructure a fractional factorial by permuting the base factors such that the leftmost base factors have a suitable alias structure for the problem at hand; meant for expert users<\/span><\/div>, <div class='tooltip'>makecatlg<span class='tooltiptext'>creates a class catlg catalogue with a single element for use in functions colpick or FrF2<\/span><\/div>, <div class='tooltip'>MEPlot<span class='tooltiptext'>Main effects plots and interaction plots are produced. The other documented functions are not intended for users.<\/span><\/div>, <div class='tooltip'>nclear.2fis<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>nfac<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>nruns<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>nrunsV<span class='tooltiptext'>Large regular fractional factorial 2-level designs in 8192 or more runs are provided: Resolution V designs in 8096 to 32768 runs with up to 120 factors according to the suggestion by Sanchez and Sanchez 2005 are automatically created (these are not necessarily optimal). Furthermore, manual generation of large regular fractional factorial designs via specification of generators is possible.<\/span><\/div>, <div class='tooltip'>pb<span class='tooltiptext'>The function generates Plackett-Burman designs and in some cases other screening designs in run numbers that are a multiple of 4. These designs are particularly suitable for screening a large number of factors, since interactions are not fully aliased with one main effect each but partially aliased. (The design in 8 runs is an exception from this rule.)<\/span><\/div>, <div class='tooltip'>phimax<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>, <div class='tooltip'>res<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>splitpick<span class='tooltiptext'>Functions to restructure a fractional factorial by permuting the base factors such that the leftmost base factors have a suitable alias structure for the problem at hand; meant for expert users<\/span><\/div>, <div class='tooltip'>WLP<span class='tooltiptext'>Functions to select elements or extract information from design catalogues of class catlg<\/span><\/div>, <div class='tooltip'>X_from_parts<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>, <div class='tooltip'>X_from_profile<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>, <div class='tooltip'>Xcalc<span class='tooltiptext'>Function colpick handles the creation of X matrices for blocking, function FF_from_X blocks a full factorial, function X_from_profile creates an X matrix from a profile, function phimax calculations the maximum number of clear 2fis from Godolphin's approach. Further helper functions support the use of the method. The functions are meant for expert users only.<\/span><\/div>","<div class='tooltip'><span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>gen2simon<span class='tooltiptext'>Calculates generalized optimal and minimax 2-stage phase II designs based on the R function ph2simon.<\/span><\/div>, <div class='tooltip'>gen2single<span class='tooltiptext'>Calculates the generalized exact one stage phase II design based on the R function ph2single.<\/span><\/div>, <div class='tooltip'>oc.gentwostage.bdry<span class='tooltiptext'>Calculates the operating characteristics of a two-stage boundary based on the R function oc.twostage.bdry.<\/span><\/div>, <div class='tooltip'>plot.gen2simon<span class='tooltiptext'>Calculates generalized optimal and minimax 2-stage phase II designs based on the R function ph2simon.<\/span><\/div>, <div class='tooltip'>print.gen2simon<span class='tooltiptext'>Calculates generalized optimal and minimax 2-stage phase II designs based on the R function ph2simon.<\/span><\/div>, <div class='tooltip'>print.gen2single<span class='tooltiptext'>Calculates the generalized exact one stage phase II design based on the R function ph2single.<\/span><\/div>","<div class='tooltip'>calculateGUI<span class='tooltiptext'>This function provides a graphical user interface for calculating statisticaltests in general factorial designs.<\/span><\/div>, <div class='tooltip'>GFD<span class='tooltiptext'>The GFD function calculates the Wald-type statistic (WTS), the ANOVA-type statistic (ATS) as well as a permutation version of the WTS for general factorial designs.<\/span><\/div>","<div class='tooltip'>find.scheme<span class='tooltiptext'>%% ~~ A concise (1-5 lines) description of what the function does. ~~Given the prior distribution (or values) of parameters,and the total/maximum number of groups (N) allowed for grouping schemes,this function finds the global optimal grouping scheme that makesthe sampling process most informative.<\/span><\/div>, <div class='tooltip'>grcmle<span class='tooltiptext'>This function infers Poisson or zero-inflated Poisson (ZIP) parameters from grouped and right-censored count data,and conducts a chi-squared goodness-of-fit test.A grouped and right-censored scheme may look like 0, 1, 2--4, 5--8, 9+.For grouped and right-censored count data collected in a survey, such as frequency of alcohol drinking, number of births or occurrence of crimes,the response category designed as the example above means never, once, 2 to 4 times, 5 to 8 times, 9 times and more.The frequency distribution from a sample corresponding to the example above may look like 3, 15, 168, 155, 15.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>","<div class='tooltip'><span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>hist_end<span class='tooltiptext'>Adjust a design based on historical controls at the end of the study using the covariate data of the patients accrued in stage 2.<\/span><\/div>, <div class='tooltip'>hist_interim<span class='tooltiptext'>Adjust a design based on historical controls at interim using the covariate data of the patients accrued in stage 1.<\/span><\/div>, <div class='tooltip'>hist_start<span class='tooltiptext'>Initializes a design based on historical controls before the start of the study.<\/span><\/div>, <div class='tooltip'>strat_end<span class='tooltiptext'>Adjust a subspace stratified design at the end of the study.<\/span><\/div>, <div class='tooltip'>strat_interim<span class='tooltiptext'>Adjust a subspace stratified design at interim.<\/span><\/div>, <div class='tooltip'>strat_start<span class='tooltiptext'>Initializes a subspace stratified design before the start of the study.<\/span><\/div>","<div class='tooltip'>diffVariantError<span class='tooltiptext'>This function finds a practical threshold to differentiate a variant call from sequencing error.<\/span><\/div>, <div class='tooltip'>FindOptPower<span class='tooltiptext'>Perform a grid search over potential design space, and find the predicted power and validity of the designs.<\/span><\/div>, <div class='tooltip'>PlotOptPower<span class='tooltiptext'>Based on the predicted values from FindOptPower, plot the 3d or contour figures. Dependes on the package rgl for 3d plots.<\/span><\/div>, <div class='tooltip'>predictPoolNCP<span class='tooltiptext'>It is based on our simulation results.<\/span><\/div>, <div class='tooltip'>probDetEqual3<span class='tooltiptext'>Find the probability of detection.<\/span><\/div>, <div class='tooltip'>ShowOptDesign<span class='tooltiptext'>Show the top [num.designs] choices of valid designs.<\/span><\/div>, <div class='tooltip'>XmeanGivenCost<span class='tooltiptext'>A function used to find best coverage given cost and other constraints<\/span><\/div>","<div class='tooltip'>A_eff<span class='tooltiptext'>This function computes lower bound to A-efficiency of a binary incomplete block design. Treatment by block incidence matrix of the design is to be supplied as input to the function.<\/span><\/div>, <div class='tooltip'>A_eff.NNP<span class='tooltiptext'>This function computes lower bound to A-efficiency of a binary proper incomplete block design given the block size, number of blocks and concurrence matrix.<\/span><\/div>, <div class='tooltip'>aov.ibd<span class='tooltiptext'>This function performs intrablock analysis of variance of data from experiments using a block design. It also computes estimated marginal means of the factor variables (e.g. treatments) and optionally estimates and tests the contrasts of factor variables (e.g treatments).<\/span><\/div>, <div class='tooltip'>bibd<span class='tooltiptext'>This function generates a balanced incomplete block design with given number of treaments(v), number of blocks(b), number of replications (r), block size(k) and number of concurrences (lambda).<\/span><\/div>, <div class='tooltip'>btib<span class='tooltiptext'>This function generates a balanced treatment incomplete block design for specified parameters.<\/span><\/div>, <div class='tooltip'>btib1<span class='tooltiptext'>This function generates a balanced treatment incomplete block design for specified parameters by searching all possible combinations.<\/span><\/div>, <div class='tooltip'>btibgen<span class='tooltiptext'>This function generate an incidence matrix of a BTIB design for given v,b,k and concurrence matrix<\/span><\/div>, <div class='tooltip'>ccmat<span class='tooltiptext'>This function generates a nearly balanced concurrence matrix from given number of treatments (v), number of blocks (b) and block size (k) by trial and error.<\/span><\/div>, <div class='tooltip'>ccmat_LP<span class='tooltiptext'>This function generates the nearly balanced concurrence matrix from given number of treatments (v), number of blocks (b) and block size (k)<\/span><\/div>, <div class='tooltip'>check.orthogonality<span class='tooltiptext'>This function checks whether rows of a given matrix are pairwise orthogonal or not. Returns a value 1 if the rows are pairwise orthogonal, else return 0.<\/span><\/div>, <div class='tooltip'>check.validity.NNP<span class='tooltiptext'>This function checks whether a v by v matrix is concurrence matrix or not. Mainly it checks whether the sum of off-diagonal elements of the matrix is (k-1) times the diagonal element for each row of the given matrix. Applicable for proper binary incomplete block design only. If the condition is satisfied, it returns a value of 1 else it returns 0.<\/span><\/div>, <div class='tooltip'>Cmatrix<span class='tooltiptext'>This function gives the information matrix from a given treatment by block incidence matrix of a block design<\/span><\/div>, <div class='tooltip'>D_eff<span class='tooltiptext'>This function computes lower bound to D-efficiency of a binary incomplete block design. Treatment by block incidence matrix of the design is to be supplied as input to the function.<\/span><\/div>, <div class='tooltip'>design_to_N<span class='tooltiptext'>This function generates treatment by block incidence matrix from a given block design<\/span><\/div>, <div class='tooltip'>do.exist.NBIB<span class='tooltiptext'>This function checks whether a nearly balanced incomplete block design exists or not for given parameters<\/span><\/div>, <div class='tooltip'>g<span class='tooltiptext'>This function computes value of g function for given v,b,k,x,z<\/span><\/div>, <div class='tooltip'>gts<span class='tooltiptext'>This function computes minimum value of g function for over permissible values of x,z for given v,b,k<\/span><\/div>, <div class='tooltip'>ibd<span class='tooltiptext'>This function generates an A- and D- efficient binary incomplete block design with given number of treaments(v), number of blocks(b) and block size(k) and optionally with a specified concurrence matrix(NNP).<\/span><\/div>, <div class='tooltip'>ibdgen<span class='tooltiptext'>This function generate an incidence matrix for given v,b,k and concurrence matrix<\/span><\/div>, <div class='tooltip'>ibdtvc<span class='tooltiptext'>This function generates an incomplete block design for test vs control(s) comparisons with specified parameters and concurrence matrix.<\/span><\/div>, <div class='tooltip'>interchange.NNP<span class='tooltiptext'>This function interchanges the distinct off-diagonal elements of a given concurrence matrix and produces a concurrence matrix which has highest A-efficiency<\/span><\/div>, <div class='tooltip'>is.connected<span class='tooltiptext'>This function checks whether an incomplete block design is connected or not. Treatment by block incidence matrix of the design is to be supplied as input to the function. If the design is connected, it returns a value of 1 else it returns 0.<\/span><\/div>, <div class='tooltip'>is.equir<span class='tooltiptext'>This function checks whether an incomplete block design is equi-replicated or not. Treatment by block incidence matrix of the design is to be supplied as input to the function. If the design is equir-eplicated, it returns a value of 1 else it returns 0.<\/span><\/div>, <div class='tooltip'>is.orthogonal<span class='tooltiptext'>This function checks whether an incomplete block design is orthogonal or not. Treatment by block incidence matrix of the design is to be supplied as input to the function. If the design is orthogonal, it returns a value of 1 else it returns 0.<\/span><\/div>, <div class='tooltip'>is.proper<span class='tooltiptext'>This function checks whether an incomplete block design is proper or not. Treatment by block incidence matrix of the design is to be supplied as input to the function. If the design is proper, it returns a value of 1 else it returns 0.<\/span><\/div>, <div class='tooltip'>is.vb<span class='tooltiptext'>This function checks whether an incomplete block design is variance balanced or not. Treatment by block incidence matrix of the design is to be supplied as input to the function. If the design is variance balanced, it returns a value of 1 else it returns 0.<\/span><\/div>, <div class='tooltip'>is.wholenumber<span class='tooltiptext'>This function checks whether a given number is whole number or not.<\/span><\/div>, <div class='tooltip'>LIP<span class='tooltiptext'>This function update an incidence matrix by adding a row through linear integer programming formulation for given number of treatments, number of blocks, block sizes and concurrence matrix and the previous rows of incidence matrix<\/span><\/div>, <div class='tooltip'>N_to_design<span class='tooltiptext'>This function generates the block contents from a given treatment by block incidence matrix<\/span><\/div>, <div class='tooltip'>NLIP<span class='tooltiptext'>This function generates a block design with specified parameters and concurrence matrix through an alternative linear integer programming approach<\/span><\/div>","<div class='tooltip'>bayes<span class='tooltiptext'>Finds (pseudo) Bayesian D-optimal designs for linear and nonlinear models. It should be used when the user assumes a (truncated) prior distribution for the unknown model parameters. If you have a discrete prior, please use the function robust.<\/span><\/div>, <div class='tooltip'>bayes.update<span class='tooltiptext'>Runs the ICA optimization algorithm on an object of class minimax for more number of iterations and updates the results.<\/span><\/div>, <div class='tooltip'>bayescomp<span class='tooltiptext'>Finds compound Bayesian DP-optimal designs that meet the dual goal of parameter estimation and increasing the probability of a particular outcome in a binary response model.A compound Bayesian DP-optimal design maximizes the product of the Bayesian efficiencies of a design \\xi with respect to D- and average P-optimality, weighted by a pre-defined mixing constant0 \\leq \\alpha \\leq 10 <= \\alpha <= 1.<\/span><\/div>, <div class='tooltip'>beff<span class='tooltiptext'>Given a prior distribution for the parameters, this function calculates the Bayesian D-and PA- efficiency of a design \\xi_1 with respect to a design \\xi_2.Usually, \\xi_2 is an optimal design.This function is especially useful for investigating the robustness of Bayesian optimal designs under different prior distributions (See 'Examples').<\/span><\/div>, <div class='tooltip'>crt.bayes.control<span class='tooltiptext'>This function returns two lists each corresponds to an implemented integration method for approximating the integrals in Bayesian criteria. The first list is named cubature and contains the hcubature control parameters to approximate the integrals with an adaptive multivariate integration method over hypercubes. The second list is named quadrature and contains the createNIGrid tuning parameters to approximate the integrals with the quadrature methods.<\/span><\/div>, <div class='tooltip'>crt.minimax.control<span class='tooltiptext'>The function crt.minimax.control returns a list of nloptr control parameters for optimizing the minimax criterion over the parameter space.The key tuning parameter for our application is maxeval.Its value should be increased when either the dimension or the size of the parameter space becomes larger to avoid pre-mature convergence in the inner optimization problem over the parameter space.If the CPU time matters, the user should find an appropriate speed-accuracy trade-off for her/his own design problem.<\/span><\/div>, <div class='tooltip'>FIM_2par_exp_censor1<span class='tooltiptext'>It provides the cpp function for the FIM introduced in Eq. (3.1) of Schmidt and Schwabe (2015) for type one censored data.<\/span><\/div>, <div class='tooltip'>FIM_2par_exp_censor2<span class='tooltiptext'>It provides the cpp function for the FIM introduced in Eq. (3.1) of Schmidt and Schwabe (2015) for random censored data (type two censored data).<\/span><\/div>, <div class='tooltip'>FIM_3par_exp_censor1<span class='tooltiptext'>It provides the cpp function for the FIM introduced in Page 247 of Schmidt and Schwabe (2015) for type one censored data.<\/span><\/div>, <div class='tooltip'>FIM_3par_exp_censor2<span class='tooltiptext'>It provides the cpp function for the FIM introduced in Page 247 of Schmidt and Schwabe (2015) for random censored data (type two censored data).<\/span><\/div>, <div class='tooltip'>FIM_exp_2par<span class='tooltiptext'>It provides the cpp function for FIM for the model ~a + exp(-b*x).<\/span><\/div>, <div class='tooltip'>FIM_kinetics_alcohol<span class='tooltiptext'>It provides the cpp function for FIM for the model ~(b3 * x1)/(1 + b1 * x1 + b2 * x2)<\/span><\/div>, <div class='tooltip'>FIM_logistic<span class='tooltiptext'>It provides the cpp function for FIM for the model ~1/(1 + exp(-b *(x - a))).In item response theory (IRT),a is the item difficulty parameter, b is the item discrimination parameter and x is the person ability parameter.<\/span><\/div>, <div class='tooltip'>FIM_logistic_2pred<span class='tooltiptext'>It provides the cpp function for FIM for the following model: ~exp(b0+ b1 * x1 + b2 * x2 + b3 * x1 * x2)/(1 + exp(b0 + b1 * x1 + b2 * x2 + b3 * x1 * x2)).<\/span><\/div>, <div class='tooltip'>FIM_logistic_4par<span class='tooltiptext'>It provides the cpp function for the FIM for the model ~theta1/(1+exp(theta2*x+theta3))+theta4. This model is another re-parameterization of the 4-parameter Hill model. For more details, see Eq. (1) and (2) in Hyun and Wong (2015).<\/span><\/div>, <div class='tooltip'>FIM_loglin<span class='tooltiptext'>It provides the cpp function for the FIM for the model ~theta0 + theta1* log(x + theta2).<\/span><\/div>, <div class='tooltip'>FIM_mixed_inhibition<span class='tooltiptext'>It provides the cpp function for FIM for the model ~ V*S/(Km * (1 + I/Kic)+ S * (1 + I/Kiu))<\/span><\/div>, <div class='tooltip'>FIM_power_logistic<span class='tooltiptext'>It provides the cpp function for FIM for the model ~1/(1 + exp(-b *(x - a)))^s, but when s is fixed (a two by two matrix).<\/span><\/div>, <div class='tooltip'>FIM_sig_emax<span class='tooltiptext'>It provides the cpp function for FIM for the model ~b1+(b2-b1)*(x^b4)/(x^b4+b3^b4)<\/span><\/div>, <div class='tooltip'>ICA.control<span class='tooltiptext'>The function ICA.control returns a list of ICA control parameters.<\/span><\/div>, <div class='tooltip'>leff<span class='tooltiptext'>Given a vector of initial estimates for the parameters, this function calculates the D-and PA- efficiency of a design \\xi_1 with respect to a design \\xi_2.Usually, \\xi_2 is an optimal design.<\/span><\/div>, <div class='tooltip'>locally<span class='tooltiptext'>Finds locally D-optimal designs for linear and nonlinear models. It should be used when a vector of initial estimates is available for the unknown model parameters. Locally optimal designs may not be efficient when the initial estimates are far away from the true values of the parameters.<\/span><\/div>, <div class='tooltip'>locallycomp<span class='tooltiptext'>Finds compound locally DP-optimal designs that meet the dual goal of parameter estimation and increasing the probability of a particular outcome in a binary response model.A compound locally DP-optimal design maximizes the product of the efficiencies of a design \\xi with respect to D- and average P-optimality, weighted by a pre-defined mixing constant0 \\leq \\alpha \\leq 10 <= \\alpha <= 1.<\/span><\/div>, <div class='tooltip'>meff<span class='tooltiptext'>Given a parameter space for the unknown parameters, this function calculates the D-efficiency of a design \\xi_1 with respect to a design \\xi_2.Usually, \\xi_2 is an optimal design.<\/span><\/div>, <div class='tooltip'>minimax<span class='tooltiptext'>Finds minimax and standardized maximin D-optimal designs for linear and nonlinear models. It should be used when the user assumes the unknown parameters belong to a parameter region \\Theta, which is called ``region of uncertainty'', and the purpose is to protect the experiment from the worst case scenario over \\Theta.<\/span><\/div>, <div class='tooltip'>multiple<span class='tooltiptext'>The 4-parameter Hill model is of the form f(D) = c + \\frac{(d-c)(\\frac{D}{a})^b}{1+(\\frac{D}{a})^b} + \\epsilon, f(D) = c + (d-c)(D/a)^b/(1 + (D/a)^b) + \\epsilon,where \\epsilon \\sim N(0, \\sigma^2)\\epsilon ~ N(0, \\sigma^2), D is the dose level and the predictor,a is the ED50, d is the upper limit of response, c is the lower limit of response and b denotes the Hill constant that control the flexibility in the slope of the response curve. Sometimes, the Hill model is re-parameterized and written as f(x) = \\frac{\\theta_1}{1 + exp(\\theta_2 x + \\theta_3)} + \\theta_4, f(x)= \\theta_1/(1 + exp(\\theta_2*x + \\theta_3)) + \\theta_4, where \\theta_1 = d - c, \\theta_2 = - b, \\theta_3 = b\\log(a)\\theta_3 = b*log(a), \\theta_4 = c, \\theta_1 > 0, \\theta_2 \\neq 0\\theta_2 not equal to 0, and -\\infty < ED50 < \\infty, where x = log(D) \\in [-M, M]x = log(D) belongs to [-M, M] for some sufficiently large value of M. The new form is sometimes called 4-parameter logistic model. The function multiple finds locally multiple-objective optimal designs for estimating the model parameters, the ED50, and the MED, simultaneously. For more details, see Hyun and Wong (2015).<\/span><\/div>, <div class='tooltip'>normal<span class='tooltiptext'>Creates a multivariate normal prior distribution for the unknown parameters as an object of class cprior.<\/span><\/div>, <div class='tooltip'>robust<span class='tooltiptext'>Finds Robust designs or optimal in-average designs for linear and nonlinear models. It is useful when a set of different vectors of initial estimates along with a discrete probability measure are available for the unknown model parameters. It is a discrete version of bayes.<\/span><\/div>, <div class='tooltip'>sens.bayes.control<span class='tooltiptext'>This function returns two lists each corresponds to an implemented integration method for approximating the integrals in the sensitivity (derivative) functions for the Bayesian optimality criteria.<\/span><\/div>, <div class='tooltip'>sens.control<span class='tooltiptext'>It returns some arguments of the nloptr function including the list of control parameters.This function is used to find the maximum of the sensitivity (derivative) function over the design space in order to calculate the efficiency lower bound (ELB).<\/span><\/div>, <div class='tooltip'>sens.minimax.control<span class='tooltiptext'>This function returns a list of control parameters that are used to findthe ``answering set'' for minimax andstandardized maximin designs. The answering set is required to obtain the sensitivity (derivative) function in order to verify the optimality of a given design.<\/span><\/div>, <div class='tooltip'>sensbayes<span class='tooltiptext'>Plots the sensitivity (derivative) function and calculates the efficiency lower bound (ELB) for a given Bayesian design.Let \\boldsymbol{x}x belongs to \\chi that denotes the design space.Based on the general equivalence theorem, a design \\xi^*\\xi* is optimal if and only if the value of the sensitivity (derivative) functionis non-positive for all \\boldsymbol{x}x in \\chi and zero when \\boldsymbol{x}x belongs to the support of \\xi^*\\xi* (be equal to the one of the design points).For an approximate (continuous) design, when the design space is one or two-dimensional, the user can visually verify the optimality of the design by observing thesensitivity plot. Furthermore, the proximity of the design to the optimal design can be measured by the ELB without knowing the latter.<\/span><\/div>, <div class='tooltip'>sensbayescomp<span class='tooltiptext'>This function plot the sensitivity (derivative) function given an approximate (continuous) design and calculate the efficiency lower bound (ELB) for Bayesian DP-optimal designs.Let \\boldsymbol{x}x belongs to \\chi that denotes the design space.Based on the general equivalence theorem, generally, a design \\xi^*\\xi* is optimal if and only if the value of its sensitivity (derivative) functionbe non-positive for all \\boldsymbol{x}x in \\chi and it only reaches zerowhen \\boldsymbol{x}x belong to the support of \\xi^*\\xi* (be equal to one of the design point).Therefore, the user can look at the sensitivity plot and the ELB and decide whether thedesign is optimal or close enough to the true optimal design (ELB tells us that without knowing the latter).<\/span><\/div>, <div class='tooltip'>senslocally<span class='tooltiptext'>It plots the sensitivity (derivative) function of the locally D-optimal criterionat a given approximate (continuous) design and also calculates its efficiency lower bound (ELB) with respectto the optimality criterion.For an approximate (continuous) design, when the design space is one or two-dimensional, the user can visually verify the optimality of the design by observing thesensitivity plot. Furthermore, the proximity of the design to the optimal design can be measured by the ELB without knowing the latter. See, for more details, Masoudi et al. (2017).<\/span><\/div>, <div class='tooltip'>senslocallycomp<span class='tooltiptext'>This function plot the sensitivity (derivative) function given an approximate (continuous) design and calculate the efficiency lower bound (ELB) for locally DP-optimal designs.Let \\boldsymbol{x}x belongs to \\chi that denotes the design space.Based on the general equivalence theorem, generally, a design \\xi^*\\xi* is optimal if and only if the value of its sensitivity (derivative) functionbe non-positive for all \\boldsymbol{x}x in \\chi and it only reaches zerowhen \\boldsymbol{x}x belong to the support of \\xi^*\\xi* (be equal to one of the design point).Therefore, the user can look at the sensitivity plot and the ELB to decide whether thedesign is optimal or close enough to the true optimal design.<\/span><\/div>, <div class='tooltip'>sensminimax<span class='tooltiptext'>It plots the sensitivity (derivative) function of the minimax or standardized maximin D-optimal criterionat a given approximate (continuous) design and also calculates its efficiency lower bound (ELB) with respectto the optimality criterion.For an approximate (continuous) design, when the design space is one or two-dimensional, the user can visually verify the optimality of the design by observing thesensitivity plot. Furthermore, the proximity of the design to the optimal design can be measured by the ELB without knowing the latter. See, for more details, Masoudi et al. (2017).<\/span><\/div>, <div class='tooltip'>sensmultiple<span class='tooltiptext'>This function uses general equivalence theorem to verifythe optimality of a multiple objective optimal design found for the 4-Parameter Hill model and the 4-parameter logistic model.For more details, See Hyun and Wong (2015).<\/span><\/div>, <div class='tooltip'>sensrobust<span class='tooltiptext'>It plots the sensitivity (derivative) function of the robust criterionat a given approximate (continuous) design and also calculates its efficiency lower bound (ELB) with respectto the optimality criterion.For an approximate (continuous) design, when the design space is one or two-dimensional, the user can visually verify the optimality of the design by observing thesensitivity plot. Furthermore, the proximity of the design to the optimal design can be measured by the ELB without knowing the latter. See, for more details, Masoudi et al. (2017).<\/span><\/div>, <div class='tooltip'>skewnormal<span class='tooltiptext'>Creates a multivariate skewed normal prior distribution for the unknown parameters as an object of class cprior.<\/span><\/div>, <div class='tooltip'>student<span class='tooltiptext'>Creates the prior distribution for the parameters as an object of class cprior.<\/span><\/div>, <div class='tooltip'>uniform<span class='tooltiptext'>Creates independent uniform prior distributions for the unknown model parameters as an object of class cprior.<\/span><\/div>","<div class='tooltip'>CEA<span class='tooltiptext'>The algorithm improves an initial start design by considering changes on anattribute-by-attribute basis. By doing this, it tries to minimize theD(B)-error based on a multinomial logit model. This routine is repeated formultiple starting designs.<\/span><\/div>, <div class='tooltip'>Datatrans<span class='tooltiptext'>Transforms the data into the desired data format required by differentestimation packages.<\/span><\/div>, <div class='tooltip'>DBerr<span class='tooltiptext'>Function to calculate the DB-error given a design, and parameter values.<\/span><\/div>, <div class='tooltip'>Decode<span class='tooltiptext'>Transforms a coded design matrix into a design containing character attributelevels, ready to be used in a survey. The frequency of each attribute levelin the design is also included in the output.<\/span><\/div>, <div class='tooltip'>ImpsampMNL<span class='tooltiptext'>This function samples from the posterior distribution using importancesampling, assuming a multivariate (truncated) normal prior distribution and aMNL likelihood.<\/span><\/div>, <div class='tooltip'>LoadData<span class='tooltiptext'>Reads all individual choice data files, created by SurveyAppfunction, from a directory and concatenatesthose files into a single data file. Files containing either \"num\" or \"char\"will be read, with num indicating numeric data and char indicating characterdata. For more information, see output of SurveyApp.<\/span><\/div>, <div class='tooltip'>Modfed<span class='tooltiptext'>The algorithm swaps every profile of an initial start design with candidateprofiles. By doing this, it tries to minimize the D(B)-error, based on amultinomial logit model. This routine is repeated for multiple startingdesigns.<\/span><\/div>, <div class='tooltip'>Profiles<span class='tooltiptext'>Function to generate all possible combinations of attribute levels (i.e. allpossible profiles).<\/span><\/div>, <div class='tooltip'>RespondMNL<span class='tooltiptext'>Function to generate random responses given parameter values and a designmatrix, assuming a MNL model.<\/span><\/div>, <div class='tooltip'>SeqCEA<span class='tooltiptext'>Selects the choice set that minimizes the DB-error when added to an initialdesign, given (updated) parameter values.<\/span><\/div>, <div class='tooltip'>SeqKL<span class='tooltiptext'>Selects the choice set that maximizes the Kullback-Leibler divergence betweenthe prior parameter values and the expected posterior, assuming a MNL model.<\/span><\/div>, <div class='tooltip'>SeqMOD<span class='tooltiptext'>Selects the choice set that minimizes the DB-error when added to an initialdesign, given (updated) parameter values.<\/span><\/div>, <div class='tooltip'>SurveyApp<span class='tooltiptext'>This function starts a shiny application which puts choice sets on screen andsaves the responses. The complete choice design can be provided in advance, orcan be generated sequentially adaptively, or can be a combination of both.<\/span><\/div>","<div class='tooltip'>applyCollineation<span class='tooltiptext'>This function relabels a balanced (t-1)-spread or a covering star St(n, \\mu, t, t_0) of PG(n-1,2) according to the specified collineation matrix.<\/span><\/div>, <div class='tooltip'>checkSpreadEquivalence<span class='tooltiptext'>This function checks the equivalence of two (t-1)-spreads of PG(n-1,2) by comparing their sorted bitstring representations.<\/span><\/div>, <div class='tooltip'>checkSpreadIsomorphism<span class='tooltiptext'>This function checks the isomorphism of two (t-1)-spreads of PG(n-1,2). If they are isomorphic, it returns the list of isomorphism establishing collineations (IECs). The option is provided to enumerate all IECs or to terminate after the first one is found.<\/span><\/div>, <div class='tooltip'>checkStarEquivalence<span class='tooltiptext'>This function checks the equivalence of two balanced covering stars of PG(n-1,2) by comparing sorted bitstring representations.<\/span><\/div>, <div class='tooltip'>checkStarIsomorphism<span class='tooltiptext'>This function checks the isomorphism of two balanced covering stars of PG(n-1,2). If they are isomorphic, it returns the list of isomorphism establishing collineations. The option is provided to enumerate all isomorphism establishing collineations or to terminate after the first one is found.<\/span><\/div>, <div class='tooltip'>getBitstrings<span class='tooltiptext'>This function computes a bitstring representation for a spread or star of PG(n-1,2)<\/span><\/div>, <div class='tooltip'>is.spread<span class='tooltiptext'>Checks whether or not the input is a proper balanced (t-1)-spread of PG(n-1,2).<\/span><\/div>, <div class='tooltip'>is.star<span class='tooltiptext'>Checks whether or not the input is a proper balanced covering star of PG(n-1,2).<\/span><\/div>, <div class='tooltip'>star_to_spread<span class='tooltiptext'>A function that converts a balanced covering star St(n, mu, t, t_0) of PG(n-1,2) to its corresponding balanced ((t-t_0)-1)-spread of PG((n-t_0)-1,2).<\/span><\/div>, <div class='tooltip'>stringtovector<span class='tooltiptext'>Converts a character string representation of a factorial effect in PG(n-1,2) into a binary vector of length n.<\/span><\/div>, <div class='tooltip'>vectortostring<span class='tooltiptext'>Converts a binary vector or matrix in PG(n-1,2) into string-vector/matrix .<\/span><\/div>","<div class='tooltip'>aov.ispd<span class='tooltiptext'>This function performs analysis of variance of data from experiments using an incomplete split-plot design for three types of situations namely (i) blocks are complete with respect to main plot treatments and mainplots are incomplete with respect to subplot treatments, (ii) blocks are incomplete with respect to main plot treatments and mainplots are complete with respect to subplot treatments and (iii) blocks are incomplete with respect to main plot treatments and mainplots are also incomplete with respect to subplot treatments.<\/span><\/div>, <div class='tooltip'>ispd<span class='tooltiptext'>This function generates an incomplete split-plot design with given number of main plot treatments(v1), number of subplot treatments (v2), number of blocks(b) and block size(k). The incomplete split-plot design may be one of the three kinds:(i) blocks are complete with respect to main plot treatments and mainplots are incomplete with respect to subplot treatments, (ii) blocks are incomplete with respect to main plot treatments and mainplots are complete with respect to subplot treatments and (iii) blocks are incomplete with respect to main plot treatments and mainplots are also incomplete with respect to subplot treatments.<\/span><\/div>","<div class='tooltip'>ACER<span class='tooltiptext'>ACER tests (two-sided) if the average cluster effectratio (ACER) is equal to lambda.<\/span><\/div>, <div class='tooltip'>double_rank<span class='tooltiptext'>double_rank returns the two-sided p-value testingFisher's sharp null hypothesis in a cluster-level proportionaltreatment effect model.<\/span><\/div>, <div class='tooltip'>double_rank_CI<span class='tooltiptext'>double_rank_CI returns the two-sided level-alpha confidenceinterval of the proportional treatment effect in a cluster-levelproportional treatment effect model.<\/span><\/div>, <div class='tooltip'>PER<span class='tooltiptext'>PER returns the two-sided p-value testingthe pooled effect ratio equal to lambda_0 in a cluster-randomizedencouragement experiment.<\/span><\/div>, <div class='tooltip'>PER_CI<span class='tooltiptext'>PER_CI returns the two-sided level-alpha confidenceinterval of the pooled effect ratio in a cluster-randomizedencouragement experiment.<\/span><\/div>","<div class='tooltip'>decision.finding<span class='tooltiptext'>This function returns the decision given the decision table, the total number of patients, the number of patients who experienced toxicity and the number of responses.<\/span><\/div>, <div class='tooltip'>example.kb<span class='tooltiptext'>This function serves solely as an example to illustrate how theKeyboard design determines dosing decisions when conducting a single-agenttrial.<\/span><\/div>, <div class='tooltip'>get.boundary.comb.kb<span class='tooltiptext'>This function generates the optimal dose escalation or de-escalation boundaries whenconducting a drug-combination trial with the Keyboard design.<\/span><\/div>, <div class='tooltip'>get.boundary.kb<span class='tooltiptext'>This function generates the optimal dose escalation or de-escalation boundaries whenconducting a single-agent trial with the Keyboard design.<\/span><\/div>, <div class='tooltip'>get.decision.obd.kb<span class='tooltiptext'>This function generates a boundary table and decision matrix for single-agent phase I/II trials using Keyboard design.<\/span><\/div>, <div class='tooltip'>get.decision.obd2.kb<span class='tooltiptext'>This function automatically generates a boundary table and a decision table for single-agent Phase I/II trials using Keyboard design.<\/span><\/div>, <div class='tooltip'>get.oc.comb.kb<span class='tooltiptext'>This function generates the operating characteristics of the Keyboard design fordrug-combination trials.<\/span><\/div>, <div class='tooltip'>get.oc.kb<span class='tooltiptext'>This function generates the operating characteristics of the Keyboard design forsingle-agent trials.<\/span><\/div>, <div class='tooltip'>get.oc.obd.kb<span class='tooltiptext'>This function generates operating characteristics to find the optimal biological dose (OBD).<\/span><\/div>, <div class='tooltip'>get.oc.obd2.kb<span class='tooltiptext'>This function reports operating characteristics for optimal biological dose (OBD) finding automatically.<\/span><\/div>, <div class='tooltip'>next.comb.kb<span class='tooltiptext'>This function determines the dose combination for the next cohort of patients indrug-combination trials.<\/span><\/div>, <div class='tooltip'>plot.kb<span class='tooltiptext'>This function plots the objects returned by the other functions in the Keyboard package.<\/span><\/div>, <div class='tooltip'>select.mtd.comb.kb<span class='tooltiptext'>This function selects the maximum tolerated dose (MTD) after the drug-combination trial iscompleted.<\/span><\/div>, <div class='tooltip'>select.mtd.kb<span class='tooltiptext'>This function selects the maximum tolerated dose (MTD) after the single-agent trial iscompleted.<\/span><\/div>, <div class='tooltip'>select.obd.kb<span class='tooltiptext'>This function selects the optimal biological dose (OBD) at the end of a single-agent phase I/II trial.<\/span><\/div>, <div class='tooltip'>summary.kb<span class='tooltiptext'>This function generates a descriptive summary for objects returned by other functions.<\/span><\/div>","<div class='tooltip'><span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>cfderiv<span class='tooltiptext'>Auto-constructs Frechet derivative of D-criterion at M(\\xi, \\beta) and in direction M(\\xi_x, \\beta) where M is Fisher information matrix, \\beta is vector of parameters, \\xi is the interested design and \\xi_x is a unique design which has only a point x. The constructed Frechet derivative is an function with argument x.%Auto-constructs derivative function of D-criterion based on general equivalence theorem (Kiefer, 1974). %models that their response variables are independent and can be writen as a function of explanatory variables and parameters. This function must be continuous and differentiable with respect to the models parameters. In addition, distribution of response variable must belong to the Natural exponential family. See 'Note' for more details.<\/span><\/div>, <div class='tooltip'>cfisher<span class='tooltiptext'>Auto-constructs Fisher information matrix for nonlinear and generalized linear models as two functions.%models that their response variables are independent and can be writen as a function of explanatory variables and parameters. This function must be continuous and differentiable with respect to the models parameters. In addition, distribution of response variable must belong to the Natural exponential family. See 'Note' for more details.<\/span><\/div>, <div class='tooltip'>eff<span class='tooltiptext'>Calculates the D-effficiency of design \\xi_1 respect to design \\xi_2 with arbitrary precision.<\/span><\/div>, <div class='tooltip'>ldemax<span class='tooltiptext'>Finds Locally D-optimal designs for Emax regression model which is defined as E(y) = a+bx/(x+c) with Var(y) = \\sigma^2, where a, b and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldexpdose<span class='tooltiptext'>Finds Locally D-optimal designs for Exponential dose-response model which is defined as E(y) = a+b\\exp(x/c) with Var(y) = \\sigma^2, where a, b and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldiq<span class='tooltiptext'>%A concise (1-5 lines) description of what the function does. ~~Finds Locally D-optimal designs for Inverse Quadratic regression model which is defined as E(y)=ax/(b+x+cx^2) or E(y)=x/(a+bx+cx^2) with Var(y) = \\sigma^2, where a, b, c and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldlogistic<span class='tooltiptext'>%Finds Locally D-optimal designs for Logistic and Logistic dose-response models which are defined as E(y) = (1/(exp(-a - b * x)+1)), V(y) = (1/(exp(-a - b * x) + 1)) * (1 - (1/(exp(-a - b * x) + 1))) and E(y) = (1/(exp(-b * (x - a)) + 1)), V(y) = (1/(exp(-b * (x - a)) + 1)) * (1 - (1/(exp(-b * (x - a)) + 1))), respectively.Finds Locally D-optimal designs for Logistic and Logistic dose-response models which are defined as E(y) = 1/(1+\\exp(-a-bx)) and E(y) = 1/(1+\\exp(-b(x-a))) with Var(y) = E(y)(1-E(y)), respectively, where a and b are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldloglin<span class='tooltiptext'>Finds Locally D-optimal designs for Log-linear regression model which is defined as E(y) = a+b\\log(x+c) with Var(y) = \\sigma^2, where a, b, c and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldmm<span class='tooltiptext'>Finds Locally D-optimal designs for Michaelis-Menten model which is defined as E(y) = (ax)/(1+bx) or E(y) = (ax)/(b+x) or E(y) = \"x/(a+bx) with Var(y) = \\sigma^2, where a, b and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldnbinom<span class='tooltiptext'>%Finds Locally D-optimal designs for Negative binomial regression model which is defined as E(y) = a * exp(-b * x) and %V(y) = sigma^2 * a * exp(-b * x) * (1 + (a * exp(-b * x)/theta)), where sigma is an unknown consonant and theta is .Finds Locally D-optimal designs for Negative Binomial regression model which is defined as E(y) = \\lambda(x) with Var(y) = \\sigma^2\\lambda(x)(1+(\\lambda(x)/\\theta)), where y ~ NB(\\theta, \\lambda(x)), \\lambda(x) = a\\exp(-bx) and a, b and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldpoisson<span class='tooltiptext'>Finds Locally D-optimal designs for Poisson and Poisson dose-response models which are defined as E(y) = \\exp(a+bx) and E(y) = a\\exp(-bx) with Var(y) = E(y), respectively, where a and b are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldrichards<span class='tooltiptext'>Finds Locally D-optimal designs for Richards regression model which is defined as E(y) = a/(1+b\\exp(-\\lambda*x))^h with Var(y) = \\sigma^2, where a, b, \\lambda, h and \\sigma are unknown parameters.<\/span><\/div>, <div class='tooltip'>ldweibull<span class='tooltiptext'>Finds Locally D-optimal designs for Weibull regression model which is defined as E(y) = a-b\\exp(-\\lambda*x^h) with Var(y) = \\sigma^2, where a, b, \\lambda, h and \\sigma are unknown parameters.<\/span><\/div>","<div class='tooltip'>AvgAbsCor<span class='tooltiptext'>AvgAbsCor returns the average absolute correlation of a matrix<\/span><\/div>, <div class='tooltip'>dij<span class='tooltiptext'>dij returns the inter-site distance of two design points of an LHD<\/span><\/div>, <div class='tooltip'>exchange<span class='tooltiptext'>exchange returns a new design matrix after two randomly selected elements are switched from a user-defined column<\/span><\/div>, <div class='tooltip'>FastMmLHD<span class='tooltiptext'>FastMmLHD returns a n by k maximin distance LHD matrix generated by the construction method of Wang, L., Xiao, Q., and Xu, H. (2018)<\/span><\/div>, <div class='tooltip'>GA<span class='tooltiptext'>GA returns a n by k LHD matrix generated by genetic algorithm (GA)<\/span><\/div>, <div class='tooltip'>GLP<span class='tooltiptext'>GLP returns a n by k design matrix generated by good lattice point (GLP)<\/span><\/div>, <div class='tooltip'>LaPSO<span class='tooltiptext'>LaPSO returns a n by k LHD matrix generated by particle swarm optimization algorithm (PSO)<\/span><\/div>, <div class='tooltip'>MaxAbsCor<span class='tooltiptext'>MaxAbsCor returns the maximum absolute correlation of a matrix<\/span><\/div>, <div class='tooltip'>MaxProCriterion<span class='tooltiptext'>MaxProCriterion returns the maximum projection criterion of an LHD<\/span><\/div>, <div class='tooltip'>OA2LHD<span class='tooltiptext'>OA2LHD transfers an OA into an LHD with corresponding size<\/span><\/div>, <div class='tooltip'>OASA<span class='tooltiptext'>OASA returns an LHD matrix generated by orthogonal-array-based simulated annealing algorithm (OASA)<\/span><\/div>, <div class='tooltip'>OLHD.B2001<span class='tooltiptext'>OLHD.B2001 returns a n by k orthogonal Latin hypercube design generated by the construction method of Butler (2001)<\/span><\/div>, <div class='tooltip'>OLHD.C2007<span class='tooltiptext'>OLHD.C2007 returns a 2^m+1 by m+{m-1 \\choose 2} orthogonal Latin hypercube design generated by the construction method of Cioppa and Lucas (2007)<\/span><\/div>, <div class='tooltip'>OLHD.L2009<span class='tooltiptext'>OLHD.L2009 returns a n^2 by 2fp orthogonal Latin hypercube design generated by the construction method of Lin et al. (2009)<\/span><\/div>, <div class='tooltip'>OLHD.S2010<span class='tooltiptext'>OLHD.S2010 returns a r2^{C+1}+1 or r2^{C+1} by 2^C orthogonal Latin hypercube design generated by the construction method of Sun et al. (2010)<\/span><\/div>, <div class='tooltip'>OLHD.Y1998<span class='tooltiptext'>OLHD.Y1998 returns a 2^m+1 by 2m-2 orthogonal Latin hypercube design generated by the construction method of Ye (1998)<\/span><\/div>, <div class='tooltip'>phi_p<span class='tooltiptext'>phi_p returns the phi_p criterion of an LHD<\/span><\/div>, <div class='tooltip'>rLHD<span class='tooltiptext'>rLHD returns a random n by k Latin hypercube design matrix<\/span><\/div>, <div class='tooltip'>SA<span class='tooltiptext'>SA returns a n by k LHD matrix generated by simulated annealing algorithm (SA)<\/span><\/div>, <div class='tooltip'>SA2008<span class='tooltiptext'>SA2008 returns a n by k LHD matrix generated by simulated annealing algorithm with multi-objective optimization approach<\/span><\/div>, <div class='tooltip'>SLHD<span class='tooltiptext'>SLHD returns a n by k LHD matrix generated by improved two-stage algorithm<\/span><\/div>, <div class='tooltip'>WT<span class='tooltiptext'>WT returns a matrix after implementing the Williams transformation<\/span><\/div>","<div class='tooltip'>CandPoints<span class='tooltiptext'>Quickly generate random candidate design points for different types of factors which can be used as an input to the MaxProAugment function.<\/span><\/div>, <div class='tooltip'>MaxPro<span class='tooltiptext'>Find the locally optimal maximum projection (MaxPro) design in the neighborhood of a given initial design for continuous factors.<\/span><\/div>, <div class='tooltip'>MaxProAugment<span class='tooltiptext'>Select the best set of design points to augment a given design matrix by optimizing the maximum projection (MaxPro) criterion sequentially. When multiple types of factors exist, all continuous factors should be placed as the leftmost columns in the design matrix, discrete numeric factors should be placed as middle columns, and all nominal factors should be placed at the rightmost columns in the design matrix. The ordinal factors, if any, should be pre-converted into discrete numeric factors through the scoring method (see, e.g., Wu and Hamada 2009, Section 14.10). This function can also be used for creating a nonadaptive sequential design, designs to run in batches, validation runs, designs in constrained regions, and nested designs.<\/span><\/div>, <div class='tooltip'>MaxProLHD<span class='tooltiptext'>Generate the maximum projection (MaxPro) Latin hypercube design for continuous factors based on a simulated annealing algorithm. If nominal, discrete numeric, or ordinal factors exist, please see the function MaxProQQ.<\/span><\/div>, <div class='tooltip'>MaxProMeasure<span class='tooltiptext'>Calculate the maximum projection (MaxPro) criterion measure for a given design.<\/span><\/div>, <div class='tooltip'>MaxProQQ<span class='tooltiptext'>Generate the maximum projection (MaxPro) design with quantitative and qualitative (QQ) factors. The quantitative factors can be continuous factors and/or discrete numeric factors, and the qualitative factors can be nominal factors and/or ordinal factors. If all factors are continuous, please use MaxProLHD directly. To use this function, first convert all the ordinal factors into discrete numeric factors through the scoring method (see, e.g., Wu and Hamada 2009, Section 14.10). That is, if the ordinal factor has levels, \"poor\", \"fair\" and \"good\", then depending on the nature of the classification, the experimenter can choose some discrete numeric levels such as (1,2,3) or (1,4,5) to represent the three ordinal levels. Also, all columns of the continuous and discrete numeric factors should be standardized into the unit range of [0,1]The users need to provide their desirable sub-design matrix for the nominal factors as part of the initial design input to this function. This function only optimizes the sub-design matrix of the continuous and discrete numeric factors as well as their row orders in joining with the pre-determined sub-design matrix of the nominal factors.<\/span><\/div>, <div class='tooltip'>MaxProRunOrder<span class='tooltiptext'>Given a computer experimental design matrix, this function searches for an optimal run (row) order based on the maximum projection (MaxPro) criterion. This optimal order enables the given design to be run in a sequential manner: when terminated at any step, the previous design points form a nearly optimal subset based on the MaxPro criterion.<\/span><\/div>","<div class='tooltip'>%ARC%<span class='tooltiptext'>All Row Combinations (ARC) function take all combination of rows of 2 matrices and bind them together<\/span><\/div>, <div class='tooltip'>addSchemes<span class='tooltiptext'>add user defined scheme to an existing SetOfSchemes-class or extend an existing set of schemes object with additional schemes<\/span><\/div>, <div class='tooltip'>check_scheme_exactNumberObsPerTimePoint<span class='tooltiptext'>check the number of observations per time points is equal specified value<\/span><\/div>, <div class='tooltip'>check_scheme_minObsPerTimePoint<span class='tooltiptext'>check the mimimum observations per time points is above a specified value<\/span><\/div>, <div class='tooltip'>check_subject_maxConsecSamples<span class='tooltiptext'>check the maximum of consecutive samples per subject falls below the specified value<\/span><\/div>, <div class='tooltip'>construct2CompModel<span class='tooltiptext'>construct a 2 compartmental PkModel-class by providing parameters and dosing info<\/span><\/div>, <div class='tooltip'>constructSetOfSchemes<span class='tooltiptext'>construct user defined SetOfSchemes-class<\/span><\/div>, <div class='tooltip'>doAllSchemeChecks<span class='tooltiptext'>check whether either a 1 subject or multiple subject microsampling scheme meets imposed constraints<\/span><\/div>, <div class='tooltip'>extractByRank<span class='tooltiptext'>extract a timepoint or Scheme choice by its rank<\/span><\/div>, <div class='tooltip'>fastRankSchemes<span class='tooltiptext'>generic function to calulate a ranking-slot<\/span><\/div>, <div class='tooltip'>flattenSetOfSchemes<span class='tooltiptext'>Transform 3 way array to 2 way array<\/span><\/div>, <div class='tooltip'>formatTimePoints<span class='tooltiptext'>Format time points as a set<\/span><\/div>, <div class='tooltip'>get2ComptModelCurve<span class='tooltiptext'>provides solution of two compartmental pharmacodynamic model at specified time points<\/span><\/div>, <div class='tooltip'>getAllTimeOptions<span class='tooltiptext'>generate all possible time options from eligible time points and number of samples per time interval ( time zone )<\/span><\/div>, <div class='tooltip'>getCoeffVariationError<span class='tooltiptext'>generic function to extract coeffVariationError slot<\/span><\/div>, <div class='tooltip'>getCombinationsWithMaxNRepetitions<span class='tooltiptext'>get all combinations with a maximum number of repetitions<\/span><\/div>, <div class='tooltip'>getConstraintsExample<span class='tooltiptext'>get a minimal example of a constraint data frame<\/span><\/div>, <div class='tooltip'>getCorrelationMatrix<span class='tooltiptext'>generic function to extract the correlationMatrix-slot<\/span><\/div>, <div class='tooltip'>getData<span class='tooltiptext'>generic function to extract the .Data-slot<\/span><\/div>, <div class='tooltip'>getDosingInfo<span class='tooltiptext'>generic function to extract dosingInfo-slot<\/span><\/div>, <div class='tooltip'>getExampleData<span class='tooltiptext'>generate an mimimal example of a Pk data without a model<\/span><\/div>, <div class='tooltip'>getExampleObjective<span class='tooltiptext'>example objective function for rankObject<\/span><\/div>, <div class='tooltip'>getExamplePkCurve<span class='tooltiptext'>example of 1 pk curve to be used to test pkCurveStat_[function]<\/span><\/div>, <div class='tooltip'>getExamplePkModel<span class='tooltiptext'>get minimal example of PkModel-class<\/span><\/div>, <div class='tooltip'>getExamplePkModelRange<span class='tooltiptext'>get minimal example of PkModelRange-class<\/span><\/div>, <div class='tooltip'>getExampleSetOfSchemes<span class='tooltiptext'>get a minimal example of a set of schemes object<\/span><\/div>, <div class='tooltip'>getExampleSetOfTimePoints<span class='tooltiptext'>get a minimal example set of time points to test functions with<\/span><\/div>, <div class='tooltip'>getExampleTimeData<span class='tooltiptext'>generate example PkData object to be used in example rankTimePoints<\/span><\/div>, <div class='tooltip'>getExampleTimeZones<span class='tooltiptext'>working example time zone dataframe to use in examples<\/span><\/div>, <div class='tooltip'>getIndividualParameters<span class='tooltiptext'>sample subject specific parameters to input in pharmacodynamic modelparamaters are sample from a log-normal distribution<\/span><\/div>, <div class='tooltip'>getMMCurve<span class='tooltiptext'>calculate Michealis-Menten relation between x and velocity and rate<\/span><\/div>, <div class='tooltip'>getModelFunction<span class='tooltiptext'>generic function to extract modelFunction slot from S4-class object<\/span><\/div>, <div class='tooltip'>getNames<span class='tooltiptext'>generic function extract the names of an S4-object<\/span><\/div>, <div class='tooltip'>getNSchemes<span class='tooltiptext'>generic function to extract nSchemes-slot<\/span><\/div>, <div class='tooltip'>getNSubjects<span class='tooltiptext'>generic function to extract nSubjects-slot<\/span><\/div>, <div class='tooltip'>getParameters<span class='tooltiptext'>generic function to extract parameter-slot<\/span><\/div>, <div class='tooltip'>getPkData<span class='tooltiptext'>simulate PkData-class from PkModel-class<\/span><\/div>, <div class='tooltip'>getPkModel<span class='tooltiptext'>generic function extract a PkModel-class<\/span><\/div>, <div class='tooltip'>getPkModelArticle<span class='tooltiptext'>reproduce the example of the article of Helen Barnet et al.<\/span><\/div>, <div class='tooltip'>getPkModels<span class='tooltiptext'>Generate all possible PkModel-class from PkModelRange-class combination of ranges<\/span><\/div>, <div class='tooltip'>getRanking<span class='tooltiptext'>generic function to extract the ranking-slot<\/span><\/div>, <div class='tooltip'>getSetOfSchemes<span class='tooltiptext'>Generate a SetOfSchemes-class object of speficified dimensions ( subjects, observations per t) for a given set of time points which meets user specified constraints<\/span><\/div>, <div class='tooltip'>getTimeChoicePerformance<span class='tooltiptext'>estimate the distance between population average an average over sample datasets with given time points (zero point included)<\/span><\/div>, <div class='tooltip'>getTimePoints<span class='tooltiptext'>generic function to extract timePoints-slot<\/span><\/div>, <div class='tooltip'>getTopNRanking<span class='tooltiptext'>extract the top n rankings as numeric vector<\/span><\/div>, <div class='tooltip'>oneCompartmentOralModel<span class='tooltiptext'>solution of one compartmental oral administration modelonly use one set of parameters, times can input can be an numeric array<\/span><\/div>, <div class='tooltip'>pkCurveStat_auc<span class='tooltiptext'>implemented statistics: auc area under the curve , between first and last time points cMax maximum concentration tMax time at maximum concentration<\/span><\/div>, <div class='tooltip'>pkCurveStat_cMax<span class='tooltiptext'>implemented statistics: auc area under the curve , between first and last time points cMax maximum concentration tMax time at maximum concentration<\/span><\/div>, <div class='tooltip'>pkCurveStat_tMax<span class='tooltiptext'>implemented statistics: auc area under the curve , between first and last time points cMax maximum concentration tMax time at maximum concentration<\/span><\/div>, <div class='tooltip'>plotAverageRat<span class='tooltiptext'>plot plasma concentration for average individual (i.e average parameter values) in function of dose at time zero<\/span><\/div>, <div class='tooltip'>plotMMCurve<span class='tooltiptext'>plot Michealis-Menten curve for either capacity dependent absorption or clearance<\/span><\/div>, <div class='tooltip'>plotMMKinetics<span class='tooltiptext'>plot MM kinetics of both absorption and clearance<\/span><\/div>, <div class='tooltip'>plotObject<span class='tooltiptext'>generic function to plot an object<\/span><\/div>, <div class='tooltip'>rankObject<span class='tooltiptext'>generic function to calulate a ranking-slot<\/span><\/div>, <div class='tooltip'>rankObjectWithRange<span class='tooltiptext'>Rank a SetOfSchemes-class or a SetOfTimePoints object using data generated per scneario defined by PkModelRange-class<\/span><\/div>, <div class='tooltip'>runMicrosamplingDesignApp<span class='tooltiptext'>Run the PkPlotApp in a new tab in your browser<\/span><\/div>, <div class='tooltip'>setCoeffVariationError<-<span class='tooltiptext'>replace coeffVariationError-slot<\/span><\/div>, <div class='tooltip'>setCorrelationMatrix<-<span class='tooltiptext'>replace correlationMatrix-slot<\/span><\/div>, <div class='tooltip'>setDosingInfo<-<span class='tooltiptext'>replace dosingInfo-slot<\/span><\/div>, <div class='tooltip'>setModelToAverageRat<span class='tooltiptext'>get a model with all variances to zero<\/span><\/div>, <div class='tooltip'>setParameters<-<span class='tooltiptext'>replace parameters-slot<\/span><\/div>, <div class='tooltip'>setRanking<-<span class='tooltiptext'>replace ranking-slot<\/span><\/div>, <div class='tooltip'>setTimePoints<-<span class='tooltiptext'>generic function to replace timePoints-slot<\/span><\/div>, <div class='tooltip'>subsetOnTimePoints<span class='tooltiptext'>generic function to subset the timePoints-slot and generate an object of the same class<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>crvtave<span class='tooltiptext'>This function calls the function Vertcen which uses Piepel's (1988) fortran code (cnvrt) for generating extreme vertices and centroids of linearly constrained mixture experimental regions.<\/span><\/div>, <div class='tooltip'>cubic<span class='tooltiptext'>Creates cubic terms that are used by the function MixModel when fitting model (3)<\/span><\/div>, <div class='tooltip'>DesignPoints<span class='tooltiptext'>This function plots design points and or constraints in the simplex mixture space. It calls the function MixturePlot that does the actual plotting.<\/span><\/div>, <div class='tooltip'>EffPlot<span class='tooltiptext'>This function makes effect plots using the Cox or Piepel directions in constrained mixture space.<\/span><\/div>, <div class='tooltip'>Eflags<span class='tooltiptext'>This function loads and runs the compiled fortran code cnvrt and prints error messages. cnvrt is Piepel's 1988 JQT fortran code for extreme vertices designs.<\/span><\/div>, <div class='tooltip'>Fillv<span class='tooltiptext'>This function creates interior points in a mixture design by averaging all possible pairs of design points. It duplicates SAS macro adxfill.<\/span><\/div>, <div class='tooltip'>MixModel<span class='tooltiptext'>This function fits mixture models (1)-(4) and mixture process models (5)-(6)described in Lawson and Willden(2015) \"Mixture Experiments in R, using mixexp\", Journal Statistical Software http://www/jstatsoft.org/, and prints the correct R square and standard errors of model coefficients.<\/span><\/div>, <div class='tooltip'>MixturePlot<span class='tooltiptext'>This function makes contour plots in the simplex mixture space, it also can draw constraint lines and show design points.<\/span><\/div>, <div class='tooltip'>ModelEff<span class='tooltiptext'>This function makes effect plots using the Cox or Piepel directions in constrained mixture space.<\/span><\/div>, <div class='tooltip'>ModelPlot<span class='tooltiptext'>This function makes contour plots in the simplex mixture space. It also can draw constraint lines and zoom in on pseudo component region.<\/span><\/div>, <div class='tooltip'>Nrows<span class='tooltiptext'>This function loads and runs the compiled fortran code cnvrt. cnvrt is Piepel's 1988 JQT fortran code for extreme vertices designs.<\/span><\/div>, <div class='tooltip'>SCD<span class='tooltiptext'>This function creates simplex centroid designs in unconstrained mixture experiment space.<\/span><\/div>, <div class='tooltip'>SLD<span class='tooltiptext'>This function creates simplex lattice designs in unconstrained mixture experiment space.<\/span><\/div>, <div class='tooltip'>Vertcen<span class='tooltiptext'>This function loads and runs the compiled fortran code cnvrt. cnvrt is Piepel's 1988 JQT fortran code for extreme vertices designs.<\/span><\/div>, <div class='tooltip'>Xvert<span class='tooltiptext'>This function calls the function crvtave to create an extreme vertices design in a constrained mixture space. If there are only three factors the function DesignPoints is called to plot the results.<\/span><\/div>","<div class='tooltip'>mkssd<span class='tooltiptext'>mkssd is a package that generates efficient balanced non-aliased multi-level k-circulant supersaturated designs by interchanging the elements of the generator vector. The package tries to generate a supersaturated design that has chisquare efficiency more than user specified efficiency level (mef). The package also displays the progress of generation of an efficient multi-level k-circulant design through a progress bar. The progress of 100 per cent means that one full round of interchange is completed. More than one full round (typically 4-5 rounds) of interchange may be required for larger designs.<\/span><\/div>","<div class='tooltip'>cpf2<span class='tooltiptext'>Combine Pareto front designs of 2 criteria<\/span><\/div>, <div class='tooltip'>cpf3<span class='tooltiptext'>Combine Pareto front designs of 3 criteria<\/span><\/div>, <div class='tooltip'>LHD<span class='tooltiptext'>Generate a random Latin Hypercube design<\/span><\/div>, <div class='tooltip'>md<span class='tooltiptext'>Minimum distance between any two points in the design<\/span><\/div>, <div class='tooltip'>miM<span class='tooltiptext'>The miniMax criterion baesd on an approximate fill distance measure<\/span><\/div>, <div class='tooltip'>miMLHD<span class='tooltiptext'>Generate the optimal Latin Hypercube Design based on the miniMax criterion.<\/span><\/div>, <div class='tooltip'>Mm<span class='tooltiptext'>Computer the approximate Maximin Criterion for a design.<\/span><\/div>, <div class='tooltip'>MmLHD<span class='tooltiptext'>Generate the optimal Maximin Latin Hypercube Design.<\/span><\/div>, <div class='tooltip'>mp<span class='tooltiptext'>Computer the MaxPro Criterion for a design.<\/span><\/div>, <div class='tooltip'>mpLHD<span class='tooltiptext'>Generate the optimal MaxPro Latin Hypercube Design.<\/span><\/div>, <div class='tooltip'>pfMm<span class='tooltiptext'>Generate the Pareto front for the optimal Latin Hypercube Designs based on both the Maximin and miniMax criteria.<\/span><\/div>, <div class='tooltip'>pfMp<span class='tooltiptext'>Generate the Pareto front for optimal Latin Hypercube Designs based on both the Maximin and the MaxPro criteria.<\/span><\/div>, <div class='tooltip'>pfMpm<span class='tooltiptext'>Generate the Pareto front for the optimal Latin Hypercube Designs based on the Maximin, MaxPro and miniMax criteria.<\/span><\/div>, <div class='tooltip'>pfpm<span class='tooltiptext'>Generate the Pareto front for the optimal Latin Hypercube Designs based on both the MaxPro and miniMax criteria.<\/span><\/div>","<div class='tooltip'>mxkssd<span class='tooltiptext'>mxkssd is a package that generates efficient balanced mixed-level k-circulant supersaturated designs by interchanging the elements of the generator vector. The package tries to generate a supersaturated design that has EfNOD efficiency more than user specified efficiency level (mef). The package also displays the progress of generation of an efficient mixed-level k-circulant design through a progress bar. The progress of 100 per cent means that one full round of interchange is completed. More than one full round (typically 4-5 rounds) of interchange may be required for larger designs.<\/span><\/div>","<div class='tooltip'>negint2<span class='tooltiptext'>This will calculate the mean exposure under different scenarios: 2: fixed follow-up with drop-out, 3: variable follow-up with a maximum (maxfu), 4: variable follow-up with a maximum and drop-out<\/span><\/div>, <div class='tooltip'>ynegbinompower<span class='tooltiptext'>This will calculate the power for the negative binomial distribution for the 2-sample case under different follow-up scenarios: 1: fixed follow-up, 2: fixed follow-up with drop-out, 3: variable follow-up with a minimum fu and a maximum fu, 4: variable follow-up with a minimum fu and a maximum fu and drop-out.<\/span><\/div>, <div class='tooltip'>ynegbinompowersim<span class='tooltiptext'>This will calculate the power for the negative binomial distribution for the 2-sample case under different follow-up scenarios: 1: fixed follow-up, 2: fixed follow-up with drop-out, 3: variable follow-up with a minimum fu and a maximum fu, 4: variable follow-up with a minimum fu and a maximum fu and drop-out.<\/span><\/div>, <div class='tooltip'>ynegbinomsize<span class='tooltiptext'>This will calculate the sample size for the negative binomial distribution for the 2-sample case under different follow-up scenarios: 1: fixed follow-up, 2: fixed follow-up with drop-out, 3: variable follow-up with a minimum fu and a maximum fu, 4: variable follow-up with a minimum fu and a maximum fu and drop-out.<\/span><\/div>","<div class='tooltip'>achievable.se.sp<span class='tooltiptext'>For a retrospective study design, given a prevalence value,produce a plot displaying the achievable contours ofeither predictive values or NNT values.The calculation uses the \"contra-Bayes\" theorem, sesp.from.pv.<\/span><\/div>, <div class='tooltip'>argmin<span class='tooltiptext'>Return the index minimizing distance from v to target.<\/span><\/div>, <div class='tooltip'>binom.confint<span class='tooltiptext'>Exact confidence intervals for a binomial proportion parameter.<\/span><\/div>, <div class='tooltip'>NNT.from.pv<span class='tooltiptext'>Compute NNT values from predictive values.<\/span><\/div>, <div class='tooltip'>NNT.from.sesp<span class='tooltiptext'>Compute NNT values from sensitivity aand specificity.<\/span><\/div>, <div class='tooltip'>NNT.to.pv<span class='tooltiptext'>Convert between (NNTpos, NNTneg) and (PPV, NPV).<\/span><\/div>, <div class='tooltip'>NNT.to.sesp<span class='tooltiptext'>Compute sensitivity aand specificity from NNT values.<\/span><\/div>, <div class='tooltip'>NNTintervalsProspective<span class='tooltiptext'>Produce Bayesian and classical intervals for NNT fromobservations in a prospective study.Useful for \"anticipated results\" when designing a study,The setting: patients will be tested immediately, and followed to determine the BestToTreat/BestToWait classification.as well as analyzing study results.There were (or will be) Npositives patients with a positive test, Nnegatives with a negative test.The observed NNTs in each group were (or will be) NNTpos and NNTneg.<\/span><\/div>, <div class='tooltip'>NNTintervalsRetrospective<span class='tooltiptext'>Bayes predictive intervals for sensitivity, specificity, NNTpos and NNTnegin a case-control retrospective study.<\/span><\/div>, <div class='tooltip'>pv.from.NNT<span class='tooltiptext'>Convert between (NNTpos, NNTneg) and (PPV, NPV).<\/span><\/div>, <div class='tooltip'>pv.from.sesp<span class='tooltiptext'>Computes predictive values from sensitivity and specificity.<\/span><\/div>, <div class='tooltip'>pv.to.NNT<span class='tooltiptext'>Compute NNT values from predictive values.<\/span><\/div>, <div class='tooltip'>pv.to.sesp<span class='tooltiptext'>Computes sensitivity and specificity from predictive values.<\/span><\/div>, <div class='tooltip'>ROCplots<span class='tooltiptext'>A variety of ROC-related plots for a binary target and a single continuous predictor.<\/span><\/div>, <div class='tooltip'>run<span class='tooltiptext'>Run a shiny app for this package.<\/span><\/div>, <div class='tooltip'>runCombinePlots<span class='tooltiptext'>Run a shiny app connecting a visual scale for NNT quantities and a \"contra-Bayes\" plot for mapping from predictive values to sensitivity/specificity (Bayes theorem in reverse).<\/span><\/div>, <div class='tooltip'>runElicit<span class='tooltiptext'>Run a shiny app outlining the process of specifying a design for a biomarker validation study.<\/span><\/div>, <div class='tooltip'>sesp.from.NNT<span class='tooltiptext'>Compute sensitivity aand specificity from NNT values.<\/span><\/div>, <div class='tooltip'>sesp.from.pv<span class='tooltiptext'>Computes sensitivity and specificity from predictive values.<\/span><\/div>, <div class='tooltip'>sesp.from.pv.feasible<span class='tooltiptext'>Computes sensitivity and specificity from predictive values.<\/span><\/div>, <div class='tooltip'>sesp.to.NNT<span class='tooltiptext'>Compute NNT values from sensitivity aand specificity.<\/span><\/div>, <div class='tooltip'>sesp.to.pv<span class='tooltiptext'>Computes predictive values from sensitivity and specificity.<\/span><\/div>","<div class='tooltip'>nparsurv_test<span class='tooltiptext'>The nparsurv_test function calculates the test statistics and the p-values as described in 'Nonparametric Methods for Factorial Designs with Censored Data'by Akritas and Brunner.<\/span><\/div>","<div class='tooltip'>npIntFactRep<span class='tooltiptext'>Nonparametric aligned rank tests for interactions in two-way factorial designs with repeated measures (e.g., Beasley & Zumbo, 2006; Higgings & Tashtoush, 1994). Five ANOVAs are reported: (1) PARAMETRIC on the original data, (2) on the interaction alignments as a CHECK, (3) on the aligned REGULAR, (4) FRIEDMAN, and (5) KOCH ranks. In the rank tests, only the resulting values for the interaction are relevant.<\/span><\/div>","<div class='tooltip'>nppbib<span class='tooltiptext'>Implements a nonparametric statistical test for rank or score data from partially-balanced incomplete block-design experiments.<\/span><\/div>","<div class='tooltip'>combinations<span class='tooltiptext'>combinations enumerates the possible combinations of a specified size from the elements of a vector.<\/span><\/div>, <div class='tooltip'>OBsProb<span class='tooltiptext'>Objective model posterior probabilities and marginal factor posterior probabilitiesfrom Bayesian screening experiments according to Consonni and Deldossi procedure.<\/span><\/div>, <div class='tooltip'>OMD<span class='tooltiptext'>Optimal follow-up experiments to discriminate between competing models. The extra-runs are derived from the maximization of the objective model discrimination criterion represented by a weighted averageof Kullback-Leibler divergences between all possible pairs of rival models<\/span><\/div>","<div class='tooltip'>od.1<span class='tooltiptext'>The optimal design of individual randomized controlled trials (RCTs) is to choose the sample allocation that minimizes the variance of treatment effect under fixed budget and cost structure. The optimal design parameter is the proportion of individuals to be assigned to treatment (p).<\/span><\/div>, <div class='tooltip'>od.2<span class='tooltiptext'>The optimal design of two-level cluster randomized trials (CRTs) is to choose the sample allocation that minimizes the variance of treatment effect under fixed budget and cost structure. The optimal design parameters include the level-1 sample size per level-2 unit (n) and the proportion of level-2 clusters/groups to be assigned to treatment (p). This function solves the optimal n and/or p with and without constraints.<\/span><\/div>, <div class='tooltip'>od.2m<span class='tooltiptext'>The optimal design of two-level multisite randomized trials (MRTs) is to choose the sample allocation that minimizes the variance of a treatment effect under a fixed budget and cost structure. The optimal design parameters include the level-one sample size per site (n) and the proportion of level-one unit to be assigned to treatment (p). This function solves the optimal n and/or p with and without constraints.<\/span><\/div>, <div class='tooltip'>od.3<span class='tooltiptext'>The optimal design of three-level cluster randomized trials (CRTs) is to choose the sample allocation that minimizes the variance of treatment effect under fixed budget and cost structure. The optimal design parameters include the level-1 sample size per level-2 unit (n), the level-2 sample size per level-3 unit (J), and the proportion of level-3 clusters/groups to be assigned to treatment (p). This function solves the optimal n, J and/or p with and without constraints.<\/span><\/div>, <div class='tooltip'>od.3m<span class='tooltiptext'>The optimal design of three-level multisite randomized trials (MRTs) is to choose the sample allocation that minimizes the variance of treatment effect under fixed budget and cost structure. The optimal design parameters include the level-1 sample size per level-2 unit (n), the level-2 sample size per level-3 unit (J), and the proportion of level-2 unit to be assigned to treatment (p). This function solves the optimal n, J and/or p with and without constraints.<\/span><\/div>, <div class='tooltip'>od.4<span class='tooltiptext'>The optimal design of four-level cluster randomized trials (CRTs) is to choose the sample allocation that minimizes the variance of treatment effect under fixed budget and cost structure. The optimal design parameters include the level-1 sample size per level-2 unit (n), the level-2 sample size per level-3 unit (J), the level-3 sample size per level-4 unit (K), and the proportion of level-4 clusters/groups to be assigned to treatment (p). This function solves the optimal n, J, K and/or p with and without constraints.<\/span><\/div>, <div class='tooltip'>od.4m<span class='tooltiptext'>The optimal design of four-level multisite randomized trials (MRTs) is to choose the sample allocation that minimizes the variance of treatment effect under fixed budget and cost structure. The optimal design parameters include the level-1 sample size per level-2 unit (n), the level-2 sample size per level-3 unit (J), the level-3 sample size per level-4 unit (K), and the proportion of level-3 units to be assigned to treatment (p). This function solves the optimal n, J, K and/or p with and without constraints.<\/span><\/div>, <div class='tooltip'>power.1<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for individual randomized controlled trials (RCTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>power.2<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for two-level cluster randomized trials (CRTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>power.2m<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for two-level multisite randomized trials (MRTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>power.3<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for three-level cluster randomized trials (CRTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>power.3m<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for three-level multisite randomized trials (MRTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>power.4<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for four-level cluster randomized trials (CRTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>power.4m<span class='tooltiptext'>This function can calculate required budget for desired power, power or minimum detectable effect size (MDES) under fixed budget for four-level multisite randomized trials (MRTs). It also can perform conventional power analyses (e.g., required sample size, power, and MDES calculation).<\/span><\/div>, <div class='tooltip'>re<span class='tooltiptext'>Calculate the relative efficiency (RE) between two designs, it returns same results as those from function rpe<\/span><\/div>, <div class='tooltip'>rpe<span class='tooltiptext'>Calculate the relative precision and efficiency (RPE) between two designs, it returns same results as those from function re<\/span><\/div>","<div class='tooltip'>add_size.mean.two_stage<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>Beta<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>bibd<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>conf.matrix<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>Confint.mean.two_stage<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>decide.triangular.test<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>defining_relation.fractional_factorial.two_levels<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>delta.t.test<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>design.reg.polynom<span class='tooltiptext'>Determines locations and number of replications for a polynomial regression design. Needs specification of order of polynom, borders of intervall and total number of measurements as input.<\/span><\/div>, <div class='tooltip'>design.regression.polynom<span class='tooltiptext'>Determines locations and number of replications for a polynomial regression design. Needs specification of order of polynom, borders of intervall and total number of measurements as input.<\/span><\/div>, <div class='tooltip'>hadamard.matrix<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>is.hadamard<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>ncp<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>optim_size_n.three_way_nested.model_4_a<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>optim_size_n.three_way_nested.model_4_c<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>plot.triangular.test<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>print.bibd<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>print.design.regression<span class='tooltiptext'>Print method for a design.regression object.<\/span><\/div>, <div class='tooltip'>print.triangular.test<span class='tooltiptext'>Prints a triangular.test object.<\/span><\/div>, <div class='tooltip'>seqadd.confint<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>seqadd.normal<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_a.three_way_mixed_cxbina.model_3_c<span class='tooltiptext'>Returns the optimal number of levels for factor A (and B).<\/span><\/div>, <div class='tooltip'>size_a.three_way_mixed_cxbina.model_7_c<span class='tooltiptext'>Returns the optimal number of levels for factor A (and B).<\/span><\/div>, <div class='tooltip'>size_ab.three_way_mixed_cxbina.model_7_c<span class='tooltiptext'>Returns the optimal number of levels for factor A (and B).<\/span><\/div>, <div class='tooltip'>size_b.three_way_mixed_ab_in_c.model_3_a<span class='tooltiptext'>Returns the optimal number of levels for factor B.<\/span><\/div>, <div class='tooltip'>size_b.three_way_mixed_cxbina.model_4_a<span class='tooltiptext'>Returns the optimal number of levels for factor B.<\/span><\/div>, <div class='tooltip'>size_b.three_way_mixed_cxbina.model_4_axc<span class='tooltiptext'>Returns the optimal number of levels for factor B.<\/span><\/div>, <div class='tooltip'>size_b.three_way_mixed_cxbina.model_4_c<span class='tooltiptext'>Returns the optimal number of levels for factor B.<\/span><\/div>, <div class='tooltip'>size_b.three_way_mixed_cxbina.model_7_c<span class='tooltiptext'>Returns the optimal number of levels for factor B.<\/span><\/div>, <div class='tooltip'>size_b.three_way_nested.model_6_a<span class='tooltiptext'>Returns the optimal number of levels for factor B.<\/span><\/div>, <div class='tooltip'>size_b.two_way_cross.mixed_model_a_fixed_a<span class='tooltiptext'>Returns the optimal number of obervations per level of factor B.<\/span><\/div>, <div class='tooltip'>size_b.two_way_nested.b_random_a_fixed_a<span class='tooltiptext'>Returns the optimal number of obervations per level of factor B.<\/span><\/div>, <div class='tooltip'>size_bc.three_way_cross.model_4_a_case1<span class='tooltiptext'>Returns the optimal number of levels for factor B and C.<\/span><\/div>, <div class='tooltip'>size_bc.three_way_cross.model_4_a_case2<span class='tooltiptext'>Returns the optimal number of levels for factor B and C.<\/span><\/div>, <div class='tooltip'>size_bc.three_way_mixed_cxbina.model_6_a_case1<span class='tooltiptext'>Returns the optimal number of levels for factor B and C.<\/span><\/div>, <div class='tooltip'>size_bc.three_way_mixed_cxbina.model_6_a_case2<span class='tooltiptext'>Returns the optimal number of levels for factor B and C.<\/span><\/div>, <div class='tooltip'>size_c.three_way_cross.model_3_a<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_cross.model_3_axb<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_ab_in_c.model_5_a<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_ab_in_c.model_5_axb<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_ab_in_c.model_5_b<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_ab_in_c.model_6_b<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_cxbina.model_5_a<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_cxbina.model_5_b<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_mixed_cxbina.model_7_b<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_nested.model_5_a<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_nested.model_5_b<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_c.three_way_nested.model_7_b<span class='tooltiptext'>Returns the optimal number of levels for .<\/span><\/div>, <div class='tooltip'>size_n.one_way.model_1<span class='tooltiptext'>Returns the optimal number of obervations per level of factor A.<\/span><\/div>, <div class='tooltip'>size_n.regII.ci_b1<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.regII.ci_exp<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.regII.ci_rho<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.regII.test_rho<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.regII.test_rho_2<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.regII.test_rho2<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.three_way_cross.model_1_a<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_cross.model_1_axb<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_cross.model_1_axbxc<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_ab_in_c.model_1_a<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_ab_in_c.model_1_axb<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_ab_in_c.model_1_b<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_ab_in_c.model_1_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_ab_in_c.model_3_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_ab_in_c.model_4_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_1_a<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_1_axc<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_1_b<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_1_bxc<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_1_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_3_b<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_mixed_cxbina.model_3_bxc<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_1_a<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_1_b<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_1_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_3_b<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_3_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_4_a<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_4_c<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size_n.three_way_nested.model_8_c<span class='tooltiptext'>Returns the optimal number of obervations per level of each factor.<\/span><\/div>, <div class='tooltip'>size_n.two_way_cross.model_1_a<span class='tooltiptext'>Returns the optimal number of obervations per level of factor A.<\/span><\/div>, <div class='tooltip'>size_n.two_way_cross.model_1_axb<span class='tooltiptext'>Returns the optimal number of obervations per level of factor A.<\/span><\/div>, <div class='tooltip'>size_n.two_way_nested.a_random_b_fixed_b<span class='tooltiptext'>Returns the optimal number of obervations per level of factor A.<\/span><\/div>, <div class='tooltip'>size_n.two_way_nested.model_1_test_factor_a<span class='tooltiptext'>Returns the optimal number of obervations per level of factor A.<\/span><\/div>, <div class='tooltip'>size_n.two_way_nested.model_1_test_factor_b<span class='tooltiptext'>Returns the optimal number of obervations per level of factor A.<\/span><\/div>, <div class='tooltip'>size.anova<span class='tooltiptext'>This function provides access to several functions returning the optimal number of levels and / or observations in different types of One-Way, Two-Way and Three-Way ANOVA.<\/span><\/div>, <div class='tooltip'>size.comparing_probs.mcnemar<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.comparing.variances<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.multiple_t.test<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.prop_test.two_sample<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.prop.confint<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.prop.test<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.selection.bechhofer<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.seq_select.mean<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.t.test<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>size.variance.confint<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>sizes.confint.welch<span class='tooltiptext'>Undocumented / internal functions<\/span><\/div>, <div class='tooltip'>triangular.test.norm<span class='tooltiptext'>Performs a sequential test, compares means of two normally distributed groups.<\/span><\/div>, <div class='tooltip'>triangular.test.prop<span class='tooltiptext'>Performs a sequential test, compares probabilities in two groups.<\/span><\/div>, <div class='tooltip'>update.triangular.test<span class='tooltiptext'>Updates a triangular.test object and executes one or more steps in the sequence of tests.<\/span><\/div>, <div class='tooltip'>update.triangular.test.norm<span class='tooltiptext'>Updates a triangular.test object and executes one or more steps in the sequence of tests.<\/span><\/div>, <div class='tooltip'>update.triangular.test.prop<span class='tooltiptext'>Updates a triangular.test object and executes one or more steps in the sequence of tests.<\/span><\/div>","<div class='tooltip'>c_weight<span class='tooltiptext'>This is a sub-part of running Newton Raphon method to search c-optimal weights for the given design points. This is function probvides the obtained c-optimal weights after one iteration of Newton Raphson method.<\/span><\/div>, <div class='tooltip'>c_weight_1<span class='tooltiptext'>The first derivative of the c-optimality criterion with respect to the model parameters. This is a sub-function of c_weight to run one iteration of Newton-Raphson method.<\/span><\/div>, <div class='tooltip'>c_weight_2<span class='tooltiptext'>The second derivative of the c-optimality criterion with respect to the model parameters. This is a sub-function of c_weight to run one iteration of Newton-Raphson method.<\/span><\/div>, <div class='tooltip'>D_weight<span class='tooltiptext'>This is a sub-part of running Newton Raphon method to search D-optimal weights for the given design points. This is function probvides the obtained D-optimal weights after one iteration of Newton Raphson method.<\/span><\/div>, <div class='tooltip'>D_weight_1<span class='tooltiptext'>The first derivative of the D-optimality criterion with respect to the model parameters. This is a sub-function of D_weight to run one iteration of Newton-Raphson method.<\/span><\/div>, <div class='tooltip'>D_weight_2<span class='tooltiptext'>The second derivative of the D-optimality criterion with respect to the model parameters. This is a sub-function of D_weight to run one iteration of Newton-Raphson method.<\/span><\/div>, <div class='tooltip'>D1<span class='tooltiptext'>This function computes each element of the vector function c_weight_1 which is the first derivative of the c-optimality criterion with respect to the model parameters.<\/span><\/div>, <div class='tooltip'>d11<span class='tooltiptext'>This function computes each element of the vector function DD_weight_1 which is the first derivative of the Ds-optimality criterion with respect to the model parameters.<\/span><\/div>, <div class='tooltip'>DD_weight<span class='tooltiptext'>This is a sub-part of running Newton Raphon method to search Ds-optimal weights for the given design points. This is function probvides the obtained Ds-optimal weights after one iteration of Newton Raphson method.<\/span><\/div>, <div class='tooltip'>DD_weight_1<span class='tooltiptext'>The first derivative of the Ds-optimality criterion with respect to the model parameters. This is a sub-function of D_weight to run one iteration of Newton-Raphson method.<\/span><\/div>, <div class='tooltip'>DD_weight_2<span class='tooltiptext'>The second derivative of the Ds-optimality criterion with respect to the model parameters. This is a sub-function of D_weight to run one iteration of Newton-Raphson method.<\/span><\/div>, <div class='tooltip'>DD1<span class='tooltiptext'>This function computes each element of the matrix function c_weight_2 which is the second derivative of the c-optimality criterion with respect to the model parameters.<\/span><\/div>, <div class='tooltip'>dd11<span class='tooltiptext'>This function computes each element of the matrix function DD_weight_2 which is the second derivative of the Ds-optimality criterion with respect to the model parameters.<\/span><\/div>, <div class='tooltip'>Deff<span class='tooltiptext'>Obtaining D-efficiency for estimating model parameters for a given design under the 3, 4, 5-parameter logistic models. For the output, the function shows three things: 1. the obtained D-optimal design for the given parameter values; 2. the D-efficiency of the given design; and 3. the number of subjects required for the given design in order to provide the same performance as the D-optimal design does.<\/span><\/div>, <div class='tooltip'>Dp<span class='tooltiptext'>This finds the target dose level EDp which is the dose producing p percentage of the difference between the maximum and the minimum responses.<\/span><\/div>, <div class='tooltip'>DS1<span class='tooltiptext'>This evaluates the sensitivity function of the c-optimality criterion for the EDp at the given value of x.<\/span><\/div>, <div class='tooltip'>ds11<span class='tooltiptext'>This evaluates the sensitivity function of the Ds-optimality criterion at the given value of x.<\/span><\/div>, <div class='tooltip'>Dseff<span class='tooltiptext'>Obtaining Ds-efficiency for estimating the asymmetric factor for a given design under the 5-parameter logistic models. For the output, the function shows three things: 1. the obtained Ds-optimal design for the given parameter values; 2. the Ds-efficiency of the given design; and 3. the number of subjects required for the given design in order to provide the same performance as the Ds-optimal design does.<\/span><\/div>, <div class='tooltip'>DsOPT<span class='tooltiptext'>Obtain Ds-optimal design (i.e., optimal dose levels and corresponding optimal weights) for estimating the asymmetric factor under the 5-parameter logistic model.<\/span><\/div>, <div class='tooltip'>EDpeff<span class='tooltiptext'>Obtaining c-efficiency for estimating the EDp for a given design under the 5-parameter logistic models. For the output, the function shows three things: 1. the obtained c-optimal design for the given parameter values; 2. the c-efficiency of the given design; and 3. the number of subjects required for the given design in order to provide the same performance as the c-optimal design does.<\/span><\/div>, <div class='tooltip'>EDpOPT<span class='tooltiptext'>Obtain c-optimal design (i.e., optimal dose levels and corresponding optimal weights) for estimating the EDp under the 5-parameter logistic model. The EDp is the dose level that achieve the 100p% of the difference between the maximum and the minimum responses.<\/span><\/div>, <div class='tooltip'>f<span class='tooltiptext'>This gives the first derivative of the mean response function (3, 4, or 5PL) with respect to the model parameters.<\/span><\/div>, <div class='tooltip'>g<span class='tooltiptext'>Partial derivative of the EDp with respect to the model parameters<\/span><\/div>, <div class='tooltip'>ginv<span class='tooltiptext'>Obtain generalized inverse of the given matrix X.<\/span><\/div>, <div class='tooltip'>infor<span class='tooltiptext'>Obtain a information matrix at a single given design point x.<\/span><\/div>, <div class='tooltip'>Inv<span class='tooltiptext'>When the inverse matrix is a singular, it is adjusted to become nonsingular matrix.<\/span><\/div>, <div class='tooltip'>Minus<span class='tooltiptext'>Matrix subtraction: M1-M2.<\/span><\/div>, <div class='tooltip'>Multiple<span class='tooltiptext'>Matrix multiplication: M1*M2.<\/span><\/div>, <div class='tooltip'>Plus<span class='tooltiptext'>Matrix addition: M1+M2.<\/span><\/div>, <div class='tooltip'>RDOPT<span class='tooltiptext'>Obtain robust D-optimal design (i.e., optimal dose levels and corresponding optimal weights) for estimating the model parameters that works well under the 3, 4, 5-parameter logistic models. In addition, the function can be used to obtain D-optimal design for each of the three models.<\/span><\/div>, <div class='tooltip'>S_weight<span class='tooltiptext'>Newton Raphson method to get optimal weights for given design points \"X\".<\/span><\/div>, <div class='tooltip'>SDM<span class='tooltiptext'>Summation of diagonal elements in a matrix M.<\/span><\/div>, <div class='tooltip'>smalld1<span class='tooltiptext'>This computes the first derivative of the D-optimality criterion with respect to the model parameters for a specified model.<\/span><\/div>, <div class='tooltip'>smalldd1<span class='tooltiptext'>This computes the second derivative of the D-optimality criterion with respect to the model parameters for a specified model.<\/span><\/div>, <div class='tooltip'>smallds1<span class='tooltiptext'>This evaluates the sensitivity function of the D-optimality criterion at the given value of x for a specified model.<\/span><\/div>, <div class='tooltip'>sMultiple<span class='tooltiptext'>Multiply a constant to a matrix: s*M.<\/span><\/div>, <div class='tooltip'>Trans<span class='tooltiptext'>Transpose of a matrix M.<\/span><\/div>, <div class='tooltip'>upinfor<span class='tooltiptext'>Obtain normalized Fisher information matrix for the 3, 4, 5PL models.<\/span><\/div>","<div class='tooltip'>Aoptbd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Aoptbd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>cmatbd.mae<span class='tooltiptext'>Computes the information matrix (C-matrix) for treatment effects under either the linear fixed effects model or the linear mixed effects model setting for a given block design of size 2.<\/span><\/div>, <div class='tooltip'>Doptbd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Doptbd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Eoptbd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Eoptbd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>fixparbd.mae<span class='tooltiptext'>Creates a tcltk widow that allow the users to set or fix values for the parametric combinations to compute optimal or near-optimal block designs.<\/span><\/div>, <div class='tooltip'>graphoptbd.mae<span class='tooltiptext'>Creates the graphical layout of resultant A-, MV-, D- or E-optimal or near-optimal block design on a separate pop-up GUI tcltk window.<\/span><\/div>, <div class='tooltip'>intcbd.mae<span class='tooltiptext'>Generates a random initial connected block design for a given number of blocks b of size k = 2 and the number of treatments v.<\/span><\/div>, <div class='tooltip'>mmenubd.mae<span class='tooltiptext'>Alternative to directly using the function optbdmaeAT on the R console, the function mmenubd.mae creates the main menu tcltk window with the list of four optimality criteria (A, MV, D and E) for search of optimal or near-optimal block designs for two-colour cDNA microarray experiments.<\/span><\/div>, <div class='tooltip'>MVoptbd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>MVoptbd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>optbdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal block designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>optbdmaeAT.default<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal block designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>print.optbdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal block designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>print.summary.optbdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal block designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>, <div class='tooltip'>summary.optbdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal block designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016).<\/span><\/div>","<div class='tooltip'>design_bakeoff<span class='tooltiptext'>A visualiation for comparing slope-divided-by-intercept estimatesfor a number of designs<\/span><\/div>, <div class='tooltip'>err_vs_theta0_plot_for_homo_design<span class='tooltiptext'>Plots a standard error estimate of thetahat (slope over intercept) over a range of possible theta0 valuesin order to investigate robustness of the the initial theta0 guess.<\/span><\/div>, <div class='tooltip'>experimental_results<span class='tooltiptext'>Report the results of the experiment as well as confidence intervals.<\/span><\/div>, <div class='tooltip'>oed_for_slope_over_intercept<span class='tooltiptext'>Create an optimal design for measuring the slope divided by the intercept<\/span><\/div>","<div class='tooltip'>optgs<span class='tooltiptext'>optgs is used to find a one-sided multi-stage design that balances four optimality criteria for a RCT with normally distributed outcomes<\/span><\/div>, <div class='tooltip'>powerfamily<span class='tooltiptext'>powerfamily is used to find a one-sided extended power-family group-sequential design<\/span><\/div>","<div class='tooltip'>dirder<span class='tooltiptext'>Computes the vector of derivatives at a normalized approximate design w of length n in the directions of singular designs e_i, where i ranges from 1 to n.<\/span><\/div>, <div class='tooltip'>effbound<span class='tooltiptext'>Computes a lower bound on the efficiency of a design w in the class of all approximate designs of the same size as w.<\/span><\/div>, <div class='tooltip'>Fx_blocks<span class='tooltiptext'>Creates the matrix of all candidate regressors of a linear regression model corresponding to the problem of the optimal block size-two design.<\/span><\/div>, <div class='tooltip'>Fx_CtoA<span class='tooltiptext'>Pre-transforms the matrix of all candidate regressors to the form suitable for computing regularized c-optimal designs via A-optimum algorithms.<\/span><\/div>, <div class='tooltip'>Fx_cube<span class='tooltiptext'>Creates the matrix of all candidate regressors for a factor regression model on a cuboid grid (up to 9 factors).<\/span><\/div>, <div class='tooltip'>Fx_dose<span class='tooltiptext'>Creates the matrix of all candidate regressors for a linearization of a dose response model.<\/span><\/div>, <div class='tooltip'>Fx_glm<span class='tooltiptext'>Creates the matrix of all candidate regressors for a linearization of a generalized linear model.<\/span><\/div>, <div class='tooltip'>Fx_ItoA<span class='tooltiptext'>Pre-transforms the matrix of all candidate regressors to the form suitable for computing I-optimal designs via A-optimum algorithms.<\/span><\/div>, <div class='tooltip'>Fx_simplex<span class='tooltiptext'>Creates the matrix of all candidate regressors for a mixture regression model on a regular simplex grid (up to 9 factors).<\/span><\/div>, <div class='tooltip'>Fx_survival<span class='tooltiptext'>Creates the matrix of all candidate regressors for a linearization of a proportional hazards survival model.<\/span><\/div>, <div class='tooltip'>ginv<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>infmat<span class='tooltiptext'>Computes the information matrix of a design w in the model determined by the matrix Fx of candidate regressors.<\/span><\/div>, <div class='tooltip'>mvee_REX<span class='tooltiptext'>Computes the shape matrix H and the center z of the minimum-volume ellipsoid enclosing a finite set of data-points.<\/span><\/div>, <div class='tooltip'>od_A_AQUA<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_DEL<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_KL<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_KL_bin<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_MISOCP<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_MUL<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_RC<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_REX<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_A_VDM<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_AQUA<span class='tooltiptext'>Computes an efficient exact design under general linear constraints via a quadratic approximation of the optimality criterion.<\/span><\/div>, <div class='tooltip'>od_c_LP<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_c_MISOCP<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_AQUA<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_DEL<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_KL<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_KL_bin<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_MISOCP<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_MUL<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_RC<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_REX<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_D_VDM<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>, <div class='tooltip'>od_DEL<span class='tooltiptext'>Removes the design points (or, equivalently, candidate regressors) that cannot support an optimal approximate design.<\/span><\/div>, <div class='tooltip'>od_KL<span class='tooltiptext'>Computes an optimal or near-optimal exact design of experiments under the standard (size) constraint on the size of the experiment.<\/span><\/div>, <div class='tooltip'>od_MISOCP<span class='tooltiptext'>Computes an optimal or nearly-optimal approximate or exact experimental design using mixed integer second-order cone programming.<\/span><\/div>, <div class='tooltip'>od_PIN<span class='tooltiptext'>Use a fast greedy method to compute an efficient saturated subset (saturated exact design).<\/span><\/div>, <div class='tooltip'>od_plot<span class='tooltiptext'>Visualizes selected aspects of an experimental design<\/span><\/div>, <div class='tooltip'>od_pool<span class='tooltiptext'>A function pool.fun is applied to all the elements of a vector val that appear within the groups formed by identical rows of a matrix X.<\/span><\/div>, <div class='tooltip'>od_print<span class='tooltiptext'>Prints various characteristics of an experimental design<\/span><\/div>, <div class='tooltip'>od_PUK<span class='tooltiptext'>Compute the classical efficient rounding of a non-normalized approximate design w such that the resulting exact design has size floor(sum(w)).<\/span><\/div>, <div class='tooltip'>od_RC<span class='tooltiptext'>Computes an efficient exact design under multiple linear resource constraints using the RC heuristic.<\/span><\/div>, <div class='tooltip'>od_REX<span class='tooltiptext'>Computes an optimal approximate design under the standard (size) constraint using one of three methods.<\/span><\/div>, <div class='tooltip'>od_SYM<span class='tooltiptext'>Attempts to \"symmetrize\" an approximate design w by minimizing its norm while keeping its information matrix.<\/span><\/div>, <div class='tooltip'>optcrit<span class='tooltiptext'>Computes the criterion value of a design w in the model determined by the matrix Fx of all regressors.<\/span><\/div>, <div class='tooltip'>varfun<span class='tooltiptext'>Computes the vector of variances (sensitivities) for a given design w.<\/span><\/div>, <div class='tooltip'>verify<span class='tooltiptext'>Internal OptimalDesign functions<\/span><\/div>","<div class='tooltip'>Aoptrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Aoptrcd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>cmatrcd.mae<span class='tooltiptext'>Computes the information matrix (C-matrix) for treatment effects under either the linear fixed effects model or the linear mixed effects model setting for a given row-column design.<\/span><\/div>, <div class='tooltip'>Doptrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Doptrcd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Eoptrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>Eoptrcd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>fixparrcd.mae<span class='tooltiptext'>Creates a GUI tcltk window that allow the users to set or fix values for the parametric combinations to compute optimal or near-optimal row-column designs.<\/span><\/div>, <div class='tooltip'>graphoptrcd.mae<span class='tooltiptext'>Creates the graphical layout of resultant A-, MV-, D- or E-optimal or near-optimal row-column design on a separate pop-up GUI tcltk window.<\/span><\/div>, <div class='tooltip'>intcrcd.mae<span class='tooltiptext'>Generates a random initial connected row-column design for a given number of arrays b of size k = 2 and the number of treatments v.<\/span><\/div>, <div class='tooltip'>mmenurcd.mae<span class='tooltiptext'>Alternative to directly using the function optrcdmaeAT on the R console, the function mmenurcd.mae creates the main menu tcltk window with the list of four optimality criteria (A, MV, D and E) for search of optimal or near-optimal row-column designs for two-colour cDNA microarray experiments.<\/span><\/div>, <div class='tooltip'>MVoptrcd.maeA<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>MVoptrcd.maeT<span class='tooltiptext'>Functions for internal usage only.<\/span><\/div>, <div class='tooltip'>optrcdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016) after adjusting to the row-column setup.<\/span><\/div>, <div class='tooltip'>optrcdmaeAT.default<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016) after adjusting to the row-column setup.<\/span><\/div>, <div class='tooltip'>print.optrcdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016) after adjusting to the row-column setup.<\/span><\/div>, <div class='tooltip'>print.summary.optrcdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016) after adjusting to the row-column setup.<\/span><\/div>, <div class='tooltip'>summary.optrcdmaeAT<span class='tooltiptext'>Used to compute A-, MV-, D- or E-optimal or near-optimal row-column designs for two-colour cDNA microarray experiments under either the linear fixed effects model or the linear mixed effects model settings using either the array exchange or treatment exchange algorithms of Debusho, Gemechu and Haines (2016) after adjusting to the row-column setup.<\/span><\/div>","<div class='tooltip'>design<span class='tooltiptext'>Construct optimal approximate designs as well as efficient exact designs forcrossover model with subject dropout, crossover model with proportional residualeffect, and interference model.<\/span><\/div>, <div class='tooltip'>design_app<span class='tooltiptext'>A function to launch graphical interface to design function.<\/span><\/div>, <div class='tooltip'>eff<span class='tooltiptext'>A generic function that returns the effciency for either exact designs to approximatedesigns or exact design to a given design<\/span><\/div>, <div class='tooltip'>effLB<span class='tooltiptext'>The function take S3 object of class 'dropout' as input and return its lowerbound of efficiency of exact design.<\/span><\/div>","<div class='tooltip'>apbibd<span class='tooltiptext'>This function calculates the different kinds of efficiencies and the overall efficiency factor of Partially Balanced Incomplete Block Designs with any number of associate classes. The total number of treatments i.e. v, replications i.e. r, block size i.e. k, vector l of lambda's (lambda i being the ith element of vector l), vector n of number of associates (n i, i.e. number of ith associates, being the ith element of vector n), a list P of P-matrices of the association scheme of the design (Pi being the ith matrix of the list P) are to be supplied as input to the function.<\/span><\/div>, <div class='tooltip'>circulant<span class='tooltiptext'>Circulant matrix, which is a special kind of Toeplitz matrix, is a square matrix of order n whose rows are obtained by cyclically rotated versions of a list ``l'' of length n such that the first row is obtained by cyclically rotating one element toward right the list ``l'' and each of the other row is the cyclically rotated one element toward the right version of the previous row. This function is used to generate a circulant matrix of order n. The order of the circulant matrix i.e. n is supplied as an argument to the function.<\/span><\/div>, <div class='tooltip'>series1<span class='tooltiptext'>Let us consider a module M of residue class mod(5) having elements 0, 1, 2, 3, 4 and all the elements of M are assigned to each of the n >= 2 classes. This function constructs PBIB designs with the following parameters:v = 5n, b = 5n, r = n+4, k = n+4lambda 1 = 5, lambda 2 = 2, lambda 3 = 5, lambda 4 = 2, lambda 5 = n<\/span><\/div>, <div class='tooltip'>series2<span class='tooltiptext'>Let us consider a module M of residue class mod(5) having elements 0, 1, 2, 3, 4 and all the elements of M are assigned to each of the n >= 2 classes. This function constructs PBIB designs with the following parameters:v = 5n, b = 5n, r = n+3, k = n+3lambda 1 = 3, lambda 2 = 1, lambda 3 = 3, lambda 4 = 2, lambda 5 = n<\/span><\/div>, <div class='tooltip'>series3<span class='tooltiptext'>Let us consider a module M of residue class mod(5) having elements 0, 1, 2, 3, 4 and all the elements of M are assigned to each of the n >= 2 classes. This function constructs PBIB designs with the following parameters:v = 5n, b = 5n, r = 2(n + 1), k = 2(n + 1)lambda 1 = n + 2, lambda 2 = n + 2, lambda 3 = 3, lambda 4 = 2, lambda 5 = 2n<\/span><\/div>, <div class='tooltip'>series4<span class='tooltiptext'>Let us consider a module M having m elements. To each element of the module, there corresponds n distinct classes, where m >= 5 and n >= 2. With these v = mn treatments following are parameters of the three-associate class PBIB design:v = mn, b = mn, r = (m+n-1), k = (m+n-1)lambda 1 = m, lambda 2 = 2, lambda 3 = n<\/span><\/div>, <div class='tooltip'>series5<span class='tooltiptext'>Consider a module M having m elements and there are n classes corresponding to each element of the module. Thus, we have a total of v = mn treatments (m is odd prime). For these v = mn treatments following are the parameters of the three-associate class PBIB design:v = mn, b = mn, r = (m+n-2), k = (m+n-2)lambda 1 = m-2, lambda 2 = 2, lambda 3 = n-2<\/span><\/div>, <div class='tooltip'>series6<span class='tooltiptext'>Consider a module M having m elements and there are n classes corresponding to each element of the module. Thus, we have a total of v = mn treatments (m is odd prime). For these v = mn treatments following are the parameters of the three-associate class PBIB design:v = mn, b = m, r = (m-1), k = (m-1)nlambda 1 = m-2, lambda 2 = m-2, lambda 3 = m-1<\/span><\/div>, <div class='tooltip'>verify<span class='tooltiptext'>There exists various necessary conditions for the existence of the PBIB design as well as the association scheme on which the PBIB design is based. This function Verifies all those necessary conditions for the existence of PBIB designs based on any association scheme. The total number of treatments i.e. v, the total number of blocks i.e. b, replications i.e. r, block size i.e. k, vector l of lambda's (lambda i being the ith element of vector l), vector n of number of associates (n i, i.e. number of ith associates, being the ith element of vector n), a list P of P-matrices of the association scheme of the design (Pi being the ith matrix of the list P) are to be supplied as input to the function.<\/span><\/div>, <div class='tooltip'>ym1<span class='tooltiptext'>If we omit the same number of rows, say t rows, from the top and the bottom of the Circulant matrix, such that we are left with atleast two rows, the resulting arrangement of rows is a Youden-m square.(A) For even-ordered Circulant matrices with order v >= 4, the columns of the Youden-m squares so obtained constitute the PBIB designs with the following parameters:v >= 4 and even, b = v, r = k = v-2tlambda 1 = v - 2(t + 1), lambda m-i = v - 2t - 1 - 2i ; i = 0, 1, ..., t-1lambda t = lambda t+1 = ... = lambda m-t = v - 4t. If t >=3 then, lambda i = v - 2(t + i); i = 2, 3, ..., t-1(B)For odd-ordered Circulant matrices with order v >= 5, the columns of the Youden-m squares so obtained constitute the PBIB designs with the following parameters:v >=5 and odd, b = v, r = k = v-2tlambda 1 = v - 2t - 1, lambda m-i = v - 2(t + 1) - i; i = 0, 1, ..., t - 1, lambda m-(t-1)-i = lambda m-(t-1) - i ; i = 0, 1, 2, ..., t-1and lambda 2 = lambda 3= ... = lambda (m-2t+1) = lambda (m-2t+2)<\/span><\/div>, <div class='tooltip'>ym2<span class='tooltiptext'>By omitting the middle 2t (t = 1, 2, ...) rows of any even ordered Circulant matrix with order v >= 6 and considering only those rows which lie either above or below the omitted 2t rows, the resulting arrangement of rows gives a new type of Youden-m square. The columns of these Youden-m squares constitute m-associate class PBIB designs, with the following parameters:v >= 6 and even, b = v, r = k = (v/2)-t, lambda 1 = r-2, lambda m = lambda 1+1 If m is even, then lambda i+1 = lambda i - 2; i = 1, 2, ..., m/2 and lambda i-1 = lambda i - 2; i = m, m-1, ..., (m/2)+1If m is odd, then lambda i+1 = lambda i - 2; i = 1, 2, ..., (m+1)/2 and lambda i-1 = lambda i - 2; i = m, m-1, ..., ((m+1)/2) + 1<\/span><\/div>, <div class='tooltip'>ym3<span class='tooltiptext'>By omitting the middle and equal number of rows, say t rows, from both ends of any odd-ordered Circulant matrix with order v >= 7and considering those rows that lie either between the middle omitted row and omitted rows from the top or between the middle omitted row and the omitted rows from the bottom of the Circulant matrix. Then this arrangement of rows gives us a Youden-m square. These Youden-m square designs are the designs for two-way elimination of heterogenity. The columns of this Youden-m square constitue PBIB design with the following parameters:v >= 7 and odd, b = v, r = k = ((v+1)/2) -1-tlambda 1 = v - 6 - (m - 4 + t), lambda i = 0 ; i = 2, 3, ..., t+2if m > 3 then, lambda j = lambda j-1+1 ; j = t+3, t+4, ...,m<\/span><\/div>","<div class='tooltip'>BIB<span class='tooltiptext'>Gives the configuration of a Balanced Incomplete binary BlocksDesigns (BIBD) using a projective geometry on a Galois Field of order 2 GF(2).<\/span><\/div>, <div class='tooltip'>Gen<span class='tooltiptext'>Gives the configuration of a Balanced Incomplete binary BlocksDesigns of seconde generation.<\/span><\/div>, <div class='tooltip'>Resolvable<span class='tooltiptext'>Extracts the Resolvable Balanced Incomplete Block Design (RBIBD) ofany BIBD constructed from a projective geometry.<\/span><\/div>, <div class='tooltip'>Steps<span class='tooltiptext'>Gives the different stages of nested design begening froma projective geometry, BIBD, BIBD seconde generation, RBIBDand associated uniform designs.<\/span><\/div>, <div class='tooltip'>Uniform<span class='tooltiptext'>Builds the uniform design (UD) using a Resolvable Balanced IncompleteBlock Design (RBIBD)<\/span><\/div>","<div class='tooltip'>postprob<span class='tooltiptext'>Thall and Simon's criterion function for determining thetrial decision cutoffs based on the posterior probability.<\/span><\/div>, <div class='tooltip'>predprob<span class='tooltiptext'>Lee and Liu's criterion function for determining thetrial decision cutoffs based on the predictive probability.<\/span><\/div>, <div class='tooltip'>stopbound_post<span class='tooltiptext'>The stopping boundaries based on Thall and Simon's criterion.<\/span><\/div>, <div class='tooltip'>stopbound_pred<span class='tooltiptext'>The stopping boundaries based on Lee and Liu's criterion.<\/span><\/div>","<div class='tooltip'>eff_suggest<span class='tooltiptext'>Suggest the input eff.structure of function SimPRMD with selectedeff.sd_tran for given efficacy mean matrix and efficacy standarddeviation<\/span><\/div>, <div class='tooltip'>eff_summary<span class='tooltiptext'>Numerically compute the Mean, mariginal standard deviance and mariginalcorrelation matrix of efficacy measure generated with specified parameters.The function provides plots of marginal density of generated efficacy andcorrelation matrix for each dose. Check details to see the efficacy datageneration procedures.<\/span><\/div>, <div class='tooltip'>nTTP_summary<span class='tooltiptext'>nTTP_summary generates the mean nTTP score and the probability ofobserving DLT for all doses and cycles<\/span><\/div>, <div class='tooltip'>nTTP.array<span class='tooltiptext'>nTTP.array generates the nTTP dictionary for all combination oftoxicity type and grade with a given toxicity weighted matrix. Used infunction nTTP_summary for checking the toxicity scenario.<\/span><\/div>, <div class='tooltip'>patlist.display<span class='tooltiptext'>Display patient records in a human-readable table. Each cell contains 3values including observed nTTP, DLT, and dose assignment. The higher thedose, the warmer the cell background color is. The black color of therecords indicates DLT equals 1.<\/span><\/div>, <div class='tooltip'>RunPRMD<span class='tooltiptext'>A function to implement a Multi-Stage Phase I Dose-Finding Design torecommend dosage selection based on the data collected in the availablepatient cohorts. The available models include 1-stage model with/withoutindividualized dose modification, 3-stage model with/without individualizeddose modification, 3-stage model with individualized dose modification onstage II and 3-stage model with individualized dose modification on stage Iand dose modification on stage II.<\/span><\/div>, <div class='tooltip'>SimPRMD<span class='tooltiptext'>A function to implement simulations for a multi-stage phase 1 dose-findingdesign incorporating a longitudinal continuous efficacy outcome andtoxicity data from multiple treatment cycles. The available models include1-stage model with/without individualized dose modification, 3-stage modelwith/without individualized dose modification, 3-stage model withindividualized dose modification on stage II and 3-stage model withindividualized dose modification on stage I and dose modification on stageII.<\/span><\/div>","<div class='tooltip'>GenToxProb<span class='tooltiptext'>%% ~~ A concise (1-5 lines) description of what the function does. ~~A function to generate toxicity probability matrices employing proportional odds model, used as input for trial simulation. The generated matrices are a four-dimension array containing the probabilities of the occurrences of certain grades for certain types of toxicities, at each dose level and cycle under consideration. Dimension 1 refers to doses; dimension 2 corresponds to cycles of the treatment; dimension 3 regards the types of toxicities while dimenion 4 relates to grades.<\/span><\/div>, <div class='tooltip'>parm<span class='tooltiptext'>parm creates an S3 object of class parm that contains all necessary prior information to estimate the phase I repeated measures model.<\/span><\/div>, <div class='tooltip'>plot.RunRMDVal<span class='tooltiptext'>plot.RunRMDVal plot the dose-nTTP estimation in cycle based on the RunRMD estimation.<\/span><\/div>, <div class='tooltip'>RunRMD<span class='tooltiptext'>RunRMD estimate the dosage for the next patient cohort using TTP measured multidimensional toxicity data collected from multiple treatemnt cycles of the existing patients.<\/span><\/div>, <div class='tooltip'>RunRMDEFF<span class='tooltiptext'>A function to interactively implement an adaptive, multi-stage phase 1 dose-finding design incorporating continuous efficacy and toxicity data from multiple treatment cycles, proposed by Du et al(2017)<\/span><\/div>, <div class='tooltip'>SimRMD<span class='tooltiptext'>SimRMD uses the design parameters to simulate the trial characters of phase I dose-finding design.<\/span><\/div>, <div class='tooltip'>SimRMDEFF<span class='tooltiptext'>A function to implement simulation for an adaptive, multi-stage phase 1 dose-finding design incorporating a continuous efficacy outcome and toxicity data from multiple treatment cycles, proposed by Du et al(2017)<\/span><\/div>, <div class='tooltip'>Tox2nTTP<span class='tooltiptext'>Tox2nTTP use the design parameters to compute the nTTP score baed on the observed toxicity.<\/span><\/div>","<div class='tooltip'>contourPlot<span class='tooltiptext'>Creates a two-dimensional contour plot of a factorial design model (linear least squares model). The two specified factors areplotted on the horizontal and vertical axes. If the model has more than two factors, then the remaining factors are set at their zero level.It is assumed the model is in coded units.<\/span><\/div>, <div class='tooltip'>grocery<span class='tooltiptext'>The hourly profit made when selling the product at price P and the product is displayed at height H [cm] on the shelf.<\/span><\/div>, <div class='tooltip'>manufacture<span class='tooltiptext'>The hourly profit made when selling the product at price P and the product is produced at a throughput rate T [parts per hour].<\/span><\/div>, <div class='tooltip'>paretoPlot<span class='tooltiptext'>Creates a plot that shows the model coefficients from the least squares model as bars.The absolute value of the coefficients are taken.The coefficient's sign is represented by the bar colour: grey for negative and black for positive coefficients.<\/span><\/div>, <div class='tooltip'>popcorn<span class='tooltiptext'>A fixed number of popcorn kernels are cooked are heated at the same temperature. This simulation returns the number ofkernels that are edible when the pot is left on the stove for a given number of T seconds.<\/span><\/div>, <div class='tooltip'>tradeoff<span class='tooltiptext'>Fractional factorial designs are a trade-off of information learned vs. the amount of work and cost (experiments) invested. This function shows which factors are confounded (confused; aliased) with each other when running a fractional factorial. The output shows the design's resolution and how to generate the fractional factorial.<\/span><\/div>, <div class='tooltip'>tradeOffTable<span class='tooltiptext'>Creates a new plot that shows a trade-off table for fractional factorial designs.<\/span><\/div>","<div class='tooltip'>pipe.design<span class='tooltiptext'>Implements the Product of Independent beta Probabilities dose Escalation (PIPE) design for dual-agent Phase I trials as described in Mander A.P. and Sweeting M.J (2015) Statistics in Medicine. <doi: 10.1002/sim.6434><\/span><\/div>, <div class='tooltip'>runShinyPIPE<span class='tooltiptext'>Runs a Shiny web browser interface of the Product of Independent beta Probabilities dose Escalation (PIPE) design<\/span><\/div>","<div class='tooltip'>a_line_search<span class='tooltiptext'>The function performs a grid search sequentially along design variables. The grid is defined by ls_step_size.<\/span><\/div>, <div class='tooltip'>bfgsb_min<span class='tooltiptext'>This is the implementation of a Broyden Fletcher Goldfarb Shanno (BFGS) method for nonlinear minimization with box constraints.<\/span><\/div>, <div class='tooltip'>blockexp<span class='tooltiptext'>Create some output to the screen and a text file that summarizes the initial design and the design spaceyou will use to optimize.<\/span><\/div>, <div class='tooltip'>blockfinal<span class='tooltiptext'>Create some output to the screen and a text file that summarizes the problem you solved.<\/span><\/div>, <div class='tooltip'>blockheader<span class='tooltiptext'>Create some output to the screen and a text file that summarizes the problem you are tying to solve.<\/span><\/div>, <div class='tooltip'>blockopt<span class='tooltiptext'>Create some output to the screen and a text file that summarizes the optimization settingsyou will use to optimize.<\/span><\/div>, <div class='tooltip'>build_sfg<span class='tooltiptext'>Build PopED parameter function from a model function<\/span><\/div>, <div class='tooltip'>calc_autofocus<span class='tooltiptext'>Compute the autofocus portion of the stochastic gradient routine<\/span><\/div>, <div class='tooltip'>calc_ofv_and_fim<span class='tooltiptext'>This function computes the expectation of the FIM and OFV(FIM) for either point values of parameter estimatesor parameter distributions given the model, parameters, distributions of parameter uncertainty, design and methods defined in the PopED database.<\/span><\/div>, <div class='tooltip'>calc_ofv_and_grad<span class='tooltiptext'>Compute an objective function and gradient with respect to the optimization parameters.This function can be passed to the Broyden Fletcher Goldfarb Shanno (BFGS) method for nonlinear minimization with box constraints implemented in bfgsb_min.<\/span><\/div>, <div class='tooltip'>cell<span class='tooltiptext'>Create a cell array as in MATLAB.<\/span><\/div>, <div class='tooltip'>convert_variables<span class='tooltiptext'>Function takes design variables from input filesand converts them to the global variables neededin PopED. Typically not used by the user. Instead use the function create.poped.database.<\/span><\/div>, <div class='tooltip'>create_design<span class='tooltiptext'>Create design variables to fully describe a design.If variables are supplied then these variables are checked for consistency and, if possible, changed to sizes that make sense if there are inconsistencies.Returns a list of matricies compatible with PopED.<\/span><\/div>, <div class='tooltip'>create_design_space<span class='tooltiptext'>create_design_space takes an initial design and arguments for a design space and creates a design and design space for design optimization.Checks the sizes of supplied design space variables and changes them to sizes that make sense if there are inconsistencies. Function arguments can use shorthand notation (single values, vectors, lists of vectors and list of list) or matricies.Returns a list of matricies compatible with PopED.<\/span><\/div>, <div class='tooltip'>create.poped.database<span class='tooltiptext'>This function takes the input file (a previously created poped database) supplied by the user, or function arguments, and creates a database that can then be used to run all other PopED functions. The function supplies default values to elements of the database that are not specified in theinput file or as function arguments. Default arguments are supplied in the Usage section (easiest to use a text search to find values you are interested in).<\/span><\/div>, <div class='tooltip'>design_summary<span class='tooltiptext'>Display a summary of output from poped_db<\/span><\/div>, <div class='tooltip'>diag_matlab<span class='tooltiptext'>There are some differences between tha MATLAB and the R version of diag.Specifically, if a 1xN or a Nx1 matrix is supplied to the Rdiag function then just the first element of this vector isreturned. This function tries to match the MATLAB version in handling vectors(matricies with one dimension equal to one), and will return a diagonalmatrix in these situations.<\/span><\/div>, <div class='tooltip'>Doptim<span class='tooltiptext'>Optimize the objective function. There are 4 different optimizationalgorithms used in this function Adaptive random search.See RS_opt. Stochastic gradient. A Broyden FletcherGoldfarb Shanno (BFGS) method for nonlinear minimization with boxconstraints. A line search. See a_line_search. Theoptimization algorithms run in series, taking as input the output from theprevious method. The stopping rule used is to test if the line searchalgorithm fids a better optimum then its initial value. If so, then the chainof algorithms is run again. If line search is not used then the argumentiter_tot defines the number of times the chain of algorithms is run. This function takes information from the PopED database supplied as anargument. The PopED database supplies information about the the model,parameters, design and methods to use. Some of the arguments coming from thePopED database can be overwritten; if they are supplied then they are usedinstead of the arguments from the PopED database.<\/span><\/div>, <div class='tooltip'>Dtrace<span class='tooltiptext'>A helper function for writing output to the screen and files when optimizing.<\/span><\/div>, <div class='tooltip'>ed_laplace_ofv<span class='tooltiptext'>Compute the expectation of the det(FIM) using the Laplaceapproximation to the expectation. Computations are made based on the model,parameters, distributions of parameter uncertainty, design and methodsdefined in the PopED database or as arguments to the function.<\/span><\/div>, <div class='tooltip'>ed_mftot<span class='tooltiptext'>Compute the expectation of the FIM given the model, parameters, distributions of parameter uncertainty, design and methods defined in the PopED database.<\/span><\/div>, <div class='tooltip'>efficiency<span class='tooltiptext'>Efficiency calculation between two designs.<\/span><\/div>, <div class='tooltip'>evaluate_design<span class='tooltiptext'>This function evaluates the design defined in a poped database.<\/span><\/div>, <div class='tooltip'>evaluate_power<span class='tooltiptext'>Evaluate the power of a design to estimate a parameter value different thansome assumed value (often the assumed value is zero). The power is calculated using the linear Wald test and the the design is defined in a poped database.<\/span><\/div>, <div class='tooltip'>evaluate.e.ofv.fim<span class='tooltiptext'>Compute the expectation of the FIM and OFV(FIM) given the model, parameters, distributions of parameter uncertainty, design and methods defined in the PopED database. Some of the arguments coming from the PopED database can be overwritten; by default these arguments are NULL in the function, if they are supplied then they are used instead of the arguments from the PopED database.<\/span><\/div>, <div class='tooltip'>evaluate.fim<span class='tooltiptext'>Compute the FIM given the model, parameters, design and methods defined in the PopED database. Some of the arguments coming from the PopED database can be overwritten; by default these arguments are NULL in the function, if they are supplied then they are used instead of the arguments from the PopED database.<\/span><\/div>, <div class='tooltip'>feps.add<span class='tooltiptext'>This is a residual unexplained variability (RUV) model function that encodes the model described above.The function is suitable for input to the create.poped.database function using the fError_file argument.<\/span><\/div>, <div class='tooltip'>feps.add.prop<span class='tooltiptext'>This is a residual unexplained variability (RUV) model function that encodes the model described above.The function is suitable for input to the create.poped.database function using the fError_file argument.<\/span><\/div>, <div class='tooltip'>feps.prop<span class='tooltiptext'>This is a residual unexplained variability (RUV) model function that encodes the model described above.The function is suitable for input to the create.poped.database function using the fError_file argument.<\/span><\/div>, <div class='tooltip'>feval<span class='tooltiptext'>This is just a wrapper for the do.call function to behave like the feval function in MATLAB.<\/span><\/div>, <div class='tooltip'>ff.PK.1.comp.oral.md.CL<span class='tooltiptext'>This is a structural model function that encodes a model that is one-compartment, oral absorption, multiple bolus dose, parameterized using CL.The function is suitable for input to the create.poped.database function using the ff_file argument.<\/span><\/div>, <div class='tooltip'>ff.PK.1.comp.oral.md.KE<span class='tooltiptext'>This is a structural model function that encodes a model that is one-compartment, oral absorption, multiple bolus dose, parameterized using KE.The function is suitable for input to the create.poped.database function using the ff_file argument.<\/span><\/div>, <div class='tooltip'>ff.PK.1.comp.oral.sd.CL<span class='tooltiptext'>This is a structural model function that encodes a model that is one-compartment, oral absorption, single bolus dose, parameterized using CL.The function is suitable for input to the create.poped.database function using the ff_file argument.<\/span><\/div>, <div class='tooltip'>ff.PK.1.comp.oral.sd.KE<span class='tooltiptext'>This is a structural model function that encodes a model that is one-compartment, oral absorption, single bolus dose, parameterized using KE.The function is suitable for input to the create.poped.database function using the ff_file argument.<\/span><\/div>, <div class='tooltip'>ff.PKPD.1.comp.oral.md.CL.imax<span class='tooltiptext'>This is a structural model function that encodes the model described above.The function is suitable for input to the create.poped.database function using the ff_file argument.<\/span><\/div>, <div class='tooltip'>ff.PKPD.1.comp.sd.CL.emax<span class='tooltiptext'>This is a structural model function that encodes the model described above.The function is suitable for input to the create.poped.database function using the ff_file argument.<\/span><\/div>, <div class='tooltip'>fileparts<span class='tooltiptext'>Get the various parts of a file with path string.<\/span><\/div>, <div class='tooltip'>get_all_params<span class='tooltiptext'>Extract all model parameters from the PopED database.<\/span><\/div>, <div class='tooltip'>get_rse<span class='tooltiptext'>This function computes the expected relative standard errors of a model given a design and a previously computedFIM.<\/span><\/div>, <div class='tooltip'>get_unfixed_params<span class='tooltiptext'>all = vector of all unfixed params var derivative is a vector of 1 and 0, 1means derivative of parameter is taken w$r.t. variance otherwise w$r.t. sd Ifparams is supplied then the parameter is taken from this vector instead ofpoped.db<\/span><\/div>, <div class='tooltip'>getfulld<span class='tooltiptext'>Create a full D (between subject variability) matrix given a vector of variances and covariances.Note, this does not test matching vector lengths.<\/span><\/div>, <div class='tooltip'>getTruncatedNormal<span class='tooltiptext'>Generate a random sample from a truncated normal distribution.<\/span><\/div>, <div class='tooltip'>gradf_eps<span class='tooltiptext'>The function performs a linearization of the model with respect to the residual variability.Derivative of model w.r.t. eps evaluated at eps=0 and b=b_ind.<\/span><\/div>, <div class='tooltip'>inv<span class='tooltiptext'>Function computes the inverse of a matrix.<\/span><\/div>, <div class='tooltip'>isempty<span class='tooltiptext'>Function written to match MATLAB's isempty function<\/span><\/div>, <div class='tooltip'>LEDoptim<span class='tooltiptext'>Optimize the objective function for D-family, E-family and Laplace approximated ED designs. Right now there is only one optimization algorithm used in this function Adaptive random search. See RS_opt.This function takes information from the PopED database supplied as an argument.The PopED database supplies information about the the model, parameters, design and methods to use.Some of the arguments coming from the PopED database can be overwritten; if they are supplied then they are used instead of the arguments from the PopED database.<\/span><\/div>, <div class='tooltip'>LinMatrixL<span class='tooltiptext'>Function computes the derivative of the model with respect to the between subject variability terms in the model (b's and bocc's) evaluated ata defined point (b_ind and bocc_ind).<\/span><\/div>, <div class='tooltip'>LinMatrixL_occ<span class='tooltiptext'>The function performs a linearization of the model with respect to the occasion variability parameter..Derivative of model w.r.t. eta_occ, evaluated bocc_ind.<\/span><\/div>, <div class='tooltip'>LinMatrixLH<span class='tooltiptext'>The function performs a linearization of the model with respect to the residual variability and then the between subject variability.Derivative of model w.r.t. eps then eta, evaluated at eps=0 and b=b_ind.<\/span><\/div>, <div class='tooltip'>mc_mean<span class='tooltiptext'>Function computes the monte-carlo mean of a function by varying the parameter inputs to the function<\/span><\/div>, <div class='tooltip'>median_hilow_poped<span class='tooltiptext'>Created for back compatibility with older versions of ggplot, and so that PopED does not have to load ggplot when started.<\/span><\/div>, <div class='tooltip'>mfea<span class='tooltiptext'>Optimize the objective function using a modified Fedorov exchange algorithm.The function works for continuous and discrete optimization variables.This function takes information from the PopED database supplied as an argument.The PopED database supplies information about the the model, parameters, design and methods to use.Some of the arguments coming from the PopED database can be overwritten; if they are supplied then they are used instead of the arguments from the PopED database.<\/span><\/div>, <div class='tooltip'>mftot<span class='tooltiptext'>Compute the FIM given specific model(s), parameters, design and methods.<\/span><\/div>, <div class='tooltip'>model_prediction<span class='tooltiptext'>Function generates a data frame of model predictions for the typical value in the population,individual predictions and data predictions. The function can also be used to generate datasetswithout predictions using the design specified in the arguments.<\/span><\/div>, <div class='tooltip'>ofv_criterion<span class='tooltiptext'>Compute a normalized OFV based on the size of the FIM matrix. This value can then be used in efficiency calculations. This is NOT the OFV used in optimization, see ofv_fim.<\/span><\/div>, <div class='tooltip'>ofv_fim<span class='tooltiptext'>Compute a criterion of the FIM given the model, parameters, design and methods defined in the PopED database.<\/span><\/div>, <div class='tooltip'>ones<span class='tooltiptext'>Create a matrix of ones of size (dim1 x dim2).<\/span><\/div>, <div class='tooltip'>optim_ARS<span class='tooltiptext'>Optimize an objective function using an adaptive random search algorithm. The function works for both discrete and continuous optimization parameters and allows for box-constraints and sets of allowed values.<\/span><\/div>, <div class='tooltip'>optim_LS<span class='tooltiptext'>optim_LS performs sequential grid search optimization of an arbitrary function with respect to each of the parameters to be optimized over. The function works for both discrete and continuous optimization parameters and allows for box-constraints (by using the upper and lower function arguments) or sets of allowed values (by using the allowed_values function argument) for all parameters, or on a parameter per parameter basis.<\/span><\/div>, <div class='tooltip'>optimize_groupsize<span class='tooltiptext'>Title Optimize the proportion of individuals in the design groups<\/span><\/div>, <div class='tooltip'>optimize_n_eff<span class='tooltiptext'>optimize HOW MANY n there should be to achieve efficiency=1 compared to a reference OFV<\/span><\/div>, <div class='tooltip'>optimize_n_rse<span class='tooltiptext'>Optimize the number of subjects, based on the current design and the desired uncertainty of a single parameter<\/span><\/div>, <div class='tooltip'>pargen<span class='tooltiptext'>Function generates random samples for a list of parameters<\/span><\/div>, <div class='tooltip'>plot_efficiency_of_windows<span class='tooltiptext'>Function plots the efficiency of windows around the sample time points. The function samples from a uniform distribution around the sample time points for each group (or each individual with deviate_by_id=TRUE, with slower calculation times) and compares the results with the design defined in poped.db. The maximal and minimal allowed values for all design variables as defined in poped.db are respected (e.g. poped.db$design_space$minxt and poped.db$design_space$maxxt).<\/span><\/div>, <div class='tooltip'>plot_model_prediction<span class='tooltiptext'>Function plots model predictions for the typical value in the population,individual predictions and data predictions.<\/span><\/div>, <div class='tooltip'>poped_gui<span class='tooltiptext'>Run the graphical interface for PopED<\/span><\/div>, <div class='tooltip'>poped_optim<span class='tooltiptext'>Optimize a design defined in a PopED database using the objective functiondescribed in the database (or in the arguments to this function). The function works for both discrete and continuous optimization variables.<\/span><\/div>, <div class='tooltip'>poped_optimize<span class='tooltiptext'>This function is an older version of poped_optim. Please use poped_optimunless you have a specific reason to use this function instead.<\/span><\/div>, <div class='tooltip'>poped.choose<span class='tooltiptext'>Function chooses arg1 unless it is NULL in which case arg2 is chosen.<\/span><\/div>, <div class='tooltip'>rand<span class='tooltiptext'>Generate random samples from a uniform distribution [0,1] and return in matrix form<\/span><\/div>, <div class='tooltip'>randn<span class='tooltiptext'>Generate random samples from a standardized normal distribution and return in matrix form.<\/span><\/div>, <div class='tooltip'>RS_opt<span class='tooltiptext'>Optimize the objective function using an adaptive random search algorithm. Optimization can be performed for both D-family and E-family designs.The function works for both discrete and continuous optimization variables.This function takes information from the PopED database supplied as an argument.The PopED database supplies information about the the model, parameters, design and methods to use.Some of the arguments coming from the PopED database can be overwritten; by default these arguments are NULL in the function, if they are supplied then they are used instead of the arguments from the PopED database.<\/span><\/div>, <div class='tooltip'>shrinkage<span class='tooltiptext'>Predict shrinkage of empirical Bayes estimates (EBEs) in a population model<\/span><\/div>, <div class='tooltip'>size<span class='tooltiptext'>Function written to match MATLAB's size function<\/span><\/div>, <div class='tooltip'>start_parallel<span class='tooltiptext'>This tool chooses the type of parallelization process to use based on thecomputer OS being used. For windows the default is \"snow\" and for Linux-likesystems the default is \"multicore\"<\/span><\/div>, <div class='tooltip'>test_mat_size<span class='tooltiptext'>Test to make sure that matricies are the right size<\/span><\/div>, <div class='tooltip'>tic<span class='tooltiptext'>Function to start a timer. Stop with toc().<\/span><\/div>, <div class='tooltip'>toc<span class='tooltiptext'>Function to stop a timer. Start with tic().<\/span><\/div>, <div class='tooltip'>zeros<span class='tooltiptext'>Create a matrix of zeros of size (dim1 x dim2).<\/span><\/div>","<div class='tooltip'>ES.anova.oneway<span class='tooltiptext'>Calculating effect size (Cohen's f) of one-way anova for means with equal observations in each group<\/span><\/div>, <div class='tooltip'>ES.chisq.assoc<span class='tooltiptext'>Compute effect size of chi-squared test of association<\/span><\/div>, <div class='tooltip'>ES.chisq.gof<span class='tooltiptext'>Compute effect size of chi-squared test of goodness of fit<\/span><\/div>, <div class='tooltip'>ES.proportions<span class='tooltiptext'>Compute effect size for a difference in proportions<\/span><\/div>, <div class='tooltip'>ES.t.one<span class='tooltiptext'>Calculating effect size (Cohen's d) of one-sample t test<\/span><\/div>, <div class='tooltip'>ES.t.paired<span class='tooltiptext'>Calculating effect size (Cohen's d) of paired two-sample t test<\/span><\/div>, <div class='tooltip'>ES.t.two<span class='tooltiptext'>Calculating effect size (Cohen's d) of independent two-sample t test<\/span><\/div>, <div class='tooltip'>power.anova.oneway<span class='tooltiptext'>Power calculations for balanced one-way analysis of variance tests<\/span><\/div>, <div class='tooltip'>power.chisq<span class='tooltiptext'>Power calculations for chi-squared test<\/span><\/div>, <div class='tooltip'>power.plot.chisq<span class='tooltiptext'>Power analysis plot of chi-squared test<\/span><\/div>, <div class='tooltip'>power.proportions<span class='tooltiptext'>Power calculations for proportion tests (two-sided)<\/span><\/div>, <div class='tooltip'>power.t<span class='tooltiptext'>Power calculations for t-test<\/span><\/div>","<div class='tooltip'>ci.length<span class='tooltiptext'>Provides expected confidence interval widths for QTL location when we have dense markers.<\/span><\/div>, <div class='tooltip'>deflate<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>deflate.bc<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>deflate.f2<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>detectable<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>detectable.bc<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>detectable.f2<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>detectable.ri<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>eff1<span class='tooltiptext'>Calculate the MMA K1, K12, and the standardized dissimilarity score (eff1).<\/span><\/div>, <div class='tooltip'>error.var<span class='tooltiptext'>The function error.var estimates the error variance usingestimates of the environmental variance and genetic variance. The effectsegregating at a locus, can be calculated using gmeans2effectThese are key inputs for power calculations. The functionprop.var calculates the proportion of variance explained by alocus given the effect size and error variance.<\/span><\/div>, <div class='tooltip'>genetic.dist<span class='tooltiptext'>Utility functions<\/span><\/div>, <div class='tooltip'>gmeans2effect<span class='tooltiptext'>The function error.var estimates the error variance usingestimates of the environmental variance and genetic variance. The effectsegregating at a locus, can be calculated using gmeans2effectThese are key inputs for power calculations. The functionprop.var calculates the proportion of variance explained by alocus given the effect size and error variance.<\/span><\/div>, <div class='tooltip'>info<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>info.bc<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>info.f2<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>info2cost<span class='tooltiptext'>Functions to calculate information cost-ratios.<\/span><\/div>, <div class='tooltip'>info2cost.bc<span class='tooltiptext'>Functions to calculate information cost-ratios.<\/span><\/div>, <div class='tooltip'>info2cost.f2<span class='tooltiptext'>Functions to calculate information cost-ratios.<\/span><\/div>, <div class='tooltip'>K1<span class='tooltiptext'>Calculate the MMA K1, K12, and the standardized dissimilarity score (eff1).<\/span><\/div>, <div class='tooltip'>K12<span class='tooltiptext'>Calculate the MMA K1, K12, and the standardized dissimilarity score (eff1).<\/span><\/div>, <div class='tooltip'>Kstat<span class='tooltiptext'>Calculate the MMA K1, K12, and the standardized dissimilarity score (eff1).<\/span><\/div>, <div class='tooltip'>mma<span class='tooltiptext'>Selective phenotyping with similarity measure 2 to select the mostdissimilar subset of individuals.<\/span><\/div>, <div class='tooltip'>mma.level<span class='tooltiptext'>This routine is for internal use. It sets 3 levels to 0,1,2.<\/span><\/div>, <div class='tooltip'>nu<span class='tooltiptext'>Internal qtlDesign functions<\/span><\/div>, <div class='tooltip'>nullinfo<span class='tooltiptext'>Functions to calculate the information under the null hypothesis of no effect. Functions for discount factors for incomplete genotyping.<\/span><\/div>, <div class='tooltip'>optselection<span class='tooltiptext'>Functions to find optimal selection fractions given cost.<\/span><\/div>, <div class='tooltip'>optselection.bc<span class='tooltiptext'>Functions to find optimal selection fractions given cost.<\/span><\/div>, <div class='tooltip'>optselection.f2<span class='tooltiptext'>Functions to find optimal selection fractions given cost.<\/span><\/div>, <div class='tooltip'>optspacing<span class='tooltiptext'>Functions to find optimal marker spacing given cost.<\/span><\/div>, <div class='tooltip'>optspacing.bc<span class='tooltiptext'>Functions to find optimal marker spacing given cost.<\/span><\/div>, <div class='tooltip'>optspacing.f2<span class='tooltiptext'>Functions to find optimal marker spacing given cost.<\/span><\/div>, <div class='tooltip'>powercalc<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>powercalc.bc<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>powercalc.f2<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>powercalc.ri<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>prop.var<span class='tooltiptext'>The function error.var estimates the error variance usingestimates of the environmental variance and genetic variance. The effectsegregating at a locus, can be calculated using gmeans2effectThese are key inputs for power calculations. The functionprop.var calculates the proportion of variance explained by alocus given the effect size and error variance.<\/span><\/div>, <div class='tooltip'>recomb<span class='tooltiptext'>Utility functions<\/span><\/div>, <div class='tooltip'>samplesize<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>samplesize.bc<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>samplesize.f2<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>samplesize.ri<span class='tooltiptext'>Power, sample size, and minimum detectable effect size calculations areperformed for backcross, F2 intercross, and recombinant inbred (RI)lines.<\/span><\/div>, <div class='tooltip'>tailprob<span class='tooltiptext'>Provides genome-wide thresholds and tail probabilities for the maxima ofgenome scans using Poisson approximations.<\/span><\/div>, <div class='tooltip'>tau<span class='tooltiptext'>Internal qtlDesign functions<\/span><\/div>, <div class='tooltip'>thresh<span class='tooltiptext'>Provides genome-wide thresholds and tail probabilities for the maxima ofgenome scans using Poisson approximations.<\/span><\/div>, <div class='tooltip'>version.qtlDesign<span class='tooltiptext'>Returns the version number for the qtlDesign package.<\/span><\/div>","<div class='tooltip'>calculateGUI<span class='tooltiptext'>This function provides a graphical user interface for calculating rank-based statistical tests in general factorial designs.<\/span><\/div>, <div class='tooltip'>noether<span class='tooltiptext'>Sample size calculation for the Wilcoxon-Mann-Whitney test using the Noether formula.<\/span><\/div>, <div class='tooltip'>psr<span class='tooltiptext'>The psr() function calculates the pseuo-ranks of data in general factorial designs.<\/span><\/div>, <div class='tooltip'>rank.two.samples<span class='tooltiptext'>The rank.two.samples() function calculates the weighted or unweighted treatment effect ina two-sample problem. In addition to rankFD, the user can specify the alternativeand choose from a variety of different possibilities to calculate confidence intervals, see the detailsbelow. Furthermore, a Wilcoxon test is calculate with the possibility to consider shift effects.<\/span><\/div>, <div class='tooltip'>rankFD<span class='tooltiptext'>The package provides the Wald-type as well as the ANOVA-type statistic for rank-based factorial designs, i.e., even for ordinal data. It isimplemented for crossed designs and allowsfor an arbitrary number of factor combinations as well as different samplesizes.<\/span><\/div>, <div class='tooltip'>WMWSSP<span class='tooltiptext'>Sample size calculation for the Wilcoxon-Mann-Whitney test.<\/span><\/div>","<div class='tooltip'>activeDataSetDesign2P<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesign2Pnoccd<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesign2PResp<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesign2Pwoc<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignccd<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignlhs<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignLongToWide<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignP<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignparam<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignparamlong<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignPResp<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignQualP<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignRemovableP<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeDataSetDesignWide<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeModelLM<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>activeModelRSM<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>checkWarnings<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>dc<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>design.info.design<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>display.design<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>display.design.std.order<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>displayblock<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>displayCentralCompositeDesigns<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>displayGlossary<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>displaysplitplot<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>exist2Designs<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existcatlgPkgs<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existDesign2<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existDesign2FrF<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existDesign2pb<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existDesigns<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existDesignsWithResp<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>existRSMs<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>fix<span class='tooltiptext'>Function showData allows to view data, function fix allows to edit data. For designs, it is only permitted to type in response data and other additional information like date or comments for experimental runs in R, such that all attributes of the design objects are updated accordingly. However, it is recommended to do such editing tasks outside of R, as R is not ideal for these.<\/span><\/div>, <div class='tooltip'>fix.default<span class='tooltiptext'>Function showData allows to view data, function fix allows to edit data. For designs, it is only permitted to type in response data and other additional information like date or comments for experimental runs in R, such that all attributes of the design objects are updated accordingly. However, it is recommended to do such editing tasks outside of R, as R is not ideal for these.<\/span><\/div>, <div class='tooltip'>fix.design<span class='tooltiptext'>Function showData allows to view data, function fix allows to edit data. For designs, it is only permitted to type in response data and other additional information like date or comments for experimental runs in R, such that all attributes of the design objects are updated accordingly. However, it is recommended to do such editing tasks outside of R, as R is not ideal for these.<\/span><\/div>, <div class='tooltip'>fpairsc<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>help2level<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpabout<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpAnalyze<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpDopt<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpExpImp<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpExport<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpgeneral<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpGeneralAnalyze<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpImport<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpInspect<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpMenus<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpModify<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpquant<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>helpQuantDesignAnalyze<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>HowToReImport<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>initializeDialogDoE<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>isDesign2FrF<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>isDesign2pb<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>justDoItDoE<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listCatlgs<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listDesigns<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listDesigns2<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listDesignsWithResp<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listLists<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listObjects<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>listRSMs<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>loadworkspace<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>Menu.addcenter<span class='tooltiptext'>This help file describes adding center points to a 2-level design from package FrF2.<\/span><\/div>, <div class='tooltip'>Menu.addresponse<span class='tooltiptext'>This help file describes how to use the menu for adding (a) response(s).<\/span><\/div>, <div class='tooltip'>Menu.aggregate<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>Menu.augmentlhs<span class='tooltiptext'>With this menu, an existing latin hypercube sample for quantitative factors can be augmented. The menu calls function lhs.augment from R-package DoE.wrapper.<\/span><\/div>, <div class='tooltip'>Menu.bbd<span class='tooltiptext'>With this menu, Box-Behnken designs for quantitative factors can be generated. The menu calls function bbd.design from R-package DoE.wrapper.<\/span><\/div>, <div class='tooltip'>Menu.cand<span class='tooltiptext'>With this menu, D-optimal designs can be generated. The menu calls function Dopt.design from R-package DoE.wrapper. Optionally, the candidate designs menu can be called before calling this menu in order to create a candidate design.<\/span><\/div>, <div class='tooltip'>Menu.ccd<span class='tooltiptext'>With this menu, central composite designs for quantitative factors can be generated. The menu calls function ccd.design from R-package DoE.wrapper.<\/span><\/div>, <div class='tooltip'>Menu.changecontr<span class='tooltiptext'>Help on using the menu for changing contrasts<\/span><\/div>, <div class='tooltip'>Menu.colremove<span class='tooltiptext'>Help on using the menu for deleting columns<\/span><\/div>, <div class='tooltip'>Menu.contour<span class='tooltiptext'>Contour plots for linear models with quantitative regressors<\/span><\/div>, <div class='tooltip'>Menu.display<span class='tooltiptext'>This help file describes displaying the active design in actual or standard run order.<\/span><\/div>, <div class='tooltip'>Menu.Dopt<span class='tooltiptext'>With this menu, D-optimal designs can be generated. The menu calls function Dopt.design from R-package DoE.wrapper. Optionally, the candidate designs menu can be called before calling this menu in order to create a candidate design.<\/span><\/div>, <div class='tooltip'>Menu.EffectPlots<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>Menu.export<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>Menu.fac<span class='tooltiptext'>This menu is for generating full factorial designs. The menu calls function function fac.design from package DoE.base and has been opened internally using the function shown in the usage section, which normal users will never have to deal with.<\/span><\/div>, <div class='tooltip'>Menu.FrF2level<span class='tooltiptext'>With this menu, regular fractional factorial 2-level designs can be generated. The menu calls function FrF2 from package FrF2 and has been opened internally using the function shown in the usage section, which normal users will never have to deal with.<\/span><\/div>, <div class='tooltip'>Menu.IAPlot<span class='tooltiptext'>Using the menu for 2-level main effects and interaction plots<\/span><\/div>, <div class='tooltip'>Menu.import<span class='tooltiptext'>This help file describes importing a design that is already in the R workspace but is not the currently active data set.<\/span><\/div>, <div class='tooltip'>Menu.importrdacsv<span class='tooltiptext'>This help file describes how to re-import an exported design that exists as an rda and a csv file, when using the R-Commander.<\/span><\/div>, <div class='tooltip'>Menu.lhs<span class='tooltiptext'>With this menu, latin hypercube samples for quantitative factors can be generated. The menu calls function lhs.design from R-package DoE.wrapper.<\/span><\/div>, <div class='tooltip'>Menu.linearModelDesign<span class='tooltiptext'>This dialog is used to specify a linear model to be fit by the lm function.<\/span><\/div>, <div class='tooltip'>Menu.loadcatlg<span class='tooltiptext'>This help file describes loading a catalogue from a loaded catalogues package. Currently, the only such package is FrF2.catlg128.<\/span><\/div>, <div class='tooltip'>Menu.model<span class='tooltiptext'>This help file describes usage of the linear model menu.<\/span><\/div>, <div class='tooltip'>Menu.oa<span class='tooltiptext'>This menu is for generating orthogonal arrays for experiments for which not all factors have 2 levels. The menu calls function function oa.design from package DoE.base and has been opened internally using the function shown in the usage section, which normal users will never have to deal with.<\/span><\/div>, <div class='tooltip'>Menu.param<span class='tooltiptext'>Using the menu for Taguchi inner-outer array designs<\/span><\/div>, <div class='tooltip'>Menu.pb2level<span class='tooltiptext'>With this menu, non-regular (=screening) orthogonal 2-level designs can be generated. The menu calls function function pb from package FrF2 and has been opened internally using the function shown in the usage section, which normal users will never have to deal with.<\/span><\/div>, <div class='tooltip'>Menu.pickcube<span class='tooltiptext'>This help file describes how to use the menu for picking the cube portion from a central composite design.<\/span><\/div>, <div class='tooltip'>Menu.plot<span class='tooltiptext'>Help on using the menu for plotting selected factors<\/span><\/div>, <div class='tooltip'>Menu.PlotMeansDoE<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>Menu.responses<span class='tooltiptext'>Help on using the menu for response selection<\/span><\/div>, <div class='tooltip'>Menu.rsm<span class='tooltiptext'>This dialog is used to specify a response surface model to be fit by the rsm function.<\/span><\/div>, <div class='tooltip'>Menu.rsmmodel<span class='tooltiptext'>This help file describes usage of the response surface model menu.<\/span><\/div>, <div class='tooltip'>Menu.steepest<span class='tooltiptext'>This help file describes usage of the steepest slope menu.<\/span><\/div>, <div class='tooltip'>Menu.summarize<span class='tooltiptext'>This help file describes summarizing a design that is already in the R workspace but is not necessarily the currently active data set.<\/span><\/div>, <div class='tooltip'>Menu.tab<span class='tooltiptext'>Help on using the menu for tabulating selected factors<\/span><\/div>, <div class='tooltip'>Menu.wide<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>modelFormulaDoE<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>onHelpSyntax<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>rsmFormula<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>showData<span class='tooltiptext'>Function showData allows to view data, function fix allows to edit data. For designs, it is only permitted to type in response data and other additional information like date or comments for experimental runs in R, such that all attributes of the design objects are updated accordingly. However, it is recommended to do such editing tasks outside of R, as R is not ideal for these.<\/span><\/div>, <div class='tooltip'>showData.default<span class='tooltiptext'>Function showData allows to view data, function fix allows to edit data. For designs, it is only permitted to type in response data and other additional information like date or comments for experimental runs in R, such that all attributes of the design objects are updated accordingly. However, it is recommended to do such editing tasks outside of R, as R is not ideal for these.<\/span><\/div>, <div class='tooltip'>showData.design<span class='tooltiptext'>Function showData allows to view data, function fix allows to edit data. For designs, it is only permitted to type in response data and other additional information like date or comments for experimental runs in R, such that all attributes of the design objects are updated accordingly. However, it is recommended to do such editing tasks outside of R, as R is not ideal for these.<\/span><\/div>, <div class='tooltip'>subsetBoxDoE<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>summarize.design<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>textcorrect<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>, <div class='tooltip'>textentry<span class='tooltiptext'>Functions used in menus - these are not needed by users<\/span><\/div>","<div class='tooltip'>rdrandinf<span class='tooltiptext'>rdrandinf implements randomization inference and related methods for RD designs,using observations in a specified or data-driven selected window around the cutoff wherelocal randomization is assumed to hold.<\/span><\/div>, <div class='tooltip'>rdrbounds<span class='tooltiptext'>rdrbounds calculates lower and upper bounds for the randomization p-value under different degrees of departure from a local randomized experiment, as suggested by Rosenbaum (2002).<\/span><\/div>, <div class='tooltip'>rdsensitivity<span class='tooltiptext'>rdsensitivity analyze the sensitivity of randomization p-valuesand confidence intervals to different window lengths.<\/span><\/div>, <div class='tooltip'>rdwinselect<span class='tooltiptext'>rdwinselect implements the window-selection procedure based on balance tests for RD designs under local randomization. Specifically, it constructs a sequence of nested windows around the RD cutoff and reports binomial tests for the running variable runvar and covariate balance tests for covariates covariates (if specified). The recommended window is the largest window around the cutoff such that the minimum p-value of the balance test is larger than a prespecified level for all nested (smaller) windows. By default, the p-values are calculated using randomization inference methods.<\/span><\/div>","<div class='tooltip'>rdmc<span class='tooltiptext'>rdmc() analyzes RD designs with multiple cutoffs.<\/span><\/div>, <div class='tooltip'>rdmcplot<span class='tooltiptext'>rdmcplot() RD plots with multiple cutoffs.<\/span><\/div>, <div class='tooltip'>rdms<span class='tooltiptext'>rdms() analyzes RD designs with cumulative cutoffs or two running variables.<\/span><\/div>","<div class='tooltip'>rdbwselect<span class='tooltiptext'>rdbwselect implements bandwidth selectors for local polynomial Regression Discontinuity (RD) point estimators and inference procedures developed in Calonico, Cattaneo and Titiunik (2014a), Calonico, Cattaneo and Farrell (2018), Calonico, Cattaneo, Farrell and Titiunik (2019) and Calonico, Cattaneo and Farrell (2020).Companion commands are: rdrobust for point estimation and inference procedures, and rdplot for data-driven RD plots (see Calonico, Cattaneo and Titiunik (2015a) for details).A detailed introduction to this command is given in Calonico, Cattaneo and Titiunik (2015b) and Calonico, Cattaneo, Farrell and Titiunik (2019). A companion Stata package is described in Calonico, Cattaneo and Titiunik (2014b). For more details, and related Stata and R packages useful for analysis of RD designs, visit https://rdpackages.github.io/<\/span><\/div>, <div class='tooltip'>rdbwselect_2014<span class='tooltiptext'>rdbwselect_2014 is a deprecated command implementing three bandwidth selectors for local polynomial Regression Discontinuity (RD) point estimators and inference procedures, as described in Calonico, Cattaneo and Titiunik (2014). This command is no longer supported or updated, and it is made available only for backward compatibility purposes. Please use rdbwselect instead. The latest version of the rdrobust package includes the following commands: rdrobust for point estimation and inference procedures. rdbwselect for data-driven bandwidth selection. rdplot for data-driven RD plots. For more details, and related Stata and R packages useful for analysis of RD designs, visit https://rdpackages.github.io/<\/span><\/div>, <div class='tooltip'>rdplot<span class='tooltiptext'>rdplot implements several data-driven Regression Discontinuity (RD) plots, using either evenly-spaced or quantile-spaced partitioning. Two type of RD plots are constructed: (i) RD plots with binned sample means tracing out the underlying regression function, and (ii) RD plots with binned sample means mimicking the underlying variability of the data. For technical and methodological details see Calonico, Cattaneo and Titiunik (2015a).Companion commands are: rdrobust for point estimation and inference procedures, and rdbwselect for data-driven bandwidth selection.A detailed introduction to this command is given in Calonico, Cattaneo and Titiunik (2015b) and Calonico, Cattaneo, Farrell and Titiunik (2017). A companion Stata package is described in Calonico, Cattaneo and Titiunik (2014).For more details, and related Stata and R packages useful for analysis of RD designs, visit https://rdpackages.github.io/<\/span><\/div>, <div class='tooltip'>rdrobust<span class='tooltiptext'>rdrobust implements local polynomial Regression Discontinuity (RD) point estimators with robust bias-corrected confidence intervals and inference procedures developed in Calonico, Cattaneo and Titiunik (2014a), Calonico, Cattaneo and Farrell (2018), Calonico, Cattaneo, Farrell and Titiunik (2019), and Calonico, Cattaneo and Farrell (2020). It also computes alternative estimation and inference procedures available in the literature. Companion commands are: rdbwselect for data-driven bandwidth selection, and rdplot for data-driven RD plots (see Calonico, Cattaneo and Titiunik (2015a) for details).A detailed introduction to this command is given in Calonico, Cattaneo and Titiunik (2015b), and Calonico, Cattaneo, Farrell and Titiunik (2017). A companion Stata package is described in Calonico, Cattaneo and Titiunik (2014b).For more details, and related Stata and R packages useful for analysis of RD designs, visit https://rdpackages.github.io/<\/span><\/div>","<div class='tooltip'>calcpk<span class='tooltiptext'>This function estimates the per-challenge probability of infection.<\/span><\/div>, <div class='tooltip'>calcVEk<span class='tooltiptext'>This function calculate the estimated per-challenge vaccine efficacy.<\/span><\/div>, <div class='tooltip'>calcVEt<span class='tooltiptext'>This function estimates the vaccine efficacy before or at the time of challenge t. VE(t)>0 indicates that the vaccine is effective in reducing the risk of infection before or at time t, whereas VE(t)<=0 indicate that the vaccine is not effective or has a negative effect.<\/span><\/div>, <div class='tooltip'>lrtest<span class='tooltiptext'>This function performs likelihood ratio test (LRT) to test regression paramters in the mean model and frailty variance paramter nu.<\/span><\/div>, <div class='tooltip'>print.calcpk<span class='tooltiptext'>This function estimates the per-challenge probability of infection.<\/span><\/div>, <div class='tooltip'>print.calcVEk<span class='tooltiptext'>This function calculate the estimated per-challenge vaccine efficacy.<\/span><\/div>, <div class='tooltip'>print.calcVEt<span class='tooltiptext'>This function estimates the vaccine efficacy before or at the time of challenge t. VE(t)>0 indicates that the vaccine is effective in reducing the risk of infection before or at time t, whereas VE(t)<=0 indicate that the vaccine is not effective or has a negative effect.<\/span><\/div>, <div class='tooltip'>print.lrtest<span class='tooltiptext'>This function performs likelihood ratio test (LRT) to test regression paramters in the mean model and frailty variance paramter nu.<\/span><\/div>, <div class='tooltip'>print.rld<span class='tooltiptext'>This function fits a discrete-time survival model with and without random effects.<\/span><\/div>, <div class='tooltip'>print.rld.design<span class='tooltiptext'>This function performs power analyis to design a repeated low-dose challenge experiment with a vaccine and a placeboarm.<\/span><\/div>, <div class='tooltip'>print.summary.calcpk<span class='tooltiptext'>This function estimates the per-challenge probability of infection.<\/span><\/div>, <div class='tooltip'>print.summary.calcVEk<span class='tooltiptext'>This function calculate the estimated per-challenge vaccine efficacy.<\/span><\/div>, <div class='tooltip'>print.summary.calcVEt<span class='tooltiptext'>This function estimates the vaccine efficacy before or at the time of challenge t. VE(t)>0 indicates that the vaccine is effective in reducing the risk of infection before or at time t, whereas VE(t)<=0 indicate that the vaccine is not effective or has a negative effect.<\/span><\/div>, <div class='tooltip'>print.summary.rld<span class='tooltiptext'>This function fits a discrete-time survival model with and without random effects.<\/span><\/div>, <div class='tooltip'>rld<span class='tooltiptext'>This function fits a discrete-time survival model with and without random effects.<\/span><\/div>, <div class='tooltip'>rld.design<span class='tooltiptext'>This function performs power analyis to design a repeated low-dose challenge experiment with a vaccine and a placeboarm.<\/span><\/div>, <div class='tooltip'>rld.fit<span class='tooltiptext'>This is the function called by rld that do the actual computation.<\/span><\/div>, <div class='tooltip'>summary.calcpk<span class='tooltiptext'>This function estimates the per-challenge probability of infection.<\/span><\/div>, <div class='tooltip'>summary.calcVEk<span class='tooltiptext'>This function calculate the estimated per-challenge vaccine efficacy.<\/span><\/div>, <div class='tooltip'>summary.calcVEt<span class='tooltiptext'>This function estimates the vaccine efficacy before or at the time of challenge t. VE(t)>0 indicates that the vaccine is effective in reducing the risk of infection before or at time t, whereas VE(t)<=0 indicate that the vaccine is not effective or has a negative effect.<\/span><\/div>, <div class='tooltip'>summary.rld<span class='tooltiptext'>This function fits a discrete-time survival model with and without random effects.<\/span><\/div>, <div class='tooltip'>transdata<span class='tooltiptext'>This function transfers the original input data frame to required data frame for model fitting.<\/span><\/div>","<div class='tooltip'>KLopt.lnorm<span class='tooltiptext'>Calculates an approximation \\xi^{**}xi^{**} of the KLKL-optimal design (in case of lognormal errors) \\xi^*xi^* for discrimination between a given list of error densities \\{f_i(x,\\theta_i),\\; i = 1,\\dots,\\nu\\}{f_i(x,theta_i), i = 1,\\dots,nu}. This procedure is based on the work [8]. This function mimics tpopt almost entirely. It is planed to combine tpopt and KLopt.lnorm in the future. See tpopt for the detailed description of the arguments marked with ``-//-''.<\/span><\/div>, <div class='tooltip'>tpopt<span class='tooltiptext'>Calculates an approximation \\xi^{**}xi^{**} of the T_{\\mathrm{P}}T_P-optimal design \\xi^*xi^* for discrimination between a given list of models \\{\\eta_i(x,\\theta_i),\\; i = 1,\\dots,\\nu\\}{eta_i(x,theta_i), i = 1,\\dots,nu}. This procedure is based on the algorithms developed by Holger Dette, Viatcheslav B. Melas and Roman Guchenko in [7]. T_{\\mathrm{P}}T_P-optimal design is a probability measure, which maximizes the functional T_{\\mathrm{P}}(\\xi) = \\sum_{i,j=1}^{\\nu} p_{i,j} \\inf_{\\theta_{i,j} \\in \\Theta_j} \\int_{\\mathcal{X}} \\Big[ \\eta_i(x,\\overline{\\theta}_{i}) - \\eta_j(x,\\theta_{i,j}) \\Big]^2 \\xi(dx),T_P(xi) = sum_ij p_ij inf_{theta_ij \\in Theta_j} int_X [ eta_i(x,theta_i^) - \\eta_j(x,theta_ij) ]^2 xi(dx),where \\xixi is an arbitrary design on \\mathcal{X}X (it is presumed here, that \\mathcal{X}X is an interval from \\mathbf{R}R), \\mathrm{P} = \\{ p_{i,j} \\}_{i,j = 1}^{\\nu}P = {p_ij} is a table of non-negative weights with zeros on the diagonal (comparison table) and \\overline{\\theta}_{i}theta_i^ are predefined fixed parameters.It was also shown in [7] that calculation of Bayesian T_{\\mathrm{P}}T_P-optimal design, which maximizes more complicated criterionT_{\\mathrm{P}}^{\\mathrm{B}}(\\xi) = \\sum_{i,j=1}^{\\nu} p_{i,j} \\int_{\\Theta_i} \\inf_{\\theta_{i,j} \\in \\Theta_j} \\int_{\\mathcal{X}} \\Big[ \\eta_i(x,\\lambda_i) - \\eta_j(x,\\theta_{i,j}) \\Big]^2 \\xi(dx) \\mathcal{P}_i(d \\lambda_i),T_P(xi) = sum_ij p_ij int_{Lambda_i} inf_{theta_ij \\in Theta_j} int_X [ eta_i(x,lambda_i^) - \\eta_j(x,theta_ij) ]^2 xi(dx) P_i(d lambda_i),can be reduced to calculation of ordinary T_{\\mathrm{P}}T_P-optimal design, when distributions \\mathcal{P}_iP_i are discrete. That is why in this case the current function is also suitable for calculation of Bayesian designs.<\/span><\/div>","<div class='tooltip'>as.coded.data<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>bbd<span class='tooltiptext'>This function can generate a Box-Behnken design in 3 to 7 factors, and optionally will block it orthogonally if there are 4 or 5 factors. It can also randomize the design.<\/span><\/div>, <div class='tooltip'>canonical<span class='tooltiptext'>Fit a linear model with a response-surface component, and produce appropriate analyses and summaries.<\/span><\/div>, <div class='tooltip'>canonical.path<span class='tooltiptext'>These functions provide the path of steepest ascent (or descent) for a fitted response surface produced by rsm.<\/span><\/div>, <div class='tooltip'>ccd<span class='tooltiptext'>These functions generate central-composite designs, or building blocks thereof.They allow for flexible choices of replications, aliasing of predictors and fractional blocks, and choices of axis or star points.<\/span><\/div>, <div class='tooltip'>ccd.pick<span class='tooltiptext'>This function looks at all combinations of specified design parametersfor central-composite designs, calculates other quantities such asthe alpha values for rotatability and orthogonal blocking, imposes specified restrictions, and outputs the best combinations in a specified order.This serves as an aid in identifying good designs. The design itself can then be generated using ccd, or in pieces using cube, star, etc.<\/span><\/div>, <div class='tooltip'>code2val<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>coded.data<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>codings<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>codings<-<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>contour.lm<span class='tooltiptext'>contour, image, and persp methods that display the fitted surface for an lm object involving two or more numerical predictors.<\/span><\/div>, <div class='tooltip'>cube<span class='tooltiptext'>These functions generate central-composite designs, or building blocks thereof.They allow for flexible choices of replications, aliasing of predictors and fractional blocks, and choices of axis or star points.<\/span><\/div>, <div class='tooltip'>decode.data<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>djoin<span class='tooltiptext'>This implements the rsm package's building-block provisions for handlingsequences of experiments. We often want to join two or more designs into one blocked design for purposes of analysis.<\/span><\/div>, <div class='tooltip'>dupe<span class='tooltiptext'>These functions generate central-composite designs, or building blocks thereof.They allow for flexible choices of replications, aliasing of predictors and fractional blocks, and choices of axis or star points.<\/span><\/div>, <div class='tooltip'>FO<span class='tooltiptext'>Use of one of these functions in a model is how you specify the portion of the model that is to be regarded as a response-surface component.<\/span><\/div>, <div class='tooltip'>foldover<span class='tooltiptext'>These functions generate central-composite designs, or building blocks thereof.They allow for flexible choices of replications, aliasing of predictors and fractional blocks, and choices of axis or star points.<\/span><\/div>, <div class='tooltip'>image.lm<span class='tooltiptext'>contour, image, and persp methods that display the fitted surface for an lm object involving two or more numerical predictors.<\/span><\/div>, <div class='tooltip'>is.coded.data<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>loftest<span class='tooltiptext'>Fit a linear model with a response-surface component, and produce appropriate analyses and summaries.<\/span><\/div>, <div class='tooltip'>model.data<span class='tooltiptext'>Create a data frame with just the variables in the formula in a lm object. This is comparable to model.matrix or model.frame except that factors, polynomials, transformations, etc. are not expanded.<\/span><\/div>, <div class='tooltip'>PE<span class='tooltiptext'>Use of one of these functions in a model is how you specify the portion of the model that is to be regarded as a response-surface component.<\/span><\/div>, <div class='tooltip'>persp.lm<span class='tooltiptext'>contour, image, and persp methods that display the fitted surface for an lm object involving two or more numerical predictors.<\/span><\/div>, <div class='tooltip'>PQ<span class='tooltiptext'>Use of one of these functions in a model is how you specify the portion of the model that is to be regarded as a response-surface component.<\/span><\/div>, <div class='tooltip'>recode.data<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>rsm<span class='tooltiptext'>Fit a linear model with a response-surface component, and produce appropriate analyses and summaries.<\/span><\/div>, <div class='tooltip'>SO<span class='tooltiptext'>Use of one of these functions in a model is how you specify the portion of the model that is to be regarded as a response-surface component.<\/span><\/div>, <div class='tooltip'>star<span class='tooltiptext'>These functions generate central-composite designs, or building blocks thereof.They allow for flexible choices of replications, aliasing of predictors and fractional blocks, and choices of axis or star points.<\/span><\/div>, <div class='tooltip'>stdorder<span class='tooltiptext'>This implements the rsm package's building-block provisions for handlingsequences of experiments. We often want to join two or more designs into one blocked design for purposes of analysis.<\/span><\/div>, <div class='tooltip'>steepest<span class='tooltiptext'>These functions provide the path of steepest ascent (or descent) for a fitted response surface produced by rsm.<\/span><\/div>, <div class='tooltip'>summary.rsm<span class='tooltiptext'>Fit a linear model with a response-surface component, and produce appropriate analyses and summaries.<\/span><\/div>, <div class='tooltip'>truenames<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>truenames<-<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>TWI<span class='tooltiptext'>Use of one of these functions in a model is how you specify the portion of the model that is to be regarded as a response-surface component.<\/span><\/div>, <div class='tooltip'>val2code<span class='tooltiptext'>These functions facilitate the use of coded data in response-surface analysis.<\/span><\/div>, <div class='tooltip'>varfcn<span class='tooltiptext'>This function computes the scaled variance function for a design, based on a specified model. Options include plotting separate curves for each of several directions from the center, or a contour plot for two of the design factors.<\/span><\/div>, <div class='tooltip'>xs<span class='tooltiptext'>Fit a linear model with a response-surface component, and produce appropriate analyses and summaries.<\/span><\/div>","<div class='tooltip'>ccd_analysis<span class='tooltiptext'>For the simultaneous production of response surface analysis output by rsm used in combination with graphics in the second-order polynomial approach. The predictor variables must be named Factor1, Factor2, etc., while the response variable must be named Response. The output includes regression model fitting and plot of the fitted response surface.<\/span><\/div>, <div class='tooltip'>design_ccd<span class='tooltiptext'>Defines a rotatable central composite design (CCD) for experimentation with k quantitative factors and j replicates of the central point (which corresponds to the average treatment combination).<\/span><\/div>","<div class='tooltip'>blockMatrixDiagonal<span class='tooltiptext'>create a diagonal block matrix<\/span><\/div>, <div class='tooltip'>calcPower.SWD<span class='tooltiptext'>Calculation of power for a lmm with cluster as random effect, fixed timepoint effects, but set to null, TP number of timepoints, I number of cluster. The design matrix has to be coded by zeros and ones.<\/span><\/div>, <div class='tooltip'>completeDataDesignMatrix<span class='tooltiptext'>create design matrix for complete data within design<\/span><\/div>, <div class='tooltip'>CovMat.Design<span class='tooltiptext'>covariance matrix of the normal distribution under cluster randomized study type given a design and a type<\/span><\/div>, <div class='tooltip'>designMatrix<span class='tooltiptext'>create design matrix for a given setup of a stepped wedge design<\/span><\/div>, <div class='tooltip'>implemMatrix.SWD<span class='tooltiptext'>Creates a implementation matrix for a given stepped wedge design and grade of intervention implementation pattern<\/span><\/div>, <div class='tooltip'>sampleData<span class='tooltiptext'>Sample data (response) for given numbers of individuals by given a model (of a parallel, cross-sectional, stepped wedge design study)<\/span><\/div>","<div class='tooltip'>CI_g<span class='tooltiptext'>Calculates a confidence interval given a g_REML, a g_HPS, or a g_mlm object using either a central t distribution (for a symmetric interval) or a non-central t distribution (for an asymmetric interval).<\/span><\/div>, <div class='tooltip'>compare_RML_HPS<span class='tooltiptext'>Simulates data from a simple linear mixed effects model, then calculates REML and HPS effect size estimators as described in Pustejovsky, Hedges, & Shadish (2014).<\/span><\/div>, <div class='tooltip'>design_matrix<span class='tooltiptext'>Create a design matrix containing a linear trend, a treatment effect, and a trend-by-treatment interaction for a single-case design with m cases and n measurement occasions.<\/span><\/div>, <div class='tooltip'>effect_size_ABk<span class='tooltiptext'>Calculates the HPS effect size estimator based on data from an (AB)^k design, as described in Hedges, Pustejovsky, & Shadish (2012). Note that the data must contain one row per measurement occasion per subject.<\/span><\/div>, <div class='tooltip'>effect_size_MB<span class='tooltiptext'>Calculates the HPS effect size estimator based on data from a multiple baseline design, as described in Hedges, Pustejovsky, & Shadish (2013). Note that the data must contain one row per measurement occasion per subject.<\/span><\/div>, <div class='tooltip'>extract_varcomp<span class='tooltiptext'>Functions imported from the lmeInfo package. extract_varcomp g_mlm varcomp_vcov<\/span><\/div>, <div class='tooltip'>g_mlm<span class='tooltiptext'>Functions imported from the lmeInfo package. extract_varcomp g_mlm varcomp_vcov<\/span><\/div>, <div class='tooltip'>g_REML<span class='tooltiptext'>Estimates a design-comparable standardized mean difference effect size based on data from a multiple baseline design, using adjusted REML method as described in Pustejovsky, Hedges, & Shadish (2014). Note that the data must contain one row per measurement occasion per case.<\/span><\/div>, <div class='tooltip'>graph_SCD<span class='tooltiptext'>Graphs single case design data for treatment reversal and multiple baseline designs.<\/span><\/div>, <div class='tooltip'>Info_Expected_lmeAR1<span class='tooltiptext'>Calculates the expected information matrix from a fitted linear mixed effectsmodel with AR(1) correlation structure in the level-1 errors.<\/span><\/div>, <div class='tooltip'>phase_pairs<span class='tooltiptext'>Calculate phase-pairs based on phases and session numbering.<\/span><\/div>, <div class='tooltip'>preprocess_SCD<span class='tooltiptext'>Clean single case design data for treatment reversal and multiple baseline designs.<\/span><\/div>, <div class='tooltip'>session_by_treatment<span class='tooltiptext'>Calculate session-by-treatment interactions based on phases and session numbering.<\/span><\/div>, <div class='tooltip'>shine_scd<span class='tooltiptext'>An interactive shiny interface for estimating design-comparable standardized mean difference effect sizes from single-case designs. Estimation methods for multiple baseline and treatment reversal designs are available.<\/span><\/div>, <div class='tooltip'>simulate_MB2<span class='tooltiptext'>Simulates data from a linear mixed effects model, then calculates REML effect size estimator as described in Pustejovsky, Hedges, & Shadish (2014).<\/span><\/div>, <div class='tooltip'>simulate_MB4<span class='tooltiptext'>Simulates data from a linear mixed effects model, then calculates REML effect size estimator as described in Pustejovsky, Hedges, & Shadish (2014).<\/span><\/div>, <div class='tooltip'>varcomp_vcov<span class='tooltiptext'>Functions imported from the lmeInfo package. extract_varcomp g_mlm varcomp_vcov<\/span><\/div>","<div class='tooltip'>bargraph.CI<span class='tooltiptext'>Barplot of the mean and standard error (or other summary statistics) of a response variable for one-way or higher experimental designs.<\/span><\/div>, <div class='tooltip'>lineplot.CI<span class='tooltiptext'>Lineplot of the mean and standard error (or other summary statistics) of a response variable for one-way or higher experimental designs.<\/span><\/div>, <div class='tooltip'>se<span class='tooltiptext'>This function computes the standard error of the values in a vector 'x'. If 'na.rm' is 'TRUE' then missing values are removed before computation proceeds.<\/span><\/div>","<div class='tooltip'>Ski.Mack<span class='tooltiptext'>The Skillings-Mack test statistic is a generalization of the statistic used in Friedman's ANOVA method and in Durbin's rank test. This nonparametric statistical test is useful for the data obtained from block designs with missing observations occurring randomly. The results are the Skillings-Mack statistic, p-values based on the chi-squared distribution and Monte Carlo method. The Monte Carlo method is recommended for approximating p-values when there are many ties and/or small designs with missing values are conducted.<\/span><\/div>","<div class='tooltip'>%>%<span class='tooltiptext'>re-export magrittr pipe operator<\/span><\/div>, <div class='tooltip'>contr.simplex<span class='tooltiptext'>Generates orthonormal (orthogonal and normalized) contrasts. Each row is the vertex of an N-dimensional simplex. The only exception are contrasts for the 2-level case, which return 1 and -1.<\/span><\/div>, <div class='tooltip'>eval_design<span class='tooltiptext'>Evaluates the power of an experimental design, for normal response variables,given the design's run matrix and the statistical model to be fit to the data.Returns a data frame of parameter and effect powers. Designs canconsist of both continuous and categorical factors. By default, eval_designassumes a signal-to-noise ratio of 2, but this can be changed with theeffectsize or anticoef parameters.<\/span><\/div>, <div class='tooltip'>eval_design_custom_mc<span class='tooltiptext'>Evaluates the power of an experimental design, given its run matrix and thestatistical model to be fit to the data, using monte carlo simulation. Simulated data is fit using auser-supplied fitting library and power is estimated by the fraction of times a parameter is significant. Returnsa data frame of parameter powers.<\/span><\/div>, <div class='tooltip'>eval_design_mc<span class='tooltiptext'>Evaluates the power of an experimental design, given the run matrix and thestatistical model to be fit to the data, using monte carlo simulation. Simulated data is fit using ageneralized linear modeland power is estimated by the fraction of times a parameter is significant.Returns a data frame of parameter powers.<\/span><\/div>, <div class='tooltip'>eval_design_survival_mc<span class='tooltiptext'>Evaluates power for an experimental design in which the response variable may beright- or left-censored. Power is evaluated with a Monte Carlo simulation,using the survival package and survreg to fit the data. Split-plot designs are not supported.<\/span><\/div>, <div class='tooltip'>gen_design<span class='tooltiptext'>Creates an experimental design given a model, desired number of runs, and a data frame of candidatetest points. gen_design chooses points from the candidate set and returns a design that is optimal for the givenstatistical model.<\/span><\/div>, <div class='tooltip'>get_attribute<span class='tooltiptext'>Returns one or more of underlying attributes used in design generation/evaluation<\/span><\/div>, <div class='tooltip'>get_optimality<span class='tooltiptext'>Returns a list of optimality values (or one value in particular)<\/span><\/div>, <div class='tooltip'>plot_correlations<span class='tooltiptext'>Plots design diagnostics<\/span><\/div>, <div class='tooltip'>plot_fds<span class='tooltiptext'>Creates a fraction of design space plot<\/span><\/div>, <div class='tooltip'>skprGUI<span class='tooltiptext'>skprGUI provides a graphical user interface to skpr, within R Studio.<\/span><\/div>, <div class='tooltip'>skprGUIbrowser<span class='tooltiptext'>skprGUI provides a graphical user interface to skpr, in an external browser.<\/span><\/div>, <div class='tooltip'>skprGUIserver<span class='tooltiptext'>skprGUI provides a graphical user interface to skpr, within R Studio.<\/span><\/div>","<div class='tooltip'>maximinSLHD<span class='tooltiptext'>Generate the optimal Latin hypercube designs when t=1 and the optimal sliced Latin hypercube designs when t>1 for computer experiments. The maximin distance criterion is adopted as the optimality criterion.<\/span><\/div>","<div class='tooltip'>computeResponseSummary<span class='tooltiptext'>Compute the response end-point summary for interim data. This will include the proportion of responses, the proportion of people on treatment and control etc.<\/span><\/div>, <div class='tooltip'>executeSP23Design<span class='tooltiptext'>This function is designed to be used in the field. Assuming a particular design is chosen, it conducts the interim analysis for a specific calendar time and provides the means for deciding whether to stop for futility or efficacy.<\/span><\/div>, <div class='tooltip'>exploreSP23Design<span class='tooltiptext'>Explore a chosen design by generating simulated datasets and storing data from simulations for further analysis<\/span><\/div>, <div class='tooltip'>generateClinicalTrialData<span class='tooltiptext'>The data is generated according to the model specified in the reference below, specifically, \\lambda(t \\mid Y,Z)=\\lambda_0(t) \\exp(\\alpha Y + \\beta Z + \\gamma Y Z)<\/span><\/div>, <div class='tooltip'>generateInterimData<span class='tooltiptext'>Generate interim data at a given time from a dataset<\/span><\/div>, <div class='tooltip'>generateSP23Design<span class='tooltiptext'>Generate a seamless Phase II-III design object given some the true parameters and clinical trial parameters<\/span><\/div>, <div class='tooltip'>performInterimLook<span class='tooltiptext'>Perform an interim look in a seamless Phase II-III clinical trial<\/span><\/div>, <div class='tooltip'>resetSP23Design<span class='tooltiptext'>Reset the design object so that counts and results are zeroed out<\/span><\/div>","<div class='tooltip'>findparameter<span class='tooltiptext'>Calibrate the design parameters for the Single-to-double Arm Transition Design<\/span><\/div>, <div class='tooltip'>rejectprob<span class='tooltiptext'>Calculate the probability of rejecting the null hypothesis at the end of the second stage in the Single-to-double Arm Transition Design<\/span><\/div>","<div class='tooltip'>alpha_standardized<span class='tooltiptext'>Compute standardized alpha level based on unstandardized alpha level and the number of observations N.<\/span><\/div>, <div class='tooltip'>ANOVA_compromise<span class='tooltiptext'>Justify your alpha level by minimizing or balancing Type 1 and Type 2 error rates for ANOVAs.<\/span><\/div>, <div class='tooltip'>ANOVA_design<span class='tooltiptext'>Design function used to specify the parameters to be used in simulations<\/span><\/div>, <div class='tooltip'>ANOVA_exact<span class='tooltiptext'>Simulates an exact dataset (mu, sd, and r represent empirical, not population, mean and covariance matrix) from the design to calculate power<\/span><\/div>, <div class='tooltip'>ANOVA_exact2<span class='tooltiptext'>Simulates an exact dataset (mu, sd, and r represent empirical, not population, mean and covariance matrix) from the design to calculate power<\/span><\/div>, <div class='tooltip'>ANOVA_power<span class='tooltiptext'>Simulation function used to estimate power<\/span><\/div>, <div class='tooltip'>emmeans_power<span class='tooltiptext'>Computes power based on t value and degrees of freedom for contrasts.Do not use to calculate \"observed power\" for empirical datasets(Hoenig & Heisey, 2001).<\/span><\/div>, <div class='tooltip'>mu_from_ES<span class='tooltiptext'>Convenience function to calculate the means for between designs with one factor (One-Way ANOVA). Can be used to determine the means that should yield a specified effect sizes (expressed in Cohen's f).<\/span><\/div>, <div class='tooltip'>optimal_alpha<span class='tooltiptext'>Justify your alpha level by minimizing or balancing Type 1 and Type 2 error rates.<\/span><\/div>, <div class='tooltip'>p_standardized<span class='tooltiptext'>Compute standardized alpha level based on unstandardized alpha level and the number of observations N.<\/span><\/div>, <div class='tooltip'>plot_power<span class='tooltiptext'>Convenience function to plot power across a range of sample sizes.<\/span><\/div>, <div class='tooltip'>power_oneway_between<span class='tooltiptext'>Analytic power calculation for one-way between designs.<\/span><\/div>, <div class='tooltip'>power_oneway_within<span class='tooltiptext'>Analytic power calculation for one-way within designs.<\/span><\/div>, <div class='tooltip'>power_standardized_alpha<span class='tooltiptext'>Because the standardized alpha depends on the sample size (N), and the power depends on the sample size, deciding upon the sample size to achieve a desired power requires an iterative procedure. Increasing the sample size reduces the standardized alpha, which requires an increase in the sample size for the power analysis, which reduces the standardized alpha. This function takes a power analysis function that outputs the power as a function of the desired power, the alpha level, as a function of N(x).<\/span><\/div>, <div class='tooltip'>power_threeway_between<span class='tooltiptext'>Analytic power calculation for three-way between designs.<\/span><\/div>, <div class='tooltip'>power_twoway_between<span class='tooltiptext'>Analytic power calculation for two-way between designs.<\/span><\/div>, <div class='tooltip'>power.ftest<span class='tooltiptext'>Compute power of test or determine parameters to obtain target power. Inspired by the pwr.f2.test function in the pwr package, but allows for varying noncentrality parameter estimates for a more liberal (default in pwr.f2.test) or conservative (default in this function) estimates (see Aberson, Chapter 5, pg 72).<\/span><\/div>, <div class='tooltip'>Superpower_options<span class='tooltiptext'>Global Superpower options are used, for example, by ANOVA_exact (et al.)and ANOVA_power. But can be changed in each functions directly usingan argument (which has precedence over the global options).<\/span><\/div>","<div class='tooltip'>gofm<span class='tooltiptext'>This function provides rho-squared and rho-squared adjusted by the number of estimated coefficients.<\/span><\/div>, <div class='tooltip'>Lma.design<span class='tooltiptext'>This function creates a choice experiment design according to the L^MA method.<\/span><\/div>, <div class='tooltip'>make.dataset<span class='tooltiptext'>This function makes a data set used for a conditional logit model analysis with the function clogit in the package survival or for a binary choice model analysis with the function glm in the package stats.<\/span><\/div>, <div class='tooltip'>make.design.matrix<span class='tooltiptext'>This function converts a choice experiment design created by the function Lma.design or rotation.design into a design matrix suitable for a conditional logit model analysis with the function clogit in the package survival, or for a binary choice model analysis with the function glm in the package stats.<\/span><\/div>, <div class='tooltip'>mwtp<span class='tooltiptext'>This function calculates the marginal willingness to pay for the attributes and/or levels of the estimated model.<\/span><\/div>, <div class='tooltip'>questionnaire<span class='tooltiptext'>This function converts a choice experiment design created by the function Lma.design or rotation.design into choice experiment questions used in a questionnaire survey.<\/span><\/div>, <div class='tooltip'>rotation.design<span class='tooltiptext'>This function creates a choice experiment design according to the rotation or mix-and-match method.<\/span><\/div>","<div class='tooltip'>blkDiag<span class='tooltiptext'>blkDiag returns block diagonal matrix based on specified square blocks (either as an array or a list).<\/span><\/div>, <div class='tooltip'>swDsn<span class='tooltiptext'>swDsn returns a SW CRT study design object based on a repeated cross-sectional sampling scheme. All clusters that start the intervention at a given time point are collectively referred to as a wave or sequence. There many be a variable number of clusters in each wave. By default, all clusters are assumed to start in the control condition. Fractional treatment effect may be specified for each time after the intervention is introduced. Additional observations may be added to the end of the trial after the intervention has been introduced in all clusters. For incorporating transition periods where no data is collected, see swPwr. swDsn is used by other functions in this package.<\/span><\/div>, <div class='tooltip'>swPlot<span class='tooltiptext'>swPlot returns plot of the mean response versus time based on waves and/or clusters from a SW CRT.<\/span><\/div>, <div class='tooltip'>swPwr<span class='tooltiptext'>swPwr returns (two-sided) power of treatment effect (\\theta) for the specified SW CRT design using a linear models weighted least squares (WLS) approach. The response/outcome of interest can be binomial or Gaussian distributed and is assumed to come from a model with random intercepts, random treatment effects, and random cluster-specific time effects. Variance components can be specified using either tau, eta, rho, and gamma, or icc and cac (see details). If a random intercepts only model is used (i.e., eta and gamma are 0 and n is constant over clusters and time), then the power calculation is comparable to the closed-form formula of [Hussey and Hughes, 2007].<\/span><\/div>, <div class='tooltip'>swSim<span class='tooltiptext'>swSim returns individual-level data set of a SW CRT study design for the specified number of clusters per wave, fractional treatment effect at each time after crossing over from control, time (standard SW CRT time computed; extending trial beyond standard time needs to be specified by the user), family (and link function), number of individuals per cluster per wave, mean in control arm, mean in treatment arm, time effect, pooled standard deviation for both arms, standard deviation of random intercepts, standard deviation of random treatment effects, correlation between random intercepts and random treatment effects, standard deviation of random time effects, time point label, and option to simulate data with time on treatment lag. Alternatively, for a Gaussian family standard deviations of random effects may be specified using ICC and CAC; see swPwr details.<\/span><\/div>, <div class='tooltip'>swSummary<span class='tooltiptext'>swSummary returns the mean, sum, and number of non-missing values for the response/outcome variable of interest for each cluster at each time point from a SW CRT.<\/span><\/div>","<div class='tooltip'>opt_alloc_RET<span class='tooltiptext'>Calculate optimal sample size allocation for Wald-type test forsuperiority or non-inferiority of the experimental treatment versusreference treatment with respect to placebo<\/span><\/div>, <div class='tooltip'>power_RET<span class='tooltiptext'>Compute power, sample size, or level of significance forWald-type test for non-inferiority or superiority of the experimental treatmentversus reference treatment with respect to placebo.<\/span><\/div>, <div class='tooltip'>test_RET<span class='tooltiptext'>Wald-type test for superiority/non-inferiority of theexperimental treatment versus reference treatment with respect to placebo.<\/span><\/div>","<div class='tooltip'>traitstats.envcorl<span class='tooltiptext'>An extract of environment correlation coefficient for between traits<\/span><\/div>, <div class='tooltip'>traitstats.envcov<span class='tooltiptext'>Computation of environment variance and covariance matrix.<\/span><\/div>, <div class='tooltip'>traitstats.genocorl<span class='tooltiptext'>An extract of genotypic correlation coefficient for between traits<\/span><\/div>, <div class='tooltip'>traitstats.genocov<span class='tooltiptext'>Computation of genotypic variance and covariance matrix.<\/span><\/div>, <div class='tooltip'>traitstats.genpar<span class='tooltiptext'>traitstats.genpar performs genetic variability analysis for all the traits in the DataFile.<\/span><\/div>, <div class='tooltip'>traitstats.phenocorl<span class='tooltiptext'>An extract of phenotypic correlation coefficient for between traits<\/span><\/div>, <div class='tooltip'>traitstats.phenocov<span class='tooltiptext'>Computation of phenotypic variance and covariance matrix.<\/span><\/div>, <div class='tooltip'>traitstats.rbd<span class='tooltiptext'>Bulk RCBD data analysis for ANOVA and Descriptive statistics<\/span><\/div>, <div class='tooltip'>traitstats.rbdmeantable<span class='tooltiptext'>Individual treatment mean table along with Grand trait mean, CV, SE(m) and CD.<\/span><\/div>, <div class='tooltip'>traitstats.rbdstats<span class='tooltiptext'>Descriptive Statistics for Trait Mean, Minimum, Maximum, SE(m), SE(d) and CV.<\/span><\/div>, <div class='tooltip'>traitstats.rbdsummary<span class='tooltiptext'>RCBD data analysis report in PDF formate compiling the ANOVA, Descriptive Statistics and Genetic Parameter along with Data distribution graph.<\/span><\/div>","<div class='tooltip'>augbin_2t_1a_fit<span class='tooltiptext'>Class used by trialr to fit Wason & Seaman's Augmented Binary method insingle arm trials with two post-baseline tumour assessments.<\/span><\/div>, <div class='tooltip'>augbin_fit<span class='tooltiptext'>Class used by trialr to fit Wason & Seaman's Augmented Binary method.<\/span><\/div>, <div class='tooltip'>binary_prob_success<span class='tooltiptext'>Calculate the binary probability of success.Calculate the binary probability of success from an augbin_2t_1a_fit object.<\/span><\/div>, <div class='tooltip'>careful_escalation<span class='tooltiptext'>Dose selection function that avoids dose-skipping in escalation andadvocates stopping when there is sufficient evidence that the risk oftoxicity at a reference dose exceeds some threshold.<\/span><\/div>, <div class='tooltip'>closest_to_target<span class='tooltiptext'>Get index of element in vector with value closest to a target<\/span><\/div>, <div class='tooltip'>crm_codified_dose_logistic<span class='tooltiptext'>Calculate the codified CRM doses that map to probability oftoxicity prob_tox in a logistic model with expected values forintercept and gradient. I.e. find x[i] such thatlogit(p[i]) = \\alpha + \\beta x[i], were p isprob_tox.<\/span><\/div>, <div class='tooltip'>crm_dtps<span class='tooltiptext'>Calculate dose-transition pathways (DTPs, Yap et al, 2017) for a dose-findingtrial using the continual reassessment method (CRM) design. DTPs are aglimpse into the future for an in-progress trial. They tell us what the modelwould advise for all feasible future outcomes. They can be used in the designstages to detect possible undesirable behaviour. They can be used during thetrial to aid planning and understanding.<\/span><\/div>, <div class='tooltip'>crm_fit<span class='tooltiptext'>Class of model fit by trialr using the CRM dose-finding design.<\/span><\/div>, <div class='tooltip'>crm_params<span class='tooltiptext'>Container class for parameters to fit the CRM models in trialr.<\/span><\/div>, <div class='tooltip'>crm_path_analysis<span class='tooltiptext'>Fit a continuous reassessment method (CRM) model to the outcomes cumulativelyobserved at the end of each cohort in a trial pathway. E.g. if the trialpathway is 1NN 2NN 3NT, we have three cohorts of two patients. This functionwill fit the model to the following four states: before any patients havebeen evaluated; after 1NN; after 1NN 2NN; and finally after 1NN 2NN 3NT.This allows us to analyse how the trial model is evolving in its estimationas trial data is accumulated.<\/span><\/div>, <div class='tooltip'>crm_prior_beliefs<span class='tooltiptext'>Infer the prior beliefs consistent with the parameters and modelform for a CRM dose-finding trial. This function could be interpreted asfitting the model to no data, thus examining the beliefs on dose-toxicitythat are suggested by the parameter priors alone. This function provides thetask analagous to stan_crm before any data has been collected.<\/span><\/div>, <div class='tooltip'>df_parse_outcomes<span class='tooltiptext'>Parse a string of dose-finding trial outcomes to the binaryvector notation required by Stan for model invocation. The outcome stringdescribes the doses given and outcomes observed. The format of the string isthe pure phase I analogue to that described in Brock et al. (2017).The letters T and N are used to represents patients that experienced(T)oxicity and (N)o toxicity. These letters are concatenated after numericaldose-levels to convey the outcomes of cohorts of patients.For instance, 2NNT represents a cohort of three patients that weretreated at dose-level 2, one of whom experienced toxicity, and two that didnot. The results of cohorts are separated by spaces. Thus, 2NNT 1NNextends our previous example, where the next cohort of two were treated atdose-level 1 and neither experienced toxicity. See examples.<\/span><\/div>, <div class='tooltip'>dose_finding_fit<span class='tooltiptext'>Class of dose-finding model fit by trialr using Stan.<\/span><\/div>, <div class='tooltip'>dose_finding_path_node<span class='tooltiptext'>A pathway in a dose-finding trial is a series of successive analyses. Forinstance, the model will likely be fit to all of the outcomes observedat the end of the first cohort, the second cohort, etc. This class holds theelements reflecting the analysis, and the place of this analysis in thepathway.<\/span><\/div>, <div class='tooltip'>eff_at_dose<span class='tooltiptext'>Get the number of efficacy events seen at the doses under investigation.<\/span><\/div>, <div class='tooltip'>efftox_analysis_to_df<span class='tooltiptext'>Convenient function to turn an efftox_fitinto a data.frame.<\/span><\/div>, <div class='tooltip'>efftox_contour_plot<span class='tooltiptext'>Plot EffTox utility contours. The probability of efficacy ison the x-axis and toxicity on the y-axis. The zero-utility curve is plottedbolder. The three \"hinge points\" are plotted as blue triangles. OptionalProb(Efficacy) vs Prob(Toxicity) points can be added; these are shown asred numerals, enumerated in the order provided.<\/span><\/div>, <div class='tooltip'>efftox_dtps<span class='tooltiptext'>Calculate dose-transition pathways for an EffTox study.The function efftox_dtps_to_dataframe performs a similarfunction, but is much less-flexible.<\/span><\/div>, <div class='tooltip'>efftox_dtps_to_dataframe<span class='tooltiptext'>Calculate dose-transition pathways for an EffTox study.Note that TODO TODO TODO<\/span><\/div>, <div class='tooltip'>efftox_fit<span class='tooltiptext'>Phase I/II dose-finding trials, i.e. those that search for a dose my efficacyand toxicity outcomes search for the optimal biological dose (OBD), ratherthan the maximum tolerated dose (MTD) that is typically sought be traditionaltoxicity-only dose-finding.<\/span><\/div>, <div class='tooltip'>efftox_get_tox<span class='tooltiptext'>Get the probability of toxicity for probability-of-efficacy andutility pairs<\/span><\/div>, <div class='tooltip'>efftox_parameters_demo<span class='tooltiptext'>Get parameters to run the EffTox demo. These match those usedto demonstrate EffTox in Thall et al. 2014.<\/span><\/div>, <div class='tooltip'>efftox_params<span class='tooltiptext'>Container class for parameters to fit the EffTox model in trialr.<\/span><\/div>, <div class='tooltip'>efftox_parse_outcomes<span class='tooltiptext'>Parse a string of EffTox outcomes to the binary vector notationrequired by Stan for model invocation. The outcome string describes the dosesgiven and outcomes observed. The format of the string is described in Brocket al. (2017). The letters E, T, N and B are used to represents patients thatexperienced (E)fficacy only, (T)oxicity only, (B)oth efficacy and toxicity,and (N)either. These letters are concatenated after numerical dose-levels toconvey the outcomes of cohorts of patients. For instance, 2ETBrepresents a cohort of three patients that were treated at dose-level 2, andexperienced efficacy, toxicity and both events, respectively. The results ofcohorts are separated by spaces. Thus, 2ETB 1NN extends our previousexample, where the next cohort of two were treated at dose-level 1 and bothpatients experienced neither efficacy nor toxicity. See examples.We present the notation in the EffTox setting but it is applicable ingeneral seamless phase I/II dose-finding scenarios.<\/span><\/div>, <div class='tooltip'>efftox_path_analysis<span class='tooltiptext'>Fit a EffTox model to the outcomes cumulatively observed at the end of eachcohort in a trial pathway. E.g. if the trial pathway is 1EN 2NN 3BT, we havethree cohorts of two patients. This function will fit the model to thefollowing four states: before any patients have been evaluated; after 1EN;after 1EN 2NN; and finally after 1EN 2NN 3BT. This allows us to analyse howthe trial model is evolving in its estimation as trial data is accumulated.<\/span><\/div>, <div class='tooltip'>efftox_priors<span class='tooltiptext'>Simple class to hold prior hyperparameters for the EffTox model.<\/span><\/div>, <div class='tooltip'>efftox_simulate<span class='tooltiptext'>Run EffTox simulations for assumed true efficacy and toxicitycurves.<\/span><\/div>, <div class='tooltip'>efftox_solve_p<span class='tooltiptext'>Calculate the p-index for EffTox utility contours so that the neutral utilitycontour intersects the following points in theProb(Efficacy) - Prob(Toxicity) plane:(eff0, 0), (1, tox1) and (eff_star, tox_star)<\/span><\/div>, <div class='tooltip'>efftox_superiority<span class='tooltiptext'>Get a dose-superiority matrix from an EffTox dose analysis.EffTox seeks to choose the dose with the highest utility, thus superiorityis inferred by posterior utility. The item in row i, col j is the posteriorprobability that the utility of dose j exceeds that of dose i.<\/span><\/div>, <div class='tooltip'>efftox_utility<span class='tooltiptext'>Get the utility of efficacy & toxicity probability pairs<\/span><\/div>, <div class='tooltip'>efftox_utility_density_plot<span class='tooltiptext'>Plot densities of EffTox dose utilities. Optionally plot only asubset of the doses by specifying the doses parameter. This functionrequires ggplot2 be installed.<\/span><\/div>, <div class='tooltip'>get_efftox_priors<span class='tooltiptext'>Get normal prior hyperparameters for the EffTox model using the algorithmpresented in Thall et al. (2014) that targets a family of priors with apre-specified effective sample size (ESS).<\/span><\/div>, <div class='tooltip'>n_at_dose<span class='tooltiptext'>Get the number of patients treated at the doses under investigation.<\/span><\/div>, <div class='tooltip'>parse_dose_finding_outcomes<span class='tooltiptext'>Parse a string of dose-finding trial outcomesParse a string of dose-finding trial outcomes to a list.The outcome string describes the doses given, outcomes observed and thetiming of analyses that recommend a dose. The format of the string isthe pure phase I analogue to that described in Brock _et al_. (2017).The letters T and N are used to represents patients that experienced(T)oxicity and (N)o toxicity. These letters are concatenated after numericaldose-levels to convey the outcomes of cohorts of patients.For instance, 2NNT represents a cohort of three patients that weretreated at dose-level 2, one of whom experienced toxicity, and two that didnot. The results of cohorts are separated by spaces and it is assumed that adose-finding decision takes place at the end of a cohort. Thus,2NNT 1NN builds on our previous example, where the next cohort of twowere treated at dose-level 1 and neither of these patients experiencedtoxicity. See examples.<\/span><\/div>, <div class='tooltip'>parse_eff_tox_dose_finding_outcomes<span class='tooltiptext'>Parse a string of phase I/II dose-finding trial outcomes.Phase I/II trials conduct dose-finding by efficacy and toxicity outcomes.Parse a string of phase I/II dose-finding outcomes to a list.The outcome string describes the doses given, efficacy and toxicity outcomesobserved and the timing of analyses that recommend a dose. The format of thestring is described in Brock _et al_. (2017). The letters E, T, N & B areused to represents patients that experienced (E)fficacy, (T)oxicity,(N)either and (B)oth. These letters are concatenated after numericaldose-levels to convey the outcomes of cohorts of patients.For instance, 2NET represents a cohort of three patients that weretreated at dose-level 2, one of whom experienced toxicity only, one thatexperienced efficacy only, and one that had neither.The results of cohorts are separated by spaces and it is assumed that adose-finding decision takes place at the end of a cohort. Thus,2NET 1NN builds on our previous example, where the next cohort of twowere treated at dose-level 1 and neither of these patients experiencedeither event. See examples.<\/span><\/div>, <div class='tooltip'>peps2_get_data<span class='tooltiptext'>Get data to run the BEBOP model in the PePS2 trial. The trialinvestigates pembrolizumab in non-small-cell lung cancer. Patients may bepreviously treated (PT) or treatment naive (TN). Pembro response rates inlung cancer have been shown to increase with PD-L1 tumour proportion score.PD-L1 score is measured at baseline. Each patient belongs to one of the Low,Medium or High categories. These two baseline variables stratify the patientpopulation and are used as predictive variables to stratify the analysis.The BEBOP model studies co-primary efficacy and toxicity outcomes in thepresence of predictive data. Thus, PePS2 studies efficacy and toxicity in6 distinct cohorts:TN Low, TN Medium, TN High, PT Low, PT Medium, PT High.The design admits all-comers and does not target specific sample sizes in theindividual cohorts.Hyperprior parameters have defaults to match those used in PePS2, but all maybe overridden.The returned object includes randomly-sampled outcomes, as well as parametersto run the model. These are all combined in the same list object for passingto RStan, as is the convention.See the accompanying vignette for a full description.<\/span><\/div>, <div class='tooltip'>peps2_process<span class='tooltiptext'>Process RStan samples from a BEBOP model fit to PePS2 data. Thisstep lets us make inferences about whether the modelled efficacy and toxicityprobabilities suggest the treatment is acceptable in each of the cohortsunder study.The parameters have default values to match those used in the PePS2 trial.See the accompanying vignette for a full description.<\/span><\/div>, <div class='tooltip'>prior_predictive_augbin_2t_1a<span class='tooltiptext'>Sample data from the prior predictive distributions of the two-period, singlearm Augmented Binary model, subject to chosen prior parameters.<\/span><\/div>, <div class='tooltip'>prob_success<span class='tooltiptext'>Calculate the probability of success.Calculate the probability of success for an augbin_2t_1a_fit object.<\/span><\/div>, <div class='tooltip'>prob_tox_exceeds<span class='tooltiptext'>Calculate the probability that the rate of toxicity exceeds some thresholdCalculate the probability that the rate of toxicity exceeds some threshold<\/span><\/div>, <div class='tooltip'>ranBin2<span class='tooltiptext'>This function is reproduced from the binarySimCLF packageon CRAN. The original package appears no longer to be maintained.View the original source at: https://github.com/cran/binarySimCLF/blob/master/R/ranBin2.R<\/span><\/div>, <div class='tooltip'>rlkjcorr<span class='tooltiptext'>This function was copied from Richard McElreath's rethinking package hostedat https://github.com/rmcelreath/rethinking. In turn, he appears to havecopied it from Ben Bolker's rLKJ function from the emdbook package, althoughI cannot find it there (else I would have imported it).<\/span><\/div>, <div class='tooltip'>spread_paths<span class='tooltiptext'>Spread the information in dose_finding_paths object to a wide data.frame format.<\/span><\/div>, <div class='tooltip'>stan_augbin<span class='tooltiptext'>Phase II clinical trials in oncology commonly assess response as a key outcomemeasure. Patients achieve a RECIST response if their tumour size post-baselinehas changed in size by some threshold amount and they do not experiencenon-shrinkage failure. An example of non-shrinkage failure is the appearanceof new lesions. As a dichtotomisation of the underlying continuous tumour sizemeasurement, RECIST response is inefficient. Wason & Seaman introduced theAugmented Binary method to incorporate mechanisms for non-shrinkage failurewhilst modelling the probability of response based on the continuous tumoursize measurements. See model-specific sections below, and the references.<\/span><\/div>, <div class='tooltip'>stan_augbin_demo<span class='tooltiptext'>This function exist mostly to demonstrate things you can do to instances ofaugbin_fit without having to paste into each example the notinconsiderable blob of code to sample outcomes and fit the model.<\/span><\/div>, <div class='tooltip'>stan_crm<span class='tooltiptext'>Fit a continual reassessment method (CRM) model for dose-finding using Stanfor full Bayesian inference. There are several likelihood and priorcombinations supported. See model-specific sections below.<\/span><\/div>, <div class='tooltip'>stan_efftox<span class='tooltiptext'>Fit an EffTox model using Stan for full Bayesian inference.<\/span><\/div>, <div class='tooltip'>stan_efftox_demo<span class='tooltiptext'>Fit the EffTox model presented in Thall et al. (2014) using Stan for fullBayesian inference.<\/span><\/div>, <div class='tooltip'>stan_hierarchical_response_thall<span class='tooltiptext'>Fit the hierarchical response model to exchangeable groups described byThall et al. (2003).<\/span><\/div>, <div class='tooltip'>stan_nbg<span class='tooltiptext'>Fit Neuenschwander, Branson & Gsponer logit model for dose-finding using Stanfor full Bayesian inference.<\/span><\/div>, <div class='tooltip'>stan_peps2<span class='tooltiptext'>The PePS2 trialinvestigates pembrolizumab in non-small-cell lung cancer. Patients may bepreviously treated (PT) or treatment naive (TN). Response rates inlung cancer have been shown to increase with PD-L1 tumour proportion score.PD-L1 score is measured at baseline. Each patient belongs to one of thecategories <1%, 1-49% or >=50%. PT vs TN status and PD-L1 category jointlystratify the patientpopulation and are used as predictive variables to stratify the analysis.The BEBOP model studies co-primary efficacy and toxicity outcomes in thepresence of predictive data. Thus, PePS2 studies efficacy and toxicity in6 distinct cohorts:TN Low, TN Medium, TN High, PT Low, PT Medium, PT High.The design admits all-comers and does not target specific sample sizes in theindividual cohorts.Hyperprior parameters have defaults to match those used in PePS2, but all maybe overridden.The returned object includes randomly-sampled outcomes, as well as parametersto run the model. These are all combined in the same list object for passingto RStan, as is the convention.See the accompanying vignette for a full description.<\/span><\/div>, <div class='tooltip'>total_weight_at_dose<span class='tooltiptext'>Get the total weight of patient outcomes at the doses under investigation.<\/span><\/div>, <div class='tooltip'>tox_at_dose<span class='tooltiptext'>Get the number of toxicity events seen at the doses under investigation.<\/span><\/div>, <div class='tooltip'>trialr_simulate<span class='tooltiptext'>This function is a fairly flexible way of running simulation studies intrialr, and beyond. It essentially uses delegates to perform this pattern:for i in 1:N: data = get_data_func() fit = fit_model_func(data) if summarise_func is null: sims[i] = fit else sims[i] = summarise_func(data, fit) endloopreturn sims<\/span><\/div>, <div class='tooltip'>weights_at_dose<span class='tooltiptext'>Get the weights of patient outcomes at the doses under investigation.<\/span><\/div>","<div class='tooltip'>adj.two<span class='tooltiptext'>adjust Zhong's 2-/3-stage design for over-/under-running<\/span><\/div>, <div class='tooltip'>dec.sim<span class='tooltiptext'>Run dose-finding simulations based on a customized decision table.<\/span><\/div>, <div class='tooltip'>dec.table<span class='tooltiptext'>Generate three stage dose finding decision table<\/span><\/div>, <div class='tooltip'>opt.design<span class='tooltiptext'>calculate optimal 2-/3-stage design given by Bob Zhong<\/span><\/div>, <div class='tooltip'>sl.sim<span class='tooltiptext'>Run dose-finding simulations based on a customized decision table for a list of scenarios.<\/span><\/div>","<div class='tooltip'>daniel.paper<span class='tooltiptext'>The daniel.paper function draws a graph-paper grid suitable for manually constructing a half-normal plot of effects. The paper is customized to the number of effects to be plotted, thus making the task as simple as possible.<\/span><\/div>, <div class='tooltip'>dot.id<span class='tooltiptext'>Flexible stacked-dot plots of relatively small samples, and provisions for identifying points. The dot plot is responsive to resizing of the plot window: a vertical resizing does not affect the vertical spacing of the dots, and a horizontal resizing may lead to re-binning to fit well within the range.<\/span><\/div>, <div class='tooltip'>dot.mod<span class='tooltiptext'>Flexible stacked-dot plots of relatively small samples, and provisions for identifying points. The dot plot is responsive to resizing of the plot window: a vertical resizing does not affect the vertical spacing of the dots, and a horizontal resizing may lead to re-binning to fit well within the range.<\/span><\/div>, <div class='tooltip'>dot.plot<span class='tooltiptext'>Flexible stacked-dot plots of relatively small samples, and provisions for identifying points. The dot plot is responsive to resizing of the plot window: a vertical resizing does not affect the vertical spacing of the dots, and a horizontal resizing may lead to re-binning to fit well within the range.<\/span><\/div>, <div class='tooltip'>eff.test<span class='tooltiptext'>These functions facilitate making inferences based on effect estimates in an unreplicated experiment, withn an underlying effect-sparsity model.<\/span><\/div>, <div class='tooltip'>gyates<span class='tooltiptext'>Implementations of Yates's method for obtaining factor effects; and reverse Yates's method for recovering response values; and a generalization for balanced unreplicated experiments having other than two-level factors.<\/span><\/div>, <div class='tooltip'>hnplot<span class='tooltiptext'>The hnplot function constructs a (half-) normal plot of effects (see Daniel 1959) that is traditionally used to identify active effects in a screening experiment. Reference lines and various other options and extensions are supported.<\/span><\/div>, <div class='tooltip'>ME<span class='tooltiptext'>Computes a pseudo standard error using any of a variety of built-in methods, or a user-supplied one.<\/span><\/div>, <div class='tooltip'>parplot<span class='tooltiptext'>Constructs a bar plot of ordered effects, along with cutoff values for the margin of error (ME) and simultaneous margin of error (SME). Such a plot is suggested in Lenth (1989), but other methods may be used for obtaining the ME and SME.<\/span><\/div>, <div class='tooltip'>PSE<span class='tooltiptext'>Computes a pseudo standard error using any of a variety of built-in methods, or a user-supplied one.<\/span><\/div>, <div class='tooltip'>ref.dist<span class='tooltiptext'>These functions facilitate making inferences based on effect estimates in an unreplicated experiment, withn an underlying effect-sparsity model.<\/span><\/div>, <div class='tooltip'>refplot<span class='tooltiptext'>The refplot function constructs a dot plot of effects along with a rweference distribution (either normal or simulated) to help in visually identifying active effects in a screening experiment.<\/span><\/div>, <div class='tooltip'>yates<span class='tooltiptext'>Implementations of Yates's method for obtaining factor effects; and reverse Yates's method for recovering response values; and a generalization for balanced unreplicated experiments having other than two-level factors.<\/span><\/div>","<div class='tooltip'>expmat<span class='tooltiptext'>Computes the matrix of spherical region moments for a given model formula and a vector of radii, and uses this tocalculate the mean spherical SPV for each of the radii. The function expmat calculatesthe matrix containing the exponents of each model factor within each model term as columns. Only simple formulae are allowed for. Only products of terms should be included incalls to I. The power operator ^ should be used insteadof sqrt. Models should contain only monomial terms.<\/span><\/div>, <div class='tooltip'>LHS<span class='tooltiptext'>Different versions of latin hypercube sampling (LHS): ordinary LHS, midpoint LHS, symmetric LHS or randomized symmetric LHS. LHS is a method for constructing space-filling designs. They can be more efficient for hypercuboidal design regions than other sampling methods.<\/span><\/div>, <div class='tooltip'>MLHS<span class='tooltiptext'>Different versions of latin hypercube sampling (LHS): ordinary LHS, midpoint LHS, symmetric LHS or randomized symmetric LHS. LHS is a method for constructing space-filling designs. They can be more efficient for hypercuboidal design regions than other sampling methods.<\/span><\/div>, <div class='tooltip'>RSLHS<span class='tooltiptext'>Different versions of latin hypercube sampling (LHS): ordinary LHS, midpoint LHS, symmetric LHS or randomized symmetric LHS. LHS is a method for constructing space-filling designs. They can be more efficient for hypercuboidal design regions than other sampling methods.<\/span><\/div>, <div class='tooltip'>runif_cube<span class='tooltiptext'>Sample uniformly in or on a hyperspheres or hypercubes.<\/span><\/div>, <div class='tooltip'>runif_sphere<span class='tooltiptext'>Sample uniformly in or on a hyperspheres or hypercubes.<\/span><\/div>, <div class='tooltip'>sampler<span class='tooltiptext'>This is a wrapper for the sampling funcions of the vdg package. It extracts design properties from the design passed to it to take appropriate samples.<\/span><\/div>, <div class='tooltip'>SLHS<span class='tooltiptext'>Different versions of latin hypercube sampling (LHS): ordinary LHS, midpoint LHS, symmetric LHS or randomized symmetric LHS. LHS is a method for constructing space-filling designs. They can be more efficient for hypercuboidal design regions than other sampling methods.<\/span><\/div>, <div class='tooltip'>spv<span class='tooltiptext'>Calculates the SPV for a sample of points in a design region of specified type. Sampling is doneby calling sampler.<\/span><\/div>, <div class='tooltip'>stdrange<span class='tooltiptext'>Simple functions for rescaling a data matrix to a coded design and back. stdrange convertsthe design in actual measurements into a coded design, while ustdrange reverses the process(if the correct arguments are given).<\/span><\/div>","<div class='tooltip'>Compare2FDS<span class='tooltiptext'>This function compares Fraction of Design Space Plots for two response surface designs with the same number of factors over the unit hypercube design space.<\/span><\/div>, <div class='tooltip'>Compare2Vdg<span class='tooltiptext'>This function calls the function Vardsgr which uses Vining's (1993) fortran code to get the coordinates of a two variance dispersion graph, and then makesthe plot.<\/span><\/div>, <div class='tooltip'>FDSPlot<span class='tooltiptext'>This function creates a Fraction of Design Space Plot over the hypercube design space from-1 to 1 on each component.<\/span><\/div>, <div class='tooltip'>Vdgraph<span class='tooltiptext'>This function calls the function Vardsgr which uses Vining's (1993) fortran code to get the coordinates of a variance dispersion graph, and then makesthe plot.<\/span><\/div>","<div class='tooltip'>Borkowski2003<span class='tooltiptext'>Retrieve small exact D-, A-, G-, and IV-Optimal designs generated by a genetic algorithm. These designs are catalogued by Borkowski (2003).<\/span><\/div>, <div class='tooltip'>cpv<span class='tooltiptext'>Create a variance dispersion graph for a response surface design in a cuboidal region.<\/span><\/div>, <div class='tooltip'>fds.cube<span class='tooltiptext'>Create fraction of design space plots for response surface designs in cuboidal regions.<\/span><\/div>, <div class='tooltip'>fds.sphere<span class='tooltiptext'>Create fraction of design space plots for response surface designs in spherical regions.<\/span><\/div>, <div class='tooltip'>gen.BBD<span class='tooltiptext'>Generate Box-Behnken designs for k = 3 to 7<\/span><\/div>, <div class='tooltip'>gen.CCD<span class='tooltiptext'>Generate central composite designs<\/span><\/div>, <div class='tooltip'>gen.Factr<span class='tooltiptext'>Generate factorial designs<\/span><\/div>, <div class='tooltip'>gen.HSCD<span class='tooltiptext'>Generate Hartley's small composite designs for k = 2 to 7<\/span><\/div>, <div class='tooltip'>gen.PBCD<span class='tooltiptext'>Generate Plackett-Burman composite designs proposed by Draper and Lin (1990) for k = 3 to 7<\/span><\/div>, <div class='tooltip'>gen.Roquemore<span class='tooltiptext'>Generate Roquemore (1976) hybrid designs for k = 3, 4, and 6. For k = 3, R310, R311A, and R311B will be produced, for k = 4, R416A, R416B, and R416C will be generated, and for k = 6 R628A and R628B will be given.<\/span><\/div>, <div class='tooltip'>gen.USD<span class='tooltiptext'>Generate uniform shell designs for k = 2 to 6<\/span><\/div>, <div class='tooltip'>hyperarcs.vdg<span class='tooltiptext'>Create a graph of scaled prediction variances for points in nested cubes (hyperarcs)<\/span><\/div>, <div class='tooltip'>spv<span class='tooltiptext'>Create variance dispesion graphs (VDGs) for response surface designs in spherical regions.<\/span><\/div>, <div class='tooltip'>spvcontour<span class='tooltiptext'>Create a contour plot of scaled prediction variances<\/span><\/div>","<div class='tooltip'>summary2<span class='tooltiptext'>S3 Summary function<\/span><\/div>, <div class='tooltip'>vtDecMap<span class='tooltiptext'>Summarize the posterior distribution of \\theta^{(l)}_{00}, \\theta^{(l)}_{01}, \\theta^{(l)}_{10}, \\theta^{(l)}_{11} and get information for making dose escalation decisions<\/span><\/div>, <div class='tooltip'>vtInterim<span class='tooltiptext'>Conduct an interim analysis for determining dose escalation actions<\/span><\/div>, <div class='tooltip'>vtPost<span class='tooltiptext'>Call STAN to draw posterior samples of the joint distribution ofimmunogenicity rate and toxicity risk<\/span><\/div>, <div class='tooltip'>vtPriorPar<span class='tooltiptext'>Get prior distribution parameters for partially parametric or partially parametric+ models<\/span><\/div>, <div class='tooltip'>vtScenario<span class='tooltiptext'>Simulation function. Get true \\theta's using marginal probabilities andodds ratio \\rho for all dose levels.<\/span><\/div>, <div class='tooltip'>vtShiny<span class='tooltiptext'>Call Shiny to run visit as a web-based application.<\/span><\/div>, <div class='tooltip'>vtSimu<span class='tooltiptext'>Simulate clinical trials with given settings for multiple times to evaluatethe study operating characteristics.<\/span><\/div>, <div class='tooltip'>vtSingleTrial<span class='tooltiptext'>Simulation function for simulating a single trial<\/span><\/div>, <div class='tooltip'>vtTrack<span class='tooltiptext'>Generate a plot representing the observed data and dose escalation decisions.<\/span><\/div>","<div class='tooltip'>ceff1<span class='tooltiptext'>Obtaining c-efficiency for estimating the ED50 for a given design under the 4-parameter logistic model. This also can be used to compute the c-efficiency for the ED50 under the 2 or 3-parameter logistic models by setting the parameters values differently.<\/span><\/div>, <div class='tooltip'>ceff2<span class='tooltiptext'>Obtaining c-efficiency for estimating the minimum effective dose MED for a given design under the 4-parameter logistic model. MED is the dose that produces a mean response of \\delta unites better than the minimum dose. This also can be used to compute the c-efficiency for MED under the 2 or 3-parameter logistic models by setting the parameter values differently.<\/span><\/div>, <div class='tooltip'>Deff<span class='tooltiptext'>Obtaining D-efficiency for estimating model parameters for a given design under the 4-parameter logistic model. This also can be used to compute the D-efficiency under the 2 or 3-parameter logistic models by setting the parameter values differently.<\/span><\/div>, <div class='tooltip'>MOPT<span class='tooltiptext'>Search the 3-objective optimal designs that maximizes a weighted average of efficiencies for estimating model parameters, the ED50, and the MED under the 4PL model and verifies the optimality of the generated design using the General Equivalence Theorem. This also can be used to search the 3-objective optimal designs under the 2 or 3-parameter logistic models by setting the parameter values differently.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>S.Weight<span class='tooltiptext'>Find optimal proportions of subjectives at the given dose levels to estimate the model parameters, the ED50 and the MED simultaneously under the 4-parameter logistiv model. This also can be used to find the optimal weights at the given dose under the 2 or 3-parameter logistic models by setting the parameter values differently.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Description<\/th>\n      <th>taskview_doe<\/th>\n      <th>taskview_survey<\/th>\n      <th>func<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="nope" class="section level3">
<h3>Nope</h3>
<div id="htmlwidget-1f261b31fcfeb50f514e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1f261b31fcfeb50f514e">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114"],["<a href='https://cran.r-project.org/web/packages/agridat/index.html'>agridat<\/a>","<a href='https://cran.r-project.org/web/packages/apdesign/index.html'>apdesign<\/a>","<a href='https://cran.r-project.org/web/packages/argonR/index.html'>argonR<\/a>","<a href='https://cran.r-project.org/web/packages/asd/index.html'>asd<\/a>","<a href='https://cran.r-project.org/web/packages/BayesFactor/index.html'>BayesFactor<\/a>","<a href='https://cran.r-project.org/web/packages/binseqtest/index.html'>binseqtest<\/a>","<a href='https://cran.r-project.org/web/packages/BsMD/index.html'>BsMD<\/a>","<a href='https://cran.r-project.org/web/packages/card/index.html'>card<\/a>","<a href='https://cran.r-project.org/web/packages/CCpop/index.html'>CCpop<\/a>","<a href='https://cran.r-project.org/web/packages/choiceDes/index.html'>choiceDes<\/a>","<a href='https://cran.r-project.org/web/packages/crispRdesignR/index.html'>crispRdesignR<\/a>","<a href='https://cran.r-project.org/web/packages/crmPack/index.html'>crmPack<\/a>","<a href='https://cran.r-project.org/web/packages/DeclareDesign/index.html'>DeclareDesign<\/a>","<a href='https://cran.r-project.org/web/packages/deepgp/index.html'>deepgp<\/a>","<a href='https://cran.r-project.org/web/packages/DesignLibrary/index.html'>DesignLibrary<\/a>","<a href='https://cran.r-project.org/web/packages/desirability/index.html'>desirability<\/a>","<a href='https://cran.r-project.org/web/packages/DiceEval/index.html'>DiceEval<\/a>","<a href='https://cran.r-project.org/web/packages/DOS/index.html'>DOS<\/a>","<a href='https://cran.r-project.org/web/packages/DOS2/index.html'>DOS2<\/a>","<a href='https://cran.r-project.org/web/packages/dssd/index.html'>dssd<\/a>","<a href='https://cran.r-project.org/web/packages/DTComPair/index.html'>DTComPair<\/a>","<a href='https://cran.r-project.org/web/packages/dynaTree/index.html'>dynaTree<\/a>","<a href='https://cran.r-project.org/web/packages/edcc/index.html'>edcc<\/a>","<a href='https://cran.r-project.org/web/packages/emon/index.html'>emon<\/a>","<a href='https://cran.r-project.org/web/packages/esDesign/index.html'>esDesign<\/a>","<a href='https://cran.r-project.org/web/packages/factorial2x2/index.html'>factorial2x2<\/a>","<a href='https://cran.r-project.org/web/packages/FisPro/index.html'>FisPro<\/a>","<a href='https://cran.r-project.org/web/packages/flowr/index.html'>flowr<\/a>","<a href='https://cran.r-project.org/web/packages/forestinventory/index.html'>forestinventory<\/a>","<a href='https://cran.r-project.org/web/packages/FSSF/index.html'>FSSF<\/a>","<a href='https://cran.r-project.org/web/packages/GAD/index.html'>GAD<\/a>","<a href='https://cran.r-project.org/web/packages/GeDS/index.html'>GeDS<\/a>","<a href='https://cran.r-project.org/web/packages/genepi/index.html'>genepi<\/a>","<a href='https://cran.r-project.org/web/packages/genotypeR/index.html'>genotypeR<\/a>","<a href='https://cran.r-project.org/web/packages/geospt/index.html'>geospt<\/a>","<a href='https://cran.r-project.org/web/packages/granova/index.html'>granova<\/a>","<a href='https://cran.r-project.org/web/packages/GroupSeq/index.html'>GroupSeq<\/a>","<a href='https://cran.r-project.org/web/packages/gsbDesign/index.html'>gsbDesign<\/a>","<a href='https://cran.r-project.org/web/packages/gscounts/index.html'>gscounts<\/a>","<a href='https://cran.r-project.org/web/packages/gsDesign/index.html'>gsDesign<\/a>","<a href='https://cran.r-project.org/web/packages/GSED/index.html'>GSED<\/a>","<a href='https://cran.r-project.org/web/packages/gset/index.html'>gset<\/a>","<a href='https://cran.r-project.org/web/packages/hetGP/index.html'>hetGP<\/a>","<a href='https://cran.r-project.org/web/packages/ICBioMark/index.html'>ICBioMark<\/a>","<a href='https://cran.r-project.org/web/packages/icensmis/index.html'>icensmis<\/a>","<a href='https://cran.r-project.org/web/packages/ICODS/index.html'>ICODS<\/a>","<a href='https://cran.r-project.org/web/packages/IDPmisc/index.html'>IDPmisc<\/a>","<a href='https://cran.r-project.org/web/packages/JMdesign/index.html'>JMdesign<\/a>","<a href='https://cran.r-project.org/web/packages/LatticeDesign/index.html'>LatticeDesign<\/a>","<a href='https://cran.r-project.org/web/packages/lhs/index.html'>lhs<\/a>","<a href='https://cran.r-project.org/web/packages/LibOPF/index.html'>LibOPF<\/a>","<a href='https://cran.r-project.org/web/packages/MAMS/index.html'>MAMS<\/a>","<a href='https://cran.r-project.org/web/packages/maximin/index.html'>maximin<\/a>","<a href='https://cran.r-project.org/web/packages/MBHdesign/index.html'>MBHdesign<\/a>","<a href='https://cran.r-project.org/web/packages/MCPMod/index.html'>MCPMod<\/a>","<a href='https://cran.r-project.org/web/packages/MCPModPack/index.html'>MCPModPack<\/a>","<a href='https://cran.r-project.org/web/packages/metaplot/index.html'>metaplot<\/a>","<a href='https://cran.r-project.org/web/packages/mined/index.html'>mined<\/a>","<a href='https://cran.r-project.org/web/packages/minimalRSD/index.html'>minimalRSD<\/a>","<a href='https://cran.r-project.org/web/packages/obfuscatoR/index.html'>obfuscatoR<\/a>","<a href='https://cran.r-project.org/web/packages/ODS/index.html'>ODS<\/a>","<a href='https://cran.r-project.org/web/packages/onewaytests/index.html'>onewaytests<\/a>","<a href='https://cran.r-project.org/web/packages/OptimaRegion/index.html'>OptimaRegion<\/a>","<a href='https://cran.r-project.org/web/packages/OptInterim/index.html'>OptInterim<\/a>","<a href='https://cran.r-project.org/web/packages/ordcrm/index.html'>ordcrm<\/a>","<a href='https://cran.r-project.org/web/packages/osDesign/index.html'>osDesign<\/a>","<a href='https://cran.r-project.org/web/packages/PharmPow/index.html'>PharmPow<\/a>","<a href='https://cran.r-project.org/web/packages/plfMA/index.html'>plfMA<\/a>","<a href='https://cran.r-project.org/web/packages/plgp/index.html'>plgp<\/a>","<a href='https://cran.r-project.org/web/packages/populationPDXdesign/index.html'>populationPDXdesign<\/a>","<a href='https://cran.r-project.org/web/packages/powerbydesign/index.html'>powerbydesign<\/a>","<a href='https://cran.r-project.org/web/packages/powerGWASinteraction/index.html'>powerGWASinteraction<\/a>","<a href='https://cran.r-project.org/web/packages/powerpkg/index.html'>powerpkg<\/a>","<a href='https://cran.r-project.org/web/packages/PracTools/index.html'>PracTools<\/a>","<a href='https://cran.r-project.org/web/packages/PRDA/index.html'>PRDA<\/a>","<a href='https://cran.r-project.org/web/packages/precautionary/index.html'>precautionary<\/a>","<a href='https://cran.r-project.org/web/packages/preference/index.html'>preference<\/a>","<a href='https://cran.r-project.org/web/packages/ProbSamplingI/index.html'>ProbSamplingI<\/a>","<a href='https://cran.r-project.org/web/packages/PWEALL/index.html'>PWEALL<\/a>","<a href='https://cran.r-project.org/web/packages/pwr2ppl/index.html'>pwr2ppl<\/a>","<a href='https://cran.r-project.org/web/packages/PwrGSD/index.html'>PwrGSD<\/a>","<a href='https://cran.r-project.org/web/packages/R6P/index.html'>R6P<\/a>","<a href='https://cran.r-project.org/web/packages/randquotes/index.html'>randquotes<\/a>","<a href='https://cran.r-project.org/web/packages/recipes/index.html'>recipes<\/a>","<a href='https://cran.r-project.org/web/packages/reservoir/index.html'>reservoir<\/a>","<a href='https://cran.r-project.org/web/packages/riskParityPortfolio/index.html'>riskParityPortfolio<\/a>","<a href='https://cran.r-project.org/web/packages/rolocISCCNBS/index.html'>rolocISCCNBS<\/a>","<a href='https://cran.r-project.org/web/packages/RPPairwiseDesign/index.html'>RPPairwiseDesign<\/a>","<a href='https://cran.r-project.org/web/packages/RSurveillance/index.html'>RSurveillance<\/a>","<a href='https://cran.r-project.org/web/packages/sampler/index.html'>sampler<\/a>","<a href='https://cran.r-project.org/web/packages/scan/index.html'>scan<\/a>","<a href='https://cran.r-project.org/web/packages/SDaA/index.html'>SDaA<\/a>","<a href='https://cran.r-project.org/web/packages/SensoMineR/index.html'>SensoMineR<\/a>","<a href='https://cran.r-project.org/web/packages/sFFLHD/index.html'>sFFLHD<\/a>","<a href='https://cran.r-project.org/web/packages/shinymaterial/index.html'>shinymaterial<\/a>","<a href='https://cran.r-project.org/web/packages/sim2Dpredictr/index.html'>sim2Dpredictr<\/a>","<a href='https://cran.r-project.org/web/packages/SimDesign/index.html'>SimDesign<\/a>","<a href='https://cran.r-project.org/web/packages/smartsizer/index.html'>smartsizer<\/a>","<a href='https://cran.r-project.org/web/packages/sparseIndexTracking/index.html'>sparseIndexTracking<\/a>","<a href='https://cran.r-project.org/web/packages/SPmlficmcm/index.html'>SPmlficmcm<\/a>","<a href='https://cran.r-project.org/web/packages/spsurvey/index.html'>spsurvey<\/a>","<a href='https://cran.r-project.org/web/packages/ssrm.logmer/index.html'>ssrm.logmer<\/a>","<a href='https://cran.r-project.org/web/packages/SunterSampling/index.html'>SunterSampling<\/a>","<a href='https://cran.r-project.org/web/packages/superb/index.html'>superb<\/a>","<a href='https://cran.r-project.org/web/packages/surveysd/index.html'>surveysd<\/a>","<a href='https://cran.r-project.org/web/packages/SvyNom/index.html'>SvyNom<\/a>","<a href='https://cran.r-project.org/web/packages/svyVGAM/index.html'>svyVGAM<\/a>","<a href='https://cran.r-project.org/web/packages/TEQR/index.html'>TEQR<\/a>","<a href='https://cran.r-project.org/web/packages/TestDesign/index.html'>TestDesign<\/a>","<a href='https://cran.r-project.org/web/packages/titeIR/index.html'>titeIR<\/a>","<a href='https://cran.r-project.org/web/packages/twowaytests/index.html'>twowaytests<\/a>","<a href='https://cran.r-project.org/web/packages/unikn/index.html'>unikn<\/a>","<a href='https://cran.r-project.org/web/packages/UScensus2000cdp/index.html'>UScensus2000cdp<\/a>","<a href='https://cran.r-project.org/web/packages/VDSPCalibration/index.html'>VDSPCalibration<\/a>"],["Agricultural Datasets","An Implementation of the Additive Polynomial Design Matrix","R Interface to Argon HTML Design","Simulations for Adaptive Seamless Designs","Computation of Bayes Factors for Common Designs","Exact Binary Sequential Designs and Analysis","Bayes Screening and Model Discrimination","Cardiovascular and Autonomic Research Design","One and two locus GWAS of binary phenotype with case-control-population design","Design Functions for Choice Studies","Guide Sequence Design for CRISPR/Cas9","Object-Oriented Implementation of CRM Designs","Declare and Diagnose Research Designs","Sequential Design for Deep Gaussian Processes using MCMC","Library of Research Designs","Function Optimization and Ranking via Desirability Functions","Construction and Evaluation of Metamodels","Design of Observational Studies","Design of Observational Studies, Companion to the Second Edition","Distance Sampling Survey Design","Comparison of Binary Diagnostic Tests in a Paired Study Design","Dynamic Trees for Learning and Design","Economic Design of Control Charts","Tools for Environmental and Ecological Survey Design","Adaptive Enrichment Designs with Sample Size Re-Estimation","Design and Analysis of a 2x2 Factorial Trial","Fuzzy Inference System Design and Optimization","Streamlining Design and Deployment of Complex Workflows","Design-Based Global and Small-Area Estimations for Multiphase Forest Inventories","Generate Fully-Sequential Space-Filling Designs Inside a Unit Hypercube","GAD: Analysis of variance from general principles","Geometrically Designed Spline Regression","Genetic Epidemiology Design and Inference","SNP Genotype Marker Design and Analysis","Geostatistical Analysis and Design of Optimal Spatial Sampling Networks","Graphical Analysis of Variance","A GUI-Based Program to Compute Probabilities Regarding Group Sequential Designs","Group Sequential Bayes Design","Group Sequential Designs with Negative Binomial Outcomes","Group Sequential Design","Group Sequential Enrichment Design","Group Sequential Design in Equivalence Studies","Heteroskedastic Gaussian Process Modeling and Design under Replication","Data-Driven Design of Targeted Gene Panels for Estimating Immunotherapy Biomarkers","Study Design and Data Analysis in the Presence of Error-Prone Diagnostic Tests and Self-Reported Outcomes","Data Analysis for ODS and Case-Cohort Designs with Interval-Censoring","'Utilities of Institute of Data Analyses and Process Design (www.zhaw.ch/idp)'","Joint Modeling of Longitudinal and Survival Data - Power Calculation","Lattice-Based Space-Filling Designs","Latin Hypercube Samples","Design of Optimum-Path Forest Classifiers","Designing Multi-Arm Multi-Stage Studies","Space-Filling Design under Maximin Distance","Spatial Designs for Ecological and Environmental Surveys","Design and Analysis of Dose-Finding Studies","Simulation-Based Design and Analysis of Dose-Finding Trials","Data-Driven Plot Design","Minimum Energy Designs","Minimally Changed CCD and BBD","Obfuscation Game Designs","Statistical Methods for Outcome-Dependent Sampling Designs","One-Way Tests in Independent Groups Designs","Confidence Regions for Optima","Optimal Two and Three Stage Designs for Single-Arm and Two-Arm Randomized Controlled Trials with a Long-Term Binary Endpoint","Likelihood-Based Continual Reassessment Method (CRM) Dose Finding Designs","Design, Planning and Analysis of Observational Studies","Pharmacometric Power calculations for mixed study designs","A GUI to View, Design and Export Various Graphs of Data","Particle Learning of Gaussian Processes","Designing Population PDX Studies","Power Estimates for ANOVA Designs","Power Calculations for GxE and GxG Interactions for GWAS","Power analyses for the affected sib pair and the TDT design","Tools for Designing and Weighting Survey Samples","Conduct a Prospective or Retrospective Design Analysis","Safety Diagnostics for Dose-Escalation Trial Designs","2-Stage Preference Trial Design and Analysis","Probabilistic Sampling Design and Strategies","Design and Monitoring of Survival Trials Accounting for Complex Situations","Power Analyses for Common Designs (Power to the People)","Power in a Group Sequential Design","Design Patterns in R","Get Random Quotes from Quotes on Design API","Preprocessing Tools to Create Design Matrices","Tools for Analysis, Design, and Operation of Water Supply Storages","Design of Risk Parity Portfolios","A Colour List and Colour Metric Based on the ISCC-NBS System of Color Designation","Resolvable partially pairwise balanced design and Space-filling design via association scheme","Design and Analysis of Disease Surveillance Activities","Sample Design, Drawing & Data Analysis Using Data Frames","Single-Case Data Analyses for Single and Multiple Baseline Designs","Sampling: Design and Analysis","Sensory Data Analysis","Sequential Full Factorial-Based Latin Hypercube Design","Implement Material Design in Shiny Applications","Simulate Outcomes Using Spatially Dependent Design Matrices","Structure for Organizing Monte Carlo Simulation Designs","Power Analysis for a SMART Design","Design of Portfolio of Stocks to Track an Index","Semiparametric Maximum Likelihood Method for Interactions Gene-Environment in Case-Mother Control-Mother Designs","Spatial Survey Design and Analysis","Sample Size Determination for Longitudinal Designs with Binary Outcome","Sunter's sampling design","Get Precision of Means Under Various Designs and Sampling Schemes","Survey Standard Error Estimation for Cumulated Estimates and their Differences in Complex Panel Designs","Nomograms for Right-Censored Outcomes from Survey Designs","Design-Based Inference in Vector Generalised Linear Models","Target Equivalence Range Design","Optimal Test Design Approach to Fixed and Adaptive Test Construction","Isotonic Designs for Phase 1 Trials with Late-Onset Toxicities","Two-Way Tests in Independent Groups Designs","Graphical Elements of the University of Konstanz's Corporate Design","US Census 2000 Designated Places Shapefiles and Additional Demographic Data","Statistical Methods for Designing and Analyzing a Calibration Study"],["Datasets from books, papers, and websites related to agriculture. Example graphics and analyses are included. Data come from small-plot trials, multi-environment trials, uniformity trials, yield monitors, and more.","An implementation of the additive polynomial (AP) design matrix. It constructs and appends an AP design matrix to a data frame for use with longitudinal data subject to seasonality.","R wrapper around the argon HTML library. More at <https://demos.creative-tim.com/argon-design-system/>.","Package runs simulations for adaptive seamless designs with and without early outcomes for treatment selection and subpopulation type designs.","A suite of functions for computing various Bayes factors for simple designs, including contingency tables, one- and two-sample designs, one-way designs, general ANOVA designs, and linear regression.","For a series of binary responses, create stopping boundary with exact results after stopping, allowing updating for missing assessments.","Bayes screening and model discrimination follow-up designs.","Tools that can aid in the assessment of the autonomic regulation of cardiovascular physiology. The aims of this package are to: 1) study electrocardiography (both intervals and morphology) as extensions of signal processing, 2) study circadian rhythms and how it effects autonomic physiology, 3) assess clinical risk of autonomic dysfunction on cardiovascular health through the perspective of epidemiology and causality. The analysis of circadian rhythms through cosinor analysis are built upon the methods by Cornelissen (2014) <doi:10.1186/1742-4682-11-16> and Refinetti, Cornelissen, Halberg (2014) <doi:10.1080/09291010600903692>.","Tests of association between SNPs or pairs of SNPs and binary phenotypes, in case-control / case-population / case-control-population studies.","Design functions for DCMs and other types of choice studies (including MaxDiff and other tradeoffs).","Designs guide sequences for CRISPR/Cas9 genome editing and provides information on sequence features pertinent to guide efficiency. Sequence features include annotated off-target predictions in a user-selected genome and a predicted efficiency score based on the model described in Doench et al. (2016) <doi:10.1038/nbt.3437>. Users are able to import additional genomes and genome annotation files to use when searching and annotating off-target hits. All guide sequences and off-target data can be generated through the 'R' console with sgRNA_Design() or through 'crispRdesignR's' user interface with crispRdesignRUI(). CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) and the associated protein Cas9 refer to a technique used in genome editing.","Implements a wide range of model-based dose escalation designs, ranging from classical and modern continual reassessment methods (CRMs) based on dose-limiting toxicity endpoints to dual-endpoint designs taking into account a biomarker/efficacy outcome. The focus is on Bayesian inference, making it very easy to setup a new design with its own JAGS code. However, it is also possible to implement 3+3 designs for comparison or models with non-Bayesian estimation. The whole package is written in a modular form in the S4 class system, making it very flexible for adaptation to new models, escalation or stopping rules.","Researchers can characterize and learn about the properties of research designs before implementation using `DeclareDesign`. Ex ante declaration and diagnosis of designs can help researchers clarify the strengths and limitations of their designs and to improve their properties, and can help readers evaluate a research strategy prior to implementation and without access to results. It can also make it easier for designs to be shared, replicated, and critiqued.","Performs model fitting and sequential design for deep Gaussian processes following Sauer, Gramacy, and Higdon (2020) <arXiv:2012.08015>. Models extend up to three layers deep; a one layer model is equivalent to typical Gaussian process regression. Sequential design criteria include integrated mean-squared error (IMSE), active learning Cohn (ALC), and expected improvement (EI). Covariance structure is based on inverse exponentiated squared euclidean distance. Applicable to noisy and deterministic functions. Incorporates SNOW parallelization and utilizes C under the hood.","A simple interface to build designs using the package 'DeclareDesign'. In one line of code, users can specify the parameters of individual designs and diagnose their properties. The designers can also be used to compare performance of a given design across a range of combinations of parameters, such as effect size, sample size, and assignment probabilities.","S3 classes for multivariate optimization using the desirability function by Derringer and Suich (1980).","Estimation, validation and prediction of models of different types : linear models, additive models, MARS,PolyMARS and Kriging.","Contains data sets, examples and software from the book Design of Observational Studies by Paul R. Rosenbaum, New York: Springer, <doi:10.1007/978-1-4419-1213-8>, ISBN 978-1-4419-1212-1.","Contains data sets, examples and software from the Second Edition of \"Design of Observational Studies\"; see Rosenbaum, P.R. (2010) <doi:10.1007/978-1-4419-1213-8>.","Creates survey designs for distance sampling surveys. These designs can be assessed for various effort and coverage statistics. Once the user is satisfied with the design characteristics they can generate a set of transects to use in their distance sampling survey. Many of the designs implemented in this R package were first made available in our 'Distance' for Windows software and are detailed in Chapter 7 of Advanced Distance Sampling, Buckland et. al. (2008, ISBN-13: 978-0199225873). Find out more about estimating animal/plant abundance with distance sampling at <http://distancesampling.org/>.","This package contains functions to compare the accuracy of two binary diagnostic tests in a \"paired\" study design, i.e. when each test is applied to each subject in the study.","Inference by sequential Monte Carlo for dynamic tree regression and classification models with hooks provided for sequential design and optimization, fully online learning with drift, variable selection, and sensitivity analysis of inputs. Illustrative examples from the original dynamic trees paper are facilitated by demos in the package; see demo(package=\"dynaTree\").","This package provides a unified approach for Economic Design of Control Charts. The main purpose of this package is to find out the optimal parameters to minimize the ECH (Expected Cost per Hour) of the process.","Statistical tools for environmental and ecological surveys. Simulation-based power and precision analysis; detection probabilities from different survey designs; visual fast count estimation.","Software of 'esDesign' is developed to implement the adaptive enrichment designs with sample size re-estimation. In details, three-proposed trial designs are provided, including the AED1-SSR (or ES1-SSR), AED2-SSR (or ES2-SSR) and AED3-SSR (or ES3-SSR). In addition, this package also contains several widely used adaptive designs, such as the Marker Sequential Test (MaST) design proposed Freidlin et al. (2014) <doi:10.1177/1740774513503739>, the adaptive enrichment designs without early stopping (AED or ES), the sample size re-estimation procedure (SSR) based on the conditional power proposed by Proschan and Hunsberger (1995), and some useful functions. In details, we can calculate the futility and/or efficacy stopping boundaries, the sample size required, calibrate the value of the threshold of the difference between subgroup-specific test statistics, conduct the simulation studies in AED, SSR, AED1-SSR, AED2-SSR and AED3-SSR.","Used for the design and analysis of a 2x2 factorial trial for a time-to-event endpoint. It performs power calculations and significance testing as well as providing estimates of the relevant hazard ratios and the corresponding 95% confidence intervals. Important reference papers include Slud EV. (1994) <https://www.ncbi.nlm.nih.gov/pubmed/8086609> Lin DY, Gong J, Gallo P, Bunn PH, Couper D. (2016) <DOI:10.1111/biom.12507> Leifer ES, Troendle JF, Kolecki A, Follmann DA. (2020) <https://github.com/EricSLeifer/factorial2x2/blob/master/Leifer%20et%20al.%20paper.pdf>.","Fuzzy inference systems are based on fuzzy rules, which have a good capability for managing progressive phenomenons. This package is a basic implementation of the main functions to use a Fuzzy Inference System (FIS) provided by the open source software 'FisPro' <https://www.fispro.org>. 'FisPro' allows to create fuzzy inference systems and to use them for reasoning purposes, especially for simulating a physical or biological system.","This framework allows you to design and implement complex pipelines, and deploy them on your institution's computing cluster. This has been built keeping in mind the needs of bioinformatics workflows. However, it is easily extendable to any field where a series of steps (shell commands) are to be executed in a (work)flow.","Extensive global and small-area estimation procedures for multiphase forest inventories under the design-based Monte-Carlo approach are provided. The implementation has been published in the Journal of Statistical Software (<doi:10.18637/jss.v097.i04>) and includes estimators for simple and cluster sampling published by Daniel Mandallaz in 2007 (<doi:10.1201/9781584889779>), 2013 (<doi:10.1139/cjfr-2012-0381>, <doi:10.1139/cjfr-2013-0181>, <doi:10.1139/cjfr-2013-0449>, <doi:10.3929/ethz-a-009990020>) and 2016 (<doi:10.3929/ethz-a-010579388>). It provides point estimates, their external- and design-based variances and confidence intervals, as well as a set of functions to analyze and visualize the produced estimates. The procedures have also been optimized for the use of remote sensing data as auxiliary information, as demonstrated in 2018 by Hill et al. (<doi:10.3390/rs10071052>).","Provides three methods proposed by Shang and Apley (2019) <doi:10.1080/00224065.2019.1705207> to generate fully-sequential space-filling designs inside a unit hypercube. A 'fully-sequential space-filling design' means a sequence of nested designs (as the design size varies from one point up to some maximum number of points) with the design points added one at a time and such that the design at each size has good space-filling properties. Two methods target the minimum pairwise distance criterion and generate maximin designs, among which one method is more efficient when design size is large. One method targets the maximum hole size criterion and uses a heuristic to generate what is closer to a minimax design.","This package analyses complex ANOVA models with any combination of orthogonal/nested and fixed/random factors, as described by Underwood (1997). There are two restrictions: (i) data must be balanced; (ii) fixed nested factors are not allowed. Homogeneity of variances is checked using Cochran's C test and 'a posteriori' comparisons of means are done using Student-Newman-Keuls (SNK) procedure.","Geometrically Designed Spline ('GeDS') Regression is a non-parametric geometrically motivated method for fitting variable knots spline predictor models in one or two independent variables, in the context of generalized (non-)linear models. 'GeDS' estimates the number and position of the knots and the order of the spline, assuming the response variable has a distribution from the exponential family. A description of the method can be found in Kaishev et al. (2016) <doi:10.1007/s00180-015-0621-7> and Dimitrova et al. (2017) <https://openaccess.city.ac.uk/18460>.","Functions for Genetic Epi Methods Developed at MSKCC","We implement a common genotyping workflow with a standardized software interface. 'genotypeR' designs genotyping markers from vcf files, outputs markers for multiplexing suitability on various platforms (Sequenom and Illumina GoldenGate), and provides various QA/QC and analysis functions. We developed this package to analyze data in Stevison LS, SA Sefick, CA Rushton, and RM Graze. 2017. Invited Review: Recombination rate plasticity: revealing mechanisms by design. Philosophical Transactions Royal Society London B Biol Sci 372:1-14. <DOI:10.1098/rstb.2016.0459>, and have published it here Sefick, S.A., M.A. Castronova, and L.S. Stevison. 2017. GENOTYPER: An integrated R packages for single nuleotide polymorphism genotype marker design and data analysis. Methods in Ecology and Evolution 9: 1318-1323. <DOI: 10.1111/2041-210X.12965>.","Estimation of the variogram through trimmed mean, radial basis functions (optimization, prediction and cross-validation), summary statistics from cross-validation, pocket plot, and design of optimal sampling networks through sequential and simultaneous points methods.","This small collection of functions provides what we call elemental graphics for display of anova results. The term elemental derives from the fact that each function is aimed at construction of graphical displays that afford direct visualizations of data with respect to the fundamental questions that drive the particular anova methods. The two main functions are granova.1w (a graphic for one way anova) and granova.2w (a corresponding graphic for two way anova). These functions were written to display data for any number of groups, regardless of their sizes (however, very large data sets or numbers of groups can be problematic). For these two functions a specialized approach is used to construct data-based contrast vectors for which anova data are displayed. The result is that the graphics use straight lines, and when appropriate flat surfaces, to facilitate clear interpretations while being faithful to the standard effect tests in anova. The graphic results are complementary to standard summary tables for these two basic kinds of analysis of variance; numerical summary results of analyses are also provided as side effects. Two additional functions are granova.ds (for comparing two dependent samples), and granova.contr (which provides graphic displays for a priori contrasts). All functions provide relevant numerical results to supplement the graphic displays of anova data. The graphics based on these functions should be especially helpful for learning how the methods have been applied to answer the question(s) posed. This means they can be particularly helpful for students and non-statistician analysts. But these methods should be quite generally helpful for work-a-day applications of all kinds, as they can help to identify outliers, clusters or patterns, as well as highlight the role of non-linear transformations of data. In the case of granova.1w and granova.ds especially, several arguments are provided to facilitate flexibility in the construction of graphics that accommodate diverse features of data, according to their corresponding display requirements. See the help files for individual functions.","A graphical user interface to compute group sequential designs based on normally distributed test statistics, particularly critical boundaries, power, drift, and confidence intervals of such designs. All computations are based on the alpha spending approach by Lan-DeMets with various alpha spending functions being available to choose among.","Group Sequential Operating Characteristics for Clinical, Bayesian two-arm Trials with known Sigma and Normal Endpoints.","Design and analysis of group sequential designs for negative binomial outcomes, as described by T Mütze, E Glimm, H Schmidli, T Friede (2018) <doi:10.1177/0962280218773115>.","Derives group sequential designs and describes their properties.","Provides function to apply \"Group sequential enrichment design incorporating subgroup selection\" (GSED) method proposed by Magnusson and Turnbull (2013) <doi:10.1002/sim.5738>.","calculate equivalence and futility boundaries based on the exact bivariate $t$ test statistics for group sequential designs in studies with equivalence hypotheses.","Performs Gaussian process regression with heteroskedastic noise following Binois, M., Gramacy, R., Ludkovski, M. (2016) <arXiv:1611.05902>. The input dependent noise is modeled as another Gaussian process. Replicated observations are encouraged as they yield computational savings. Sequential design procedures based on the integrated mean square prediction error and lookahead heuristics are provided, and notably fast update functions when adding new observations.","Implementation of the methodology proposed in 'Data-driven design of targeted gene panels for estimating immunotherapy biomarkers', Bradley and Cannings (2021) <arXiv:2102.04296>. This package allows the user to fit generative models of mutation from an annotated mutation dataset, and then further to produce tunable linear estimators of exome-wide biomarkers. It also contains functions to simulate mutation annotated format (MAF) data, as well as to analyse the output and performance of models.","We consider studies in which information from error-prone diagnostic tests or self-reports are gathered sequentially to determine the occurrence of a silent event. Using a likelihood-based approach incorporating the proportional hazards assumption, we provide functions to estimate the survival distribution and covariate effects. We also provide functions for power and sample size calculations for this setting. Please refer to Xiangdong Gu, Yunsheng Ma, and Raji Balasubramanian (2015) <doi: 10.1214/15-AOAS810>, Xiangdong Gu and Raji Balasubramanian (2016) <doi: 10.1002/sim.6962>, Xiangdong Gu, Mahlet G Tadesse, Andrea S Foulkes, Yunsheng Ma, and Raji Balasubramanian (2020) <doi: 10.1186/s12911-020-01223-w>.","Sieve semiparametric likelihood methods for analyzing interval-censored failure time data from an outcome-dependent sampling (ODS) design and from a case-cohort design. Zhou, Q., Cai, J., and Zhou, H. (2018) <doi:10.1111/biom.12744>; Zhou, Q., Zhou, H., and Cai, J. (2017) <doi:10.1093/biomet/asw067>.","Different high-level graphics functions for displaying large datasets, displaying circular data in a very flexible way, finding local maxima, brewing color ramps, drawing nice arrows, zooming 2D-plots, creating figures with differently colored margin and plot region. In addition, the package contains auxiliary functions for data manipulation like omitting observations with irregular values or selecting data by logical vectors, which include NAs. Other functions are especially useful in spectroscopy and analyses of environmental data: robust baseline fitting, finding peaks in spectra, converting humidity measures.","Performs power calculations for joint modeling of longitudinal and survival data with k-th order trajectories when the variance-covariance matrix, Sigma_theta, is unknown.","Lattice-based space-filling designs with fill or separation distance properties including interleaved lattice-based minimax distance designs proposed in Xu He (2017) <doi:10.1093/biomet/asx036>, interleaved lattice-based maximin distance designs proposed in Xu He (2018) <doi:10.1093/biomet/asy069>, (sliced) rotated sphere packing designs proposed in Xu He (2017) <doi:10.1080/01621459.2016.1222289> and Xu He (2019) <doi:10.1080/00401706.2018.1458655>, and densest packing-based maximum projections designs proposed in Xu He (2020) <doi:10.1093/biomet/asaa057> and Xu He (2018) <arXiv:1709.02062v2>.","Provides a number of methods for creating and augmenting Latin Hypercube Samples and Orthogonal Array Latin Hypercube Samples.","The 'LibOPF' is a framework to develop pattern recognition techniques based on optimum-path forests (OPF), João P. Papa and Alexandre X. Falcão (2008) <doi:10.1007/978-3-540-89639-5_89>, with methods for supervised learning and data clustering.","Designing multi-arm multi-stage studies with (asymptotically) normal endpoints and known variance.","Constructs a space-filling design under the criterion of maximum-minimum distance. Both discrete and continuous searches are provided.","Provides spatially balanced designs from a set of (contiguous) potential sampling locations in a study region for point-based and for transect-based surveys. Accommodates, without detrimental effects on spatial balance, sites that the researcher wishes to include in the survey for reasons other than the current randomisation (legacy sites).","Implements a methodology for the design and analysis of dose-response studies that combines aspects of multiple comparison procedures and modeling approaches (Bretz, Pinheiro and Branson, 2005, Biometrics 61, 738-748, <doi: 10.1111/j.1541-0420.2005.00344.x>). The package provides tools for the analysis of dose finding trials as well as a variety of tools necessary to plan a trial to be conducted with the MCP-Mod methodology. Please note: The 'MCPMod' package will not be further developed, all future development of the MCP-Mod methodology will be done in the 'DoseFinding' R-package.","An efficient implementation of the MCPMod (Multiple Comparisons and Modeling) method to support a simulation-based design and analysis of dose-finding trials with normally distributed, binary and count endpoints (Bretz et al. (2005) <doi:10.1111/j.1541-0420.2005.00344.x>).","Designs plots in terms of core structure. See 'example(metaplot)'. Primary arguments are (unquoted) column names; order and type (numeric or not) dictate the resulting plot. Specify any y variables, x variable, any groups variable, and any conditioning variables to metaplot() to generate density plots, boxplots, mosaic plots, scatterplots, scatterplot matrices, or conditioned plots. Use multiplot() to arrange plots in grids. Wherever present, scalar column attributes 'label' and 'guide' are honored, producing fully annotated plots with minimal effort. Attribute 'guide' is typically units, but may be encoded() to provide interpretations of categorical values (see '?encode'). Utility unpack() transforms scalar column attributes to row values and pack() does the reverse, supporting tool-neutral storage of metadata along with primary data. The package supports customizable aesthetics such as such as reference lines, unity lines, smooths, log transformation, and linear fits. The user may choose between trellis and ggplot output. Compact syntax and integrated metadata promote workflow scalability.","This is a method (MinED) for mining probability distributions using deterministic sampling which is proposed by Joseph, Wang, Gu, Lv, and Tuo (2018). The MinED samples can be used for approximating the target distribution. They can be generated from a density function that is known only up to a proportionality constant and thus, it might find applications in Bayesian computation. Moreover, the MinED samples are generated with much fewer evaluations of the density function compared to random sampling-based methods such as MCMC and therefore, this method will be especially useful when the unnormalized posterior is expensive or time consuming to evaluate.","Generate central composite designs (CCD)with full as well as fractional factorial points (half replicate) and Box Behnken designs (BBD) with minimally changed run sequence.","When people make decisions, they may do so using a wide variety of decision rules. The package allows users to easily create obfuscation games to test the obfuscation hypothesis. It provides an easy to use interface and multiple options designed to vary the difficulty of the game and tailor it to the user's needs. For more detail: Chorus et al., 2021, Obfuscation maximization-based decision-making: Theory, methodology and first empirical evidence, Mathematical Social Sciences, 109, 28-44.","Outcome-dependent sampling (ODS) schemes are cost-effective ways to enhance study efficiency. In ODS designs, one observes the exposure/covariates with a probability that depends on the outcome variable. Popular ODS designs include case-control for binary outcome, case-cohort for time-to-event outcome, and continuous outcome ODS design (Zhou et al. 2002) <doi: 10.1111/j.0006-341X.2002.00413.x>. Because ODS data has biased sampling nature, standard statistical analysis such as linear regression will lead to biases estimates of the population parameters. This package implements four statistical methods related to ODS designs: (1) An empirical likelihood method analyzing the primary continuous outcome with respect to exposure variables in continuous ODS design (Zhou et al., 2002). (2) A partial linear model analyzing the primary outcome in continuous ODS design (Zhou, Qin and Longnecker, 2011) <doi: 10.1111/j.1541-0420.2010.01500.x>. (3) Analyze a secondary outcome in continuous ODS design (Pan et al. 2018) <doi: 10.1002/sim.7672>. (4) An estimated likelihood method analyzing a secondary outcome in case-cohort data (Pan et al. 2017) <doi: 10.1111/biom.12838>.","Performs one-way tests in independent groups designs; one-way analysis of variance (ANOVA), Welch's heteroscedastic F test, Welch's heteroscedastic F test with trimmed means and Winsorized variances, Brown-Forsythe test, Alexander-Govern test, James second order test, Kruskal-Wallis test, Scott-Smith test, Box F test and Johansen F test, Generalized tests equivalent to Parametric Bootstrap and Fiducial tests. The package performs pairwise comparisons and graphical approaches. Also, the package includes Student's t test, Welch's t test and Mann-Whitney U test for two samples. Moreover, it assesses variance homogeneity and normality of data in each group via tests and plots (Dag et al., 2018, <https://journal.r-project.org/archive/2018/RJ-2018-022/RJ-2018-022.pdf>).","Computes confidence regions on the location of response surface optima.","Optimal two and three stage designs monitoring time-to-event endpoints at a specified timepoint","Provides the setup and calculations needed to run a likelihood-based continual reassessment method (CRM) dose finding trial and performs simulations to assess design performance under various scenarios. 3 dose finding designs are included in this package: ordinal proportional odds model (POM) CRM, ordinal continuation ratio (CR) model CRM, and the binary 2-parameter logistic model CRM. These functions allow customization of design characteristics to vary sample size, cohort sizes, target dose-limiting toxicity (DLT) rates, discrete or continuous dose levels, combining ordinal grades 0 and 1 into one category, and incorporate safety and/or stopping rules. For POM and CR model designs, ordinal toxicity grades are specified by common terminology criteria for adverse events (CTCAE) version 4.0. Function 'pseudodata' creates the necessary starting models for these 3 designs, and function 'nextdose' estimates the next dose to test in a cohort of patients for a target DLT rate. We also provide the function 'crmsimulations' to assess the performance of these 3 dose finding designs under various scenarios.","A suite of functions for the design of case-control and two-phase studies, and the analysis of data that arise from them. Functions in this packages provides Monte Carlo based evaluation of operating characteristics such as powers for estimators of the components of a logistic regression model. For additional detail see: Haneuse S, Saegusa T and Lumley T (2011)<doi:10.18637/jss.v043.i11>.","This package contains functions performing power calculations for mixed (sparse/dense sampled) pharmacokinetic study designs. The input data for these functions is tailored for NONMEM .phi files.","Provides a graphical user interface for viewing and designing various types of graphs of the data. The graphs can be saved in different formats of an image.","Sequential Monte Carlo inference for fully Bayesian Gaussian process (GP) regression and classification models by particle learning (PL). The sequential nature of inference and the active learning (AL) hooks provided facilitate thrifty sequential design (by entropy) and optimization (by improvement) for classification and regression models, respectively. This package essentially provides a generic PL interface, and functions (arguments to the interface) which implement the GP models and AL heuristics. Functions for a special, linked, regression/classification GP model and an integrated expected conditional improvement (IECI) statistic is provides for optimization in the presence of unknown constraints. Separable and isotropic Gaussian, and single-index correlation functions are supported. See the examples section of ?plgp and demo(package=\"plgp\") for an index of demos","Run simulations to assess the impact of various designs features and the underlying biological behaviour on the outcome of a Patient Derived Xenograft (PDX) population study. This project can either be deployed to a server as a 'shiny' app or installed locally as a package and run the app using the command 'populationPDXdesignApp()'.","Functions for bootstrapping the power of ANOVA designs based on estimated means and standard deviations of the conditions. Please refer to the documentation of the boot.power.anova() function for further details.","Analytical power calculations for GxE and GxG interactions for case-control studies of candidate genes and genome-wide association studies (GWAS). This includes power calculation for four two-step screening and testing procedures. It can also calculate power for GxE and GxG without any screening.","(1) To estimate the power of testing for linkage using an affected sib pair design, as a function of the recurrence risk ratios. We will use analytical power formulae as implemented in R. These are based on a Mathematica notebook created by Martin Farrall. (2) To examine how the power of the transmission disequilibrium test (TDT) depends on the disease allele frequency, the marker allele frequency, the strength of the linkage disequilibrium, and the magnitude of the genetic effect. We will use an R program that implements the power formulae of Abel and Muller-Myhsok (1998). These formulae allow one to quickly compute power of the TDT approach under a variety of different conditions. This R program was modeled on Martin Farrall's Mathematica notebook.","Functions and datasets to support Valliant, Dever, and Kreuter, ``Practical Tools for Designing and Weighting Survey Samples'' (2nd edition, 2018). Contains functions for sample size calculation for survey samples using stratified or clustered one-, two-, and three-stage sample designs. Other functions compute variance components for multistage designs and sample sizes in two-phase designs. A number of example data sets are included.","An implementation of the \"Design Analysis\" proposed by Gelman and Carlin (2014) <doi:10.1177/1745691614551642>. It combines the evaluation of Power-Analysis with other inferential-risks as Type-M error (i.e. Magnitude) and Type-S error (i.e. Sign). See also Altoè et al. (2020) <doi:10.3389/fpsyg.2019.02893> and Bertoldo et al. (2020) <doi:10.31234/osf.io/q9f86>.","Enhances various R packages that support the design and simulation of phase 1 dose-escalation trials, adding diagnostics to examine the safety characteristics of these designs in light of expected inter-individual variation in pharmacokinetics and pharmacodynamics. See Norris (2020b), \"Retrospective analysis of a fatal dose-finding trial\" <arXiv:2004.12755> and (2020c) \"What Were They Thinking? Pharmacologic priors implicit in a choice of 3+3 dose-escalation design\" <arXiv:2012.05301>.","Design and analyze two-stage randomized trials with a continuous outcome measure. The package contains functions to compute the required sample size needed to detect a given preference, treatment, and selection effect; alternatively, the package contains functions that can report the study power given a fixed sample size. Finally, analysis functions are provided to test each effect using either summary data (i.e. means, variances) or raw study data <doi:10.18637/jss.v094.c02>.","It allows the user to determine sample sizes, select probabilistic samples, make estimates of different parameters for the total finite population and in studio domains, using the main design drawings.","Calculates various functions needed for design and monitoring survival trials accounting for complex situations such as delayed treatment effect, treatment crossover, non-uniform accrual, and different censoring distributions between groups. The event time distribution is assumed to be piecewise exponential (PWE) distribution and the entry time is assumed to be piecewise uniform distribution. As compared with Version 1.2.1, two more types of hybrid crossover are added. A bug is corrected in the function \"pwecx\" that calculates the crossover-adjusted survival, distribution, density, hazard and cumulative hazard functions. Also, to generate the crossover-adjusted event time random variable, a more efficient algorithm is used and the output includes crossover indicators.","Statistical power analysis for designs including t-tests, correlations, multiple regression, ANOVA, mediation, and logistic regression. Functions accompany Aberson (2019) <doi:10.4324/9781315171500>.","Tools for the evaluation of interim analysis plans for sequentially monitored trials on a survival endpoint; tools to construct efficacy and futility boundaries, for deriving power of a sequential design at a specified alternative, template for evaluating the performance of candidate plans at a set of time varying alternatives. See Izmirlian, G. (2014) <doi:10.4310/SII.2014.v7.n1.a4>.","Build robust and maintainable software with object-oriented design patterns in R. Design patterns abstract and present in neat, well-defined components and interfaces the experience of many software designers and architects over many years of solving similar problems. These are solutions that have withstood the test of time with respect to re-usability, flexibility, and maintainability. 'R6P' provides abstract base classes with examples for a few known design patterns. The patterns were selected by their applicability to analytic projects in R. Using these patterns in R projects have proven effective in dealing with the complexity that data-driven applications possess.","Connects to the site <http://quotesondesign.com/> that uses the 'WordPress' built-in REST API to provide a way for you to grab quotes.","An extensible framework to create and preprocess design matrices. Recipes consist of one or more data manipulation and analysis \"steps\". Statistical parameters for the steps can be estimated from an initial data set and then applied to other data sets. The resulting design matrices can then be used as inputs into statistical or machine learning models.","Measure single-storage water supply system performance using resilience, reliability, and vulnerability metrics; assess storage-yield-reliability relationships; determine no-fail storage with sequent peak analysis; optimize release decisions for water supply, hydropower, and multi-objective reservoirs using deterministic and stochastic dynamic programming; generate inflow replicates using parametric and non-parametric models; evaluate inflow persistence using the Hurst coefficient.","Fast design of risk parity portfolios for financial investment. The goal of the risk parity portfolio formulation is to equalize or distribute the risk contributions of the different assets, which is missing if we simply consider the overall volatility of the portfolio as in the mean-variance Markowitz portfolio. In addition to the vanilla formulation, where the risk contributions are perfectly equalized subject to no shortselling and budget constraints, many other formulations are considered that allow for box constraints and shortselling, as well as the inclusion of additional objectives like the expected return and overall variance. See vignette for a detailed documentation and comparison, with several illustrative examples. The package is based on the papers: Y. Feng, and D. P. Palomar (2015). SCRIP: Successive Convex Optimization Methods for Risk Parity Portfolio Design. IEEE Trans. on Signal Processing, vol. 63, no. 19, pp. 5285-5300. <doi:10.1109/TSP.2015.2452219>. F. Spinu (2013), An Algorithm for Computing Risk Parity Weights. <doi:10.2139/ssrn.2297383>. T. Griveau-Billion, J. Richard, and T. Roncalli (2013). A fast algorithm for computing High-dimensional risk parity portfolios. <arXiv:1311.4057>.","A colour list and colour metric based on the ISCC-NBS System of Color Designation for use with the 'roloc' package for converting colour specifications to colour names.","Using some association schemes to obtain a new series of resolvable partially pairwise balanced designs (RPPBD) and space-filling designs.","A range of functions for the design and analysis of disease surveillance activities. These functions were originally developed for animal health surveillance activities but can be equally applied to aquatic animal, wildlife, plant and human health surveillance activities. Utilities are included for sample size calculation and analysis of representative surveys for disease freedom, risk-based studies for disease freedom and for prevalence estimation. This package is based on Cameron A., Conraths F., Frohlich A., Schauer B., Schulz K., Sergeant E., Sonnenburg J., Staubach C. (2015). R package of functions for risk-based surveillance. Deliverable 6.24, WP 6 - Decision making tools for implementing risk-based surveillance, Grant Number no. 310806, RISKSUR (<https://www.fp7-risksur.eu/sites/default/files/documents/Deliverables/RISKSUR_%28310806%29_D6.24.pdf>). Many of the 'RSurveillance' functions are incorporated into the 'epitools' website: Sergeant, ESG, 2019. Epitools epidemiological calculators. Ausvet Pty Ltd. Available at: <http://epitools.ausvet.com.au>.","Determine sample sizes, draw samples, and conduct data analysis using data frames. It specifically enables you to determine simple random sample sizes, stratified sample sizes, and complex stratified sample sizes using a secondary variable such as population; draw simple random samples and stratified random samples from sampling data frames; determine which observations are missing from a random sample, missing by strata, duplicated within a dataset; and perform data analysis, including proportions, margins of error and upper and lower bounds for simple, stratified and cluster sample designs.","A collection of procedures for analysing, visualising, and managing single-case data. These include piecewise linear regression models, multilevel models, overlap indices (PND, PEM, PAND, PET, tauU, baseline corrected tau), and randomization tests. Data preparation functions support outlier detection, handling missing values, scaling, truncating, rank transformation, and smoothing. An exporting function helps to generate html and latex tables in a publication friendly style. More details can be found at <https://jazznbass.github.io/scan-Book/>.","Functions and Datasets from Lohr, S. (1999), Sampling: Design and Analysis, Duxbury.","Statistical Methods to Analyse Sensory Data. SensoMineR: A package for sensory data analysis. S. Le and F. Husson (2008) <DOI:10.1111/j.1745-459X.2007.00137.x>.","Gives design points from a sequential full factorial-based Latin hypercube design, as described in Duan, Ankenman, Sanchez, and Sanchez (2015, Technometrics, <doi:10.1080/00401706.2015.1108233>).","Allows shiny developers to incorporate UI elements based on Google's Material design. See <https://material.io/guidelines/> for more information.","Provides tools for simulating spatially dependent predictors (continuous or binary), which are used to generate scalar outcomes in a (generalized) linear model framework. Continuous predictors are generated using traditional multivariate normal distributions or Gauss Markov random fields with several correlation function approaches (e.g., see Rue (2001) <doi:10.1111/1467-9868.00288> and Furrer and Sain (2010) <doi:10.18637/jss.v036.i10>), while binary predictors are generated using a Boolean model (see Cressie and Wikle (2011, ISBN: 978-0-471-69274-4)). Parameter vectors exhibiting spatial clustering can also be easily specified by the user.","Provides tools to help safely and efficiently organize Monte Carlo simulations in R. The package controls the structure and back-end of Monte Carlo simulations by utilizing a general generate-analyse-summarise strategy. The functions provided control common simulation issues such as re-simulating non-convergent results, support parallel back-end and MPI distributed computations, save and restore temporary files, aggregate results across independent nodes, and provide native support for debugging. For a pedagogical introduction to the package refer to Sigal and Chalmers (2016) <doi:10.1080/10691898.2016.1246953>, and for an in-depth overview of the package and its design philosophy see Chalmers and Adkins (2020) <doi:10.20982/tqmp.16.4.p248>.","A set of tools for determining the necessary sample size in order to identify the optimal dynamic treatment regime in a sequential, multiple assignment, randomized trial (SMART). Utilizes multiple comparisons with the best methodology to adjust for multiple comparisons. Designed for an arbitrary SMART design. Please see Artman (2018) <doi:10.1093/biostatistics/kxy064> for more details.","Computation of sparse portfolios for financial index tracking, i.e., joint selection of a subset of the assets that compose the index and computation of their relative weights (capital allocation). The level of sparsity of the portfolios, i.e., the number of selected assets, is controlled through a regularization parameter. Different tracking measures are available, namely, the empirical tracking error (ETE), downside risk (DR), Huber empirical tracking error (HETE), and Huber downside risk (HDR). See vignette for a detailed documentation and comparison, with several illustrative examples. The package is based on the paper: K. Benidis, Y. Feng, and D. P. Palomar, \"Sparse Portfolios for High-Dimensional Financial Index Tracking,\" IEEE Trans. on Signal Processing, vol. 66, no. 1, pp. 155-170, Jan. 2018. <doi:10.1109/TSP.2017.2762286>.","Implements the method of general semiparametric maximum likelihood estimation for logistic models in case-mother control-mother designs.","These functions provide procedures for selecting sites for spatial surveys using spatially balanced algorithms applied to discrete points, linear networks, or polygons. The probability survey designs available include independent random samples, stratified random samples, and unequal probability random samples (categorical or probability proportional to size). Design-based estimation based on the results from surveys is available for estimating totals, means, quantiles, CDFs, and linear models. The analyses rely on package survey for most results. Variance estimation options include a local neighborhood variance estimator that is appropriate for spatially-balanced survey designs. A reference for the survey design portion of the package is: D. L. Stevens, Jr. and A. R. Olsen (2004), \"Spatially-balanced sampling of natural resources.\", Journal of the American Statistical Association 99(465): 262-278, <DOI:10.1198/016214504000000250>. Additional helpful references for this package are A. R. Olsen, T. M. Kincaid, and Q. Payton (2012) and T. M. Kincaid and A. R. Olsen (2012), both of which are chapters in the book \"Design and Analysis of Long-Term Ecological Monitoring Studies\" (R. A. Gitzen, J. J. Millspaugh, A. B. Cooper, and D. S. Licht (eds.), Cambridge University Press, New York, <Online ISBN:9781139022422>).","Provides the necessary sample size for a longitudinal study with binary outcome in order to attain a pre-specified power while strictly maintaining the Type I error rate. Kapur K, Bhaumik R, Tang XC, Hur K, Reda DJ, Bhaumik D (2014) <doi:10.1002/sim.6203>.","Functions for drawing samples according to Sunter's sampling design, and for computing first and second order inclusion probabilities","Computes standard error and confidence interval of various descriptive statistics under various designs and sampling schemes. The main function, superbPlot(), can either return a plot or a dataframe with the statistic and its precision interval so that other plotting package can be used. See Cousineau (2017) <doi:10.5709/acp-0214-z> for a review or Cousineau (2005) <doi:10.20982/tqmp.01.1.p042>, Morey (2008) <doi:10.20982/tqmp.04.2.p061>, Baguley (2012) <doi:10.3758/s13428-011-0123-7>, Cousineau & Laurencelle (2016) <doi:10.1037/met0000055>, Cousineau & O'Brien (2014) <doi:10.3758/s13428-013-0441-z>.","Calculate point estimates and their standard errors in complex household surveys using bootstrap replicates. Bootstrapping considers survey design with a rotating panel. A comprehensive description of the methodology can be found under <https://statistikat.github.io/surveysd/articles/methodology.html>.","Builds, evaluates and validates a nomogram with survey data and right-censored outcomes.","Provides inference based on the survey package for the wide range of parametric models in the 'VGAM' package.","The TEQR package contains software to calculate the operating characteristics for the TEQR and the ACT designs.The TEQR (toxicity equivalence range) design is a toxicity based cumulative cohort design with added safety rules. The ACT (Activity constrained for toxicity) design is also a cumulative cohort design with additional safety rules. The unique feature of this design is that dose is escalated based on lack of activity rather than on lack of toxicity and is de-escalated only if an unacceptable level of toxicity is experienced.","Use the optimal test design approach by Birnbaum (1968, ISBN:9781593119348) and van der Linden (2018) <doi:10.1201/9781315117430> in constructing fixed and adaptive tests. Supports the following mixed-integer programming (MIP) solver packages: 'lpsymphony', 'Rsymphony', 'gurobi', 'lpSolve', and 'Rglpk'. The 'gurobi' package is not available from CRAN; see <https://www.gurobi.com/downloads/>. See vignette for installing 'Rsymphony' package on Mac systems.","Functions to design phase 1 trials using an isotonic regression based design incorporating time-to-event information. Simulation and design functions are available, which incorporate information about followup and DLTs, and apply isotonic regression to devise estimates of DLT probability.","Performs two-way tests in independent groups designs; Parametric Bootstrap based Generalized Test and Generalized Pivotal Quantity based Generalized Test (Weerahandi and Krishnamoorthy, 2019) <doi:10.1080/03610926.2017.1419264>. The package performs descriptive statistics and graphical approaches. Moreover, it assesses variance homogeneity and normality of data in each group via tests and plots. All 'twowaytests' functions are designed for two-way layout.","Define and use graphical elements of corporate design manuals in R. The 'unikn' package provides color functions (by defining dedicated colors and color palettes, and commands for changing, viewing, and using them) and styled text elements (e.g., for marking, underlining, or plotting colored titles). The pre-defined range of colors and text functions is based on the corporate design of the University of Konstanz <https://www.uni-konstanz.de/>, but can be adapted and extended for other institutions and purposes.","US Census 2000 Designated Places shapefiles and additional demographic data from the SF1 100 percent files. This data set contains polygon files in lat/lon coordinates and the corresponding demographic data for a number of different variables.","Provides statistical methods for the design and analysis of a calibration study, which aims for calibrating measurements using two different methods. The package includes sample size calculation, sample selection, regression analysis with error-in measurements and change-point regression. The method is described in Tian, Durazo-Arvizu, Myers, et al. (2014) <DOI:10.1002/sim.6235>."],["yes","no","no","yes","no","yes","yes","no","no","yes","no","yes","no","no","no","yes","yes","no","no","no","no","yes","no","no","no","no","no","no","no","no","yes","no","no","no","yes","yes","yes","yes","no","yes","no","yes","no","no","no","no","no","yes","no","yes","no","yes","no","yes","no","no","no","no","yes","no","no","no","yes","yes","no","no","no","no","yes","no","yes","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","yes","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no"],["no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no"],["<div class='tooltip'>libs<span class='tooltiptext'>Load multiple packages and install if neededThe 'agridat' package uses dozens of packages in the examples for each dataset.The 'libs' function provides a simple way to load multiple packages at once,and can install any missing packages on-the-fly.This is very similar to the `pacman::p_load` function.<\/span><\/div>","<div class='tooltip'>apdesign<span class='tooltiptext'>apdesign: AP codingapdesign returns a data frame with additive polynomial coding<\/span><\/div>, <div class='tooltip'>apdesign_i<span class='tooltiptext'>apdesign_i: AP coding for a single subject<\/span><\/div>","<div class='tooltip'>argonAlert<span class='tooltiptext'>Build an argon alert<\/span><\/div>, <div class='tooltip'>argonBadge<span class='tooltiptext'>Build an argon badge<\/span><\/div>, <div class='tooltip'>argonBlur<span class='tooltiptext'>Build an argon blur effect<\/span><\/div>, <div class='tooltip'>argonButton<span class='tooltiptext'>Build an argon button<\/span><\/div>, <div class='tooltip'>argonCard<span class='tooltiptext'>Build an argon card<\/span><\/div>, <div class='tooltip'>argonCarousel<span class='tooltiptext'>Build an argon carousel<\/span><\/div>, <div class='tooltip'>argonCarouselItem<span class='tooltiptext'>Build an argon carousel item<\/span><\/div>, <div class='tooltip'>argonCascade<span class='tooltiptext'>Build an argon cascade effect<\/span><\/div>, <div class='tooltip'>argonCascadeItem<span class='tooltiptext'>Build an argon cascade item<\/span><\/div>, <div class='tooltip'>argonColumn<span class='tooltiptext'>Build an argon column<\/span><\/div>, <div class='tooltip'>argonContact<span class='tooltiptext'>Build an argon contact form<\/span><\/div>, <div class='tooltip'>argonContainer<span class='tooltiptext'>Create a boostrap 4 container<\/span><\/div>, <div class='tooltip'>argonDropdown<span class='tooltiptext'>Build an argon dropdown menu<\/span><\/div>, <div class='tooltip'>argonDropdownItem<span class='tooltiptext'>Build an argon dropdown item<\/span><\/div>, <div class='tooltip'>argonFooter<span class='tooltiptext'>Build an argon footer<\/span><\/div>, <div class='tooltip'>argonH1<span class='tooltiptext'>A custom h1 tag<\/span><\/div>, <div class='tooltip'>argonIcon<span class='tooltiptext'>Build an argon Nucleo icon<\/span><\/div>, <div class='tooltip'>argonIconWrapper<span class='tooltiptext'>Build an argon Nucleo icon wrapper<\/span><\/div>, <div class='tooltip'>argonImage<span class='tooltiptext'>Build an argon image container<\/span><\/div>, <div class='tooltip'>argonInstall<span class='tooltiptext'>Install argon assets<\/span><\/div>, <div class='tooltip'>argonLead<span class='tooltiptext'>A custom p tag<\/span><\/div>, <div class='tooltip'>argonMargin<span class='tooltiptext'>Change tag margins<\/span><\/div>, <div class='tooltip'>argonModal<span class='tooltiptext'>Build an argon modal<\/span><\/div>, <div class='tooltip'>argonMuted<span class='tooltiptext'>Apply the class muted to a text<\/span><\/div>, <div class='tooltip'>argonNavbar<span class='tooltiptext'>Build an argon navbar<\/span><\/div>, <div class='tooltip'>argonNavItem<span class='tooltiptext'>Build an argon navbar item<\/span><\/div>, <div class='tooltip'>argonNavMenu<span class='tooltiptext'>Build an argon navbar menu<\/span><\/div>, <div class='tooltip'>argonPadding<span class='tooltiptext'>Change tag padding<\/span><\/div>, <div class='tooltip'>argonPage<span class='tooltiptext'>Build an argon page<\/span><\/div>, <div class='tooltip'>argonPageTemplate<span class='tooltiptext'>Create an HTML version of the argonPage<\/span><\/div>, <div class='tooltip'>argonPagination<span class='tooltiptext'>Build an argon pagination container<\/span><\/div>, <div class='tooltip'>argonPaginationItem<span class='tooltiptext'>Build an argon pagination item<\/span><\/div>, <div class='tooltip'>argonPersp<span class='tooltiptext'>Build an argon perspective effect<\/span><\/div>, <div class='tooltip'>argonProfile<span class='tooltiptext'>Build an argon profile card<\/span><\/div>, <div class='tooltip'>argonProfileStat<span class='tooltiptext'>Build an argon profile card stat item<\/span><\/div>, <div class='tooltip'>argonProfileStats<span class='tooltiptext'>Build an argon profile card stats container<\/span><\/div>, <div class='tooltip'>argonProgress<span class='tooltiptext'>Build an argon progress bar<\/span><\/div>, <div class='tooltip'>argonQuote<span class='tooltiptext'>A custom quotation tool<\/span><\/div>, <div class='tooltip'>argonRow<span class='tooltiptext'>Build an argon row<\/span><\/div>, <div class='tooltip'>argonSection<span class='tooltiptext'>Build an argon section container<\/span><\/div>, <div class='tooltip'>argonSocialButton<span class='tooltiptext'>Build an argon social button<\/span><\/div>, <div class='tooltip'>argonTab<span class='tooltiptext'>Build an argon tab item<\/span><\/div>, <div class='tooltip'>argonTabSet<span class='tooltiptext'>Build an argon tabs<\/span><\/div>, <div class='tooltip'>argonTextColor<span class='tooltiptext'>A custom color fo your text<\/span><\/div>, <div class='tooltip'>argonTooltip<span class='tooltiptext'>Create a boostrap 4 tooltip<\/span><\/div>, <div class='tooltip'>argonUser<span class='tooltiptext'>Build an argon user image<\/span><\/div>","<div class='tooltip'>combn.test<span class='tooltiptext'>Implements weighted inverse normal and Fisher combination tests for combining p-values for adaptive seamless designs.<\/span><\/div>, <div class='tooltip'>dunnett.test<span class='tooltiptext'>Implements Dunnett's test (Dunnett, 1955) for many-to-one comparisons.<\/span><\/div>, <div class='tooltip'>gsubpop.sim<span class='tooltiptext'>Function subpop.sim runs simulations for a trial design that tests an experimental treatment against a single control treatment group in a seamless adaptive trial with co-primary analyses in a pre-defined subgroup and the full population. An interim analysis is undertaken using an early outcome measure and a decision is made on whether to continue with both full and subpopulations, the subpopulation only or the full population, using a pre-defined selection rule. A number of different methods to control the family wise error rate are implemented; (i) the treatment is compared to the control in the subpopulation and full populations using Simes test and the inverse normal combination function used to combine p-values before and after design adaptation, (ii) as (i) but the bivariate normal method of Spiessens and Debois (2010) is used to control the type I error rate, (iii) as (i) but a Bonferroni test is used and (iv) a conditional error function approach using the Spiessens and Debois test. Data are simulated for the early and final outcome measures, subpopulation prevalence and correlation between the final and the early outcomes. This function should not generally be called by the user. The more user-friendly function subpop.sim covers most common applications.<\/span><\/div>, <div class='tooltip'>gtreatsel.sim<span class='tooltiptext'>Function treatsel.sim runs simulations for a trial design that tests a number of experimental treatments against a single control treatment group in a seamless adaptive trial. Test treatments are compared to the control treatment using Dunnett's many-to-one testing procedure. An interim analysis is undertaken using an early outcome measure for each treatment (and control). A decision is made on which of the treatments to take forward, using a pre-defined selection rule. Data are simulated for the final outcome measure, and data from the interim and final analyses for the final outcome measure are combined together using either the inverse normal or Fisher combination test, and hypotheses tested at the selected level.This function should not generally be called by the user. The more user-friendly function treatsel.sim covers most common applications.<\/span><\/div>, <div class='tooltip'>hyp.test<span class='tooltiptext'>Implements the closure principle (Marcus et al., 1976) for controlling the familywise type I error rate in ASD.<\/span><\/div>, <div class='tooltip'>select.rule<span class='tooltiptext'>Function select.rule provides a number of options for selecting treatments at an interim analysis in ASD.<\/span><\/div>, <div class='tooltip'>simeans.binormal<span class='tooltiptext'>Simulates bivariate normal means; for use with asd.sim and gasd.sim in ASD.<\/span><\/div>, <div class='tooltip'>subpop.sim<span class='tooltiptext'>Function subpop.sim runs simulations for a trial design that tests an experimental treatment against a single control treatment group in a seamless adaptive trial with co-primary analyses in a pre-defined subgroup and the full population. An interim analysis is undertaken using an early outcome measure and a decision is made on whether to continue with both full and subpopulations, the subpopulation only or the full population, using a pre-defined selection rule. A number of different methods to control the family wise error rate are implemented; (i) the treatment is compared to the control in the subpopulation and full populations using Simes test and the inverse normal combination function used to combine p-values before and after design adaptation, (ii) as (i) but the bivariate normal method of Spiessens and Debois (2010) is used to control the type I error rate, (iii) as (i) but a Bonferroni test is used and (iv) a conditional error function approach using the Spiessens and Debois test. Data are simulated for the early and final outcome measures, subpopulation prevalence and correlation between the final and the early outcomes.<\/span><\/div>, <div class='tooltip'>treatsel.sim<span class='tooltiptext'>Function treatsel.sim runs simulations for a trial design that tests a number of experimental treatments against a single control treatment group in a seamless adaptive trial. Test treatments are compared to the control treatment using Dunnett's many-to-one testing procedure. An interim analysis is undertaken using an early outcome measure for each treatment (and control). A decision is made on which of the treatments to take forward, using a pre-defined selection rule. Data are simulated for the final outcome measure, and data from the interim and final analyses for the final outcome measure are combined together using either the inverse normal or Fisher combination test, and hypotheses tested at the selected level.<\/span><\/div>","<div class='tooltip'>anovaBF<span class='tooltiptext'>This function computes Bayes factors for all main-effects and interactioncontrasts in an ANOVA design.<\/span><\/div>, <div class='tooltip'>as.BFBayesFactor<span class='tooltiptext'>This function coerces objects to the BFBayesFactor class<\/span><\/div>, <div class='tooltip'>as.BFprobability<span class='tooltiptext'>This function coerces objects to the BFprobability class<\/span><\/div>, <div class='tooltip'>BFInfo<span class='tooltiptext'>Prints the version, revision, and date information for the BayesFactor package<\/span><\/div>, <div class='tooltip'>BFManual<span class='tooltiptext'>This function opens the HTML manual for the BayesFactor package in whateverbrowser is configured.<\/span><\/div>, <div class='tooltip'>compare<span class='tooltiptext'>This method is used primarily in the backend, and will only rarely be calledby the end user. But see the examples below for a demonstration.<\/span><\/div>, <div class='tooltip'>contingencyTableBF<span class='tooltiptext'>This function computes Bayes factors for contingency tables.<\/span><\/div>, <div class='tooltip'>correlationBF<span class='tooltiptext'>Bayes factors or posterior samples for correlations.<\/span><\/div>, <div class='tooltip'>enumerateGeneralModels<span class='tooltiptext'>Generate lists of nested models, given a model formula<\/span><\/div>, <div class='tooltip'>extractBF<span class='tooltiptext'>Extract the Bayes factor from an object<\/span><\/div>, <div class='tooltip'>extractOdds<span class='tooltiptext'>Extract the odds from an object<\/span><\/div>, <div class='tooltip'>extractProbabilities<span class='tooltiptext'>Extract the probabilities from an object<\/span><\/div>, <div class='tooltip'>filterBF<span class='tooltiptext'>Filter the elements of an object according to some pre-specified criteria<\/span><\/div>, <div class='tooltip'>generalTestBF<span class='tooltiptext'>This function computes Bayes factors corresponding to restrictions on a full model.<\/span><\/div>, <div class='tooltip'>linearReg.R2stat<span class='tooltiptext'>Using the classical R^2 test statistic for (linear) regression designs, thisfunction computes the corresponding Bayes factor test.<\/span><\/div>, <div class='tooltip'>lmBF<span class='tooltiptext'>This function computes Bayes factors, or samples from the posterior, ofspecific linear models (either ANOVA or regression).<\/span><\/div>, <div class='tooltip'>logCumMeanExpLogs<span class='tooltiptext'>Given a vector of numeric values of real values represented in log form,logMeanExpLogs computes the logarithm of the mean of the(exponentiated) values. logCumMeanExpLogs computes the logarithm ofthe cumulative mean.<\/span><\/div>, <div class='tooltip'>logMeanExpLogs<span class='tooltiptext'>Given a vector of numeric values of real values represented in log form,logMeanExpLogs computes the logarithm of the mean of the(exponentiated) values. logCumMeanExpLogs computes the logarithm ofthe cumulative mean.<\/span><\/div>, <div class='tooltip'>logSummaryStats<span class='tooltiptext'>Given a vector of numeric values of real values represented in log form,logMeanExpLogs computes the logarithm of the mean of the(exponentiated) values. logCumMeanExpLogs computes the logarithm ofthe cumulative mean.<\/span><\/div>, <div class='tooltip'>meta.ttestBF<span class='tooltiptext'>This function computes mata-analytic Bayes factors, or samples from the posterior, forone- and two-sample designs where multiple t values have been observed.<\/span><\/div>, <div class='tooltip'>model.matrix<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>newPriorOdds<span class='tooltiptext'>Create a prior odds object from a Bayes factor object<\/span><\/div>, <div class='tooltip'>nWayAOV<span class='tooltiptext'>Computes a single Bayes factor, or samples from the posterior, for an ANOVAmodel defined by a design matrix<\/span><\/div>, <div class='tooltip'>oneWayAOV.Fstat<span class='tooltiptext'>Using the classical F test statistic for a balanced one-way design, this function computes the corresponding Bayes factor test.<\/span><\/div>, <div class='tooltip'>posterior<span class='tooltiptext'>This function samples from the posterior distribution of a BFmodel,which can be obtained from a BFBayesFactor object. If there is morethan one numerator in the BFBayesFactor object, the indexargument can be passed to select one numerator.<\/span><\/div>, <div class='tooltip'>priorLogodds<-<span class='tooltiptext'>Set prior log odds in an object<\/span><\/div>, <div class='tooltip'>priorOdds<-<span class='tooltiptext'>Set prior odds in an object<\/span><\/div>, <div class='tooltip'>proportionBF<span class='tooltiptext'>Bayes factors or posterior samples for binomial, geometric, or neg. binomial data.<\/span><\/div>, <div class='tooltip'>recompute<span class='tooltiptext'>Take an object and redo the computation (useful for sampling). In cases where sampling isused to compute the Bayes factor, the estimate of the precision of new samples will be addedto the estimate precision of the old sample will be added to produce a new estimate of theprecision.<\/span><\/div>, <div class='tooltip'>regressionBF<span class='tooltiptext'>This function simultaneously computes Bayes factors for groups of models inregression designs<\/span><\/div>, <div class='tooltip'>t<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>ttest.tstat<span class='tooltiptext'>Using the classical t test statistic for a one- or two-sample design, thisfunction computes the corresponding Bayes factor test.<\/span><\/div>, <div class='tooltip'>ttestBF<span class='tooltiptext'>This function computes Bayes factors, or samples from the posterior, forone- and two-sample designs.<\/span><\/div>, <div class='tooltip'>which.max<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>which.min<span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>analyze<span class='tooltiptext'>For routine use, these functions will not need to be called directly but are called from within the design functions (see designOBF). If needed, use analyze for any class representing a binary sequential boundary (see bound-class),and the appropriate function is called.<\/span><\/div>, <div class='tooltip'>analyzeBound<span class='tooltiptext'>For routine use, these functions will not need to be called directly but are called from within the design functions (see designOBF). If needed, use analyze for any class representing a binary sequential boundary (see bound-class),and the appropriate function is called.<\/span><\/div>, <div class='tooltip'>analyzeBoundNBF<span class='tooltiptext'>For routine use, these functions will not need to be called directly but are called from within the design functions (see designOBF). If needed, use analyze for any class representing a binary sequential boundary (see bound-class),and the appropriate function is called.<\/span><\/div>, <div class='tooltip'>designAb<span class='tooltiptext'>There are several functions that create binary sequential boundaries. The function designAb allows great flexibility in creating user defined boundaries. The functions designOBF and designOBFpower create boudaries of the O-Brien-Fleming type, extending those boundaries to allow looks after every observation. The former (designOBF) uses a user defined maximum number of observations (Nmax), while the latter (designOBFpower) uses the power argument to try to find a design with a smaller maximum that achieves the desired power. The functions designFixed and designFixedpower are analogous for fixed sample designs. The function designSimon uses the ph2simonfrom the clinfun package to create boundaries using Simon's (1989) two-stage design.<\/span><\/div>, <div class='tooltip'>designFixed<span class='tooltiptext'>There are several functions that create binary sequential boundaries. The function designAb allows great flexibility in creating user defined boundaries. The functions designOBF and designOBFpower create boudaries of the O-Brien-Fleming type, extending those boundaries to allow looks after every observation. The former (designOBF) uses a user defined maximum number of observations (Nmax), while the latter (designOBFpower) uses the power argument to try to find a design with a smaller maximum that achieves the desired power. The functions designFixed and designFixedpower are analogous for fixed sample designs. The function designSimon uses the ph2simonfrom the clinfun package to create boundaries using Simon's (1989) two-stage design.<\/span><\/div>, <div class='tooltip'>designFixedpower<span class='tooltiptext'>There are several functions that create binary sequential boundaries. The function designAb allows great flexibility in creating user defined boundaries. The functions designOBF and designOBFpower create boudaries of the O-Brien-Fleming type, extending those boundaries to allow looks after every observation. The former (designOBF) uses a user defined maximum number of observations (Nmax), while the latter (designOBFpower) uses the power argument to try to find a design with a smaller maximum that achieves the desired power. The functions designFixed and designFixedpower are analogous for fixed sample designs. The function designSimon uses the ph2simonfrom the clinfun package to create boundaries using Simon's (1989) two-stage design.<\/span><\/div>, <div class='tooltip'>designOBF<span class='tooltiptext'>There are several functions that create binary sequential boundaries. The function designAb allows great flexibility in creating user defined boundaries. The functions designOBF and designOBFpower create boudaries of the O-Brien-Fleming type, extending those boundaries to allow looks after every observation. The former (designOBF) uses a user defined maximum number of observations (Nmax), while the latter (designOBFpower) uses the power argument to try to find a design with a smaller maximum that achieves the desired power. The functions designFixed and designFixedpower are analogous for fixed sample designs. The function designSimon uses the ph2simonfrom the clinfun package to create boundaries using Simon's (1989) two-stage design.<\/span><\/div>, <div class='tooltip'>designOBFpower<span class='tooltiptext'>There are several functions that create binary sequential boundaries. The function designAb allows great flexibility in creating user defined boundaries. The functions designOBF and designOBFpower create boudaries of the O-Brien-Fleming type, extending those boundaries to allow looks after every observation. The former (designOBF) uses a user defined maximum number of observations (Nmax), while the latter (designOBFpower) uses the power argument to try to find a design with a smaller maximum that achieves the desired power. The functions designFixed and designFixedpower are analogous for fixed sample designs. The function designSimon uses the ph2simonfrom the clinfun package to create boundaries using Simon's (1989) two-stage design.<\/span><\/div>, <div class='tooltip'>designSimon<span class='tooltiptext'>There are several functions that create binary sequential boundaries. The function designAb allows great flexibility in creating user defined boundaries. The functions designOBF and designOBFpower create boudaries of the O-Brien-Fleming type, extending those boundaries to allow looks after every observation. The former (designOBF) uses a user defined maximum number of observations (Nmax), while the latter (designOBFpower) uses the power argument to try to find a design with a smaller maximum that achieves the desired power. The functions designFixed and designFixedpower are analogous for fixed sample designs. The function designSimon uses the ph2simonfrom the clinfun package to create boundaries using Simon's (1989) two-stage design.<\/span><\/div>, <div class='tooltip'>EN<span class='tooltiptext'>Calculate expected sample size for bound object, after inputing theta.<\/span><\/div>, <div class='tooltip'>getAlternative<span class='tooltiptext'>Two functions to find tsalpha and alternative.<\/span><\/div>, <div class='tooltip'>getTSalpha<span class='tooltiptext'>Two functions to find tsalpha and alternative.<\/span><\/div>, <div class='tooltip'>modify<span class='tooltiptext'>Modify several different aspects of a binary sequential boundary. Most modifications do not change the stopping boundaries. The exceptions are 'missN', which allows modifications for missing assessments,and 'closeout', which allows for early stopping of the trial for administrative reasons (i.e., reasons that do not depend on the responses in the trial). Other modifications possible: level of the confidence intervals (using tsalpha, conf.level, or alternative), which boundaries are binding (i.e., can change from a boundary with binding futility boundaries to one with non-binding futility boundaries), null hypothesis value (theta0), and whether the confidence intervals should match the non-binding futility p-values on the superiority boundaries (cipMatch).<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>points<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>powerBsb<span class='tooltiptext'>Calculate power from boundEst object for vector of alternatives<\/span><\/div>, <div class='tooltip'>print.stopTable<span class='tooltiptext'>Takes boundEst object and creates data frame with p-values and confidence intervals for stopping points between Nmin and Nmax.<\/span><\/div>, <div class='tooltip'>prStop<span class='tooltiptext'>Calculates the probability of stopping at any point in a binary sequential boundary.<\/span><\/div>, <div class='tooltip'>stopTable<span class='tooltiptext'>Takes boundEst object and creates data frame with p-values and confidence intervals for stopping points between Nmin and Nmax.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>unirootDiscrete<span class='tooltiptext'>Let f be a non-increasing (or non-decreasing) function that changes sign within the interval specified. If 'pos.side'=TRUE (or FALSE) then unirootDiscrete finds the value x such that f(x) is closest to the sign change and is positive (or negative).<\/span><\/div>","<div class='tooltip'>BsProb<span class='tooltiptext'>Marginal factor posterior probabilities and model posterior probabilitiesfrom designed screening experiments are calculated according to Box and Meyer'sBayesian procedure.<\/span><\/div>, <div class='tooltip'>DanielPlot<span class='tooltiptext'>Normal plot of effects from a two level factorial experiment.<\/span><\/div>, <div class='tooltip'>LenthPlot<span class='tooltiptext'>Plot of the factor effects with significance levels based on robustestimation of contrast standard errors.<\/span><\/div>, <div class='tooltip'>MD<span class='tooltiptext'>Best follow-up experiments based on the MD criterionare suggested to discriminate between competing models.<\/span><\/div>","<div class='tooltip'>augment<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsaugment, tidy<\/span><\/div>, <div class='tooltip'>build_sequential_models<span class='tooltiptext'>Simplify the process of building multiple models in a sequentialorder. This is particularly helpful in epidemiological cases of testingeffect of additional parameters. Every parameter should be theoretically apart of the causal model for the exposure-outcome relationship.<\/span><\/div>, <div class='tooltip'>circ_center<span class='tooltiptext'>Based on a centering time point, shifts a vector to a \"before\"and \"after\" system to help align multiple individuals to a universal time,like the sunrise or any other appropriatehttps://en.wikipedia.org/wiki/Zeitgeberzeitgeiber. Originally intendedto expand upon the card::circ_sun function.<\/span><\/div>, <div class='tooltip'>circ_compare_groups<span class='tooltiptext'>Takes data and returns a summary table of continuousvariable based on a categorical variable. This summary is repeat bytime groups to help describe a circadian pattern.<\/span><\/div>, <div class='tooltip'>circ_odds<span class='tooltiptext'>Creates an OR table for each time point of data given, initiallyapplied to any grouping variable (particularly hour/time of day).<\/span><\/div>, <div class='tooltip'>circ_rad<span class='tooltiptext'>Converts a time unit to radians given a known period. Allows for vectorization.<\/span><\/div>, <div class='tooltip'>circ_sun<span class='tooltiptext'>Gets sunrise and sunset times based on date and location usingthe algorithm by thehttp://www.edwilliams.org/sunrise_sunset_algorithm.htmUnited StatesNaval Observatory. Uses the sensible default of official zenith forcalculations. Requires geographic position in latitude and longitude tocalculate sunrise or sunset. Uses the function lutz::tz_lookup_coords.<\/span><\/div>, <div class='tooltip'>cosinor<span class='tooltiptext'>cosinor() fits a regression model of a time variable to acontinuous outcome use trigonometric features. This approaches uses thelinearization of the parameters to assess their statistics anddistribution.<\/span><\/div>, <div class='tooltip'>cosinor_area<span class='tooltiptext'>Formulas for creating the area of the ellipse to identifyconfidence intervals, direction, and graphing purposes.<\/span><\/div>, <div class='tooltip'>cosinor_features<span class='tooltiptext'>Extract the special/global features of a multiple component cosinor. In a multiple component model, there are specific parameters that are not within the model itself, but must be extracted from the model fit. When extracted, can be used to improve the plot of a multiple component cosinor. However, this is only possible if the cosinor is harmonic (see details). For single-component models, the orthophase is the same as the acrophase and the global amplitude Global Amplitude (Ag) = the overall amplitude is defined as half the difference between the peak and trough values Orthophase (Po) = the lag until the peak time Bathyphase (Pb) = the lag until the trough time<\/span><\/div>, <div class='tooltip'>cosinor_goodness_of_fit<span class='tooltiptext'>Goodness of fit of a cosinor from data that has multiplecollections at different timepoints or from multiple cycles. The RSS ispartitioned into pure error (SSPE) and lack of fit (SSLOF). An F-testcompares the SSPE and SSLOF to detect appropriateness of model.SSLOF = RSS - SSPESSPE = \\sum_{i} \\sum_{l} ( Y_{il} - \\overline{Y}_{i} )^2The fitted values for each time point are:\\overline{Y}_{i} = \\frac{ \\sum_{l} Y_{il} }{ n_{i}}<\/span><\/div>, <div class='tooltip'>cosinor_zero_amplitude<span class='tooltiptext'>Zero amplitude test assesses how well the circadian pattern fitsthe data, essentially detecting the present of a rhythm to the data.<\/span><\/div>, <div class='tooltip'>geom_residuals<span class='tooltiptext'>geom_residuals makes a diagnostic plot of residuals versusfitted data for linear models. Does not yet accept logistic models<\/span><\/div>, <div class='tooltip'>ggcircadian<span class='tooltiptext'>Converts the output ofcard::circ_compare_groups into a complex geomthat is broken down by time/hour and HRV (or any other continuousvariable). Each hour is then separated by the grouping variable.<\/span><\/div>, <div class='tooltip'>ggcosinor<span class='tooltiptext'>ggplot of cosinor model that can visualize a variety of cosinormodel subtypes, including single-component, multiple-component, individual,and population cosinor models, built using card::cosinor. For singlecomponent cosinor, the following values are plotted: M = midline estimating statistic of rhythm A = amplitude P = phi or acrophase (shift from 0 to peak)If using a multiple-component cosinor, the terms are different. If theperiods or frequencies resonate or are harmonic, then the following arecalculated. If the periods are not harmonic, the values are justdescriptors of the curve. M = midline estimating statistic of rhythm Ag = global amplitude, which is the distance between peak and trough(this is the same value as the amplitude from single component) Po = orthophase (the equivalent of the acrophase in a single component),the lag time to peak value Pb = bathyphase, the lag time to trough value<\/span><\/div>, <div class='tooltip'>ggellipse<span class='tooltiptext'>This is a ggplot-styled graphical representation of the ellipseregion generated by the cosinor analysis. It requires the same data used bycosinor model to be fit with the model card::cosinor. This includesthe amplitude, acrophase,<\/span><\/div>, <div class='tooltip'>ggerror<span class='tooltiptext'>Creates a ggplot geom that can be extended and accept otherggplot layers. Shows residual error from the regression mean for differenttypes of regression models.<\/span><\/div>, <div class='tooltip'>ggforest<span class='tooltiptext'>Creates an OR plot for each hour of data given. Its aggplot format so additional variables, like titles, can be added in.<\/span><\/div>, <div class='tooltip'>hrv_linear_model<span class='tooltiptext'>hrv_linear_model Linear models for each HRV measure.<\/span><\/div>, <div class='tooltip'>recur_followup_table<span class='tooltiptext'>Makes a before/after dataset using a unique ID that followspatients between studies, to allow for comparison over time.<\/span><\/div>, <div class='tooltip'>recur_summary<span class='tooltiptext'>recur_summary Creates a table with summary ofrecurrent events<\/span><\/div>, <div class='tooltip'>recur_survival_table<span class='tooltiptext'>Reformats recurrent event data (wide) into different models forsurvival analysis, but can also be used for simple survival analysis tablesas well. Importantly, for large datasets, this function will showsignificant slow-down since it uses an intuitive approach on defining thedatasets. Future iterations will create a vectorized approach that shouldprovide performance speed-ups. For recurrent events, the final censoring event can include death, or canbe ignored if its not considered a failure event. For simple survival analysis, death censoring should be left as NULL, andthe event (e.g. \"date_of_death\"), should be used as a single event.datesparameter. The function will do the rest.<\/span><\/div>, <div class='tooltip'>recurrent_model_building<span class='tooltiptext'>Takes a different covariate groups to generate several modelsfor recurrent event survival analyses.<\/span><\/div>, <div class='tooltip'>recurrent_propensity<span class='tooltiptext'>recurrent_propensity Adds propensity score to any data set that is being regressed upon.<\/span><\/div>, <div class='tooltip'>tidy<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsaugment, tidy<\/span><\/div>","<div class='tooltip'>conditional.assoc.test.pure.pop.hwe.le.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>marginal.assoc.test.hwe.in.controls<span class='tooltiptext'>Tests for association of single SNP locus and a binary phenotype in case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>marginal.assoc.test.kpx.kpy<span class='tooltiptext'>Tests for association of single SNP locus and a binary phenotype in case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>marginal.assoc.test.pop.hwe.kpy<span class='tooltiptext'>Tests for association of single SNP locus and a binary phenotype in case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>marginal.assoc.test.unconstrained.chisq<span class='tooltiptext'>Tests for association of single SNP locus and a binary phenotype in case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>marginal.assoc.test.unconstrained.gsq<span class='tooltiptext'>Tests for association of single SNP locus and a binary phenotype in case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.case.only<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.hwe.le.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.ind.3d<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.kpx.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.pop.hwe.le.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.pop.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.pure.pop.hwe.le.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.pure.pop.kpy<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.pure.unconstrained<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.unconstrained.chisq<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>, <div class='tooltip'>pairwise.assoc.test.unconstrained.gsq<span class='tooltiptext'>Tests for joint association of a pair of SNP loci and a binary phenotype in case-only, case-control, case-population, and case-control-population designs, under various assumptions.<\/span><\/div>","<div class='tooltip'>cp.scree<span class='tooltiptext'>Line plot showing the relationship between the criterion used to assess column position balance and the number of iterations in the column position balancing routine.<\/span><\/div>, <div class='tooltip'>dcm.design<span class='tooltiptext'>Generate an optimal fractional factorial design given vectors of factor lengths.<\/span><\/div>, <div class='tooltip'>dcm.design.cand<span class='tooltiptext'>Generate an optimal restricted fractional factorial design given a pre-generated candidate set.<\/span><\/div>, <div class='tooltip'>dcm.design.effcy<span class='tooltiptext'>Internal function to calculate mathematical efficiencies of designs.<\/span><\/div>, <div class='tooltip'>dcm.design.sort<span class='tooltiptext'>Internal function to append card, version and task variables to design.<\/span><\/div>, <div class='tooltip'>optBlockC<span class='tooltiptext'>INTERNAL: Simplified wrapper for blocking of experimental designs using optBlock in the AlgDesign package.<\/span><\/div>, <div class='tooltip'>optFederovC<span class='tooltiptext'>INTERNAL: Simplified wrapper for calculating exact algorithmic designs using Federov's exchange algorithm. Based on optFederov in the AlgDesign package.<\/span><\/div>, <div class='tooltip'>pw.eval<span class='tooltiptext'>Internal function to evaluate whether the two-way (pairwise) frequencies of items in a matrix are balanced.<\/span><\/div>, <div class='tooltip'>tradeoff.des<span class='tooltiptext'>Generate a design to be used for MaxDiff and related tradeoff exercises.<\/span><\/div>, <div class='tooltip'>write.tab<span class='tooltiptext'>Internal function that acts as an alias to write.table, appending extra arguments.<\/span><\/div>","<div class='tooltip'>crispRdesignRUI<span class='tooltiptext'>Activates the shiny UI for the crispRdesignR package<\/span><\/div>, <div class='tooltip'>Doench_2016_processing<span class='tooltiptext'>Warning: This function is not designed to be directly called by the user. This function is used internally in sgRNA_design() and sgRNA_design_function().Internal function that encodes all sgRNA sequence information into a data frame. This data frame is then used in conjunction with the Rule_Set_2_Model to predict effciency scores for the generated sgRNA.<\/span><\/div>, <div class='tooltip'>getofftargetdata<span class='tooltiptext'>Will provide a data frame with all information about the generated sgRNA returned by the sgRNA_design function.<\/span><\/div>, <div class='tooltip'>getsgRNAdata<span class='tooltiptext'>Will provide a data frame with all information about the generated sgRNA returned by the sgRNA_design function.<\/span><\/div>, <div class='tooltip'>sgRNA_design<span class='tooltiptext'>sgRNA_design returns information to design sgRNA sequences based on a given target sequence, a genome to annotate off-target information, and a genome annoation file (.gtf), to annotate the off-target findings.<\/span><\/div>, <div class='tooltip'>sgRNA_design_function<span class='tooltiptext'>Warning: This function should not be directly called by the user - it must be called though RunShiny.RDesigns sgRNA based on inputs provided in the Shiny App.<\/span><\/div>","<div class='tooltip'>%~%<span class='tooltiptext'>This is a modified version of match that supports tolerance.<\/span><\/div>, <div class='tooltip'>approximate<span class='tooltiptext'>It is recommended to use set.seed before, in orderto be able to reproduce the resulting approximating model exactly.<\/span><\/div>, <div class='tooltip'>as.list<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>biomLevel<span class='tooltiptext'>Compute the biomarker level for a given dose, given model and samples<\/span><\/div>, <div class='tooltip'>CohortSizeConst<span class='tooltiptext'>Initialization function for \"CohortSizeConst\"<\/span><\/div>, <div class='tooltip'>CohortSizeDLT<span class='tooltiptext'>Initialization function for \"CohortSizeDLT\"<\/span><\/div>, <div class='tooltip'>CohortSizeMax<span class='tooltiptext'>Initialization function for \"CohortSizeMax\"<\/span><\/div>, <div class='tooltip'>CohortSizeMin<span class='tooltiptext'>Initialization function for \"CohortSizeMin\"<\/span><\/div>, <div class='tooltip'>CohortSizeParts<span class='tooltiptext'>Initialization function for \"CohortSizeParts\"<\/span><\/div>, <div class='tooltip'>CohortSizeRange<span class='tooltiptext'>Initialization function for \"CohortSizeRange\"<\/span><\/div>, <div class='tooltip'>crmPackExample<span class='tooltiptext'>Calling this helper function should open the example.pdf document,residing in the doc subfolder of the package installation directory.<\/span><\/div>, <div class='tooltip'>crmPackHelp<span class='tooltiptext'>This convenience function opens your browser with the help pages forcrmPack.<\/span><\/div>, <div class='tooltip'>Data<span class='tooltiptext'>This is the function for initializing a \"Data\" class object.<\/span><\/div>, <div class='tooltip'>DataDual<span class='tooltiptext'>This is the function for initializing a \"DataDual\" class object.<\/span><\/div>, <div class='tooltip'>DataMixture<span class='tooltiptext'>This is the function for initializing a \"DataMixture\" class object.<\/span><\/div>, <div class='tooltip'>DataParts<span class='tooltiptext'>This is the function for initializing a DataPartsobject.<\/span><\/div>, <div class='tooltip'>Design<span class='tooltiptext'>Initialization function for \"Design\"<\/span><\/div>, <div class='tooltip'>dose<span class='tooltiptext'>Compute the doses for a given probability, given model and samples<\/span><\/div>, <div class='tooltip'>DualDesign<span class='tooltiptext'>Initialization function for \"DualDesign\"<\/span><\/div>, <div class='tooltip'>DualEndpoint<span class='tooltiptext'>Initialization function for the \"DualEndpoint\" class<\/span><\/div>, <div class='tooltip'>DualEndpointBeta<span class='tooltiptext'>Initialization function for the \"DualEndpointBeta\" class<\/span><\/div>, <div class='tooltip'>DualEndpointEmax<span class='tooltiptext'>Initialization function for the \"DualEndpointEmax\" class<\/span><\/div>, <div class='tooltip'>DualEndpointRW<span class='tooltiptext'>Initialization function for the \"DualEndpointRW\" class<\/span><\/div>, <div class='tooltip'>DualResponsesDesign<span class='tooltiptext'>Initialization function for 'DualResponsesDesign\"<\/span><\/div>, <div class='tooltip'>DualResponsesSamplesDesign<span class='tooltiptext'>Initialization function for 'DualResponsesSamplesDesign\"<\/span><\/div>, <div class='tooltip'>DualSimulations<span class='tooltiptext'>Initialization function for \"DualSimulations\"<\/span><\/div>, <div class='tooltip'>EffFlexi<span class='tooltiptext'>Initialization function for the \"EffFlexi\" class<\/span><\/div>, <div class='tooltip'>Effloglog<span class='tooltiptext'>Initialization function for the \"Effloglog\" class<\/span><\/div>, <div class='tooltip'>examine<span class='tooltiptext'>This generic function takes a design and generates a dataframeshowing the beginning of several hypothetical trial courses underthe design. This means, from the generated dataframe one can read off:- how many cohorts are required in the optimal case (no DLTs observed) in order to reach the highest dose of the specified dose grid (or until the stopping rule is fulfilled)- assuming no DLTs are observed until a certain dose level, what the next recommended dose is for all possible number of DLTs observed- the actual relative increments that will be used in these cases- whether the trial would stop at a certain cohortExamining the \"single trial\" behavior of a dose escalation design isthe first important step in evaluating a design, and cannot be replaced bystudying solely the operating characteristics in \"many trials\". The cohortsizes are also taken from the design, assuming no DLTs occur until the doselisted.<\/span><\/div>, <div class='tooltip'>ExpEff<span class='tooltiptext'>Compute the expected efficacy based on a given dose, a given pseudo Efficacy log-log model and a given efficacy sample<\/span><\/div>, <div class='tooltip'>fit<span class='tooltiptext'>Note this new generic function is necessary because the fittedfunction only allows the first argument object to appear in thesignature. But we need also other arguments in the signature.<\/span><\/div>, <div class='tooltip'>fitGain<span class='tooltiptext'>Get the fiited values for the gain values at all dose levels based on a given pseudo DLE model, DLE sample, a pseudo efficacy model, a Efficacy sample and data. This method returns a data frame with dose, middle, lower and upper quantiles of the gain value samples<\/span><\/div>, <div class='tooltip'>gain<span class='tooltiptext'>Compute the gain value with a given dose level, given a pseudo DLE model, a DLE sample, a pseudo Efficacy log-log model and a Efficacy sample<\/span><\/div>, <div class='tooltip'>GeneralSimulations<span class='tooltiptext'>Initialization function for \"GeneralSimulations\"<\/span><\/div>, <div class='tooltip'>get<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>getEff<span class='tooltiptext'>Extracting efficacy responses for subjects without or with a DLE. This is a class where we separateefficacy responses with or without a DLE. It outputs the efficacy responses and their corresponding dose levels treated at in two categories (with or without DLE)<\/span><\/div>, <div class='tooltip'>getMinInfBeta<span class='tooltiptext'>As defined in Neuenschwander et al (2008), this function computes theparameters of the minimal informative unimodal beta distribution, given therequest that the p-quantile should be q, i.e. X ~ Be(a, b) with Pr(X <= q) =p.<\/span><\/div>, <div class='tooltip'>IncrementMin<span class='tooltiptext'>Initialization function for \"IncrementMin\"<\/span><\/div>, <div class='tooltip'>IncrementsNumDoseLevels<span class='tooltiptext'>Initialization function for \"IncrementsNumDoseLevels\"<\/span><\/div>, <div class='tooltip'>IncrementsRelative<span class='tooltiptext'>Initialization function for \"IncrementsRelative\"<\/span><\/div>, <div class='tooltip'>IncrementsRelativeDLT<span class='tooltiptext'>Initialization function for \"IncrementsRelativeDLT\"<\/span><\/div>, <div class='tooltip'>IncrementsRelativeParts<span class='tooltiptext'>Initialization function for \"IncrementsRelativeParts\"<\/span><\/div>, <div class='tooltip'>initialize<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>LogisticIndepBeta<span class='tooltiptext'>Intialization function for \"LogisticIndepBeta\" class<\/span><\/div>, <div class='tooltip'>LogisticKadane<span class='tooltiptext'>Initialization function for the \"LogisticKadane\" class<\/span><\/div>, <div class='tooltip'>LogisticLogNormal<span class='tooltiptext'>Initialization function for the \"LogisticLogNormal\" class<\/span><\/div>, <div class='tooltip'>LogisticLogNormalMixture<span class='tooltiptext'>Initialization function for the \"LogisticLogNormalMixture\" class<\/span><\/div>, <div class='tooltip'>LogisticLogNormalSub<span class='tooltiptext'>Initialization function for the \"LogisticLogNormalSub\" class<\/span><\/div>, <div class='tooltip'>LogisticNormal<span class='tooltiptext'>Initialization function for the \"LogisticNormal\" class<\/span><\/div>, <div class='tooltip'>LogisticNormalFixedMixture<span class='tooltiptext'>Initialization function for the \"LogisticNormalFixedMixture\" class<\/span><\/div>, <div class='tooltip'>LogisticNormalMixture<span class='tooltiptext'>Initialization function for the \"LogisticNormalMixture\" class<\/span><\/div>, <div class='tooltip'>logit<span class='tooltiptext'>Shorthand for logit function<\/span><\/div>, <div class='tooltip'>matchTolerance<span class='tooltiptext'>This is a modified version of match that supports tolerance.<\/span><\/div>, <div class='tooltip'>maxDose<span class='tooltiptext'>Determine the upper limit of the next dose based on the increments rule.<\/span><\/div>, <div class='tooltip'>maxSize<span class='tooltiptext'>This function combines cohort size rules by takingthe maximum of all sizes.<\/span><\/div>, <div class='tooltip'>mcmc<span class='tooltiptext'>This is the function to actually run the MCMC machinery to produce posteriorsamples from all model parameters and required derived values. It is ageneric function, so that customized versions may be conveniently definedfor specific subclasses of GeneralData, GeneralModel, and McmcOptions input.<\/span><\/div>, <div class='tooltip'>McmcOptions<span class='tooltiptext'>Initialization function for the \"McmcOptions\" class<\/span><\/div>, <div class='tooltip'>MinimalInformative<span class='tooltiptext'>This function constructs a minimally informative prior, which is captured ina LogisticNormal (orLogisticLogNormal) object.<\/span><\/div>, <div class='tooltip'>minSize<span class='tooltiptext'>This function combines cohort size rules by takingthe minimum of all sizes.<\/span><\/div>, <div class='tooltip'>multiplot<span class='tooltiptext'>ggplot objects can be passed in ..., or to plotlist (as a list of ggplot objects).If the layout is something like matrix(c(1,2,3,3), nrow=2, byrow=TRUE),then plot 1 will go in the upper left, 2 will go in the upper right, and3 will go all the way across the bottom.<\/span><\/div>, <div class='tooltip'>nextBest<span class='tooltiptext'>Compute the recommended next best dose.<\/span><\/div>, <div class='tooltip'>NextBestDualEndpoint<span class='tooltiptext'>Initialization function for \"NextBestDualEndpoint\"<\/span><\/div>, <div class='tooltip'>NextBestMaxGain<span class='tooltiptext'>Initialization function for the class 'NextBestMaxGain'<\/span><\/div>, <div class='tooltip'>NextBestMaxGainSamples<span class='tooltiptext'>Initialization function for class \"NextBestMaxGainSamples\"<\/span><\/div>, <div class='tooltip'>NextBestMTD<span class='tooltiptext'>Initialization function for class \"NextBestMTD\"<\/span><\/div>, <div class='tooltip'>NextBestNCRM<span class='tooltiptext'>Initialization function for \"NextBestNCRM\"<\/span><\/div>, <div class='tooltip'>NextBestTD<span class='tooltiptext'>Initialization function for the class \"NextBestTD\"<\/span><\/div>, <div class='tooltip'>NextBestTDsamples<span class='tooltiptext'>Initialization function for class \"NextBestTDsamples\"<\/span><\/div>, <div class='tooltip'>NextBestThreePlusThree<span class='tooltiptext'>Initialization function for \"NextBestThreePlusThree\"<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>plotDualResponses<span class='tooltiptext'>Plot of the DLE and efficacy curve side by side given a DLE pseudo model,a DLE sample, an efficacy pseudo model and a given efficacy samplePlot of the dose-DLE and dose-efficacy curve side by side given a DLE pseudo model and a given pseudo efficacy model without DLE and efficacy samples<\/span><\/div>, <div class='tooltip'>plotGain<span class='tooltiptext'>Plot the gain curve in addition with the dose-DLE and dose-efficacy curve using a given DLE pseudo model,a DLE sample, a given efficacy pseudo model and an efficacy samplePlot the gain curve in addition with the dose-DLE and dose-efficacy curve using a given DLE pseudo model,and a given efficacy pseudo model<\/span><\/div>, <div class='tooltip'>prob<span class='tooltiptext'>Compute the probability for a given dose, given model and samples<\/span><\/div>, <div class='tooltip'>probit<span class='tooltiptext'>Shorthand for probit function<\/span><\/div>, <div class='tooltip'>ProbitLogNormal<span class='tooltiptext'>Initialization function for the \"ProbitLogNormal\" class<\/span><\/div>, <div class='tooltip'>PseudoSimulations<span class='tooltiptext'>Initialization function of the 'PseudoSimulations' class<\/span><\/div>, <div class='tooltip'>Quantiles2LogisticNormal<span class='tooltiptext'>This function uses generalised simulated annealing to optimisea LogisticNormal model to be as close as possibleto the given prior quantiles.<\/span><\/div>, <div class='tooltip'>RuleDesign<span class='tooltiptext'>Initialization function for \"RuleDesign\"<\/span><\/div>, <div class='tooltip'>Samples<span class='tooltiptext'>Initialization function for \"Samples\"<\/span><\/div>, <div class='tooltip'>sampleSize<span class='tooltiptext'>Compute the number of samples for a given MCMC options triple<\/span><\/div>, <div class='tooltip'>saveSample<span class='tooltiptext'>Determine if we should save this sample<\/span><\/div>, <div class='tooltip'>setSeed<span class='tooltiptext'>This is basically copied from simulate.lm<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>simulate<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>Simulations<span class='tooltiptext'>Initialization function for the \"Simulations\" class<\/span><\/div>, <div class='tooltip'>size<span class='tooltiptext'>This function determines the size of the next cohort.<\/span><\/div>, <div class='tooltip'>StoppingAll<span class='tooltiptext'>Initialization function for \"StoppingAll\"<\/span><\/div>, <div class='tooltip'>StoppingAny<span class='tooltiptext'>Initialization function for \"StoppingAny\"<\/span><\/div>, <div class='tooltip'>StoppingCohortsNearDose<span class='tooltiptext'>Initialization function for \"StoppingCohortsNearDose\"<\/span><\/div>, <div class='tooltip'>StoppingGstarCIRatio<span class='tooltiptext'>Initialization function for \"StoppingGstarCIRatio\"<\/span><\/div>, <div class='tooltip'>StoppingHighestDose<span class='tooltiptext'>Initialization function for \"StoppingHighestDose\"<\/span><\/div>, <div class='tooltip'>StoppingList<span class='tooltiptext'>Initialization function for \"StoppingList\"<\/span><\/div>, <div class='tooltip'>StoppingMinCohorts<span class='tooltiptext'>Initialization function for \"StoppingMinCohorts\"<\/span><\/div>, <div class='tooltip'>StoppingMinPatients<span class='tooltiptext'>Initialization function for \"StoppingMinPatients\"<\/span><\/div>, <div class='tooltip'>StoppingMTDdistribution<span class='tooltiptext'>Initialization function for \"StoppingMTDdistribution\"<\/span><\/div>, <div class='tooltip'>StoppingPatientsNearDose<span class='tooltiptext'>Initialization function for \"StoppingPatientsNearDose\"<\/span><\/div>, <div class='tooltip'>StoppingTargetBiomarker<span class='tooltiptext'>Initialization function for \"StoppingTargetBiomarker\"<\/span><\/div>, <div class='tooltip'>StoppingTargetProb<span class='tooltiptext'>Initialization function for \"StoppingTargetProb\"<\/span><\/div>, <div class='tooltip'>StoppingTDCIRatio<span class='tooltiptext'>Initialization function for \"StoppingTDCIRatio\"<\/span><\/div>, <div class='tooltip'>stopTrial<span class='tooltiptext'>This function returns whether to stop the trial.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>TDDesign<span class='tooltiptext'>Initialization function for 'TDDesign' class<\/span><\/div>, <div class='tooltip'>TDsamplesDesign<span class='tooltiptext'>Initialization function for 'TDsamplesDesign' class<\/span><\/div>, <div class='tooltip'>ThreePlusThreeDesign<span class='tooltiptext'>Creates a new 3+3 design object from a dose grid<\/span><\/div>, <div class='tooltip'>update<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>writeModel<span class='tooltiptext'>Convert R function to a WinBUGS model file. BUGS models followclosely S syntax. It is therefore possible to write most BUGS models as Rfunctions.As a difference, BUGS syntax allows truncation specification like this:dnorm(...) I(...) but this is illegal in R. To overcome thisincompatibility, use dummy operator %_% before I(...):dnorm(...) %_% I(...). The dummy operator %_% will beremoved before the BUGS code is saved.In S-PLUS, a warning is generated when the model function is defined if thelast statement in the model is an assignment. To avoid this warning, add theline invisible() to the end of the model definition. This line will beremoved before the BUGS code is saved.<\/span><\/div>","<div class='tooltip'>cite_design<span class='tooltiptext'>Obtain the preferred citation for a design<\/span><\/div>, <div class='tooltip'>compare_design_code<span class='tooltiptext'>Compare two designs<\/span><\/div>, <div class='tooltip'>compare_design_data<span class='tooltiptext'>Compare two designs<\/span><\/div>, <div class='tooltip'>compare_design_estimates<span class='tooltiptext'>Compare two designs<\/span><\/div>, <div class='tooltip'>compare_design_inquiries<span class='tooltiptext'>Compare two designs<\/span><\/div>, <div class='tooltip'>compare_design_summaries<span class='tooltiptext'>Compare two designs<\/span><\/div>, <div class='tooltip'>compare_designs<span class='tooltiptext'>Compare two designs<\/span><\/div>, <div class='tooltip'>compare_diagnoses<span class='tooltiptext'>Diagnose and compare designs.<\/span><\/div>, <div class='tooltip'>declare_assignment<span class='tooltiptext'>Declare Data Strategy: Assignment<\/span><\/div>, <div class='tooltip'>declare_diagnosands<span class='tooltiptext'>Declare diagnosands<\/span><\/div>, <div class='tooltip'>declare_estimand<span class='tooltiptext'>Declares inquiries, or the inferential target of interest. Conceptually very close to \"estimand\" or \"quantity of interest\".<\/span><\/div>, <div class='tooltip'>declare_estimands<span class='tooltiptext'>Declares inquiries, or the inferential target of interest. Conceptually very close to \"estimand\" or \"quantity of interest\".<\/span><\/div>, <div class='tooltip'>declare_estimator<span class='tooltiptext'>Declares an estimator which generates estimates and associated statistics.Use of declare_test is identical to use of declare_estimator. Use declare_test for hypothesis testing with no specific inquiry in mind; use declare_estimator for hypothesis testing when you can link each estimate to an inquiry. For example, declare_test could be used for a K-S test of distributional equality and declare_estimator for a difference-in-means estimate of an average treatment effect.<\/span><\/div>, <div class='tooltip'>declare_estimators<span class='tooltiptext'>Declares an estimator which generates estimates and associated statistics.Use of declare_test is identical to use of declare_estimator. Use declare_test for hypothesis testing with no specific inquiry in mind; use declare_estimator for hypothesis testing when you can link each estimate to an inquiry. For example, declare_test could be used for a K-S test of distributional equality and declare_estimator for a difference-in-means estimate of an average treatment effect.<\/span><\/div>, <div class='tooltip'>declare_inquiries<span class='tooltiptext'>Declares inquiries, or the inferential target of interest. Conceptually very close to \"estimand\" or \"quantity of interest\".<\/span><\/div>, <div class='tooltip'>declare_inquiry<span class='tooltiptext'>Declares inquiries, or the inferential target of interest. Conceptually very close to \"estimand\" or \"quantity of interest\".<\/span><\/div>, <div class='tooltip'>declare_measurement<span class='tooltiptext'>This function adds measured data columns that can be functions of unmeasured data columns.<\/span><\/div>, <div class='tooltip'>declare_model<span class='tooltiptext'>Declare the size and features of the population<\/span><\/div>, <div class='tooltip'>declare_population<span class='tooltiptext'>Declare the size and features of the population<\/span><\/div>, <div class='tooltip'>declare_potential_outcomes<span class='tooltiptext'>Declare potential outcomes<\/span><\/div>, <div class='tooltip'>declare_reveal<span class='tooltiptext'>Potential outcomes declarations indicate what outcomes would obtain for different possible values of assignment variables. But realized outcomes need to be \"revealed.\" declare_reveal generates these realized outcomes using information on potential outcomes (for instance generated via declare_potential_outcomes) and the relevant assignment variables (for example created by declare_assignment). Revelation steps are usefully included after declaration of all assignments of conditions required to determine the realized outcome.If a revelation is not declared, DeclareDesign will try to guess appropriate revelations. Explicit revelation is recommended however.<\/span><\/div>, <div class='tooltip'>declare_sampling<span class='tooltiptext'>Declare sampling procedure<\/span><\/div>, <div class='tooltip'>declare_step<span class='tooltiptext'>With declare_step, you can include any function that takes data as one of its arguments and returns data in a design declaration. The first argument is always a \"handler\", which is the name of the data-in, data-out function.For handy data manipulations use declare_step(fabricate, ...).<\/span><\/div>, <div class='tooltip'>declare_test<span class='tooltiptext'>Declares an test which generates a test statistic and associated inferential statistics. Use of declare_test is identical to use of declare_estimator. Use declare_test for hypothesis testing with no specific inquiry in mind; use declare_estimator for hypothesis testing when you can link each estimate to an inquiry. For example, declare_test could be used for a K-S test of distributional equality and declare_estimator for a difference-in-means estimate of an average treatment effect.See declare_estimator help for an explanation of how to use model_handler, which is used identically in both declare_estimator and declare_test. The main difference between declare_estimator and declare_test is that declare_test does not link with an explicit inquiry.<\/span><\/div>, <div class='tooltip'>delete_step<span class='tooltiptext'>Insert, delete and replace steps in an (already declared) design object.<\/span><\/div>, <div class='tooltip'>diagnose_design<span class='tooltiptext'>Generates diagnosands from a design or simulations of a design.<\/span><\/div>, <div class='tooltip'>diagnose_designs<span class='tooltiptext'>Generates diagnosands from a design or simulations of a design.<\/span><\/div>, <div class='tooltip'>draw_assignment<span class='tooltiptext'>Get estimates, inquiries, assignment vectors, or samples from a design given data<\/span><\/div>, <div class='tooltip'>draw_data<span class='tooltiptext'>Draw data, estimates, and inquiries from a design<\/span><\/div>, <div class='tooltip'>draw_estimand<span class='tooltiptext'>Draw data, estimates, and inquiries from a design<\/span><\/div>, <div class='tooltip'>draw_estimands<span class='tooltiptext'>Draw data, estimates, and inquiries from a design<\/span><\/div>, <div class='tooltip'>draw_estimates<span class='tooltiptext'>Draw data, estimates, and inquiries from a design<\/span><\/div>, <div class='tooltip'>draw_inquiries<span class='tooltiptext'>Draw data, estimates, and inquiries from a design<\/span><\/div>, <div class='tooltip'>draw_inquiry<span class='tooltiptext'>Draw data, estimates, and inquiries from a design<\/span><\/div>, <div class='tooltip'>draw_sample<span class='tooltiptext'>Get estimates, inquiries, assignment vectors, or samples from a design given data<\/span><\/div>, <div class='tooltip'>expand_design<span class='tooltiptext'>expand_design easily generates a set of design from a designer function.<\/span><\/div>, <div class='tooltip'>get_diagnosands<span class='tooltiptext'>Explore your design diagnosis<\/span><\/div>, <div class='tooltip'>get_estimates<span class='tooltiptext'>Get estimates, inquiries, assignment vectors, or samples from a design given data<\/span><\/div>, <div class='tooltip'>get_simulations<span class='tooltiptext'>Explore your design diagnosis<\/span><\/div>, <div class='tooltip'>insert_step<span class='tooltiptext'>Insert, delete and replace steps in an (already declared) design object.<\/span><\/div>, <div class='tooltip'>label_estimator<span class='tooltiptext'>Declares an estimator which generates estimates and associated statistics.Use of declare_test is identical to use of declare_estimator. Use declare_test for hypothesis testing with no specific inquiry in mind; use declare_estimator for hypothesis testing when you can link each estimate to an inquiry. For example, declare_test could be used for a K-S test of distributional equality and declare_estimator for a difference-in-means estimate of an average treatment effect.<\/span><\/div>, <div class='tooltip'>label_test<span class='tooltiptext'>Declares an test which generates a test statistic and associated inferential statistics. Use of declare_test is identical to use of declare_estimator. Use declare_test for hypothesis testing with no specific inquiry in mind; use declare_estimator for hypothesis testing when you can link each estimate to an inquiry. For example, declare_test could be used for a K-S test of distributional equality and declare_estimator for a difference-in-means estimate of an average treatment effect.See declare_estimator help for an explanation of how to use model_handler, which is used identically in both declare_estimator and declare_test. The main difference between declare_estimator and declare_test is that declare_test does not link with an explicit inquiry.<\/span><\/div>, <div class='tooltip'>print_code<span class='tooltiptext'>Explore your designPrint code to recreate a design<\/span><\/div>, <div class='tooltip'>redesign<span class='tooltiptext'>redesign quickly generates a design from an existing one by resetting symbols used in design handler parameters in a step's environment (Advanced).<\/span><\/div>, <div class='tooltip'>replace_step<span class='tooltiptext'>Insert, delete and replace steps in an (already declared) design object.<\/span><\/div>, <div class='tooltip'>reshape_diagnosis<span class='tooltiptext'>Take a diagnosis object and returns a pretty output table. If diagnosands are bootstrapped, se's are put in parentheses on a second line and rounded to digits.<\/span><\/div>, <div class='tooltip'>run_design<span class='tooltiptext'>Run a design one time<\/span><\/div>, <div class='tooltip'>set_citation<span class='tooltiptext'>Set the citation of a design<\/span><\/div>, <div class='tooltip'>set_diagnosands<span class='tooltiptext'>A researcher often has a set of diagnosands in mind to appropriately assess the quality of a design. set_diagnosands sets the default diagnosands for a design, so that later readers can assess the design on the same terms as the original author. Readers can also use diagnose_design to diagnose the design using any other set of diagnosands.<\/span><\/div>, <div class='tooltip'>simulate_design<span class='tooltiptext'>Runs many simulations of a design and returns a simulations data.frame.<\/span><\/div>, <div class='tooltip'>simulate_designs<span class='tooltiptext'>Runs many simulations of a design and returns a simulations data.frame.<\/span><\/div>, <div class='tooltip'>tidy<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericstidy<\/span><\/div>, <div class='tooltip'>tidy_estimator<span class='tooltiptext'>The functions listed below are deprecated and will be defunct in the near future. When possible, alternative functions with similar functionality are also mentioned. Help pages for deprecated functions are available at help(\"-deprecated\").<\/span><\/div>, <div class='tooltip'>tidy_try<span class='tooltiptext'>Tidy function that returns a tidy data.frame of model results and allows filtering to relevant coefficients. The function will attempt to tidy model objects even when they do not have a tidy method available. For best results, first load the broom package via library(broom).<\/span><\/div>, <div class='tooltip'>vars<span class='tooltiptext'>Generates diagnosands from a design or simulations of a design.<\/span><\/div>","<div class='tooltip'>ALC<span class='tooltiptext'>Acts on a \"gp\", \"dgp2\", or \"dgp3\" object. Calculates ALC over the input locations x_new using specified reference grid. If no reference grid is specified, x_new is used as the reference. Optionally utilizes SNOW parallelization. User should select the point with the highest ALC to add to the design.<\/span><\/div>, <div class='tooltip'>calc_K<span class='tooltiptext'>Calculates covariance matrix based on inverse exponentiated squared euclidean distance with specified hyperparameters.<\/span><\/div>, <div class='tooltip'>continue<span class='tooltiptext'>Acts on a \"gp\", \"dgp2\", or \"dgp3\" object. Continues MCMC sampling of hyperparameters and hidden layers and appends results to existing object.<\/span><\/div>, <div class='tooltip'>EI<span class='tooltiptext'>Acts on a \"gp\", \"dgp2\", or \"dgp3\" object. Calculates expected improvement over input locations x_new with the goal of MINIMIZING the function. Optionally utilizes SNOW parallelization. User should select the point with the highest EI to add to the design.<\/span><\/div>, <div class='tooltip'>fit_one_layer<span class='tooltiptext'>Conducts MCMC sampling of hyperparameters for a one layer GP. Covariance structure is based on inverse exponentiated squared euclidean distance with length scale parameter \"theta\" governing the strength of the correlation and nugget parameter \"g\" governing noise.<\/span><\/div>, <div class='tooltip'>fit_three_layer<span class='tooltiptext'>Conducts MCMC sampling of hyperparameters, hidden layer \"z\", and hidden layer \"w\" for a three layer deep GP. Covariance structure is based on inverse exponentiated squared euclidean distance. Separate length scale parameters \"theta_z\", \"theta_w\", and \"theta_y\" govern the correlation strength of the inner layer, middle layer, and outer layer respectively. Nugget parameter \"g\" governs noise on the outer layer.<\/span><\/div>, <div class='tooltip'>fit_two_layer<span class='tooltiptext'>Conducts MCMC sampling of hyperparameters and hidden layer \"w\" for a two layer deep GP. Covariance structure is based on inverse exponentiated squared euclidean distance. Separate length scale parameters \"theta_w\" and \"theta_y\" govern the correlation strength of the hidden layer and outer layer respectively. Nugget parameter \"g\" governs noise on the outer layer.<\/span><\/div>, <div class='tooltip'>IMSE<span class='tooltiptext'>Acts on a \"gp\", \"dgp2\", or \"dgp3\" object. Calculates IMSE over the input locations x_new. Optionally utilizes SNOW parallelization. User should select the point with the lowest IMSE to add to the design.<\/span><\/div>, <div class='tooltip'>rmse<span class='tooltiptext'>Calculates root mean square error (lower RMSE indicate better fits).<\/span><\/div>, <div class='tooltip'>score<span class='tooltiptext'>Calculates score (higher scores indicate better fits). Only applicable to noisy data. Requires full covariance matrix (e.g. predict with lite = FALSE).<\/span><\/div>, <div class='tooltip'>sq_dist<span class='tooltiptext'>Calculates squared pairwise euclidean distances using C.<\/span><\/div>, <div class='tooltip'>trim<span class='tooltiptext'>Acts on a \"gp\", \"dgp2\", or \"dgp3\" object. Removes the specified number of MCMC iterations (starting at the first iteration). After these samples are removed, the remaining samples may be thinned.<\/span><\/div>","<div class='tooltip'>binary_iv_designer<span class='tooltiptext'>Builds a design with one instrument, one binary explanatory variable, and one outcome.<\/span><\/div>, <div class='tooltip'>block_cluster_two_arm_designer<span class='tooltiptext'>Builds a two-arm design with blocks and clusters.<\/span><\/div>, <div class='tooltip'>cluster_sampling_designer<span class='tooltiptext'>Builds a cluster sampling design for an ordinal outcome variable for a population with N_blocks strata, each with N_clusters_in_block clusters, each of which contains N_i_in_cluster units. The sampling strategy involves sampling n_clusters_in_block clusters in each stratum, and then sampling n_i_in_cluster units in each cluster. Outcomes within clusters have intra-cluster correlation approximately equal to ICC.<\/span><\/div>, <div class='tooltip'>factorial_designer<span class='tooltiptext'>A 2^k factorial designer with k factors assigned with independent probabilities. Results in 2^k treatment combinations, each with independent, normally distributed shocks. Estimands are average effects and average interactions of given conditions, averaged over other conditions. Estimation uses regression of demeaned variables with propensity weights.<\/span><\/div>, <div class='tooltip'>get_design_code<span class='tooltiptext'>Get the code from a design<\/span><\/div>, <div class='tooltip'>match.call.defaults<span class='tooltiptext'>This is a version of match.call which also includes default arguments.<\/span><\/div>, <div class='tooltip'>mediation_analysis_designer<span class='tooltiptext'>A mediation analysis design that examines the effect of treatment (Z) on mediator (M) and the effect of mediator (M) on outcome (Y) (given Z=0) as well as direct effect of treatment (Z) on outcome (Y) (given M=0). Analysis is implemented using an interacted regression model. Note this model is not guaranteed to be unbiased despite randomization of Z because of possible violations of sequential ignorability.<\/span><\/div>, <div class='tooltip'>multi_arm_designer<span class='tooltiptext'>Creates a design with m_arms experimental arms, each assigned with equal probability.<\/span><\/div>, <div class='tooltip'>pretest_posttest_designer<span class='tooltiptext'>Produces a design in which an outcome Y is observed pre- and post-treatment.The design allows for individual post-treatment outcomes to be correlated with pre-treatment outcomesand for at-random missingness in the observation of post-treatment outcomes.<\/span><\/div>, <div class='tooltip'>process_tracing_designer<span class='tooltiptext'>Builds a design in which two pieces of evidence are sought and used to update about whether X caused Y using Bayes' rule.<\/span><\/div>, <div class='tooltip'>randomized_response_designer<span class='tooltiptext'>Produces a (forced) randomized response design that measures the share of individuals with a given trait prevalence_trait in a population of size N. Probability of forced response (\"Yes\") is given by prob_forced_yes, and rate at which individuals with trait lie is given by withholding_rate.<\/span><\/div>, <div class='tooltip'>regression_discontinuity_designer<span class='tooltiptext'>Builds a design with sample from population of size N. The average treatment effect local to the cutpoint is equal to tau. It allows for specification of the order of the polynomial regression (poly_reg_order), cutoff value on the running variable (cutoff), and size of bandwidth around the cutoff (bandwidth). By providing a vector of numbers to control_coefs and treatment_coefs, users can also specify polynomial regression coefficients that generate the expected control and treatment potential outcomes given the running variable.<\/span><\/div>, <div class='tooltip'>simple_factorial_designer<span class='tooltiptext'>Builds a two-by-two factorial design in which assignments to each factor are independent of each other.<\/span><\/div>, <div class='tooltip'>simple_iv_designer<span class='tooltiptext'>Builds a design with one instrument, one binary explanatory variable, and one outcome.<\/span><\/div>, <div class='tooltip'>simple_spillover_designer<span class='tooltiptext'>Builds a design with N_groups groups each containing N_i_group individuals. Potential outcomes exhibit spillovers: if any individual in a group receives treatment, the effect is spread equally among members of the group.<\/span><\/div>, <div class='tooltip'>simple_two_arm_designer<span class='tooltiptext'>Builds a design with one treatment and one control arm.Treatment effects can be specified either by providing control_mean and treatment_meanor by specifying a control_mean and ate.<\/span><\/div>, <div class='tooltip'>spillover_designer<span class='tooltiptext'>Builds a design with N_groups groups each containing N_i_group individuals. Potential outcomes exhibit spillovers: if any individual in a group receives treatment, the effect is spread equally among members of the group.<\/span><\/div>, <div class='tooltip'>two_arm_attrition_designer<span class='tooltiptext'>Creates a two-arm design with application for when estimand of interest is conditional on a post-treatment outcome (the effect on Y given R) or data is conditionally observed (Y given R). See `Details` for more information on the data generating process.<\/span><\/div>, <div class='tooltip'>two_arm_covariate_designer<span class='tooltiptext'>Builds a design with one treatment and one control arm.Treatment effects can be specified either by providing control_mean and treatment_meanor by specifying a control_mean and ate.Non random assignment is specified by a possible correlation, rho_WZ, between W and a latent variable that determines the probability of Z. Nonignorability is specified by a possible correlation, rho_WY, between W and outcome Y.<\/span><\/div>, <div class='tooltip'>two_arm_designer<span class='tooltiptext'>Builds a design with one treatment and one control arm.Treatment effects can be specified either by providing control_mean and treatment_meanor by specifying a control_mean and ate.<\/span><\/div>, <div class='tooltip'>two_by_two_designer<span class='tooltiptext'>Builds a two-by-two factorial design in which assignments to each factor are independent of each other.<\/span><\/div>","<div class='tooltip'>dArb<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dArb.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dBox<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dBox.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dCategorical<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dCategorical.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dMax<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dMax.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dMin<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dMin.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dOverall<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dOverall.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dTarget<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>, <div class='tooltip'>dTarget.default<span class='tooltiptext'>Functions implementing multivariate optimization and ranking using the desirability function approach described in Derringer and Suich (1980)<\/span><\/div>","<div class='tooltip'>crossValidation<span class='tooltiptext'>This function calculates the predicted values at each point of the design and gives an estimation of criterion using K-fold cross-validation.<\/span><\/div>, <div class='tooltip'>formulaAm<span class='tooltiptext'>This function constructs a formula based on splines for additive models.<\/span><\/div>, <div class='tooltip'>formulaLm<span class='tooltiptext'>This function constructs a formula containing only the principal factors.<\/span><\/div>, <div class='tooltip'>MAE<span class='tooltiptext'>The mean of absolute errors between real values and predictions.<\/span><\/div>, <div class='tooltip'>modelComparison<span class='tooltiptext'>modelComparison fits different metamodels and returns R2 and RMSE criteria relating to each.<\/span><\/div>, <div class='tooltip'>modelFit<span class='tooltiptext'>modelFit is used to fit a metamodel of class lm, gam, mars, polymars or km.<\/span><\/div>, <div class='tooltip'>modelPredict<span class='tooltiptext'>modelPredict computes predicted values based on the model given in argument.<\/span><\/div>, <div class='tooltip'>penaltyPolyMARS<span class='tooltiptext'>This function fits a PolyMARS model for different values of the penalty parameterand compute criteria.<\/span><\/div>, <div class='tooltip'>R2<span class='tooltiptext'>Coefficient of determination R^{2}R2<\/span><\/div>, <div class='tooltip'>residualsStudy<span class='tooltiptext'>residualsStudy analyzes the residuals of a model: a plot of the residuals against the index, a plotof the residuals against the fitted values, the representation of thedensity and a normal Q-Q plot.<\/span><\/div>, <div class='tooltip'>RMA<span class='tooltiptext'>Relative Maximal Absolute Error<\/span><\/div>, <div class='tooltip'>RMSE<span class='tooltiptext'>The root of the Mean Squared Error between the exact value and the predicted one.<\/span><\/div>, <div class='tooltip'>stepEvolution<span class='tooltiptext'>Graphical representation of the selected terms using stepwiseprocedure for different values of the penalty parameter.<\/span><\/div>, <div class='tooltip'>testCrossValidation<span class='tooltiptext'>This function calculates the estimated K-fold cross-validation for differentvalues of K.<\/span><\/div>","<div class='tooltip'>addalmostexact<span class='tooltiptext'>A near-exact match (also known as an almost exact match) for a nominal covariate maximizes the numberof pairs that are exactly matched for the nominal covariate. This is done by adding a large penaltyto the covariate distance between individuals with different values of the nominal covariate.The topic is discussed in Section 9.2 of Design of Observational Studies (2010).<\/span><\/div>, <div class='tooltip'>addcaliper<span class='tooltiptext'>A caliper forces a close match on a continuous covariate, often the propensity score (Rosenbaum and Rubin 1985). Alternatively or additionally, a caliper may be imposed on a prognostic score estimated from some other independent data set (Hansen 2008). Rosenbaum and Rubin (1985) suggested minimizing the Mahalanobis distance within calipers defined by the propensity score. If the caliper is implemented using a penalty function, the caliper cannot create infeasibility. Implementation of a caliper using a penalty function is discussed in Chapter 8 of Design of Observational Studies (2010).<\/span><\/div>, <div class='tooltip'>fine<span class='tooltiptext'>In optimal pair matching with fine balance, expand a distance matrix to become a square matrix to enforce fine balance. The method is discussed in Chapter 10 of Design of Observational Studies (2010), and it is conceptually the simplest way to implement fine balance; therefore, it remains very useful for teaching and for self-study. See details. For practical work, consider also the rcbalance package, the designmatch package and the bigmatch package; see the references.<\/span><\/div>, <div class='tooltip'>mahal<span class='tooltiptext'>Computes a Mahalanobis distance matrix between treated individuals and potential controls. The method is discussed in Chapter 8 of Design of Observational Studies (2010).<\/span><\/div>, <div class='tooltip'>senU<span class='tooltiptext'>Sensitivity analysis for the new U statistic of Rosenbaum (2011). For m=m1=m2, this is the test of Stephenson (1981). The ranks proposed by Stephenson closely approximate the optimal ranks proposed by Conover and Salzburg (1988) for detecting a treatment that has a large effect on a small subpopulation and no effect on most of the population; see Rosenbaum (2007). The example reproduces some results from Chapter 16 of Design of Observational Studies (2010).<\/span><\/div>, <div class='tooltip'>senWilcox<span class='tooltiptext'>Sensitivity analysis for Wilcoxon's signed rank statistic in observational studies.Performs a sensitivity analysis for the P-value, the Hodges-Lehmann estimate ofan additive treatment effect, and the confidence interval for the effect.<\/span><\/div>, <div class='tooltip'>senWilcoxExact<span class='tooltiptext'>Exact sensitivity analysis for Wilcoxon's signed rank statistic in observational studies.Performs a sensitivity analysis for the one-sided P-value. The method can be used insmall samples without ties; however, it is primarily of theoretical interest, as thelarge sample approximation in senWilcox() is fine for most samples of practical size.<\/span><\/div>, <div class='tooltip'>smahal<span class='tooltiptext'>Computes a robust Mahalanobis distance matrix between treated individuals and potential controls. The usual Mahalnaobis distance may ignore a variable because it contains one extreme outlier, and it pays toomuch attention to rare binary variables, but smahal addresses both issues.<\/span><\/div>","<div class='tooltip'>addalmostexact<span class='tooltiptext'>A near-exact match (also known as an almost exact match) for a nominal covariate maximizes the numberof pairs that are exactly matched for the nominal covariate. This is done by adding a large penaltyto the covariate distance between individuals with different values of the nominal covariate.The topic is discussed in Section 10.2 of the second edition of \"Design of Observational Studies\".<\/span><\/div>, <div class='tooltip'>addcaliper<span class='tooltiptext'>A caliper forces a close match on a continuous covariate, often the propensity score (Rosenbaum and Rubin 1985). Alternatively or additionally, a caliper may be imposed on a prognostic score estimated from some other independent data set (Hansen 2008). Rosenbaum and Rubin (1985) suggested minimizing the Mahalanobis distance within calipers defined by the propensity score. If the caliper is implemented using a penalty function, the caliper cannot create infeasibility. Implementation of a caliper using a penalty function is discussed in Chapter 9 of \"Design of Observational Studies\", second edition.<\/span><\/div>, <div class='tooltip'>cohere<span class='tooltiptext'>For multiple outcomes in an observation study, computesa weighted combination of signed rank statistics, one for each outcome,and performs either a one-sided randomization test or an analysisof sensitivity to departures from random assignment; see Rosenbaum (1997).Each matched set contains one treated individual and one control.The signed rank statistics can be either Wilcoxon's signed rankstatistic or the new U-statistic in Rosenbaum (2011).The Scheffe method is described in Rosenbaum (2016). For one outcome,use the functions 'senWilcox' or 'senU', instead of 'cohere'.The method is discussed in Sections 5.2.3, 18.2 and 18.3 of\"Design of Observational Studies\", second edition.<\/span><\/div>, <div class='tooltip'>crosscut<span class='tooltiptext'>Computes the cross-cut test and its sensitivity analysis. The cross-cut test isa nonparametric test of dose-response correlation with good design sensitivitywhen used for causal inference in observational studies.<\/span><\/div>, <div class='tooltip'>crosscutplot<span class='tooltiptext'>A scatterplot of y against x, with the points used by thecross-cut test in black,and the remaining points in grey. See Figure 19.2 in\"Design of Observational Studies\", second edition.<\/span><\/div>, <div class='tooltip'>fine<span class='tooltiptext'>In optimal pair matching with fine balance, expand a distance matrix to become a square matrix to enforce fine balance. The method is discussed in Chapter 11 of \"Design of Observational Studies\", second edition, and it is conceptually the simplest way to implement fine balance; therefore, it remains very useful for teaching and for self-study. See details.<\/span><\/div>, <div class='tooltip'>mahal<span class='tooltiptext'>Computes a Mahalanobis distance matrix between treated individuals and potential controls; see Rubin (1980) and Rosenbaum and Rubin (1985). The method is discussed in Section 9.3 of \"Design of Observational Studies\", second edition.<\/span><\/div>, <div class='tooltip'>planScheffe<span class='tooltiptext'>The function planScheffe() computes the critical values for a levelalpha test that combines one planned linear combination ofa K-dimensional multivariate Normal outcome and consideration ofall possible combinations correcting for multiple testing usinga Scheffe projection. The function is the same as the planScheffe()function in the sensitivitymult package, but the examples aredifferent. The method is discussed in Section 18.3 of \"Design ofObservational Studies\", second edition, and in Rosenbaum (2019).<\/span><\/div>, <div class='tooltip'>senU<span class='tooltiptext'>Sensitivity analysis for the new U statistic of Rosenbaum (2011). For m=m1=m2, this is the test of Stephenson (1981). The ranks proposed by Stephenson closely approximate the optimal ranks proposed by Conover and Salzburg (1988) for detecting a treatment that has a large effect on a small subpopulation and no effect on most of the population; see Rosenbaum (2007). The example reproduces some results from Chapter 17 of \"Design of Observational Studies\", second edition, and the method is discusssed in Section 19.2.<\/span><\/div>, <div class='tooltip'>senWilcox<span class='tooltiptext'>Sensitivity analysis for Wilcoxon's signed rank statistic in observational studies.Performs a sensitivity analysis for the P-value, the Hodges-Lehmann estimate ofan additive treatment effect, and the confidence interval for the effect.The example is from Section 3.5 of \"Design of Observational Studies\", 2nd ed.,and it also illustrates the amplification in Section 3.6 using the'amplify' function from the 'sensitivitymult' package.<\/span><\/div>, <div class='tooltip'>senWilcoxExact<span class='tooltiptext'>Exact sensitivity analysis for Wilcoxon's signed rank statistic in observational studies. Performs a sensitivity analysis for the one-sided P-value. The method can be used insmall samples without ties; however, it is primarily of theoretical interest, as the large sample approximation in 'senWilcox' is fine for most samples of practical size.<\/span><\/div>, <div class='tooltip'>smahal<span class='tooltiptext'>Computes a robust Mahalanobis distance matrix between treated individuals and potential controls. The usual Mahalnaobis distance may ignore a variable because it contains one extreme outlier, and it pays toomuch attention to rare binary variables, but smahal addresses both issues.See Section 9.3 of \"Design of Observational Studies\", second edition.<\/span><\/div>","<div class='tooltip'>generate.transects<span class='tooltiptext'>Uses the survey design details in the design class to generatea set of transects, i.e. a single survey.<\/span><\/div>, <div class='tooltip'>get.coverage<span class='tooltiptext'>Obtains the coverage scores from the survey design object.<\/span><\/div>, <div class='tooltip'>make.coverage<span class='tooltiptext'>This creates an instance of the Coverage.Grid class.<\/span><\/div>, <div class='tooltip'>make.design<span class='tooltiptext'>Creates a description of a survey design. Designs may use differenttypes of either point or line transect designs across strata but cannot mixpoint and line transect design types within a single design object.<\/span><\/div>, <div class='tooltip'>make.region<span class='tooltiptext'>This creates an instance of the Region class which defines the studyarea for the survey.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>run.coverage<span class='tooltiptext'>This function can be used to assess the coverage ofa design and also assess design statistics, such as how the numberof samplers, the line length, trackline length or percentagecoverage varies between surveys generated from the same design.It generates the specified number of surveys from thedesign and looks to see which of the coverage grid points, asystematic grid of points across the survey region, are includedin each survey. When calculating coverage scores if more thanone sampler falls on a grid point then that grid point getsallocated the appropriate count. These counts are then averagedover the number of surveys which have been generated. At the sametime it records the relevant statistics for the design. While 100repetitions may be sufficient to get an idea of design statistics1000 or even more repetitions may be needed to gain a goodrepresentation of the coverage scores across the study region.<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>write.transects<span class='tooltiptext'>This function will write a set of transects to file, either asa shapefile or gpx file, or it will write the transect coordinates (centrepoints for point transects or end points for line transects) to acomma-separated values 'csv' file or a text file 'txt' with tabular spacingbetween columns. For line transects which have been split across geographicalfeatures (such as islands or lakes) there will be two or more rows in thecsv / txt file with all rows having the same transect ID.<\/span><\/div>","<div class='tooltip'>acc.1test<span class='tooltiptext'>Sensitivity and specificity, (positive and negative) predictive values and (positive and negative) diagnostic likelihood ratios of a single binary diagnostic test.<\/span><\/div>, <div class='tooltip'>acc.paired<span class='tooltiptext'>Sensitivity and specificity, (positive and negative) predictive values and (positive and negative) diagnostic likelihood ratios of a two binary diagnostic tests in a paired study design.<\/span><\/div>, <div class='tooltip'>DLR<span class='tooltiptext'>This function allows for estimating the log diagnostic likelihood ratio in a regression model approach. It can be used to assess the gain in diagnostic accuracy for a new binary or continuous diagnostic marker comparedto established markers, to determine the impact of covariates on the risk prediction model, and to estimate theDLR for selected marker/covariate values.<\/span><\/div>, <div class='tooltip'>dlr.regtest<span class='tooltiptext'>Performs a test for differences in (positive and negative) diagnostic likelihood ratios (DLRs) of two binary diagnostic tests in a paired study using a regression model approach proposed by Gu and Pepe (2009).<\/span><\/div>, <div class='tooltip'>generate.paired<span class='tooltiptext'>Generates a dataset from contingency tables of binary diagnostic test results in a paired study design.<\/span><\/div>, <div class='tooltip'>print.acc.1test<span class='tooltiptext'>Prints objects of class acc.1test in an easy-to-read form.<\/span><\/div>, <div class='tooltip'>print.acc.paired<span class='tooltiptext'>Prints objects of class acc.paired in an easy-to-read form.<\/span><\/div>, <div class='tooltip'>print.tab.1test<span class='tooltiptext'>Prints objects of class tab.1test in an easy-to-read form.<\/span><\/div>, <div class='tooltip'>print.tab.paired<span class='tooltiptext'>Prints objects of class tab.paired in an easy-to-read form.<\/span><\/div>, <div class='tooltip'>pv.gs<span class='tooltiptext'>Performs a test for differences in (positive and negative) predictive values of two binary diagnostic tests using a generalized score statistic proposed by Leisenring, Alonzo and Pepe (2000).<\/span><\/div>, <div class='tooltip'>pv.rpv<span class='tooltiptext'>Performs a test for differences in (positive and negative) predictive values of two binary diagnostic tests in a paired study design using relative predictive values, as proposed by Moskowitz and Pepe (2006).<\/span><\/div>, <div class='tooltip'>pv.wgs<span class='tooltiptext'>Performs a test for differences in (positive and negative) predictive values of two binary diagnostic tests using a weighted generalized score statistic proposed by Kosinski (2013).<\/span><\/div>, <div class='tooltip'>read.tab.1test<span class='tooltiptext'>Reads in objects of class tab.1test using cell frequencies.<\/span><\/div>, <div class='tooltip'>read.tab.paired<span class='tooltiptext'>Reads in objects of class tab.paired using cell frequencies.<\/span><\/div>, <div class='tooltip'>represent.long<span class='tooltiptext'>Long representation of results from two binary diagnostic tests.<\/span><\/div>, <div class='tooltip'>sesp.diff.ci<span class='tooltiptext'>Calculates confidence intervals for differences in sensitivity and specificity of two binary diagnostic tests in a paired study design.<\/span><\/div>, <div class='tooltip'>sesp.exactbinom<span class='tooltiptext'>Performs an exact binomial test for differences in sensitivity and specificity of two binary diagnostic tests in a paired study design.<\/span><\/div>, <div class='tooltip'>sesp.mcnemar<span class='tooltiptext'>Performs a McNemar Test for comparison of sensitivities and specificities of two binary diagnostic tests in a paired study design.<\/span><\/div>, <div class='tooltip'>tab.1test<span class='tooltiptext'>Produces a contingency table of results from a single binary diagnostic test vs. the gold-standard results.<\/span><\/div>, <div class='tooltip'>tab.paired<span class='tooltiptext'>Produces contingency tables of results from two binary diagnostic tests evaluated in a paired study design.<\/span><\/div>","<div class='tooltip'>alc<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>alc.dynaTree<span class='tooltiptext'>Uses analytic integration (at the leaves) to calculate the (regression) ALC statistic, or calculates the predictive (class) entropy at the input (X) locations; or calculate ALC at new predictive locations either analytically or numerically<\/span><\/div>, <div class='tooltip'>alcalc<span class='tooltiptext'>Internal dynaTree functions<\/span><\/div>, <div class='tooltip'>alcX<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>alcX.dynaTree<span class='tooltiptext'>Uses analytic integration (at the leaves) to calculate the (regression) ALC statistic, or calculates the predictive (class) entropy at the input (X) locations; or calculate ALC at new predictive locations either analytically or numerically<\/span><\/div>, <div class='tooltip'>copy<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>copy.dynaTree<span class='tooltiptext'>Function(s) to free the memory used to represent particle clouds stored on the C-side of a \"dynaTree\"-class object -- essential to avoid memory leaks; also a function to copy an entire \"dynaTree\"-class object, which is required since explicit code is needed to copy the C-side memory<\/span><\/div>, <div class='tooltip'>deletecloud<span class='tooltiptext'>Function(s) to free the memory used to represent particle clouds stored on the C-side of a \"dynaTree\"-class object -- essential to avoid memory leaks; also a function to copy an entire \"dynaTree\"-class object, which is required since explicit code is needed to copy the C-side memory<\/span><\/div>, <div class='tooltip'>deleteclouds<span class='tooltiptext'>Function(s) to free the memory used to represent particle clouds stored on the C-side of a \"dynaTree\"-class object -- essential to avoid memory leaks; also a function to copy an entire \"dynaTree\"-class object, which is required since explicit code is needed to copy the C-side memory<\/span><\/div>, <div class='tooltip'>dynaTree<span class='tooltiptext'>A function to initialize and fit dynamic tree modelsto regression and classification data by the sequentialMonte Carlo (SMC) method of particle learning (PL)<\/span><\/div>, <div class='tooltip'>dynaTrees<span class='tooltiptext'>A function to initialize and fit dynamic tree modelsto regression and classification data by the sequentialMonte Carlo (SMC) method of particle learning (PL)<\/span><\/div>, <div class='tooltip'>getBF<span class='tooltiptext'>Extract a path (log) Bayes factors (BFs) from the log marginal posterior probabilities of two \"dynaTree\"-class objects<\/span><\/div>, <div class='tooltip'>plotprogress<span class='tooltiptext'>Internal dynaTree functions<\/span><\/div>, <div class='tooltip'>qEI<span class='tooltiptext'>Internal dynaTree functions<\/span><\/div>, <div class='tooltip'>qEI.dynaTree<span class='tooltiptext'>Internal dynaTree functions<\/span><\/div>, <div class='tooltip'>qEntropy<span class='tooltiptext'>Internal dynaTree functions<\/span><\/div>, <div class='tooltip'>qEntropy.dynaTree<span class='tooltiptext'>Internal dynaTree functions<\/span><\/div>, <div class='tooltip'>rejuvenate<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>rejuvenate.dynaTree<span class='tooltiptext'>Re-pass the X-y pairs in the object in a random (or specified) order to temporarily double the size of the particle set<\/span><\/div>, <div class='tooltip'>relevance<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>relevance.dynaTree<span class='tooltiptext'>Computes relevance statistics for each input coordinate by calculating their particle-averaged mean reduction in variance each time that coordinate is used as a splitting variable in (an internal node of) the tree(s)<\/span><\/div>, <div class='tooltip'>retire<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>retire.dynaTree<span class='tooltiptext'>Allows the removal (or retireing of X-y pairs from a \"dynaTree\"-class object to facilitate online learning; retireed pairs ar absorbed into the leaf prior(s)<\/span><\/div>, <div class='tooltip'>sens<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>sens.dynaTree<span class='tooltiptext'>A Monte Carlo sensitivity analysis using random Latin hypercube samples (LHSs) or bootstrap resamples for each particle to estimate main effects as well as 1st order and total sensitivity indices<\/span><\/div>, <div class='tooltip'>treestats<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>treestats.dynaTree<span class='tooltiptext'>Calculates the proportion of particles which use each input to make a tree split and the proportion of all splits in trees of each particle that correspond to each input variable; also provides tree height and leaf size summary information<\/span><\/div>, <div class='tooltip'>varproptotal<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>varproptotal.dynaTree<span class='tooltiptext'>Calculates the proportion of particles which use each input to make a tree split and the proportion of all splits in trees of each particle that correspond to each input variable; also provides tree height and leaf size summary information<\/span><\/div>, <div class='tooltip'>varpropuse<span class='tooltiptext'>A stub for class dynaTree and its custom generic methods<\/span><\/div>, <div class='tooltip'>varpropuse.dynaTree<span class='tooltiptext'>Calculates the proportion of particles which use each input to make a tree split and the proportion of all splits in trees of each particle that correspond to each input variable; also provides tree height and leaf size summary information<\/span><\/div>","<div class='tooltip'>echCusum<span class='tooltiptext'>Calculate the optimum parameters of n(sample size), h(sampling interval), k(reference value) and H(decision interval) for Economic Design of the CUSUM control chart. For more information about the reference value see 'Details'.<\/span><\/div>, <div class='tooltip'>echEwma<span class='tooltiptext'>Calculate the optimum parameters, n(sample size), h(sampling interval), w(weight to the present sample) and k(number of s.d. from control limits to center line) for economic Design of the EWMA control chart .<\/span><\/div>, <div class='tooltip'>echXbar<span class='tooltiptext'>Calculate the optimum parameters, n(sample size), h(sampling interval) and L(number of s.d. from control limits to center line) for Economic Design of the X-bar control chart .<\/span><\/div>, <div class='tooltip'>ecoCusum<span class='tooltiptext'>Calculate the optimum parameters of n(sample size), h(sampling interval), k(reference value) and H(decision interval) for Economic Design of the CUSUM control chart. For more information about the reference value see 'Details'.<\/span><\/div>, <div class='tooltip'>ecoEwma<span class='tooltiptext'>Calculate the optimum parameters, n(sample size), h(sampling interval), w(weight to the present sample) and k(number of s.d. from control limits to center line) for economic Design of the EWMA control chart .<\/span><\/div>, <div class='tooltip'>ecoXbar<span class='tooltiptext'>Calculate the optimum parameters, n(sample size), h(sampling interval) and L(number of s.d. from control limits to center line) for Economic Design of the X-bar control chart .<\/span><\/div>","<div class='tooltip'>addnoise<span class='tooltiptext'>This function is used within power.trend.Distribution of noise can be either Normal, Poisson or Negative Binomial. The mean values are enteredas a parameter (possibly generated by function generate.trend). Other parameters (sd for Normal,nbsizefor Negative Binomial) need to be given. Values in meanvalues are used as the mean values for one of the threespecified distributions and then a random allocation is made for each of the nobs values, where nobs is thelength of meanvalues.<\/span><\/div>, <div class='tooltip'>detect<span class='tooltiptext'>The function can calculate the probability of detection of a circular patch of specified radius for a specified density of points; the density needed to achieve a specified probability of detection; or theradius of the patch that will be detected with specified probability and sampling density.This is done forrandom, square lattice, and triangular lattice spatial sampling designs.<\/span><\/div>, <div class='tooltip'>detect.prop<span class='tooltiptext'>The function can calculate the probability of a feature that occupiesa proportion theta of the sampling area and where the sampling point density of the survey isspecified; the sampling point density needed to achieve a specified probability of detection, wheretheta is also specified ; or the value of theta that will be detected with specifiedprobability and sampling density.Unless the feature is made of a large number of random segments(see below for how to deal with this situation), these methods apply only when the pattern of pointsin the sampling deisgn is random.<\/span><\/div>, <div class='tooltip'>expected.nb<span class='tooltiptext'>The function is used to obtain the method of moments estimator within the function GVFCMOM.Calculates the expected value of the Visual Fast Count method.The function assumes that the count per segment is Negative Binomial with mean m/s and size k,and that segment counts are independent. The expected value is also a function of the number of positivesd before segment counting is stopped.<\/span><\/div>, <div class='tooltip'>expected.pois<span class='tooltiptext'>%% ~~ A concise (1-5 lines) description of what the function does. ~~The function is used to obtain the method of moments estimator within the function GVFCMOM.Calculates the expected value of the Visual Fast Count method.The function assumes that the count per segment is Poissonwith mean m/s and that segment counts are independent. The expected value is also a function of the number ofpositives d before segment counting is stopped.<\/span><\/div>, <div class='tooltip'>fS.detect<span class='tooltiptext'>Used in the function detect for calculating the sample size for a square lattice design.It is used by the optimize function.<\/span><\/div>, <div class='tooltip'>fT.detect<span class='tooltiptext'>Used in the function detect for calculating the sample size for a triangular lattice design.It is used by the optimize function.<\/span><\/div>, <div class='tooltip'>generate.trend<span class='tooltiptext'>This function is used to generate mean value scenarios for use in power.trend.<\/span><\/div>, <div class='tooltip'>GVFC<span class='tooltiptext'>The function considers the counts per segment and uses them sequentially until d positive countsare obtained or until all s segments have been considered. If we assume that u counts are used (ofwhich some may be zero) thenthe visual fast count estimator is the mean over the u counts multiplied by s. This function is usedby GVFCMOM to obtain the method of moments VFC estimator - which has reduced bias compared to theraw VFC estimator.<\/span><\/div>, <div class='tooltip'>GVFCMOM<span class='tooltiptext'>The function takes data in the form of counts per segment along a transect and uses the raw Generalised Visual Fast Count estimator (as calculated by GVFCMOM and itsexpectation (as calculated by expected.pois for Poisson or expected.nb fornegative binomial) to calculatea method of moments estimator. This effectively,adjusts the biased raw GVFC estimate. The function allows counts to have either a Poisson or a Negative Binomialdistribution. The method is a generalisation of the methods in Barry and Coggan (2010).<\/span><\/div>, <div class='tooltip'>is.wholenumber<span class='tooltiptext'>Used in error checking to ascertain whether a function argument is an integer.<\/span><\/div>, <div class='tooltip'>mannkendall<span class='tooltiptext'>Calculates the Mann-Kendall statistic for monotonic trend and also the p-value against the null hypothesis of no trend.Unlike the function MannKendall, works for repeat values of time.<\/span><\/div>, <div class='tooltip'>mannkendall.stat<span class='tooltiptext'>Calculates the Mann-Kendall statistic for monotonic trend. Unlike the function MannKendall, worksfor repeat values of time. Used in function mannkendall, which also calculates the p-value by simulation.<\/span><\/div>, <div class='tooltip'>mom.min.nb<span class='tooltiptext'>Used by the optimize function to obtain the method of moments estimator within the functionGVFCMOM.<\/span><\/div>, <div class='tooltip'>mom.min.pois<span class='tooltiptext'>Used by the optimize function to obtain the method of moments estimator within the functionGVFCMOM.<\/span><\/div>, <div class='tooltip'>n.min<span class='tooltiptext'>Used by the optimize function to obtain the correct sample size within the functionprecision given that the t-distributions is also a function of sample size.<\/span><\/div>, <div class='tooltip'>permute.BACI<span class='tooltiptext'>We have control and treatment data from time 1 in a BACI design, plus control and treatmentdata from time 2. The interaction the amount that the difference in the control andtreatment meansis different between times 1 and 2.<\/span><\/div>, <div class='tooltip'>permute.groups<span class='tooltiptext'>Does randomisation test for the difference in means mu1, mu2of two vectors v1 and v2. Can do one or two sided tests.<\/span><\/div>, <div class='tooltip'>power.BACI<span class='tooltiptext'>BACI designs are commonly used in environmental monitoring. They are relevant where youwant to measure the effect of an impact (e.g. the effect on benthic ecology of dredgingin an area). Observations for treatment and control areas are measured BEFORE andafter the impact. This function allows you to examine the power of particular BACI designsto detect differences between the control and the treatment.<\/span><\/div>, <div class='tooltip'>power.groups<span class='tooltiptext'>Calculates the power by simulation for comparing the mean of two groups of independent observations.<\/span><\/div>, <div class='tooltip'>power.trend<span class='tooltiptext'>Calculates power for a specified trend wherethe signal for the trend is specified by xvalues and meanvalues (possiblygenerated by generate.trend), and the error distribution is specified by distribution. The statistical method to detectthe trendis specified by method.The power is the proportion of repeat simulations for which the trend is detected with a p-valueless than alpha (two-sided test).<\/span><\/div>, <div class='tooltip'>precision<span class='tooltiptext'>Precision is measured by the width of a 100(1-alpha)% confidence interval for a single population mean.The function generates the sample size needed to achieve this or the precision achieved fora specified sample size.<\/span><\/div>, <div class='tooltip'>size2.samevar<span class='tooltiptext'>Calculates the Negative Binomial size parameter s2 such that the variance of the distributionwith mean mu2 and size s2 is the same as the Negative Binomial distribution with meanmu1 and size s1. This can be useful when computing power for a Negative Binomialdistribution in the packages power.groups and power.BACI.<\/span><\/div>, <div class='tooltip'>svariog<span class='tooltiptext'>Calculates empirical semi-variogram cloud plus classical, robust and median estimators from bins.<\/span><\/div>","<div class='tooltip'>AED.boundary<span class='tooltiptext'>AED.boundary() is used to calculate the critical value used at the final analysis in APE design, meanwhile preserving the overall type I error rate at \\alpha level<\/span><\/div>, <div class='tooltip'>AED.sim<span class='tooltiptext'>The AED.sim() is used to conduct the simulation studies of the Adaptive Enrichment Design without early stopping boundary. The AED design is quite similar with the AED1_SSR design. But, in the AED design, the futility stopping boundary and the Sample Size Re-estimation Procedure are removed. On the contrary, a fixed sample size is used to replace the sample size re-estimated procedure. In addition, an \\epsilon-rule is also introduced to select the subgroup with larger subgroup-specific test statistic.<\/span><\/div>, <div class='tooltip'>AED1_SSR.boundary<span class='tooltiptext'>The AED1_SSR.boundary() is used to calculate the critical value required at the final analysis of the Adaptive Enrichment Design (Strategy 1) with sample size re-estimation procedure. In the AED1-SSR design, the adaptive enrichment strategy is guided by a pre-specified futility stopping boundary and a threshold of the difference between the subgroup-specific test statistics.<\/span><\/div>, <div class='tooltip'>AED1_SSR.CP<span class='tooltiptext'>The AED1_SSR.CP() is used to calculate the conditional power of the Adaptive Enrichment Design (Strategy 1) with sample size re-estimation procedure<\/span><\/div>, <div class='tooltip'>AED1_SSR.N2<span class='tooltiptext'>The AED1_SSR.N2() is used to calculated the sample size required at the second stage of the Adaptive Enrichment Design (Strategy 1) with Sample Size Re-estimation Procedure.<\/span><\/div>, <div class='tooltip'>AED1_SSR.sim<span class='tooltiptext'>The AED1_SSR.sim() is used to conduct the simulation study of the Adaptive Enrichment Design (Strategy 1) with Sample Size Re-estimation procedure<\/span><\/div>, <div class='tooltip'>AED2_SSR.boundary<span class='tooltiptext'>The AED2_SSR.boundary() is used to calculate the futility and efficacy stopping boundaries of the Adaptive Enrichment Design (strategy 2) with Sample Size Re-estimation Procedure. In the AED2-SSR design, an \\epsilon-rule is introduced to select the subgroup with larger test statistic. In practice, the value of \\epsilon should be calibrated to fit the requirement of the trial.<\/span><\/div>, <div class='tooltip'>AED2_SSR.CP<span class='tooltiptext'>The AED2_SSR.CP() is used to calculate the sample size required at the second stage and the critical value used at the final analysis in the Adaptive Enrichment Design with Sample Size Re-estimation Procedure. In addition, this function can also used to conduct the conditional power analysis in terms of N2<\/span><\/div>, <div class='tooltip'>AED2_SSR.sim<span class='tooltiptext'>The AED2_SSR.sim() is used to conduct the simulation studies of the Adaptive Enrichment Design (Strategy) with sample size re-estimation procedure. The AED2-SSR is different from the AED3-SSR, in which an \\epsilon-rule is introduced to select the subgroup with larger subgroup-specific test statistic.<\/span><\/div>, <div class='tooltip'>AED3_SSR.boundary<span class='tooltiptext'>The AED3_SSR.boundary() is used to calculate the futility and efficacy stopping boundaries in the Adaptive Enrichment Design with Sample Size Re-estimation Procedure.<\/span><\/div>, <div class='tooltip'>AED3_SSR.CP<span class='tooltiptext'>The AED3_SSR.CP() is used to calculate the sample size required at the second stage and the critical value used at the final analysis in the Adaptive Enrichment Design with Sample Size Re-estimation Procedure. In addition, this function can also used to conduct the conditional power analysis in terms of N2<\/span><\/div>, <div class='tooltip'>AED3_SSR.sim<span class='tooltiptext'>The AED3_SSR.sim() is used to conduct the adaptive enrichment design with Sample Size Re-estimation, in which futility and efficacy stopping boundaries are used to guide the adaptive enrichment process. For the adaptively enriched subgroup, we re-estimate the sample size to maintain an adequate conditional power meanwhile protect the overall Type I error rate.<\/span><\/div>, <div class='tooltip'>MaST.sim<span class='tooltiptext'>The MaST.sim() is used to conduct the simulation studies of the marker sequential test design (MaST).<\/span><\/div>, <div class='tooltip'>SD.sim<span class='tooltiptext'>The SD.sim() is used to implement the simulation studies of the standard design.<\/span><\/div>, <div class='tooltip'>SigP<span class='tooltiptext'>The SigP() is used to calculate the reduced significant level based on several widely used \\alpha-spending functions, such as the \"Pocock\", \"Lan-DeMets\", \"O`Brein-Fleming\" and \"Power\" functions.<\/span><\/div>, <div class='tooltip'>sSize.norm<span class='tooltiptext'>The sSize.norm() is used to calculate the sample size used in the standard design with continuous endpoint.<\/span><\/div>, <div class='tooltip'>SSR.boundary<span class='tooltiptext'>The SSD.boundary() is used to calculate the futility and efficacy stopping boundaries, meanwhile protect the overall Type I error rate at the pre-specified level.<\/span><\/div>, <div class='tooltip'>SSR.CP<span class='tooltiptext'>The SSR.CP() is used to calculate the sample size required at the second stage and the critical value used at the final analysis. In addition, this function can also used to conduct the conditional power analysis in terms of N2<\/span><\/div>, <div class='tooltip'>SSR.sim<span class='tooltiptext'>The SSR.sim() is used to implement the simulation studies based on the Sample Size Re-estimation Procedure.<\/span><\/div>","<div class='tooltip'>cor2x2<span class='tooltiptext'>Computes the hazard ratios, confidence intervals, p-values, and correlationsfor the overall A, simple A, and simple AB logrank statistics.<\/span><\/div>, <div class='tooltip'>crit2x2<span class='tooltiptext'>Computes the critical valuesfor null hypotheses rejection and corresponding nominal two-sided significancelevels for the Equal Allocation 3, Proportional Allocation 2,and Equal Allocation 2 procedures<\/span><\/div>, <div class='tooltip'>eventProb<span class='tooltiptext'>Calculates the event probabilities for each of the four factorial groupsC, A, B, AB. The time unit is in years, but of course, any timeunit could be used. Average event probabilities across variouscombinations of the groups are also calculated.The event times are assumed to be exponentially distributed. Thecensoring times are assumed to be uniformly distributed and indepedent ofthe event times.<\/span><\/div>, <div class='tooltip'>fac2x2analyze<span class='tooltiptext'>Performs significance testing for the Proportional Allocation 2, Equal Allocation 3,Equal Allocation 2 procedures.Also reports the hazard ratios, 95% confidence intervals, p-values,nominal significance levels, and correlations for the overall and simpletest statistics.<\/span><\/div>, <div class='tooltip'>fac2x2design<span class='tooltiptext'>Power for the Equal Allocation 3, Proportional Allocation 2, andEqual Allocation 2 procedures.<\/span><\/div>, <div class='tooltip'>lgrkPower<span class='tooltiptext'>Computes the power for the unstratified (ordinary) logrank statisticfor two group comparison.<\/span><\/div>, <div class='tooltip'>powerEA2<span class='tooltiptext'>Computes the Equal Allocation 2's procedure power todetect the simple A effect and the simple AB effect, respectively.<\/span><\/div>, <div class='tooltip'>powerEA3<span class='tooltiptext'>Computes the Equal Allocation 3 procedure's power todetect the overall A effect, the simple A effect, or the simple AB effect, respectively.<\/span><\/div>, <div class='tooltip'>powerPA2<span class='tooltiptext'>Computes the Proportional Allocation 2 procedure's power todetect the overall A effect or the simple AB effect, respectively.<\/span><\/div>, <div class='tooltip'>roundDown<span class='tooltiptext'>Rounds a negative number to the nearest, more negativenumber to a specified decimal place.<\/span><\/div>, <div class='tooltip'>strLgrkPower<span class='tooltiptext'>Computes the power for the overall treatment A effect based on the stratified logrank test based on Slud (1994).<\/span><\/div>","<div class='tooltip'>NewFis<span class='tooltiptext'>Function to create object of class Fis<\/span><\/div>, <div class='tooltip'>NewFisIn<span class='tooltiptext'>Function to create object of class FisIn<\/span><\/div>, <div class='tooltip'>NewFisOutCrisp<span class='tooltiptext'>Function to create object of class FisOutCrisp<\/span><\/div>, <div class='tooltip'>NewFisOutFuzzy<span class='tooltiptext'>Function to create object of class FisOutFuzzy<\/span><\/div>, <div class='tooltip'>NewMfTrapezoidal<span class='tooltiptext'>Function to create object of class MfTrapezoidal<\/span><\/div>, <div class='tooltip'>NewMfTrapezoidalInf<span class='tooltiptext'>Function to create object of class MfTrapezoidalInf<\/span><\/div>, <div class='tooltip'>NewMfTrapezoidalSup<span class='tooltiptext'>Function to create object of class MfTrapezoidalSup<\/span><\/div>, <div class='tooltip'>NewMfTriangular<span class='tooltiptext'>Function to create object of class MfTriangular<\/span><\/div>, <div class='tooltip'>NewRule<span class='tooltiptext'>Function to create object of class Rule<\/span><\/div>","<div class='tooltip'>as.flowdef<span class='tooltiptext'>This function enables creation of a skeleton flow definition with several default values, using a flowmat.To customize the flowdef, one may supply parameters such as sub_type and dep_type upfront.As such, these params must be of the same length as number of unique jobs using in the flowmat.Each row in this table refers to one step of the pipeline. It describes the resources used by the step and also its relationship with other steps, especially, the step immediately prior to it. <br><br>Submission types: This refers to the sub_type column in flow definition.<br>Consider an example with three steps A, B and C. A has 10 commands from A1 to A10, similarly B has 10 commands B1 through B10 and C has a single command, C1.Consider another step D (with D1-D3), which comes after C.step (number of sub-processes)A (10) ----> B (10) -----> C (1) -----> D (3) scatter: submit all commands as parallel, independent jobs. Submit A1 through A10 as independent jobs serial: run these commands sequentially one after the other. - Wrap A1 through A10, into a single job.Dependency typesThis refers to the dep_type column in flow definition. none: independent job. Initial step A has no dependency serial: one to one relationship with previous job. B1 can start as soon as A1 completes, and B2 starts just after A2 and so on. gather: many to one, wait for all commands in the previous job to finish then start the current step. All jobs of B (1-10), need to complete before C1 starts burst: one to many wait for the previous step which has one job and start processing all cmds in the current step. - D1 to D3 are started as soon as C1 finishes.<\/span><\/div>, <div class='tooltip'>as.flowmat<span class='tooltiptext'>Create a flowmat (data.frame) using a named list a commands.as.flowmat(): reads a file and checks for required columns. If x is data.frame checks for required columns.<\/span><\/div>, <div class='tooltip'>check<span class='tooltiptext'>Check consistency of flowdef and flowmat, using various rules.<\/span><\/div>, <div class='tooltip'>check_args<span class='tooltiptext'>Checks all the arguments in the parent function and makes sure that none of themare NULL<\/span><\/div>, <div class='tooltip'>fetch<span class='tooltiptext'>These functions help in searching for specific files in the user's space.fetch_pipes(): Fetches pipelines in the following places, in this specific order: user's folder: ~/flowr/pipelines current wd: ./NOTE: If same pipeline is available in multiple places; intuitively, one from the laterfolder would be selected. As such, giving priority to user's home, and current working directories.<br>fetch_conf(): Fetches configuration files in ALL of the following places: package: conf folders in flowr and ngsflows packages. user's folder: ~/flowr/conf folder. current wd: ./NOTE:This function would greedily return all matching conf files. One would load all of them in the order returned by this function. If the same variable isrepeated in multiple files, value from later files would replace those formerly defined.Thus ( as explained above ), giving priority to options defined in user's home and current working directories.By default flowr loads, flowr.conf and ngsflows.conf. See the details sections, for more explanation on this.<\/span><\/div>, <div class='tooltip'>fetch_conf<span class='tooltiptext'>These functions help in searching for specific files in the user's space.fetch_pipes(): Fetches pipelines in the following places, in this specific order: user's folder: ~/flowr/pipelines current wd: ./NOTE: If same pipeline is available in multiple places; intuitively, one from the laterfolder would be selected. As such, giving priority to user's home, and current working directories.<br>fetch_conf(): Fetches configuration files in ALL of the following places: package: conf folders in flowr and ngsflows packages. user's folder: ~/flowr/conf folder. current wd: ./NOTE:This function would greedily return all matching conf files. One would load all of them in the order returned by this function. If the same variable isrepeated in multiple files, value from later files would replace those formerly defined.Thus ( as explained above ), giving priority to options defined in user's home and current working directories.By default flowr loads, flowr.conf and ngsflows.conf. See the details sections, for more explanation on this.<\/span><\/div>, <div class='tooltip'>fetch_pipes<span class='tooltiptext'>These functions help in searching for specific files in the user's space.fetch_pipes(): Fetches pipelines in the following places, in this specific order: user's folder: ~/flowr/pipelines current wd: ./NOTE: If same pipeline is available in multiple places; intuitively, one from the laterfolder would be selected. As such, giving priority to user's home, and current working directories.<br>fetch_conf(): Fetches configuration files in ALL of the following places: package: conf folders in flowr and ngsflows packages. user's folder: ~/flowr/conf folder. current wd: ./NOTE:This function would greedily return all matching conf files. One would load all of them in the order returned by this function. If the same variable isrepeated in multiple files, value from later files would replace those formerly defined.Thus ( as explained above ), giving priority to options defined in user's home and current working directories.By default flowr loads, flowr.conf and ngsflows.conf. See the details sections, for more explanation on this.<\/span><\/div>, <div class='tooltip'>flow<span class='tooltiptext'>Internal function (used by to_flow), which aids in creating a flow object.<\/span><\/div>, <div class='tooltip'>get_opts<span class='tooltiptext'>There are three helper functions which attempt to manage parameters used by flowr and ngsflows: get_opts OR opts_flow\\$get(): show all default options set_opts OR opts_flow\\$set(): set default options load_opts OR opts_flow\\$load(): load options specified in a tab separated text fileFor more details regarding these functions refer to http://sahilseth.com/params/params package.<\/span><\/div>, <div class='tooltip'>get_resources<span class='tooltiptext'>Extract resources used by each job of a flowget_resources currently this only works on LSF<\/span><\/div>, <div class='tooltip'>get_resources_lsf<span class='tooltiptext'>get_resources_lsf<\/span><\/div>, <div class='tooltip'>get_status<span class='tooltiptext'>Summarize status of a flow OR multiple flows OR a high-level summary of all flows in a folder.<\/span><\/div>, <div class='tooltip'>get_unique_id<span class='tooltiptext'>get_unique_id<\/span><\/div>, <div class='tooltip'>get_wds<span class='tooltiptext'>Get all the (sub)directories in a folder<\/span><\/div>, <div class='tooltip'>is.flow<span class='tooltiptext'>Use a set of shell commands (flow mat) and flow definition to create flow object.<\/span><\/div>, <div class='tooltip'>is.flowdef<span class='tooltiptext'>This function enables creation of a skeleton flow definition with several default values, using a flowmat.To customize the flowdef, one may supply parameters such as sub_type and dep_type upfront.As such, these params must be of the same length as number of unique jobs using in the flowmat.Each row in this table refers to one step of the pipeline. It describes the resources used by the step and also its relationship with other steps, especially, the step immediately prior to it. <br><br>Submission types: This refers to the sub_type column in flow definition.<br>Consider an example with three steps A, B and C. A has 10 commands from A1 to A10, similarly B has 10 commands B1 through B10 and C has a single command, C1.Consider another step D (with D1-D3), which comes after C.step (number of sub-processes)A (10) ----> B (10) -----> C (1) -----> D (3) scatter: submit all commands as parallel, independent jobs. Submit A1 through A10 as independent jobs serial: run these commands sequentially one after the other. - Wrap A1 through A10, into a single job.Dependency typesThis refers to the dep_type column in flow definition. none: independent job. Initial step A has no dependency serial: one to one relationship with previous job. B1 can start as soon as A1 completes, and B2 starts just after A2 and so on. gather: many to one, wait for all commands in the previous job to finish then start the current step. All jobs of B (1-10), need to complete before C1 starts burst: one to many wait for the previous step which has one job and start processing all cmds in the current step. - D1 to D3 are started as soon as C1 finishes.<\/span><\/div>, <div class='tooltip'>is.flowmat<span class='tooltiptext'>Create a flowmat (data.frame) using a named list a commands.as.flowmat(): reads a file and checks for required columns. If x is data.frame checks for required columns.<\/span><\/div>, <div class='tooltip'>job<span class='tooltiptext'>Internal function (used by to_flow), which aids in creating a job object.<\/span><\/div>, <div class='tooltip'>kill<span class='tooltiptext'>NOTE:This requires files which are created at the end of the submit_flow command.Even if you want to kill the flow, its best to let submit_flow do its job, when done simply use kill(flow_wd). If submit_flow is interrupted, files like flow_details.rds etc are not created, thus flowr looses the association of jobs with flow instance and cannot monitor, kill or re-run the flow.<\/span><\/div>, <div class='tooltip'>load_opts<span class='tooltiptext'>There are three helper functions which attempt to manage parameters used by flowr and ngsflows: get_opts OR opts_flow\\$get(): show all default options set_opts OR opts_flow\\$set(): set default options load_opts OR opts_flow\\$load(): load options specified in a tab separated text fileFor more details regarding these functions refer to http://sahilseth.com/params/params package.<\/span><\/div>, <div class='tooltip'>plot_flow<span class='tooltiptext'>Plot a flowchart using a flow object or flowdef<\/span><\/div>, <div class='tooltip'>queue<span class='tooltiptext'>Internal function (used by to_flow), to define the format used to submit a job.<\/span><\/div>, <div class='tooltip'>rerun<span class='tooltiptext'>hardware no change required, simple rerun: rerun(x=flow_wd) software either a change to flowmat or flowdef has been made: rerun(x=flow_wd, mat = new_flowmat, def = new_flowdef)NOTE:flow_wd: flow working directory, same input as used for status<\/span><\/div>, <div class='tooltip'>run<span class='tooltiptext'>Run complete pipelines, by wrapping several steps into one convenient function.NOTE: please use flowr version 0.9.8.9010 or higher.In summary, this function performs the following steps: the argument x defines the name of the pipeline. Say, for example sleep_pipe. fetch_pipes: finds the pipeline definition (sleep_pipe.R, sleep_pipe.def and sleep_pipe.conf files) sleep_pipe(): Create all the required commands (flowmat) to_flow: Use flowmat and sleep_pipe.def to create a flow object. submit_flow: Submit the flow to the cluster.<\/span><\/div>, <div class='tooltip'>run_pipe<span class='tooltiptext'>Run complete pipelines, by wrapping several steps into one convenient function.NOTE: please use flowr version 0.9.8.9010 or higher.In summary, this function performs the following steps: the argument x defines the name of the pipeline. Say, for example sleep_pipe. fetch_pipes: finds the pipeline definition (sleep_pipe.R, sleep_pipe.def and sleep_pipe.conf files) sleep_pipe(): Create all the required commands (flowmat) to_flow: Use flowmat and sleep_pipe.def to create a flow object. submit_flow: Submit the flow to the cluster.<\/span><\/div>, <div class='tooltip'>run_pipe_v2<span class='tooltiptext'>Run complete pipelines, by wrapping several steps into one convenient function.NOTE: please use flowr version 0.9.8.9010 or higher.In summary, this function performs the following steps: the argument x defines the name of the pipeline. Say, for example sleep_pipe. fetch_pipes: finds the pipeline definition (sleep_pipe.R, sleep_pipe.def and sleep_pipe.conf files) sleep_pipe(): Create all the required commands (flowmat) to_flow: Use flowmat and sleep_pipe.def to create a flow object. submit_flow: Submit the flow to the cluster.<\/span><\/div>, <div class='tooltip'>set_opts<span class='tooltiptext'>There are three helper functions which attempt to manage parameters used by flowr and ngsflows: get_opts OR opts_flow\\$get(): show all default options set_opts OR opts_flow\\$set(): set default options load_opts OR opts_flow\\$load(): load options specified in a tab separated text fileFor more details regarding these functions refer to http://sahilseth.com/params/params package.<\/span><\/div>, <div class='tooltip'>setup<span class='tooltiptext'>This functions creates a directory structure in user's home directory. Additionally it creates a shortcut to the flowr helper script in ~/bin.<\/span><\/div>, <div class='tooltip'>status<span class='tooltiptext'>Summarize status of a flow OR multiple flows OR a high-level summary of all flows in a folder.<\/span><\/div>, <div class='tooltip'>submit_flow<span class='tooltiptext'>Submit a flow to the cluster or perform a dry-run to check and debug issues.<\/span><\/div>, <div class='tooltip'>submit_run<span class='tooltiptext'>Submit several flow objects, limit the max running concurrently<\/span><\/div>, <div class='tooltip'>test_queue<span class='tooltiptext'>This function attempts to test the submission of a job to the queue.We would first submit one single job, then submit another with a dependency to see if configuration works. This would create a folder in home called 'flows'.[Depreciated]: This function has been superseded by run(\"sleep_pipe\", platform = \"lsf\", execute=TRUE)<\/span><\/div>, <div class='tooltip'>to_df.status<span class='tooltiptext'>convert the status to a df. To be called from R<\/span><\/div>, <div class='tooltip'>to_flow<span class='tooltiptext'>Use a set of shell commands (flow mat) and flow definition to create flow object.<\/span><\/div>, <div class='tooltip'>to_flowdef<span class='tooltiptext'>This function enables creation of a skeleton flow definition with several default values, using a flowmat.To customize the flowdef, one may supply parameters such as sub_type and dep_type upfront.As such, these params must be of the same length as number of unique jobs using in the flowmat.Each row in this table refers to one step of the pipeline. It describes the resources used by the step and also its relationship with other steps, especially, the step immediately prior to it. <br><br>Submission types: This refers to the sub_type column in flow definition.<br>Consider an example with three steps A, B and C. A has 10 commands from A1 to A10, similarly B has 10 commands B1 through B10 and C has a single command, C1.Consider another step D (with D1-D3), which comes after C.step (number of sub-processes)A (10) ----> B (10) -----> C (1) -----> D (3) scatter: submit all commands as parallel, independent jobs. Submit A1 through A10 as independent jobs serial: run these commands sequentially one after the other. - Wrap A1 through A10, into a single job.Dependency typesThis refers to the dep_type column in flow definition. none: independent job. Initial step A has no dependency serial: one to one relationship with previous job. B1 can start as soon as A1 completes, and B2 starts just after A2 and so on. gather: many to one, wait for all commands in the previous job to finish then start the current step. All jobs of B (1-10), need to complete before C1 starts burst: one to many wait for the previous step which has one job and start processing all cmds in the current step. - D1 to D3 are started as soon as C1 finishes.<\/span><\/div>, <div class='tooltip'>to_flowdet<span class='tooltiptext'>Create a file describing details regarding jobs ids, submission scripts etc.<\/span><\/div>, <div class='tooltip'>to_flowmat<span class='tooltiptext'>Create a flowmat (data.frame) using a named list a commands.as.flowmat(): reads a file and checks for required columns. If x is data.frame checks for required columns.<\/span><\/div>, <div class='tooltip'>whisker_render<span class='tooltiptext'>Internal function (used by submit_job), which creates a submission script using platform specific templates.This is a wrapper around whisker.render<\/span><\/div>","<div class='tooltip'>estTable<span class='tooltiptext'>estTable can be used to compare the results of onephase to multiphase estimations(twophase, threephase). It restructures the estimation results into a table that canbe used to plot the estimation results and provides the basis for further analysis.<\/span><\/div>, <div class='tooltip'>mphase.gain<span class='tooltiptext'>mphase.gain takes as input an object created by the estTable functionand returns a validation of which multiphase method and estimator performed best in comparisonto the onephase estimation (baseline) in terms of estimation precision.<\/span><\/div>, <div class='tooltip'>onephase<span class='tooltiptext'>onephase is used to calculate estimations exclusively based onterrestrial observations of a forest inventory (i.e. the local densities).The estimation method is available for simple and cluster-samplingand provides point estimates of the sample mean and their variances.<\/span><\/div>, <div class='tooltip'>threephase<span class='tooltiptext'>threephase is used to calculate estimations based on triple sampling under themodel-assisted Monte Carlo approach. A zero phase of auxiliary information(e.g. taken from remote sensing data) is used to generate model predictions based on multiple linearregression using the method of ordinary least squares. A subsample of the zero phase comprisesfurther auxiliary information that produces another set of model predictions.A further subsample produces a second phase based on terrestrial observations(i.e. the local densities of the ground truth) and is used to correct for bias in the design-based sense.The estimation method is available for simple and cluster sampling and includesthe special case where the first phase is based on an exhaustive sample (i.e. a census).Small-area applications are supported for synthetic estimation as well as two varietiesof bias-corrected estimators: the traditional small-area estimator and an asymptoticallyequivalent version derived under Mandallaz' extended model approach.<\/span><\/div>, <div class='tooltip'>twophase<span class='tooltiptext'>twophase is used to calculate estimations based on double sampling under themodel-assisted Monte Carlo approach. A first phase of auxiliary information(e.g. taken from remote sensing data) is used to generate model predictions based on multiple linearregression using the method of ordinary least squares. A subsample of the first phase comprisesthe second phase which contains terrestrial observations (i.e. the local densitiesof the ground truth) that is used to correct for bias in the design-based sense.The estimation method is available for simple and cluster sampling and includesthe special case where the first phase is based on an exhaustive sample (i.e. a census).Small-area applications are supported for synthetic estimation as well as two varietiesof bias-corrected estimators: the traditional small-area estimator and an asymptoticallyequivalent version derived under Mandallaz' extended model approach.<\/span><\/div>","<div class='tooltip'>fssf_b<span class='tooltiptext'>Produces a random fully-sequential design (a nested sequence of designs with points added one at a time) inside a unit hypercube such that the design points are as far away from each other as possible (Shang and Apley, 2019). The \"b\" stands for \"backward\", since the algorithm begins with the largest design and removes points one at a time. fssf_b is much faster than fssf_f when the design size is large.<\/span><\/div>, <div class='tooltip'>fssf_f<span class='tooltiptext'>Produces a random fully-sequential design (a nested sequence of designs with points added one at a time) inside a unit hypercube such that the design points are as far away from each other as possible (Shang and Apley, 2019). The \"f\" stands for \"forward\", since the algorithm begins with the smallest design and adds points one at a time.<\/span><\/div>, <div class='tooltip'>fssf_fr<span class='tooltiptext'>Produces a random fully-sequential design (a nested sequence of designs with points added one at a time) inside a unit hypercube such that the largest distance between any point inside the unit hypercube to its closest design point is as small as possible (Shang and Apley, 2019). The \"fr\" stands for \"forward reflected\", since the algorithm uses a modification of the criterion used in the fssf-f algorithm.<\/span><\/div>","<div class='tooltip'>as.fixed<span class='tooltiptext'>Assigns a class \"fixed\" to a vector<\/span><\/div>, <div class='tooltip'>as.random<span class='tooltiptext'>Assigns a class \"random\" to a vector<\/span><\/div>, <div class='tooltip'>C.test<span class='tooltiptext'>Performs a Cochran's test of the null hypothesis that the largest variance in several sampled variances are the same.<\/span><\/div>, <div class='tooltip'>estimates<span class='tooltiptext'>This function is used to construct the mean squares estimates of an ANOVA design, considering the complications imposed by nested/orthogonal and fixed/random factors.<\/span><\/div>, <div class='tooltip'>gad<span class='tooltiptext'>Fits a general ANOVA design with any combination of orthogonal/nested and fixed/random factors through function estimates<\/span><\/div>, <div class='tooltip'>is.fixed<span class='tooltiptext'>This function works the same way of is.factor.<\/span><\/div>, <div class='tooltip'>is.random<span class='tooltiptext'>This function works the same way of is.factor<\/span><\/div>, <div class='tooltip'>snk.test<span class='tooltiptext'>This function perforns a SNK post-hoc test of means on the factors of a chosen term of the model, comparing among levels of one factor within each level of other factor or combination of factors.<\/span><\/div>","<div class='tooltip'>Derive<span class='tooltiptext'>This function computes derivatives of a fitted GeDS regression model.<\/span><\/div>, <div class='tooltip'>GenUnivariateFitter<span class='tooltiptext'>These are computing engines called by NGeDS and GGeDS, needed forthe underlying fitting procedures.<\/span><\/div>, <div class='tooltip'>GGeDS<span class='tooltiptext'>GGeDS constructs a Geometrically Designed (univariate) variableknots spline regression model for the predictor in the context of Generalized (Non-)Linear Models,referred to as a GeDS model, for a response with a pre-specified distribution fromthe Exponential Family.<\/span><\/div>, <div class='tooltip'>Integrate<span class='tooltiptext'>This function computes defined integrals of a fitted GeDS regression model.<\/span><\/div>, <div class='tooltip'>IRLSfit<span class='tooltiptext'>This function is an implementation of the IRLS estimation algorithm adjusted to the specific usage in the functionSplineReg_GLM.<\/span><\/div>, <div class='tooltip'>lines<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>NGeDS<span class='tooltiptext'>NGeDS constructs a Geometrically Designed variableknots spline regression model referred to asa GeDS model, for a response having a Normal distribution.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>PPolyRep<span class='tooltiptext'>The function converts a GeDS fit which has a B-spline representationto a piecewise polynomial form.<\/span><\/div>, <div class='tooltip'>SplineReg_GLM<span class='tooltiptext'>Functions that estimate the coefficients of a predictor model involving a spline componentand possibly a parametric component applying (Iteratively Re-weighted) Least Squares (IR)LS iteration.<\/span><\/div>, <div class='tooltip'>SplineReg_LM<span class='tooltiptext'>Functions that estimate the coefficients of a predictor model involving a spline componentand possibly a parametric component applying (Iteratively Re-weighted) Least Squares (IR)LS iteration.<\/span><\/div>, <div class='tooltip'>UnivariateFitter<span class='tooltiptext'>These are computing engines called by NGeDS and GGeDS, needed forthe underlying fitting procedures.<\/span><\/div>","<div class='tooltip'>haplotypeOddsRatio<span class='tooltiptext'>Haplotype disease risk is calculated resolving haplotype ambiguity and adjusting for covariates and population stratification.<\/span><\/div>, <div class='tooltip'>twoStagePower<span class='tooltiptext'>Calculate the power of a two stage design for GWAS under sample size or cost constraints. Implements methods in the refereces below.<\/span><\/div>","<div class='tooltip'>binary_coding<span class='tooltiptext'>binary_coding codes genotypes contained in a genotypeR objectand places them into a genotypeR object's binary_genotype slot.<\/span><\/div>, <div class='tooltip'>binary_genotypes<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>binary_genotypes<-<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>CO<span class='tooltiptext'>CO is an internal function used in count_CO to countcrossovers. CO is provided in case there is a use case forthe user. We used this function for QA and it can be used forestimates of crossover assurance.<\/span><\/div>, <div class='tooltip'>convert2qtl_table<span class='tooltiptext'>convert2qtl_table will take a genotypeR object thatcontains binary coded genotypes, and export this to a csvfile suitable for use with Rqtl.<\/span><\/div>, <div class='tooltip'>count_CO<span class='tooltiptext'>count_CO counts crossovers from binary coded genotypes ina genotypeR object. This function assigns crossovers tothe counted_crossovers slot in a genotypeR object.<\/span><\/div>, <div class='tooltip'>counted_crossovers<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>counted_crossovers<-<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>genotypes<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>genotypes<-<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>GoldenGate2iCOM_design<span class='tooltiptext'>GoldenGate2iCOM_design outputs GoldenGate markers forSNP genotyping assay development with illumina iCOM.<\/span><\/div>, <div class='tooltip'>grep_df_subset<span class='tooltiptext'>grep_df_subset is an internal function that subsetsa data frame based on supplied pattern. This function isprovided in case it is found useful.<\/span><\/div>, <div class='tooltip'>Heterogametic_Genotype_Warnings<span class='tooltiptext'>Heterogametic_Genotype_Warnings provides QA forback cross designs by determine those organisms that havean impossible genotypes based on their sex.<\/span><\/div>, <div class='tooltip'>illumina_Genotype_Table<span class='tooltiptext'>illumina_Genotype_Table produces the Alt_Ref_Table neededby initialize_genotypeR_data from illumina'sgoldengate platform.<\/span><\/div>, <div class='tooltip'>impossible_genotype<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>initialize_genotypeR_data<span class='tooltiptext'>This initializes the genotypeR data structure used throughoutthe package.<\/span><\/div>, <div class='tooltip'>make_marker_names<span class='tooltiptext'>make_marker_names makes genotypeR compliant names. Thisis used for input into SNP assay design software. The output isalso used in Ref_Alt_Table.<\/span><\/div>, <div class='tooltip'>read_in_illumina_GoldenGate<span class='tooltiptext'>read_in_illumina_GoldenGate reads in a tab delimitedoutput file from illumina GoldenGate SNP genotyping platform foruse in genotypeR.<\/span><\/div>, <div class='tooltip'>read_in_Master_SNPs_data<span class='tooltiptext'>This reads in single nucleotide polymorphism markers generatedby the GrandMasterSNPs Perl program.<\/span><\/div>, <div class='tooltip'>read_in_sequenom_data<span class='tooltiptext'>read_in_sequenom_data reads in a csv file produced fromthe Sequenom platform (i.e., sequenom excel output saved as a csv).This function is a wrapper function around read.csv in order to read genotype data from the Sequenom Platform,and provide data compatible with the genotypeR package.<\/span><\/div>, <div class='tooltip'>Ref_Alt_Table<span class='tooltiptext'>Ref_Alt_Table makes the ref/alt table usedin for proper genotype coding andQA/QC initialize_genotypeR_data.<\/span><\/div>, <div class='tooltip'>SequenomMarkers<span class='tooltiptext'>SequenomMarkers runs the SNP genotyping marker designportion of the genotypeR pipeline.This function designs Sequenom markers.<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'>Class genotypeR Defines a class and data structure for working with genotyping data.Generic genotypes.Method impossible_genotype.Method binary_genotypes.Method counted_crossovers.Method binary_genotypes<-.Method genotypes<-.Method counted_crossovers<-.Method show.<\/span><\/div>, <div class='tooltip'>sort_sequenom_df<span class='tooltiptext'>This function sorts Sequenom Data at the read-in stage.<\/span><\/div>, <div class='tooltip'>subsetChromosome<span class='tooltiptext'>subsetChromosome subsets a genotypeR object based onthe supplied chromosome name (must be the same as that in the data).<\/span><\/div>, <div class='tooltip'>zero_one_two_coding<span class='tooltiptext'>zero_one_two_coding code homozygous reference as 0, heterozygous as 1, and homozygous alternate as 2 using a genotypeR object created with initialize_genotypeR_data with the pass_through argument.<\/span><\/div>","<div class='tooltip'>bestnet<span class='tooltiptext'>Generate a SpatialPoints object with the x and y coordinates corresponding to the best result obtained in an optimized network. The parameter to be passed to this function must be the result of seqPtsOptNet or simPtsOptNet<\/span><\/div>, <div class='tooltip'>bp.with.outlier.label<span class='tooltiptext'>geospt internal function<\/span><\/div>, <div class='tooltip'>criterio.cv<span class='tooltiptext'>Generate a data frame of statistical values associated with cross-validation<\/span><\/div>, <div class='tooltip'>est.variograms<span class='tooltiptext'>Calculate empirical variogram estimates. An object of classvariogram contains empirical variogram estimates which are generatedfrom a point object and a pair object. A variogram object is storedas a data frame containing seven columns: lags, bins,classic, robust,med, trim andn. The length of each vector is equal to the number of lagsin the pair object used to create the variogram object, say l. Thelags vector contains the lag numbers for each lag, beginningwith one (1) and going to the number of lags (l). The bins vectorcontains the spatial midpoint of each lag. The classic, robust,med and trimmed.mean vectors contain: the classical,robust, median, and trimmed mean, respectively, which are given, respectively,by (see Cressie, 1993, p. 75)classical\\gamma_{c}(h)=\\frac{1}{n}\\sum_{(i,j)\\in N(h)}(z(x_{i})-z(x_{j}))^{2}robust,\\gamma_{m}(h)=\\frac{(\\frac{1}{n}\\sum_{(i,j)\\in N(h)} (\\sqrt{|z(x_{i})-z(x_{j})|}))^{4}}{0.457+\\frac{0.494}{n}}median\\gamma_{me}(h)=\\frac{\\mbox(median_{(i,j)\\in N(h)} (\\sqrt{|z(x_{i})-z(x_{j})|}))^{4}}{0.457+\\frac{0.494}{|N(h)|}}and trimmed mean\\gamma_{tm}(h)=\\frac{(trimmed.mean_{(i,j)\\in N(h)}(\\sqrt{|z(x_{i})-z(x_{j})|}))^{4}}{0.457+\\frac{0.494}{|N(h)|}}The n vector contains the number |N(h)| of pairs of pointsin each lag N(h).<\/span><\/div>, <div class='tooltip'>extractFormula<span class='tooltiptext'>geospt internal function<\/span><\/div>, <div class='tooltip'>graph.rbf<span class='tooltiptext'>Function for plotting the RMSPE for several values of the smoothing parametereta with the same dataset. A curve is fitted to the points, andthen the optimal eta that provides the smallestRMSPE is determined from the curve, by the optimize function from the statspackage.<\/span><\/div>, <div class='tooltip'>network.design<span class='tooltiptext'>Generates a sampling network for a given sampling distance or type (configuration), and calculates the average kriging standard error prediction errors (ASEPE) associated with the spatial configuration for a given predefined variogram<\/span><\/div>, <div class='tooltip'>pocket.plot<span class='tooltiptext'>The pocket-plot (so named because of its use in detecting pocketsof non-stationarity) is a technique necessary to identify a localizedarea that is atypical with respect to the stationarity model. It is builtto exploit the spatial nature of the data through the coordinatesof rows and columns (east \"X\" and north \"Y\", respectively).<\/span><\/div>, <div class='tooltip'>rbf<span class='tooltiptext'>Function for gaussian (GAU), exponential (EXPON), trigonometric (TRI), thin plate spline (TPS), completely regularized spline (CRS), spline with tension (ST), inverse multiquadratic (IM), and multiquadratic (M) radial basis function (rbf),where rbf is in a local neighbourhood<\/span><\/div>, <div class='tooltip'>rbf.cv<span class='tooltiptext'>Generate the RMSPE value, which is given by the radial basis functionwith smoothing parameter eta and robustness parameter rho.<\/span><\/div>, <div class='tooltip'>rbf.cv1<span class='tooltiptext'>Generate the RMSPE value, which is given by the radial basis functionwith smoothing parameter eta and robustness parameter rho.<\/span><\/div>, <div class='tooltip'>RBF.phi<span class='tooltiptext'>generate the value associated with radial basis functions; gaussian GAU), exponential (EXPON),trigonometric (TRI), thin plate spline (TPS), completely regularized spline (CRS),spline with tension (ST), inverse multiquadratic (IM), and multiquadratic (M)<\/span><\/div>, <div class='tooltip'>rbf.tcv<span class='tooltiptext'>Generates a table with the results of the evaluation of radial basis functions (rbf): gaussian (GAU), exponential (EXPON), trigonometric (TRI), thin plate spline (TPS), completely regularized spline (CRS), spline with tension (ST), inverse multiquadratic (IM), and multiquadratic (M) from the leave-one-out cross validation method.<\/span><\/div>, <div class='tooltip'>samplePts<span class='tooltiptext'>sample location points within a square area, a grid, a polygon, or on a spatial line, using regular or random sampling methods. The function spsample from the package sp is used iteratively to find exactly n sample locations<\/span><\/div>, <div class='tooltip'>seqPtsOptNet<span class='tooltiptext'>Search for the optimum location of one additional point to be added to an initial network, minimizing the average standard error of kriging through a genetic algorithm. It takes, as input for the optimization, the minimum and maximum values of the coordinates that enclose the study area. This function uses previous samples information to direct additional sampling. The location of the new point is searched randomly.<\/span><\/div>, <div class='tooltip'>simPtsOptNet<span class='tooltiptext'>Search for an optimum set of simultaneous additional points to an initial network that minimize the average standard error of kriging, using a genetic algorithm. It takes, as input for the optimization, the minimum and maximum values of the coordinates that enclose the study area. This function uses previous samples information to directadditional sampling for minimum average standard error. The algorithm generates random sampling schemes.<\/span><\/div>","<div class='tooltip'>granova.1w<span class='tooltiptext'>Graphic to display data for a one-way analysis of variance, and also to help understand how ANOVA works, how the F statistic is generated for the data in hand, etc. The graphic may be called 'elemental' or 'natural' because it is builtupon the key question that drives one-way ANOVA.<\/span><\/div>, <div class='tooltip'>granova.2w<span class='tooltiptext'>Produces a rotatable graphic (controlled by the mouse) to display all data points for any two way analysis of variance.<\/span><\/div>, <div class='tooltip'>granova.contr<span class='tooltiptext'>Provides graphic displays that shows data and effects for a priori contrasts in ANOVA contexts; also corresponding numerical results.<\/span><\/div>, <div class='tooltip'>granova.ds<span class='tooltiptext'>Plots dependent sample data beginning from a scatterplot for the X,Y pairs; proceeds to display difference scores as point projections; also X and Y means, as well as the mean of the difference scores. Also prints various summary statistics including: effect size, means for X and Y, a 95% confidence interval for the mean difference as well as the t-statistic and degrees of freedom.<\/span><\/div>","<div class='tooltip'>groupseq<span class='tooltiptext'>This function is invoked when the package is loaded into R workspace. It starts the main menuusing a graphical user interface (GUI). If, for any reason, the GUI does not work,user can choose an interactive mode by typing groupseq(mode=\"c\") to work within the R console.However, the console mode has not been further developed since GroupSeq version 1.0.The program itself refers to an earlier Fortran Implementation by Reboussin et al.and was completely recoded in R. Following the maxim of modular programming,it was decomposed in various functions. Furthermore, some algorithms could be improvedand the whole application was embedded in a GUI affording much more convenient and faster working.Last but not least, computation results can be saved into common tables of *.html file formatto eventually process them further on.<\/span><\/div>","<div class='tooltip'>gsb<span class='tooltiptext'>The gsbDesign package allows to evaluate the operating characteristics for a group sequential design with Bayesian success/futility criteria and prior information. A clinical trial with two arms, a normal endpoint and an arbitrary number of interim analyses can be evaluated. The success and futility criteria at each interim analysis are based on the posterior distribution for the true treatment difference (delta). An arbitrary number of success and futility criteria can be specified at each interim analysis. The success criteria are of the form: P( \\delta > ds | data ) > ps And the futility criteria are of the form: P( \\delta < df | data ) > pf Here ds and df are user-specified effect thresholds, and ps and pf are user-defined probability thresholds. Prior information can either be specified for the true treatmentdifference (delta), or for the true effects in the two treatmentarms. Only normal prior distributions can be used. In all cases informative priors are specified in terms of a mean and an effective sample size defined relative to sigma.The user also has to specify the number of patients for each stage ofthe group sequential design, and the standard deviation (sigma) of the endpoint (assumed to be known).The operating characteristics are either evaluated for a user-specified grid of true treatment differences, or for a grid or set of true treatment means for the two arms. The operating characteristics of main interest are the probabilities of success and futility at each interim analysis, and the expected sample size.The main function of the package is gsb().<\/span><\/div>, <div class='tooltip'>gsbBayesUpdate<span class='tooltiptext'>Bayesian update from prior and data to posterior for normallydistributed data with known sigma.<\/span><\/div>, <div class='tooltip'>gsbCriteria<span class='tooltiptext'>Transforms the criteria on posterior-scale.<\/span><\/div>, <div class='tooltip'>gsbDesign<span class='tooltiptext'>The gsbDesign package allows to evaluate the operating characteristics for a group sequential design with Bayesian success/futility criteria and prior information. A clinical trial with two arms, a normal endpoint and an arbitrary number of interim analyses can be evaluated. The success and futility criteria at each interim analysis are based on the posterior distribution for the true treatment difference (delta). An arbitrary number of success and futility criteria can be specified at each interim analysis. The success criteria are of the form: P( \\delta > ds | data ) > ps And the futility criteria are of the form: P( \\delta < df | data ) > pf Here ds and df are user-specified effect thresholds, and ps and pf are user-defined probability thresholds. Prior information can either be specified for the true treatmentdifference (delta), or for the true effects in the two treatmentarms. Only normal prior distributions can be used. In all cases informative priors are specified in terms of a mean and an effective sample size defined relative to sigma.The user also has to specify the number of patients for each stage ofthe group sequential design, and the standard deviation (sigma) of the endpoint (assumed to be known).The operating characteristics are either evaluated for a user-specified grid of true treatment differences, or for a grid or set of true treatment means for the two arms. The operating characteristics of main interest are the probabilities of success and futility at each interim analysis, and the expected sample size.The main function of the package is gsb().<\/span><\/div>, <div class='tooltip'>gsbSimulation<span class='tooltiptext'>The gsbDesign package allows to evaluate the operating characteristics for a group sequential design with Bayesian success/futility criteria and prior information. A clinical trial with two arms, a normal endpoint and an arbitrary number of interim analyses can be evaluated. The success and futility criteria at each interim analysis are based on the posterior distribution for the true treatment difference (delta). An arbitrary number of success and futility criteria can be specified at each interim analysis. The success criteria are of the form: P( \\delta > ds | data ) > ps And the futility criteria are of the form: P( \\delta < df | data ) > pf Here ds and df are user-specified effect thresholds, and ps and pf are user-defined probability thresholds. Prior information can either be specified for the true treatmentdifference (delta), or for the true effects in the two treatmentarms. Only normal prior distributions can be used. In all cases informative priors are specified in terms of a mean and an effective sample size defined relative to sigma.The user also has to specify the number of patients for each stage ofthe group sequential design, and the standard deviation (sigma) of the endpoint (assumed to be known).The operating characteristics are either evaluated for a user-specified grid of true treatment differences, or for a grid or set of true treatment means for the two arms. The operating characteristics of main interest are the probabilities of success and futility at each interim analysis, and the expected sample size.The main function of the package is gsb().<\/span><\/div>, <div class='tooltip'>tab<span class='tooltiptext'>This function creates tables from the output of functiongsb().<\/span><\/div>","<div class='tooltip'>design_gsnb<span class='tooltiptext'>Design a group sequential trial with negative binomial outcomes<\/span><\/div>, <div class='tooltip'>design_nb<span class='tooltiptext'>Design a clinical trial with negative binomial outcomes<\/span><\/div>, <div class='tooltip'>get_calendartime_gsnb<span class='tooltiptext'>Calculate the calendar time of looks given the information time<\/span><\/div>, <div class='tooltip'>get_info_gsnb<span class='tooltiptext'>Calculates the information level for the log rate ratio ofthe negative binomial model<\/span><\/div>, <div class='tooltip'>obrien<span class='tooltiptext'>Error spending function mimicking O'Brien & Fleming critical values<\/span><\/div>, <div class='tooltip'>pocock<span class='tooltiptext'>Error spending function mimicking Pococks critical values<\/span><\/div>","<div class='tooltip'>binomialSPRT<span class='tooltiptext'>gsBinomialExact computes power/Type I error and expected sample sizefor a group sequential design in a single-arm trial with a binary outcome.This can also be used to compare event rates in two-arm studies. The printfunction has been extended using print.gsBinomialExact to printgsBinomialExact objects. Similarly, a plot function hasbeen extended using plot.gsBinomialExact to plotgsBinomialExact objects.binomialSPRT computes a truncated binomial sequential probabilityratio test (SPRT) which is a specific instance of an exact binomial groupsequential design for a single arm trial with a binary outcome.%\\code{gsBinomialPP} computes a truncated binomial (group) sequential designbased on predictive probability.nBinomial1Sample uses exact binomial calculations to compute powerand sample size for single arm binomial experiments.gsBinomialExact is based on the book \"Group Sequential Methods withApplications to Clinical Trials,\" Christopher Jennison and Bruce W.Turnbull, Chapter 12, Section 12.1.2 Exact Calculations for Binary Data.This computation is often used as an approximation for the distribution ofthe number of events in one treatment group out of all events when theprobability of an event is small and sample size is large.An object of class gsBinomialExact is returned. On output, the valuesof theta input to gsBinomialExact will be the parameter valuesfor which the boundary crossing probabilities and expected sample sizes arecomputed.Note that a[1] equal to -1 lower bound at n.I[1] means 0 successes continuesat interim 1; a[2]==0 at interim 2 means 0 successes stops trial forfutility at 2nd analysis. For final analysis, set a[k] equal to b[k]-1 toincorporate all possibilities into non-positive trial; see example.The sequential probability ratio test (SPRT) is a sequential testing schemeallowing testing after each observation. This likelihood ratio is used todetermine upper and lower cutoffs which are linear and parallel in thenumber of responses as a function of sample size. binomialSPRTproduces a variation the the SPRT that tests only within a range of samplesizes. While the linear SPRT bounds are continuous, actual bounds are theinteger number of response at or beyond each linear bound for each samplesize where testing is performed. Because of the truncation anddiscretization of the bounds, power and Type I error achieve will be lowerthan the nominal levels specified by alpha and beta which canbe altered to produce desired values that are achieved by the planned samplesize. See also example that shows computation of Type I error when futilitybound is considered non-binding.Note that if the objective of a design is to demonstrate that a rate (e.g.,failure rate) is lower than a certain level, two approaches can be taken.First, 1 minus the failure rate is the success rate and this can be used forplanning. Second, the role of beta becomes to express Type I errorand alpha is used to express Type II error.Plots produced include boundary plots, expected sample size, response rateat the boundary and power.gsBinomial1Sample uses exact binomial computations based on the baseR functions qbinom() and pbinom(). The tabular output may beconvenient for plotting. Note that input variables are largely not checked,so the user is largely responsible for results; it is a good idea to do arun with outtype=3 to check that you have done things appropriately.If n is not ordered (a bad idea) or not sequential (maybe OK), beaware of possible consequences.nBinomial1Sample is based on code from Marc Schwartz marc_schwartz@me.com. The possible sample size vector n needs to be selected in such a fashionthat it covers the possible range of values that include the true minima. NOTE: the one-sided evaluation of significance is more conservative than using the 2-sided exact test in binom.test.<\/span><\/div>, <div class='tooltip'>checkLengths<span class='tooltiptext'>Utility functions to verify an objects's properties including whether it isa scalar or vector, the class, the length, and (if numeric) whether therange of values is on a specified interval. Additionally, thecheckLengths function can be used to ensure that all the suppliedinputs have equal lengths.isInteger is similar to is.integer except thatisInteger(1) returns TRUE whereas is.integer(1) returnsFALSE.checkScalar is used to verify that the input object is a scalar aswell as the other properties specified above.checkVector is used to verify that the input object is an atomicvector as well as the other properties as defined above.checkRange is used to check whether the numeric input object's valuesreside on the specified interval. If any of the values are outside thespecified interval, a FALSE is returned.checkLength is used to check whether all of the supplied inputs haveequal lengths.<\/span><\/div>, <div class='tooltip'>checkRange<span class='tooltiptext'>Utility functions to verify an objects's properties including whether it isa scalar or vector, the class, the length, and (if numeric) whether therange of values is on a specified interval. Additionally, thecheckLengths function can be used to ensure that all the suppliedinputs have equal lengths.isInteger is similar to is.integer except thatisInteger(1) returns TRUE whereas is.integer(1) returnsFALSE.checkScalar is used to verify that the input object is a scalar aswell as the other properties specified above.checkVector is used to verify that the input object is an atomicvector as well as the other properties as defined above.checkRange is used to check whether the numeric input object's valuesreside on the specified interval. If any of the values are outside thespecified interval, a FALSE is returned.checkLength is used to check whether all of the supplied inputs haveequal lengths.<\/span><\/div>, <div class='tooltip'>checkScalar<span class='tooltiptext'>Utility functions to verify an objects's properties including whether it isa scalar or vector, the class, the length, and (if numeric) whether therange of values is on a specified interval. Additionally, thecheckLengths function can be used to ensure that all the suppliedinputs have equal lengths.isInteger is similar to is.integer except thatisInteger(1) returns TRUE whereas is.integer(1) returnsFALSE.checkScalar is used to verify that the input object is a scalar aswell as the other properties specified above.checkVector is used to verify that the input object is an atomicvector as well as the other properties as defined above.checkRange is used to check whether the numeric input object's valuesreside on the specified interval. If any of the values are outside thespecified interval, a FALSE is returned.checkLength is used to check whether all of the supplied inputs haveequal lengths.<\/span><\/div>, <div class='tooltip'>checkVector<span class='tooltiptext'>Utility functions to verify an objects's properties including whether it isa scalar or vector, the class, the length, and (if numeric) whether therange of values is on a specified interval. Additionally, thecheckLengths function can be used to ensure that all the suppliedinputs have equal lengths.isInteger is similar to is.integer except thatisInteger(1) returns TRUE whereas is.integer(1) returnsFALSE.checkScalar is used to verify that the input object is a scalar aswell as the other properties specified above.checkVector is used to verify that the input object is an atomicvector as well as the other properties as defined above.checkRange is used to check whether the numeric input object's valuesreside on the specified interval. If any of the values are outside thespecified interval, a FALSE is returned.checkLength is used to check whether all of the supplied inputs haveequal lengths.<\/span><\/div>, <div class='tooltip'>ciBinomial<span class='tooltiptext'>Support is provided for sample size estimation, power, testing, confidenceintervals and simulation for fixed sample size trials (that is, not groupsequential or adaptive) with two arms and binary outcomes. Both superiorityand non-inferiority trials are considered. While all routines default tocomparisons of risk-difference, options to base computations on risk-ratioand odds-ratio are also included.nBinomial() computes sample size or power using the method ofFarrington and Manning (1990) for a trial to test the difference between twobinomial event rates. The routine can be used for a test of superiority ornon-inferiority. For a design that tests for superiority nBinomial()is consistent with the method of Fleiss, Tytun, and Ury (but without thecontinuity correction) to test for differences between event rates. Thisroutine is consistent with the Hmisc package routines bsamsize andbpower for superiority designs. Vector arguments allow computingsample sizes for multiple scenarios for comparative purposes.testBinomial() computes a Z- or Chi-square-statistic that comparestwo binomial event rates using the method of Miettinen and Nurminen (1980).This can be used for superiority or non-inferiority testing. Vectorarguments allow easy incorporation into simulation routines for fixed, groupsequential and adaptive designs.ciBinomial() computes confidence intervals for 1) the differencebetween two rates, 2) the risk-ratio for two rates or 3) the odds-ratio fortwo rates. This procedure provides inference that is consistent withtestBinomial() in that the confidence intervals are produced byinverting the testing procedures in testBinomial(). The Type I erroralpha input to ciBinomial is always interpreted as 2-sided.simBinomial() performs simulations to estimate the power for aMiettinen and Nurminen (1985) test comparing two binomial rates forsuperiority or non-inferiority. As noted in documentation forbpower.sim() in the HMisc package, by using testBinomial() youcan see that the formulas without any continuity correction are quiteaccurate. In fact, Type I error for a continuity-corrected test issignificantly lower (Gordon and Watson, 1996) than the nominal rate. Thus,as a default no continuity corrections are performed.varBinomial computes blinded estimates of the variance of theestimate of 1) event rate differences, 2) logarithm of the risk ratio, or 3)logarithm of the odds ratio. This is intended for blinded sample sizere-estimation for comparative trials with a binary outcome.Testing is 2-sided when a Chi-square statistic is used and 1-sided when aZ-statistic is used. Thus, these 2 options will produce substantiallydifferent results, in general. For non-inferiority, 1-sided testing isappropriate.You may wish to round sample sizes up using ceiling().Farrington and Manning (1990) begin with event rates p1 and p2under the alternative hypothesis and a difference between these rates underthe null hypothesis, delta0. From these values, actual rates underthe null hypothesis are computed, which are labeled p10 andp20 when outtype=3. The rates p1 and p2 are usedto compute a variance for a Z-test comparing rates under the alternativehypothesis, while p10 and p20 are used under the nullhypothesis. This computational method is also used to estimate variances invarBinomial() based on the overall event rate observed and the inputtreatment difference specified in delta0.Sample size with scale=\"Difference\" produces an error ifp1-p2=delta0. Normally, the alternative hypothesis underconsideration would be p1-p2-delta0$>0$. However, the alternative canhave p1-p2-delta0$<0$.<\/span><\/div>, <div class='tooltip'>condPower<span class='tooltiptext'>ssrCP() adapts 2-stage group sequential designs to 2-stage samplesize re-estimation designs based on interim analysis conditional power. Thisis a simple case of designs developed by Lehmacher and Wassmer, Biometrics(1999). The conditional power designs of Bauer and Kohne (1994), Proschan and Hunsberger (1995), Cui, Hung and Wang (1999) and Liu and Chi (2001), Gao, Ware and Mehta (2008), and Mehta and Pocock (2011). Either the estimatedtreatment effect at the interim analysis or any chosen effect size can beused for sample size re-estimation. If not done carefully, these designs canbe very inefficient. It is probably a good idea to compare any design to asimpler group sequential design; see, for example, Jennison and Turnbull(2003). The a assumes a small Type I error is included with the interimanalysis and that the design is an adaptation from a 2-stage groupsequential design Related functions include 3 pre-defined combination testfunctions (z2NC, z2Z, z2Fisher) that represent theinverse normal combination test (Lehmacher and Wassmer, 1999), thesufficient statistic for the complete data, and Fisher's combination test.Power.ssrCP computes unconditional power for a conditional powerdesign derived by ssrCP.condPower is a supportive routine that also is interesting in its ownright; it computes conditional power of a combination test given an interimtest statistic, stage 2 sample size and combination test statistic. Whilethe returned data frame should make general plotting easy, the functionplot.ssrCP() prints a plot of study sample size by stage 1 outcomewith multiple x-axis labels for stage 1 z-value, conditional power, andstage 1 effect size relative to the effect size for which the underlyinggroup sequential design was powered.Sample size re-estimation using conditional power and an interim estimate oftreatment effect was proposed by several authors in the 1990's (seereferences below). Statistical testing for these original methods was basedon combination tests since Type I error could be inflated by using asufficient statistic for testing at the end of the trial. Since 2000, moreefficient variations of these conditional power designs have been developed.Fully optimized designs have also been derived (Posch et all, 2003,Lokhnygina and Tsiatis, 2008). Some of the later conditional power methodsallow use of sufficient statistics for testing (Chen, DeMets and Lan, 2004,Gao, Ware and Mehta, 2008, and Mehta and Pocock, 2011).The methods considered here are extensions of 2-stage group sequentialdesigns that include both an efficacy and a futility bound at the plannedinterim analysis. A maximum fold-increase in sample size (maxinc)fromthe supplied group sequential design (x) is specified by the user, aswell as a range of conditional power (cpadj) where sample size shouldbe re-estimated at the interim analysis, 1-the targeted conditional power tobe used for sample size re-estimation (beta) and a combination teststatistic (z2) to be used for testing at the end of the trial. Theinput value overrun represents incremental enrollment not included inthe interim analysis that is not included in the analysis; this is used forcalculating the required number of patients enrolled to complete the trial.Whereas most of the methods proposed have been based on using the interimestimated treatment effect size (default for ssrCP), the variabletheta allows the user to specify an alternative; Liu and Chi (2001)suggest that using the parameter value for which the trial was originallypowered is a good choice.<\/span><\/div>, <div class='tooltip'>eEvents<span class='tooltiptext'>eEvents() is used to calculate the expected number of events for apopulation with a time-to-event endpoint. It is based on calculationsdemonstrated in Lachin and Foulkes (1986) and is fundamental in computationsfor the sample size method they propose. Piecewise exponential survival anddropout rates are supported as well as piecewise uniform enrollment. Astratified population is allowed. Output is the expected number of eventsobserved given a trial duration and the above rate parameters.eEvents() produces an object of class eEvents with the numberof subjects and events for a set of pre-specified trial parameters, such asaccrual duration and follow-up period. The underlying power calculation isbased on Lachin and Foulkes (1986) method for proportional hazards assuminga fixed underlying hazard ratio between 2 treatment groups. The method hasbeen extended here to enable designs to test non-inferiority. Piecewiseconstant enrollment and failure rates are assumed and a stratifiedpopulation is allowed. See also nSurvival for other Lachin andFoulkes (1986) methods assuming a constant hazard difference or exponentialenrollment rate.print.eEvents() formats the output for an object of classeEvents and returns the input value.<\/span><\/div>, <div class='tooltip'>gsBinomialExact<span class='tooltiptext'>gsBinomialExact computes power/Type I error and expected sample sizefor a group sequential design in a single-arm trial with a binary outcome.This can also be used to compare event rates in two-arm studies. The printfunction has been extended using print.gsBinomialExact to printgsBinomialExact objects. Similarly, a plot function hasbeen extended using plot.gsBinomialExact to plotgsBinomialExact objects.binomialSPRT computes a truncated binomial sequential probabilityratio test (SPRT) which is a specific instance of an exact binomial groupsequential design for a single arm trial with a binary outcome.%\\code{gsBinomialPP} computes a truncated binomial (group) sequential designbased on predictive probability.nBinomial1Sample uses exact binomial calculations to compute powerand sample size for single arm binomial experiments.gsBinomialExact is based on the book \"Group Sequential Methods withApplications to Clinical Trials,\" Christopher Jennison and Bruce W.Turnbull, Chapter 12, Section 12.1.2 Exact Calculations for Binary Data.This computation is often used as an approximation for the distribution ofthe number of events in one treatment group out of all events when theprobability of an event is small and sample size is large.An object of class gsBinomialExact is returned. On output, the valuesof theta input to gsBinomialExact will be the parameter valuesfor which the boundary crossing probabilities and expected sample sizes arecomputed.Note that a[1] equal to -1 lower bound at n.I[1] means 0 successes continuesat interim 1; a[2]==0 at interim 2 means 0 successes stops trial forfutility at 2nd analysis. For final analysis, set a[k] equal to b[k]-1 toincorporate all possibilities into non-positive trial; see example.The sequential probability ratio test (SPRT) is a sequential testing schemeallowing testing after each observation. This likelihood ratio is used todetermine upper and lower cutoffs which are linear and parallel in thenumber of responses as a function of sample size. binomialSPRTproduces a variation the the SPRT that tests only within a range of samplesizes. While the linear SPRT bounds are continuous, actual bounds are theinteger number of response at or beyond each linear bound for each samplesize where testing is performed. Because of the truncation anddiscretization of the bounds, power and Type I error achieve will be lowerthan the nominal levels specified by alpha and beta which canbe altered to produce desired values that are achieved by the planned samplesize. See also example that shows computation of Type I error when futilitybound is considered non-binding.Note that if the objective of a design is to demonstrate that a rate (e.g.,failure rate) is lower than a certain level, two approaches can be taken.First, 1 minus the failure rate is the success rate and this can be used forplanning. Second, the role of beta becomes to express Type I errorand alpha is used to express Type II error.Plots produced include boundary plots, expected sample size, response rateat the boundary and power.gsBinomial1Sample uses exact binomial computations based on the baseR functions qbinom() and pbinom(). The tabular output may beconvenient for plotting. Note that input variables are largely not checked,so the user is largely responsible for results; it is a good idea to do arun with outtype=3 to check that you have done things appropriately.If n is not ordered (a bad idea) or not sequential (maybe OK), beaware of possible consequences.nBinomial1Sample is based on code from Marc Schwartz marc_schwartz@me.com. The possible sample size vector n needs to be selected in such a fashionthat it covers the possible range of values that include the true minima. NOTE: the one-sided evaluation of significance is more conservative than using the 2-sided exact test in binom.test.<\/span><\/div>, <div class='tooltip'>gsBound<span class='tooltiptext'>gsBound() and gsBound1() are lower-level functions used tofind boundaries for a group sequential design. They are not recommended(especially gsBound1()) for casual users. These functions do notadjust sample size as gsDesign() does to ensure appropriate power fora design.gsBound() computes upper and lower bounds given boundary crossingprobabilities assuming a mean of 0, the usual null hypothesis.gsBound1() computes the upper bound given a lower boundary, upperboundary crossing probabilities and an arbitrary mean (theta).The function gsBound1() requires special attention to detail andknowledge of behavior when a design corresponding to the input parametersdoes not exist.<\/span><\/div>, <div class='tooltip'>gsBound1<span class='tooltiptext'>gsBound() and gsBound1() are lower-level functions used tofind boundaries for a group sequential design. They are not recommended(especially gsBound1()) for casual users. These functions do notadjust sample size as gsDesign() does to ensure appropriate power fora design.gsBound() computes upper and lower bounds given boundary crossingprobabilities assuming a mean of 0, the usual null hypothesis.gsBound1() computes the upper bound given a lower boundary, upperboundary crossing probabilities and an arbitrary mean (theta).The function gsBound1() requires special attention to detail andknowledge of behavior when a design corresponding to the input parametersdoes not exist.<\/span><\/div>, <div class='tooltip'>gsBoundCP<span class='tooltiptext'>gsBoundCP() computes the total probability of crossing future upperbounds given an interim test statistic at an interim bound. For each interimboundary, assumes an interim test statistic at the boundary and computes theprobability of crossing any of the later upper boundaries.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.<\/span><\/div>, <div class='tooltip'>gsBoundSummary<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>gsBValue<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>gsCP<span class='tooltiptext'>gsCP() computes conditional boundary crossing probabilities at futureplanned analyses for a given group sequential design assuming an interimz-statistic at a specified interim analysis. While gsCP() is designedtoward computing conditional power for a variety of underlying parametervalues, condPower is built to compute conditional power for avariety of interim test statistic values which is useful for sample sizeadaptation (see ssrCP). gsPP() averages conditionalpower across a posterior distribution to compute predictive power.gsPI() computes Bayesian prediction intervals for future analysescorresponding to results produced by gsPP(). gsPosterior()computes the posterior density for the group sequential design parameter ofinterest given a prior density and an interim outcome that is exact or in aninterval. gsPOS() computes the probability of success for a trialusing a prior distribution to average power over a set of thetavalues of interest. gsCPOS() assumes no boundary has been crossedbefore and including an interim analysis of interest, and computes theprobability of success based on this event. Note that gsCP() andgsPP() take only the interim test statistic into account in computingconditional probabilities, while gsCPOS() conditions on not crossingany bound through a specified interim analysis.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.For gsPP(), gsPI(), gsPOS() and gsCPOS(), theprior distribution for the standardized parameter theta () for agroup sequential design specified through a gsDesign object is specifiedthrough the arguments theta and wgts. This can be a discreteor a continuous probability density function. For a discrete function,generally all weights would be 1. For a continuous density, the wgtswould contain integration grid weights, such as those provided bynormalGrid.For gsPosterior, a prior distribution in prior must becomposed of the vectors z density. The vector zcontains points where the prior is evaluated and density thecorresponding density or, for a discrete distribution, the probabilities ofeach point in z. Densities may be supplied as fromnormalGrid() where grid weights for numerical integration aresupplied in gridwgts. If gridwgts are not supplied, they aredefaulted to 1 (equal weighting). To ensure a proper prior distribution, youmust have sum(gridwgts * density) equal to 1; this is NOT checked,however.<\/span><\/div>, <div class='tooltip'>gsCPOS<span class='tooltiptext'>gsCP() computes conditional boundary crossing probabilities at futureplanned analyses for a given group sequential design assuming an interimz-statistic at a specified interim analysis. While gsCP() is designedtoward computing conditional power for a variety of underlying parametervalues, condPower is built to compute conditional power for avariety of interim test statistic values which is useful for sample sizeadaptation (see ssrCP). gsPP() averages conditionalpower across a posterior distribution to compute predictive power.gsPI() computes Bayesian prediction intervals for future analysescorresponding to results produced by gsPP(). gsPosterior()computes the posterior density for the group sequential design parameter ofinterest given a prior density and an interim outcome that is exact or in aninterval. gsPOS() computes the probability of success for a trialusing a prior distribution to average power over a set of thetavalues of interest. gsCPOS() assumes no boundary has been crossedbefore and including an interim analysis of interest, and computes theprobability of success based on this event. Note that gsCP() andgsPP() take only the interim test statistic into account in computingconditional probabilities, while gsCPOS() conditions on not crossingany bound through a specified interim analysis.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.For gsPP(), gsPI(), gsPOS() and gsCPOS(), theprior distribution for the standardized parameter theta () for agroup sequential design specified through a gsDesign object is specifiedthrough the arguments theta and wgts. This can be a discreteor a continuous probability density function. For a discrete function,generally all weights would be 1. For a continuous density, the wgtswould contain integration grid weights, such as those provided bynormalGrid.For gsPosterior, a prior distribution in prior must becomposed of the vectors z density. The vector zcontains points where the prior is evaluated and density thecorresponding density or, for a discrete distribution, the probabilities ofeach point in z. Densities may be supplied as fromnormalGrid() where grid weights for numerical integration aresupplied in gridwgts. If gridwgts are not supplied, they aredefaulted to 1 (equal weighting). To ensure a proper prior distribution, youmust have sum(gridwgts * density) equal to 1; this is NOT checked,however.<\/span><\/div>, <div class='tooltip'>gsCPz<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>gsDelta<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>gsDensity<span class='tooltiptext'>Given an interim analysis i of a group sequential design and a vectorof real values zi, gsDensity() computes an interim densityfunction at analysis i at the values in zi. For each value inzi, this interim density is the derivative of the probability thatthe group sequential trial does not cross a boundary prior to thei-th analysis and at the i-th analysis the interim Z-statisticis less than that value. When integrated over the real line, this densitycomputes the probability of not crossing a bound at a previous analysis. Itcorresponds to the subdistribution function at analysis i thatexcludes the probability of crossing a bound at an earlier analysis.The initial purpose of this routine was as a component needed to compute thepredictive power for a trial given an interim result; seegsPP.See Jennison and Turnbull (2000) for details on how these computations areperformed.<\/span><\/div>, <div class='tooltip'>gsDesign<span class='tooltiptext'>gsDesign() is used to find boundaries and trial size required for agroup sequential design.Many parameters normally take on default values and thus do not requireexplicit specification. One- and two-sided designs are supported. Two-sideddesigns may be symmetric or asymmetric. Wang-Tsiatis designs, includingO'Brien-Fleming and Pocock designs can be generated. Designs with commonspending functions as well as other built-in and user-specified functionsfor Type I error and futility are supported. Type I error computations forasymmetric designs may assume binding or non-binding lower bounds. The printfunction has been extended using print.gsDesign() to printgsDesign objects; see examples.The user may ignore the structure of the value returned by gsDesign()if the standard printing and plotting suffice; see examples.delta and n.fix are used together to determine what samplesize output options the user seeks. The default, delta=0 andn.fix=1, results in a generic design that may be used withany sampling situation. Sample size ratios are provided and the usermultiplies these times the sample size for a fixed design to obtain thecorresponding group sequential analysis times. If delta>0,n.fix is ignored, and delta is taken as the standardizedeffect size - the signal to noise ratio for a single observation; forexample, the mean divided by the standard deviation for a one-sample normalproblem. In this case, the sample size at each analysis is computed. Whendelta=0 and n.fix>1, n.fix is assumed to be the samplesize for a fixed design with no interim analyses. See examples below.Following are further comments on the input argument test.type whichis used to control what type of error measurements are used in trial design.The manual may also be worth some review in order to see actual formulas forboundary crossing probabilities for the various options. Options 3 and 5assume the trial stops if the lower bound is crossed for Type I and Type IIerror computation (binding lower bound). For the purpose of computing TypeI error, options 4 and 6 assume the trial continues if the lower bound iscrossed (non-binding lower bound); that is a Type I error can be made bycrossing an upper bound after crossing a previous lower bound.Beta-spending refers to error spending for the lower bound crossingprobabilities under the alternative hypothesis (options 3 and 4). In thiscase, the final analysis lower and upper boundaries are assumed to be thesame. The appropriate total beta spending (power) is determined by adjustingthe maximum sample size through an iterative process for all options. Sinceoptions 3 and 4 must compute boundary crossing probabilities under both thenull and alternative hypotheses, deriving these designs can take longer thanother options. Options 5 and 6 compute lower bound spending under the nullhypothesis.<\/span><\/div>, <div class='tooltip'>gsHR<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>gsPI<span class='tooltiptext'>gsCP() computes conditional boundary crossing probabilities at futureplanned analyses for a given group sequential design assuming an interimz-statistic at a specified interim analysis. While gsCP() is designedtoward computing conditional power for a variety of underlying parametervalues, condPower is built to compute conditional power for avariety of interim test statistic values which is useful for sample sizeadaptation (see ssrCP). gsPP() averages conditionalpower across a posterior distribution to compute predictive power.gsPI() computes Bayesian prediction intervals for future analysescorresponding to results produced by gsPP(). gsPosterior()computes the posterior density for the group sequential design parameter ofinterest given a prior density and an interim outcome that is exact or in aninterval. gsPOS() computes the probability of success for a trialusing a prior distribution to average power over a set of thetavalues of interest. gsCPOS() assumes no boundary has been crossedbefore and including an interim analysis of interest, and computes theprobability of success based on this event. Note that gsCP() andgsPP() take only the interim test statistic into account in computingconditional probabilities, while gsCPOS() conditions on not crossingany bound through a specified interim analysis.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.For gsPP(), gsPI(), gsPOS() and gsCPOS(), theprior distribution for the standardized parameter theta () for agroup sequential design specified through a gsDesign object is specifiedthrough the arguments theta and wgts. This can be a discreteor a continuous probability density function. For a discrete function,generally all weights would be 1. For a continuous density, the wgtswould contain integration grid weights, such as those provided bynormalGrid.For gsPosterior, a prior distribution in prior must becomposed of the vectors z density. The vector zcontains points where the prior is evaluated and density thecorresponding density or, for a discrete distribution, the probabilities ofeach point in z. Densities may be supplied as fromnormalGrid() where grid weights for numerical integration aresupplied in gridwgts. If gridwgts are not supplied, they aredefaulted to 1 (equal weighting). To ensure a proper prior distribution, youmust have sum(gridwgts * density) equal to 1; this is NOT checked,however.<\/span><\/div>, <div class='tooltip'>gsPOS<span class='tooltiptext'>gsCP() computes conditional boundary crossing probabilities at futureplanned analyses for a given group sequential design assuming an interimz-statistic at a specified interim analysis. While gsCP() is designedtoward computing conditional power for a variety of underlying parametervalues, condPower is built to compute conditional power for avariety of interim test statistic values which is useful for sample sizeadaptation (see ssrCP). gsPP() averages conditionalpower across a posterior distribution to compute predictive power.gsPI() computes Bayesian prediction intervals for future analysescorresponding to results produced by gsPP(). gsPosterior()computes the posterior density for the group sequential design parameter ofinterest given a prior density and an interim outcome that is exact or in aninterval. gsPOS() computes the probability of success for a trialusing a prior distribution to average power over a set of thetavalues of interest. gsCPOS() assumes no boundary has been crossedbefore and including an interim analysis of interest, and computes theprobability of success based on this event. Note that gsCP() andgsPP() take only the interim test statistic into account in computingconditional probabilities, while gsCPOS() conditions on not crossingany bound through a specified interim analysis.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.For gsPP(), gsPI(), gsPOS() and gsCPOS(), theprior distribution for the standardized parameter theta () for agroup sequential design specified through a gsDesign object is specifiedthrough the arguments theta and wgts. This can be a discreteor a continuous probability density function. For a discrete function,generally all weights would be 1. For a continuous density, the wgtswould contain integration grid weights, such as those provided bynormalGrid.For gsPosterior, a prior distribution in prior must becomposed of the vectors z density. The vector zcontains points where the prior is evaluated and density thecorresponding density or, for a discrete distribution, the probabilities ofeach point in z. Densities may be supplied as fromnormalGrid() where grid weights for numerical integration aresupplied in gridwgts. If gridwgts are not supplied, they aredefaulted to 1 (equal weighting). To ensure a proper prior distribution, youmust have sum(gridwgts * density) equal to 1; this is NOT checked,however.<\/span><\/div>, <div class='tooltip'>gsPosterior<span class='tooltiptext'>gsCP() computes conditional boundary crossing probabilities at futureplanned analyses for a given group sequential design assuming an interimz-statistic at a specified interim analysis. While gsCP() is designedtoward computing conditional power for a variety of underlying parametervalues, condPower is built to compute conditional power for avariety of interim test statistic values which is useful for sample sizeadaptation (see ssrCP). gsPP() averages conditionalpower across a posterior distribution to compute predictive power.gsPI() computes Bayesian prediction intervals for future analysescorresponding to results produced by gsPP(). gsPosterior()computes the posterior density for the group sequential design parameter ofinterest given a prior density and an interim outcome that is exact or in aninterval. gsPOS() computes the probability of success for a trialusing a prior distribution to average power over a set of thetavalues of interest. gsCPOS() assumes no boundary has been crossedbefore and including an interim analysis of interest, and computes theprobability of success based on this event. Note that gsCP() andgsPP() take only the interim test statistic into account in computingconditional probabilities, while gsCPOS() conditions on not crossingany bound through a specified interim analysis.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.For gsPP(), gsPI(), gsPOS() and gsCPOS(), theprior distribution for the standardized parameter theta () for agroup sequential design specified through a gsDesign object is specifiedthrough the arguments theta and wgts. This can be a discreteor a continuous probability density function. For a discrete function,generally all weights would be 1. For a continuous density, the wgtswould contain integration grid weights, such as those provided bynormalGrid.For gsPosterior, a prior distribution in prior must becomposed of the vectors z density. The vector zcontains points where the prior is evaluated and density thecorresponding density or, for a discrete distribution, the probabilities ofeach point in z. Densities may be supplied as fromnormalGrid() where grid weights for numerical integration aresupplied in gridwgts. If gridwgts are not supplied, they aredefaulted to 1 (equal weighting). To ensure a proper prior distribution, youmust have sum(gridwgts * density) equal to 1; this is NOT checked,however.<\/span><\/div>, <div class='tooltip'>gsPP<span class='tooltiptext'>gsCP() computes conditional boundary crossing probabilities at futureplanned analyses for a given group sequential design assuming an interimz-statistic at a specified interim analysis. While gsCP() is designedtoward computing conditional power for a variety of underlying parametervalues, condPower is built to compute conditional power for avariety of interim test statistic values which is useful for sample sizeadaptation (see ssrCP). gsPP() averages conditionalpower across a posterior distribution to compute predictive power.gsPI() computes Bayesian prediction intervals for future analysescorresponding to results produced by gsPP(). gsPosterior()computes the posterior density for the group sequential design parameter ofinterest given a prior density and an interim outcome that is exact or in aninterval. gsPOS() computes the probability of success for a trialusing a prior distribution to average power over a set of thetavalues of interest. gsCPOS() assumes no boundary has been crossedbefore and including an interim analysis of interest, and computes theprobability of success based on this event. Note that gsCP() andgsPP() take only the interim test statistic into account in computingconditional probabilities, while gsCPOS() conditions on not crossingany bound through a specified interim analysis.See Conditional power section of manual for further clarification. See alsoMuller and Schaffer (2001) for background theory.For gsPP(), gsPI(), gsPOS() and gsCPOS(), theprior distribution for the standardized parameter theta () for agroup sequential design specified through a gsDesign object is specifiedthrough the arguments theta and wgts. This can be a discreteor a continuous probability density function. For a discrete function,generally all weights would be 1. For a continuous density, the wgtswould contain integration grid weights, such as those provided bynormalGrid.For gsPosterior, a prior distribution in prior must becomposed of the vectors z density. The vector zcontains points where the prior is evaluated and density thecorresponding density or, for a discrete distribution, the probabilities ofeach point in z. Densities may be supplied as fromnormalGrid() where grid weights for numerical integration aresupplied in gridwgts. If gridwgts are not supplied, they aredefaulted to 1 (equal weighting). To ensure a proper prior distribution, youmust have sum(gridwgts * density) equal to 1; this is NOT checked,however.<\/span><\/div>, <div class='tooltip'>gsProbability<span class='tooltiptext'>Computes power/Type I error and expected sample size for a group sequentialdesign across a selected set of parameter values for a given set of analysesand boundaries. The print function has been extended usingprint.gsProbability to print gsProbability objects; seeexamples.Depending on the calling sequence, an object of class gsProbabilityor class gsDesign is returned. If it is of class gsDesign thenthe members of the object will be the same as described ingsDesign. If d is input as NULL (the default),all other arguments (other than r) must be specified and an object ofclass gsProbability is returned. If d is passed as an objectof class gsProbability or gsDesign the only other argumentrequired is theta; the object returned has the same class as theinput d. On output, the values of theta input togsProbability will be the parameter values for which the design ischaracterized.<\/span><\/div>, <div class='tooltip'>gsRR<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>gsSurv<span class='tooltiptext'>nSurv() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint and an assumption of proportional hazards.This set of routines is new with version 2.7 and will continue to bemodified and refined to improve input error checking and output format withsubsequent versions. It allows both the Lachin and Foulkes (1986) method(fixed trial duration) as well as the Kim and Tsiatis(1990) method (fixedenrollment rates and either fixed enrollment duration or fixed minimumfollow-up). Piecewise exponential survival is supported as well as piecewiseconstant enrollment and dropout rates. The methods are for a 2-arm trialwith treatment groups referred to as experimental and control. A stratifiedpopulation is allowed as in Lachin and Foulkes (1986); this method has beenextended to derive non-inferiority as well as superiority trials.Stratification also allows power calculation for meta-analyses.gsSurv() combines nSurv() with gsDesign() to derive agroup sequential design for a study with a time-to-event endpoint.<\/span><\/div>, <div class='tooltip'>hGraph<span class='tooltiptext'>hGraph() plots a multiplicity graph defined by user inputs.The graph can also be used with the ***gMCP*** package to evaluate a set of nominal p-values for the tests of the hypotheses in the graph<\/span><\/div>, <div class='tooltip'>hrn2z<span class='tooltiptext'>nSurvival() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint. The Lachin and Foulkes (1986) method is used.nEvents uses the Schoenfeld (1981) approximation to provide samplesize and power in terms of the underlying hazard ratio and the number ofevents observed in a survival analysis. The functions hrz2n(),hrn2z() and zn2hr() also use the Schoenfeld approximation toprovide simple translations between hazard ratios, z-values and the numberof events in an analysis; input variables can be given as vectors.nSurvival() produces an object of class \"nSurvival\" with the numberof subjects and events for a set of pre-specified trial parameters, such asaccrual duration and follow-up period. The calculation is based on Lachinand Foulkes (1986) method and can be used for risk ratio or risk difference.The function also consider non-uniform (exponential) entry as well asuniform entry.If the logical approx is TRUE, the variance under alternativehypothesis is used to replace the variance under null hypothesis. Fornon-uniform entry, a non-zero value of gamma for exponential entrymust be supplied. For positive gamma, the entry distribution isconvex, whereas for negative gamma, the entry distribution isconcave.nEvents() uses the Schoenfeld (1981) method to approximate the numberof events n (given beta) or the power (given n).Arguments may be vectors or scalars, but any vectors must have the samelength.The functions hrz2n, hrn2z and zn2hr also all apply theSchoenfeld approximation for proportional hazards modeling. Thisapproximation is based on the asymptotic normal distribtuion of the logrankstatistic as well as related statistics are asymptotically normal. Let\\lambda denote the underlying hazard ratio (lambda1/lambda2 interms of the arguments to nSurvival). Further, let n denote thenumber of events observed when computing the statistic of interest andr the ratio of the sample size in an experimental group relative to acontrol. The estimated natural logarithm of the hazard ratio from aproportional hazards ratio is approximately normal with a mean oflog{\\lambda} and variance (1+r)^2/nr. Let z denote alogrank statistic (or a Wald statistic or score statistic from aproportional hazards regression model). The same asymptotic theory impliesz is asymptotically equivalent to a normalized estimate of the hazardratio \\lambda and thus z is asymptotically normal with variance1 and mean \\frac{log{\\lambda}r}{(1+r)^2}. Plugging the estimatedhazard ratio into the above equation allows approximating any one of thefollowing based on the other two: the estimate hazard ratio, the number ofevents and the z-statistic. That is, \\hat{\\lambda}=\\exp(z(1+r)/\\sqrt{rn}) z=\\log(\\hat{\\lambda})\\sqrt{nr}/(1+r) n=(z(1+r)/\\log(\\hat{\\lambda}))^2/r.hrz2n() translates an observed interim hazard ratio and interimz-value into the number of events required for the Z-value and hazard ratioto correspond to each other. hrn2z() translates a hazard ratio andnumber of events into an approximate corresponding Z-value. zn2hr()translates a Z-value and number of events into an approximate correspondinghazard ratio. Each of these functions has a default assumption of anunderlying hazard ratio of 1 which can be changed using the argumenthr0. hrn2z() and zn2hr() also have an argumenthr1 which is only used to compute the sign of the computed Z-value inthe case of hrn2z() and whether or not a z-value > 0 corresponds to ahazard ratio > or < the null hazard ratio hr0.<\/span><\/div>, <div class='tooltip'>hrz2n<span class='tooltiptext'>nSurvival() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint. The Lachin and Foulkes (1986) method is used.nEvents uses the Schoenfeld (1981) approximation to provide samplesize and power in terms of the underlying hazard ratio and the number ofevents observed in a survival analysis. The functions hrz2n(),hrn2z() and zn2hr() also use the Schoenfeld approximation toprovide simple translations between hazard ratios, z-values and the numberof events in an analysis; input variables can be given as vectors.nSurvival() produces an object of class \"nSurvival\" with the numberof subjects and events for a set of pre-specified trial parameters, such asaccrual duration and follow-up period. The calculation is based on Lachinand Foulkes (1986) method and can be used for risk ratio or risk difference.The function also consider non-uniform (exponential) entry as well asuniform entry.If the logical approx is TRUE, the variance under alternativehypothesis is used to replace the variance under null hypothesis. Fornon-uniform entry, a non-zero value of gamma for exponential entrymust be supplied. For positive gamma, the entry distribution isconvex, whereas for negative gamma, the entry distribution isconcave.nEvents() uses the Schoenfeld (1981) method to approximate the numberof events n (given beta) or the power (given n).Arguments may be vectors or scalars, but any vectors must have the samelength.The functions hrz2n, hrn2z and zn2hr also all apply theSchoenfeld approximation for proportional hazards modeling. Thisapproximation is based on the asymptotic normal distribtuion of the logrankstatistic as well as related statistics are asymptotically normal. Let\\lambda denote the underlying hazard ratio (lambda1/lambda2 interms of the arguments to nSurvival). Further, let n denote thenumber of events observed when computing the statistic of interest andr the ratio of the sample size in an experimental group relative to acontrol. The estimated natural logarithm of the hazard ratio from aproportional hazards ratio is approximately normal with a mean oflog{\\lambda} and variance (1+r)^2/nr. Let z denote alogrank statistic (or a Wald statistic or score statistic from aproportional hazards regression model). The same asymptotic theory impliesz is asymptotically equivalent to a normalized estimate of the hazardratio \\lambda and thus z is asymptotically normal with variance1 and mean \\frac{log{\\lambda}r}{(1+r)^2}. Plugging the estimatedhazard ratio into the above equation allows approximating any one of thefollowing based on the other two: the estimate hazard ratio, the number ofevents and the z-statistic. That is, \\hat{\\lambda}=\\exp(z(1+r)/\\sqrt{rn}) z=\\log(\\hat{\\lambda})\\sqrt{nr}/(1+r) n=(z(1+r)/\\log(\\hat{\\lambda}))^2/r.hrz2n() translates an observed interim hazard ratio and interimz-value into the number of events required for the Z-value and hazard ratioto correspond to each other. hrn2z() translates a hazard ratio andnumber of events into an approximate corresponding Z-value. zn2hr()translates a Z-value and number of events into an approximate correspondinghazard ratio. Each of these functions has a default assumption of anunderlying hazard ratio of 1 which can be changed using the argumenthr0. hrn2z() and zn2hr() also have an argumenthr1 which is only used to compute the sign of the computed Z-value inthe case of hrn2z() and whether or not a z-value > 0 corresponds to ahazard ratio > or < the null hazard ratio hr0.<\/span><\/div>, <div class='tooltip'>isInteger<span class='tooltiptext'>Utility functions to verify an objects's properties including whether it isa scalar or vector, the class, the length, and (if numeric) whether therange of values is on a specified interval. Additionally, thecheckLengths function can be used to ensure that all the suppliedinputs have equal lengths.isInteger is similar to is.integer except thatisInteger(1) returns TRUE whereas is.integer(1) returnsFALSE.checkScalar is used to verify that the input object is a scalar aswell as the other properties specified above.checkVector is used to verify that the input object is an atomicvector as well as the other properties as defined above.checkRange is used to check whether the numeric input object's valuesreside on the specified interval. If any of the values are outside thespecified interval, a FALSE is returned.checkLength is used to check whether all of the supplied inputs haveequal lengths.<\/span><\/div>, <div class='tooltip'>nBinomial<span class='tooltiptext'>Support is provided for sample size estimation, power, testing, confidenceintervals and simulation for fixed sample size trials (that is, not groupsequential or adaptive) with two arms and binary outcomes. Both superiorityand non-inferiority trials are considered. While all routines default tocomparisons of risk-difference, options to base computations on risk-ratioand odds-ratio are also included.nBinomial() computes sample size or power using the method ofFarrington and Manning (1990) for a trial to test the difference between twobinomial event rates. The routine can be used for a test of superiority ornon-inferiority. For a design that tests for superiority nBinomial()is consistent with the method of Fleiss, Tytun, and Ury (but without thecontinuity correction) to test for differences between event rates. Thisroutine is consistent with the Hmisc package routines bsamsize andbpower for superiority designs. Vector arguments allow computingsample sizes for multiple scenarios for comparative purposes.testBinomial() computes a Z- or Chi-square-statistic that comparestwo binomial event rates using the method of Miettinen and Nurminen (1980).This can be used for superiority or non-inferiority testing. Vectorarguments allow easy incorporation into simulation routines for fixed, groupsequential and adaptive designs.ciBinomial() computes confidence intervals for 1) the differencebetween two rates, 2) the risk-ratio for two rates or 3) the odds-ratio fortwo rates. This procedure provides inference that is consistent withtestBinomial() in that the confidence intervals are produced byinverting the testing procedures in testBinomial(). The Type I erroralpha input to ciBinomial is always interpreted as 2-sided.simBinomial() performs simulations to estimate the power for aMiettinen and Nurminen (1985) test comparing two binomial rates forsuperiority or non-inferiority. As noted in documentation forbpower.sim() in the HMisc package, by using testBinomial() youcan see that the formulas without any continuity correction are quiteaccurate. In fact, Type I error for a continuity-corrected test issignificantly lower (Gordon and Watson, 1996) than the nominal rate. Thus,as a default no continuity corrections are performed.varBinomial computes blinded estimates of the variance of theestimate of 1) event rate differences, 2) logarithm of the risk ratio, or 3)logarithm of the odds ratio. This is intended for blinded sample sizere-estimation for comparative trials with a binary outcome.Testing is 2-sided when a Chi-square statistic is used and 1-sided when aZ-statistic is used. Thus, these 2 options will produce substantiallydifferent results, in general. For non-inferiority, 1-sided testing isappropriate.You may wish to round sample sizes up using ceiling().Farrington and Manning (1990) begin with event rates p1 and p2under the alternative hypothesis and a difference between these rates underthe null hypothesis, delta0. From these values, actual rates underthe null hypothesis are computed, which are labeled p10 andp20 when outtype=3. The rates p1 and p2 are usedto compute a variance for a Z-test comparing rates under the alternativehypothesis, while p10 and p20 are used under the nullhypothesis. This computational method is also used to estimate variances invarBinomial() based on the overall event rate observed and the inputtreatment difference specified in delta0.Sample size with scale=\"Difference\" produces an error ifp1-p2=delta0. Normally, the alternative hypothesis underconsideration would be p1-p2-delta0$>0$. However, the alternative canhave p1-p2-delta0$<0$.<\/span><\/div>, <div class='tooltip'>nBinomial1Sample<span class='tooltiptext'>gsBinomialExact computes power/Type I error and expected sample sizefor a group sequential design in a single-arm trial with a binary outcome.This can also be used to compare event rates in two-arm studies. The printfunction has been extended using print.gsBinomialExact to printgsBinomialExact objects. Similarly, a plot function hasbeen extended using plot.gsBinomialExact to plotgsBinomialExact objects.binomialSPRT computes a truncated binomial sequential probabilityratio test (SPRT) which is a specific instance of an exact binomial groupsequential design for a single arm trial with a binary outcome.%\\code{gsBinomialPP} computes a truncated binomial (group) sequential designbased on predictive probability.nBinomial1Sample uses exact binomial calculations to compute powerand sample size for single arm binomial experiments.gsBinomialExact is based on the book \"Group Sequential Methods withApplications to Clinical Trials,\" Christopher Jennison and Bruce W.Turnbull, Chapter 12, Section 12.1.2 Exact Calculations for Binary Data.This computation is often used as an approximation for the distribution ofthe number of events in one treatment group out of all events when theprobability of an event is small and sample size is large.An object of class gsBinomialExact is returned. On output, the valuesof theta input to gsBinomialExact will be the parameter valuesfor which the boundary crossing probabilities and expected sample sizes arecomputed.Note that a[1] equal to -1 lower bound at n.I[1] means 0 successes continuesat interim 1; a[2]==0 at interim 2 means 0 successes stops trial forfutility at 2nd analysis. For final analysis, set a[k] equal to b[k]-1 toincorporate all possibilities into non-positive trial; see example.The sequential probability ratio test (SPRT) is a sequential testing schemeallowing testing after each observation. This likelihood ratio is used todetermine upper and lower cutoffs which are linear and parallel in thenumber of responses as a function of sample size. binomialSPRTproduces a variation the the SPRT that tests only within a range of samplesizes. While the linear SPRT bounds are continuous, actual bounds are theinteger number of response at or beyond each linear bound for each samplesize where testing is performed. Because of the truncation anddiscretization of the bounds, power and Type I error achieve will be lowerthan the nominal levels specified by alpha and beta which canbe altered to produce desired values that are achieved by the planned samplesize. See also example that shows computation of Type I error when futilitybound is considered non-binding.Note that if the objective of a design is to demonstrate that a rate (e.g.,failure rate) is lower than a certain level, two approaches can be taken.First, 1 minus the failure rate is the success rate and this can be used forplanning. Second, the role of beta becomes to express Type I errorand alpha is used to express Type II error.Plots produced include boundary plots, expected sample size, response rateat the boundary and power.gsBinomial1Sample uses exact binomial computations based on the baseR functions qbinom() and pbinom(). The tabular output may beconvenient for plotting. Note that input variables are largely not checked,so the user is largely responsible for results; it is a good idea to do arun with outtype=3 to check that you have done things appropriately.If n is not ordered (a bad idea) or not sequential (maybe OK), beaware of possible consequences.nBinomial1Sample is based on code from Marc Schwartz marc_schwartz@me.com. The possible sample size vector n needs to be selected in such a fashionthat it covers the possible range of values that include the true minima. NOTE: the one-sided evaluation of significance is more conservative than using the 2-sided exact test in binom.test.<\/span><\/div>, <div class='tooltip'>nEvents<span class='tooltiptext'>nSurvival() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint. The Lachin and Foulkes (1986) method is used.nEvents uses the Schoenfeld (1981) approximation to provide samplesize and power in terms of the underlying hazard ratio and the number ofevents observed in a survival analysis. The functions hrz2n(),hrn2z() and zn2hr() also use the Schoenfeld approximation toprovide simple translations between hazard ratios, z-values and the numberof events in an analysis; input variables can be given as vectors.nSurvival() produces an object of class \"nSurvival\" with the numberof subjects and events for a set of pre-specified trial parameters, such asaccrual duration and follow-up period. The calculation is based on Lachinand Foulkes (1986) method and can be used for risk ratio or risk difference.The function also consider non-uniform (exponential) entry as well asuniform entry.If the logical approx is TRUE, the variance under alternativehypothesis is used to replace the variance under null hypothesis. Fornon-uniform entry, a non-zero value of gamma for exponential entrymust be supplied. For positive gamma, the entry distribution isconvex, whereas for negative gamma, the entry distribution isconcave.nEvents() uses the Schoenfeld (1981) method to approximate the numberof events n (given beta) or the power (given n).Arguments may be vectors or scalars, but any vectors must have the samelength.The functions hrz2n, hrn2z and zn2hr also all apply theSchoenfeld approximation for proportional hazards modeling. Thisapproximation is based on the asymptotic normal distribtuion of the logrankstatistic as well as related statistics are asymptotically normal. Let\\lambda denote the underlying hazard ratio (lambda1/lambda2 interms of the arguments to nSurvival). Further, let n denote thenumber of events observed when computing the statistic of interest andr the ratio of the sample size in an experimental group relative to acontrol. The estimated natural logarithm of the hazard ratio from aproportional hazards ratio is approximately normal with a mean oflog{\\lambda} and variance (1+r)^2/nr. Let z denote alogrank statistic (or a Wald statistic or score statistic from aproportional hazards regression model). The same asymptotic theory impliesz is asymptotically equivalent to a normalized estimate of the hazardratio \\lambda and thus z is asymptotically normal with variance1 and mean \\frac{log{\\lambda}r}{(1+r)^2}. Plugging the estimatedhazard ratio into the above equation allows approximating any one of thefollowing based on the other two: the estimate hazard ratio, the number ofevents and the z-statistic. That is, \\hat{\\lambda}=\\exp(z(1+r)/\\sqrt{rn}) z=\\log(\\hat{\\lambda})\\sqrt{nr}/(1+r) n=(z(1+r)/\\log(\\hat{\\lambda}))^2/r.hrz2n() translates an observed interim hazard ratio and interimz-value into the number of events required for the Z-value and hazard ratioto correspond to each other. hrn2z() translates a hazard ratio andnumber of events into an approximate corresponding Z-value. zn2hr()translates a Z-value and number of events into an approximate correspondinghazard ratio. Each of these functions has a default assumption of anunderlying hazard ratio of 1 which can be changed using the argumenthr0. hrn2z() and zn2hr() also have an argumenthr1 which is only used to compute the sign of the computed Z-value inthe case of hrn2z() and whether or not a z-value > 0 corresponds to ahazard ratio > or < the null hazard ratio hr0.<\/span><\/div>, <div class='tooltip'>nEventsIA<span class='tooltiptext'>nSurv() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint and an assumption of proportional hazards.This set of routines is new with version 2.7 and will continue to bemodified and refined to improve input error checking and output format withsubsequent versions. It allows both the Lachin and Foulkes (1986) method(fixed trial duration) as well as the Kim and Tsiatis(1990) method (fixedenrollment rates and either fixed enrollment duration or fixed minimumfollow-up). Piecewise exponential survival is supported as well as piecewiseconstant enrollment and dropout rates. The methods are for a 2-arm trialwith treatment groups referred to as experimental and control. A stratifiedpopulation is allowed as in Lachin and Foulkes (1986); this method has beenextended to derive non-inferiority as well as superiority trials.Stratification also allows power calculation for meta-analyses.gsSurv() combines nSurv() with gsDesign() to derive agroup sequential design for a study with a time-to-event endpoint.<\/span><\/div>, <div class='tooltip'>nNormal<span class='tooltiptext'>nNormal() computes a fixed design sample size for comparing 2 meanswhere variance is known. T The function allows computation of sample sizefor a non-inferiority hypothesis. Note that you may wish to investigate other R packages such as the pwr package which uses the t-distribution.In the examples below we show how to set up a 2-arm group sequential design with a normal outcome. nNormal() computes sample size for comparing two normal means whenthe variance for observations in<\/span><\/div>, <div class='tooltip'>normalGrid<span class='tooltiptext'>normalGrid() is intended to be used for computation of the expected value ofa function of a normal random variable. The function produces grid pointsand weights to be used for numerical integration.This is a utility function to provide a normal density function and a gridto integrate over as described by Jennison and Turnbull (2000), Chapter 19.While integration can be performed over the real line or over any portion ofit, the numerical integration does not extend beyond 6 standard deviationsfrom the mean. The grid used for integration uses equally spaced points overthe middle of the distribution function, and spreads points further apart inthe tails. The values returned in gridwgts may be used to integrateany function over the given grid, although the user should take care thatthe function integrated is not large in the tails of the grid where pointsare spread further apart.<\/span><\/div>, <div class='tooltip'>nSurv<span class='tooltiptext'>nSurv() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint and an assumption of proportional hazards.This set of routines is new with version 2.7 and will continue to bemodified and refined to improve input error checking and output format withsubsequent versions. It allows both the Lachin and Foulkes (1986) method(fixed trial duration) as well as the Kim and Tsiatis(1990) method (fixedenrollment rates and either fixed enrollment duration or fixed minimumfollow-up). Piecewise exponential survival is supported as well as piecewiseconstant enrollment and dropout rates. The methods are for a 2-arm trialwith treatment groups referred to as experimental and control. A stratifiedpopulation is allowed as in Lachin and Foulkes (1986); this method has beenextended to derive non-inferiority as well as superiority trials.Stratification also allows power calculation for meta-analyses.gsSurv() combines nSurv() with gsDesign() to derive agroup sequential design for a study with a time-to-event endpoint.<\/span><\/div>, <div class='tooltip'>nSurvival<span class='tooltiptext'>nSurvival() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint. The Lachin and Foulkes (1986) method is used.nEvents uses the Schoenfeld (1981) approximation to provide samplesize and power in terms of the underlying hazard ratio and the number ofevents observed in a survival analysis. The functions hrz2n(),hrn2z() and zn2hr() also use the Schoenfeld approximation toprovide simple translations between hazard ratios, z-values and the numberof events in an analysis; input variables can be given as vectors.nSurvival() produces an object of class \"nSurvival\" with the numberof subjects and events for a set of pre-specified trial parameters, such asaccrual duration and follow-up period. The calculation is based on Lachinand Foulkes (1986) method and can be used for risk ratio or risk difference.The function also consider non-uniform (exponential) entry as well asuniform entry.If the logical approx is TRUE, the variance under alternativehypothesis is used to replace the variance under null hypothesis. Fornon-uniform entry, a non-zero value of gamma for exponential entrymust be supplied. For positive gamma, the entry distribution isconvex, whereas for negative gamma, the entry distribution isconcave.nEvents() uses the Schoenfeld (1981) method to approximate the numberof events n (given beta) or the power (given n).Arguments may be vectors or scalars, but any vectors must have the samelength.The functions hrz2n, hrn2z and zn2hr also all apply theSchoenfeld approximation for proportional hazards modeling. Thisapproximation is based on the asymptotic normal distribtuion of the logrankstatistic as well as related statistics are asymptotically normal. Let\\lambda denote the underlying hazard ratio (lambda1/lambda2 interms of the arguments to nSurvival). Further, let n denote thenumber of events observed when computing the statistic of interest andr the ratio of the sample size in an experimental group relative to acontrol. The estimated natural logarithm of the hazard ratio from aproportional hazards ratio is approximately normal with a mean oflog{\\lambda} and variance (1+r)^2/nr. Let z denote alogrank statistic (or a Wald statistic or score statistic from aproportional hazards regression model). The same asymptotic theory impliesz is asymptotically equivalent to a normalized estimate of the hazardratio \\lambda and thus z is asymptotically normal with variance1 and mean \\frac{log{\\lambda}r}{(1+r)^2}. Plugging the estimatedhazard ratio into the above equation allows approximating any one of thefollowing based on the other two: the estimate hazard ratio, the number ofevents and the z-statistic. That is, \\hat{\\lambda}=\\exp(z(1+r)/\\sqrt{rn}) z=\\log(\\hat{\\lambda})\\sqrt{nr}/(1+r) n=(z(1+r)/\\log(\\hat{\\lambda}))^2/r.hrz2n() translates an observed interim hazard ratio and interimz-value into the number of events required for the Z-value and hazard ratioto correspond to each other. hrn2z() translates a hazard ratio andnumber of events into an approximate corresponding Z-value. zn2hr()translates a Z-value and number of events into an approximate correspondinghazard ratio. Each of these functions has a default assumption of anunderlying hazard ratio of 1 which can be changed using the argumenthr0. hrn2z() and zn2hr() also have an argumenthr1 which is only used to compute the sign of the computed Z-value inthe case of hrn2z() and whether or not a z-value > 0 corresponds to ahazard ratio > or < the null hazard ratio hr0.<\/span><\/div>, <div class='tooltip'>Power.ssrCP<span class='tooltiptext'>ssrCP() adapts 2-stage group sequential designs to 2-stage samplesize re-estimation designs based on interim analysis conditional power. Thisis a simple case of designs developed by Lehmacher and Wassmer, Biometrics(1999). The conditional power designs of Bauer and Kohne (1994), Proschan and Hunsberger (1995), Cui, Hung and Wang (1999) and Liu and Chi (2001), Gao, Ware and Mehta (2008), and Mehta and Pocock (2011). Either the estimatedtreatment effect at the interim analysis or any chosen effect size can beused for sample size re-estimation. If not done carefully, these designs canbe very inefficient. It is probably a good idea to compare any design to asimpler group sequential design; see, for example, Jennison and Turnbull(2003). The a assumes a small Type I error is included with the interimanalysis and that the design is an adaptation from a 2-stage groupsequential design Related functions include 3 pre-defined combination testfunctions (z2NC, z2Z, z2Fisher) that represent theinverse normal combination test (Lehmacher and Wassmer, 1999), thesufficient statistic for the complete data, and Fisher's combination test.Power.ssrCP computes unconditional power for a conditional powerdesign derived by ssrCP.condPower is a supportive routine that also is interesting in its ownright; it computes conditional power of a combination test given an interimtest statistic, stage 2 sample size and combination test statistic. Whilethe returned data frame should make general plotting easy, the functionplot.ssrCP() prints a plot of study sample size by stage 1 outcomewith multiple x-axis labels for stage 1 z-value, conditional power, andstage 1 effect size relative to the effect size for which the underlyinggroup sequential design was powered.Sample size re-estimation using conditional power and an interim estimate oftreatment effect was proposed by several authors in the 1990's (seereferences below). Statistical testing for these original methods was basedon combination tests since Type I error could be inflated by using asufficient statistic for testing at the end of the trial. Since 2000, moreefficient variations of these conditional power designs have been developed.Fully optimized designs have also been derived (Posch et all, 2003,Lokhnygina and Tsiatis, 2008). Some of the later conditional power methodsallow use of sufficient statistics for testing (Chen, DeMets and Lan, 2004,Gao, Ware and Mehta, 2008, and Mehta and Pocock, 2011).The methods considered here are extensions of 2-stage group sequentialdesigns that include both an efficacy and a futility bound at the plannedinterim analysis. A maximum fold-increase in sample size (maxinc)fromthe supplied group sequential design (x) is specified by the user, aswell as a range of conditional power (cpadj) where sample size shouldbe re-estimated at the interim analysis, 1-the targeted conditional power tobe used for sample size re-estimation (beta) and a combination teststatistic (z2) to be used for testing at the end of the trial. Theinput value overrun represents incremental enrollment not included inthe interim analysis that is not included in the analysis; this is used forcalculating the required number of patients enrolled to complete the trial.Whereas most of the methods proposed have been based on using the interimestimated treatment effect size (default for ssrCP), the variabletheta allows the user to specify an alternative; Liu and Chi (2001)suggest that using the parameter value for which the trial was originallypowered is a good choice.<\/span><\/div>, <div class='tooltip'>sequentialPValue<span class='tooltiptext'>sequentialPValue computes a sequential p-value for a group sequential design using a spending function as described in Maurer and Bretz (2013) and previously defined by Liu and Anderson (2008).It is the minimum of repeated p-values computed at each analysis (Jennison and Turnbull, 2000).This is particularly useful for multiplicity methods such as the graphical method for group sequential designs where sequential p-values for multiple hypotheses can be used as nominal p-values to plug into a multiplicity graph. A sequential p-value is described as the minimum alpha level at which a one-sided group sequential bound would be rejected given interim and final observed results.It is meaningful for both one-sided designs and designs with non-binding futility bounds (test.type 1, 4, 6), but not for 2-sided designs with binding futility bounds (test.type 2, 3 or 5).Mild restrictions are required on spending functions used, but these are satisfied for commonly used spending functionssuch as the Lan-DeMets spending function approximating an O'Brien-Fleming bound or a Hwang-Shih-DeCani spending function; see Maurer and Bretz (2013).<\/span><\/div>, <div class='tooltip'>sfBetaDist<span class='tooltiptext'>The functions sfLogistic(), sfNormal(),sfExtremeValue(), sfExtremeValue2(), sfCauchy(), andsfBetaDist() are all 2-parameter spending function families. Theseprovide increased flexibility in some situations where the flexibility of aone-parameter spending function family is not sufficient. These functionsall allow fitting of two points on a cumulative spending function curve; inthis case, four parameters are specified indicating an x and a y coordinatefor each of 2 points. Normally each of these functions will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated in the examples; note, however, that anautomatic \\alphaalpha- and \\betabeta-spending function plotis also available.sfBetaDist(alpha,t,param) is simply alpha times the incompletebeta cumulative distribution function with parameters a and bpassed in param evaluated at values passed in t.The other spending functions take the form f(t;\\alpha,a,b)=\\alphaF(a+bF^{-1}(t))f(t;alpha,a,b)=alpha F(a+bF^{-1}(t)) where F() is acumulative distribution function with values > 0 on the real line(logistic for sfLogistic(), normal for sfNormal(), extremevalue for sfExtremeValue() and Cauchy for sfCauchy()) andF^{-1}() is its inverse.For the logistic spending function this simplifies tof(t;\\alpha,a,b)=\\alpha (1-(1+e^a(t/(1-t))^b)^{-1}).For the extreme value distribution with F(x)=\\exp(-\\exp(-x)) thissimplifies to f(t;\\alpha,a,b)=\\alpha \\exp(-e^a (-\\ln t)^b). Since theextreme value distribution is not symmetric, there is also a version wherethe standard distribution is flipped about 0. This is reflected insfExtremeValue2() where F(x)=1-\\exp(-\\exp(x)).<\/span><\/div>, <div class='tooltip'>sfCauchy<span class='tooltiptext'>The functions sfLogistic(), sfNormal(),sfExtremeValue(), sfExtremeValue2(), sfCauchy(), andsfBetaDist() are all 2-parameter spending function families. Theseprovide increased flexibility in some situations where the flexibility of aone-parameter spending function family is not sufficient. These functionsall allow fitting of two points on a cumulative spending function curve; inthis case, four parameters are specified indicating an x and a y coordinatefor each of 2 points. Normally each of these functions will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated in the examples; note, however, that anautomatic \\alphaalpha- and \\betabeta-spending function plotis also available.sfBetaDist(alpha,t,param) is simply alpha times the incompletebeta cumulative distribution function with parameters a and bpassed in param evaluated at values passed in t.The other spending functions take the form f(t;\\alpha,a,b)=\\alphaF(a+bF^{-1}(t))f(t;alpha,a,b)=alpha F(a+bF^{-1}(t)) where F() is acumulative distribution function with values > 0 on the real line(logistic for sfLogistic(), normal for sfNormal(), extremevalue for sfExtremeValue() and Cauchy for sfCauchy()) andF^{-1}() is its inverse.For the logistic spending function this simplifies tof(t;\\alpha,a,b)=\\alpha (1-(1+e^a(t/(1-t))^b)^{-1}).For the extreme value distribution with F(x)=\\exp(-\\exp(-x)) thissimplifies to f(t;\\alpha,a,b)=\\alpha \\exp(-e^a (-\\ln t)^b). Since theextreme value distribution is not symmetric, there is also a version wherethe standard distribution is flipped about 0. This is reflected insfExtremeValue2() where F(x)=1-\\exp(-\\exp(x)).<\/span><\/div>, <div class='tooltip'>sfExponential<span class='tooltiptext'>The function sfExponential implements the exponential spendingfunction (Anderson and Clark, 2009). Normally sfExponential will bepassed to gsDesign in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated below in examples.An exponential spending function is defined for any positive nu and0\\le t\\le 1 asf(t;\\alpha,\\nu)=\\alpha(t)=\\alpha^{t^{-\\nu}}.f(t;alpha,nu)=alpha^(t^(-nu)).A value of nu=0.8 approximates an O'Brien-Fleming spending functionwell.The general class of spending functions this family is derived from requiresa continuously increasing cumulative distribution function defined forx>0 and is defined as f(t;\\alpha,\\nu)=1-F\\left(F^{-1}(1-\\alpha)/ t^\\nu\\right).% f(t; alpha,nu)=1-F(F^(-1)(1-alpha)/ t^nu). The exponential spending function can bederived by letting F(x)=1-\\exp(-x), the exponential cumulativedistribution function. This function was derived as a generalization of theLan-DeMets (1983) spending function used to approximate an O'Brien-Flemingspending function (sfLDOF()), f(t; \\alpha)=2-2\\Phi \\left(\\Phi^{-1}(1-\\alpha/2)/ t^{1/2} \\right).% f(t;alpha)=2-2*Phi(Phi^(-1)(1-alpha/2)/t^(1/2)).<\/span><\/div>, <div class='tooltip'>sfExtremeValue<span class='tooltiptext'>The functions sfLogistic(), sfNormal(),sfExtremeValue(), sfExtremeValue2(), sfCauchy(), andsfBetaDist() are all 2-parameter spending function families. Theseprovide increased flexibility in some situations where the flexibility of aone-parameter spending function family is not sufficient. These functionsall allow fitting of two points on a cumulative spending function curve; inthis case, four parameters are specified indicating an x and a y coordinatefor each of 2 points. Normally each of these functions will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated in the examples; note, however, that anautomatic \\alphaalpha- and \\betabeta-spending function plotis also available.sfBetaDist(alpha,t,param) is simply alpha times the incompletebeta cumulative distribution function with parameters a and bpassed in param evaluated at values passed in t.The other spending functions take the form f(t;\\alpha,a,b)=\\alphaF(a+bF^{-1}(t))f(t;alpha,a,b)=alpha F(a+bF^{-1}(t)) where F() is acumulative distribution function with values > 0 on the real line(logistic for sfLogistic(), normal for sfNormal(), extremevalue for sfExtremeValue() and Cauchy for sfCauchy()) andF^{-1}() is its inverse.For the logistic spending function this simplifies tof(t;\\alpha,a,b)=\\alpha (1-(1+e^a(t/(1-t))^b)^{-1}).For the extreme value distribution with F(x)=\\exp(-\\exp(-x)) thissimplifies to f(t;\\alpha,a,b)=\\alpha \\exp(-e^a (-\\ln t)^b). Since theextreme value distribution is not symmetric, there is also a version wherethe standard distribution is flipped about 0. This is reflected insfExtremeValue2() where F(x)=1-\\exp(-\\exp(x)).<\/span><\/div>, <div class='tooltip'>sfExtremeValue2<span class='tooltiptext'>The functions sfLogistic(), sfNormal(),sfExtremeValue(), sfExtremeValue2(), sfCauchy(), andsfBetaDist() are all 2-parameter spending function families. Theseprovide increased flexibility in some situations where the flexibility of aone-parameter spending function family is not sufficient. These functionsall allow fitting of two points on a cumulative spending function curve; inthis case, four parameters are specified indicating an x and a y coordinatefor each of 2 points. Normally each of these functions will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated in the examples; note, however, that anautomatic \\alphaalpha- and \\betabeta-spending function plotis also available.sfBetaDist(alpha,t,param) is simply alpha times the incompletebeta cumulative distribution function with parameters a and bpassed in param evaluated at values passed in t.The other spending functions take the form f(t;\\alpha,a,b)=\\alphaF(a+bF^{-1}(t))f(t;alpha,a,b)=alpha F(a+bF^{-1}(t)) where F() is acumulative distribution function with values > 0 on the real line(logistic for sfLogistic(), normal for sfNormal(), extremevalue for sfExtremeValue() and Cauchy for sfCauchy()) andF^{-1}() is its inverse.For the logistic spending function this simplifies tof(t;\\alpha,a,b)=\\alpha (1-(1+e^a(t/(1-t))^b)^{-1}).For the extreme value distribution with F(x)=\\exp(-\\exp(-x)) thissimplifies to f(t;\\alpha,a,b)=\\alpha \\exp(-e^a (-\\ln t)^b). Since theextreme value distribution is not symmetric, there is also a version wherethe standard distribution is flipped about 0. This is reflected insfExtremeValue2() where F(x)=1-\\exp(-\\exp(x)).<\/span><\/div>, <div class='tooltip'>sfGapped<span class='tooltiptext'>The functions sfTruncated() and sfTrimmed apply any otherspending function over a restricted range. This allows eliminating spendingfor early interim analyses when you desire not to stop for the bound beingspecified; this is usually applied to eliminate early tests for a positiveefficacy finding. The truncation can come late in the trial if you desire tostop a trial any time after, say, 90 percent of information is available andan analysis is performed. This allows full Type I error spending if thefinal analysis occurs early. Both functions set cumulative spending to 0below a 'spending interval' in the interval [0,1], and set cumulativespending to 1 above this range. sfTrimmed() otherwise does not changean input spending function that is specified; probably the preferred andmore intuitive method in most cases. sfTruncated() resets the timescale on which the input spending function is computed to the 'spendinginterval.'sfGapped() allows elimination of analyses after some time point inthe trial; see details and examples.sfTrimmed simply computes the value of the input spending functionand parameters in the sub-range of [0,1], sets spending to 0 below thisrange and sets spending to 1 above this range.sfGapped spends outside of the range provided in trange. Belowtrange, the input spending function is used. Above trange, full spending isused; i.e., the first analysis performed above the interval in trange is thefinal analysis. As long as the input spending function is strictlyincreasing, this means that the first interim in the interval trange is thefinal interim analysis for the bound being specified.sfTruncated compresses spending into a sub-range of [0,1]. Theparameter param$trange specifies the range over which spending is tooccur. Within this range, spending is spent according to the spendingfunction specified in param$sf along with the corresponding spendingfunction parameter(s) in param$param. See example usingsfLinear that spends uniformly over specified range.<\/span><\/div>, <div class='tooltip'>sfHSD<span class='tooltiptext'>The function sfHSD implements a Hwang-Shih-DeCani spending function.This is the default spending function for gsDesign(). Normally itwill be passed to gsDesign in the parameter sfu for the upperbound or sfl for the lower bound to specify a spending functionfamily for a design. In this case, the user does not need to know thecalling sequence. The calling sequence is useful, however, when the userwishes to plot a spending function as demonstrated below in examples.A Hwang-Shih-DeCani spending function takes the form f(t;\\alpha,\\gamma)=\\alpha(1-e^{-\\gamma t})/(1-e^{-\\gamma})f(t; alpha, gamma) = alpha* (1-exp(-gamma * t))/(1 - exp(-gamma)) where \\gammagamma is thevalue passed in param. A value of \\gamma=-4gamma=-4 is usedto approximate an O'Brien-Fleming design (see sfExponentialfor a better fit), while a value of \\gamma=1gamma=1 approximates aPocock design well.<\/span><\/div>, <div class='tooltip'>sfLDOF<span class='tooltiptext'>Lan and DeMets (1983) first published the method of using spending functions to set boundaries for group sequential trials. In this publication theyproposed two specific spending functions: one to approximate anO'Brien-Fleming design and the other to approximate a Pocock design.The spending function to approximate O'Brien-Fleming has been generalized as proposed by Liu, et al (2012)With param=1=rho, the Lan-DeMets (1983) spending function to approximate an O'Brien-Flemingbound is implemented in the function (sfLDOF()): f(t; \\alpha)=2-2\\Phi\\left(\\Phi^{-1}(1-\\alpha/2)/ t^{\\rho/2}\\right).%f(t; alpha)=2-2*Phi(Phi^(-1)(1-alpha/2)/t^(rho/2)\\right)For rho otherwise in [.005,2], this is the generalized version of Liu et al (2012).For param outside of [.005,2], rho is set to 1. The Lan-DeMets (1983)spending function to approximate a Pocock design is implemented in thefunction sfLDPocock():f(t;\\alpha)=ln(1+(e-1)t).f(t;alpha)=ln(1+(e-1)t). As shown inexamples below, other spending functions can be used to get as good orbetter approximations to Pocock and O'Brien-Fleming bounds. In particular,O'Brien-Fleming bounds can be closely approximated usingsfExponential.<\/span><\/div>, <div class='tooltip'>sfLDPocock<span class='tooltiptext'>Lan and DeMets (1983) first published the method of using spending functions to set boundaries for group sequential trials. In this publication theyproposed two specific spending functions: one to approximate anO'Brien-Fleming design and the other to approximate a Pocock design.The spending function to approximate O'Brien-Fleming has been generalized as proposed by Liu, et al (2012)With param=1=rho, the Lan-DeMets (1983) spending function to approximate an O'Brien-Flemingbound is implemented in the function (sfLDOF()): f(t; \\alpha)=2-2\\Phi\\left(\\Phi^{-1}(1-\\alpha/2)/ t^{\\rho/2}\\right).%f(t; alpha)=2-2*Phi(Phi^(-1)(1-alpha/2)/t^(rho/2)\\right)For rho otherwise in [.005,2], this is the generalized version of Liu et al (2012).For param outside of [.005,2], rho is set to 1. The Lan-DeMets (1983)spending function to approximate a Pocock design is implemented in thefunction sfLDPocock():f(t;\\alpha)=ln(1+(e-1)t).f(t;alpha)=ln(1+(e-1)t). As shown inexamples below, other spending functions can be used to get as good orbetter approximations to Pocock and O'Brien-Fleming bounds. In particular,O'Brien-Fleming bounds can be closely approximated usingsfExponential.<\/span><\/div>, <div class='tooltip'>sfLinear<span class='tooltiptext'>The function sfLinear() allows specification of a piecewise linearspending function. The function sfStep() specifies a step functionspending function. Both functions provide complete flexibility in settingspending at desired timepoints in a group sequential design. Normally thesefunction will be passed to gsDesign() in the parameter sfu forthe upper bound or sfl for the lower bound to specify a spendingfunction family for a design. When passed to gsDesign(), the value ofparam would be passed to sfLinear() or sfStep() throughthe gsDesign() arguments sfupar for the upper bound andsflpar for the lower bound.Note that sfStep() allows setting a particular level of spending whenthe timing is not strictly known; an example shows how this can inflate TypeI error when timing of analyses are changed based on knowing the treatmenteffect at an interim.<\/span><\/div>, <div class='tooltip'>sfLogistic<span class='tooltiptext'>The functions sfLogistic(), sfNormal(),sfExtremeValue(), sfExtremeValue2(), sfCauchy(), andsfBetaDist() are all 2-parameter spending function families. Theseprovide increased flexibility in some situations where the flexibility of aone-parameter spending function family is not sufficient. These functionsall allow fitting of two points on a cumulative spending function curve; inthis case, four parameters are specified indicating an x and a y coordinatefor each of 2 points. Normally each of these functions will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated in the examples; note, however, that anautomatic \\alphaalpha- and \\betabeta-spending function plotis also available.sfBetaDist(alpha,t,param) is simply alpha times the incompletebeta cumulative distribution function with parameters a and bpassed in param evaluated at values passed in t.The other spending functions take the form f(t;\\alpha,a,b)=\\alphaF(a+bF^{-1}(t))f(t;alpha,a,b)=alpha F(a+bF^{-1}(t)) where F() is acumulative distribution function with values > 0 on the real line(logistic for sfLogistic(), normal for sfNormal(), extremevalue for sfExtremeValue() and Cauchy for sfCauchy()) andF^{-1}() is its inverse.For the logistic spending function this simplifies tof(t;\\alpha,a,b)=\\alpha (1-(1+e^a(t/(1-t))^b)^{-1}).For the extreme value distribution with F(x)=\\exp(-\\exp(-x)) thissimplifies to f(t;\\alpha,a,b)=\\alpha \\exp(-e^a (-\\ln t)^b). Since theextreme value distribution is not symmetric, there is also a version wherethe standard distribution is flipped about 0. This is reflected insfExtremeValue2() where F(x)=1-\\exp(-\\exp(x)).<\/span><\/div>, <div class='tooltip'>sfNormal<span class='tooltiptext'>The functions sfLogistic(), sfNormal(),sfExtremeValue(), sfExtremeValue2(), sfCauchy(), andsfBetaDist() are all 2-parameter spending function families. Theseprovide increased flexibility in some situations where the flexibility of aone-parameter spending function family is not sufficient. These functionsall allow fitting of two points on a cumulative spending function curve; inthis case, four parameters are specified indicating an x and a y coordinatefor each of 2 points. Normally each of these functions will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated in the examples; note, however, that anautomatic \\alphaalpha- and \\betabeta-spending function plotis also available.sfBetaDist(alpha,t,param) is simply alpha times the incompletebeta cumulative distribution function with parameters a and bpassed in param evaluated at values passed in t.The other spending functions take the form f(t;\\alpha,a,b)=\\alphaF(a+bF^{-1}(t))f(t;alpha,a,b)=alpha F(a+bF^{-1}(t)) where F() is acumulative distribution function with values > 0 on the real line(logistic for sfLogistic(), normal for sfNormal(), extremevalue for sfExtremeValue() and Cauchy for sfCauchy()) andF^{-1}() is its inverse.For the logistic spending function this simplifies tof(t;\\alpha,a,b)=\\alpha (1-(1+e^a(t/(1-t))^b)^{-1}).For the extreme value distribution with F(x)=\\exp(-\\exp(-x)) thissimplifies to f(t;\\alpha,a,b)=\\alpha \\exp(-e^a (-\\ln t)^b). Since theextreme value distribution is not symmetric, there is also a version wherethe standard distribution is flipped about 0. This is reflected insfExtremeValue2() where F(x)=1-\\exp(-\\exp(x)).<\/span><\/div>, <div class='tooltip'>sfPoints<span class='tooltiptext'>The function sfPoints implements a spending function with valuesspecified for an arbitrary set of specified points. It is now recommended touse sfLinear rather than sfPoints. Normally sfPoints will be passedto gsDesign in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence,just the points they wish to specify. If using sfPoints() in adesign, it is recommended to specify how to interpolate between thespecified points (e.g,, linear interpolation); also consider fitting smoothspending functions; see Spending_Function_Overview.<\/span><\/div>, <div class='tooltip'>sfPower<span class='tooltiptext'>The function sfPower() implements a Kim-DeMets (power) spendingfunction. This is a flexible, one-parameter spending function recommended byJennison and Turnbull (2000). Normally it will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated below in examples.A Kim-DeMets spending function takes the form f(t;\\alpha,\\rho)=\\alphat^\\rhof(t; alpha, rho) = alpha t^rho where \\rhorho is the valuepassed in param. See examples below for a range of values of\\rhorho that may be of interest (param=0.75 to 3 aredocumented there).<\/span><\/div>, <div class='tooltip'>sfStep<span class='tooltiptext'>The function sfLinear() allows specification of a piecewise linearspending function. The function sfStep() specifies a step functionspending function. Both functions provide complete flexibility in settingspending at desired timepoints in a group sequential design. Normally thesefunction will be passed to gsDesign() in the parameter sfu forthe upper bound or sfl for the lower bound to specify a spendingfunction family for a design. When passed to gsDesign(), the value ofparam would be passed to sfLinear() or sfStep() throughthe gsDesign() arguments sfupar for the upper bound andsflpar for the lower bound.Note that sfStep() allows setting a particular level of spending whenthe timing is not strictly known; an example shows how this can inflate TypeI error when timing of analyses are changed based on knowing the treatmenteffect at an interim.<\/span><\/div>, <div class='tooltip'>sfTDist<span class='tooltiptext'>The function sfTDist() provides perhaps the maximum flexibility amongspending functions provided in the gsDesign package. This functionallows fitting of three points on a cumulative spending function curve; inthis case, six parameters are specified indicating an x and a y coordinatefor each of 3 points. Normally this function will be passed togsDesign() in the parameter sfu for the upper bound orsfl for the lower bound to specify a spending function family for adesign. In this case, the user does not need to know the calling sequence.The calling sequence is useful, however, when the user wishes to plot aspending function as demonstrated below in examples.The t-distribution spending function takes the form f(t;\\alpha)=\\alphaF(a+bF^{-1}(t)) where F() is a cumulative t-distribution functionwith df degrees of freedom and F^{-1}() is its inverse.<\/span><\/div>, <div class='tooltip'>sfTrimmed<span class='tooltiptext'>The functions sfTruncated() and sfTrimmed apply any otherspending function over a restricted range. This allows eliminating spendingfor early interim analyses when you desire not to stop for the bound beingspecified; this is usually applied to eliminate early tests for a positiveefficacy finding. The truncation can come late in the trial if you desire tostop a trial any time after, say, 90 percent of information is available andan analysis is performed. This allows full Type I error spending if thefinal analysis occurs early. Both functions set cumulative spending to 0below a 'spending interval' in the interval [0,1], and set cumulativespending to 1 above this range. sfTrimmed() otherwise does not changean input spending function that is specified; probably the preferred andmore intuitive method in most cases. sfTruncated() resets the timescale on which the input spending function is computed to the 'spendinginterval.'sfGapped() allows elimination of analyses after some time point inthe trial; see details and examples.sfTrimmed simply computes the value of the input spending functionand parameters in the sub-range of [0,1], sets spending to 0 below thisrange and sets spending to 1 above this range.sfGapped spends outside of the range provided in trange. Belowtrange, the input spending function is used. Above trange, full spending isused; i.e., the first analysis performed above the interval in trange is thefinal analysis. As long as the input spending function is strictlyincreasing, this means that the first interim in the interval trange is thefinal interim analysis for the bound being specified.sfTruncated compresses spending into a sub-range of [0,1]. Theparameter param$trange specifies the range over which spending is tooccur. Within this range, spending is spent according to the spendingfunction specified in param$sf along with the corresponding spendingfunction parameter(s) in param$param. See example usingsfLinear that spends uniformly over specified range.<\/span><\/div>, <div class='tooltip'>sfTruncated<span class='tooltiptext'>The functions sfTruncated() and sfTrimmed apply any otherspending function over a restricted range. This allows eliminating spendingfor early interim analyses when you desire not to stop for the bound beingspecified; this is usually applied to eliminate early tests for a positiveefficacy finding. The truncation can come late in the trial if you desire tostop a trial any time after, say, 90 percent of information is available andan analysis is performed. This allows full Type I error spending if thefinal analysis occurs early. Both functions set cumulative spending to 0below a 'spending interval' in the interval [0,1], and set cumulativespending to 1 above this range. sfTrimmed() otherwise does not changean input spending function that is specified; probably the preferred andmore intuitive method in most cases. sfTruncated() resets the timescale on which the input spending function is computed to the 'spendinginterval.'sfGapped() allows elimination of analyses after some time point inthe trial; see details and examples.sfTrimmed simply computes the value of the input spending functionand parameters in the sub-range of [0,1], sets spending to 0 below thisrange and sets spending to 1 above this range.sfGapped spends outside of the range provided in trange. Belowtrange, the input spending function is used. Above trange, full spending isused; i.e., the first analysis performed above the interval in trange is thefinal analysis. As long as the input spending function is strictlyincreasing, this means that the first interim in the interval trange is thefinal interim analysis for the bound being specified.sfTruncated compresses spending into a sub-range of [0,1]. Theparameter param$trange specifies the range over which spending is tooccur. Within this range, spending is spent according to the spendingfunction specified in param$sf along with the corresponding spendingfunction parameter(s) in param$param. See example usingsfLinear that spends uniformly over specified range.<\/span><\/div>, <div class='tooltip'>simBinomial<span class='tooltiptext'>Support is provided for sample size estimation, power, testing, confidenceintervals and simulation for fixed sample size trials (that is, not groupsequential or adaptive) with two arms and binary outcomes. Both superiorityand non-inferiority trials are considered. While all routines default tocomparisons of risk-difference, options to base computations on risk-ratioand odds-ratio are also included.nBinomial() computes sample size or power using the method ofFarrington and Manning (1990) for a trial to test the difference between twobinomial event rates. The routine can be used for a test of superiority ornon-inferiority. For a design that tests for superiority nBinomial()is consistent with the method of Fleiss, Tytun, and Ury (but without thecontinuity correction) to test for differences between event rates. Thisroutine is consistent with the Hmisc package routines bsamsize andbpower for superiority designs. Vector arguments allow computingsample sizes for multiple scenarios for comparative purposes.testBinomial() computes a Z- or Chi-square-statistic that comparestwo binomial event rates using the method of Miettinen and Nurminen (1980).This can be used for superiority or non-inferiority testing. Vectorarguments allow easy incorporation into simulation routines for fixed, groupsequential and adaptive designs.ciBinomial() computes confidence intervals for 1) the differencebetween two rates, 2) the risk-ratio for two rates or 3) the odds-ratio fortwo rates. This procedure provides inference that is consistent withtestBinomial() in that the confidence intervals are produced byinverting the testing procedures in testBinomial(). The Type I erroralpha input to ciBinomial is always interpreted as 2-sided.simBinomial() performs simulations to estimate the power for aMiettinen and Nurminen (1985) test comparing two binomial rates forsuperiority or non-inferiority. As noted in documentation forbpower.sim() in the HMisc package, by using testBinomial() youcan see that the formulas without any continuity correction are quiteaccurate. In fact, Type I error for a continuity-corrected test issignificantly lower (Gordon and Watson, 1996) than the nominal rate. Thus,as a default no continuity corrections are performed.varBinomial computes blinded estimates of the variance of theestimate of 1) event rate differences, 2) logarithm of the risk ratio, or 3)logarithm of the odds ratio. This is intended for blinded sample sizere-estimation for comparative trials with a binary outcome.Testing is 2-sided when a Chi-square statistic is used and 1-sided when aZ-statistic is used. Thus, these 2 options will produce substantiallydifferent results, in general. For non-inferiority, 1-sided testing isappropriate.You may wish to round sample sizes up using ceiling().Farrington and Manning (1990) begin with event rates p1 and p2under the alternative hypothesis and a difference between these rates underthe null hypothesis, delta0. From these values, actual rates underthe null hypothesis are computed, which are labeled p10 andp20 when outtype=3. The rates p1 and p2 are usedto compute a variance for a Z-test comparing rates under the alternativehypothesis, while p10 and p20 are used under the nullhypothesis. This computational method is also used to estimate variances invarBinomial() based on the overall event rate observed and the inputtreatment difference specified in delta0.Sample size with scale=\"Difference\" produces an error ifp1-p2=delta0. Normally, the alternative hypothesis underconsideration would be p1-p2-delta0$>0$. However, the alternative canhave p1-p2-delta0$<0$.<\/span><\/div>, <div class='tooltip'>spendingFunction<span class='tooltiptext'>Spending functions are used to set boundaries for group sequential designs.Using the spending function approach to design offers a natural way toprovide interim testing boundaries when unplanned interim analyses are addedor when the timing of an interim analysis changes. Many standard andinvestigational spending functions are provided in the gsDesign package.These offer a great deal of flexibility in setting up stopping boundariesfor a design.<\/span><\/div>, <div class='tooltip'>ssrCP<span class='tooltiptext'>ssrCP() adapts 2-stage group sequential designs to 2-stage samplesize re-estimation designs based on interim analysis conditional power. Thisis a simple case of designs developed by Lehmacher and Wassmer, Biometrics(1999). The conditional power designs of Bauer and Kohne (1994), Proschan and Hunsberger (1995), Cui, Hung and Wang (1999) and Liu and Chi (2001), Gao, Ware and Mehta (2008), and Mehta and Pocock (2011). Either the estimatedtreatment effect at the interim analysis or any chosen effect size can beused for sample size re-estimation. If not done carefully, these designs canbe very inefficient. It is probably a good idea to compare any design to asimpler group sequential design; see, for example, Jennison and Turnbull(2003). The a assumes a small Type I error is included with the interimanalysis and that the design is an adaptation from a 2-stage groupsequential design Related functions include 3 pre-defined combination testfunctions (z2NC, z2Z, z2Fisher) that represent theinverse normal combination test (Lehmacher and Wassmer, 1999), thesufficient statistic for the complete data, and Fisher's combination test.Power.ssrCP computes unconditional power for a conditional powerdesign derived by ssrCP.condPower is a supportive routine that also is interesting in its ownright; it computes conditional power of a combination test given an interimtest statistic, stage 2 sample size and combination test statistic. Whilethe returned data frame should make general plotting easy, the functionplot.ssrCP() prints a plot of study sample size by stage 1 outcomewith multiple x-axis labels for stage 1 z-value, conditional power, andstage 1 effect size relative to the effect size for which the underlyinggroup sequential design was powered.Sample size re-estimation using conditional power and an interim estimate oftreatment effect was proposed by several authors in the 1990's (seereferences below). Statistical testing for these original methods was basedon combination tests since Type I error could be inflated by using asufficient statistic for testing at the end of the trial. Since 2000, moreefficient variations of these conditional power designs have been developed.Fully optimized designs have also been derived (Posch et all, 2003,Lokhnygina and Tsiatis, 2008). Some of the later conditional power methodsallow use of sufficient statistics for testing (Chen, DeMets and Lan, 2004,Gao, Ware and Mehta, 2008, and Mehta and Pocock, 2011).The methods considered here are extensions of 2-stage group sequentialdesigns that include both an efficacy and a futility bound at the plannedinterim analysis. A maximum fold-increase in sample size (maxinc)fromthe supplied group sequential design (x) is specified by the user, aswell as a range of conditional power (cpadj) where sample size shouldbe re-estimated at the interim analysis, 1-the targeted conditional power tobe used for sample size re-estimation (beta) and a combination teststatistic (z2) to be used for testing at the end of the trial. Theinput value overrun represents incremental enrollment not included inthe interim analysis that is not included in the analysis; this is used forcalculating the required number of patients enrolled to complete the trial.Whereas most of the methods proposed have been based on using the interimestimated treatment effect size (default for ssrCP), the variabletheta allows the user to specify an alternative; Liu and Chi (2001)suggest that using the parameter value for which the trial was originallypowered is a good choice.<\/span><\/div>, <div class='tooltip'>testBinomial<span class='tooltiptext'>Support is provided for sample size estimation, power, testing, confidenceintervals and simulation for fixed sample size trials (that is, not groupsequential or adaptive) with two arms and binary outcomes. Both superiorityand non-inferiority trials are considered. While all routines default tocomparisons of risk-difference, options to base computations on risk-ratioand odds-ratio are also included.nBinomial() computes sample size or power using the method ofFarrington and Manning (1990) for a trial to test the difference between twobinomial event rates. The routine can be used for a test of superiority ornon-inferiority. For a design that tests for superiority nBinomial()is consistent with the method of Fleiss, Tytun, and Ury (but without thecontinuity correction) to test for differences between event rates. Thisroutine is consistent with the Hmisc package routines bsamsize andbpower for superiority designs. Vector arguments allow computingsample sizes for multiple scenarios for comparative purposes.testBinomial() computes a Z- or Chi-square-statistic that comparestwo binomial event rates using the method of Miettinen and Nurminen (1980).This can be used for superiority or non-inferiority testing. Vectorarguments allow easy incorporation into simulation routines for fixed, groupsequential and adaptive designs.ciBinomial() computes confidence intervals for 1) the differencebetween two rates, 2) the risk-ratio for two rates or 3) the odds-ratio fortwo rates. This procedure provides inference that is consistent withtestBinomial() in that the confidence intervals are produced byinverting the testing procedures in testBinomial(). The Type I erroralpha input to ciBinomial is always interpreted as 2-sided.simBinomial() performs simulations to estimate the power for aMiettinen and Nurminen (1985) test comparing two binomial rates forsuperiority or non-inferiority. As noted in documentation forbpower.sim() in the HMisc package, by using testBinomial() youcan see that the formulas without any continuity correction are quiteaccurate. In fact, Type I error for a continuity-corrected test issignificantly lower (Gordon and Watson, 1996) than the nominal rate. Thus,as a default no continuity corrections are performed.varBinomial computes blinded estimates of the variance of theestimate of 1) event rate differences, 2) logarithm of the risk ratio, or 3)logarithm of the odds ratio. This is intended for blinded sample sizere-estimation for comparative trials with a binary outcome.Testing is 2-sided when a Chi-square statistic is used and 1-sided when aZ-statistic is used. Thus, these 2 options will produce substantiallydifferent results, in general. For non-inferiority, 1-sided testing isappropriate.You may wish to round sample sizes up using ceiling().Farrington and Manning (1990) begin with event rates p1 and p2under the alternative hypothesis and a difference between these rates underthe null hypothesis, delta0. From these values, actual rates underthe null hypothesis are computed, which are labeled p10 andp20 when outtype=3. The rates p1 and p2 are usedto compute a variance for a Z-test comparing rates under the alternativehypothesis, while p10 and p20 are used under the nullhypothesis. This computational method is also used to estimate variances invarBinomial() based on the overall event rate observed and the inputtreatment difference specified in delta0.Sample size with scale=\"Difference\" produces an error ifp1-p2=delta0. Normally, the alternative hypothesis underconsideration would be p1-p2-delta0$>0$. However, the alternative canhave p1-p2-delta0$<0$.<\/span><\/div>, <div class='tooltip'>tEventsIA<span class='tooltiptext'>nSurv() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint and an assumption of proportional hazards.This set of routines is new with version 2.7 and will continue to bemodified and refined to improve input error checking and output format withsubsequent versions. It allows both the Lachin and Foulkes (1986) method(fixed trial duration) as well as the Kim and Tsiatis(1990) method (fixedenrollment rates and either fixed enrollment duration or fixed minimumfollow-up). Piecewise exponential survival is supported as well as piecewiseconstant enrollment and dropout rates. The methods are for a 2-arm trialwith treatment groups referred to as experimental and control. A stratifiedpopulation is allowed as in Lachin and Foulkes (1986); this method has beenextended to derive non-inferiority as well as superiority trials.Stratification also allows power calculation for meta-analyses.gsSurv() combines nSurv() with gsDesign() to derive agroup sequential design for a study with a time-to-event endpoint.<\/span><\/div>, <div class='tooltip'>varBinomial<span class='tooltiptext'>Support is provided for sample size estimation, power, testing, confidenceintervals and simulation for fixed sample size trials (that is, not groupsequential or adaptive) with two arms and binary outcomes. Both superiorityand non-inferiority trials are considered. While all routines default tocomparisons of risk-difference, options to base computations on risk-ratioand odds-ratio are also included.nBinomial() computes sample size or power using the method ofFarrington and Manning (1990) for a trial to test the difference between twobinomial event rates. The routine can be used for a test of superiority ornon-inferiority. For a design that tests for superiority nBinomial()is consistent with the method of Fleiss, Tytun, and Ury (but without thecontinuity correction) to test for differences between event rates. Thisroutine is consistent with the Hmisc package routines bsamsize andbpower for superiority designs. Vector arguments allow computingsample sizes for multiple scenarios for comparative purposes.testBinomial() computes a Z- or Chi-square-statistic that comparestwo binomial event rates using the method of Miettinen and Nurminen (1980).This can be used for superiority or non-inferiority testing. Vectorarguments allow easy incorporation into simulation routines for fixed, groupsequential and adaptive designs.ciBinomial() computes confidence intervals for 1) the differencebetween two rates, 2) the risk-ratio for two rates or 3) the odds-ratio fortwo rates. This procedure provides inference that is consistent withtestBinomial() in that the confidence intervals are produced byinverting the testing procedures in testBinomial(). The Type I erroralpha input to ciBinomial is always interpreted as 2-sided.simBinomial() performs simulations to estimate the power for aMiettinen and Nurminen (1985) test comparing two binomial rates forsuperiority or non-inferiority. As noted in documentation forbpower.sim() in the HMisc package, by using testBinomial() youcan see that the formulas without any continuity correction are quiteaccurate. In fact, Type I error for a continuity-corrected test issignificantly lower (Gordon and Watson, 1996) than the nominal rate. Thus,as a default no continuity corrections are performed.varBinomial computes blinded estimates of the variance of theestimate of 1) event rate differences, 2) logarithm of the risk ratio, or 3)logarithm of the odds ratio. This is intended for blinded sample sizere-estimation for comparative trials with a binary outcome.Testing is 2-sided when a Chi-square statistic is used and 1-sided when aZ-statistic is used. Thus, these 2 options will produce substantiallydifferent results, in general. For non-inferiority, 1-sided testing isappropriate.You may wish to round sample sizes up using ceiling().Farrington and Manning (1990) begin with event rates p1 and p2under the alternative hypothesis and a difference between these rates underthe null hypothesis, delta0. From these values, actual rates underthe null hypothesis are computed, which are labeled p10 andp20 when outtype=3. The rates p1 and p2 are usedto compute a variance for a Z-test comparing rates under the alternativehypothesis, while p10 and p20 are used under the nullhypothesis. This computational method is also used to estimate variances invarBinomial() based on the overall event rate observed and the inputtreatment difference specified in delta0.Sample size with scale=\"Difference\" produces an error ifp1-p2=delta0. Normally, the alternative hypothesis underconsideration would be p1-p2-delta0$>0$. However, the alternative canhave p1-p2-delta0$<0$.<\/span><\/div>, <div class='tooltip'>xprint<span class='tooltiptext'>A tabular summary of a group sequential design's bounds and their propertiesare often useful. The 'vintage' print.gsDesign() function provides acomplete but minimally formatted summary of a group sequential designderived by gsDesign(). A brief description of the overall design canalso be useful (summary.gsDesign(). A tabular summary of boundarycharacteristics oriented only towards LaTeX output is produced byxtable.gsSurv. More flexibility is provided bygsBoundSummary() which produces a tabular summary of auser-specifiable set of package-provided boundary properties in a dataframe. This can also be used to along with functions such asprint.data.frame(), write.table(),write.csv(), write.csv2() or, from the RTFpackage, addTable.RTF() (from the rtf package) to produce console orR Markdown output or output to a variety of file types. xprint() isprovided for LaTeX output by setting default options forprint.xtable() when producing tables summarizing designbounds.Individual transformation of z-value test statistics for interim and finalanalyses are obtained from gsBValue(), gsDelta(),gsHR() and gsCPz() for B-values, approximate treatment effect(see details), approximate hazard ratio and conditional power, respectively.The print.gsDesign function is intended to provide an easier outputto review than is available from a simple list of all the output components.The gsBoundSummary function is intended to provide a summary ofboundary characteristics that is often useful for evaluating boundaryselection; this outputs an extension of the data.frame class thatsets up default printing without row names usingprint.gsBoundSummary. summary.gsDesign, on the other hand,provides a summary of the overall design at a higher level; this providescharacteristics not included in the gsBoundSummary summary and nodetail concerning interim analysis bounds.In brief, the computed descriptions of group sequential design bounds are asfollows: Z: Standardized normal test statistic at design bound.p (1-sided): 1-sided p-value for Z. This will be computed asthe probability of a greater EXCEPT for lower bound when a 2-sided design isbeing summarized.delta at bound: Approximate value of the natural parameter at thebound. The approximate standardized effect size at the bound is generallycomputed as Z/sqrt(n). Calling this theta, this is translatedto the delta using the values delta0 and delta1 fromthe input x by the formula delta0 +(delta1-delta0)/theta1*theta where theta1 is the alternatehypothesis value of the standardized parameter. Note that this value will beexponentiated in the case of relative risks, hazard ratios or when the userspecifies logdelta=TRUE. In the case of hazard ratios, the value iscomputed instead by gsHR() to be consistent withplot.gsDesign(). Similarly, the value is computed by gsRR()when the relative risk is the natural parameter.Spending: Incremental error spending at each given analysis. Forasymmetric designs, futility bound will have beta-spending summarized.Efficacy bound always has alpha-spending summarized.B-value: sqrt(t)*Z where t is the proportion ofinformation at the analysis divided by the final analysis plannedinformation. The expected value for B-values is directly proportional tot.CP: Conditional power under the estimated treatment differenceassuming the interim Z-statistic is at the study boundCP H1: Conditional power under the alternate hypothesis treatmenteffect assuming the interim test statistic is at the study bound.PP: Predictive power assuming the interim test statistic is at thestudy bound and the input prior distribution for the standardized effectsize. This is the conditional power averaged across the posteriordistribution for the treatment effect given the interim test statisticvalue. P{Cross if delta=xx}: For each of the parameter values inx, the probability of crossing either bound given that treatmenteffect is computed. This value is cumulative for each bound. For example,the probability of crossing the efficacy bound at or before the analysis ofinterest.<\/span><\/div>, <div class='tooltip'>xtable<span class='tooltiptext'>xtable<\/span><\/div>, <div class='tooltip'>z2Fisher<span class='tooltiptext'>ssrCP() adapts 2-stage group sequential designs to 2-stage samplesize re-estimation designs based on interim analysis conditional power. Thisis a simple case of designs developed by Lehmacher and Wassmer, Biometrics(1999). The conditional power designs of Bauer and Kohne (1994), Proschan and Hunsberger (1995), Cui, Hung and Wang (1999) and Liu and Chi (2001), Gao, Ware and Mehta (2008), and Mehta and Pocock (2011). Either the estimatedtreatment effect at the interim analysis or any chosen effect size can beused for sample size re-estimation. If not done carefully, these designs canbe very inefficient. It is probably a good idea to compare any design to asimpler group sequential design; see, for example, Jennison and Turnbull(2003). The a assumes a small Type I error is included with the interimanalysis and that the design is an adaptation from a 2-stage groupsequential design Related functions include 3 pre-defined combination testfunctions (z2NC, z2Z, z2Fisher) that represent theinverse normal combination test (Lehmacher and Wassmer, 1999), thesufficient statistic for the complete data, and Fisher's combination test.Power.ssrCP computes unconditional power for a conditional powerdesign derived by ssrCP.condPower is a supportive routine that also is interesting in its ownright; it computes conditional power of a combination test given an interimtest statistic, stage 2 sample size and combination test statistic. Whilethe returned data frame should make general plotting easy, the functionplot.ssrCP() prints a plot of study sample size by stage 1 outcomewith multiple x-axis labels for stage 1 z-value, conditional power, andstage 1 effect size relative to the effect size for which the underlyinggroup sequential design was powered.Sample size re-estimation using conditional power and an interim estimate oftreatment effect was proposed by several authors in the 1990's (seereferences below). Statistical testing for these original methods was basedon combination tests since Type I error could be inflated by using asufficient statistic for testing at the end of the trial. Since 2000, moreefficient variations of these conditional power designs have been developed.Fully optimized designs have also been derived (Posch et all, 2003,Lokhnygina and Tsiatis, 2008). Some of the later conditional power methodsallow use of sufficient statistics for testing (Chen, DeMets and Lan, 2004,Gao, Ware and Mehta, 2008, and Mehta and Pocock, 2011).The methods considered here are extensions of 2-stage group sequentialdesigns that include both an efficacy and a futility bound at the plannedinterim analysis. A maximum fold-increase in sample size (maxinc)fromthe supplied group sequential design (x) is specified by the user, aswell as a range of conditional power (cpadj) where sample size shouldbe re-estimated at the interim analysis, 1-the targeted conditional power tobe used for sample size re-estimation (beta) and a combination teststatistic (z2) to be used for testing at the end of the trial. Theinput value overrun represents incremental enrollment not included inthe interim analysis that is not included in the analysis; this is used forcalculating the required number of patients enrolled to complete the trial.Whereas most of the methods proposed have been based on using the interimestimated treatment effect size (default for ssrCP), the variabletheta allows the user to specify an alternative; Liu and Chi (2001)suggest that using the parameter value for which the trial was originallypowered is a good choice.<\/span><\/div>, <div class='tooltip'>z2NC<span class='tooltiptext'>ssrCP() adapts 2-stage group sequential designs to 2-stage samplesize re-estimation designs based on interim analysis conditional power. Thisis a simple case of designs developed by Lehmacher and Wassmer, Biometrics(1999). The conditional power designs of Bauer and Kohne (1994), Proschan and Hunsberger (1995), Cui, Hung and Wang (1999) and Liu and Chi (2001), Gao, Ware and Mehta (2008), and Mehta and Pocock (2011). Either the estimatedtreatment effect at the interim analysis or any chosen effect size can beused for sample size re-estimation. If not done carefully, these designs canbe very inefficient. It is probably a good idea to compare any design to asimpler group sequential design; see, for example, Jennison and Turnbull(2003). The a assumes a small Type I error is included with the interimanalysis and that the design is an adaptation from a 2-stage groupsequential design Related functions include 3 pre-defined combination testfunctions (z2NC, z2Z, z2Fisher) that represent theinverse normal combination test (Lehmacher and Wassmer, 1999), thesufficient statistic for the complete data, and Fisher's combination test.Power.ssrCP computes unconditional power for a conditional powerdesign derived by ssrCP.condPower is a supportive routine that also is interesting in its ownright; it computes conditional power of a combination test given an interimtest statistic, stage 2 sample size and combination test statistic. Whilethe returned data frame should make general plotting easy, the functionplot.ssrCP() prints a plot of study sample size by stage 1 outcomewith multiple x-axis labels for stage 1 z-value, conditional power, andstage 1 effect size relative to the effect size for which the underlyinggroup sequential design was powered.Sample size re-estimation using conditional power and an interim estimate oftreatment effect was proposed by several authors in the 1990's (seereferences below). Statistical testing for these original methods was basedon combination tests since Type I error could be inflated by using asufficient statistic for testing at the end of the trial. Since 2000, moreefficient variations of these conditional power designs have been developed.Fully optimized designs have also been derived (Posch et all, 2003,Lokhnygina and Tsiatis, 2008). Some of the later conditional power methodsallow use of sufficient statistics for testing (Chen, DeMets and Lan, 2004,Gao, Ware and Mehta, 2008, and Mehta and Pocock, 2011).The methods considered here are extensions of 2-stage group sequentialdesigns that include both an efficacy and a futility bound at the plannedinterim analysis. A maximum fold-increase in sample size (maxinc)fromthe supplied group sequential design (x) is specified by the user, aswell as a range of conditional power (cpadj) where sample size shouldbe re-estimated at the interim analysis, 1-the targeted conditional power tobe used for sample size re-estimation (beta) and a combination teststatistic (z2) to be used for testing at the end of the trial. Theinput value overrun represents incremental enrollment not included inthe interim analysis that is not included in the analysis; this is used forcalculating the required number of patients enrolled to complete the trial.Whereas most of the methods proposed have been based on using the interimestimated treatment effect size (default for ssrCP), the variabletheta allows the user to specify an alternative; Liu and Chi (2001)suggest that using the parameter value for which the trial was originallypowered is a good choice.<\/span><\/div>, <div class='tooltip'>z2Z<span class='tooltiptext'>ssrCP() adapts 2-stage group sequential designs to 2-stage samplesize re-estimation designs based on interim analysis conditional power. Thisis a simple case of designs developed by Lehmacher and Wassmer, Biometrics(1999). The conditional power designs of Bauer and Kohne (1994), Proschan and Hunsberger (1995), Cui, Hung and Wang (1999) and Liu and Chi (2001), Gao, Ware and Mehta (2008), and Mehta and Pocock (2011). Either the estimatedtreatment effect at the interim analysis or any chosen effect size can beused for sample size re-estimation. If not done carefully, these designs canbe very inefficient. It is probably a good idea to compare any design to asimpler group sequential design; see, for example, Jennison and Turnbull(2003). The a assumes a small Type I error is included with the interimanalysis and that the design is an adaptation from a 2-stage groupsequential design Related functions include 3 pre-defined combination testfunctions (z2NC, z2Z, z2Fisher) that represent theinverse normal combination test (Lehmacher and Wassmer, 1999), thesufficient statistic for the complete data, and Fisher's combination test.Power.ssrCP computes unconditional power for a conditional powerdesign derived by ssrCP.condPower is a supportive routine that also is interesting in its ownright; it computes conditional power of a combination test given an interimtest statistic, stage 2 sample size and combination test statistic. Whilethe returned data frame should make general plotting easy, the functionplot.ssrCP() prints a plot of study sample size by stage 1 outcomewith multiple x-axis labels for stage 1 z-value, conditional power, andstage 1 effect size relative to the effect size for which the underlyinggroup sequential design was powered.Sample size re-estimation using conditional power and an interim estimate oftreatment effect was proposed by several authors in the 1990's (seereferences below). Statistical testing for these original methods was basedon combination tests since Type I error could be inflated by using asufficient statistic for testing at the end of the trial. Since 2000, moreefficient variations of these conditional power designs have been developed.Fully optimized designs have also been derived (Posch et all, 2003,Lokhnygina and Tsiatis, 2008). Some of the later conditional power methodsallow use of sufficient statistics for testing (Chen, DeMets and Lan, 2004,Gao, Ware and Mehta, 2008, and Mehta and Pocock, 2011).The methods considered here are extensions of 2-stage group sequentialdesigns that include both an efficacy and a futility bound at the plannedinterim analysis. A maximum fold-increase in sample size (maxinc)fromthe supplied group sequential design (x) is specified by the user, aswell as a range of conditional power (cpadj) where sample size shouldbe re-estimated at the interim analysis, 1-the targeted conditional power tobe used for sample size re-estimation (beta) and a combination teststatistic (z2) to be used for testing at the end of the trial. Theinput value overrun represents incremental enrollment not included inthe interim analysis that is not included in the analysis; this is used forcalculating the required number of patients enrolled to complete the trial.Whereas most of the methods proposed have been based on using the interimestimated treatment effect size (default for ssrCP), the variabletheta allows the user to specify an alternative; Liu and Chi (2001)suggest that using the parameter value for which the trial was originallypowered is a good choice.<\/span><\/div>, <div class='tooltip'>zn2hr<span class='tooltiptext'>nSurvival() is used to calculate the sample size for a clinical trialwith a time-to-event endpoint. The Lachin and Foulkes (1986) method is used.nEvents uses the Schoenfeld (1981) approximation to provide samplesize and power in terms of the underlying hazard ratio and the number ofevents observed in a survival analysis. The functions hrz2n(),hrn2z() and zn2hr() also use the Schoenfeld approximation toprovide simple translations between hazard ratios, z-values and the numberof events in an analysis; input variables can be given as vectors.nSurvival() produces an object of class \"nSurvival\" with the numberof subjects and events for a set of pre-specified trial parameters, such asaccrual duration and follow-up period. The calculation is based on Lachinand Foulkes (1986) method and can be used for risk ratio or risk difference.The function also consider non-uniform (exponential) entry as well asuniform entry.If the logical approx is TRUE, the variance under alternativehypothesis is used to replace the variance under null hypothesis. Fornon-uniform entry, a non-zero value of gamma for exponential entrymust be supplied. For positive gamma, the entry distribution isconvex, whereas for negative gamma, the entry distribution isconcave.nEvents() uses the Schoenfeld (1981) method to approximate the numberof events n (given beta) or the power (given n).Arguments may be vectors or scalars, but any vectors must have the samelength.The functions hrz2n, hrn2z and zn2hr also all apply theSchoenfeld approximation for proportional hazards modeling. Thisapproximation is based on the asymptotic normal distribtuion of the logrankstatistic as well as related statistics are asymptotically normal. Let\\lambda denote the underlying hazard ratio (lambda1/lambda2 interms of the arguments to nSurvival). Further, let n denote thenumber of events observed when computing the statistic of interest andr the ratio of the sample size in an experimental group relative to acontrol. The estimated natural logarithm of the hazard ratio from aproportional hazards ratio is approximately normal with a mean oflog{\\lambda} and variance (1+r)^2/nr. Let z denote alogrank statistic (or a Wald statistic or score statistic from aproportional hazards regression model). The same asymptotic theory impliesz is asymptotically equivalent to a normalized estimate of the hazardratio \\lambda and thus z is asymptotically normal with variance1 and mean \\frac{log{\\lambda}r}{(1+r)^2}. Plugging the estimatedhazard ratio into the above equation allows approximating any one of thefollowing based on the other two: the estimate hazard ratio, the number ofevents and the z-statistic. That is, \\hat{\\lambda}=\\exp(z(1+r)/\\sqrt{rn}) z=\\log(\\hat{\\lambda})\\sqrt{nr}/(1+r) n=(z(1+r)/\\log(\\hat{\\lambda}))^2/r.hrz2n() translates an observed interim hazard ratio and interimz-value into the number of events required for the Z-value and hazard ratioto correspond to each other. hrn2z() translates a hazard ratio andnumber of events into an approximate corresponding Z-value. zn2hr()translates a Z-value and number of events into an approximate correspondinghazard ratio. Each of these functions has a default assumption of anunderlying hazard ratio of 1 which can be changed using the argumenthr0. hrn2z() and zn2hr() also have an argumenthr1 which is only used to compute the sign of the computed Z-value inthe case of hrn2z() and whether or not a z-value > 0 corresponds to ahazard ratio > or < the null hazard ratio hr0.<\/span><\/div>","<div class='tooltip'>boundaries_sim<span class='tooltiptext'>boundaries_sim is used to estimate lower and upper boundaries for GSED based on simulations of trials.<\/span><\/div>, <div class='tooltip'>magnusson_turnbull<span class='tooltiptext'>magnusson_turnbull is used apply GSED design, selection or evaluation at each stage, on data.<\/span><\/div>, <div class='tooltip'>max_FI<span class='tooltiptext'>max_FI is used to estimate maximum Fisher information based on two power criteria. - The first criterion consider the maxmimum Fisher information such that there is a pre-defined power to declare efficacy in the entire population for a given vector of parameters representing treatment effetcs in each subgroup.- The second criterion consider the maxmimum Fisher information such that there is a pre-defined power to declare efficacy in at least one subgroup for a given vector of parameters representing treatment effetcs in each subgroup.<\/span><\/div>, <div class='tooltip'>sim_magnusson_turnbull<span class='tooltiptext'>sim_magnusson_turnbull is used to simulate clincal trials with GSED for different type of outcome (survival, binary, continuous).<\/span><\/div>, <div class='tooltip'>stage_1_evaluation<span class='tooltiptext'>stage_1_evaluation is used to evaluate the efficacy of the subgroup selected at the end of the first stage of GSED.<\/span><\/div>, <div class='tooltip'>stage_1_selection<span class='tooltiptext'>stage_1_selection is used to determine the subgroup selected at the end of the first stage of GSED.<\/span><\/div>, <div class='tooltip'>test_BC<span class='tooltiptext'>For internal use<\/span><\/div>","<div class='tooltip'>binding<span class='tooltiptext'>Computation of equivalence and binding futility boundaries for group sequential designs in studies with equivalence hypothesis via Monte Carlo simulations<\/span><\/div>, <div class='tooltip'>equivonly<span class='tooltiptext'>Computation of equivalence boundaries for group sequential designs in studies with equivalence hypothesis that only stop for equivalence via Monte Carlo simulations<\/span><\/div>, <div class='tooltip'>figureE<span class='tooltiptext'>produces the 2-dimenisonal equivalence boundary plot for each stage in group sequential designs of equivalence studies<\/span><\/div>, <div class='tooltip'>figureEF<span class='tooltiptext'>produces the 2-dimenisonal equivalence and futility boundary plot for each stage in group sequential designs of equivalence studies<\/span><\/div>, <div class='tooltip'>nfix<span class='tooltiptext'>calculates sample size for equivalence studies in the traditional non-sequential setting<\/span><\/div>, <div class='tooltip'>nminmax<span class='tooltiptext'>calculates sample size for group sequential designs in equivalence studies that can stop for equivalence, or for either equivalence or futility (binding or non-binding). The calculated samples size is referred to as n.minmax, \"min\" in the sense that the calculcated n is the minimunm required sample size to reach a given power level, \"max\" in the sense that it would the max spent sample size which only happens if the study stop in the last stage<\/span><\/div>, <div class='tooltip'>nonbinding<span class='tooltiptext'>Computation of equivalence and non-binding futility boundaries for group sequential designs in studies with equivalence hypothesis via Monte Carlo simulations<\/span><\/div>, <div class='tooltip'>oc<span class='tooltiptext'>empirical examination of the operating characteristics of a group sequential design via the Monte Carlo simulation approach. users need to supply the design, including the lower and upper bounds of the equivalence hypothesis, the true difference between 2 groups, and the corresponding variance parameter, and the number of looks (including the final look), the group sizes, and the equivalence and futility boundaries. It outputs empirical type I or type II error rates, expected sample sizes in the 2 groups, and the probability of stopping at each stage due to either equivalence or futility<\/span><\/div>","<div class='tooltip'>allocate_mult<span class='tooltiptext'>Allocation of replicates on existing design locations, based on (29) from (Ankenman et al, 2010)<\/span><\/div>, <div class='tooltip'>compareGP<span class='tooltiptext'>Compare two models based on the log-likelihood for hetGP and homGP models<\/span><\/div>, <div class='tooltip'>cov_gen<span class='tooltiptext'>Correlation function of selected type, supporting both isotropic and product forms<\/span><\/div>, <div class='tooltip'>crit_cSUR<span class='tooltiptext'>Computes cSUR infill criterion<\/span><\/div>, <div class='tooltip'>crit_EI<span class='tooltiptext'>Computes EI for minimization<\/span><\/div>, <div class='tooltip'>crit_ICU<span class='tooltiptext'>Computes ICU infill criterion<\/span><\/div>, <div class='tooltip'>crit_IMSPE<span class='tooltiptext'>Compute the integrated mean square prediction error after adding a new design<\/span><\/div>, <div class='tooltip'>crit_MCU<span class='tooltiptext'>Computes MCU infill criterion<\/span><\/div>, <div class='tooltip'>crit_MEE<span class='tooltiptext'>Computes MEE infill criterion<\/span><\/div>, <div class='tooltip'>crit_optim<span class='tooltiptext'>Search for the best value of available criterion, possibly using a h-steps lookahead strategy to favor designs with replication<\/span><\/div>, <div class='tooltip'>crit_qEI<span class='tooltiptext'>Fast approximated batch-Expected Improvement criterion (for minimization)<\/span><\/div>, <div class='tooltip'>crit_tMSE<span class='tooltiptext'>Computes targeted mean squared error infill criterion<\/span><\/div>, <div class='tooltip'>deriv_crit_EI<span class='tooltiptext'>Derivative of EI criterion for GP models<\/span><\/div>, <div class='tooltip'>deriv_crit_IMSPE<span class='tooltiptext'>Derivative of crit_IMSPE<\/span><\/div>, <div class='tooltip'>f1d<span class='tooltiptext'>1d test function (1)<\/span><\/div>, <div class='tooltip'>f1d_n<span class='tooltiptext'>Noisy 1d test function (1)Add Gaussian noise with variance r(x) = scale * (1.1 + sin(2 pi x))^2 to f1d<\/span><\/div>, <div class='tooltip'>f1d2<span class='tooltiptext'>1d test function (2)<\/span><\/div>, <div class='tooltip'>f1d2_n<span class='tooltiptext'>Noisy 1d test function (2)Add Gaussian noise with variance r(x) = scale * (exp(sin(2 pi x)))^2 to f1d2<\/span><\/div>, <div class='tooltip'>find_reps<span class='tooltiptext'>Prepare data for use with mleHetGP, in particular to find replicated observations<\/span><\/div>, <div class='tooltip'>horizon<span class='tooltiptext'>Adapt the look-ahead horizon depending on the replicate allocation or a target ratio<\/span><\/div>, <div class='tooltip'>IMSPE<span class='tooltiptext'>IMSPE of a given design<\/span><\/div>, <div class='tooltip'>IMSPE_optim<span class='tooltiptext'>Search for the best value of the IMSPE criterion, possibly using a h-steps lookahead strategy to favor designs with replication<\/span><\/div>, <div class='tooltip'>logLikH<span class='tooltiptext'>Generic Log-likelihood functionThis function can be used to compute loglikelihood for homGP/hetGP models<\/span><\/div>, <div class='tooltip'>LOO_preds<span class='tooltiptext'>Provide leave one out predictions, e.g., for model testing and diagnostics. This is used in the method plot available on GP and TP models.<\/span><\/div>, <div class='tooltip'>mleHetGP<span class='tooltiptext'>Gaussian process regression under input dependent noise based on maximum likelihood estimation of the hyperparameters. A second GP is used to model latent (log-) variances. This function is enhanced to deal with replicated observations.<\/span><\/div>, <div class='tooltip'>mleHetTP<span class='tooltiptext'>Student-t process regression under input dependent noise based on maximum likelihood estimation of the hyperparameters.A GP is used to model latent (log-) variances.This function is enhanced to deal with replicated observations.<\/span><\/div>, <div class='tooltip'>mleHomGP<span class='tooltiptext'>Gaussian process regression under homoskedastic noise based on maximum likelihood estimation of the hyperparameters. This function is enhanced to deal with replicated observations.<\/span><\/div>, <div class='tooltip'>mleHomTP<span class='tooltiptext'>Student-t process regression under homoskedastic noise based on maximum likelihood estimation of the hyperparameters. This function is enhanced to deal with replicated observations.<\/span><\/div>, <div class='tooltip'>pred_noisy_input<span class='tooltiptext'>Gaussian process prediction prediction at a noisy input x, with centered Gaussian noise of variance sigma_x. Several options are available, with different efficiency/accuracy tradeoffs.<\/span><\/div>, <div class='tooltip'>rebuild<span class='tooltiptext'>Functions to make hetGP objects lighter before exporting them, and to reverse this after import.The rebuild function may also be used to obtain more robust inverse of covariance matrices using ginv.<\/span><\/div>, <div class='tooltip'>scores<span class='tooltiptext'>Score and RMSE functionTo asses the performance of the prediction, this function computes the root mean squared error and proper score function (also known as negative log-probability density).<\/span><\/div>, <div class='tooltip'>sirEval<span class='tooltiptext'>Epidemiology problem, initial and rescaled to [0,1]^2 versions.<\/span><\/div>, <div class='tooltip'>sirSimulate<span class='tooltiptext'>Epidemiology problem, initial and rescaled to [0,1]^2 versions.<\/span><\/div>, <div class='tooltip'>strip<span class='tooltiptext'>Functions to make hetGP objects lighter before exporting them, and to reverse this after import.The rebuild function may also be used to obtain more robust inverse of covariance matrices using ginv.<\/span><\/div>, <div class='tooltip'>Wij<span class='tooltiptext'>Compute double integral of the covariance kernel over a [0,1]^d domain<\/span><\/div>","<div class='tooltip'>fit_gen_model<span class='tooltiptext'>A function to fit a generative model to a mutation dataset. At its heart, requires a gene_lengths dataframe (for examples of the correct format for this see the pre-loaded datasets example_maf_data$gene_lengths and ensembl_gene_lengths), and a mutation dataset. This is best supplied through the 'table' argument, and constructed via the function get_mutation_tables().<\/span><\/div>, <div class='tooltip'>fit_gen_model_uninteract<span class='tooltiptext'>A function to fit a generative model to a mutation dataset that does not incorporate gene/variant-specific effects.Otherwise acts similarly to the function fit_gen_model().NOTE: fits produced by this model will not be compatible with predictive model fits downstream - it is purely for comparingwith full models.<\/span><\/div>, <div class='tooltip'>fit_gen_model_unisamp<span class='tooltiptext'>A function to fit a generative model to a mutation dataset that does not incorporate sample-specific effects.Otherwise acts similarly to the function fit_gen_model().NOTE: fits produced by this model will not be compatible with predictive model fits downstream - it is purely for comparingwith full models.<\/span><\/div>, <div class='tooltip'>generate_maf_data<span class='tooltiptext'>A function to randomly simulate an (abridged) annotated mutation file, containing information on sample of origin, gene and mutation type, as well as a dataframe of gene lengths.<\/span><\/div>, <div class='tooltip'>get_auprc<span class='tooltiptext'>A function to return AUPRC metrics for predictions vs actual values. Works well when piped to straight from get_predictions().<\/span><\/div>, <div class='tooltip'>get_biomarker_from_maf<span class='tooltiptext'>A function to recover true biomarker values from a mutation annotation file.<\/span><\/div>, <div class='tooltip'>get_biomarker_tables<span class='tooltiptext'>A function, similar to get_mutation_tables(), but returning the true biomarker values for a training, validation and test sets.<\/span><\/div>, <div class='tooltip'>get_gen_estimates<span class='tooltiptext'>Given a generative model of the type we propose, and an alternate version (saturated \"S\", sample-independent \"US\", gene-independent \"UG\" or gene/variant interaction independent \"UI\"), either produces the estimated observations on the training dataset or calculates residual deviance between models.<\/span><\/div>, <div class='tooltip'>get_K<span class='tooltiptext'>An internal function, producing the correct bias penalisation for use in predictive model fitting.<\/span><\/div>, <div class='tooltip'>get_mutation_dictionary<span class='tooltiptext'>A function to create a mutation dictionary to group and filter mutation types: this can be useful for computational practicality. It is often not practical to modeleach distinct mutation type together, so for practicality one may group multiple classes together (e.g. all indel mutations, all nonsynonymous mutations).Additionally, some mutation types may be excluded from modelling (for example, one may wish not to use synonymous mutations in the model fitting process).<\/span><\/div>, <div class='tooltip'>get_mutation_tables<span class='tooltiptext'>This function allows for i) separation of a mutation dataset into training, validation and testing components, and ii) conversion from annotated mutation formatto sparse mutation matrices, as described in the function get_table_from_maf().<\/span><\/div>, <div class='tooltip'>get_p<span class='tooltiptext'>An internal function. From the learned generative model and training data, produces a vector of weights p to be used inthe subsequent group lasso optimisation, alongside a biomarker-dependent normalisation quantity p_norm.<\/span><\/div>, <div class='tooltip'>get_panels_from_fit<span class='tooltiptext'>An internal function for analysing a group Lasso fit as part of the predictive model learning procedure, which returns the sets of genes identified by different iterations of the group Lasso algorithm.<\/span><\/div>, <div class='tooltip'>get_predictions<span class='tooltiptext'>A function taking a predictive model(s) and new observations, and applying the predictive model to them to return predicted biomarker values.<\/span><\/div>, <div class='tooltip'>get_r_squared<span class='tooltiptext'>A function to return R^2 metrics for predictions vs actual values. Works well when piped to straight from get_predictions().<\/span><\/div>, <div class='tooltip'>get_stats<span class='tooltiptext'>A function to return a variety metrics for predictions vs actual values. Works well when piped to straight from get_predictions().<\/span><\/div>, <div class='tooltip'>get_table_from_maf<span class='tooltiptext'>A function to, given a mutation annotation dataset with columns for sample barcode, gene name and mutation type, to reformulate this as a mutation matrix,with rows denoting samples, columns denoting gene/mutation type combinations, and the individual entries giving the number of mutations observed. Thiswill likely be very sparse, so we save it as a sparse matrix for efficiency.<\/span><\/div>, <div class='tooltip'>pred_first_fit<span class='tooltiptext'>This function implements the first-fit procedure described in Bradley and Cannings, 2021. It requires at least a generative model and a dataframe containing gene lengths as input.<\/span><\/div>, <div class='tooltip'>pred_intervals<span class='tooltiptext'>A function to produce a confidence region for a linear predictor. In upcoming versions will (hopefully) be greatly simplified.<\/span><\/div>, <div class='tooltip'>pred_refit_panel<span class='tooltiptext'>A function taking the output of a call to pred_first_fit(), as well as gene length information, and a specified panel (list of genes), and producing a refitted predictive model on that given panel.<\/span><\/div>, <div class='tooltip'>pred_refit_range<span class='tooltiptext'>A function producing a refitted predictive model for each panel produced by usage of the function pred_first_fit(), by repeatedly applying the function pred_refit_panel().<\/span><\/div>, <div class='tooltip'>vis_model_fit<span class='tooltiptext'>A function to visualise how well a general model has fitted to a mutation dataset across cross-validation folds. Designed to produce a similar output to glmnet's function plot.cv.glmnet.<\/span><\/div>","<div class='tooltip'>bayesmc<span class='tooltiptext'>Bayesian method for high-dimensional variable selection<\/span><\/div>, <div class='tooltip'>datasim<span class='tooltiptext'>This function simulates a data of N subjects with misclassified outcomes, assuming each subject receives a sequence of pre-scheduled tests for disease status ascertainment. Each test is subject to error, characterized by sensitivity and specificity. An exponential distribution is assumed for the time to event of interest. Three kinds of covariate settings can be generated: one sample setting, two group setting, and continuous covariates setting with each covariate sampled from i.i.d. N(0, 1). Two missing mechanisms can be assumed, namely MCAR and NTFP. The MCAR setting assumes that each test is subject to a constant, independent probability of missingness. The NTFP mechanism includes two types of missingness - (1) incorporates a constant, independent, probability of missing for each test prior to the first positive test result; and (2) all test results after first positive are missing. The simulated data is in longitudinal form with one row per test time. Covariate values, by default, are assumed to be constant. However, this function can simulate a special case of time varying covariates. Under time varying covariates setting, each subject is assumed to have a change time point, which is sampled from the visit times. We assume that each subject has two sets of covariate values. Before his change time point, the covariate values take from the first set, and second set after change time point. Thus, each subject's distribution of survival time is two-piece exponential distribution with different hazard rates.<\/span><\/div>, <div class='tooltip'>fitsurv<span class='tooltiptext'>Fit survival function, used for Bayesian simulation<\/span><\/div>, <div class='tooltip'>icmis<span class='tooltiptext'>This function estimates the baseline survival function evaluated at each testtime in the presence of error-prone diagnostic tests and self-reported outcomes. If there are covariates included in the dataset, it also estimates their coefficients assuming proportional hazards. The covariate values can beeither time independent or time varying The function can also be used toincorporate misclassification of disease status at baseline (due to anerror-prone diagnostic procedure).<\/span><\/div>, <div class='tooltip'>icpower<span class='tooltiptext'>This function calculates the power and sample in the presence of error-prone diagnostic tests and self-reported outcomes. Two missing mechanisms can be assumed, namely MCAR and NTFP. The MCAR setting assumes that each test is subject to a constant, independent probability of missingness. The NTFP mechanism includes two types of missingness - (1) incorporates a constant, independent, probability of missing for each test prior to the first positivetest result; and (2) all test results after first positive are missing.<\/span><\/div>, <div class='tooltip'>icpower_weibull<span class='tooltiptext'>This functions works same way as icpower function except that it assumes the survival function follows Weibulldistribution. The scale parameter is assumed to be same forboth treatment and control groups.This can be used estimate power and sample size for intervalcensored data using Weibull model, which is a cpecial casewhen both sensitivity and specificity being 1.<\/span><\/div>, <div class='tooltip'>icpower.val<span class='tooltiptext'>This function calculates the power and sample size in the presence of error-prone diagnostic tests and self-reported outcomes when both sensitivityand specificity are unknown. In this case, a subject of the subjects receive both gold standard test and error-prone test at each non-missing visit. The remaining subjects receive only error-prone test. Here, for the validation set, NTFP refers to no test after first positive result from the gold standard test. Both sensitivity and specificity are treated as unknown parameters in this setting.<\/span><\/div>, <div class='tooltip'>icpowerpf<span class='tooltiptext'>This function implements power and sample size calculations for intervalcensored time-to-event outcomes, when the diagnostic tests are assumed to beperfect (i.e. sensitivity=1 and specificity=1). This is a special case of themore general study design function icpower. However, for thespecial case of perfect diagnostic tests, this function can be used withsignificantly improved computational efficiency.<\/span><\/div>","<div class='tooltip'>CaseCohortIC<span class='tooltiptext'>Provides a sieve semiparametric likelihood approach under the proportional hazards model for analyzing data from a case-cohort design with failure times subject to interval-censoring. The likelihood function is constructed using inverse probability weighting, and the sieves are built with Bernstein polynomials. A weighted bootstrap procedure is implemented for variance estimation.<\/span><\/div>, <div class='tooltip'>estimate<span class='tooltiptext'>Retrieves the estimated beta parameters for the m value that minimizes the AIC.<\/span><\/div>, <div class='tooltip'>minAIC<span class='tooltiptext'>Retrieves the minimum AIC.<\/span><\/div>, <div class='tooltip'>mVal<span class='tooltiptext'>Retrieves the degree of the Bernstein polynomial basis provided as input that minimizes the AIC.<\/span><\/div>, <div class='tooltip'>ODSDesignIC<span class='tooltiptext'>Provides an outcome-dependent sampling (ODS) design with interval-censored failure time data, where the observed sample is enriched by selectively including certain more informative failure subjects. The method is a sieve semiparametric maximum empirical likelihood approach for fitting the proportional hazards model to data from the interval- censoring ODS design.<\/span><\/div>, <div class='tooltip'>optimObj<span class='tooltiptext'>Retrieves the final optimization results for the m value that minimizes the AIC.<\/span><\/div>, <div class='tooltip'>print<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'>Retrieves the estimated beta parameters for the m value that minimizes the AIC; the m value; and the AIC value.<\/span><\/div>","<div class='tooltip'>Arrows<span class='tooltiptext'>Draws a set of open or closed arrows which can be shaped by many arguments. Arrows is an extended version of p.arrows.<\/span><\/div>, <div class='tooltip'>cart2clock<span class='tooltiptext'>Converts cartesian coordinates (x, y to clock coordinates (rho, phi)<\/span><\/div>, <div class='tooltip'>clock2cart<span class='tooltiptext'>Converts clock coordinates (rho, phi) to cartesian coordinates (x, y).<\/span><\/div>, <div class='tooltip'>col2hsv<span class='tooltiptext'>Converts color names or values to hsv code.<\/span><\/div>, <div class='tooltip'>data.sheet<span class='tooltiptext'>Coerces a list with vectors of different length into a data.frame. Fills the shorter vectors with NA.<\/span><\/div>, <div class='tooltip'>draw.leg<span class='tooltiptext'>Produces (and possibly draws) a Grid frame grob which is a legend that can be placed in other Grid plots. draw.leg is a slightly enhanced version of draw.key in package lattice V 0.12-3.<\/span><\/div>, <div class='tooltip'>general.control<span class='tooltiptext'>Defines the General Appearance of rose plot.<\/span><\/div>, <div class='tooltip'>getXY<span class='tooltiptext'>The function accepts one- or two dimensional data, checks them for compatibility and gives a dataframe back.<\/span><\/div>, <div class='tooltip'>grid.control<span class='tooltiptext'>Defines the appearance of the guiding elements of rose plots such as circles, rays and labels.<\/span><\/div>, <div class='tooltip'>hr<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>IDPcolorRamp<span class='tooltiptext'>Produces color ramps which change simultanously hues, saturation and values as defined in the hsv modus. This allows to produce especially smooth transitions from one color to the next. The default color ramp starts with light blue, continues with green, yellow, red and ends with dark violet.<\/span><\/div>, <div class='tooltip'>ilagplot<span class='tooltiptext'>Produces an image lag plot matrix of large timeseries where the colors encode the density of the points in the lag plots.<\/span><\/div>, <div class='tooltip'>Image<span class='tooltiptext'>The density of points in a scatter plot is encoded by color.<\/span><\/div>, <div class='tooltip'>ipairs<span class='tooltiptext'>Produces an image scatter plot matrix of large datasets where the colors encode the density of the points in the scatter plots.<\/span><\/div>, <div class='tooltip'>ipanel.smooth<span class='tooltiptext'>An example of a useful panel function for huge datasets<\/span><\/div>, <div class='tooltip'>iplot<span class='tooltiptext'>Produces an image scatter plot of large datasets where the colors encode the density of the points in the scatter plot. Works also with factors.<\/span><\/div>, <div class='tooltip'>iplotLegend<span class='tooltiptext'>Plots legend for color ramp.<\/span><\/div>, <div class='tooltip'>itermplot<span class='tooltiptext'>Plots regression terms against their predictors, optionally withstandard errors and partial residuals in a density plot.<\/span><\/div>, <div class='tooltip'>key.control<span class='tooltiptext'>Defines the appearance of the legend of a Rose Plot.<\/span><\/div>, <div class='tooltip'>longtsPlot<span class='tooltiptext'>Plot one or more regular time series in multiple figures on one or more pages.<\/span><\/div>, <div class='tooltip'>NaRV.omit<span class='tooltiptext'>Omits observations with values which a not regular (=Not a Regular Value) when object is a vector, a factor, a data.frame or a matrix.<\/span><\/div>, <div class='tooltip'>ok<span class='tooltiptext'>Sets NAs in logical vectors and matrices to FALSE. This is especially useful for conditional selections of data when the variables the condition is based on contain NAs.<\/span><\/div>, <div class='tooltip'>peaks<span class='tooltiptext'>Returns position, signal height and approximate width at half maximum peak height.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'>Describes plot method for class \"rose\"<\/span><\/div>, <div class='tooltip'>poster.plot<span class='tooltiptext'>Convenient xyplot with Colored Background. Background of margin may be chosen independently from background in plot region.<\/span><\/div>, <div class='tooltip'>pw<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>pw.ai<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>pw.aw<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>rfbaseline<span class='tooltiptext'>Robust fitting of local regression models for estimating a baseline or a background signal<\/span><\/div>, <div class='tooltip'>rfbaselineScale<span class='tooltiptext'>Estimation of the scale parameter based on data smaller than its first mode. Mainly used in rfbaseline.<\/span><\/div>, <div class='tooltip'>rose<span class='tooltiptext'>rose splits data into subsets according to one or two grouping elements, computes summary statistics for each, and returns the result in a rose object.<\/span><\/div>, <div class='tooltip'>showColors<span class='tooltiptext'>Displays colors produced by a color vector col and labels them by the corresponding number of the element of col.<\/span><\/div>, <div class='tooltip'>Td<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>Td.aw<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>Tf.ai<span class='tooltiptext'>Converting dew point (of water in air) into vapor pressure or relative humidity and vice versa.<\/span><\/div>, <div class='tooltip'>title.control<span class='tooltiptext'>Auxilary for controlling the title of a rose plot.<\/span><\/div>, <div class='tooltip'>zoom<span class='tooltiptext'>Function to zoom in and out by mouse click in a 2D-plot.<\/span><\/div>","<div class='tooltip'>powerLongSurv<span class='tooltiptext'>Compute the power in joint modeling of longitudinal and survival data when the variance-covariance matrix Sigma_Theta is unknown and the trajectories are order k.The function computes power for a one-sided test, either H_0: \\beta = 0 \\quad \\mbox{ and } \\quad H_{1A}: \\beta > 0% H_0: \\beta = 0 and H_1A: \\beta > 0or H_0: \\beta = 0 \\quad \\mbox{ and } \\quad H_{1B}: \\beta < 0% H_0: \\beta = 0 and H_1B: \\beta < 0with Type I error \\alpha. The choice of the alternative is determined by the sign of \\beta. Negative values for \\beta indicate that the alternative hypothesis is H_{1B}, while \\beta \\ge 0 indicates that it is H_{1A}.It creates a powerLongSurv object.<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>AdaptiveRSPD<span class='tooltiptext'>Generates a sliced rotated sphere packing design by enlarging one rotated sphere packing design.<\/span><\/div>, <div class='tooltip'>DPMPD<span class='tooltiptext'>Generates a densest packing-based maximum projection design.<\/span><\/div>, <div class='tooltip'>InterleavedMaximinD<span class='tooltiptext'>Generates an interleaved lattice-based maximin distance design.<\/span><\/div>, <div class='tooltip'>InterleavedMaximinDAlg1<span class='tooltiptext'>Generates an interleaved lattice-based maximin distance design.<\/span><\/div>, <div class='tooltip'>InterleavedMaximinDAlg2<span class='tooltiptext'>Generates an interleaved lattice-based maximin distance design.<\/span><\/div>, <div class='tooltip'>InterleavedMaximinDAlg3<span class='tooltiptext'>Generates an interleaved lattice-based maximin distance design.<\/span><\/div>, <div class='tooltip'>InterleavedMinimaxD<span class='tooltiptext'>Generates an interleaved lattice-based minimax distance design.<\/span><\/div>, <div class='tooltip'>LRS<span class='tooltiptext'>Computes the radius, widths, and vertexes of a polytope giving halfspace definition. The program is a R shell of LRS (v.5.1a with lrsmp.h), a reverse search vertex enumeration program/CH package in C which is developed by David Avis <http://cgm.cs.mcgill.ca/~avis/C/lrs.html>. Consider the problem of Ax<=b, where A is an n*p matrix, x is a p-vector, and b is an n-vector. Please make sure that the solution of x is nonempty and bounded. Then the nonequalities give the halfspace definition of a polytope. Also make sure that A and b are rational numbers.<\/span><\/div>, <div class='tooltip'>ProjSepD<span class='tooltiptext'>Computes the projective separation distance of a design.<\/span><\/div>, <div class='tooltip'>RSPD<span class='tooltiptext'>Generates a rotated sphere packing design.<\/span><\/div>, <div class='tooltip'>SlicedRSPD<span class='tooltiptext'>Generates a sliced rotated sphere packing design by partitioning one rotated sphere packing design.<\/span><\/div>","<div class='tooltip'>augmentLHS<span class='tooltiptext'>Augments an existing Latin Hypercube Sample, adding points to the design, whilemaintaining the latin properties of the design.<\/span><\/div>, <div class='tooltip'>create_oalhs<span class='tooltiptext'>Create an orthogonal array Latin hypercube<\/span><\/div>, <div class='tooltip'>createAddelKemp<span class='tooltiptext'>The addelkemp program produces OA( 2q^2, k, q, 2 ), k <= 2q+1,for odd prime powers q.<\/span><\/div>, <div class='tooltip'>createAddelKemp3<span class='tooltiptext'>The addelkemp3 program producesOA( 2*q^3, k, q, 2 ), k <= 2q^2+2q+1, for prime powers q.q may be an odd prime power, or q may be 2 or 4.<\/span><\/div>, <div class='tooltip'>createAddelKempN<span class='tooltiptext'>The addelkempn program producesOA( 2*q^n, k, q, 2 ), k <= 2(q^n - 1)/(q-1)-1, for prime powers q.q may be an odd prime power, or q may be 2 or 4.<\/span><\/div>, <div class='tooltip'>createBose<span class='tooltiptext'>The bose programproduces OA( q^2, k, q, 2 ), k <= q+1 for prime powers q.<\/span><\/div>, <div class='tooltip'>createBoseBush<span class='tooltiptext'>The bosebush programproduces OA( 2q^2, k, q, 2 ), k <= 2q+1, for powers of 2, q=2^r.<\/span><\/div>, <div class='tooltip'>createBoseBushl<span class='tooltiptext'>The bosebushl program produces OA( lambda*q^2, k, q, 2 ),k <= lambda*q+1, for prime powers q and lambda > 1. Both q andlambda must be powers of the same prime.<\/span><\/div>, <div class='tooltip'>createBush<span class='tooltiptext'>The bush programproduces OA( q^3, k, q, 3 ), k <= q+1 for prime powers q.<\/span><\/div>, <div class='tooltip'>createBusht<span class='tooltiptext'>The busht program produces OA( q^t, k, q, t ), k <= q+1, t>=3,for prime powers q.<\/span><\/div>, <div class='tooltip'>geneticLHS<span class='tooltiptext'>Draws a Latin Hypercube Sample from a set of uniform distributions for use increating a Latin Hypercube Design. This function attempts to optimize thesample with respect to the S optimality criterion through a genetic typealgorithm.<\/span><\/div>, <div class='tooltip'>improvedLHS<span class='tooltiptext'>Draws a Latin Hypercube Sample from a set of uniform distributions for use increating a Latin Hypercube Design. This function attempts to optimize thesample with respect to an optimum euclidean distance between design points.<\/span><\/div>, <div class='tooltip'>maximinLHS<span class='tooltiptext'>Draws a Latin Hypercube Sample from a set of uniform distributions for use increating a Latin Hypercube Design. This function attempts to optimize thesample by maximizing the minium distance between design points (maximin criteria).<\/span><\/div>, <div class='tooltip'>oa_to_oalhs<span class='tooltiptext'>Create a Latin hypercube from an orthogonal array<\/span><\/div>, <div class='tooltip'>optAugmentLHS<span class='tooltiptext'>Augments an existing Latin Hypercube Sample, adding points to the design, whilemaintaining the latin properties of the design. This function attempts toadd the points to the design in an optimal way.<\/span><\/div>, <div class='tooltip'>optimumLHS<span class='tooltiptext'>Draws a Latin Hypercube Sample from a set of uniform distributions for use increating a Latin Hypercube Design. This function uses the ColumnwisePairwise (CP) algorithm to generate an optimal design with respect to the Soptimality criterion.<\/span><\/div>, <div class='tooltip'>optSeededLHS<span class='tooltiptext'>Augments an existing Latin Hypercube Sample, adding points to the design, whilemaintaining the latin properties of the design. This function then uses thecolumnwise pairwise (CP) algoritm to optimize the design. The original design is not necessarily maintained.<\/span><\/div>, <div class='tooltip'>randomLHS<span class='tooltiptext'>randomLHS(4,3) returns a 4x3 matrix with each columnconstructed as follows: A random permutation of (1,2,3,4) isgenerated, say (3,1,2,4) for each of K columns. Then a uniformrandom number is picked from each indicated quartile. In thisexample a random number between .5 and .75 is chosen, then one between0 and .25, then one between .25 and .5, finally one between.75 and 1.<\/span><\/div>, <div class='tooltip'>runifint<span class='tooltiptext'>Create a Random Sample of Uniform Integers<\/span><\/div>","<div class='tooltip'>opf_accuracy<span class='tooltiptext'>Computes the OPF accuracy<\/span><\/div>, <div class='tooltip'>opf_accuracy4label<span class='tooltiptext'>Computes the OPF accuracy for each class of a given set<\/span><\/div>, <div class='tooltip'>opf_check<span class='tooltiptext'>Checks the OPF file for proper formatting purposes<\/span><\/div>, <div class='tooltip'>opf_classify<span class='tooltiptext'>Executes the test phase of the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_cluster<span class='tooltiptext'>Computes clusters by unsupervised OPF<\/span><\/div>, <div class='tooltip'>opf_create_subGraph<span class='tooltiptext'>Creates an empty subGraph structure<\/span><\/div>, <div class='tooltip'>opf_distance<span class='tooltiptext'>Generates the precomputed distance file for the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_fold<span class='tooltiptext'>Generates k folds (objects) for the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_info<span class='tooltiptext'>Gives information about the OPF file<\/span><\/div>, <div class='tooltip'>opf_knn_classify<span class='tooltiptext'>Executes the test phase of the OPF classifier with knn adjacency<\/span><\/div>, <div class='tooltip'>opf_knn_train<span class='tooltiptext'>Executes the training phase of the OPF classifier with knn adjacency<\/span><\/div>, <div class='tooltip'>opf_learn<span class='tooltiptext'>Executes the learning phase of the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_merge<span class='tooltiptext'>Merge subGraphs<\/span><\/div>, <div class='tooltip'>opf_normalize<span class='tooltiptext'>Normalizes data for the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_pruning<span class='tooltiptext'>Executes the pruning algorithm<\/span><\/div>, <div class='tooltip'>opf_read_classification<span class='tooltiptext'>Reads a file which contains the nodes' predicted labels<\/span><\/div>, <div class='tooltip'>opf_read_distances<span class='tooltiptext'>Reads a file which contains the precalculated distances<\/span><\/div>, <div class='tooltip'>opf_read_modelfile<span class='tooltiptext'>Reads a file which contains the learned model<\/span><\/div>, <div class='tooltip'>opf_read_subGraph<span class='tooltiptext'>Reads a file which contains the subGraph<\/span><\/div>, <div class='tooltip'>opf_run_example<span class='tooltiptext'>This function will run this example:dat <- opf_read_subGraph(dataset) (dataset is the subgraph file)X <- opf_split(dat,0.5,0,0.5,0)T <- X$trainingT2 <- X$testingY <- opf_train(T)class <- opf_classify(T2, Y$classifier)acc <- opf_accuracy(T2, class)<\/span><\/div>, <div class='tooltip'>opf_semi<span class='tooltiptext'>Executes the semi supervised training phase<\/span><\/div>, <div class='tooltip'>opf_split<span class='tooltiptext'>Generates training, evaluation and test sets for the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_train<span class='tooltiptext'>Executes the training phase of the OPF classifier<\/span><\/div>, <div class='tooltip'>opf_write_classification<span class='tooltiptext'>Writes into a file the predicted labels produced by the opf classificator<\/span><\/div>, <div class='tooltip'>opf_write_distances<span class='tooltiptext'>Writes into a file the precalculated distances computed by opf_distances function<\/span><\/div>, <div class='tooltip'>opf_write_modelfile<span class='tooltiptext'>Writes into a file the trained OPF classifier<\/span><\/div>, <div class='tooltip'>opf_write_subGraph<span class='tooltiptext'>Writes into a file a subGraph<\/span><\/div>, <div class='tooltip'>opf2svm<span class='tooltiptext'>Converts an OPF subGraph object to a LIBSVM file<\/span><\/div>, <div class='tooltip'>opf2txt<span class='tooltiptext'>Converts an OPF subGraph object to a text file<\/span><\/div>, <div class='tooltip'>SNode<span class='tooltiptext'>Subgraphs' node class<\/span><\/div>, <div class='tooltip'>subGraph<span class='tooltiptext'>Subgraph class<\/span><\/div>, <div class='tooltip'>svm2opf<span class='tooltiptext'>Converts a LIBSVM file to an OPF subGraph object<\/span><\/div>, <div class='tooltip'>txt2opf<span class='tooltiptext'>Converts a text file to an OPF subGraph object<\/span><\/div>","<div class='tooltip'>mams<span class='tooltiptext'>The function determines the boundaries of a multi-arm multi-stage study for a given boundary shape and finds the required number of subjects.<\/span><\/div>, <div class='tooltip'>mams.sim<span class='tooltiptext'>The function simulates multi-arm multi-stage designs and estimates power and expected sample size.<\/span><\/div>, <div class='tooltip'>MAMSNews<span class='tooltiptext'>Functions showing changes since previous versions.<\/span><\/div>, <div class='tooltip'>new.bounds<span class='tooltiptext'>The function determines updated boundaries of a multi-arm multi-stage study based on observed number of observations per arm.<\/span><\/div>, <div class='tooltip'>ordinal.mams<span class='tooltiptext'>The function determines (approximately) the boundaries of a multi-arm multi-stage study with ordinal or binary endpoints for a given boundary shape and finds the required number of subjects.<\/span><\/div>, <div class='tooltip'>stepdown.mams<span class='tooltiptext'>The function determines stopping boundaries for all intersection hypothesis tests in a multi-arm multi-stage study, given the amount of alpha (familywise error rate) to be spent at each analysis.<\/span><\/div>, <div class='tooltip'>stepdown.update<span class='tooltiptext'>Function to update a planned multi-arm multi-stage design to account for unplanned adaptations.<\/span><\/div>, <div class='tooltip'>tite.mams<span class='tooltiptext'>The function determines (approximately) the boundaries of a multi-arm multi-stage study with time-to-event endpoints for a given boundary shape and finds the required number of events.<\/span><\/div>","<div class='tooltip'>maximin<span class='tooltiptext'>Generates a space-filling design under the criterion of maximum-minimum distance; both discrete and continuous searches are provided.<\/span><\/div>, <div class='tooltip'>maximin.cand<span class='tooltiptext'>Generates a space-filling design under the criterion of maximum-minimum distance; both discrete and continuous searches are provided.<\/span><\/div>","<div class='tooltip'>alterInclProbs<span class='tooltiptext'>Alters inclusion probabilities to accommodate legacy sites. Inclusion probabilities are deflated around legacy sites, and the resulting set of new sites (and legacy sites) are spatially balanced.<\/span><\/div>, <div class='tooltip'>findDescendingTrans<span class='tooltiptext'>Creates a 'mask' for the extended sampling frame (sites by rotations) for two reasons: Some sampling gears that can only be deployed down-slope, and those designs where a 'wagon-wheel' is required from a small set of points (e.g. the top of a seamount).<\/span><\/div>, <div class='tooltip'>findTransFromPoint<span class='tooltiptext'>Creates a 'mask' for the extended sampling frame (sites by rotations) for two reasons: Some sampling gears that can only be deployed down-slope, and those designs where a 'wagon-wheel' is required from a small set of points (e.g. the top of a seamount).<\/span><\/div>, <div class='tooltip'>modEsti<span class='tooltiptext'>For a given survey design in any number of dimensions, calculate the mean prediction (plus SE plus 95% CI) for the area.<\/span><\/div>, <div class='tooltip'>quasiSamp<span class='tooltiptext'>Generates a spatially balanced design for given inclusion probabilities over a grid of potential sampling locations<\/span><\/div>, <div class='tooltip'>transectSamp<span class='tooltiptext'>For arbitrary transect patterns, generates a spatially balanced design for given inclusion probabilities over a grid of potential sampling locations<\/span><\/div>","<div class='tooltip'>betaMod<span class='tooltiptext'>The beta model is defined as f(d,\\theta)=E_0+E_{max}B(\\delta_1,\\delta_2)(d/scal)^{\\delta_1}(1-d/scal)^{\\delta_2} f(d,theta)=E0+Emax B(delta1,delta2)(d/scal)^delta1(1-d/scal)^delta2 where B(\\delta_1,\\delta_2)=(\\delta_1+\\delta_2)^{\\delta_1+\\delta_2}/(\\delta_1^{\\delta_1} \\delta_2^{\\delta_2})B(delta1,delta2)=(delta1+delta2)^(delta1+delta2)/(delta1^delta1 delta2^delta2).<\/span><\/div>, <div class='tooltip'>critVal<span class='tooltiptext'>This function calculates the critical value for a multiple contrast test via numerical integration (using the methods implemented in the mvtnorm package).<\/span><\/div>, <div class='tooltip'>emax<span class='tooltiptext'>The model function for the Emax model is defined as f(d,\\theta)=E_0+E_{max}\\frac{d}{ED_{50}+d}f(d,theta)=E0+Emax d/(ED50 + d).<\/span><\/div>, <div class='tooltip'>exponential<span class='tooltiptext'>The model function for the exponential model is defined as f(d,\\theta)=E_0+E_1(exp(d/\\delta)-1)f(d,theta)=E0+E1 (exp(d/delta)-1).<\/span><\/div>, <div class='tooltip'>fullMod<span class='tooltiptext'>Calculates location and scale parameters for all models in the candidate set using the maximum approach from Pinheiro et al. (2006). This is done by repeatedly calling the getPars function.<\/span><\/div>, <div class='tooltip'>genDFdata<span class='tooltiptext'>The function simulates normally distributed dose-response data, according to a prespecified dose-response model (or mean vector) and a given standard deviation.<\/span><\/div>, <div class='tooltip'>getPars<span class='tooltiptext'>Given the baseline, the maximum effect and the standardized model parameters this function calculates the location and scale parameters in the model function using the maximum approach, see Pinheiro et al. (2006) for the basic idea.<\/span><\/div>, <div class='tooltip'>guesst<span class='tooltiptext'>Calculates guesstimates for standardized model parameter(s) using the general approach described in Pinheiro et al. (2006).<\/span><\/div>, <div class='tooltip'>linear<span class='tooltiptext'>The model function for the linear model is defined as f(d,\\theta)=E_0+\\delta df(d,theta)=E0+delta d.<\/span><\/div>, <div class='tooltip'>linlog<span class='tooltiptext'>The model function for the linear in log model is defined as f(d,\\theta)=E_0+\\delta \\log(d + off)f(d,theta)=E0+delta log(d + off), where offoff is an offset parameter not estimated in the code.<\/span><\/div>, <div class='tooltip'>logistic<span class='tooltiptext'>The model function for the logistic model is defined as f(d, \\theta) = E_0 + E_{\\max}/\\left\\{1 + \\exp\\left[ \\left(ED_{50} - d \\right)/\\delta \\right] \\right\\}f(d,theta)=E0+Emax/(1 + exp((ED50-d)/delta)).<\/span><\/div>, <div class='tooltip'>LP<span class='tooltiptext'>Calculates the loss in power associated with misspecification of the standardized model parameters for a specific model.<\/span><\/div>, <div class='tooltip'>MCPMod<span class='tooltiptext'>Tests for a dose-response effect with a model-based multiple contrast test andestimates a target dose with regression techniques. For details seeBretz et al. (2005) or the enclosed vignette, available via the command vignette(\"MCPMod\").<\/span><\/div>, <div class='tooltip'>modelMeans<span class='tooltiptext'>Calculates the mean or standardized mean vectors for a candidate set of models. This function is mainly for internal use.<\/span><\/div>, <div class='tooltip'>mvtnorm.control<span class='tooltiptext'>Returns a list with control parameters (an object of class GenzBretz) for the pmvt and qmvt functions from the mvtnorm package, see the corresponding documentation for more information.<\/span><\/div>, <div class='tooltip'>planMM<span class='tooltiptext'>Calculates the optimal model contrasts, the critical value and the contrast correlation matrix, i.e. the quantities necessary to conduct the multiple contrast test for a given candidate set of dose-response models.<\/span><\/div>, <div class='tooltip'>plotModels<span class='tooltiptext'>Produces a trellis display of the model functions in the candidate set. The location and scale parameters of the models are determined by the base and maxEff arguments.<\/span><\/div>, <div class='tooltip'>powCalc<span class='tooltiptext'>Given the optimal contrasts, the sample size and a certain `alternative' (i.e. a mean vector and sigma), the function calculates the power to detect this alternative. See Pinheiro et al. (2006) for details. The function is the building block for the functions powerMM, sampSize and LP. Numerical integration routines from the mvtnorm package are used to calculate the underlying multivariate integrals.<\/span><\/div>, <div class='tooltip'>powerMM<span class='tooltiptext'>Calculates the power under the assumed candidate set for different sample sizes.<\/span><\/div>, <div class='tooltip'>quadratic<span class='tooltiptext'>The model function for the quadratic model is defined as f(d,\\theta)=E_0+\\beta_1d+\\beta_2d^2f(d,theta)=E0+beta1 d+beta2 d^2.<\/span><\/div>, <div class='tooltip'>sampSize<span class='tooltiptext'>Given a candidate set, the baseline effect, the maximum effect and the standard deviation, the sampSize function returns the smallest sample size achieving a certain combined power value. See Pinheiro et al. (2006) for details.<\/span><\/div>, <div class='tooltip'>sigEmax<span class='tooltiptext'>The model function for the sigmoid Emax model is defined as f(d,\\theta)=E_0+E_{max}\\frac{d^h}{ED^h_{50}+d^h}f(d,theta)=E0+Emax d^h/(ED50^h + d^h).<\/span><\/div>","<div class='tooltip'>AnalysisApp<span class='tooltiptext'>This function creates a Shiny-based graphical user interface to perform MCPMod-based analysis of a dose-finding trial. The application requires data sets with the dose and response information (comma-separated value file). The data set is required to include the dose and resp variables with a single record per patient. The first row of the file should contain the required variable names as follows: \"dose\",\"resp\" (including the quotation marks and the comma). Subsequent rows should contain comma-separated values corresponding to the dose and response values for each patient (quotation marks are not required for the rows of data). See for example the normal data set.<\/span><\/div>, <div class='tooltip'>AnalysisReport<span class='tooltiptext'>This function creates a detailed summary of MCPMod analysis results in a Microsoft Word format.<\/span><\/div>, <div class='tooltip'>ComputeDRFunctionParameters<span class='tooltiptext'>This function implements the simulation-based analysis of dose-finding clinical trials with normally distributed, binary and count endpoints using the MCPMod methodology. For more information, see the technical manual in the package's doc folder.<\/span><\/div>, <div class='tooltip'>EvaluateDRFunction<span class='tooltiptext'>This function implements the simulation-based analysis of dose-finding clinical trials with normally distributed, binary and count endpoints using the MCPMod methodology. For more information, see the technical manual in the package's doc folder.<\/span><\/div>, <div class='tooltip'>GenerateAnalysisReport<span class='tooltiptext'>This function creates a detailed summary of MCPMod analysis results in a Microsoft Word format.<\/span><\/div>, <div class='tooltip'>GenerateSimulationReport<span class='tooltiptext'>This function creates a detailed summary of MCPMod simulation results in a Microsoft Word format.<\/span><\/div>, <div class='tooltip'>MCPModAnalysis<span class='tooltiptext'>This function implements the MCPMod-based analysis of dose-finding clinical trials with normally distributed, binary and count endpoints, including derivation of the optimal contrasts for the candidate dose-response models, evaluation of dose-response tests based on the optimal contrasts, selection of the significant dose-response models and estimation of the target dose. For more information, see the technical manual in the package's doc folder.<\/span><\/div>, <div class='tooltip'>MCPModSimulation<span class='tooltiptext'>This function implements the simulation-based analysis of dose-finding clinical trials with normally distributed, binary and count endpoints using the MCPMod methodology. For more information, see the technical manual in the package's doc folder.<\/span><\/div>, <div class='tooltip'>SimulationApp<span class='tooltiptext'>This function creates a Shiny-based graphical user interface to perform a simulation-based evaluation of dose-finding trial designs using MCPMod.<\/span><\/div>, <div class='tooltip'>SimulationReport<span class='tooltiptext'>This function creates a detailed summary of MCPMod simulation results in a Microsoft Word format.<\/span><\/div>, <div class='tooltip'>TestDoseResponseFunction<span class='tooltiptext'>The MCPModPack package facilitates the design and analysis of dose-finding clinical trials with normally distributed, binary and count endpoints using the MCPMod methodology.<\/span><\/div>, <div class='tooltip'>TestFindTargetDose<span class='tooltiptext'>The MCPModPack package facilitates the design and analysis of dose-finding clinical trials with normally distributed, binary and count endpoints using the MCPMod methodology.<\/span><\/div>","<div class='tooltip'>axislabel<span class='tooltiptext'>Generic axis label, with method for 'data.frame'.<\/span><\/div>, <div class='tooltip'>boxplot_data_frame<span class='tooltiptext'>Boxplot for data.frame. Creates a boxplot using boxplot_panel by default.<\/span><\/div>, <div class='tooltip'>boxplot_panel<span class='tooltiptext'>Panel function for metaplot boxplot. Calls, panel.bwplot and plots reference lines if ref has length.<\/span><\/div>, <div class='tooltip'>categorical<span class='tooltiptext'>Categorical Plot. Generic, with method for 'data.frame'.<\/span><\/div>, <div class='tooltip'>categorical_data_frame<span class='tooltiptext'>Categorical function for class 'data.frame'. Implements a simple mosaic plot.<\/span><\/div>, <div class='tooltip'>categorical_panel<span class='tooltiptext'>Default panel function for categorical_data_frame. Implements a simple mosaic plot.Global options are supported but typically are supplied by the calling function andmay therefore be unreachable.<\/span><\/div>, <div class='tooltip'>cax<span class='tooltiptext'>Calculates axis labels and positions for categorical values.<\/span><\/div>, <div class='tooltip'>corsplom<span class='tooltiptext'>Scatterplot matrix with correlations.<\/span><\/div>, <div class='tooltip'>corsplom_data_frame<span class='tooltiptext'>Creates a scatterplot matrix with correlations in lower panel, by default.<\/span><\/div>, <div class='tooltip'>corsplom_gg_correlation<span class='tooltiptext'>Default lower panel function for corsplom_data_frame() with gg = TRUE. Plots Pearson correlation coefficient.corsplom_data_frame() typically supplies smooth aesthetics in case smooth is desired; smooth arguments are defined tosuppress corresponding warnings.<\/span><\/div>, <div class='tooltip'>corsplom_gg_diagonal<span class='tooltiptext'>Default diagonal panel function for corsplom_data_frame with gg = TRUE. Plots a density smooth against the corresponding axis from within the diagonal panel. Plots a grey pin at each axis zero. Formats and displays the variable name.<\/span><\/div>, <div class='tooltip'>corsplom_gg_scatter<span class='tooltiptext'>Default upper panel function for corsplom_data_frame with gg = TRUE. Plots data with loess smooth.<\/span><\/div>, <div class='tooltip'>corsplom_panel_correlation<span class='tooltiptext'>Default lower panel function for corsplom_data_frame. Plots Pearson correlation coefficient.<\/span><\/div>, <div class='tooltip'>corsplom_panel_diagonal<span class='tooltiptext'>Default diagonal panel function for corsplom_data_frame. Plots a density smooth against the corresponding axis from within the diagonal panel. Plots a grey pin at each axis zero.<\/span><\/div>, <div class='tooltip'>corsplom_panel_scatter<span class='tooltiptext'>Default upper panel function for corsplom_data_frame. Plots data with loess smooth.<\/span><\/div>, <div class='tooltip'>dens_panel<span class='tooltiptext'>Default panel function for dens_data_frame. Calls panel.densityplot, and plots reference lines if ref has length.<\/span><\/div>, <div class='tooltip'>densplot<span class='tooltiptext'>Creates a density plot.<\/span><\/div>, <div class='tooltip'>densplot_data_frame<span class='tooltiptext'>Plot density for object of class 'data.frame' using dens_panel by default.<\/span><\/div>, <div class='tooltip'>diag_label<span class='tooltiptext'>Formats a diagonal label. Can return a simple column name, a column label (if attribute defined), a fractured column label (split on spaces), or a processed symbol (over-rides label).<\/span><\/div>, <div class='tooltip'>diag_pin<span class='tooltiptext'>Calculates pin placement in the density region, inside margin of diagonal panels.<\/span><\/div>, <div class='tooltip'>iso_prepanel<span class='tooltiptext'>Prepanel function for isometric axes. Returns join minimum and maximum for limits on both axes.<\/span><\/div>, <div class='tooltip'>metaplot<span class='tooltiptext'>Metaplot creates univariate, bivariate, or multivariate plots depending on the number and types of variables represented by the anonymous arguments. Types are either numeric (NUM, e.g. real, integer) or categorical (CAT, e.g. factor, character). A variable stored as numeric that nonetheless has an encoded guide attribute will be treated as categorical. Mnemonic: x %>% metaplot(yvars, xvar, groupvar, facets) where arguments are unquoted column names, and only xvar is required. Column attributes label, guide, reference, and symbol modify the behavior of the default handlers.<\/span><\/div>, <div class='tooltip'>metaplot_key<span class='tooltiptext'>Default key function for constructing scatterplot legends.<\/span><\/div>, <div class='tooltip'>metaplot_ref<span class='tooltiptext'>Calculates reference values for x and y axes. Coerces column attribute 'reference' to numeric: a single value or an encoding giving multiple numeric values (decodes are ignored).<\/span><\/div>, <div class='tooltip'>metastats<span class='tooltiptext'>Formats GLM statistics. Uses a gaussian family by default, or binomial family if all y are 0 or 1, to fit a general linear model. Formats number of observations, p-value, and Pearson correlation coefficient into a string for printing.<\/span><\/div>, <div class='tooltip'>metOption<span class='tooltiptext'>Gets a metaplot option value from the named list getOption('metaplot').If an exact match is not found, trailing elements of x, separated by underscore,are removed one at a time in search of a partial match. Thus 'ref.col' will matchfor 'ref.col_dens' and 'ref.col_scatter' if neither of these is set (allowingselective override). However, global' will never match 'global.col'.<\/span><\/div>, <div class='tooltip'>model<span class='tooltiptext'>Executes a linear model, automatically choosing binomial family as necessary.<\/span><\/div>, <div class='tooltip'>multiplot<span class='tooltiptext'>Arranges mutiple trellis plots or ggplots in a grid, automatically choosing number of rows and columns. By default, number of rows is one less than or equal to the number of columns.<\/span><\/div>, <div class='tooltip'>pack<span class='tooltiptext'>Pack Something. Generic, with method for data.frame.<\/span><\/div>, <div class='tooltip'>panel_tile<span class='tooltiptext'>Draws a tile in a mosaic.<\/span><\/div>, <div class='tooltip'>panel.meta_densityplot<span class='tooltiptext'>Variant of panel.densityplot that supports filled area and alpha points.<\/span><\/div>, <div class='tooltip'>region<span class='tooltiptext'>Calculates a confidence region. se.fit from predict.glm is multiplied by z and added or subtracted from fits to give hi and lo columns in return value. z is normal quantile for the one-tailed probablitity corresponding to conf, e.g. ~ 1.96 for conf = 0.95. If non-missing y is only 0 or 1, the model family is binomial and resulting confidence intervals are back-transformed using plogis.<\/span><\/div>, <div class='tooltip'>scatter<span class='tooltiptext'>Scatterplot.<\/span><\/div>, <div class='tooltip'>scatter_data_frame<span class='tooltiptext'>Scatterplot function for class 'data.frame'.<\/span><\/div>, <div class='tooltip'>scatter_panel<span class='tooltiptext'>Default panel function for scatter_data_frame. Calls panel.xyplotand optionally plots linear fit, confidence region, reference lines, and statistics.Note that, although global options are supported, typically these are unreachablesince the calling function supplies appropriate values.<\/span><\/div>, <div class='tooltip'>scatter_panel_ref<span class='tooltiptext'>Calculates reference values for x and y axes at the panel level.<\/span><\/div>, <div class='tooltip'>setOption<span class='tooltiptext'>Sets an option value in the list getOption('metaplot').If invoked without named arguments, option 'metaplot' is set to NULL.Setting an existing option moves it to the end of the list (breaks ties in metOption).<\/span><\/div>, <div class='tooltip'>test_metaplot<span class='tooltiptext'>Tests metaplot variants by example. Returns null. Use example(test_metaplot).<\/span><\/div>, <div class='tooltip'>tiles<span class='tooltiptext'>Calculates limits for mosaic tiles<\/span><\/div>, <div class='tooltip'>tilestats<span class='tooltiptext'>Formats statistics for a mosaic tile.<\/span><\/div>, <div class='tooltip'>unpack<span class='tooltiptext'>Unpack Something. Generic, with method for data.frame.<\/span><\/div>, <div class='tooltip'>wikisym2plotmath<span class='tooltiptext'>Converts wiki symbol to plotmath. Vectorized version of wikisym2plotmath_.<\/span><\/div>, <div class='tooltip'>wikisym2plotmath_<span class='tooltiptext'>Converts one wiki symbol to plotmath. A Wiki symbol is simple text with arbitrarily nested subscript (_) and superscript (^) groupings. Use dot (.) to explicitly terminate a grouping, and use backslash-dot (\\.) for a literal dot. Examples: V_c./F. Trailing dots need not be supplied. Leading/trailing whitespace is removed. Tab character not allowed.<\/span><\/div>","<div class='tooltip'>Lattice<span class='tooltiptext'>Generate good rank-1 lattice points with prime number of points by using the fast component-by-component construction algorithm of Nuyens and Cools (2006). Refer Nuyens (2007) for more details.<\/span><\/div>, <div class='tooltip'>mined<span class='tooltiptext'>Generate MinED samples from an unnormalized density function.%% ~~ A concise (1-5 lines) description of what the function does. ~~<\/span><\/div>, <div class='tooltip'>SelectMinED<span class='tooltiptext'>Select MinED samples from candidates by optimizing the generalized MinED criterion in Joseph et al. (2018).<\/span><\/div>","<div class='tooltip'>fact.pts<span class='tooltiptext'>For a given number of factors, say k, it generates 2^k factorialpoints such that the number of changes in the run sequenceis minimum.<\/span><\/div>, <div class='tooltip'>fact.pts_2<span class='tooltiptext'>For a given number of factors, say k, it generates 2^(k-1) factorialpoints such that the number of changes in levels in the halfreplicate is minimum.<\/span><\/div>, <div class='tooltip'>min_bbd<span class='tooltiptext'>Generate Box Behnken design (BBD) with minimum level changesin the run sequence.<\/span><\/div>, <div class='tooltip'>min_ccd<span class='tooltiptext'>Generate minimally changed run sequence for central compositedesigns (CCD).<\/span><\/div>, <div class='tooltip'>min_ccd_2<span class='tooltiptext'>Generate Minimally Changed Central Composite designs (CCD) with Fractional Factorial Points (Half Replicate)<\/span><\/div>, <div class='tooltip'>num.ch<span class='tooltiptext'>For a given design in matrix form, it counts the number of changesin levels of each factor as well as overall number of changes inthe design.<\/span><\/div>","<div class='tooltip'>calculate_entropy<span class='tooltiptext'>The function is a wrapper for calc_entropy and is meant forexternal use by the user. The goal for the decision maker isto choose an action such that the observer is left as clueless as possible as to which rule governs his actions, i.e. maximize entropy.<\/span><\/div>, <div class='tooltip'>calculate_payouts<span class='tooltiptext'>The function is a wrapper function for calc_payout_obs andcalc_payout_dm, and exported to be used by the user. It calculates the expected payout to both observers and decision makers for each possible action undertaken by the decision maker, and the observers choice of whether or not to try and guessthe rule.<\/span><\/div>, <div class='tooltip'>extract_attr<span class='tooltiptext'>Extracts the attributes of objects nested in a list<\/span><\/div>, <div class='tooltip'>generate_designs<span class='tooltiptext'>The function takes the list of design options design_opt_input and generates one or more obfuscation designs subject to the specified restrictions. A full specification of all the options available can be foundin the manual along with detailed examples of different designs. At a minimumthe user must supply the number of rules and actions, i.e. the dimensions ofthe design problem.<\/span><\/div>, <div class='tooltip'>last<span class='tooltiptext'>last extracts the last element of a vector<\/span><\/div>, <div class='tooltip'>print_design<span class='tooltiptext'>Takes a design or list of designs and prints them to the console. To store a design, please see save_design. Depending on the print options,additional text is provided with information on the considered rule and/orthe design generation process.<\/span><\/div>, <div class='tooltip'>print_entropy<span class='tooltiptext'>The function prints the vector of entropies for each possible action. Depending on printing options, additional information about the probabilitycalculations can be provided.<\/span><\/div>, <div class='tooltip'>print_payout<span class='tooltiptext'>The function formats and prints the payout to the observer and decisionmaker.<\/span><\/div>, <div class='tooltip'>save_design<span class='tooltiptext'>The function takes a design or a list of designs and stores them in .csvfiles in the specified folder.<\/span><\/div>","<div class='tooltip'>Bfct<span class='tooltiptext'>Bfct returns the power basis functions of a spline of given degree.<\/span><\/div>, <div class='tooltip'>Estimate_PLMODS<span class='tooltiptext'>Estimate_PLMODS computes the estimate of parameters in a partiallinear model in the setting of outcome-dependent sampling. See details inZhou, Qin and Longnecker (2011).<\/span><\/div>, <div class='tooltip'>gcv_ODS<span class='tooltiptext'>gcv_ODS calculates the generalized cross-validation (GCV) forselecting the smoothing parameter in the setting of outcome-dependentsampling. The details can be seen in Zhou, Qin and Longnecker (2011) and itssupplementary materials.<\/span><\/div>, <div class='tooltip'>logspace<span class='tooltiptext'>logspace generates n logarithmically spaced points between 10^d1 and10^d2. The utility of this function is equivalent to logspace function inmatlab.<\/span><\/div>, <div class='tooltip'>odsmle<span class='tooltiptext'>odsmle provides a maximum semiparametric empirical likelihood estimator(MSELE) for analyzing the primary outcome Y with respect to expensive exposureand other covariates in ODS design (Zhou et al. 2002).<\/span><\/div>, <div class='tooltip'>quantileknots<span class='tooltiptext'>quantileknots creates knots at sample quantiles<\/span><\/div>, <div class='tooltip'>se.spmle<span class='tooltiptext'>se.spmle calculates the standard error for MSELE estimator in Zhou etal. 2002<\/span><\/div>, <div class='tooltip'>secondary_casecohort<span class='tooltiptext'>secondary_casecohort performs the secondary analysis which describesthe association between a continuous secondary outcome and the expensiveexposure for case-cohort data.<\/span><\/div>, <div class='tooltip'>secondary_ODS<span class='tooltiptext'>secondary_ODS performs the secondary analysis which describes theassociation between a continuous scale secondary outcome and the expensiveexposure for data obtained with ODS (outcome dependent sampling) design.<\/span><\/div>","<div class='tooltip'>ag.test<span class='tooltiptext'>ag.test performs Alexander-Govern test.<\/span><\/div>, <div class='tooltip'>aov.test<span class='tooltiptext'>aov.test performs one-way analysis of variance (ANOVA).<\/span><\/div>, <div class='tooltip'>bf.test<span class='tooltiptext'>bf.test performs Brown-Forsythe test.<\/span><\/div>, <div class='tooltip'>box.test<span class='tooltiptext'>box.test performs Box F test.<\/span><\/div>, <div class='tooltip'>describe<span class='tooltiptext'>describe produces basic descritive statistics including sample size, mean, standard deviation, median, minimum value, maximum value, 25th quantile, 75th quantile, skewness, kurtosis, the number of missing value.<\/span><\/div>, <div class='tooltip'>gp.test<span class='tooltiptext'>gp.test tests whether two or more samples from normal distributions have the same means when the variances are not necessarily equal.<\/span><\/div>, <div class='tooltip'>gplot<span class='tooltiptext'>gplot produce box-and-whisker plots and error bars of the given grouped values.<\/span><\/div>, <div class='tooltip'>homog.test<span class='tooltiptext'>homog.test performs variance homogeneity tests including Levene, Bartlett, Fligner-Killeen tests.<\/span><\/div>, <div class='tooltip'>james.test<span class='tooltiptext'>james.test performs James second order test.<\/span><\/div>, <div class='tooltip'>johansen.test<span class='tooltiptext'>johansen.test performs Johansen F test.<\/span><\/div>, <div class='tooltip'>kw.test<span class='tooltiptext'>kw.test performs Kruskal-Wallis test.<\/span><\/div>, <div class='tooltip'>mw.test<span class='tooltiptext'>mw.test performs Mann-Whitney U test for two samples.<\/span><\/div>, <div class='tooltip'>nor.test<span class='tooltiptext'>nor.test performs normality tests including Shapiro-Wilk, Shapiro-Francia, Kolmogorov-Smirnov, Anderson-Darling, Cramer-von Mises, Pearson Chi-square tests, and also assess the normality of each group through plots.<\/span><\/div>, <div class='tooltip'>paircomp<span class='tooltiptext'>paircomp is a generic function for pairwise comparisons by adjusting p-values.<\/span><\/div>, <div class='tooltip'>ss.test<span class='tooltiptext'>ss.test performs Scott-Smith test.<\/span><\/div>, <div class='tooltip'>st.test<span class='tooltiptext'>st.test performs student's t-test for two samples.<\/span><\/div>, <div class='tooltip'>welch.test<span class='tooltiptext'>welch.test performs Welch's heteroscedastic F test and Welch's heteroscedastic F test with trimmed means and Winsorized variances.<\/span><\/div>, <div class='tooltip'>wt.test<span class='tooltiptext'>wt.test performs Welch's t-test for two samples.<\/span><\/div>","<div class='tooltip'>CRcompare<span class='tooltiptext'>Computes bootstrapped confidence intervals for the mean and mediandistance between the optima of two response surface models. Models can be thinplate splines or quadratic polynomials\\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}DelCastilloCROptimaRegion(Del Castillo et al. 2019).<\/span><\/div>, <div class='tooltip'>GloptiPolyR<span class='tooltiptext'>Optimize a quadratic or cubic polynomial functionin 2 ~ 5 variableswith bound constraints\\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}DelCastilloCROptimaRegion(Del Castillo et al. 2019).<\/span><\/div>, <div class='tooltip'>GloptiPolyRegion<span class='tooltiptext'>Computes and displays an approximated (1 - alpha) confidence region (CR) forthe bound-constrained maximum of a polynomial regression model in up to cubic orderwith up to 5 controllable factors\\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}DelCastilloCROptimaRegion(Del Castillo et al. 2019).<\/span><\/div>, <div class='tooltip'>OptRegionQuad<span class='tooltiptext'>Computes and displays an approximated (1 - alpha) confidence region (CR) forthe linear-constrained maximum of a quadratic polynomial regression modelin 2 controllable factors\\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}DelCastilloCROptimaRegion(Del Castillo et al. 2019).Grey region on output plot is the approximate CR.The CR is computed as the convex hull of the coordinates of the optima foundfrom simulating nosim quadratic polynomial regressions to the data (therefore,it is an approximate CR). The mean value of the optimum is shown as a red point,and a smoothed contour plot of the X,y data obtained via thin plate splines isshown as well.<\/span><\/div>, <div class='tooltip'>OptRegionTps<span class='tooltiptext'>Computes and displays an approximated (1 - alpha) confidence region (CR) forthe linear-constrained maximum of a penalized Thin Plate Spline (TPS) modelin 2 controllable factors\\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}DelCastilloCROptimaRegion(Del Castillo et al. 2019).Generates a PDF file with a graph displaying the CR.Grey region on output plot is the approximate CR. The mean coordinates (centroid)of the optima is shown as a red point.<\/span><\/div>","<div class='tooltip'>FixDes<span class='tooltiptext'>Find the sample size, duration of accrual, and test boundary for a single-stage design with an event-free endpoint versus either a known standard control or a randomized comparative control. Testing is one-sided based on the Kaplan-Meier estimator evaluated at a user-specified time point.<\/span><\/div>, <div class='tooltip'>np.OptimDes<span class='tooltiptext'>Construct a two-stage or three-stage design with a time-to-event endpoint evaluated at a pre-specified time (e.g., 6-month progression-free survival) comparing treatment versus either a historical control rate with possible stopping for futility (single-arm), or an active control arm with possible stopping for both futility and superiority (two-arm), after the end of Stage 1 utilizing time to event data. The design minimizes either the expectedduration of accrual (EDA), the expected sample size (ES), or the expected total study length(ETSL). The maximum combined sample size for both stages ispre-specifed by the user.<\/span><\/div>, <div class='tooltip'>OptimDes<span class='tooltiptext'>Construct an optimal two-stage or three-stage designs with a time-to-event endpoint evaluated at a pre-specified time (e.g., 6 months) comparing treatment versus either a historical control rate with possible stopping for futility (single-arm), or an active control arm with possible stopping for both futility and superiority (two-arm), after the end of Stage I utilizing time to event data. The design minimizes either the expectedduration of accrual (EDA), expected sample size (ES), or the expected total study length(ETSL).<\/span><\/div>, <div class='tooltip'>OptimDesControl<span class='tooltiptext'>Set parameters controlling numerical methods for OptimDes and np.OptimDes<\/span><\/div>, <div class='tooltip'>plot.OptimDes<span class='tooltiptext'>Output from function OptimDes is used to display the ETSL, ES andEDA for a two-stage or three-stage design relative to a single-stage design as afunction of the combined-stage sample size or study length.<\/span><\/div>, <div class='tooltip'>print.OptimDes<span class='tooltiptext'>Print an object output by OptimDes.<\/span><\/div>, <div class='tooltip'>SimDes<span class='tooltiptext'>Simulation experiments to compare the alpha level, power and other features of two-stage or three-stage designs from functionOptimDes with the targetted values.<\/span><\/div>, <div class='tooltip'>TestStage<span class='tooltiptext'>This function performs the hypothesis tests for the two-stage or three-stage designswith event-free endpoint from OptimDes.<\/span><\/div>, <div class='tooltip'>weibPmatch<span class='tooltiptext'>Determine the shape or scale parameter of a Weibull distribution soit has event-free rate P0 at time x. If the shape is specified, the scaleparameter is computed, and if the scale is specified, the shapeparameter is computed.<\/span><\/div>, <div class='tooltip'>weibull.plot<span class='tooltiptext'>Plot Weibull survival curves with differences at a target time highlighted. Designed for use with the paramvalues input to function OptimDes.<\/span><\/div>","<div class='tooltip'>crmsimulations<span class='tooltiptext'>This function performs simulations to assess design performance characteristics of the proportional odds model (POM),continuation ratio (CR) Model, and binary CRM likelihood-based dose finding designs.There are many different options in this functionto vary sample size, cohort sizes, target dose-limiting toxicity (DLT) rates, true underlying dose-toxicity models,discrete or continuous dose levels, combining ordianl grades 0 and 1 into one category, and the design you wish to assess.Returns results collected over all simulations run for these specified safety and dose estimation performance criteria.<\/span><\/div>, <div class='tooltip'>nextdose<span class='tooltiptext'>This function incorporates pseudodata created in pseudodata with the collected cohortsof patient information and updates the fitted model selected (choice of POM, CR, or binary 2-parameterlogistic model). Using this newly updated model, it then re-estimates the next dose to be tested fora specified target dose limiting toxicity (DLT) rate. It has the flexibility to add in varioussafety constraints and can accommodate continuous or discrete doses. This function also has to option tocombine toxicity grades 0 and 1 as specified by CTCAEv4.0 into one category should clinical investigatorswish to do so.<\/span><\/div>, <div class='tooltip'>pseudodata<span class='tooltiptext'>This function creates 2 pseudodata vectors (toxicity and dose levels) necessary to run any of the3 variations of the likelihood-based CRM design. If using the ordinal POM or CR model design,ordinal toxicity grades are specified by CTCAEv4.0. This function also specifies the starting dosefor a trial with the choice of continuous or discrete dose levels for a specified target dose limitingtoxicity (DLT) rate.<\/span><\/div>","<div class='tooltip'>beta0<span class='tooltiptext'>When conducting power calculations, one is often interested in examining power for various 'effect sizes'. Suppose the logistic regression is specified via the vector of coefficients (beta0, betaX); the first element is the intercept and the second consists of a vector of log odds ratio parameters. In many settings, the overall outcome prevalence in the population of interest is known or, at least, fixed. Modifying any given element of betaX will automatically modify the overall prevalence, unless there is a corresponding change in beta0. The function beta0() calculates the value of beta0 that minimizes the difference between the target outcome prevalence, rhoY, and prevalence induced by the model in conjuction with the assumed marginal exposure distribution.<\/span><\/div>, <div class='tooltip'>ccPower<span class='tooltiptext'>Monte Carlo based estimation of statistical power for maximum likelihood estimator (MLE) of the components of a logistic regression model, based on the case-control design.<\/span><\/div>, <div class='tooltip'>ccSim<span class='tooltiptext'>Monte Carlo based evaluation of operating characteristics of the maximum likelihood estimator (MLE) for the coefficients of a logistic regression model, based on the case-control.<\/span><\/div>, <div class='tooltip'>phaseI<span class='tooltiptext'>phaseI() provides the expected phase I counts, based on a pre-specified population and outcome model. If phase II sample sizes are provided, the (expected) phase II sampling probabilities are also reported.<\/span><\/div>, <div class='tooltip'>plotPower<span class='tooltiptext'>The plotPower function plots estimates of power obtained from objects returned by either the tpsPower or ccPower functions.<\/span><\/div>, <div class='tooltip'>rmvhyper<span class='tooltiptext'>Generates a single random deviate from a multivariate hypergeometric distribution.<\/span><\/div>, <div class='tooltip'>tps<span class='tooltiptext'>Fits a logistic regression model to data arising from two phase designs<\/span><\/div>, <div class='tooltip'>tpsPower<span class='tooltiptext'>Monte Carlo based estimation of statistical power for estimators of the components of a logistic regression model, based on balanced two-phase and case-control study designs (Breslow and Chatterjee, 1999; Prentice and Pykle, 1979).<\/span><\/div>, <div class='tooltip'>tpsSim<span class='tooltiptext'>Monte Carlo based evaluation of operating characteristics for estimators of the components of a logistic regression model, based on the two-phase and case-control study designs (Breslow and Chatterjee, 1999; Prentice and Pykle, 1979).<\/span><\/div>","<div class='tooltip'>fig_PharmPow_crossover<span class='tooltiptext'>This funcion enables the figure customisation for power calculations of a mixed crossover study design<\/span><\/div>, <div class='tooltip'>fig_PharmPow_parallel<span class='tooltiptext'>This funcion enables the figure customisation for power calculations of a mixed parallel study design<\/span><\/div>, <div class='tooltip'>PharmPow_crossover<span class='tooltiptext'>This function performs power calculations for mixed crossover study designs. The input data for these functions is taylored for NONMEM .phi files.<\/span><\/div>, <div class='tooltip'>PharmPow_parallel<span class='tooltiptext'>This function performs power calculations for mixed parallel study designs. The input data for these functions is taylored for NONMEM .phi files.<\/span><\/div>","<div class='tooltip'>plfMA<span class='tooltiptext'>A function to launch the plfMA GUI.<\/span><\/div>","<div class='tooltip'>addpall.CGP<span class='tooltiptext'>Add sufficient data common to all particles to the global pall variable, a mnemonic for particles-all, for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>addpall.ConstGP<span class='tooltiptext'>Add sufficient data common to all particles to the global pall variable, a mnemonic for particles-all, for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>addpall.GP<span class='tooltiptext'>Add sufficient data common to all particles to the global pall variable, a mnemonic for particles-all, for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>alc.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>alc.const.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>alc.ConstGP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>alc.GP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.alcs<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.ecis<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.eis<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.ents<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.ieci<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.iecis<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.ktKik.x<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc.vars<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>calc2.ktKik.x<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>covar<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>covar.sep<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>covar.sim<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>cv.folds<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>data.CGP<span class='tooltiptext'>Functions to supply data to PL for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>data.CGP.adapt<span class='tooltiptext'>Functions to supply data to PL for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>data.ConstGP.improv<span class='tooltiptext'>Functions to supply data to PL for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>data.GP<span class='tooltiptext'>Functions to supply data to PL for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>data.GP.improv<span class='tooltiptext'>Functions to supply data to PL for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>dist2covar.symm<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>distance<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>draw.CGP<span class='tooltiptext'>Functions for using Metropolis-Hastings (MH) to evolve a particle according to the posterior distribution given by a Gaussian process (GP) for regression, classification, or combined unknown constraint model<\/span><\/div>, <div class='tooltip'>draw.ConstGP<span class='tooltiptext'>Functions for using Metropolis-Hastings (MH) to evolve a particle according to the posterior distribution given by a Gaussian process (GP) for regression, classification, or combined unknown constraint model<\/span><\/div>, <div class='tooltip'>draw.GP<span class='tooltiptext'>Functions for using Metropolis-Hastings (MH) to evolve a particle according to the posterior distribution given by a Gaussian process (GP) for regression, classification, or combined unknown constraint model<\/span><\/div>, <div class='tooltip'>EI<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>ei.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>entropy<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>entropy.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>entropy.bvsb<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>exp2d.C<span class='tooltiptext'>Generates 2-d classification data with two or three class labels, based on the Hessian data from a 2-d real-valued response<\/span><\/div>, <div class='tooltip'>findmin.ConstGP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>findmin.GP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>getmap.CGP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>getmap.GP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>ieci.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>ieci.const.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>ieci.ConstGP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>ieci.GP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>init.CGP<span class='tooltiptext'>Functions for initializing particles for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>init.ConstGP<span class='tooltiptext'>Functions for initializing particles for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>init.GP<span class='tooltiptext'>Functions for initializing particles for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>latents.CGP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>lpost.GP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>lpredprob.CGP<span class='tooltiptext'>Log-predictive probability calculation for Gaussian process (GP) regression, classification, or combined unknown constraint models; primarily to be used particle learning (PL) re-sample step<\/span><\/div>, <div class='tooltip'>lpredprob.ConstGP<span class='tooltiptext'>Log-predictive probability calculation for Gaussian process (GP) regression, classification, or combined unknown constraint models; primarily to be used particle learning (PL) re-sample step<\/span><\/div>, <div class='tooltip'>lpredprob.GP<span class='tooltiptext'>Log-predictive probability calculation for Gaussian process (GP) regression, classification, or combined unknown constraint models; primarily to be used particle learning (PL) re-sample step<\/span><\/div>, <div class='tooltip'>mindist.adapt<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>papply<span class='tooltiptext'>Applies a user-specified function to each particle contained in the global variables peach and pall, collecting the output in a data.frame<\/span><\/div>, <div class='tooltip'>params.CGP<span class='tooltiptext'>Extract parameters from particles for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>params.ConstGP<span class='tooltiptext'>Extract parameters from particles for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>params.GP<span class='tooltiptext'>Extract parameters from particles for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>phist<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>PL<span class='tooltiptext'>Implements the Particle Learning sequential Monte Carlo algorithm on the data sequence provided, using re-sample and propagate steps<\/span><\/div>, <div class='tooltip'>PL.clear<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>pred.CGP<span class='tooltiptext'>Prediction on a per-particle basis for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>pred.ConstGP<span class='tooltiptext'>Prediction on a per-particle basis for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>pred.GP<span class='tooltiptext'>Prediction on a per-particle basis for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>pred.mean.GP<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>prior.CGP<span class='tooltiptext'>Generate priors for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>prior.ConstGP<span class='tooltiptext'>Generate priors for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>prior.GP<span class='tooltiptext'>Generate priors for Gaussian process (GP) regression, classification, or combined unknown constraint models<\/span><\/div>, <div class='tooltip'>propagate.CGP<span class='tooltiptext'>Incorporation of a new data point for Gaussian process (GP) regression, classification, or combined unknown constraint models; primarily to be used particle learning (PL) propagate step<\/span><\/div>, <div class='tooltip'>propagate.ConstGP<span class='tooltiptext'>Incorporation of a new data point for Gaussian process (GP) regression, classification, or combined unknown constraint models; primarily to be used particle learning (PL) propagate step<\/span><\/div>, <div class='tooltip'>propagate.GP<span class='tooltiptext'>Incorporation of a new data point for Gaussian process (GP) regression, classification, or combined unknown constraint models; primarily to be used particle learning (PL) propagate step<\/span><\/div>, <div class='tooltip'>rectscale<span class='tooltiptext'>Scale data lying in an arbitrary rectangle to lie in the unit rectangle, and back again<\/span><\/div>, <div class='tooltip'>rectunscale<span class='tooltiptext'>Scale data lying in an arbitrary rectangle to lie in the unit rectangle, and back again<\/span><\/div>, <div class='tooltip'>renorm.lweights<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>renorm.weights<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>resample<span class='tooltiptext'>Internal plgp functions<\/span><\/div>, <div class='tooltip'>tquants<span class='tooltiptext'>Internal plgp functions<\/span><\/div>","<div class='tooltip'>populationPDXdesignApp<span class='tooltiptext'>Runs the 'shiny' app.<\/span><\/div>","<div class='tooltip'>boot.power.anova<span class='tooltiptext'>This function bootstraps the power of each effect in an ANOVA design for a given rangeof sample sizes. Power is computed by randomly drawing samples from a multivariate normaldistribution specified according to the values supplied by the design.anovaobject. Power is defined as the proportion of bootstrap iterations the p-values of eacheffect lie below the supplied alpha level. Note that this function runs many ANOVAs whichmight be slow for large sample size ranges or bootstrap iterations (see Details below).Further note that this function does not check for assumptions such as sphericity.<\/span><\/div>, <div class='tooltip'>design.anova<span class='tooltiptext'>Constructs an \"design.anova\" object required by the boot.power.anova function.<\/span><\/div>","<div class='tooltip'>powerGE<span class='tooltiptext'>This routine carries out (analytical, approximate) power calculations for identifying Gene-Environment interactions in Genome Wide Association Studies<\/span><\/div>, <div class='tooltip'>powerGG<span class='tooltiptext'>This routine carries out (analytical, approximate) power calculations for identifying Gene-Gene interactions in Genome Wide Association Studies<\/span><\/div>, <div class='tooltip'>powerGWASinteraction<span class='tooltiptext'>This function is depreciated and has been replaced by powerGG and powerGE<\/span><\/div>","<div class='tooltip'>fnfam<span class='tooltiptext'>An internal function called by the 'ntdt' function based on formulae from Abel and Muller-Myhsok (1998) Am J Hum Genet 63:664-667<\/span><\/div>, <div class='tooltip'>fx<span class='tooltiptext'>Internal function for the ptdt function, based on formulae from Abel and Muller-Myhsok (1998)Am J Hum Genet 63:664-667<\/span><\/div>, <div class='tooltip'>normplot<span class='tooltiptext'>This is an interactive program designed to illustratethe basic principles of what controls power. Here, we assume thatunder the null, the statistic is distributed as a standard normalwith mean zero and variance one, while under the alternative, thestatistic is distributed as a standard normal with a shifted mean.Power is a function of the mean under the alternative hypothesisand the desired significance level. The 'Options' boxallows the user to select the desired significance level using the radiobuttons, and to setthe mean under the alternative hypothesis using the slider.<\/span><\/div>, <div class='tooltip'>nsibs<span class='tooltiptext'>The nsibs() function will compute the number of affected sibpairsneeded to detect linkage to a susceptibility gene attributed with auser-specified lambda sib (ls) and lambda off (lo). Lambda sib is therecurrence risk ratio for full- sibs (includes dominance variance),lambda off is the recurrence risk ratio for parent- offspring pairs (sodoes not include dominance variance). In these calculations we assume thatparents and sibs are completely genotyped with markers that perfectlydefine the IBD configurations. alpha and beta correspond to the type 1and 1 - type 2 error rates respectively.<\/span><\/div>, <div class='tooltip'>ntdt<span class='tooltiptext'>Calculation of how many TDT trios need to be studied in order to havean adequately powered TDT study. The user specifies the susceptibilitygene allele frequency (q), the frequency of the marker allele (m) inlinkage disequilibrium with the susceptibility allele, the strength ofthe linkage disequilibrium between marker and susceptibility allele (ld),the genotype risk ratio for the susceptibility gene (g), the desired power(power) and the type 1 error rate (alpha) allowing for multiple testing.<\/span><\/div>, <div class='tooltip'>ntdt.q<span class='tooltiptext'>Power for a TDT study will be highest when the frequency of thesusceptibility allele (q) matches the frequency of the associated allele(m). We can now examine this by using the ntdt.q() function, whichreturns a table give the required sample sizes (and log sample sizes) fora range of values of q, at three different levels of ld. These levels are(1) the maximum (dmax), (2) 75% of the maximum (dmax.75), and (3) 50%of the maximum (dmax.50).<\/span><\/div>, <div class='tooltip'>plotDom<span class='tooltiptext'>plotDom plots the effects of dominance on sample size in an affected sibpair linkage study.<\/span><\/div>, <div class='tooltip'>plotNsibs<span class='tooltiptext'>Plots the required sample sizes needed to detect linkagein an affected sib pair linakge study as a function of lambda sib,under the assumption that lambda sib = lambda off.<\/span><\/div>, <div class='tooltip'>plotNtdt<span class='tooltiptext'>This function plots the results from ntdt.q<\/span><\/div>, <div class='tooltip'>ptb<span class='tooltiptext'>Create a power table for different values of q.<\/span><\/div>, <div class='tooltip'>ptdt<span class='tooltiptext'>Function to compute power for a TDT designusing formulae from Abel and Muller-Myhsok (1998)Am J Hum Genet 63:664-667<\/span><\/div>","<div class='tooltip'>BW2stagePPS<span class='tooltiptext'>Compute components of relvariance for a sample design where primary sampling units (PSUs) are selected with probability proportional to size (pps) and elements are selected via simple random sampling (srs). The input is an entire sampling frame.<\/span><\/div>, <div class='tooltip'>BW2stagePPSe<span class='tooltiptext'>Estimate components of relvariance for a sample design where primary sampling units (PSUs) are selected with pps and elements are selected via srs. The input is a sample selected in this way.<\/span><\/div>, <div class='tooltip'>BW2stageSRS<span class='tooltiptext'>Compute components of relvariance for a sample design where primary sampling units (PSUs) and elements are selected via srs. The input is an entire sampling frame.<\/span><\/div>, <div class='tooltip'>BW3stagePPS<span class='tooltiptext'>Compute components of relvariance for a sample design where primary sampling units (PSUs) are selected with ppswr and secondary sampling units (SSUs) and elements within SSUs are selected via srs. The input is an entire sampling frame.<\/span><\/div>, <div class='tooltip'>BW3stagePPSe<span class='tooltiptext'>Estimate components of relvariance for a sample design where primary sampling units (PSUs) are selected with probability proportional to size with replacement (ppswr) and secondary sampling units (SSUs) and elements within SSUs are selected via simple random sampling (srs). The input is a sample selected in this way.<\/span><\/div>, <div class='tooltip'>clusOpt2<span class='tooltiptext'>Compute the sample sizes that minimize the variance of the pwr-estimator of a total in a two-stage sample.<\/span><\/div>, <div class='tooltip'>clusOpt2fixedPSU<span class='tooltiptext'>Compute the optimum number of sample elements per primary sampling unit (PSU) for a fixed set of PSUs<\/span><\/div>, <div class='tooltip'>clusOpt3<span class='tooltiptext'>Compute the sample sizes that minimize the variance of the pwr-estimator of a total in a three-stage sample.<\/span><\/div>, <div class='tooltip'>clusOpt3fixedPSU<span class='tooltiptext'>Compute the sample sizes that minimize the variance of the pwr-estimator of a total in a three-stage sample when the PSU sample is fixed.<\/span><\/div>, <div class='tooltip'>CVcalc2<span class='tooltiptext'>Compute the coefficient of variation of an estimated total in a two-stage design. Primary sampling units (PSUs) can be selected either with probability proportional to size (pps) or with equal probability. Elements are selected via simple random sampling (srs).<\/span><\/div>, <div class='tooltip'>CVcalc3<span class='tooltiptext'>Compute the coefficient of variation of an estimated total in a three-stage design. Primary sampling units (PSUs) can be selected either with probability proportional to size (pps) or with equal probability. Secondary units and elements within SSUs are selected via simple random sampling (srs).<\/span><\/div>, <div class='tooltip'>deff<span class='tooltiptext'>Compute the Kish, Henry, Spencer, or Chen-Rust design effects.<\/span><\/div>, <div class='tooltip'>deffCR<span class='tooltiptext'>Chen-Rust design effect for stratified, clustered, two-stage samples<\/span><\/div>, <div class='tooltip'>deffH<span class='tooltiptext'>Compute the Henry design effect for single-stage samples when a general regression estimator is used for a total.<\/span><\/div>, <div class='tooltip'>deffK<span class='tooltiptext'>Compute the Kish design effect due to having unequal weights.<\/span><\/div>, <div class='tooltip'>deffS<span class='tooltiptext'>Compute the Spencer design effect for single-stage samples selected with probability proportional to a measure of size.<\/span><\/div>, <div class='tooltip'>dub<span class='tooltiptext'>Compute samples sizes at each phase of a two-phase design where strata are created using the first phase.<\/span><\/div>, <div class='tooltip'>gamEst<span class='tooltiptext'>Regresses a y on a set of covariates X where Var_M(y)=\\sigma^2x^\\gamma and thenregresses the squared residuals on log(x) to estimate \\gamma.<\/span><\/div>, <div class='tooltip'>gammaFit<span class='tooltiptext'>Iteratively computes estimate of \\gamma in a model with E_M(y)=x^T\\beta andVar_M(y)=\\sigma^2x^\\gamma.<\/span><\/div>, <div class='tooltip'>HMT<span class='tooltiptext'>Generate a population that follows the model in Hansen, Madow, and Tepping (1983)<\/span><\/div>, <div class='tooltip'>nCont<span class='tooltiptext'>Compute a simple random sample size using either a target coefficient of variation, CV_0, or target variance, V_0, for an estimated mean.<\/span><\/div>, <div class='tooltip'>nContMoe<span class='tooltiptext'>Compute a simple random sample size using a margin of error specified as the half-width of a normal approximation confidence interval or the half-width relative to the population mean.<\/span><\/div>, <div class='tooltip'>nDep2sam<span class='tooltiptext'>Compute a simple random sample size for estimating the difference in means when samples overlap<\/span><\/div>, <div class='tooltip'>nDomain<span class='tooltiptext'>Compute a simple random sample size using either a target coefficient of variation, CV_{0}(d), or target variance, V_{0}(d), for an estimated mean or total for a domain.<\/span><\/div>, <div class='tooltip'>nLogOdds<span class='tooltiptext'>Calculate the simple random sample size for estimating a proportion using the log-odds transformation.<\/span><\/div>, <div class='tooltip'>nProp<span class='tooltiptext'>Compute the simple random sample size for estimating a proportion based on different precision requirements.<\/span><\/div>, <div class='tooltip'>nProp2sam<span class='tooltiptext'>Compute a simple random sample size for estimating the difference in proportions when samples overlap<\/span><\/div>, <div class='tooltip'>nPropMoe<span class='tooltiptext'>Calculates a simple random sample size based on a specified margin of error.<\/span><\/div>, <div class='tooltip'>NRadjClass<span class='tooltiptext'>Compute separate nonresponse adjustments in a set of classes.<\/span><\/div>, <div class='tooltip'>NRFUopt<span class='tooltiptext'>Compute optimal values of the first-phase sample size and the second-phase sampling fraction in a two-phase sample.<\/span><\/div>, <div class='tooltip'>nWilson<span class='tooltiptext'>Calculate a simple random sample size for estimating a proportion using the Wilson method.<\/span><\/div>, <div class='tooltip'>pclass<span class='tooltiptext'>Fit a binary regression model for response probabilities and divide units into a specified number of classes.<\/span><\/div>, <div class='tooltip'>strAlloc<span class='tooltiptext'>Compute the proportional, Neyman, cost-constrained, and variance-constrainedallocations in a stratified simple random sample.<\/span><\/div>, <div class='tooltip'>wtdvar<span class='tooltiptext'>Compute an estimate of a population unit variance from a complex sample with survey weights.<\/span><\/div>","<div class='tooltip'>prospective<span class='tooltiptext'>Given the hypothetical population effect size and the required power level,the function prospective() performs a prospective design analysis forPearson's correlation test between two variables or t-test comparinggroup means (Cohen's d). According to the defined alternativehypothesis and the significance level, the required sample size is computedtogether with the associated Type M error, Type S error, and the criticaleffect value (i.e., the minimum absolute effect size value that wouldresult significant).<\/span><\/div>, <div class='tooltip'>retrospective<span class='tooltiptext'>Given the hypothetical population effect size and the study sample size, thefunction retrospective() performs a retrospective design analysis forPearson's correlation test between two variables or t-test comparinggroup means (Cohen's d). According to the defined alternativehypothesis and the significance level, inferential risks (i.e., Power level,Type M error, and Type S error) are computed together with the criticaleffect value (i.e., the minimum absolute effect size value that would resultsignificant).<\/span><\/div>","<div class='tooltip'>cohorts_of_n<span class='tooltiptext'>The original function in package \\href{https://CRAN.R-project.org/package=#1}{\\pkg{#1}}escalationhttps://CRAN.R-project.org/package=escalationescalation recognizes that individualtrial participants arrive at distinct times. Building upon this acknowledgmentof individuality, this override adds an extra line of code to draw as well alatent toxicity tolerance u_i for each individual participant.<\/span><\/div>, <div class='tooltip'>exact<span class='tooltiptext'>Implemented currently only for the (most?) common variant of the 3+3 design,which requires that at least 6 patients be treated at a dose before it may bedeclared as the MTD.<\/span><\/div>, <div class='tooltip'>extend<span class='tooltiptext'>A trial simulation carried through a predetermined number of replicationsmay not achieve desired precision as judged by Monte Carlo standard errors(MCSE) of estimated toxicity counts. This method enables simulations ofclass c('precautionary','simulations') to be extended until a givenlevel of precision has been achieved on expected counts of enrollment andDLTs, or (optionally) for a fixed additional number of simulations.<\/span><\/div>, <div class='tooltip'>hyper_mtdi_lognormal<span class='tooltiptext'>This hyperprior generates lognormal MTDi distributions with theircoefficients of variation being drawn from a Rayleigh distributionwith mode parameter set to\\sigma := CV.Because the standard deviation of this distribution issd = \\sigma \\sqrt{(2 - \\pi/2)} \\approx 0.655 \\sigma,% sd = \\sigma \\sqrt(2 - \\pi/2) ~ 0.655 \\sigma,this conveniently links our uncertainty about CV to its mode,and indeed to other measures of centrality that are proportional to this:mean = \\sigma \\sqrt{\\pi/2} \\approx 1.25 \\sigma% mean = \\sigma \\sqrt(\\pi/2) ~ 1.25 \\sigmamedian = \\sigma \\sqrt{2 log(2)} \\approx 1.18 \\sigma.% median = \\sigma \\sqrt(2 log(2)) ~ 1.18 \\sigma.The medians of the lognormal distributions generated are themselvesdrawn from a lognormal distribution with meanlog = log(median_mtd)and sdlog = median_sdlog. Thus, parameter median_sdlogrepresents a proportional uncertainty in median_mtd.<\/span><\/div>, <div class='tooltip'>mtdi_lognormal<span class='tooltiptext'>A lognormal MTDi distribution<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>safety_kable<span class='tooltiptext'>Output a kable for a simulation summary of class safetytab<\/span><\/div>, <div class='tooltip'>simulate_trials<span class='tooltiptext'>An S4 generic method providing a more abstract interface to trial simulation thanthe simulate_trials function of package escalation.This abstraction is needed to support simulations in which escalation's simplevectors of true toxicity probabilities are replaced by precautionary'smore realistic toxicity-distribution generators.<\/span><\/div>, <div class='tooltip'>test_draw_samples<span class='tooltiptext'>Test performance of draw_samples function<\/span><\/div>","<div class='tooltip'>effect_size<span class='tooltiptext'>Accessor function have been created to get the sample size (sample_size), power (power), effect size (effect_size), arm proportion(proportion), significance (significance), and trial variance estimates(sigma2) for a set of preference trials.Note that these methods are preferred over accessing the underlyingdata frame directly since the structure is slightly non-standard (somecolumns are lists) and some values, like power, are not stored directly.<\/span><\/div>, <div class='tooltip'>effects_from_means<span class='tooltiptext'>Calculates the preference, selection and treatment effects given the meansof each treatment group in the choice and random arms for the 2-stage randomized study.<\/span><\/div>, <div class='tooltip'>fit_preference<span class='tooltiptext'>Computes the test statistics and p-values for the preference, selection, and treatment effects for the two-stage randomized trial using collected outcome, random, treatment, and strata values for specifiedsignificance level.<\/span><\/div>, <div class='tooltip'>fit_preference_summary<span class='tooltiptext'>Computes the test statistics and p-values for the preference, selection, and treatment effects in a two-stage randomized trial using summary data.<\/span><\/div>, <div class='tooltip'>optimal_proportion<span class='tooltiptext'>Calculates the optimal proportion of patients assigned to the choice armin an unstratified two-stage randomized trial<\/span><\/div>, <div class='tooltip'>overall_power_binom<span class='tooltiptext'>Calculates the study power to detect the preference effect given a particular sample size in a two-stage randomized clinical trial with a binary outcome measure<\/span><\/div>, <div class='tooltip'>overall_sample_size_bin<span class='tooltiptext'>Calculates the sample size required to detect a given set of effectsin a two-stage randomize clinical trial with a binary outcome. Returns the sample size for each of the three effects: preference,selection and treatment<\/span><\/div>, <div class='tooltip'>overall_sample_size_pois<span class='tooltiptext'>Calculates the sample size required to detect a given set of effects in a two-stage randomized clinical trial. Returns the largestof the required sample sizes for a given set of treatment, selection, and preference effects.<\/span><\/div>, <div class='tooltip'>power<span class='tooltiptext'>Accessor function have been created to get the sample size (sample_size), power (power), effect size (effect_size), arm proportion(proportion), significance (significance), and trial variance estimates(sigma2) for a set of preference trials.Note that these methods are preferred over accessing the underlyingdata frame directly since the structure is slightly non-standard (somecolumns are lists) and some values, like power, are not stored directly.<\/span><\/div>, <div class='tooltip'>preference<span class='tooltiptext'>The variables in the formula should reference columns in thedata parameter and should have the following characteristics. outcome: Numeric values giving the outcome of interest. treatment: Character, categorical, or integer values denoting the treatment received by an individual. random: Logical value indicating whether the sample was from the random arm (TRUE) or choice (FALSE). strata: An optional integer value denoting which strata individuals belong to.<\/span><\/div>, <div class='tooltip'>preference.trial<span class='tooltiptext'>Create a Preference Trial<\/span><\/div>, <div class='tooltip'>proportion<span class='tooltiptext'>Accessor function have been created to get the sample size (sample_size), power (power), effect size (effect_size), arm proportion(proportion), significance (significance), and trial variance estimates(sigma2) for a set of preference trials.Note that these methods are preferred over accessing the underlyingdata frame directly since the structure is slightly non-standard (somecolumns are lists) and some values, like power, are not stored directly.<\/span><\/div>, <div class='tooltip'>pt_from_power<span class='tooltiptext'>Create a set of preference trials with specified power.The power parameter guarantees that the power will be at least what isspecified for each of the three arms.<\/span><\/div>, <div class='tooltip'>pt_from_ss<span class='tooltiptext'>Create a set of preference trials where the maximum sample size for an arm is specified.<\/span><\/div>, <div class='tooltip'>pwr_overall_pois<span class='tooltiptext'>Calculates the study power to detect a set of effects given a particular sample size in a two-stage randomized clinical trial<\/span><\/div>, <div class='tooltip'>sample_size<span class='tooltiptext'>Accessor function have been created to get the sample size (sample_size), power (power), effect size (effect_size), arm proportion(proportion), significance (significance), and trial variance estimates(sigma2) for a set of preference trials.Note that these methods are preferred over accessing the underlyingdata frame directly since the structure is slightly non-standard (somecolumns are lists) and some values, like power, are not stored directly.<\/span><\/div>, <div class='tooltip'>sigma2<span class='tooltiptext'>Accessor function have been created to get the sample size (sample_size), power (power), effect size (effect_size), arm proportion(proportion), significance (significance), and trial variance estimates(sigma2) for a set of preference trials.Note that these methods are preferred over accessing the underlyingdata frame directly since the structure is slightly non-standard (somecolumns are lists) and some values, like power, are not stored directly.<\/span><\/div>, <div class='tooltip'>significance<span class='tooltiptext'>Accessor function have been created to get the sample size (sample_size), power (power), effect size (effect_size), arm proportion(proportion), significance (significance), and trial variance estimates(sigma2) for a set of preference trials.Note that these methods are preferred over accessing the underlyingdata frame directly since the structure is slightly non-standard (somecolumns are lists) and some values, like power, are not stored directly.<\/span><\/div>, <div class='tooltip'>treatment_effect_size<span class='tooltiptext'>Calculates the treatment effect that can be detected given a desired study power and overall study sample size for the two-stage randomized design<\/span><\/div>","<div class='tooltip'>BER<span class='tooltiptext'>The BER function selects a random sample or estimates an interest parameter under a Bernoulli design.<\/span><\/div>, <div class='tooltip'>CONGL<span class='tooltiptext'>The CONGL function selects a random sample or estimates a parameter of interest under a cluster sampling design<\/span><\/div>, <div class='tooltip'>ESTRAT<span class='tooltiptext'>The ESTRAT function selects a random sample or estimates an interest parameter under a stratified sampling.<\/span><\/div>, <div class='tooltip'>M.MET<span class='tooltiptext'>The M.MET function selects a random sample or estimates an interest parameter under multi-stage sampling (up to four stages).<\/span><\/div>, <div class='tooltip'>MAS<span class='tooltiptext'>The MAS function selects a random sample or estimates a parameter of interest under a simple random sampling design without replacement.<\/span><\/div>, <div class='tooltip'>MCR<span class='tooltiptext'>The MCR function selects a random sample or estimates an interest parameter under a simple random sampling design without replacement.<\/span><\/div>, <div class='tooltip'>n.ESTMAS<span class='tooltiptext'>The n.ESTMAS function determines the sample size with its corresponding allocation by stratum, using a stratified sampling strategy, where a simple random sampling design with no replacement (ESTMAS) is applied in each stratum; taking into account whether the parameter of interest is the average (or total) or a proportion.<\/span><\/div>, <div class='tooltip'>n.MAS<span class='tooltiptext'>The n.MAS function determines the sample size by a simple random sample design without replacement, taking into account whether the parameter of interest is the mean (or total) or a proportion.<\/span><\/div>, <div class='tooltip'>n.MASC<span class='tooltiptext'>The n.MASC function determines sample size using a simple random sampling design without replacement of Conglomerates.<\/span><\/div>, <div class='tooltip'>PiPT<span class='tooltiptext'>The PiPT function selects a random sample or estimates an interest parameter under a sampling design with proportional inclusion probabilities proportional to size.<\/span><\/div>, <div class='tooltip'>PPT<span class='tooltiptext'>The PPT function selects a random sample or estimates a parameter of interest under a sampling design with proportional proportional selection probabilities (PPT).<\/span><\/div>, <div class='tooltip'>R.SIS<span class='tooltiptext'>The R.SIS function selects a random sample or estimates a parameter of interest under a r-systematic sampling design.<\/span><\/div>, <div class='tooltip'>WHICH1<span class='tooltiptext'>The WHICH1 function returns the positions in which the vector components (V1) are located in another vector (V2).<\/span><\/div>","<div class='tooltip'>cp<span class='tooltiptext'>This will calculate the conditional power given the observed log hazard ratio based on Cox model<\/span><\/div>, <div class='tooltip'>cpboundary<span class='tooltiptext'>This will calculate the stopping boundary based on the conditional power criteria, i.e. if observed HR is above the boundary, the conditional power will be lower than the designated level. All the calculation is based on the proportional hazards assumption and the Cox model.<\/span><\/div>, <div class='tooltip'>cpstop<span class='tooltiptext'>This will calculate the stopping probability given the stopping boundary. All the calculation is based on the proportional hazards assumption and the Cox model.<\/span><\/div>, <div class='tooltip'>fourhr<span class='tooltiptext'>This will calculate the more complex integration<\/span><\/div>, <div class='tooltip'>hxbeta<span class='tooltiptext'>A function to calculate the beta-smoothed hazard rate<\/span><\/div>, <div class='tooltip'>innercov<span class='tooltiptext'>This will calculate the inner integration of the overall covariance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>innervar<span class='tooltiptext'>This will calculate the inner integration of the overall variance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>instudyfindt<span class='tooltiptext'>This will calculate the timeline from some timepoint in study when some/all subjects have entered accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>ovbeta<span class='tooltiptext'>This will calculate the overall (log) hazard ratio accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>overallcov<span class='tooltiptext'>This will calculate the overall covariance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>overallcovp1<span class='tooltiptext'>This will calculate the first part of the overall covariance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>overallcovp2<span class='tooltiptext'>This will calculate the other parts of the overall covariance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>overallvar<span class='tooltiptext'>This will calculate the overall variance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwe<span class='tooltiptext'>This will provide the related functions of the specified piecewise exponential distribution.<\/span><\/div>, <div class='tooltip'>pwecx<span class='tooltiptext'>This will calculate the functions according to the piecewise exponential distribution with crossover<\/span><\/div>, <div class='tooltip'>pwecxcens<span class='tooltiptext'>This will calculate the functions according to the piecewise exponential distribution with crossover<\/span><\/div>, <div class='tooltip'>pwecxpwu<span class='tooltiptext'>This will calculate the functions according to the piecewise exponential distribution with crossover<\/span><\/div>, <div class='tooltip'>pwecxpwufindt<span class='tooltiptext'>This will calculate the timeline from study inception accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwecxpwuforvar<span class='tooltiptext'>This is a utility function to calculate the overall variance accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwefv2<span class='tooltiptext'>This will $int_0^t s^k lambda_1(s)S_2(s)ds$ where k=0,1,2and rate1=lambda_1 and S_2 has hazard rate2<\/span><\/div>, <div class='tooltip'>pwefvplus<span class='tooltiptext'>This will calculate the more complex integration accounting for crossover<\/span><\/div>, <div class='tooltip'>pwepower<span class='tooltiptext'>This will calculate the powers for the test statistics accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwepowereq<span class='tooltiptext'>This will calculate the powers for the test statistics accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwepowerfindt<span class='tooltiptext'>This will calculate the timepoint where a certain power of the specified test statistics is obtained accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwepowerni<span class='tooltiptext'>This will calculate the powers for the test statistics accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwesim<span class='tooltiptext'>This will simulate the test statistics accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>pwu<span class='tooltiptext'>This will calculate the distribution function of the piecewise uniform distribution<\/span><\/div>, <div class='tooltip'>qpwe<span class='tooltiptext'>This will provide the quantile function of the specified piecewise exponential distribution<\/span><\/div>, <div class='tooltip'>qpwu<span class='tooltiptext'>This will provide the quantile function of the specified piecewise uniform distribution<\/span><\/div>, <div class='tooltip'>rmstcov<span class='tooltiptext'>A function to calculate the variance and covariance of estimated restricted mean survival time using data from different cut-off points accounting for delayed treatment, discontinued treatment and non-uniform entry<\/span><\/div>, <div class='tooltip'>rmsth<span class='tooltiptext'>A function to estimate the restricted mean survival time (RMST) and its variance from data<\/span><\/div>, <div class='tooltip'>rmstpower<span class='tooltiptext'>A function to calculate powers at different cut-points based on difference of restricted mean survival times (RMST)account for delayed treatment, discontinued treatment and non-uniform entry<\/span><\/div>, <div class='tooltip'>rmstpowerfindt<span class='tooltiptext'>This will calculate the timepoint where a certain power of the mean difference of RMSTs is obtained accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>rmstsim<span class='tooltiptext'>This will simulate the test statistics accouting for staggered entry, delayed treatment effect, treatment crossover and loss to follow-up.<\/span><\/div>, <div class='tooltip'>rmstutil<span class='tooltiptext'>A utility function to calculate the true restricted mean survival time (RMST) and its variance account for delayed treatment, discontinued treatment and non-uniform entry<\/span><\/div>, <div class='tooltip'>rpwe<span class='tooltiptext'>This will generate random numbers according to the specified piecewise exponential distribution<\/span><\/div>, <div class='tooltip'>rpwecx<span class='tooltiptext'>This will generate random numbers according to the piecewise exponential distribution with crossover<\/span><\/div>, <div class='tooltip'>rpwu<span class='tooltiptext'>This will generate random numbers according to the specified piecewise uniform distribution<\/span><\/div>, <div class='tooltip'>spf<span class='tooltiptext'>A utility function to calculate a ratio.<\/span><\/div>, <div class='tooltip'>wlrcal<span class='tooltiptext'>A utility function to calculate the weighted log-rank statistics and their varainces given the weights<\/span><\/div>, <div class='tooltip'>wlrcom<span class='tooltiptext'>A function to calculate the weighted log-rank statistics and their varainces given the weights including log-rank, gehan, Tarone-Ware, Peto-Peto, mPeto-Peto and Fleming-Harrington<\/span><\/div>, <div class='tooltip'>wlrutil<span class='tooltiptext'>A utility function to calculate some common functions in contructing weights<\/span><\/div>","<div class='tooltip'>anc<span class='tooltiptext'>Compute Power for One or Two Factor ANCOVA with a single covariateTakes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by userFactor A can have up to four levels, Factor B, if used, can only be two<\/span><\/div>, <div class='tooltip'>anova1f_3<span class='tooltiptext'>Compute power for a One Factor ANOVA with three levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>anova1f_3c<span class='tooltiptext'>Compute power for a One Factor ANOVA with three levels and contrasts.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>anova1f_4<span class='tooltiptext'>Compute power for a One Factor Between Subjects ANOVA with four levelsTakes means, sds, and sample sizes for each group<\/span><\/div>, <div class='tooltip'>anova1f_4c<span class='tooltiptext'>Compute power for a One Factor ANOVA with four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>anova2x2<span class='tooltiptext'>Compute power for a Two by Two Between Subjects ANOVA.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>anova2x2_se<span class='tooltiptext'>Compute power for Simple Effects in a Two by Two Between Subjects ANOVA with two levels for each factor.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>Chi2x2<span class='tooltiptext'>Compute power for an Chi Square 2x2Takes proportions for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>Chi2X3<span class='tooltiptext'>Compute power for an Chi Square 2x3Takes proportions for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>ChiES<span class='tooltiptext'>Compute power for Chi Square Based on Effect SizeTakes phi, degrees of freedom, and a range of sample sizes. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>ChiGOF<span class='tooltiptext'>Compute power for an Chi Square Goodness of FitTakes proportions for up to six group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>corr<span class='tooltiptext'>Compute power for Pearson's CorrelationTakes correlation and range of values<\/span><\/div>, <div class='tooltip'>d_prec<span class='tooltiptext'>Compute Precision Analyses for Standardized Mean Differences<\/span><\/div>, <div class='tooltip'>depb<span class='tooltiptext'>Power for Comparing Dependent Coefficients in Multiple Regression with Two or Three PredictorsRequires correlations between all variables as sample size. Means, sds, and alpha are option. Also computes Power(All)<\/span><\/div>, <div class='tooltip'>depcorr0<span class='tooltiptext'>Compute Power for Comparing Two Dependent Correlations, No Variables in CommonTakes correlations and range of values. First variable in each pair is termed predictor, second is DV<\/span><\/div>, <div class='tooltip'>depcorr1<span class='tooltiptext'>Compute Power for Comparing Two Dependent Correlations, One Variable in CommonTakes correlations and range of values<\/span><\/div>, <div class='tooltip'>indb<span class='tooltiptext'>Power for Comparing Independent Coefficients in Multiple Regression with Two or Three PredictorsRequires correlations between all variables as sample size. Means, sds, and alpha are option. Also computes Power(All)<\/span><\/div>, <div class='tooltip'>indcorr<span class='tooltiptext'>Compute Power for Comparing Two Independent CorrelationsTakes correlations and range of values<\/span><\/div>, <div class='tooltip'>indR2<span class='tooltiptext'>Power for Comparing Independent R2 in Multiple Regression with Two or Three PredictorsRequires correlations between all variables as sample size. Means, sds, and alpha are option. Also computes Power(All)<\/span><\/div>, <div class='tooltip'>indt<span class='tooltiptext'>Compute power for an Independent Samples t-testTakes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>lmm1F<span class='tooltiptext'>Compute power for a One Factor Within Subjects Linear Mixed Model with up to four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>lmm1Ftrends<span class='tooltiptext'>Compute power for a One Factor Within Subjects LMM Trends with up to four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>lmm1w1b<span class='tooltiptext'>Compute power for a One Factor Within Subjects and One Factor Between LMM with up to two by four levels (within).Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>lmm2F<span class='tooltiptext'>Compute power for a Two Factor Within Subjects Using Linear Mixed Models with up to two by four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>lmm2Fse<span class='tooltiptext'>Compute power for a Two Factor Within Subjects Using Linear Mixed Models with up to two by four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>LRcat<span class='tooltiptext'>Compute Power for Logistic Regression with a Single Categorical Predictor<\/span><\/div>, <div class='tooltip'>LRcont<span class='tooltiptext'>Compute Power for Logistic Regression with Continuous Predictors<\/span><\/div>, <div class='tooltip'>MANOVA1f<span class='tooltiptext'>Compute power for a One Factor MANOVA with up to two levels and up to four measures.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>md_prec<span class='tooltiptext'>Compute Precision Analyses for Mean Differences<\/span><\/div>, <div class='tooltip'>med<span class='tooltiptext'>Compute Power for Mediated (Indirect) EffectsRequires correlations between all variables as sample size.This approach calculates power for the Sobel test.The medjs function calculates power based on joint significance (recommended)<\/span><\/div>, <div class='tooltip'>medjs<span class='tooltiptext'>Compute Power for Mediated (Indirect) Effects Using Joint SignificanceRequires correlations between all variables as sample size.This is the recommended approach for determining power<\/span><\/div>, <div class='tooltip'>medserial<span class='tooltiptext'>Compute Power for Serial Mediation EffectsRequires correlations between all variables as sample size.This approach calculates power for the serial mediation usingjoint significance (recommended)<\/span><\/div>, <div class='tooltip'>MRC<span class='tooltiptext'>Compute power for Multiple Regression with up to Five PredictorsExample code below for three predictors. Expand as needed for four or five<\/span><\/div>, <div class='tooltip'>MRC_all<span class='tooltiptext'>Compute power for Multiple Regression with Up to Five PredictorsRequires correlations between all variables as sample size. Means, sds, and alpha are option. Also computes Power(All)<\/span><\/div>, <div class='tooltip'>MRC_short2<span class='tooltiptext'>Compute Multiple Regression shortcuts with three predictors for Ind CoefficientsRequires correlations between all variables as sample size. Means and sds are option. Also computes Power(All)<\/span><\/div>, <div class='tooltip'>MRC_shortcuts<span class='tooltiptext'>Compute Multiple Regression shortcuts with three predictors (will expand to handle two to five)Requires correlations between all variables as sample size. Means and sds are option. Also computes Power(All)<\/span><\/div>, <div class='tooltip'>pairt<span class='tooltiptext'>Compute power for a Paired t-testTakes means, sd, and sample sizes. Alpha is .05 by default, alternative values may be entered by user.correlation (r) defaults to .50.<\/span><\/div>, <div class='tooltip'>prop1<span class='tooltiptext'>Compute power for a single sample proportion testTakes phi, degrees of freedom, and a range of sample sizes. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>propind<span class='tooltiptext'>Compute power for Tests of Two Independent ProportionsTakes phi, degrees of freedom, and a range of sample sizes. Alpha is .05 by default, alternative values may be entered by userThis test uses what is sometimes called the chi-square test for comparing proportions<\/span><\/div>, <div class='tooltip'>r_prec<span class='tooltiptext'>Compute Precision Analyses for CorrelationsThis approach simply loops a function from MBESS<\/span><\/div>, <div class='tooltip'>R2_prec<span class='tooltiptext'>Compute Precision Analyses for R-SquaredThis approach simply loops a function from MBESS<\/span><\/div>, <div class='tooltip'>R2ch<span class='tooltiptext'>Compute power for R2 change in Multiple Regression (up to three predictors)Requires correlations between all variables as sample size. Means, sds, and alpha are option. Also computes Power(All)Example code below for three predictors. Expand as needed for four or five<\/span><\/div>, <div class='tooltip'>regint<span class='tooltiptext'>Compute Power for Regression Interaction (Correlation/Coefficient Approach)<\/span><\/div>, <div class='tooltip'>regintR2<span class='tooltiptext'>Compute Power for Regression Interaction (R2 Change Approach)<\/span><\/div>, <div class='tooltip'>tfromd<span class='tooltiptext'>Compute power for a t test using d statisticTakes d, sample size range, type of test, and tails.<\/span><\/div>, <div class='tooltip'>win1bg1<span class='tooltiptext'>Compute power for a One Factor Within Subjects and One Factor Between ANOVA with up to two by four levels (within).Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>win1F<span class='tooltiptext'>Compute power for a One Factor Within Subjects ANOVA with up to four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>win1Ftrends<span class='tooltiptext'>Compute power for a One Factor Within Subjects Trends with up to four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>win2F<span class='tooltiptext'>Compute power for a Two Factor Within Subjects ANOVA with up to two by four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>, <div class='tooltip'>win2Fse<span class='tooltiptext'>Compute power for Simple Effects in Two Factor Within Subjects ANOVA with up to two by four levels.Takes means, sds, and sample sizes for each group. Alpha is .05 by default, alternative values may be entered by user<\/span><\/div>","<div class='tooltip'>%,%<span class='tooltiptext'>A binary operator shortcut for paste(x,y)<\/span><\/div>, <div class='tooltip'>agghaz<span class='tooltiptext'>Computes the MLE for the model that assumes piecewise constant hazards on intervals defined by a grid of points. One applications for example is to calculate monthly hazard rates given numbers of events, numbers at risk and event times reported to the day. Can also handle time to event data stratified on a blocking factor.<\/span><\/div>, <div class='tooltip'>as.boundaries<span class='tooltiptext'>Convert a PwrGSD object to a boundaries object<\/span><\/div>, <div class='tooltip'>CDFOR2LRR<span class='tooltiptext'>Given the values of the baseline hazard and odds ratio of the CDF at a grid of time points find the corresponding logged risk ratio.<\/span><\/div>, <div class='tooltip'>CondPower<span class='tooltiptext'>Computes conditional type I and type II error probabilities given current value of the test statistic for monitoring based upon stochastic curtailment. This is now obsolete and included in the functionality of GrpSeqBnds and is here for instructional purposes only.<\/span><\/div>, <div class='tooltip'>cpd.PwrGSD<span class='tooltiptext'>Given a user defined indexing dataframe as its only argument, creates a skeleton compound PwrGSD object having a component Elements, a list of PwrGSD objects, of length equal to the number of rows in the indexing dataframe<\/span><\/div>, <div class='tooltip'>CRRtoRR<span class='tooltiptext'>Given a vector of cumulative-risk ratios, computes risk ratios<\/span><\/div>, <div class='tooltip'>CY2TOShaz<span class='tooltiptext'>Given the cutpoints at which the hazard is to be constant, the values taken by the calender year rates and the calender time offset from the start of the trial at which randomization ended, this function converts to time on study rates, assuming uniform accrual.<\/span><\/div>, <div class='tooltip'>detail<span class='tooltiptext'>Extracts the 'detail' component from an object of class PwrGSD<\/span><\/div>, <div class='tooltip'>DX<span class='tooltiptext'>DX(x) returns c(x[1], diff(x))<\/span><\/div>, <div class='tooltip'>Elements<span class='tooltiptext'>Create a subset of a cpd.PwrGSD object<\/span><\/div>, <div class='tooltip'>EX1gXK<span class='tooltiptext'>A function for computing the bias adjusted point estimate for a statistic, on the Brownian scale, observed to cross the efficacy boundary.<\/span><\/div>, <div class='tooltip'>GrpSeqBnds<span class='tooltiptext'>This computes efficacy and futility boundaries for interim analysis and sequential designs. Two sided symmetric efficacy boundaries can be computed by specifying half of the intended total type I error probability in the argument, Alpha.Efficacy. Otherwise, especially in the case of efficacy and futility bounds only one sided boundaries are currently computed. The computation allows for two different time scales--one must be the variance ratio, and the second can be a user chosen increasing scale beginning with 0 that takes the value 1 at the conclusion of the trial.<\/span><\/div>, <div class='tooltip'>gsd.dens<span class='tooltiptext'>A function for computing the probability density for a sequentially monitored test. This is the joint density, in the rejection region, of (X_K, K), where X_K is the observed value of the test statistic upon efficacy boundary crossing, and K is the analysis number at which the efficacy boundary was crossed.<\/span><\/div>, <div class='tooltip'>Haybittle<span class='tooltiptext'>The function Haybittle is used in calls to the functions GrpSeqBnds and PwrGSD as a possible setting for the argument EfficacyBoundary. NOTE: the Haybittle method is not implemented as a futility boundary method The Haybittle method is one of four currently availiable choices (efficacy only), the others being LanDemets, SC (stochastic curtailment), and user specified.<\/span><\/div>, <div class='tooltip'>IntSurvDiff<span class='tooltiptext'>Computes a two sample weighted integrated survival function log-rank statistic with events weighted according to one of the available weighting function choices<\/span><\/div>, <div class='tooltip'>LanDemets<span class='tooltiptext'>The function LanDemets is used in calls to the functions GrpSeqBnds and PwrGSD as a possible setting for the arguments EfficacyBoundary and FutilityBoundary, in specification of the method whereby efficacy and or futility boundaries are to be constructed. The Lan-Demets method is one of four currently availiable choices, the others being SC (stochastic curtailment), Haybittle (efficacy only) and user specified.<\/span><\/div>, <div class='tooltip'>lookup<span class='tooltiptext'>Given the values and lefthand endpoints for intervals of constancy, lookup values of the function at arbitrary values of the independent variable.<\/span><\/div>, <div class='tooltip'>mystack<span class='tooltiptext'>Given a dataframe containing one or more variables named with a common prefix, this function creates a stacked dataset with one set of observed values of the variables (in order of occurence) per line.<\/span><\/div>, <div class='tooltip'>mysurvfit<span class='tooltiptext'>Computes numbers at risk, numbers of events at each unique event time within levels of a blocking factor<\/span><\/div>, <div class='tooltip'>ObrienFleming<span class='tooltiptext'>Stipulates alpha spending according to the O'Brien-Fleming spending function in the Lan-Demets boundary construction method. Its intended purpose is in constructing calls to GrpSeqBnds and PwrGSD.<\/span><\/div>, <div class='tooltip'>Pocock<span class='tooltiptext'>Stipulates alpha spending according to the Pocock spending function in the Lan-Demets boundary construction method. Its intended purpose is in constructing calls to GrpSeqBnds and PwrGSD.<\/span><\/div>, <div class='tooltip'>Pow<span class='tooltiptext'>Stipulates alpha spending according to the Wang-Tsiatis Power function in the Lan-Demets boundary construction method. Its intended purpose is in constructing calls to GrpSeqBnds and PwrGSD.<\/span><\/div>, <div class='tooltip'>Power<span class='tooltiptext'>The function `Power' is used to summarize the `cpd.PwrGSD' object into a dataframe containing power and type II error, summed over analysis times. The data frame is stacked by rows of `descr' and by `stat' (if there are multiple statistics being profiled per each component of `Elements'), for generating tables or performing other computations.<\/span><\/div>, <div class='tooltip'>PwrGSD<span class='tooltiptext'>Derives power in a two arm clinical trial under a group sequential design. Allows for arbitrary number of interim analyses, arbitrary specification of arm-0/arm-1 time to event distributions (via survival or hazard), arm-0/arm-1 censoring distribution, provisions for two types of continuous time non-compliance according to arm-0/arm-1 rate followed by switch to new hazard rate. Allows for analyses using (I) weighted log-rank statistic, with weighting function (a) a member of the Flemming-Harrington G-Rho class, or (b) a stopped version thereof, or (c) the ramp-plateau deterministic weights, or (II) the integrated survival distance (currently under method==\"S\" without futility only). Stopping boundaries are computed via the Lan-Demets method, Haybittle method, converted from the stochastic curtailment procedure, or be completely specified by the user. The Lan-Demets boundaries can be constructed usign either O'Brien-Flemming, Pocock or Wang-Tsiatis power alpha-spending. The C kernel is readily extensible, and further options will become available in the near future.<\/span><\/div>, <div class='tooltip'>RCM2RR<span class='tooltiptext'>Given the relative cumulative mortality (ratio of CDFs), the baseline hazard and censoring hazard at a grid of time points, calculates the corresponding risk ratio at a second specified grid of time points.<\/span><\/div>, <div class='tooltip'>RR2RCM<span class='tooltiptext'>Relative risk to Relative Cumulative Mortality<\/span><\/div>, <div class='tooltip'>SC<span class='tooltiptext'>The function SC is used in calls to the functions GrpSeqBnds and PwrGSD as a possible setting for the arguments EfficacyBoundary and FutilityBoundary, in specification of the method whereby efficacy and or futility boundaries are to be constructed. The Stochastic Curtailment method is one of four currently availiable choices, the others being LanDemets, Haybittle (efficacy only) and user specified.<\/span><\/div>, <div class='tooltip'>SCtoBdry<span class='tooltiptext'>Converts a stochastic curtailment boundary (conditional type I or II error probability) into a (efficacy or futility) boundary on the standardized Z scale<\/span><\/div>, <div class='tooltip'>SimGSB<span class='tooltiptext'>Verifies the results of GrpSeqBnds via simulation<\/span><\/div>, <div class='tooltip'>wtdlogrank<span class='tooltiptext'>Computes a two sample weighted log-rank statistic with events weighted according to one of the available weighting function choices<\/span><\/div>","<div class='tooltip'>%>%<span class='tooltiptext'>Pipe operator<\/span><\/div>, <div class='tooltip'>NullObject<span class='tooltiptext'>Model a domain concept using natural lingo of the domainexperts, such as “Passenger”, “Address”, and “Money”.<\/span><\/div>, <div class='tooltip'>ValueObject<span class='tooltiptext'>Model a domain concept using natural lingo of the domainexperts, such as “Passenger”, “Address”, and “Money”.<\/span><\/div>","<div class='tooltip'>randquote<span class='tooltiptext'>Get Random Quotes<\/span><\/div>, <div class='tooltip'>randquote_simple<span class='tooltiptext'>Minimal version of getting random quotes (only quote and author)<\/span><\/div>","<div class='tooltip'>%>%<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsrequired_pkgs, tidy, tunable magrittr%>% statsupdate<\/span><\/div>, <div class='tooltip'>add_check<span class='tooltiptext'>add_step adds a step to the last location in the recipe.add_check does the same for checks.<\/span><\/div>, <div class='tooltip'>add_role<span class='tooltiptext'>update_role() alters an existing role in the recipe or assigns an initialrole to variables that do not yet have a declared role.add_role() adds an additional role to variables that already have a rolein the recipe. It does not overwrite old roles, as a single variable can havemultiple roles.remove_role() eliminates a single existing role in the recipe.<\/span><\/div>, <div class='tooltip'>add_step<span class='tooltiptext'>add_step adds a step to the last location in the recipe.add_check does the same for checks.<\/span><\/div>, <div class='tooltip'>all_nominal<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>all_numeric<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>all_outcomes<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>all_predictors<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>bake<span class='tooltiptext'>For a recipe with at least one preprocessing operation that has been trained byprep.recipe(), apply the computations to new data.<\/span><\/div>, <div class='tooltip'>check<span class='tooltiptext'>step sets the class of the step and check is for checks.<\/span><\/div>, <div class='tooltip'>check_class<span class='tooltiptext'>check_class creates a specification of a recipecheck that will check if a variable is of a designated class.<\/span><\/div>, <div class='tooltip'>check_cols<span class='tooltiptext'>check_cols creates a specification of a recipestep that will check if all the columns of the training frame arepresent in the new data.<\/span><\/div>, <div class='tooltip'>check_missing<span class='tooltiptext'>check_missing creates a specification of a recipeoperation that will check if variables contain missing values.<\/span><\/div>, <div class='tooltip'>check_name<span class='tooltiptext'>check_name is to be used in the bake function to ensure thatnewly created variable names don't overlap with existing names.Throws an error if check fails.<\/span><\/div>, <div class='tooltip'>check_new_values<span class='tooltiptext'>check_new_values creates a specification of a recipeoperation that will check if variables contain new values.<\/span><\/div>, <div class='tooltip'>check_range<span class='tooltiptext'>check_range creates a specification of a recipecheck that will check if the range of a numericvariable changed in the new data.<\/span><\/div>, <div class='tooltip'>check_type<span class='tooltiptext'>This internal function is to be used in the prep function to ensure thatthe type of the variables matches the expectation. Throws an error ifcheck fails.<\/span><\/div>, <div class='tooltip'>current_info<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>denom_vars<span class='tooltiptext'>step_ratio creates a specification of a recipestep that will create one or more ratios out of numericvariables.<\/span><\/div>, <div class='tooltip'>detect_step<span class='tooltiptext'>Detect if a particular step or check is used in a recipe<\/span><\/div>, <div class='tooltip'>discretize<span class='tooltiptext'>discretize converts a numeric vector into a factor withbins having approximately the same number of data points (basedon a training set).<\/span><\/div>, <div class='tooltip'>dummy_names<span class='tooltiptext'>names0 creates a series of num names with a common prefix.The names are numbered with leading zeros (e.g.prefix01-prefix10 instead of prefix1-prefix10).dummy_names can be used for renaming unordered and ordereddummy variables (in step_dummy()).<\/span><\/div>, <div class='tooltip'>ellipse_check<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>, <div class='tooltip'>estimate_yj<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>, <div class='tooltip'>fixed<span class='tooltiptext'>Helper Functions for Profile Data Sets<\/span><\/div>, <div class='tooltip'>format_ch_vec<span class='tooltiptext'>Helpers for printing step functions<\/span><\/div>, <div class='tooltip'>format_selectors<span class='tooltiptext'>Helpers for printing step functions<\/span><\/div>, <div class='tooltip'>fully_trained<span class='tooltiptext'>Check to see if a recipe is trained/prepared<\/span><\/div>, <div class='tooltip'>has_role<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>has_type<span class='tooltiptext'>has_role(), all_predictors(), and all_outcomes() can be used toselect variables in a formula that have certain roles.Similarly, has_type(), all_numeric(), and all_nominal() are usedto select columns based on their data type.See ?selections for more details.current_info() is an internal function.All of these functions have have limited utilityoutside of column selection in step functions.<\/span><\/div>, <div class='tooltip'>imp_vars<span class='tooltiptext'>step_bagimpute creates a specification of a recipe step that willcreate bagged tree models to impute missing data.<\/span><\/div>, <div class='tooltip'>is_trained<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>, <div class='tooltip'>juice<span class='tooltiptext'>As of recipes version 0.1.14, juice() is superseded in favor ofbake(object, new_data = NULL).<\/span><\/div>, <div class='tooltip'>names0<span class='tooltiptext'>names0 creates a series of num names with a common prefix.The names are numbered with leading zeros (e.g.prefix01-prefix10 instead of prefix1-prefix10).dummy_names can be used for renaming unordered and ordereddummy variables (in step_dummy()).<\/span><\/div>, <div class='tooltip'>prep<span class='tooltiptext'>For a recipe with at least one preprocessing operation, estimate the requiredparameters from a training set that can be later applied to other datasets.<\/span><\/div>, <div class='tooltip'>prepare<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>, <div class='tooltip'>prepper<span class='tooltiptext'>When working with the rsample package, a simple recipemust be prepared using the prepfunction first. When using recipes with rsample itis helpful to have a function that can prepare a recipeacross a series of split objects that are producedin this package. prepper is a wrapper functionaround prep that can be used to dothis. See the vignette on \"Recipes and rsample\" for anexample.<\/span><\/div>, <div class='tooltip'>printer<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>, <div class='tooltip'>prof<span class='tooltiptext'>Helper Functions for Profile Data Sets<\/span><\/div>, <div class='tooltip'>rand_id<span class='tooltiptext'>Make a random identification field for steps<\/span><\/div>, <div class='tooltip'>recipe<span class='tooltiptext'>A recipe is a description of what steps should be applied to a data set inorder to get it ready for data analysis.<\/span><\/div>, <div class='tooltip'>recipes_pkg_check<span class='tooltiptext'>This will check to see if all required packages are installed.<\/span><\/div>, <div class='tooltip'>remove_role<span class='tooltiptext'>update_role() alters an existing role in the recipe or assigns an initialrole to variables that do not yet have a declared role.add_role() adds an additional role to variables that already have a rolein the recipe. It does not overwrite old roles, as a single variable can havemultiple roles.remove_role() eliminates a single existing role in the recipe.<\/span><\/div>, <div class='tooltip'>required_pkgs<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsrequired_pkgs, tidy, tunable magrittr%>% statsupdate<\/span><\/div>, <div class='tooltip'>sel2char<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>, <div class='tooltip'>step<span class='tooltiptext'>step sets the class of the step and check is for checks.<\/span><\/div>, <div class='tooltip'>step_arrange<span class='tooltiptext'>step_arrange creates a specification of a recipe stepthat will sort rows using dplyr::arrange().<\/span><\/div>, <div class='tooltip'>step_bagimpute<span class='tooltiptext'>step_bagimpute creates a specification of a recipe step that willcreate bagged tree models to impute missing data.<\/span><\/div>, <div class='tooltip'>step_bin2factor<span class='tooltiptext'>step_bin2factor creates a specification of arecipe step that will create a two-level factor from a singledummy variable.<\/span><\/div>, <div class='tooltip'>step_BoxCox<span class='tooltiptext'>step_BoxCox creates a specification of a recipestep that will transform data using a simple Box-Coxtransformation.<\/span><\/div>, <div class='tooltip'>step_bs<span class='tooltiptext'>step_bs creates a specification of a recipe stepthat will create new columns that are basis expansions ofvariables using B-splines.<\/span><\/div>, <div class='tooltip'>step_center<span class='tooltiptext'>step_center creates a specification of a recipestep that will normalize numeric data to have a mean of zero.<\/span><\/div>, <div class='tooltip'>step_classdist<span class='tooltiptext'>step_classdist creates a specification of arecipe step that will convert numeric data into Mahalanobisdistance measurements to the data centroid. This is done foreach value of a categorical class variable.<\/span><\/div>, <div class='tooltip'>step_corr<span class='tooltiptext'>step_corr creates a specification of a recipestep that will potentially remove variables that have largeabsolute correlations with other variables.<\/span><\/div>, <div class='tooltip'>step_count<span class='tooltiptext'>step_count creates a specification of a recipestep that will create a variable that counts instances of aregular expression pattern in text.<\/span><\/div>, <div class='tooltip'>step_cut<span class='tooltiptext'>step_cut() creates a specification of a recipe step that cuts a numericvariable into a factor based on provided boundary values<\/span><\/div>, <div class='tooltip'>step_date<span class='tooltiptext'>step_date creates a specification of a recipestep that will convert date data into one or more factor ornumeric variables.<\/span><\/div>, <div class='tooltip'>step_depth<span class='tooltiptext'>step_depth creates a specification of a recipestep that will convert numeric data into measurement ofdata depth. This is done for each value of a categoricalclass variable.<\/span><\/div>, <div class='tooltip'>step_discretize<span class='tooltiptext'>step_discretize creates a specification of a recipestep that will convert numeric data into a factor withbins having approximately the same number of data points (basedon a training set).<\/span><\/div>, <div class='tooltip'>step_downsample<span class='tooltiptext'>htmllifecycle-soft-deprecated.svgalt=\"lifecycle-soft-deprecated\"step_downsample is now available as themis::step_downsample(). Thisfunction creates a specification of a recipe step that will removerows of a data set to make the occurrence of levels in a specific factorlevel equal.<\/span><\/div>, <div class='tooltip'>step_dummy<span class='tooltiptext'>step_dummy creates a specification of a recipestep that will convert nominal data (e.g. character or factors)into one or more numeric binary model terms for the levels ofthe original data.<\/span><\/div>, <div class='tooltip'>step_factor2string<span class='tooltiptext'>step_factor2string will convert one or more factorvectors to strings.<\/span><\/div>, <div class='tooltip'>step_filter<span class='tooltiptext'>step_filter creates a specification of a recipe stepthat will remove rows using dplyr::filter().<\/span><\/div>, <div class='tooltip'>step_geodist<span class='tooltiptext'>step_geodist creates a specification of arecipe step that will calculate the distance betweenpoints on a map to a reference location.<\/span><\/div>, <div class='tooltip'>step_holiday<span class='tooltiptext'>step_holiday creates a specification of arecipe step that will convert date data into one or more binaryindicator variables for common holidays.<\/span><\/div>, <div class='tooltip'>step_hyperbolic<span class='tooltiptext'>step_hyperbolic creates a specification of arecipe step that will transform data using a hyperbolicfunction.<\/span><\/div>, <div class='tooltip'>step_ica<span class='tooltiptext'>step_ica creates a specification of a recipe stepthat will convert numeric data into one or more independentcomponents.<\/span><\/div>, <div class='tooltip'>step_impute_linear<span class='tooltiptext'>step_impute_linear creates a specification of a recipe step that willcreate linear regression models to impute missing data.<\/span><\/div>, <div class='tooltip'>step_integer<span class='tooltiptext'>step_integer creates a specification of a recipestep that will convert new data into a set of integers basedon the original data values.<\/span><\/div>, <div class='tooltip'>step_interact<span class='tooltiptext'>step_interact creates a specification of a recipestep that will create new columns that are interaction termsbetween two or more variables.<\/span><\/div>, <div class='tooltip'>step_intercept<span class='tooltiptext'>step_intercept creates a specification of a recipe step thatwill add an intercept or constant term in the first column of a datamatrix. step_intercept has defaults to predictor role sothat it is by default called in the bake step. Be careful to avoidunintentional transformations when calling steps withall_predictors.<\/span><\/div>, <div class='tooltip'>step_inverse<span class='tooltiptext'>step_inverse creates a specification of a recipestep that will inverse transform the data.<\/span><\/div>, <div class='tooltip'>step_invlogit<span class='tooltiptext'>step_invlogit creates a specification of a recipestep that will transform the data from real values to be betweenzero and one.<\/span><\/div>, <div class='tooltip'>step_isomap<span class='tooltiptext'>step_isomap creates a specification of a recipestep that will convert numeric data into one or more newdimensions.<\/span><\/div>, <div class='tooltip'>step_knnimpute<span class='tooltiptext'>step_knnimpute creates a specification of a recipe step that willimpute missing data using nearest neighbors.<\/span><\/div>, <div class='tooltip'>step_kpca<span class='tooltiptext'>step_kpca a specification of a recipe step thatwill convert numeric data into one or more principal componentsusing a kernel basis expansion.<\/span><\/div>, <div class='tooltip'>step_kpca_poly<span class='tooltiptext'>step_kpca_poly a specification of a recipe step thatwill convert numeric data into one or more principal componentsusing a polynomial kernel basis expansion.<\/span><\/div>, <div class='tooltip'>step_kpca_rbf<span class='tooltiptext'>step_kpca_rbf a specification of a recipe step thatwill convert numeric data into one or more principal componentsusing a radial basis function kernel basis expansion.<\/span><\/div>, <div class='tooltip'>step_lag<span class='tooltiptext'>step_lag creates a specification of a recipe step thatwill add new columns of lagged data. Lagged data willby default include NA values where the lag was induced.These can be removed with step_naomit(), or you mayspecify an alternative filler value with the defaultargument.<\/span><\/div>, <div class='tooltip'>step_lincomb<span class='tooltiptext'>step_lincomb creates a specification of a recipestep that will potentially remove numeric variables that havelinear combinations between them.<\/span><\/div>, <div class='tooltip'>step_log<span class='tooltiptext'>step_log creates a specification of a recipe stepthat will log transform data.<\/span><\/div>, <div class='tooltip'>step_logit<span class='tooltiptext'>step_logit creates a specification of a recipestep that will logit transform the data.<\/span><\/div>, <div class='tooltip'>step_lowerimpute<span class='tooltiptext'>step_lowerimpute creates a specification of a recipe stepdesigned for cases where the non-negative numeric data cannot bemeasured below a known value. In these cases, one method forimputing the data is to substitute the truncated value by arandom uniform number between zero and the truncation point.<\/span><\/div>, <div class='tooltip'>step_meanimpute<span class='tooltiptext'>step_meanimpute creates a specification of a recipe step that willsubstitute missing values of numeric variables by the training set mean ofthose variables.<\/span><\/div>, <div class='tooltip'>step_medianimpute<span class='tooltiptext'>step_medianimpute creates a specification of a recipe step that willsubstitute missing values of numeric variables by the training set median ofthose variables.<\/span><\/div>, <div class='tooltip'>step_modeimpute<span class='tooltiptext'>step_modeimpute creates a specification of arecipe step that will substitute missing values of nominalvariables by the training set mode of those variables.<\/span><\/div>, <div class='tooltip'>step_mutate<span class='tooltiptext'>step_mutate creates a specification of a recipe stepthat will add variables using dplyr::mutate().<\/span><\/div>, <div class='tooltip'>step_mutate_at<span class='tooltiptext'>step_mutate_at creates a specification of a recipe step that will modifythe selected variables using a common function.<\/span><\/div>, <div class='tooltip'>step_naomit<span class='tooltiptext'>step_naomit creates a specification of a recipe step thatwill remove observations (rows of data) if they contain NAor NaN values.<\/span><\/div>, <div class='tooltip'>step_nnmf<span class='tooltiptext'>step_nnmf creates a specification of a recipe stepthat will convert numeric data into one or more non-negativecomponents.<\/span><\/div>, <div class='tooltip'>step_normalize<span class='tooltiptext'>step_normalize creates a specification of a recipestep that will normalize numeric data to have a standarddeviation of one and a mean of zero.<\/span><\/div>, <div class='tooltip'>step_novel<span class='tooltiptext'>step_novel creates a specification of a recipestep that will assign a previously unseen factor level to anew value.<\/span><\/div>, <div class='tooltip'>step_ns<span class='tooltiptext'>step_ns creates a specification of a recipe stepthat will create new columns that are basis expansions ofvariables using natural splines.<\/span><\/div>, <div class='tooltip'>step_num2factor<span class='tooltiptext'>step_num2factor will convert one or more numeric vectors to factors(ordered or unordered). This can be useful when categories are encoded asintegers.<\/span><\/div>, <div class='tooltip'>step_nzv<span class='tooltiptext'>step_nzv creates a specification of a recipe stepthat will potentially remove variables that are highly sparseand unbalanced.<\/span><\/div>, <div class='tooltip'>step_ordinalscore<span class='tooltiptext'>step_ordinalscore creates a specification of arecipe step that will convert ordinal factor variables intonumeric scores.<\/span><\/div>, <div class='tooltip'>step_other<span class='tooltiptext'>step_other creates a specification of a recipestep that will potentially pool infrequently occurring valuesinto an \"other\" category.<\/span><\/div>, <div class='tooltip'>step_pca<span class='tooltiptext'>step_pca creates a specification of a recipe stepthat will convert numeric data into one or more principalcomponents.<\/span><\/div>, <div class='tooltip'>step_pls<span class='tooltiptext'>step_pls creates a specification of a recipe step that willconvert numeric data into one or more new dimensions.<\/span><\/div>, <div class='tooltip'>step_poly<span class='tooltiptext'>step_poly creates a specification of a recipestep that will create new columns that are basis expansions ofvariables using orthogonal polynomials.<\/span><\/div>, <div class='tooltip'>step_profile<span class='tooltiptext'>step_profile creates a specification of a recipe step thatwill fix the levels of all variables but one and will create asequence of values for the remaining variable. This step can behelpful when creating partial regression plots for additivemodels.<\/span><\/div>, <div class='tooltip'>step_range<span class='tooltiptext'>step_range creates a specification of a recipestep that will normalize numeric data to be within a pre-definedrange of values.<\/span><\/div>, <div class='tooltip'>step_ratio<span class='tooltiptext'>step_ratio creates a specification of a recipestep that will create one or more ratios out of numericvariables.<\/span><\/div>, <div class='tooltip'>step_regex<span class='tooltiptext'>step_regex creates a specification of a recipe step that willcreate a new dummy variable based on a regular expression.<\/span><\/div>, <div class='tooltip'>step_relevel<span class='tooltiptext'>step_relevel creates a specification of a recipestep that will reorder the provided factor columns so thatthe level specified by ref_level is first. This is usefulfor contr.treatment contrasts which take the first level as thereference.<\/span><\/div>, <div class='tooltip'>step_relu<span class='tooltiptext'>step_relu creates a specification of a recipe step thatwill apply the rectified linear or softplus transformations to numericdata. The transformed data is added as new columns to the data matrix.<\/span><\/div>, <div class='tooltip'>step_rename<span class='tooltiptext'>step_rename creates a specification of a recipe step that will addvariables using dplyr::rename().<\/span><\/div>, <div class='tooltip'>step_rename_at<span class='tooltiptext'>step_rename_at creates a specification of a recipe step that will renamethe selected variables using a common function.<\/span><\/div>, <div class='tooltip'>step_rm<span class='tooltiptext'>step_rm creates a specification of a recipe stepthat will remove variables based on their name, type, or role.<\/span><\/div>, <div class='tooltip'>step_rollimpute<span class='tooltiptext'>step_rollimpute creates a specification of arecipe step that will substitute missing values of numericvariables by the measure of location (e.g. median) within a moving window.<\/span><\/div>, <div class='tooltip'>step_sample<span class='tooltiptext'>step_sample creates a specification of a recipe stepthat will sample rows using dplyr::sample_n() ordplyr::sample_frac().<\/span><\/div>, <div class='tooltip'>step_scale<span class='tooltiptext'>step_scale creates a specification of a recipestep that will normalize numeric data to have a standarddeviation of one.<\/span><\/div>, <div class='tooltip'>step_shuffle<span class='tooltiptext'>step_shuffle creates a specification of a recipestep that will randomly change the order of rows for selectedvariables.<\/span><\/div>, <div class='tooltip'>step_slice<span class='tooltiptext'>step_slice creates a specification of a recipe stepthat will filter rows using dplyr::slice().<\/span><\/div>, <div class='tooltip'>step_spatialsign<span class='tooltiptext'>step_spatialsign is a specification of a recipestep that will convert numeric data into a projection on to aunit sphere.<\/span><\/div>, <div class='tooltip'>step_sqrt<span class='tooltiptext'>step_sqrt creates a specification of a recipestep that will square root transform the data.<\/span><\/div>, <div class='tooltip'>step_string2factor<span class='tooltiptext'>step_string2factor will convert one or more charactervectors to factors (ordered or unordered).<\/span><\/div>, <div class='tooltip'>step_unknown<span class='tooltiptext'>step_unknown creates a specification of a recipestep that will assign a missing value in a factor level to\"unknown\".<\/span><\/div>, <div class='tooltip'>step_unorder<span class='tooltiptext'>step_unorder creates a specification of a recipestep that will transform the data.<\/span><\/div>, <div class='tooltip'>step_upsample<span class='tooltiptext'>htmllifecycle-soft-deprecated.svgalt=\"lifecycle-soft-deprecated\"step_upsample is now available as themis::step_upsample(). Thisfunction creates a specification of a recipe step thatwill replicate rows of a data set to make the occurrence oflevels in a specific factor level equal.<\/span><\/div>, <div class='tooltip'>step_window<span class='tooltiptext'>step_window creates a specification of a recipestep that will create new columns that are the results offunctions that compute statistics across moving windows.<\/span><\/div>, <div class='tooltip'>step_YeoJohnson<span class='tooltiptext'>step_YeoJohnson creates a specification of arecipe step that will transform data using a simple Yeo-Johnsontransformation.<\/span><\/div>, <div class='tooltip'>step_zv<span class='tooltiptext'>step_zv creates a specification of a recipe stepthat will remove variables that contain only a single value.<\/span><\/div>, <div class='tooltip'>terms_select<span class='tooltiptext'>This function bakes the step function selectors and might beuseful when creating custom steps.<\/span><\/div>, <div class='tooltip'>tidy<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsrequired_pkgs, tidy, tunable magrittr%>% statsupdate<\/span><\/div>, <div class='tooltip'>tunable<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsrequired_pkgs, tidy, tunable magrittr%>% statsupdate<\/span><\/div>, <div class='tooltip'>update<span class='tooltiptext'>These objects are imported from other packages. Follow the linksbelow to see their documentation. genericsrequired_pkgs, tidy, tunable magrittr%>% statsupdate<\/span><\/div>, <div class='tooltip'>update_role<span class='tooltiptext'>update_role() alters an existing role in the recipe or assigns an initialrole to variables that do not yet have a declared role.add_role() adds an additional role to variables that already have a rolein the recipe. It does not overwrite old roles, as a single variable can havemultiple roles.remove_role() eliminates a single existing role in the recipe.<\/span><\/div>, <div class='tooltip'>yj_transform<span class='tooltiptext'>These are not to be used directly by the users.This internal function is used for printing steps.This internal function takes a list of selectors (e.g. termsin most steps) and returns a character vector version forprinting.<\/span><\/div>","<div class='tooltip'>dirtyreps<span class='tooltiptext'>Generates seasonal time series using either the kNN Bootstrap (non-parametric) or a numerically-fitted PARMA(1,1) (parametric) model. For the parametric model, the function automatically transforms the seasonal sub-series to normal and deseasonalizes prior to model fitting.<\/span><\/div>, <div class='tooltip'>dp<span class='tooltiptext'>Determines the optimal sequence of releases from the reservoir to minimise a penalty cost function based on water supply defict.<\/span><\/div>, <div class='tooltip'>dp_hydro<span class='tooltiptext'>Determines the optimal sequence of turbined releases to maximise the total energy produced by the reservoir.<\/span><\/div>, <div class='tooltip'>dp_multi<span class='tooltiptext'>Determines the optimal sequence of releases from the reservoir to minimise a penalty cost function based on water supply, spill, and water level. For water supply: Cost[t] = ((target - release[t]) / target) ^ loss_exp[1]). For flood control: Cost[t] = (Spill[t] / quantile(Q, spill_targ)) ^ loss_exp[2]. For amenity: Cost[t] = abs(((storage[t] - (vol_targ * capacity)) / (vol_targ * capacity))) ^ loss_exp[3].<\/span><\/div>, <div class='tooltip'>dp_supply<span class='tooltiptext'>Determines the optimal sequence of releases from the reservoir to minimise a penalty cost function based on water supply defict.<\/span><\/div>, <div class='tooltip'>Hurst<span class='tooltiptext'>Hurst coefficient estimation.<\/span><\/div>, <div class='tooltip'>Rippl<span class='tooltiptext'>Computes the Rippl no-failure storage for given time series of inflows and releases using the sequent peak algorithm.<\/span><\/div>, <div class='tooltip'>rrv<span class='tooltiptext'>Computes time-based, annual, and volumetric reliability, as well as resilience and dimensionless vulnerability for a single reservoir.<\/span><\/div>, <div class='tooltip'>sdp<span class='tooltiptext'>Derives the optimal release policy based on storage state, inflow class and within-year period.<\/span><\/div>, <div class='tooltip'>sdp_hydro<span class='tooltiptext'>Determines the optimal policy of turbined releases to maximise the total energy produced by the reservoir. The policy can be based on season and storage level, or season, storage level, and current-period inflow.<\/span><\/div>, <div class='tooltip'>sdp_multi<span class='tooltiptext'>Determines the optimal sequence of releases from the reservoir to minimise a penalty cost function based on water supply, spill, and water level. For water supply: Cost[t] = ((target - release[t]) / target) ^ loss_exp[1]). For flood control: Cost[t] = (Spill[t] / quantile(Q, spill_targ)) ^ loss_exp[2]. For amenity: Cost[t] = abs(((storage[t] - (vol_targ * capacity)) / (vol_targ * capacity))) ^ loss_exp[3].<\/span><\/div>, <div class='tooltip'>sdp_supply<span class='tooltiptext'>Derives the optimal release policy based on storage state and within-year period only.<\/span><\/div>, <div class='tooltip'>simRes<span class='tooltiptext'>Simulates a reservoir for a given inflow time series and assuming Standard Operating Policy (meet target at all times, unless constrained by available water in reservoir plus incoming flows) or an optimised policy deived using sdp_supply.<\/span><\/div>, <div class='tooltip'>storage<span class='tooltiptext'>Returns the required storage for given inflow time series, yield, and target time-based reliability. Assumes standard operating policy. Storage is computed iteratively using the bi-section method.<\/span><\/div>, <div class='tooltip'>yield<span class='tooltiptext'>Returns the yield for given inflow time series, reservoir capacity, and required time-based reliability. Assumes standard operating policy. Yield is computed iteratively using the bi-section method.<\/span><\/div>","<div class='tooltip'>barplotPortfolioRisk<span class='tooltiptext'>Creates a barplot on top with the portfolio capital allocation and anotherat the bottom with the risk contribution allocation whose profile is the target of therisk parity portfolio design with riskParityPortfolio.By default the plot is based on the package ggplot2, but the usercan also specify a simple base plot.<\/span><\/div>, <div class='tooltip'>riskParityPortfolio<span class='tooltiptext'>This function designs risk parity portfolios to equalize/distributethe risk contributions of the different assets, which is missing if we simplyconsider the overall volatility of the portfolio as in the mean-varianceMarkowitz portfolio. In addition to the vanilla formulation, where the riskcontributions are perfectly equalized subject to no shortselling and budgetconstraints, many other formulations are considered that allow for boxconstraints, as well as the inclusion of additional objectives like theexpected return and overall variance. In short, this function solves thefollowing problem: minimize R(w) - lmd_mu * t(w) %*% mu + lmd_var * t(w) %*% Sigma %*% w subject to sum(w) = 1, w_lb <= w <= w_ub, Cmat %*% w = cvec, Dmat %*% w <= dvec,where R(w) denotes the risk concentration,t(w) %*% mu is the expected return, t(w) %*% Sigma %*% w is theoverall variance, lmd_mu and lmd_var are the trade-off weightsfor the expected return and the variance terms, respectively, w_lb andw_ub are the lower and upper bound vector values for the portfolio vector w,Cmat %*% w = cvec denotes arbitrary linear equality constrains, andDmat %*% w = dvec denotes arbitrary linear inequality constrains.<\/span><\/div>","<div class='tooltip'>ISCCNBSblock<span class='tooltiptext'>A colour metric based on the ISCC-NBS System of Colour Designation.<\/span><\/div>","<div class='tooltip'>PPdiv<span class='tooltiptext'>The configuration of group divisible RPPBD obtained by applying the (ASC-RPPBD) method on a group divisible association scheme.<\/span><\/div>, <div class='tooltip'>PPGrectRightAng4<span class='tooltiptext'>The configuration of Generalized rectangular right angular RPPBD (4) obtained by applying the (ASC-RPPBD) method on a Generalized rectangular right angular association scheme (4).<\/span><\/div>, <div class='tooltip'>PPGrectRightAng5<span class='tooltiptext'>The configuration of Generalized rectangular right angular RPPBD (5) obtained by applying the (ASC-RPPBD) method on a Generalized rectangular right angular association scheme (5).<\/span><\/div>, <div class='tooltip'>PPGrectRightAng7<span class='tooltiptext'>The configuration of Generalized rectangular right angular RPPBD (7) obtained by applying the (ASC-RPPBD) method on a Generalized rectangular right angular association scheme (7).<\/span><\/div>, <div class='tooltip'>PPNestdiv<span class='tooltiptext'>The configuration of Nested group divisible RPPBD obtained by applying the (ASC-RPPBD) method on a nested group divisible association scheme.<\/span><\/div>, <div class='tooltip'>PPrect<span class='tooltiptext'>The configuration of rectangular RPPBD obtained by applying the (ASC-RPPBD) method on a rectangular association scheme.<\/span><\/div>, <div class='tooltip'>PPrightAng<span class='tooltiptext'>The configuration of right angular RPPBD obtained by applying the (ASC-RPPBD) method on a right angular association scheme.<\/span><\/div>, <div class='tooltip'>SpaceFilling<span class='tooltiptext'>The application of the (ASC-SF) algorithm on some association schemes to obtain new series of Space-filling Design.<\/span><\/div>","<div class='tooltip'>adj.risk<span class='tooltiptext'>Calculates adjusted risk for given relative risk and population proportions. This is an intermediate calculation in the calculation of effective probability of infection for risk-based surveillance activities<\/span><\/div>, <div class='tooltip'>adj.risk.sim<span class='tooltiptext'>Calculates adjusted risk for given relative risk and population proportions. This is an intermediate calculation in the calculation of effective probability of infection for risk-based surveillance activities. This function is similar to Adj.risk, except is adapted for use with multiple simulations instead of single RR values.<\/span><\/div>, <div class='tooltip'>ap<span class='tooltiptext'>Estimates apparent prevalence and confidence limits forgiven sample size and result, assuming representative sampling<\/span><\/div>, <div class='tooltip'>binom.agresti<span class='tooltiptext'>Calculates Agresti-Coull confidence limits for a simple proportion (apparent prevalence)<\/span><\/div>, <div class='tooltip'>binom.cp<span class='tooltiptext'>Calculates Clopper-Pearson exact binomial confidence limits for a simple proportion (apparent prevalence)<\/span><\/div>, <div class='tooltip'>binom.jeffreys<span class='tooltiptext'>Calculates Jeffreys confidence limits for a simple proportion (apparent prevalence)<\/span><\/div>, <div class='tooltip'>disc.prior<span class='tooltiptext'>Calculates the discounted prior probability of disease freedom, after adjusting for the probability of disease exceeding the design prevalence during the time period of the surveillance data being analysed<\/span><\/div>, <div class='tooltip'>epi.calc<span class='tooltiptext'>Calculates effective probability of infection (adjusted design prevalence) for each risk group for risk-based surveillance activities<\/span><\/div>, <div class='tooltip'>n.2stage<span class='tooltiptext'>Calculates sample sizes for a 2-stage representative survey (sampling of clusters and units within clusters) for disease freedom or detection, assuming imperfect test sensitivity, perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>n.ap<span class='tooltiptext'>Calculates sample size for estimating apparent prevalence (simple proportion)<\/span><\/div>, <div class='tooltip'>n.binom<span class='tooltiptext'>Calculates sample size for demonstrating freedom or detecting disease using binomial approach and assuming imperfect test sensitivity, perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>n.c.freecalc<span class='tooltiptext'>Calculates optimum sample size and cut-point number of positives to achieve specified population sensitivity, for given population size and other parameters, using freecalc algorithm, all paramaters must be scalars<\/span><\/div>, <div class='tooltip'>n.c.hp<span class='tooltiptext'>Calculates optimum sample size and cut-point positives to achieve specified population sensitivity, for given population size and other parameters, all paramaters must be scalars<\/span><\/div>, <div class='tooltip'>n.freecalc<span class='tooltiptext'>Calculates sample size required for a specified population sensitivity,for a given population size, cut-point number of positives and other parameters, using Freecalc algorithm. All paramaters must be scalars<\/span><\/div>, <div class='tooltip'>n.freedom<span class='tooltiptext'>Calculates sample size for demonstrating freedom or detecting disease using the appropriate method, depending on whether or not N provided (hypergeometric if N provided, binomial otherwise), assuming imperfect test sensitivity, perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>n.hp<span class='tooltiptext'>Calculates sample size to achieve specified population sensitivity with population specificity >= specified minimum value, for given population size, cut-point number of positives and other parameters, all paramaters must be scalars<\/span><\/div>, <div class='tooltip'>n.hypergeo<span class='tooltiptext'>Calculates sample size for demonstrating freedom or detecting disease using hypergeometric approximation and assuming imperfect test sensitivity, perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>n.pfree<span class='tooltiptext'>Calculates the sample size required to achieve a given value for probability of disease freedom<\/span><\/div>, <div class='tooltip'>n.pooled<span class='tooltiptext'>Calculates sample size to achieve desired population-level sensitivity, assuming pooled sampling and allowing for imperfect sensitivity and specificity of the pooled test<\/span><\/div>, <div class='tooltip'>n.rb<span class='tooltiptext'>Calculates sample size for risk-based sampling for a single risk factor and using binomial method<\/span><\/div>, <div class='tooltip'>n.rb.2stage.1<span class='tooltiptext'>Calculates sample size required (clusters and units)for a 2-stage risk-based survey with a single risk factor at the cluster level only.<\/span><\/div>, <div class='tooltip'>n.rb.2stage.2<span class='tooltiptext'>Calculates sample size required (clusters and units)for a 2-stage risk-based survey with risk factors at either cluster level or unit level, or both.<\/span><\/div>, <div class='tooltip'>n.rb.varse<span class='tooltiptext'>Calculates sample size for risk-based sampling for a single risk factor and varying unit sensitivity, using binomial method<\/span><\/div>, <div class='tooltip'>n.tp<span class='tooltiptext'>Calculates sample size for estimating true prevalence using normal approximation<\/span><\/div>, <div class='tooltip'>pfree.1<span class='tooltiptext'>Calculates the posterior probability (confidence) of disease freedom (negative predictive value) for a single time period<\/span><\/div>, <div class='tooltip'>pfree.calc<span class='tooltiptext'>Calculates the probability (confidence) of disease freedom forgiven prior, sep and p.intro over 1 or more time periods<\/span><\/div>, <div class='tooltip'>pfree.equ<span class='tooltiptext'>Calculates equilibrium probability of disease freedom and equilibrium prior probability of freedom, after discounting for probability of introduction<\/span><\/div>, <div class='tooltip'>pstar.calc<span class='tooltiptext'>Calculates design prevalence required for given sample size and desired population-level sensitivity, assuming imperfect test sensitivity, perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>se.parallel<span class='tooltiptext'>Calculates the combined sensitivity for multiple tests interpreted in parallel (assuming independence)<\/span><\/div>, <div class='tooltip'>se.series<span class='tooltiptext'>Calculates the combined sensitivity for multiple tests interpreted in series (assuming independence)<\/span><\/div>, <div class='tooltip'>sep<span class='tooltiptext'>Calculates population sensitivity using appropriate method, depending on whether or not N provided (hypergeometric if N provided, binomial otherwise), assuming perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>sep.binom<span class='tooltiptext'>Calculates population sensitivity for detecting disease, assuming imperfect test sensitivity and specificity and representative sampling, using binomial distribution (assumes large or unknown population size and that cut-point number of reactors for a positive result = 1)<\/span><\/div>, <div class='tooltip'>sep.binom.imperfect<span class='tooltiptext'>Calculates population sensitivity for a large or unknown population and allowing for imperfect test sensitivity and specificity, using Binomial distribution an allowing for a variable cut-point number of positives to classify as positive<\/span><\/div>, <div class='tooltip'>sep.exact<span class='tooltiptext'>Calculates population sensitivity for detecting disease assuming imperfect test sensitivity, perfect test specificity and a census of all units in the population<\/span><\/div>, <div class='tooltip'>sep.freecalc<span class='tooltiptext'>Calculates population sensitivity for a finite population and allowing for imperfect test sensitivity and specificity, using Freecalc method<\/span><\/div>, <div class='tooltip'>sep.hp<span class='tooltiptext'>Calculates population sensitivity for a finite population and allowing for imperfect test sensitivity and specificity, using Hypergeometric distribution<\/span><\/div>, <div class='tooltip'>sep.hypergeo<span class='tooltiptext'>Calculates population sensitivity for detecting disease, assuming imperfect test sensitivity, perfect test specificity and representative sampling, using hypergeometric approximation (assumes known population size)<\/span><\/div>, <div class='tooltip'>sep.passive<span class='tooltiptext'>Estimates the population sensitivity of a passive surveillance system. Assumes comprehensive population coverage and samling of representative affected units from infected clusters<\/span><\/div>, <div class='tooltip'>sep.pfree<span class='tooltiptext'>Calculates the population sensitivity required to achieve a given value for probability of disease freedom<\/span><\/div>, <div class='tooltip'>sep.pooled<span class='tooltiptext'>Calculates population sensitivity (sep) and population specificity (spp) assuming pooled sampling and allowing for imperfect sensitivity and specificity of the pooled test<\/span><\/div>, <div class='tooltip'>sep.prior<span class='tooltiptext'>Calculates the population sensitivity required to achieve a given value for the prior (discounted) probability of disease freedom<\/span><\/div>, <div class='tooltip'>sep.rb.bin<span class='tooltiptext'>Calculates risk-based population sensitivity with a single risk factor, using binomial method (assumes a large population), allows for unit sensitivity to vary among risk strata<\/span><\/div>, <div class='tooltip'>sep.rb.bin.varse<span class='tooltiptext'>Calculates population sensitivity for a single risk factor and varying unit sensitivity using binomial method (assumes large population)<\/span><\/div>, <div class='tooltip'>sep.rb.hypergeo<span class='tooltiptext'>Calculates risk-based population sensitivity with a single risk factor, using the hypergeometric method (assuming a finite and known population size), allows for unit sensitivity to vary among risk strata<\/span><\/div>, <div class='tooltip'>sep.rb.hypergeo.varse<span class='tooltiptext'>Calculates population sensitivity for a single risk factor and varying unit sensitivity using hypergeometric approximation method (assumes known population size)<\/span><\/div>, <div class='tooltip'>sep.rb2.binom<span class='tooltiptext'>Calculates risk-based population sensitivity for two risk factors, using binomial method (assumes a large population)<\/span><\/div>, <div class='tooltip'>sep.rb2.hypergeo<span class='tooltiptext'>Calculates risk-based population sensitivity for two risk factors, using hypergeometric approximation method (assumes a known population size)<\/span><\/div>, <div class='tooltip'>sep.sys<span class='tooltiptext'>Calculates population-level (system) sensitivity for representative 2-stage sampling (sampling of clusters and units within clusters), assuming imperfect test sensitivity and perfect test specificity<\/span><\/div>, <div class='tooltip'>sep.var.se<span class='tooltiptext'>Calculates population-level sensitivity where unit sensitivity varies and using the appropriate method, depending on whether or not N provided (hypergeometric if N provided, binomial otherwise), assuming perfect test specificity and representative sampling<\/span><\/div>, <div class='tooltip'>sp.parallel<span class='tooltiptext'>Calculates the combined specificity for multiple tests interpreted in parallel (assuming independence)<\/span><\/div>, <div class='tooltip'>sp.series<span class='tooltiptext'>Calculates the combined specificity for multiple tests interpreted in series (assuming independence)<\/span><\/div>, <div class='tooltip'>sph.binom<span class='tooltiptext'>Calculates population specificity for a large or unknown population, using the Binomial distribution and adjusting for cut-point number of positives<\/span><\/div>, <div class='tooltip'>sph.hp<span class='tooltiptext'>Calculates population specificity for a finite population and imperfect test, using Hypergeometric distribution<\/span><\/div>, <div class='tooltip'>spp<span class='tooltiptext'>Calculates population specificity assuming representative sampling<\/span><\/div>, <div class='tooltip'>sse.combined<span class='tooltiptext'>Calculates overall system sensitivity for multiple components, accounting for lack of independence (overlap) between components<\/span><\/div>, <div class='tooltip'>sse.rb.2stage<span class='tooltiptext'>Calculates system sensitivity for 2 stage risk-based sampling, llowing for a single risk factor at each stage and using either binomial or hypergeometric approxiation<\/span><\/div>, <div class='tooltip'>tp<span class='tooltiptext'>Estimates true prevalence and confidence limits forgiven sample size and result, according to specified method<\/span><\/div>, <div class='tooltip'>tp.normal<span class='tooltiptext'>Estimates true prevalence and confidence limits for estimates based on normal approximation<\/span><\/div>","<div class='tooltip'>cpro<span class='tooltiptext'>Calculate proportion and margin of error (unequal-sized cluster sample)<\/span><\/div>, <div class='tooltip'>dedupe<span class='tooltiptext'>Removes duplicate observations within collected data<\/span><\/div>, <div class='tooltip'>dupe<span class='tooltiptext'>Identifies duplicate values within collected data<\/span><\/div>, <div class='tooltip'>psampcalc<span class='tooltiptext'>Determines sample size by strata using sub-units<\/span><\/div>, <div class='tooltip'>rmissing<span class='tooltiptext'>Identifies missing points between sample and collected data<\/span><\/div>, <div class='tooltip'>rpro<span class='tooltiptext'>Calculate proportion and margin of error (simple random sample)<\/span><\/div>, <div class='tooltip'>rsamp<span class='tooltiptext'>Draws simple random sample without replacement<\/span><\/div>, <div class='tooltip'>rsampcalc<span class='tooltiptext'>Determines random sample size<\/span><\/div>, <div class='tooltip'>smissing<span class='tooltiptext'>Identifies number of missing points by strata between sample and collected data<\/span><\/div>, <div class='tooltip'>spro<span class='tooltiptext'>Calculate proportion and margin of error (stratified sample)<\/span><\/div>, <div class='tooltip'>ssamp<span class='tooltiptext'>Draws stratifed sample without replacement using proportional allocation<\/span><\/div>, <div class='tooltip'>ssampcalc<span class='tooltiptext'>Determines sample size by strata using proportional allocation<\/span><\/div>","<div class='tooltip'>as_scdf<span class='tooltiptext'>as_scdfConverts a data frame to an scdf object<\/span><\/div>, <div class='tooltip'>autocorr<span class='tooltiptext'>The autocorr function calculates autocorrelations within each phase andacross all phases.<\/span><\/div>, <div class='tooltip'>autocorrSC<span class='tooltiptext'>The autocorr function calculates autocorrelations within each phase andacross all phases.<\/span><\/div>, <div class='tooltip'>combine<span class='tooltiptext'>Combine single-case data frames into a multiple case study<\/span><\/div>, <div class='tooltip'>corrected_tau<span class='tooltiptext'>Kendalls tau correlation for the dependent variable and the phase variable is calculatedafter correcting for a baseline trend.<\/span><\/div>, <div class='tooltip'>corrected_tauSC<span class='tooltiptext'>Kendalls tau correlation for the dependent variable and the phase variable is calculatedafter correcting for a baseline trend.<\/span><\/div>, <div class='tooltip'>describeSC<span class='tooltiptext'>The describeSC function provides common descriptive statistics forsingle-case data.<\/span><\/div>, <div class='tooltip'>design_rSC<span class='tooltiptext'>The rSC function generates random single-case data framesfor monte-carlo studies and demonstration purposes.design_rSC is used to set up a design matrix with all parameters needed for the rSC function.<\/span><\/div>, <div class='tooltip'>estimate_design<span class='tooltiptext'>This functions takes an scdf and extracts design parameters. The resultingobject can be unsed to randomly create new scdf files with the sameunderlying parameters. This is usefull for monte-carlo studies andbootstrapping procedures.<\/span><\/div>, <div class='tooltip'>estimateSC<span class='tooltiptext'>This functions takes an scdf an extracts design parameters. The resulting inobject can be unsed to randomly create new scdf files with the sameunderlying parameters. This might be usefull for monte-carlo studies andbootstrapping procedures.<\/span><\/div>, <div class='tooltip'>export<span class='tooltiptext'>This function is in an experimental status.Export creates html files of tables or displayes them directly in the viewer pane of rstudio.When applied in rmarkdown, tables can also be created for pdf/latex output.<\/span><\/div>, <div class='tooltip'>fill_missing<span class='tooltiptext'>The fillmissingSC function replaces missing measurements insingle-case data.<\/span><\/div>, <div class='tooltip'>fillmissingSC<span class='tooltiptext'>The fillmissingSC function replaces missing measurements insingle-case data.<\/span><\/div>, <div class='tooltip'>hplm<span class='tooltiptext'>The hplm function computes a hierarchical piecewise regression model.<\/span><\/div>, <div class='tooltip'>longSCDF<span class='tooltiptext'>The longSCDF function transposes a scdf into one long data frame.Additionally, a data frame can be merged that includes data of the subjects.This might be helpful to prepare data to be used with other packages than scan.<\/span><\/div>, <div class='tooltip'>makeSCDF<span class='tooltiptext'>The class scdf stores single-case study data with one or moresingle-cases.<\/span><\/div>, <div class='tooltip'>makesingleSC<span class='tooltiptext'>The makesingleSC function combines multiple single-case data frames into onesingle-case data frame.<\/span><\/div>, <div class='tooltip'>mplm<span class='tooltiptext'>This function is in an experimental status.The mplm function computes a multivariate piecewise regression model.<\/span><\/div>, <div class='tooltip'>nap<span class='tooltiptext'>The nap function calculates the nonoverlap of all pairs (NAP; Parker& Vannest, 2009). NAP summarizes the overlap between all pairs of phase Aand phase B data points. If an increase of phase B scores is expected, anon-overlapping pair has a higher phase B data point. The NAP equalsnumber of pairs showing no overlap / number of pairs. Because NAP canonly take values between 50 and 100 percent, a rescaled and therefore moreintuitive NAP (0-100%) is also displayed.<\/span><\/div>, <div class='tooltip'>outlier<span class='tooltiptext'>Identifies and drops outliers within a single-case data frame (scdf).<\/span><\/div>, <div class='tooltip'>outlierSC<span class='tooltiptext'>Identifies and drops outliers within a single-case data frame (scdf).<\/span><\/div>, <div class='tooltip'>overlap<span class='tooltiptext'>The overlap function provides the most common overlap indices forsingle-case data and some additional statistics.<\/span><\/div>, <div class='tooltip'>overlapSC<span class='tooltiptext'>The overlap function provides the most common overlap indices forsingle-case data and some additional statistics.<\/span><\/div>, <div class='tooltip'>pand<span class='tooltiptext'>The pand function calculates the percentage of all non-overlappingdata (PAND; Parker, Hagan-Burke, & Vannest, 2007), an index to quantify alevel increase (or decrease) in performance after the onset of anintervention.<\/span><\/div>, <div class='tooltip'>pem<span class='tooltiptext'>The pem function returns the percentage of phase B data exceeding thephase A median. Additionally, a chi square test against a 50/50distribution is computed. Different measures of central tendency can beaddressed for alternative analyses.<\/span><\/div>, <div class='tooltip'>pet<span class='tooltiptext'>The pet function provides the percentage of phase B data pointsexceeding the prediction based on the phase A trend. A binomial test againsta 50/50 distribution is computed. Furthermore, the percentage of phase Bdata points exceeding the upper (or lower) 95 percent confidence interval ofthe predicted progress is computed.<\/span><\/div>, <div class='tooltip'>plm<span class='tooltiptext'>The plm function computes a piecewise regression model (see Huitema &McKean, 2000).<\/span><\/div>, <div class='tooltip'>plotSC<span class='tooltiptext'>This function provides a plot of a single-case or multiplesingle-cases.<\/span><\/div>, <div class='tooltip'>pnd<span class='tooltiptext'>This function returns the percentage of non-overlapping data. Due to itserror-proneness the PND should not be used, but nap orpand instead (see Parker & Vannest, 2009).<\/span><\/div>, <div class='tooltip'>power_test<span class='tooltiptext'>The power_test command conducts a Monte-Carlo study on thetest-power and alpha-error of a set of single-cases. The distribution values of the Monte-Carlo sample are either specified by the user or estimated based on actual data.<\/span><\/div>, <div class='tooltip'>power_testSC<span class='tooltiptext'>The power_test command conducts a Monte-Carlo study on thetest-power and alpha-error of a set of single-cases. The distribution values of the Monte-Carlo sample are either specified by the user or estimated based on actual data.<\/span><\/div>, <div class='tooltip'>power.testSC<span class='tooltiptext'>!! This function is deprecated. Please use the power_testSC fucntion !!The power.testSC command conducts a Monte-Carlo study on thetest-power and alpha-error of a randomization-test and apiecewise-regression model. The distribution values of the Monte-Carlosample are either specified by the user or estimated based on actual data.<\/span><\/div>, <div class='tooltip'>rand_test<span class='tooltiptext'>The randSC function computes a randomization test for single ormultiple baseline single-case data. The function is based on an algorithmfrom the SCRT package (Bulte & Onghena, 2009, 2012), but rewrittenand extended for the use in AB designs.<\/span><\/div>, <div class='tooltip'>rand.test<span class='tooltiptext'>The randSC function computes a randomization test for single ormultiple baseline single-case data. The function is based on an algorithmfrom the SCRT package (Bulte & Onghena, 2009, 2012), but rewrittenand extended for the use in AB designs.<\/span><\/div>, <div class='tooltip'>randSC<span class='tooltiptext'>The randSC function computes a randomization test for single ormultiple baseline single-case data. The function is based on an algorithmfrom the SCRT package (Bulte & Onghena, 2009, 2012), but rewrittenand extended for the use in AB designs.<\/span><\/div>, <div class='tooltip'>ranks<span class='tooltiptext'>Rank-transformation of single-case data files<\/span><\/div>, <div class='tooltip'>rankSC<span class='tooltiptext'>Rank-transformation of single-case data files<\/span><\/div>, <div class='tooltip'>rci<span class='tooltiptext'>CAUTION! This function is still under development and not ready foruse! The rciSC function computes three indices of reliable change(Wise, 2004) and corresponding descriptive statistics.<\/span><\/div>, <div class='tooltip'>rCi<span class='tooltiptext'>CAUTION! This function is still under development and not ready foruse! The rciSC function computes three indices of reliable change(Wise, 2004) and corresponding descriptive statistics.<\/span><\/div>, <div class='tooltip'>rciSC<span class='tooltiptext'>CAUTION! This function is still under development and not ready foruse! The rciSC function computes three indices of reliable change(Wise, 2004) and corresponding descriptive statistics.<\/span><\/div>, <div class='tooltip'>readSC<span class='tooltiptext'>Use the readSC function to import single-case data from structured.csv or the readSC.excel function for importing excel files.<\/span><\/div>, <div class='tooltip'>readSC.excel<span class='tooltiptext'>Use the readSC function to import single-case data from structured.csv or the readSC.excel function for importing excel files.<\/span><\/div>, <div class='tooltip'>rSC<span class='tooltiptext'>The rSC function generates random single-case data framesfor monte-carlo studies and demonstration purposes.design_rSC is used to set up a design matrix with all parameters needed for the rSC function.<\/span><\/div>, <div class='tooltip'>scdf<span class='tooltiptext'>The class scdf stores single-case study data with one or moresingle-cases.<\/span><\/div>, <div class='tooltip'>scdf_attr<span class='tooltiptext'>Set and get scdf attributes<\/span><\/div>, <div class='tooltip'>scdf_attr<-<span class='tooltiptext'>Set and get scdf attributes<\/span><\/div>, <div class='tooltip'>select_cases<span class='tooltiptext'>Select a subset of cases<\/span><\/div>, <div class='tooltip'>shift<span class='tooltiptext'>Shifting the values might be helpful in cases where the measurement timeis given as a time variable (see example below).<\/span><\/div>, <div class='tooltip'>shiftSC<span class='tooltiptext'>Shifting the values might be helpful in cases where the measurement timeis given as a time variable (see example below).<\/span><\/div>, <div class='tooltip'>smooth_cases<span class='tooltiptext'>The smooth_cases function provides procedures to smooth single-case data(i.e., to eliminate noise). A moving average function (mean- ormedian-based) replaces each data point by the average of the surroundingdata points step-by-step. With a local regression function, each data pointis regressed by its surrounding data points.<\/span><\/div>, <div class='tooltip'>smoothSC<span class='tooltiptext'>The smooth_cases function provides procedures to smooth single-case data(i.e., to eliminate noise). A moving average function (mean- ormedian-based) replaces each data point by the average of the surroundingdata points step-by-step. With a local regression function, each data pointis regressed by its surrounding data points.<\/span><\/div>, <div class='tooltip'>standardize<span class='tooltiptext'>This function scales the measured values of an scdf file. It allows for meancentering and standardization based on each single-case data set or ascaling across all cases included in an scdf.<\/span><\/div>, <div class='tooltip'>style_plot<span class='tooltiptext'>The style_plot function is used to create graphical styles for a single-case plot<\/span><\/div>, <div class='tooltip'>style_plotSC<span class='tooltiptext'>The style_plot function is used to create graphical styles for a single-case plot<\/span><\/div>, <div class='tooltip'>style.plotSC<span class='tooltiptext'>The style_plot function is used to create graphical styles for a single-case plot<\/span><\/div>, <div class='tooltip'>tau_u<span class='tooltiptext'>This function calculates indices of the Tau-U family as proposed by Parkeret al. (2011).<\/span><\/div>, <div class='tooltip'>tauUSC<span class='tooltiptext'>This function calculates indices of the Tau-U family as proposed by Parkeret al. (2011).<\/span><\/div>, <div class='tooltip'>trend<span class='tooltiptext'>The trendSC function provides an overview of linear trends insingle-case data. By default, it gives you the intercept and slope of alinear and a squared regression of measurement-time on scores. Models arecomputed separately for each phase and across all phases. For amore advanced application, you can add regression models using the Rspecific formula class.<\/span><\/div>, <div class='tooltip'>trendSC<span class='tooltiptext'>The trendSC function provides an overview of linear trends insingle-case data. By default, it gives you the intercept and slope of alinear and a squared regression of measurement-time on scores. Models arecomputed separately for each phase and across all phases. For amore advanced application, you can add regression models using the Rspecific formula class.<\/span><\/div>, <div class='tooltip'>truncate_phase<span class='tooltiptext'>This function truncates data points at the beginning and / or end each phase.<\/span><\/div>, <div class='tooltip'>truncateSC<span class='tooltiptext'>This function truncates data points at the beginning and / or end each phase.<\/span><\/div>, <div class='tooltip'>writeSC<span class='tooltiptext'>This function restructures and exports single-case data into a .csv-file.<\/span><\/div>","<div class='tooltip'>lahiri.design<span class='tooltiptext'>Draw Samples Using Lahiri's Method<\/span><\/div>","<div class='tooltip'>analyse_holos<span class='tooltiptext'>Provide numeric tools and graphical tools to analyse Holos data.<\/span><\/div>, <div class='tooltip'>ardi<span class='tooltiptext'>Spot the most singular or particular data with respect to all descriptors and to twoqualitative variables and all their possible categories combinations.Computes the highest differences between all the categories of the variables product, panelist and all their possible combinations,with respect to a set of quantitative variables (the sensory descriptors).<\/span><\/div>, <div class='tooltip'>averagetable<span class='tooltiptext'>Returns the (products,descriptors) matrix with entries the means over panelists and sessions. Computes analyses of variance automatically for a given model and a set of quantitative variables. Returns a data matrix where each row is associated with each category of a given categorical variable (in most cases, the categorical variable is the product variable), each column is associated with a quantitative variable, and each cell is the corresponding adjusted mean or mean. Computes the average data table with respect to a categorical variable and a set of quantitative variables.<\/span><\/div>, <div class='tooltip'>barrow<span class='tooltiptext'>Returns as many barplots as there are rows in a matrix. The barplots are automatically generated for all the quantitative variables.<\/span><\/div>, <div class='tooltip'>boot<span class='tooltiptext'>Simulate virtual panels for the sorting task, the napping, the sorting napping, the free choice profiling, the hierarchical sorting task<\/span><\/div>, <div class='tooltip'>boxprod<span class='tooltiptext'>Returns as many boxplots as there are categories for a given categorical variable of interest (in most cases, the product variable). The boxplots are automatically generated for all the quantitative variables (in our type of applications, variables are often sensory descriptors).<\/span><\/div>, <div class='tooltip'>CA_JAR<span class='tooltiptext'>Plot the CA graph.<\/span><\/div>, <div class='tooltip'>calculate.average<span class='tooltiptext'>Fonction intermediaire<\/span><\/div>, <div class='tooltip'>carto<span class='tooltiptext'>Performs preference mapping techniques based on multidimensional exploratory data analysis.<\/span><\/div>, <div class='tooltip'>cartoconsumer<span class='tooltiptext'>Performs preference mapping techniques based on multidimensional exploratory data analysis and segmentation of consumers.<\/span><\/div>, <div class='tooltip'>coltable<span class='tooltiptext'>Return a colored display of a data frame according to 4 threshold levels.<\/span><\/div>, <div class='tooltip'>ConsensualWords<span class='tooltiptext'>This function is designed to point out the words that are used in a consensual way by consumers from a sorting task.<\/span><\/div>, <div class='tooltip'>ConsistencyIdeal<span class='tooltiptext'>Evaluate the sensory and hedonic consistency of the ideal data, both at the consumer and panel level.<\/span><\/div>, <div class='tooltip'>construct.axes<span class='tooltiptext'>This function is especially designed to be used in a sensory data analysis context.Returns the coordinates of the products when performing either PCA or MFA and the coordinates of the \"partial\" products when performing MFA.Returns also the panelists' coordinates when projected as illustrative rows onto the products' space.Produces graphs of products and descriptors from the output of PCA or MFA.<\/span><\/div>, <div class='tooltip'>cpa<span class='tooltiptext'>Performs preference mapping techniques based on multidimensional exploratory data analysis. This methodology isoriented towards consumers' preferences; here consumers are pictured according only to their preferences. In this manner, the distance between two consumers is very natural and easy to interpret, and a clustering of the consumers is also very easy to obtain.<\/span><\/div>, <div class='tooltip'>decat<span class='tooltiptext'>This function is designed to point out the variables that are the most characteristic according to the set of products in its whole, and to each of the products in particular. This function is designed to test the main effect of a categorical variable (F-test) and the significance of its coefficients (T-test) for a set of endogenous variables and a given analysis of variance model. In most cases, the main effect is the product effect and the endogenous variables are the sensory descriptors.<\/span><\/div>, <div class='tooltip'>fahst<span class='tooltiptext'>Perform Factorial Approach for Hierarchical Sorting Task data (FAHST) on a table where the rows (i) are products and the columns (j) are for each consumer the partitionning variables associated with nested sorting. The columns are grouped by consumer.For the partitionning variables, the label associated with a group can be an arbirary label (for example G1 for group 1, etc.) or the words associated with the group in the case of qualified hierarchical sorting.<\/span><\/div>, <div class='tooltip'>fasnt<span class='tooltiptext'>Perform Factorial Approach for Sorting Napping Task data (FASNT) on a table where the rows (i) are products and the columns (j) are for each consumer the coordinates of the products on the tablecloth associated with napping on the one hand and the partitionning variable associated with categorization on the other hand. The columns are grouped by consumer.For the partitionning variable, the label associated with a group can be an arbirary label (for example G1 for group 1, etc.) or the words associated with the group in the case of qualified sorted napping.<\/span><\/div>, <div class='tooltip'>fast<span class='tooltiptext'>Perform Factorial Approach for Sorting Task data (FAST) on a table where the rows (i) are products and the columns (j) are consumers.A cell (i,j) corresponds either to the number of the group to which the product i belongs for the consumer j, or, in the case of \"qualified\" categorization, to the sequence of words associted with the group to which the product i belongs for the consumer j.<\/span><\/div>, <div class='tooltip'>fcp<span class='tooltiptext'>Free choice profiling with confidence ellipses<\/span><\/div>, <div class='tooltip'>format_holos<span class='tooltiptext'>Format the data exported from the Holos platform.<\/span><\/div>, <div class='tooltip'>graphinter<span class='tooltiptext'>This function is designed to display the interaction between two qualitative variables, in most casesthe product and the session variables.<\/span><\/div>, <div class='tooltip'>histprod<span class='tooltiptext'>Computes automatically histograms for a set of quantitative variables.<\/span><\/div>, <div class='tooltip'>hsortplot<span class='tooltiptext'>Plot consumers' hierarchical sorting<\/span><\/div>, <div class='tooltip'>IdMap<span class='tooltiptext'>Create the ideal map, a map based on the ideal profiles provided by the consumers.<\/span><\/div>, <div class='tooltip'>IdMapConsumer<span class='tooltiptext'>Create the ideal map and plot the ideal areas of the categories of qualitative variables. And perform 2 tests: a global test in order to highlight the significance of the difference between ideals of all the categories of the same variable; a pair comparison test to highlight the significance between 2 categories of the same variable.<\/span><\/div>, <div class='tooltip'>indscal<span class='tooltiptext'>This version of the Indscal model is specially adapted to Napping data type, i.e. products (stimuli)are positioned on a tableclothe by panelists, then their coordinates are used as input for the Indscal model.<\/span><\/div>, <div class='tooltip'>interact<span class='tooltiptext'>Computes automatically the interaction coefficients between two quantitative variablescol.p and col.j for the following model:\"~col.p+col.j+col.p:col.j\".<\/span><\/div>, <div class='tooltip'>JAR<span class='tooltiptext'>Just About Right<\/span><\/div>, <div class='tooltip'>magicsort<span class='tooltiptext'>Sort the rows and columns of a matrix in a \"magic\" order or by ascending (or descending) mean or median or geometrical mean.<\/span><\/div>, <div class='tooltip'>MultiIdeal<span class='tooltiptext'>By the use of confidence ellipses, this procedure checks whether consumersassociate the different products tested to a single or to multiple ideals.<\/span><\/div>, <div class='tooltip'>nappeplot<span class='tooltiptext'>Plot panelists' tableclothe.<\/span><\/div>, <div class='tooltip'>nappesortplot<span class='tooltiptext'>Plot consumers' sorted tablecloth.<\/span><\/div>, <div class='tooltip'>optimaldesign<span class='tooltiptext'>Construction of an optimal design balanced for first order of carry-over effect.<\/span><\/div>, <div class='tooltip'>paneliperf<span class='tooltiptext'>Computes automatically P-values, Vtests, residuals, r-square for each category of a given qualitative variable (e.g. the panelist variable);Computes he agreement between each panelist and the panel results;Gives the panel results (optional).<\/span><\/div>, <div class='tooltip'>panellipse<span class='tooltiptext'>Virtual panels are generated using Boostrap techniques in order to displayconfidence ellipses around products.<\/span><\/div>, <div class='tooltip'>panellipse.session<span class='tooltiptext'>Virtual panels are generated using Boostrap techniques in order to displayconfidence ellipses around products.<\/span><\/div>, <div class='tooltip'>panelmatch<span class='tooltiptext'>Comparison of panels.<\/span><\/div>, <div class='tooltip'>panelperf<span class='tooltiptext'>Computes automatically P-values associated with the F-test as well as the residual term for a given analysis of variance model.<\/span><\/div>, <div class='tooltip'>plot.fahst<span class='tooltiptext'>Plot the graphs for Factorial Approach for Hierarchical Sorting Task data (FAHST).<\/span><\/div>, <div class='tooltip'>plot.fasnt<span class='tooltiptext'>Plot the graphs for Factorial Approach for Sorting Napping Task data (FASNT).<\/span><\/div>, <div class='tooltip'>plot.fast<span class='tooltiptext'>Plot the graphs for Factorial Approach for Sorting Task data (FAST).<\/span><\/div>, <div class='tooltip'>plot.IdMap<span class='tooltiptext'>Option to replot by zooming on the ideal map.<\/span><\/div>, <div class='tooltip'>plot.JAR<span class='tooltiptext'>Plot the graphs for the penalty analysis.<\/span><\/div>, <div class='tooltip'>plot.WordCountAna<span class='tooltiptext'>Draw the Word-Count based methods Analysis (WordCountAna) graphs<\/span><\/div>, <div class='tooltip'>plotellipse<span class='tooltiptext'>Plot confidence ellipses.<\/span><\/div>, <div class='tooltip'>plotellipseinter<span class='tooltiptext'>Construit l'ensemble des ellipses de confiance<\/span><\/div>, <div class='tooltip'>plotellipseinterhmfa<span class='tooltiptext'>Construit l'ensemble des ellipses de confiance<\/span><\/div>, <div class='tooltip'>plotpanelist<span class='tooltiptext'>Displays panelists' sensory profiles onto the products' space<\/span><\/div>, <div class='tooltip'>pmfa<span class='tooltiptext'>Performs Multiple Factor Analysis combined with Procrustean Analysis.<\/span><\/div>, <div class='tooltip'>print.fahst<span class='tooltiptext'>Print Factorial Approach for Hierarchical Sorting Task data (FAHST) results.<\/span><\/div>, <div class='tooltip'>print.fasnt<span class='tooltiptext'>Print Factorial Approach for Sorting Napping Task data (FASNT) results.<\/span><\/div>, <div class='tooltip'>print.fast<span class='tooltiptext'>Print Factorial Approach for Sorting Task data (FAST) results.<\/span><\/div>, <div class='tooltip'>scalebypanelist<span class='tooltiptext'>Returns a data frame with entries the means of the products over the sessions for the whole panel and for each panelist.<\/span><\/div>, <div class='tooltip'>search.desc<span class='tooltiptext'>This function is designed to select the significant descriptors in a data frame<\/span><\/div>, <div class='tooltip'>simulation<span class='tooltiptext'>Simulate virtual panels<\/span><\/div>, <div class='tooltip'>triangle.design<span class='tooltiptext'>Construct a design to make triangle tests.<\/span><\/div>, <div class='tooltip'>triangle.pair.test<span class='tooltiptext'>Make a Triangle test for two products.<\/span><\/div>, <div class='tooltip'>triangle.test<span class='tooltiptext'>Make a Triangle test for a set of products.<\/span><\/div>, <div class='tooltip'>WilliamsDesign<span class='tooltiptext'>Construct a Williams design<\/span><\/div>, <div class='tooltip'>WordCountAna<span class='tooltiptext'>Sensory methods as labelled sorting task, check-all-that-apply (CATA), ultra-flash profiling (UFP) and open-ended questions can be used to collect free-text descriptions of products through word-count based methods. A data frame with rows-products and columns-panellists is considered for the analysis. WordCountAna performs a multiple factor analysis for contingency tables keeping all the information in the comparison of the products. The identification of the consensual words which have the same meaning for most of the panellists eases the interpretation of the word-count based methods and solves the problems arising from the large diversity of vocabulary as the different meanings possibly associated to a same word. A test, based on resampling techniques, allows for assessing the significance of the consensus.<\/span><\/div>","<div class='tooltip'>sFFLHD<span class='tooltiptext'>sFFLHD object that gives a batch of points at a time.<\/span><\/div>, <div class='tooltip'>split_matrix<span class='tooltiptext'>Split a matrix by rows, based on either the number of rows per groupor number of splits.<\/span><\/div>","<div class='tooltip'>close_material_modal<span class='tooltiptext'>Close a material modal programmatically (server side).<\/span><\/div>, <div class='tooltip'>material_button<span class='tooltiptext'>Build a shinymaterial button. The initial value is zero, and increases by one on each press.<\/span><\/div>, <div class='tooltip'>material_card<span class='tooltiptext'>UI content can be placed in cards to organize items on a page.<\/span><\/div>, <div class='tooltip'>material_checkbox<span class='tooltiptext'>Build a shinymaterial checkbox. The value is a boolean (TRUE if checked, FALSE if not checked).<\/span><\/div>, <div class='tooltip'>material_column<span class='tooltiptext'>UI content can be placed in columns to organize items on a page.<\/span><\/div>, <div class='tooltip'>material_date_picker<span class='tooltiptext'>Build a shinymaterial date picker.<\/span><\/div>, <div class='tooltip'>material_depth<span class='tooltiptext'>Give a UI element the perception of depth by creating a shadow.<\/span><\/div>, <div class='tooltip'>material_dropdown<span class='tooltiptext'>Build a shinymaterial dropdown.<\/span><\/div>, <div class='tooltip'>material_file_input<span class='tooltiptext'>Build a shinymaterial file input.<\/span><\/div>, <div class='tooltip'>material_floating_button<span class='tooltiptext'>Build a shinymaterial floating button. The initial value is zero, and increases by one on each press.<\/span><\/div>, <div class='tooltip'>material_input<span class='tooltiptext'>Build a shinymaterial input.<\/span><\/div>, <div class='tooltip'>material_modal<span class='tooltiptext'>Put any UI object inside of a modal. The modal will open when the button is pressed.<\/span><\/div>, <div class='tooltip'>material_number_box<span class='tooltiptext'>Build a shinymaterial number box.<\/span><\/div>, <div class='tooltip'>material_page<span class='tooltiptext'>Build a shinymaterial page.<\/span><\/div>, <div class='tooltip'>material_parallax<span class='tooltiptext'>Use this function to create a parallax effect in your application.<\/span><\/div>, <div class='tooltip'>material_password_box<span class='tooltiptext'>Build a shinymaterial password box.<\/span><\/div>, <div class='tooltip'>material_radio_button<span class='tooltiptext'>Build a shinymaterial radio button.<\/span><\/div>, <div class='tooltip'>material_row<span class='tooltiptext'>UI content can be placed in a row to organize items on a page.<\/span><\/div>, <div class='tooltip'>material_side_nav<span class='tooltiptext'>UI content can be placed in side-nav.<\/span><\/div>, <div class='tooltip'>material_side_nav_tab_content<span class='tooltiptext'>Use this function to place UI content within a specific side-nav tab.<\/span><\/div>, <div class='tooltip'>material_side_nav_tabs<span class='tooltiptext'>Use this function to create side-nav tabs in your application.<\/span><\/div>, <div class='tooltip'>material_slider<span class='tooltiptext'>Build a shinymaterial slider.<\/span><\/div>, <div class='tooltip'>material_spinner_hide<span class='tooltiptext'>Display a preloader (spinner) while server is busy.<\/span><\/div>, <div class='tooltip'>material_spinner_show<span class='tooltiptext'>Display a preloader (spinner) while server is busy.<\/span><\/div>, <div class='tooltip'>material_switch<span class='tooltiptext'>Build a shinymaterial switch. The value is a boolean (TRUE if 'on', FALSE if 'off').<\/span><\/div>, <div class='tooltip'>material_tab_content<span class='tooltiptext'>Use this function to place UI content within a specific tab.<\/span><\/div>, <div class='tooltip'>material_tabs<span class='tooltiptext'>Use this function to create tabs in your application.<\/span><\/div>, <div class='tooltip'>material_text_box<span class='tooltiptext'>Build a shinymaterial text box.<\/span><\/div>, <div class='tooltip'>open_material_modal<span class='tooltiptext'>Open a material modal programmatically (server side).<\/span><\/div>, <div class='tooltip'>render_material_from_server<span class='tooltiptext'>This function is used within a shiny::renderUI(). The corresponding output is referenced using shiny::uiOutput().<\/span><\/div>, <div class='tooltip'>side_nav_tabs_info<span class='tooltiptext'>Query information about the side-nav tabs (e.g. which tab is active).<\/span><\/div>, <div class='tooltip'>update_material_button<span class='tooltiptext'>Change the value text, icon of a material_button on the client. Allow to disable the button and then enable.<\/span><\/div>, <div class='tooltip'>update_material_checkbox<span class='tooltiptext'>Change the value of a material_checkbox on the client.<\/span><\/div>, <div class='tooltip'>update_material_date_picker<span class='tooltiptext'>Change the value of a material_date_picker on the client.<\/span><\/div>, <div class='tooltip'>update_material_dropdown<span class='tooltiptext'>Change the value of a material_dropdown on the client.<\/span><\/div>, <div class='tooltip'>update_material_number_box<span class='tooltiptext'>Change the value of a material_number_box on the client.<\/span><\/div>, <div class='tooltip'>update_material_password_box<span class='tooltiptext'>Change the value of a material_password_box on the client.<\/span><\/div>, <div class='tooltip'>update_material_radio_button<span class='tooltiptext'>Change the value of a material_radio_button on the client.<\/span><\/div>, <div class='tooltip'>update_material_slider<span class='tooltiptext'>Change the value of a material_slider on the client.<\/span><\/div>, <div class='tooltip'>update_material_switch<span class='tooltiptext'>Change the value of a material_switch on the client.<\/span><\/div>, <div class='tooltip'>update_material_text_box<span class='tooltiptext'>Change the value of a material_text_box on the client.<\/span><\/div>","<div class='tooltip'>beta_builder<span class='tooltiptext'>Specify the locations in the lattice/image that have non-zero parameters as well as the values for those parameters, and the function creates the parameter vector that matches the correct locations in the design matrix.<\/span><\/div>, <div class='tooltip'>chol_s2Dp<span class='tooltiptext'>The function first builds a correlation matrix with correlation.builder, convertsthat matrix to a covariance matrix if necessary, and then takes the Cholesky decompositionof the matrix using either base R or the R package spam. Note that spamis particularly effective when the matrix is sparse.<\/span><\/div>, <div class='tooltip'>corr_fun<span class='tooltiptext'>This is primarily for use within correlation builder, and may be altered/expanded tohandle more complicated correlation functions if desired.<\/span><\/div>, <div class='tooltip'>correlation_builder<span class='tooltiptext'>This function \"builds\" a correlation matrix based on user specifications.<\/span><\/div>, <div class='tooltip'>generate_grid<span class='tooltiptext'>Input the limits of a 2D space and the desired image resolution, thenthe function outputs the appropriate grid/lattice coordinates.<\/span><\/div>, <div class='tooltip'>inf_2D_image<span class='tooltiptext'>Provide graphics for spatial extent of predictor parameters, rejectionsand/or the truth/falsity of the rejections.<\/span><\/div>, <div class='tooltip'>neighbors_by_dist<span class='tooltiptext'>Determine and store neighbors by Euclidean Distance Constraints<\/span><\/div>, <div class='tooltip'>precision_builder<span class='tooltiptext'>This function constructs the precision matrix for a Conditional Autoregression (CAR).<\/span><\/div>, <div class='tooltip'>proximity_builder<span class='tooltiptext'>Generates a proximity matrix where non-zero entries are the weights associated with neighbors, andzero entries are not neighbors.<\/span><\/div>, <div class='tooltip'>sample_FP_Power<span class='tooltiptext'>This function calculates sample FDR, FWER, and Power for large numbers of predictors,given a vector of \"true\" parameter values and a vector of associated rejections.In the case that more than 1 predictor has a \"true\" non-zero parameter, then Poweris defined as the proportion/percentage of those \"true\" parameters identified.<\/span><\/div>, <div class='tooltip'>sim_MVN_X<span class='tooltiptext'>Takes N draws from a Multivariate Normal (MVN) distribution using eitherbase R or the R package spam. This function requires the Choleskydecomposition of the desired covariance matrix.<\/span><\/div>, <div class='tooltip'>sim_Y_Binary_X<span class='tooltiptext'>N spatially dependent binary design vectors are simulated using sim2D_binarymap.These design vectors are used to then simulate scalar outcomes that haveone of Gaussian, Binomial, or Poisson distributions.<\/span><\/div>, <div class='tooltip'>sim_Y_MVN_X<span class='tooltiptext'>N spatially correlated design vectors are simulated from an MVN. Thesedesign vectors are used to then simulate scalar outcomes that haveone of Gaussian, Binomial, or Poisson distributions.<\/span><\/div>, <div class='tooltip'>sim2D_binarymap<span class='tooltiptext'>Use a Homogenous Poisson Process to generate random \"events\", a uniform distributionto generate circles of random radii about the events, and take the union to obtaina random set. This is mapped onto a lattice to obtain a binary map.<\/span><\/div>, <div class='tooltip'>sim2D_RandSet_HPPP<span class='tooltiptext'>A random set is generated by using a Poisson process in 2D space to choose'event' locations, about which a circle of random radius is 'drawn'. The unionof the circles defines ultimately defines the set.<\/span><\/div>, <div class='tooltip'>within_area<span class='tooltiptext'>Determine whether locations in the image/lattice (from generate.grid) are withinor without the union of a random set generated by sim2D_HPPP_coords(). If theEuclidean distance between a lattice location and any 'event' is less than the radiusabout the 'event', then the location is said to be within the random set. Otherwise,it is without the random set.<\/span><\/div>","<div class='tooltip'>add_missing<span class='tooltiptext'>Given an input vector, replace elements of this vector with missing values according to some scheme.Default method replaces input values with a MCAR scheme (where on average 10% of the values will bereplaced with NAs). MAR and MNAR are supported by replacing the default FUN argument.<\/span><\/div>, <div class='tooltip'>aggregate_simulations<span class='tooltiptext'>This function grabs all .rds files in the working directory and aggregates them into a singledata.frame object or combines all the saved results directories and combines them into one.This is generally useful when results are run piecewise on one node or run independently acrossdifferent nodes/computers which are not on the same network.<\/span><\/div>, <div class='tooltip'>Attach<span class='tooltiptext'>This function accepts the condition object used to indicate the design conditionsand makes the variable names available in the environment from which it is called. Thisis useful primarily as a convenience function when you prefer to call the variable namesin condition directly rather than indexing with condition$sample_size orwith(condition, sample_size), for example.<\/span><\/div>, <div class='tooltip'>bias<span class='tooltiptext'>Computes the (relative) bias of a sample estimate from the parameter value.Accepts estimate and parameter values, as well as estimate values which are in deviation form.If relative bias is requested the estimate and parameter inputs are both required.<\/span><\/div>, <div class='tooltip'>boot_predict<span class='tooltiptext'>This function computes bootstrap mean-square error estimates to approximate the sampling behaviorof the meta-statistics in SimDesign's summarise functions. A single design condition issupplied, and a simulation with max(Rstar) replications is performed whereby thegenerate-analyse results are collected. After obtaining these replication values, thereplications are further drawn from (with replacement) using the differing sizes in Rstarto approximate the bootstrap MSE behavior given different replication sizes. Finally, given thesebootstrap estimates linear regression models are fitted using the predictor termone_sqrtR = 1 / sqrt(Rstar) to allow extrapolation to replication sizes not observed inRstar. For more information about the method and subsequent bootstrap MSE plots,refer to Koehler, Brown, and Haneuse (2009).<\/span><\/div>, <div class='tooltip'>CC<span class='tooltiptext'>Computes the congruence coefficient, also known as an \"unadjusted\" correlationor Tucker's congruence coefficient.<\/span><\/div>, <div class='tooltip'>createDesign<span class='tooltiptext'>Create a partially or fully-crossed data object reflecting the uniquesimulation design conditions. Each row of the returned object representsa unique simulation condition, and each column represents the named factorvariables under study.<\/span><\/div>, <div class='tooltip'>ECR<span class='tooltiptext'>Computes the detection rate for determining empirical coverage rates given a set of estimatedconfidence intervals. Note that using 1 - ECR(CIs, parameter) will provide the empiricaldetection rate. Also supports computing the average width of the CIs, which may be useful when comparingthe efficiency of CI estimators.<\/span><\/div>, <div class='tooltip'>EDR<span class='tooltiptext'>Computes the detection rate for determining empirical Type I error and power rates using informationfrom p-values.<\/span><\/div>, <div class='tooltip'>IRMSE<span class='tooltiptext'>Computes the average/cumulative deviation given two continuous functions and an optionalfunction representing the probability density function. Only one-dimensional integrationis supported.<\/span><\/div>, <div class='tooltip'>MAE<span class='tooltiptext'>Computes the average absolute deviation of a sample estimate from the parameter value.Accepts estimate and parameter values, as well as estimate values which are in deviation form.<\/span><\/div>, <div class='tooltip'>MSRSE<span class='tooltiptext'>The mean-square relative standard error (MSRSE) compares standard errorestimates to the standard deviation of the respectiveparameter estimates. Values close to 1 indicate that the behavior of the standard errorsclosely matched the sampling variability of the parameter estimates.<\/span><\/div>, <div class='tooltip'>quiet<span class='tooltiptext'>This function is used to suppress information printed from external functionsthat make internal use of link{message} and cat, whichprovide information in interactive R sessions. For simulations, the sessionis not interactive, and therefore this type of output should be suppressed.For similar behavior for suppressing warning messages seesuppressWarnings, though use this function carefully as somewarnings can be meaningful and unexpected.<\/span><\/div>, <div class='tooltip'>RAB<span class='tooltiptext'>Computes the relative absolute bias given the bias estimates for multiple estimators.<\/span><\/div>, <div class='tooltip'>RD<span class='tooltiptext'>Computes the relative difference statistic of the form (est - pop)/ pop, whichis equivalent to the form est/pop - 1. If matrices are supplied thenan equivalent matrix variant will be used of the form(est - pop) * solve(pop). Values closer to 0 indicate betterrelative parameter recovery. Note that for single variable inputs this is equivalent tobias(..., type = 'relative').<\/span><\/div>, <div class='tooltip'>RE<span class='tooltiptext'>Computes the relative efficiency given the RMSE (default) or MSE values for multiple estimators.<\/span><\/div>, <div class='tooltip'>rejectionSampling<span class='tooltiptext'>This function supports the rejection sampling (i.e., accept-reject) approach to drawing valuesfrom seemingly difficult, and potentially non-normed, probability density functions bysampling values from a more manageable proxy distribution. This function isoptimized to work efficiently when the defined functions are vectorized; otherwise,the accept-reject algorithm will loop over candidate sample-draws in isolation.<\/span><\/div>, <div class='tooltip'>reSummarise<span class='tooltiptext'>When runSimulation() uses the option save_results = TRUEthe R replication results from the Generate-Analyse functions arestored to the hard drive. As such, additional summarise componentsmay be required at a later time, whereby the respective .rds filesmust be read back into R to be summarised. This function performsthe reading of these files, application of a provided summarise function,and final collection of the respective results.<\/span><\/div>, <div class='tooltip'>rHeadrick<span class='tooltiptext'>Generate multivariate non-normal distributions using the fifth-order polynomial method described by Headrick (2002).<\/span><\/div>, <div class='tooltip'>rint<span class='tooltiptext'>Efficiently generate positive and negative integer values with (default) or without replacement.This function is mainly a wrapper to the sample.int function (which itself is muchmore efficient integer sampler than the more general sample), however is intendedto work with both positive and negative integer ranges since sample.int only returnspositive integer values that must begin at 1L.<\/span><\/div>, <div class='tooltip'>rinvWishart<span class='tooltiptext'>Function generates data in the form of symmetric matrices from the inverseWishart distribution given a covariance matrix and degrees of freedom.<\/span><\/div>, <div class='tooltip'>rmgh<span class='tooltiptext'>Generate non-normal distributions using the multivariate g-and-h distribution. Can be used togenerate several different classes of univariate and multivariate distributions.<\/span><\/div>, <div class='tooltip'>RMSD<span class='tooltiptext'>Computes the average deviation (root mean square error; also known as the root mean square deviation)of a sample estimate from the parameter value. Accepts estimate and parameter values,as well as estimate values which are in deviation form.<\/span><\/div>, <div class='tooltip'>RMSE<span class='tooltiptext'>Computes the average deviation (root mean square error; also known as the root mean square deviation)of a sample estimate from the parameter value. Accepts estimate and parameter values,as well as estimate values which are in deviation form.<\/span><\/div>, <div class='tooltip'>rmvnorm<span class='tooltiptext'>Function generates data from the multivariate normal distribution given some mean vector and/orcovariance matrix.<\/span><\/div>, <div class='tooltip'>rmvt<span class='tooltiptext'>Function generates data from the multivariate t distribution given a covariance matrix,non-centrality parameter (or mode), and degrees of freedom.<\/span><\/div>, <div class='tooltip'>rtruncate<span class='tooltiptext'>Function generates data given a supplied random number generating function thatare constructed to fall within a particular range. Sampled values outside this rangeare discarded and re-sampled until the desired criteria has been met.<\/span><\/div>, <div class='tooltip'>runSimulation<span class='tooltiptext'>This function runs a Monte Carlo simulation study given a set of predefined simulation functions,design conditions, and number of replications. Results can be saved as temporary files in case of interruptionsand may be restored by re-running runSimulation, provided that the respective tempfile can be found in the working directory. runSimulation supports paralleland cluster computing, global and local debugging, error handling (including fail-safestopping when functions fail too often, even across nodes), provides bootstrap estimates of thesampling variability (optional), and automatic tracking of error and warning messages and their associated.Random.seed states.For convenience, all functions available in the R work-space are exported across all computational nodesso that they are more easily accessible (however, other R objects are not, and thereforemust be passed to the fixed_objects input to become available across nodes).For an in-depth tutorial of the package please refer to Chalmers and Adkins (2020;\\Sexpr[results=rd,stage=build]{tools:::Rd_expr_doi(\"#1\")}10.20982/tqmp.16.4.p248textdoi: 10.20982/tqmp.16.4.p248 (URL: https://doi.org/10.20982/tqmp.16.4.p248)latexdoi:\\nobreakspace{}https://doi.org/10.20982/tqmp.16.4.p24810.20982\\slash{}tqmp.16.4.p248doi: https://doi.org/10.20982/tqmp.16.4.p24810.20982/tqmp.16.4.p248).For an earlier didactic presentation of the package refer to Sigal and Chalmers(2016; \\Sexpr[results=rd,stage=build]{tools:::Rd_expr_doi(\"#1\")}10.1080/10691898.2016.1246953textdoi: 10.1080/10691898.2016.1246953 (URL: https://doi.org/10.1080/10691898.2016.1246953)latexdoi:\\nobreakspace{}https://doi.org/10.1080/10691898.2016.124695310.1080\\slash{}10691898.2016.1246953doi: https://doi.org/10.1080/10691898.2016.124695310.1080/10691898.2016.1246953). Finally, see the associatedwiki on Github (https://github.com/philchalmers/SimDesign/wiki)for tutorial material, examples, and applications of SimDesign to real-world simulation experiments.<\/span><\/div>, <div class='tooltip'>rValeMaurelli<span class='tooltiptext'>Generate multivariate non-normal distributions using the third-order polynomial method describedby Vale & Maurelli (1983). If only a single variable is generated then this functionis equivalent to the method described by Fleishman (1978).<\/span><\/div>, <div class='tooltip'>Serlin2000<span class='tooltiptext'>Hypothesis test to determine whether an observed empirical detection rate,coupled with a given robustness interval, statistically differs from thepopulation value. Uses the methods described by Serlin (2000) as well togenerate critical values (similar to confidence intervals, but define a fixedwindow of robustness). Critical values may be computed without performing the simulationexperiment (hence, can be obtained a priori).<\/span><\/div>, <div class='tooltip'>SimAnova<span class='tooltiptext'>Given the results from a simulation with runSimulation form an ANOVA table (withoutp-values) with effect sizes based on the eta-squared statistic. These results provide approximateindications of observable simulation effects, therefore these ANOVA-based results are generally usefulas exploratory rather than inferential tools.<\/span><\/div>, <div class='tooltip'>SimClean<span class='tooltiptext'>This function is mainly used in pilot studies where results and datasets have been temporarily savedby runSimulation but should be removed before beginning the fullMonte Carlo simulation (e.g., remove files and folders which contained bugs/biased results).<\/span><\/div>, <div class='tooltip'>SimExtract<span class='tooltiptext'>Function used to extract any error or warnings messages, the seeds associatedwith any error or warning messages, and any analysis results that were stored in thefinal simulation object.<\/span><\/div>, <div class='tooltip'>SimFunctions<span class='tooltiptext'>This function prints template versions of the required Design and Generate-Analyse-Summarise functionsfor SimDesign to run simulations. Templated output comes complete with the correct inputs,class of outputs, and optional comments to help with the initial definitions.Use this at the start of your Monte Carlo simulation study. Followingthe definition of the SimDesign template file please refer to detailed the informationin runSimulation for how to edit this template to make a working simulation study.<\/span><\/div>, <div class='tooltip'>SimResults<span class='tooltiptext'>If runSimulation was passed the flag save_results = TRUE then therow results corresponding to the design object will be stored to a suitablesub-directory as individual .rds files. While users could use readRDS directlyto read these files in themselves, this convenience function will read the desired rows inautomatically given the returned objectfrom the simulation. Can be used to read in 1 or more .rds files at once (if more than 1 fileis read in then the result will be stored in a list).<\/span><\/div>, <div class='tooltip'>SimShiny<span class='tooltiptext'>This function generates suitable stand-alone code from the shiny package to create simpleweb-interfaces for performing single condition Monte Carlo simulations. The templategenerated is relatively minimalistic, but allows the user to quickly and easilyedit the saved files to customize the associated shiny elements as they see fit.<\/span><\/div>","<div class='tooltip'>computePower<span class='tooltiptext'>Computes the power in an arbitrary SMART design withthe goal of identifying optimal embedded dynamic treatment regime (EDTR).The power is the probability of excluding from the set of best EDTRsall EDTRs which are inferior to the best EDTR by min_Delta or more.<\/span><\/div>, <div class='tooltip'>computePowerBySampleSize<span class='tooltiptext'>Computes the power overa grid of sample size values.<\/span><\/div>, <div class='tooltip'>computeSampleSize<span class='tooltiptext'>Computes the necessary sample size to enrollin an arbitrary SMART design for a specified power with the goal of determining optimalembedded dynamic treatment regime (EDTR). The power is theprobability of excluding from the set of best EDTRs all EDTRs inferior to the bestby min_Delta or more.<\/span><\/div>, <div class='tooltip'>plotPowerByN<span class='tooltiptext'>Plots the power over a grid of sample sizes.<\/span><\/div>","<div class='tooltip'>spIndexTrack<span class='tooltiptext'>Computes the weights of assets (relative capital allocation) for a sparse approximation of a financial index.<\/span><\/div>","<div class='tooltip'>Est.Inpar<span class='tooltiptext'>Computes initial values of the model parameters.<\/span><\/div>, <div class='tooltip'>FtSmlrmCMCM<span class='tooltiptext'>The function generates data from a logistic regression model. The data obtained contain: an outcome variable, the mother and child genotype coded as the number of minor allele and the environmental factors. For simulation of each environmental variable, the user can specify the coefficients of linear dependency between the mother genotype and the environmental factors.<\/span><\/div>, <div class='tooltip'>SeltcEch<span class='tooltiptext'>The function draws randomly n1+n0 individuals among N1 + N0 individuals with N0 > n0 representing the numbers of non-cases and N1 > n1 representing the number of cases.<\/span><\/div>, <div class='tooltip'>Spmlficmcm<span class='tooltiptext'>The function builds the nonlinear system from the data, solves the system and assesses the effect of each factor of the model, computes the variance - covariance matrix and deduces from it the standard deviations of each factor.<\/span><\/div>","<div class='tooltip'>adjwgt<span class='tooltiptext'>Purpose of this function is to adjust initial survey design weights when implementation results in use of oversample sites or when it is desired to have final weights sum to known frame size. Adjusted weights are equal to initial weight * framesize/sum(initial weights). The adjustment is done separately for each category specified in wtcat.<\/span><\/div>, <div class='tooltip'>albersgeod<span class='tooltiptext'>Convert x-coordinates and y-coordinates given in Albers projection tolatitude and longitude in Clarke1866, GRS80 or WGS84 spheriod withspecified parameters.<\/span><\/div>, <div class='tooltip'>ash1.wgt<span class='tooltiptext'>Calculate the average shifted histogram estimate of a density based on data from a survey design with weights.<\/span><\/div>, <div class='tooltip'>attrisk.analysis<span class='tooltiptext'>This function organizes input and output for attributable risk analysis ofcategorical data generated by a probability survey.<\/span><\/div>, <div class='tooltip'>attrisk.est<span class='tooltiptext'>This function calculates the attributable risk estimate for a 2x2 table ofcell counts defined by a categorical response variable and a categoricalexplanatory (stressor) variable for an unequal probability design (Van Sickleand Paulsen, 2008). The attributable risk of the stressor variable is thepercent reduction in the first level of the response variable that wouldresult from elimination of the stressor variable. Cell totals are estimatedusing the Horvitz-Thompson estimator. The standard error of the log of theattributable risk estimate and confidence limits for the estimate also arecalculated. The standard error is calculated using a first-order Taylorseries linearization (Sarndal et al, 1992).<\/span><\/div>, <div class='tooltip'>attrisk.var<span class='tooltiptext'>This function calculates the variance-covariance estimate for the cell totalsused to calculate the attributable risk estimate. Either the simple randomsampling (SRS) variance estimator or the local mean variance estimator iscalculated, which is subject to user control. The SRS variance estimatoruses the independent random sample approximation to calculate joint inclusionprobabilities. The function can accomodate single-stage and two-stagesamples.<\/span><\/div>, <div class='tooltip'>cat.analysis<span class='tooltiptext'>This function organizes input and output for analysis of categorical datagenerated by a probability survey. Input can be either an object of classspsurvey.analysis (see the documentation for function spsurvey.analysis)or through use of the other arguments to this function.<\/span><\/div>, <div class='tooltip'>category.est<span class='tooltiptext'>This function estimates proportion (expressed as percent) and size of aresource in each of a set of categories and can also be used to estimateproportion and size for site status categories. Upper and lower confidencebounds also are estimated. Proportion estimates are calculated using theHorvitz-Thompson ratio estimator, i.e., the ratio of two Horvitz-Thompsonestimators. The numerator of the ratio estimates the size of the category.The denominator of the ratio estimates the size of the resource. Varianceestimates for the proportion estimates are calculated using either the localmean variance estimator or the simple random sampling (SRS) varianceestimator. The choice of variance estimator is subject to user control. Thelocal mean variance estimator requires the x-coordinate and the y-coordinateof each site. The SRS variance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. Confidence boundsare calculated using a Normal distribution multiplier. For a finite resourcesize is the number of units in the resource. For an extensive resource sizeis the measure (extent) of the resource, i.e., length, area, or volume. Sizeestimates are calculated using the Horvitz-Thompson estimator. Varianceestimates for the size estimates are calculated using either the local meanvariance estimator or the SRS variance estimator. The function canaccommodate a stratified sample. For a stratified sample, separate estimatesand standard errors are calculated for each stratum, which are used toproduce estimates and standard errors for all strata combined. Strata thatcontain a single value are removed. For a stratified sample, when either thesize of the resource or the sum of the size-weights for the resource isprovided for each stratum, those values are used as stratum weights forcalculating the estimates and standard errors for all strata combined. Inaddition, when either of those known values is provided for each stratum,size estimates are obtained by multiplying the proportion estimate, i.e., theHorvitz- Thompson ratio estimator, by the known value for the stratum. For astratified sample when neither the size of the resource nor the sum of thesize-weights of the resource is provided for each stratum, estimated valuesare used as stratum weights for calculating the estimates and standard errorsfor all strata combined. The function can accommodate single-stage andtwo-stage samples for both stratified and unstratified sampling designs.Finite population and continuous population correction factors can beutilized in variance estimation. The function checks for compatibility ofinput values and removes missing values.<\/span><\/div>, <div class='tooltip'>catvar.prop<span class='tooltiptext'>This function calculates variance estimates of the estimated proportion ineach of a set of categories. Either the simple random sampling (SRS)variance estimator or the local mean variance estimator is calculated, whichis subject to user control. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities. Thefunction can accomodate single-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>catvar.size<span class='tooltiptext'>This function calculates variance estimates of the estimated size in each ofa set of categories. Either the simple random sampling (SRS) varianceestimator or the local mean variance estimator is calculated, which issubject to user control. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities. Thefunction can accomodate single-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>cdf.decon<span class='tooltiptext'>This function calculates an estimate of the deconvoluted cumulative distribution function (CDF) for the proportion (expressed as percent) and the total of a response variable, where the response variable may be defined for either a finite or an extensive resource. Optionally, for a finite resource, the size-weighted CDF can be calculated. In addition the standard error of the estimated CDF and confidence bounds are calculated. The simulation extrapolation deconvolution method (Stefanski and Bay, 1996) is used to deconvolute measurement error variance from the response. The user can supply the set of values at which the CDF is estimated. For the CDF of a proportion, the Horvitz-Thompson ratio estimator, i.e., the ratio of two Horvitz-Thompson estimators, is used to calculate the CDF estimate. For the CDF of a total, the user can supply the known size of the resource or the known sum of the size-weights of the resource, as appropriate. For the CDF of a total when either the size of the resource or the sum of the size- weights of the resource is provided, the classic ratio estimator is used to calculate the CDF estimate, where that estimator is the product of the known value and the Horvitz-Thompson ratio estimator. For the CDF of a total when neither the size of the resource nor the sum of the size-weights of the resource is provided, the Horvitz-Thompson estimator is used to calculate the CDF estimate. Variance estimates for the estimated CDF are calculated using either the local mean variance estimator or the simple random sampling (SRS) variance estimator. The choice of variance estimator is subject to user control. The local mean variance estimator requires the x-coordinate and the y-coordinate of each site. The SRS variance estimator uses the independent random sample approximation to calculate joint inclusion probabilities. Confidence bounds are calculated using a Normal distribution multiplier. In addition the function uses the estimated CDF to calculate percentile estimates. Estimated confidence bounds for the percentile estimates are calculated. The user can supply the set of values for which percentiles estimates are desired. Optionally, the user can use the default set of percentiles. The function can accommodate a stratified sample. For a stratified sample, separate estimates and standard errors are calculated for each stratum, which are used to produce estimates and standard errors for all strata combined. Strata that contain a single value are removed. For a stratified sample, when either the size of the resource or the sum of the size-weights of the resource is provided for each stratum, those values are used as stratum weights for calculating the estimates and standard errors for all strata combined. For a stratified sample when neither the size of the resource nor the sum of the size-weights of the resource is provided for each stratum, estimated values are used as stratum weights for calculating the estimates and standard errors for all strata combined. The function can accommodate single-stage and two-stage samples for both stratified and unstratified sampling designs. Finite population and continuous population correction factors can be utilized in variance estimation. The function checks for compatibility of input values and removes missing values.<\/span><\/div>, <div class='tooltip'>cdf.est<span class='tooltiptext'>This function calculates an estimate of the cumulative distribution function(CDF) for the proportion (expressed as percent) and the total of a responsevariable, where the response variable may be defined for either a finite oran extensive resource. Optionally, for a finite resource, the size- weightedCDF can be calculated. In addition the standard error of the estimated CDFand confidence bounds are calculated. The user can supply the set of valuesat which the CDF is estimated. For the CDF of a proportion, theHorvitz-Thompson ratio estimator, i.e., the ratio of two Horvitz-Thompsonestimators, is used to calculate the CDF estimate. For the CDF of a total,the user can supply the known size of the resource or the known sum of thesize-weights of the resource, as appropriate. For the CDF of a total wheneither the size of the resource or the sum of the size-weights of theresource is provided, the classic ratio estimator is used to calculate theCDF estimate, where that estimator is the product of the known value and theHorvitz-Thompson ratio estimator. For the CDF of a total when neither thesize of the resource nor the sum of the size-weights of the resource isprovided, the Horvitz-Thompson estimator is used to calculate the CDFestimate. Variance estimates for the estimated CDF are calculated usingeither the local mean variance estimator or the simple random sampling (SRS)variance estimator. The choice of variance estimator is subject to usercontrol. The local mean variance estimator requires the x-coordinate and they-coordinate of each site. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities.Confidence bounds are calculated using a Normal distribution multiplier. Inaddition the function uses the estimated CDF to calculate percentileestimates. Estimated confidence bounds for the percentile estimates arecalculated. The user can supply the set of values for which percentilesestimates are desired. Optionally, the user can use the default set ofpercentiles. The function can accommodate a stratified sample. For astratified sample, separate estimates and standard errors are calculated foreach stratum, which are used to produce estimates and standard errors for allstrata combined. Strata that contain a single value are removed. For astratified sample, when either the size of the resource or the sum of thesize-weights of the resource is provided for each stratum, those values areused as stratum weights for calculating the estimates and standard errors forall strata combined. For a stratified sample when neither the size of theresource nor the sum of the size-weights of the resource is provided for eachstratum, estimated values are used as stratum weights for calculating theestimates and standard errors for all strata combined. The function canaccommodate single-stage and two-stage samples for both stratified andunstratified sampling designs. Finite population and continuous populationcorrection factors can be utilized in variance estimation. The functionchecks for compatibility of input values and removes missing values.<\/span><\/div>, <div class='tooltip'>cdf.nresp<span class='tooltiptext'>This function calculates the number of response values less than or equal toeach of the set of values at which the cumulative distribution function (CDF)is estimated.<\/span><\/div>, <div class='tooltip'>cdf.plot<span class='tooltiptext'>This function creates a CDF plot. Input data for the plots is provided by adata frame utilizing the same structure as the data frame named \"CDF\" that isincluded in the output object produced by function cont.analysis, but thedata frame includes only the values for a single CDF. Confidence limits forthe CDF also are plotted.<\/span><\/div>, <div class='tooltip'>cdf.prop<span class='tooltiptext'>This function calculates an estimate of the cumulative distribution function(CDF) for the proportion of a finite or an extensive resource. The set ofvalues at which the CDF is estimated is supplied to the function. TheHorvitz-Thompson ratio estimator, i.e., the ratio of two Horvitz-Thompsonestimators, is used to calculate the estimate. The numerator of the ratioestimates the total of the resource equal to or less than a specified value.The denominator of the ratio estimates the size of the resource. For afinite resource size is the number of units in the resource. For anextensive resource size is the extent (measure) of the resource, i.e.,length, area, or volume. The function can accomodate single-stage andtwo-stage samples.<\/span><\/div>, <div class='tooltip'>cdf.size.prop<span class='tooltiptext'>This function calculates an estimate of the size-weighted cumulativedistribution function (CDF) for the proportion of a finite resource. The setof values at which the CDF is estimated is supplied to the function. TheHorvitz-Thompson ratio estimator, i.e., the ratio of two Horvitz-Thompsonestimators, is used to calculate the estimate. The numerator of the ratioestimates the size-weighted total of the resource equal to or less than aspecified value. The denominator of the ratio estimates the sum of thesize-weights for the resource. The function can accomodate single-stage andtwo-stage samples.<\/span><\/div>, <div class='tooltip'>cdf.size.total<span class='tooltiptext'>This function calculates an estimate of the size-weighted cumulativedistribution function (CDF) for the total of a finite resource. The set ofvalues at which the CDF is estimated is supplied to the function. If theknown sum of the size-weights of the resource is provided, the classic ratioestimator is used to calculate the estimate. That estimator is the product ofthe known sum of the size-weights of the resource and the Horvitz- Thompsonratio estimator, where the latter is the ratio of two Horvitz- Thompsonestimators. The numerator of the ratio estimates the size-weighted total ofthe resource equal to or less than a specified value. The denominator of theratio estimates the sum of the size-weights of the resource. If the knownsum of the size-weights of the resource is not provided, the Horvitz-Thompsonestimator of the size-weighted total of the resource equal to or less than aspecified value is used to calculate the estimate. The function canaccomodate single-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>cdf.test<span class='tooltiptext'>This function tests for differences between cumulative distribution functions(CDFs) generated by probability surveys. The function returns a variety oftest statistics along with their degrees of freedom and p values. Theinferential procedures divide the CDFs into a discrete set of intervals(classes) and then utilize procedures that have been developed for analysisof categorical data from probability surveys. The function calculates theWald, Rao-Scott first order corrected (mean eigenvalue corrected), andRao-Scott second order corrected (Satterthwaite corrected) test statistics.Both standard versions of the three statistics, which are distributed asChi-squared random variables, and alternate version of the statistics, whichare distributed as F random variables, are available. The default teststatistic is the F distribution version of the Wald statistic. The usersupplies the set of upper bounds that define the intervals (classes) intowhich the CDFs are divided (binned). The minimum number of classes is three.The Horvitz-Thompson ratio estimator, i.e., the ratio of two Horvitz-Thompson estimators, is used to calculate estimates of the class proportionsfor the CDFs. Variance estimates for the test statistics are calculatedusing either the local mean variance estimator or the simple random sampling(SRS) variance estimator. The choice of variance estimator is subject touser control. The SRS variance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. The function canaccommodate a stratified sample. For a stratified sample, separate classproportion estimates and associated covariance estimates are calculated foreach stratum, which are used to produce estimates for all strata combined.Strata that contain a single value are removed. For a stratified sample,when either the size of the resource or the sum of the size-weights of theresource is provided for each stratum, those values are used as stratumweights for calculating the estimates for all strata combined. For astratified sample when neither the size of the resource nor the sum of thesize-weights of the resource is provided for each stratum, estimated valuesare used as stratum weights for calculating the estimates for all stratacombined. The function can accommodate single-stage and two-stage samplesfor both stratified and unstratified sampling designs. Finite population andcontinuous population correction factors can be utilized in varianceestimation. The function checks for compatability of input values andremoves missing values.<\/span><\/div>, <div class='tooltip'>cdf.test.prop<span class='tooltiptext'>This function calculates an estimate of the population proportion in a setof intervals (classes). The set of values defining the upper bound of eachclass is supplied to the function. The Horvitz-Thompson ratio estimator,i.e., the ratio of two Horvitz-Thompson estimators, is used to calculate theestimate. The numerator of the ratio estimates the total of the resourcewithin a class. The denominator of the ratio estimates the size of theresource. For a finite resource size is the number of units in the resource.For an extensive resource size is the extent (measure) of the resource, i.e.,length, area, or volume. The function can accomodate single stage andtwo-stage samples.<\/span><\/div>, <div class='tooltip'>cdf.test.size.prop<span class='tooltiptext'>This function calculates a size-weighted estimate of the populationproportions in a set of intervals (classes). The set of values defining theupper bound of each class is supplied to the function. The Horvitz-Thompsonratio estimator, i.e., the ratio of two Horvitz-Thompson estimators, is usedto calculate the estimate. The numerator of the ratio estimates the total ofthe resource within a class. The denominator of the ratio estimates the sizeof the resource. For a finite resource size is the number of units in theresource. For an extensive resource size is the extent (measure) of theresource, i.e., length, area, or volume. The function can accomodate singlestage and two-stage samples.<\/span><\/div>, <div class='tooltip'>cdf.total<span class='tooltiptext'>This function calculates an estimate of the cumulative distribution function(CDF) for the total of a finite or an extensive resource. The set of valuesat which the CDF is estimated is supplied to the function. If the knownextent of the resource is provided, the classic ratio estimator is used tocalculate the estimate. That estimator is the product of the known extent ofthe resource and the Horvitz-Thompson ratio estimator, where the latter isthe ratio of two Horvitz-Thompson estimators. The numerator of the ratioestimates the total of the resource equal to or less than a specified value.The denominator of the ratio estimates the extent of the resource. If theknown extent of the resource is not provided, the Horvitz- Thompson estimatorof the total of the resource equal to or less than a specified value is usedto calculate the estimate. For a finite resource, size is the number of unitsin the resource. For an extensive resource, size is the measure of theresource, i.e., length, area, or volume. The function can accomodatesingle-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>cdfvar.prop<span class='tooltiptext'>This function calculates variance estimates of the estimated cumulativedistribution function (CDF) for the proportion of a finite or a continuousresource. The set of values at which the CDF is estimated is supplied to thefunction. Either the simple random sampling (SRS) variance estimator or thelocal mean variance estimator is calculated, which is subject to usercontrol. The SRS variance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>cdfvar.size.prop<span class='tooltiptext'>This function calculates variance estimates of the estimated size-weightedcumulative distribution function (CDF) for the proportion of a finiteresource. The set of values at which the CDF is estimated is supplied to thefunction. Either the simple random sampling (SRS) variance estimator or thelocal mean variance estimator is calculated, which is subject to usercontrol. The SRS variance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>cdfvar.size.total<span class='tooltiptext'>This function calculates variance estimates of the estimated size-weightedcumulative distribution function (CDF) for the total of a finite resource.The set of values at which the CDF is estimated is supplied to the function.Either the simple random sampling (SRS) variance estimator or the local meanvariance estimator is calculated, which is subject to user control. The SRSvariance estimator uses the independent random sample approximation tocalculate joint inclusion probabilities. The function can accomodatesingle-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>cdfvar.test<span class='tooltiptext'>This function calculates estimates of the variance-covariance matrix of thepopulation proportions in a set of intervals (classes). The set of valuesdefining upper bounds for the classes is supplied to the function. Either thesimple random sampling (SRS) variance estimator or the local mean varianceestimator is calculated, which is subject to user control. The simple randomsampling variance estimator uses the independent random sample approximationto calculate joint inclusion probabilities. The function can accomodatesingle-stage and two-stage samples. Finite population and continuouspopulation correction factors can be utilized in variance estimation.<\/span><\/div>, <div class='tooltip'>cdfvar.total<span class='tooltiptext'>This function calculates variance estimates of the estimated cumulativedistribution function (CDF) for the total of a finite or a continuousresource. The set of values at which the CDF is estimated is supplied to thefunction. Either the simple random sampling (SRS) variance estimator or thelocal mean variance estimator is calculated, which is subject to usercontrol. The SRS variance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>cell.wt<span class='tooltiptext'>Total Inclusion Probablity for Matrix of Cells<\/span><\/div>, <div class='tooltip'>cellWeight<span class='tooltiptext'>Calculates the total inclusion probability for each grid cell from a GRTSsurvey design.<\/span><\/div>, <div class='tooltip'>change.analysis<span class='tooltiptext'>This function organizes input and output for estimation of change between twoprobability surveys.<\/span><\/div>, <div class='tooltip'>change.est<span class='tooltiptext'>This function estimates change between two probability surveys. The functioncan accommodate both categorical and continuous response variables. For acategorical response variable, change is estimated by the difference incategory estimates for the two surveys, where a category estimate is theestimated proportion of values in a category. Note that a separate changeestimate is calculated for each category of a categorical response variable.For a continuous response variable, change can be estimated for the mean, themedian, or for both the mean and median. For a continuous response variableusing the mean, change is estimated by the difference in estimated meanvalues for the two surveys. For change estimates using the median, the firststep is to calculate an estimate of the median for the first survey. Theestimated median from the first survey is then used to define two categories:(1) values that are less than or equal to the estimated median and (2) valuesthat are greater than the estimated median. Once the categories are defined,change analysis for the median is identical to change analysis for acategorical variable, i.e., change is estimated by the difference in categoryestimates for the two surveys. In addition to change estimates, the standarderror of the change estimates and confidence bounds are calculated. Varianceestimates are calculated using either the local mean variance estimator orthe simple random sampling (SRS) variance estimator. The choice of varianceestimator is subject to user control. The local mean variance estimatorrequires the x-coordinate and y-coordinate of each site. The SRS varianceestimator uses the independent random sample approximation to calculate jointinclusion probabilities. Confidence bounds are calculated using a Normaldistribution multiplier. The function can accommodate a stratified sample.For a stratified sample, separate estimates and standard errors arecalculated for each stratum, which are used to produce estimates and standarderrors for all strata combined. Strata that contain a single value areremoved. For a stratified sample, when either the size of the resource orthe sum of the size-weights of the resource is provided for each stratum,those values are used as stratum weights for calculating the estimates andstandard errors for all strata combined. For a stratified sample whenneither the size of the resource nor the sum of the size-weights of theresource is provided for each stratum, estimated values are used as stratumweights for calculating the estimates and standard errors for all stratacombined. The function can accommodate single-stage and two-stage samplesfor both stratified and unstratified sampling designs. It is assumed thatboth surveys employ the same type of survey design. Finite population andcontinuous population correction factors can be utilized in varianceestimation. The function checks for compatibility of input values andremoves missing values.<\/span><\/div>, <div class='tooltip'>changevar.mean<span class='tooltiptext'>This function calculates estimates of the variance-covariance matrix of thepopulation proportions in a set of intervals (classes). The set of valuesdefining upper bounds for the classes is supplied to the function. Either thesimple random sampling (SRS) variance estimator or the local mean varianceestimator is calculated, which is subject to user control. The simple randomsampling variance estimator uses the independent random sample approximationto calculate joint inclusion probabilities. The function can accomodatesingle-stage and two-stage samples. Finite population and continuouspopulation correction factors can be utilized in variance estimation.<\/span><\/div>, <div class='tooltip'>changevar.prop<span class='tooltiptext'>This function uses the repeat visit sites for two probability surveys tocalculate either covariance or correlation estimates of estimated change inthe proportion in each of a set of categories. Covariance estimates arecalculated when the resivit sites have the same survey design weight in bothsurveys. Correlation estimates are calculated when the revisit sites do nothave the same weight in both surveys, in which case the sites are assignedequal weights. The revisitwgt argument controls whether covariance orcorrelation estimates are calculated. Either the simple random sampling(SRS) variance/covariance estimator or the local mean variance/covarianceestimator is calculated, which is subject to user control. The simple randomsampling variance/covariance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>changevar.size<span class='tooltiptext'>This function uses the repeat visit sites for two probability surveys tocalculate either covariance or correlation estimates of estimated change inresource size in each of a set of categories. Covariance estimates arecalculated when the resivit sites have the same survey design weight in bothsurveys. Correlation estimates are calculated when the revisit sites do nothave the same weight in both surveys, in which case the sites are assignedequal weights. The revisitwgt argument controls whether covariance orcorrelation estimates are calculated. Either the simple random sampling(SRS) variance/covariance estimator or the local mean variance/covarianceestimator is calculated, which is subject to user control. The simple randomsampling variance/covariance estimator uses the independent random sampleapproximation to calculate joint inclusion probabilities. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>constructAddr<span class='tooltiptext'>This function constructS the hierarchical addresses for a GRTS survey design.<\/span><\/div>, <div class='tooltip'>cont.analysis<span class='tooltiptext'>This function organizes input and output for analysis of continuous datagenerated by a probability survey. Input can be either an object of classspsurvey.analysis (see the documentation for function spsurvey.analysis) orthrough use of the other arguments to this function.<\/span><\/div>, <div class='tooltip'>cont.cdfplot<span class='tooltiptext'>This function creates CDF plots. Input data for the plots is provided by adata frame utilizing the same structure as the data frame named \"CDF\" that isincluded in the output object produced by function cont.analysis. Plots areproduced for every combination of Type of population, subpopulation withinType, and indicator. Output from the function is placed in a PDF file.<\/span><\/div>, <div class='tooltip'>cont.cdftest<span class='tooltiptext'>This function organizes input and output for conducting inference regardingcumulative distribution functions (CDFs) generated by a probability survey.Input can be either an object of class spsurvey.analysis (see thedocumentation for function spsurvey.analysis) or through use of the otherarguments to this function. For every response variable and every populationType, differences between CDFs are tested for every pair of subpopulationswithin a Type. The inferential procedures divide the CDFs into a discreteset of intervals (classes) and then utilize procedures that have beendeveloped for analysis of categorical data from probability surveys. Choicesfor inference are the Wald, Rao-Scott first order corrected (mean eigenvaluecorrected), and Rao-Scott second order corrected (Satterthwaite corrected)test statistics. Both standard versions of the three statistics, which aredistributed as Chi-squared random variables, and alternate version of thestatistics, which are distributed as F random variables, are available. Thedefault test statistic is the F distribution version of the Wald statistic.<\/span><\/div>, <div class='tooltip'>cov.panel.dsgn<span class='tooltiptext'>Covariance structure accounts for the panel design and the four variancecomponents: unit variation, period variation, unit by period interactionvariation and index (or residual) variation. The model incorporates unit,period, unit by period, and index variance components. It also includes aprovision for unit correlation and period autocorrelation.<\/span><\/div>, <div class='tooltip'>dcdf.prop<span class='tooltiptext'>This function calculates an estimate of the deconvoluted cumulativedistribution function (CDF) for the proportion of a discrete or an extensiveresource. The simulation extrapolation deconvolution method (Stefanski andBay, 1996) is use to deconvolute measurement error variance from theresponse. The Horvitz-Thompson ratio estimator, i.e., the ratio of twoHorvitz-Thompson estimators, is used to calculate the estimate. Thenumerator of the ratio estimates the total of the resource equal to or lessthan a specified value. The denominator of the ratio estimates the size ofthe resource. For a discrete resource size is the number of units in theresource. For an extensive resource size is the extent (measure) of theresource, i.e., length, area, or volume. The function can accomodatesingle-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>dcdf.size.prop<span class='tooltiptext'>This function calculates an estimate of the size-weighted, deconvolutedcumulative distribution function (CDF) for the proportion of a discreteresource. The simulation extrapolation deconvolution method (Stefanski andBay, 1996) is use to deconvolute measurement error variance from theresponse. The Horvitz-Thompson ratio estimator, i.e., the ratio of twoHorvitz-Thompson estimators, is used to calculate the estimate. Thenumerator of the ratio estimates the size-weighted total of the resourceequal to or less than a specified value. The denominator of the ratioestimates the sum of the size-weights for the resource. The function canaccomodate single-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>dcdf.size.total<span class='tooltiptext'>This function calculates an estimate of the size-weighted, deconvolutedcumulative distribution function (CDF) for the total of a discrete resource.The simulation extrapolation deconvolution method (Stefanski and Bay, 1996)is use to deconvolute measurement error variance from the response. If theknown sum of the size-weights of the resource is provided, the classic ratioestimator is used to calculate the estimate. That estimator is the product ofthe known sum of the size-weights of the resource and the Horvitz- Thompsonratio estimator, where the latter is the ratio of two Horvitz- Thompsonestimators. The numerator of the ratio estimates the size-weighted total ofthe resource equal to or less than a specified value. The denominator of theratio estimates the sum of the size-weights of the resource. If the knownsum of the size-weights of the resource is not provided, the Horvitz-Thompsonestimator of the size-weighted total of the resource equal to or less than aspecified value is used to calculate the estimate. The function canaccomodate single-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>dcdf.total<span class='tooltiptext'>This function calculates an estimate of the deconvoluted cumulativedistribution function (CDF) for the total of a discrete or an extensiveresource. The simulation extrapolation deconvolution method (Stefanski andBay, 1996) is use to deconvolute measurement error variance from theresponse. If the known extent of the resource is provided, the classic ratioestimator is used to calculate the estimate. That estimator is the product ofthe known extent of the resource and the Horvitz-Thompson ratio estimator,where the latter is the ratio of two Horvitz-Thompson estimators. Thenumerator of the ratio estimates the total of the resource equal to or lessthan a specified value. The denominator of the ratio estimates the extent ofthe resource. If the known extent of the resource is not provided, theHorvitz-Thompson estimator of the total of the resource equal to or less thana specified value is used to calculate the estimate. For a discrete resource,size is the number of units in the resource. For an extensive resource, sizeis the measure of the resource, i.e., length, area, or volume. The functioncan accomodate single-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>dcdfvar.prop<span class='tooltiptext'>This function calculates variance estimates of the estimated, deconvolutedcumulative distribution function (CDF) for the proportion of a discrete or acontinuous resource. Either the simple random sampling (SRS) varianceestimator or the local mean variance estimator is calculated, which issubject to user control. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities. Whenvariance of the estimated measurement error variance is nonzero, a correctionfactor is added to the estimated variance of the CDF. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>dcdfvar.size.prop<span class='tooltiptext'>This function calculates variance estimates of the estimated, deconvoluted,size-weighted cumulative distribution function (CDF) for the proportion of adiscrete resource. Either the simple random sampling (SRS) varianceestimator or the local mean variance estimator is calculated, which issubject to user control. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities. Whenvariance of the estimated measurement error variance is nonzero, a correctionfactor is added to the estimated variance of the CDF. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>dcdfvar.size.total<span class='tooltiptext'>This function calculates variance estimates of the estimated, deconvoluted,size-weighted cumulative distribution function (CDF) for the total of adiscrete resource. Either the simple random sampling (SRS) varianceestimator or the local mean variance estimator is calculated, which issubject to user control. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities. Whenvariance of the estimated measurement error variance is nonzero, a correctionfactor is added to the estimated variance of the CDF. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>dcdfvar.total<span class='tooltiptext'>This function calculates variance estimates of the estimated, deconvolutedcumulative distribution function (CDF) for the total of a discrete or acontinuous resource. Either the simple random sampling (SRS) varianceestimator or the local mean variance estimator is calculated, which issubject to user control. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities. Whenvariance of the estimated measurement error variance is nonzero, a correctionfactor is added to the estimated variance of the CDF. The function canaccomodate single-stage and two-stage samples. Finite population andcontinuous population correction factors can be utilized in varianceestimation.<\/span><\/div>, <div class='tooltip'>dframe.check<span class='tooltiptext'>This function checks site IDs, the sites data frame, the subpop data frame,the data.cat data frame, the data.cont data frame, the data.ar data frame,and the data.rr data frame to assure valid contents. If they do not exist,then the sites data frame and the subpop data frame are created.<\/span><\/div>, <div class='tooltip'>dsgnsum<span class='tooltiptext'>This function summarizes the sites selected for a survey design by producingcontingency tables containing the cross-tabluation of number of sites forsurvey design variables and, optionally, for auxiliary variables.<\/span><\/div>, <div class='tooltip'>examine<span class='tooltiptext'>This function examines variables in a data frame by printing either a tableor the results of a call to the describe function in the Hmisc package.<\/span><\/div>, <div class='tooltip'>framesum<span class='tooltiptext'>This function summarizes the frame for a survey design. When type.frameequals \"finite\", summary is a count of number of units in att.frame forcross-tabulation of stratum, mdcaty, and auxvar. When type.frame equals\"linear\" or \"area\", summary is the sum of length or area for units for cross-tabulation of stratum, mdcaty, and auxvar. If argument mdcaty or argumentstratum equals NULL or if both arguments equal NULL, then the cross-tabulationis performed without use of the design variable(s).<\/span><\/div>, <div class='tooltip'>geodalbers<span class='tooltiptext'>Project Latitude/Longitude to Albers Projection<\/span><\/div>, <div class='tooltip'>grts<span class='tooltiptext'>This function select a GRTS sample of a finite, linear, or area resource.Frame elements must be located in 1- or 2-dimensional coordinate system.Sample may be equal probability or unequal probability (either categorical orproportional to auxiliary variable). May designate panels of sites forsurveys over time.<\/span><\/div>, <div class='tooltip'>grtsarea<span class='tooltiptext'>This function select a GRTS sample of an area resource. The function useshierarchical randomization to ensure that the sample will include no morethan one point per cell and then picks a point in selected cells.<\/span><\/div>, <div class='tooltip'>grtslin<span class='tooltiptext'>This function select a GRTS sample of a linear resource. The function useshierarchical randomization to ensure that the sample will include no morethan one point per cell and then picks a point in selected cells.<\/span><\/div>, <div class='tooltip'>grtspts<span class='tooltiptext'>This function select a GRTS sample of a finite resource. This function useshierarchical randomization to ensure that the sample will include no morethan one point per grid cell and then picks a point in selected cells.<\/span><\/div>, <div class='tooltip'>input.check<span class='tooltiptext'>This function checks input values for errors, consistency, and compatibilitywith analytical functions.<\/span><\/div>, <div class='tooltip'>input.format<span class='tooltiptext'>This function formats an input value of class numeric, character, or factor.For a numeric value, the number of digits after the decimal point can bespecified. A factor value is converted to character. Missing values areallowed.<\/span><\/div>, <div class='tooltip'>insideAreaGridCell<span class='tooltiptext'>For each grid cell, this function calculates the clipped area of each polygonfeature contained in the cell.<\/span><\/div>, <div class='tooltip'>insideLinearGridCell<span class='tooltiptext'>For each grid cell, this function calculates the clipped length of eachlinestring feature contained in the cell.<\/span><\/div>, <div class='tooltip'>interp.axis<span class='tooltiptext'>This function creates right side y-axis labels for a CDF plot. It assumesthat arguments cdfest.l and cdfest.r are strictly increasing. If argumentyl.lab is less than the first cdfest.l vlaue, then the funciton assumes 0 forboth cdfest.l and cdfest.r.<\/span><\/div>, <div class='tooltip'>interp.cdf<span class='tooltiptext'>This function interpolates CDF values at a set of percentiles. The CDFvalues can be CDF estimates, CDF confidence bound estimates, or values atwhich the CDF is estimated (i.e., x-axis values). It is assumed thatarguments cdfest.p and cdf.value are strictly increasing.<\/span><\/div>, <div class='tooltip'>irs<span class='tooltiptext'>Select an independent random sample from a point, linear, or areal frame.Frame features must be located in a 1-dimensional or 2-dimensionalcoordinate system. Sample may be equal probability or unequal probability(either categorical or proportional to an auxiliary variable). May designatepanels of sites for surveys over time.<\/span><\/div>, <div class='tooltip'>irsarea<span class='tooltiptext'>This function selects an IRS of an area resource.<\/span><\/div>, <div class='tooltip'>irslin<span class='tooltiptext'>This function selects an IRS of a linear resource.<\/span><\/div>, <div class='tooltip'>irspts<span class='tooltiptext'>This function selects an IRS of a finite resource (discrete points).<\/span><\/div>, <div class='tooltip'>isotonic<span class='tooltiptext'>This function performs isotonic regression of an input set of values so thatthe output set of values is a nondecreasing sequence. The output set ofvalues is truncated to the range: [minval, maxval].<\/span><\/div>, <div class='tooltip'>localmean.cov<span class='tooltiptext'>This function calculates the variance-covariance matrix using the local meanestimator.<\/span><\/div>, <div class='tooltip'>localmean.df<span class='tooltiptext'>This function calculates the degrees of freedom of the local mean variance-covariance estimator.<\/span><\/div>, <div class='tooltip'>localmean.var<span class='tooltiptext'>This function calculates the local mean variance estimator.<\/span><\/div>, <div class='tooltip'>localmean.weight<span class='tooltiptext'>This function calculates the index values of neighboring points andassociated weights required by the local mean variance estimator.<\/span><\/div>, <div class='tooltip'>localmean.weight2<span class='tooltiptext'>This function calculates the initial section of the localmean.weight functionand serves to allow recovery from an error in the singular valuedecomposition function (La.svd) that is called by the generalized inversefunction (ginv) in the MASS package.<\/span><\/div>, <div class='tooltip'>make_grid<span class='tooltiptext'>Creates the grid for a GRTS suvey design.<\/span><\/div>, <div class='tooltip'>marinus<span class='tooltiptext'>This function converts x,y coordinates measured in units of latitude andlongitude, i.e., geographic coordinates measured in decimal degrees, tocoordinates in the equidistant, cylindric map projection measured in units ofkilometers. The projection center is defined as the midpoint in latitude-longitude space. The map projection is here named after Marinus of Tyre (seeJ.P. Snyder. USGS Prof Paper 1395, p. 90).<\/span><\/div>, <div class='tooltip'>mdmarea<span class='tooltiptext'>GRTS Multiplieers for Multi-density Categories for areas<\/span><\/div>, <div class='tooltip'>mdmlin<span class='tooltiptext'>Internal Function: GRTS Multipliers for Multi-Density Categories for Linear Network<\/span><\/div>, <div class='tooltip'>mdmpts<span class='tooltiptext'>Internal Function: GRTS Multipliers for Multi-Density Categories for Points<\/span><\/div>, <div class='tooltip'>numLevels<span class='tooltiptext'>This function determine the bumber of levels of hierarchical randomization for a GRTS survey design.<\/span><\/div>, <div class='tooltip'>panel_summary<span class='tooltiptext'>Revisit panel design characteristics are summarized: number of panels, numberof time periods, total number of sample events for the revisit design, totalnumber of sample events for each panel, total number of sample events foreach time period and cumulative number of unique units sampled by timeperiods.<\/span><\/div>, <div class='tooltip'>pickFiniteSamplePoints<span class='tooltiptext'>This function selects sample points from an sf object of geometry typepoint.<\/span><\/div>, <div class='tooltip'>pickGridCells<span class='tooltiptext'>This function identifies grid cells from which sample points will be selected.<\/span><\/div>, <div class='tooltip'>pickSamplePoints<span class='tooltiptext'>This function selects sample points from an sf object of geometry typepolygon or linestring.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>plot_powerpaneldesign<span class='tooltiptext'>Plot power curves and relative power curves for trend detection for set ofpanel designs, time periods, indicators, signifance levels and trend. Trendmay be based on percent change per period in mean or percent change inproportion of cumulative distribution function above or below a fixed cutpoint. Types of plots are combinations of standard/relative, mean/percent,period/change and design/indicator. Input must be be of classpowerpaneldesign and is normally the output of function power.dsgn.<\/span><\/div>, <div class='tooltip'>power.dsgn<span class='tooltiptext'>Calculates the power for trend detection for one or more variables, for oneor more panel designs, for one or more linear trends, and for one or moresignficance levels. The panel designs create a covarance model where themodel includes variance components for units, periods, the interaction ofunits and periods, and the residual (or index) variance.<\/span><\/div>, <div class='tooltip'>ranho<span class='tooltiptext'>This function constructs randomized hierarchical addresses for a GRTS surveydesign.<\/span><\/div>, <div class='tooltip'>read.dbf<span class='tooltiptext'>This function reads the dbf file of an ESRI shapefile and creates a data frame.<\/span><\/div>, <div class='tooltip'>read.sas<span class='tooltiptext'>This function reads either a SAS dataset or a SAS XPORT (transport) file andcreates a data frame.<\/span><\/div>, <div class='tooltip'>read.shape<span class='tooltiptext'>This function reads an ESRI shapefile and creates a simple features (sf) object.<\/span><\/div>, <div class='tooltip'>relrisk.analysis<span class='tooltiptext'>This function organizes input and output for relative risk analysis ofcategorical data generated by a probability survey.<\/span><\/div>, <div class='tooltip'>relrisk.est<span class='tooltiptext'>This function calculates the relative risk estimate for a 2x2 table of cellcounts defined by a categorical response variable and a categoricalexplanatory (stressor) variable for an unequal probability design. Relativerisk is the ratio of two probabilities: the numerator is the probability thatthe first level of the response variable is observed given occurrence of thefirst level of the stressor variable, and the denominator is the probabilitythat the first level of the response variable is observed given occurrence ofthe second level of the stressor variable. The numerator probability anddenominator probability are estimated using cell and marginal totals from a2x2 table of cell counts defined by a categorical response variable and acategorical stressor variable. An estimate of the numerator probability isprovided by the ratio of the cell total defined by the first level ofresponse variable and the first level of the stressor variable to themarginal total for the first level of the stressor variable. An estimate ofthe denominator probability is provided by the ratio of the cell totaldefined by the first level of response variable and the second level of thestressor variable to the marginal total for the second level of the stressorvariable. Cell and marginal totals are estimated using the Horvitz-Thompsonestimator. The standard error of the log of the relative risk estimate andconfidence limits for the estimate also are calculated. The standard error iscalculated using a first-order Taylor series linearization (Sarndal et al.,1992).<\/span><\/div>, <div class='tooltip'>relrisk.var<span class='tooltiptext'>This function calculates the variance-covariance estimate for the cell andmarginal totals used to calculate the relative risk estimate. Either thesimple random sampling (SRS) variance estimator or the local mean varianceestimator is calculated, which is subject to user control. The SRS varianceestimator uses the independent random sample approximation to calculate jointinclusion probabilities. The function can accomodate single-stage andtwo-stage samples.<\/span><\/div>, <div class='tooltip'>revisit_bibd<span class='tooltiptext'>Create a revisit design for panels in a survey that specifies the timeperiods for the units of each panel to be sampled based on searching for aD-optimal block design that is a member of the class of generalized Youdendesigns. The resulting design need not be a balanced incomplete blockdesign. Based on algorithmic idea by Cook and Nachtsheim (1989) andimplemented by Robert Wheeler.<\/span><\/div>, <div class='tooltip'>revisit_dsgn<span class='tooltiptext'>Create a revisit design for panels in a survey that specifies the timeperiods that members of each panel will be sampled. Three basic panel designstructures may be created: always revisit panel, serially alternating panels,or rotating panels.<\/span><\/div>, <div class='tooltip'>revisit_rand<span class='tooltiptext'>Create a revisit design for a survey that specifies the panels and timeperiods that will be sampled by random selection of panels and time periods.Three options for random assignments are \"period\" where the number of timeperiods to be sampled in a panel is fixed, \"panel\" where the number panels tobe sampled in a time period is fixed, and \"none\" where the number ofpanel-period combinations is fixed.<\/span><\/div>, <div class='tooltip'>sbcframe<span class='tooltiptext'>This function calculates spatial balance grid cell extent and proportionfor the sample frame.<\/span><\/div>, <div class='tooltip'>sbcsamp<span class='tooltiptext'>This function calculates spatial balance grid cell extent and proportionsfor a survey design. The user must provide either sbc.frame or values fordx, dy, xc, and yc.<\/span><\/div>, <div class='tooltip'>selectFeatureID<span class='tooltiptext'>This function determines the ID value of the feature in an sf object fromwhich a sample point will be selected.<\/span><\/div>, <div class='tooltip'>selectframe<span class='tooltiptext'>Internal Function: Select All Points in Frame<\/span><\/div>, <div class='tooltip'>selectpts<span class='tooltiptext'>Selects a sample of size one or larger from a set of cells based on inclusionprobabilities.<\/span><\/div>, <div class='tooltip'>simex<span class='tooltiptext'>This function executes the extrapolation step of the simulation extrapolationdeconvolution method (Stefanski and Bay, 1996). The function can accomodatesingle-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>sorted<span class='tooltiptext'>This function determines whether the input set of values is a nondecreasingsequence.<\/span><\/div>, <div class='tooltip'>sp2shape<span class='tooltiptext'>This function creates an ESRI shapefile from an sp package object. Thefunction can also accommodate an object created by the grts or irs functionsin spsurvey. The type of shapefile, i.e., point, polyline, or polygon, isdetermined by the class of the sp object.<\/span><\/div>, <div class='tooltip'>SpatialDesign<span class='tooltiptext'>Create the definition for class SpatialDesign, a class for spatial survey designs.Create the wrapper function for class SpatialDesign.Define S3 and S4 methods for summary and plot for class SpatialDesign.<\/span><\/div>, <div class='tooltip'>spbalance<span class='tooltiptext'>This function calculates spatial balance metrics for a survey design. Twooptions for calculation of spatial balance metrics are available: (1) useproportions obtained from the intersection of Dirichlet tesselation polygonsfor the sample points with the frame object and (2) use proportions obtainedfrom a rectangular grid superimposed on the sample points and the frameobject. In both cases the proportions are used to calculate the spatialbalance metrics. Two metrics are calculated: (1) the Pielou evenness measureand (2) the chi-square statistic.<\/span><\/div>, <div class='tooltip'>spsurvey.analysis<span class='tooltiptext'>This function creates an object of class spsurvey.analysis that contains allof the information necessary to use the analysis functions in the spsurveypackage.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>total.est<span class='tooltiptext'>This function calculates estimates of the population total, mean, variance,and standard deviation of a response variable, where the response variablemay be defined for either a finite or an extensive resource. In addition thestandard error of the population estimates and confidence bounds arecalculated. The Horvitz-Thompson estimator is used to calculate the total,variance, and standard deviation estimates. The Horvitz-Thompson ratioestimator, i.e., the ratio of two Horvitz-Thompson estimators, is used tocalculate the mean estimate. Variance estimates are calculated using eitherthe local mean variance estimator or the simple random sampling (SRS)variance estimator. The choice of variance estimator is subject to usercontrol. The local mean variance estimator requires the x-coordinate and they-coordinate of each site. The SRS variance estimator uses the independentrandom sample approximation to calculate joint inclusion probabilities.Confidence bounds are calculated using a Normal distribution multiplier. Thefunction can accommodate a stratified sample. For a stratified sample,separate estimates and standard errors are calculated for each stratum, whichare used to produce estimates and standard errors for all strata combined.Strata that contain a single value are removed. For a stratified sample,when either the size of the resource or the sum of the size-weights of theresource is provided for each stratum, those values are used as stratumweights for calculating the estimates and standard errors for all stratacombined. For a stratified sample when neither the size of the resource northe sum of the size-weights of the resource is provided for each stratum,estimated values are used as stratum weights for calculating the estimatesand standard errors for all strata combined. The function can accommodatesingle-stage and two-stage samples for both stratified and unstratifiedsampling designs. Finite population and continuous population correctionfactors can be utilized in variance estimation. The function checks forcompatibility of input values and removes missing values.<\/span><\/div>, <div class='tooltip'>total.var<span class='tooltiptext'>This function calculates variance estimates of the estimated populationtotal, mean, variance, and standard deviation of a response variable. Eitherthe simple random sampling (SRS) variance estimator or the local meanvariance estimator is calculated, which is subject to user control. The SRSvariance estimator uses the independent random sample approximation tocalculate joint inclusion probabilities. The function can accomodatesingle-stage and two-stage samples.<\/span><\/div>, <div class='tooltip'>uniqueID<span class='tooltiptext'>This function creates unique site IDs by appending a unique number to eachoccurrence of a site ID. It is intended for survey designs that have repeatvisits to sites.<\/span><\/div>, <div class='tooltip'>vecprint<span class='tooltiptext'>This function takes an input vector and outputs a character string with linebreaks inserted so that, whenever possible, no line in the string exceeds theinput value n.char, which is set to 78 characters by default. The inputvector is coereced to mode character. When an element of the input vector isgreater than n.char characters in length, then that element is inserted inthe output character string as an individual line.<\/span><\/div>, <div class='tooltip'>warnprnt<span class='tooltiptext'>This function prints the warnings data frame.<\/span><\/div>, <div class='tooltip'>wnas<span class='tooltiptext'>This function removes missing values from data for which the mode is one ofthe following: numeric, logical, or character. Data that is not one of thosemodes will cause the function to terminate with an error message. For numericdata this function removes values that are not finite, i.e., missing value(NA), not a number (NaN), infinity (Inf), and minus infinity (-Inf). Forlogical data this function removes missing values (NA). For character datathe following values are removed: \"\", \"NA\", NA (R only), \"NaN\", \"Inf\", and\"-Inf\". For a factor this function removes the following values: NA, NaN,Inf, and -Inf. For a vector this function returns the vector with theindicated values removed. For a data frame this function returns the dataframe with rows removed that contain at least one indicated value. For alist with components that are the same length, the list is converted to adata frame. For a list with components that are not the same length, thefunction prints an error message and terminates. When the process ofremoving missing values produces an object that no longer contains anyelements (vector) or rows (data frame or list), then a NULL object isreturned.<\/span><\/div>, <div class='tooltip'>write.object<span class='tooltiptext'>This function writes the contents of an object to a plot. The object may beeither a data frame or a matrix. Values in the input data frame or matrixmust be of class numeric, character, or factor.<\/span><\/div>","<div class='tooltip'>ssrm.logmer<span class='tooltiptext'>Provides the necessary sample size for a longitudinal study with binary outcome in order to attain a pre-specified power while strictly maintaining the Type I error rate.The sample size computation requires the user to define a column of design matrix relating to the slope of time as a monotonic function of time, such as linear, log, sqrt etc., along with the respective beta parameters. The underlying model is assumed to be a two-level logistic mixed-effects regression model with random intercept and/or slope of time to account for within-subject correlations and between-subject variability.Gaussian quadrature is used to compute the marginal likelihood integrals and to evaluate Fisher Information matrix.<\/span><\/div>","<div class='tooltip'>sunter<span class='tooltiptext'>The function uses the Sunter's method to select a sample of units (unequal probabilities, without replacement, fixed sample size).<\/span><\/div>, <div class='tooltip'>sunterpi<span class='tooltiptext'>The function computes the inclusion probabilities for Sunter sampling.<\/span><\/div>, <div class='tooltip'>sunterpi2<span class='tooltiptext'>The function computes the joint (second-order) inclusion probabilities for Sunter sampling.<\/span><\/div>","<div class='tooltip'>bias_correction_transform<span class='tooltiptext'>two_step_transform, subject_centering_transform, bias_correction_transform and pool_sd_transform are four transformations that can be applied to a matrix of data.<\/span><\/div>, <div class='tooltip'>custom<span class='tooltiptext'>There is four ways that effects can be definedin GRD. \"factor = slope(s)\" will vary the means by an amount of s for each step of the factor; \"factor = extent(s)\" will vary the meansuniformly so that there is a difference of s between the first andthe last factor level; \"factor = custom(a,b,c..)\" will alter eachmeans by an amount of a for the first, b for the second, etc. Finally\"factor = Rexpression(\"R code\")\" will apply R code to all levels of the factors. R code result alters the base mean.<\/span><\/div>, <div class='tooltip'>epsilon<span class='tooltiptext'>epsilon is a measure of sphericity created by Greenhouse and Geiser (1958). The original measure was biasedand therefore, Hyunh and Feldt (1976) produced a revised version(note that the 1976 paper contained typos that were uncorrectedin SPSS; Lecoutre, 1991)<\/span><\/div>, <div class='tooltip'>extent<span class='tooltiptext'>There is four ways that effects can be definedin GRD. \"factor = slope(s)\" will vary the means by an amount of s for each step of the factor; \"factor = extent(s)\" will vary the meansuniformly so that there is a difference of s between the first andthe last factor level; \"factor = custom(a,b,c..)\" will alter eachmeans by an amount of a for the first, b for the second, etc. Finally\"factor = Rexpression(\"R code\")\" will apply R code to all levels of the factors. R code result alters the base mean.<\/span><\/div>, <div class='tooltip'>GRD<span class='tooltiptext'>Generate Random Data (GRD) is a function thatgenerates a data frame containing random data suitable for analyses.The data can be from within-subject or between-group designs.Within-subject designs are in wide format. The function was originallypresented in \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}ch19;textualsuperbCalderini and Harding (2019).<\/span><\/div>, <div class='tooltip'>lambda<span class='tooltiptext'>The functions lambda returns the correction factor for cluster-randomized sampling. This correction is then used in a variety of ways, for example, to get the effective number of participants (in a power study) or to correct a t-test. See \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}cl16superb(Cousineau and Laurencelle 2016).<\/span><\/div>, <div class='tooltip'>MauchlySphericityTest<span class='tooltiptext'>Performs a test of sphericity on a dataframe with multiple measures, one subject per line. It assesses the significance of the null hypothesis that the covariance matrix is spherical. This test is described in \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}a10superb(Abdi 2010)<\/span><\/div>, <div class='tooltip'>pool_sd_transform<span class='tooltiptext'>two_step_transform, subject_centering_transform, bias_correction_transform and pool_sd_transform are four transformations that can be applied to a matrix of data.<\/span><\/div>, <div class='tooltip'>Rexpression<span class='tooltiptext'>There is four ways that effects can be definedin GRD. \"factor = slope(s)\" will vary the means by an amount of s for each step of the factor; \"factor = extent(s)\" will vary the meansuniformly so that there is a difference of s between the first andthe last factor level; \"factor = custom(a,b,c..)\" will alter eachmeans by an amount of a for the first, b for the second, etc. Finally\"factor = Rexpression(\"R code\")\" will apply R code to all levels of the factors. R code result alters the base mean.<\/span><\/div>, <div class='tooltip'>runDebug<span class='tooltiptext'>runDebug is an internal function used by GRD and superbPlot to help in debugging the functions. It assigns in the global environment the variables that are local to a function so that they become visible.<\/span><\/div>, <div class='tooltip'>ShroutFleissICC1<span class='tooltiptext'>The functions ShroutFleissICC1, ShroutFleissICC11 and ShroutFleissICC1k computes the intra-class correlation ICC for a given data frame containing repeated measures in columns cols when the measures are in distinct clusters, identified in column clustercol. See \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}sf79superb(Shrout and Fleiss 1979).<\/span><\/div>, <div class='tooltip'>ShroutFleissICC11<span class='tooltiptext'>The functions ShroutFleissICC1, ShroutFleissICC11 and ShroutFleissICC1k computes the intra-class correlation ICC for a given data frame containing repeated measures in columns cols when the measures are in distinct clusters, identified in column clustercol. See \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}sf79superb(Shrout and Fleiss 1979).<\/span><\/div>, <div class='tooltip'>ShroutFleissICC1k<span class='tooltiptext'>The functions ShroutFleissICC1, ShroutFleissICC11 and ShroutFleissICC1k computes the intra-class correlation ICC for a given data frame containing repeated measures in columns cols when the measures are in distinct clusters, identified in column clustercol. See \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}sf79superb(Shrout and Fleiss 1979).<\/span><\/div>, <div class='tooltip'>slope<span class='tooltiptext'>There is four ways that effects can be definedin GRD. \"factor = slope(s)\" will vary the means by an amount of s for each step of the factor; \"factor = extent(s)\" will vary the meansuniformly so that there is a difference of s between the first andthe last factor level; \"factor = custom(a,b,c..)\" will alter eachmeans by an amount of a for the first, b for the second, etc. Finally\"factor = Rexpression(\"R code\")\" will apply R code to all levels of the factors. R code result alters the base mean.<\/span><\/div>, <div class='tooltip'>subject_centering_transform<span class='tooltiptext'>two_step_transform, subject_centering_transform, bias_correction_transform and pool_sd_transform are four transformations that can be applied to a matrix of data.<\/span><\/div>, <div class='tooltip'>superbPlot<span class='tooltiptext'>plotsuberb plots standard error or confidence interval for various descriptive statistics under various designs, sampling schemes, population size and purposes, according to the suberb framework. See \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}c17superb(Cousineau 2017) for more.<\/span><\/div>, <div class='tooltip'>superbPlot.bar<span class='tooltiptext'>superbPlot comes with a few built-in templates forfor making the final plots. All produces ggplot objects thatcan be further customized. Additionally, it is possible to addcustom-make templates (see vignette 6). The functions, to be \"superbPlot-compatible\", must have these parameters:If you have useful styles, do contribute github.com/dcousin3/superbby opening an issue.Finally, any number of optional list of parameters with default value set to list()can be provided depending on the template style. These includes errorbarParams,pointParams, barParams, etc.<\/span><\/div>, <div class='tooltip'>superbPlot.line<span class='tooltiptext'>superbPlot comes with a few built-in templates forfor making the final plots. All produces ggplot objects thatcan be further customized. Additionally, it is possible to addcustom-make templates (see vignette 6). The functions, to be \"superbPlot-compatible\", must have these parameters:If you have useful styles, do contribute github.com/dcousin3/superbby opening an issue.Finally, any number of optional list of parameters with default value set to list()can be provided depending on the template style. These includes errorbarParams,pointParams, barParams, etc.<\/span><\/div>, <div class='tooltip'>superbPlot.point<span class='tooltiptext'>superbPlot comes with a few built-in templates forfor making the final plots. All produces ggplot objects thatcan be further customized. Additionally, it is possible to addcustom-make templates (see vignette 6). The functions, to be \"superbPlot-compatible\", must have these parameters:If you have useful styles, do contribute github.com/dcousin3/superbby opening an issue.Finally, any number of optional list of parameters with default value set to list()can be provided depending on the template style. These includes errorbarParams,pointParams, barParams, etc.<\/span><\/div>, <div class='tooltip'>superbPlot.pointindividualline<span class='tooltiptext'>superbPlot comes with a few built-in templates forfor making the final plots. All produces ggplot objects thatcan be further customized. Additionally, it is possible to addcustom-make templates (see vignette 6). The functions, to be \"superbPlot-compatible\", must have these parameters:If you have useful styles, do contribute github.com/dcousin3/superbby opening an issue.Finally, any number of optional list of parameters with default value set to list()can be provided depending on the template style. These includes errorbarParams,pointParams, barParams, etc.<\/span><\/div>, <div class='tooltip'>superbPlot.pointjitter<span class='tooltiptext'>superbPlot comes with a few built-in templates forfor making the final plots. All produces ggplot objects thatcan be further customized. Additionally, it is possible to addcustom-make templates (see vignette 6). The functions, to be \"superbPlot-compatible\", must have these parameters:If you have useful styles, do contribute github.com/dcousin3/superbby opening an issue.Finally, any number of optional list of parameters with default value set to list()can be provided depending on the template style. These includes errorbarParams,pointParams, barParams, etc.<\/span><\/div>, <div class='tooltip'>superbPlot.pointjitterviolin<span class='tooltiptext'>superbPlot comes with a few built-in templates forfor making the final plots. All produces ggplot objects thatcan be further customized. Additionally, it is possible to addcustom-make templates (see vignette 6). The functions, to be \"superbPlot-compatible\", must have these parameters:If you have useful styles, do contribute github.com/dcousin3/superbby opening an issue.Finally, any number of optional list of parameters with default value set to list()can be provided depending on the template style. These includes errorbarParams,pointParams, barParams, etc.<\/span><\/div>, <div class='tooltip'>two_step_transform<span class='tooltiptext'>two_step_transform, subject_centering_transform, bias_correction_transform and pool_sd_transform are four transformations that can be applied to a matrix of data.<\/span><\/div>, <div class='tooltip'>WinerCompoundSymmetryTest<span class='tooltiptext'>Run a test of compound symmetry. generates a data frame of random data suitable for analyses. It assesses the significance of the null hypothesis that the covariance matrix is compound symmetric. This test is given without demonstration in \\Sexpr[results=rd,stage=build]{if(!exists(\".Rdpack.currefs\")) .Rdpack.currefs <- new.env();Rdpack::insert_citeOnly(key=\"#1\",package=\"#2\", cached_env=.Rdpack.currefs)}w91superb(Winer et al. 1991), p. 517.<\/span><\/div>","<div class='tooltip'>calc.stError<span class='tooltiptext'>Calculate point estimates as well as standard errors of variables in surveys.Standard errors are estimated using bootstrap weights (see draw.bootstrapand recalib). In addition the standard error of an estimate can becalcualted using the survey data for 3 or more consecutive periods, whichresults in a reduction of the standard error.<\/span><\/div>, <div class='tooltip'>combine_factors<span class='tooltiptext'>C++ routines to invoke a single iteration of the Iterative proportional updating (IPU) scheme. Targets and classesare assumed to be one dimensional in the ipf_step functions. combine_factors aggregates several vectors oftype factor into a single one to allow multidimensional ipu-steps. See examples.<\/span><\/div>, <div class='tooltip'>computeFrac<span class='tooltiptext'>Customize weight-updating within factor levels in case of numericalcalibration. The functions described here serve as inputs for ipf.<\/span><\/div>, <div class='tooltip'>computeLinear<span class='tooltiptext'>Customize weight-updating within factor levels in case of numericalcalibration. The functions described here serve as inputs for ipf.<\/span><\/div>, <div class='tooltip'>computeLinearG1<span class='tooltiptext'>Customize weight-updating within factor levels in case of numericalcalibration. The functions described here serve as inputs for ipf.<\/span><\/div>, <div class='tooltip'>demo.eusilc<span class='tooltiptext'>Create a dummy dataset to be used for demonstrating the functionalities ofthe surveysd package based on laeken::eusilc. Please refer to thedocumentation page of the original data for details about the variables.<\/span><\/div>, <div class='tooltip'>draw.bootstrap<span class='tooltiptext'>Draw bootstrap replicates from survey data with rotating paneldesign. Survey information, like ID, sample weights, strata and populationtotals per strata, should be specified to ensure meaningfull surveybootstraping.<\/span><\/div>, <div class='tooltip'>generate.HHID<span class='tooltiptext'>Generating a new houshold ID for survey data using a houshold ID and apersonal ID.For surveys with rotating panel design containing housholds, houshold memberscan move from an existing household to a new one, that was not originally inthe sample. This leads to the creation of so called split households. Using aperonal ID (that stays fixed over the whole survey), an indicator fordifferent time steps and a houshold ID, a new houshold ID is assigned to theoriginal and the split household.<\/span><\/div>, <div class='tooltip'>ipf<span class='tooltiptext'>Adjust sampling weights to given totals based on household-level and/orindividual level constraints.<\/span><\/div>, <div class='tooltip'>ipf_step<span class='tooltiptext'>C++ routines to invoke a single iteration of the Iterative proportional updating (IPU) scheme. Targets and classesare assumed to be one dimensional in the ipf_step functions. combine_factors aggregates several vectors oftype factor into a single one to allow multidimensional ipu-steps. See examples.<\/span><\/div>, <div class='tooltip'>ipf_step_f<span class='tooltiptext'>C++ routines to invoke a single iteration of the Iterative proportional updating (IPU) scheme. Targets and classesare assumed to be one dimensional in the ipf_step functions. combine_factors aggregates several vectors oftype factor into a single one to allow multidimensional ipu-steps. See examples.<\/span><\/div>, <div class='tooltip'>ipf_step_ref<span class='tooltiptext'>C++ routines to invoke a single iteration of the Iterative proportional updating (IPU) scheme. Targets and classesare assumed to be one dimensional in the ipf_step functions. combine_factors aggregates several vectors oftype factor into a single one to allow multidimensional ipu-steps. See examples.<\/span><\/div>, <div class='tooltip'>kishFactor<span class='tooltiptext'>Compute the design effect due to unequal weighting.<\/span><\/div>, <div class='tooltip'>recalib<span class='tooltiptext'>Calibrate weights for bootstrap replicates by using iterative proportionalupdating to match population totals on various household and personal levels.<\/span><\/div>, <div class='tooltip'>rescaled.bootstrap<span class='tooltiptext'>Draw bootstrap replicates from survey data using the rescaledbootstrap for stratified multistage sampling, presented by Preston, J.(2009).<\/span><\/div>, <div class='tooltip'>weightedRatio<span class='tooltiptext'>Predefined functions for weighted point estimates in package surveysd.<\/span><\/div>, <div class='tooltip'>weightedSum<span class='tooltiptext'>Predefined functions for weighted point estimates in package surveysd.<\/span><\/div>","<div class='tooltip'>svycox.calibrate<span class='tooltiptext'>Checks the calibration of a nomogram for a survey-weighted Cox model.<\/span><\/div>, <div class='tooltip'>svycox.nomogram<span class='tooltiptext'>Builds a nomogram for a survey-weighted Cox model.<\/span><\/div>, <div class='tooltip'>svycox.validate<span class='tooltiptext'>Validates a nomogram for a survey-weighted Cox model using bootstrap.<\/span><\/div>","<div class='tooltip'>svy_vglm<span class='tooltiptext'>This function provides design-based (survey) inference for Thomas Yee'svector generalised linear models. It works by calling vglm withsampling weights, and then either using resampling (replicate weights)or extracting the influence functions and using a Horvitz-Thompson-typesandwich estimator.<\/span><\/div>","<div class='tooltip'>desc<span class='tooltiptext'>desc caculates select descriptive statistics for the variable y.<\/span><\/div>, <div class='tooltip'>print.teqrDG<span class='tooltiptext'>Print file for the class teqrDG.<\/span><\/div>, <div class='tooltip'>print.teqrOCact<span class='tooltiptext'>Print file for the class teqrOCact.<\/span><\/div>, <div class='tooltip'>print.teqrOCtox<span class='tooltiptext'>Print file for the class teqrOCtox.<\/span><\/div>, <div class='tooltip'>teqr.DLdata<span class='tooltiptext'>teqr.DLdata takes the simData output and creates the dose level data.<\/span><\/div>, <div class='tooltip'>teqr.DLdataconACT<span class='tooltiptext'>teqr.DLdata takes the simData output and creates the dose level data.<\/span><\/div>, <div class='tooltip'>teqr.MTDdataconACT<span class='tooltiptext'>teqr.MTDdatatox takes the simData and DLdata output and creates the MTD level data.<\/span><\/div>, <div class='tooltip'>teqr.MTDdatatox<span class='tooltiptext'>teqr.MTDdatatox takes the simData and DLdata output and creates the MTD level data.<\/span><\/div>, <div class='tooltip'>teqrDG<span class='tooltiptext'>teqrDG prints the TEQR Design Dose Escalation/Expansion/De-escalation Guidelines.The toxicity equivalence range (TEQR) design, is a frequentist up-down design patterned after the modified toxicity probability interval design. The TEQR design extends the well known 3+3 design to allow for: an explicit target range for the dose limiting toxicity (DLT) rate, more than 6 subjects at the maximum tolerated dose (MTD), and specification of a too-toxic rate, which closes a dose level. The TEQR is better than the 3+3 when compared on: 1) the number of times the dose at or nearest the target toxicity level was selected as the MTD, 2) the number of subjects assigned to doses levels, at, or nearest the MTD, and 3) the overall trial DLT rate. For sample sizes at the MTD greater that 6 TEQR more accurately and more precisely estimates the rate of toxicity at the MTD because a larger number of subjects are studied at the MTD dose. The TEQR on average uses fewer subjects and provide reasonably comparable results to the continual reassessment method (CRM) in the number of times the dose at or nearest the target toxicity level was selected as the MTD and the number of subjects assigned doses, at, or nearest that target and in overall DLT rate.<\/span><\/div>, <div class='tooltip'>teqrOCact<span class='tooltiptext'>The activity constrained for toxicity (ACT) design implements a cumulative cohort design in conjunction with safety rules similar to those from the toxicity equivalence range design or the modified toxicity probability interval design. The unique feature of this design is that dose is escalated based on lack of activity rather than on lack of toxicity and is de-escalated only if an unacceptable level of toxicity is experienced.<\/span><\/div>, <div class='tooltip'>teqrOCtox<span class='tooltiptext'>teqrOCtox: Runs clinical trial simulations for the TEQR Design implemented as a Phase I toxicity design and provides a listing of the operating characteristics.The toxicity equivalence range (TEQR) design is a frequentist up-down design patterned after the modified toxicity probability interval design. The TEQR design extends the well known 3+3 design to allow for: an explicit target range for the dose limiting toxicity (DLT) rate, more than 6 subjects at the maximum tolerated dose (MTD), and specification of a too-toxic rate, which closes a dose level. The TEQR is better than the 3+3 when compared on: 1) the number of times the dose at or nearest the target toxicity level was selected as the MTD, 2) the number of subjects assigned to doses levels, at, or nearest the MTD, and 3) the overall trial DLT rate. For sample sizes at the MTD greater that 6 TEQR more accurately and more precisely estimates the rate of toxicity at the MTD because a larger number of subjects are studied at the MTD dose. The TEQR on average uses fewer subjects and provide reasonably comparable results to the continual reassessment method (CRM) in the number of times the dose at or nearest the target toxicity level was selected as the MTD and the number of subjects assigned doses, at, or nearest that target and in overall DLT rate.<\/span><\/div>","<div class='tooltip'>app<span class='tooltiptext'>app and OAT are aliases of TestDesign.<\/span><\/div>, <div class='tooltip'>array_info_1pl<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>array_info_2pl<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>array_info_3pl<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>array_info_gpc<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>array_info_gr<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>array_info_pc<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>array_p_1pl<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>array_p_2pl<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>array_p_3pl<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>array_p_gpc<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>array_p_gr<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>array_p_pc<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>as.data.frame<span class='tooltiptext'><\/span><\/div>, <div class='tooltip'>buildConstraints<span class='tooltiptext'>buildConstraints is a data loading function to create a constraints object.buildConstraints is a shortcut that calls other data loading functions.The constraints must be in the expected format; see the vignette in vignette(\"constraints\").<\/span><\/div>, <div class='tooltip'>calc_info<span class='tooltiptext'>calc_info and calc_info_matrix are functions to calculate Fisher information.These functions are designed for multiple items.<\/span><\/div>, <div class='tooltip'>calc_info_matrix<span class='tooltiptext'>calc_info and calc_info_matrix are functions to calculate Fisher information.These functions are designed for multiple items.<\/span><\/div>, <div class='tooltip'>calc_likelihood<span class='tooltiptext'>calc_likelihood and calc_likelihood_function are functions to calculate likelihoods.<\/span><\/div>, <div class='tooltip'>calc_likelihood_function<span class='tooltiptext'>calc_likelihood and calc_likelihood_function are functions to calculate likelihoods.<\/span><\/div>, <div class='tooltip'>calc_log_likelihood<span class='tooltiptext'>calc_likelihood and calc_likelihood_function are functions to calculate likelihoods.<\/span><\/div>, <div class='tooltip'>calc_log_likelihood_function<span class='tooltiptext'>calc_likelihood and calc_likelihood_function are functions to calculate likelihoods.<\/span><\/div>, <div class='tooltip'>calcEscore<span class='tooltiptext'>calcEscore is a function to calculate expected scores.<\/span><\/div>, <div class='tooltip'>calcFisher<span class='tooltiptext'>calcFisher is a function to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>calcHessian<span class='tooltiptext'>calcHessian is a function to calculate the second derivative of the log-likelihood function.<\/span><\/div>, <div class='tooltip'>calcJacobian<span class='tooltiptext'>calcJacobian is a function to calculate the first derivative of the log-likelihood function.<\/span><\/div>, <div class='tooltip'>calcLocation<span class='tooltiptext'>calcLocation is a function to calculate the central location (overall difficulty) of items.<\/span><\/div>, <div class='tooltip'>calcLogLikelihood<span class='tooltiptext'>calcLogLikelihood is a function to calculate log-likelihood values.<\/span><\/div>, <div class='tooltip'>calcProb<span class='tooltiptext'>calcProb is a function to calculate item response probabilities.<\/span><\/div>, <div class='tooltip'>combineConstraints<span class='tooltiptext'>Create a subset of a constraints object: constraints[i] subsetConstraints(constraints, 1:10)Combine two constraints objects: c(constraints1, constraints2) combineConstraints(constraints1, constraints2)<\/span><\/div>, <div class='tooltip'>combineItemPool<span class='tooltiptext'>Create a subset of an item_pool object: pool[i] subsetItemPool(pool, i)Combine two item_pool objects: c(pool1, pool2) combineItemPool(pool1, pool2) pool1 + pool2pool1 - pool2 excludes items in pool2 from pool1.pool1 == pool2 tests whether two item_pool objects are identical.<\/span><\/div>, <div class='tooltip'>createShadowTestConfig<span class='tooltiptext'>createShadowTestConfig is a config function to create a config_Shadow object for Shadow test assembly.Default values are used for any unspecified parameters/slots.<\/span><\/div>, <div class='tooltip'>createStaticTestConfig<span class='tooltiptext'>createStaticTestConfig is a config function to create a config_Static object for Static (fixed-form) test assembly.Default values are used for any unspecified parameters/slots.<\/span><\/div>, <div class='tooltip'>eap<span class='tooltiptext'>eap is a function to compute expected a posteriori estimates of theta.<\/span><\/div>, <div class='tooltip'>find_segment<span class='tooltiptext'>find_segment is a function to classify theta values into segments based on supplied cutpoints.<\/span><\/div>, <div class='tooltip'>getSolution<span class='tooltiptext'>Print solution items<\/span><\/div>, <div class='tooltip'>getSolutionAttributes<span class='tooltiptext'>getSolutionAttributes is a helper function to retrieve constraints-related attributes from a solution.<\/span><\/div>, <div class='tooltip'>info_1pl<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>info_2pl<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>info_3pl<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>info_gpc<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>info_gr<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>info_pc<span class='tooltiptext'>info_* and array_info_* are functions to calculate Fisher information.<\/span><\/div>, <div class='tooltip'>iparPosteriorSample<span class='tooltiptext'>Sample item parameter estimates from their posterior distributions.<\/span><\/div>, <div class='tooltip'>lnHyperPars<span class='tooltiptext'>Calculate hyperparameters for log-normal distribution.<\/span><\/div>, <div class='tooltip'>loadConstraints<span class='tooltiptext'>loadConstraints is a data loading function to create a constraints object.loadConstraints can read constraints from a data.frame or a .csv file.The contents must be in the expected format; see the vignette in vignette(\"constraints\").<\/span><\/div>, <div class='tooltip'>loadItemAttrib<span class='tooltiptext'>loadItemAttrib is a data loading function to create an item_attrib object.loadItemAttrib can read item attributes a data.frame or a .csv file.<\/span><\/div>, <div class='tooltip'>loadItemPool<span class='tooltiptext'>loadItemPool is a data loading function to create an item_pool object.loadItemPool can read item parameters and standard errors from a data.frame or a .csv file.<\/span><\/div>, <div class='tooltip'>loadStAttrib<span class='tooltiptext'>loadStAttrib is a data loading function to create an st_attrib object.loadStAttrib can read stimulus attributes a data.frame or a .csv file.<\/span><\/div>, <div class='tooltip'>logitHyperPars<span class='tooltiptext'>Calculate hyperparameters for logit-normal distribution.<\/span><\/div>, <div class='tooltip'>makeItemPoolCluster<span class='tooltiptext'>Create a item_pool_cluster object.item_pool_cluster1 == item_pool_cluster2 tests equality of two item_pool_cluster objects.<\/span><\/div>, <div class='tooltip'>makeTest<span class='tooltiptext'>makeTest is a function for creating a test object.This is used in Shadow to determine all necessary data prior to the main simulation, so that they are not affected by random number generation.<\/span><\/div>, <div class='tooltip'>mle<span class='tooltiptext'>mle is a function to compute maximum likelihood estimates of theta.<\/span><\/div>, <div class='tooltip'>mlef<span class='tooltiptext'>mlef is a function to compute maximum likelihood estimates of theta using fence items.<\/span><\/div>, <div class='tooltip'>OAT<span class='tooltiptext'>app and OAT are aliases of TestDesign.<\/span><\/div>, <div class='tooltip'>p_1pl<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>p_2pl<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>p_3pl<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>p_gpc<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>p_gr<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>p_pc<span class='tooltiptext'>p_* and array_p_* are functions to calculate item response probability.<\/span><\/div>, <div class='tooltip'>plot<span class='tooltiptext'>Extension of plot() for objects in TestDesign package<\/span><\/div>, <div class='tooltip'>plotExposure<span class='tooltiptext'>(deprecated) Use plot with type = 'exposure' instead.<\/span><\/div>, <div class='tooltip'>plotInfo<span class='tooltiptext'>(deprecated) Use plot.<\/span><\/div>, <div class='tooltip'>print<span class='tooltiptext'>Extension of print() for objects in TestDesign package<\/span><\/div>, <div class='tooltip'>Shadow<span class='tooltiptext'>Shadow is a test assembly function to perform adaptive test assembly based on the generalized shadow-test framework.<\/span><\/div>, <div class='tooltip'>show<span class='tooltiptext'>Extension of show() for objects in TestDesign package<\/span><\/div>, <div class='tooltip'>simResp<span class='tooltiptext'>simResp is a function to simulate item response data.<\/span><\/div>, <div class='tooltip'>Static<span class='tooltiptext'>Static is a test assembly function to perform fixed-form test assembly based on the generalized shadow-test framework.<\/span><\/div>, <div class='tooltip'>subsetConstraints<span class='tooltiptext'>Create a subset of a constraints object: constraints[i] subsetConstraints(constraints, 1:10)Combine two constraints objects: c(constraints1, constraints2) combineConstraints(constraints1, constraints2)<\/span><\/div>, <div class='tooltip'>subsetItemPool<span class='tooltiptext'>Create a subset of an item_pool object: pool[i] subsetItemPool(pool, i)Combine two item_pool objects: c(pool1, pool2) combineItemPool(pool1, pool2) pool1 + pool2pool1 - pool2 excludes items in pool2 from pool1.pool1 == pool2 tests whether two item_pool objects are identical.<\/span><\/div>, <div class='tooltip'>subsetTest<span class='tooltiptext'>Create a subset of a test object.<\/span><\/div>, <div class='tooltip'>summary<span class='tooltiptext'>Extension of summary() for objects in TestDesign package<\/span><\/div>, <div class='tooltip'>TestDesign<span class='tooltiptext'>TestDesign is a caller function to open the Shiny interface of TestDesign package.<\/span><\/div>, <div class='tooltip'>theta_EB<span class='tooltiptext'>theta_EB_single and theta_EB are functions to calculate theta estimates using EB (Empirical Bayes) method.<\/span><\/div>, <div class='tooltip'>theta_EB_single<span class='tooltiptext'>theta_EB_single and theta_EB are functions to calculate theta estimates using EB (Empirical Bayes) method.<\/span><\/div>, <div class='tooltip'>toggleConstraints<span class='tooltiptext'>toggleConstraints is a function to toggle individual constraints in a constraints object.<\/span><\/div>","<div class='tooltip'>isotitedose<span class='tooltiptext'>Calculate the next dose assignment for a TITE-IR design.<\/span><\/div>, <div class='tooltip'>isotitesim<span class='tooltiptext'>Simulates trials based on the TITE-IR design.<\/span><\/div>","<div class='tooltip'>descTwoWay<span class='tooltiptext'>descTwoWay produces basic descritive statistics including sample size, mean, standard deviation, median, minimum value, maximum value, 25th quantile, 75th quantile, skewness, kurtosis, the number of missing value for two-way layout.<\/span><\/div>, <div class='tooltip'>gplotTwoWay<span class='tooltiptext'>gplotTwoWay produce box-and-whisker plots and error bars for two-way layout.<\/span><\/div>, <div class='tooltip'>gpTwoWay<span class='tooltiptext'>gpTwoWay The gpTwoWay function computes a two-way ANOVA for main effects and interaction effect under heteroscedasticity.<\/span><\/div>, <div class='tooltip'>homogtestTwoWay<span class='tooltiptext'>homogtestTwoWay performs variance homogeneity tests including Levene, Bartlett, Fligner-Killeen tests for two-way layout.<\/span><\/div>, <div class='tooltip'>nortestTwoWay<span class='tooltiptext'>nortestTwoWay performs normality tests including Shapiro-Wilk, Shapiro-Francia, Kolmogorov-Smirnov, Anderson-Darling, Cramer-von Mises, Pearson Chi-square tests, and also assess the normality of each group through plots for two-way layout.<\/span><\/div>","<div class='tooltip'>grepal<span class='tooltiptext'>grepal returns a vector of colors whose names match a regular expression (regex).<\/span><\/div>, <div class='tooltip'>heading<span class='tooltiptext'>heading plots 1 or more text strings (provided as a character vector labels) as a heading to an (existing or new) plot and places a colored box behindeach label to mark it (i.e., highlighting the heading).<\/span><\/div>, <div class='tooltip'>mark<span class='tooltiptext'>mark plots 1 or more text strings (provided as a character vector labels) to an (existing or new) plot and places a colored box behindeach label to mark it (i.e., highlight or make it stand out from the background).<\/span><\/div>, <div class='tooltip'>newpal<span class='tooltiptext'>newpal allows defining new color palettes (as data frames).<\/span><\/div>, <div class='tooltip'>post<span class='tooltiptext'>post plots 1 or more text strings (provided as a character vector labels) to an (existing or new) xbox.<\/span><\/div>, <div class='tooltip'>seecol<span class='tooltiptext'>seecol provides an interface to plotting (or \"seeing\") the colors of a palette or comparing multiple color palettes.<\/span><\/div>, <div class='tooltip'>slide<span class='tooltiptext'>slide plots an empty slide (or frame) as a colored rectangle.<\/span><\/div>, <div class='tooltip'>theme_grau<span class='tooltiptext'>theme_grau provides an alternative unikn theme to use in ggplot2 commands.<\/span><\/div>, <div class='tooltip'>theme_unikn<span class='tooltiptext'>theme_unikn provides a basic unikn theme to use in ggplot2 commands.<\/span><\/div>, <div class='tooltip'>uline<span class='tooltiptext'>uline plots 1 or more text strings (provided as a character vector labels) to an (existing or new) plot and places a colored line underneath each label (to underline it).<\/span><\/div>, <div class='tooltip'>unikn.guide<span class='tooltiptext'>Opens the unikn package guides<\/span><\/div>, <div class='tooltip'>url_unikn<span class='tooltiptext'>url_unikn removes various patterns (e.g., \"http\", \"https\", \"://\", \"www.\") from the front of a given URL and returns the remaining character string with a figure dash prefix.<\/span><\/div>, <div class='tooltip'>usecol<span class='tooltiptext'>usecol allows using a color or color palette pal (e.g., for plotting).<\/span><\/div>, <div class='tooltip'>xbox<span class='tooltiptext'>xbox plots a box with a cross (x) in its top-right corner.<\/span><\/div>","<div class='tooltip'><span class='tooltiptext'><\/span><\/div>","<div class='tooltip'>calfun<span class='tooltiptext'>Estimates the calibration equation based on CV information<\/span><\/div>, <div class='tooltip'>chngpt<span class='tooltiptext'>Estimate a piecewise linear regression equation<\/span><\/div>, <div class='tooltip'>samplefun<span class='tooltiptext'>Draws samples uniformly (for internal use only)<\/span><\/div>, <div class='tooltip'>samplesize<span class='tooltiptext'>Estimates the sample size to achived the specified precision in the estimated calibration equation.<\/span><\/div>, <div class='tooltip'>sampletot<span class='tooltiptext'>Selects samples used in a calibration study<\/span><\/div>"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Description<\/th>\n      <th>taskview_doe<\/th>\n      <th>taskview_survey<\/th>\n      <th>func<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
